<!DOCTYPE html><html><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta name="viewport" content="width=device-width"><title>Endogenous FGFs drive ERK-dependent cell fate patterning in 2D human gastruloids</title><link rel="preload" href="print.css" as="style"><script type="text/javascript">/**
 * @license Paged.js v0.4.3 | MIT | https://pagedjs.org
 */

(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? module.exports = factory() :
	typeof define === 'function' && define.amd ? define(factory) :
	(global = typeof globalThis !== 'undefined' ? globalThis : global || self, global.PagedPolyfill = factory());
})(this, (function () { 'use strict';

	function getBoundingClientRect(element) {
		if (!element) {
			return;
		}
		let rect;
		if (typeof element.getBoundingClientRect !== "undefined") {
			rect = element.getBoundingClientRect();
		} else {
			let range = document.createRange();
			range.selectNode(element);
			rect = range.getBoundingClientRect();
		}
		return rect;
	}

	function getClientRects(element) {
		if (!element) {
			return;
		}
		let rect;
		if (typeof element.getClientRects !== "undefined") {
			rect = element.getClientRects();
		} else {
			let range = document.createRange();
			range.selectNode(element);
			rect = range.getClientRects();
		}
		return rect;
	}

	/**
	 * Generates a UUID
	 * based on: http://stackoverflow.com/questions/105034/how-to-create-a-guid-uuid-in-javascript
	 * @returns {string} uuid
	 */
	function UUID() {
		var d = new Date().getTime();
		if (typeof performance !== "undefined" && typeof performance.now === "function") {
			d += performance.now(); //use high-precision timer if available
		}
		return "xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx".replace(/[xy]/g, function (c) {
			var r = (d + Math.random() * 16) % 16 | 0;
			d = Math.floor(d / 16);
			return (c === "x" ? r : (r & 0x3 | 0x8)).toString(16);
		});
	}

	function attr(element, attributes) {
		for (var i = 0; i < attributes.length; i++) {
			if (element.hasAttribute(attributes[i])) {
				return element.getAttribute(attributes[i]);
			}
		}
	}

	/* Based on by https://mths.be/cssescape v1.5.1 by @mathias | MIT license
	 * Allows # and .
	 */
	function querySelectorEscape(value) {
		if (arguments.length == 0) {
			throw new TypeError("`CSS.escape` requires an argument.");
		}
		var string = String(value);

		var length = string.length;
		var index = -1;
		var codeUnit;
		var result = "";
		var firstCodeUnit = string.charCodeAt(0);
		while (++index < length) {
			codeUnit = string.charCodeAt(index);



			// Note: there’s no need to special-case astral symbols, surrogate
			// pairs, or lone surrogates.

			// If the character is NULL (U+0000), then the REPLACEMENT CHARACTER
			// (U+FFFD).
			if (codeUnit == 0x0000) {
				result += "\uFFFD";
				continue;
			}

			if (
				// If the character is in the range [\1-\1F] (U+0001 to U+001F) or is
				// U+007F, […]
				(codeUnit >= 0x0001 && codeUnit <= 0x001F) || codeUnit == 0x007F ||
				// If the character is the first character and is in the range [0-9]
				// (U+0030 to U+0039), […]
				(index == 0 && codeUnit >= 0x0030 && codeUnit <= 0x0039) ||
				// If the character is the second character and is in the range [0-9]
				// (U+0030 to U+0039) and the first character is a `-` (U+002D), […]
				(
					index == 1 &&
					codeUnit >= 0x0030 && codeUnit <= 0x0039 &&
					firstCodeUnit == 0x002D
				)
			) {
				// https://drafts.csswg.org/cssom/#escape-a-character-as-code-point
				result += "\\" + codeUnit.toString(16) + " ";
				continue;
			}

			if (
				// If the character is the first character and is a `-` (U+002D), and
				// there is no second character, […]
				index == 0 &&
				length == 1 &&
				codeUnit == 0x002D
			) {
				result += "\\" + string.charAt(index);
				continue;
			}

			// support for period character in id
			if (codeUnit == 0x002E) {
				if (string.charAt(0) == "#") {
					result += "\\.";
					continue;
				}
			}


			// If the character is not handled by one of the above rules and is
			// greater than or equal to U+0080, is `-` (U+002D) or `_` (U+005F), or
			// is in one of the ranges [0-9] (U+0030 to U+0039), [A-Z] (U+0041 to
			// U+005A), or [a-z] (U+0061 to U+007A), […]
			if (
				codeUnit >= 0x0080 ||
				codeUnit == 0x002D ||
				codeUnit == 0x005F ||
				codeUnit == 35 || // Allow #
				codeUnit == 46 || // Allow .
				codeUnit >= 0x0030 && codeUnit <= 0x0039 ||
				codeUnit >= 0x0041 && codeUnit <= 0x005A ||
				codeUnit >= 0x0061 && codeUnit <= 0x007A
			) {
				// the character itself
				result += string.charAt(index);
				continue;
			}

			// Otherwise, the escaped character.
			// https://drafts.csswg.org/cssom/#escape-a-character
			result += "\\" + string.charAt(index);

		}
		return result;
	}

	/**
	 * Creates a new pending promise and provides methods to resolve or reject it.
	 * From: https://developer.mozilla.org/en-US/docs/Mozilla/JavaScript_code_modules/Promise.jsm/Deferred#backwards_forwards_compatible
	 * @returns {object} defered
	 */
	function defer() {
		this.resolve = null;

		this.reject = null;

		this.id = UUID();

		this.promise = new Promise((resolve, reject) => {
			this.resolve = resolve;
			this.reject = reject;
		});
		Object.freeze(this);
	}

	const requestIdleCallback = typeof window !== "undefined" && ("requestIdleCallback" in window ? window.requestIdleCallback : window.requestAnimationFrame);

	function CSSValueToString(obj) {
		return obj.value + (obj.unit || "");
	}

	function isElement(node) {
		return node && node.nodeType === 1;
	}

	function isText(node) {
		return node && node.nodeType === 3;
	}

	function* walk$2(start, limiter) {
		let node = start;

		while (node) {

			yield node;

			if (node.childNodes.length) {
				node = node.firstChild;
			} else if (node.nextSibling) {
				if (limiter && node === limiter) {
					node = undefined;
					break;
				}
				node = node.nextSibling;
			} else {
				while (node) {
					node = node.parentNode;
					if (limiter && node === limiter) {
						node = undefined;
						break;
					}
					if (node && node.nextSibling) {
						node = node.nextSibling;
						break;
					}

				}
			}
		}
	}

	function nodeAfter(node, limiter) {
		if (limiter && node === limiter) {
			return;
		}
		let significantNode = nextSignificantNode(node);
		if (significantNode) {
			return significantNode;
		}
		if (node.parentNode) {
			while ((node = node.parentNode)) {
				if (limiter && node === limiter) {
					return;
				}
				significantNode = nextSignificantNode(node);
				if (significantNode) {
					return significantNode;
				}
			}
		}
	}

	function nodeBefore(node, limiter) {
		if (limiter && node === limiter) {
			return;
		}
		let significantNode = previousSignificantNode(node);
		if (significantNode) {
			return significantNode;
		}
		if (node.parentNode) {
			while ((node = node.parentNode)) {
				if (limiter && node === limiter) {
					return;
				}
				significantNode = previousSignificantNode(node);
				if (significantNode) {
					return significantNode;
				}
			}
		}
	}

	function elementAfter(node, limiter) {
		let after = nodeAfter(node, limiter);

		while (after && after.nodeType !== 1) {
			after = nodeAfter(after, limiter);
		}

		return after;
	}

	function elementBefore(node, limiter) {
		let before = nodeBefore(node, limiter);

		while (before && before.nodeType !== 1) {
			before = nodeBefore(before, limiter);
		}

		return before;
	}

	function displayedElementAfter(node, limiter) {
		let after = elementAfter(node, limiter);

		while (after && after.dataset.undisplayed) {
			after = elementAfter(after, limiter);
		}

		return after;
	}

	function displayedElementBefore(node, limiter) {
		let before = elementBefore(node, limiter);

		while (before && before.dataset.undisplayed) {
			before = elementBefore(before, limiter);
		}

		return before;
	}

	function rebuildAncestors(node) {
		let parent, ancestor;
		let ancestors = [];
		let added = [];

		let fragment = document.createDocumentFragment();

		// Handle rowspan on table
		if (node.nodeName === "TR") {
			let previousRow = node.previousElementSibling;
			let previousRowDistance = 1;
			while (previousRow) {
				// previous row has more columns, might indicate a rowspan.
				if (previousRow.childElementCount > node.childElementCount) {
					const initialColumns = Array.from(node.children);
					while (node.firstChild) {
						node.firstChild.remove();
					}
					let k = 0;
					for (let j = 0; j < previousRow.children.length; j++) {
						let column = previousRow.children[j];
						if (column.rowSpan && column.rowSpan > previousRowDistance) {
							const duplicatedColumn = column.cloneNode(true);
							// Adjust rowspan value
							duplicatedColumn.rowSpan = column.rowSpan - previousRowDistance;
							// Add the column to the row
							node.appendChild(duplicatedColumn);
						} else {
							// Fill the gap with the initial columns (if exists)
							const initialColumn = initialColumns[k++];
							// The initial column can be undefined if the newly created table has less columns than the original table
							if (initialColumn) {
								node.appendChild(initialColumn);
							}
						}
					}
				}
				previousRow = previousRow.previousElementSibling;
				previousRowDistance++;
			}
		}

		// Gather all ancestors
		let element = node;
		while(element.parentNode && element.parentNode.nodeType === 1) {
			ancestors.unshift(element.parentNode);
			element = element.parentNode;
		}

		for (var i = 0; i < ancestors.length; i++) {
			ancestor = ancestors[i];
			parent = ancestor.cloneNode(false);
		
			parent.setAttribute("data-split-from", parent.getAttribute("data-ref"));
			// ancestor.setAttribute("data-split-to", parent.getAttribute("data-ref"));

			if (parent.hasAttribute("id")) {
				let dataID = parent.getAttribute("id");
				parent.setAttribute("data-id", dataID);
				parent.removeAttribute("id");
			}

			// This is handled by css :not, but also tidied up here
			if (parent.hasAttribute("data-break-before")) {
				parent.removeAttribute("data-break-before");
			}

			if (parent.hasAttribute("data-previous-break-after")) {
				parent.removeAttribute("data-previous-break-after");
			}

			if (added.length) {
				let container = added[added.length-1];
				container.appendChild(parent);
			} else {
				fragment.appendChild(parent);
			}
			added.push(parent);

			// rebuild table rows
			if (parent.nodeName === "TD" && ancestor.parentElement.contains(ancestor)) {
				let td = ancestor;
				let prev = parent;
				while ((td = td.previousElementSibling)) {
					let sib = td.cloneNode(false);
					parent.parentElement.insertBefore(sib, prev);
					prev = sib;
				}
				
			}
		}

		added = undefined;
		return fragment;
	}
	/*
	export function split(bound, cutElement, breakAfter) {
			let needsRemoval = [];
			let index = indexOf(cutElement);

			if (!breakAfter && index === 0) {
				return;
			}

			if (breakAfter && index === (cutElement.parentNode.children.length - 1)) {
				return;
			}

			// Create a fragment with rebuilt ancestors
			let fragment = rebuildAncestors(cutElement);

			// Clone cut
			if (!breakAfter) {
				let clone = cutElement.cloneNode(true);
				let ref = cutElement.parentNode.getAttribute('data-ref');
				let parent = fragment.querySelector("[data-ref='" + ref + "']");
				parent.appendChild(clone);
				needsRemoval.push(cutElement);
			}

			// Remove all after cut
			let next = nodeAfter(cutElement, bound);
			while (next) {
				let clone = next.cloneNode(true);
				let ref = next.parentNode.getAttribute('data-ref');
				let parent = fragment.querySelector("[data-ref='" + ref + "']");
				parent.appendChild(clone);
				needsRemoval.push(next);
				next = nodeAfter(next, bound);
			}

			// Remove originals
			needsRemoval.forEach((node) => {
				if (node) {
					node.remove();
				}
			});

			// Insert after bounds
			bound.parentNode.insertBefore(fragment, bound.nextSibling);
			return [bound, bound.nextSibling];
	}
	*/

	function needsBreakBefore(node) {
		if( typeof node !== "undefined" &&
				typeof node.dataset !== "undefined" &&
				typeof node.dataset.breakBefore !== "undefined" &&
				(node.dataset.breakBefore === "always" ||
				 node.dataset.breakBefore === "page" ||
				 node.dataset.breakBefore === "left" ||
				 node.dataset.breakBefore === "right" ||
				 node.dataset.breakBefore === "recto" ||
				 node.dataset.breakBefore === "verso")
			 ) {
			return true;
		}

		return false;
	}

	function needsPreviousBreakAfter(node) {
		if( typeof node !== "undefined" &&
				typeof node.dataset !== "undefined" &&
				typeof node.dataset.previousBreakAfter !== "undefined" &&
				(node.dataset.previousBreakAfter === "always" ||
				 node.dataset.previousBreakAfter === "page" ||
				 node.dataset.previousBreakAfter === "left" ||
				 node.dataset.previousBreakAfter === "right" ||
				 node.dataset.previousBreakAfter === "recto" ||
				 node.dataset.previousBreakAfter === "verso")
			 ) {
			return true;
		}

		return false;
	}

	function needsPageBreak(node, previousSignificantNode) {
		if (typeof node === "undefined" || !previousSignificantNode || isIgnorable(node)) {
			return false;
		}
		if (node.dataset && node.dataset.undisplayed) {
			return false;
		}
		let previousSignificantNodePage = previousSignificantNode.dataset ? previousSignificantNode.dataset.page : undefined;
		if (typeof previousSignificantNodePage === "undefined") {
			const nodeWithNamedPage = getNodeWithNamedPage(previousSignificantNode);
			if (nodeWithNamedPage) {
				previousSignificantNodePage = nodeWithNamedPage.dataset.page;
			}
		}
		let currentNodePage = node.dataset ? node.dataset.page : undefined;
		if (typeof currentNodePage === "undefined") {
			const nodeWithNamedPage = getNodeWithNamedPage(node, previousSignificantNode);
			if (nodeWithNamedPage) {
				currentNodePage = nodeWithNamedPage.dataset.page;
			}
		}
		return currentNodePage !== previousSignificantNodePage;
	}

	function *words(node) {
		let currentText = node.nodeValue;
		let max = currentText.length;
		let currentOffset = 0;
		let currentLetter;

		let range;
		const significantWhitespaces = node.parentElement && node.parentElement.nodeName === "PRE";

		while (currentOffset < max) {
			currentLetter = currentText[currentOffset];
			if (/^[\S\u202F\u00A0]$/.test(currentLetter) || significantWhitespaces) {
				if (!range) {
					range = document.createRange();
					range.setStart(node, currentOffset);
				}
			} else {
				if (range) {
					range.setEnd(node, currentOffset);
					yield range;
					range = undefined;
				}
			}

			currentOffset += 1;
		}

		if (range) {
			range.setEnd(node, currentOffset);
			yield range;
		}
	}

	function *letters(wordRange) {
		let currentText = wordRange.startContainer;
		let max = currentText.length;
		let currentOffset = wordRange.startOffset;
		// let currentLetter;

		let range;

		while(currentOffset < max) {
			 // currentLetter = currentText[currentOffset];
			 range = document.createRange();
			 range.setStart(currentText, currentOffset);
			 range.setEnd(currentText, currentOffset+1);

			 yield range;

			 currentOffset += 1;
		}
	}

	function isContainer(node) {
		let container;

		if (typeof node.tagName === "undefined") {
			return true;
		}

		if (node.style && node.style.display === "none") {
			return false;
		}

		switch (node.tagName) {
			// Inline
			case "A":
			case "ABBR":
			case "ACRONYM":
			case "B":
			case "BDO":
			case "BIG":
			case "BR":
			case "BUTTON":
			case "CITE":
			case "CODE":
			case "DFN":
			case "EM":
			case "I":
			case "IMG":
			case "INPUT":
			case "KBD":
			case "LABEL":
			case "MAP":
			case "OBJECT":
			case "Q":
			case "SAMP":
			case "SCRIPT":
			case "SELECT":
			case "SMALL":
			case "SPAN":
			case "STRONG":
			case "SUB":
			case "SUP":
			case "TEXTAREA":
			case "TIME":
			case "TT":
			case "VAR":
			case "P":
			case "H1":
			case "H2":
			case "H3":
			case "H4":
			case "H5":
			case "H6":
			case "FIGCAPTION":
			case "BLOCKQUOTE":
			case "PRE":
			case "LI":
			case "TD":
			case "DT":
			case "DD":
			case "VIDEO":
			case "CANVAS":
				container = false;
				break;
			default:
				container = true;
		}

		return container;
	}

	function cloneNode(n, deep=false) {
		return n.cloneNode(deep);
	}

	function findElement(node, doc, forceQuery) {
		const ref = node.getAttribute("data-ref");
		return findRef(ref, doc, forceQuery);
	}

	function findRef(ref, doc, forceQuery) {
		if (!forceQuery && doc.indexOfRefs && doc.indexOfRefs[ref]) {
			return doc.indexOfRefs[ref];
		} else {
			return doc.querySelector(`[data-ref='${ref}']`);
		}
	}

	function validNode(node) {
		if (isText(node)) {
			return true;
		}

		if (isElement(node) && node.dataset.ref) {
			return true;
		}

		return false;
	}

	function prevValidNode(node) {
		while (!validNode(node)) {
			if (node.previousSibling) {
				node = node.previousSibling;
			} else {
				node = node.parentNode;
			}

			if (!node) {
				break;
			}
		}

		return node;
	}


	function indexOf$2(node) {
		let parent = node.parentNode;
		if (!parent) {
			return 0;
		}
		return Array.prototype.indexOf.call(parent.childNodes, node);
	}

	function child(node, index) {
		return node.childNodes[index];
	}

	function hasContent(node) {
		if (isElement(node)) {
			return true;
		} else if (isText(node) &&
				node.textContent.trim().length) {
			return true;
		}
		return false;
	}

	function indexOfTextNode(node, parent) {
		if (!isText(node)) {
			return -1;
		}
		let nodeTextContent = node.textContent;
		let child;
		let index = -1;
		for (var i = 0; i < parent.childNodes.length; i++) {
			child = parent.childNodes[i];
			if (child.nodeType === 3) {
				let text = parent.childNodes[i].textContent;
				if (text.includes(nodeTextContent)) {
					index = i;
					break;
				}
			}
		}

		return index;
	}


	/**
	 * Throughout, whitespace is defined as one of the characters
	 *  "\t" TAB \u0009
	 *  "\n" LF  \u000A
	 *  "\r" CR  \u000D
	 *  " "  SPC \u0020
	 *
	 * This does not use Javascript's "\s" because that includes non-breaking
	 * spaces (and also some other characters).
	 */

	/**
	 * Determine if a node should be ignored by the iterator functions.
	 * taken from https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Whitespace#Whitespace_helper_functions
	 *
	 * @param {Node} node An object implementing the DOM1 |Node| interface.
	 * @return {boolean} true if the node is:
	 *  1) A |Text| node that is all whitespace
	 *  2) A |Comment| node
	 *  and otherwise false.
	 */
	function isIgnorable(node) {
		return (node.nodeType === 8) || // A comment node
			((node.nodeType === 3) && isAllWhitespace(node)); // a text node, all whitespace
	}

	/**
	 * Determine whether a node's text content is entirely whitespace.
	 *
	 * @param {Node} node  A node implementing the |CharacterData| interface (i.e., a |Text|, |Comment|, or |CDATASection| node
	 * @return {boolean} true if all of the text content of |nod| is whitespace, otherwise false.
	 */
	function isAllWhitespace(node) {
		return !(/[^\t\n\r ]/.test(node.textContent));
	}

	/**
	 * Version of |previousSibling| that skips nodes that are entirely
	 * whitespace or comments.  (Normally |previousSibling| is a property
	 * of all DOM nodes that gives the sibling node, the node that is
	 * a child of the same parent, that occurs immediately before the
	 * reference node.)
	 *
	 * @param {ChildNode} sib  The reference node.
	 * @return {Node|null} Either:
	 *  1) The closest previous sibling to |sib| that is not ignorable according to |is_ignorable|, or
	 *  2) null if no such node exists.
	 */
	function previousSignificantNode(sib) {
		while ((sib = sib.previousSibling)) {
			if (!isIgnorable(sib)) return sib;
		}
		return null;
	}

	function getNodeWithNamedPage(node, limiter) {
		if (node && node.dataset && node.dataset.page) {
			return node;
		}
		if (node.parentNode) {
			while ((node = node.parentNode)) {
				if (limiter && node === limiter) {
					return;
				}
				if (node.dataset && node.dataset.page) {
					return node;
				}
			}
		}
		return null;
	}

	function breakInsideAvoidParentNode(node) {
		while ((node = node.parentNode)) {
			if (node && node.dataset && node.dataset.breakInside === "avoid") {
				return node;
			}
		}
		return null;
	}

	/**
	 * Find a parent with a given node name.
	 * @param {Node} node - initial Node
	 * @param {string} nodeName - node name (eg. "TD", "TABLE", "STRONG"...)
	 * @param {Node} limiter - go up to the parent until there's no more parent or the current node is equals to the limiter
	 * @returns {Node|undefined} - Either:
	 *  1) The closest parent for a the given node name, or
	 *  2) undefined if no such node exists.
	 */
	function parentOf(node, nodeName, limiter) {
		if (limiter && node === limiter) {
			return;
		}
		if (node.parentNode) {
			while ((node = node.parentNode)) {
				if (limiter && node === limiter) {
					return;
				}
				if (node.nodeName === nodeName) {
					return node;
				}
			}
		}
	}

	/**
	 * Version of |nextSibling| that skips nodes that are entirely
	 * whitespace or comments.
	 *
	 * @param {ChildNode} sib  The reference node.
	 * @return {Node|null} Either:
	 *  1) The closest next sibling to |sib| that is not ignorable according to |is_ignorable|, or
	 *  2) null if no such node exists.
	 */
	function nextSignificantNode(sib) {
		while ((sib = sib.nextSibling)) {
			if (!isIgnorable(sib)) return sib;
		}
		return null;
	}

	function filterTree(content, func, what) {
		const treeWalker = document.createTreeWalker(
			content || this.dom,
			what || NodeFilter.SHOW_ALL,
			func ? { acceptNode: func } : null,
			false
		);

		let node;
		let current;
		node = treeWalker.nextNode();
		while(node) {
			current = node;
			node = treeWalker.nextNode();
			current.parentNode.removeChild(current);
		}
	}

	/**
	 * BreakToken
	 * @class
	 */
	class BreakToken {

		constructor(node, offset) {
			this.node = node;
			this.offset = offset;
		}

		equals(otherBreakToken) {
			if (!otherBreakToken) {
				return false;
			}
			if (this["node"] && otherBreakToken["node"] &&
				this["node"] !== otherBreakToken["node"]) {
				return false;
			}
			if (this["offset"] && otherBreakToken["offset"] &&
				this["offset"] !== otherBreakToken["offset"]) {
				return false;
			}
			return true;
		}

		toJSON(hash) {
			let node;
			let index = 0;
			if (!this.node) {
				return {};
			}
			if (isElement(this.node) && this.node.dataset.ref) {
				node = this.node.dataset.ref;
			} else if (hash) {
				node = this.node.parentElement.dataset.ref;
			}

			if (this.node.parentElement) {
				const children = Array.from(this.node.parentElement.childNodes);
				index = children.indexOf(this.node);
			}

			return JSON.stringify({
				"node": node,
				"index" : index,
				"offset": this.offset
			});
		}

	}

	/**
	 * Render result.
	 * @class
	 */
	class RenderResult {

		constructor(breakToken, error) {
			this.breakToken = breakToken;
			this.error = error;
		}
	}

	class OverflowContentError extends Error {
		constructor(message, items) {
			super(message);
			this.items = items;
		}
	}

	function getDefaultExportFromCjs (x) {
		return x && x.__esModule && Object.prototype.hasOwnProperty.call(x, 'default') ? x['default'] : x;
	}

	var eventEmitter = {exports: {}};

	var d$2 = {exports: {}};

	// ES3 safe
	var _undefined$1 = void 0;

	var is$4 = function (value) { return value !== _undefined$1 && value !== null; };

	var isValue$5 = is$4;

	// prettier-ignore
	var possibleTypes = { "object": true, "function": true, "undefined": true /* document.all */ };

	var is$3 = function (value) {
		if (!isValue$5(value)) return false;
		return hasOwnProperty.call(possibleTypes, typeof value);
	};

	var isObject$3 = is$3;

	var is$2 = function (value) {
		if (!isObject$3(value)) return false;
		try {
			if (!value.constructor) return false;
			return value.constructor.prototype === value;
		} catch (error) {
			return false;
		}
	};

	var isPrototype = is$2;

	var is$1 = function (value) {
		if (typeof value !== "function") return false;

		if (!hasOwnProperty.call(value, "length")) return false;

		try {
			if (typeof value.length !== "number") return false;
			if (typeof value.call !== "function") return false;
			if (typeof value.apply !== "function") return false;
		} catch (error) {
			return false;
		}

		return !isPrototype(value);
	};

	var isFunction$1 = is$1;

	var classRe = /^\s*class[\s{/}]/, functionToString = Function.prototype.toString;

	var is = function (value) {
		if (!isFunction$1(value)) return false;
		if (classRe.test(functionToString.call(value))) return false;
		return true;
	};

	var isImplemented$7 = function () {
		var assign = Object.assign, obj;
		if (typeof assign !== "function") return false;
		obj = { foo: "raz" };
		assign(obj, { bar: "dwa" }, { trzy: "trzy" });
		return obj.foo + obj.bar + obj.trzy === "razdwatrzy";
	};

	var isImplemented$6;
	var hasRequiredIsImplemented$2;

	function requireIsImplemented$2 () {
		if (hasRequiredIsImplemented$2) return isImplemented$6;
		hasRequiredIsImplemented$2 = 1;

		isImplemented$6 = function () {
			try {
				Object.keys("primitive");
				return true;
			} catch (e) {
				return false;
			}
		};
		return isImplemented$6;
	}

	// eslint-disable-next-line no-empty-function
	var noop$4 = function () {};

	var _undefined = noop$4(); // Support ES3 engines

	var isValue$4 = function (val) { return val !== _undefined && val !== null; };

	var shim$5;
	var hasRequiredShim$5;

	function requireShim$5 () {
		if (hasRequiredShim$5) return shim$5;
		hasRequiredShim$5 = 1;

		var isValue = isValue$4;

		var keys = Object.keys;

		shim$5 = function (object) { return keys(isValue(object) ? Object(object) : object); };
		return shim$5;
	}

	var keys;
	var hasRequiredKeys;

	function requireKeys () {
		if (hasRequiredKeys) return keys;
		hasRequiredKeys = 1;

		keys = requireIsImplemented$2()() ? Object.keys : requireShim$5();
		return keys;
	}

	var isValue$3 = isValue$4;

	var validValue = function (value) {
		if (!isValue$3(value)) throw new TypeError("Cannot use null or undefined");
		return value;
	};

	var shim$4;
	var hasRequiredShim$4;

	function requireShim$4 () {
		if (hasRequiredShim$4) return shim$4;
		hasRequiredShim$4 = 1;

		var keys  = requireKeys()
		  , value = validValue
		  , max   = Math.max;

		shim$4 = function (dest, src /*, …srcn*/) {
			var error, i, length = max(arguments.length, 2), assign;
			dest = Object(value(dest));
			assign = function (key) {
				try {
					dest[key] = src[key];
				} catch (e) {
					if (!error) error = e;
				}
			};
			for (i = 1; i < length; ++i) {
				src = arguments[i];
				keys(src).forEach(assign);
			}
			if (error !== undefined) throw error;
			return dest;
		};
		return shim$4;
	}

	var assign$2 = isImplemented$7() ? Object.assign : requireShim$4();

	var isValue$2 = isValue$4;

	var forEach$1 = Array.prototype.forEach, create$5 = Object.create;

	var process = function (src, obj) {
		var key;
		for (key in src) obj[key] = src[key];
	};

	// eslint-disable-next-line no-unused-vars
	var normalizeOptions = function (opts1 /*, …options*/) {
		var result = create$5(null);
		forEach$1.call(arguments, function (options) {
			if (!isValue$2(options)) return;
			process(Object(options), result);
		});
		return result;
	};

	var str = "razdwatrzy";

	var isImplemented$5 = function () {
		if (typeof str.contains !== "function") return false;
		return str.contains("dwa") === true && str.contains("foo") === false;
	};

	var shim$3;
	var hasRequiredShim$3;

	function requireShim$3 () {
		if (hasRequiredShim$3) return shim$3;
		hasRequiredShim$3 = 1;

		var indexOf = String.prototype.indexOf;

		shim$3 = function (searchString /*, position*/) {
			return indexOf.call(this, searchString, arguments[1]) > -1;
		};
		return shim$3;
	}

	var contains$1 = isImplemented$5() ? String.prototype.contains : requireShim$3();

	var isValue$1         = is$4
	  , isPlainFunction = is
	  , assign$1          = assign$2
	  , normalizeOpts   = normalizeOptions
	  , contains        = contains$1;

	var d$1 = (d$2.exports = function (dscr, value/*, options*/) {
		var c, e, w, options, desc;
		if (arguments.length < 2 || typeof dscr !== "string") {
			options = value;
			value = dscr;
			dscr = null;
		} else {
			options = arguments[2];
		}
		if (isValue$1(dscr)) {
			c = contains.call(dscr, "c");
			e = contains.call(dscr, "e");
			w = contains.call(dscr, "w");
		} else {
			c = w = true;
			e = false;
		}

		desc = { value: value, configurable: c, enumerable: e, writable: w };
		return !options ? desc : assign$1(normalizeOpts(options), desc);
	});

	d$1.gs = function (dscr, get, set/*, options*/) {
		var c, e, options, desc;
		if (typeof dscr !== "string") {
			options = set;
			set = get;
			get = dscr;
			dscr = null;
		} else {
			options = arguments[3];
		}
		if (!isValue$1(get)) {
			get = undefined;
		} else if (!isPlainFunction(get)) {
			options = get;
			get = set = undefined;
		} else if (!isValue$1(set)) {
			set = undefined;
		} else if (!isPlainFunction(set)) {
			options = set;
			set = undefined;
		}
		if (isValue$1(dscr)) {
			c = contains.call(dscr, "c");
			e = contains.call(dscr, "e");
		} else {
			c = true;
			e = false;
		}

		desc = { get: get, set: set, configurable: c, enumerable: e };
		return !options ? desc : assign$1(normalizeOpts(options), desc);
	};

	var dExports = d$2.exports;

	var validCallable = function (fn) {
		if (typeof fn !== "function") throw new TypeError(fn + " is not a function");
		return fn;
	};

	(function (module, exports) {

		var d        = dExports
		  , callable = validCallable

		  , apply = Function.prototype.apply, call = Function.prototype.call
		  , create = Object.create, defineProperty = Object.defineProperty
		  , defineProperties = Object.defineProperties
		  , hasOwnProperty = Object.prototype.hasOwnProperty
		  , descriptor = { configurable: true, enumerable: false, writable: true }

		  , on, once, off, emit, methods, descriptors, base;

		on = function (type, listener) {
			var data;

			callable(listener);

			if (!hasOwnProperty.call(this, '__ee__')) {
				data = descriptor.value = create(null);
				defineProperty(this, '__ee__', descriptor);
				descriptor.value = null;
			} else {
				data = this.__ee__;
			}
			if (!data[type]) data[type] = listener;
			else if (typeof data[type] === 'object') data[type].push(listener);
			else data[type] = [data[type], listener];

			return this;
		};

		once = function (type, listener) {
			var once, self;

			callable(listener);
			self = this;
			on.call(this, type, once = function () {
				off.call(self, type, once);
				apply.call(listener, this, arguments);
			});

			once.__eeOnceListener__ = listener;
			return this;
		};

		off = function (type, listener) {
			var data, listeners, candidate, i;

			callable(listener);

			if (!hasOwnProperty.call(this, '__ee__')) return this;
			data = this.__ee__;
			if (!data[type]) return this;
			listeners = data[type];

			if (typeof listeners === 'object') {
				for (i = 0; (candidate = listeners[i]); ++i) {
					if ((candidate === listener) ||
							(candidate.__eeOnceListener__ === listener)) {
						if (listeners.length === 2) data[type] = listeners[i ? 0 : 1];
						else listeners.splice(i, 1);
					}
				}
			} else {
				if ((listeners === listener) ||
						(listeners.__eeOnceListener__ === listener)) {
					delete data[type];
				}
			}

			return this;
		};

		emit = function (type) {
			var i, l, listener, listeners, args;

			if (!hasOwnProperty.call(this, '__ee__')) return;
			listeners = this.__ee__[type];
			if (!listeners) return;

			if (typeof listeners === 'object') {
				l = arguments.length;
				args = new Array(l - 1);
				for (i = 1; i < l; ++i) args[i - 1] = arguments[i];

				listeners = listeners.slice();
				for (i = 0; (listener = listeners[i]); ++i) {
					apply.call(listener, this, args);
				}
			} else {
				switch (arguments.length) {
				case 1:
					call.call(listeners, this);
					break;
				case 2:
					call.call(listeners, this, arguments[1]);
					break;
				case 3:
					call.call(listeners, this, arguments[1], arguments[2]);
					break;
				default:
					l = arguments.length;
					args = new Array(l - 1);
					for (i = 1; i < l; ++i) {
						args[i - 1] = arguments[i];
					}
					apply.call(listeners, this, args);
				}
			}
		};

		methods = {
			on: on,
			once: once,
			off: off,
			emit: emit
		};

		descriptors = {
			on: d(on),
			once: d(once),
			off: d(off),
			emit: d(emit)
		};

		base = defineProperties({}, descriptors);

		module.exports = exports = function (o) {
			return (o == null) ? create(base) : defineProperties(Object(o), descriptors);
		};
		exports.methods = methods; 
	} (eventEmitter, eventEmitter.exports));

	var eventEmitterExports = eventEmitter.exports;
	var EventEmitter = /*@__PURE__*/getDefaultExportFromCjs(eventEmitterExports);

	/**
	 * Hooks allow for injecting functions that must all complete in order before finishing
	 * They will execute in parallel but all must finish before continuing
	 * Functions may return a promise if they are asycn.
	 * From epubjs/src/utils/hooks
	 * @param {any} context scope of this
	 * @example this.content = new Hook(this);
	 */
	class Hook {
		constructor(context){
			this.context = context || this;
			this.hooks = [];
		}

		/**
		 * Adds a function to be run before a hook completes
		 * @example this.content.register(function(){...});
		 * @return {undefined} void
		 */
		register(){
			for(var i = 0; i < arguments.length; ++i) {
				if (typeof arguments[i]  === "function") {
					this.hooks.push(arguments[i]);
				} else {
					// unpack array
					for(var j = 0; j < arguments[i].length; ++j) {
						this.hooks.push(arguments[i][j]);
					}
				}
			}
		}

		/**
		 * Triggers a hook to run all functions
		 * @example this.content.trigger(args).then(function(){...});
		 * @return {Promise} results
		 */
		trigger(){
			var args = arguments;
			var context = this.context;
			var promises = [];

			this.hooks.forEach(function(task) {
				var executing = task.apply(context, args);

				if(executing && typeof executing["then"] === "function") {
					// Task is a function that returns a promise
					promises.push(executing);
				} else {
					// Otherwise Task resolves immediately, add resolved promise with result
					promises.push(new Promise((resolve, reject) => {
						resolve(executing);
					}));
				}
			});


			return Promise.all(promises);
		}

		/**
	   * Triggers a hook to run all functions synchronously
	   * @example this.content.trigger(args).then(function(){...});
	   * @return {Array} results
	   */
		triggerSync(){
			var args = arguments;
			var context = this.context;
			var results = [];

			this.hooks.forEach(function(task) {
				var executing = task.apply(context, args);

				results.push(executing);
			});


			return results;
		}

		// Adds a function to be run before a hook completes
		list(){
			return this.hooks;
		}

		clear(){
			return this.hooks = [];
		}
	}

	const MAX_CHARS_PER_BREAK = 1500;

	/**
	 * Layout
	 * @class
	 */
	class Layout {

		constructor(element, hooks, options) {
			this.element = element;

			this.bounds = this.element.getBoundingClientRect();
			this.parentBounds = this.element.offsetParent.getBoundingClientRect();
			let gap = parseFloat(window.getComputedStyle(this.element).columnGap);
		
			if (gap) {
				let leftMargin = this.bounds.left - this.parentBounds.left;
				this.gap =  gap - leftMargin;	
			} else {
				this.gap = 0;
			}

			if (hooks) {
				this.hooks = hooks;
			} else {
				this.hooks = {};
				this.hooks.onPageLayout = new Hook();
				this.hooks.layout = new Hook();
				this.hooks.renderNode = new Hook();
				this.hooks.layoutNode = new Hook();
				this.hooks.beforeOverflow = new Hook();
				this.hooks.onOverflow = new Hook();
				this.hooks.afterOverflowRemoved = new Hook();
				this.hooks.onBreakToken = new Hook();
				this.hooks.beforeRenderResult = new Hook();
			}

			this.settings = options || {};

			this.maxChars = this.settings.maxChars || MAX_CHARS_PER_BREAK;
			this.forceRenderBreak = false;
		}

		async renderTo(wrapper, source, breakToken, bounds = this.bounds) {
			let start = this.getStart(source, breakToken);
			let walker = walk$2(start, source);

			let node;
			let prevNode;
			let done;
			let next;

			let hasRenderedContent = false;
			let newBreakToken;

			let length = 0;

			let prevBreakToken = breakToken || new BreakToken(start);

			this.hooks && this.hooks.onPageLayout.trigger(wrapper, prevBreakToken, this);

			while (!done && !newBreakToken) {
				next = walker.next();
				prevNode = node;
				node = next.value;
				done = next.done;

				if (!node) {
					this.hooks && this.hooks.layout.trigger(wrapper, this);

					let imgs = wrapper.querySelectorAll("img");
					if (imgs.length) {
						await this.waitForImages(imgs);
					}

					newBreakToken = this.findBreakToken(wrapper, source, bounds, prevBreakToken);

					if (newBreakToken && newBreakToken.equals(prevBreakToken)) {
						console.warn("Unable to layout item: ", prevNode);
						this.hooks && this.hooks.beforeRenderResult.trigger(undefined, wrapper, this);
						return new RenderResult(undefined, new OverflowContentError("Unable to layout item", [prevNode]));
					}

					this.rebuildTableFromBreakToken(newBreakToken, wrapper);

					this.hooks && this.hooks.beforeRenderResult.trigger(newBreakToken, wrapper, this);
					return new RenderResult(newBreakToken);
				}

				this.hooks && this.hooks.layoutNode.trigger(node);

				// Check if the rendered element has a break set
				if (hasRenderedContent && this.shouldBreak(node, start)) {
					this.hooks && this.hooks.layout.trigger(wrapper, this);

					let imgs = wrapper.querySelectorAll("img");
					if (imgs.length) {
						await this.waitForImages(imgs);
					}

					newBreakToken = this.findBreakToken(wrapper, source, bounds, prevBreakToken);

					if (!newBreakToken) {
						newBreakToken = this.breakAt(node);
					} else {
						this.rebuildTableFromBreakToken(newBreakToken, wrapper);
					}

					if (newBreakToken && newBreakToken.equals(prevBreakToken)) {
						console.warn("Unable to layout item: ", node);
						let after = newBreakToken.node && nodeAfter(newBreakToken.node);
						if (after) {
							newBreakToken = new BreakToken(after);
						} else {
							return new RenderResult(undefined, new OverflowContentError("Unable to layout item", [node]));
						}
					}

					length = 0;

					break;
				}

				if (node.dataset && node.dataset.page) {
					let named = node.dataset.page;
					let page = this.element.closest(".pagedjs_page");
					page.classList.add("pagedjs_named_page");
					page.classList.add("pagedjs_" + named + "_page");

					if (!node.dataset.splitFrom) {
						page.classList.add("pagedjs_" + named + "_first_page");
					}
				}

				// Should the Node be a shallow or deep clone
				let shallow = isContainer(node);

				let rendered = this.append(node, wrapper, breakToken, shallow);

				length += rendered.textContent.length;

				// Check if layout has content yet
				if (!hasRenderedContent) {
					hasRenderedContent = hasContent(node);
				}

				// Skip to the next node if a deep clone was rendered
				if (!shallow) {
					walker = walk$2(nodeAfter(node, source), source);
				}

				if (this.forceRenderBreak) {
					this.hooks && this.hooks.layout.trigger(wrapper, this);

					newBreakToken = this.findBreakToken(wrapper, source, bounds, prevBreakToken);

					if (!newBreakToken) {
						newBreakToken = this.breakAt(node);
					} else {
						this.rebuildTableFromBreakToken(newBreakToken, wrapper);
					}

					length = 0;
					this.forceRenderBreak = false;

					break;
				}

				// Only check x characters
				if (length >= this.maxChars) {

					this.hooks && this.hooks.layout.trigger(wrapper, this);

					let imgs = wrapper.querySelectorAll("img");
					if (imgs.length) {
						await this.waitForImages(imgs);
					}

					newBreakToken = this.findBreakToken(wrapper, source, bounds, prevBreakToken);

					if (newBreakToken) {
						length = 0;
						this.rebuildTableFromBreakToken(newBreakToken, wrapper);
					}

					if (newBreakToken && newBreakToken.equals(prevBreakToken)) {
						console.warn("Unable to layout item: ", node);
						let after = newBreakToken.node && nodeAfter(newBreakToken.node);
						if (after) {
							newBreakToken = new BreakToken(after);
						} else {
							this.hooks && this.hooks.beforeRenderResult.trigger(undefined, wrapper, this);
							return new RenderResult(undefined, new OverflowContentError("Unable to layout item", [node]));
						}
					}
				}

			}

			this.hooks && this.hooks.beforeRenderResult.trigger(newBreakToken, wrapper, this);
			return new RenderResult(newBreakToken);
		}

		breakAt(node, offset = 0) {
			let newBreakToken = new BreakToken(
				node,
				offset
			);
			let breakHooks = this.hooks.onBreakToken.triggerSync(newBreakToken, undefined, node, this);
			breakHooks.forEach((newToken) => {
				if (typeof newToken != "undefined") {
					newBreakToken = newToken;
				}
			});

			return newBreakToken;
		}

		shouldBreak(node, limiter) {
			let previousNode = nodeBefore(node, limiter);
			let parentNode = node.parentNode;
			let parentBreakBefore = needsBreakBefore(node) && parentNode && !previousNode && needsBreakBefore(parentNode);
			let doubleBreakBefore;

			if (parentBreakBefore) {
				doubleBreakBefore = node.dataset.breakBefore === parentNode.dataset.breakBefore;
			}

			return !doubleBreakBefore && needsBreakBefore(node) || needsPreviousBreakAfter(node) || needsPageBreak(node, previousNode);
		}

		forceBreak() {
			this.forceRenderBreak = true;
		}

		getStart(source, breakToken) {
			let start;
			let node = breakToken && breakToken.node;

			if (node) {
				start = node;
			} else {
				start = source.firstChild;
			}

			return start;
		}

		append(node, dest, breakToken, shallow = true, rebuild = true) {

			let clone = cloneNode(node, !shallow);

			if (node.parentNode && isElement(node.parentNode)) {
				let parent = findElement(node.parentNode, dest);
				// Rebuild chain
				if (parent) {
					parent.appendChild(clone);
				} else if (rebuild) {
					let fragment = rebuildAncestors(node);
					parent = findElement(node.parentNode, fragment);
					if (!parent) {
						dest.appendChild(clone);
					} else if (breakToken && isText(breakToken.node) && breakToken.offset > 0) {
						clone.textContent = clone.textContent.substring(breakToken.offset);
						parent.appendChild(clone);
					} else {
						parent.appendChild(clone);
					}

					dest.appendChild(fragment);
				} else {
					dest.appendChild(clone);
				}


			} else {
				dest.appendChild(clone);
			}

			if (clone.dataset && clone.dataset.ref) {
				if (!dest.indexOfRefs) {
					dest.indexOfRefs = {};
				}
				dest.indexOfRefs[clone.dataset.ref] = clone;
			}

			let nodeHooks = this.hooks.renderNode.triggerSync(clone, node, this);
			nodeHooks.forEach((newNode) => {
				if (typeof newNode != "undefined") {
					clone = newNode;
				}
			});

			return clone;
		}

		rebuildTableFromBreakToken(breakToken, dest) {
			if (!breakToken || !breakToken.node) {
				return;
			}
			let node = breakToken.node;
			let td = isElement(node) ? node.closest("td") : node.parentElement.closest("td");
			if (td) {
				let rendered = findElement(td, dest, true);
				if (!rendered) {
					return;
				}
				while ((td = td.nextElementSibling)) {
					this.append(td, dest, null, true);
				}
			}
		}

		async waitForImages(imgs) {
			let results = Array.from(imgs).map(async (img) => {
				return this.awaitImageLoaded(img);
			});
			await Promise.all(results);
		}

		async awaitImageLoaded(image) {
			return new Promise(resolve => {
				if (image.complete !== true) {
					image.onload = function () {
						let {width, height} = window.getComputedStyle(image);
						resolve(width, height);
					};
					image.onerror = function (e) {
						let {width, height} = window.getComputedStyle(image);
						resolve(width, height, e);
					};
				} else {
					let {width, height} = window.getComputedStyle(image);
					resolve(width, height);
				}
			});
		}

		avoidBreakInside(node, limiter) {
			let breakNode;

			if (node === limiter) {
				return;
			}

			while (node.parentNode) {
				node = node.parentNode;

				if (node === limiter) {
					break;
				}

				if (window.getComputedStyle(node)["break-inside"] === "avoid") {
					breakNode = node;
					break;
				}

			}
			return breakNode;
		}

		createBreakToken(overflow, rendered, source) {
			let container = overflow.startContainer;
			let offset = overflow.startOffset;
			let node, renderedNode, parent, index, temp;

			if (isElement(container)) {
				temp = child(container, offset);

				if (isElement(temp)) {
					renderedNode = findElement(temp, rendered);

					if (!renderedNode) {
						// Find closest element with data-ref
						let prevNode = prevValidNode(temp);
						if (!isElement(prevNode)) {
							prevNode = prevNode.parentElement;
						}
						renderedNode = findElement(prevNode, rendered);
						// Check if temp is the last rendered node at its level.
						if (!temp.nextSibling) {
							// We need to ensure that the previous sibling of temp is fully rendered.
							const renderedNodeFromSource = findElement(renderedNode, source);
							const walker = document.createTreeWalker(renderedNodeFromSource, NodeFilter.SHOW_ELEMENT);
							const lastChildOfRenderedNodeFromSource = walker.lastChild();
							const lastChildOfRenderedNodeMatchingFromRendered = findElement(lastChildOfRenderedNodeFromSource, rendered);
							// Check if we found that the last child in source
							if (!lastChildOfRenderedNodeMatchingFromRendered) {
								// Pending content to be rendered before virtual break token
								return;
							}
							// Otherwise we will return a break token as per below
						}
						// renderedNode is actually the last unbroken box that does not overflow.
						// Break Token is therefore the next sibling of renderedNode within source node.
						node = findElement(renderedNode, source).nextSibling;
						offset = 0;
					} else {
						node = findElement(renderedNode, source);
						offset = 0;
					}
				} else {
					renderedNode = findElement(container, rendered);

					if (!renderedNode) {
						renderedNode = findElement(prevValidNode(container), rendered);
					}

					parent = findElement(renderedNode, source);
					index = indexOfTextNode(temp, parent);
					// No seperatation for the first textNode of an element
					if(index === 0) {
						node = parent;
						offset = 0;
					} else {
						node = child(parent, index);
						offset = 0;
					}
				}
			} else {
				renderedNode = findElement(container.parentNode, rendered);

				if (!renderedNode) {
					renderedNode = findElement(prevValidNode(container.parentNode), rendered);
				}

				parent = findElement(renderedNode, source);
				index = indexOfTextNode(container, parent);

				if (index === -1) {
					return;
				}

				node = child(parent, index);

				offset += node.textContent.indexOf(container.textContent);
			}

			if (!node) {
				return;
			}

			return new BreakToken(
				node,
				offset
			);

		}

		findBreakToken(rendered, source, bounds = this.bounds, prevBreakToken, extract = true) {
			let overflow = this.findOverflow(rendered, bounds);
			let breakToken, breakLetter;

			let overflowHooks = this.hooks.onOverflow.triggerSync(overflow, rendered, bounds, this);
			overflowHooks.forEach((newOverflow) => {
				if (typeof newOverflow != "undefined") {
					overflow = newOverflow;
				}
			});

			if (overflow) {
				breakToken = this.createBreakToken(overflow, rendered, source);
				// breakToken is nullable
				let breakHooks = this.hooks.onBreakToken.triggerSync(breakToken, overflow, rendered, this);
				breakHooks.forEach((newToken) => {
					if (typeof newToken != "undefined") {
						breakToken = newToken;
					}
				});

				// Stop removal if we are in a loop
				if (breakToken && breakToken.equals(prevBreakToken)) {
					return breakToken;
				}

				if (breakToken && breakToken["node"] && breakToken["offset"] && breakToken["node"].textContent) {
					breakLetter = breakToken["node"].textContent.charAt(breakToken["offset"]);
				} else {
					breakLetter = undefined;
				}

				if (breakToken && breakToken.node && extract) {
					let removed = this.removeOverflow(overflow, breakLetter);
					this.hooks && this.hooks.afterOverflowRemoved.trigger(removed, rendered, this);
				}

			}
			return breakToken;
		}

		hasOverflow(element, bounds = this.bounds) {
			let constrainingElement = element && element.parentNode; // this gets the element, instead of the wrapper for the width workaround
			let {width, height} = element.getBoundingClientRect();
			let scrollWidth = constrainingElement ? constrainingElement.scrollWidth : 0;
			let scrollHeight = constrainingElement ? constrainingElement.scrollHeight : 0;
			return Math.max(Math.floor(width), scrollWidth) > Math.round(bounds.width) ||
				Math.max(Math.floor(height), scrollHeight) > Math.round(bounds.height);
		}

		findOverflow(rendered, bounds = this.bounds, gap = this.gap) {
			if (!this.hasOverflow(rendered, bounds)) return;

			let start = Math.floor(bounds.left);
			let end = Math.round(bounds.right + gap);
			let vStart = Math.round(bounds.top);
			let vEnd = Math.round(bounds.bottom);
			let range;

			let walker = walk$2(rendered.firstChild, rendered);

			// Find Start
			let next, done, node, offset, skip, breakAvoid, prev, br;
			while (!done) {
				next = walker.next();
				done = next.done;
				node = next.value;
				skip = false;
				breakAvoid = false;
				prev = undefined;
				br = undefined;

				if (node) {
					let pos = getBoundingClientRect(node);
					let left = Math.round(pos.left);
					let right = Math.floor(pos.right);
					let top = Math.round(pos.top);
					let bottom = Math.floor(pos.bottom);

					if (!range && (left >= end || top >= vEnd)) {
						// Check if it is a float
						let isFloat = false;

						// Check if the node is inside a break-inside: avoid table cell
						const insideTableCell = parentOf(node, "TD", rendered);
						if (insideTableCell && window.getComputedStyle(insideTableCell)["break-inside"] === "avoid") {
							// breaking inside a table cell produces unexpected result, as a workaround, we forcibly avoid break inside in a cell.
							// But we take the whole row, not just the cell that is causing the break.
							prev = insideTableCell.parentElement;
						} else if (isElement(node)) {
							let styles = window.getComputedStyle(node);
							isFloat = styles.getPropertyValue("float") !== "none";
							skip = styles.getPropertyValue("break-inside") === "avoid";
							breakAvoid = node.dataset.breakBefore === "avoid" || node.dataset.previousBreakAfter === "avoid";
							prev = breakAvoid && nodeBefore(node, rendered);
							br = node.tagName === "BR" || node.tagName === "WBR";
						}

						let tableRow;
						if (node.nodeName === "TR") {
							tableRow = node;
						} else {
							tableRow = parentOf(node, "TR", rendered);
						}
						if (tableRow) {
							// honor break-inside="avoid" in parent tbody/thead
							let container = tableRow.parentElement;
							if (["TBODY", "THEAD"].includes(container.nodeName)) {
								let styles = window.getComputedStyle(container);
								if (styles.getPropertyValue("break-inside") === "avoid") prev = container;
							}

							// Check if the node is inside a row with a rowspan
							const table = parentOf(tableRow, "TABLE", rendered);
							const rowspan = table.querySelector("[colspan]");
							if (table && rowspan) {
								let columnCount = 0;
								for (const cell of Array.from(table.rows[0].cells)) {
									columnCount += parseInt(cell.getAttribute("colspan") || "1");
								}
								if (tableRow.cells.length !== columnCount) {
									let previousRow = tableRow.previousElementSibling;
									let previousRowColumnCount;
									while (previousRow !== null) {
										previousRowColumnCount = 0;
										for (const cell of Array.from(previousRow.cells)) {
											previousRowColumnCount += parseInt(cell.getAttribute("colspan") || "1");
										}
										if (previousRowColumnCount === columnCount) {
											break;
										}
										previousRow = previousRow.previousElementSibling;
									}
									if (previousRowColumnCount === columnCount) {
										prev = previousRow;
									}
								}
							}
						}

						if (prev) {
							range = document.createRange();
							range.selectNode(prev);
							break;
						}

						if (!br && !isFloat && isElement(node)) {
							range = document.createRange();
							range.selectNode(node);
							break;
						}

						if (isText(node) && node.textContent.trim().length) {
							range = document.createRange();
							range.selectNode(node);
							break;
						}

					}

					if (!range && isText(node) &&
						node.textContent.trim().length &&
						!breakInsideAvoidParentNode(node.parentNode)) {

						let rects = getClientRects(node);
						let rect;
						left = 0;
						top = 0;
						for (var i = 0; i != rects.length; i++) {
							rect = rects[i];
							if (rect.width > 0 && (!left || rect.left > left)) {
								left = rect.left;
							}
							if (rect.height > 0 && (!top || rect.top > top)) {
								top = rect.top;
							}
						}

						if (left >= end || top >= vEnd) {
							range = document.createRange();
							offset = this.textBreak(node, start, end, vStart, vEnd);
							if (!offset) {
								range = undefined;
							} else {
								range.setStart(node, offset);
							}
							break;
						}
					}

					// Skip children
					if (skip || (right <= end && bottom <= vEnd)) {
						next = nodeAfter(node, rendered);
						if (next) {
							walker = walk$2(next, rendered);
						}

					}

				}
			}

			// Find End
			if (range) {
				range.setEndAfter(rendered.lastChild);
				return range;
			}

		}

		findEndToken(rendered, source) {
			if (rendered.childNodes.length === 0) {
				return;
			}

			let lastChild = rendered.lastChild;

			let lastNodeIndex;
			while (lastChild && lastChild.lastChild) {
				if (!validNode(lastChild)) {
					// Only get elements with refs
					lastChild = lastChild.previousSibling;
				} else if (!validNode(lastChild.lastChild)) {
					// Deal with invalid dom items
					lastChild = prevValidNode(lastChild.lastChild);
					break;
				} else {
					lastChild = lastChild.lastChild;
				}
			}

			if (isText(lastChild)) {

				if (lastChild.parentNode.dataset.ref) {
					lastNodeIndex = indexOf$2(lastChild);
					lastChild = lastChild.parentNode;
				} else {
					lastChild = lastChild.previousSibling;
				}
			}

			let original = findElement(lastChild, source);

			if (lastNodeIndex) {
				original = original.childNodes[lastNodeIndex];
			}

			let after = nodeAfter(original);

			return this.breakAt(after);
		}

		textBreak(node, start, end, vStart, vEnd) {
			let wordwalker = words(node);
			let left = 0;
			let right = 0;
			let top = 0;
			let bottom = 0;
			let word, next, done, pos;
			let offset;
			while (!done) {
				next = wordwalker.next();
				word = next.value;
				done = next.done;

				if (!word) {
					break;
				}

				pos = getBoundingClientRect(word);

				left = Math.floor(pos.left);
				right = Math.floor(pos.right);
				top = Math.floor(pos.top);
				bottom = Math.floor(pos.bottom);

				if (left >= end || top >= vEnd) {
					offset = word.startOffset;
					break;
				}

				if (right > end || bottom > vEnd) {
					let letterwalker = letters(word);
					let letter, nextLetter, doneLetter;

					while (!doneLetter) {
						nextLetter = letterwalker.next();
						letter = nextLetter.value;
						doneLetter = nextLetter.done;

						if (!letter) {
							break;
						}

						pos = getBoundingClientRect(letter);
						left = Math.floor(pos.left);
						top = Math.floor(pos.top);

						if (left >= end || top >= vEnd) {
							offset = letter.startOffset;
							done = true;

							break;
						}
					}
				}

			}

			return offset;
		}

		removeOverflow(overflow, breakLetter) {
			let {startContainer} = overflow;
			let extracted = overflow.extractContents();

			this.hyphenateAtBreak(startContainer, breakLetter);

			return extracted;
		}

		hyphenateAtBreak(startContainer, breakLetter) {
			if (isText(startContainer)) {
				let startText = startContainer.textContent;
				let prevLetter = startText[startText.length - 1];

				// Add a hyphen if previous character is a letter or soft hyphen
				if (
					(breakLetter && /^\w|\u00AD$/.test(prevLetter) && /^\w|\u00AD$/.test(breakLetter)) ||
					(!breakLetter && /^\w|\u00AD$/.test(prevLetter))
				) {
					startContainer.parentNode.classList.add("pagedjs_hyphen");
					startContainer.textContent += this.settings.hyphenGlyph || "\u2011";
				}
			}
		}

		equalTokens(a, b) {
			if (!a || !b) {
				return false;
			}
			if (a["node"] && b["node"] && a["node"] !== b["node"]) {
				return false;
			}
			if (a["offset"] && b["offset"] && a["offset"] !== b["offset"]) {
				return false;
			}
			return true;
		}
	}

	EventEmitter(Layout.prototype);

	/**
	 * Render a page
	 * @class
	 */
	class Page {
		constructor(pagesArea, pageTemplate, blank, hooks, options) {
			this.pagesArea = pagesArea;
			this.pageTemplate = pageTemplate;
			this.blank = blank;

			this.width = undefined;
			this.height = undefined;

			this.hooks = hooks;

			this.settings = options || {};

			// this.element = this.create(this.pageTemplate);
		}

		create(template, after) {
			//let documentFragment = document.createRange().createContextualFragment( TEMPLATE );
			//let page = documentFragment.children[0];
			let clone = document.importNode(this.pageTemplate.content, true);

			let page, index;
			if (after) {
				this.pagesArea.insertBefore(clone, after.nextElementSibling);
				index = Array.prototype.indexOf.call(this.pagesArea.children, after.nextElementSibling);
				page = this.pagesArea.children[index];
			} else {
				this.pagesArea.appendChild(clone);
				page = this.pagesArea.lastChild;
			}

			let pagebox = page.querySelector(".pagedjs_pagebox");
			let area = page.querySelector(".pagedjs_page_content");
			let footnotesArea = page.querySelector(".pagedjs_footnote_area");


			let size = area.getBoundingClientRect();


			area.style.columnWidth = Math.round(size.width) + "px";
			area.style.columnGap = "calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset))";
			// area.style.overflow = "scroll";

			this.width = Math.round(size.width);
			this.height = Math.round(size.height);

			this.element = page;
			this.pagebox = pagebox;
			this.area = area;
			this.footnotesArea = footnotesArea;

			return page;
		}

		createWrapper() {
			let wrapper = document.createElement("div");

			this.area.appendChild(wrapper);

			this.wrapper = wrapper;

			return wrapper;
		}

		index(pgnum) {
			this.position = pgnum;

			let page = this.element;
			// let pagebox = this.pagebox;

			let index = pgnum + 1;

			let id = `page-${index}`;

			this.id = id;

			// page.dataset.pageNumber = index;

			page.dataset.pageNumber = index;
			page.setAttribute("id", id);

			if (this.name) {
				page.classList.add("pagedjs_" + this.name + "_page");
			}

			if (this.blank) {
				page.classList.add("pagedjs_blank_page");
			}

			if (pgnum === 0) {
				page.classList.add("pagedjs_first_page");
			}

			if (pgnum % 2 !== 1) {
				page.classList.remove("pagedjs_left_page");
				page.classList.add("pagedjs_right_page");
			} else {
				page.classList.remove("pagedjs_right_page");
				page.classList.add("pagedjs_left_page");
			}
		}

		/*
		size(width, height) {
			if (width === this.width && height === this.height) {
				return;
			}
			this.width = width;
			this.height = height;

			this.element.style.width = Math.round(width) + "px";
			this.element.style.height = Math.round(height) + "px";
			this.element.style.columnWidth = Math.round(width) + "px";
		}
		*/

		async layout(contents, breakToken, maxChars) {

			this.clear();

			this.startToken = breakToken;

			let settings = this.settings;
			if (!settings.maxChars && maxChars) {
				settings.maxChars = maxChars;
			}

			this.layoutMethod = new Layout(this.area, this.hooks, settings);

			let renderResult = await this.layoutMethod.renderTo(this.wrapper, contents, breakToken);
			let newBreakToken = renderResult.breakToken;

			this.addListeners(contents);

			this.endToken = newBreakToken;

			return newBreakToken;
		}

		async append(contents, breakToken) {

			if (!this.layoutMethod) {
				return this.layout(contents, breakToken);
			}

			let renderResult = await this.layoutMethod.renderTo(this.wrapper, contents, breakToken);
			let newBreakToken = renderResult.breakToken;

			this.endToken = newBreakToken;

			return newBreakToken;
		}

		getByParent(ref, entries) {
			let e;
			for (var i = 0; i < entries.length; i++) {
				e = entries[i];
				if (e.dataset.ref === ref) {
					return e;
				}
			}
		}

		onOverflow(func) {
			this._onOverflow = func;
		}

		onUnderflow(func) {
			this._onUnderflow = func;
		}

		clear() {
			this.removeListeners();
			this.wrapper && this.wrapper.remove();
			this.createWrapper();
		}

		addListeners(contents) {
			if (typeof ResizeObserver !== "undefined") {
				this.addResizeObserver(contents);
			} else {
				this._checkOverflowAfterResize = this.checkOverflowAfterResize.bind(this, contents);
				this.element.addEventListener("overflow", this._checkOverflowAfterResize, false);
				this.element.addEventListener("underflow", this._checkOverflowAfterResize, false);
			}
			// TODO: fall back to mutation observer?

			this._onScroll = function () {
				if (this.listening) {
					this.element.scrollLeft = 0;
				}
			}.bind(this);

			// Keep scroll left from changing
			this.element.addEventListener("scroll", this._onScroll);

			this.listening = true;

			return true;
		}

		removeListeners() {
			this.listening = false;

			if (typeof ResizeObserver !== "undefined" && this.ro) {
				this.ro.disconnect();
			} else if (this.element) {
				this.element.removeEventListener("overflow", this._checkOverflowAfterResize, false);
				this.element.removeEventListener("underflow", this._checkOverflowAfterResize, false);
			}

			this.element && this.element.removeEventListener("scroll", this._onScroll);

		}

		addResizeObserver(contents) {
			let wrapper = this.wrapper;
			let prevHeight = wrapper.getBoundingClientRect().height;
			this.ro = new ResizeObserver(entries => {

				if (!this.listening) {
					return;
				}
				requestAnimationFrame(() => {
					for (let entry of entries) {
						const cr = entry.contentRect;

						if (cr.height > prevHeight) {
							this.checkOverflowAfterResize(contents);
							prevHeight = wrapper.getBoundingClientRect().height;
						} else if (cr.height < prevHeight) { // TODO: calc line height && (prevHeight - cr.height) >= 22
							this.checkUnderflowAfterResize(contents);
							prevHeight = cr.height;
						}
					}
				});
			});

			this.ro.observe(wrapper);
		}

		checkOverflowAfterResize(contents) {
			if (!this.listening || !this.layoutMethod) {
				return;
			}

			let newBreakToken = this.layoutMethod.findBreakToken(this.wrapper, contents, this.startToken);

			if (newBreakToken) {
				this.endToken = newBreakToken;
				this._onOverflow && this._onOverflow(newBreakToken);
			}
		}

		checkUnderflowAfterResize(contents) {
			if (!this.listening || !this.layoutMethod) {
				return;
			}

			let endToken = this.layoutMethod.findEndToken(this.wrapper, contents);

			if (endToken) {
				this._onUnderflow && this._onUnderflow(endToken);
			}
		}


		destroy() {
			this.removeListeners();

			this.element.remove();

			this.element = undefined;
			this.wrapper = undefined;
		}
	}

	EventEmitter(Page.prototype);

	/**
	 * Render a flow of text offscreen
	 * @class
	 */
	class ContentParser {

		constructor(content, cb) {
			if (content && content.nodeType) {
				// handle dom
				this.dom = this.add(content);
			} else if (typeof content === "string") {
				this.dom = this.parse(content);
			}

			return this.dom;
		}

		parse(markup, mime) {
			let range = document.createRange();
			let fragment = range.createContextualFragment(markup);

			this.addRefs(fragment);

			return fragment;
		}

		add(contents) {
			// let fragment = document.createDocumentFragment();
			//
			// let children = [...contents.childNodes];
			// for (let child of children) {
			// 	let clone = child.cloneNode(true);
			// 	fragment.appendChild(clone);
			// }

			this.addRefs(contents);

			return contents;
		}

		addRefs(content) {
			var treeWalker = document.createTreeWalker(
				content,
				NodeFilter.SHOW_ELEMENT,
				null,
				false
			);

			let node = treeWalker.nextNode();
			while(node) {

				if (!node.hasAttribute("data-ref")) {
					let uuid = UUID();
					node.setAttribute("data-ref", uuid);
				}

				if (node.id) {
					node.setAttribute("data-id", node.id);
				}

				// node.setAttribute("data-children", node.childNodes.length);

				// node.setAttribute("data-text", node.textContent.trim().length);
				node = treeWalker.nextNode();
			}
		}

		find(ref) {
			return this.refs[ref];
		}

		destroy() {
			this.refs = undefined;
			this.dom = undefined;
		}
	}

	/**
	 * Queue for handling tasks one at a time
	 * @class
	 * @param {scope} context what this will resolve to in the tasks
	 */
	class Queue {
		constructor(context){
			this._q = [];
			this.context = context;
			this.tick = requestAnimationFrame;
			this.running = false;
			this.paused = false;
		}

		/**
		 * Add an item to the queue
		 * @return {Promise} enqueued
		 */
		enqueue() {
			var deferred, promise;
			var queued;
			var task = [].shift.call(arguments);
			var args = arguments;

			// Handle single args without context
			// if(args && !Array.isArray(args)) {
			//   args = [args];
			// }
			if(!task) {
				throw new Error("No Task Provided");
			}

			if(typeof task === "function"){

				deferred = new defer();
				promise = deferred.promise;

				queued = {
					"task" : task,
					"args"     : args,
					//"context"  : context,
					"deferred" : deferred,
					"promise" : promise
				};

			} else {
				// Task is a promise
				queued = {
					"promise" : task
				};

			}

			this._q.push(queued);

			// Wait to start queue flush
			if (this.paused == false && !this.running) {
				this.run();
			}

			return queued.promise;
		}

		/**
		 * Run one item
		 * @return {Promise} dequeued
		 */
		dequeue(){
			var inwait, task, result;

			if(this._q.length && !this.paused) {
				inwait = this._q.shift();
				task = inwait.task;
				if(task){
					// console.log(task)

					result = task.apply(this.context, inwait.args);

					if(result && typeof result["then"] === "function") {
						// Task is a function that returns a promise
						return result.then(function(){
							inwait.deferred.resolve.apply(this.context, arguments);
						}.bind(this), function() {
							inwait.deferred.reject.apply(this.context, arguments);
						}.bind(this));
					} else {
						// Task resolves immediately
						inwait.deferred.resolve.apply(this.context, result);
						return inwait.promise;
					}



				} else if(inwait.promise) {
					// Task is a promise
					return inwait.promise;
				}

			} else {
				inwait = new defer();
				inwait.deferred.resolve();
				return inwait.promise;
			}

		}

		// Run All Immediately
		dump(){
			while(this._q.length) {
				this.dequeue();
			}
		}

		/**
		 * Run all tasks sequentially, at convince
		 * @return {Promise} all run
		 */
		run(){

			if(!this.running){
				this.running = true;
				this.defered = new defer();
			}

			this.tick.call(window, () => {

				if(this._q.length) {

					this.dequeue()
						.then(function(){
							this.run();
						}.bind(this));

				} else {
					this.defered.resolve();
					this.running = undefined;
				}

			});

			// Unpause
			if(this.paused == true) {
				this.paused = false;
			}

			return this.defered.promise;
		}

		/**
		 * Flush all, as quickly as possible
		 * @return {Promise} ran
		 */
		flush(){

			if(this.running){
				return this.running;
			}

			if(this._q.length) {
				this.running = this.dequeue()
					.then(function(){
						this.running = undefined;
						return this.flush();
					}.bind(this));

				return this.running;
			}

		}

		/**
		 * Clear all items in wait
		 * @return {void}
		 */
		clear(){
			this._q = [];
		}

		/**
		 * Get the number of tasks in the queue
		 * @return {number} tasks
		 */
		length(){
			return this._q.length;
		}

		/**
		 * Pause a running queue
		 * @return {void}
		 */
		pause(){
			this.paused = true;
		}

		/**
		 * End the queue
		 * @return {void}
		 */
		stop(){
			this._q = [];
			this.running = false;
			this.paused = true;
		}
	}

	const TEMPLATE = `
<div class="pagedjs_page">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox">
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top">
				<div class="pagedjs_margin pagedjs_margin-top-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content"></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>`;

	/**
	 * Chop up text into flows
	 * @class
	 */
	class Chunker {
		constructor(content, renderTo, options) {
			// this.preview = preview;

			this.settings = options || {};

			this.hooks = {};
			this.hooks.beforeParsed = new Hook(this);
			this.hooks.filter = new Hook(this);
			this.hooks.afterParsed = new Hook(this);
			this.hooks.beforePageLayout = new Hook(this);
			this.hooks.onPageLayout = new Hook(this);
			this.hooks.layout = new Hook(this);
			this.hooks.renderNode = new Hook(this);
			this.hooks.layoutNode = new Hook(this);
			this.hooks.onOverflow = new Hook(this);
			this.hooks.afterOverflowRemoved = new Hook(this);
			this.hooks.onBreakToken = new Hook();
			this.hooks.beforeRenderResult = new Hook(this);
			this.hooks.afterPageLayout = new Hook(this);
			this.hooks.finalizePage = new Hook(this);
			this.hooks.afterRendered = new Hook(this);

			this.pages = [];
			this.total = 0;

			this.q = new Queue(this);
			this.stopped = false;
			this.rendered = false;

			this.content = content;

			this.charsPerBreak = [];
			this.maxChars;

			if (content) {
				this.flow(content, renderTo);
			}
		}

		setup(renderTo) {
			this.pagesArea = document.createElement("div");
			this.pagesArea.classList.add("pagedjs_pages");

			if (renderTo) {
				renderTo.appendChild(this.pagesArea);
			} else {
				document.querySelector("body").appendChild(this.pagesArea);
			}

			this.pageTemplate = document.createElement("template");
			this.pageTemplate.innerHTML = TEMPLATE;

		}

		async flow(content, renderTo) {
			let parsed;

			await this.hooks.beforeParsed.trigger(content, this);

			parsed = new ContentParser(content);

			this.hooks.filter.triggerSync(parsed);

			this.source = parsed;
			this.breakToken = undefined;

			if (this.pagesArea && this.pageTemplate) {
				this.q.clear();
				this.removePages();
			} else {
				this.setup(renderTo);
			}

			this.emit("rendering", parsed);

			await this.hooks.afterParsed.trigger(parsed, this);

			await this.loadFonts();

			let rendered = await this.render(parsed, this.breakToken);
			while (rendered.canceled) {
				this.start();
				rendered = await this.render(parsed, this.breakToken);
			}

			this.rendered = true;
			this.pagesArea.style.setProperty("--pagedjs-page-count", this.total);

			await this.hooks.afterRendered.trigger(this.pages, this);

			this.emit("rendered", this.pages);



			return this;
		}

		// oversetPages() {
		// 	let overset = [];
		// 	for (let i = 0; i < this.pages.length; i++) {
		// 		let page = this.pages[i];
		// 		if (page.overset) {
		// 			overset.push(page);
		// 			// page.overset = false;
		// 		}
		// 	}
		// 	return overset;
		// }
		//
		// async handleOverset(parsed) {
		// 	let overset = this.oversetPages();
		// 	if (overset.length) {
		// 		console.log("overset", overset);
		// 		let index = this.pages.indexOf(overset[0]) + 1;
		// 		console.log("INDEX", index);
		//
		// 		// Remove pages
		// 		// this.removePages(index);
		//
		// 		// await this.render(parsed, overset[0].overset);
		//
		// 		// return this.handleOverset(parsed);
		// 	}
		// }

		async render(parsed, startAt) {
			let renderer = this.layout(parsed, startAt);

			let done = false;
			let result;
			while (!done) {
				result = await this.q.enqueue(() => { return this.renderAsync(renderer); });
				done = result.done;
			}

			return result;
		}

		start() {
			this.rendered = false;
			this.stopped = false;
		}

		stop() {
			this.stopped = true;
			// this.q.clear();
		}

		renderOnIdle(renderer) {
			return new Promise(resolve => {
				requestIdleCallback(async () => {
					if (this.stopped) {
						return resolve({ done: true, canceled: true });
					}
					let result = await renderer.next();
					if (this.stopped) {
						resolve({ done: true, canceled: true });
					} else {
						resolve(result);
					}
				});
			});
		}

		async renderAsync(renderer) {
			if (this.stopped) {
				return { done: true, canceled: true };
			}
			let result = await renderer.next();
			if (this.stopped) {
				return { done: true, canceled: true };
			} else {
				return result;
			}
		}

		async handleBreaks(node, force) {
			let currentPage = this.total + 1;
			let currentPosition = currentPage % 2 === 0 ? "left" : "right";
			// TODO: Recto and Verso should reverse for rtl languages
			let currentSide = currentPage % 2 === 0 ? "verso" : "recto";
			let previousBreakAfter;
			let breakBefore;
			let page;

			if (currentPage === 1) {
				return;
			}

			if (node &&
					typeof node.dataset !== "undefined" &&
					typeof node.dataset.previousBreakAfter !== "undefined") {
				previousBreakAfter = node.dataset.previousBreakAfter;
			}

			if (node &&
					typeof node.dataset !== "undefined" &&
					typeof node.dataset.breakBefore !== "undefined") {
				breakBefore = node.dataset.breakBefore;
			}

			if (force) {
				page = this.addPage(true);
			} else if( previousBreakAfter &&
					(previousBreakAfter === "left" || previousBreakAfter === "right") &&
					previousBreakAfter !== currentPosition) {
				page = this.addPage(true);
			} else if( previousBreakAfter &&
					(previousBreakAfter === "verso" || previousBreakAfter === "recto") &&
					previousBreakAfter !== currentSide) {
				page = this.addPage(true);
			} else if( breakBefore &&
					(breakBefore === "left" || breakBefore === "right") &&
					breakBefore !== currentPosition) {
				page = this.addPage(true);
			} else if( breakBefore &&
					(breakBefore === "verso" || breakBefore === "recto") &&
					breakBefore !== currentSide) {
				page = this.addPage(true);
			}

			if (page) {
				await this.hooks.beforePageLayout.trigger(page, undefined, undefined, this);
				this.emit("page", page);
				// await this.hooks.layout.trigger(page.element, page, undefined, this);
				await this.hooks.afterPageLayout.trigger(page.element, page, undefined, this);
				await this.hooks.finalizePage.trigger(page.element, page, undefined, this);
				this.emit("renderedPage", page);
			}
		}

		async *layout(content, startAt) {
			let breakToken = startAt || false;
			let tokens = [];

			while (breakToken !== undefined && (true)) {

				if (breakToken && breakToken.node) {
					await this.handleBreaks(breakToken.node);
				} else {
					await this.handleBreaks(content.firstChild);
				}

				let page = this.addPage();

				await this.hooks.beforePageLayout.trigger(page, content, breakToken, this);
				this.emit("page", page);

				// Layout content in the page, starting from the breakToken
				breakToken = await page.layout(content, breakToken, this.maxChars);

				if (breakToken) {
					let newToken = breakToken.toJSON(true);
					if (tokens.lastIndexOf(newToken) > -1) {
						// loop
						let err = new OverflowContentError("Layout repeated", [breakToken.node]);
						console.error("Layout repeated at: ", breakToken.node);
						return err;
					} else {
						tokens.push(newToken);
					}
				}

				await this.hooks.afterPageLayout.trigger(page.element, page, breakToken, this);
				await this.hooks.finalizePage.trigger(page.element, page, undefined, this);
				this.emit("renderedPage", page);

				this.recoredCharLength(page.wrapper.textContent.length);

				yield breakToken;

				// Stop if we get undefined, showing we have reached the end of the content
			}


		}

		recoredCharLength(length) {
			if (length === 0) {
				return;
			}

			this.charsPerBreak.push(length);

			// Keep the length of the last few breaks
			if (this.charsPerBreak.length > 4) {
				this.charsPerBreak.shift();
			}

			this.maxChars = this.charsPerBreak.reduce((a, b) => a + b, 0) / (this.charsPerBreak.length);
		}

		removePages(fromIndex=0) {

			if (fromIndex >= this.pages.length) {
				return;
			}

			// Remove pages
			for (let i = fromIndex; i < this.pages.length; i++) {
				this.pages[i].destroy();
			}

			if (fromIndex > 0) {
				this.pages.splice(fromIndex);
			} else {
				this.pages = [];
			}

			this.total = this.pages.length;
		}

		addPage(blank) {
			let lastPage = this.pages[this.pages.length - 1];
			// Create a new page from the template
			let page = new Page(this.pagesArea, this.pageTemplate, blank, this.hooks, this.settings);

			this.pages.push(page);

			// Create the pages
			page.create(undefined, lastPage && lastPage.element);

			page.index(this.total);

			if (!blank) {
				// Listen for page overflow
				page.onOverflow((overflowToken) => {
					console.warn("overflow on", page.id, overflowToken);

					// Only reflow while rendering
					if (this.rendered) {
						return;
					}

					let index = this.pages.indexOf(page) + 1;

					// Stop the rendering
					this.stop();

					// Set the breakToken to resume at
					this.breakToken = overflowToken;

					// Remove pages
					this.removePages(index);

					if (this.rendered === true) {
						this.rendered = false;

						this.q.enqueue(async () => {

							this.start();

							await this.render(this.source, this.breakToken);

							this.rendered = true;

						});
					}


				});

				page.onUnderflow((overflowToken) => {
					// console.log("underflow on", page.id, overflowToken);

					// page.append(this.source, overflowToken);

				});
			}

			this.total = this.pages.length;

			return page;
		}
		/*
		insertPage(index, blank) {
			let lastPage = this.pages[index];
			// Create a new page from the template
			let page = new Page(this.pagesArea, this.pageTemplate, blank, this.hooks);

			let total = this.pages.splice(index, 0, page);

			// Create the pages
			page.create(undefined, lastPage && lastPage.element);

			page.index(index + 1);

			for (let i = index + 2; i < this.pages.length; i++) {
				this.pages[i].index(i);
			}

			if (!blank) {
				// Listen for page overflow
				page.onOverflow((overflowToken) => {
					if (total < this.pages.length) {
						this.pages[total].layout(this.source, overflowToken);
					} else {
						let newPage = this.addPage();
						newPage.layout(this.source, overflowToken);
					}
				});

				page.onUnderflow(() => {
					// console.log("underflow on", page.id);
				});
			}

			this.total += 1;

			return page;
		}
		*/

		async clonePage(originalPage) {
			let lastPage = this.pages[this.pages.length - 1];

			let page = new Page(this.pagesArea, this.pageTemplate, false, this.hooks);

			this.pages.push(page);

			// Create the pages
			page.create(undefined, lastPage && lastPage.element);

			page.index(this.total);

			await this.hooks.beforePageLayout.trigger(page, undefined, undefined, this);
			this.emit("page", page);

			for (const className of originalPage.element.classList) {
				if (className !== "pagedjs_left_page" && className !== "pagedjs_right_page") {
					page.element.classList.add(className);
				}
			}

			await this.hooks.afterPageLayout.trigger(page.element, page, undefined, this);
			await this.hooks.finalizePage.trigger(page.element, page, undefined, this);
			this.emit("renderedPage", page);
		}

		loadFonts() {
			let fontPromises = [];
			(document.fonts || []).forEach((fontFace) => {
				if (fontFace.status !== "loaded") {
					let fontLoaded = fontFace.load().then((r) => {
						return fontFace.family;
					}, (r) => {
						console.warn("Failed to preload font-family:", fontFace.family);
						return fontFace.family;
					});
					fontPromises.push(fontLoaded);
				}
			});
			return Promise.all(fontPromises).catch((err) => {
				console.warn(err);
			});
		}

		destroy() {
			this.pagesArea.remove();
			this.pageTemplate.remove();
		}

	}

	EventEmitter(Chunker.prototype);

	var syntax = {exports: {}};

	var create$4 = {};

	//
	//                              list
	//                            ┌──────┐
	//             ┌──────────────┼─head │
	//             │              │ tail─┼──────────────┐
	//             │              └──────┘              │
	//             ▼                                    ▼
	//            item        item        item        item
	//          ┌──────┐    ┌──────┐    ┌──────┐    ┌──────┐
	//  null ◀──┼─prev │◀───┼─prev │◀───┼─prev │◀───┼─prev │
	//          │ next─┼───▶│ next─┼───▶│ next─┼───▶│ next─┼──▶ null
	//          ├──────┤    ├──────┤    ├──────┤    ├──────┤
	//          │ data │    │ data │    │ data │    │ data │
	//          └──────┘    └──────┘    └──────┘    └──────┘
	//

	function createItem(data) {
	    return {
	        prev: null,
	        next: null,
	        data: data
	    };
	}

	function allocateCursor(node, prev, next) {
	    var cursor;

	    if (cursors !== null) {
	        cursor = cursors;
	        cursors = cursors.cursor;
	        cursor.prev = prev;
	        cursor.next = next;
	        cursor.cursor = node.cursor;
	    } else {
	        cursor = {
	            prev: prev,
	            next: next,
	            cursor: node.cursor
	        };
	    }

	    node.cursor = cursor;

	    return cursor;
	}

	function releaseCursor(node) {
	    var cursor = node.cursor;

	    node.cursor = cursor.cursor;
	    cursor.prev = null;
	    cursor.next = null;
	    cursor.cursor = cursors;
	    cursors = cursor;
	}

	var cursors = null;
	var List$6 = function() {
	    this.cursor = null;
	    this.head = null;
	    this.tail = null;
	};

	List$6.createItem = createItem;
	List$6.prototype.createItem = createItem;

	List$6.prototype.updateCursors = function(prevOld, prevNew, nextOld, nextNew) {
	    var cursor = this.cursor;

	    while (cursor !== null) {
	        if (cursor.prev === prevOld) {
	            cursor.prev = prevNew;
	        }

	        if (cursor.next === nextOld) {
	            cursor.next = nextNew;
	        }

	        cursor = cursor.cursor;
	    }
	};

	List$6.prototype.getSize = function() {
	    var size = 0;
	    var cursor = this.head;

	    while (cursor) {
	        size++;
	        cursor = cursor.next;
	    }

	    return size;
	};

	List$6.prototype.fromArray = function(array) {
	    var cursor = null;

	    this.head = null;

	    for (var i = 0; i < array.length; i++) {
	        var item = createItem(array[i]);

	        if (cursor !== null) {
	            cursor.next = item;
	        } else {
	            this.head = item;
	        }

	        item.prev = cursor;
	        cursor = item;
	    }

	    this.tail = cursor;

	    return this;
	};

	List$6.prototype.toArray = function() {
	    var cursor = this.head;
	    var result = [];

	    while (cursor) {
	        result.push(cursor.data);
	        cursor = cursor.next;
	    }

	    return result;
	};

	List$6.prototype.toJSON = List$6.prototype.toArray;

	List$6.prototype.isEmpty = function() {
	    return this.head === null;
	};

	List$6.prototype.first = function() {
	    return this.head && this.head.data;
	};

	List$6.prototype.last = function() {
	    return this.tail && this.tail.data;
	};

	List$6.prototype.each = function(fn, context) {
	    var item;

	    if (context === undefined) {
	        context = this;
	    }

	    // push cursor
	    var cursor = allocateCursor(this, null, this.head);

	    while (cursor.next !== null) {
	        item = cursor.next;
	        cursor.next = item.next;

	        fn.call(context, item.data, item, this);
	    }

	    // pop cursor
	    releaseCursor(this);
	};

	List$6.prototype.forEach = List$6.prototype.each;

	List$6.prototype.eachRight = function(fn, context) {
	    var item;

	    if (context === undefined) {
	        context = this;
	    }

	    // push cursor
	    var cursor = allocateCursor(this, this.tail, null);

	    while (cursor.prev !== null) {
	        item = cursor.prev;
	        cursor.prev = item.prev;

	        fn.call(context, item.data, item, this);
	    }

	    // pop cursor
	    releaseCursor(this);
	};

	List$6.prototype.forEachRight = List$6.prototype.eachRight;

	List$6.prototype.reduce = function(fn, initialValue, context) {
	    var item;

	    if (context === undefined) {
	        context = this;
	    }

	    // push cursor
	    var cursor = allocateCursor(this, null, this.head);
	    var acc = initialValue;

	    while (cursor.next !== null) {
	        item = cursor.next;
	        cursor.next = item.next;

	        acc = fn.call(context, acc, item.data, item, this);
	    }

	    // pop cursor
	    releaseCursor(this);

	    return acc;
	};

	List$6.prototype.reduceRight = function(fn, initialValue, context) {
	    var item;

	    if (context === undefined) {
	        context = this;
	    }

	    // push cursor
	    var cursor = allocateCursor(this, this.tail, null);
	    var acc = initialValue;

	    while (cursor.prev !== null) {
	        item = cursor.prev;
	        cursor.prev = item.prev;

	        acc = fn.call(context, acc, item.data, item, this);
	    }

	    // pop cursor
	    releaseCursor(this);

	    return acc;
	};

	List$6.prototype.nextUntil = function(start, fn, context) {
	    if (start === null) {
	        return;
	    }

	    var item;

	    if (context === undefined) {
	        context = this;
	    }

	    // push cursor
	    var cursor = allocateCursor(this, null, start);

	    while (cursor.next !== null) {
	        item = cursor.next;
	        cursor.next = item.next;

	        if (fn.call(context, item.data, item, this)) {
	            break;
	        }
	    }

	    // pop cursor
	    releaseCursor(this);
	};

	List$6.prototype.prevUntil = function(start, fn, context) {
	    if (start === null) {
	        return;
	    }

	    var item;

	    if (context === undefined) {
	        context = this;
	    }

	    // push cursor
	    var cursor = allocateCursor(this, start, null);

	    while (cursor.prev !== null) {
	        item = cursor.prev;
	        cursor.prev = item.prev;

	        if (fn.call(context, item.data, item, this)) {
	            break;
	        }
	    }

	    // pop cursor
	    releaseCursor(this);
	};

	List$6.prototype.some = function(fn, context) {
	    var cursor = this.head;

	    if (context === undefined) {
	        context = this;
	    }

	    while (cursor !== null) {
	        if (fn.call(context, cursor.data, cursor, this)) {
	            return true;
	        }

	        cursor = cursor.next;
	    }

	    return false;
	};

	List$6.prototype.map = function(fn, context) {
	    var result = new List$6();
	    var cursor = this.head;

	    if (context === undefined) {
	        context = this;
	    }

	    while (cursor !== null) {
	        result.appendData(fn.call(context, cursor.data, cursor, this));
	        cursor = cursor.next;
	    }

	    return result;
	};

	List$6.prototype.filter = function(fn, context) {
	    var result = new List$6();
	    var cursor = this.head;

	    if (context === undefined) {
	        context = this;
	    }

	    while (cursor !== null) {
	        if (fn.call(context, cursor.data, cursor, this)) {
	            result.appendData(cursor.data);
	        }
	        cursor = cursor.next;
	    }

	    return result;
	};

	List$6.prototype.clear = function() {
	    this.head = null;
	    this.tail = null;
	};

	List$6.prototype.copy = function() {
	    var result = new List$6();
	    var cursor = this.head;

	    while (cursor !== null) {
	        result.insert(createItem(cursor.data));
	        cursor = cursor.next;
	    }

	    return result;
	};

	List$6.prototype.prepend = function(item) {
	    //      head
	    //    ^
	    // item
	    this.updateCursors(null, item, this.head, item);

	    // insert to the beginning of the list
	    if (this.head !== null) {
	        // new item <- first item
	        this.head.prev = item;

	        // new item -> first item
	        item.next = this.head;
	    } else {
	        // if list has no head, then it also has no tail
	        // in this case tail points to the new item
	        this.tail = item;
	    }

	    // head always points to new item
	    this.head = item;

	    return this;
	};

	List$6.prototype.prependData = function(data) {
	    return this.prepend(createItem(data));
	};

	List$6.prototype.append = function(item) {
	    return this.insert(item);
	};

	List$6.prototype.appendData = function(data) {
	    return this.insert(createItem(data));
	};

	List$6.prototype.insert = function(item, before) {
	    if (before !== undefined && before !== null) {
	        // prev   before
	        //      ^
	        //     item
	        this.updateCursors(before.prev, item, before, item);

	        if (before.prev === null) {
	            // insert to the beginning of list
	            if (this.head !== before) {
	                throw new Error('before doesn\'t belong to list');
	            }

	            // since head points to before therefore list doesn't empty
	            // no need to check tail
	            this.head = item;
	            before.prev = item;
	            item.next = before;

	            this.updateCursors(null, item);
	        } else {

	            // insert between two items
	            before.prev.next = item;
	            item.prev = before.prev;

	            before.prev = item;
	            item.next = before;
	        }
	    } else {
	        // tail
	        //      ^
	        //      item
	        this.updateCursors(this.tail, item, null, item);

	        // insert to the ending of the list
	        if (this.tail !== null) {
	            // last item -> new item
	            this.tail.next = item;

	            // last item <- new item
	            item.prev = this.tail;
	        } else {
	            // if list has no tail, then it also has no head
	            // in this case head points to new item
	            this.head = item;
	        }

	        // tail always points to new item
	        this.tail = item;
	    }

	    return this;
	};

	List$6.prototype.insertData = function(data, before) {
	    return this.insert(createItem(data), before);
	};

	List$6.prototype.remove = function(item) {
	    //      item
	    //       ^
	    // prev     next
	    this.updateCursors(item, item.prev, item, item.next);

	    if (item.prev !== null) {
	        item.prev.next = item.next;
	    } else {
	        if (this.head !== item) {
	            throw new Error('item doesn\'t belong to list');
	        }

	        this.head = item.next;
	    }

	    if (item.next !== null) {
	        item.next.prev = item.prev;
	    } else {
	        if (this.tail !== item) {
	            throw new Error('item doesn\'t belong to list');
	        }

	        this.tail = item.prev;
	    }

	    item.prev = null;
	    item.next = null;

	    return item;
	};

	List$6.prototype.push = function(data) {
	    this.insert(createItem(data));
	};

	List$6.prototype.pop = function() {
	    if (this.tail !== null) {
	        return this.remove(this.tail);
	    }
	};

	List$6.prototype.unshift = function(data) {
	    this.prepend(createItem(data));
	};

	List$6.prototype.shift = function() {
	    if (this.head !== null) {
	        return this.remove(this.head);
	    }
	};

	List$6.prototype.prependList = function(list) {
	    return this.insertList(list, this.head);
	};

	List$6.prototype.appendList = function(list) {
	    return this.insertList(list);
	};

	List$6.prototype.insertList = function(list, before) {
	    // ignore empty lists
	    if (list.head === null) {
	        return this;
	    }

	    if (before !== undefined && before !== null) {
	        this.updateCursors(before.prev, list.tail, before, list.head);

	        // insert in the middle of dist list
	        if (before.prev !== null) {
	            // before.prev <-> list.head
	            before.prev.next = list.head;
	            list.head.prev = before.prev;
	        } else {
	            this.head = list.head;
	        }

	        before.prev = list.tail;
	        list.tail.next = before;
	    } else {
	        this.updateCursors(this.tail, list.tail, null, list.head);

	        // insert to end of the list
	        if (this.tail !== null) {
	            // if destination list has a tail, then it also has a head,
	            // but head doesn't change

	            // dest tail -> source head
	            this.tail.next = list.head;

	            // dest tail <- source head
	            list.head.prev = this.tail;
	        } else {
	            // if list has no a tail, then it also has no a head
	            // in this case points head to new item
	            this.head = list.head;
	        }

	        // tail always start point to new item
	        this.tail = list.tail;
	    }

	    list.head = null;
	    list.tail = null;

	    return this;
	};

	List$6.prototype.replace = function(oldItem, newItemOrList) {
	    if ('head' in newItemOrList) {
	        this.insertList(newItemOrList, oldItem);
	    } else {
	        this.insert(newItemOrList, oldItem);
	    }

	    this.remove(oldItem);
	};

	var List_1 = List$6;

	var createCustomError$3 = function createCustomError(name, message) {
	    // use Object.create(), because some VMs prevent setting line/column otherwise
	    // (iOS Safari 10 even throws an exception)
	    var error = Object.create(SyntaxError.prototype);
	    var errorStack = new Error();

	    error.name = name;
	    error.message = message;

	    Object.defineProperty(error, 'stack', {
	        get: function() {
	            return (errorStack.stack || '').replace(/^(.+\n){1,3}/, name + ': ' + message + '\n');
	        }
	    });

	    return error;
	};

	var createCustomError$2 = createCustomError$3;
	var MAX_LINE_LENGTH = 100;
	var OFFSET_CORRECTION = 60;
	var TAB_REPLACEMENT = '    ';

	function sourceFragment(error, extraLines) {
	    function processLines(start, end) {
	        return lines.slice(start, end).map(function(line, idx) {
	            var num = String(start + idx + 1);

	            while (num.length < maxNumLength) {
	                num = ' ' + num;
	            }

	            return num + ' |' + line;
	        }).join('\n');
	    }

	    var lines = error.source.split(/\r\n?|\n|\f/);
	    var line = error.line;
	    var column = error.column;
	    var startLine = Math.max(1, line - extraLines) - 1;
	    var endLine = Math.min(line + extraLines, lines.length + 1);
	    var maxNumLength = Math.max(4, String(endLine).length) + 1;
	    var cutLeft = 0;

	    // column correction according to replaced tab before column
	    column += (TAB_REPLACEMENT.length - 1) * (lines[line - 1].substr(0, column - 1).match(/\t/g) || []).length;

	    if (column > MAX_LINE_LENGTH) {
	        cutLeft = column - OFFSET_CORRECTION + 3;
	        column = OFFSET_CORRECTION - 2;
	    }

	    for (var i = startLine; i <= endLine; i++) {
	        if (i >= 0 && i < lines.length) {
	            lines[i] = lines[i].replace(/\t/g, TAB_REPLACEMENT);
	            lines[i] =
	                (cutLeft > 0 && lines[i].length > cutLeft ? '\u2026' : '') +
	                lines[i].substr(cutLeft, MAX_LINE_LENGTH - 2) +
	                (lines[i].length > cutLeft + MAX_LINE_LENGTH - 1 ? '\u2026' : '');
	        }
	    }

	    return [
	        processLines(startLine, line),
	        new Array(column + maxNumLength + 2).join('-') + '^',
	        processLines(line, endLine)
	    ].filter(Boolean).join('\n');
	}

	var SyntaxError$4 = function(message, source, offset, line, column) {
	    var error = createCustomError$2('SyntaxError', message);

	    error.source = source;
	    error.offset = offset;
	    error.line = line;
	    error.column = column;

	    error.sourceFragment = function(extraLines) {
	        return sourceFragment(error, isNaN(extraLines) ? 0 : extraLines);
	    };
	    Object.defineProperty(error, 'formattedMessage', {
	        get: function() {
	            return (
	                'Parse error: ' + error.message + '\n' +
	                sourceFragment(error, 2)
	            );
	        }
	    });

	    // for backward capability
	    error.parseError = {
	        offset: offset,
	        line: line,
	        column: column
	    };

	    return error;
	};

	var _SyntaxError$1 = SyntaxError$4;

	// CSS Syntax Module Level 3
	// https://www.w3.org/TR/css-syntax-3/
	var TYPE$H = {
	    EOF: 0,                 // <EOF-token>
	    Ident: 1,               // <ident-token>
	    Function: 2,            // <function-token>
	    AtKeyword: 3,           // <at-keyword-token>
	    Hash: 4,                // <hash-token>
	    String: 5,              // <string-token>
	    BadString: 6,           // <bad-string-token>
	    Url: 7,                 // <url-token>
	    BadUrl: 8,              // <bad-url-token>
	    Delim: 9,               // <delim-token>
	    Number: 10,             // <number-token>
	    Percentage: 11,         // <percentage-token>
	    Dimension: 12,          // <dimension-token>
	    WhiteSpace: 13,         // <whitespace-token>
	    CDO: 14,                // <CDO-token>
	    CDC: 15,                // <CDC-token>
	    Colon: 16,              // <colon-token>     :
	    Semicolon: 17,          // <semicolon-token> ;
	    Comma: 18,              // <comma-token>     ,
	    LeftSquareBracket: 19,  // <[-token>
	    RightSquareBracket: 20, // <]-token>
	    LeftParenthesis: 21,    // <(-token>
	    RightParenthesis: 22,   // <)-token>
	    LeftCurlyBracket: 23,   // <{-token>
	    RightCurlyBracket: 24,  // <}-token>
	    Comment: 25
	};

	var NAME$3 = Object.keys(TYPE$H).reduce(function(result, key) {
	    result[TYPE$H[key]] = key;
	    return result;
	}, {});

	var _const = {
	    TYPE: TYPE$H,
	    NAME: NAME$3
	};

	var EOF$1 = 0;

	// https://drafts.csswg.org/css-syntax-3/
	// § 4.2. Definitions

	// digit
	// A code point between U+0030 DIGIT ZERO (0) and U+0039 DIGIT NINE (9).
	function isDigit$5(code) {
	    return code >= 0x0030 && code <= 0x0039;
	}

	// hex digit
	// A digit, or a code point between U+0041 LATIN CAPITAL LETTER A (A) and U+0046 LATIN CAPITAL LETTER F (F),
	// or a code point between U+0061 LATIN SMALL LETTER A (a) and U+0066 LATIN SMALL LETTER F (f).
	function isHexDigit$4(code) {
	    return (
	        isDigit$5(code) || // 0 .. 9
	        (code >= 0x0041 && code <= 0x0046) || // A .. F
	        (code >= 0x0061 && code <= 0x0066)    // a .. f
	    );
	}

	// uppercase letter
	// A code point between U+0041 LATIN CAPITAL LETTER A (A) and U+005A LATIN CAPITAL LETTER Z (Z).
	function isUppercaseLetter$1(code) {
	    return code >= 0x0041 && code <= 0x005A;
	}

	// lowercase letter
	// A code point between U+0061 LATIN SMALL LETTER A (a) and U+007A LATIN SMALL LETTER Z (z).
	function isLowercaseLetter(code) {
	    return code >= 0x0061 && code <= 0x007A;
	}

	// letter
	// An uppercase letter or a lowercase letter.
	function isLetter(code) {
	    return isUppercaseLetter$1(code) || isLowercaseLetter(code);
	}

	// non-ASCII code point
	// A code point with a value equal to or greater than U+0080 <control>.
	function isNonAscii(code) {
	    return code >= 0x0080;
	}

	// name-start code point
	// A letter, a non-ASCII code point, or U+005F LOW LINE (_).
	function isNameStart(code) {
	    return isLetter(code) || isNonAscii(code) || code === 0x005F;
	}

	// name code point
	// A name-start code point, a digit, or U+002D HYPHEN-MINUS (-).
	function isName$2(code) {
	    return isNameStart(code) || isDigit$5(code) || code === 0x002D;
	}

	// non-printable code point
	// A code point between U+0000 NULL and U+0008 BACKSPACE, or U+000B LINE TABULATION,
	// or a code point between U+000E SHIFT OUT and U+001F INFORMATION SEPARATOR ONE, or U+007F DELETE.
	function isNonPrintable(code) {
	    return (
	        (code >= 0x0000 && code <= 0x0008) ||
	        (code === 0x000B) ||
	        (code >= 0x000E && code <= 0x001F) ||
	        (code === 0x007F)
	    );
	}

	// newline
	// U+000A LINE FEED. Note that U+000D CARRIAGE RETURN and U+000C FORM FEED are not included in this definition,
	// as they are converted to U+000A LINE FEED during preprocessing.
	// TODO: we doesn't do a preprocessing, so check a code point for U+000D CARRIAGE RETURN and U+000C FORM FEED
	function isNewline$1(code) {
	    return code === 0x000A || code === 0x000D || code === 0x000C;
	}

	// whitespace
	// A newline, U+0009 CHARACTER TABULATION, or U+0020 SPACE.
	function isWhiteSpace$2(code) {
	    return isNewline$1(code) || code === 0x0020 || code === 0x0009;
	}

	// § 4.3.8. Check if two code points are a valid escape
	function isValidEscape$2(first, second) {
	    // If the first code point is not U+005C REVERSE SOLIDUS (\), return false.
	    if (first !== 0x005C) {
	        return false;
	    }

	    // Otherwise, if the second code point is a newline or EOF, return false.
	    if (isNewline$1(second) || second === EOF$1) {
	        return false;
	    }

	    // Otherwise, return true.
	    return true;
	}

	// § 4.3.9. Check if three code points would start an identifier
	function isIdentifierStart$2(first, second, third) {
	    // Look at the first code point:

	    // U+002D HYPHEN-MINUS
	    if (first === 0x002D) {
	        // If the second code point is a name-start code point or a U+002D HYPHEN-MINUS,
	        // or the second and third code points are a valid escape, return true. Otherwise, return false.
	        return (
	            isNameStart(second) ||
	            second === 0x002D ||
	            isValidEscape$2(second, third)
	        );
	    }

	    // name-start code point
	    if (isNameStart(first)) {
	        // Return true.
	        return true;
	    }

	    // U+005C REVERSE SOLIDUS (\)
	    if (first === 0x005C) {
	        // If the first and second code points are a valid escape, return true. Otherwise, return false.
	        return isValidEscape$2(first, second);
	    }

	    // anything else
	    // Return false.
	    return false;
	}

	// § 4.3.10. Check if three code points would start a number
	function isNumberStart$1(first, second, third) {
	    // Look at the first code point:

	    // U+002B PLUS SIGN (+)
	    // U+002D HYPHEN-MINUS (-)
	    if (first === 0x002B || first === 0x002D) {
	        // If the second code point is a digit, return true.
	        if (isDigit$5(second)) {
	            return 2;
	        }

	        // Otherwise, if the second code point is a U+002E FULL STOP (.)
	        // and the third code point is a digit, return true.
	        // Otherwise, return false.
	        return second === 0x002E && isDigit$5(third) ? 3 : 0;
	    }

	    // U+002E FULL STOP (.)
	    if (first === 0x002E) {
	        // If the second code point is a digit, return true. Otherwise, return false.
	        return isDigit$5(second) ? 2 : 0;
	    }

	    // digit
	    if (isDigit$5(first)) {
	        // Return true.
	        return 1;
	    }

	    // anything else
	    // Return false.
	    return 0;
	}

	//
	// Misc
	//

	// detect BOM (https://en.wikipedia.org/wiki/Byte_order_mark)
	function isBOM$2(code) {
	    // UTF-16BE
	    if (code === 0xFEFF) {
	        return 1;
	    }

	    // UTF-16LE
	    if (code === 0xFFFE) {
	        return 1;
	    }

	    return 0;
	}

	// Fast code category
	//
	// https://drafts.csswg.org/css-syntax/#tokenizer-definitions
	// > non-ASCII code point
	// >   A code point with a value equal to or greater than U+0080 <control>
	// > name-start code point
	// >   A letter, a non-ASCII code point, or U+005F LOW LINE (_).
	// > name code point
	// >   A name-start code point, a digit, or U+002D HYPHEN-MINUS (-)
	// That means only ASCII code points has a special meaning and we define a maps for 0..127 codes only
	var CATEGORY = new Array(0x80);
	charCodeCategory$1.Eof = 0x80;
	charCodeCategory$1.WhiteSpace = 0x82;
	charCodeCategory$1.Digit = 0x83;
	charCodeCategory$1.NameStart = 0x84;
	charCodeCategory$1.NonPrintable = 0x85;

	for (var i = 0; i < CATEGORY.length; i++) {
	    switch (true) {
	        case isWhiteSpace$2(i):
	            CATEGORY[i] = charCodeCategory$1.WhiteSpace;
	            break;

	        case isDigit$5(i):
	            CATEGORY[i] = charCodeCategory$1.Digit;
	            break;

	        case isNameStart(i):
	            CATEGORY[i] = charCodeCategory$1.NameStart;
	            break;

	        case isNonPrintable(i):
	            CATEGORY[i] = charCodeCategory$1.NonPrintable;
	            break;

	        default:
	            CATEGORY[i] = i || charCodeCategory$1.Eof;
	    }
	}

	function charCodeCategory$1(code) {
	    return code < 0x80 ? CATEGORY[code] : charCodeCategory$1.NameStart;
	}
	var charCodeDefinitions$1 = {
	    isDigit: isDigit$5,
	    isHexDigit: isHexDigit$4,
	    isUppercaseLetter: isUppercaseLetter$1,
	    isLowercaseLetter: isLowercaseLetter,
	    isLetter: isLetter,
	    isNonAscii: isNonAscii,
	    isNameStart: isNameStart,
	    isName: isName$2,
	    isNonPrintable: isNonPrintable,
	    isNewline: isNewline$1,
	    isWhiteSpace: isWhiteSpace$2,
	    isValidEscape: isValidEscape$2,
	    isIdentifierStart: isIdentifierStart$2,
	    isNumberStart: isNumberStart$1,

	    isBOM: isBOM$2,
	    charCodeCategory: charCodeCategory$1
	};

	var charCodeDef = charCodeDefinitions$1;
	var isDigit$4 = charCodeDef.isDigit;
	var isHexDigit$3 = charCodeDef.isHexDigit;
	var isUppercaseLetter = charCodeDef.isUppercaseLetter;
	var isName$1 = charCodeDef.isName;
	var isWhiteSpace$1 = charCodeDef.isWhiteSpace;
	var isValidEscape$1 = charCodeDef.isValidEscape;

	function getCharCode(source, offset) {
	    return offset < source.length ? source.charCodeAt(offset) : 0;
	}

	function getNewlineLength$1(source, offset, code) {
	    if (code === 13 /* \r */ && getCharCode(source, offset + 1) === 10 /* \n */) {
	        return 2;
	    }

	    return 1;
	}

	function cmpChar$5(testStr, offset, referenceCode) {
	    var code = testStr.charCodeAt(offset);

	    // code.toLowerCase() for A..Z
	    if (isUppercaseLetter(code)) {
	        code = code | 32;
	    }

	    return code === referenceCode;
	}

	function cmpStr$6(testStr, start, end, referenceStr) {
	    if (end - start !== referenceStr.length) {
	        return false;
	    }

	    if (start < 0 || end > testStr.length) {
	        return false;
	    }

	    for (var i = start; i < end; i++) {
	        var testCode = testStr.charCodeAt(i);
	        var referenceCode = referenceStr.charCodeAt(i - start);

	        // testCode.toLowerCase() for A..Z
	        if (isUppercaseLetter(testCode)) {
	            testCode = testCode | 32;
	        }

	        if (testCode !== referenceCode) {
	            return false;
	        }
	    }

	    return true;
	}

	function findWhiteSpaceStart$1(source, offset) {
	    for (; offset >= 0; offset--) {
	        if (!isWhiteSpace$1(source.charCodeAt(offset))) {
	            break;
	        }
	    }

	    return offset + 1;
	}

	function findWhiteSpaceEnd$1(source, offset) {
	    for (; offset < source.length; offset++) {
	        if (!isWhiteSpace$1(source.charCodeAt(offset))) {
	            break;
	        }
	    }

	    return offset;
	}

	function findDecimalNumberEnd(source, offset) {
	    for (; offset < source.length; offset++) {
	        if (!isDigit$4(source.charCodeAt(offset))) {
	            break;
	        }
	    }

	    return offset;
	}

	// § 4.3.7. Consume an escaped code point
	function consumeEscaped$1(source, offset) {
	    // It assumes that the U+005C REVERSE SOLIDUS (\) has already been consumed and
	    // that the next input code point has already been verified to be part of a valid escape.
	    offset += 2;

	    // hex digit
	    if (isHexDigit$3(getCharCode(source, offset - 1))) {
	        // Consume as many hex digits as possible, but no more than 5.
	        // Note that this means 1-6 hex digits have been consumed in total.
	        for (var maxOffset = Math.min(source.length, offset + 5); offset < maxOffset; offset++) {
	            if (!isHexDigit$3(getCharCode(source, offset))) {
	                break;
	            }
	        }

	        // If the next input code point is whitespace, consume it as well.
	        var code = getCharCode(source, offset);
	        if (isWhiteSpace$1(code)) {
	            offset += getNewlineLength$1(source, offset, code);
	        }
	    }

	    return offset;
	}

	// §4.3.11. Consume a name
	// Note: This algorithm does not do the verification of the first few code points that are necessary
	// to ensure the returned code points would constitute an <ident-token>. If that is the intended use,
	// ensure that the stream starts with an identifier before calling this algorithm.
	function consumeName$1(source, offset) {
	    // Let result initially be an empty string.
	    // Repeatedly consume the next input code point from the stream:
	    for (; offset < source.length; offset++) {
	        var code = source.charCodeAt(offset);

	        // name code point
	        if (isName$1(code)) {
	            // Append the code point to result.
	            continue;
	        }

	        // the stream starts with a valid escape
	        if (isValidEscape$1(code, getCharCode(source, offset + 1))) {
	            // Consume an escaped code point. Append the returned code point to result.
	            offset = consumeEscaped$1(source, offset) - 1;
	            continue;
	        }

	        // anything else
	        // Reconsume the current input code point. Return result.
	        break;
	    }

	    return offset;
	}

	// §4.3.12. Consume a number
	function consumeNumber$5(source, offset) {
	    var code = source.charCodeAt(offset);

	    // 2. If the next input code point is U+002B PLUS SIGN (+) or U+002D HYPHEN-MINUS (-),
	    // consume it and append it to repr.
	    if (code === 0x002B || code === 0x002D) {
	        code = source.charCodeAt(offset += 1);
	    }

	    // 3. While the next input code point is a digit, consume it and append it to repr.
	    if (isDigit$4(code)) {
	        offset = findDecimalNumberEnd(source, offset + 1);
	        code = source.charCodeAt(offset);
	    }

	    // 4. If the next 2 input code points are U+002E FULL STOP (.) followed by a digit, then:
	    if (code === 0x002E && isDigit$4(source.charCodeAt(offset + 1))) {
	        // 4.1 Consume them.
	        // 4.2 Append them to repr.
	        code = source.charCodeAt(offset += 2);

	        // 4.3 Set type to "number".
	        // TODO

	        // 4.4 While the next input code point is a digit, consume it and append it to repr.

	        offset = findDecimalNumberEnd(source, offset);
	    }

	    // 5. If the next 2 or 3 input code points are U+0045 LATIN CAPITAL LETTER E (E)
	    // or U+0065 LATIN SMALL LETTER E (e), ... , followed by a digit, then:
	    if (cmpChar$5(source, offset, 101 /* e */)) {
	        var sign = 0;
	        code = source.charCodeAt(offset + 1);

	        // ... optionally followed by U+002D HYPHEN-MINUS (-) or U+002B PLUS SIGN (+) ...
	        if (code === 0x002D || code === 0x002B) {
	            sign = 1;
	            code = source.charCodeAt(offset + 2);
	        }

	        // ... followed by a digit
	        if (isDigit$4(code)) {
	            // 5.1 Consume them.
	            // 5.2 Append them to repr.

	            // 5.3 Set type to "number".
	            // TODO

	            // 5.4 While the next input code point is a digit, consume it and append it to repr.
	            offset = findDecimalNumberEnd(source, offset + 1 + sign + 1);
	        }
	    }

	    return offset;
	}

	// § 4.3.14. Consume the remnants of a bad url
	// ... its sole use is to consume enough of the input stream to reach a recovery point
	// where normal tokenizing can resume.
	function consumeBadUrlRemnants$1(source, offset) {
	    // Repeatedly consume the next input code point from the stream:
	    for (; offset < source.length; offset++) {
	        var code = source.charCodeAt(offset);

	        // U+0029 RIGHT PARENTHESIS ())
	        // EOF
	        if (code === 0x0029) {
	            // Return.
	            offset++;
	            break;
	        }

	        if (isValidEscape$1(code, getCharCode(source, offset + 1))) {
	            // Consume an escaped code point.
	            // Note: This allows an escaped right parenthesis ("\)") to be encountered
	            // without ending the <bad-url-token>. This is otherwise identical to
	            // the "anything else" clause.
	            offset = consumeEscaped$1(source, offset);
	        }
	    }

	    return offset;
	}

	var utils$2 = {
	    consumeEscaped: consumeEscaped$1,
	    consumeName: consumeName$1,
	    consumeNumber: consumeNumber$5,
	    consumeBadUrlRemnants: consumeBadUrlRemnants$1,

	    cmpChar: cmpChar$5,
	    cmpStr: cmpStr$6,

	    getNewlineLength: getNewlineLength$1,
	    findWhiteSpaceStart: findWhiteSpaceStart$1,
	    findWhiteSpaceEnd: findWhiteSpaceEnd$1
	};

	var constants$2 = _const;
	var TYPE$G = constants$2.TYPE;
	var NAME$2 = constants$2.NAME;

	var utils$1 = utils$2;
	var cmpStr$5 = utils$1.cmpStr;

	var EOF = TYPE$G.EOF;
	var WHITESPACE$c = TYPE$G.WhiteSpace;
	var COMMENT$a = TYPE$G.Comment;

	var OFFSET_MASK$1 = 0x00FFFFFF;
	var TYPE_SHIFT$1 = 24;

	var TokenStream$4 = function() {
	    this.offsetAndType = null;
	    this.balance = null;

	    this.reset();
	};

	TokenStream$4.prototype = {
	    reset: function() {
	        this.eof = false;
	        this.tokenIndex = -1;
	        this.tokenType = 0;
	        this.tokenStart = this.firstCharOffset;
	        this.tokenEnd = this.firstCharOffset;
	    },

	    lookupType: function(offset) {
	        offset += this.tokenIndex;

	        if (offset < this.tokenCount) {
	            return this.offsetAndType[offset] >> TYPE_SHIFT$1;
	        }

	        return EOF;
	    },
	    lookupOffset: function(offset) {
	        offset += this.tokenIndex;

	        if (offset < this.tokenCount) {
	            return this.offsetAndType[offset - 1] & OFFSET_MASK$1;
	        }

	        return this.source.length;
	    },
	    lookupValue: function(offset, referenceStr) {
	        offset += this.tokenIndex;

	        if (offset < this.tokenCount) {
	            return cmpStr$5(
	                this.source,
	                this.offsetAndType[offset - 1] & OFFSET_MASK$1,
	                this.offsetAndType[offset] & OFFSET_MASK$1,
	                referenceStr
	            );
	        }

	        return false;
	    },
	    getTokenStart: function(tokenIndex) {
	        if (tokenIndex === this.tokenIndex) {
	            return this.tokenStart;
	        }

	        if (tokenIndex > 0) {
	            return tokenIndex < this.tokenCount
	                ? this.offsetAndType[tokenIndex - 1] & OFFSET_MASK$1
	                : this.offsetAndType[this.tokenCount] & OFFSET_MASK$1;
	        }

	        return this.firstCharOffset;
	    },

	    // TODO: -> skipUntilBalanced
	    getRawLength: function(startToken, mode) {
	        var cursor = startToken;
	        var balanceEnd;
	        var offset = this.offsetAndType[Math.max(cursor - 1, 0)] & OFFSET_MASK$1;
	        var type;

	        loop:
	        for (; cursor < this.tokenCount; cursor++) {
	            balanceEnd = this.balance[cursor];

	            // stop scanning on balance edge that points to offset before start token
	            if (balanceEnd < startToken) {
	                break loop;
	            }

	            type = this.offsetAndType[cursor] >> TYPE_SHIFT$1;

	            // check token is stop type
	            switch (mode(type, this.source, offset)) {
	                case 1:
	                    break loop;

	                case 2:
	                    cursor++;
	                    break loop;

	                default:
	                    // fast forward to the end of balanced block
	                    if (this.balance[balanceEnd] === cursor) {
	                        cursor = balanceEnd;
	                    }

	                    offset = this.offsetAndType[cursor] & OFFSET_MASK$1;
	            }
	        }

	        return cursor - this.tokenIndex;
	    },
	    isBalanceEdge: function(pos) {
	        return this.balance[this.tokenIndex] < pos;
	    },
	    isDelim: function(code, offset) {
	        if (offset) {
	            return (
	                this.lookupType(offset) === TYPE$G.Delim &&
	                this.source.charCodeAt(this.lookupOffset(offset)) === code
	            );
	        }

	        return (
	            this.tokenType === TYPE$G.Delim &&
	            this.source.charCodeAt(this.tokenStart) === code
	        );
	    },

	    getTokenValue: function() {
	        return this.source.substring(this.tokenStart, this.tokenEnd);
	    },
	    getTokenLength: function() {
	        return this.tokenEnd - this.tokenStart;
	    },
	    substrToCursor: function(start) {
	        return this.source.substring(start, this.tokenStart);
	    },

	    skipWS: function() {
	        for (var i = this.tokenIndex, skipTokenCount = 0; i < this.tokenCount; i++, skipTokenCount++) {
	            if ((this.offsetAndType[i] >> TYPE_SHIFT$1) !== WHITESPACE$c) {
	                break;
	            }
	        }

	        if (skipTokenCount > 0) {
	            this.skip(skipTokenCount);
	        }
	    },
	    skipSC: function() {
	        while (this.tokenType === WHITESPACE$c || this.tokenType === COMMENT$a) {
	            this.next();
	        }
	    },
	    skip: function(tokenCount) {
	        var next = this.tokenIndex + tokenCount;

	        if (next < this.tokenCount) {
	            this.tokenIndex = next;
	            this.tokenStart = this.offsetAndType[next - 1] & OFFSET_MASK$1;
	            next = this.offsetAndType[next];
	            this.tokenType = next >> TYPE_SHIFT$1;
	            this.tokenEnd = next & OFFSET_MASK$1;
	        } else {
	            this.tokenIndex = this.tokenCount;
	            this.next();
	        }
	    },
	    next: function() {
	        var next = this.tokenIndex + 1;

	        if (next < this.tokenCount) {
	            this.tokenIndex = next;
	            this.tokenStart = this.tokenEnd;
	            next = this.offsetAndType[next];
	            this.tokenType = next >> TYPE_SHIFT$1;
	            this.tokenEnd = next & OFFSET_MASK$1;
	        } else {
	            this.tokenIndex = this.tokenCount;
	            this.eof = true;
	            this.tokenType = EOF;
	            this.tokenStart = this.tokenEnd = this.source.length;
	        }
	    },

	    forEachToken(fn) {
	        for (var i = 0, offset = this.firstCharOffset; i < this.tokenCount; i++) {
	            var start = offset;
	            var item = this.offsetAndType[i];
	            var end = item & OFFSET_MASK$1;
	            var type = item >> TYPE_SHIFT$1;

	            offset = end;

	            fn(type, start, end, i);
	        }
	    },

	    dump() {
	        var tokens = new Array(this.tokenCount);

	        this.forEachToken((type, start, end, index) => {
	            tokens[index] = {
	                idx: index,
	                type: NAME$2[type],
	                chunk: this.source.substring(start, end),
	                balance: this.balance[index]
	            };
	        });

	        return tokens;
	    }
	};

	var TokenStream_1 = TokenStream$4;

	function noop$3(value) {
	    return value;
	}

	function generateMultiplier(multiplier) {
	    if (multiplier.min === 0 && multiplier.max === 0) {
	        return '*';
	    }

	    if (multiplier.min === 0 && multiplier.max === 1) {
	        return '?';
	    }

	    if (multiplier.min === 1 && multiplier.max === 0) {
	        return multiplier.comma ? '#' : '+';
	    }

	    if (multiplier.min === 1 && multiplier.max === 1) {
	        return '';
	    }

	    return (
	        (multiplier.comma ? '#' : '') +
	        (multiplier.min === multiplier.max
	            ? '{' + multiplier.min + '}'
	            : '{' + multiplier.min + ',' + (multiplier.max !== 0 ? multiplier.max : '') + '}'
	        )
	    );
	}

	function generateTypeOpts(node) {
	    switch (node.type) {
	        case 'Range':
	            return (
	                ' [' +
	                (node.min === null ? '-∞' : node.min) +
	                ',' +
	                (node.max === null ? '∞' : node.max) +
	                ']'
	            );

	        default:
	            throw new Error('Unknown node type `' + node.type + '`');
	    }
	}

	function generateSequence(node, decorate, forceBraces, compact) {
	    var combinator = node.combinator === ' ' || compact ? node.combinator : ' ' + node.combinator + ' ';
	    var result = node.terms.map(function(term) {
	        return generate$2(term, decorate, forceBraces, compact);
	    }).join(combinator);

	    if (node.explicit || forceBraces) {
	        result = (compact || result[0] === ',' ? '[' : '[ ') + result + (compact ? ']' : ' ]');
	    }

	    return result;
	}

	function generate$2(node, decorate, forceBraces, compact) {
	    var result;

	    switch (node.type) {
	        case 'Group':
	            result =
	                generateSequence(node, decorate, forceBraces, compact) +
	                (node.disallowEmpty ? '!' : '');
	            break;

	        case 'Multiplier':
	            // return since node is a composition
	            return (
	                generate$2(node.term, decorate, forceBraces, compact) +
	                decorate(generateMultiplier(node), node)
	            );

	        case 'Type':
	            result = '<' + node.name + (node.opts ? decorate(generateTypeOpts(node.opts), node.opts) : '') + '>';
	            break;

	        case 'Property':
	            result = '<\'' + node.name + '\'>';
	            break;

	        case 'Keyword':
	            result = node.name;
	            break;

	        case 'AtKeyword':
	            result = '@' + node.name;
	            break;

	        case 'Function':
	            result = node.name + '(';
	            break;

	        case 'String':
	        case 'Token':
	            result = node.value;
	            break;

	        case 'Comma':
	            result = ',';
	            break;

	        default:
	            throw new Error('Unknown node type `' + node.type + '`');
	    }

	    return decorate(result, node);
	}

	var generate_1 = function(node, options) {
	    var decorate = noop$3;
	    var forceBraces = false;
	    var compact = false;

	    if (typeof options === 'function') {
	        decorate = options;
	    } else if (options) {
	        forceBraces = Boolean(options.forceBraces);
	        compact = Boolean(options.compact);
	        if (typeof options.decorate === 'function') {
	            decorate = options.decorate;
	        }
	    }

	    return generate$2(node, decorate, forceBraces, compact);
	};

	const createCustomError$1 = createCustomError$3;
	const generate$1 = generate_1;
	const defaultLoc = { offset: 0, line: 1, column: 1 };

	function locateMismatch(matchResult, node) {
	    const tokens = matchResult.tokens;
	    const longestMatch = matchResult.longestMatch;
	    const mismatchNode = longestMatch < tokens.length ? tokens[longestMatch].node || null : null;
	    const badNode = mismatchNode !== node ? mismatchNode : null;
	    let mismatchOffset = 0;
	    let mismatchLength = 0;
	    let entries = 0;
	    let css = '';
	    let start;
	    let end;

	    for (let i = 0; i < tokens.length; i++) {
	        const token = tokens[i].value;

	        if (i === longestMatch) {
	            mismatchLength = token.length;
	            mismatchOffset = css.length;
	        }

	        if (badNode !== null && tokens[i].node === badNode) {
	            if (i <= longestMatch) {
	                entries++;
	            } else {
	                entries = 0;
	            }
	        }

	        css += token;
	    }

	    if (longestMatch === tokens.length || entries > 1) { // last
	        start = fromLoc(badNode || node, 'end') || buildLoc(defaultLoc, css);
	        end = buildLoc(start);
	    } else {
	        start = fromLoc(badNode, 'start') ||
	            buildLoc(fromLoc(node, 'start') || defaultLoc, css.slice(0, mismatchOffset));
	        end = fromLoc(badNode, 'end') ||
	            buildLoc(start, css.substr(mismatchOffset, mismatchLength));
	    }

	    return {
	        css,
	        mismatchOffset,
	        mismatchLength,
	        start,
	        end
	    };
	}

	function fromLoc(node, point) {
	    const value = node && node.loc && node.loc[point];

	    if (value) {
	        return 'line' in value ? buildLoc(value) : value;
	    }

	    return null;
	}

	function buildLoc({ offset, line, column }, extra) {
	    const loc = {
	        offset,
	        line,
	        column
	    };

	    if (extra) {
	        const lines = extra.split(/\n|\r\n?|\f/);

	        loc.offset += extra.length;
	        loc.line += lines.length - 1;
	        loc.column = lines.length === 1 ? loc.column + extra.length : lines.pop().length + 1;
	    }

	    return loc;
	}

	const SyntaxReferenceError$1 = function(type, referenceName) {
	    const error = createCustomError$1(
	        'SyntaxReferenceError',
	        type + (referenceName ? ' `' + referenceName + '`' : '')
	    );

	    error.reference = referenceName;

	    return error;
	};

	const SyntaxMatchError$1 = function(message, syntax, node, matchResult) {
	    const error = createCustomError$1('SyntaxMatchError', message);
	    const {
	        css,
	        mismatchOffset,
	        mismatchLength,
	        start,
	        end
	    } = locateMismatch(matchResult, node);

	    error.rawMessage = message;
	    error.syntax = syntax ? generate$1(syntax) : '<generic>';
	    error.css = css;
	    error.mismatchOffset = mismatchOffset;
	    error.mismatchLength = mismatchLength;
	    error.message = message + '\n' +
	        '  syntax: ' + error.syntax + '\n' +
	        '   value: ' + (css || '<empty string>') + '\n' +
	        '  --------' + new Array(error.mismatchOffset + 1).join('-') + '^';

	    Object.assign(error, start);
	    error.loc = {
	        source: (node && node.loc && node.loc.source) || '<unknown>',
	        start,
	        end
	    };

	    return error;
	};

	var error = {
	    SyntaxReferenceError: SyntaxReferenceError$1,
	    SyntaxMatchError: SyntaxMatchError$1
	};

	var hasOwnProperty$7 = Object.prototype.hasOwnProperty;
	var keywords$1 = Object.create(null);
	var properties$1 = Object.create(null);
	var HYPHENMINUS$5 = 45; // '-'.charCodeAt()

	function isCustomProperty$1(str, offset) {
	    offset = offset || 0;

	    return str.length - offset >= 2 &&
	           str.charCodeAt(offset) === HYPHENMINUS$5 &&
	           str.charCodeAt(offset + 1) === HYPHENMINUS$5;
	}

	function getVendorPrefix(str, offset) {
	    offset = offset || 0;

	    // verdor prefix should be at least 3 chars length
	    if (str.length - offset >= 3) {
	        // vendor prefix starts with hyper minus following non-hyper minus
	        if (str.charCodeAt(offset) === HYPHENMINUS$5 &&
	            str.charCodeAt(offset + 1) !== HYPHENMINUS$5) {
	            // vendor prefix should contain a hyper minus at the ending
	            var secondDashIndex = str.indexOf('-', offset + 2);

	            if (secondDashIndex !== -1) {
	                return str.substring(offset, secondDashIndex + 1);
	            }
	        }
	    }

	    return '';
	}

	function getKeywordDescriptor(keyword) {
	    if (hasOwnProperty$7.call(keywords$1, keyword)) {
	        return keywords$1[keyword];
	    }

	    var name = keyword.toLowerCase();

	    if (hasOwnProperty$7.call(keywords$1, name)) {
	        return keywords$1[keyword] = keywords$1[name];
	    }

	    var custom = isCustomProperty$1(name, 0);
	    var vendor = !custom ? getVendorPrefix(name, 0) : '';

	    return keywords$1[keyword] = Object.freeze({
	        basename: name.substr(vendor.length),
	        name: name,
	        vendor: vendor,
	        prefix: vendor,
	        custom: custom
	    });
	}

	function getPropertyDescriptor(property) {
	    if (hasOwnProperty$7.call(properties$1, property)) {
	        return properties$1[property];
	    }

	    var name = property;
	    var hack = property[0];

	    if (hack === '/') {
	        hack = property[1] === '/' ? '//' : '/';
	    } else if (hack !== '_' &&
	               hack !== '*' &&
	               hack !== '$' &&
	               hack !== '#' &&
	               hack !== '+' &&
	               hack !== '&') {
	        hack = '';
	    }

	    var custom = isCustomProperty$1(name, hack.length);

	    // re-use result when possible (the same as for lower case)
	    if (!custom) {
	        name = name.toLowerCase();
	        if (hasOwnProperty$7.call(properties$1, name)) {
	            return properties$1[property] = properties$1[name];
	        }
	    }

	    var vendor = !custom ? getVendorPrefix(name, hack.length) : '';
	    var prefix = name.substr(0, hack.length + vendor.length);

	    return properties$1[property] = Object.freeze({
	        basename: name.substr(prefix.length),
	        name: name.substr(hack.length),
	        hack: hack,
	        vendor: vendor,
	        prefix: prefix,
	        custom: custom
	    });
	}

	var names$2 = {
	    keyword: getKeywordDescriptor,
	    property: getPropertyDescriptor,
	    isCustomProperty: isCustomProperty$1,
	    vendorPrefix: getVendorPrefix
	};

	var MIN_SIZE = 16 * 1024;
	var SafeUint32Array = typeof Uint32Array !== 'undefined' ? Uint32Array : Array; // fallback on Array when TypedArray is not supported

	var adoptBuffer$2 = function adoptBuffer(buffer, size) {
	    if (buffer === null || buffer.length < size) {
	        return new SafeUint32Array(Math.max(size + 1024, MIN_SIZE));
	    }

	    return buffer;
	};

	var TokenStream$3 = TokenStream_1;
	var adoptBuffer$1 = adoptBuffer$2;

	var constants$1 = _const;
	var TYPE$F = constants$1.TYPE;

	var charCodeDefinitions = charCodeDefinitions$1;
	var isNewline = charCodeDefinitions.isNewline;
	var isName = charCodeDefinitions.isName;
	var isValidEscape = charCodeDefinitions.isValidEscape;
	var isNumberStart = charCodeDefinitions.isNumberStart;
	var isIdentifierStart$1 = charCodeDefinitions.isIdentifierStart;
	var charCodeCategory = charCodeDefinitions.charCodeCategory;
	var isBOM$1 = charCodeDefinitions.isBOM;

	var utils = utils$2;
	var cmpStr$4 = utils.cmpStr;
	var getNewlineLength = utils.getNewlineLength;
	var findWhiteSpaceEnd = utils.findWhiteSpaceEnd;
	var consumeEscaped = utils.consumeEscaped;
	var consumeName = utils.consumeName;
	var consumeNumber$4 = utils.consumeNumber;
	var consumeBadUrlRemnants = utils.consumeBadUrlRemnants;

	var OFFSET_MASK = 0x00FFFFFF;
	var TYPE_SHIFT = 24;

	function tokenize$3(source, stream) {
	    function getCharCode(offset) {
	        return offset < sourceLength ? source.charCodeAt(offset) : 0;
	    }

	    // § 4.3.3. Consume a numeric token
	    function consumeNumericToken() {
	        // Consume a number and let number be the result.
	        offset = consumeNumber$4(source, offset);

	        // If the next 3 input code points would start an identifier, then:
	        if (isIdentifierStart$1(getCharCode(offset), getCharCode(offset + 1), getCharCode(offset + 2))) {
	            // Create a <dimension-token> with the same value and type flag as number, and a unit set initially to the empty string.
	            // Consume a name. Set the <dimension-token>’s unit to the returned value.
	            // Return the <dimension-token>.
	            type = TYPE$F.Dimension;
	            offset = consumeName(source, offset);
	            return;
	        }

	        // Otherwise, if the next input code point is U+0025 PERCENTAGE SIGN (%), consume it.
	        if (getCharCode(offset) === 0x0025) {
	            // Create a <percentage-token> with the same value as number, and return it.
	            type = TYPE$F.Percentage;
	            offset++;
	            return;
	        }

	        // Otherwise, create a <number-token> with the same value and type flag as number, and return it.
	        type = TYPE$F.Number;
	    }

	    // § 4.3.4. Consume an ident-like token
	    function consumeIdentLikeToken() {
	        const nameStartOffset = offset;

	        // Consume a name, and let string be the result.
	        offset = consumeName(source, offset);

	        // If string’s value is an ASCII case-insensitive match for "url",
	        // and the next input code point is U+0028 LEFT PARENTHESIS ((), consume it.
	        if (cmpStr$4(source, nameStartOffset, offset, 'url') && getCharCode(offset) === 0x0028) {
	            // While the next two input code points are whitespace, consume the next input code point.
	            offset = findWhiteSpaceEnd(source, offset + 1);

	            // If the next one or two input code points are U+0022 QUOTATION MARK ("), U+0027 APOSTROPHE ('),
	            // or whitespace followed by U+0022 QUOTATION MARK (") or U+0027 APOSTROPHE ('),
	            // then create a <function-token> with its value set to string and return it.
	            if (getCharCode(offset) === 0x0022 ||
	                getCharCode(offset) === 0x0027) {
	                type = TYPE$F.Function;
	                offset = nameStartOffset + 4;
	                return;
	            }

	            // Otherwise, consume a url token, and return it.
	            consumeUrlToken();
	            return;
	        }

	        // Otherwise, if the next input code point is U+0028 LEFT PARENTHESIS ((), consume it.
	        // Create a <function-token> with its value set to string and return it.
	        if (getCharCode(offset) === 0x0028) {
	            type = TYPE$F.Function;
	            offset++;
	            return;
	        }

	        // Otherwise, create an <ident-token> with its value set to string and return it.
	        type = TYPE$F.Ident;
	    }

	    // § 4.3.5. Consume a string token
	    function consumeStringToken(endingCodePoint) {
	        // This algorithm may be called with an ending code point, which denotes the code point
	        // that ends the string. If an ending code point is not specified,
	        // the current input code point is used.
	        if (!endingCodePoint) {
	            endingCodePoint = getCharCode(offset++);
	        }

	        // Initially create a <string-token> with its value set to the empty string.
	        type = TYPE$F.String;

	        // Repeatedly consume the next input code point from the stream:
	        for (; offset < source.length; offset++) {
	            var code = source.charCodeAt(offset);

	            switch (charCodeCategory(code)) {
	                // ending code point
	                case endingCodePoint:
	                    // Return the <string-token>.
	                    offset++;
	                    return;

	                // EOF
	                case charCodeCategory.Eof:
	                    // This is a parse error. Return the <string-token>.
	                    return;

	                // newline
	                case charCodeCategory.WhiteSpace:
	                    if (isNewline(code)) {
	                        // This is a parse error. Reconsume the current input code point,
	                        // create a <bad-string-token>, and return it.
	                        offset += getNewlineLength(source, offset, code);
	                        type = TYPE$F.BadString;
	                        return;
	                    }
	                    break;

	                // U+005C REVERSE SOLIDUS (\)
	                case 0x005C:
	                    // If the next input code point is EOF, do nothing.
	                    if (offset === source.length - 1) {
	                        break;
	                    }

	                    var nextCode = getCharCode(offset + 1);

	                    // Otherwise, if the next input code point is a newline, consume it.
	                    if (isNewline(nextCode)) {
	                        offset += getNewlineLength(source, offset + 1, nextCode);
	                    } else if (isValidEscape(code, nextCode)) {
	                        // Otherwise, (the stream starts with a valid escape) consume
	                        // an escaped code point and append the returned code point to
	                        // the <string-token>’s value.
	                        offset = consumeEscaped(source, offset) - 1;
	                    }
	                    break;

	                // anything else
	                // Append the current input code point to the <string-token>’s value.
	            }
	        }
	    }

	    // § 4.3.6. Consume a url token
	    // Note: This algorithm assumes that the initial "url(" has already been consumed.
	    // This algorithm also assumes that it’s being called to consume an "unquoted" value, like url(foo).
	    // A quoted value, like url("foo"), is parsed as a <function-token>. Consume an ident-like token
	    // automatically handles this distinction; this algorithm shouldn’t be called directly otherwise.
	    function consumeUrlToken() {
	        // Initially create a <url-token> with its value set to the empty string.
	        type = TYPE$F.Url;

	        // Consume as much whitespace as possible.
	        offset = findWhiteSpaceEnd(source, offset);

	        // Repeatedly consume the next input code point from the stream:
	        for (; offset < source.length; offset++) {
	            var code = source.charCodeAt(offset);

	            switch (charCodeCategory(code)) {
	                // U+0029 RIGHT PARENTHESIS ())
	                case 0x0029:
	                    // Return the <url-token>.
	                    offset++;
	                    return;

	                // EOF
	                case charCodeCategory.Eof:
	                    // This is a parse error. Return the <url-token>.
	                    return;

	                // whitespace
	                case charCodeCategory.WhiteSpace:
	                    // Consume as much whitespace as possible.
	                    offset = findWhiteSpaceEnd(source, offset);

	                    // If the next input code point is U+0029 RIGHT PARENTHESIS ()) or EOF,
	                    // consume it and return the <url-token>
	                    // (if EOF was encountered, this is a parse error);
	                    if (getCharCode(offset) === 0x0029 || offset >= source.length) {
	                        if (offset < source.length) {
	                            offset++;
	                        }
	                        return;
	                    }

	                    // otherwise, consume the remnants of a bad url, create a <bad-url-token>,
	                    // and return it.
	                    offset = consumeBadUrlRemnants(source, offset);
	                    type = TYPE$F.BadUrl;
	                    return;

	                // U+0022 QUOTATION MARK (")
	                // U+0027 APOSTROPHE (')
	                // U+0028 LEFT PARENTHESIS (()
	                // non-printable code point
	                case 0x0022:
	                case 0x0027:
	                case 0x0028:
	                case charCodeCategory.NonPrintable:
	                    // This is a parse error. Consume the remnants of a bad url,
	                    // create a <bad-url-token>, and return it.
	                    offset = consumeBadUrlRemnants(source, offset);
	                    type = TYPE$F.BadUrl;
	                    return;

	                // U+005C REVERSE SOLIDUS (\)
	                case 0x005C:
	                    // If the stream starts with a valid escape, consume an escaped code point and
	                    // append the returned code point to the <url-token>’s value.
	                    if (isValidEscape(code, getCharCode(offset + 1))) {
	                        offset = consumeEscaped(source, offset) - 1;
	                        break;
	                    }

	                    // Otherwise, this is a parse error. Consume the remnants of a bad url,
	                    // create a <bad-url-token>, and return it.
	                    offset = consumeBadUrlRemnants(source, offset);
	                    type = TYPE$F.BadUrl;
	                    return;

	                // anything else
	                // Append the current input code point to the <url-token>’s value.
	            }
	        }
	    }

	    if (!stream) {
	        stream = new TokenStream$3();
	    }

	    // ensure source is a string
	    source = String(source || '');

	    var sourceLength = source.length;
	    var offsetAndType = adoptBuffer$1(stream.offsetAndType, sourceLength + 1); // +1 because of eof-token
	    var balance = adoptBuffer$1(stream.balance, sourceLength + 1);
	    var tokenCount = 0;
	    var start = isBOM$1(getCharCode(0));
	    var offset = start;
	    var balanceCloseType = 0;
	    var balanceStart = 0;
	    var balancePrev = 0;

	    // https://drafts.csswg.org/css-syntax-3/#consume-token
	    // § 4.3.1. Consume a token
	    while (offset < sourceLength) {
	        var code = source.charCodeAt(offset);
	        var type = 0;

	        balance[tokenCount] = sourceLength;

	        switch (charCodeCategory(code)) {
	            // whitespace
	            case charCodeCategory.WhiteSpace:
	                // Consume as much whitespace as possible. Return a <whitespace-token>.
	                type = TYPE$F.WhiteSpace;
	                offset = findWhiteSpaceEnd(source, offset + 1);
	                break;

	            // U+0022 QUOTATION MARK (")
	            case 0x0022:
	                // Consume a string token and return it.
	                consumeStringToken();
	                break;

	            // U+0023 NUMBER SIGN (#)
	            case 0x0023:
	                // If the next input code point is a name code point or the next two input code points are a valid escape, then:
	                if (isName(getCharCode(offset + 1)) || isValidEscape(getCharCode(offset + 1), getCharCode(offset + 2))) {
	                    // Create a <hash-token>.
	                    type = TYPE$F.Hash;

	                    // If the next 3 input code points would start an identifier, set the <hash-token>’s type flag to "id".
	                    // if (isIdentifierStart(getCharCode(offset + 1), getCharCode(offset + 2), getCharCode(offset + 3))) {
	                    //     // TODO: set id flag
	                    // }

	                    // Consume a name, and set the <hash-token>’s value to the returned string.
	                    offset = consumeName(source, offset + 1);

	                    // Return the <hash-token>.
	                } else {
	                    // Otherwise, return a <delim-token> with its value set to the current input code point.
	                    type = TYPE$F.Delim;
	                    offset++;
	                }

	                break;

	            // U+0027 APOSTROPHE (')
	            case 0x0027:
	                // Consume a string token and return it.
	                consumeStringToken();
	                break;

	            // U+0028 LEFT PARENTHESIS (()
	            case 0x0028:
	                // Return a <(-token>.
	                type = TYPE$F.LeftParenthesis;
	                offset++;
	                break;

	            // U+0029 RIGHT PARENTHESIS ())
	            case 0x0029:
	                // Return a <)-token>.
	                type = TYPE$F.RightParenthesis;
	                offset++;
	                break;

	            // U+002B PLUS SIGN (+)
	            case 0x002B:
	                // If the input stream starts with a number, ...
	                if (isNumberStart(code, getCharCode(offset + 1), getCharCode(offset + 2))) {
	                    // ... reconsume the current input code point, consume a numeric token, and return it.
	                    consumeNumericToken();
	                } else {
	                    // Otherwise, return a <delim-token> with its value set to the current input code point.
	                    type = TYPE$F.Delim;
	                    offset++;
	                }
	                break;

	            // U+002C COMMA (,)
	            case 0x002C:
	                // Return a <comma-token>.
	                type = TYPE$F.Comma;
	                offset++;
	                break;

	            // U+002D HYPHEN-MINUS (-)
	            case 0x002D:
	                // If the input stream starts with a number, reconsume the current input code point, consume a numeric token, and return it.
	                if (isNumberStart(code, getCharCode(offset + 1), getCharCode(offset + 2))) {
	                    consumeNumericToken();
	                } else {
	                    // Otherwise, if the next 2 input code points are U+002D HYPHEN-MINUS U+003E GREATER-THAN SIGN (->), consume them and return a <CDC-token>.
	                    if (getCharCode(offset + 1) === 0x002D &&
	                        getCharCode(offset + 2) === 0x003E) {
	                        type = TYPE$F.CDC;
	                        offset = offset + 3;
	                    } else {
	                        // Otherwise, if the input stream starts with an identifier, ...
	                        if (isIdentifierStart$1(code, getCharCode(offset + 1), getCharCode(offset + 2))) {
	                            // ... reconsume the current input code point, consume an ident-like token, and return it.
	                            consumeIdentLikeToken();
	                        } else {
	                            // Otherwise, return a <delim-token> with its value set to the current input code point.
	                            type = TYPE$F.Delim;
	                            offset++;
	                        }
	                    }
	                }
	                break;

	            // U+002E FULL STOP (.)
	            case 0x002E:
	                // If the input stream starts with a number, ...
	                if (isNumberStart(code, getCharCode(offset + 1), getCharCode(offset + 2))) {
	                    // ... reconsume the current input code point, consume a numeric token, and return it.
	                    consumeNumericToken();
	                } else {
	                    // Otherwise, return a <delim-token> with its value set to the current input code point.
	                    type = TYPE$F.Delim;
	                    offset++;
	                }

	                break;

	            // U+002F SOLIDUS (/)
	            case 0x002F:
	                // If the next two input code point are U+002F SOLIDUS (/) followed by a U+002A ASTERISK (*),
	                if (getCharCode(offset + 1) === 0x002A) {
	                    // ... consume them and all following code points up to and including the first U+002A ASTERISK (*)
	                    // followed by a U+002F SOLIDUS (/), or up to an EOF code point.
	                    type = TYPE$F.Comment;
	                    offset = source.indexOf('*/', offset + 2) + 2;
	                    if (offset === 1) {
	                        offset = source.length;
	                    }
	                } else {
	                    type = TYPE$F.Delim;
	                    offset++;
	                }
	                break;

	            // U+003A COLON (:)
	            case 0x003A:
	                // Return a <colon-token>.
	                type = TYPE$F.Colon;
	                offset++;
	                break;

	            // U+003B SEMICOLON (;)
	            case 0x003B:
	                // Return a <semicolon-token>.
	                type = TYPE$F.Semicolon;
	                offset++;
	                break;

	            // U+003C LESS-THAN SIGN (<)
	            case 0x003C:
	                // If the next 3 input code points are U+0021 EXCLAMATION MARK U+002D HYPHEN-MINUS U+002D HYPHEN-MINUS (!--), ...
	                if (getCharCode(offset + 1) === 0x0021 &&
	                    getCharCode(offset + 2) === 0x002D &&
	                    getCharCode(offset + 3) === 0x002D) {
	                    // ... consume them and return a <CDO-token>.
	                    type = TYPE$F.CDO;
	                    offset = offset + 4;
	                } else {
	                    // Otherwise, return a <delim-token> with its value set to the current input code point.
	                    type = TYPE$F.Delim;
	                    offset++;
	                }

	                break;

	            // U+0040 COMMERCIAL AT (@)
	            case 0x0040:
	                // If the next 3 input code points would start an identifier, ...
	                if (isIdentifierStart$1(getCharCode(offset + 1), getCharCode(offset + 2), getCharCode(offset + 3))) {
	                    // ... consume a name, create an <at-keyword-token> with its value set to the returned value, and return it.
	                    type = TYPE$F.AtKeyword;
	                    offset = consumeName(source, offset + 1);
	                } else {
	                    // Otherwise, return a <delim-token> with its value set to the current input code point.
	                    type = TYPE$F.Delim;
	                    offset++;
	                }

	                break;

	            // U+005B LEFT SQUARE BRACKET ([)
	            case 0x005B:
	                // Return a <[-token>.
	                type = TYPE$F.LeftSquareBracket;
	                offset++;
	                break;

	            // U+005C REVERSE SOLIDUS (\)
	            case 0x005C:
	                // If the input stream starts with a valid escape, ...
	                if (isValidEscape(code, getCharCode(offset + 1))) {
	                    // ... reconsume the current input code point, consume an ident-like token, and return it.
	                    consumeIdentLikeToken();
	                } else {
	                    // Otherwise, this is a parse error. Return a <delim-token> with its value set to the current input code point.
	                    type = TYPE$F.Delim;
	                    offset++;
	                }
	                break;

	            // U+005D RIGHT SQUARE BRACKET (])
	            case 0x005D:
	                // Return a <]-token>.
	                type = TYPE$F.RightSquareBracket;
	                offset++;
	                break;

	            // U+007B LEFT CURLY BRACKET ({)
	            case 0x007B:
	                // Return a <{-token>.
	                type = TYPE$F.LeftCurlyBracket;
	                offset++;
	                break;

	            // U+007D RIGHT CURLY BRACKET (})
	            case 0x007D:
	                // Return a <}-token>.
	                type = TYPE$F.RightCurlyBracket;
	                offset++;
	                break;

	            // digit
	            case charCodeCategory.Digit:
	                // Reconsume the current input code point, consume a numeric token, and return it.
	                consumeNumericToken();
	                break;

	            // name-start code point
	            case charCodeCategory.NameStart:
	                // Reconsume the current input code point, consume an ident-like token, and return it.
	                consumeIdentLikeToken();
	                break;

	            // EOF
	            case charCodeCategory.Eof:
	                // Return an <EOF-token>.
	                break;

	            // anything else
	            default:
	                // Return a <delim-token> with its value set to the current input code point.
	                type = TYPE$F.Delim;
	                offset++;
	        }

	        switch (type) {
	            case balanceCloseType:
	                balancePrev = balanceStart & OFFSET_MASK;
	                balanceStart = balance[balancePrev];
	                balanceCloseType = balanceStart >> TYPE_SHIFT;
	                balance[tokenCount] = balancePrev;
	                balance[balancePrev++] = tokenCount;
	                for (; balancePrev < tokenCount; balancePrev++) {
	                    if (balance[balancePrev] === sourceLength) {
	                        balance[balancePrev] = tokenCount;
	                    }
	                }
	                break;

	            case TYPE$F.LeftParenthesis:
	            case TYPE$F.Function:
	                balance[tokenCount] = balanceStart;
	                balanceCloseType = TYPE$F.RightParenthesis;
	                balanceStart = (balanceCloseType << TYPE_SHIFT) | tokenCount;
	                break;

	            case TYPE$F.LeftSquareBracket:
	                balance[tokenCount] = balanceStart;
	                balanceCloseType = TYPE$F.RightSquareBracket;
	                balanceStart = (balanceCloseType << TYPE_SHIFT) | tokenCount;
	                break;

	            case TYPE$F.LeftCurlyBracket:
	                balance[tokenCount] = balanceStart;
	                balanceCloseType = TYPE$F.RightCurlyBracket;
	                balanceStart = (balanceCloseType << TYPE_SHIFT) | tokenCount;
	                break;
	        }

	        offsetAndType[tokenCount++] = (type << TYPE_SHIFT) | offset;
	    }

	    // finalize buffers
	    offsetAndType[tokenCount] = (TYPE$F.EOF << TYPE_SHIFT) | offset; // <EOF-token>
	    balance[tokenCount] = sourceLength;
	    balance[sourceLength] = sourceLength; // prevents false positive balance match with any token
	    while (balanceStart !== 0) {
	        balancePrev = balanceStart & OFFSET_MASK;
	        balanceStart = balance[balancePrev];
	        balance[balancePrev] = sourceLength;
	    }

	    // update stream
	    stream.source = source;
	    stream.firstCharOffset = start;
	    stream.offsetAndType = offsetAndType;
	    stream.tokenCount = tokenCount;
	    stream.balance = balance;
	    stream.reset();
	    stream.next();

	    return stream;
	}

	// extend tokenizer with constants
	Object.keys(constants$1).forEach(function(key) {
	    tokenize$3[key] = constants$1[key];
	});

	// extend tokenizer with static methods from utils
	Object.keys(charCodeDefinitions).forEach(function(key) {
	    tokenize$3[key] = charCodeDefinitions[key];
	});
	Object.keys(utils).forEach(function(key) {
	    tokenize$3[key] = utils[key];
	});

	var tokenizer$3 = tokenize$3;

	var isDigit$3 = tokenizer$3.isDigit;
	var cmpChar$4 = tokenizer$3.cmpChar;
	var TYPE$E = tokenizer$3.TYPE;

	var DELIM$6 = TYPE$E.Delim;
	var WHITESPACE$b = TYPE$E.WhiteSpace;
	var COMMENT$9 = TYPE$E.Comment;
	var IDENT$i = TYPE$E.Ident;
	var NUMBER$9 = TYPE$E.Number;
	var DIMENSION$7 = TYPE$E.Dimension;
	var PLUSSIGN$8 = 0x002B;    // U+002B PLUS SIGN (+)
	var HYPHENMINUS$4 = 0x002D; // U+002D HYPHEN-MINUS (-)
	var N$4 = 0x006E;           // U+006E LATIN SMALL LETTER N (n)
	var DISALLOW_SIGN$1 = true;
	var ALLOW_SIGN$1 = false;

	function isDelim$1(token, code) {
	    return token !== null && token.type === DELIM$6 && token.value.charCodeAt(0) === code;
	}

	function skipSC(token, offset, getNextToken) {
	    while (token !== null && (token.type === WHITESPACE$b || token.type === COMMENT$9)) {
	        token = getNextToken(++offset);
	    }

	    return offset;
	}

	function checkInteger$1(token, valueOffset, disallowSign, offset) {
	    if (!token) {
	        return 0;
	    }

	    var code = token.value.charCodeAt(valueOffset);

	    if (code === PLUSSIGN$8 || code === HYPHENMINUS$4) {
	        if (disallowSign) {
	            // Number sign is not allowed
	            return 0;
	        }
	        valueOffset++;
	    }

	    for (; valueOffset < token.value.length; valueOffset++) {
	        if (!isDigit$3(token.value.charCodeAt(valueOffset))) {
	            // Integer is expected
	            return 0;
	        }
	    }

	    return offset + 1;
	}

	// ... <signed-integer>
	// ... ['+' | '-'] <signless-integer>
	function consumeB$1(token, offset_, getNextToken) {
	    var sign = false;
	    var offset = skipSC(token, offset_, getNextToken);

	    token = getNextToken(offset);

	    if (token === null) {
	        return offset_;
	    }

	    if (token.type !== NUMBER$9) {
	        if (isDelim$1(token, PLUSSIGN$8) || isDelim$1(token, HYPHENMINUS$4)) {
	            sign = true;
	            offset = skipSC(getNextToken(++offset), offset, getNextToken);
	            token = getNextToken(offset);

	            if (token === null && token.type !== NUMBER$9) {
	                return 0;
	            }
	        } else {
	            return offset_;
	        }
	    }

	    if (!sign) {
	        var code = token.value.charCodeAt(0);
	        if (code !== PLUSSIGN$8 && code !== HYPHENMINUS$4) {
	            // Number sign is expected
	            return 0;
	        }
	    }

	    return checkInteger$1(token, sign ? 0 : 1, sign, offset);
	}

	// An+B microsyntax https://www.w3.org/TR/css-syntax-3/#anb
	var genericAnPlusB = function anPlusB(token, getNextToken) {
	    /* eslint-disable brace-style*/
	    var offset = 0;

	    if (!token) {
	        return 0;
	    }

	    // <integer>
	    if (token.type === NUMBER$9) {
	        return checkInteger$1(token, 0, ALLOW_SIGN$1, offset); // b
	    }

	    // -n
	    // -n <signed-integer>
	    // -n ['+' | '-'] <signless-integer>
	    // -n- <signless-integer>
	    // <dashndashdigit-ident>
	    else if (token.type === IDENT$i && token.value.charCodeAt(0) === HYPHENMINUS$4) {
	        // expect 1st char is N
	        if (!cmpChar$4(token.value, 1, N$4)) {
	            return 0;
	        }

	        switch (token.value.length) {
	            // -n
	            // -n <signed-integer>
	            // -n ['+' | '-'] <signless-integer>
	            case 2:
	                return consumeB$1(getNextToken(++offset), offset, getNextToken);

	            // -n- <signless-integer>
	            case 3:
	                if (token.value.charCodeAt(2) !== HYPHENMINUS$4) {
	                    return 0;
	                }

	                offset = skipSC(getNextToken(++offset), offset, getNextToken);
	                token = getNextToken(offset);

	                return checkInteger$1(token, 0, DISALLOW_SIGN$1, offset);

	            // <dashndashdigit-ident>
	            default:
	                if (token.value.charCodeAt(2) !== HYPHENMINUS$4) {
	                    return 0;
	                }

	                return checkInteger$1(token, 3, DISALLOW_SIGN$1, offset);
	        }
	    }

	    // '+'? n
	    // '+'? n <signed-integer>
	    // '+'? n ['+' | '-'] <signless-integer>
	    // '+'? n- <signless-integer>
	    // '+'? <ndashdigit-ident>
	    else if (token.type === IDENT$i || (isDelim$1(token, PLUSSIGN$8) && getNextToken(offset + 1).type === IDENT$i)) {
	        // just ignore a plus
	        if (token.type !== IDENT$i) {
	            token = getNextToken(++offset);
	        }

	        if (token === null || !cmpChar$4(token.value, 0, N$4)) {
	            return 0;
	        }

	        switch (token.value.length) {
	            // '+'? n
	            // '+'? n <signed-integer>
	            // '+'? n ['+' | '-'] <signless-integer>
	            case 1:
	                return consumeB$1(getNextToken(++offset), offset, getNextToken);

	            // '+'? n- <signless-integer>
	            case 2:
	                if (token.value.charCodeAt(1) !== HYPHENMINUS$4) {
	                    return 0;
	                }

	                offset = skipSC(getNextToken(++offset), offset, getNextToken);
	                token = getNextToken(offset);

	                return checkInteger$1(token, 0, DISALLOW_SIGN$1, offset);

	            // '+'? <ndashdigit-ident>
	            default:
	                if (token.value.charCodeAt(1) !== HYPHENMINUS$4) {
	                    return 0;
	                }

	                return checkInteger$1(token, 2, DISALLOW_SIGN$1, offset);
	        }
	    }

	    // <ndashdigit-dimension>
	    // <ndash-dimension> <signless-integer>
	    // <n-dimension>
	    // <n-dimension> <signed-integer>
	    // <n-dimension> ['+' | '-'] <signless-integer>
	    else if (token.type === DIMENSION$7) {
	        var code = token.value.charCodeAt(0);
	        var sign = code === PLUSSIGN$8 || code === HYPHENMINUS$4 ? 1 : 0;

	        for (var i = sign; i < token.value.length; i++) {
	            if (!isDigit$3(token.value.charCodeAt(i))) {
	                break;
	            }
	        }

	        if (i === sign) {
	            // Integer is expected
	            return 0;
	        }

	        if (!cmpChar$4(token.value, i, N$4)) {
	            return 0;
	        }

	        // <n-dimension>
	        // <n-dimension> <signed-integer>
	        // <n-dimension> ['+' | '-'] <signless-integer>
	        if (i + 1 === token.value.length) {
	            return consumeB$1(getNextToken(++offset), offset, getNextToken);
	        } else {
	            if (token.value.charCodeAt(i + 1) !== HYPHENMINUS$4) {
	                return 0;
	            }

	            // <ndash-dimension> <signless-integer>
	            if (i + 2 === token.value.length) {
	                offset = skipSC(getNextToken(++offset), offset, getNextToken);
	                token = getNextToken(offset);

	                return checkInteger$1(token, 0, DISALLOW_SIGN$1, offset);
	            }
	            // <ndashdigit-dimension>
	            else {
	                return checkInteger$1(token, i + 2, DISALLOW_SIGN$1, offset);
	            }
	        }
	    }

	    return 0;
	};

	var isHexDigit$2 = tokenizer$3.isHexDigit;
	var cmpChar$3 = tokenizer$3.cmpChar;
	var TYPE$D = tokenizer$3.TYPE;

	var IDENT$h = TYPE$D.Ident;
	var DELIM$5 = TYPE$D.Delim;
	var NUMBER$8 = TYPE$D.Number;
	var DIMENSION$6 = TYPE$D.Dimension;
	var PLUSSIGN$7 = 0x002B;     // U+002B PLUS SIGN (+)
	var HYPHENMINUS$3 = 0x002D;  // U+002D HYPHEN-MINUS (-)
	var QUESTIONMARK$2 = 0x003F; // U+003F QUESTION MARK (?)
	var U$2 = 0x0075;            // U+0075 LATIN SMALL LETTER U (u)

	function isDelim(token, code) {
	    return token !== null && token.type === DELIM$5 && token.value.charCodeAt(0) === code;
	}

	function startsWith$1(token, code) {
	    return token.value.charCodeAt(0) === code;
	}

	function hexSequence(token, offset, allowDash) {
	    for (var pos = offset, hexlen = 0; pos < token.value.length; pos++) {
	        var code = token.value.charCodeAt(pos);

	        if (code === HYPHENMINUS$3 && allowDash && hexlen !== 0) {
	            if (hexSequence(token, offset + hexlen + 1, false) > 0) {
	                return 6; // dissallow following question marks
	            }

	            return 0; // dash at the ending of a hex sequence is not allowed
	        }

	        if (!isHexDigit$2(code)) {
	            return 0; // not a hex digit
	        }

	        if (++hexlen > 6) {
	            return 0; // too many hex digits
	        }    }

	    return hexlen;
	}

	function withQuestionMarkSequence(consumed, length, getNextToken) {
	    if (!consumed) {
	        return 0; // nothing consumed
	    }

	    while (isDelim(getNextToken(length), QUESTIONMARK$2)) {
	        if (++consumed > 6) {
	            return 0; // too many question marks
	        }

	        length++;
	    }

	    return length;
	}

	// https://drafts.csswg.org/css-syntax/#urange
	// Informally, the <urange> production has three forms:
	// U+0001
	//      Defines a range consisting of a single code point, in this case the code point "1".
	// U+0001-00ff
	//      Defines a range of codepoints between the first and the second value, in this case
	//      the range between "1" and "ff" (255 in decimal) inclusive.
	// U+00??
	//      Defines a range of codepoints where the "?" characters range over all hex digits,
	//      in this case defining the same as the value U+0000-00ff.
	// In each form, a maximum of 6 digits is allowed for each hexadecimal number (if you treat "?" as a hexadecimal digit).
	//
	// <urange> =
	//   u '+' <ident-token> '?'* |
	//   u <dimension-token> '?'* |
	//   u <number-token> '?'* |
	//   u <number-token> <dimension-token> |
	//   u <number-token> <number-token> |
	//   u '+' '?'+
	var genericUrange = function urange(token, getNextToken) {
	    var length = 0;

	    // should start with `u` or `U`
	    if (token === null || token.type !== IDENT$h || !cmpChar$3(token.value, 0, U$2)) {
	        return 0;
	    }

	    token = getNextToken(++length);
	    if (token === null) {
	        return 0;
	    }

	    // u '+' <ident-token> '?'*
	    // u '+' '?'+
	    if (isDelim(token, PLUSSIGN$7)) {
	        token = getNextToken(++length);
	        if (token === null) {
	            return 0;
	        }

	        if (token.type === IDENT$h) {
	            // u '+' <ident-token> '?'*
	            return withQuestionMarkSequence(hexSequence(token, 0, true), ++length, getNextToken);
	        }

	        if (isDelim(token, QUESTIONMARK$2)) {
	            // u '+' '?'+
	            return withQuestionMarkSequence(1, ++length, getNextToken);
	        }

	        // Hex digit or question mark is expected
	        return 0;
	    }

	    // u <number-token> '?'*
	    // u <number-token> <dimension-token>
	    // u <number-token> <number-token>
	    if (token.type === NUMBER$8) {
	        if (!startsWith$1(token, PLUSSIGN$7)) {
	            return 0;
	        }

	        var consumedHexLength = hexSequence(token, 1, true);
	        if (consumedHexLength === 0) {
	            return 0;
	        }

	        token = getNextToken(++length);
	        if (token === null) {
	            // u <number-token> <eof>
	            return length;
	        }

	        if (token.type === DIMENSION$6 || token.type === NUMBER$8) {
	            // u <number-token> <dimension-token>
	            // u <number-token> <number-token>
	            if (!startsWith$1(token, HYPHENMINUS$3) || !hexSequence(token, 1, false)) {
	                return 0;
	            }

	            return length + 1;
	        }

	        // u <number-token> '?'*
	        return withQuestionMarkSequence(consumedHexLength, length, getNextToken);
	    }

	    // u <dimension-token> '?'*
	    if (token.type === DIMENSION$6) {
	        if (!startsWith$1(token, PLUSSIGN$7)) {
	            return 0;
	        }

	        return withQuestionMarkSequence(hexSequence(token, 1, true), ++length, getNextToken);
	    }

	    return 0;
	};

	var tokenizer$2 = tokenizer$3;
	var isIdentifierStart = tokenizer$2.isIdentifierStart;
	var isHexDigit$1 = tokenizer$2.isHexDigit;
	var isDigit$2 = tokenizer$2.isDigit;
	var cmpStr$3 = tokenizer$2.cmpStr;
	var consumeNumber$3 = tokenizer$2.consumeNumber;
	var TYPE$C = tokenizer$2.TYPE;
	var anPlusB = genericAnPlusB;
	var urange = genericUrange;

	var cssWideKeywords$1 = ['unset', 'initial', 'inherit'];
	var calcFunctionNames = ['calc(', '-moz-calc(', '-webkit-calc('];

	// https://www.w3.org/TR/css-values-3/#lengths
	var LENGTH = {
	    // absolute length units
	    'px': true,
	    'mm': true,
	    'cm': true,
	    'in': true,
	    'pt': true,
	    'pc': true,
	    'q': true,

	    // relative length units
	    'em': true,
	    'ex': true,
	    'ch': true,
	    'rem': true,

	    // viewport-percentage lengths
	    'vh': true,
	    'vw': true,
	    'vmin': true,
	    'vmax': true,
	    'vm': true
	};

	var ANGLE = {
	    'deg': true,
	    'grad': true,
	    'rad': true,
	    'turn': true
	};

	var TIME = {
	    's': true,
	    'ms': true
	};

	var FREQUENCY = {
	    'hz': true,
	    'khz': true
	};

	// https://www.w3.org/TR/css-values-3/#resolution (https://drafts.csswg.org/css-values/#resolution)
	var RESOLUTION = {
	    'dpi': true,
	    'dpcm': true,
	    'dppx': true,
	    'x': true      // https://github.com/w3c/csswg-drafts/issues/461
	};

	// https://drafts.csswg.org/css-grid/#fr-unit
	var FLEX = {
	    'fr': true
	};

	// https://www.w3.org/TR/css3-speech/#mixing-props-voice-volume
	var DECIBEL = {
	    'db': true
	};

	// https://www.w3.org/TR/css3-speech/#voice-props-voice-pitch
	var SEMITONES = {
	    'st': true
	};

	// safe char code getter
	function charCode(str, index) {
	    return index < str.length ? str.charCodeAt(index) : 0;
	}

	function eqStr(actual, expected) {
	    return cmpStr$3(actual, 0, actual.length, expected);
	}

	function eqStrAny(actual, expected) {
	    for (var i = 0; i < expected.length; i++) {
	        if (eqStr(actual, expected[i])) {
	            return true;
	        }
	    }

	    return false;
	}

	// IE postfix hack, i.e. 123\0 or 123px\9
	function isPostfixIeHack(str, offset) {
	    if (offset !== str.length - 2) {
	        return false;
	    }

	    return (
	        str.charCodeAt(offset) === 0x005C &&  // U+005C REVERSE SOLIDUS (\)
	        isDigit$2(str.charCodeAt(offset + 1))
	    );
	}

	function outOfRange(opts, value, numEnd) {
	    if (opts && opts.type === 'Range') {
	        var num = Number(
	            numEnd !== undefined && numEnd !== value.length
	                ? value.substr(0, numEnd)
	                : value
	        );

	        if (isNaN(num)) {
	            return true;
	        }

	        if (opts.min !== null && num < opts.min) {
	            return true;
	        }

	        if (opts.max !== null && num > opts.max) {
	            return true;
	        }
	    }

	    return false;
	}

	function consumeFunction(token, getNextToken) {
	    var startIdx = token.index;
	    var length = 0;

	    // balanced token consuming
	    do {
	        length++;

	        if (token.balance <= startIdx) {
	            break;
	        }
	    } while (token = getNextToken(length));

	    return length;
	}

	// TODO: implement
	// can be used wherever <length>, <frequency>, <angle>, <time>, <percentage>, <number>, or <integer> values are allowed
	// https://drafts.csswg.org/css-values/#calc-notation
	function calc(next) {
	    return function(token, getNextToken, opts) {
	        if (token === null) {
	            return 0;
	        }

	        if (token.type === TYPE$C.Function && eqStrAny(token.value, calcFunctionNames)) {
	            return consumeFunction(token, getNextToken);
	        }

	        return next(token, getNextToken, opts);
	    };
	}

	function tokenType(expectedTokenType) {
	    return function(token) {
	        if (token === null || token.type !== expectedTokenType) {
	            return 0;
	        }

	        return 1;
	    };
	}

	function func(name) {
	    name = name + '(';

	    return function(token, getNextToken) {
	        if (token !== null && eqStr(token.value, name)) {
	            return consumeFunction(token, getNextToken);
	        }

	        return 0;
	    };
	}

	// =========================
	// Complex types
	//

	// https://drafts.csswg.org/css-values-4/#custom-idents
	// 4.2. Author-defined Identifiers: the <custom-ident> type
	// Some properties accept arbitrary author-defined identifiers as a component value.
	// This generic data type is denoted by <custom-ident>, and represents any valid CSS identifier
	// that would not be misinterpreted as a pre-defined keyword in that property’s value definition.
	//
	// See also: https://developer.mozilla.org/en-US/docs/Web/CSS/custom-ident
	function customIdent(token) {
	    if (token === null || token.type !== TYPE$C.Ident) {
	        return 0;
	    }

	    var name = token.value.toLowerCase();

	    // The CSS-wide keywords are not valid <custom-ident>s
	    if (eqStrAny(name, cssWideKeywords$1)) {
	        return 0;
	    }

	    // The default keyword is reserved and is also not a valid <custom-ident>
	    if (eqStr(name, 'default')) {
	        return 0;
	    }

	    // TODO: ignore property specific keywords (as described https://developer.mozilla.org/en-US/docs/Web/CSS/custom-ident)
	    // Specifications using <custom-ident> must specify clearly what other keywords
	    // are excluded from <custom-ident>, if any—for example by saying that any pre-defined keywords
	    // in that property’s value definition are excluded. Excluded keywords are excluded
	    // in all ASCII case permutations.

	    return 1;
	}

	// https://drafts.csswg.org/css-variables/#typedef-custom-property-name
	// A custom property is any property whose name starts with two dashes (U+002D HYPHEN-MINUS), like --foo.
	// The <custom-property-name> production corresponds to this: it’s defined as any valid identifier
	// that starts with two dashes, except -- itself, which is reserved for future use by CSS.
	// NOTE: Current implementation treat `--` as a valid name since most (all?) major browsers treat it as valid.
	function customPropertyName(token) {
	    // ... defined as any valid identifier
	    if (token === null || token.type !== TYPE$C.Ident) {
	        return 0;
	    }

	    // ... that starts with two dashes (U+002D HYPHEN-MINUS)
	    if (charCode(token.value, 0) !== 0x002D || charCode(token.value, 1) !== 0x002D) {
	        return 0;
	    }

	    return 1;
	}

	// https://drafts.csswg.org/css-color-4/#hex-notation
	// The syntax of a <hex-color> is a <hash-token> token whose value consists of 3, 4, 6, or 8 hexadecimal digits.
	// In other words, a hex color is written as a hash character, "#", followed by some number of digits 0-9 or
	// letters a-f (the case of the letters doesn’t matter - #00ff00 is identical to #00FF00).
	function hexColor(token) {
	    if (token === null || token.type !== TYPE$C.Hash) {
	        return 0;
	    }

	    var length = token.value.length;

	    // valid values (length): #rgb (4), #rgba (5), #rrggbb (7), #rrggbbaa (9)
	    if (length !== 4 && length !== 5 && length !== 7 && length !== 9) {
	        return 0;
	    }

	    for (var i = 1; i < length; i++) {
	        if (!isHexDigit$1(token.value.charCodeAt(i))) {
	            return 0;
	        }
	    }

	    return 1;
	}

	function idSelector(token) {
	    if (token === null || token.type !== TYPE$C.Hash) {
	        return 0;
	    }

	    if (!isIdentifierStart(charCode(token.value, 1), charCode(token.value, 2), charCode(token.value, 3))) {
	        return 0;
	    }

	    return 1;
	}

	// https://drafts.csswg.org/css-syntax/#any-value
	// It represents the entirety of what a valid declaration can have as its value.
	function declarationValue(token, getNextToken) {
	    if (!token) {
	        return 0;
	    }

	    var length = 0;
	    var level = 0;
	    var startIdx = token.index;

	    // The <declaration-value> production matches any sequence of one or more tokens,
	    // so long as the sequence ...
	    scan:
	    do {
	        switch (token.type) {
	            // ... does not contain <bad-string-token>, <bad-url-token>,
	            case TYPE$C.BadString:
	            case TYPE$C.BadUrl:
	                break scan;

	            // ... unmatched <)-token>, <]-token>, or <}-token>,
	            case TYPE$C.RightCurlyBracket:
	            case TYPE$C.RightParenthesis:
	            case TYPE$C.RightSquareBracket:
	                if (token.balance > token.index || token.balance < startIdx) {
	                    break scan;
	                }

	                level--;
	                break;

	            // ... or top-level <semicolon-token> tokens
	            case TYPE$C.Semicolon:
	                if (level === 0) {
	                    break scan;
	                }

	                break;

	            // ... or <delim-token> tokens with a value of "!"
	            case TYPE$C.Delim:
	                if (token.value === '!' && level === 0) {
	                    break scan;
	                }

	                break;

	            case TYPE$C.Function:
	            case TYPE$C.LeftParenthesis:
	            case TYPE$C.LeftSquareBracket:
	            case TYPE$C.LeftCurlyBracket:
	                level++;
	                break;
	        }

	        length++;

	        // until balance closing
	        if (token.balance <= startIdx) {
	            break;
	        }
	    } while (token = getNextToken(length));

	    return length;
	}

	// https://drafts.csswg.org/css-syntax/#any-value
	// The <any-value> production is identical to <declaration-value>, but also
	// allows top-level <semicolon-token> tokens and <delim-token> tokens
	// with a value of "!". It represents the entirety of what valid CSS can be in any context.
	function anyValue(token, getNextToken) {
	    if (!token) {
	        return 0;
	    }

	    var startIdx = token.index;
	    var length = 0;

	    // The <any-value> production matches any sequence of one or more tokens,
	    // so long as the sequence ...
	    scan:
	    do {
	        switch (token.type) {
	            // ... does not contain <bad-string-token>, <bad-url-token>,
	            case TYPE$C.BadString:
	            case TYPE$C.BadUrl:
	                break scan;

	            // ... unmatched <)-token>, <]-token>, or <}-token>,
	            case TYPE$C.RightCurlyBracket:
	            case TYPE$C.RightParenthesis:
	            case TYPE$C.RightSquareBracket:
	                if (token.balance > token.index || token.balance < startIdx) {
	                    break scan;
	                }

	                break;
	        }

	        length++;

	        // until balance closing
	        if (token.balance <= startIdx) {
	            break;
	        }
	    } while (token = getNextToken(length));

	    return length;
	}

	// =========================
	// Dimensions
	//

	function dimension(type) {
	    return function(token, getNextToken, opts) {
	        if (token === null || token.type !== TYPE$C.Dimension) {
	            return 0;
	        }

	        var numberEnd = consumeNumber$3(token.value, 0);

	        // check unit
	        if (type !== null) {
	            // check for IE postfix hack, i.e. 123px\0 or 123px\9
	            var reverseSolidusOffset = token.value.indexOf('\\', numberEnd);
	            var unit = reverseSolidusOffset === -1 || !isPostfixIeHack(token.value, reverseSolidusOffset)
	                ? token.value.substr(numberEnd)
	                : token.value.substring(numberEnd, reverseSolidusOffset);

	            if (type.hasOwnProperty(unit.toLowerCase()) === false) {
	                return 0;
	            }
	        }

	        // check range if specified
	        if (outOfRange(opts, token.value, numberEnd)) {
	            return 0;
	        }

	        return 1;
	    };
	}

	// =========================
	// Percentage
	//

	// §5.5. Percentages: the <percentage> type
	// https://drafts.csswg.org/css-values-4/#percentages
	function percentage(token, getNextToken, opts) {
	    // ... corresponds to the <percentage-token> production
	    if (token === null || token.type !== TYPE$C.Percentage) {
	        return 0;
	    }

	    // check range if specified
	    if (outOfRange(opts, token.value, token.value.length - 1)) {
	        return 0;
	    }

	    return 1;
	}

	// =========================
	// Numeric
	//

	// https://drafts.csswg.org/css-values-4/#numbers
	// The value <zero> represents a literal number with the value 0. Expressions that merely
	// evaluate to a <number> with the value 0 (for example, calc(0)) do not match <zero>;
	// only literal <number-token>s do.
	function zero(next) {
	    if (typeof next !== 'function') {
	        next = function() {
	            return 0;
	        };
	    }

	    return function(token, getNextToken, opts) {
	        if (token !== null && token.type === TYPE$C.Number) {
	            if (Number(token.value) === 0) {
	                return 1;
	            }
	        }

	        return next(token, getNextToken, opts);
	    };
	}

	// § 5.3. Real Numbers: the <number> type
	// https://drafts.csswg.org/css-values-4/#numbers
	// Number values are denoted by <number>, and represent real numbers, possibly with a fractional component.
	// ... It corresponds to the <number-token> production
	function number(token, getNextToken, opts) {
	    if (token === null) {
	        return 0;
	    }

	    var numberEnd = consumeNumber$3(token.value, 0);
	    var isNumber = numberEnd === token.value.length;
	    if (!isNumber && !isPostfixIeHack(token.value, numberEnd)) {
	        return 0;
	    }

	    // check range if specified
	    if (outOfRange(opts, token.value, numberEnd)) {
	        return 0;
	    }

	    return 1;
	}

	// §5.2. Integers: the <integer> type
	// https://drafts.csswg.org/css-values-4/#integers
	function integer(token, getNextToken, opts) {
	    // ... corresponds to a subset of the <number-token> production
	    if (token === null || token.type !== TYPE$C.Number) {
	        return 0;
	    }

	    // The first digit of an integer may be immediately preceded by `-` or `+` to indicate the integer’s sign.
	    var i = token.value.charCodeAt(0) === 0x002B ||       // U+002B PLUS SIGN (+)
	            token.value.charCodeAt(0) === 0x002D ? 1 : 0; // U+002D HYPHEN-MINUS (-)

	    // When written literally, an integer is one or more decimal digits 0 through 9 ...
	    for (; i < token.value.length; i++) {
	        if (!isDigit$2(token.value.charCodeAt(i))) {
	            return 0;
	        }
	    }

	    // check range if specified
	    if (outOfRange(opts, token.value, i)) {
	        return 0;
	    }

	    return 1;
	}

	var generic$1 = {
	    // token types
	    'ident-token': tokenType(TYPE$C.Ident),
	    'function-token': tokenType(TYPE$C.Function),
	    'at-keyword-token': tokenType(TYPE$C.AtKeyword),
	    'hash-token': tokenType(TYPE$C.Hash),
	    'string-token': tokenType(TYPE$C.String),
	    'bad-string-token': tokenType(TYPE$C.BadString),
	    'url-token': tokenType(TYPE$C.Url),
	    'bad-url-token': tokenType(TYPE$C.BadUrl),
	    'delim-token': tokenType(TYPE$C.Delim),
	    'number-token': tokenType(TYPE$C.Number),
	    'percentage-token': tokenType(TYPE$C.Percentage),
	    'dimension-token': tokenType(TYPE$C.Dimension),
	    'whitespace-token': tokenType(TYPE$C.WhiteSpace),
	    'CDO-token': tokenType(TYPE$C.CDO),
	    'CDC-token': tokenType(TYPE$C.CDC),
	    'colon-token': tokenType(TYPE$C.Colon),
	    'semicolon-token': tokenType(TYPE$C.Semicolon),
	    'comma-token': tokenType(TYPE$C.Comma),
	    '[-token': tokenType(TYPE$C.LeftSquareBracket),
	    ']-token': tokenType(TYPE$C.RightSquareBracket),
	    '(-token': tokenType(TYPE$C.LeftParenthesis),
	    ')-token': tokenType(TYPE$C.RightParenthesis),
	    '{-token': tokenType(TYPE$C.LeftCurlyBracket),
	    '}-token': tokenType(TYPE$C.RightCurlyBracket),

	    // token type aliases
	    'string': tokenType(TYPE$C.String),
	    'ident': tokenType(TYPE$C.Ident),

	    // complex types
	    'custom-ident': customIdent,
	    'custom-property-name': customPropertyName,
	    'hex-color': hexColor,
	    'id-selector': idSelector, // element( <id-selector> )
	    'an-plus-b': anPlusB,
	    'urange': urange,
	    'declaration-value': declarationValue,
	    'any-value': anyValue,

	    // dimensions
	    'dimension': calc(dimension(null)),
	    'angle': calc(dimension(ANGLE)),
	    'decibel': calc(dimension(DECIBEL)),
	    'frequency': calc(dimension(FREQUENCY)),
	    'flex': calc(dimension(FLEX)),
	    'length': calc(zero(dimension(LENGTH))),
	    'resolution': calc(dimension(RESOLUTION)),
	    'semitones': calc(dimension(SEMITONES)),
	    'time': calc(dimension(TIME)),

	    // percentage
	    'percentage': calc(percentage),

	    // numeric
	    'zero': zero(),
	    'number': calc(number),
	    'integer': calc(integer),

	    // old IE stuff
	    '-ms-legacy-expression': func('expression')
	};

	var createCustomError = createCustomError$3;

	var _SyntaxError = function SyntaxError(message, input, offset) {
	    var error = createCustomError('SyntaxError', message);

	    error.input = input;
	    error.offset = offset;
	    error.rawMessage = message;
	    error.message = error.rawMessage + '\n' +
	        '  ' + error.input + '\n' +
	        '--' + new Array((error.offset || error.input.length) + 1).join('-') + '^';

	    return error;
	};

	var SyntaxError$3 = _SyntaxError;

	var TAB$1 = 9;
	var N$3 = 10;
	var F$2 = 12;
	var R$2 = 13;
	var SPACE$2 = 32;

	var Tokenizer$1 = function(str) {
	    this.str = str;
	    this.pos = 0;
	};

	Tokenizer$1.prototype = {
	    charCodeAt: function(pos) {
	        return pos < this.str.length ? this.str.charCodeAt(pos) : 0;
	    },
	    charCode: function() {
	        return this.charCodeAt(this.pos);
	    },
	    nextCharCode: function() {
	        return this.charCodeAt(this.pos + 1);
	    },
	    nextNonWsCode: function(pos) {
	        return this.charCodeAt(this.findWsEnd(pos));
	    },
	    findWsEnd: function(pos) {
	        for (; pos < this.str.length; pos++) {
	            var code = this.str.charCodeAt(pos);
	            if (code !== R$2 && code !== N$3 && code !== F$2 && code !== SPACE$2 && code !== TAB$1) {
	                break;
	            }
	        }

	        return pos;
	    },
	    substringToPos: function(end) {
	        return this.str.substring(this.pos, this.pos = end);
	    },
	    eat: function(code) {
	        if (this.charCode() !== code) {
	            this.error('Expect `' + String.fromCharCode(code) + '`');
	        }

	        this.pos++;
	    },
	    peek: function() {
	        return this.pos < this.str.length ? this.str.charAt(this.pos++) : '';
	    },
	    error: function(message) {
	        throw new SyntaxError$3(message, this.str, this.pos);
	    }
	};

	var tokenizer$1 = Tokenizer$1;

	var Tokenizer = tokenizer$1;
	var TAB = 9;
	var N$2 = 10;
	var F$1 = 12;
	var R$1 = 13;
	var SPACE$1 = 32;
	var EXCLAMATIONMARK$3 = 33;    // !
	var NUMBERSIGN$4 = 35;         // #
	var AMPERSAND$1 = 38;          // &
	var APOSTROPHE = 39;         // '
	var LEFTPARENTHESIS$7 = 40;    // (
	var RIGHTPARENTHESIS$7 = 41;   // )
	var ASTERISK$6 = 42;           // *
	var PLUSSIGN$6 = 43;           // +
	var COMMA$4 = 44;              // ,
	var HYPERMINUS = 45;         // -
	var LESSTHANSIGN = 60;       // <
	var GREATERTHANSIGN$2 = 62;    // >
	var QUESTIONMARK$1 = 63;       // ?
	var COMMERCIALAT = 64;       // @
	var LEFTSQUAREBRACKET$4 = 91;  // [
	var RIGHTSQUAREBRACKET$2 = 93; // ]
	var LEFTCURLYBRACKET$4 = 123;  // {
	var VERTICALLINE$3 = 124;      // |
	var RIGHTCURLYBRACKET$2 = 125; // }
	var INFINITY = 8734;         // ∞
	var NAME_CHAR = createCharMap(function(ch) {
	    return /[a-zA-Z0-9\-]/.test(ch);
	});
	var COMBINATOR_PRECEDENCE = {
	    ' ': 1,
	    '&&': 2,
	    '||': 3,
	    '|': 4
	};

	function createCharMap(fn) {
	    var array = typeof Uint32Array === 'function' ? new Uint32Array(128) : new Array(128);
	    for (var i = 0; i < 128; i++) {
	        array[i] = fn(String.fromCharCode(i)) ? 1 : 0;
	    }
	    return array;
	}

	function scanSpaces(tokenizer) {
	    return tokenizer.substringToPos(
	        tokenizer.findWsEnd(tokenizer.pos)
	    );
	}

	function scanWord(tokenizer) {
	    var end = tokenizer.pos;

	    for (; end < tokenizer.str.length; end++) {
	        var code = tokenizer.str.charCodeAt(end);
	        if (code >= 128 || NAME_CHAR[code] === 0) {
	            break;
	        }
	    }

	    if (tokenizer.pos === end) {
	        tokenizer.error('Expect a keyword');
	    }

	    return tokenizer.substringToPos(end);
	}

	function scanNumber(tokenizer) {
	    var end = tokenizer.pos;

	    for (; end < tokenizer.str.length; end++) {
	        var code = tokenizer.str.charCodeAt(end);
	        if (code < 48 || code > 57) {
	            break;
	        }
	    }

	    if (tokenizer.pos === end) {
	        tokenizer.error('Expect a number');
	    }

	    return tokenizer.substringToPos(end);
	}

	function scanString(tokenizer) {
	    var end = tokenizer.str.indexOf('\'', tokenizer.pos + 1);

	    if (end === -1) {
	        tokenizer.pos = tokenizer.str.length;
	        tokenizer.error('Expect an apostrophe');
	    }

	    return tokenizer.substringToPos(end + 1);
	}

	function readMultiplierRange(tokenizer) {
	    var min = null;
	    var max = null;

	    tokenizer.eat(LEFTCURLYBRACKET$4);

	    min = scanNumber(tokenizer);

	    if (tokenizer.charCode() === COMMA$4) {
	        tokenizer.pos++;
	        if (tokenizer.charCode() !== RIGHTCURLYBRACKET$2) {
	            max = scanNumber(tokenizer);
	        }
	    } else {
	        max = min;
	    }

	    tokenizer.eat(RIGHTCURLYBRACKET$2);

	    return {
	        min: Number(min),
	        max: max ? Number(max) : 0
	    };
	}

	function readMultiplier(tokenizer) {
	    var range = null;
	    var comma = false;

	    switch (tokenizer.charCode()) {
	        case ASTERISK$6:
	            tokenizer.pos++;

	            range = {
	                min: 0,
	                max: 0
	            };

	            break;

	        case PLUSSIGN$6:
	            tokenizer.pos++;

	            range = {
	                min: 1,
	                max: 0
	            };

	            break;

	        case QUESTIONMARK$1:
	            tokenizer.pos++;

	            range = {
	                min: 0,
	                max: 1
	            };

	            break;

	        case NUMBERSIGN$4:
	            tokenizer.pos++;

	            comma = true;

	            if (tokenizer.charCode() === LEFTCURLYBRACKET$4) {
	                range = readMultiplierRange(tokenizer);
	            } else {
	                range = {
	                    min: 1,
	                    max: 0
	                };
	            }

	            break;

	        case LEFTCURLYBRACKET$4:
	            range = readMultiplierRange(tokenizer);
	            break;

	        default:
	            return null;
	    }

	    return {
	        type: 'Multiplier',
	        comma: comma,
	        min: range.min,
	        max: range.max,
	        term: null
	    };
	}

	function maybeMultiplied(tokenizer, node) {
	    var multiplier = readMultiplier(tokenizer);

	    if (multiplier !== null) {
	        multiplier.term = node;
	        return multiplier;
	    }

	    return node;
	}

	function maybeToken(tokenizer) {
	    var ch = tokenizer.peek();

	    if (ch === '') {
	        return null;
	    }

	    return {
	        type: 'Token',
	        value: ch
	    };
	}

	function readProperty$1(tokenizer) {
	    var name;

	    tokenizer.eat(LESSTHANSIGN);
	    tokenizer.eat(APOSTROPHE);

	    name = scanWord(tokenizer);

	    tokenizer.eat(APOSTROPHE);
	    tokenizer.eat(GREATERTHANSIGN$2);

	    return maybeMultiplied(tokenizer, {
	        type: 'Property',
	        name: name
	    });
	}

	// https://drafts.csswg.org/css-values-3/#numeric-ranges
	// 4.1. Range Restrictions and Range Definition Notation
	//
	// Range restrictions can be annotated in the numeric type notation using CSS bracketed
	// range notation—[min,max]—within the angle brackets, after the identifying keyword,
	// indicating a closed range between (and including) min and max.
	// For example, <integer [0, 10]> indicates an integer between 0 and 10, inclusive.
	function readTypeRange(tokenizer) {
	    // use null for Infinity to make AST format JSON serializable/deserializable
	    var min = null; // -Infinity
	    var max = null; // Infinity
	    var sign = 1;

	    tokenizer.eat(LEFTSQUAREBRACKET$4);

	    if (tokenizer.charCode() === HYPERMINUS) {
	        tokenizer.peek();
	        sign = -1;
	    }

	    if (sign == -1 && tokenizer.charCode() === INFINITY) {
	        tokenizer.peek();
	    } else {
	        min = sign * Number(scanNumber(tokenizer));
	    }

	    scanSpaces(tokenizer);
	    tokenizer.eat(COMMA$4);
	    scanSpaces(tokenizer);

	    if (tokenizer.charCode() === INFINITY) {
	        tokenizer.peek();
	    } else {
	        sign = 1;

	        if (tokenizer.charCode() === HYPERMINUS) {
	            tokenizer.peek();
	            sign = -1;
	        }

	        max = sign * Number(scanNumber(tokenizer));
	    }

	    tokenizer.eat(RIGHTSQUAREBRACKET$2);

	    // If no range is indicated, either by using the bracketed range notation
	    // or in the property description, then [−∞,∞] is assumed.
	    if (min === null && max === null) {
	        return null;
	    }

	    return {
	        type: 'Range',
	        min: min,
	        max: max
	    };
	}

	function readType(tokenizer) {
	    var name;
	    var opts = null;

	    tokenizer.eat(LESSTHANSIGN);
	    name = scanWord(tokenizer);

	    if (tokenizer.charCode() === LEFTPARENTHESIS$7 &&
	        tokenizer.nextCharCode() === RIGHTPARENTHESIS$7) {
	        tokenizer.pos += 2;
	        name += '()';
	    }

	    if (tokenizer.charCodeAt(tokenizer.findWsEnd(tokenizer.pos)) === LEFTSQUAREBRACKET$4) {
	        scanSpaces(tokenizer);
	        opts = readTypeRange(tokenizer);
	    }

	    tokenizer.eat(GREATERTHANSIGN$2);

	    return maybeMultiplied(tokenizer, {
	        type: 'Type',
	        name: name,
	        opts: opts
	    });
	}

	function readKeywordOrFunction(tokenizer) {
	    var name;

	    name = scanWord(tokenizer);

	    if (tokenizer.charCode() === LEFTPARENTHESIS$7) {
	        tokenizer.pos++;

	        return {
	            type: 'Function',
	            name: name
	        };
	    }

	    return maybeMultiplied(tokenizer, {
	        type: 'Keyword',
	        name: name
	    });
	}

	function regroupTerms(terms, combinators) {
	    function createGroup(terms, combinator) {
	        return {
	            type: 'Group',
	            terms: terms,
	            combinator: combinator,
	            disallowEmpty: false,
	            explicit: false
	        };
	    }

	    combinators = Object.keys(combinators).sort(function(a, b) {
	        return COMBINATOR_PRECEDENCE[a] - COMBINATOR_PRECEDENCE[b];
	    });

	    while (combinators.length > 0) {
	        var combinator = combinators.shift();
	        for (var i = 0, subgroupStart = 0; i < terms.length; i++) {
	            var term = terms[i];
	            if (term.type === 'Combinator') {
	                if (term.value === combinator) {
	                    if (subgroupStart === -1) {
	                        subgroupStart = i - 1;
	                    }
	                    terms.splice(i, 1);
	                    i--;
	                } else {
	                    if (subgroupStart !== -1 && i - subgroupStart > 1) {
	                        terms.splice(
	                            subgroupStart,
	                            i - subgroupStart,
	                            createGroup(terms.slice(subgroupStart, i), combinator)
	                        );
	                        i = subgroupStart + 1;
	                    }
	                    subgroupStart = -1;
	                }
	            }
	        }

	        if (subgroupStart !== -1 && combinators.length) {
	            terms.splice(
	                subgroupStart,
	                i - subgroupStart,
	                createGroup(terms.slice(subgroupStart, i), combinator)
	            );
	        }
	    }

	    return combinator;
	}

	function readImplicitGroup(tokenizer) {
	    var terms = [];
	    var combinators = {};
	    var token;
	    var prevToken = null;
	    var prevTokenPos = tokenizer.pos;

	    while (token = peek(tokenizer)) {
	        if (token.type !== 'Spaces') {
	            if (token.type === 'Combinator') {
	                // check for combinator in group beginning and double combinator sequence
	                if (prevToken === null || prevToken.type === 'Combinator') {
	                    tokenizer.pos = prevTokenPos;
	                    tokenizer.error('Unexpected combinator');
	                }

	                combinators[token.value] = true;
	            } else if (prevToken !== null && prevToken.type !== 'Combinator') {
	                combinators[' '] = true;  // a b
	                terms.push({
	                    type: 'Combinator',
	                    value: ' '
	                });
	            }

	            terms.push(token);
	            prevToken = token;
	            prevTokenPos = tokenizer.pos;
	        }
	    }

	    // check for combinator in group ending
	    if (prevToken !== null && prevToken.type === 'Combinator') {
	        tokenizer.pos -= prevTokenPos;
	        tokenizer.error('Unexpected combinator');
	    }

	    return {
	        type: 'Group',
	        terms: terms,
	        combinator: regroupTerms(terms, combinators) || ' ',
	        disallowEmpty: false,
	        explicit: false
	    };
	}

	function readGroup(tokenizer) {
	    var result;

	    tokenizer.eat(LEFTSQUAREBRACKET$4);
	    result = readImplicitGroup(tokenizer);
	    tokenizer.eat(RIGHTSQUAREBRACKET$2);

	    result.explicit = true;

	    if (tokenizer.charCode() === EXCLAMATIONMARK$3) {
	        tokenizer.pos++;
	        result.disallowEmpty = true;
	    }

	    return result;
	}

	function peek(tokenizer) {
	    var code = tokenizer.charCode();

	    if (code < 128 && NAME_CHAR[code] === 1) {
	        return readKeywordOrFunction(tokenizer);
	    }

	    switch (code) {
	        case RIGHTSQUAREBRACKET$2:
	            // don't eat, stop scan a group
	            break;

	        case LEFTSQUAREBRACKET$4:
	            return maybeMultiplied(tokenizer, readGroup(tokenizer));

	        case LESSTHANSIGN:
	            return tokenizer.nextCharCode() === APOSTROPHE
	                ? readProperty$1(tokenizer)
	                : readType(tokenizer);

	        case VERTICALLINE$3:
	            return {
	                type: 'Combinator',
	                value: tokenizer.substringToPos(
	                    tokenizer.nextCharCode() === VERTICALLINE$3
	                        ? tokenizer.pos + 2
	                        : tokenizer.pos + 1
	                )
	            };

	        case AMPERSAND$1:
	            tokenizer.pos++;
	            tokenizer.eat(AMPERSAND$1);

	            return {
	                type: 'Combinator',
	                value: '&&'
	            };

	        case COMMA$4:
	            tokenizer.pos++;
	            return {
	                type: 'Comma'
	            };

	        case APOSTROPHE:
	            return maybeMultiplied(tokenizer, {
	                type: 'String',
	                value: scanString(tokenizer)
	            });

	        case SPACE$1:
	        case TAB:
	        case N$2:
	        case R$1:
	        case F$1:
	            return {
	                type: 'Spaces',
	                value: scanSpaces(tokenizer)
	            };

	        case COMMERCIALAT:
	            code = tokenizer.nextCharCode();

	            if (code < 128 && NAME_CHAR[code] === 1) {
	                tokenizer.pos++;
	                return {
	                    type: 'AtKeyword',
	                    name: scanWord(tokenizer)
	                };
	            }

	            return maybeToken(tokenizer);

	        case ASTERISK$6:
	        case PLUSSIGN$6:
	        case QUESTIONMARK$1:
	        case NUMBERSIGN$4:
	        case EXCLAMATIONMARK$3:
	            // prohibited tokens (used as a multiplier start)
	            break;

	        case LEFTCURLYBRACKET$4:
	            // LEFTCURLYBRACKET is allowed since mdn/data uses it w/o quoting
	            // check next char isn't a number, because it's likely a disjoined multiplier
	            code = tokenizer.nextCharCode();

	            if (code < 48 || code > 57) {
	                return maybeToken(tokenizer);
	            }

	            break;

	        default:
	            return maybeToken(tokenizer);
	    }
	}

	function parse$2(source) {
	    var tokenizer = new Tokenizer(source);
	    var result = readImplicitGroup(tokenizer);

	    if (tokenizer.pos !== source.length) {
	        tokenizer.error('Unexpected input');
	    }

	    // reduce redundant groups with single group term
	    if (result.terms.length === 1 && result.terms[0].type === 'Group') {
	        result = result.terms[0];
	    }

	    return result;
	}

	// warm up parse to elimitate code branches that never execute
	// fix soft deoptimizations (insufficient type feedback)
	parse$2('[a&&<b>#|<\'c\'>*||e() f{2} /,(% g#{1,2} h{2,})]!');

	var parse_1 = parse$2;

	var noop$2 = function() {};

	function ensureFunction$1(value) {
	    return typeof value === 'function' ? value : noop$2;
	}

	var walk$1 = function(node, options, context) {
	    function walk(node) {
	        enter.call(context, node);

	        switch (node.type) {
	            case 'Group':
	                node.terms.forEach(walk);
	                break;

	            case 'Multiplier':
	                walk(node.term);
	                break;

	            case 'Type':
	            case 'Property':
	            case 'Keyword':
	            case 'AtKeyword':
	            case 'Function':
	            case 'String':
	            case 'Token':
	            case 'Comma':
	                break;

	            default:
	                throw new Error('Unknown type: ' + node.type);
	        }

	        leave.call(context, node);
	    }

	    var enter = noop$2;
	    var leave = noop$2;

	    if (typeof options === 'function') {
	        enter = options;
	    } else if (options) {
	        enter = ensureFunction$1(options.enter);
	        leave = ensureFunction$1(options.leave);
	    }

	    if (enter === noop$2 && leave === noop$2) {
	        throw new Error('Neither `enter` nor `leave` walker handler is set or both aren\'t a function');
	    }

	    walk(node);
	};

	var tokenize$2 = tokenizer$3;
	var TokenStream$2 = TokenStream_1;
	var tokenStream = new TokenStream$2();
	var astToTokens = {
	    decorator: function(handlers) {
	        var curNode = null;
	        var prev = { len: 0, node: null };
	        var nodes = [prev];
	        var buffer = '';

	        return {
	            children: handlers.children,
	            node: function(node) {
	                var tmp = curNode;
	                curNode = node;
	                handlers.node.call(this, node);
	                curNode = tmp;
	            },
	            chunk: function(chunk) {
	                buffer += chunk;
	                if (prev.node !== curNode) {
	                    nodes.push({
	                        len: chunk.length,
	                        node: curNode
	                    });
	                } else {
	                    prev.len += chunk.length;
	                }
	            },
	            result: function() {
	                return prepareTokens$1(buffer, nodes);
	            }
	        };
	    }
	};

	function prepareTokens$1(str, nodes) {
	    var tokens = [];
	    var nodesOffset = 0;
	    var nodesIndex = 0;
	    var currentNode = nodes ? nodes[nodesIndex].node : null;

	    tokenize$2(str, tokenStream);

	    while (!tokenStream.eof) {
	        if (nodes) {
	            while (nodesIndex < nodes.length && nodesOffset + nodes[nodesIndex].len <= tokenStream.tokenStart) {
	                nodesOffset += nodes[nodesIndex++].len;
	                currentNode = nodes[nodesIndex].node;
	            }
	        }

	        tokens.push({
	            type: tokenStream.tokenType,
	            value: tokenStream.getTokenValue(),
	            index: tokenStream.tokenIndex, // TODO: remove it, temporary solution
	            balance: tokenStream.balance[tokenStream.tokenIndex], // TODO: remove it, temporary solution
	            node: currentNode
	        });
	        tokenStream.next();
	        // console.log({ ...tokens[tokens.length - 1], node: undefined });
	    }

	    return tokens;
	}

	var prepareTokens_1 = function(value, syntax) {
	    if (typeof value === 'string') {
	        return prepareTokens$1(value, null);
	    }

	    return syntax.generate(value, astToTokens);
	};

	var parse$1 = parse_1;

	var MATCH$1 = { type: 'Match' };
	var MISMATCH$1 = { type: 'Mismatch' };
	var DISALLOW_EMPTY$1 = { type: 'DisallowEmpty' };
	var LEFTPARENTHESIS$6 = 40;  // (
	var RIGHTPARENTHESIS$6 = 41; // )

	function createCondition(match, thenBranch, elseBranch) {
	    // reduce node count
	    if (thenBranch === MATCH$1 && elseBranch === MISMATCH$1) {
	        return match;
	    }

	    if (match === MATCH$1 && thenBranch === MATCH$1 && elseBranch === MATCH$1) {
	        return match;
	    }

	    if (match.type === 'If' && match.else === MISMATCH$1 && thenBranch === MATCH$1) {
	        thenBranch = match.then;
	        match = match.match;
	    }

	    return {
	        type: 'If',
	        match: match,
	        then: thenBranch,
	        else: elseBranch
	    };
	}

	function isFunctionType(name) {
	    return (
	        name.length > 2 &&
	        name.charCodeAt(name.length - 2) === LEFTPARENTHESIS$6 &&
	        name.charCodeAt(name.length - 1) === RIGHTPARENTHESIS$6
	    );
	}

	function isEnumCapatible(term) {
	    return (
	        term.type === 'Keyword' ||
	        term.type === 'AtKeyword' ||
	        term.type === 'Function' ||
	        term.type === 'Type' && isFunctionType(term.name)
	    );
	}

	function buildGroupMatchGraph(combinator, terms, atLeastOneTermMatched) {
	    switch (combinator) {
	        case ' ':
	            // Juxtaposing components means that all of them must occur, in the given order.
	            //
	            // a b c
	            // =
	            // match a
	            //   then match b
	            //     then match c
	            //       then MATCH
	            //       else MISMATCH
	            //     else MISMATCH
	            //   else MISMATCH
	            var result = MATCH$1;

	            for (var i = terms.length - 1; i >= 0; i--) {
	                var term = terms[i];

	                result = createCondition(
	                    term,
	                    result,
	                    MISMATCH$1
	                );
	            }
	            return result;

	        case '|':
	            // A bar (|) separates two or more alternatives: exactly one of them must occur.
	            //
	            // a | b | c
	            // =
	            // match a
	            //   then MATCH
	            //   else match b
	            //     then MATCH
	            //     else match c
	            //       then MATCH
	            //       else MISMATCH

	            var result = MISMATCH$1;
	            var map = null;

	            for (var i = terms.length - 1; i >= 0; i--) {
	                var term = terms[i];

	                // reduce sequence of keywords into a Enum
	                if (isEnumCapatible(term)) {
	                    if (map === null && i > 0 && isEnumCapatible(terms[i - 1])) {
	                        map = Object.create(null);
	                        result = createCondition(
	                            {
	                                type: 'Enum',
	                                map: map
	                            },
	                            MATCH$1,
	                            result
	                        );
	                    }

	                    if (map !== null) {
	                        var key = (isFunctionType(term.name) ? term.name.slice(0, -1) : term.name).toLowerCase();
	                        if (key in map === false) {
	                            map[key] = term;
	                            continue;
	                        }
	                    }
	                }

	                map = null;

	                // create a new conditonal node
	                result = createCondition(
	                    term,
	                    MATCH$1,
	                    result
	                );
	            }
	            return result;

	        case '&&':
	            // A double ampersand (&&) separates two or more components,
	            // all of which must occur, in any order.

	            // Use MatchOnce for groups with a large number of terms,
	            // since &&-groups produces at least N!-node trees
	            if (terms.length > 5) {
	                return {
	                    type: 'MatchOnce',
	                    terms: terms,
	                    all: true
	                };
	            }

	            // Use a combination tree for groups with small number of terms
	            //
	            // a && b && c
	            // =
	            // match a
	            //   then [b && c]
	            //   else match b
	            //     then [a && c]
	            //     else match c
	            //       then [a && b]
	            //       else MISMATCH
	            //
	            // a && b
	            // =
	            // match a
	            //   then match b
	            //     then MATCH
	            //     else MISMATCH
	            //   else match b
	            //     then match a
	            //       then MATCH
	            //       else MISMATCH
	            //     else MISMATCH
	            var result = MISMATCH$1;

	            for (var i = terms.length - 1; i >= 0; i--) {
	                var term = terms[i];
	                var thenClause;

	                if (terms.length > 1) {
	                    thenClause = buildGroupMatchGraph(
	                        combinator,
	                        terms.filter(function(newGroupTerm) {
	                            return newGroupTerm !== term;
	                        }),
	                        false
	                    );
	                } else {
	                    thenClause = MATCH$1;
	                }

	                result = createCondition(
	                    term,
	                    thenClause,
	                    result
	                );
	            }
	            return result;

	        case '||':
	            // A double bar (||) separates two or more options:
	            // one or more of them must occur, in any order.

	            // Use MatchOnce for groups with a large number of terms,
	            // since ||-groups produces at least N!-node trees
	            if (terms.length > 5) {
	                return {
	                    type: 'MatchOnce',
	                    terms: terms,
	                    all: false
	                };
	            }

	            // Use a combination tree for groups with small number of terms
	            //
	            // a || b || c
	            // =
	            // match a
	            //   then [b || c]
	            //   else match b
	            //     then [a || c]
	            //     else match c
	            //       then [a || b]
	            //       else MISMATCH
	            //
	            // a || b
	            // =
	            // match a
	            //   then match b
	            //     then MATCH
	            //     else MATCH
	            //   else match b
	            //     then match a
	            //       then MATCH
	            //       else MATCH
	            //     else MISMATCH
	            var result = atLeastOneTermMatched ? MATCH$1 : MISMATCH$1;

	            for (var i = terms.length - 1; i >= 0; i--) {
	                var term = terms[i];
	                var thenClause;

	                if (terms.length > 1) {
	                    thenClause = buildGroupMatchGraph(
	                        combinator,
	                        terms.filter(function(newGroupTerm) {
	                            return newGroupTerm !== term;
	                        }),
	                        true
	                    );
	                } else {
	                    thenClause = MATCH$1;
	                }

	                result = createCondition(
	                    term,
	                    thenClause,
	                    result
	                );
	            }
	            return result;
	    }
	}

	function buildMultiplierMatchGraph(node) {
	    var result = MATCH$1;
	    var matchTerm = buildMatchGraph$1(node.term);

	    if (node.max === 0) {
	        // disable repeating of empty match to prevent infinite loop
	        matchTerm = createCondition(
	            matchTerm,
	            DISALLOW_EMPTY$1,
	            MISMATCH$1
	        );

	        // an occurrence count is not limited, make a cycle;
	        // to collect more terms on each following matching mismatch
	        result = createCondition(
	            matchTerm,
	            null, // will be a loop
	            MISMATCH$1
	        );

	        result.then = createCondition(
	            MATCH$1,
	            MATCH$1,
	            result // make a loop
	        );

	        if (node.comma) {
	            result.then.else = createCondition(
	                { type: 'Comma', syntax: node },
	                result,
	                MISMATCH$1
	            );
	        }
	    } else {
	        // create a match node chain for [min .. max] interval with optional matches
	        for (var i = node.min || 1; i <= node.max; i++) {
	            if (node.comma && result !== MATCH$1) {
	                result = createCondition(
	                    { type: 'Comma', syntax: node },
	                    result,
	                    MISMATCH$1
	                );
	            }

	            result = createCondition(
	                matchTerm,
	                createCondition(
	                    MATCH$1,
	                    MATCH$1,
	                    result
	                ),
	                MISMATCH$1
	            );
	        }
	    }

	    if (node.min === 0) {
	        // allow zero match
	        result = createCondition(
	            MATCH$1,
	            MATCH$1,
	            result
	        );
	    } else {
	        // create a match node chain to collect [0 ... min - 1] required matches
	        for (var i = 0; i < node.min - 1; i++) {
	            if (node.comma && result !== MATCH$1) {
	                result = createCondition(
	                    { type: 'Comma', syntax: node },
	                    result,
	                    MISMATCH$1
	                );
	            }

	            result = createCondition(
	                matchTerm,
	                result,
	                MISMATCH$1
	            );
	        }
	    }

	    return result;
	}

	function buildMatchGraph$1(node) {
	    if (typeof node === 'function') {
	        return {
	            type: 'Generic',
	            fn: node
	        };
	    }

	    switch (node.type) {
	        case 'Group':
	            var result = buildGroupMatchGraph(
	                node.combinator,
	                node.terms.map(buildMatchGraph$1),
	                false
	            );

	            if (node.disallowEmpty) {
	                result = createCondition(
	                    result,
	                    DISALLOW_EMPTY$1,
	                    MISMATCH$1
	                );
	            }

	            return result;

	        case 'Multiplier':
	            return buildMultiplierMatchGraph(node);

	        case 'Type':
	        case 'Property':
	            return {
	                type: node.type,
	                name: node.name,
	                syntax: node
	            };

	        case 'Keyword':
	            return {
	                type: node.type,
	                name: node.name.toLowerCase(),
	                syntax: node
	            };

	        case 'AtKeyword':
	            return {
	                type: node.type,
	                name: '@' + node.name.toLowerCase(),
	                syntax: node
	            };

	        case 'Function':
	            return {
	                type: node.type,
	                name: node.name.toLowerCase() + '(',
	                syntax: node
	            };

	        case 'String':
	            // convert a one char length String to a Token
	            if (node.value.length === 3) {
	                return {
	                    type: 'Token',
	                    value: node.value.charAt(1),
	                    syntax: node
	                };
	            }

	            // otherwise use it as is
	            return {
	                type: node.type,
	                value: node.value.substr(1, node.value.length - 2).replace(/\\'/g, '\''),
	                syntax: node
	            };

	        case 'Token':
	            return {
	                type: node.type,
	                value: node.value,
	                syntax: node
	            };

	        case 'Comma':
	            return {
	                type: node.type,
	                syntax: node
	            };

	        default:
	            throw new Error('Unknown node type:', node.type);
	    }
	}

	var matchGraph$1 = {
	    MATCH: MATCH$1,
	    MISMATCH: MISMATCH$1,
	    DISALLOW_EMPTY: DISALLOW_EMPTY$1,
	    buildMatchGraph: function(syntaxTree, ref) {
	        if (typeof syntaxTree === 'string') {
	            syntaxTree = parse$1(syntaxTree);
	        }

	        return {
	            type: 'MatchGraph',
	            match: buildMatchGraph$1(syntaxTree),
	            syntax: ref || null,
	            source: syntaxTree
	        };
	    }
	};

	var hasOwnProperty$6 = Object.prototype.hasOwnProperty;
	var matchGraph = matchGraph$1;
	var MATCH = matchGraph.MATCH;
	var MISMATCH = matchGraph.MISMATCH;
	var DISALLOW_EMPTY = matchGraph.DISALLOW_EMPTY;
	var TYPE$B = _const.TYPE;

	var STUB = 0;
	var TOKEN = 1;
	var OPEN_SYNTAX = 2;
	var CLOSE_SYNTAX = 3;

	var EXIT_REASON_MATCH = 'Match';
	var EXIT_REASON_MISMATCH = 'Mismatch';
	var EXIT_REASON_ITERATION_LIMIT = 'Maximum iteration number exceeded (please fill an issue on https://github.com/csstree/csstree/issues)';

	var ITERATION_LIMIT = 15000;
	var totalIterationCount = 0;

	function reverseList(list) {
	    var prev = null;
	    var next = null;
	    var item = list;

	    while (item !== null) {
	        next = item.prev;
	        item.prev = prev;
	        prev = item;
	        item = next;
	    }

	    return prev;
	}

	function areStringsEqualCaseInsensitive(testStr, referenceStr) {
	    if (testStr.length !== referenceStr.length) {
	        return false;
	    }

	    for (var i = 0; i < testStr.length; i++) {
	        var testCode = testStr.charCodeAt(i);
	        var referenceCode = referenceStr.charCodeAt(i);

	        // testCode.toLowerCase() for U+0041 LATIN CAPITAL LETTER A (A) .. U+005A LATIN CAPITAL LETTER Z (Z).
	        if (testCode >= 0x0041 && testCode <= 0x005A) {
	            testCode = testCode | 32;
	        }

	        if (testCode !== referenceCode) {
	            return false;
	        }
	    }

	    return true;
	}

	function isContextEdgeDelim(token) {
	    if (token.type !== TYPE$B.Delim) {
	        return false;
	    }

	    // Fix matching for unicode-range: U+30??, U+FF00-FF9F
	    // Probably we need to check out previous match instead
	    return token.value !== '?';
	}

	function isCommaContextStart(token) {
	    if (token === null) {
	        return true;
	    }

	    return (
	        token.type === TYPE$B.Comma ||
	        token.type === TYPE$B.Function ||
	        token.type === TYPE$B.LeftParenthesis ||
	        token.type === TYPE$B.LeftSquareBracket ||
	        token.type === TYPE$B.LeftCurlyBracket ||
	        isContextEdgeDelim(token)
	    );
	}

	function isCommaContextEnd(token) {
	    if (token === null) {
	        return true;
	    }

	    return (
	        token.type === TYPE$B.RightParenthesis ||
	        token.type === TYPE$B.RightSquareBracket ||
	        token.type === TYPE$B.RightCurlyBracket ||
	        token.type === TYPE$B.Delim
	    );
	}

	function internalMatch(tokens, state, syntaxes) {
	    function moveToNextToken() {
	        do {
	            tokenIndex++;
	            token = tokenIndex < tokens.length ? tokens[tokenIndex] : null;
	        } while (token !== null && (token.type === TYPE$B.WhiteSpace || token.type === TYPE$B.Comment));
	    }

	    function getNextToken(offset) {
	        var nextIndex = tokenIndex + offset;

	        return nextIndex < tokens.length ? tokens[nextIndex] : null;
	    }

	    function stateSnapshotFromSyntax(nextState, prev) {
	        return {
	            nextState: nextState,
	            matchStack: matchStack,
	            syntaxStack: syntaxStack,
	            thenStack: thenStack,
	            tokenIndex: tokenIndex,
	            prev: prev
	        };
	    }

	    function pushThenStack(nextState) {
	        thenStack = {
	            nextState: nextState,
	            matchStack: matchStack,
	            syntaxStack: syntaxStack,
	            prev: thenStack
	        };
	    }

	    function pushElseStack(nextState) {
	        elseStack = stateSnapshotFromSyntax(nextState, elseStack);
	    }

	    function addTokenToMatch() {
	        matchStack = {
	            type: TOKEN,
	            syntax: state.syntax,
	            token: token,
	            prev: matchStack
	        };

	        moveToNextToken();
	        syntaxStash = null;

	        if (tokenIndex > longestMatch) {
	            longestMatch = tokenIndex;
	        }
	    }

	    function openSyntax() {
	        syntaxStack = {
	            syntax: state.syntax,
	            opts: state.syntax.opts || (syntaxStack !== null && syntaxStack.opts) || null,
	            prev: syntaxStack
	        };

	        matchStack = {
	            type: OPEN_SYNTAX,
	            syntax: state.syntax,
	            token: matchStack.token,
	            prev: matchStack
	        };
	    }

	    function closeSyntax() {
	        if (matchStack.type === OPEN_SYNTAX) {
	            matchStack = matchStack.prev;
	        } else {
	            matchStack = {
	                type: CLOSE_SYNTAX,
	                syntax: syntaxStack.syntax,
	                token: matchStack.token,
	                prev: matchStack
	            };
	        }

	        syntaxStack = syntaxStack.prev;
	    }

	    var syntaxStack = null;
	    var thenStack = null;
	    var elseStack = null;

	    // null – stashing allowed, nothing stashed
	    // false – stashing disabled, nothing stashed
	    // anithing else – fail stashable syntaxes, some syntax stashed
	    var syntaxStash = null;

	    var iterationCount = 0; // count iterations and prevent infinite loop
	    var exitReason = null;

	    var token = null;
	    var tokenIndex = -1;
	    var longestMatch = 0;
	    var matchStack = {
	        type: STUB,
	        syntax: null,
	        token: null,
	        prev: null
	    };

	    moveToNextToken();

	    while (exitReason === null && ++iterationCount < ITERATION_LIMIT) {
	        // function mapList(list, fn) {
	        //     var result = [];
	        //     while (list) {
	        //         result.unshift(fn(list));
	        //         list = list.prev;
	        //     }
	        //     return result;
	        // }
	        // console.log('--\n',
	        //     '#' + iterationCount,
	        //     require('util').inspect({
	        //         match: mapList(matchStack, x => x.type === TOKEN ? x.token && x.token.value : x.syntax ? ({ [OPEN_SYNTAX]: '<', [CLOSE_SYNTAX]: '</' }[x.type] || x.type) + '!' + x.syntax.name : null),
	        //         token: token && token.value,
	        //         tokenIndex,
	        //         syntax: syntax.type + (syntax.id ? ' #' + syntax.id : '')
	        //     }, { depth: null })
	        // );
	        switch (state.type) {
	            case 'Match':
	                if (thenStack === null) {
	                    // turn to MISMATCH when some tokens left unmatched
	                    if (token !== null) {
	                        // doesn't mismatch if just one token left and it's an IE hack
	                        if (tokenIndex !== tokens.length - 1 || (token.value !== '\\0' && token.value !== '\\9')) {
	                            state = MISMATCH;
	                            break;
	                        }
	                    }

	                    // break the main loop, return a result - MATCH
	                    exitReason = EXIT_REASON_MATCH;
	                    break;
	                }

	                // go to next syntax (`then` branch)
	                state = thenStack.nextState;

	                // check match is not empty
	                if (state === DISALLOW_EMPTY) {
	                    if (thenStack.matchStack === matchStack) {
	                        state = MISMATCH;
	                        break;
	                    } else {
	                        state = MATCH;
	                    }
	                }

	                // close syntax if needed
	                while (thenStack.syntaxStack !== syntaxStack) {
	                    closeSyntax();
	                }

	                // pop stack
	                thenStack = thenStack.prev;
	                break;

	            case 'Mismatch':
	                // when some syntax is stashed
	                if (syntaxStash !== null && syntaxStash !== false) {
	                    // there is no else branches or a branch reduce match stack
	                    if (elseStack === null || tokenIndex > elseStack.tokenIndex) {
	                        // restore state from the stash
	                        elseStack = syntaxStash;
	                        syntaxStash = false; // disable stashing
	                    }
	                } else if (elseStack === null) {
	                    // no else branches -> break the main loop
	                    // return a result - MISMATCH
	                    exitReason = EXIT_REASON_MISMATCH;
	                    break;
	                }

	                // go to next syntax (`else` branch)
	                state = elseStack.nextState;

	                // restore all the rest stack states
	                thenStack = elseStack.thenStack;
	                syntaxStack = elseStack.syntaxStack;
	                matchStack = elseStack.matchStack;
	                tokenIndex = elseStack.tokenIndex;
	                token = tokenIndex < tokens.length ? tokens[tokenIndex] : null;

	                // pop stack
	                elseStack = elseStack.prev;
	                break;

	            case 'MatchGraph':
	                state = state.match;
	                break;

	            case 'If':
	                // IMPORTANT: else stack push must go first,
	                // since it stores the state of thenStack before changes
	                if (state.else !== MISMATCH) {
	                    pushElseStack(state.else);
	                }

	                if (state.then !== MATCH) {
	                    pushThenStack(state.then);
	                }

	                state = state.match;
	                break;

	            case 'MatchOnce':
	                state = {
	                    type: 'MatchOnceBuffer',
	                    syntax: state,
	                    index: 0,
	                    mask: 0
	                };
	                break;

	            case 'MatchOnceBuffer':
	                var terms = state.syntax.terms;

	                if (state.index === terms.length) {
	                    // no matches at all or it's required all terms to be matched
	                    if (state.mask === 0 || state.syntax.all) {
	                        state = MISMATCH;
	                        break;
	                    }

	                    // a partial match is ok
	                    state = MATCH;
	                    break;
	                }

	                // all terms are matched
	                if (state.mask === (1 << terms.length) - 1) {
	                    state = MATCH;
	                    break;
	                }

	                for (; state.index < terms.length; state.index++) {
	                    var matchFlag = 1 << state.index;

	                    if ((state.mask & matchFlag) === 0) {
	                        // IMPORTANT: else stack push must go first,
	                        // since it stores the state of thenStack before changes
	                        pushElseStack(state);
	                        pushThenStack({
	                            type: 'AddMatchOnce',
	                            syntax: state.syntax,
	                            mask: state.mask | matchFlag
	                        });

	                        // match
	                        state = terms[state.index++];
	                        break;
	                    }
	                }
	                break;

	            case 'AddMatchOnce':
	                state = {
	                    type: 'MatchOnceBuffer',
	                    syntax: state.syntax,
	                    index: 0,
	                    mask: state.mask
	                };
	                break;

	            case 'Enum':
	                if (token !== null) {
	                    var name = token.value.toLowerCase();

	                    // drop \0 and \9 hack from keyword name
	                    if (name.indexOf('\\') !== -1) {
	                        name = name.replace(/\\[09].*$/, '');
	                    }

	                    if (hasOwnProperty$6.call(state.map, name)) {
	                        state = state.map[name];
	                        break;
	                    }
	                }

	                state = MISMATCH;
	                break;

	            case 'Generic':
	                var opts = syntaxStack !== null ? syntaxStack.opts : null;
	                var lastTokenIndex = tokenIndex + Math.floor(state.fn(token, getNextToken, opts));

	                if (!isNaN(lastTokenIndex) && lastTokenIndex > tokenIndex) {
	                    while (tokenIndex < lastTokenIndex) {
	                        addTokenToMatch();
	                    }

	                    state = MATCH;
	                } else {
	                    state = MISMATCH;
	                }

	                break;

	            case 'Type':
	            case 'Property':
	                var syntaxDict = state.type === 'Type' ? 'types' : 'properties';
	                var dictSyntax = hasOwnProperty$6.call(syntaxes, syntaxDict) ? syntaxes[syntaxDict][state.name] : null;

	                if (!dictSyntax || !dictSyntax.match) {
	                    throw new Error(
	                        'Bad syntax reference: ' +
	                        (state.type === 'Type'
	                            ? '<' + state.name + '>'
	                            : '<\'' + state.name + '\'>')
	                    );
	                }

	                // stash a syntax for types with low priority
	                if (syntaxStash !== false && token !== null && state.type === 'Type') {
	                    var lowPriorityMatching =
	                        // https://drafts.csswg.org/css-values-4/#custom-idents
	                        // When parsing positionally-ambiguous keywords in a property value, a <custom-ident> production
	                        // can only claim the keyword if no other unfulfilled production can claim it.
	                        (state.name === 'custom-ident' && token.type === TYPE$B.Ident) ||

	                        // https://drafts.csswg.org/css-values-4/#lengths
	                        // ... if a `0` could be parsed as either a <number> or a <length> in a property (such as line-height),
	                        // it must parse as a <number>
	                        (state.name === 'length' && token.value === '0');

	                    if (lowPriorityMatching) {
	                        if (syntaxStash === null) {
	                            syntaxStash = stateSnapshotFromSyntax(state, elseStack);
	                        }

	                        state = MISMATCH;
	                        break;
	                    }
	                }

	                openSyntax();
	                state = dictSyntax.match;
	                break;

	            case 'Keyword':
	                var name = state.name;

	                if (token !== null) {
	                    var keywordName = token.value;

	                    // drop \0 and \9 hack from keyword name
	                    if (keywordName.indexOf('\\') !== -1) {
	                        keywordName = keywordName.replace(/\\[09].*$/, '');
	                    }

	                    if (areStringsEqualCaseInsensitive(keywordName, name)) {
	                        addTokenToMatch();
	                        state = MATCH;
	                        break;
	                    }
	                }

	                state = MISMATCH;
	                break;

	            case 'AtKeyword':
	            case 'Function':
	                if (token !== null && areStringsEqualCaseInsensitive(token.value, state.name)) {
	                    addTokenToMatch();
	                    state = MATCH;
	                    break;
	                }

	                state = MISMATCH;
	                break;

	            case 'Token':
	                if (token !== null && token.value === state.value) {
	                    addTokenToMatch();
	                    state = MATCH;
	                    break;
	                }

	                state = MISMATCH;
	                break;

	            case 'Comma':
	                if (token !== null && token.type === TYPE$B.Comma) {
	                    if (isCommaContextStart(matchStack.token)) {
	                        state = MISMATCH;
	                    } else {
	                        addTokenToMatch();
	                        state = isCommaContextEnd(token) ? MISMATCH : MATCH;
	                    }
	                } else {
	                    state = isCommaContextStart(matchStack.token) || isCommaContextEnd(token) ? MATCH : MISMATCH;
	                }

	                break;

	            case 'String':
	                var string = '';

	                for (var lastTokenIndex = tokenIndex; lastTokenIndex < tokens.length && string.length < state.value.length; lastTokenIndex++) {
	                    string += tokens[lastTokenIndex].value;
	                }

	                if (areStringsEqualCaseInsensitive(string, state.value)) {
	                    while (tokenIndex < lastTokenIndex) {
	                        addTokenToMatch();
	                    }

	                    state = MATCH;
	                } else {
	                    state = MISMATCH;
	                }

	                break;

	            default:
	                throw new Error('Unknown node type: ' + state.type);
	        }
	    }

	    totalIterationCount += iterationCount;

	    switch (exitReason) {
	        case null:
	            console.warn('[csstree-match] BREAK after ' + ITERATION_LIMIT + ' iterations');
	            exitReason = EXIT_REASON_ITERATION_LIMIT;
	            matchStack = null;
	            break;

	        case EXIT_REASON_MATCH:
	            while (syntaxStack !== null) {
	                closeSyntax();
	            }
	            break;

	        default:
	            matchStack = null;
	    }

	    return {
	        tokens: tokens,
	        reason: exitReason,
	        iterations: iterationCount,
	        match: matchStack,
	        longestMatch: longestMatch
	    };
	}

	function matchAsList(tokens, matchGraph, syntaxes) {
	    var matchResult = internalMatch(tokens, matchGraph, syntaxes || {});

	    if (matchResult.match !== null) {
	        var item = reverseList(matchResult.match).prev;

	        matchResult.match = [];

	        while (item !== null) {
	            switch (item.type) {
	                case STUB:
	                    break;

	                case OPEN_SYNTAX:
	                case CLOSE_SYNTAX:
	                    matchResult.match.push({
	                        type: item.type,
	                        syntax: item.syntax
	                    });
	                    break;

	                default:
	                    matchResult.match.push({
	                        token: item.token.value,
	                        node: item.token.node
	                    });
	                    break;
	            }

	            item = item.prev;
	        }
	    }

	    return matchResult;
	}

	function matchAsTree$1(tokens, matchGraph, syntaxes) {
	    var matchResult = internalMatch(tokens, matchGraph, syntaxes || {});

	    if (matchResult.match === null) {
	        return matchResult;
	    }

	    var item = matchResult.match;
	    var host = matchResult.match = {
	        syntax: matchGraph.syntax || null,
	        match: []
	    };
	    var hostStack = [host];

	    // revert a list and start with 2nd item since 1st is a stub item
	    item = reverseList(item).prev;

	    // build a tree
	    while (item !== null) {
	        switch (item.type) {
	            case OPEN_SYNTAX:
	                host.match.push(host = {
	                    syntax: item.syntax,
	                    match: []
	                });
	                hostStack.push(host);
	                break;

	            case CLOSE_SYNTAX:
	                hostStack.pop();
	                host = hostStack[hostStack.length - 1];
	                break;

	            default:
	                host.match.push({
	                    syntax: item.syntax || null,
	                    token: item.token.value,
	                    node: item.token.node
	                });
	        }

	        item = item.prev;
	    }

	    return matchResult;
	}

	var match = {
	    matchAsList: matchAsList,
	    matchAsTree: matchAsTree$1,
	    getTotalIterationCount: function() {
	        return totalIterationCount;
	    }
	};

	function getTrace(node) {
	    function shouldPutToTrace(syntax) {
	        if (syntax === null) {
	            return false;
	        }

	        return (
	            syntax.type === 'Type' ||
	            syntax.type === 'Property' ||
	            syntax.type === 'Keyword'
	        );
	    }

	    function hasMatch(matchNode) {
	        if (Array.isArray(matchNode.match)) {
	            // use for-loop for better perfomance
	            for (var i = 0; i < matchNode.match.length; i++) {
	                if (hasMatch(matchNode.match[i])) {
	                    if (shouldPutToTrace(matchNode.syntax)) {
	                        result.unshift(matchNode.syntax);
	                    }

	                    return true;
	                }
	            }
	        } else if (matchNode.node === node) {
	            result = shouldPutToTrace(matchNode.syntax)
	                ? [matchNode.syntax]
	                : [];

	            return true;
	        }

	        return false;
	    }

	    var result = null;

	    if (this.matched !== null) {
	        hasMatch(this.matched);
	    }

	    return result;
	}

	function testNode(match, node, fn) {
	    var trace = getTrace.call(match, node);

	    if (trace === null) {
	        return false;
	    }

	    return trace.some(fn);
	}

	function isType(node, type) {
	    return testNode(this, node, function(matchNode) {
	        return matchNode.type === 'Type' && matchNode.name === type;
	    });
	}

	function isProperty(node, property) {
	    return testNode(this, node, function(matchNode) {
	        return matchNode.type === 'Property' && matchNode.name === property;
	    });
	}

	function isKeyword(node) {
	    return testNode(this, node, function(matchNode) {
	        return matchNode.type === 'Keyword';
	    });
	}

	var trace$1 = {
	    getTrace: getTrace,
	    isType: isType,
	    isProperty: isProperty,
	    isKeyword: isKeyword
	};

	var List$5 = List_1;

	function getFirstMatchNode(matchNode) {
	    if ('node' in matchNode) {
	        return matchNode.node;
	    }

	    return getFirstMatchNode(matchNode.match[0]);
	}

	function getLastMatchNode(matchNode) {
	    if ('node' in matchNode) {
	        return matchNode.node;
	    }

	    return getLastMatchNode(matchNode.match[matchNode.match.length - 1]);
	}

	function matchFragments(lexer, ast, match, type, name) {
	    function findFragments(matchNode) {
	        if (matchNode.syntax !== null &&
	            matchNode.syntax.type === type &&
	            matchNode.syntax.name === name) {
	            var start = getFirstMatchNode(matchNode);
	            var end = getLastMatchNode(matchNode);

	            lexer.syntax.walk(ast, function(node, item, list) {
	                if (node === start) {
	                    var nodes = new List$5();

	                    do {
	                        nodes.appendData(item.data);

	                        if (item.data === end) {
	                            break;
	                        }

	                        item = item.next;
	                    } while (item !== null);

	                    fragments.push({
	                        parent: list,
	                        nodes: nodes
	                    });
	                }
	            });
	        }

	        if (Array.isArray(matchNode.match)) {
	            matchNode.match.forEach(findFragments);
	        }
	    }

	    var fragments = [];

	    if (match.matched !== null) {
	        findFragments(match.matched);
	    }

	    return fragments;
	}

	var search$1 = {
	    matchFragments: matchFragments
	};

	var List$4 = List_1;
	var hasOwnProperty$5 = Object.prototype.hasOwnProperty;

	function isValidNumber(value) {
	    // Number.isInteger(value) && value >= 0
	    return (
	        typeof value === 'number' &&
	        isFinite(value) &&
	        Math.floor(value) === value &&
	        value >= 0
	    );
	}

	function isValidLocation(loc) {
	    return (
	        Boolean(loc) &&
	        isValidNumber(loc.offset) &&
	        isValidNumber(loc.line) &&
	        isValidNumber(loc.column)
	    );
	}

	function createNodeStructureChecker(type, fields) {
	    return function checkNode(node, warn) {
	        if (!node || node.constructor !== Object) {
	            return warn(node, 'Type of node should be an Object');
	        }

	        for (var key in node) {
	            var valid = true;

	            if (hasOwnProperty$5.call(node, key) === false) {
	                continue;
	            }

	            if (key === 'type') {
	                if (node.type !== type) {
	                    warn(node, 'Wrong node type `' + node.type + '`, expected `' + type + '`');
	                }
	            } else if (key === 'loc') {
	                if (node.loc === null) {
	                    continue;
	                } else if (node.loc && node.loc.constructor === Object) {
	                    if (typeof node.loc.source !== 'string') {
	                        key += '.source';
	                    } else if (!isValidLocation(node.loc.start)) {
	                        key += '.start';
	                    } else if (!isValidLocation(node.loc.end)) {
	                        key += '.end';
	                    } else {
	                        continue;
	                    }
	                }

	                valid = false;
	            } else if (fields.hasOwnProperty(key)) {
	                for (var i = 0, valid = false; !valid && i < fields[key].length; i++) {
	                    var fieldType = fields[key][i];

	                    switch (fieldType) {
	                        case String:
	                            valid = typeof node[key] === 'string';
	                            break;

	                        case Boolean:
	                            valid = typeof node[key] === 'boolean';
	                            break;

	                        case null:
	                            valid = node[key] === null;
	                            break;

	                        default:
	                            if (typeof fieldType === 'string') {
	                                valid = node[key] && node[key].type === fieldType;
	                            } else if (Array.isArray(fieldType)) {
	                                valid = node[key] instanceof List$4;
	                            }
	                    }
	                }
	            } else {
	                warn(node, 'Unknown field `' + key + '` for ' + type + ' node type');
	            }

	            if (!valid) {
	                warn(node, 'Bad value for `' + type + '.' + key + '`');
	            }
	        }

	        for (var key in fields) {
	            if (hasOwnProperty$5.call(fields, key) &&
	                hasOwnProperty$5.call(node, key) === false) {
	                warn(node, 'Field `' + type + '.' + key + '` is missed');
	            }
	        }
	    };
	}

	function processStructure(name, nodeType) {
	    var structure = nodeType.structure;
	    var fields = {
	        type: String,
	        loc: true
	    };
	    var docs = {
	        type: '"' + name + '"'
	    };

	    for (var key in structure) {
	        if (hasOwnProperty$5.call(structure, key) === false) {
	            continue;
	        }

	        var docsTypes = [];
	        var fieldTypes = fields[key] = Array.isArray(structure[key])
	            ? structure[key].slice()
	            : [structure[key]];

	        for (var i = 0; i < fieldTypes.length; i++) {
	            var fieldType = fieldTypes[i];
	            if (fieldType === String || fieldType === Boolean) {
	                docsTypes.push(fieldType.name);
	            } else if (fieldType === null) {
	                docsTypes.push('null');
	            } else if (typeof fieldType === 'string') {
	                docsTypes.push('<' + fieldType + '>');
	            } else if (Array.isArray(fieldType)) {
	                docsTypes.push('List'); // TODO: use type enum
	            } else {
	                throw new Error('Wrong value `' + fieldType + '` in `' + name + '.' + key + '` structure definition');
	            }
	        }

	        docs[key] = docsTypes.join(' | ');
	    }

	    return {
	        docs: docs,
	        check: createNodeStructureChecker(name, fields)
	    };
	}

	var structure = {
	    getStructureFromConfig: function(config) {
	        var structure = {};

	        if (config.node) {
	            for (var name in config.node) {
	                if (hasOwnProperty$5.call(config.node, name)) {
	                    var nodeType = config.node[name];

	                    if (nodeType.structure) {
	                        structure[name] = processStructure(name, nodeType);
	                    } else {
	                        throw new Error('Missed `structure` field in `' + name + '` node type definition');
	                    }
	                }
	            }
	        }

	        return structure;
	    }
	};

	var SyntaxReferenceError = error.SyntaxReferenceError;
	var SyntaxMatchError = error.SyntaxMatchError;
	var names$1 = names$2;
	var generic = generic$1;
	var parse = parse_1;
	var generate = generate_1;
	var walk = walk$1;
	var prepareTokens = prepareTokens_1;
	var buildMatchGraph = matchGraph$1.buildMatchGraph;
	var matchAsTree = match.matchAsTree;
	var trace = trace$1;
	var search = search$1;
	var getStructureFromConfig = structure.getStructureFromConfig;
	var cssWideKeywords = buildMatchGraph('inherit | initial | unset');
	var cssWideKeywordsWithExpression = buildMatchGraph('inherit | initial | unset | <-ms-legacy-expression>');

	function dumpMapSyntax(map, compact, syntaxAsAst) {
	    var result = {};

	    for (var name in map) {
	        if (map[name].syntax) {
	            result[name] = syntaxAsAst
	                ? map[name].syntax
	                : generate(map[name].syntax, { compact: compact });
	        }
	    }

	    return result;
	}

	function dumpAtruleMapSyntax(map, compact, syntaxAsAst) {
	    const result = {};

	    for (const [name, atrule] of Object.entries(map)) {
	        result[name] = {
	            prelude: atrule.prelude && (
	                syntaxAsAst
	                    ? atrule.prelude.syntax
	                    : generate(atrule.prelude.syntax, { compact })
	            ),
	            descriptors: atrule.descriptors && dumpMapSyntax(atrule.descriptors, compact, syntaxAsAst)
	        };
	    }

	    return result;
	}

	function valueHasVar(tokens) {
	    for (var i = 0; i < tokens.length; i++) {
	        if (tokens[i].value.toLowerCase() === 'var(') {
	            return true;
	        }
	    }

	    return false;
	}

	function buildMatchResult(match, error, iterations) {
	    return {
	        matched: match,
	        iterations: iterations,
	        error: error,
	        getTrace: trace.getTrace,
	        isType: trace.isType,
	        isProperty: trace.isProperty,
	        isKeyword: trace.isKeyword
	    };
	}

	function matchSyntax(lexer, syntax, value, useCommon) {
	    var tokens = prepareTokens(value, lexer.syntax);
	    var result;

	    if (valueHasVar(tokens)) {
	        return buildMatchResult(null, new Error('Matching for a tree with var() is not supported'));
	    }

	    if (useCommon) {
	        result = matchAsTree(tokens, lexer.valueCommonSyntax, lexer);
	    }

	    if (!useCommon || !result.match) {
	        result = matchAsTree(tokens, syntax.match, lexer);
	        if (!result.match) {
	            return buildMatchResult(
	                null,
	                new SyntaxMatchError(result.reason, syntax.syntax, value, result),
	                result.iterations
	            );
	        }
	    }

	    return buildMatchResult(result.match, null, result.iterations);
	}

	var Lexer$1 = function(config, syntax, structure) {
	    this.valueCommonSyntax = cssWideKeywords;
	    this.syntax = syntax;
	    this.generic = false;
	    this.atrules = {};
	    this.properties = {};
	    this.types = {};
	    this.structure = structure || getStructureFromConfig(config);

	    if (config) {
	        if (config.types) {
	            for (var name in config.types) {
	                this.addType_(name, config.types[name]);
	            }
	        }

	        if (config.generic) {
	            this.generic = true;
	            for (var name in generic) {
	                this.addType_(name, generic[name]);
	            }
	        }

	        if (config.atrules) {
	            for (var name in config.atrules) {
	                this.addAtrule_(name, config.atrules[name]);
	            }
	        }

	        if (config.properties) {
	            for (var name in config.properties) {
	                this.addProperty_(name, config.properties[name]);
	            }
	        }
	    }
	};

	Lexer$1.prototype = {
	    structure: {},
	    checkStructure: function(ast) {
	        function collectWarning(node, message) {
	            warns.push({
	                node: node,
	                message: message
	            });
	        }

	        var structure = this.structure;
	        var warns = [];

	        this.syntax.walk(ast, function(node) {
	            if (structure.hasOwnProperty(node.type)) {
	                structure[node.type].check(node, collectWarning);
	            } else {
	                collectWarning(node, 'Unknown node type `' + node.type + '`');
	            }
	        });

	        return warns.length ? warns : false;
	    },

	    createDescriptor: function(syntax, type, name, parent = null) {
	        var ref = {
	            type: type,
	            name: name
	        };
	        var descriptor = {
	            type: type,
	            name: name,
	            parent: parent,
	            syntax: null,
	            match: null
	        };

	        if (typeof syntax === 'function') {
	            descriptor.match = buildMatchGraph(syntax, ref);
	        } else {
	            if (typeof syntax === 'string') {
	                // lazy parsing on first access
	                Object.defineProperty(descriptor, 'syntax', {
	                    get: function() {
	                        Object.defineProperty(descriptor, 'syntax', {
	                            value: parse(syntax)
	                        });

	                        return descriptor.syntax;
	                    }
	                });
	            } else {
	                descriptor.syntax = syntax;
	            }

	            // lazy graph build on first access
	            Object.defineProperty(descriptor, 'match', {
	                get: function() {
	                    Object.defineProperty(descriptor, 'match', {
	                        value: buildMatchGraph(descriptor.syntax, ref)
	                    });

	                    return descriptor.match;
	                }
	            });
	        }

	        return descriptor;
	    },
	    addAtrule_: function(name, syntax) {
	        if (!syntax) {
	            return;
	        }

	        this.atrules[name] = {
	            type: 'Atrule',
	            name: name,
	            prelude: syntax.prelude ? this.createDescriptor(syntax.prelude, 'AtrulePrelude', name) : null,
	            descriptors: syntax.descriptors
	                ? Object.keys(syntax.descriptors).reduce((res, descName) => {
	                    res[descName] = this.createDescriptor(syntax.descriptors[descName], 'AtruleDescriptor', descName, name);
	                    return res;
	                }, {})
	                : null
	        };
	    },
	    addProperty_: function(name, syntax) {
	        if (!syntax) {
	            return;
	        }

	        this.properties[name] = this.createDescriptor(syntax, 'Property', name);
	    },
	    addType_: function(name, syntax) {
	        if (!syntax) {
	            return;
	        }

	        this.types[name] = this.createDescriptor(syntax, 'Type', name);

	        if (syntax === generic['-ms-legacy-expression']) {
	            this.valueCommonSyntax = cssWideKeywordsWithExpression;
	        }
	    },

	    checkAtruleName: function(atruleName) {
	        if (!this.getAtrule(atruleName)) {
	            return new SyntaxReferenceError('Unknown at-rule', '@' + atruleName);
	        }
	    },
	    checkAtrulePrelude: function(atruleName, prelude) {
	        let error = this.checkAtruleName(atruleName);

	        if (error) {
	            return error;
	        }

	        var atrule = this.getAtrule(atruleName);

	        if (!atrule.prelude && prelude) {
	            return new SyntaxError('At-rule `@' + atruleName + '` should not contain a prelude');
	        }

	        if (atrule.prelude && !prelude) {
	            return new SyntaxError('At-rule `@' + atruleName + '` should contain a prelude');
	        }
	    },
	    checkAtruleDescriptorName: function(atruleName, descriptorName) {
	        let error = this.checkAtruleName(atruleName);

	        if (error) {
	            return error;
	        }

	        var atrule = this.getAtrule(atruleName);
	        var descriptor = names$1.keyword(descriptorName);

	        if (!atrule.descriptors) {
	            return new SyntaxError('At-rule `@' + atruleName + '` has no known descriptors');
	        }

	        if (!atrule.descriptors[descriptor.name] &&
	            !atrule.descriptors[descriptor.basename]) {
	            return new SyntaxReferenceError('Unknown at-rule descriptor', descriptorName);
	        }
	    },
	    checkPropertyName: function(propertyName) {
	        var property = names$1.property(propertyName);

	        // don't match syntax for a custom property
	        if (property.custom) {
	            return new Error('Lexer matching doesn\'t applicable for custom properties');
	        }

	        if (!this.getProperty(propertyName)) {
	            return new SyntaxReferenceError('Unknown property', propertyName);
	        }
	    },

	    matchAtrulePrelude: function(atruleName, prelude) {
	        var error = this.checkAtrulePrelude(atruleName, prelude);

	        if (error) {
	            return buildMatchResult(null, error);
	        }

	        if (!prelude) {
	            return buildMatchResult(null, null);
	        }

	        return matchSyntax(this, this.getAtrule(atruleName).prelude, prelude, false);
	    },
	    matchAtruleDescriptor: function(atruleName, descriptorName, value) {
	        var error = this.checkAtruleDescriptorName(atruleName, descriptorName);

	        if (error) {
	            return buildMatchResult(null, error);
	        }

	        var atrule = this.getAtrule(atruleName);
	        var descriptor = names$1.keyword(descriptorName);

	        return matchSyntax(this, atrule.descriptors[descriptor.name] || atrule.descriptors[descriptor.basename], value, false);
	    },
	    matchDeclaration: function(node) {
	        if (node.type !== 'Declaration') {
	            return buildMatchResult(null, new Error('Not a Declaration node'));
	        }

	        return this.matchProperty(node.property, node.value);
	    },
	    matchProperty: function(propertyName, value) {
	        var error = this.checkPropertyName(propertyName);

	        if (error) {
	            return buildMatchResult(null, error);
	        }

	        return matchSyntax(this, this.getProperty(propertyName), value, true);
	    },
	    matchType: function(typeName, value) {
	        var typeSyntax = this.getType(typeName);

	        if (!typeSyntax) {
	            return buildMatchResult(null, new SyntaxReferenceError('Unknown type', typeName));
	        }

	        return matchSyntax(this, typeSyntax, value, false);
	    },
	    match: function(syntax, value) {
	        if (typeof syntax !== 'string' && (!syntax || !syntax.type)) {
	            return buildMatchResult(null, new SyntaxReferenceError('Bad syntax'));
	        }

	        if (typeof syntax === 'string' || !syntax.match) {
	            syntax = this.createDescriptor(syntax, 'Type', 'anonymous');
	        }

	        return matchSyntax(this, syntax, value, false);
	    },

	    findValueFragments: function(propertyName, value, type, name) {
	        return search.matchFragments(this, value, this.matchProperty(propertyName, value), type, name);
	    },
	    findDeclarationValueFragments: function(declaration, type, name) {
	        return search.matchFragments(this, declaration.value, this.matchDeclaration(declaration), type, name);
	    },
	    findAllFragments: function(ast, type, name) {
	        var result = [];

	        this.syntax.walk(ast, {
	            visit: 'Declaration',
	            enter: function(declaration) {
	                result.push.apply(result, this.findDeclarationValueFragments(declaration, type, name));
	            }.bind(this)
	        });

	        return result;
	    },

	    getAtrule: function(atruleName, fallbackBasename = true) {
	        var atrule = names$1.keyword(atruleName);
	        var atruleEntry = atrule.vendor && fallbackBasename
	            ? this.atrules[atrule.name] || this.atrules[atrule.basename]
	            : this.atrules[atrule.name];

	        return atruleEntry || null;
	    },
	    getAtrulePrelude: function(atruleName, fallbackBasename = true) {
	        const atrule = this.getAtrule(atruleName, fallbackBasename);

	        return atrule && atrule.prelude || null;
	    },
	    getAtruleDescriptor: function(atruleName, name) {
	        return this.atrules.hasOwnProperty(atruleName) && this.atrules.declarators
	            ? this.atrules[atruleName].declarators[name] || null
	            : null;
	    },
	    getProperty: function(propertyName, fallbackBasename = true) {
	        var property = names$1.property(propertyName);
	        var propertyEntry = property.vendor && fallbackBasename
	            ? this.properties[property.name] || this.properties[property.basename]
	            : this.properties[property.name];

	        return propertyEntry || null;
	    },
	    getType: function(name) {
	        return this.types.hasOwnProperty(name) ? this.types[name] : null;
	    },

	    validate: function() {
	        function validate(syntax, name, broken, descriptor) {
	            if (broken.hasOwnProperty(name)) {
	                return broken[name];
	            }

	            broken[name] = false;
	            if (descriptor.syntax !== null) {
	                walk(descriptor.syntax, function(node) {
	                    if (node.type !== 'Type' && node.type !== 'Property') {
	                        return;
	                    }

	                    var map = node.type === 'Type' ? syntax.types : syntax.properties;
	                    var brokenMap = node.type === 'Type' ? brokenTypes : brokenProperties;

	                    if (!map.hasOwnProperty(node.name) || validate(syntax, node.name, brokenMap, map[node.name])) {
	                        broken[name] = true;
	                    }
	                }, this);
	            }
	        }

	        var brokenTypes = {};
	        var brokenProperties = {};

	        for (var key in this.types) {
	            validate(this, key, brokenTypes, this.types[key]);
	        }

	        for (var key in this.properties) {
	            validate(this, key, brokenProperties, this.properties[key]);
	        }

	        brokenTypes = Object.keys(brokenTypes).filter(function(name) {
	            return brokenTypes[name];
	        });
	        brokenProperties = Object.keys(brokenProperties).filter(function(name) {
	            return brokenProperties[name];
	        });

	        if (brokenTypes.length || brokenProperties.length) {
	            return {
	                types: brokenTypes,
	                properties: brokenProperties
	            };
	        }

	        return null;
	    },
	    dump: function(syntaxAsAst, pretty) {
	        return {
	            generic: this.generic,
	            types: dumpMapSyntax(this.types, !pretty, syntaxAsAst),
	            properties: dumpMapSyntax(this.properties, !pretty, syntaxAsAst),
	            atrules: dumpAtruleMapSyntax(this.atrules, !pretty, syntaxAsAst)
	        };
	    },
	    toString: function() {
	        return JSON.stringify(this.dump());
	    }
	};

	var Lexer_1 = Lexer$1;

	var definitionSyntax$1 = {
	    SyntaxError: _SyntaxError,
	    parse: parse_1,
	    generate: generate_1,
	    walk: walk$1
	};

	var adoptBuffer = adoptBuffer$2;
	var isBOM = tokenizer$3.isBOM;

	var N$1 = 10;
	var F = 12;
	var R = 13;

	function computeLinesAndColumns(host, source) {
	    var sourceLength = source.length;
	    var lines = adoptBuffer(host.lines, sourceLength); // +1
	    var line = host.startLine;
	    var columns = adoptBuffer(host.columns, sourceLength);
	    var column = host.startColumn;
	    var startOffset = source.length > 0 ? isBOM(source.charCodeAt(0)) : 0;

	    for (var i = startOffset; i < sourceLength; i++) { // -1
	        var code = source.charCodeAt(i);

	        lines[i] = line;
	        columns[i] = column++;

	        if (code === N$1 || code === R || code === F) {
	            if (code === R && i + 1 < sourceLength && source.charCodeAt(i + 1) === N$1) {
	                i++;
	                lines[i] = line;
	                columns[i] = column;
	            }

	            line++;
	            column = 1;
	        }
	    }

	    lines[i] = line;
	    columns[i] = column;

	    host.lines = lines;
	    host.columns = columns;
	}

	var OffsetToLocation$1 = function() {
	    this.lines = null;
	    this.columns = null;
	    this.linesAndColumnsComputed = false;
	};

	OffsetToLocation$1.prototype = {
	    setSource: function(source, startOffset, startLine, startColumn) {
	        this.source = source;
	        this.startOffset = typeof startOffset === 'undefined' ? 0 : startOffset;
	        this.startLine = typeof startLine === 'undefined' ? 1 : startLine;
	        this.startColumn = typeof startColumn === 'undefined' ? 1 : startColumn;
	        this.linesAndColumnsComputed = false;
	    },

	    ensureLinesAndColumnsComputed: function() {
	        if (!this.linesAndColumnsComputed) {
	            computeLinesAndColumns(this, this.source);
	            this.linesAndColumnsComputed = true;
	        }
	    },
	    getLocation: function(offset, filename) {
	        this.ensureLinesAndColumnsComputed();

	        return {
	            source: filename,
	            offset: this.startOffset + offset,
	            line: this.lines[offset],
	            column: this.columns[offset]
	        };
	    },
	    getLocationRange: function(start, end, filename) {
	        this.ensureLinesAndColumnsComputed();

	        return {
	            source: filename,
	            start: {
	                offset: this.startOffset + start,
	                line: this.lines[start],
	                column: this.columns[start]
	            },
	            end: {
	                offset: this.startOffset + end,
	                line: this.lines[end],
	                column: this.columns[end]
	            }
	        };
	    }
	};

	var OffsetToLocation_1 = OffsetToLocation$1;

	var TYPE$A = tokenizer$3.TYPE;
	var WHITESPACE$a = TYPE$A.WhiteSpace;
	var COMMENT$8 = TYPE$A.Comment;

	var sequence$1 = function readSequence(recognizer) {
	    var children = this.createList();
	    var child = null;
	    var context = {
	        recognizer: recognizer,
	        space: null,
	        ignoreWS: false,
	        ignoreWSAfter: false
	    };

	    this.scanner.skipSC();

	    while (!this.scanner.eof) {
	        switch (this.scanner.tokenType) {
	            case COMMENT$8:
	                this.scanner.next();
	                continue;

	            case WHITESPACE$a:
	                if (context.ignoreWS) {
	                    this.scanner.next();
	                } else {
	                    context.space = this.WhiteSpace();
	                }
	                continue;
	        }

	        child = recognizer.getNode.call(this, context);

	        if (child === undefined) {
	            break;
	        }

	        if (context.space !== null) {
	            children.push(context.space);
	            context.space = null;
	        }

	        children.push(child);

	        if (context.ignoreWSAfter) {
	            context.ignoreWSAfter = false;
	            context.ignoreWS = true;
	        } else {
	            context.ignoreWS = false;
	        }
	    }

	    return children;
	};

	var OffsetToLocation = OffsetToLocation_1;
	var SyntaxError$2 = _SyntaxError$1;
	var TokenStream$1 = TokenStream_1;
	var List$3 = List_1;
	var tokenize$1 = tokenizer$3;
	var constants = _const;
	var { findWhiteSpaceStart, cmpStr: cmpStr$2 } = utils$2;
	var sequence = sequence$1;
	var noop$1 = function() {};

	var TYPE$z = constants.TYPE;
	var NAME$1 = constants.NAME;
	var WHITESPACE$9 = TYPE$z.WhiteSpace;
	var COMMENT$7 = TYPE$z.Comment;
	var IDENT$g = TYPE$z.Ident;
	var FUNCTION$6 = TYPE$z.Function;
	var URL$4 = TYPE$z.Url;
	var HASH$5 = TYPE$z.Hash;
	var PERCENTAGE$3 = TYPE$z.Percentage;
	var NUMBER$7 = TYPE$z.Number;
	var NUMBERSIGN$3 = 0x0023; // U+0023 NUMBER SIGN (#)
	var NULL = 0;

	function createParseContext(name) {
	    return function() {
	        return this[name]();
	    };
	}

	function processConfig(config) {
	    var parserConfig = {
	        context: {},
	        scope: {},
	        atrule: {},
	        pseudo: {}
	    };

	    if (config.parseContext) {
	        for (var name in config.parseContext) {
	            switch (typeof config.parseContext[name]) {
	                case 'function':
	                    parserConfig.context[name] = config.parseContext[name];
	                    break;

	                case 'string':
	                    parserConfig.context[name] = createParseContext(config.parseContext[name]);
	                    break;
	            }
	        }
	    }

	    if (config.scope) {
	        for (var name in config.scope) {
	            parserConfig.scope[name] = config.scope[name];
	        }
	    }

	    if (config.atrule) {
	        for (var name in config.atrule) {
	            var atrule = config.atrule[name];

	            if (atrule.parse) {
	                parserConfig.atrule[name] = atrule.parse;
	            }
	        }
	    }

	    if (config.pseudo) {
	        for (var name in config.pseudo) {
	            var pseudo = config.pseudo[name];

	            if (pseudo.parse) {
	                parserConfig.pseudo[name] = pseudo.parse;
	            }
	        }
	    }

	    if (config.node) {
	        for (var name in config.node) {
	            parserConfig[name] = config.node[name].parse;
	        }
	    }

	    return parserConfig;
	}

	var create$3 = function createParser(config) {
	    var parser = {
	        scanner: new TokenStream$1(),
	        locationMap: new OffsetToLocation(),

	        filename: '<unknown>',
	        needPositions: false,
	        onParseError: noop$1,
	        onParseErrorThrow: false,
	        parseAtrulePrelude: true,
	        parseRulePrelude: true,
	        parseValue: true,
	        parseCustomProperty: false,

	        readSequence: sequence,

	        createList: function() {
	            return new List$3();
	        },
	        createSingleNodeList: function(node) {
	            return new List$3().appendData(node);
	        },
	        getFirstListNode: function(list) {
	            return list && list.first();
	        },
	        getLastListNode: function(list) {
	            return list.last();
	        },

	        parseWithFallback: function(consumer, fallback) {
	            var startToken = this.scanner.tokenIndex;

	            try {
	                return consumer.call(this);
	            } catch (e) {
	                if (this.onParseErrorThrow) {
	                    throw e;
	                }

	                var fallbackNode = fallback.call(this, startToken);

	                this.onParseErrorThrow = true;
	                this.onParseError(e, fallbackNode);
	                this.onParseErrorThrow = false;

	                return fallbackNode;
	            }
	        },

	        lookupNonWSType: function(offset) {
	            do {
	                var type = this.scanner.lookupType(offset++);
	                if (type !== WHITESPACE$9) {
	                    return type;
	                }
	            } while (type !== NULL);

	            return NULL;
	        },

	        eat: function(tokenType) {
	            if (this.scanner.tokenType !== tokenType) {
	                var offset = this.scanner.tokenStart;
	                var message = NAME$1[tokenType] + ' is expected';

	                // tweak message and offset
	                switch (tokenType) {
	                    case IDENT$g:
	                        // when identifier is expected but there is a function or url
	                        if (this.scanner.tokenType === FUNCTION$6 || this.scanner.tokenType === URL$4) {
	                            offset = this.scanner.tokenEnd - 1;
	                            message = 'Identifier is expected but function found';
	                        } else {
	                            message = 'Identifier is expected';
	                        }
	                        break;

	                    case HASH$5:
	                        if (this.scanner.isDelim(NUMBERSIGN$3)) {
	                            this.scanner.next();
	                            offset++;
	                            message = 'Name is expected';
	                        }
	                        break;

	                    case PERCENTAGE$3:
	                        if (this.scanner.tokenType === NUMBER$7) {
	                            offset = this.scanner.tokenEnd;
	                            message = 'Percent sign is expected';
	                        }
	                        break;

	                    default:
	                        // when test type is part of another token show error for current position + 1
	                        // e.g. eat(HYPHENMINUS) will fail on "-foo", but pointing on "-" is odd
	                        if (this.scanner.source.charCodeAt(this.scanner.tokenStart) === tokenType) {
	                            offset = offset + 1;
	                        }
	                }

	                this.error(message, offset);
	            }

	            this.scanner.next();
	        },

	        consume: function(tokenType) {
	            var value = this.scanner.getTokenValue();

	            this.eat(tokenType);

	            return value;
	        },
	        consumeFunctionName: function() {
	            var name = this.scanner.source.substring(this.scanner.tokenStart, this.scanner.tokenEnd - 1);

	            this.eat(FUNCTION$6);

	            return name;
	        },

	        getLocation: function(start, end) {
	            if (this.needPositions) {
	                return this.locationMap.getLocationRange(
	                    start,
	                    end,
	                    this.filename
	                );
	            }

	            return null;
	        },
	        getLocationFromList: function(list) {
	            if (this.needPositions) {
	                var head = this.getFirstListNode(list);
	                var tail = this.getLastListNode(list);
	                return this.locationMap.getLocationRange(
	                    head !== null ? head.loc.start.offset - this.locationMap.startOffset : this.scanner.tokenStart,
	                    tail !== null ? tail.loc.end.offset - this.locationMap.startOffset : this.scanner.tokenStart,
	                    this.filename
	                );
	            }

	            return null;
	        },

	        error: function(message, offset) {
	            var location = typeof offset !== 'undefined' && offset < this.scanner.source.length
	                ? this.locationMap.getLocation(offset)
	                : this.scanner.eof
	                    ? this.locationMap.getLocation(findWhiteSpaceStart(this.scanner.source, this.scanner.source.length - 1))
	                    : this.locationMap.getLocation(this.scanner.tokenStart);

	            throw new SyntaxError$2(
	                message || 'Unexpected input',
	                this.scanner.source,
	                location.offset,
	                location.line,
	                location.column
	            );
	        }
	    };

	    config = processConfig(config || {});
	    for (var key in config) {
	        parser[key] = config[key];
	    }

	    return function(source, options) {
	        options = options || {};

	        var context = options.context || 'default';
	        var onComment = options.onComment;
	        var ast;

	        tokenize$1(source, parser.scanner);
	        parser.locationMap.setSource(
	            source,
	            options.offset,
	            options.line,
	            options.column
	        );

	        parser.filename = options.filename || '<unknown>';
	        parser.needPositions = Boolean(options.positions);
	        parser.onParseError = typeof options.onParseError === 'function' ? options.onParseError : noop$1;
	        parser.onParseErrorThrow = false;
	        parser.parseAtrulePrelude = 'parseAtrulePrelude' in options ? Boolean(options.parseAtrulePrelude) : true;
	        parser.parseRulePrelude = 'parseRulePrelude' in options ? Boolean(options.parseRulePrelude) : true;
	        parser.parseValue = 'parseValue' in options ? Boolean(options.parseValue) : true;
	        parser.parseCustomProperty = 'parseCustomProperty' in options ? Boolean(options.parseCustomProperty) : false;

	        if (!parser.context.hasOwnProperty(context)) {
	            throw new Error('Unknown context `' + context + '`');
	        }

	        if (typeof onComment === 'function') {
	            parser.scanner.forEachToken((type, start, end) => {
	                if (type === COMMENT$7) {
	                    const loc = parser.getLocation(start, end);
	                    const value = cmpStr$2(source, end - 2, end, '*/')
	                        ? source.slice(start + 2, end - 2)
	                        : source.slice(start + 2, end);

	                    onComment(value, loc);
	                }
	            });
	        }

	        ast = parser.context[context].call(parser, options);

	        if (!parser.scanner.eof) {
	            parser.error();
	        }

	        return ast;
	    };
	};

	var sourceMapGenerator = {};

	var base64Vlq = {};

	var base64$1 = {};

	/* -*- Mode: js; js-indent-level: 2; -*- */

	/*
	 * Copyright 2011 Mozilla Foundation and contributors
	 * Licensed under the New BSD license. See LICENSE or:
	 * http://opensource.org/licenses/BSD-3-Clause
	 */

	var intToCharMap = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/'.split('');

	/**
	 * Encode an integer in the range of 0 to 63 to a single base 64 digit.
	 */
	base64$1.encode = function (number) {
	  if (0 <= number && number < intToCharMap.length) {
	    return intToCharMap[number];
	  }
	  throw new TypeError("Must be between 0 and 63: " + number);
	};

	/**
	 * Decode a single base 64 character code digit to an integer. Returns -1 on
	 * failure.
	 */
	base64$1.decode = function (charCode) {
	  var bigA = 65;     // 'A'
	  var bigZ = 90;     // 'Z'

	  var littleA = 97;  // 'a'
	  var littleZ = 122; // 'z'

	  var zero = 48;     // '0'
	  var nine = 57;     // '9'

	  var plus = 43;     // '+'
	  var slash = 47;    // '/'

	  var littleOffset = 26;
	  var numberOffset = 52;

	  // 0 - 25: ABCDEFGHIJKLMNOPQRSTUVWXYZ
	  if (bigA <= charCode && charCode <= bigZ) {
	    return (charCode - bigA);
	  }

	  // 26 - 51: abcdefghijklmnopqrstuvwxyz
	  if (littleA <= charCode && charCode <= littleZ) {
	    return (charCode - littleA + littleOffset);
	  }

	  // 52 - 61: 0123456789
	  if (zero <= charCode && charCode <= nine) {
	    return (charCode - zero + numberOffset);
	  }

	  // 62: +
	  if (charCode == plus) {
	    return 62;
	  }

	  // 63: /
	  if (charCode == slash) {
	    return 63;
	  }

	  // Invalid base64 digit.
	  return -1;
	};

	/* -*- Mode: js; js-indent-level: 2; -*- */

	/*
	 * Copyright 2011 Mozilla Foundation and contributors
	 * Licensed under the New BSD license. See LICENSE or:
	 * http://opensource.org/licenses/BSD-3-Clause
	 *
	 * Based on the Base 64 VLQ implementation in Closure Compiler:
	 * https://code.google.com/p/closure-compiler/source/browse/trunk/src/com/google/debugging/sourcemap/Base64VLQ.java
	 *
	 * Copyright 2011 The Closure Compiler Authors. All rights reserved.
	 * Redistribution and use in source and binary forms, with or without
	 * modification, are permitted provided that the following conditions are
	 * met:
	 *
	 *  * Redistributions of source code must retain the above copyright
	 *    notice, this list of conditions and the following disclaimer.
	 *  * Redistributions in binary form must reproduce the above
	 *    copyright notice, this list of conditions and the following
	 *    disclaimer in the documentation and/or other materials provided
	 *    with the distribution.
	 *  * Neither the name of Google Inc. nor the names of its
	 *    contributors may be used to endorse or promote products derived
	 *    from this software without specific prior written permission.
	 *
	 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
	 * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
	 * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
	 * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
	 * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
	 * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
	 * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
	 * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
	 * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
	 * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
	 * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
	 */

	var base64 = base64$1;

	// A single base 64 digit can contain 6 bits of data. For the base 64 variable
	// length quantities we use in the source map spec, the first bit is the sign,
	// the next four bits are the actual value, and the 6th bit is the
	// continuation bit. The continuation bit tells us whether there are more
	// digits in this value following this digit.
	//
	//   Continuation
	//   |    Sign
	//   |    |
	//   V    V
	//   101011

	var VLQ_BASE_SHIFT = 5;

	// binary: 100000
	var VLQ_BASE = 1 << VLQ_BASE_SHIFT;

	// binary: 011111
	var VLQ_BASE_MASK = VLQ_BASE - 1;

	// binary: 100000
	var VLQ_CONTINUATION_BIT = VLQ_BASE;

	/**
	 * Converts from a two-complement value to a value where the sign bit is
	 * placed in the least significant bit.  For example, as decimals:
	 *   1 becomes 2 (10 binary), -1 becomes 3 (11 binary)
	 *   2 becomes 4 (100 binary), -2 becomes 5 (101 binary)
	 */
	function toVLQSigned(aValue) {
	  return aValue < 0
	    ? ((-aValue) << 1) + 1
	    : (aValue << 1) + 0;
	}

	/**
	 * Converts to a two-complement value from a value where the sign bit is
	 * placed in the least significant bit.  For example, as decimals:
	 *   2 (10 binary) becomes 1, 3 (11 binary) becomes -1
	 *   4 (100 binary) becomes 2, 5 (101 binary) becomes -2
	 */
	function fromVLQSigned(aValue) {
	  var isNegative = (aValue & 1) === 1;
	  var shifted = aValue >> 1;
	  return isNegative
	    ? -shifted
	    : shifted;
	}

	/**
	 * Returns the base 64 VLQ encoded value.
	 */
	base64Vlq.encode = function base64VLQ_encode(aValue) {
	  var encoded = "";
	  var digit;

	  var vlq = toVLQSigned(aValue);

	  do {
	    digit = vlq & VLQ_BASE_MASK;
	    vlq >>>= VLQ_BASE_SHIFT;
	    if (vlq > 0) {
	      // There are still more digits in this value, so we must make sure the
	      // continuation bit is marked.
	      digit |= VLQ_CONTINUATION_BIT;
	    }
	    encoded += base64.encode(digit);
	  } while (vlq > 0);

	  return encoded;
	};

	/**
	 * Decodes the next base 64 VLQ value from the given string and returns the
	 * value and the rest of the string via the out parameter.
	 */
	base64Vlq.decode = function base64VLQ_decode(aStr, aIndex, aOutParam) {
	  var strLen = aStr.length;
	  var result = 0;
	  var shift = 0;
	  var continuation, digit;

	  do {
	    if (aIndex >= strLen) {
	      throw new Error("Expected more digits in base 64 VLQ value.");
	    }

	    digit = base64.decode(aStr.charCodeAt(aIndex++));
	    if (digit === -1) {
	      throw new Error("Invalid base64 digit: " + aStr.charAt(aIndex - 1));
	    }

	    continuation = !!(digit & VLQ_CONTINUATION_BIT);
	    digit &= VLQ_BASE_MASK;
	    result = result + (digit << shift);
	    shift += VLQ_BASE_SHIFT;
	  } while (continuation);

	  aOutParam.value = fromVLQSigned(result);
	  aOutParam.rest = aIndex;
	};

	var util$3 = {};

	/* -*- Mode: js; js-indent-level: 2; -*- */

	(function (exports) {
		/*
		 * Copyright 2011 Mozilla Foundation and contributors
		 * Licensed under the New BSD license. See LICENSE or:
		 * http://opensource.org/licenses/BSD-3-Clause
		 */

		/**
		 * This is a helper function for getting values from parameter/options
		 * objects.
		 *
		 * @param args The object we are extracting values from
		 * @param name The name of the property we are getting.
		 * @param defaultValue An optional value to return if the property is missing
		 * from the object. If this is not specified and the property is missing, an
		 * error will be thrown.
		 */
		function getArg(aArgs, aName, aDefaultValue) {
		  if (aName in aArgs) {
		    return aArgs[aName];
		  } else if (arguments.length === 3) {
		    return aDefaultValue;
		  } else {
		    throw new Error('"' + aName + '" is a required argument.');
		  }
		}
		exports.getArg = getArg;

		var urlRegexp = /^(?:([\w+\-.]+):)?\/\/(?:(\w+:\w+)@)?([\w.-]*)(?::(\d+))?(.*)$/;
		var dataUrlRegexp = /^data:.+\,.+$/;

		function urlParse(aUrl) {
		  var match = aUrl.match(urlRegexp);
		  if (!match) {
		    return null;
		  }
		  return {
		    scheme: match[1],
		    auth: match[2],
		    host: match[3],
		    port: match[4],
		    path: match[5]
		  };
		}
		exports.urlParse = urlParse;

		function urlGenerate(aParsedUrl) {
		  var url = '';
		  if (aParsedUrl.scheme) {
		    url += aParsedUrl.scheme + ':';
		  }
		  url += '//';
		  if (aParsedUrl.auth) {
		    url += aParsedUrl.auth + '@';
		  }
		  if (aParsedUrl.host) {
		    url += aParsedUrl.host;
		  }
		  if (aParsedUrl.port) {
		    url += ":" + aParsedUrl.port;
		  }
		  if (aParsedUrl.path) {
		    url += aParsedUrl.path;
		  }
		  return url;
		}
		exports.urlGenerate = urlGenerate;

		/**
		 * Normalizes a path, or the path portion of a URL:
		 *
		 * - Replaces consecutive slashes with one slash.
		 * - Removes unnecessary '.' parts.
		 * - Removes unnecessary '<dir>/..' parts.
		 *
		 * Based on code in the Node.js 'path' core module.
		 *
		 * @param aPath The path or url to normalize.
		 */
		function normalize(aPath) {
		  var path = aPath;
		  var url = urlParse(aPath);
		  if (url) {
		    if (!url.path) {
		      return aPath;
		    }
		    path = url.path;
		  }
		  var isAbsolute = exports.isAbsolute(path);

		  var parts = path.split(/\/+/);
		  for (var part, up = 0, i = parts.length - 1; i >= 0; i--) {
		    part = parts[i];
		    if (part === '.') {
		      parts.splice(i, 1);
		    } else if (part === '..') {
		      up++;
		    } else if (up > 0) {
		      if (part === '') {
		        // The first part is blank if the path is absolute. Trying to go
		        // above the root is a no-op. Therefore we can remove all '..' parts
		        // directly after the root.
		        parts.splice(i + 1, up);
		        up = 0;
		      } else {
		        parts.splice(i, 2);
		        up--;
		      }
		    }
		  }
		  path = parts.join('/');

		  if (path === '') {
		    path = isAbsolute ? '/' : '.';
		  }

		  if (url) {
		    url.path = path;
		    return urlGenerate(url);
		  }
		  return path;
		}
		exports.normalize = normalize;

		/**
		 * Joins two paths/URLs.
		 *
		 * @param aRoot The root path or URL.
		 * @param aPath The path or URL to be joined with the root.
		 *
		 * - If aPath is a URL or a data URI, aPath is returned, unless aPath is a
		 *   scheme-relative URL: Then the scheme of aRoot, if any, is prepended
		 *   first.
		 * - Otherwise aPath is a path. If aRoot is a URL, then its path portion
		 *   is updated with the result and aRoot is returned. Otherwise the result
		 *   is returned.
		 *   - If aPath is absolute, the result is aPath.
		 *   - Otherwise the two paths are joined with a slash.
		 * - Joining for example 'http://' and 'www.example.com' is also supported.
		 */
		function join(aRoot, aPath) {
		  if (aRoot === "") {
		    aRoot = ".";
		  }
		  if (aPath === "") {
		    aPath = ".";
		  }
		  var aPathUrl = urlParse(aPath);
		  var aRootUrl = urlParse(aRoot);
		  if (aRootUrl) {
		    aRoot = aRootUrl.path || '/';
		  }

		  // `join(foo, '//www.example.org')`
		  if (aPathUrl && !aPathUrl.scheme) {
		    if (aRootUrl) {
		      aPathUrl.scheme = aRootUrl.scheme;
		    }
		    return urlGenerate(aPathUrl);
		  }

		  if (aPathUrl || aPath.match(dataUrlRegexp)) {
		    return aPath;
		  }

		  // `join('http://', 'www.example.com')`
		  if (aRootUrl && !aRootUrl.host && !aRootUrl.path) {
		    aRootUrl.host = aPath;
		    return urlGenerate(aRootUrl);
		  }

		  var joined = aPath.charAt(0) === '/'
		    ? aPath
		    : normalize(aRoot.replace(/\/+$/, '') + '/' + aPath);

		  if (aRootUrl) {
		    aRootUrl.path = joined;
		    return urlGenerate(aRootUrl);
		  }
		  return joined;
		}
		exports.join = join;

		exports.isAbsolute = function (aPath) {
		  return aPath.charAt(0) === '/' || urlRegexp.test(aPath);
		};

		/**
		 * Make a path relative to a URL or another path.
		 *
		 * @param aRoot The root path or URL.
		 * @param aPath The path or URL to be made relative to aRoot.
		 */
		function relative(aRoot, aPath) {
		  if (aRoot === "") {
		    aRoot = ".";
		  }

		  aRoot = aRoot.replace(/\/$/, '');

		  // It is possible for the path to be above the root. In this case, simply
		  // checking whether the root is a prefix of the path won't work. Instead, we
		  // need to remove components from the root one by one, until either we find
		  // a prefix that fits, or we run out of components to remove.
		  var level = 0;
		  while (aPath.indexOf(aRoot + '/') !== 0) {
		    var index = aRoot.lastIndexOf("/");
		    if (index < 0) {
		      return aPath;
		    }

		    // If the only part of the root that is left is the scheme (i.e. http://,
		    // file:///, etc.), one or more slashes (/), or simply nothing at all, we
		    // have exhausted all components, so the path is not relative to the root.
		    aRoot = aRoot.slice(0, index);
		    if (aRoot.match(/^([^\/]+:\/)?\/*$/)) {
		      return aPath;
		    }

		    ++level;
		  }

		  // Make sure we add a "../" for each component we removed from the root.
		  return Array(level + 1).join("../") + aPath.substr(aRoot.length + 1);
		}
		exports.relative = relative;

		var supportsNullProto = (function () {
		  var obj = Object.create(null);
		  return !('__proto__' in obj);
		}());

		function identity (s) {
		  return s;
		}

		/**
		 * Because behavior goes wacky when you set `__proto__` on objects, we
		 * have to prefix all the strings in our set with an arbitrary character.
		 *
		 * See https://github.com/mozilla/source-map/pull/31 and
		 * https://github.com/mozilla/source-map/issues/30
		 *
		 * @param String aStr
		 */
		function toSetString(aStr) {
		  if (isProtoString(aStr)) {
		    return '$' + aStr;
		  }

		  return aStr;
		}
		exports.toSetString = supportsNullProto ? identity : toSetString;

		function fromSetString(aStr) {
		  if (isProtoString(aStr)) {
		    return aStr.slice(1);
		  }

		  return aStr;
		}
		exports.fromSetString = supportsNullProto ? identity : fromSetString;

		function isProtoString(s) {
		  if (!s) {
		    return false;
		  }

		  var length = s.length;

		  if (length < 9 /* "__proto__".length */) {
		    return false;
		  }

		  if (s.charCodeAt(length - 1) !== 95  /* '_' */ ||
		      s.charCodeAt(length - 2) !== 95  /* '_' */ ||
		      s.charCodeAt(length - 3) !== 111 /* 'o' */ ||
		      s.charCodeAt(length - 4) !== 116 /* 't' */ ||
		      s.charCodeAt(length - 5) !== 111 /* 'o' */ ||
		      s.charCodeAt(length - 6) !== 114 /* 'r' */ ||
		      s.charCodeAt(length - 7) !== 112 /* 'p' */ ||
		      s.charCodeAt(length - 8) !== 95  /* '_' */ ||
		      s.charCodeAt(length - 9) !== 95  /* '_' */) {
		    return false;
		  }

		  for (var i = length - 10; i >= 0; i--) {
		    if (s.charCodeAt(i) !== 36 /* '$' */) {
		      return false;
		    }
		  }

		  return true;
		}

		/**
		 * Comparator between two mappings where the original positions are compared.
		 *
		 * Optionally pass in `true` as `onlyCompareGenerated` to consider two
		 * mappings with the same original source/line/column, but different generated
		 * line and column the same. Useful when searching for a mapping with a
		 * stubbed out mapping.
		 */
		function compareByOriginalPositions(mappingA, mappingB, onlyCompareOriginal) {
		  var cmp = strcmp(mappingA.source, mappingB.source);
		  if (cmp !== 0) {
		    return cmp;
		  }

		  cmp = mappingA.originalLine - mappingB.originalLine;
		  if (cmp !== 0) {
		    return cmp;
		  }

		  cmp = mappingA.originalColumn - mappingB.originalColumn;
		  if (cmp !== 0 || onlyCompareOriginal) {
		    return cmp;
		  }

		  cmp = mappingA.generatedColumn - mappingB.generatedColumn;
		  if (cmp !== 0) {
		    return cmp;
		  }

		  cmp = mappingA.generatedLine - mappingB.generatedLine;
		  if (cmp !== 0) {
		    return cmp;
		  }

		  return strcmp(mappingA.name, mappingB.name);
		}
		exports.compareByOriginalPositions = compareByOriginalPositions;

		/**
		 * Comparator between two mappings with deflated source and name indices where
		 * the generated positions are compared.
		 *
		 * Optionally pass in `true` as `onlyCompareGenerated` to consider two
		 * mappings with the same generated line and column, but different
		 * source/name/original line and column the same. Useful when searching for a
		 * mapping with a stubbed out mapping.
		 */
		function compareByGeneratedPositionsDeflated(mappingA, mappingB, onlyCompareGenerated) {
		  var cmp = mappingA.generatedLine - mappingB.generatedLine;
		  if (cmp !== 0) {
		    return cmp;
		  }

		  cmp = mappingA.generatedColumn - mappingB.generatedColumn;
		  if (cmp !== 0 || onlyCompareGenerated) {
		    return cmp;
		  }

		  cmp = strcmp(mappingA.source, mappingB.source);
		  if (cmp !== 0) {
		    return cmp;
		  }

		  cmp = mappingA.originalLine - mappingB.originalLine;
		  if (cmp !== 0) {
		    return cmp;
		  }

		  cmp = mappingA.originalColumn - mappingB.originalColumn;
		  if (cmp !== 0) {
		    return cmp;
		  }

		  return strcmp(mappingA.name, mappingB.name);
		}
		exports.compareByGeneratedPositionsDeflated = compareByGeneratedPositionsDeflated;

		function strcmp(aStr1, aStr2) {
		  if (aStr1 === aStr2) {
		    return 0;
		  }

		  if (aStr1 === null) {
		    return 1; // aStr2 !== null
		  }

		  if (aStr2 === null) {
		    return -1; // aStr1 !== null
		  }

		  if (aStr1 > aStr2) {
		    return 1;
		  }

		  return -1;
		}

		/**
		 * Comparator between two mappings with inflated source and name strings where
		 * the generated positions are compared.
		 */
		function compareByGeneratedPositionsInflated(mappingA, mappingB) {
		  var cmp = mappingA.generatedLine - mappingB.generatedLine;
		  if (cmp !== 0) {
		    return cmp;
		  }

		  cmp = mappingA.generatedColumn - mappingB.generatedColumn;
		  if (cmp !== 0) {
		    return cmp;
		  }

		  cmp = strcmp(mappingA.source, mappingB.source);
		  if (cmp !== 0) {
		    return cmp;
		  }

		  cmp = mappingA.originalLine - mappingB.originalLine;
		  if (cmp !== 0) {
		    return cmp;
		  }

		  cmp = mappingA.originalColumn - mappingB.originalColumn;
		  if (cmp !== 0) {
		    return cmp;
		  }

		  return strcmp(mappingA.name, mappingB.name);
		}
		exports.compareByGeneratedPositionsInflated = compareByGeneratedPositionsInflated;

		/**
		 * Strip any JSON XSSI avoidance prefix from the string (as documented
		 * in the source maps specification), and then parse the string as
		 * JSON.
		 */
		function parseSourceMapInput(str) {
		  return JSON.parse(str.replace(/^\)]}'[^\n]*\n/, ''));
		}
		exports.parseSourceMapInput = parseSourceMapInput;

		/**
		 * Compute the URL of a source given the the source root, the source's
		 * URL, and the source map's URL.
		 */
		function computeSourceURL(sourceRoot, sourceURL, sourceMapURL) {
		  sourceURL = sourceURL || '';

		  if (sourceRoot) {
		    // This follows what Chrome does.
		    if (sourceRoot[sourceRoot.length - 1] !== '/' && sourceURL[0] !== '/') {
		      sourceRoot += '/';
		    }
		    // The spec says:
		    //   Line 4: An optional source root, useful for relocating source
		    //   files on a server or removing repeated values in the
		    //   “sources” entry.  This value is prepended to the individual
		    //   entries in the “source” field.
		    sourceURL = sourceRoot + sourceURL;
		  }

		  // Historically, SourceMapConsumer did not take the sourceMapURL as
		  // a parameter.  This mode is still somewhat supported, which is why
		  // this code block is conditional.  However, it's preferable to pass
		  // the source map URL to SourceMapConsumer, so that this function
		  // can implement the source URL resolution algorithm as outlined in
		  // the spec.  This block is basically the equivalent of:
		  //    new URL(sourceURL, sourceMapURL).toString()
		  // ... except it avoids using URL, which wasn't available in the
		  // older releases of node still supported by this library.
		  //
		  // The spec says:
		  //   If the sources are not absolute URLs after prepending of the
		  //   “sourceRoot”, the sources are resolved relative to the
		  //   SourceMap (like resolving script src in a html document).
		  if (sourceMapURL) {
		    var parsed = urlParse(sourceMapURL);
		    if (!parsed) {
		      throw new Error("sourceMapURL could not be parsed");
		    }
		    if (parsed.path) {
		      // Strip the last path component, but keep the "/".
		      var index = parsed.path.lastIndexOf('/');
		      if (index >= 0) {
		        parsed.path = parsed.path.substring(0, index + 1);
		      }
		    }
		    sourceURL = join(urlGenerate(parsed), sourceURL);
		  }

		  return normalize(sourceURL);
		}
		exports.computeSourceURL = computeSourceURL; 
	} (util$3));

	var arraySet = {};

	/* -*- Mode: js; js-indent-level: 2; -*- */

	/*
	 * Copyright 2011 Mozilla Foundation and contributors
	 * Licensed under the New BSD license. See LICENSE or:
	 * http://opensource.org/licenses/BSD-3-Clause
	 */

	var util$2 = util$3;
	var has$1 = Object.prototype.hasOwnProperty;
	var hasNativeMap = typeof Map !== "undefined";

	/**
	 * A data structure which is a combination of an array and a set. Adding a new
	 * member is O(1), testing for membership is O(1), and finding the index of an
	 * element is O(1). Removing elements from the set is not supported. Only
	 * strings are supported for membership.
	 */
	function ArraySet$1() {
	  this._array = [];
	  this._set = hasNativeMap ? new Map() : Object.create(null);
	}

	/**
	 * Static method for creating ArraySet instances from an existing array.
	 */
	ArraySet$1.fromArray = function ArraySet_fromArray(aArray, aAllowDuplicates) {
	  var set = new ArraySet$1();
	  for (var i = 0, len = aArray.length; i < len; i++) {
	    set.add(aArray[i], aAllowDuplicates);
	  }
	  return set;
	};

	/**
	 * Return how many unique items are in this ArraySet. If duplicates have been
	 * added, than those do not count towards the size.
	 *
	 * @returns Number
	 */
	ArraySet$1.prototype.size = function ArraySet_size() {
	  return hasNativeMap ? this._set.size : Object.getOwnPropertyNames(this._set).length;
	};

	/**
	 * Add the given string to this set.
	 *
	 * @param String aStr
	 */
	ArraySet$1.prototype.add = function ArraySet_add(aStr, aAllowDuplicates) {
	  var sStr = hasNativeMap ? aStr : util$2.toSetString(aStr);
	  var isDuplicate = hasNativeMap ? this.has(aStr) : has$1.call(this._set, sStr);
	  var idx = this._array.length;
	  if (!isDuplicate || aAllowDuplicates) {
	    this._array.push(aStr);
	  }
	  if (!isDuplicate) {
	    if (hasNativeMap) {
	      this._set.set(aStr, idx);
	    } else {
	      this._set[sStr] = idx;
	    }
	  }
	};

	/**
	 * Is the given string a member of this set?
	 *
	 * @param String aStr
	 */
	ArraySet$1.prototype.has = function ArraySet_has(aStr) {
	  if (hasNativeMap) {
	    return this._set.has(aStr);
	  } else {
	    var sStr = util$2.toSetString(aStr);
	    return has$1.call(this._set, sStr);
	  }
	};

	/**
	 * What is the index of the given string in the array?
	 *
	 * @param String aStr
	 */
	ArraySet$1.prototype.indexOf = function ArraySet_indexOf(aStr) {
	  if (hasNativeMap) {
	    var idx = this._set.get(aStr);
	    if (idx >= 0) {
	        return idx;
	    }
	  } else {
	    var sStr = util$2.toSetString(aStr);
	    if (has$1.call(this._set, sStr)) {
	      return this._set[sStr];
	    }
	  }

	  throw new Error('"' + aStr + '" is not in the set.');
	};

	/**
	 * What is the element at the given index?
	 *
	 * @param Number aIdx
	 */
	ArraySet$1.prototype.at = function ArraySet_at(aIdx) {
	  if (aIdx >= 0 && aIdx < this._array.length) {
	    return this._array[aIdx];
	  }
	  throw new Error('No element indexed by ' + aIdx);
	};

	/**
	 * Returns the array representation of this set (which has the proper indices
	 * indicated by indexOf). Note that this is a copy of the internal array used
	 * for storing the members so that no one can mess with internal state.
	 */
	ArraySet$1.prototype.toArray = function ArraySet_toArray() {
	  return this._array.slice();
	};

	arraySet.ArraySet = ArraySet$1;

	var mappingList = {};

	/* -*- Mode: js; js-indent-level: 2; -*- */

	/*
	 * Copyright 2014 Mozilla Foundation and contributors
	 * Licensed under the New BSD license. See LICENSE or:
	 * http://opensource.org/licenses/BSD-3-Clause
	 */

	var util$1 = util$3;

	/**
	 * Determine whether mappingB is after mappingA with respect to generated
	 * position.
	 */
	function generatedPositionAfter(mappingA, mappingB) {
	  // Optimized for most common case
	  var lineA = mappingA.generatedLine;
	  var lineB = mappingB.generatedLine;
	  var columnA = mappingA.generatedColumn;
	  var columnB = mappingB.generatedColumn;
	  return lineB > lineA || lineB == lineA && columnB >= columnA ||
	         util$1.compareByGeneratedPositionsInflated(mappingA, mappingB) <= 0;
	}

	/**
	 * A data structure to provide a sorted view of accumulated mappings in a
	 * performance conscious manner. It trades a neglibable overhead in general
	 * case for a large speedup in case of mappings being added in order.
	 */
	function MappingList$1() {
	  this._array = [];
	  this._sorted = true;
	  // Serves as infimum
	  this._last = {generatedLine: -1, generatedColumn: 0};
	}

	/**
	 * Iterate through internal items. This method takes the same arguments that
	 * `Array.prototype.forEach` takes.
	 *
	 * NOTE: The order of the mappings is NOT guaranteed.
	 */
	MappingList$1.prototype.unsortedForEach =
	  function MappingList_forEach(aCallback, aThisArg) {
	    this._array.forEach(aCallback, aThisArg);
	  };

	/**
	 * Add the given source mapping.
	 *
	 * @param Object aMapping
	 */
	MappingList$1.prototype.add = function MappingList_add(aMapping) {
	  if (generatedPositionAfter(this._last, aMapping)) {
	    this._last = aMapping;
	    this._array.push(aMapping);
	  } else {
	    this._sorted = false;
	    this._array.push(aMapping);
	  }
	};

	/**
	 * Returns the flat, sorted array of mappings. The mappings are sorted by
	 * generated position.
	 *
	 * WARNING: This method returns internal data without copying, for
	 * performance. The return value must NOT be mutated, and should be treated as
	 * an immutable borrow. If you want to take ownership, you must make your own
	 * copy.
	 */
	MappingList$1.prototype.toArray = function MappingList_toArray() {
	  if (!this._sorted) {
	    this._array.sort(util$1.compareByGeneratedPositionsInflated);
	    this._sorted = true;
	  }
	  return this._array;
	};

	mappingList.MappingList = MappingList$1;

	/* -*- Mode: js; js-indent-level: 2; -*- */

	/*
	 * Copyright 2011 Mozilla Foundation and contributors
	 * Licensed under the New BSD license. See LICENSE or:
	 * http://opensource.org/licenses/BSD-3-Clause
	 */

	var base64VLQ = base64Vlq;
	var util = util$3;
	var ArraySet = arraySet.ArraySet;
	var MappingList = mappingList.MappingList;

	/**
	 * An instance of the SourceMapGenerator represents a source map which is
	 * being built incrementally. You may pass an object with the following
	 * properties:
	 *
	 *   - file: The filename of the generated source.
	 *   - sourceRoot: A root for all relative URLs in this source map.
	 */
	function SourceMapGenerator$1(aArgs) {
	  if (!aArgs) {
	    aArgs = {};
	  }
	  this._file = util.getArg(aArgs, 'file', null);
	  this._sourceRoot = util.getArg(aArgs, 'sourceRoot', null);
	  this._skipValidation = util.getArg(aArgs, 'skipValidation', false);
	  this._sources = new ArraySet();
	  this._names = new ArraySet();
	  this._mappings = new MappingList();
	  this._sourcesContents = null;
	}

	SourceMapGenerator$1.prototype._version = 3;

	/**
	 * Creates a new SourceMapGenerator based on a SourceMapConsumer
	 *
	 * @param aSourceMapConsumer The SourceMap.
	 */
	SourceMapGenerator$1.fromSourceMap =
	  function SourceMapGenerator_fromSourceMap(aSourceMapConsumer) {
	    var sourceRoot = aSourceMapConsumer.sourceRoot;
	    var generator = new SourceMapGenerator$1({
	      file: aSourceMapConsumer.file,
	      sourceRoot: sourceRoot
	    });
	    aSourceMapConsumer.eachMapping(function (mapping) {
	      var newMapping = {
	        generated: {
	          line: mapping.generatedLine,
	          column: mapping.generatedColumn
	        }
	      };

	      if (mapping.source != null) {
	        newMapping.source = mapping.source;
	        if (sourceRoot != null) {
	          newMapping.source = util.relative(sourceRoot, newMapping.source);
	        }

	        newMapping.original = {
	          line: mapping.originalLine,
	          column: mapping.originalColumn
	        };

	        if (mapping.name != null) {
	          newMapping.name = mapping.name;
	        }
	      }

	      generator.addMapping(newMapping);
	    });
	    aSourceMapConsumer.sources.forEach(function (sourceFile) {
	      var sourceRelative = sourceFile;
	      if (sourceRoot !== null) {
	        sourceRelative = util.relative(sourceRoot, sourceFile);
	      }

	      if (!generator._sources.has(sourceRelative)) {
	        generator._sources.add(sourceRelative);
	      }

	      var content = aSourceMapConsumer.sourceContentFor(sourceFile);
	      if (content != null) {
	        generator.setSourceContent(sourceFile, content);
	      }
	    });
	    return generator;
	  };

	/**
	 * Add a single mapping from original source line and column to the generated
	 * source's line and column for this source map being created. The mapping
	 * object should have the following properties:
	 *
	 *   - generated: An object with the generated line and column positions.
	 *   - original: An object with the original line and column positions.
	 *   - source: The original source file (relative to the sourceRoot).
	 *   - name: An optional original token name for this mapping.
	 */
	SourceMapGenerator$1.prototype.addMapping =
	  function SourceMapGenerator_addMapping(aArgs) {
	    var generated = util.getArg(aArgs, 'generated');
	    var original = util.getArg(aArgs, 'original', null);
	    var source = util.getArg(aArgs, 'source', null);
	    var name = util.getArg(aArgs, 'name', null);

	    if (!this._skipValidation) {
	      this._validateMapping(generated, original, source, name);
	    }

	    if (source != null) {
	      source = String(source);
	      if (!this._sources.has(source)) {
	        this._sources.add(source);
	      }
	    }

	    if (name != null) {
	      name = String(name);
	      if (!this._names.has(name)) {
	        this._names.add(name);
	      }
	    }

	    this._mappings.add({
	      generatedLine: generated.line,
	      generatedColumn: generated.column,
	      originalLine: original != null && original.line,
	      originalColumn: original != null && original.column,
	      source: source,
	      name: name
	    });
	  };

	/**
	 * Set the source content for a source file.
	 */
	SourceMapGenerator$1.prototype.setSourceContent =
	  function SourceMapGenerator_setSourceContent(aSourceFile, aSourceContent) {
	    var source = aSourceFile;
	    if (this._sourceRoot != null) {
	      source = util.relative(this._sourceRoot, source);
	    }

	    if (aSourceContent != null) {
	      // Add the source content to the _sourcesContents map.
	      // Create a new _sourcesContents map if the property is null.
	      if (!this._sourcesContents) {
	        this._sourcesContents = Object.create(null);
	      }
	      this._sourcesContents[util.toSetString(source)] = aSourceContent;
	    } else if (this._sourcesContents) {
	      // Remove the source file from the _sourcesContents map.
	      // If the _sourcesContents map is empty, set the property to null.
	      delete this._sourcesContents[util.toSetString(source)];
	      if (Object.keys(this._sourcesContents).length === 0) {
	        this._sourcesContents = null;
	      }
	    }
	  };

	/**
	 * Applies the mappings of a sub-source-map for a specific source file to the
	 * source map being generated. Each mapping to the supplied source file is
	 * rewritten using the supplied source map. Note: The resolution for the
	 * resulting mappings is the minimium of this map and the supplied map.
	 *
	 * @param aSourceMapConsumer The source map to be applied.
	 * @param aSourceFile Optional. The filename of the source file.
	 *        If omitted, SourceMapConsumer's file property will be used.
	 * @param aSourceMapPath Optional. The dirname of the path to the source map
	 *        to be applied. If relative, it is relative to the SourceMapConsumer.
	 *        This parameter is needed when the two source maps aren't in the same
	 *        directory, and the source map to be applied contains relative source
	 *        paths. If so, those relative source paths need to be rewritten
	 *        relative to the SourceMapGenerator.
	 */
	SourceMapGenerator$1.prototype.applySourceMap =
	  function SourceMapGenerator_applySourceMap(aSourceMapConsumer, aSourceFile, aSourceMapPath) {
	    var sourceFile = aSourceFile;
	    // If aSourceFile is omitted, we will use the file property of the SourceMap
	    if (aSourceFile == null) {
	      if (aSourceMapConsumer.file == null) {
	        throw new Error(
	          'SourceMapGenerator.prototype.applySourceMap requires either an explicit source file, ' +
	          'or the source map\'s "file" property. Both were omitted.'
	        );
	      }
	      sourceFile = aSourceMapConsumer.file;
	    }
	    var sourceRoot = this._sourceRoot;
	    // Make "sourceFile" relative if an absolute Url is passed.
	    if (sourceRoot != null) {
	      sourceFile = util.relative(sourceRoot, sourceFile);
	    }
	    // Applying the SourceMap can add and remove items from the sources and
	    // the names array.
	    var newSources = new ArraySet();
	    var newNames = new ArraySet();

	    // Find mappings for the "sourceFile"
	    this._mappings.unsortedForEach(function (mapping) {
	      if (mapping.source === sourceFile && mapping.originalLine != null) {
	        // Check if it can be mapped by the source map, then update the mapping.
	        var original = aSourceMapConsumer.originalPositionFor({
	          line: mapping.originalLine,
	          column: mapping.originalColumn
	        });
	        if (original.source != null) {
	          // Copy mapping
	          mapping.source = original.source;
	          if (aSourceMapPath != null) {
	            mapping.source = util.join(aSourceMapPath, mapping.source);
	          }
	          if (sourceRoot != null) {
	            mapping.source = util.relative(sourceRoot, mapping.source);
	          }
	          mapping.originalLine = original.line;
	          mapping.originalColumn = original.column;
	          if (original.name != null) {
	            mapping.name = original.name;
	          }
	        }
	      }

	      var source = mapping.source;
	      if (source != null && !newSources.has(source)) {
	        newSources.add(source);
	      }

	      var name = mapping.name;
	      if (name != null && !newNames.has(name)) {
	        newNames.add(name);
	      }

	    }, this);
	    this._sources = newSources;
	    this._names = newNames;

	    // Copy sourcesContents of applied map.
	    aSourceMapConsumer.sources.forEach(function (sourceFile) {
	      var content = aSourceMapConsumer.sourceContentFor(sourceFile);
	      if (content != null) {
	        if (aSourceMapPath != null) {
	          sourceFile = util.join(aSourceMapPath, sourceFile);
	        }
	        if (sourceRoot != null) {
	          sourceFile = util.relative(sourceRoot, sourceFile);
	        }
	        this.setSourceContent(sourceFile, content);
	      }
	    }, this);
	  };

	/**
	 * A mapping can have one of the three levels of data:
	 *
	 *   1. Just the generated position.
	 *   2. The Generated position, original position, and original source.
	 *   3. Generated and original position, original source, as well as a name
	 *      token.
	 *
	 * To maintain consistency, we validate that any new mapping being added falls
	 * in to one of these categories.
	 */
	SourceMapGenerator$1.prototype._validateMapping =
	  function SourceMapGenerator_validateMapping(aGenerated, aOriginal, aSource,
	                                              aName) {
	    // When aOriginal is truthy but has empty values for .line and .column,
	    // it is most likely a programmer error. In this case we throw a very
	    // specific error message to try to guide them the right way.
	    // For example: https://github.com/Polymer/polymer-bundler/pull/519
	    if (aOriginal && typeof aOriginal.line !== 'number' && typeof aOriginal.column !== 'number') {
	        throw new Error(
	            'original.line and original.column are not numbers -- you probably meant to omit ' +
	            'the original mapping entirely and only map the generated position. If so, pass ' +
	            'null for the original mapping instead of an object with empty or null values.'
	        );
	    }

	    if (aGenerated && 'line' in aGenerated && 'column' in aGenerated
	        && aGenerated.line > 0 && aGenerated.column >= 0
	        && !aOriginal && !aSource && !aName) {
	      // Case 1.
	      return;
	    }
	    else if (aGenerated && 'line' in aGenerated && 'column' in aGenerated
	             && aOriginal && 'line' in aOriginal && 'column' in aOriginal
	             && aGenerated.line > 0 && aGenerated.column >= 0
	             && aOriginal.line > 0 && aOriginal.column >= 0
	             && aSource) {
	      // Cases 2 and 3.
	      return;
	    }
	    else {
	      throw new Error('Invalid mapping: ' + JSON.stringify({
	        generated: aGenerated,
	        source: aSource,
	        original: aOriginal,
	        name: aName
	      }));
	    }
	  };

	/**
	 * Serialize the accumulated mappings in to the stream of base 64 VLQs
	 * specified by the source map format.
	 */
	SourceMapGenerator$1.prototype._serializeMappings =
	  function SourceMapGenerator_serializeMappings() {
	    var previousGeneratedColumn = 0;
	    var previousGeneratedLine = 1;
	    var previousOriginalColumn = 0;
	    var previousOriginalLine = 0;
	    var previousName = 0;
	    var previousSource = 0;
	    var result = '';
	    var next;
	    var mapping;
	    var nameIdx;
	    var sourceIdx;

	    var mappings = this._mappings.toArray();
	    for (var i = 0, len = mappings.length; i < len; i++) {
	      mapping = mappings[i];
	      next = '';

	      if (mapping.generatedLine !== previousGeneratedLine) {
	        previousGeneratedColumn = 0;
	        while (mapping.generatedLine !== previousGeneratedLine) {
	          next += ';';
	          previousGeneratedLine++;
	        }
	      }
	      else {
	        if (i > 0) {
	          if (!util.compareByGeneratedPositionsInflated(mapping, mappings[i - 1])) {
	            continue;
	          }
	          next += ',';
	        }
	      }

	      next += base64VLQ.encode(mapping.generatedColumn
	                                 - previousGeneratedColumn);
	      previousGeneratedColumn = mapping.generatedColumn;

	      if (mapping.source != null) {
	        sourceIdx = this._sources.indexOf(mapping.source);
	        next += base64VLQ.encode(sourceIdx - previousSource);
	        previousSource = sourceIdx;

	        // lines are stored 0-based in SourceMap spec version 3
	        next += base64VLQ.encode(mapping.originalLine - 1
	                                   - previousOriginalLine);
	        previousOriginalLine = mapping.originalLine - 1;

	        next += base64VLQ.encode(mapping.originalColumn
	                                   - previousOriginalColumn);
	        previousOriginalColumn = mapping.originalColumn;

	        if (mapping.name != null) {
	          nameIdx = this._names.indexOf(mapping.name);
	          next += base64VLQ.encode(nameIdx - previousName);
	          previousName = nameIdx;
	        }
	      }

	      result += next;
	    }

	    return result;
	  };

	SourceMapGenerator$1.prototype._generateSourcesContent =
	  function SourceMapGenerator_generateSourcesContent(aSources, aSourceRoot) {
	    return aSources.map(function (source) {
	      if (!this._sourcesContents) {
	        return null;
	      }
	      if (aSourceRoot != null) {
	        source = util.relative(aSourceRoot, source);
	      }
	      var key = util.toSetString(source);
	      return Object.prototype.hasOwnProperty.call(this._sourcesContents, key)
	        ? this._sourcesContents[key]
	        : null;
	    }, this);
	  };

	/**
	 * Externalize the source map.
	 */
	SourceMapGenerator$1.prototype.toJSON =
	  function SourceMapGenerator_toJSON() {
	    var map = {
	      version: this._version,
	      sources: this._sources.toArray(),
	      names: this._names.toArray(),
	      mappings: this._serializeMappings()
	    };
	    if (this._file != null) {
	      map.file = this._file;
	    }
	    if (this._sourceRoot != null) {
	      map.sourceRoot = this._sourceRoot;
	    }
	    if (this._sourcesContents) {
	      map.sourcesContent = this._generateSourcesContent(map.sources, map.sourceRoot);
	    }

	    return map;
	  };

	/**
	 * Render the source map being generated to a string.
	 */
	SourceMapGenerator$1.prototype.toString =
	  function SourceMapGenerator_toString() {
	    return JSON.stringify(this.toJSON());
	  };

	sourceMapGenerator.SourceMapGenerator = SourceMapGenerator$1;

	var SourceMapGenerator = sourceMapGenerator.SourceMapGenerator;
	var trackNodes = {
	    Atrule: true,
	    Selector: true,
	    Declaration: true
	};

	var sourceMap$1 = function generateSourceMap(handlers) {
	    var map = new SourceMapGenerator();
	    var line = 1;
	    var column = 0;
	    var generated = {
	        line: 1,
	        column: 0
	    };
	    var original = {
	        line: 0, // should be zero to add first mapping
	        column: 0
	    };
	    var sourceMappingActive = false;
	    var activatedGenerated = {
	        line: 1,
	        column: 0
	    };
	    var activatedMapping = {
	        generated: activatedGenerated
	    };

	    var handlersNode = handlers.node;
	    handlers.node = function(node) {
	        if (node.loc && node.loc.start && trackNodes.hasOwnProperty(node.type)) {
	            var nodeLine = node.loc.start.line;
	            var nodeColumn = node.loc.start.column - 1;

	            if (original.line !== nodeLine ||
	                original.column !== nodeColumn) {
	                original.line = nodeLine;
	                original.column = nodeColumn;

	                generated.line = line;
	                generated.column = column;

	                if (sourceMappingActive) {
	                    sourceMappingActive = false;
	                    if (generated.line !== activatedGenerated.line ||
	                        generated.column !== activatedGenerated.column) {
	                        map.addMapping(activatedMapping);
	                    }
	                }

	                sourceMappingActive = true;
	                map.addMapping({
	                    source: node.loc.source,
	                    original: original,
	                    generated: generated
	                });
	            }
	        }

	        handlersNode.call(this, node);

	        if (sourceMappingActive && trackNodes.hasOwnProperty(node.type)) {
	            activatedGenerated.line = line;
	            activatedGenerated.column = column;
	        }
	    };

	    var handlersChunk = handlers.chunk;
	    handlers.chunk = function(chunk) {
	        for (var i = 0; i < chunk.length; i++) {
	            if (chunk.charCodeAt(i) === 10) { // \n
	                line++;
	                column = 0;
	            } else {
	                column++;
	            }
	        }

	        handlersChunk(chunk);
	    };

	    var handlersResult = handlers.result;
	    handlers.result = function() {
	        if (sourceMappingActive) {
	            map.addMapping(activatedMapping);
	        }

	        return {
	            css: handlersResult(),
	            map: map
	        };
	    };

	    return handlers;
	};

	var sourceMap = sourceMap$1;
	var hasOwnProperty$4 = Object.prototype.hasOwnProperty;

	function processChildren(node, delimeter) {
	    var list = node.children;
	    var prev = null;

	    if (typeof delimeter !== 'function') {
	        list.forEach(this.node, this);
	    } else {
	        list.forEach(function(node) {
	            if (prev !== null) {
	                delimeter.call(this, prev);
	            }

	            this.node(node);
	            prev = node;
	        }, this);
	    }
	}

	var create$2 = function createGenerator(config) {
	    function processNode(node) {
	        if (hasOwnProperty$4.call(types, node.type)) {
	            types[node.type].call(this, node);
	        } else {
	            throw new Error('Unknown node type: ' + node.type);
	        }
	    }

	    var types = {};

	    if (config.node) {
	        for (var name in config.node) {
	            types[name] = config.node[name].generate;
	        }
	    }

	    return function(node, options) {
	        var buffer = '';
	        var handlers = {
	            children: processChildren,
	            node: processNode,
	            chunk: function(chunk) {
	                buffer += chunk;
	            },
	            result: function() {
	                return buffer;
	            }
	        };

	        if (options) {
	            if (typeof options.decorator === 'function') {
	                handlers = options.decorator(handlers);
	            }

	            if (options.sourceMap) {
	                handlers = sourceMap(handlers);
	            }
	        }

	        handlers.node(node);

	        return handlers.result();
	    };
	};

	var List$2 = List_1;

	var create$1 = function createConvertors(walk) {
	    return {
	        fromPlainObject: function(ast) {
	            walk(ast, {
	                enter: function(node) {
	                    if (node.children && node.children instanceof List$2 === false) {
	                        node.children = new List$2().fromArray(node.children);
	                    }
	                }
	            });

	            return ast;
	        },
	        toPlainObject: function(ast) {
	            walk(ast, {
	                leave: function(node) {
	                    if (node.children && node.children instanceof List$2) {
	                        node.children = node.children.toArray();
	                    }
	                }
	            });

	            return ast;
	        }
	    };
	};

	var hasOwnProperty$3 = Object.prototype.hasOwnProperty;
	var noop = function() {};

	function ensureFunction(value) {
	    return typeof value === 'function' ? value : noop;
	}

	function invokeForType(fn, type) {
	    return function(node, item, list) {
	        if (node.type === type) {
	            fn.call(this, node, item, list);
	        }
	    };
	}

	function getWalkersFromStructure(name, nodeType) {
	    var structure = nodeType.structure;
	    var walkers = [];

	    for (var key in structure) {
	        if (hasOwnProperty$3.call(structure, key) === false) {
	            continue;
	        }

	        var fieldTypes = structure[key];
	        var walker = {
	            name: key,
	            type: false,
	            nullable: false
	        };

	        if (!Array.isArray(structure[key])) {
	            fieldTypes = [structure[key]];
	        }

	        for (var i = 0; i < fieldTypes.length; i++) {
	            var fieldType = fieldTypes[i];
	            if (fieldType === null) {
	                walker.nullable = true;
	            } else if (typeof fieldType === 'string') {
	                walker.type = 'node';
	            } else if (Array.isArray(fieldType)) {
	                walker.type = 'list';
	            }
	        }

	        if (walker.type) {
	            walkers.push(walker);
	        }
	    }

	    if (walkers.length) {
	        return {
	            context: nodeType.walkContext,
	            fields: walkers
	        };
	    }

	    return null;
	}

	function getTypesFromConfig(config) {
	    var types = {};

	    for (var name in config.node) {
	        if (hasOwnProperty$3.call(config.node, name)) {
	            var nodeType = config.node[name];

	            if (!nodeType.structure) {
	                throw new Error('Missed `structure` field in `' + name + '` node type definition');
	            }

	            types[name] = getWalkersFromStructure(name, nodeType);
	        }
	    }

	    return types;
	}

	function createTypeIterator(config, reverse) {
	    var fields = config.fields.slice();
	    var contextName = config.context;
	    var useContext = typeof contextName === 'string';

	    if (reverse) {
	        fields.reverse();
	    }

	    return function(node, context, walk, walkReducer) {
	        var prevContextValue;

	        if (useContext) {
	            prevContextValue = context[contextName];
	            context[contextName] = node;
	        }

	        for (var i = 0; i < fields.length; i++) {
	            var field = fields[i];
	            var ref = node[field.name];

	            if (!field.nullable || ref) {
	                if (field.type === 'list') {
	                    var breakWalk = reverse
	                        ? ref.reduceRight(walkReducer, false)
	                        : ref.reduce(walkReducer, false);

	                    if (breakWalk) {
	                        return true;
	                    }
	                } else if (walk(ref)) {
	                    return true;
	                }
	            }
	        }

	        if (useContext) {
	            context[contextName] = prevContextValue;
	        }
	    };
	}

	function createFastTraveralMap(iterators) {
	    return {
	        Atrule: {
	            StyleSheet: iterators.StyleSheet,
	            Atrule: iterators.Atrule,
	            Rule: iterators.Rule,
	            Block: iterators.Block
	        },
	        Rule: {
	            StyleSheet: iterators.StyleSheet,
	            Atrule: iterators.Atrule,
	            Rule: iterators.Rule,
	            Block: iterators.Block
	        },
	        Declaration: {
	            StyleSheet: iterators.StyleSheet,
	            Atrule: iterators.Atrule,
	            Rule: iterators.Rule,
	            Block: iterators.Block,
	            DeclarationList: iterators.DeclarationList
	        }
	    };
	}

	var create = function createWalker(config) {
	    var types = getTypesFromConfig(config);
	    var iteratorsNatural = {};
	    var iteratorsReverse = {};
	    var breakWalk = Symbol('break-walk');
	    var skipNode = Symbol('skip-node');

	    for (var name in types) {
	        if (hasOwnProperty$3.call(types, name) && types[name] !== null) {
	            iteratorsNatural[name] = createTypeIterator(types[name], false);
	            iteratorsReverse[name] = createTypeIterator(types[name], true);
	        }
	    }

	    var fastTraversalIteratorsNatural = createFastTraveralMap(iteratorsNatural);
	    var fastTraversalIteratorsReverse = createFastTraveralMap(iteratorsReverse);

	    var walk = function(root, options) {
	        function walkNode(node, item, list) {
	            var enterRet = enter.call(context, node, item, list);

	            if (enterRet === breakWalk) {
	                debugger;
	                return true;
	            }

	            if (enterRet === skipNode) {
	                return false;
	            }

	            if (iterators.hasOwnProperty(node.type)) {
	                if (iterators[node.type](node, context, walkNode, walkReducer)) {
	                    return true;
	                }
	            }

	            if (leave.call(context, node, item, list) === breakWalk) {
	                return true;
	            }

	            return false;
	        }

	        var walkReducer = (ret, data, item, list) => ret || walkNode(data, item, list);
	        var enter = noop;
	        var leave = noop;
	        var iterators = iteratorsNatural;
	        var context = {
	            break: breakWalk,
	            skip: skipNode,

	            root: root,
	            stylesheet: null,
	            atrule: null,
	            atrulePrelude: null,
	            rule: null,
	            selector: null,
	            block: null,
	            declaration: null,
	            function: null
	        };

	        if (typeof options === 'function') {
	            enter = options;
	        } else if (options) {
	            enter = ensureFunction(options.enter);
	            leave = ensureFunction(options.leave);

	            if (options.reverse) {
	                iterators = iteratorsReverse;
	            }

	            if (options.visit) {
	                if (fastTraversalIteratorsNatural.hasOwnProperty(options.visit)) {
	                    iterators = options.reverse
	                        ? fastTraversalIteratorsReverse[options.visit]
	                        : fastTraversalIteratorsNatural[options.visit];
	                } else if (!types.hasOwnProperty(options.visit)) {
	                    throw new Error('Bad value `' + options.visit + '` for `visit` option (should be: ' + Object.keys(types).join(', ') + ')');
	                }

	                enter = invokeForType(enter, options.visit);
	                leave = invokeForType(leave, options.visit);
	            }
	        }

	        if (enter === noop && leave === noop) {
	            throw new Error('Neither `enter` nor `leave` walker handler is set or both aren\'t a function');
	        }

	        walkNode(root);
	    };

	    walk.break = breakWalk;
	    walk.skip = skipNode;

	    walk.find = function(ast, fn) {
	        var found = null;

	        walk(ast, function(node, item, list) {
	            if (fn.call(this, node, item, list)) {
	                found = node;
	                return breakWalk;
	            }
	        });

	        return found;
	    };

	    walk.findLast = function(ast, fn) {
	        var found = null;

	        walk(ast, {
	            reverse: true,
	            enter: function(node, item, list) {
	                if (fn.call(this, node, item, list)) {
	                    found = node;
	                    return breakWalk;
	                }
	            }
	        });

	        return found;
	    };

	    walk.findAll = function(ast, fn) {
	        var found = [];

	        walk(ast, function(node, item, list) {
	            if (fn.call(this, node, item, list)) {
	                found.push(node);
	            }
	        });

	        return found;
	    };

	    return walk;
	};

	var List$1 = List_1;

	var clone$1 = function clone(node) {
	    var result = {};

	    for (var key in node) {
	        var value = node[key];

	        if (value) {
	            if (Array.isArray(value) || value instanceof List$1) {
	                value = value.map(clone);
	            } else if (value.constructor === Object) {
	                value = clone(value);
	            }
	        }

	        result[key] = value;
	    }

	    return result;
	};

	const hasOwnProperty$2 = Object.prototype.hasOwnProperty;
	const shape$1 = {
	    generic: true,
	    types: appendOrAssign,
	    atrules: {
	        prelude: appendOrAssignOrNull,
	        descriptors: appendOrAssignOrNull
	    },
	    properties: appendOrAssign,
	    parseContext: assign,
	    scope: deepAssign,
	    atrule: ['parse'],
	    pseudo: ['parse'],
	    node: ['name', 'structure', 'parse', 'generate', 'walkContext']
	};

	function isObject$2(value) {
	    return value && value.constructor === Object;
	}

	function copy(value) {
	    return isObject$2(value)
	        ? Object.assign({}, value)
	        : value;
	}

	function assign(dest, src) {
	    return Object.assign(dest, src);
	}

	function deepAssign(dest, src) {
	    for (const key in src) {
	        if (hasOwnProperty$2.call(src, key)) {
	            if (isObject$2(dest[key])) {
	                deepAssign(dest[key], copy(src[key]));
	            } else {
	                dest[key] = copy(src[key]);
	            }
	        }
	    }

	    return dest;
	}

	function append(a, b) {
	    if (typeof b === 'string' && /^\s*\|/.test(b)) {
	        return typeof a === 'string'
	            ? a + b
	            : b.replace(/^\s*\|\s*/, '');
	    }

	    return b || null;
	}

	function appendOrAssign(a, b) {
	    if (typeof b === 'string') {
	        return append(a, b);
	    }

	    const result = Object.assign({}, a);
	    for (let key in b) {
	        if (hasOwnProperty$2.call(b, key)) {
	            result[key] = append(hasOwnProperty$2.call(a, key) ? a[key] : undefined, b[key]);
	        }
	    }

	    return result;
	}

	function appendOrAssignOrNull(a, b) {
	    const result = appendOrAssign(a, b);

	    return !isObject$2(result) || Object.keys(result).length
	        ? result
	        : null;
	}

	function mix$1(dest, src, shape) {
	    for (const key in shape) {
	        if (hasOwnProperty$2.call(shape, key) === false) {
	            continue;
	        }

	        if (shape[key] === true) {
	            if (key in src) {
	                if (hasOwnProperty$2.call(src, key)) {
	                    dest[key] = copy(src[key]);
	                }
	            }
	        } else if (shape[key]) {
	            if (typeof shape[key] === 'function') {
	                const fn = shape[key];
	                dest[key] = fn({}, dest[key]);
	                dest[key] = fn(dest[key] || {}, src[key]);
	            } else if (isObject$2(shape[key])) {
	                const result = {};

	                for (let name in dest[key]) {
	                    result[name] = mix$1({}, dest[key][name], shape[key]);
	                }

	                for (let name in src[key]) {
	                    result[name] = mix$1(result[name] || {}, src[key][name], shape[key]);
	                }

	                dest[key] = result;
	            } else if (Array.isArray(shape[key])) {
	                const res = {};
	                const innerShape = shape[key].reduce(function(s, k) {
	                    s[k] = true;
	                    return s;
	                }, {});

	                for (const [name, value] of Object.entries(dest[key] || {})) {
	                    res[name] = {};
	                    if (value) {
	                        mix$1(res[name], value, innerShape);
	                    }
	                }

	                for (const name in src[key]) {
	                    if (hasOwnProperty$2.call(src[key], name)) {
	                        if (!res[name]) {
	                            res[name] = {};
	                        }

	                        if (src[key] && src[key][name]) {
	                            mix$1(res[name], src[key][name], innerShape);
	                        }
	                    }
	                }

	                dest[key] = res;
	            }
	        }
	    }
	    return dest;
	}

	var mix_1 = (dest, src) => mix$1(dest, src, shape$1);

	var List = List_1;
	var SyntaxError$1 = _SyntaxError$1;
	var TokenStream = TokenStream_1;
	var Lexer = Lexer_1;
	var definitionSyntax = definitionSyntax$1;
	var tokenize = tokenizer$3;
	var createParser = create$3;
	var createGenerator = create$2;
	var createConvertor = create$1;
	var createWalker = create;
	var clone = clone$1;
	var names = names$2;
	var mix = mix_1;

	function createSyntax(config) {
	    var parse = createParser(config);
	    var walk = createWalker(config);
	    var generate = createGenerator(config);
	    var convert = createConvertor(walk);

	    var syntax = {
	        List: List,
	        SyntaxError: SyntaxError$1,
	        TokenStream: TokenStream,
	        Lexer: Lexer,

	        vendorPrefix: names.vendorPrefix,
	        keyword: names.keyword,
	        property: names.property,
	        isCustomProperty: names.isCustomProperty,

	        definitionSyntax: definitionSyntax,
	        lexer: null,
	        createLexer: function(config) {
	            return new Lexer(config, syntax, syntax.lexer.structure);
	        },

	        tokenize: tokenize,
	        parse: parse,
	        walk: walk,
	        generate: generate,

	        find: walk.find,
	        findLast: walk.findLast,
	        findAll: walk.findAll,

	        clone: clone,
	        fromPlainObject: convert.fromPlainObject,
	        toPlainObject: convert.toPlainObject,

	        createSyntax: function(config) {
	            return createSyntax(mix({}, config));
	        },
	        fork: function(extension) {
	            var base = mix({}, config); // copy of config
	            return createSyntax(
	                typeof extension === 'function'
	                    ? extension(base, Object.assign)
	                    : mix(base, extension)
	            );
	        }
	    };

	    syntax.lexer = new Lexer({
	        generic: true,
	        types: config.types,
	        atrules: config.atrules,
	        properties: config.properties,
	        node: config.node
	    }, syntax);

	    return syntax;
	}
	create$4.create = function(config) {
	    return createSyntax(mix({}, config));
	};

	var require$$0 = {
		"@charset": {
		syntax: "@charset \"<charset>\";",
		groups: [
			"CSS Charsets"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/@charset"
	},
		"@counter-style": {
		syntax: "@counter-style <counter-style-name> {\n  [ system: <counter-system>; ] ||\n  [ symbols: <counter-symbols>; ] ||\n  [ additive-symbols: <additive-symbols>; ] ||\n  [ negative: <negative-symbol>; ] ||\n  [ prefix: <prefix>; ] ||\n  [ suffix: <suffix>; ] ||\n  [ range: <range>; ] ||\n  [ pad: <padding>; ] ||\n  [ speak-as: <speak-as>; ] ||\n  [ fallback: <counter-style-name>; ]\n}",
		interfaces: [
			"CSSCounterStyleRule"
		],
		groups: [
			"CSS Counter Styles"
		],
		descriptors: {
			"additive-symbols": {
				syntax: "[ <integer> && <symbol> ]#",
				media: "all",
				initial: "n/a (required)",
				percentages: "no",
				computed: "asSpecified",
				order: "orderOfAppearance",
				status: "standard"
			},
			fallback: {
				syntax: "<counter-style-name>",
				media: "all",
				initial: "decimal",
				percentages: "no",
				computed: "asSpecified",
				order: "uniqueOrder",
				status: "standard"
			},
			negative: {
				syntax: "<symbol> <symbol>?",
				media: "all",
				initial: "\"-\" hyphen-minus",
				percentages: "no",
				computed: "asSpecified",
				order: "orderOfAppearance",
				status: "standard"
			},
			pad: {
				syntax: "<integer> && <symbol>",
				media: "all",
				initial: "0 \"\"",
				percentages: "no",
				computed: "asSpecified",
				order: "uniqueOrder",
				status: "standard"
			},
			prefix: {
				syntax: "<symbol>",
				media: "all",
				initial: "\"\"",
				percentages: "no",
				computed: "asSpecified",
				order: "uniqueOrder",
				status: "standard"
			},
			range: {
				syntax: "[ [ <integer> | infinite ]{2} ]# | auto",
				media: "all",
				initial: "auto",
				percentages: "no",
				computed: "asSpecified",
				order: "orderOfAppearance",
				status: "standard"
			},
			"speak-as": {
				syntax: "auto | bullets | numbers | words | spell-out | <counter-style-name>",
				media: "all",
				initial: "auto",
				percentages: "no",
				computed: "asSpecified",
				order: "uniqueOrder",
				status: "standard"
			},
			suffix: {
				syntax: "<symbol>",
				media: "all",
				initial: "\". \"",
				percentages: "no",
				computed: "asSpecified",
				order: "uniqueOrder",
				status: "standard"
			},
			symbols: {
				syntax: "<symbol>+",
				media: "all",
				initial: "n/a (required)",
				percentages: "no",
				computed: "asSpecified",
				order: "orderOfAppearance",
				status: "standard"
			},
			system: {
				syntax: "cyclic | numeric | alphabetic | symbolic | additive | [ fixed <integer>? ] | [ extends <counter-style-name> ]",
				media: "all",
				initial: "symbolic",
				percentages: "no",
				computed: "asSpecified",
				order: "uniqueOrder",
				status: "standard"
			}
		},
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/@counter-style"
	},
		"@document": {
		syntax: "@document [ <url> | url-prefix(<string>) | domain(<string>) | media-document(<string>) | regexp(<string>) ]# {\n  <group-rule-body>\n}",
		interfaces: [
			"CSSGroupingRule",
			"CSSConditionRule"
		],
		groups: [
			"CSS Conditional Rules"
		],
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/@document"
	},
		"@font-face": {
		syntax: "@font-face {\n  [ font-family: <family-name>; ] ||\n  [ src: <src>; ] ||\n  [ unicode-range: <unicode-range>; ] ||\n  [ font-variant: <font-variant>; ] ||\n  [ font-feature-settings: <font-feature-settings>; ] ||\n  [ font-variation-settings: <font-variation-settings>; ] ||\n  [ font-stretch: <font-stretch>; ] ||\n  [ font-weight: <font-weight>; ] ||\n  [ font-style: <font-style>; ]\n}",
		interfaces: [
			"CSSFontFaceRule"
		],
		groups: [
			"CSS Fonts"
		],
		descriptors: {
			"font-display": {
				syntax: "[ auto | block | swap | fallback | optional ]",
				media: "visual",
				percentages: "no",
				initial: "auto",
				computed: "asSpecified",
				order: "uniqueOrder",
				status: "experimental"
			},
			"font-family": {
				syntax: "<family-name>",
				media: "all",
				initial: "n/a (required)",
				percentages: "no",
				computed: "asSpecified",
				order: "uniqueOrder",
				status: "standard"
			},
			"font-feature-settings": {
				syntax: "normal | <feature-tag-value>#",
				media: "all",
				initial: "normal",
				percentages: "no",
				computed: "asSpecified",
				order: "orderOfAppearance",
				status: "standard"
			},
			"font-variation-settings": {
				syntax: "normal | [ <string> <number> ]#",
				media: "all",
				initial: "normal",
				percentages: "no",
				computed: "asSpecified",
				order: "orderOfAppearance",
				status: "standard"
			},
			"font-stretch": {
				syntax: "<font-stretch-absolute>{1,2}",
				media: "all",
				initial: "normal",
				percentages: "no",
				computed: "asSpecified",
				order: "uniqueOrder",
				status: "standard"
			},
			"font-style": {
				syntax: "normal | italic | oblique <angle>{0,2}",
				media: "all",
				initial: "normal",
				percentages: "no",
				computed: "asSpecified",
				order: "uniqueOrder",
				status: "standard"
			},
			"font-weight": {
				syntax: "<font-weight-absolute>{1,2}",
				media: "all",
				initial: "normal",
				percentages: "no",
				computed: "asSpecified",
				order: "uniqueOrder",
				status: "standard"
			},
			"font-variant": {
				syntax: "normal | none | [ <common-lig-values> || <discretionary-lig-values> || <historical-lig-values> || <contextual-alt-values> || stylistic(<feature-value-name>) || historical-forms || styleset(<feature-value-name>#) || character-variant(<feature-value-name>#) || swash(<feature-value-name>) || ornaments(<feature-value-name>) || annotation(<feature-value-name>) || [ small-caps | all-small-caps | petite-caps | all-petite-caps | unicase | titling-caps ] || <numeric-figure-values> || <numeric-spacing-values> || <numeric-fraction-values> || ordinal || slashed-zero || <east-asian-variant-values> || <east-asian-width-values> || ruby ]",
				media: "all",
				initial: "normal",
				percentages: "no",
				computed: "asSpecified",
				order: "orderOfAppearance",
				status: "standard"
			},
			src: {
				syntax: "[ <url> [ format( <string># ) ]? | local( <family-name> ) ]#",
				media: "all",
				initial: "n/a (required)",
				percentages: "no",
				computed: "asSpecified",
				order: "orderOfAppearance",
				status: "standard"
			},
			"unicode-range": {
				syntax: "<unicode-range>#",
				media: "all",
				initial: "U+0-10FFFF",
				percentages: "no",
				computed: "asSpecified",
				order: "orderOfAppearance",
				status: "standard"
			}
		},
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/@font-face"
	},
		"@font-feature-values": {
		syntax: "@font-feature-values <family-name># {\n  <feature-value-block-list>\n}",
		interfaces: [
			"CSSFontFeatureValuesRule"
		],
		groups: [
			"CSS Fonts"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/@font-feature-values"
	},
		"@import": {
		syntax: "@import [ <string> | <url> ] [ <media-query-list> ]?;",
		groups: [
			"Media Queries"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/@import"
	},
		"@keyframes": {
		syntax: "@keyframes <keyframes-name> {\n  <keyframe-block-list>\n}",
		interfaces: [
			"CSSKeyframeRule",
			"CSSKeyframesRule"
		],
		groups: [
			"CSS Animations"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/@keyframes"
	},
		"@media": {
		syntax: "@media <media-query-list> {\n  <group-rule-body>\n}",
		interfaces: [
			"CSSGroupingRule",
			"CSSConditionRule",
			"CSSMediaRule",
			"CSSCustomMediaRule"
		],
		groups: [
			"CSS Conditional Rules",
			"Media Queries"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/@media"
	},
		"@namespace": {
		syntax: "@namespace <namespace-prefix>? [ <string> | <url> ];",
		groups: [
			"CSS Namespaces"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/@namespace"
	},
		"@page": {
		syntax: "@page <page-selector-list> {\n  <page-body>\n}",
		interfaces: [
			"CSSPageRule"
		],
		groups: [
			"CSS Pages"
		],
		descriptors: {
			bleed: {
				syntax: "auto | <length>",
				media: [
					"visual",
					"paged"
				],
				initial: "auto",
				percentages: "no",
				computed: "asSpecified",
				order: "uniqueOrder",
				status: "standard"
			},
			marks: {
				syntax: "none | [ crop || cross ]",
				media: [
					"visual",
					"paged"
				],
				initial: "none",
				percentages: "no",
				computed: "asSpecified",
				order: "orderOfAppearance",
				status: "standard"
			},
			size: {
				syntax: "<length>{1,2} | auto | [ <page-size> || [ portrait | landscape ] ]",
				media: [
					"visual",
					"paged"
				],
				initial: "auto",
				percentages: "no",
				computed: "asSpecifiedRelativeToAbsoluteLengths",
				order: "orderOfAppearance",
				status: "standard"
			}
		},
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/@page"
	},
		"@property": {
		syntax: "@property <custom-property-name> {\n  <declaration-list>\n}",
		interfaces: [
			"CSS",
			"CSSPropertyRule"
		],
		groups: [
			"CSS Houdini"
		],
		descriptors: {
			syntax: {
				syntax: "<string>",
				media: "all",
				percentages: "no",
				initial: "n/a (required)",
				computed: "asSpecified",
				order: "uniqueOrder",
				status: "experimental"
			},
			inherits: {
				syntax: "true | false",
				media: "all",
				percentages: "no",
				initial: "auto",
				computed: "asSpecified",
				order: "uniqueOrder",
				status: "experimental"
			},
			"initial-value": {
				syntax: "<string>",
				media: "all",
				initial: "n/a (required)",
				percentages: "no",
				computed: "asSpecified",
				order: "uniqueOrder",
				status: "experimental"
			}
		},
		status: "experimental",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/@property"
	},
		"@supports": {
		syntax: "@supports <supports-condition> {\n  <group-rule-body>\n}",
		interfaces: [
			"CSSGroupingRule",
			"CSSConditionRule",
			"CSSSupportsRule"
		],
		groups: [
			"CSS Conditional Rules"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/@supports"
	},
		"@viewport": {
		syntax: "@viewport {\n  <group-rule-body>\n}",
		interfaces: [
			"CSSViewportRule"
		],
		groups: [
			"CSS Device Adaptation"
		],
		descriptors: {
			height: {
				syntax: "<viewport-length>{1,2}",
				media: [
					"visual",
					"continuous"
				],
				initial: [
					"min-height",
					"max-height"
				],
				percentages: [
					"min-height",
					"max-height"
				],
				computed: [
					"min-height",
					"max-height"
				],
				order: "orderOfAppearance",
				status: "standard"
			},
			"max-height": {
				syntax: "<viewport-length>",
				media: [
					"visual",
					"continuous"
				],
				initial: "auto",
				percentages: "referToHeightOfInitialViewport",
				computed: "lengthAbsolutePercentageAsSpecifiedOtherwiseAuto",
				order: "uniqueOrder",
				status: "standard"
			},
			"max-width": {
				syntax: "<viewport-length>",
				media: [
					"visual",
					"continuous"
				],
				initial: "auto",
				percentages: "referToWidthOfInitialViewport",
				computed: "lengthAbsolutePercentageAsSpecifiedOtherwiseAuto",
				order: "uniqueOrder",
				status: "standard"
			},
			"max-zoom": {
				syntax: "auto | <number> | <percentage>",
				media: [
					"visual",
					"continuous"
				],
				initial: "auto",
				percentages: "the zoom factor itself",
				computed: "autoNonNegativeOrPercentage",
				order: "uniqueOrder",
				status: "standard"
			},
			"min-height": {
				syntax: "<viewport-length>",
				media: [
					"visual",
					"continuous"
				],
				initial: "auto",
				percentages: "referToHeightOfInitialViewport",
				computed: "lengthAbsolutePercentageAsSpecifiedOtherwiseAuto",
				order: "uniqueOrder",
				status: "standard"
			},
			"min-width": {
				syntax: "<viewport-length>",
				media: [
					"visual",
					"continuous"
				],
				initial: "auto",
				percentages: "referToWidthOfInitialViewport",
				computed: "lengthAbsolutePercentageAsSpecifiedOtherwiseAuto",
				order: "uniqueOrder",
				status: "standard"
			},
			"min-zoom": {
				syntax: "auto | <number> | <percentage>",
				media: [
					"visual",
					"continuous"
				],
				initial: "auto",
				percentages: "the zoom factor itself",
				computed: "autoNonNegativeOrPercentage",
				order: "uniqueOrder",
				status: "standard"
			},
			orientation: {
				syntax: "auto | portrait | landscape",
				media: [
					"visual",
					"continuous"
				],
				initial: "auto",
				percentages: "referToSizeOfBoundingBox",
				computed: "asSpecified",
				order: "uniqueOrder",
				status: "standard"
			},
			"user-zoom": {
				syntax: "zoom | fixed",
				media: [
					"visual",
					"continuous"
				],
				initial: "zoom",
				percentages: "referToSizeOfBoundingBox",
				computed: "asSpecified",
				order: "uniqueOrder",
				status: "standard"
			},
			"viewport-fit": {
				syntax: "auto | contain | cover",
				media: [
					"visual",
					"continuous"
				],
				initial: "auto",
				percentages: "no",
				computed: "asSpecified",
				order: "uniqueOrder",
				status: "standard"
			},
			width: {
				syntax: "<viewport-length>{1,2}",
				media: [
					"visual",
					"continuous"
				],
				initial: [
					"min-width",
					"max-width"
				],
				percentages: [
					"min-width",
					"max-width"
				],
				computed: [
					"min-width",
					"max-width"
				],
				order: "orderOfAppearance",
				status: "standard"
			},
			zoom: {
				syntax: "auto | <number> | <percentage>",
				media: [
					"visual",
					"continuous"
				],
				initial: "auto",
				percentages: "the zoom factor itself",
				computed: "autoNonNegativeOrPercentage",
				order: "uniqueOrder",
				status: "standard"
			}
		},
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/@viewport"
	}
	};

	var all = {
		syntax: "initial | inherit | unset | revert",
		media: "noPracticalMedia",
		inherited: false,
		animationType: "eachOfShorthandPropertiesExceptUnicodeBiDiAndDirection",
		percentages: "no",
		groups: [
			"CSS Miscellaneous"
		],
		initial: "noPracticalInitialValue",
		appliesto: "allElements",
		computed: "asSpecifiedAppliesToEachProperty",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/all"
	};
	var animation = {
		syntax: "<single-animation>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Animations"
		],
		initial: [
			"animation-name",
			"animation-duration",
			"animation-timing-function",
			"animation-delay",
			"animation-iteration-count",
			"animation-direction",
			"animation-fill-mode",
			"animation-play-state"
		],
		appliesto: "allElementsAndPseudos",
		computed: [
			"animation-name",
			"animation-duration",
			"animation-timing-function",
			"animation-delay",
			"animation-direction",
			"animation-iteration-count",
			"animation-fill-mode",
			"animation-play-state"
		],
		order: "orderOfAppearance",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/animation"
	};
	var appearance = {
		syntax: "none | auto | textfield | menulist-button | <compat-auto>",
		media: "all",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Basic User Interface"
		],
		initial: "auto",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "experimental",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/appearance"
	};
	var azimuth = {
		syntax: "<angle> | [ [ left-side | far-left | left | center-left | center | center-right | right | far-right | right-side ] || behind ] | leftwards | rightwards",
		media: "aural",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Speech"
		],
		initial: "center",
		appliesto: "allElements",
		computed: "normalizedAngle",
		order: "orderOfAppearance",
		status: "obsolete",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/azimuth"
	};
	var background = {
		syntax: "[ <bg-layer> , ]* <final-bg-layer>",
		media: "visual",
		inherited: false,
		animationType: [
			"background-color",
			"background-image",
			"background-clip",
			"background-position",
			"background-size",
			"background-repeat",
			"background-attachment"
		],
		percentages: [
			"background-position",
			"background-size"
		],
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: [
			"background-image",
			"background-position",
			"background-size",
			"background-repeat",
			"background-origin",
			"background-clip",
			"background-attachment",
			"background-color"
		],
		appliesto: "allElements",
		computed: [
			"background-image",
			"background-position",
			"background-size",
			"background-repeat",
			"background-origin",
			"background-clip",
			"background-attachment",
			"background-color"
		],
		order: "orderOfAppearance",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/background"
	};
	var border = {
		syntax: "<line-width> || <line-style> || <color>",
		media: "visual",
		inherited: false,
		animationType: [
			"border-color",
			"border-style",
			"border-width"
		],
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: [
			"border-width",
			"border-style",
			"border-color"
		],
		appliesto: "allElements",
		computed: [
			"border-width",
			"border-style",
			"border-color"
		],
		order: "orderOfAppearance",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border"
	};
	var bottom = {
		syntax: "<length> | <percentage> | auto",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "referToContainingBlockHeight",
		groups: [
			"CSS Positioning"
		],
		initial: "auto",
		appliesto: "positionedElements",
		computed: "lengthAbsolutePercentageAsSpecifiedOtherwiseAuto",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/bottom"
	};
	var clear = {
		syntax: "none | left | right | both | inline-start | inline-end",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Positioning"
		],
		initial: "none",
		appliesto: "blockLevelElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/clear"
	};
	var clip = {
		syntax: "<shape> | auto",
		media: "visual",
		inherited: false,
		animationType: "rectangle",
		percentages: "no",
		groups: [
			"CSS Masking"
		],
		initial: "auto",
		appliesto: "absolutelyPositionedElements",
		computed: "autoOrRectangle",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/clip"
	};
	var color$1 = {
		syntax: "<color>",
		media: "visual",
		inherited: true,
		animationType: "color",
		percentages: "no",
		groups: [
			"CSS Color"
		],
		initial: "variesFromBrowserToBrowser",
		appliesto: "allElements",
		computed: "translucentValuesRGBAOtherwiseRGB",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/color"
	};
	var columns = {
		syntax: "<'column-width'> || <'column-count'>",
		media: "visual",
		inherited: false,
		animationType: [
			"column-width",
			"column-count"
		],
		percentages: "no",
		groups: [
			"CSS Columns"
		],
		initial: [
			"column-width",
			"column-count"
		],
		appliesto: "blockContainersExceptTableWrappers",
		computed: [
			"column-width",
			"column-count"
		],
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/columns"
	};
	var contain = {
		syntax: "none | strict | content | [ size || layout || style || paint ]",
		media: "all",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Containment"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/contain"
	};
	var content = {
		syntax: "normal | none | [ <content-replacement> | <content-list> ] [/ <string> ]?",
		media: "all",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Generated Content"
		],
		initial: "normal",
		appliesto: "beforeAndAfterPseudos",
		computed: "normalOnElementsForPseudosNoneAbsoluteURIStringOrAsSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/content"
	};
	var cursor = {
		syntax: "[ [ <url> [ <x> <y> ]? , ]* [ auto | default | none | context-menu | help | pointer | progress | wait | cell | crosshair | text | vertical-text | alias | copy | move | no-drop | not-allowed | e-resize | n-resize | ne-resize | nw-resize | s-resize | se-resize | sw-resize | w-resize | ew-resize | ns-resize | nesw-resize | nwse-resize | col-resize | row-resize | all-scroll | zoom-in | zoom-out | grab | grabbing ] ]",
		media: [
			"visual",
			"interactive"
		],
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Basic User Interface"
		],
		initial: "auto",
		appliesto: "allElements",
		computed: "asSpecifiedURLsAbsolute",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/cursor"
	};
	var direction = {
		syntax: "ltr | rtl",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Writing Modes"
		],
		initial: "ltr",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/direction"
	};
	var display = {
		syntax: "[ <display-outside> || <display-inside> ] | <display-listitem> | <display-internal> | <display-box> | <display-legacy>",
		media: "all",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Display"
		],
		initial: "inline",
		appliesto: "allElements",
		computed: "asSpecifiedExceptPositionedFloatingAndRootElementsKeywordMaybeDifferent",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/display"
	};
	var filter = {
		syntax: "none | <filter-function-list>",
		media: "visual",
		inherited: false,
		animationType: "filterList",
		percentages: "no",
		groups: [
			"Filter Effects"
		],
		initial: "none",
		appliesto: "allElementsSVGContainerElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/filter"
	};
	var flex = {
		syntax: "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
		media: "visual",
		inherited: false,
		animationType: [
			"flex-grow",
			"flex-shrink",
			"flex-basis"
		],
		percentages: "no",
		groups: [
			"CSS Flexible Box Layout"
		],
		initial: [
			"flex-grow",
			"flex-shrink",
			"flex-basis"
		],
		appliesto: "flexItemsAndInFlowPseudos",
		computed: [
			"flex-grow",
			"flex-shrink",
			"flex-basis"
		],
		order: "orderOfAppearance",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/flex"
	};
	var float = {
		syntax: "left | right | none | inline-start | inline-end",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Positioning"
		],
		initial: "none",
		appliesto: "allElementsNoEffectIfDisplayNone",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/float"
	};
	var font = {
		syntax: "[ [ <'font-style'> || <font-variant-css21> || <'font-weight'> || <'font-stretch'> ]? <'font-size'> [ / <'line-height'> ]? <'font-family'> ] | caption | icon | menu | message-box | small-caption | status-bar",
		media: "visual",
		inherited: true,
		animationType: [
			"font-style",
			"font-variant",
			"font-weight",
			"font-stretch",
			"font-size",
			"line-height",
			"font-family"
		],
		percentages: [
			"font-size",
			"line-height"
		],
		groups: [
			"CSS Fonts"
		],
		initial: [
			"font-style",
			"font-variant",
			"font-weight",
			"font-stretch",
			"font-size",
			"line-height",
			"font-family"
		],
		appliesto: "allElements",
		computed: [
			"font-style",
			"font-variant",
			"font-weight",
			"font-stretch",
			"font-size",
			"line-height",
			"font-family"
		],
		order: "orderOfAppearance",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/font"
	};
	var gap = {
		syntax: "<'row-gap'> <'column-gap'>?",
		media: "visual",
		inherited: false,
		animationType: [
			"row-gap",
			"column-gap"
		],
		percentages: "no",
		groups: [
			"CSS Box Alignment"
		],
		initial: [
			"row-gap",
			"column-gap"
		],
		appliesto: "multiColumnElementsFlexContainersGridContainers",
		computed: [
			"row-gap",
			"column-gap"
		],
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/gap"
	};
	var grid = {
		syntax: "<'grid-template'> | <'grid-template-rows'> / [ auto-flow && dense? ] <'grid-auto-columns'>? | [ auto-flow && dense? ] <'grid-auto-rows'>? / <'grid-template-columns'>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: [
			"grid-template-rows",
			"grid-template-columns",
			"grid-auto-rows",
			"grid-auto-columns"
		],
		groups: [
			"CSS Grid Layout"
		],
		initial: [
			"grid-template-rows",
			"grid-template-columns",
			"grid-template-areas",
			"grid-auto-rows",
			"grid-auto-columns",
			"grid-auto-flow",
			"grid-column-gap",
			"grid-row-gap",
			"column-gap",
			"row-gap"
		],
		appliesto: "gridContainers",
		computed: [
			"grid-template-rows",
			"grid-template-columns",
			"grid-template-areas",
			"grid-auto-rows",
			"grid-auto-columns",
			"grid-auto-flow",
			"grid-column-gap",
			"grid-row-gap",
			"column-gap",
			"row-gap"
		],
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/grid"
	};
	var height = {
		syntax: "auto | <length> | <percentage> | min-content | max-content | fit-content(<length-percentage>)",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "regardingHeightOfGeneratedBoxContainingBlockPercentagesRelativeToContainingBlock",
		groups: [
			"CSS Box Model"
		],
		initial: "auto",
		appliesto: "allElementsButNonReplacedAndTableColumns",
		computed: "percentageAutoOrAbsoluteLength",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/height"
	};
	var hyphens = {
		syntax: "none | manual | auto",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Text"
		],
		initial: "manual",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/hyphens"
	};
	var inset = {
		syntax: "<'top'>{1,4}",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "logicalHeightOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "auto",
		appliesto: "positionedElements",
		computed: "sameAsBoxOffsets",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/inset"
	};
	var isolation = {
		syntax: "auto | isolate",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Compositing and Blending"
		],
		initial: "auto",
		appliesto: "allElementsSVGContainerGraphicsAndGraphicsReferencingElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/isolation"
	};
	var left = {
		syntax: "<length> | <percentage> | auto",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "referToWidthOfContainingBlock",
		groups: [
			"CSS Positioning"
		],
		initial: "auto",
		appliesto: "positionedElements",
		computed: "lengthAbsolutePercentageAsSpecifiedOtherwiseAuto",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/left"
	};
	var margin = {
		syntax: "[ <length> | <percentage> | auto ]{1,4}",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "referToWidthOfContainingBlock",
		groups: [
			"CSS Box Model"
		],
		initial: [
			"margin-bottom",
			"margin-left",
			"margin-right",
			"margin-top"
		],
		appliesto: "allElementsExceptTableDisplayTypes",
		computed: [
			"margin-bottom",
			"margin-left",
			"margin-right",
			"margin-top"
		],
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/margin"
	};
	var mask = {
		syntax: "<mask-layer>#",
		media: "visual",
		inherited: false,
		animationType: [
			"mask-image",
			"mask-mode",
			"mask-repeat",
			"mask-position",
			"mask-clip",
			"mask-origin",
			"mask-size",
			"mask-composite"
		],
		percentages: [
			"mask-position"
		],
		groups: [
			"CSS Masking"
		],
		initial: [
			"mask-image",
			"mask-mode",
			"mask-repeat",
			"mask-position",
			"mask-clip",
			"mask-origin",
			"mask-size",
			"mask-composite"
		],
		appliesto: "allElementsSVGContainerElements",
		computed: [
			"mask-image",
			"mask-mode",
			"mask-repeat",
			"mask-position",
			"mask-clip",
			"mask-origin",
			"mask-size",
			"mask-composite"
		],
		order: "perGrammar",
		stacking: true,
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mask"
	};
	var offset = {
		syntax: "[ <'offset-position'>? [ <'offset-path'> [ <'offset-distance'> || <'offset-rotate'> ]? ]? ]! [ / <'offset-anchor'> ]?",
		media: "visual",
		inherited: false,
		animationType: [
			"offset-position",
			"offset-path",
			"offset-distance",
			"offset-anchor",
			"offset-rotate"
		],
		percentages: [
			"offset-position",
			"offset-distance",
			"offset-anchor"
		],
		groups: [
			"CSS Motion Path"
		],
		initial: [
			"offset-position",
			"offset-path",
			"offset-distance",
			"offset-anchor",
			"offset-rotate"
		],
		appliesto: "transformableElements",
		computed: [
			"offset-position",
			"offset-path",
			"offset-distance",
			"offset-anchor",
			"offset-rotate"
		],
		order: "perGrammar",
		stacking: true,
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/offset"
	};
	var opacity = {
		syntax: "<alpha-value>",
		media: "visual",
		inherited: false,
		animationType: "number",
		percentages: "no",
		groups: [
			"CSS Color"
		],
		initial: "1.0",
		appliesto: "allElements",
		computed: "specifiedValueClipped0To1",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/opacity"
	};
	var order = {
		syntax: "<integer>",
		media: "visual",
		inherited: false,
		animationType: "integer",
		percentages: "no",
		groups: [
			"CSS Flexible Box Layout"
		],
		initial: "0",
		appliesto: "flexItemsGridItemsAbsolutelyPositionedContainerChildren",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/order"
	};
	var orphans = {
		syntax: "<integer>",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Fragmentation"
		],
		initial: "2",
		appliesto: "blockContainerElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/orphans"
	};
	var outline = {
		syntax: "[ <'outline-color'> || <'outline-style'> || <'outline-width'> ]",
		media: [
			"visual",
			"interactive"
		],
		inherited: false,
		animationType: [
			"outline-color",
			"outline-width",
			"outline-style"
		],
		percentages: "no",
		groups: [
			"CSS Basic User Interface"
		],
		initial: [
			"outline-color",
			"outline-style",
			"outline-width"
		],
		appliesto: "allElements",
		computed: [
			"outline-color",
			"outline-width",
			"outline-style"
		],
		order: "orderOfAppearance",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/outline"
	};
	var overflow = {
		syntax: "[ visible | hidden | clip | scroll | auto ]{1,2}",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Overflow"
		],
		initial: "visible",
		appliesto: "blockContainersFlexContainersGridContainers",
		computed: [
			"overflow-x",
			"overflow-y"
		],
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/overflow"
	};
	var padding = {
		syntax: "[ <length> | <percentage> ]{1,4}",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "referToWidthOfContainingBlock",
		groups: [
			"CSS Box Model"
		],
		initial: [
			"padding-bottom",
			"padding-left",
			"padding-right",
			"padding-top"
		],
		appliesto: "allElementsExceptInternalTableDisplayTypes",
		computed: [
			"padding-bottom",
			"padding-left",
			"padding-right",
			"padding-top"
		],
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/padding"
	};
	var perspective = {
		syntax: "none | <length>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "no",
		groups: [
			"CSS Transforms"
		],
		initial: "none",
		appliesto: "transformableElements",
		computed: "absoluteLengthOrNone",
		order: "uniqueOrder",
		stacking: true,
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/perspective"
	};
	var position$1 = {
		syntax: "static | relative | absolute | sticky | fixed",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Positioning"
		],
		initial: "static",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		stacking: true,
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/position"
	};
	var quotes = {
		syntax: "none | auto | [ <string> <string> ]+",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Generated Content"
		],
		initial: "dependsOnUserAgent",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/quotes"
	};
	var resize = {
		syntax: "none | both | horizontal | vertical | block | inline",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Basic User Interface"
		],
		initial: "none",
		appliesto: "elementsWithOverflowNotVisibleAndReplacedElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/resize"
	};
	var right = {
		syntax: "<length> | <percentage> | auto",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "referToWidthOfContainingBlock",
		groups: [
			"CSS Positioning"
		],
		initial: "auto",
		appliesto: "positionedElements",
		computed: "lengthAbsolutePercentageAsSpecifiedOtherwiseAuto",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/right"
	};
	var rotate = {
		syntax: "none | <angle> | [ x | y | z | <number>{3} ] && <angle>",
		media: "visual",
		inherited: false,
		animationType: "transform",
		percentages: "no",
		groups: [
			"CSS Transforms"
		],
		initial: "none",
		appliesto: "transformableElements",
		computed: "asSpecified",
		order: "perGrammar",
		stacking: true,
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/rotate"
	};
	var scale = {
		syntax: "none | <number>{1,3}",
		media: "visual",
		inherited: false,
		animationType: "transform",
		percentages: "no",
		groups: [
			"CSS Transforms"
		],
		initial: "none",
		appliesto: "transformableElements",
		computed: "asSpecified",
		order: "perGrammar",
		stacking: true,
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scale"
	};
	var top = {
		syntax: "<length> | <percentage> | auto",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "referToContainingBlockHeight",
		groups: [
			"CSS Positioning"
		],
		initial: "auto",
		appliesto: "positionedElements",
		computed: "lengthAbsolutePercentageAsSpecifiedOtherwiseAuto",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/top"
	};
	var transform = {
		syntax: "none | <transform-list>",
		media: "visual",
		inherited: false,
		animationType: "transform",
		percentages: "referToSizeOfBoundingBox",
		groups: [
			"CSS Transforms"
		],
		initial: "none",
		appliesto: "transformableElements",
		computed: "asSpecifiedRelativeToAbsoluteLengths",
		order: "uniqueOrder",
		stacking: true,
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/transform"
	};
	var transition = {
		syntax: "<single-transition>#",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Transitions"
		],
		initial: [
			"transition-delay",
			"transition-duration",
			"transition-property",
			"transition-timing-function"
		],
		appliesto: "allElementsAndPseudos",
		computed: [
			"transition-delay",
			"transition-duration",
			"transition-property",
			"transition-timing-function"
		],
		order: "orderOfAppearance",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/transition"
	};
	var translate = {
		syntax: "none | <length-percentage> [ <length-percentage> <length>? ]?",
		media: "visual",
		inherited: false,
		animationType: "transform",
		percentages: "referToSizeOfBoundingBox",
		groups: [
			"CSS Transforms"
		],
		initial: "none",
		appliesto: "transformableElements",
		computed: "asSpecifiedRelativeToAbsoluteLengths",
		order: "perGrammar",
		stacking: true,
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/translate"
	};
	var visibility = {
		syntax: "visible | hidden | collapse",
		media: "visual",
		inherited: true,
		animationType: "visibility",
		percentages: "no",
		groups: [
			"CSS Box Model"
		],
		initial: "visible",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/visibility"
	};
	var widows = {
		syntax: "<integer>",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Fragmentation"
		],
		initial: "2",
		appliesto: "blockContainerElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/widows"
	};
	var width = {
		syntax: "auto | <length> | <percentage> | min-content | max-content | fit-content(<length-percentage>)",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "referToWidthOfContainingBlock",
		groups: [
			"CSS Box Model"
		],
		initial: "auto",
		appliesto: "allElementsButNonReplacedAndTableRows",
		computed: "percentageAutoOrAbsoluteLength",
		order: "lengthOrPercentageBeforeKeywordIfBothPresent",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/width"
	};
	var zoom = {
		syntax: "normal | reset | <number> | <percentage>",
		media: "visual",
		inherited: false,
		animationType: "integer",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "normal",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/zoom"
	};
	var require$$1 = {
		"--*": {
		syntax: "<declaration-value>",
		media: "all",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Variables"
		],
		initial: "seeProse",
		appliesto: "allElements",
		computed: "asSpecifiedWithVarsSubstituted",
		order: "perGrammar",
		status: "experimental",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/--*"
	},
		"-ms-accelerator": {
		syntax: "false | true",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "false",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-accelerator"
	},
		"-ms-block-progression": {
		syntax: "tb | rl | bt | lr",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "tb",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-block-progression"
	},
		"-ms-content-zoom-chaining": {
		syntax: "none | chained",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "none",
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-content-zoom-chaining"
	},
		"-ms-content-zooming": {
		syntax: "none | zoom",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "zoomForTheTopLevelNoneForTheRest",
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-content-zooming"
	},
		"-ms-content-zoom-limit": {
		syntax: "<'-ms-content-zoom-limit-min'> <'-ms-content-zoom-limit-max'>",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: [
			"-ms-content-zoom-limit-max",
			"-ms-content-zoom-limit-min"
		],
		groups: [
			"Microsoft Extensions"
		],
		initial: [
			"-ms-content-zoom-limit-max",
			"-ms-content-zoom-limit-min"
		],
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: [
			"-ms-content-zoom-limit-max",
			"-ms-content-zoom-limit-min"
		],
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-content-zoom-limit"
	},
		"-ms-content-zoom-limit-max": {
		syntax: "<percentage>",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "maxZoomFactor",
		groups: [
			"Microsoft Extensions"
		],
		initial: "400%",
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-content-zoom-limit-max"
	},
		"-ms-content-zoom-limit-min": {
		syntax: "<percentage>",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "minZoomFactor",
		groups: [
			"Microsoft Extensions"
		],
		initial: "100%",
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-content-zoom-limit-min"
	},
		"-ms-content-zoom-snap": {
		syntax: "<'-ms-content-zoom-snap-type'> || <'-ms-content-zoom-snap-points'>",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: [
			"-ms-content-zoom-snap-type",
			"-ms-content-zoom-snap-points"
		],
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: [
			"-ms-content-zoom-snap-type",
			"-ms-content-zoom-snap-points"
		],
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-content-zoom-snap"
	},
		"-ms-content-zoom-snap-points": {
		syntax: "snapInterval( <percentage>, <percentage> ) | snapList( <percentage># )",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "snapInterval(0%, 100%)",
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-content-zoom-snap-points"
	},
		"-ms-content-zoom-snap-type": {
		syntax: "none | proximity | mandatory",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "none",
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-content-zoom-snap-type"
	},
		"-ms-filter": {
		syntax: "<string>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "\"\"",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-filter"
	},
		"-ms-flow-from": {
		syntax: "[ none | <custom-ident> ]#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "none",
		appliesto: "nonReplacedElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-flow-from"
	},
		"-ms-flow-into": {
		syntax: "[ none | <custom-ident> ]#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "none",
		appliesto: "iframeElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-flow-into"
	},
		"-ms-grid-columns": {
		syntax: "none | <track-list> | <auto-track-list>",
		media: "visual",
		inherited: false,
		animationType: "simpleListOfLpcDifferenceLpc",
		percentages: "referToDimensionOfContentArea",
		groups: [
			"CSS Grid Layout"
		],
		initial: "none",
		appliesto: "gridContainers",
		computed: "asSpecifiedRelativeToAbsoluteLengths",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-grid-columns"
	},
		"-ms-grid-rows": {
		syntax: "none | <track-list> | <auto-track-list>",
		media: "visual",
		inherited: false,
		animationType: "simpleListOfLpcDifferenceLpc",
		percentages: "referToDimensionOfContentArea",
		groups: [
			"CSS Grid Layout"
		],
		initial: "none",
		appliesto: "gridContainers",
		computed: "asSpecifiedRelativeToAbsoluteLengths",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-grid-rows"
	},
		"-ms-high-contrast-adjust": {
		syntax: "auto | none",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "auto",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-high-contrast-adjust"
	},
		"-ms-hyphenate-limit-chars": {
		syntax: "auto | <integer>{1,3}",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "auto",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-hyphenate-limit-chars"
	},
		"-ms-hyphenate-limit-lines": {
		syntax: "no-limit | <integer>",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "no-limit",
		appliesto: "blockContainerElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-hyphenate-limit-lines"
	},
		"-ms-hyphenate-limit-zone": {
		syntax: "<percentage> | <length>",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "referToLineBoxWidth",
		groups: [
			"Microsoft Extensions"
		],
		initial: "0",
		appliesto: "blockContainerElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-hyphenate-limit-zone"
	},
		"-ms-ime-align": {
		syntax: "auto | after",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "auto",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-ime-align"
	},
		"-ms-overflow-style": {
		syntax: "auto | none | scrollbar | -ms-autohiding-scrollbar",
		media: "interactive",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "auto",
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-overflow-style"
	},
		"-ms-scrollbar-3dlight-color": {
		syntax: "<color>",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "dependsOnUserAgent",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-scrollbar-3dlight-color"
	},
		"-ms-scrollbar-arrow-color": {
		syntax: "<color>",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "ButtonText",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-scrollbar-arrow-color"
	},
		"-ms-scrollbar-base-color": {
		syntax: "<color>",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "dependsOnUserAgent",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-scrollbar-base-color"
	},
		"-ms-scrollbar-darkshadow-color": {
		syntax: "<color>",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "ThreeDDarkShadow",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-scrollbar-darkshadow-color"
	},
		"-ms-scrollbar-face-color": {
		syntax: "<color>",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "ThreeDFace",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-scrollbar-face-color"
	},
		"-ms-scrollbar-highlight-color": {
		syntax: "<color>",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "ThreeDHighlight",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-scrollbar-highlight-color"
	},
		"-ms-scrollbar-shadow-color": {
		syntax: "<color>",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "ThreeDDarkShadow",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-scrollbar-shadow-color"
	},
		"-ms-scrollbar-track-color": {
		syntax: "<color>",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "Scrollbar",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-scrollbar-track-color"
	},
		"-ms-scroll-chaining": {
		syntax: "chained | none",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "chained",
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-scroll-chaining"
	},
		"-ms-scroll-limit": {
		syntax: "<'-ms-scroll-limit-x-min'> <'-ms-scroll-limit-y-min'> <'-ms-scroll-limit-x-max'> <'-ms-scroll-limit-y-max'>",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: [
			"-ms-scroll-limit-x-min",
			"-ms-scroll-limit-y-min",
			"-ms-scroll-limit-x-max",
			"-ms-scroll-limit-y-max"
		],
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: [
			"-ms-scroll-limit-x-min",
			"-ms-scroll-limit-y-min",
			"-ms-scroll-limit-x-max",
			"-ms-scroll-limit-y-max"
		],
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-scroll-limit"
	},
		"-ms-scroll-limit-x-max": {
		syntax: "auto | <length>",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "auto",
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-scroll-limit-x-max"
	},
		"-ms-scroll-limit-x-min": {
		syntax: "<length>",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "0",
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-scroll-limit-x-min"
	},
		"-ms-scroll-limit-y-max": {
		syntax: "auto | <length>",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "auto",
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-scroll-limit-y-max"
	},
		"-ms-scroll-limit-y-min": {
		syntax: "<length>",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "0",
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-scroll-limit-y-min"
	},
		"-ms-scroll-rails": {
		syntax: "none | railed",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "railed",
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-scroll-rails"
	},
		"-ms-scroll-snap-points-x": {
		syntax: "snapInterval( <length-percentage>, <length-percentage> ) | snapList( <length-percentage># )",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "snapInterval(0px, 100%)",
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-scroll-snap-points-x"
	},
		"-ms-scroll-snap-points-y": {
		syntax: "snapInterval( <length-percentage>, <length-percentage> ) | snapList( <length-percentage># )",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "snapInterval(0px, 100%)",
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-scroll-snap-points-y"
	},
		"-ms-scroll-snap-type": {
		syntax: "none | proximity | mandatory",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "none",
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-scroll-snap-type"
	},
		"-ms-scroll-snap-x": {
		syntax: "<'-ms-scroll-snap-type'> <'-ms-scroll-snap-points-x'>",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: [
			"-ms-scroll-snap-type",
			"-ms-scroll-snap-points-x"
		],
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: [
			"-ms-scroll-snap-type",
			"-ms-scroll-snap-points-x"
		],
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-scroll-snap-x"
	},
		"-ms-scroll-snap-y": {
		syntax: "<'-ms-scroll-snap-type'> <'-ms-scroll-snap-points-y'>",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: [
			"-ms-scroll-snap-type",
			"-ms-scroll-snap-points-y"
		],
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: [
			"-ms-scroll-snap-type",
			"-ms-scroll-snap-points-y"
		],
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-scroll-snap-y"
	},
		"-ms-scroll-translation": {
		syntax: "none | vertical-to-horizontal",
		media: "interactive",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-scroll-translation"
	},
		"-ms-text-autospace": {
		syntax: "none | ideograph-alpha | ideograph-numeric | ideograph-parenthesis | ideograph-space",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-text-autospace"
	},
		"-ms-touch-select": {
		syntax: "grippers | none",
		media: "interactive",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "grippers",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-touch-select"
	},
		"-ms-user-select": {
		syntax: "none | element | text",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "text",
		appliesto: "nonReplacedElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-user-select"
	},
		"-ms-wrap-flow": {
		syntax: "auto | both | start | end | maximum | clear",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "auto",
		appliesto: "blockLevelElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-wrap-flow"
	},
		"-ms-wrap-margin": {
		syntax: "<length>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "0",
		appliesto: "exclusionElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-wrap-margin"
	},
		"-ms-wrap-through": {
		syntax: "wrap | none",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Microsoft Extensions"
		],
		initial: "wrap",
		appliesto: "blockLevelElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-ms-wrap-through"
	},
		"-moz-appearance": {
		syntax: "none | button | button-arrow-down | button-arrow-next | button-arrow-previous | button-arrow-up | button-bevel | button-focus | caret | checkbox | checkbox-container | checkbox-label | checkmenuitem | dualbutton | groupbox | listbox | listitem | menuarrow | menubar | menucheckbox | menuimage | menuitem | menuitemtext | menulist | menulist-button | menulist-text | menulist-textfield | menupopup | menuradio | menuseparator | meterbar | meterchunk | progressbar | progressbar-vertical | progresschunk | progresschunk-vertical | radio | radio-container | radio-label | radiomenuitem | range | range-thumb | resizer | resizerpanel | scale-horizontal | scalethumbend | scalethumb-horizontal | scalethumbstart | scalethumbtick | scalethumb-vertical | scale-vertical | scrollbarbutton-down | scrollbarbutton-left | scrollbarbutton-right | scrollbarbutton-up | scrollbarthumb-horizontal | scrollbarthumb-vertical | scrollbartrack-horizontal | scrollbartrack-vertical | searchfield | separator | sheet | spinner | spinner-downbutton | spinner-textfield | spinner-upbutton | splitter | statusbar | statusbarpanel | tab | tabpanel | tabpanels | tab-scroll-arrow-back | tab-scroll-arrow-forward | textfield | textfield-multiline | toolbar | toolbarbutton | toolbarbutton-dropdown | toolbargripper | toolbox | tooltip | treeheader | treeheadercell | treeheadersortarrow | treeitem | treeline | treetwisty | treetwistyopen | treeview | -moz-mac-unified-toolbar | -moz-win-borderless-glass | -moz-win-browsertabbar-toolbox | -moz-win-communicationstext | -moz-win-communications-toolbox | -moz-win-exclude-glass | -moz-win-glass | -moz-win-mediatext | -moz-win-media-toolbox | -moz-window-button-box | -moz-window-button-box-maximized | -moz-window-button-close | -moz-window-button-maximize | -moz-window-button-minimize | -moz-window-button-restore | -moz-window-frame-bottom | -moz-window-frame-left | -moz-window-frame-right | -moz-window-titlebar | -moz-window-titlebar-maximized",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions",
			"WebKit Extensions"
		],
		initial: "noneButOverriddenInUserAgentCSS",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/appearance"
	},
		"-moz-binding": {
		syntax: "<url> | none",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions"
		],
		initial: "none",
		appliesto: "allElementsExceptGeneratedContentOrPseudoElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-moz-binding"
	},
		"-moz-border-bottom-colors": {
		syntax: "<color>+ | none",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-moz-border-bottom-colors"
	},
		"-moz-border-left-colors": {
		syntax: "<color>+ | none",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-moz-border-left-colors"
	},
		"-moz-border-right-colors": {
		syntax: "<color>+ | none",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-moz-border-right-colors"
	},
		"-moz-border-top-colors": {
		syntax: "<color>+ | none",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-moz-border-top-colors"
	},
		"-moz-context-properties": {
		syntax: "none | [ fill | fill-opacity | stroke | stroke-opacity ]#",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions"
		],
		initial: "none",
		appliesto: "allElementsThatCanReferenceImages",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-moz-context-properties"
	},
		"-moz-float-edge": {
		syntax: "border-box | content-box | margin-box | padding-box",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions"
		],
		initial: "content-box",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-moz-float-edge"
	},
		"-moz-force-broken-image-icon": {
		syntax: "<integer [0,1]>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions"
		],
		initial: "0",
		appliesto: "images",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-moz-force-broken-image-icon"
	},
		"-moz-image-region": {
		syntax: "<shape> | auto",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions"
		],
		initial: "auto",
		appliesto: "xulImageElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-moz-image-region"
	},
		"-moz-orient": {
		syntax: "inline | block | horizontal | vertical",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions"
		],
		initial: "inline",
		appliesto: "anyElementEffectOnProgressAndMeter",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-moz-orient"
	},
		"-moz-outline-radius": {
		syntax: "<outline-radius>{1,4} [ / <outline-radius>{1,4} ]?",
		media: "visual",
		inherited: false,
		animationType: [
			"-moz-outline-radius-topleft",
			"-moz-outline-radius-topright",
			"-moz-outline-radius-bottomright",
			"-moz-outline-radius-bottomleft"
		],
		percentages: [
			"-moz-outline-radius-topleft",
			"-moz-outline-radius-topright",
			"-moz-outline-radius-bottomright",
			"-moz-outline-radius-bottomleft"
		],
		groups: [
			"Mozilla Extensions"
		],
		initial: [
			"-moz-outline-radius-topleft",
			"-moz-outline-radius-topright",
			"-moz-outline-radius-bottomright",
			"-moz-outline-radius-bottomleft"
		],
		appliesto: "allElements",
		computed: [
			"-moz-outline-radius-topleft",
			"-moz-outline-radius-topright",
			"-moz-outline-radius-bottomright",
			"-moz-outline-radius-bottomleft"
		],
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-moz-outline-radius"
	},
		"-moz-outline-radius-bottomleft": {
		syntax: "<outline-radius>",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "referToDimensionOfBorderBox",
		groups: [
			"Mozilla Extensions"
		],
		initial: "0",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-moz-outline-radius-bottomleft"
	},
		"-moz-outline-radius-bottomright": {
		syntax: "<outline-radius>",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "referToDimensionOfBorderBox",
		groups: [
			"Mozilla Extensions"
		],
		initial: "0",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-moz-outline-radius-bottomright"
	},
		"-moz-outline-radius-topleft": {
		syntax: "<outline-radius>",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "referToDimensionOfBorderBox",
		groups: [
			"Mozilla Extensions"
		],
		initial: "0",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-moz-outline-radius-topleft"
	},
		"-moz-outline-radius-topright": {
		syntax: "<outline-radius>",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "referToDimensionOfBorderBox",
		groups: [
			"Mozilla Extensions"
		],
		initial: "0",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-moz-outline-radius-topright"
	},
		"-moz-stack-sizing": {
		syntax: "ignore | stretch-to-fit",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions"
		],
		initial: "stretch-to-fit",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-moz-stack-sizing"
	},
		"-moz-text-blink": {
		syntax: "none | blink",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-moz-text-blink"
	},
		"-moz-user-focus": {
		syntax: "ignore | normal | select-after | select-before | select-menu | select-same | select-all | none",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-moz-user-focus"
	},
		"-moz-user-input": {
		syntax: "auto | none | enabled | disabled",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions"
		],
		initial: "auto",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-moz-user-input"
	},
		"-moz-user-modify": {
		syntax: "read-only | read-write | write-only",
		media: "interactive",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions"
		],
		initial: "read-only",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-moz-user-modify"
	},
		"-moz-window-dragging": {
		syntax: "drag | no-drag",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions"
		],
		initial: "drag",
		appliesto: "allElementsCreatingNativeWindows",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-moz-window-dragging"
	},
		"-moz-window-shadow": {
		syntax: "default | menu | tooltip | sheet | none",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions"
		],
		initial: "default",
		appliesto: "allElementsCreatingNativeWindows",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-moz-window-shadow"
	},
		"-webkit-appearance": {
		syntax: "none | button | button-bevel | caret | checkbox | default-button | inner-spin-button | listbox | listitem | media-controls-background | media-controls-fullscreen-background | media-current-time-display | media-enter-fullscreen-button | media-exit-fullscreen-button | media-fullscreen-button | media-mute-button | media-overlay-play-button | media-play-button | media-seek-back-button | media-seek-forward-button | media-slider | media-sliderthumb | media-time-remaining-display | media-toggle-closed-captions-button | media-volume-slider | media-volume-slider-container | media-volume-sliderthumb | menulist | menulist-button | menulist-text | menulist-textfield | meter | progress-bar | progress-bar-value | push-button | radio | searchfield | searchfield-cancel-button | searchfield-decoration | searchfield-results-button | searchfield-results-decoration | slider-horizontal | slider-vertical | sliderthumb-horizontal | sliderthumb-vertical | square-button | textarea | textfield | -apple-pay-button",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"WebKit Extensions"
		],
		initial: "noneButOverriddenInUserAgentCSS",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/appearance"
	},
		"-webkit-border-before": {
		syntax: "<'border-width'> || <'border-style'> || <'color'>",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: [
			"-webkit-border-before-width"
		],
		groups: [
			"WebKit Extensions"
		],
		initial: [
			"border-width",
			"border-style",
			"color"
		],
		appliesto: "allElements",
		computed: [
			"border-width",
			"border-style",
			"color"
		],
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-webkit-border-before"
	},
		"-webkit-border-before-color": {
		syntax: "<'color'>",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"WebKit Extensions"
		],
		initial: "currentcolor",
		appliesto: "allElements",
		computed: "computedColor",
		order: "uniqueOrder",
		status: "nonstandard"
	},
		"-webkit-border-before-style": {
		syntax: "<'border-style'>",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"WebKit Extensions"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard"
	},
		"-webkit-border-before-width": {
		syntax: "<'border-width'>",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "logicalWidthOfContainingBlock",
		groups: [
			"WebKit Extensions"
		],
		initial: "medium",
		appliesto: "allElements",
		computed: "absoluteLengthZeroIfBorderStyleNoneOrHidden",
		order: "uniqueOrder",
		status: "nonstandard"
	},
		"-webkit-box-reflect": {
		syntax: "[ above | below | right | left ]? <length>? <image>?",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"WebKit Extensions"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-webkit-box-reflect"
	},
		"-webkit-line-clamp": {
		syntax: "none | <integer>",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "no",
		groups: [
			"WebKit Extensions",
			"CSS Overflow"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-webkit-line-clamp"
	},
		"-webkit-mask": {
		syntax: "[ <mask-reference> || <position> [ / <bg-size> ]? || <repeat-style> || [ <box> | border | padding | content | text ] || [ <box> | border | padding | content ] ]#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"WebKit Extensions"
		],
		initial: [
			"-webkit-mask-image",
			"-webkit-mask-repeat",
			"-webkit-mask-attachment",
			"-webkit-mask-position",
			"-webkit-mask-origin",
			"-webkit-mask-clip"
		],
		appliesto: "allElements",
		computed: [
			"-webkit-mask-image",
			"-webkit-mask-repeat",
			"-webkit-mask-attachment",
			"-webkit-mask-position",
			"-webkit-mask-origin",
			"-webkit-mask-clip"
		],
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mask"
	},
		"-webkit-mask-attachment": {
		syntax: "<attachment>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"WebKit Extensions"
		],
		initial: "scroll",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "orderOfAppearance",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-webkit-mask-attachment"
	},
		"-webkit-mask-clip": {
		syntax: "[ <box> | border | padding | content | text ]#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"WebKit Extensions"
		],
		initial: "border",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "orderOfAppearance",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mask-clip"
	},
		"-webkit-mask-composite": {
		syntax: "<composite-style>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"WebKit Extensions"
		],
		initial: "source-over",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "orderOfAppearance",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-webkit-mask-composite"
	},
		"-webkit-mask-image": {
		syntax: "<mask-reference>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"WebKit Extensions"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "absoluteURIOrNone",
		order: "orderOfAppearance",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mask-image"
	},
		"-webkit-mask-origin": {
		syntax: "[ <box> | border | padding | content ]#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"WebKit Extensions"
		],
		initial: "padding",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "orderOfAppearance",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mask-origin"
	},
		"-webkit-mask-position": {
		syntax: "<position>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "referToSizeOfElement",
		groups: [
			"WebKit Extensions"
		],
		initial: "0% 0%",
		appliesto: "allElements",
		computed: "absoluteLengthOrPercentage",
		order: "orderOfAppearance",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mask-position"
	},
		"-webkit-mask-position-x": {
		syntax: "[ <length-percentage> | left | center | right ]#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "referToSizeOfElement",
		groups: [
			"WebKit Extensions"
		],
		initial: "0%",
		appliesto: "allElements",
		computed: "absoluteLengthOrPercentage",
		order: "orderOfAppearance",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-webkit-mask-position-x"
	},
		"-webkit-mask-position-y": {
		syntax: "[ <length-percentage> | top | center | bottom ]#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "referToSizeOfElement",
		groups: [
			"WebKit Extensions"
		],
		initial: "0%",
		appliesto: "allElements",
		computed: "absoluteLengthOrPercentage",
		order: "orderOfAppearance",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-webkit-mask-position-y"
	},
		"-webkit-mask-repeat": {
		syntax: "<repeat-style>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"WebKit Extensions"
		],
		initial: "repeat",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "orderOfAppearance",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mask-repeat"
	},
		"-webkit-mask-repeat-x": {
		syntax: "repeat | no-repeat | space | round",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"WebKit Extensions"
		],
		initial: "repeat",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "orderOfAppearance",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-webkit-mask-repeat-x"
	},
		"-webkit-mask-repeat-y": {
		syntax: "repeat | no-repeat | space | round",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"WebKit Extensions"
		],
		initial: "repeat",
		appliesto: "allElements",
		computed: "absoluteLengthOrPercentage",
		order: "orderOfAppearance",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-webkit-mask-repeat-y"
	},
		"-webkit-mask-size": {
		syntax: "<bg-size>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "relativeToBackgroundPositioningArea",
		groups: [
			"WebKit Extensions"
		],
		initial: "auto auto",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "orderOfAppearance",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mask-size"
	},
		"-webkit-overflow-scrolling": {
		syntax: "auto | touch",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"WebKit Extensions"
		],
		initial: "auto",
		appliesto: "scrollingBoxes",
		computed: "asSpecified",
		order: "orderOfAppearance",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-webkit-overflow-scrolling"
	},
		"-webkit-tap-highlight-color": {
		syntax: "<color>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"WebKit Extensions"
		],
		initial: "black",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-webkit-tap-highlight-color"
	},
		"-webkit-text-fill-color": {
		syntax: "<color>",
		media: "visual",
		inherited: true,
		animationType: "color",
		percentages: "no",
		groups: [
			"WebKit Extensions"
		],
		initial: "currentcolor",
		appliesto: "allElements",
		computed: "computedColor",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-webkit-text-fill-color"
	},
		"-webkit-text-stroke": {
		syntax: "<length> || <color>",
		media: "visual",
		inherited: true,
		animationType: [
			"-webkit-text-stroke-width",
			"-webkit-text-stroke-color"
		],
		percentages: "no",
		groups: [
			"WebKit Extensions"
		],
		initial: [
			"-webkit-text-stroke-width",
			"-webkit-text-stroke-color"
		],
		appliesto: "allElements",
		computed: [
			"-webkit-text-stroke-width",
			"-webkit-text-stroke-color"
		],
		order: "canonicalOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-webkit-text-stroke"
	},
		"-webkit-text-stroke-color": {
		syntax: "<color>",
		media: "visual",
		inherited: true,
		animationType: "color",
		percentages: "no",
		groups: [
			"WebKit Extensions"
		],
		initial: "currentcolor",
		appliesto: "allElements",
		computed: "computedColor",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-webkit-text-stroke-color"
	},
		"-webkit-text-stroke-width": {
		syntax: "<length>",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"WebKit Extensions"
		],
		initial: "0",
		appliesto: "allElements",
		computed: "absoluteLength",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-webkit-text-stroke-width"
	},
		"-webkit-touch-callout": {
		syntax: "default | none",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"WebKit Extensions"
		],
		initial: "default",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/-webkit-touch-callout"
	},
		"-webkit-user-modify": {
		syntax: "read-only | read-write | read-write-plaintext-only",
		media: "interactive",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"WebKit Extensions"
		],
		initial: "read-only",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard"
	},
		"align-content": {
		syntax: "normal | <baseline-position> | <content-distribution> | <overflow-position>? <content-position>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Box Alignment"
		],
		initial: "normal",
		appliesto: "multilineFlexContainers",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/align-content"
	},
		"align-items": {
		syntax: "normal | stretch | <baseline-position> | [ <overflow-position>? <self-position> ]",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Box Alignment"
		],
		initial: "normal",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/align-items"
	},
		"align-self": {
		syntax: "auto | normal | stretch | <baseline-position> | <overflow-position>? <self-position>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Box Alignment"
		],
		initial: "auto",
		appliesto: "flexItemsGridItemsAndAbsolutelyPositionedBoxes",
		computed: "autoOnAbsolutelyPositionedElementsValueOfAlignItemsOnParent",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/align-self"
	},
		"align-tracks": {
		syntax: "[ normal | <baseline-position> | <content-distribution> | <overflow-position>? <content-position> ]#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Grid Layout"
		],
		initial: "normal",
		appliesto: "gridContainersWithMasonryLayoutInTheirBlockAxis",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "experimental",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/align-tracks"
	},
		all: all,
		animation: animation,
		"animation-delay": {
		syntax: "<time>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Animations"
		],
		initial: "0s",
		appliesto: "allElementsAndPseudos",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/animation-delay"
	},
		"animation-direction": {
		syntax: "<single-animation-direction>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Animations"
		],
		initial: "normal",
		appliesto: "allElementsAndPseudos",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/animation-direction"
	},
		"animation-duration": {
		syntax: "<time>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Animations"
		],
		initial: "0s",
		appliesto: "allElementsAndPseudos",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/animation-duration"
	},
		"animation-fill-mode": {
		syntax: "<single-animation-fill-mode>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Animations"
		],
		initial: "none",
		appliesto: "allElementsAndPseudos",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/animation-fill-mode"
	},
		"animation-iteration-count": {
		syntax: "<single-animation-iteration-count>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Animations"
		],
		initial: "1",
		appliesto: "allElementsAndPseudos",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/animation-iteration-count"
	},
		"animation-name": {
		syntax: "[ none | <keyframes-name> ]#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Animations"
		],
		initial: "none",
		appliesto: "allElementsAndPseudos",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/animation-name"
	},
		"animation-play-state": {
		syntax: "<single-animation-play-state>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Animations"
		],
		initial: "running",
		appliesto: "allElementsAndPseudos",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/animation-play-state"
	},
		"animation-timing-function": {
		syntax: "<timing-function>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Animations"
		],
		initial: "ease",
		appliesto: "allElementsAndPseudos",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/animation-timing-function"
	},
		appearance: appearance,
		"aspect-ratio": {
		syntax: "auto | <ratio>",
		media: "all",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Basic User Interface"
		],
		initial: "auto",
		appliesto: "allElementsExceptInlineBoxesAndInternalRubyOrTableBoxes",
		computed: "asSpecified",
		order: "perGrammar",
		status: "experimental",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/aspect-ratio"
	},
		azimuth: azimuth,
		"backdrop-filter": {
		syntax: "none | <filter-function-list>",
		media: "visual",
		inherited: false,
		animationType: "filterList",
		percentages: "no",
		groups: [
			"Filter Effects"
		],
		initial: "none",
		appliesto: "allElementsSVGContainerElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/backdrop-filter"
	},
		"backface-visibility": {
		syntax: "visible | hidden",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Transforms"
		],
		initial: "visible",
		appliesto: "transformableElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/backface-visibility"
	},
		background: background,
		"background-attachment": {
		syntax: "<attachment>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "scroll",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/background-attachment"
	},
		"background-blend-mode": {
		syntax: "<blend-mode>#",
		media: "none",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Compositing and Blending"
		],
		initial: "normal",
		appliesto: "allElementsSVGContainerGraphicsAndGraphicsReferencingElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/background-blend-mode"
	},
		"background-clip": {
		syntax: "<box>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "border-box",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/background-clip"
	},
		"background-color": {
		syntax: "<color>",
		media: "visual",
		inherited: false,
		animationType: "color",
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "transparent",
		appliesto: "allElements",
		computed: "computedColor",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/background-color"
	},
		"background-image": {
		syntax: "<bg-image>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecifiedURLsAbsolute",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/background-image"
	},
		"background-origin": {
		syntax: "<box>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "padding-box",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/background-origin"
	},
		"background-position": {
		syntax: "<bg-position>#",
		media: "visual",
		inherited: false,
		animationType: "repeatableListOfSimpleListOfLpc",
		percentages: "referToSizeOfBackgroundPositioningAreaMinusBackgroundImageSize",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "0% 0%",
		appliesto: "allElements",
		computed: "listEachItemTwoKeywordsOriginOffsets",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/background-position"
	},
		"background-position-x": {
		syntax: "[ center | [ [ left | right | x-start | x-end ]? <length-percentage>? ]! ]#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "referToWidthOfBackgroundPositioningAreaMinusBackgroundImageHeight",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "left",
		appliesto: "allElements",
		computed: "listEachItemConsistingOfAbsoluteLengthPercentageAndOrigin",
		order: "uniqueOrder",
		status: "experimental",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/background-position-x"
	},
		"background-position-y": {
		syntax: "[ center | [ [ top | bottom | y-start | y-end ]? <length-percentage>? ]! ]#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "referToHeightOfBackgroundPositioningAreaMinusBackgroundImageHeight",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "top",
		appliesto: "allElements",
		computed: "listEachItemConsistingOfAbsoluteLengthPercentageAndOrigin",
		order: "uniqueOrder",
		status: "experimental",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/background-position-y"
	},
		"background-repeat": {
		syntax: "<repeat-style>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "repeat",
		appliesto: "allElements",
		computed: "listEachItemHasTwoKeywordsOnePerDimension",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/background-repeat"
	},
		"background-size": {
		syntax: "<bg-size>#",
		media: "visual",
		inherited: false,
		animationType: "repeatableListOfSimpleListOfLpc",
		percentages: "relativeToBackgroundPositioningArea",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "auto auto",
		appliesto: "allElements",
		computed: "asSpecifiedRelativeToAbsoluteLengths",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/background-size"
	},
		"block-overflow": {
		syntax: "clip | ellipsis | <string>",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Overflow"
		],
		initial: "clip",
		appliesto: "blockContainers",
		computed: "asSpecified",
		order: "perGrammar",
		status: "experimental"
	},
		"block-size": {
		syntax: "<'width'>",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "blockSizeOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "auto",
		appliesto: "sameAsWidthAndHeight",
		computed: "sameAsWidthAndHeight",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/block-size"
	},
		border: border,
		"border-block": {
		syntax: "<'border-top-width'> || <'border-top-style'> || <'color'>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Logical Properties"
		],
		initial: [
			"border-top-width",
			"border-top-style",
			"border-top-color"
		],
		appliesto: "allElements",
		computed: [
			"border-top-width",
			"border-top-style",
			"border-top-color"
		],
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-block"
	},
		"border-block-color": {
		syntax: "<'border-top-color'>{1,2}",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Logical Properties"
		],
		initial: "currentcolor",
		appliesto: "allElements",
		computed: "computedColor",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-block-color"
	},
		"border-block-style": {
		syntax: "<'border-top-style'>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Logical Properties"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-block-style"
	},
		"border-block-width": {
		syntax: "<'border-top-width'>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "logicalWidthOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "medium",
		appliesto: "allElements",
		computed: "absoluteLengthZeroIfBorderStyleNoneOrHidden",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-block-width"
	},
		"border-block-end": {
		syntax: "<'border-top-width'> || <'border-top-style'> || <'color'>",
		media: "visual",
		inherited: false,
		animationType: [
			"border-block-end-color",
			"border-block-end-style",
			"border-block-end-width"
		],
		percentages: "no",
		groups: [
			"CSS Logical Properties"
		],
		initial: [
			"border-top-width",
			"border-top-style",
			"border-top-color"
		],
		appliesto: "allElements",
		computed: [
			"border-top-width",
			"border-top-style",
			"border-top-color"
		],
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-block-end"
	},
		"border-block-end-color": {
		syntax: "<'border-top-color'>",
		media: "visual",
		inherited: false,
		animationType: "color",
		percentages: "no",
		groups: [
			"CSS Logical Properties"
		],
		initial: "currentcolor",
		appliesto: "allElements",
		computed: "computedColor",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-block-end-color"
	},
		"border-block-end-style": {
		syntax: "<'border-top-style'>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Logical Properties"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-block-end-style"
	},
		"border-block-end-width": {
		syntax: "<'border-top-width'>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "logicalWidthOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "medium",
		appliesto: "allElements",
		computed: "absoluteLengthZeroIfBorderStyleNoneOrHidden",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-block-end-width"
	},
		"border-block-start": {
		syntax: "<'border-top-width'> || <'border-top-style'> || <'color'>",
		media: "visual",
		inherited: false,
		animationType: [
			"border-block-start-color",
			"border-block-start-style",
			"border-block-start-width"
		],
		percentages: "no",
		groups: [
			"CSS Logical Properties"
		],
		initial: [
			"border-width",
			"border-style",
			"color"
		],
		appliesto: "allElements",
		computed: [
			"border-width",
			"border-style",
			"border-block-start-color"
		],
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-block-start"
	},
		"border-block-start-color": {
		syntax: "<'border-top-color'>",
		media: "visual",
		inherited: false,
		animationType: "color",
		percentages: "no",
		groups: [
			"CSS Logical Properties"
		],
		initial: "currentcolor",
		appliesto: "allElements",
		computed: "computedColor",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-block-start-color"
	},
		"border-block-start-style": {
		syntax: "<'border-top-style'>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Logical Properties"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-block-start-style"
	},
		"border-block-start-width": {
		syntax: "<'border-top-width'>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "logicalWidthOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "medium",
		appliesto: "allElements",
		computed: "absoluteLengthZeroIfBorderStyleNoneOrHidden",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-block-start-width"
	},
		"border-bottom": {
		syntax: "<line-width> || <line-style> || <color>",
		media: "visual",
		inherited: false,
		animationType: [
			"border-bottom-color",
			"border-bottom-style",
			"border-bottom-width"
		],
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: [
			"border-bottom-width",
			"border-bottom-style",
			"border-bottom-color"
		],
		appliesto: "allElements",
		computed: [
			"border-bottom-width",
			"border-bottom-style",
			"border-bottom-color"
		],
		order: "orderOfAppearance",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-bottom"
	},
		"border-bottom-color": {
		syntax: "<'border-top-color'>",
		media: "visual",
		inherited: false,
		animationType: "color",
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "currentcolor",
		appliesto: "allElements",
		computed: "computedColor",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-bottom-color"
	},
		"border-bottom-left-radius": {
		syntax: "<length-percentage>{1,2}",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "referToDimensionOfBorderBox",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "0",
		appliesto: "allElementsUAsNotRequiredWhenCollapse",
		computed: "twoAbsoluteLengthOrPercentages",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-bottom-left-radius"
	},
		"border-bottom-right-radius": {
		syntax: "<length-percentage>{1,2}",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "referToDimensionOfBorderBox",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "0",
		appliesto: "allElementsUAsNotRequiredWhenCollapse",
		computed: "twoAbsoluteLengthOrPercentages",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-bottom-right-radius"
	},
		"border-bottom-style": {
		syntax: "<line-style>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-bottom-style"
	},
		"border-bottom-width": {
		syntax: "<line-width>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "medium",
		appliesto: "allElements",
		computed: "absoluteLengthOr0IfBorderBottomStyleNoneOrHidden",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-bottom-width"
	},
		"border-collapse": {
		syntax: "collapse | separate",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Table"
		],
		initial: "separate",
		appliesto: "tableElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-collapse"
	},
		"border-color": {
		syntax: "<color>{1,4}",
		media: "visual",
		inherited: false,
		animationType: [
			"border-bottom-color",
			"border-left-color",
			"border-right-color",
			"border-top-color"
		],
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: [
			"border-top-color",
			"border-right-color",
			"border-bottom-color",
			"border-left-color"
		],
		appliesto: "allElements",
		computed: [
			"border-bottom-color",
			"border-left-color",
			"border-right-color",
			"border-top-color"
		],
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-color"
	},
		"border-end-end-radius": {
		syntax: "<length-percentage>{1,2}",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "referToDimensionOfBorderBox",
		groups: [
			"CSS Logical Properties"
		],
		initial: "0",
		appliesto: "allElementsUAsNotRequiredWhenCollapse",
		computed: "twoAbsoluteLengthOrPercentages",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-end-end-radius"
	},
		"border-end-start-radius": {
		syntax: "<length-percentage>{1,2}",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "referToDimensionOfBorderBox",
		groups: [
			"CSS Logical Properties"
		],
		initial: "0",
		appliesto: "allElementsUAsNotRequiredWhenCollapse",
		computed: "twoAbsoluteLengthOrPercentages",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-end-start-radius"
	},
		"border-image": {
		syntax: "<'border-image-source'> || <'border-image-slice'> [ / <'border-image-width'> | / <'border-image-width'>? / <'border-image-outset'> ]? || <'border-image-repeat'>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: [
			"border-image-slice",
			"border-image-width"
		],
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: [
			"border-image-source",
			"border-image-slice",
			"border-image-width",
			"border-image-outset",
			"border-image-repeat"
		],
		appliesto: "allElementsExceptTableElementsWhenCollapse",
		computed: [
			"border-image-outset",
			"border-image-repeat",
			"border-image-slice",
			"border-image-source",
			"border-image-width"
		],
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-image"
	},
		"border-image-outset": {
		syntax: "[ <length> | <number> ]{1,4}",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "0",
		appliesto: "allElementsExceptTableElementsWhenCollapse",
		computed: "asSpecifiedRelativeToAbsoluteLengths",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-image-outset"
	},
		"border-image-repeat": {
		syntax: "[ stretch | repeat | round | space ]{1,2}",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "stretch",
		appliesto: "allElementsExceptTableElementsWhenCollapse",
		computed: "asSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-image-repeat"
	},
		"border-image-slice": {
		syntax: "<number-percentage>{1,4} && fill?",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "referToSizeOfBorderImage",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "100%",
		appliesto: "allElementsExceptTableElementsWhenCollapse",
		computed: "oneToFourPercentagesOrAbsoluteLengthsPlusFill",
		order: "percentagesOrLengthsFollowedByFill",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-image-slice"
	},
		"border-image-source": {
		syntax: "none | <image>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "none",
		appliesto: "allElementsExceptTableElementsWhenCollapse",
		computed: "noneOrImageWithAbsoluteURI",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-image-source"
	},
		"border-image-width": {
		syntax: "[ <length-percentage> | <number> | auto ]{1,4}",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "referToWidthOrHeightOfBorderImageArea",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "1",
		appliesto: "allElementsExceptTableElementsWhenCollapse",
		computed: "asSpecifiedRelativeToAbsoluteLengths",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-image-width"
	},
		"border-inline": {
		syntax: "<'border-top-width'> || <'border-top-style'> || <'color'>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Logical Properties"
		],
		initial: [
			"border-top-width",
			"border-top-style",
			"border-top-color"
		],
		appliesto: "allElements",
		computed: [
			"border-top-width",
			"border-top-style",
			"border-top-color"
		],
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-inline"
	},
		"border-inline-end": {
		syntax: "<'border-top-width'> || <'border-top-style'> || <'color'>",
		media: "visual",
		inherited: false,
		animationType: [
			"border-inline-end-color",
			"border-inline-end-style",
			"border-inline-end-width"
		],
		percentages: "no",
		groups: [
			"CSS Logical Properties"
		],
		initial: [
			"border-width",
			"border-style",
			"color"
		],
		appliesto: "allElements",
		computed: [
			"border-width",
			"border-style",
			"border-inline-end-color"
		],
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-inline-end"
	},
		"border-inline-color": {
		syntax: "<'border-top-color'>{1,2}",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Logical Properties"
		],
		initial: "currentcolor",
		appliesto: "allElements",
		computed: "computedColor",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-inline-color"
	},
		"border-inline-style": {
		syntax: "<'border-top-style'>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Logical Properties"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-inline-style"
	},
		"border-inline-width": {
		syntax: "<'border-top-width'>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "logicalWidthOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "medium",
		appliesto: "allElements",
		computed: "absoluteLengthZeroIfBorderStyleNoneOrHidden",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-inline-width"
	},
		"border-inline-end-color": {
		syntax: "<'border-top-color'>",
		media: "visual",
		inherited: false,
		animationType: "color",
		percentages: "no",
		groups: [
			"CSS Logical Properties"
		],
		initial: "currentcolor",
		appliesto: "allElements",
		computed: "computedColor",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-inline-end-color"
	},
		"border-inline-end-style": {
		syntax: "<'border-top-style'>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Logical Properties"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-inline-end-style"
	},
		"border-inline-end-width": {
		syntax: "<'border-top-width'>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "logicalWidthOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "medium",
		appliesto: "allElements",
		computed: "absoluteLengthZeroIfBorderStyleNoneOrHidden",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-inline-end-width"
	},
		"border-inline-start": {
		syntax: "<'border-top-width'> || <'border-top-style'> || <'color'>",
		media: "visual",
		inherited: false,
		animationType: [
			"border-inline-start-color",
			"border-inline-start-style",
			"border-inline-start-width"
		],
		percentages: "no",
		groups: [
			"CSS Logical Properties"
		],
		initial: [
			"border-width",
			"border-style",
			"color"
		],
		appliesto: "allElements",
		computed: [
			"border-width",
			"border-style",
			"border-inline-start-color"
		],
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-inline-start"
	},
		"border-inline-start-color": {
		syntax: "<'border-top-color'>",
		media: "visual",
		inherited: false,
		animationType: "color",
		percentages: "no",
		groups: [
			"CSS Logical Properties"
		],
		initial: "currentcolor",
		appliesto: "allElements",
		computed: "computedColor",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-inline-start-color"
	},
		"border-inline-start-style": {
		syntax: "<'border-top-style'>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Logical Properties"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-inline-start-style"
	},
		"border-inline-start-width": {
		syntax: "<'border-top-width'>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "logicalWidthOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "medium",
		appliesto: "allElements",
		computed: "absoluteLengthZeroIfBorderStyleNoneOrHidden",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-inline-start-width"
	},
		"border-left": {
		syntax: "<line-width> || <line-style> || <color>",
		media: "visual",
		inherited: false,
		animationType: [
			"border-left-color",
			"border-left-style",
			"border-left-width"
		],
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: [
			"border-left-width",
			"border-left-style",
			"border-left-color"
		],
		appliesto: "allElements",
		computed: [
			"border-left-width",
			"border-left-style",
			"border-left-color"
		],
		order: "orderOfAppearance",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-left"
	},
		"border-left-color": {
		syntax: "<color>",
		media: "visual",
		inherited: false,
		animationType: "color",
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "currentcolor",
		appliesto: "allElements",
		computed: "computedColor",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-left-color"
	},
		"border-left-style": {
		syntax: "<line-style>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-left-style"
	},
		"border-left-width": {
		syntax: "<line-width>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "medium",
		appliesto: "allElements",
		computed: "absoluteLengthOr0IfBorderLeftStyleNoneOrHidden",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-left-width"
	},
		"border-radius": {
		syntax: "<length-percentage>{1,4} [ / <length-percentage>{1,4} ]?",
		media: "visual",
		inherited: false,
		animationType: [
			"border-top-left-radius",
			"border-top-right-radius",
			"border-bottom-right-radius",
			"border-bottom-left-radius"
		],
		percentages: "referToDimensionOfBorderBox",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: [
			"border-top-left-radius",
			"border-top-right-radius",
			"border-bottom-right-radius",
			"border-bottom-left-radius"
		],
		appliesto: "allElementsUAsNotRequiredWhenCollapse",
		computed: [
			"border-bottom-left-radius",
			"border-bottom-right-radius",
			"border-top-left-radius",
			"border-top-right-radius"
		],
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-radius"
	},
		"border-right": {
		syntax: "<line-width> || <line-style> || <color>",
		media: "visual",
		inherited: false,
		animationType: [
			"border-right-color",
			"border-right-style",
			"border-right-width"
		],
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: [
			"border-right-width",
			"border-right-style",
			"border-right-color"
		],
		appliesto: "allElements",
		computed: [
			"border-right-width",
			"border-right-style",
			"border-right-color"
		],
		order: "orderOfAppearance",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-right"
	},
		"border-right-color": {
		syntax: "<color>",
		media: "visual",
		inherited: false,
		animationType: "color",
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "currentcolor",
		appliesto: "allElements",
		computed: "computedColor",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-right-color"
	},
		"border-right-style": {
		syntax: "<line-style>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-right-style"
	},
		"border-right-width": {
		syntax: "<line-width>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "medium",
		appliesto: "allElements",
		computed: "absoluteLengthOr0IfBorderRightStyleNoneOrHidden",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-right-width"
	},
		"border-spacing": {
		syntax: "<length> <length>?",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Table"
		],
		initial: "0",
		appliesto: "tableElements",
		computed: "twoAbsoluteLengths",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-spacing"
	},
		"border-start-end-radius": {
		syntax: "<length-percentage>{1,2}",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "referToDimensionOfBorderBox",
		groups: [
			"CSS Logical Properties"
		],
		initial: "0",
		appliesto: "allElementsUAsNotRequiredWhenCollapse",
		computed: "twoAbsoluteLengthOrPercentages",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-start-end-radius"
	},
		"border-start-start-radius": {
		syntax: "<length-percentage>{1,2}",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "referToDimensionOfBorderBox",
		groups: [
			"CSS Logical Properties"
		],
		initial: "0",
		appliesto: "allElementsUAsNotRequiredWhenCollapse",
		computed: "twoAbsoluteLengthOrPercentages",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-start-start-radius"
	},
		"border-style": {
		syntax: "<line-style>{1,4}",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: [
			"border-top-style",
			"border-right-style",
			"border-bottom-style",
			"border-left-style"
		],
		appliesto: "allElements",
		computed: [
			"border-bottom-style",
			"border-left-style",
			"border-right-style",
			"border-top-style"
		],
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-style"
	},
		"border-top": {
		syntax: "<line-width> || <line-style> || <color>",
		media: "visual",
		inherited: false,
		animationType: [
			"border-top-color",
			"border-top-style",
			"border-top-width"
		],
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: [
			"border-top-width",
			"border-top-style",
			"border-top-color"
		],
		appliesto: "allElements",
		computed: [
			"border-top-width",
			"border-top-style",
			"border-top-color"
		],
		order: "orderOfAppearance",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-top"
	},
		"border-top-color": {
		syntax: "<color>",
		media: "visual",
		inherited: false,
		animationType: "color",
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "currentcolor",
		appliesto: "allElements",
		computed: "computedColor",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-top-color"
	},
		"border-top-left-radius": {
		syntax: "<length-percentage>{1,2}",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "referToDimensionOfBorderBox",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "0",
		appliesto: "allElementsUAsNotRequiredWhenCollapse",
		computed: "twoAbsoluteLengthOrPercentages",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-top-left-radius"
	},
		"border-top-right-radius": {
		syntax: "<length-percentage>{1,2}",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "referToDimensionOfBorderBox",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "0",
		appliesto: "allElementsUAsNotRequiredWhenCollapse",
		computed: "twoAbsoluteLengthOrPercentages",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-top-right-radius"
	},
		"border-top-style": {
		syntax: "<line-style>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-top-style"
	},
		"border-top-width": {
		syntax: "<line-width>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "medium",
		appliesto: "allElements",
		computed: "absoluteLengthOr0IfBorderTopStyleNoneOrHidden",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-top-width"
	},
		"border-width": {
		syntax: "<line-width>{1,4}",
		media: "visual",
		inherited: false,
		animationType: [
			"border-bottom-width",
			"border-left-width",
			"border-right-width",
			"border-top-width"
		],
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: [
			"border-top-width",
			"border-right-width",
			"border-bottom-width",
			"border-left-width"
		],
		appliesto: "allElements",
		computed: [
			"border-bottom-width",
			"border-left-width",
			"border-right-width",
			"border-top-width"
		],
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/border-width"
	},
		bottom: bottom,
		"box-align": {
		syntax: "start | center | end | baseline | stretch",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions",
			"WebKit Extensions"
		],
		initial: "stretch",
		appliesto: "elementsWithDisplayBoxOrInlineBox",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/box-align"
	},
		"box-decoration-break": {
		syntax: "slice | clone",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Fragmentation"
		],
		initial: "slice",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/box-decoration-break"
	},
		"box-direction": {
		syntax: "normal | reverse | inherit",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions",
			"WebKit Extensions"
		],
		initial: "normal",
		appliesto: "elementsWithDisplayBoxOrInlineBox",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/box-direction"
	},
		"box-flex": {
		syntax: "<number>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions",
			"WebKit Extensions"
		],
		initial: "0",
		appliesto: "directChildrenOfElementsWithDisplayMozBoxMozInlineBox",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/box-flex"
	},
		"box-flex-group": {
		syntax: "<integer>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions",
			"WebKit Extensions"
		],
		initial: "1",
		appliesto: "inFlowChildrenOfBoxElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/box-flex-group"
	},
		"box-lines": {
		syntax: "single | multiple",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions",
			"WebKit Extensions"
		],
		initial: "single",
		appliesto: "boxElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/box-lines"
	},
		"box-ordinal-group": {
		syntax: "<integer>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions",
			"WebKit Extensions"
		],
		initial: "1",
		appliesto: "childrenOfBoxElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/box-ordinal-group"
	},
		"box-orient": {
		syntax: "horizontal | vertical | inline-axis | block-axis | inherit",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions",
			"WebKit Extensions"
		],
		initial: "inlineAxisHorizontalInXUL",
		appliesto: "elementsWithDisplayBoxOrInlineBox",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/box-orient"
	},
		"box-pack": {
		syntax: "start | center | end | justify",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions",
			"WebKit Extensions"
		],
		initial: "start",
		appliesto: "elementsWithDisplayMozBoxMozInlineBox",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/box-pack"
	},
		"box-shadow": {
		syntax: "none | <shadow>#",
		media: "visual",
		inherited: false,
		animationType: "shadowList",
		percentages: "no",
		groups: [
			"CSS Backgrounds and Borders"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "absoluteLengthsSpecifiedColorAsSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/box-shadow"
	},
		"box-sizing": {
		syntax: "content-box | border-box",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Basic User Interface"
		],
		initial: "content-box",
		appliesto: "allElementsAcceptingWidthOrHeight",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/box-sizing"
	},
		"break-after": {
		syntax: "auto | avoid | always | all | avoid-page | page | left | right | recto | verso | avoid-column | column | avoid-region | region",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Fragmentation"
		],
		initial: "auto",
		appliesto: "blockLevelElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/break-after"
	},
		"break-before": {
		syntax: "auto | avoid | always | all | avoid-page | page | left | right | recto | verso | avoid-column | column | avoid-region | region",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Fragmentation"
		],
		initial: "auto",
		appliesto: "blockLevelElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/break-before"
	},
		"break-inside": {
		syntax: "auto | avoid | avoid-page | avoid-column | avoid-region",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Fragmentation"
		],
		initial: "auto",
		appliesto: "blockLevelElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/break-inside"
	},
		"caption-side": {
		syntax: "top | bottom | block-start | block-end | inline-start | inline-end",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Table"
		],
		initial: "top",
		appliesto: "tableCaptionElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/caption-side"
	},
		"caret-color": {
		syntax: "auto | <color>",
		media: "interactive",
		inherited: true,
		animationType: "color",
		percentages: "no",
		groups: [
			"CSS Basic User Interface"
		],
		initial: "auto",
		appliesto: "allElements",
		computed: "asAutoOrColor",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/caret-color"
	},
		clear: clear,
		clip: clip,
		"clip-path": {
		syntax: "<clip-source> | [ <basic-shape> || <geometry-box> ] | none",
		media: "visual",
		inherited: false,
		animationType: "basicShapeOtherwiseNo",
		percentages: "referToReferenceBoxWhenSpecifiedOtherwiseBorderBox",
		groups: [
			"CSS Masking"
		],
		initial: "none",
		appliesto: "allElementsSVGContainerElements",
		computed: "asSpecifiedURLsAbsolute",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/clip-path"
	},
		color: color$1,
		"color-adjust": {
		syntax: "economy | exact",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Color"
		],
		initial: "economy",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/color-adjust"
	},
		"column-count": {
		syntax: "<integer> | auto",
		media: "visual",
		inherited: false,
		animationType: "integer",
		percentages: "no",
		groups: [
			"CSS Columns"
		],
		initial: "auto",
		appliesto: "blockContainersExceptTableWrappers",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/column-count"
	},
		"column-fill": {
		syntax: "auto | balance | balance-all",
		media: "visualInContinuousMediaNoEffectInOverflowColumns",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Columns"
		],
		initial: "balance",
		appliesto: "multicolElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/column-fill"
	},
		"column-gap": {
		syntax: "normal | <length-percentage>",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "referToDimensionOfContentArea",
		groups: [
			"CSS Box Alignment"
		],
		initial: "normal",
		appliesto: "multiColumnElementsFlexContainersGridContainers",
		computed: "asSpecifiedWithLengthsAbsoluteAndNormalComputingToZeroExceptMultiColumn",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/column-gap"
	},
		"column-rule": {
		syntax: "<'column-rule-width'> || <'column-rule-style'> || <'column-rule-color'>",
		media: "visual",
		inherited: false,
		animationType: [
			"column-rule-color",
			"column-rule-style",
			"column-rule-width"
		],
		percentages: "no",
		groups: [
			"CSS Columns"
		],
		initial: [
			"column-rule-width",
			"column-rule-style",
			"column-rule-color"
		],
		appliesto: "multicolElements",
		computed: [
			"column-rule-color",
			"column-rule-style",
			"column-rule-width"
		],
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/column-rule"
	},
		"column-rule-color": {
		syntax: "<color>",
		media: "visual",
		inherited: false,
		animationType: "color",
		percentages: "no",
		groups: [
			"CSS Columns"
		],
		initial: "currentcolor",
		appliesto: "multicolElements",
		computed: "computedColor",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/column-rule-color"
	},
		"column-rule-style": {
		syntax: "<'border-style'>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Columns"
		],
		initial: "none",
		appliesto: "multicolElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/column-rule-style"
	},
		"column-rule-width": {
		syntax: "<'border-width'>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "no",
		groups: [
			"CSS Columns"
		],
		initial: "medium",
		appliesto: "multicolElements",
		computed: "absoluteLength0IfColumnRuleStyleNoneOrHidden",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/column-rule-width"
	},
		"column-span": {
		syntax: "none | all",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Columns"
		],
		initial: "none",
		appliesto: "inFlowBlockLevelElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/column-span"
	},
		"column-width": {
		syntax: "<length> | auto",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "no",
		groups: [
			"CSS Columns"
		],
		initial: "auto",
		appliesto: "blockContainersExceptTableWrappers",
		computed: "absoluteLengthZeroOrLarger",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/column-width"
	},
		columns: columns,
		contain: contain,
		content: content,
		"counter-increment": {
		syntax: "[ <custom-ident> <integer>? ]+ | none",
		media: "all",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Counter Styles"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/counter-increment"
	},
		"counter-reset": {
		syntax: "[ <custom-ident> <integer>? ]+ | none",
		media: "all",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Counter Styles"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/counter-reset"
	},
		"counter-set": {
		syntax: "[ <custom-ident> <integer>? ]+ | none",
		media: "all",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Counter Styles"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/counter-set"
	},
		cursor: cursor,
		direction: direction,
		display: display,
		"empty-cells": {
		syntax: "show | hide",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Table"
		],
		initial: "show",
		appliesto: "tableCellElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/empty-cells"
	},
		filter: filter,
		flex: flex,
		"flex-basis": {
		syntax: "content | <'width'>",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "referToFlexContainersInnerMainSize",
		groups: [
			"CSS Flexible Box Layout"
		],
		initial: "auto",
		appliesto: "flexItemsAndInFlowPseudos",
		computed: "asSpecifiedRelativeToAbsoluteLengths",
		order: "lengthOrPercentageBeforeKeywordIfBothPresent",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/flex-basis"
	},
		"flex-direction": {
		syntax: "row | row-reverse | column | column-reverse",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Flexible Box Layout"
		],
		initial: "row",
		appliesto: "flexContainers",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/flex-direction"
	},
		"flex-flow": {
		syntax: "<'flex-direction'> || <'flex-wrap'>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Flexible Box Layout"
		],
		initial: [
			"flex-direction",
			"flex-wrap"
		],
		appliesto: "flexContainers",
		computed: [
			"flex-direction",
			"flex-wrap"
		],
		order: "orderOfAppearance",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/flex-flow"
	},
		"flex-grow": {
		syntax: "<number>",
		media: "visual",
		inherited: false,
		animationType: "number",
		percentages: "no",
		groups: [
			"CSS Flexible Box Layout"
		],
		initial: "0",
		appliesto: "flexItemsAndInFlowPseudos",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/flex-grow"
	},
		"flex-shrink": {
		syntax: "<number>",
		media: "visual",
		inherited: false,
		animationType: "number",
		percentages: "no",
		groups: [
			"CSS Flexible Box Layout"
		],
		initial: "1",
		appliesto: "flexItemsAndInFlowPseudos",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/flex-shrink"
	},
		"flex-wrap": {
		syntax: "nowrap | wrap | wrap-reverse",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Flexible Box Layout"
		],
		initial: "nowrap",
		appliesto: "flexContainers",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/flex-wrap"
	},
		float: float,
		font: font,
		"font-family": {
		syntax: "[ <family-name> | <generic-family> ]#",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Fonts"
		],
		initial: "dependsOnUserAgent",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/font-family"
	},
		"font-feature-settings": {
		syntax: "normal | <feature-tag-value>#",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Fonts"
		],
		initial: "normal",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/font-feature-settings"
	},
		"font-kerning": {
		syntax: "auto | normal | none",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Fonts"
		],
		initial: "auto",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/font-kerning"
	},
		"font-language-override": {
		syntax: "normal | <string>",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Fonts"
		],
		initial: "normal",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/font-language-override"
	},
		"font-optical-sizing": {
		syntax: "auto | none",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Fonts"
		],
		initial: "auto",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "perGrammar",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/font-optical-sizing"
	},
		"font-variation-settings": {
		syntax: "normal | [ <string> <number> ]#",
		media: "visual",
		inherited: true,
		animationType: "transform",
		percentages: "no",
		groups: [
			"CSS Fonts"
		],
		initial: "normal",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "perGrammar",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/font-variation-settings"
	},
		"font-size": {
		syntax: "<absolute-size> | <relative-size> | <length-percentage>",
		media: "visual",
		inherited: true,
		animationType: "length",
		percentages: "referToParentElementsFontSize",
		groups: [
			"CSS Fonts"
		],
		initial: "medium",
		appliesto: "allElements",
		computed: "asSpecifiedRelativeToAbsoluteLengths",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/font-size"
	},
		"font-size-adjust": {
		syntax: "none | <number>",
		media: "visual",
		inherited: true,
		animationType: "number",
		percentages: "no",
		groups: [
			"CSS Fonts"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/font-size-adjust"
	},
		"font-smooth": {
		syntax: "auto | never | always | <absolute-size> | <length>",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Fonts"
		],
		initial: "auto",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/font-smooth"
	},
		"font-stretch": {
		syntax: "<font-stretch-absolute>",
		media: "visual",
		inherited: true,
		animationType: "fontStretch",
		percentages: "no",
		groups: [
			"CSS Fonts"
		],
		initial: "normal",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/font-stretch"
	},
		"font-style": {
		syntax: "normal | italic | oblique <angle>?",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Fonts"
		],
		initial: "normal",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/font-style"
	},
		"font-synthesis": {
		syntax: "none | [ weight || style ]",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Fonts"
		],
		initial: "weight style",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "orderOfAppearance",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/font-synthesis"
	},
		"font-variant": {
		syntax: "normal | none | [ <common-lig-values> || <discretionary-lig-values> || <historical-lig-values> || <contextual-alt-values> || stylistic( <feature-value-name> ) || historical-forms || styleset( <feature-value-name># ) || character-variant( <feature-value-name># ) || swash( <feature-value-name> ) || ornaments( <feature-value-name> ) || annotation( <feature-value-name> ) || [ small-caps | all-small-caps | petite-caps | all-petite-caps | unicase | titling-caps ] || <numeric-figure-values> || <numeric-spacing-values> || <numeric-fraction-values> || ordinal || slashed-zero || <east-asian-variant-values> || <east-asian-width-values> || ruby ]",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Fonts"
		],
		initial: "normal",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/font-variant"
	},
		"font-variant-alternates": {
		syntax: "normal | [ stylistic( <feature-value-name> ) || historical-forms || styleset( <feature-value-name># ) || character-variant( <feature-value-name># ) || swash( <feature-value-name> ) || ornaments( <feature-value-name> ) || annotation( <feature-value-name> ) ]",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Fonts"
		],
		initial: "normal",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "orderOfAppearance",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/font-variant-alternates"
	},
		"font-variant-caps": {
		syntax: "normal | small-caps | all-small-caps | petite-caps | all-petite-caps | unicase | titling-caps",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Fonts"
		],
		initial: "normal",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/font-variant-caps"
	},
		"font-variant-east-asian": {
		syntax: "normal | [ <east-asian-variant-values> || <east-asian-width-values> || ruby ]",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Fonts"
		],
		initial: "normal",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "orderOfAppearance",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/font-variant-east-asian"
	},
		"font-variant-ligatures": {
		syntax: "normal | none | [ <common-lig-values> || <discretionary-lig-values> || <historical-lig-values> || <contextual-alt-values> ]",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Fonts"
		],
		initial: "normal",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "orderOfAppearance",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/font-variant-ligatures"
	},
		"font-variant-numeric": {
		syntax: "normal | [ <numeric-figure-values> || <numeric-spacing-values> || <numeric-fraction-values> || ordinal || slashed-zero ]",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Fonts"
		],
		initial: "normal",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "orderOfAppearance",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/font-variant-numeric"
	},
		"font-variant-position": {
		syntax: "normal | sub | super",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Fonts"
		],
		initial: "normal",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/font-variant-position"
	},
		"font-weight": {
		syntax: "<font-weight-absolute> | bolder | lighter",
		media: "visual",
		inherited: true,
		animationType: "fontWeight",
		percentages: "no",
		groups: [
			"CSS Fonts"
		],
		initial: "normal",
		appliesto: "allElements",
		computed: "keywordOrNumericalValueBolderLighterTransformedToRealValue",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/font-weight"
	},
		gap: gap,
		grid: grid,
		"grid-area": {
		syntax: "<grid-line> [ / <grid-line> ]{0,3}",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Grid Layout"
		],
		initial: [
			"grid-row-start",
			"grid-column-start",
			"grid-row-end",
			"grid-column-end"
		],
		appliesto: "gridItemsAndBoxesWithinGridContainer",
		computed: [
			"grid-row-start",
			"grid-column-start",
			"grid-row-end",
			"grid-column-end"
		],
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/grid-area"
	},
		"grid-auto-columns": {
		syntax: "<track-size>+",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "referToDimensionOfContentArea",
		groups: [
			"CSS Grid Layout"
		],
		initial: "auto",
		appliesto: "gridContainers",
		computed: "percentageAsSpecifiedOrAbsoluteLength",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/grid-auto-columns"
	},
		"grid-auto-flow": {
		syntax: "[ row | column ] || dense",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Grid Layout"
		],
		initial: "row",
		appliesto: "gridContainers",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/grid-auto-flow"
	},
		"grid-auto-rows": {
		syntax: "<track-size>+",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "referToDimensionOfContentArea",
		groups: [
			"CSS Grid Layout"
		],
		initial: "auto",
		appliesto: "gridContainers",
		computed: "percentageAsSpecifiedOrAbsoluteLength",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/grid-auto-rows"
	},
		"grid-column": {
		syntax: "<grid-line> [ / <grid-line> ]?",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Grid Layout"
		],
		initial: [
			"grid-column-start",
			"grid-column-end"
		],
		appliesto: "gridItemsAndBoxesWithinGridContainer",
		computed: [
			"grid-column-start",
			"grid-column-end"
		],
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/grid-column"
	},
		"grid-column-end": {
		syntax: "<grid-line>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Grid Layout"
		],
		initial: "auto",
		appliesto: "gridItemsAndBoxesWithinGridContainer",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/grid-column-end"
	},
		"grid-column-gap": {
		syntax: "<length-percentage>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "referToDimensionOfContentArea",
		groups: [
			"CSS Grid Layout"
		],
		initial: "0",
		appliesto: "gridContainers",
		computed: "percentageAsSpecifiedOrAbsoluteLength",
		order: "uniqueOrder",
		status: "obsolete",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/column-gap"
	},
		"grid-column-start": {
		syntax: "<grid-line>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Grid Layout"
		],
		initial: "auto",
		appliesto: "gridItemsAndBoxesWithinGridContainer",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/grid-column-start"
	},
		"grid-gap": {
		syntax: "<'grid-row-gap'> <'grid-column-gap'>?",
		media: "visual",
		inherited: false,
		animationType: [
			"grid-row-gap",
			"grid-column-gap"
		],
		percentages: "no",
		groups: [
			"CSS Grid Layout"
		],
		initial: [
			"grid-row-gap",
			"grid-column-gap"
		],
		appliesto: "gridContainers",
		computed: [
			"grid-row-gap",
			"grid-column-gap"
		],
		order: "uniqueOrder",
		status: "obsolete",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/gap"
	},
		"grid-row": {
		syntax: "<grid-line> [ / <grid-line> ]?",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Grid Layout"
		],
		initial: [
			"grid-row-start",
			"grid-row-end"
		],
		appliesto: "gridItemsAndBoxesWithinGridContainer",
		computed: [
			"grid-row-start",
			"grid-row-end"
		],
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/grid-row"
	},
		"grid-row-end": {
		syntax: "<grid-line>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Grid Layout"
		],
		initial: "auto",
		appliesto: "gridItemsAndBoxesWithinGridContainer",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/grid-row-end"
	},
		"grid-row-gap": {
		syntax: "<length-percentage>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "referToDimensionOfContentArea",
		groups: [
			"CSS Grid Layout"
		],
		initial: "0",
		appliesto: "gridContainers",
		computed: "percentageAsSpecifiedOrAbsoluteLength",
		order: "uniqueOrder",
		status: "obsolete",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/row-gap"
	},
		"grid-row-start": {
		syntax: "<grid-line>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Grid Layout"
		],
		initial: "auto",
		appliesto: "gridItemsAndBoxesWithinGridContainer",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/grid-row-start"
	},
		"grid-template": {
		syntax: "none | [ <'grid-template-rows'> / <'grid-template-columns'> ] | [ <line-names>? <string> <track-size>? <line-names>? ]+ [ / <explicit-track-list> ]?",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: [
			"grid-template-columns",
			"grid-template-rows"
		],
		groups: [
			"CSS Grid Layout"
		],
		initial: [
			"grid-template-columns",
			"grid-template-rows",
			"grid-template-areas"
		],
		appliesto: "gridContainers",
		computed: [
			"grid-template-columns",
			"grid-template-rows",
			"grid-template-areas"
		],
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/grid-template"
	},
		"grid-template-areas": {
		syntax: "none | <string>+",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Grid Layout"
		],
		initial: "none",
		appliesto: "gridContainers",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/grid-template-areas"
	},
		"grid-template-columns": {
		syntax: "none | <track-list> | <auto-track-list> | subgrid <line-name-list>?",
		media: "visual",
		inherited: false,
		animationType: "simpleListOfLpcDifferenceLpc",
		percentages: "referToDimensionOfContentArea",
		groups: [
			"CSS Grid Layout"
		],
		initial: "none",
		appliesto: "gridContainers",
		computed: "asSpecifiedRelativeToAbsoluteLengths",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/grid-template-columns"
	},
		"grid-template-rows": {
		syntax: "none | <track-list> | <auto-track-list> | subgrid <line-name-list>?",
		media: "visual",
		inherited: false,
		animationType: "simpleListOfLpcDifferenceLpc",
		percentages: "referToDimensionOfContentArea",
		groups: [
			"CSS Grid Layout"
		],
		initial: "none",
		appliesto: "gridContainers",
		computed: "asSpecifiedRelativeToAbsoluteLengths",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/grid-template-rows"
	},
		"hanging-punctuation": {
		syntax: "none | [ first || [ force-end | allow-end ] || last ]",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Text"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/hanging-punctuation"
	},
		height: height,
		hyphens: hyphens,
		"image-orientation": {
		syntax: "from-image | <angle> | [ <angle>? flip ]",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Images"
		],
		initial: "from-image",
		appliesto: "allElements",
		computed: "angleRoundedToNextQuarter",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/image-orientation"
	},
		"image-rendering": {
		syntax: "auto | crisp-edges | pixelated",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Images"
		],
		initial: "auto",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/image-rendering"
	},
		"image-resolution": {
		syntax: "[ from-image || <resolution> ] && snap?",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Images"
		],
		initial: "1dppx",
		appliesto: "allElements",
		computed: "asSpecifiedWithExceptionOfResolution",
		order: "uniqueOrder",
		status: "experimental"
	},
		"ime-mode": {
		syntax: "auto | normal | active | inactive | disabled",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Basic User Interface"
		],
		initial: "auto",
		appliesto: "textFields",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "obsolete",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/ime-mode"
	},
		"initial-letter": {
		syntax: "normal | [ <number> <integer>? ]",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Inline"
		],
		initial: "normal",
		appliesto: "firstLetterPseudoElementsAndInlineLevelFirstChildren",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "experimental",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/initial-letter"
	},
		"initial-letter-align": {
		syntax: "[ auto | alphabetic | hanging | ideographic ]",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Inline"
		],
		initial: "auto",
		appliesto: "firstLetterPseudoElementsAndInlineLevelFirstChildren",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "experimental",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/initial-letter-align"
	},
		"inline-size": {
		syntax: "<'width'>",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "inlineSizeOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "auto",
		appliesto: "sameAsWidthAndHeight",
		computed: "sameAsWidthAndHeight",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/inline-size"
	},
		inset: inset,
		"inset-block": {
		syntax: "<'top'>{1,2}",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "logicalHeightOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "auto",
		appliesto: "positionedElements",
		computed: "sameAsBoxOffsets",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/inset-block"
	},
		"inset-block-end": {
		syntax: "<'top'>",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "logicalHeightOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "auto",
		appliesto: "positionedElements",
		computed: "sameAsBoxOffsets",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/inset-block-end"
	},
		"inset-block-start": {
		syntax: "<'top'>",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "logicalHeightOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "auto",
		appliesto: "positionedElements",
		computed: "sameAsBoxOffsets",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/inset-block-start"
	},
		"inset-inline": {
		syntax: "<'top'>{1,2}",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "logicalWidthOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "auto",
		appliesto: "positionedElements",
		computed: "sameAsBoxOffsets",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/inset-inline"
	},
		"inset-inline-end": {
		syntax: "<'top'>",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "logicalWidthOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "auto",
		appliesto: "positionedElements",
		computed: "sameAsBoxOffsets",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/inset-inline-end"
	},
		"inset-inline-start": {
		syntax: "<'top'>",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "logicalWidthOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "auto",
		appliesto: "positionedElements",
		computed: "sameAsBoxOffsets",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/inset-inline-start"
	},
		isolation: isolation,
		"justify-content": {
		syntax: "normal | <content-distribution> | <overflow-position>? [ <content-position> | left | right ]",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Box Alignment"
		],
		initial: "normal",
		appliesto: "flexContainers",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/justify-content"
	},
		"justify-items": {
		syntax: "normal | stretch | <baseline-position> | <overflow-position>? [ <self-position> | left | right ] | legacy | legacy && [ left | right | center ]",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Box Alignment"
		],
		initial: "legacy",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/justify-items"
	},
		"justify-self": {
		syntax: "auto | normal | stretch | <baseline-position> | <overflow-position>? [ <self-position> | left | right ]",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Box Alignment"
		],
		initial: "auto",
		appliesto: "blockLevelBoxesAndAbsolutelyPositionedBoxesAndGridItems",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/justify-self"
	},
		"justify-tracks": {
		syntax: "[ normal | <content-distribution> | <overflow-position>? [ <content-position> | left | right ] ]#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Grid Layout"
		],
		initial: "normal",
		appliesto: "gridContainersWithMasonryLayoutInTheirInlineAxis",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "experimental",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/justify-tracks"
	},
		left: left,
		"letter-spacing": {
		syntax: "normal | <length>",
		media: "visual",
		inherited: true,
		animationType: "length",
		percentages: "no",
		groups: [
			"CSS Text"
		],
		initial: "normal",
		appliesto: "allElements",
		computed: "optimumValueOfAbsoluteLengthOrNormal",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/letter-spacing"
	},
		"line-break": {
		syntax: "auto | loose | normal | strict | anywhere",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Text"
		],
		initial: "auto",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/line-break"
	},
		"line-clamp": {
		syntax: "none | <integer>",
		media: "visual",
		inherited: false,
		animationType: "integer",
		percentages: "no",
		groups: [
			"CSS Overflow"
		],
		initial: "none",
		appliesto: "blockContainersExceptMultiColumnContainers",
		computed: "asSpecified",
		order: "perGrammar",
		status: "experimental"
	},
		"line-height": {
		syntax: "normal | <number> | <length> | <percentage>",
		media: "visual",
		inherited: true,
		animationType: "numberOrLength",
		percentages: "referToElementFontSize",
		groups: [
			"CSS Fonts"
		],
		initial: "normal",
		appliesto: "allElements",
		computed: "absoluteLengthOrAsSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/line-height"
	},
		"line-height-step": {
		syntax: "<length>",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Fonts"
		],
		initial: "0",
		appliesto: "blockContainers",
		computed: "absoluteLength",
		order: "perGrammar",
		status: "experimental",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/line-height-step"
	},
		"list-style": {
		syntax: "<'list-style-type'> || <'list-style-position'> || <'list-style-image'>",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Lists and Counters"
		],
		initial: [
			"list-style-type",
			"list-style-position",
			"list-style-image"
		],
		appliesto: "listItems",
		computed: [
			"list-style-image",
			"list-style-position",
			"list-style-type"
		],
		order: "orderOfAppearance",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/list-style"
	},
		"list-style-image": {
		syntax: "<url> | none",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Lists and Counters"
		],
		initial: "none",
		appliesto: "listItems",
		computed: "noneOrImageWithAbsoluteURI",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/list-style-image"
	},
		"list-style-position": {
		syntax: "inside | outside",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Lists and Counters"
		],
		initial: "outside",
		appliesto: "listItems",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/list-style-position"
	},
		"list-style-type": {
		syntax: "<counter-style> | <string> | none",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Lists and Counters"
		],
		initial: "disc",
		appliesto: "listItems",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/list-style-type"
	},
		margin: margin,
		"margin-block": {
		syntax: "<'margin-left'>{1,2}",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "dependsOnLayoutModel",
		groups: [
			"CSS Logical Properties"
		],
		initial: "0",
		appliesto: "sameAsMargin",
		computed: "lengthAbsolutePercentageAsSpecifiedOtherwiseAuto",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/margin-block"
	},
		"margin-block-end": {
		syntax: "<'margin-left'>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "dependsOnLayoutModel",
		groups: [
			"CSS Logical Properties"
		],
		initial: "0",
		appliesto: "sameAsMargin",
		computed: "lengthAbsolutePercentageAsSpecifiedOtherwiseAuto",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/margin-block-end"
	},
		"margin-block-start": {
		syntax: "<'margin-left'>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "dependsOnLayoutModel",
		groups: [
			"CSS Logical Properties"
		],
		initial: "0",
		appliesto: "sameAsMargin",
		computed: "lengthAbsolutePercentageAsSpecifiedOtherwiseAuto",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/margin-block-start"
	},
		"margin-bottom": {
		syntax: "<length> | <percentage> | auto",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "referToWidthOfContainingBlock",
		groups: [
			"CSS Box Model"
		],
		initial: "0",
		appliesto: "allElementsExceptTableDisplayTypes",
		computed: "percentageAsSpecifiedOrAbsoluteLength",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/margin-bottom"
	},
		"margin-inline": {
		syntax: "<'margin-left'>{1,2}",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "dependsOnLayoutModel",
		groups: [
			"CSS Logical Properties"
		],
		initial: "0",
		appliesto: "sameAsMargin",
		computed: "lengthAbsolutePercentageAsSpecifiedOtherwiseAuto",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/margin-inline"
	},
		"margin-inline-end": {
		syntax: "<'margin-left'>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "dependsOnLayoutModel",
		groups: [
			"CSS Logical Properties"
		],
		initial: "0",
		appliesto: "sameAsMargin",
		computed: "lengthAbsolutePercentageAsSpecifiedOtherwiseAuto",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/margin-inline-end"
	},
		"margin-inline-start": {
		syntax: "<'margin-left'>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "dependsOnLayoutModel",
		groups: [
			"CSS Logical Properties"
		],
		initial: "0",
		appliesto: "sameAsMargin",
		computed: "lengthAbsolutePercentageAsSpecifiedOtherwiseAuto",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/margin-inline-start"
	},
		"margin-left": {
		syntax: "<length> | <percentage> | auto",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "referToWidthOfContainingBlock",
		groups: [
			"CSS Box Model"
		],
		initial: "0",
		appliesto: "allElementsExceptTableDisplayTypes",
		computed: "percentageAsSpecifiedOrAbsoluteLength",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/margin-left"
	},
		"margin-right": {
		syntax: "<length> | <percentage> | auto",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "referToWidthOfContainingBlock",
		groups: [
			"CSS Box Model"
		],
		initial: "0",
		appliesto: "allElementsExceptTableDisplayTypes",
		computed: "percentageAsSpecifiedOrAbsoluteLength",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/margin-right"
	},
		"margin-top": {
		syntax: "<length> | <percentage> | auto",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "referToWidthOfContainingBlock",
		groups: [
			"CSS Box Model"
		],
		initial: "0",
		appliesto: "allElementsExceptTableDisplayTypes",
		computed: "percentageAsSpecifiedOrAbsoluteLength",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/margin-top"
	},
		"margin-trim": {
		syntax: "none | in-flow | all",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Box Model"
		],
		initial: "none",
		appliesto: "blockContainersAndMultiColumnContainers",
		computed: "asSpecified",
		order: "perGrammar",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line"
		],
		status: "experimental",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/margin-trim"
	},
		mask: mask,
		"mask-border": {
		syntax: "<'mask-border-source'> || <'mask-border-slice'> [ / <'mask-border-width'>? [ / <'mask-border-outset'> ]? ]? || <'mask-border-repeat'> || <'mask-border-mode'>",
		media: "visual",
		inherited: false,
		animationType: [
			"mask-border-mode",
			"mask-border-outset",
			"mask-border-repeat",
			"mask-border-slice",
			"mask-border-source",
			"mask-border-width"
		],
		percentages: [
			"mask-border-slice",
			"mask-border-width"
		],
		groups: [
			"CSS Masking"
		],
		initial: [
			"mask-border-mode",
			"mask-border-outset",
			"mask-border-repeat",
			"mask-border-slice",
			"mask-border-source",
			"mask-border-width"
		],
		appliesto: "allElementsSVGContainerElements",
		computed: [
			"mask-border-mode",
			"mask-border-outset",
			"mask-border-repeat",
			"mask-border-slice",
			"mask-border-source",
			"mask-border-width"
		],
		order: "perGrammar",
		stacking: true,
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mask-border"
	},
		"mask-border-mode": {
		syntax: "luminance | alpha",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Masking"
		],
		initial: "alpha",
		appliesto: "allElementsSVGContainerElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mask-border-mode"
	},
		"mask-border-outset": {
		syntax: "[ <length> | <number> ]{1,4}",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Masking"
		],
		initial: "0",
		appliesto: "allElementsSVGContainerElements",
		computed: "asSpecifiedRelativeToAbsoluteLengths",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mask-border-outset"
	},
		"mask-border-repeat": {
		syntax: "[ stretch | repeat | round | space ]{1,2}",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Masking"
		],
		initial: "stretch",
		appliesto: "allElementsSVGContainerElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mask-border-repeat"
	},
		"mask-border-slice": {
		syntax: "<number-percentage>{1,4} fill?",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "referToSizeOfMaskBorderImage",
		groups: [
			"CSS Masking"
		],
		initial: "0",
		appliesto: "allElementsSVGContainerElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mask-border-slice"
	},
		"mask-border-source": {
		syntax: "none | <image>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Masking"
		],
		initial: "none",
		appliesto: "allElementsSVGContainerElements",
		computed: "asSpecifiedURLsAbsolute",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mask-border-source"
	},
		"mask-border-width": {
		syntax: "[ <length-percentage> | <number> | auto ]{1,4}",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "relativeToMaskBorderImageArea",
		groups: [
			"CSS Masking"
		],
		initial: "auto",
		appliesto: "allElementsSVGContainerElements",
		computed: "asSpecifiedRelativeToAbsoluteLengths",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mask-border-width"
	},
		"mask-clip": {
		syntax: "[ <geometry-box> | no-clip ]#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Masking"
		],
		initial: "border-box",
		appliesto: "allElementsSVGContainerElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mask-clip"
	},
		"mask-composite": {
		syntax: "<compositing-operator>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Masking"
		],
		initial: "add",
		appliesto: "allElementsSVGContainerElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mask-composite"
	},
		"mask-image": {
		syntax: "<mask-reference>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Masking"
		],
		initial: "none",
		appliesto: "allElementsSVGContainerElements",
		computed: "asSpecifiedURLsAbsolute",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mask-image"
	},
		"mask-mode": {
		syntax: "<masking-mode>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Masking"
		],
		initial: "match-source",
		appliesto: "allElementsSVGContainerElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mask-mode"
	},
		"mask-origin": {
		syntax: "<geometry-box>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Masking"
		],
		initial: "border-box",
		appliesto: "allElementsSVGContainerElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mask-origin"
	},
		"mask-position": {
		syntax: "<position>#",
		media: "visual",
		inherited: false,
		animationType: "repeatableListOfSimpleListOfLpc",
		percentages: "referToSizeOfMaskPaintingArea",
		groups: [
			"CSS Masking"
		],
		initial: "center",
		appliesto: "allElementsSVGContainerElements",
		computed: "consistsOfTwoKeywordsForOriginAndOffsets",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mask-position"
	},
		"mask-repeat": {
		syntax: "<repeat-style>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Masking"
		],
		initial: "no-repeat",
		appliesto: "allElementsSVGContainerElements",
		computed: "consistsOfTwoDimensionKeywords",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mask-repeat"
	},
		"mask-size": {
		syntax: "<bg-size>#",
		media: "visual",
		inherited: false,
		animationType: "repeatableListOfSimpleListOfLpc",
		percentages: "no",
		groups: [
			"CSS Masking"
		],
		initial: "auto",
		appliesto: "allElementsSVGContainerElements",
		computed: "asSpecifiedRelativeToAbsoluteLengths",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mask-size"
	},
		"mask-type": {
		syntax: "luminance | alpha",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Masking"
		],
		initial: "luminance",
		appliesto: "maskElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mask-type"
	},
		"masonry-auto-flow": {
		syntax: "[ pack | next ] || [ definite-first | ordered ]",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Grid Layout"
		],
		initial: "pack",
		appliesto: "gridContainersWithMasonryLayout",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "experimental",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/masonry-auto-flow"
	},
		"math-style": {
		syntax: "normal | compact",
		media: "visual",
		inherited: true,
		animationType: "notAnimatable",
		percentages: "no",
		groups: [
			"MathML"
		],
		initial: "normal",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/math-style"
	},
		"max-block-size": {
		syntax: "<'max-width'>",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "blockSizeOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "0",
		appliesto: "sameAsWidthAndHeight",
		computed: "sameAsMaxWidthAndMaxHeight",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/max-block-size"
	},
		"max-height": {
		syntax: "none | <length-percentage> | min-content | max-content | fit-content(<length-percentage>)",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "regardingHeightOfGeneratedBoxContainingBlockPercentagesNone",
		groups: [
			"CSS Box Model"
		],
		initial: "none",
		appliesto: "allElementsButNonReplacedAndTableColumns",
		computed: "percentageAsSpecifiedAbsoluteLengthOrNone",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/max-height"
	},
		"max-inline-size": {
		syntax: "<'max-width'>",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "inlineSizeOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "0",
		appliesto: "sameAsWidthAndHeight",
		computed: "sameAsMaxWidthAndMaxHeight",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/max-inline-size"
	},
		"max-lines": {
		syntax: "none | <integer>",
		media: "visual",
		inherited: false,
		animationType: "integer",
		percentages: "no",
		groups: [
			"CSS Overflow"
		],
		initial: "none",
		appliesto: "blockContainersExceptMultiColumnContainers",
		computed: "asSpecified",
		order: "perGrammar",
		status: "experimental"
	},
		"max-width": {
		syntax: "none | <length-percentage> | min-content | max-content | fit-content(<length-percentage>)",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "referToWidthOfContainingBlock",
		groups: [
			"CSS Box Model"
		],
		initial: "none",
		appliesto: "allElementsButNonReplacedAndTableRows",
		computed: "percentageAsSpecifiedAbsoluteLengthOrNone",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/max-width"
	},
		"min-block-size": {
		syntax: "<'min-width'>",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "blockSizeOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "0",
		appliesto: "sameAsWidthAndHeight",
		computed: "sameAsMinWidthAndMinHeight",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/min-block-size"
	},
		"min-height": {
		syntax: "auto | <length> | <percentage> | min-content | max-content | fit-content(<length-percentage>)",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "regardingHeightOfGeneratedBoxContainingBlockPercentages0",
		groups: [
			"CSS Box Model"
		],
		initial: "auto",
		appliesto: "allElementsButNonReplacedAndTableColumns",
		computed: "percentageAsSpecifiedOrAbsoluteLength",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/min-height"
	},
		"min-inline-size": {
		syntax: "<'min-width'>",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "inlineSizeOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "0",
		appliesto: "sameAsWidthAndHeight",
		computed: "sameAsMinWidthAndMinHeight",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/min-inline-size"
	},
		"min-width": {
		syntax: "auto | <length> | <percentage> | min-content | max-content | fit-content(<length-percentage>)",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "referToWidthOfContainingBlock",
		groups: [
			"CSS Box Model"
		],
		initial: "auto",
		appliesto: "allElementsButNonReplacedAndTableRows",
		computed: "percentageAsSpecifiedOrAbsoluteLength",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/min-width"
	},
		"mix-blend-mode": {
		syntax: "<blend-mode>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Compositing and Blending"
		],
		initial: "normal",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		stacking: true,
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/mix-blend-mode"
	},
		"object-fit": {
		syntax: "fill | contain | cover | none | scale-down",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Images"
		],
		initial: "fill",
		appliesto: "replacedElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/object-fit"
	},
		"object-position": {
		syntax: "<position>",
		media: "visual",
		inherited: true,
		animationType: "repeatableListOfSimpleListOfLpc",
		percentages: "referToWidthAndHeightOfElement",
		groups: [
			"CSS Images"
		],
		initial: "50% 50%",
		appliesto: "replacedElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/object-position"
	},
		offset: offset,
		"offset-anchor": {
		syntax: "auto | <position>",
		media: "visual",
		inherited: false,
		animationType: "position",
		percentages: "relativeToWidthAndHeight",
		groups: [
			"CSS Motion Path"
		],
		initial: "auto",
		appliesto: "transformableElements",
		computed: "forLengthAbsoluteValueOtherwisePercentage",
		order: "perGrammar",
		status: "standard"
	},
		"offset-distance": {
		syntax: "<length-percentage>",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "referToTotalPathLength",
		groups: [
			"CSS Motion Path"
		],
		initial: "0",
		appliesto: "transformableElements",
		computed: "forLengthAbsoluteValueOtherwisePercentage",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/offset-distance"
	},
		"offset-path": {
		syntax: "none | ray( [ <angle> && <size> && contain? ] ) | <path()> | <url> | [ <basic-shape> || <geometry-box> ]",
		media: "visual",
		inherited: false,
		animationType: "angleOrBasicShapeOrPath",
		percentages: "no",
		groups: [
			"CSS Motion Path"
		],
		initial: "none",
		appliesto: "transformableElements",
		computed: "asSpecified",
		order: "perGrammar",
		stacking: true,
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/offset-path"
	},
		"offset-position": {
		syntax: "auto | <position>",
		media: "visual",
		inherited: false,
		animationType: "position",
		percentages: "referToSizeOfContainingBlock",
		groups: [
			"CSS Motion Path"
		],
		initial: "auto",
		appliesto: "transformableElements",
		computed: "forLengthAbsoluteValueOtherwisePercentage",
		order: "perGrammar",
		status: "experimental"
	},
		"offset-rotate": {
		syntax: "[ auto | reverse ] || <angle>",
		media: "visual",
		inherited: false,
		animationType: "angleOrBasicShapeOrPath",
		percentages: "no",
		groups: [
			"CSS Motion Path"
		],
		initial: "auto",
		appliesto: "transformableElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/offset-rotate"
	},
		opacity: opacity,
		order: order,
		orphans: orphans,
		outline: outline,
		"outline-color": {
		syntax: "<color> | invert",
		media: [
			"visual",
			"interactive"
		],
		inherited: false,
		animationType: "color",
		percentages: "no",
		groups: [
			"CSS Basic User Interface"
		],
		initial: "invertOrCurrentColor",
		appliesto: "allElements",
		computed: "invertForTranslucentColorRGBAOtherwiseRGB",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/outline-color"
	},
		"outline-offset": {
		syntax: "<length>",
		media: [
			"visual",
			"interactive"
		],
		inherited: false,
		animationType: "length",
		percentages: "no",
		groups: [
			"CSS Basic User Interface"
		],
		initial: "0",
		appliesto: "allElements",
		computed: "asSpecifiedRelativeToAbsoluteLengths",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/outline-offset"
	},
		"outline-style": {
		syntax: "auto | <'border-style'>",
		media: [
			"visual",
			"interactive"
		],
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Basic User Interface"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/outline-style"
	},
		"outline-width": {
		syntax: "<line-width>",
		media: [
			"visual",
			"interactive"
		],
		inherited: false,
		animationType: "length",
		percentages: "no",
		groups: [
			"CSS Basic User Interface"
		],
		initial: "medium",
		appliesto: "allElements",
		computed: "absoluteLength0ForNone",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/outline-width"
	},
		overflow: overflow,
		"overflow-anchor": {
		syntax: "auto | none",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Scroll Anchoring"
		],
		initial: "auto",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard"
	},
		"overflow-block": {
		syntax: "visible | hidden | clip | scroll | auto",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Overflow"
		],
		initial: "auto",
		appliesto: "blockContainersFlexContainersGridContainers",
		computed: "asSpecifiedButVisibleOrClipReplacedToAutoOrHiddenIfOtherValueDifferent",
		order: "perGrammar",
		status: "standard"
	},
		"overflow-clip-box": {
		syntax: "padding-box | content-box",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Mozilla Extensions"
		],
		initial: "padding-box",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Mozilla/CSS/overflow-clip-box"
	},
		"overflow-inline": {
		syntax: "visible | hidden | clip | scroll | auto",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Overflow"
		],
		initial: "auto",
		appliesto: "blockContainersFlexContainersGridContainers",
		computed: "asSpecifiedButVisibleOrClipReplacedToAutoOrHiddenIfOtherValueDifferent",
		order: "perGrammar",
		status: "standard"
	},
		"overflow-wrap": {
		syntax: "normal | break-word | anywhere",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Text"
		],
		initial: "normal",
		appliesto: "nonReplacedInlineElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/overflow-wrap"
	},
		"overflow-x": {
		syntax: "visible | hidden | clip | scroll | auto",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Overflow"
		],
		initial: "visible",
		appliesto: "blockContainersFlexContainersGridContainers",
		computed: "asSpecifiedButVisibleOrClipReplacedToAutoOrHiddenIfOtherValueDifferent",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/overflow-x"
	},
		"overflow-y": {
		syntax: "visible | hidden | clip | scroll | auto",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Overflow"
		],
		initial: "visible",
		appliesto: "blockContainersFlexContainersGridContainers",
		computed: "asSpecifiedButVisibleOrClipReplacedToAutoOrHiddenIfOtherValueDifferent",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/overflow-y"
	},
		"overscroll-behavior": {
		syntax: "[ contain | none | auto ]{1,2}",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Box Model"
		],
		initial: "auto",
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/overscroll-behavior"
	},
		"overscroll-behavior-block": {
		syntax: "contain | none | auto",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Box Model"
		],
		initial: "auto",
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/overscroll-behavior-block"
	},
		"overscroll-behavior-inline": {
		syntax: "contain | none | auto",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Box Model"
		],
		initial: "auto",
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/overscroll-behavior-inline"
	},
		"overscroll-behavior-x": {
		syntax: "contain | none | auto",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Box Model"
		],
		initial: "auto",
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/overscroll-behavior-x"
	},
		"overscroll-behavior-y": {
		syntax: "contain | none | auto",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Box Model"
		],
		initial: "auto",
		appliesto: "nonReplacedBlockAndInlineBlockElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/overscroll-behavior-y"
	},
		padding: padding,
		"padding-block": {
		syntax: "<'padding-left'>{1,2}",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "logicalWidthOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "0",
		appliesto: "allElements",
		computed: "asLength",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/padding-block"
	},
		"padding-block-end": {
		syntax: "<'padding-left'>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "logicalWidthOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "0",
		appliesto: "allElements",
		computed: "asLength",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/padding-block-end"
	},
		"padding-block-start": {
		syntax: "<'padding-left'>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "logicalWidthOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "0",
		appliesto: "allElements",
		computed: "asLength",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/padding-block-start"
	},
		"padding-bottom": {
		syntax: "<length> | <percentage>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "referToWidthOfContainingBlock",
		groups: [
			"CSS Box Model"
		],
		initial: "0",
		appliesto: "allElementsExceptInternalTableDisplayTypes",
		computed: "percentageAsSpecifiedOrAbsoluteLength",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/padding-bottom"
	},
		"padding-inline": {
		syntax: "<'padding-left'>{1,2}",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "logicalWidthOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "0",
		appliesto: "allElements",
		computed: "asLength",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/padding-inline"
	},
		"padding-inline-end": {
		syntax: "<'padding-left'>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "logicalWidthOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "0",
		appliesto: "allElements",
		computed: "asLength",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/padding-inline-end"
	},
		"padding-inline-start": {
		syntax: "<'padding-left'>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "logicalWidthOfContainingBlock",
		groups: [
			"CSS Logical Properties"
		],
		initial: "0",
		appliesto: "allElements",
		computed: "asLength",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/padding-inline-start"
	},
		"padding-left": {
		syntax: "<length> | <percentage>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "referToWidthOfContainingBlock",
		groups: [
			"CSS Box Model"
		],
		initial: "0",
		appliesto: "allElementsExceptInternalTableDisplayTypes",
		computed: "percentageAsSpecifiedOrAbsoluteLength",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/padding-left"
	},
		"padding-right": {
		syntax: "<length> | <percentage>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "referToWidthOfContainingBlock",
		groups: [
			"CSS Box Model"
		],
		initial: "0",
		appliesto: "allElementsExceptInternalTableDisplayTypes",
		computed: "percentageAsSpecifiedOrAbsoluteLength",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/padding-right"
	},
		"padding-top": {
		syntax: "<length> | <percentage>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "referToWidthOfContainingBlock",
		groups: [
			"CSS Box Model"
		],
		initial: "0",
		appliesto: "allElementsExceptInternalTableDisplayTypes",
		computed: "percentageAsSpecifiedOrAbsoluteLength",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/padding-top"
	},
		"page-break-after": {
		syntax: "auto | always | avoid | left | right | recto | verso",
		media: [
			"visual",
			"paged"
		],
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Pages"
		],
		initial: "auto",
		appliesto: "blockElementsInNormalFlow",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/page-break-after"
	},
		"page-break-before": {
		syntax: "auto | always | avoid | left | right | recto | verso",
		media: [
			"visual",
			"paged"
		],
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Pages"
		],
		initial: "auto",
		appliesto: "blockElementsInNormalFlow",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/page-break-before"
	},
		"page-break-inside": {
		syntax: "auto | avoid",
		media: [
			"visual",
			"paged"
		],
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Pages"
		],
		initial: "auto",
		appliesto: "blockElementsInNormalFlow",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/page-break-inside"
	},
		"paint-order": {
		syntax: "normal | [ fill || stroke || markers ]",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Text"
		],
		initial: "normal",
		appliesto: "textElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/paint-order"
	},
		perspective: perspective,
		"perspective-origin": {
		syntax: "<position>",
		media: "visual",
		inherited: false,
		animationType: "simpleListOfLpc",
		percentages: "referToSizeOfBoundingBox",
		groups: [
			"CSS Transforms"
		],
		initial: "50% 50%",
		appliesto: "transformableElements",
		computed: "forLengthAbsoluteValueOtherwisePercentage",
		order: "oneOrTwoValuesLengthAbsoluteKeywordsPercentages",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/perspective-origin"
	},
		"place-content": {
		syntax: "<'align-content'> <'justify-content'>?",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Box Alignment"
		],
		initial: "normal",
		appliesto: "multilineFlexContainers",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/place-content"
	},
		"place-items": {
		syntax: "<'align-items'> <'justify-items'>?",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Box Alignment"
		],
		initial: [
			"align-items",
			"justify-items"
		],
		appliesto: "allElements",
		computed: [
			"align-items",
			"justify-items"
		],
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/place-items"
	},
		"place-self": {
		syntax: "<'align-self'> <'justify-self'>?",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Box Alignment"
		],
		initial: [
			"align-self",
			"justify-self"
		],
		appliesto: "blockLevelBoxesAndAbsolutelyPositionedBoxesAndGridItems",
		computed: [
			"align-self",
			"justify-self"
		],
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/place-self"
	},
		"pointer-events": {
		syntax: "auto | none | visiblePainted | visibleFill | visibleStroke | visible | painted | fill | stroke | all | inherit",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Pointer Events"
		],
		initial: "auto",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/pointer-events"
	},
		position: position$1,
		quotes: quotes,
		resize: resize,
		right: right,
		rotate: rotate,
		"row-gap": {
		syntax: "normal | <length-percentage>",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "referToDimensionOfContentArea",
		groups: [
			"CSS Box Alignment"
		],
		initial: "normal",
		appliesto: "multiColumnElementsFlexContainersGridContainers",
		computed: "asSpecifiedWithLengthsAbsoluteAndNormalComputingToZeroExceptMultiColumn",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/row-gap"
	},
		"ruby-align": {
		syntax: "start | center | space-between | space-around",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Ruby"
		],
		initial: "space-around",
		appliesto: "rubyBasesAnnotationsBaseAnnotationContainers",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "experimental",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/ruby-align"
	},
		"ruby-merge": {
		syntax: "separate | collapse | auto",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Ruby"
		],
		initial: "separate",
		appliesto: "rubyAnnotationsContainers",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "experimental"
	},
		"ruby-position": {
		syntax: "over | under | inter-character",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Ruby"
		],
		initial: "over",
		appliesto: "rubyAnnotationsContainers",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "experimental",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/ruby-position"
	},
		scale: scale,
		"scrollbar-color": {
		syntax: "auto | dark | light | <color>{2}",
		media: "visual",
		inherited: true,
		animationType: "color",
		percentages: "no",
		groups: [
			"CSS Scrollbars"
		],
		initial: "auto",
		appliesto: "scrollingBoxes",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scrollbar-color"
	},
		"scrollbar-gutter": {
		syntax: "auto | [ stable | always ] && both? && force?",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Overflow"
		],
		initial: "auto",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scrollbar-gutter"
	},
		"scrollbar-width": {
		syntax: "auto | thin | none",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Scrollbars"
		],
		initial: "auto",
		appliesto: "scrollingBoxes",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scrollbar-width"
	},
		"scroll-behavior": {
		syntax: "auto | smooth",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSSOM View"
		],
		initial: "auto",
		appliesto: "scrollingBoxes",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-behavior"
	},
		"scroll-margin": {
		syntax: "<length>{1,4}",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "no",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "0",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-margin"
	},
		"scroll-margin-block": {
		syntax: "<length>{1,2}",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "no",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "0",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-margin-block"
	},
		"scroll-margin-block-start": {
		syntax: "<length>",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "no",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "0",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-margin-block-start"
	},
		"scroll-margin-block-end": {
		syntax: "<length>",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "no",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "0",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-margin-block-end"
	},
		"scroll-margin-bottom": {
		syntax: "<length>",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "no",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "0",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-margin-bottom"
	},
		"scroll-margin-inline": {
		syntax: "<length>{1,2}",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "no",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "0",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-margin-inline"
	},
		"scroll-margin-inline-start": {
		syntax: "<length>",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "no",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "0",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-margin-inline-start"
	},
		"scroll-margin-inline-end": {
		syntax: "<length>",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "no",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "0",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-margin-inline-end"
	},
		"scroll-margin-left": {
		syntax: "<length>",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "no",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "0",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-margin-left"
	},
		"scroll-margin-right": {
		syntax: "<length>",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "no",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "0",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-margin-right"
	},
		"scroll-margin-top": {
		syntax: "<length>",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "no",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "0",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-margin-top"
	},
		"scroll-padding": {
		syntax: "[ auto | <length-percentage> ]{1,4}",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "relativeToTheScrollContainersScrollport",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "auto",
		appliesto: "scrollContainers",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-padding"
	},
		"scroll-padding-block": {
		syntax: "[ auto | <length-percentage> ]{1,2}",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "relativeToTheScrollContainersScrollport",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "auto",
		appliesto: "scrollContainers",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-padding-block"
	},
		"scroll-padding-block-start": {
		syntax: "auto | <length-percentage>",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "relativeToTheScrollContainersScrollport",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "auto",
		appliesto: "scrollContainers",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-padding-block-start"
	},
		"scroll-padding-block-end": {
		syntax: "auto | <length-percentage>",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "relativeToTheScrollContainersScrollport",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "auto",
		appliesto: "scrollContainers",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-padding-block-end"
	},
		"scroll-padding-bottom": {
		syntax: "auto | <length-percentage>",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "relativeToTheScrollContainersScrollport",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "auto",
		appliesto: "scrollContainers",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-padding-bottom"
	},
		"scroll-padding-inline": {
		syntax: "[ auto | <length-percentage> ]{1,2}",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "relativeToTheScrollContainersScrollport",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "auto",
		appliesto: "scrollContainers",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-padding-inline"
	},
		"scroll-padding-inline-start": {
		syntax: "auto | <length-percentage>",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "relativeToTheScrollContainersScrollport",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "auto",
		appliesto: "scrollContainers",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-padding-inline-start"
	},
		"scroll-padding-inline-end": {
		syntax: "auto | <length-percentage>",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "relativeToTheScrollContainersScrollport",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "auto",
		appliesto: "scrollContainers",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-padding-inline-end"
	},
		"scroll-padding-left": {
		syntax: "auto | <length-percentage>",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "relativeToTheScrollContainersScrollport",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "auto",
		appliesto: "scrollContainers",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-padding-left"
	},
		"scroll-padding-right": {
		syntax: "auto | <length-percentage>",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "relativeToTheScrollContainersScrollport",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "auto",
		appliesto: "scrollContainers",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-padding-right"
	},
		"scroll-padding-top": {
		syntax: "auto | <length-percentage>",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "relativeToTheScrollContainersScrollport",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "auto",
		appliesto: "scrollContainers",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-padding-top"
	},
		"scroll-snap-align": {
		syntax: "[ none | start | end | center ]{1,2}",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-snap-align"
	},
		"scroll-snap-coordinate": {
		syntax: "none | <position>#",
		media: "interactive",
		inherited: false,
		animationType: "position",
		percentages: "referToBorderBox",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecifiedRelativeToAbsoluteLengths",
		order: "uniqueOrder",
		status: "obsolete",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-snap-coordinate"
	},
		"scroll-snap-destination": {
		syntax: "<position>",
		media: "interactive",
		inherited: false,
		animationType: "position",
		percentages: "relativeToScrollContainerPaddingBoxAxis",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "0px 0px",
		appliesto: "scrollContainers",
		computed: "asSpecifiedRelativeToAbsoluteLengths",
		order: "uniqueOrder",
		status: "obsolete",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-snap-destination"
	},
		"scroll-snap-points-x": {
		syntax: "none | repeat( <length-percentage> )",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "relativeToScrollContainerPaddingBoxAxis",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "none",
		appliesto: "scrollContainers",
		computed: "asSpecifiedRelativeToAbsoluteLengths",
		order: "uniqueOrder",
		status: "obsolete",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-snap-points-x"
	},
		"scroll-snap-points-y": {
		syntax: "none | repeat( <length-percentage> )",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "relativeToScrollContainerPaddingBoxAxis",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "none",
		appliesto: "scrollContainers",
		computed: "asSpecifiedRelativeToAbsoluteLengths",
		order: "uniqueOrder",
		status: "obsolete",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-snap-points-y"
	},
		"scroll-snap-stop": {
		syntax: "normal | always",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "normal",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-snap-stop"
	},
		"scroll-snap-type": {
		syntax: "none | [ x | y | block | inline | both ] [ mandatory | proximity ]?",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-snap-type"
	},
		"scroll-snap-type-x": {
		syntax: "none | mandatory | proximity",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "none",
		appliesto: "scrollContainers",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "obsolete",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-snap-type-x"
	},
		"scroll-snap-type-y": {
		syntax: "none | mandatory | proximity",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Scroll Snap"
		],
		initial: "none",
		appliesto: "scrollContainers",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "obsolete",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/scroll-snap-type-y"
	},
		"shape-image-threshold": {
		syntax: "<alpha-value>",
		media: "visual",
		inherited: false,
		animationType: "number",
		percentages: "no",
		groups: [
			"CSS Shapes"
		],
		initial: "0.0",
		appliesto: "floats",
		computed: "specifiedValueNumberClipped0To1",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/shape-image-threshold"
	},
		"shape-margin": {
		syntax: "<length-percentage>",
		media: "visual",
		inherited: false,
		animationType: "lpc",
		percentages: "referToWidthOfContainingBlock",
		groups: [
			"CSS Shapes"
		],
		initial: "0",
		appliesto: "floats",
		computed: "asSpecifiedRelativeToAbsoluteLengths",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/shape-margin"
	},
		"shape-outside": {
		syntax: "none | <shape-box> || <basic-shape> | <image>",
		media: "visual",
		inherited: false,
		animationType: "basicShapeOtherwiseNo",
		percentages: "no",
		groups: [
			"CSS Shapes"
		],
		initial: "none",
		appliesto: "floats",
		computed: "asDefinedForBasicShapeWithAbsoluteURIOtherwiseAsSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/shape-outside"
	},
		"tab-size": {
		syntax: "<integer> | <length>",
		media: "visual",
		inherited: true,
		animationType: "length",
		percentages: "no",
		groups: [
			"CSS Text"
		],
		initial: "8",
		appliesto: "blockContainers",
		computed: "specifiedIntegerOrAbsoluteLength",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/tab-size"
	},
		"table-layout": {
		syntax: "auto | fixed",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Table"
		],
		initial: "auto",
		appliesto: "tableElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/table-layout"
	},
		"text-align": {
		syntax: "start | end | left | right | center | justify | match-parent",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Text"
		],
		initial: "startOrNamelessValueIfLTRRightIfRTL",
		appliesto: "blockContainers",
		computed: "asSpecifiedExceptMatchParent",
		order: "orderOfAppearance",
		alsoAppliesTo: [
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/text-align"
	},
		"text-align-last": {
		syntax: "auto | start | end | left | right | center | justify",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Text"
		],
		initial: "auto",
		appliesto: "blockContainers",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/text-align-last"
	},
		"text-combine-upright": {
		syntax: "none | all | [ digits <integer>? ]",
		media: "visual",
		inherited: true,
		animationType: "notAnimatable",
		percentages: "no",
		groups: [
			"CSS Writing Modes"
		],
		initial: "none",
		appliesto: "nonReplacedInlineElements",
		computed: "keywordPlusIntegerIfDigits",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/text-combine-upright"
	},
		"text-decoration": {
		syntax: "<'text-decoration-line'> || <'text-decoration-style'> || <'text-decoration-color'> || <'text-decoration-thickness'>",
		media: "visual",
		inherited: false,
		animationType: [
			"text-decoration-color",
			"text-decoration-style",
			"text-decoration-line",
			"text-decoration-thickness"
		],
		percentages: "no",
		groups: [
			"CSS Text Decoration"
		],
		initial: [
			"text-decoration-color",
			"text-decoration-style",
			"text-decoration-line"
		],
		appliesto: "allElements",
		computed: [
			"text-decoration-line",
			"text-decoration-style",
			"text-decoration-color",
			"text-decoration-thickness"
		],
		order: "orderOfAppearance",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/text-decoration"
	},
		"text-decoration-color": {
		syntax: "<color>",
		media: "visual",
		inherited: false,
		animationType: "color",
		percentages: "no",
		groups: [
			"CSS Text Decoration"
		],
		initial: "currentcolor",
		appliesto: "allElements",
		computed: "computedColor",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/text-decoration-color"
	},
		"text-decoration-line": {
		syntax: "none | [ underline || overline || line-through || blink ] | spelling-error | grammar-error",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Text Decoration"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "orderOfAppearance",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/text-decoration-line"
	},
		"text-decoration-skip": {
		syntax: "none | [ objects || [ spaces | [ leading-spaces || trailing-spaces ] ] || edges || box-decoration ]",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Text Decoration"
		],
		initial: "objects",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "orderOfAppearance",
		status: "experimental",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/text-decoration-skip"
	},
		"text-decoration-skip-ink": {
		syntax: "auto | all | none",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Text Decoration"
		],
		initial: "auto",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "orderOfAppearance",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/text-decoration-skip-ink"
	},
		"text-decoration-style": {
		syntax: "solid | double | dotted | dashed | wavy",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Text Decoration"
		],
		initial: "solid",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/text-decoration-style"
	},
		"text-decoration-thickness": {
		syntax: "auto | from-font | <length> | <percentage> ",
		media: "visual",
		inherited: false,
		animationType: "byComputedValueType",
		percentages: "referToElementFontSize",
		groups: [
			"CSS Text Decoration"
		],
		initial: "auto",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/text-decoration-thickness"
	},
		"text-emphasis": {
		syntax: "<'text-emphasis-style'> || <'text-emphasis-color'>",
		media: "visual",
		inherited: false,
		animationType: [
			"text-emphasis-color",
			"text-emphasis-style"
		],
		percentages: "no",
		groups: [
			"CSS Text Decoration"
		],
		initial: [
			"text-emphasis-style",
			"text-emphasis-color"
		],
		appliesto: "allElements",
		computed: [
			"text-emphasis-style",
			"text-emphasis-color"
		],
		order: "orderOfAppearance",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/text-emphasis"
	},
		"text-emphasis-color": {
		syntax: "<color>",
		media: "visual",
		inherited: false,
		animationType: "color",
		percentages: "no",
		groups: [
			"CSS Text Decoration"
		],
		initial: "currentcolor",
		appliesto: "allElements",
		computed: "computedColor",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/text-emphasis-color"
	},
		"text-emphasis-position": {
		syntax: "[ over | under ] && [ right | left ]",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Text Decoration"
		],
		initial: "over right",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/text-emphasis-position"
	},
		"text-emphasis-style": {
		syntax: "none | [ [ filled | open ] || [ dot | circle | double-circle | triangle | sesame ] ] | <string>",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Text Decoration"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/text-emphasis-style"
	},
		"text-indent": {
		syntax: "<length-percentage> && hanging? && each-line?",
		media: "visual",
		inherited: true,
		animationType: "lpc",
		percentages: "referToWidthOfContainingBlock",
		groups: [
			"CSS Text"
		],
		initial: "0",
		appliesto: "blockContainers",
		computed: "percentageOrAbsoluteLengthPlusKeywords",
		order: "lengthOrPercentageBeforeKeywords",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/text-indent"
	},
		"text-justify": {
		syntax: "auto | inter-character | inter-word | none",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Text"
		],
		initial: "auto",
		appliesto: "inlineLevelAndTableCellElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/text-justify"
	},
		"text-orientation": {
		syntax: "mixed | upright | sideways",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Writing Modes"
		],
		initial: "mixed",
		appliesto: "allElementsExceptTableRowGroupsRowsColumnGroupsAndColumns",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/text-orientation"
	},
		"text-overflow": {
		syntax: "[ clip | ellipsis | <string> ]{1,2}",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Basic User Interface"
		],
		initial: "clip",
		appliesto: "blockContainerElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/text-overflow"
	},
		"text-rendering": {
		syntax: "auto | optimizeSpeed | optimizeLegibility | geometricPrecision",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Miscellaneous"
		],
		initial: "auto",
		appliesto: "textElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/text-rendering"
	},
		"text-shadow": {
		syntax: "none | <shadow-t>#",
		media: "visual",
		inherited: true,
		animationType: "shadowList",
		percentages: "no",
		groups: [
			"CSS Text Decoration"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "colorPlusThreeAbsoluteLengths",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/text-shadow"
	},
		"text-size-adjust": {
		syntax: "none | auto | <percentage>",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "referToSizeOfFont",
		groups: [
			"CSS Text"
		],
		initial: "autoForSmartphoneBrowsersSupportingInflation",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "experimental",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/text-size-adjust"
	},
		"text-transform": {
		syntax: "none | capitalize | uppercase | lowercase | full-width | full-size-kana",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Text"
		],
		initial: "none",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/text-transform"
	},
		"text-underline-offset": {
		syntax: "auto | <length> | <percentage> ",
		media: "visual",
		inherited: true,
		animationType: "byComputedValueType",
		percentages: "referToElementFontSize",
		groups: [
			"CSS Text Decoration"
		],
		initial: "auto",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/text-underline-offset"
	},
		"text-underline-position": {
		syntax: "auto | from-font | [ under || [ left | right ] ]",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Text Decoration"
		],
		initial: "auto",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "orderOfAppearance",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/text-underline-position"
	},
		top: top,
		"touch-action": {
		syntax: "auto | none | [ [ pan-x | pan-left | pan-right ] || [ pan-y | pan-up | pan-down ] || pinch-zoom ] | manipulation",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"Pointer Events"
		],
		initial: "auto",
		appliesto: "allElementsExceptNonReplacedInlineElementsTableRowsColumnsRowColumnGroups",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/touch-action"
	},
		transform: transform,
		"transform-box": {
		syntax: "content-box | border-box | fill-box | stroke-box | view-box",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Transforms"
		],
		initial: "view-box",
		appliesto: "transformableElements",
		computed: "asSpecified",
		order: "perGrammar",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/transform-box"
	},
		"transform-origin": {
		syntax: "[ <length-percentage> | left | center | right | top | bottom ] | [ [ <length-percentage> | left | center | right ] && [ <length-percentage> | top | center | bottom ] ] <length>?",
		media: "visual",
		inherited: false,
		animationType: "simpleListOfLpc",
		percentages: "referToSizeOfBoundingBox",
		groups: [
			"CSS Transforms"
		],
		initial: "50% 50% 0",
		appliesto: "transformableElements",
		computed: "forLengthAbsoluteValueOtherwisePercentage",
		order: "oneOrTwoValuesLengthAbsoluteKeywordsPercentages",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/transform-origin"
	},
		"transform-style": {
		syntax: "flat | preserve-3d",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Transforms"
		],
		initial: "flat",
		appliesto: "transformableElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		stacking: true,
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/transform-style"
	},
		transition: transition,
		"transition-delay": {
		syntax: "<time>#",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Transitions"
		],
		initial: "0s",
		appliesto: "allElementsAndPseudos",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/transition-delay"
	},
		"transition-duration": {
		syntax: "<time>#",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Transitions"
		],
		initial: "0s",
		appliesto: "allElementsAndPseudos",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/transition-duration"
	},
		"transition-property": {
		syntax: "none | <single-transition-property>#",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Transitions"
		],
		initial: "all",
		appliesto: "allElementsAndPseudos",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/transition-property"
	},
		"transition-timing-function": {
		syntax: "<timing-function>#",
		media: "interactive",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Transitions"
		],
		initial: "ease",
		appliesto: "allElementsAndPseudos",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/transition-timing-function"
	},
		translate: translate,
		"unicode-bidi": {
		syntax: "normal | embed | isolate | bidi-override | isolate-override | plaintext",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Writing Modes"
		],
		initial: "normal",
		appliesto: "allElementsSomeValuesNoEffectOnNonInlineElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/unicode-bidi"
	},
		"user-select": {
		syntax: "auto | text | none | contain | all",
		media: "visual",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Basic User Interface"
		],
		initial: "auto",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "nonstandard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/user-select"
	},
		"vertical-align": {
		syntax: "baseline | sub | super | text-top | text-bottom | middle | top | bottom | <percentage> | <length>",
		media: "visual",
		inherited: false,
		animationType: "length",
		percentages: "referToLineHeight",
		groups: [
			"CSS Table"
		],
		initial: "baseline",
		appliesto: "inlineLevelAndTableCellElements",
		computed: "absoluteLengthOrKeyword",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/vertical-align"
	},
		visibility: visibility,
		"white-space": {
		syntax: "normal | pre | nowrap | pre-wrap | pre-line | break-spaces",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Text"
		],
		initial: "normal",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/white-space"
	},
		widows: widows,
		width: width,
		"will-change": {
		syntax: "auto | <animateable-feature>#",
		media: "all",
		inherited: false,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Will Change"
		],
		initial: "auto",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/will-change"
	},
		"word-break": {
		syntax: "normal | break-all | keep-all | break-word",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Text"
		],
		initial: "normal",
		appliesto: "allElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/word-break"
	},
		"word-spacing": {
		syntax: "normal | <length-percentage>",
		media: "visual",
		inherited: true,
		animationType: "length",
		percentages: "referToWidthOfAffectedGlyph",
		groups: [
			"CSS Text"
		],
		initial: "normal",
		appliesto: "allElements",
		computed: "optimumMinAndMaxValueOfAbsoluteLengthPercentageOrNormal",
		order: "uniqueOrder",
		alsoAppliesTo: [
			"::first-letter",
			"::first-line",
			"::placeholder"
		],
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/word-spacing"
	},
		"word-wrap": {
		syntax: "normal | break-word",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Text"
		],
		initial: "normal",
		appliesto: "nonReplacedInlineElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/overflow-wrap"
	},
		"writing-mode": {
		syntax: "horizontal-tb | vertical-rl | vertical-lr | sideways-rl | sideways-lr",
		media: "visual",
		inherited: true,
		animationType: "discrete",
		percentages: "no",
		groups: [
			"CSS Writing Modes"
		],
		initial: "horizontal-tb",
		appliesto: "allElementsExceptTableRowColumnGroupsTableRowsColumns",
		computed: "asSpecified",
		order: "uniqueOrder",
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/writing-mode"
	},
		"z-index": {
		syntax: "auto | <integer>",
		media: "visual",
		inherited: false,
		animationType: "integer",
		percentages: "no",
		groups: [
			"CSS Positioning"
		],
		initial: "auto",
		appliesto: "positionedElements",
		computed: "asSpecified",
		order: "uniqueOrder",
		stacking: true,
		status: "standard",
		mdn_url: "https://developer.mozilla.org/docs/Web/CSS/z-index"
	},
		zoom: zoom
	};

	var attachment = {
		syntax: "scroll | fixed | local"
	};
	var box = {
		syntax: "border-box | padding-box | content-box"
	};
	var color = {
		syntax: "<rgb()> | <rgba()> | <hsl()> | <hsla()> | <hex-color> | <named-color> | currentcolor | <deprecated-system-color>"
	};
	var combinator = {
		syntax: "'>' | '+' | '~' | [ '||' ]"
	};
	var gradient = {
		syntax: "<linear-gradient()> | <repeating-linear-gradient()> | <radial-gradient()> | <repeating-radial-gradient()> | <conic-gradient()>"
	};
	var hue = {
		syntax: "<number> | <angle>"
	};
	var image = {
		syntax: "<url> | <image()> | <image-set()> | <element()> | <paint()> | <cross-fade()> | <gradient>"
	};
	var nth$1 = {
		syntax: "<an-plus-b> | even | odd"
	};
	var position = {
		syntax: "[ [ left | center | right ] || [ top | center | bottom ] | [ left | center | right | <length-percentage> ] [ top | center | bottom | <length-percentage> ]? | [ [ left | right ] <length-percentage> ] && [ [ top | bottom ] <length-percentage> ] ]"
	};
	var quote = {
		syntax: "open-quote | close-quote | no-open-quote | no-close-quote"
	};
	var shadow = {
		syntax: "inset? && <length>{2,4} && <color>?"
	};
	var shape = {
		syntax: "rect(<top>, <right>, <bottom>, <left>)"
	};
	var size = {
		syntax: "closest-side | farthest-side | closest-corner | farthest-corner | <length> | <length-percentage>{2}"
	};
	var symbol = {
		syntax: "<string> | <image> | <custom-ident>"
	};
	var target = {
		syntax: "<target-counter()> | <target-counters()> | <target-text()>"
	};
	var require$$2 = {
		"absolute-size": {
		syntax: "xx-small | x-small | small | medium | large | x-large | xx-large | xxx-large"
	},
		"alpha-value": {
		syntax: "<number> | <percentage>"
	},
		"angle-percentage": {
		syntax: "<angle> | <percentage>"
	},
		"angular-color-hint": {
		syntax: "<angle-percentage>"
	},
		"angular-color-stop": {
		syntax: "<color> && <color-stop-angle>?"
	},
		"angular-color-stop-list": {
		syntax: "[ <angular-color-stop> [, <angular-color-hint>]? ]# , <angular-color-stop>"
	},
		"animateable-feature": {
		syntax: "scroll-position | contents | <custom-ident>"
	},
		attachment: attachment,
		"attr()": {
		syntax: "attr( <attr-name> <type-or-unit>? [, <attr-fallback> ]? )"
	},
		"attr-matcher": {
		syntax: "[ '~' | '|' | '^' | '$' | '*' ]? '='"
	},
		"attr-modifier": {
		syntax: "i | s"
	},
		"attribute-selector": {
		syntax: "'[' <wq-name> ']' | '[' <wq-name> <attr-matcher> [ <string-token> | <ident-token> ] <attr-modifier>? ']'"
	},
		"auto-repeat": {
		syntax: "repeat( [ auto-fill | auto-fit ] , [ <line-names>? <fixed-size> ]+ <line-names>? )"
	},
		"auto-track-list": {
		syntax: "[ <line-names>? [ <fixed-size> | <fixed-repeat> ] ]* <line-names>? <auto-repeat>\n[ <line-names>? [ <fixed-size> | <fixed-repeat> ] ]* <line-names>?"
	},
		"baseline-position": {
		syntax: "[ first | last ]? baseline"
	},
		"basic-shape": {
		syntax: "<inset()> | <circle()> | <ellipse()> | <polygon()> | <path()>"
	},
		"bg-image": {
		syntax: "none | <image>"
	},
		"bg-layer": {
		syntax: "<bg-image> || <bg-position> [ / <bg-size> ]? || <repeat-style> || <attachment> || <box> || <box>"
	},
		"bg-position": {
		syntax: "[ [ left | center | right | top | bottom | <length-percentage> ] | [ left | center | right | <length-percentage> ] [ top | center | bottom | <length-percentage> ] | [ center | [ left | right ] <length-percentage>? ] && [ center | [ top | bottom ] <length-percentage>? ] ]"
	},
		"bg-size": {
		syntax: "[ <length-percentage> | auto ]{1,2} | cover | contain"
	},
		"blur()": {
		syntax: "blur( <length> )"
	},
		"blend-mode": {
		syntax: "normal | multiply | screen | overlay | darken | lighten | color-dodge | color-burn | hard-light | soft-light | difference | exclusion | hue | saturation | color | luminosity"
	},
		box: box,
		"brightness()": {
		syntax: "brightness( <number-percentage> )"
	},
		"calc()": {
		syntax: "calc( <calc-sum> )"
	},
		"calc-sum": {
		syntax: "<calc-product> [ [ '+' | '-' ] <calc-product> ]*"
	},
		"calc-product": {
		syntax: "<calc-value> [ '*' <calc-value> | '/' <number> ]*"
	},
		"calc-value": {
		syntax: "<number> | <dimension> | <percentage> | ( <calc-sum> )"
	},
		"cf-final-image": {
		syntax: "<image> | <color>"
	},
		"cf-mixing-image": {
		syntax: "<percentage>? && <image>"
	},
		"circle()": {
		syntax: "circle( [ <shape-radius> ]? [ at <position> ]? )"
	},
		"clamp()": {
		syntax: "clamp( <calc-sum>#{3} )"
	},
		"class-selector": {
		syntax: "'.' <ident-token>"
	},
		"clip-source": {
		syntax: "<url>"
	},
		color: color,
		"color-stop": {
		syntax: "<color-stop-length> | <color-stop-angle>"
	},
		"color-stop-angle": {
		syntax: "<angle-percentage>{1,2}"
	},
		"color-stop-length": {
		syntax: "<length-percentage>{1,2}"
	},
		"color-stop-list": {
		syntax: "[ <linear-color-stop> [, <linear-color-hint>]? ]# , <linear-color-stop>"
	},
		combinator: combinator,
		"common-lig-values": {
		syntax: "[ common-ligatures | no-common-ligatures ]"
	},
		"compat-auto": {
		syntax: "searchfield | textarea | push-button | slider-horizontal | checkbox | radio | square-button | menulist | listbox | meter | progress-bar | button"
	},
		"composite-style": {
		syntax: "clear | copy | source-over | source-in | source-out | source-atop | destination-over | destination-in | destination-out | destination-atop | xor"
	},
		"compositing-operator": {
		syntax: "add | subtract | intersect | exclude"
	},
		"compound-selector": {
		syntax: "[ <type-selector>? <subclass-selector>* [ <pseudo-element-selector> <pseudo-class-selector>* ]* ]!"
	},
		"compound-selector-list": {
		syntax: "<compound-selector>#"
	},
		"complex-selector": {
		syntax: "<compound-selector> [ <combinator>? <compound-selector> ]*"
	},
		"complex-selector-list": {
		syntax: "<complex-selector>#"
	},
		"conic-gradient()": {
		syntax: "conic-gradient( [ from <angle> ]? [ at <position> ]?, <angular-color-stop-list> )"
	},
		"contextual-alt-values": {
		syntax: "[ contextual | no-contextual ]"
	},
		"content-distribution": {
		syntax: "space-between | space-around | space-evenly | stretch"
	},
		"content-list": {
		syntax: "[ <string> | contents | <image> | <quote> | <target> | <leader()> ]+"
	},
		"content-position": {
		syntax: "center | start | end | flex-start | flex-end"
	},
		"content-replacement": {
		syntax: "<image>"
	},
		"contrast()": {
		syntax: "contrast( [ <number-percentage> ] )"
	},
		"counter()": {
		syntax: "counter( <custom-ident>, <counter-style>? )"
	},
		"counter-style": {
		syntax: "<counter-style-name> | symbols()"
	},
		"counter-style-name": {
		syntax: "<custom-ident>"
	},
		"counters()": {
		syntax: "counters( <custom-ident>, <string>, <counter-style>? )"
	},
		"cross-fade()": {
		syntax: "cross-fade( <cf-mixing-image> , <cf-final-image>? )"
	},
		"cubic-bezier-timing-function": {
		syntax: "ease | ease-in | ease-out | ease-in-out | cubic-bezier(<number [0,1]>, <number>, <number [0,1]>, <number>)"
	},
		"deprecated-system-color": {
		syntax: "ActiveBorder | ActiveCaption | AppWorkspace | Background | ButtonFace | ButtonHighlight | ButtonShadow | ButtonText | CaptionText | GrayText | Highlight | HighlightText | InactiveBorder | InactiveCaption | InactiveCaptionText | InfoBackground | InfoText | Menu | MenuText | Scrollbar | ThreeDDarkShadow | ThreeDFace | ThreeDHighlight | ThreeDLightShadow | ThreeDShadow | Window | WindowFrame | WindowText"
	},
		"discretionary-lig-values": {
		syntax: "[ discretionary-ligatures | no-discretionary-ligatures ]"
	},
		"display-box": {
		syntax: "contents | none"
	},
		"display-inside": {
		syntax: "flow | flow-root | table | flex | grid | ruby"
	},
		"display-internal": {
		syntax: "table-row-group | table-header-group | table-footer-group | table-row | table-cell | table-column-group | table-column | table-caption | ruby-base | ruby-text | ruby-base-container | ruby-text-container"
	},
		"display-legacy": {
		syntax: "inline-block | inline-list-item | inline-table | inline-flex | inline-grid"
	},
		"display-listitem": {
		syntax: "<display-outside>? && [ flow | flow-root ]? && list-item"
	},
		"display-outside": {
		syntax: "block | inline | run-in"
	},
		"drop-shadow()": {
		syntax: "drop-shadow( <length>{2,3} <color>? )"
	},
		"east-asian-variant-values": {
		syntax: "[ jis78 | jis83 | jis90 | jis04 | simplified | traditional ]"
	},
		"east-asian-width-values": {
		syntax: "[ full-width | proportional-width ]"
	},
		"element()": {
		syntax: "element( <id-selector> )"
	},
		"ellipse()": {
		syntax: "ellipse( [ <shape-radius>{2} ]? [ at <position> ]? )"
	},
		"ending-shape": {
		syntax: "circle | ellipse"
	},
		"env()": {
		syntax: "env( <custom-ident> , <declaration-value>? )"
	},
		"explicit-track-list": {
		syntax: "[ <line-names>? <track-size> ]+ <line-names>?"
	},
		"family-name": {
		syntax: "<string> | <custom-ident>+"
	},
		"feature-tag-value": {
		syntax: "<string> [ <integer> | on | off ]?"
	},
		"feature-type": {
		syntax: "@stylistic | @historical-forms | @styleset | @character-variant | @swash | @ornaments | @annotation"
	},
		"feature-value-block": {
		syntax: "<feature-type> '{' <feature-value-declaration-list> '}'"
	},
		"feature-value-block-list": {
		syntax: "<feature-value-block>+"
	},
		"feature-value-declaration": {
		syntax: "<custom-ident>: <integer>+;"
	},
		"feature-value-declaration-list": {
		syntax: "<feature-value-declaration>"
	},
		"feature-value-name": {
		syntax: "<custom-ident>"
	},
		"fill-rule": {
		syntax: "nonzero | evenodd"
	},
		"filter-function": {
		syntax: "<blur()> | <brightness()> | <contrast()> | <drop-shadow()> | <grayscale()> | <hue-rotate()> | <invert()> | <opacity()> | <saturate()> | <sepia()>"
	},
		"filter-function-list": {
		syntax: "[ <filter-function> | <url> ]+"
	},
		"final-bg-layer": {
		syntax: "<'background-color'> || <bg-image> || <bg-position> [ / <bg-size> ]? || <repeat-style> || <attachment> || <box> || <box>"
	},
		"fit-content()": {
		syntax: "fit-content( [ <length> | <percentage> ] )"
	},
		"fixed-breadth": {
		syntax: "<length-percentage>"
	},
		"fixed-repeat": {
		syntax: "repeat( [ <positive-integer> ] , [ <line-names>? <fixed-size> ]+ <line-names>? )"
	},
		"fixed-size": {
		syntax: "<fixed-breadth> | minmax( <fixed-breadth> , <track-breadth> ) | minmax( <inflexible-breadth> , <fixed-breadth> )"
	},
		"font-stretch-absolute": {
		syntax: "normal | ultra-condensed | extra-condensed | condensed | semi-condensed | semi-expanded | expanded | extra-expanded | ultra-expanded | <percentage>"
	},
		"font-variant-css21": {
		syntax: "[ normal | small-caps ]"
	},
		"font-weight-absolute": {
		syntax: "normal | bold | <number [1,1000]>"
	},
		"frequency-percentage": {
		syntax: "<frequency> | <percentage>"
	},
		"general-enclosed": {
		syntax: "[ <function-token> <any-value> ) ] | ( <ident> <any-value> )"
	},
		"generic-family": {
		syntax: "serif | sans-serif | cursive | fantasy | monospace"
	},
		"generic-name": {
		syntax: "serif | sans-serif | cursive | fantasy | monospace"
	},
		"geometry-box": {
		syntax: "<shape-box> | fill-box | stroke-box | view-box"
	},
		gradient: gradient,
		"grayscale()": {
		syntax: "grayscale( <number-percentage> )"
	},
		"grid-line": {
		syntax: "auto | <custom-ident> | [ <integer> && <custom-ident>? ] | [ span && [ <integer> || <custom-ident> ] ]"
	},
		"historical-lig-values": {
		syntax: "[ historical-ligatures | no-historical-ligatures ]"
	},
		"hsl()": {
		syntax: "hsl( <hue> <percentage> <percentage> [ / <alpha-value> ]? ) | hsl( <hue>, <percentage>, <percentage>, <alpha-value>? )"
	},
		"hsla()": {
		syntax: "hsla( <hue> <percentage> <percentage> [ / <alpha-value> ]? ) | hsla( <hue>, <percentage>, <percentage>, <alpha-value>? )"
	},
		hue: hue,
		"hue-rotate()": {
		syntax: "hue-rotate( <angle> )"
	},
		"id-selector": {
		syntax: "<hash-token>"
	},
		image: image,
		"image()": {
		syntax: "image( <image-tags>? [ <image-src>? , <color>? ]! )"
	},
		"image-set()": {
		syntax: "image-set( <image-set-option># )"
	},
		"image-set-option": {
		syntax: "[ <image> | <string> ] <resolution>"
	},
		"image-src": {
		syntax: "<url> | <string>"
	},
		"image-tags": {
		syntax: "ltr | rtl"
	},
		"inflexible-breadth": {
		syntax: "<length> | <percentage> | min-content | max-content | auto"
	},
		"inset()": {
		syntax: "inset( <length-percentage>{1,4} [ round <'border-radius'> ]? )"
	},
		"invert()": {
		syntax: "invert( <number-percentage> )"
	},
		"keyframes-name": {
		syntax: "<custom-ident> | <string>"
	},
		"keyframe-block": {
		syntax: "<keyframe-selector># {\n  <declaration-list>\n}"
	},
		"keyframe-block-list": {
		syntax: "<keyframe-block>+"
	},
		"keyframe-selector": {
		syntax: "from | to | <percentage>"
	},
		"leader()": {
		syntax: "leader( <leader-type> )"
	},
		"leader-type": {
		syntax: "dotted | solid | space | <string>"
	},
		"length-percentage": {
		syntax: "<length> | <percentage>"
	},
		"line-names": {
		syntax: "'[' <custom-ident>* ']'"
	},
		"line-name-list": {
		syntax: "[ <line-names> | <name-repeat> ]+"
	},
		"line-style": {
		syntax: "none | hidden | dotted | dashed | solid | double | groove | ridge | inset | outset"
	},
		"line-width": {
		syntax: "<length> | thin | medium | thick"
	},
		"linear-color-hint": {
		syntax: "<length-percentage>"
	},
		"linear-color-stop": {
		syntax: "<color> <color-stop-length>?"
	},
		"linear-gradient()": {
		syntax: "linear-gradient( [ <angle> | to <side-or-corner> ]? , <color-stop-list> )"
	},
		"mask-layer": {
		syntax: "<mask-reference> || <position> [ / <bg-size> ]? || <repeat-style> || <geometry-box> || [ <geometry-box> | no-clip ] || <compositing-operator> || <masking-mode>"
	},
		"mask-position": {
		syntax: "[ <length-percentage> | left | center | right ] [ <length-percentage> | top | center | bottom ]?"
	},
		"mask-reference": {
		syntax: "none | <image> | <mask-source>"
	},
		"mask-source": {
		syntax: "<url>"
	},
		"masking-mode": {
		syntax: "alpha | luminance | match-source"
	},
		"matrix()": {
		syntax: "matrix( <number>#{6} )"
	},
		"matrix3d()": {
		syntax: "matrix3d( <number>#{16} )"
	},
		"max()": {
		syntax: "max( <calc-sum># )"
	},
		"media-and": {
		syntax: "<media-in-parens> [ and <media-in-parens> ]+"
	},
		"media-condition": {
		syntax: "<media-not> | <media-and> | <media-or> | <media-in-parens>"
	},
		"media-condition-without-or": {
		syntax: "<media-not> | <media-and> | <media-in-parens>"
	},
		"media-feature": {
		syntax: "( [ <mf-plain> | <mf-boolean> | <mf-range> ] )"
	},
		"media-in-parens": {
		syntax: "( <media-condition> ) | <media-feature> | <general-enclosed>"
	},
		"media-not": {
		syntax: "not <media-in-parens>"
	},
		"media-or": {
		syntax: "<media-in-parens> [ or <media-in-parens> ]+"
	},
		"media-query": {
		syntax: "<media-condition> | [ not | only ]? <media-type> [ and <media-condition-without-or> ]?"
	},
		"media-query-list": {
		syntax: "<media-query>#"
	},
		"media-type": {
		syntax: "<ident>"
	},
		"mf-boolean": {
		syntax: "<mf-name>"
	},
		"mf-name": {
		syntax: "<ident>"
	},
		"mf-plain": {
		syntax: "<mf-name> : <mf-value>"
	},
		"mf-range": {
		syntax: "<mf-name> [ '<' | '>' ]? '='? <mf-value>\n| <mf-value> [ '<' | '>' ]? '='? <mf-name>\n| <mf-value> '<' '='? <mf-name> '<' '='? <mf-value>\n| <mf-value> '>' '='? <mf-name> '>' '='? <mf-value>"
	},
		"mf-value": {
		syntax: "<number> | <dimension> | <ident> | <ratio>"
	},
		"min()": {
		syntax: "min( <calc-sum># )"
	},
		"minmax()": {
		syntax: "minmax( [ <length> | <percentage> | min-content | max-content | auto ] , [ <length> | <percentage> | <flex> | min-content | max-content | auto ] )"
	},
		"named-color": {
		syntax: "transparent | aliceblue | antiquewhite | aqua | aquamarine | azure | beige | bisque | black | blanchedalmond | blue | blueviolet | brown | burlywood | cadetblue | chartreuse | chocolate | coral | cornflowerblue | cornsilk | crimson | cyan | darkblue | darkcyan | darkgoldenrod | darkgray | darkgreen | darkgrey | darkkhaki | darkmagenta | darkolivegreen | darkorange | darkorchid | darkred | darksalmon | darkseagreen | darkslateblue | darkslategray | darkslategrey | darkturquoise | darkviolet | deeppink | deepskyblue | dimgray | dimgrey | dodgerblue | firebrick | floralwhite | forestgreen | fuchsia | gainsboro | ghostwhite | gold | goldenrod | gray | green | greenyellow | grey | honeydew | hotpink | indianred | indigo | ivory | khaki | lavender | lavenderblush | lawngreen | lemonchiffon | lightblue | lightcoral | lightcyan | lightgoldenrodyellow | lightgray | lightgreen | lightgrey | lightpink | lightsalmon | lightseagreen | lightskyblue | lightslategray | lightslategrey | lightsteelblue | lightyellow | lime | limegreen | linen | magenta | maroon | mediumaquamarine | mediumblue | mediumorchid | mediumpurple | mediumseagreen | mediumslateblue | mediumspringgreen | mediumturquoise | mediumvioletred | midnightblue | mintcream | mistyrose | moccasin | navajowhite | navy | oldlace | olive | olivedrab | orange | orangered | orchid | palegoldenrod | palegreen | paleturquoise | palevioletred | papayawhip | peachpuff | peru | pink | plum | powderblue | purple | rebeccapurple | red | rosybrown | royalblue | saddlebrown | salmon | sandybrown | seagreen | seashell | sienna | silver | skyblue | slateblue | slategray | slategrey | snow | springgreen | steelblue | tan | teal | thistle | tomato | turquoise | violet | wheat | white | whitesmoke | yellow | yellowgreen"
	},
		"namespace-prefix": {
		syntax: "<ident>"
	},
		"ns-prefix": {
		syntax: "[ <ident-token> | '*' ]? '|'"
	},
		"number-percentage": {
		syntax: "<number> | <percentage>"
	},
		"numeric-figure-values": {
		syntax: "[ lining-nums | oldstyle-nums ]"
	},
		"numeric-fraction-values": {
		syntax: "[ diagonal-fractions | stacked-fractions ]"
	},
		"numeric-spacing-values": {
		syntax: "[ proportional-nums | tabular-nums ]"
	},
		nth: nth$1,
		"opacity()": {
		syntax: "opacity( [ <number-percentage> ] )"
	},
		"overflow-position": {
		syntax: "unsafe | safe"
	},
		"outline-radius": {
		syntax: "<length> | <percentage>"
	},
		"page-body": {
		syntax: "<declaration>? [ ; <page-body> ]? | <page-margin-box> <page-body>"
	},
		"page-margin-box": {
		syntax: "<page-margin-box-type> '{' <declaration-list> '}'"
	},
		"page-margin-box-type": {
		syntax: "@top-left-corner | @top-left | @top-center | @top-right | @top-right-corner | @bottom-left-corner | @bottom-left | @bottom-center | @bottom-right | @bottom-right-corner | @left-top | @left-middle | @left-bottom | @right-top | @right-middle | @right-bottom"
	},
		"page-selector-list": {
		syntax: "[ <page-selector># ]?"
	},
		"page-selector": {
		syntax: "<pseudo-page>+ | <ident> <pseudo-page>*"
	},
		"path()": {
		syntax: "path( [ <fill-rule>, ]? <string> )"
	},
		"paint()": {
		syntax: "paint( <ident>, <declaration-value>? )"
	},
		"perspective()": {
		syntax: "perspective( <length> )"
	},
		"polygon()": {
		syntax: "polygon( <fill-rule>? , [ <length-percentage> <length-percentage> ]# )"
	},
		position: position,
		"pseudo-class-selector": {
		syntax: "':' <ident-token> | ':' <function-token> <any-value> ')'"
	},
		"pseudo-element-selector": {
		syntax: "':' <pseudo-class-selector>"
	},
		"pseudo-page": {
		syntax: ": [ left | right | first | blank ]"
	},
		quote: quote,
		"radial-gradient()": {
		syntax: "radial-gradient( [ <ending-shape> || <size> ]? [ at <position> ]? , <color-stop-list> )"
	},
		"relative-selector": {
		syntax: "<combinator>? <complex-selector>"
	},
		"relative-selector-list": {
		syntax: "<relative-selector>#"
	},
		"relative-size": {
		syntax: "larger | smaller"
	},
		"repeat-style": {
		syntax: "repeat-x | repeat-y | [ repeat | space | round | no-repeat ]{1,2}"
	},
		"repeating-linear-gradient()": {
		syntax: "repeating-linear-gradient( [ <angle> | to <side-or-corner> ]? , <color-stop-list> )"
	},
		"repeating-radial-gradient()": {
		syntax: "repeating-radial-gradient( [ <ending-shape> || <size> ]? [ at <position> ]? , <color-stop-list> )"
	},
		"rgb()": {
		syntax: "rgb( <percentage>{3} [ / <alpha-value> ]? ) | rgb( <number>{3} [ / <alpha-value> ]? ) | rgb( <percentage>#{3} , <alpha-value>? ) | rgb( <number>#{3} , <alpha-value>? )"
	},
		"rgba()": {
		syntax: "rgba( <percentage>{3} [ / <alpha-value> ]? ) | rgba( <number>{3} [ / <alpha-value> ]? ) | rgba( <percentage>#{3} , <alpha-value>? ) | rgba( <number>#{3} , <alpha-value>? )"
	},
		"rotate()": {
		syntax: "rotate( [ <angle> | <zero> ] )"
	},
		"rotate3d()": {
		syntax: "rotate3d( <number> , <number> , <number> , [ <angle> | <zero> ] )"
	},
		"rotateX()": {
		syntax: "rotateX( [ <angle> | <zero> ] )"
	},
		"rotateY()": {
		syntax: "rotateY( [ <angle> | <zero> ] )"
	},
		"rotateZ()": {
		syntax: "rotateZ( [ <angle> | <zero> ] )"
	},
		"saturate()": {
		syntax: "saturate( <number-percentage> )"
	},
		"scale()": {
		syntax: "scale( <number> , <number>? )"
	},
		"scale3d()": {
		syntax: "scale3d( <number> , <number> , <number> )"
	},
		"scaleX()": {
		syntax: "scaleX( <number> )"
	},
		"scaleY()": {
		syntax: "scaleY( <number> )"
	},
		"scaleZ()": {
		syntax: "scaleZ( <number> )"
	},
		"self-position": {
		syntax: "center | start | end | self-start | self-end | flex-start | flex-end"
	},
		"shape-radius": {
		syntax: "<length-percentage> | closest-side | farthest-side"
	},
		"skew()": {
		syntax: "skew( [ <angle> | <zero> ] , [ <angle> | <zero> ]? )"
	},
		"skewX()": {
		syntax: "skewX( [ <angle> | <zero> ] )"
	},
		"skewY()": {
		syntax: "skewY( [ <angle> | <zero> ] )"
	},
		"sepia()": {
		syntax: "sepia( <number-percentage> )"
	},
		shadow: shadow,
		"shadow-t": {
		syntax: "[ <length>{2,3} && <color>? ]"
	},
		shape: shape,
		"shape-box": {
		syntax: "<box> | margin-box"
	},
		"side-or-corner": {
		syntax: "[ left | right ] || [ top | bottom ]"
	},
		"single-animation": {
		syntax: "<time> || <timing-function> || <time> || <single-animation-iteration-count> || <single-animation-direction> || <single-animation-fill-mode> || <single-animation-play-state> || [ none | <keyframes-name> ]"
	},
		"single-animation-direction": {
		syntax: "normal | reverse | alternate | alternate-reverse"
	},
		"single-animation-fill-mode": {
		syntax: "none | forwards | backwards | both"
	},
		"single-animation-iteration-count": {
		syntax: "infinite | <number>"
	},
		"single-animation-play-state": {
		syntax: "running | paused"
	},
		"single-transition": {
		syntax: "[ none | <single-transition-property> ] || <time> || <timing-function> || <time>"
	},
		"single-transition-property": {
		syntax: "all | <custom-ident>"
	},
		size: size,
		"step-position": {
		syntax: "jump-start | jump-end | jump-none | jump-both | start | end"
	},
		"step-timing-function": {
		syntax: "step-start | step-end | steps(<integer>[, <step-position>]?)"
	},
		"subclass-selector": {
		syntax: "<id-selector> | <class-selector> | <attribute-selector> | <pseudo-class-selector>"
	},
		"supports-condition": {
		syntax: "not <supports-in-parens> | <supports-in-parens> [ and <supports-in-parens> ]* | <supports-in-parens> [ or <supports-in-parens> ]*"
	},
		"supports-in-parens": {
		syntax: "( <supports-condition> ) | <supports-feature> | <general-enclosed>"
	},
		"supports-feature": {
		syntax: "<supports-decl> | <supports-selector-fn>"
	},
		"supports-decl": {
		syntax: "( <declaration> )"
	},
		"supports-selector-fn": {
		syntax: "selector( <complex-selector> )"
	},
		symbol: symbol,
		target: target,
		"target-counter()": {
		syntax: "target-counter( [ <string> | <url> ] , <custom-ident> , <counter-style>? )"
	},
		"target-counters()": {
		syntax: "target-counters( [ <string> | <url> ] , <custom-ident> , <string> , <counter-style>? )"
	},
		"target-text()": {
		syntax: "target-text( [ <string> | <url> ] , [ content | before | after | first-letter ]? )"
	},
		"time-percentage": {
		syntax: "<time> | <percentage>"
	},
		"timing-function": {
		syntax: "linear | <cubic-bezier-timing-function> | <step-timing-function>"
	},
		"track-breadth": {
		syntax: "<length-percentage> | <flex> | min-content | max-content | auto"
	},
		"track-list": {
		syntax: "[ <line-names>? [ <track-size> | <track-repeat> ] ]+ <line-names>?"
	},
		"track-repeat": {
		syntax: "repeat( [ <positive-integer> ] , [ <line-names>? <track-size> ]+ <line-names>? )"
	},
		"track-size": {
		syntax: "<track-breadth> | minmax( <inflexible-breadth> , <track-breadth> ) | fit-content( [ <length> | <percentage> ] )"
	},
		"transform-function": {
		syntax: "<matrix()> | <translate()> | <translateX()> | <translateY()> | <scale()> | <scaleX()> | <scaleY()> | <rotate()> | <skew()> | <skewX()> | <skewY()> | <matrix3d()> | <translate3d()> | <translateZ()> | <scale3d()> | <scaleZ()> | <rotate3d()> | <rotateX()> | <rotateY()> | <rotateZ()> | <perspective()>"
	},
		"transform-list": {
		syntax: "<transform-function>+"
	},
		"translate()": {
		syntax: "translate( <length-percentage> , <length-percentage>? )"
	},
		"translate3d()": {
		syntax: "translate3d( <length-percentage> , <length-percentage> , <length> )"
	},
		"translateX()": {
		syntax: "translateX( <length-percentage> )"
	},
		"translateY()": {
		syntax: "translateY( <length-percentage> )"
	},
		"translateZ()": {
		syntax: "translateZ( <length> )"
	},
		"type-or-unit": {
		syntax: "string | color | url | integer | number | length | angle | time | frequency | cap | ch | em | ex | ic | lh | rlh | rem | vb | vi | vw | vh | vmin | vmax | mm | Q | cm | in | pt | pc | px | deg | grad | rad | turn | ms | s | Hz | kHz | %"
	},
		"type-selector": {
		syntax: "<wq-name> | <ns-prefix>? '*'"
	},
		"var()": {
		syntax: "var( <custom-property-name> , <declaration-value>? )"
	},
		"viewport-length": {
		syntax: "auto | <length-percentage>"
	},
		"wq-name": {
		syntax: "<ns-prefix>? <ident-token>"
	}
	};

	var atrules = {
		charset: {
			prelude: "<string>"
		},
		"font-face": {
			descriptors: {
				"unicode-range": {
					comment: "replaces <unicode-range>, an old production name",
					syntax: "<urange>#"
				}
			}
		}
	};
	var properties = {
		"-moz-background-clip": {
			comment: "deprecated syntax in old Firefox, https://developer.mozilla.org/en/docs/Web/CSS/background-clip",
			syntax: "padding | border"
		},
		"-moz-border-radius-bottomleft": {
			comment: "https://developer.mozilla.org/en-US/docs/Web/CSS/border-bottom-left-radius",
			syntax: "<'border-bottom-left-radius'>"
		},
		"-moz-border-radius-bottomright": {
			comment: "https://developer.mozilla.org/en-US/docs/Web/CSS/border-bottom-right-radius",
			syntax: "<'border-bottom-right-radius'>"
		},
		"-moz-border-radius-topleft": {
			comment: "https://developer.mozilla.org/en-US/docs/Web/CSS/border-top-left-radius",
			syntax: "<'border-top-left-radius'>"
		},
		"-moz-border-radius-topright": {
			comment: "https://developer.mozilla.org/en-US/docs/Web/CSS/border-bottom-right-radius",
			syntax: "<'border-bottom-right-radius'>"
		},
		"-moz-control-character-visibility": {
			comment: "firefox specific keywords, https://bugzilla.mozilla.org/show_bug.cgi?id=947588",
			syntax: "visible | hidden"
		},
		"-moz-osx-font-smoothing": {
			comment: "misssed old syntax https://developer.mozilla.org/en-US/docs/Web/CSS/font-smooth",
			syntax: "auto | grayscale"
		},
		"-moz-user-select": {
			comment: "https://developer.mozilla.org/en-US/docs/Web/CSS/user-select",
			syntax: "none | text | all | -moz-none"
		},
		"-ms-flex-align": {
			comment: "misssed old syntax implemented in IE, https://www.w3.org/TR/2012/WD-css3-flexbox-20120322/#flex-align",
			syntax: "start | end | center | baseline | stretch"
		},
		"-ms-flex-item-align": {
			comment: "misssed old syntax implemented in IE, https://www.w3.org/TR/2012/WD-css3-flexbox-20120322/#flex-align",
			syntax: "auto | start | end | center | baseline | stretch"
		},
		"-ms-flex-line-pack": {
			comment: "misssed old syntax implemented in IE, https://www.w3.org/TR/2012/WD-css3-flexbox-20120322/#flex-line-pack",
			syntax: "start | end | center | justify | distribute | stretch"
		},
		"-ms-flex-negative": {
			comment: "misssed old syntax implemented in IE; TODO: find references for comfirmation",
			syntax: "<'flex-shrink'>"
		},
		"-ms-flex-pack": {
			comment: "misssed old syntax implemented in IE, https://www.w3.org/TR/2012/WD-css3-flexbox-20120322/#flex-pack",
			syntax: "start | end | center | justify | distribute"
		},
		"-ms-flex-order": {
			comment: "misssed old syntax implemented in IE; https://msdn.microsoft.com/en-us/library/jj127303(v=vs.85).aspx",
			syntax: "<integer>"
		},
		"-ms-flex-positive": {
			comment: "misssed old syntax implemented in IE; TODO: find references for comfirmation",
			syntax: "<'flex-grow'>"
		},
		"-ms-flex-preferred-size": {
			comment: "misssed old syntax implemented in IE; TODO: find references for comfirmation",
			syntax: "<'flex-basis'>"
		},
		"-ms-interpolation-mode": {
			comment: "https://msdn.microsoft.com/en-us/library/ff521095(v=vs.85).aspx",
			syntax: "nearest-neighbor | bicubic"
		},
		"-ms-grid-column-align": {
			comment: "add this property first since it uses as fallback for flexbox, https://msdn.microsoft.com/en-us/library/windows/apps/hh466338.aspx",
			syntax: "start | end | center | stretch"
		},
		"-ms-grid-row-align": {
			comment: "add this property first since it uses as fallback for flexbox, https://msdn.microsoft.com/en-us/library/windows/apps/hh466348.aspx",
			syntax: "start | end | center | stretch"
		},
		"-ms-hyphenate-limit-last": {
			comment: "misssed old syntax implemented in IE; https://www.w3.org/TR/css-text-4/#hyphenate-line-limits",
			syntax: "none | always | column | page | spread"
		},
		"-webkit-appearance": {
			comment: "webkit specific keywords",
			references: [
				"http://css-infos.net/property/-webkit-appearance"
			],
			syntax: "none | button | button-bevel | caps-lock-indicator | caret | checkbox | default-button | inner-spin-button | listbox | listitem | media-controls-background | media-controls-fullscreen-background | media-current-time-display | media-enter-fullscreen-button | media-exit-fullscreen-button | media-fullscreen-button | media-mute-button | media-overlay-play-button | media-play-button | media-seek-back-button | media-seek-forward-button | media-slider | media-sliderthumb | media-time-remaining-display | media-toggle-closed-captions-button | media-volume-slider | media-volume-slider-container | media-volume-sliderthumb | menulist | menulist-button | menulist-text | menulist-textfield | meter | progress-bar | progress-bar-value | push-button | radio | scrollbarbutton-down | scrollbarbutton-left | scrollbarbutton-right | scrollbarbutton-up | scrollbargripper-horizontal | scrollbargripper-vertical | scrollbarthumb-horizontal | scrollbarthumb-vertical | scrollbartrack-horizontal | scrollbartrack-vertical | searchfield | searchfield-cancel-button | searchfield-decoration | searchfield-results-button | searchfield-results-decoration | slider-horizontal | slider-vertical | sliderthumb-horizontal | sliderthumb-vertical | square-button | textarea | textfield | -apple-pay-button"
		},
		"-webkit-background-clip": {
			comment: "https://developer.mozilla.org/en/docs/Web/CSS/background-clip",
			syntax: "[ <box> | border | padding | content | text ]#"
		},
		"-webkit-column-break-after": {
			comment: "added, http://help.dottoro.com/lcrthhhv.php",
			syntax: "always | auto | avoid"
		},
		"-webkit-column-break-before": {
			comment: "added, http://help.dottoro.com/lcxquvkf.php",
			syntax: "always | auto | avoid"
		},
		"-webkit-column-break-inside": {
			comment: "added, http://help.dottoro.com/lclhnthl.php",
			syntax: "always | auto | avoid"
		},
		"-webkit-font-smoothing": {
			comment: "https://developer.mozilla.org/en-US/docs/Web/CSS/font-smooth",
			syntax: "auto | none | antialiased | subpixel-antialiased"
		},
		"-webkit-mask-box-image": {
			comment: "missed; https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-mask-box-image",
			syntax: "[ <url> | <gradient> | none ] [ <length-percentage>{4} <-webkit-mask-box-repeat>{2} ]?"
		},
		"-webkit-print-color-adjust": {
			comment: "missed",
			references: [
				"https://developer.mozilla.org/en/docs/Web/CSS/-webkit-print-color-adjust"
			],
			syntax: "economy | exact"
		},
		"-webkit-text-security": {
			comment: "missed; http://help.dottoro.com/lcbkewgt.php",
			syntax: "none | circle | disc | square"
		},
		"-webkit-user-drag": {
			comment: "missed; http://help.dottoro.com/lcbixvwm.php",
			syntax: "none | element | auto"
		},
		"-webkit-user-select": {
			comment: "auto is supported by old webkit, https://developer.mozilla.org/en-US/docs/Web/CSS/user-select",
			syntax: "auto | none | text | all"
		},
		"alignment-baseline": {
			comment: "added SVG property",
			references: [
				"https://www.w3.org/TR/SVG/text.html#AlignmentBaselineProperty"
			],
			syntax: "auto | baseline | before-edge | text-before-edge | middle | central | after-edge | text-after-edge | ideographic | alphabetic | hanging | mathematical"
		},
		"baseline-shift": {
			comment: "added SVG property",
			references: [
				"https://www.w3.org/TR/SVG/text.html#BaselineShiftProperty"
			],
			syntax: "baseline | sub | super | <svg-length>"
		},
		behavior: {
			comment: "added old IE property https://msdn.microsoft.com/en-us/library/ms530723(v=vs.85).aspx",
			syntax: "<url>+"
		},
		"clip-rule": {
			comment: "added SVG property",
			references: [
				"https://www.w3.org/TR/SVG/masking.html#ClipRuleProperty"
			],
			syntax: "nonzero | evenodd"
		},
		cue: {
			comment: "https://www.w3.org/TR/css3-speech/#property-index",
			syntax: "<'cue-before'> <'cue-after'>?"
		},
		"cue-after": {
			comment: "https://www.w3.org/TR/css3-speech/#property-index",
			syntax: "<url> <decibel>? | none"
		},
		"cue-before": {
			comment: "https://www.w3.org/TR/css3-speech/#property-index",
			syntax: "<url> <decibel>? | none"
		},
		cursor: {
			comment: "added legacy keywords: hand, -webkit-grab. -webkit-grabbing, -webkit-zoom-in, -webkit-zoom-out, -moz-grab, -moz-grabbing, -moz-zoom-in, -moz-zoom-out",
			references: [
				"https://www.sitepoint.com/css3-cursor-styles/"
			],
			syntax: "[ [ <url> [ <x> <y> ]? , ]* [ auto | default | none | context-menu | help | pointer | progress | wait | cell | crosshair | text | vertical-text | alias | copy | move | no-drop | not-allowed | e-resize | n-resize | ne-resize | nw-resize | s-resize | se-resize | sw-resize | w-resize | ew-resize | ns-resize | nesw-resize | nwse-resize | col-resize | row-resize | all-scroll | zoom-in | zoom-out | grab | grabbing | hand | -webkit-grab | -webkit-grabbing | -webkit-zoom-in | -webkit-zoom-out | -moz-grab | -moz-grabbing | -moz-zoom-in | -moz-zoom-out ] ]"
		},
		display: {
			comment: "extended with -ms-flexbox",
			syntax: "| <-non-standard-display>"
		},
		position: {
			comment: "extended with -webkit-sticky",
			syntax: "| -webkit-sticky"
		},
		"dominant-baseline": {
			comment: "added SVG property",
			references: [
				"https://www.w3.org/TR/SVG/text.html#DominantBaselineProperty"
			],
			syntax: "auto | use-script | no-change | reset-size | ideographic | alphabetic | hanging | mathematical | central | middle | text-after-edge | text-before-edge"
		},
		"image-rendering": {
			comment: "extended with <-non-standard-image-rendering>, added SVG keywords optimizeSpeed and optimizeQuality",
			references: [
				"https://developer.mozilla.org/en/docs/Web/CSS/image-rendering",
				"https://www.w3.org/TR/SVG/painting.html#ImageRenderingProperty"
			],
			syntax: "| optimizeSpeed | optimizeQuality | <-non-standard-image-rendering>"
		},
		fill: {
			comment: "added SVG property",
			references: [
				"https://www.w3.org/TR/SVG/painting.html#FillProperty"
			],
			syntax: "<paint>"
		},
		"fill-opacity": {
			comment: "added SVG property",
			references: [
				"https://www.w3.org/TR/SVG/painting.html#FillProperty"
			],
			syntax: "<number-zero-one>"
		},
		"fill-rule": {
			comment: "added SVG property",
			references: [
				"https://www.w3.org/TR/SVG/painting.html#FillProperty"
			],
			syntax: "nonzero | evenodd"
		},
		filter: {
			comment: "extend with IE legacy syntaxes",
			syntax: "| <-ms-filter-function-list>"
		},
		"glyph-orientation-horizontal": {
			comment: "added SVG property",
			references: [
				"https://www.w3.org/TR/SVG/text.html#GlyphOrientationHorizontalProperty"
			],
			syntax: "<angle>"
		},
		"glyph-orientation-vertical": {
			comment: "added SVG property",
			references: [
				"https://www.w3.org/TR/SVG/text.html#GlyphOrientationVerticalProperty"
			],
			syntax: "<angle>"
		},
		kerning: {
			comment: "added SVG property",
			references: [
				"https://www.w3.org/TR/SVG/text.html#KerningProperty"
			],
			syntax: "auto | <svg-length>"
		},
		"letter-spacing": {
			comment: "fix syntax <length> -> <length-percentage>",
			references: [
				"https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/letter-spacing"
			],
			syntax: "normal | <length-percentage>"
		},
		marker: {
			comment: "added SVG property",
			references: [
				"https://www.w3.org/TR/SVG/painting.html#MarkerProperties"
			],
			syntax: "none | <url>"
		},
		"marker-end": {
			comment: "added SVG property",
			references: [
				"https://www.w3.org/TR/SVG/painting.html#MarkerProperties"
			],
			syntax: "none | <url>"
		},
		"marker-mid": {
			comment: "added SVG property",
			references: [
				"https://www.w3.org/TR/SVG/painting.html#MarkerProperties"
			],
			syntax: "none | <url>"
		},
		"marker-start": {
			comment: "added SVG property",
			references: [
				"https://www.w3.org/TR/SVG/painting.html#MarkerProperties"
			],
			syntax: "none | <url>"
		},
		"max-width": {
			comment: "fix auto -> none (https://github.com/mdn/data/pull/431); extend by non-standard width keywords https://developer.mozilla.org/en-US/docs/Web/CSS/max-width",
			syntax: "none | <length-percentage> | min-content | max-content | fit-content(<length-percentage>) | <-non-standard-width>"
		},
		width: {
			comment: "per spec fit-content should be a function, however browsers are supporting it as a keyword (https://github.com/csstree/stylelint-validator/issues/29)",
			syntax: "| fit-content | -moz-fit-content | -webkit-fit-content"
		},
		"min-width": {
			comment: "extend by non-standard width keywords https://developer.mozilla.org/en-US/docs/Web/CSS/width",
			syntax: "auto | <length-percentage> | min-content | max-content | fit-content(<length-percentage>) | <-non-standard-width>"
		},
		overflow: {
			comment: "extend by vendor keywords https://developer.mozilla.org/en-US/docs/Web/CSS/overflow",
			syntax: "| <-non-standard-overflow>"
		},
		pause: {
			comment: "https://www.w3.org/TR/css3-speech/#property-index",
			syntax: "<'pause-before'> <'pause-after'>?"
		},
		"pause-after": {
			comment: "https://www.w3.org/TR/css3-speech/#property-index",
			syntax: "<time> | none | x-weak | weak | medium | strong | x-strong"
		},
		"pause-before": {
			comment: "https://www.w3.org/TR/css3-speech/#property-index",
			syntax: "<time> | none | x-weak | weak | medium | strong | x-strong"
		},
		rest: {
			comment: "https://www.w3.org/TR/css3-speech/#property-index",
			syntax: "<'rest-before'> <'rest-after'>?"
		},
		"rest-after": {
			comment: "https://www.w3.org/TR/css3-speech/#property-index",
			syntax: "<time> | none | x-weak | weak | medium | strong | x-strong"
		},
		"rest-before": {
			comment: "https://www.w3.org/TR/css3-speech/#property-index",
			syntax: "<time> | none | x-weak | weak | medium | strong | x-strong"
		},
		"shape-rendering": {
			comment: "added SVG property",
			references: [
				"https://www.w3.org/TR/SVG/painting.html#ShapeRenderingPropert"
			],
			syntax: "auto | optimizeSpeed | crispEdges | geometricPrecision"
		},
		src: {
			comment: "added @font-face's src property https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/src",
			syntax: "[ <url> [ format( <string># ) ]? | local( <family-name> ) ]#"
		},
		speak: {
			comment: "https://www.w3.org/TR/css3-speech/#property-index",
			syntax: "auto | none | normal"
		},
		"speak-as": {
			comment: "https://www.w3.org/TR/css3-speech/#property-index",
			syntax: "normal | spell-out || digits || [ literal-punctuation | no-punctuation ]"
		},
		stroke: {
			comment: "added SVG property",
			references: [
				"https://www.w3.org/TR/SVG/painting.html#StrokeProperties"
			],
			syntax: "<paint>"
		},
		"stroke-dasharray": {
			comment: "added SVG property; a list of comma and/or white space separated <length>s and <percentage>s",
			references: [
				"https://www.w3.org/TR/SVG/painting.html#StrokeProperties"
			],
			syntax: "none | [ <svg-length>+ ]#"
		},
		"stroke-dashoffset": {
			comment: "added SVG property",
			references: [
				"https://www.w3.org/TR/SVG/painting.html#StrokeProperties"
			],
			syntax: "<svg-length>"
		},
		"stroke-linecap": {
			comment: "added SVG property",
			references: [
				"https://www.w3.org/TR/SVG/painting.html#StrokeProperties"
			],
			syntax: "butt | round | square"
		},
		"stroke-linejoin": {
			comment: "added SVG property",
			references: [
				"https://www.w3.org/TR/SVG/painting.html#StrokeProperties"
			],
			syntax: "miter | round | bevel"
		},
		"stroke-miterlimit": {
			comment: "added SVG property (<miterlimit> = <number-one-or-greater>) ",
			references: [
				"https://www.w3.org/TR/SVG/painting.html#StrokeProperties"
			],
			syntax: "<number-one-or-greater>"
		},
		"stroke-opacity": {
			comment: "added SVG property",
			references: [
				"https://www.w3.org/TR/SVG/painting.html#StrokeProperties"
			],
			syntax: "<number-zero-one>"
		},
		"stroke-width": {
			comment: "added SVG property",
			references: [
				"https://www.w3.org/TR/SVG/painting.html#StrokeProperties"
			],
			syntax: "<svg-length>"
		},
		"text-anchor": {
			comment: "added SVG property",
			references: [
				"https://www.w3.org/TR/SVG/text.html#TextAlignmentProperties"
			],
			syntax: "start | middle | end"
		},
		"unicode-bidi": {
			comment: "added prefixed keywords https://developer.mozilla.org/en-US/docs/Web/CSS/unicode-bidi",
			syntax: "| -moz-isolate | -moz-isolate-override | -moz-plaintext | -webkit-isolate | -webkit-isolate-override | -webkit-plaintext"
		},
		"unicode-range": {
			comment: "added missed property https://developer.mozilla.org/en-US/docs/Web/CSS/%40font-face/unicode-range",
			syntax: "<urange>#"
		},
		"voice-balance": {
			comment: "https://www.w3.org/TR/css3-speech/#property-index",
			syntax: "<number> | left | center | right | leftwards | rightwards"
		},
		"voice-duration": {
			comment: "https://www.w3.org/TR/css3-speech/#property-index",
			syntax: "auto | <time>"
		},
		"voice-family": {
			comment: "<name> -> <family-name>, https://www.w3.org/TR/css3-speech/#property-index",
			syntax: "[ [ <family-name> | <generic-voice> ] , ]* [ <family-name> | <generic-voice> ] | preserve"
		},
		"voice-pitch": {
			comment: "https://www.w3.org/TR/css3-speech/#property-index",
			syntax: "<frequency> && absolute | [ [ x-low | low | medium | high | x-high ] || [ <frequency> | <semitones> | <percentage> ] ]"
		},
		"voice-range": {
			comment: "https://www.w3.org/TR/css3-speech/#property-index",
			syntax: "<frequency> && absolute | [ [ x-low | low | medium | high | x-high ] || [ <frequency> | <semitones> | <percentage> ] ]"
		},
		"voice-rate": {
			comment: "https://www.w3.org/TR/css3-speech/#property-index",
			syntax: "[ normal | x-slow | slow | medium | fast | x-fast ] || <percentage>"
		},
		"voice-stress": {
			comment: "https://www.w3.org/TR/css3-speech/#property-index",
			syntax: "normal | strong | moderate | none | reduced"
		},
		"voice-volume": {
			comment: "https://www.w3.org/TR/css3-speech/#property-index",
			syntax: "silent | [ [ x-soft | soft | medium | loud | x-loud ] || <decibel> ]"
		},
		"writing-mode": {
			comment: "extend with SVG keywords",
			syntax: "| <svg-writing-mode>"
		}
	};
	var syntaxes = {
		"-legacy-gradient": {
			comment: "added collection of legacy gradient syntaxes",
			syntax: "<-webkit-gradient()> | <-legacy-linear-gradient> | <-legacy-repeating-linear-gradient> | <-legacy-radial-gradient> | <-legacy-repeating-radial-gradient>"
		},
		"-legacy-linear-gradient": {
			comment: "like standard syntax but w/o `to` keyword https://developer.mozilla.org/en-US/docs/Web/CSS/linear-gradient",
			syntax: "-moz-linear-gradient( <-legacy-linear-gradient-arguments> ) | -webkit-linear-gradient( <-legacy-linear-gradient-arguments> ) | -o-linear-gradient( <-legacy-linear-gradient-arguments> )"
		},
		"-legacy-repeating-linear-gradient": {
			comment: "like standard syntax but w/o `to` keyword https://developer.mozilla.org/en-US/docs/Web/CSS/linear-gradient",
			syntax: "-moz-repeating-linear-gradient( <-legacy-linear-gradient-arguments> ) | -webkit-repeating-linear-gradient( <-legacy-linear-gradient-arguments> ) | -o-repeating-linear-gradient( <-legacy-linear-gradient-arguments> )"
		},
		"-legacy-linear-gradient-arguments": {
			comment: "like standard syntax but w/o `to` keyword https://developer.mozilla.org/en-US/docs/Web/CSS/linear-gradient",
			syntax: "[ <angle> | <side-or-corner> ]? , <color-stop-list>"
		},
		"-legacy-radial-gradient": {
			comment: "deprecated syntax that implemented by some browsers https://www.w3.org/TR/2011/WD-css3-images-20110908/#radial-gradients",
			syntax: "-moz-radial-gradient( <-legacy-radial-gradient-arguments> ) | -webkit-radial-gradient( <-legacy-radial-gradient-arguments> ) | -o-radial-gradient( <-legacy-radial-gradient-arguments> )"
		},
		"-legacy-repeating-radial-gradient": {
			comment: "deprecated syntax that implemented by some browsers https://www.w3.org/TR/2011/WD-css3-images-20110908/#radial-gradients",
			syntax: "-moz-repeating-radial-gradient( <-legacy-radial-gradient-arguments> ) | -webkit-repeating-radial-gradient( <-legacy-radial-gradient-arguments> ) | -o-repeating-radial-gradient( <-legacy-radial-gradient-arguments> )"
		},
		"-legacy-radial-gradient-arguments": {
			comment: "deprecated syntax that implemented by some browsers https://www.w3.org/TR/2011/WD-css3-images-20110908/#radial-gradients",
			syntax: "[ <position> , ]? [ [ [ <-legacy-radial-gradient-shape> || <-legacy-radial-gradient-size> ] | [ <length> | <percentage> ]{2} ] , ]? <color-stop-list>"
		},
		"-legacy-radial-gradient-size": {
			comment: "before a standard it contains 2 extra keywords (`contain` and `cover`) https://www.w3.org/TR/2011/WD-css3-images-20110908/#ltsize",
			syntax: "closest-side | closest-corner | farthest-side | farthest-corner | contain | cover"
		},
		"-legacy-radial-gradient-shape": {
			comment: "define to double sure it doesn't extends in future https://www.w3.org/TR/2011/WD-css3-images-20110908/#ltshape",
			syntax: "circle | ellipse"
		},
		"-non-standard-font": {
			comment: "non standard fonts",
			references: [
				"https://webkit.org/blog/3709/using-the-system-font-in-web-content/"
			],
			syntax: "-apple-system-body | -apple-system-headline | -apple-system-subheadline | -apple-system-caption1 | -apple-system-caption2 | -apple-system-footnote | -apple-system-short-body | -apple-system-short-headline | -apple-system-short-subheadline | -apple-system-short-caption1 | -apple-system-short-footnote | -apple-system-tall-body"
		},
		"-non-standard-color": {
			comment: "non standard colors",
			references: [
				"http://cssdot.ru/%D0%A1%D0%BF%D1%80%D0%B0%D0%B2%D0%BE%D1%87%D0%BD%D0%B8%D0%BA_CSS/color-i305.html",
				"https://developer.mozilla.org/en-US/docs/Web/CSS/color_value#Mozilla_Color_Preference_Extensions"
			],
			syntax: "-moz-ButtonDefault | -moz-ButtonHoverFace | -moz-ButtonHoverText | -moz-CellHighlight | -moz-CellHighlightText | -moz-Combobox | -moz-ComboboxText | -moz-Dialog | -moz-DialogText | -moz-dragtargetzone | -moz-EvenTreeRow | -moz-Field | -moz-FieldText | -moz-html-CellHighlight | -moz-html-CellHighlightText | -moz-mac-accentdarkestshadow | -moz-mac-accentdarkshadow | -moz-mac-accentface | -moz-mac-accentlightesthighlight | -moz-mac-accentlightshadow | -moz-mac-accentregularhighlight | -moz-mac-accentregularshadow | -moz-mac-chrome-active | -moz-mac-chrome-inactive | -moz-mac-focusring | -moz-mac-menuselect | -moz-mac-menushadow | -moz-mac-menutextselect | -moz-MenuHover | -moz-MenuHoverText | -moz-MenuBarText | -moz-MenuBarHoverText | -moz-nativehyperlinktext | -moz-OddTreeRow | -moz-win-communicationstext | -moz-win-mediatext | -moz-activehyperlinktext | -moz-default-background-color | -moz-default-color | -moz-hyperlinktext | -moz-visitedhyperlinktext | -webkit-activelink | -webkit-focus-ring-color | -webkit-link | -webkit-text"
		},
		"-non-standard-image-rendering": {
			comment: "non-standard keywords http://phrogz.net/tmp/canvas_image_zoom.html",
			syntax: "optimize-contrast | -moz-crisp-edges | -o-crisp-edges | -webkit-optimize-contrast"
		},
		"-non-standard-overflow": {
			comment: "non-standard keywords https://developer.mozilla.org/en-US/docs/Web/CSS/overflow",
			syntax: "-moz-scrollbars-none | -moz-scrollbars-horizontal | -moz-scrollbars-vertical | -moz-hidden-unscrollable"
		},
		"-non-standard-width": {
			comment: "non-standard keywords https://developer.mozilla.org/en-US/docs/Web/CSS/width",
			syntax: "fill-available | min-intrinsic | intrinsic | -moz-available | -moz-fit-content | -moz-min-content | -moz-max-content | -webkit-min-content | -webkit-max-content"
		},
		"-webkit-gradient()": {
			comment: "first Apple proposal gradient syntax https://webkit.org/blog/175/introducing-css-gradients/ - TODO: simplify when after match algorithm improvement ( [, point, radius | , point] -> [, radius]? , point )",
			syntax: "-webkit-gradient( <-webkit-gradient-type>, <-webkit-gradient-point> [, <-webkit-gradient-point> | , <-webkit-gradient-radius>, <-webkit-gradient-point> ] [, <-webkit-gradient-radius>]? [, <-webkit-gradient-color-stop>]* )"
		},
		"-webkit-gradient-color-stop": {
			comment: "first Apple proposal gradient syntax https://webkit.org/blog/175/introducing-css-gradients/",
			syntax: "from( <color> ) | color-stop( [ <number-zero-one> | <percentage> ] , <color> ) | to( <color> )"
		},
		"-webkit-gradient-point": {
			comment: "first Apple proposal gradient syntax https://webkit.org/blog/175/introducing-css-gradients/",
			syntax: "[ left | center | right | <length-percentage> ] [ top | center | bottom | <length-percentage> ]"
		},
		"-webkit-gradient-radius": {
			comment: "first Apple proposal gradient syntax https://webkit.org/blog/175/introducing-css-gradients/",
			syntax: "<length> | <percentage>"
		},
		"-webkit-gradient-type": {
			comment: "first Apple proposal gradient syntax https://webkit.org/blog/175/introducing-css-gradients/",
			syntax: "linear | radial"
		},
		"-webkit-mask-box-repeat": {
			comment: "missed; https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-mask-box-image",
			syntax: "repeat | stretch | round"
		},
		"-webkit-mask-clip-style": {
			comment: "missed; there is no enough information about `-webkit-mask-clip` property, but looks like all those keywords are working",
			syntax: "border | border-box | padding | padding-box | content | content-box | text"
		},
		"-ms-filter-function-list": {
			comment: "https://developer.mozilla.org/en-US/docs/Web/CSS/-ms-filter",
			syntax: "<-ms-filter-function>+"
		},
		"-ms-filter-function": {
			comment: "https://developer.mozilla.org/en-US/docs/Web/CSS/-ms-filter",
			syntax: "<-ms-filter-function-progid> | <-ms-filter-function-legacy>"
		},
		"-ms-filter-function-progid": {
			comment: "https://developer.mozilla.org/en-US/docs/Web/CSS/-ms-filter",
			syntax: "'progid:' [ <ident-token> '.' ]* [ <ident-token> | <function-token> <any-value>? ) ]"
		},
		"-ms-filter-function-legacy": {
			comment: "https://developer.mozilla.org/en-US/docs/Web/CSS/-ms-filter",
			syntax: "<ident-token> | <function-token> <any-value>? )"
		},
		"-ms-filter": {
			syntax: "<string>"
		},
		age: {
			comment: "https://www.w3.org/TR/css3-speech/#voice-family",
			syntax: "child | young | old"
		},
		"attr-name": {
			syntax: "<wq-name>"
		},
		"attr-fallback": {
			syntax: "<any-value>"
		},
		"border-radius": {
			comment: "missed, https://drafts.csswg.org/css-backgrounds-3/#the-border-radius",
			syntax: "<length-percentage>{1,2}"
		},
		bottom: {
			comment: "missed; not sure we should add it, but no others except `shape` is using it so it's ok for now; https://drafts.fxtf.org/css-masking-1/#funcdef-clip-rect",
			syntax: "<length> | auto"
		},
		"content-list": {
			comment: "missed -> https://drafts.csswg.org/css-content/#typedef-content-list (document-url, <target> and leader() is omitted util stabilization)",
			syntax: "[ <string> | contents | <image> | <quote> | <target> | <leader()> | <attr()> | counter( <ident>, <'list-style-type'>? ) ]+"
		},
		"element()": {
			comment: "https://drafts.csswg.org/css-gcpm/#element-syntax & https://drafts.csswg.org/css-images-4/#element-notation",
			syntax: "element( <custom-ident> , [ first | start | last | first-except ]? ) | element( <id-selector> )"
		},
		"generic-voice": {
			comment: "https://www.w3.org/TR/css3-speech/#voice-family",
			syntax: "[ <age>? <gender> <integer>? ]"
		},
		gender: {
			comment: "https://www.w3.org/TR/css3-speech/#voice-family",
			syntax: "male | female | neutral"
		},
		"generic-family": {
			comment: "added -apple-system",
			references: [
				"https://webkit.org/blog/3709/using-the-system-font-in-web-content/"
			],
			syntax: "| -apple-system"
		},
		gradient: {
			comment: "added legacy syntaxes support",
			syntax: "| <-legacy-gradient>"
		},
		left: {
			comment: "missed; not sure we should add it, but no others except `shape` is using it so it's ok for now; https://drafts.fxtf.org/css-masking-1/#funcdef-clip-rect",
			syntax: "<length> | auto"
		},
		"mask-image": {
			comment: "missed; https://drafts.fxtf.org/css-masking-1/#the-mask-image",
			syntax: "<mask-reference>#"
		},
		"name-repeat": {
			comment: "missed, and looks like obsolete, keep it as is since other property syntaxes should be changed too; https://www.w3.org/TR/2015/WD-css-grid-1-20150917/#typedef-name-repeat",
			syntax: "repeat( [ <positive-integer> | auto-fill ], <line-names>+)"
		},
		"named-color": {
			comment: "added non standard color names",
			syntax: "| <-non-standard-color>"
		},
		paint: {
			comment: "used by SVG https://www.w3.org/TR/SVG/painting.html#SpecifyingPaint",
			syntax: "none | <color> | <url> [ none | <color> ]? | context-fill | context-stroke"
		},
		"page-size": {
			comment: "https://www.w3.org/TR/css-page-3/#typedef-page-size-page-size",
			syntax: "A5 | A4 | A3 | B5 | B4 | JIS-B5 | JIS-B4 | letter | legal | ledger"
		},
		ratio: {
			comment: "missed, https://drafts.csswg.org/mediaqueries-4/#typedef-ratio",
			syntax: "<integer> / <integer>"
		},
		right: {
			comment: "missed; not sure we should add it, but no others except `shape` is using it so it's ok for now; https://drafts.fxtf.org/css-masking-1/#funcdef-clip-rect",
			syntax: "<length> | auto"
		},
		shape: {
			comment: "missed spaces in function body and add backwards compatible syntax",
			syntax: "rect( <top>, <right>, <bottom>, <left> ) | rect( <top> <right> <bottom> <left> )"
		},
		"svg-length": {
			comment: "All coordinates and lengths in SVG can be specified with or without a unit identifier",
			references: [
				"https://www.w3.org/TR/SVG11/coords.html#Units"
			],
			syntax: "<percentage> | <length> | <number>"
		},
		"svg-writing-mode": {
			comment: "SVG specific keywords (deprecated for CSS)",
			references: [
				"https://developer.mozilla.org/en/docs/Web/CSS/writing-mode",
				"https://www.w3.org/TR/SVG/text.html#WritingModeProperty"
			],
			syntax: "lr-tb | rl-tb | tb-rl | lr | rl | tb"
		},
		top: {
			comment: "missed; not sure we should add it, but no others except `shape` is using it so it's ok for now; https://drafts.fxtf.org/css-masking-1/#funcdef-clip-rect",
			syntax: "<length> | auto"
		},
		"track-group": {
			comment: "used by old grid-columns and grid-rows syntax v0",
			syntax: "'(' [ <string>* <track-minmax> <string>* ]+ ')' [ '[' <positive-integer> ']' ]? | <track-minmax>"
		},
		"track-list-v0": {
			comment: "used by old grid-columns and grid-rows syntax v0",
			syntax: "[ <string>* <track-group> <string>* ]+ | none"
		},
		"track-minmax": {
			comment: "used by old grid-columns and grid-rows syntax v0",
			syntax: "minmax( <track-breadth> , <track-breadth> ) | auto | <track-breadth> | fit-content"
		},
		x: {
			comment: "missed; not sure we should add it, but no others except `cursor` is using it so it's ok for now; https://drafts.csswg.org/css-ui-3/#cursor",
			syntax: "<number>"
		},
		y: {
			comment: "missed; not sure we should add it, but no others except `cursor` is using so it's ok for now; https://drafts.csswg.org/css-ui-3/#cursor",
			syntax: "<number>"
		},
		declaration: {
			comment: "missed, restored by https://drafts.csswg.org/css-syntax",
			syntax: "<ident-token> : <declaration-value>? [ '!' important ]?"
		},
		"declaration-list": {
			comment: "missed, restored by https://drafts.csswg.org/css-syntax",
			syntax: "[ <declaration>? ';' ]* <declaration>?"
		},
		url: {
			comment: "https://drafts.csswg.org/css-values-4/#urls",
			syntax: "url( <string> <url-modifier>* ) | <url-token>"
		},
		"url-modifier": {
			comment: "https://drafts.csswg.org/css-values-4/#typedef-url-modifier",
			syntax: "<ident> | <function-token> <any-value> )"
		},
		"number-zero-one": {
			syntax: "<number [0,1]>"
		},
		"number-one-or-greater": {
			syntax: "<number [1,∞]>"
		},
		"positive-integer": {
			syntax: "<integer [0,∞]>"
		},
		"-non-standard-display": {
			syntax: "-ms-inline-flexbox | -ms-grid | -ms-inline-grid | -webkit-flex | -webkit-inline-flex | -webkit-box | -webkit-inline-box | -moz-inline-stack | -moz-box | -moz-inline-box"
		}
	};
	var require$$3 = {
		atrules: atrules,
		properties: properties,
		syntaxes: syntaxes
	};

	const mdnAtrules = require$$0;
	const mdnProperties = require$$1;
	const mdnSyntaxes = require$$2;
	const patch = require$$3;
	const extendSyntax = /^\s*\|\s*/;

	function preprocessAtrules(dict) {
	    const result = Object.create(null);

	    for (const atruleName in dict) {
	        const atrule = dict[atruleName];
	        let descriptors = null;

	        if (atrule.descriptors) {
	            descriptors = Object.create(null);

	            for (const descriptor in atrule.descriptors) {
	                descriptors[descriptor] = atrule.descriptors[descriptor].syntax;
	            }
	        }

	        result[atruleName.substr(1)] = {
	            prelude: atrule.syntax.trim().match(/^@\S+\s+([^;\{]*)/)[1].trim() || null,
	            descriptors
	        };
	    }

	    return result;
	}

	function patchDictionary(dict, patchDict) {
	    const result = {};

	    // copy all syntaxes for an original dict
	    for (const key in dict) {
	        result[key] = dict[key].syntax || dict[key];
	    }

	    // apply a patch
	    for (const key in patchDict) {
	        if (key in dict) {
	            if (patchDict[key].syntax) {
	                result[key] = extendSyntax.test(patchDict[key].syntax)
	                    ? result[key] + ' ' + patchDict[key].syntax.trim()
	                    : patchDict[key].syntax;
	            } else {
	                delete result[key];
	            }
	        } else {
	            if (patchDict[key].syntax) {
	                result[key] = patchDict[key].syntax.replace(extendSyntax, '');
	            }
	        }
	    }

	    return result;
	}

	function unpackSyntaxes(dict) {
	    const result = {};

	    for (const key in dict) {
	        result[key] = dict[key].syntax;
	    }

	    return result;
	}

	function patchAtrules(dict, patchDict) {
	    const result = {};

	    // copy all syntaxes for an original dict
	    for (const key in dict) {
	        const patchDescriptors = (patchDict[key] && patchDict[key].descriptors) || null;

	        result[key] = {
	            prelude: key in patchDict && 'prelude' in patchDict[key]
	                ? patchDict[key].prelude
	                : dict[key].prelude || null,
	            descriptors: dict[key].descriptors
	                ? patchDictionary(dict[key].descriptors, patchDescriptors || {})
	                : patchDescriptors && unpackSyntaxes(patchDescriptors)
	        };
	    }

	    // apply a patch
	    for (const key in patchDict) {
	        if (!hasOwnProperty.call(dict, key)) {
	            result[key] = {
	                prelude: patchDict[key].prelude || null,
	                descriptors: patchDict[key].descriptors && unpackSyntaxes(patchDict[key].descriptors)
	            };
	        }
	    }

	    return result;
	}

	var data$1 = {
	    types: patchDictionary(mdnSyntaxes, patch.syntaxes),
	    atrules: patchAtrules(preprocessAtrules(mdnAtrules), patch.atrules),
	    properties: patchDictionary(mdnProperties, patch.properties)
	};

	var cmpChar$2 = tokenizer$3.cmpChar;
	var isDigit$1 = tokenizer$3.isDigit;
	var TYPE$y = tokenizer$3.TYPE;

	var WHITESPACE$8 = TYPE$y.WhiteSpace;
	var COMMENT$6 = TYPE$y.Comment;
	var IDENT$f = TYPE$y.Ident;
	var NUMBER$6 = TYPE$y.Number;
	var DIMENSION$5 = TYPE$y.Dimension;
	var PLUSSIGN$5 = 0x002B;    // U+002B PLUS SIGN (+)
	var HYPHENMINUS$2 = 0x002D; // U+002D HYPHEN-MINUS (-)
	var N = 0x006E;           // U+006E LATIN SMALL LETTER N (n)
	var DISALLOW_SIGN = true;
	var ALLOW_SIGN = false;

	function checkInteger(offset, disallowSign) {
	    var pos = this.scanner.tokenStart + offset;
	    var code = this.scanner.source.charCodeAt(pos);

	    if (code === PLUSSIGN$5 || code === HYPHENMINUS$2) {
	        if (disallowSign) {
	            this.error('Number sign is not allowed');
	        }
	        pos++;
	    }

	    for (; pos < this.scanner.tokenEnd; pos++) {
	        if (!isDigit$1(this.scanner.source.charCodeAt(pos))) {
	            this.error('Integer is expected', pos);
	        }
	    }
	}

	function checkTokenIsInteger(disallowSign) {
	    return checkInteger.call(this, 0, disallowSign);
	}

	function expectCharCode(offset, code) {
	    if (!cmpChar$2(this.scanner.source, this.scanner.tokenStart + offset, code)) {
	        var msg = '';

	        switch (code) {
	            case N:
	                msg = 'N is expected';
	                break;
	            case HYPHENMINUS$2:
	                msg = 'HyphenMinus is expected';
	                break;
	        }

	        this.error(msg, this.scanner.tokenStart + offset);
	    }
	}

	// ... <signed-integer>
	// ... ['+' | '-'] <signless-integer>
	function consumeB() {
	    var offset = 0;
	    var sign = 0;
	    var type = this.scanner.tokenType;

	    while (type === WHITESPACE$8 || type === COMMENT$6) {
	        type = this.scanner.lookupType(++offset);
	    }

	    if (type !== NUMBER$6) {
	        if (this.scanner.isDelim(PLUSSIGN$5, offset) ||
	            this.scanner.isDelim(HYPHENMINUS$2, offset)) {
	            sign = this.scanner.isDelim(PLUSSIGN$5, offset) ? PLUSSIGN$5 : HYPHENMINUS$2;

	            do {
	                type = this.scanner.lookupType(++offset);
	            } while (type === WHITESPACE$8 || type === COMMENT$6);

	            if (type !== NUMBER$6) {
	                this.scanner.skip(offset);
	                checkTokenIsInteger.call(this, DISALLOW_SIGN);
	            }
	        } else {
	            return null;
	        }
	    }

	    if (offset > 0) {
	        this.scanner.skip(offset);
	    }

	    if (sign === 0) {
	        type = this.scanner.source.charCodeAt(this.scanner.tokenStart);
	        if (type !== PLUSSIGN$5 && type !== HYPHENMINUS$2) {
	            this.error('Number sign is expected');
	        }
	    }

	    checkTokenIsInteger.call(this, sign !== 0);
	    return sign === HYPHENMINUS$2 ? '-' + this.consume(NUMBER$6) : this.consume(NUMBER$6);
	}

	// An+B microsyntax https://www.w3.org/TR/css-syntax-3/#anb
	var AnPlusB = {
	    name: 'AnPlusB',
	    structure: {
	        a: [String, null],
	        b: [String, null]
	    },
	    parse: function() {
	        /* eslint-disable brace-style*/
	        var start = this.scanner.tokenStart;
	        var a = null;
	        var b = null;

	        // <integer>
	        if (this.scanner.tokenType === NUMBER$6) {
	            checkTokenIsInteger.call(this, ALLOW_SIGN);
	            b = this.consume(NUMBER$6);
	        }

	        // -n
	        // -n <signed-integer>
	        // -n ['+' | '-'] <signless-integer>
	        // -n- <signless-integer>
	        // <dashndashdigit-ident>
	        else if (this.scanner.tokenType === IDENT$f && cmpChar$2(this.scanner.source, this.scanner.tokenStart, HYPHENMINUS$2)) {
	            a = '-1';

	            expectCharCode.call(this, 1, N);

	            switch (this.scanner.getTokenLength()) {
	                // -n
	                // -n <signed-integer>
	                // -n ['+' | '-'] <signless-integer>
	                case 2:
	                    this.scanner.next();
	                    b = consumeB.call(this);
	                    break;

	                // -n- <signless-integer>
	                case 3:
	                    expectCharCode.call(this, 2, HYPHENMINUS$2);

	                    this.scanner.next();
	                    this.scanner.skipSC();

	                    checkTokenIsInteger.call(this, DISALLOW_SIGN);

	                    b = '-' + this.consume(NUMBER$6);
	                    break;

	                // <dashndashdigit-ident>
	                default:
	                    expectCharCode.call(this, 2, HYPHENMINUS$2);
	                    checkInteger.call(this, 3, DISALLOW_SIGN);
	                    this.scanner.next();

	                    b = this.scanner.substrToCursor(start + 2);
	            }
	        }

	        // '+'? n
	        // '+'? n <signed-integer>
	        // '+'? n ['+' | '-'] <signless-integer>
	        // '+'? n- <signless-integer>
	        // '+'? <ndashdigit-ident>
	        else if (this.scanner.tokenType === IDENT$f || (this.scanner.isDelim(PLUSSIGN$5) && this.scanner.lookupType(1) === IDENT$f)) {
	            var sign = 0;
	            a = '1';

	            // just ignore a plus
	            if (this.scanner.isDelim(PLUSSIGN$5)) {
	                sign = 1;
	                this.scanner.next();
	            }

	            expectCharCode.call(this, 0, N);

	            switch (this.scanner.getTokenLength()) {
	                // '+'? n
	                // '+'? n <signed-integer>
	                // '+'? n ['+' | '-'] <signless-integer>
	                case 1:
	                    this.scanner.next();
	                    b = consumeB.call(this);
	                    break;

	                // '+'? n- <signless-integer>
	                case 2:
	                    expectCharCode.call(this, 1, HYPHENMINUS$2);

	                    this.scanner.next();
	                    this.scanner.skipSC();

	                    checkTokenIsInteger.call(this, DISALLOW_SIGN);

	                    b = '-' + this.consume(NUMBER$6);
	                    break;

	                // '+'? <ndashdigit-ident>
	                default:
	                    expectCharCode.call(this, 1, HYPHENMINUS$2);
	                    checkInteger.call(this, 2, DISALLOW_SIGN);
	                    this.scanner.next();

	                    b = this.scanner.substrToCursor(start + sign + 1);
	            }
	        }

	        // <ndashdigit-dimension>
	        // <ndash-dimension> <signless-integer>
	        // <n-dimension>
	        // <n-dimension> <signed-integer>
	        // <n-dimension> ['+' | '-'] <signless-integer>
	        else if (this.scanner.tokenType === DIMENSION$5) {
	            var code = this.scanner.source.charCodeAt(this.scanner.tokenStart);
	            var sign = code === PLUSSIGN$5 || code === HYPHENMINUS$2;

	            for (var i = this.scanner.tokenStart + sign; i < this.scanner.tokenEnd; i++) {
	                if (!isDigit$1(this.scanner.source.charCodeAt(i))) {
	                    break;
	                }
	            }

	            if (i === this.scanner.tokenStart + sign) {
	                this.error('Integer is expected', this.scanner.tokenStart + sign);
	            }

	            expectCharCode.call(this, i - this.scanner.tokenStart, N);
	            a = this.scanner.source.substring(start, i);

	            // <n-dimension>
	            // <n-dimension> <signed-integer>
	            // <n-dimension> ['+' | '-'] <signless-integer>
	            if (i + 1 === this.scanner.tokenEnd) {
	                this.scanner.next();
	                b = consumeB.call(this);
	            } else {
	                expectCharCode.call(this, i - this.scanner.tokenStart + 1, HYPHENMINUS$2);

	                // <ndash-dimension> <signless-integer>
	                if (i + 2 === this.scanner.tokenEnd) {
	                    this.scanner.next();
	                    this.scanner.skipSC();
	                    checkTokenIsInteger.call(this, DISALLOW_SIGN);
	                    b = '-' + this.consume(NUMBER$6);
	                }
	                // <ndashdigit-dimension>
	                else {
	                    checkInteger.call(this, i - this.scanner.tokenStart + 2, DISALLOW_SIGN);
	                    this.scanner.next();
	                    b = this.scanner.substrToCursor(i + 1);
	                }
	            }
	        } else {
	            this.error();
	        }

	        if (a !== null && a.charCodeAt(0) === PLUSSIGN$5) {
	            a = a.substr(1);
	        }

	        if (b !== null && b.charCodeAt(0) === PLUSSIGN$5) {
	            b = b.substr(1);
	        }

	        return {
	            type: 'AnPlusB',
	            loc: this.getLocation(start, this.scanner.tokenStart),
	            a: a,
	            b: b
	        };
	    },
	    generate: function(node) {
	        var a = node.a !== null && node.a !== undefined;
	        var b = node.b !== null && node.b !== undefined;

	        if (a) {
	            this.chunk(
	                node.a === '+1' ? '+n' : // eslint-disable-line operator-linebreak, indent
	                node.a ===  '1' ?  'n' : // eslint-disable-line operator-linebreak, indent
	                node.a === '-1' ? '-n' : // eslint-disable-line operator-linebreak, indent
	                node.a + 'n'             // eslint-disable-line operator-linebreak, indent
	            );

	            if (b) {
	                b = String(node.b);
	                if (b.charAt(0) === '-' || b.charAt(0) === '+') {
	                    this.chunk(b.charAt(0));
	                    this.chunk(b.substr(1));
	                } else {
	                    this.chunk('+');
	                    this.chunk(b);
	                }
	            }
	        } else {
	            this.chunk(String(node.b));
	        }
	    }
	};

	var tokenizer = tokenizer$3;
	var TYPE$x = tokenizer.TYPE;

	var WhiteSpace$1 = TYPE$x.WhiteSpace;
	var Semicolon = TYPE$x.Semicolon;
	var LeftCurlyBracket = TYPE$x.LeftCurlyBracket;
	var Delim = TYPE$x.Delim;
	var EXCLAMATIONMARK$2 = 0x0021; // U+0021 EXCLAMATION MARK (!)

	function getOffsetExcludeWS() {
	    if (this.scanner.tokenIndex > 0) {
	        if (this.scanner.lookupType(-1) === WhiteSpace$1) {
	            return this.scanner.tokenIndex > 1
	                ? this.scanner.getTokenStart(this.scanner.tokenIndex - 1)
	                : this.scanner.firstCharOffset;
	        }
	    }

	    return this.scanner.tokenStart;
	}

	// 0, 0, false
	function balanceEnd() {
	    return 0;
	}

	// LEFTCURLYBRACKET, 0, false
	function leftCurlyBracket(tokenType) {
	    return tokenType === LeftCurlyBracket ? 1 : 0;
	}

	// LEFTCURLYBRACKET, SEMICOLON, false
	function leftCurlyBracketOrSemicolon(tokenType) {
	    return tokenType === LeftCurlyBracket || tokenType === Semicolon ? 1 : 0;
	}

	// EXCLAMATIONMARK, SEMICOLON, false
	function exclamationMarkOrSemicolon(tokenType, source, offset) {
	    if (tokenType === Delim && source.charCodeAt(offset) === EXCLAMATIONMARK$2) {
	        return 1;
	    }

	    return tokenType === Semicolon ? 1 : 0;
	}

	// 0, SEMICOLON, true
	function semicolonIncluded(tokenType) {
	    return tokenType === Semicolon ? 2 : 0;
	}

	var Raw = {
	    name: 'Raw',
	    structure: {
	        value: String
	    },
	    parse: function(startToken, mode, excludeWhiteSpace) {
	        var startOffset = this.scanner.getTokenStart(startToken);
	        var endOffset;

	        this.scanner.skip(
	            this.scanner.getRawLength(startToken, mode || balanceEnd)
	        );

	        if (excludeWhiteSpace && this.scanner.tokenStart > startOffset) {
	            endOffset = getOffsetExcludeWS.call(this);
	        } else {
	            endOffset = this.scanner.tokenStart;
	        }

	        return {
	            type: 'Raw',
	            loc: this.getLocation(startOffset, endOffset),
	            value: this.scanner.source.substring(startOffset, endOffset)
	        };
	    },
	    generate: function(node) {
	        this.chunk(node.value);
	    },

	    mode: {
	        default: balanceEnd,
	        leftCurlyBracket: leftCurlyBracket,
	        leftCurlyBracketOrSemicolon: leftCurlyBracketOrSemicolon,
	        exclamationMarkOrSemicolon: exclamationMarkOrSemicolon,
	        semicolonIncluded: semicolonIncluded
	    }
	};

	var TYPE$w = tokenizer$3.TYPE;
	var rawMode$5 = Raw.mode;

	var ATKEYWORD$2 = TYPE$w.AtKeyword;
	var SEMICOLON$4 = TYPE$w.Semicolon;
	var LEFTCURLYBRACKET$3 = TYPE$w.LeftCurlyBracket;
	var RIGHTCURLYBRACKET$1 = TYPE$w.RightCurlyBracket;

	function consumeRaw$5(startToken) {
	    return this.Raw(startToken, rawMode$5.leftCurlyBracketOrSemicolon, true);
	}

	function isDeclarationBlockAtrule() {
	    for (var offset = 1, type; type = this.scanner.lookupType(offset); offset++) {
	        if (type === RIGHTCURLYBRACKET$1) {
	            return true;
	        }

	        if (type === LEFTCURLYBRACKET$3 ||
	            type === ATKEYWORD$2) {
	            return false;
	        }
	    }

	    return false;
	}

	var Atrule = {
	    name: 'Atrule',
	    structure: {
	        name: String,
	        prelude: ['AtrulePrelude', 'Raw', null],
	        block: ['Block', null]
	    },
	    parse: function() {
	        var start = this.scanner.tokenStart;
	        var name;
	        var nameLowerCase;
	        var prelude = null;
	        var block = null;

	        this.eat(ATKEYWORD$2);

	        name = this.scanner.substrToCursor(start + 1);
	        nameLowerCase = name.toLowerCase();
	        this.scanner.skipSC();

	        // parse prelude
	        if (this.scanner.eof === false &&
	            this.scanner.tokenType !== LEFTCURLYBRACKET$3 &&
	            this.scanner.tokenType !== SEMICOLON$4) {
	            if (this.parseAtrulePrelude) {
	                prelude = this.parseWithFallback(this.AtrulePrelude.bind(this, name), consumeRaw$5);

	                // turn empty AtrulePrelude into null
	                if (prelude.type === 'AtrulePrelude' && prelude.children.head === null) {
	                    prelude = null;
	                }
	            } else {
	                prelude = consumeRaw$5.call(this, this.scanner.tokenIndex);
	            }

	            this.scanner.skipSC();
	        }

	        switch (this.scanner.tokenType) {
	            case SEMICOLON$4:
	                this.scanner.next();
	                break;

	            case LEFTCURLYBRACKET$3:
	                if (this.atrule.hasOwnProperty(nameLowerCase) &&
	                    typeof this.atrule[nameLowerCase].block === 'function') {
	                    block = this.atrule[nameLowerCase].block.call(this);
	                } else {
	                    // TODO: should consume block content as Raw?
	                    block = this.Block(isDeclarationBlockAtrule.call(this));
	                }

	                break;
	        }

	        return {
	            type: 'Atrule',
	            loc: this.getLocation(start, this.scanner.tokenStart),
	            name: name,
	            prelude: prelude,
	            block: block
	        };
	    },
	    generate: function(node) {
	        this.chunk('@');
	        this.chunk(node.name);

	        if (node.prelude !== null) {
	            this.chunk(' ');
	            this.node(node.prelude);
	        }

	        if (node.block) {
	            this.node(node.block);
	        } else {
	            this.chunk(';');
	        }
	    },
	    walkContext: 'atrule'
	};

	var TYPE$v = tokenizer$3.TYPE;

	var SEMICOLON$3 = TYPE$v.Semicolon;
	var LEFTCURLYBRACKET$2 = TYPE$v.LeftCurlyBracket;

	var AtrulePrelude = {
	    name: 'AtrulePrelude',
	    structure: {
	        children: [[]]
	    },
	    parse: function(name) {
	        var children = null;

	        if (name !== null) {
	            name = name.toLowerCase();
	        }

	        this.scanner.skipSC();

	        if (this.atrule.hasOwnProperty(name) &&
	            typeof this.atrule[name].prelude === 'function') {
	            // custom consumer
	            children = this.atrule[name].prelude.call(this);
	        } else {
	            // default consumer
	            children = this.readSequence(this.scope.AtrulePrelude);
	        }

	        this.scanner.skipSC();

	        if (this.scanner.eof !== true &&
	            this.scanner.tokenType !== LEFTCURLYBRACKET$2 &&
	            this.scanner.tokenType !== SEMICOLON$3) {
	            this.error('Semicolon or block is expected');
	        }

	        if (children === null) {
	            children = this.createList();
	        }

	        return {
	            type: 'AtrulePrelude',
	            loc: this.getLocationFromList(children),
	            children: children
	        };
	    },
	    generate: function(node) {
	        this.children(node);
	    },
	    walkContext: 'atrulePrelude'
	};

	var TYPE$u = tokenizer$3.TYPE;

	var IDENT$e = TYPE$u.Ident;
	var STRING$3 = TYPE$u.String;
	var COLON$6 = TYPE$u.Colon;
	var LEFTSQUAREBRACKET$3 = TYPE$u.LeftSquareBracket;
	var RIGHTSQUAREBRACKET$1 = TYPE$u.RightSquareBracket;
	var DOLLARSIGN$1 = 0x0024;       // U+0024 DOLLAR SIGN ($)
	var ASTERISK$5 = 0x002A;         // U+002A ASTERISK (*)
	var EQUALSSIGN = 0x003D;       // U+003D EQUALS SIGN (=)
	var CIRCUMFLEXACCENT = 0x005E; // U+005E (^)
	var VERTICALLINE$2 = 0x007C;     // U+007C VERTICAL LINE (|)
	var TILDE$2 = 0x007E;            // U+007E TILDE (~)

	function getAttributeName() {
	    if (this.scanner.eof) {
	        this.error('Unexpected end of input');
	    }

	    var start = this.scanner.tokenStart;
	    var expectIdent = false;
	    var checkColon = true;

	    if (this.scanner.isDelim(ASTERISK$5)) {
	        expectIdent = true;
	        checkColon = false;
	        this.scanner.next();
	    } else if (!this.scanner.isDelim(VERTICALLINE$2)) {
	        this.eat(IDENT$e);
	    }

	    if (this.scanner.isDelim(VERTICALLINE$2)) {
	        if (this.scanner.source.charCodeAt(this.scanner.tokenStart + 1) !== EQUALSSIGN) {
	            this.scanner.next();
	            this.eat(IDENT$e);
	        } else if (expectIdent) {
	            this.error('Identifier is expected', this.scanner.tokenEnd);
	        }
	    } else if (expectIdent) {
	        this.error('Vertical line is expected');
	    }

	    if (checkColon && this.scanner.tokenType === COLON$6) {
	        this.scanner.next();
	        this.eat(IDENT$e);
	    }

	    return {
	        type: 'Identifier',
	        loc: this.getLocation(start, this.scanner.tokenStart),
	        name: this.scanner.substrToCursor(start)
	    };
	}

	function getOperator() {
	    var start = this.scanner.tokenStart;
	    var code = this.scanner.source.charCodeAt(start);

	    if (code !== EQUALSSIGN &&        // =
	        code !== TILDE$2 &&             // ~=
	        code !== CIRCUMFLEXACCENT &&  // ^=
	        code !== DOLLARSIGN$1 &&        // $=
	        code !== ASTERISK$5 &&          // *=
	        code !== VERTICALLINE$2         // |=
	    ) {
	        this.error('Attribute selector (=, ~=, ^=, $=, *=, |=) is expected');
	    }

	    this.scanner.next();

	    if (code !== EQUALSSIGN) {
	        if (!this.scanner.isDelim(EQUALSSIGN)) {
	            this.error('Equal sign is expected');
	        }

	        this.scanner.next();
	    }

	    return this.scanner.substrToCursor(start);
	}

	// '[' <wq-name> ']'
	// '[' <wq-name> <attr-matcher> [ <string-token> | <ident-token> ] <attr-modifier>? ']'
	var AttributeSelector = {
	    name: 'AttributeSelector',
	    structure: {
	        name: 'Identifier',
	        matcher: [String, null],
	        value: ['String', 'Identifier', null],
	        flags: [String, null]
	    },
	    parse: function() {
	        var start = this.scanner.tokenStart;
	        var name;
	        var matcher = null;
	        var value = null;
	        var flags = null;

	        this.eat(LEFTSQUAREBRACKET$3);
	        this.scanner.skipSC();

	        name = getAttributeName.call(this);
	        this.scanner.skipSC();

	        if (this.scanner.tokenType !== RIGHTSQUAREBRACKET$1) {
	            // avoid case `[name i]`
	            if (this.scanner.tokenType !== IDENT$e) {
	                matcher = getOperator.call(this);

	                this.scanner.skipSC();

	                value = this.scanner.tokenType === STRING$3
	                    ? this.String()
	                    : this.Identifier();

	                this.scanner.skipSC();
	            }

	            // attribute flags
	            if (this.scanner.tokenType === IDENT$e) {
	                flags = this.scanner.getTokenValue();
	                this.scanner.next();

	                this.scanner.skipSC();
	            }
	        }

	        this.eat(RIGHTSQUAREBRACKET$1);

	        return {
	            type: 'AttributeSelector',
	            loc: this.getLocation(start, this.scanner.tokenStart),
	            name: name,
	            matcher: matcher,
	            value: value,
	            flags: flags
	        };
	    },
	    generate: function(node) {
	        var flagsPrefix = ' ';

	        this.chunk('[');
	        this.node(node.name);

	        if (node.matcher !== null) {
	            this.chunk(node.matcher);

	            if (node.value !== null) {
	                this.node(node.value);

	                // space between string and flags is not required
	                if (node.value.type === 'String') {
	                    flagsPrefix = '';
	                }
	            }
	        }

	        if (node.flags !== null) {
	            this.chunk(flagsPrefix);
	            this.chunk(node.flags);
	        }

	        this.chunk(']');
	    }
	};

	var TYPE$t = tokenizer$3.TYPE;
	var rawMode$4 = Raw.mode;

	var WHITESPACE$7 = TYPE$t.WhiteSpace;
	var COMMENT$5 = TYPE$t.Comment;
	var SEMICOLON$2 = TYPE$t.Semicolon;
	var ATKEYWORD$1 = TYPE$t.AtKeyword;
	var LEFTCURLYBRACKET$1 = TYPE$t.LeftCurlyBracket;
	var RIGHTCURLYBRACKET = TYPE$t.RightCurlyBracket;

	function consumeRaw$4(startToken) {
	    return this.Raw(startToken, null, true);
	}
	function consumeRule() {
	    return this.parseWithFallback(this.Rule, consumeRaw$4);
	}
	function consumeRawDeclaration(startToken) {
	    return this.Raw(startToken, rawMode$4.semicolonIncluded, true);
	}
	function consumeDeclaration() {
	    if (this.scanner.tokenType === SEMICOLON$2) {
	        return consumeRawDeclaration.call(this, this.scanner.tokenIndex);
	    }

	    var node = this.parseWithFallback(this.Declaration, consumeRawDeclaration);

	    if (this.scanner.tokenType === SEMICOLON$2) {
	        this.scanner.next();
	    }

	    return node;
	}

	var Block = {
	    name: 'Block',
	    structure: {
	        children: [[
	            'Atrule',
	            'Rule',
	            'Declaration'
	        ]]
	    },
	    parse: function(isDeclaration) {
	        var consumer = isDeclaration ? consumeDeclaration : consumeRule;

	        var start = this.scanner.tokenStart;
	        var children = this.createList();

	        this.eat(LEFTCURLYBRACKET$1);

	        scan:
	        while (!this.scanner.eof) {
	            switch (this.scanner.tokenType) {
	                case RIGHTCURLYBRACKET:
	                    break scan;

	                case WHITESPACE$7:
	                case COMMENT$5:
	                    this.scanner.next();
	                    break;

	                case ATKEYWORD$1:
	                    children.push(this.parseWithFallback(this.Atrule, consumeRaw$4));
	                    break;

	                default:
	                    children.push(consumer.call(this));
	            }
	        }

	        if (!this.scanner.eof) {
	            this.eat(RIGHTCURLYBRACKET);
	        }

	        return {
	            type: 'Block',
	            loc: this.getLocation(start, this.scanner.tokenStart),
	            children: children
	        };
	    },
	    generate: function(node) {
	        this.chunk('{');
	        this.children(node, function(prev) {
	            if (prev.type === 'Declaration') {
	                this.chunk(';');
	            }
	        });
	        this.chunk('}');
	    },
	    walkContext: 'block'
	};

	var TYPE$s = tokenizer$3.TYPE;

	var LEFTSQUAREBRACKET$2 = TYPE$s.LeftSquareBracket;
	var RIGHTSQUAREBRACKET = TYPE$s.RightSquareBracket;

	var Brackets = {
	    name: 'Brackets',
	    structure: {
	        children: [[]]
	    },
	    parse: function(readSequence, recognizer) {
	        var start = this.scanner.tokenStart;
	        var children = null;

	        this.eat(LEFTSQUAREBRACKET$2);

	        children = readSequence.call(this, recognizer);

	        if (!this.scanner.eof) {
	            this.eat(RIGHTSQUAREBRACKET);
	        }

	        return {
	            type: 'Brackets',
	            loc: this.getLocation(start, this.scanner.tokenStart),
	            children: children
	        };
	    },
	    generate: function(node) {
	        this.chunk('[');
	        this.children(node);
	        this.chunk(']');
	    }
	};

	var CDC$1 = tokenizer$3.TYPE.CDC;

	var CDC_1 = {
	    name: 'CDC',
	    structure: [],
	    parse: function() {
	        var start = this.scanner.tokenStart;

	        this.eat(CDC$1); // -->

	        return {
	            type: 'CDC',
	            loc: this.getLocation(start, this.scanner.tokenStart)
	        };
	    },
	    generate: function() {
	        this.chunk('-->');
	    }
	};

	var CDO$1 = tokenizer$3.TYPE.CDO;

	var CDO_1 = {
	    name: 'CDO',
	    structure: [],
	    parse: function() {
	        var start = this.scanner.tokenStart;

	        this.eat(CDO$1); // <!--

	        return {
	            type: 'CDO',
	            loc: this.getLocation(start, this.scanner.tokenStart)
	        };
	    },
	    generate: function() {
	        this.chunk('<!--');
	    }
	};

	var TYPE$r = tokenizer$3.TYPE;

	var IDENT$d = TYPE$r.Ident;
	var FULLSTOP$2 = 0x002E; // U+002E FULL STOP (.)

	// '.' ident
	var ClassSelector = {
	    name: 'ClassSelector',
	    structure: {
	        name: String
	    },
	    parse: function() {
	        if (!this.scanner.isDelim(FULLSTOP$2)) {
	            this.error('Full stop is expected');
	        }

	        this.scanner.next();

	        return {
	            type: 'ClassSelector',
	            loc: this.getLocation(this.scanner.tokenStart - 1, this.scanner.tokenEnd),
	            name: this.consume(IDENT$d)
	        };
	    },
	    generate: function(node) {
	        this.chunk('.');
	        this.chunk(node.name);
	    }
	};

	var TYPE$q = tokenizer$3.TYPE;

	var IDENT$c = TYPE$q.Ident;
	var PLUSSIGN$4 = 0x002B;        // U+002B PLUS SIGN (+)
	var SOLIDUS$5 = 0x002F;         // U+002F SOLIDUS (/)
	var GREATERTHANSIGN$1 = 0x003E; // U+003E GREATER-THAN SIGN (>)
	var TILDE$1 = 0x007E;           // U+007E TILDE (~)

	// + | > | ~ | /deep/
	var Combinator = {
	    name: 'Combinator',
	    structure: {
	        name: String
	    },
	    parse: function() {
	        var start = this.scanner.tokenStart;
	        var code = this.scanner.source.charCodeAt(this.scanner.tokenStart);

	        switch (code) {
	            case GREATERTHANSIGN$1:
	            case PLUSSIGN$4:
	            case TILDE$1:
	                this.scanner.next();
	                break;

	            case SOLIDUS$5:
	                this.scanner.next();

	                if (this.scanner.tokenType !== IDENT$c || this.scanner.lookupValue(0, 'deep') === false) {
	                    this.error('Identifier `deep` is expected');
	                }

	                this.scanner.next();

	                if (!this.scanner.isDelim(SOLIDUS$5)) {
	                    this.error('Solidus is expected');
	                }

	                this.scanner.next();
	                break;

	            default:
	                this.error('Combinator is expected');
	        }

	        return {
	            type: 'Combinator',
	            loc: this.getLocation(start, this.scanner.tokenStart),
	            name: this.scanner.substrToCursor(start)
	        };
	    },
	    generate: function(node) {
	        this.chunk(node.name);
	    }
	};

	var TYPE$p = tokenizer$3.TYPE;

	var COMMENT$4 = TYPE$p.Comment;
	var ASTERISK$4 = 0x002A;        // U+002A ASTERISK (*)
	var SOLIDUS$4 = 0x002F;         // U+002F SOLIDUS (/)

	// '/*' .* '*/'
	var Comment = {
	    name: 'Comment',
	    structure: {
	        value: String
	    },
	    parse: function() {
	        var start = this.scanner.tokenStart;
	        var end = this.scanner.tokenEnd;

	        this.eat(COMMENT$4);

	        if ((end - start + 2) >= 2 &&
	            this.scanner.source.charCodeAt(end - 2) === ASTERISK$4 &&
	            this.scanner.source.charCodeAt(end - 1) === SOLIDUS$4) {
	            end -= 2;
	        }

	        return {
	            type: 'Comment',
	            loc: this.getLocation(start, this.scanner.tokenStart),
	            value: this.scanner.source.substring(start + 2, end)
	        };
	    },
	    generate: function(node) {
	        this.chunk('/*');
	        this.chunk(node.value);
	        this.chunk('*/');
	    }
	};

	var isCustomProperty = names$2.isCustomProperty;
	var TYPE$o = tokenizer$3.TYPE;
	var rawMode$3 = Raw.mode;

	var IDENT$b = TYPE$o.Ident;
	var HASH$4 = TYPE$o.Hash;
	var COLON$5 = TYPE$o.Colon;
	var SEMICOLON$1 = TYPE$o.Semicolon;
	var DELIM$4 = TYPE$o.Delim;
	var WHITESPACE$6 = TYPE$o.WhiteSpace;
	var EXCLAMATIONMARK$1 = 0x0021; // U+0021 EXCLAMATION MARK (!)
	var NUMBERSIGN$2 = 0x0023;      // U+0023 NUMBER SIGN (#)
	var DOLLARSIGN = 0x0024;      // U+0024 DOLLAR SIGN ($)
	var AMPERSAND = 0x0026;       // U+0026 ANPERSAND (&)
	var ASTERISK$3 = 0x002A;        // U+002A ASTERISK (*)
	var PLUSSIGN$3 = 0x002B;        // U+002B PLUS SIGN (+)
	var SOLIDUS$3 = 0x002F;         // U+002F SOLIDUS (/)

	function consumeValueRaw(startToken) {
	    return this.Raw(startToken, rawMode$3.exclamationMarkOrSemicolon, true);
	}

	function consumeCustomPropertyRaw(startToken) {
	    return this.Raw(startToken, rawMode$3.exclamationMarkOrSemicolon, false);
	}

	function consumeValue() {
	    var startValueToken = this.scanner.tokenIndex;
	    var value = this.Value();

	    if (value.type !== 'Raw' &&
	        this.scanner.eof === false &&
	        this.scanner.tokenType !== SEMICOLON$1 &&
	        this.scanner.isDelim(EXCLAMATIONMARK$1) === false &&
	        this.scanner.isBalanceEdge(startValueToken) === false) {
	        this.error();
	    }

	    return value;
	}

	var Declaration = {
	    name: 'Declaration',
	    structure: {
	        important: [Boolean, String],
	        property: String,
	        value: ['Value', 'Raw']
	    },
	    parse: function() {
	        var start = this.scanner.tokenStart;
	        var startToken = this.scanner.tokenIndex;
	        var property = readProperty.call(this);
	        var customProperty = isCustomProperty(property);
	        var parseValue = customProperty ? this.parseCustomProperty : this.parseValue;
	        var consumeRaw = customProperty ? consumeCustomPropertyRaw : consumeValueRaw;
	        var important = false;
	        var value;

	        this.scanner.skipSC();
	        this.eat(COLON$5);

	        const valueStart = this.scanner.tokenIndex;

	        if (!customProperty) {
	            this.scanner.skipSC();
	        }

	        if (parseValue) {
	            value = this.parseWithFallback(consumeValue, consumeRaw);
	        } else {
	            value = consumeRaw.call(this, this.scanner.tokenIndex);
	        }

	        if (customProperty && value.type === 'Value' && value.children.isEmpty()) {
	            for (let offset = valueStart - this.scanner.tokenIndex; offset <= 0; offset++) {
	                if (this.scanner.lookupType(offset) === WHITESPACE$6) {
	                    value.children.appendData({
	                        type: 'WhiteSpace',
	                        loc: null,
	                        value: ' '
	                    });
	                    break;
	                }
	            }
	        }

	        if (this.scanner.isDelim(EXCLAMATIONMARK$1)) {
	            important = getImportant.call(this);
	            this.scanner.skipSC();
	        }

	        // Do not include semicolon to range per spec
	        // https://drafts.csswg.org/css-syntax/#declaration-diagram

	        if (this.scanner.eof === false &&
	            this.scanner.tokenType !== SEMICOLON$1 &&
	            this.scanner.isBalanceEdge(startToken) === false) {
	            this.error();
	        }

	        return {
	            type: 'Declaration',
	            loc: this.getLocation(start, this.scanner.tokenStart),
	            important: important,
	            property: property,
	            value: value
	        };
	    },
	    generate: function(node) {
	        this.chunk(node.property);
	        this.chunk(':');
	        this.node(node.value);

	        if (node.important) {
	            this.chunk(node.important === true ? '!important' : '!' + node.important);
	        }
	    },
	    walkContext: 'declaration'
	};

	function readProperty() {
	    var start = this.scanner.tokenStart;

	    // hacks
	    if (this.scanner.tokenType === DELIM$4) {
	        switch (this.scanner.source.charCodeAt(this.scanner.tokenStart)) {
	            case ASTERISK$3:
	            case DOLLARSIGN:
	            case PLUSSIGN$3:
	            case NUMBERSIGN$2:
	            case AMPERSAND:
	                this.scanner.next();
	                break;

	            // TODO: not sure we should support this hack
	            case SOLIDUS$3:
	                this.scanner.next();
	                if (this.scanner.isDelim(SOLIDUS$3)) {
	                    this.scanner.next();
	                }
	                break;
	        }
	    }

	    if (this.scanner.tokenType === HASH$4) {
	        this.eat(HASH$4);
	    } else {
	        this.eat(IDENT$b);
	    }

	    return this.scanner.substrToCursor(start);
	}

	// ! ws* important
	function getImportant() {
	    this.eat(DELIM$4);
	    this.scanner.skipSC();

	    var important = this.consume(IDENT$b);

	    // store original value in case it differ from `important`
	    // for better original source restoring and hacks like `!ie` support
	    return important === 'important' ? true : important;
	}

	var TYPE$n = tokenizer$3.TYPE;
	var rawMode$2 = Raw.mode;

	var WHITESPACE$5 = TYPE$n.WhiteSpace;
	var COMMENT$3 = TYPE$n.Comment;
	var SEMICOLON = TYPE$n.Semicolon;

	function consumeRaw$3(startToken) {
	    return this.Raw(startToken, rawMode$2.semicolonIncluded, true);
	}

	var DeclarationList = {
	    name: 'DeclarationList',
	    structure: {
	        children: [[
	            'Declaration'
	        ]]
	    },
	    parse: function() {
	        var children = this.createList();

	        while (!this.scanner.eof) {
	            switch (this.scanner.tokenType) {
	                case WHITESPACE$5:
	                case COMMENT$3:
	                case SEMICOLON:
	                    this.scanner.next();
	                    break;

	                default:
	                    children.push(this.parseWithFallback(this.Declaration, consumeRaw$3));
	            }
	        }

	        return {
	            type: 'DeclarationList',
	            loc: this.getLocationFromList(children),
	            children: children
	        };
	    },
	    generate: function(node) {
	        this.children(node, function(prev) {
	            if (prev.type === 'Declaration') {
	                this.chunk(';');
	            }
	        });
	    }
	};

	var consumeNumber$2 = utils$2.consumeNumber;
	var TYPE$m = tokenizer$3.TYPE;

	var DIMENSION$4 = TYPE$m.Dimension;

	var Dimension = {
	    name: 'Dimension',
	    structure: {
	        value: String,
	        unit: String
	    },
	    parse: function() {
	        var start = this.scanner.tokenStart;
	        var numberEnd = consumeNumber$2(this.scanner.source, start);

	        this.eat(DIMENSION$4);

	        return {
	            type: 'Dimension',
	            loc: this.getLocation(start, this.scanner.tokenStart),
	            value: this.scanner.source.substring(start, numberEnd),
	            unit: this.scanner.source.substring(numberEnd, this.scanner.tokenStart)
	        };
	    },
	    generate: function(node) {
	        this.chunk(node.value);
	        this.chunk(node.unit);
	    }
	};

	var TYPE$l = tokenizer$3.TYPE;

	var RIGHTPARENTHESIS$5 = TYPE$l.RightParenthesis;

	// <function-token> <sequence> )
	var _Function = {
	    name: 'Function',
	    structure: {
	        name: String,
	        children: [[]]
	    },
	    parse: function(readSequence, recognizer) {
	        var start = this.scanner.tokenStart;
	        var name = this.consumeFunctionName();
	        var nameLowerCase = name.toLowerCase();
	        var children;

	        children = recognizer.hasOwnProperty(nameLowerCase)
	            ? recognizer[nameLowerCase].call(this, recognizer)
	            : readSequence.call(this, recognizer);

	        if (!this.scanner.eof) {
	            this.eat(RIGHTPARENTHESIS$5);
	        }

	        return {
	            type: 'Function',
	            loc: this.getLocation(start, this.scanner.tokenStart),
	            name: name,
	            children: children
	        };
	    },
	    generate: function(node) {
	        this.chunk(node.name);
	        this.chunk('(');
	        this.children(node);
	        this.chunk(')');
	    },
	    walkContext: 'function'
	};

	var TYPE$k = tokenizer$3.TYPE;

	var HASH$3 = TYPE$k.Hash;

	// '#' ident
	var Hash = {
	    name: 'Hash',
	    structure: {
	        value: String
	    },
	    parse: function() {
	        var start = this.scanner.tokenStart;

	        this.eat(HASH$3);

	        return {
	            type: 'Hash',
	            loc: this.getLocation(start, this.scanner.tokenStart),
	            value: this.scanner.substrToCursor(start + 1)
	        };
	    },
	    generate: function(node) {
	        this.chunk('#');
	        this.chunk(node.value);
	    }
	};

	var TYPE$j = tokenizer$3.TYPE;

	var IDENT$a = TYPE$j.Ident;

	var Identifier = {
	    name: 'Identifier',
	    structure: {
	        name: String
	    },
	    parse: function() {
	        return {
	            type: 'Identifier',
	            loc: this.getLocation(this.scanner.tokenStart, this.scanner.tokenEnd),
	            name: this.consume(IDENT$a)
	        };
	    },
	    generate: function(node) {
	        this.chunk(node.name);
	    }
	};

	var TYPE$i = tokenizer$3.TYPE;

	var HASH$2 = TYPE$i.Hash;

	// <hash-token>
	var IdSelector = {
	    name: 'IdSelector',
	    structure: {
	        name: String
	    },
	    parse: function() {
	        var start = this.scanner.tokenStart;

	        // TODO: check value is an ident
	        this.eat(HASH$2);

	        return {
	            type: 'IdSelector',
	            loc: this.getLocation(start, this.scanner.tokenStart),
	            name: this.scanner.substrToCursor(start + 1)
	        };
	    },
	    generate: function(node) {
	        this.chunk('#');
	        this.chunk(node.name);
	    }
	};

	var TYPE$h = tokenizer$3.TYPE;

	var IDENT$9 = TYPE$h.Ident;
	var NUMBER$5 = TYPE$h.Number;
	var DIMENSION$3 = TYPE$h.Dimension;
	var LEFTPARENTHESIS$5 = TYPE$h.LeftParenthesis;
	var RIGHTPARENTHESIS$4 = TYPE$h.RightParenthesis;
	var COLON$4 = TYPE$h.Colon;
	var DELIM$3 = TYPE$h.Delim;

	var MediaFeature = {
	    name: 'MediaFeature',
	    structure: {
	        name: String,
	        value: ['Identifier', 'Number', 'Dimension', 'Ratio', null]
	    },
	    parse: function() {
	        var start = this.scanner.tokenStart;
	        var name;
	        var value = null;

	        this.eat(LEFTPARENTHESIS$5);
	        this.scanner.skipSC();

	        name = this.consume(IDENT$9);
	        this.scanner.skipSC();

	        if (this.scanner.tokenType !== RIGHTPARENTHESIS$4) {
	            this.eat(COLON$4);
	            this.scanner.skipSC();

	            switch (this.scanner.tokenType) {
	                case NUMBER$5:
	                    if (this.lookupNonWSType(1) === DELIM$3) {
	                        value = this.Ratio();
	                    } else {
	                        value = this.Number();
	                    }

	                    break;

	                case DIMENSION$3:
	                    value = this.Dimension();
	                    break;

	                case IDENT$9:
	                    value = this.Identifier();

	                    break;

	                default:
	                    this.error('Number, dimension, ratio or identifier is expected');
	            }

	            this.scanner.skipSC();
	        }

	        this.eat(RIGHTPARENTHESIS$4);

	        return {
	            type: 'MediaFeature',
	            loc: this.getLocation(start, this.scanner.tokenStart),
	            name: name,
	            value: value
	        };
	    },
	    generate: function(node) {
	        this.chunk('(');
	        this.chunk(node.name);
	        if (node.value !== null) {
	            this.chunk(':');
	            this.node(node.value);
	        }
	        this.chunk(')');
	    }
	};

	var TYPE$g = tokenizer$3.TYPE;

	var WHITESPACE$4 = TYPE$g.WhiteSpace;
	var COMMENT$2 = TYPE$g.Comment;
	var IDENT$8 = TYPE$g.Ident;
	var LEFTPARENTHESIS$4 = TYPE$g.LeftParenthesis;

	var MediaQuery = {
	    name: 'MediaQuery',
	    structure: {
	        children: [[
	            'Identifier',
	            'MediaFeature',
	            'WhiteSpace'
	        ]]
	    },
	    parse: function() {
	        this.scanner.skipSC();

	        var children = this.createList();
	        var child = null;
	        var space = null;

	        scan:
	        while (!this.scanner.eof) {
	            switch (this.scanner.tokenType) {
	                case COMMENT$2:
	                    this.scanner.next();
	                    continue;

	                case WHITESPACE$4:
	                    space = this.WhiteSpace();
	                    continue;

	                case IDENT$8:
	                    child = this.Identifier();
	                    break;

	                case LEFTPARENTHESIS$4:
	                    child = this.MediaFeature();
	                    break;

	                default:
	                    break scan;
	            }

	            if (space !== null) {
	                children.push(space);
	                space = null;
	            }

	            children.push(child);
	        }

	        if (child === null) {
	            this.error('Identifier or parenthesis is expected');
	        }

	        return {
	            type: 'MediaQuery',
	            loc: this.getLocationFromList(children),
	            children: children
	        };
	    },
	    generate: function(node) {
	        this.children(node);
	    }
	};

	var COMMA$3 = tokenizer$3.TYPE.Comma;

	var MediaQueryList = {
	    name: 'MediaQueryList',
	    structure: {
	        children: [[
	            'MediaQuery'
	        ]]
	    },
	    parse: function(relative) {
	        var children = this.createList();

	        this.scanner.skipSC();

	        while (!this.scanner.eof) {
	            children.push(this.MediaQuery(relative));

	            if (this.scanner.tokenType !== COMMA$3) {
	                break;
	            }

	            this.scanner.next();
	        }

	        return {
	            type: 'MediaQueryList',
	            loc: this.getLocationFromList(children),
	            children: children
	        };
	    },
	    generate: function(node) {
	        this.children(node, function() {
	            this.chunk(',');
	        });
	    }
	};

	var Nth = {
	    name: 'Nth',
	    structure: {
	        nth: ['AnPlusB', 'Identifier'],
	        selector: ['SelectorList', null]
	    },
	    parse: function(allowOfClause) {
	        this.scanner.skipSC();

	        var start = this.scanner.tokenStart;
	        var end = start;
	        var selector = null;
	        var query;

	        if (this.scanner.lookupValue(0, 'odd') || this.scanner.lookupValue(0, 'even')) {
	            query = this.Identifier();
	        } else {
	            query = this.AnPlusB();
	        }

	        this.scanner.skipSC();

	        if (allowOfClause && this.scanner.lookupValue(0, 'of')) {
	            this.scanner.next();

	            selector = this.SelectorList();

	            if (this.needPositions) {
	                end = this.getLastListNode(selector.children).loc.end.offset;
	            }
	        } else {
	            if (this.needPositions) {
	                end = query.loc.end.offset;
	            }
	        }

	        return {
	            type: 'Nth',
	            loc: this.getLocation(start, end),
	            nth: query,
	            selector: selector
	        };
	    },
	    generate: function(node) {
	        this.node(node.nth);
	        if (node.selector !== null) {
	            this.chunk(' of ');
	            this.node(node.selector);
	        }
	    }
	};

	var NUMBER$4 = tokenizer$3.TYPE.Number;

	var _Number = {
	    name: 'Number',
	    structure: {
	        value: String
	    },
	    parse: function() {
	        return {
	            type: 'Number',
	            loc: this.getLocation(this.scanner.tokenStart, this.scanner.tokenEnd),
	            value: this.consume(NUMBER$4)
	        };
	    },
	    generate: function(node) {
	        this.chunk(node.value);
	    }
	};

	// '/' | '*' | ',' | ':' | '+' | '-'
	var Operator = {
	    name: 'Operator',
	    structure: {
	        value: String
	    },
	    parse: function() {
	        var start = this.scanner.tokenStart;

	        this.scanner.next();

	        return {
	            type: 'Operator',
	            loc: this.getLocation(start, this.scanner.tokenStart),
	            value: this.scanner.substrToCursor(start)
	        };
	    },
	    generate: function(node) {
	        this.chunk(node.value);
	    }
	};

	var TYPE$f = tokenizer$3.TYPE;

	var LEFTPARENTHESIS$3 = TYPE$f.LeftParenthesis;
	var RIGHTPARENTHESIS$3 = TYPE$f.RightParenthesis;

	var Parentheses = {
	    name: 'Parentheses',
	    structure: {
	        children: [[]]
	    },
	    parse: function(readSequence, recognizer) {
	        var start = this.scanner.tokenStart;
	        var children = null;

	        this.eat(LEFTPARENTHESIS$3);

	        children = readSequence.call(this, recognizer);

	        if (!this.scanner.eof) {
	            this.eat(RIGHTPARENTHESIS$3);
	        }

	        return {
	            type: 'Parentheses',
	            loc: this.getLocation(start, this.scanner.tokenStart),
	            children: children
	        };
	    },
	    generate: function(node) {
	        this.chunk('(');
	        this.children(node);
	        this.chunk(')');
	    }
	};

	var consumeNumber$1 = utils$2.consumeNumber;
	var TYPE$e = tokenizer$3.TYPE;

	var PERCENTAGE$2 = TYPE$e.Percentage;

	var Percentage = {
	    name: 'Percentage',
	    structure: {
	        value: String
	    },
	    parse: function() {
	        var start = this.scanner.tokenStart;
	        var numberEnd = consumeNumber$1(this.scanner.source, start);

	        this.eat(PERCENTAGE$2);

	        return {
	            type: 'Percentage',
	            loc: this.getLocation(start, this.scanner.tokenStart),
	            value: this.scanner.source.substring(start, numberEnd)
	        };
	    },
	    generate: function(node) {
	        this.chunk(node.value);
	        this.chunk('%');
	    }
	};

	var TYPE$d = tokenizer$3.TYPE;

	var IDENT$7 = TYPE$d.Ident;
	var FUNCTION$5 = TYPE$d.Function;
	var COLON$3 = TYPE$d.Colon;
	var RIGHTPARENTHESIS$2 = TYPE$d.RightParenthesis;

	// : [ <ident> | <function-token> <any-value>? ) ]
	var PseudoClassSelector = {
	    name: 'PseudoClassSelector',
	    structure: {
	        name: String,
	        children: [['Raw'], null]
	    },
	    parse: function() {
	        var start = this.scanner.tokenStart;
	        var children = null;
	        var name;
	        var nameLowerCase;

	        this.eat(COLON$3);

	        if (this.scanner.tokenType === FUNCTION$5) {
	            name = this.consumeFunctionName();
	            nameLowerCase = name.toLowerCase();

	            if (this.pseudo.hasOwnProperty(nameLowerCase)) {
	                this.scanner.skipSC();
	                children = this.pseudo[nameLowerCase].call(this);
	                this.scanner.skipSC();
	            } else {
	                children = this.createList();
	                children.push(
	                    this.Raw(this.scanner.tokenIndex, null, false)
	                );
	            }

	            this.eat(RIGHTPARENTHESIS$2);
	        } else {
	            name = this.consume(IDENT$7);
	        }

	        return {
	            type: 'PseudoClassSelector',
	            loc: this.getLocation(start, this.scanner.tokenStart),
	            name: name,
	            children: children
	        };
	    },
	    generate: function(node) {
	        this.chunk(':');
	        this.chunk(node.name);

	        if (node.children !== null) {
	            this.chunk('(');
	            this.children(node);
	            this.chunk(')');
	        }
	    },
	    walkContext: 'function'
	};

	var TYPE$c = tokenizer$3.TYPE;

	var IDENT$6 = TYPE$c.Ident;
	var FUNCTION$4 = TYPE$c.Function;
	var COLON$2 = TYPE$c.Colon;
	var RIGHTPARENTHESIS$1 = TYPE$c.RightParenthesis;

	// :: [ <ident> | <function-token> <any-value>? ) ]
	var PseudoElementSelector = {
	    name: 'PseudoElementSelector',
	    structure: {
	        name: String,
	        children: [['Raw'], null]
	    },
	    parse: function() {
	        var start = this.scanner.tokenStart;
	        var children = null;
	        var name;
	        var nameLowerCase;

	        this.eat(COLON$2);
	        this.eat(COLON$2);

	        if (this.scanner.tokenType === FUNCTION$4) {
	            name = this.consumeFunctionName();
	            nameLowerCase = name.toLowerCase();

	            if (this.pseudo.hasOwnProperty(nameLowerCase)) {
	                this.scanner.skipSC();
	                children = this.pseudo[nameLowerCase].call(this);
	                this.scanner.skipSC();
	            } else {
	                children = this.createList();
	                children.push(
	                    this.Raw(this.scanner.tokenIndex, null, false)
	                );
	            }

	            this.eat(RIGHTPARENTHESIS$1);
	        } else {
	            name = this.consume(IDENT$6);
	        }

	        return {
	            type: 'PseudoElementSelector',
	            loc: this.getLocation(start, this.scanner.tokenStart),
	            name: name,
	            children: children
	        };
	    },
	    generate: function(node) {
	        this.chunk('::');
	        this.chunk(node.name);

	        if (node.children !== null) {
	            this.chunk('(');
	            this.children(node);
	            this.chunk(')');
	        }
	    },
	    walkContext: 'function'
	};

	var isDigit = tokenizer$3.isDigit;
	var TYPE$b = tokenizer$3.TYPE;

	var NUMBER$3 = TYPE$b.Number;
	var DELIM$2 = TYPE$b.Delim;
	var SOLIDUS$2 = 0x002F;  // U+002F SOLIDUS (/)
	var FULLSTOP$1 = 0x002E; // U+002E FULL STOP (.)

	// Terms of <ratio> should be a positive numbers (not zero or negative)
	// (see https://drafts.csswg.org/mediaqueries-3/#values)
	// However, -o-min-device-pixel-ratio takes fractional values as a ratio's term
	// and this is using by various sites. Therefore we relax checking on parse
	// to test a term is unsigned number without an exponent part.
	// Additional checking may be applied on lexer validation.
	function consumeNumber() {
	    this.scanner.skipWS();

	    var value = this.consume(NUMBER$3);

	    for (var i = 0; i < value.length; i++) {
	        var code = value.charCodeAt(i);
	        if (!isDigit(code) && code !== FULLSTOP$1) {
	            this.error('Unsigned number is expected', this.scanner.tokenStart - value.length + i);
	        }
	    }

	    if (Number(value) === 0) {
	        this.error('Zero number is not allowed', this.scanner.tokenStart - value.length);
	    }

	    return value;
	}

	// <positive-integer> S* '/' S* <positive-integer>
	var Ratio = {
	    name: 'Ratio',
	    structure: {
	        left: String,
	        right: String
	    },
	    parse: function() {
	        var start = this.scanner.tokenStart;
	        var left = consumeNumber.call(this);
	        var right;

	        this.scanner.skipWS();

	        if (!this.scanner.isDelim(SOLIDUS$2)) {
	            this.error('Solidus is expected');
	        }
	        this.eat(DELIM$2);
	        right = consumeNumber.call(this);

	        return {
	            type: 'Ratio',
	            loc: this.getLocation(start, this.scanner.tokenStart),
	            left: left,
	            right: right
	        };
	    },
	    generate: function(node) {
	        this.chunk(node.left);
	        this.chunk('/');
	        this.chunk(node.right);
	    }
	};

	var TYPE$a = tokenizer$3.TYPE;
	var rawMode$1 = Raw.mode;

	var LEFTCURLYBRACKET = TYPE$a.LeftCurlyBracket;

	function consumeRaw$2(startToken) {
	    return this.Raw(startToken, rawMode$1.leftCurlyBracket, true);
	}

	function consumePrelude() {
	    var prelude = this.SelectorList();

	    if (prelude.type !== 'Raw' &&
	        this.scanner.eof === false &&
	        this.scanner.tokenType !== LEFTCURLYBRACKET) {
	        this.error();
	    }

	    return prelude;
	}

	var Rule = {
	    name: 'Rule',
	    structure: {
	        prelude: ['SelectorList', 'Raw'],
	        block: ['Block']
	    },
	    parse: function() {
	        var startToken = this.scanner.tokenIndex;
	        var startOffset = this.scanner.tokenStart;
	        var prelude;
	        var block;

	        if (this.parseRulePrelude) {
	            prelude = this.parseWithFallback(consumePrelude, consumeRaw$2);
	        } else {
	            prelude = consumeRaw$2.call(this, startToken);
	        }

	        block = this.Block(true);

	        return {
	            type: 'Rule',
	            loc: this.getLocation(startOffset, this.scanner.tokenStart),
	            prelude: prelude,
	            block: block
	        };
	    },
	    generate: function(node) {
	        this.node(node.prelude);
	        this.node(node.block);
	    },
	    walkContext: 'rule'
	};

	var Selector = {
	    name: 'Selector',
	    structure: {
	        children: [[
	            'TypeSelector',
	            'IdSelector',
	            'ClassSelector',
	            'AttributeSelector',
	            'PseudoClassSelector',
	            'PseudoElementSelector',
	            'Combinator',
	            'WhiteSpace'
	        ]]
	    },
	    parse: function() {
	        var children = this.readSequence(this.scope.Selector);

	        // nothing were consumed
	        if (this.getFirstListNode(children) === null) {
	            this.error('Selector is expected');
	        }

	        return {
	            type: 'Selector',
	            loc: this.getLocationFromList(children),
	            children: children
	        };
	    },
	    generate: function(node) {
	        this.children(node);
	    }
	};

	var TYPE$9 = tokenizer$3.TYPE;

	var COMMA$2 = TYPE$9.Comma;

	var SelectorList = {
	    name: 'SelectorList',
	    structure: {
	        children: [[
	            'Selector',
	            'Raw'
	        ]]
	    },
	    parse: function() {
	        var children = this.createList();

	        while (!this.scanner.eof) {
	            children.push(this.Selector());

	            if (this.scanner.tokenType === COMMA$2) {
	                this.scanner.next();
	                continue;
	            }

	            break;
	        }

	        return {
	            type: 'SelectorList',
	            loc: this.getLocationFromList(children),
	            children: children
	        };
	    },
	    generate: function(node) {
	        this.children(node, function() {
	            this.chunk(',');
	        });
	    },
	    walkContext: 'selector'
	};

	var STRING$2 = tokenizer$3.TYPE.String;

	var _String = {
	    name: 'String',
	    structure: {
	        value: String
	    },
	    parse: function() {
	        return {
	            type: 'String',
	            loc: this.getLocation(this.scanner.tokenStart, this.scanner.tokenEnd),
	            value: this.consume(STRING$2)
	        };
	    },
	    generate: function(node) {
	        this.chunk(node.value);
	    }
	};

	var TYPE$8 = tokenizer$3.TYPE;

	var WHITESPACE$3 = TYPE$8.WhiteSpace;
	var COMMENT$1 = TYPE$8.Comment;
	var ATKEYWORD = TYPE$8.AtKeyword;
	var CDO = TYPE$8.CDO;
	var CDC = TYPE$8.CDC;
	var EXCLAMATIONMARK = 0x0021; // U+0021 EXCLAMATION MARK (!)

	function consumeRaw$1(startToken) {
	    return this.Raw(startToken, null, false);
	}

	var StyleSheet = {
	    name: 'StyleSheet',
	    structure: {
	        children: [[
	            'Comment',
	            'CDO',
	            'CDC',
	            'Atrule',
	            'Rule',
	            'Raw'
	        ]]
	    },
	    parse: function() {
	        var start = this.scanner.tokenStart;
	        var children = this.createList();
	        var child;

	        while (!this.scanner.eof) {
	            switch (this.scanner.tokenType) {
	                case WHITESPACE$3:
	                    this.scanner.next();
	                    continue;

	                case COMMENT$1:
	                    // ignore comments except exclamation comments (i.e. /*! .. */) on top level
	                    if (this.scanner.source.charCodeAt(this.scanner.tokenStart + 2) !== EXCLAMATIONMARK) {
	                        this.scanner.next();
	                        continue;
	                    }

	                    child = this.Comment();
	                    break;

	                case CDO: // <!--
	                    child = this.CDO();
	                    break;

	                case CDC: // -->
	                    child = this.CDC();
	                    break;

	                // CSS Syntax Module Level 3
	                // §2.2 Error handling
	                // At the "top level" of a stylesheet, an <at-keyword-token> starts an at-rule.
	                case ATKEYWORD:
	                    child = this.parseWithFallback(this.Atrule, consumeRaw$1);
	                    break;

	                // Anything else starts a qualified rule ...
	                default:
	                    child = this.parseWithFallback(this.Rule, consumeRaw$1);
	            }

	            children.push(child);
	        }

	        return {
	            type: 'StyleSheet',
	            loc: this.getLocation(start, this.scanner.tokenStart),
	            children: children
	        };
	    },
	    generate: function(node) {
	        this.children(node);
	    },
	    walkContext: 'stylesheet'
	};

	var TYPE$7 = tokenizer$3.TYPE;

	var IDENT$5 = TYPE$7.Ident;
	var ASTERISK$2 = 0x002A;     // U+002A ASTERISK (*)
	var VERTICALLINE$1 = 0x007C; // U+007C VERTICAL LINE (|)

	function eatIdentifierOrAsterisk() {
	    if (this.scanner.tokenType !== IDENT$5 &&
	        this.scanner.isDelim(ASTERISK$2) === false) {
	        this.error('Identifier or asterisk is expected');
	    }

	    this.scanner.next();
	}

	// ident
	// ident|ident
	// ident|*
	// *
	// *|ident
	// *|*
	// |ident
	// |*
	var TypeSelector = {
	    name: 'TypeSelector',
	    structure: {
	        name: String
	    },
	    parse: function() {
	        var start = this.scanner.tokenStart;

	        if (this.scanner.isDelim(VERTICALLINE$1)) {
	            this.scanner.next();
	            eatIdentifierOrAsterisk.call(this);
	        } else {
	            eatIdentifierOrAsterisk.call(this);

	            if (this.scanner.isDelim(VERTICALLINE$1)) {
	                this.scanner.next();
	                eatIdentifierOrAsterisk.call(this);
	            }
	        }

	        return {
	            type: 'TypeSelector',
	            loc: this.getLocation(start, this.scanner.tokenStart),
	            name: this.scanner.substrToCursor(start)
	        };
	    },
	    generate: function(node) {
	        this.chunk(node.name);
	    }
	};

	var isHexDigit = tokenizer$3.isHexDigit;
	var cmpChar$1 = tokenizer$3.cmpChar;
	var TYPE$6 = tokenizer$3.TYPE;
	var NAME = tokenizer$3.NAME;

	var IDENT$4 = TYPE$6.Ident;
	var NUMBER$2 = TYPE$6.Number;
	var DIMENSION$2 = TYPE$6.Dimension;
	var PLUSSIGN$2 = 0x002B;     // U+002B PLUS SIGN (+)
	var HYPHENMINUS$1 = 0x002D;  // U+002D HYPHEN-MINUS (-)
	var QUESTIONMARK = 0x003F; // U+003F QUESTION MARK (?)
	var U$1 = 0x0075;            // U+0075 LATIN SMALL LETTER U (u)

	function eatHexSequence(offset, allowDash) {
	    for (var pos = this.scanner.tokenStart + offset, len = 0; pos < this.scanner.tokenEnd; pos++) {
	        var code = this.scanner.source.charCodeAt(pos);

	        if (code === HYPHENMINUS$1 && allowDash && len !== 0) {
	            if (eatHexSequence.call(this, offset + len + 1, false) === 0) {
	                this.error();
	            }

	            return -1;
	        }

	        if (!isHexDigit(code)) {
	            this.error(
	                allowDash && len !== 0
	                    ? 'HyphenMinus' + (len < 6 ? ' or hex digit' : '') + ' is expected'
	                    : (len < 6 ? 'Hex digit is expected' : 'Unexpected input'),
	                pos
	            );
	        }

	        if (++len > 6) {
	            this.error('Too many hex digits', pos);
	        }    }

	    this.scanner.next();
	    return len;
	}

	function eatQuestionMarkSequence(max) {
	    var count = 0;

	    while (this.scanner.isDelim(QUESTIONMARK)) {
	        if (++count > max) {
	            this.error('Too many question marks');
	        }

	        this.scanner.next();
	    }
	}

	function startsWith(code) {
	    if (this.scanner.source.charCodeAt(this.scanner.tokenStart) !== code) {
	        this.error(NAME[code] + ' is expected');
	    }
	}

	// https://drafts.csswg.org/css-syntax/#urange
	// Informally, the <urange> production has three forms:
	// U+0001
	//      Defines a range consisting of a single code point, in this case the code point "1".
	// U+0001-00ff
	//      Defines a range of codepoints between the first and the second value, in this case
	//      the range between "1" and "ff" (255 in decimal) inclusive.
	// U+00??
	//      Defines a range of codepoints where the "?" characters range over all hex digits,
	//      in this case defining the same as the value U+0000-00ff.
	// In each form, a maximum of 6 digits is allowed for each hexadecimal number (if you treat "?" as a hexadecimal digit).
	//
	// <urange> =
	//   u '+' <ident-token> '?'* |
	//   u <dimension-token> '?'* |
	//   u <number-token> '?'* |
	//   u <number-token> <dimension-token> |
	//   u <number-token> <number-token> |
	//   u '+' '?'+
	function scanUnicodeRange() {
	    var hexLength = 0;

	    // u '+' <ident-token> '?'*
	    // u '+' '?'+
	    if (this.scanner.isDelim(PLUSSIGN$2)) {
	        this.scanner.next();

	        if (this.scanner.tokenType === IDENT$4) {
	            hexLength = eatHexSequence.call(this, 0, true);
	            if (hexLength > 0) {
	                eatQuestionMarkSequence.call(this, 6 - hexLength);
	            }
	            return;
	        }

	        if (this.scanner.isDelim(QUESTIONMARK)) {
	            this.scanner.next();
	            eatQuestionMarkSequence.call(this, 5);
	            return;
	        }

	        this.error('Hex digit or question mark is expected');
	        return;
	    }

	    // u <number-token> '?'*
	    // u <number-token> <dimension-token>
	    // u <number-token> <number-token>
	    if (this.scanner.tokenType === NUMBER$2) {
	        startsWith.call(this, PLUSSIGN$2);
	        hexLength = eatHexSequence.call(this, 1, true);

	        if (this.scanner.isDelim(QUESTIONMARK)) {
	            eatQuestionMarkSequence.call(this, 6 - hexLength);
	            return;
	        }

	        if (this.scanner.tokenType === DIMENSION$2 ||
	            this.scanner.tokenType === NUMBER$2) {
	            startsWith.call(this, HYPHENMINUS$1);
	            eatHexSequence.call(this, 1, false);
	            return;
	        }

	        return;
	    }

	    // u <dimension-token> '?'*
	    if (this.scanner.tokenType === DIMENSION$2) {
	        startsWith.call(this, PLUSSIGN$2);
	        hexLength = eatHexSequence.call(this, 1, true);

	        if (hexLength > 0) {
	            eatQuestionMarkSequence.call(this, 6 - hexLength);
	        }

	        return;
	    }

	    this.error();
	}

	var UnicodeRange = {
	    name: 'UnicodeRange',
	    structure: {
	        value: String
	    },
	    parse: function() {
	        var start = this.scanner.tokenStart;

	        // U or u
	        if (!cmpChar$1(this.scanner.source, start, U$1)) {
	            this.error('U is expected');
	        }

	        if (!cmpChar$1(this.scanner.source, start + 1, PLUSSIGN$2)) {
	            this.error('Plus sign is expected');
	        }

	        this.scanner.next();
	        scanUnicodeRange.call(this);

	        return {
	            type: 'UnicodeRange',
	            loc: this.getLocation(start, this.scanner.tokenStart),
	            value: this.scanner.substrToCursor(start)
	        };
	    },
	    generate: function(node) {
	        this.chunk(node.value);
	    }
	};

	var isWhiteSpace = tokenizer$3.isWhiteSpace;
	var cmpStr$1 = tokenizer$3.cmpStr;
	var TYPE$5 = tokenizer$3.TYPE;

	var FUNCTION$3 = TYPE$5.Function;
	var URL$3 = TYPE$5.Url;
	var RIGHTPARENTHESIS = TYPE$5.RightParenthesis;

	// <url-token> | <function-token> <string> )
	var Url = {
	    name: 'Url',
	    structure: {
	        value: ['String', 'Raw']
	    },
	    parse: function() {
	        var start = this.scanner.tokenStart;
	        var value;

	        switch (this.scanner.tokenType) {
	            case URL$3:
	                var rawStart = start + 4;
	                var rawEnd = this.scanner.tokenEnd - 1;

	                while (rawStart < rawEnd && isWhiteSpace(this.scanner.source.charCodeAt(rawStart))) {
	                    rawStart++;
	                }

	                while (rawStart < rawEnd && isWhiteSpace(this.scanner.source.charCodeAt(rawEnd - 1))) {
	                    rawEnd--;
	                }

	                value = {
	                    type: 'Raw',
	                    loc: this.getLocation(rawStart, rawEnd),
	                    value: this.scanner.source.substring(rawStart, rawEnd)
	                };

	                this.eat(URL$3);
	                break;

	            case FUNCTION$3:
	                if (!cmpStr$1(this.scanner.source, this.scanner.tokenStart, this.scanner.tokenEnd, 'url(')) {
	                    this.error('Function name must be `url`');
	                }

	                this.eat(FUNCTION$3);
	                this.scanner.skipSC();
	                value = this.String();
	                this.scanner.skipSC();
	                this.eat(RIGHTPARENTHESIS);
	                break;

	            default:
	                this.error('Url or Function is expected');
	        }

	        return {
	            type: 'Url',
	            loc: this.getLocation(start, this.scanner.tokenStart),
	            value: value
	        };
	    },
	    generate: function(node) {
	        this.chunk('url');
	        this.chunk('(');
	        this.node(node.value);
	        this.chunk(')');
	    }
	};

	var Value = {
	    name: 'Value',
	    structure: {
	        children: [[]]
	    },
	    parse: function() {
	        var start = this.scanner.tokenStart;
	        var children = this.readSequence(this.scope.Value);

	        return {
	            type: 'Value',
	            loc: this.getLocation(start, this.scanner.tokenStart),
	            children: children
	        };
	    },
	    generate: function(node) {
	        this.children(node);
	    }
	};

	var WHITESPACE$2 = tokenizer$3.TYPE.WhiteSpace;
	var SPACE = Object.freeze({
	    type: 'WhiteSpace',
	    loc: null,
	    value: ' '
	});

	var WhiteSpace = {
	    name: 'WhiteSpace',
	    structure: {
	        value: String
	    },
	    parse: function() {
	        this.eat(WHITESPACE$2);
	        return SPACE;

	        // return {
	        //     type: 'WhiteSpace',
	        //     loc: this.getLocation(this.scanner.tokenStart, this.scanner.tokenEnd),
	        //     value: this.consume(WHITESPACE)
	        // };
	    },
	    generate: function(node) {
	        this.chunk(node.value);
	    }
	};

	var node = {
	    AnPlusB: AnPlusB,
	    Atrule: Atrule,
	    AtrulePrelude: AtrulePrelude,
	    AttributeSelector: AttributeSelector,
	    Block: Block,
	    Brackets: Brackets,
	    CDC: CDC_1,
	    CDO: CDO_1,
	    ClassSelector: ClassSelector,
	    Combinator: Combinator,
	    Comment: Comment,
	    Declaration: Declaration,
	    DeclarationList: DeclarationList,
	    Dimension: Dimension,
	    Function: _Function,
	    Hash: Hash,
	    Identifier: Identifier,
	    IdSelector: IdSelector,
	    MediaFeature: MediaFeature,
	    MediaQuery: MediaQuery,
	    MediaQueryList: MediaQueryList,
	    Nth: Nth,
	    Number: _Number,
	    Operator: Operator,
	    Parentheses: Parentheses,
	    Percentage: Percentage,
	    PseudoClassSelector: PseudoClassSelector,
	    PseudoElementSelector: PseudoElementSelector,
	    Ratio: Ratio,
	    Raw: Raw,
	    Rule: Rule,
	    Selector: Selector,
	    SelectorList: SelectorList,
	    String: _String,
	    StyleSheet: StyleSheet,
	    TypeSelector: TypeSelector,
	    UnicodeRange: UnicodeRange,
	    Url: Url,
	    Value: Value,
	    WhiteSpace: WhiteSpace
	};

	var data = data$1;

	var lexer = {
	    generic: true,
	    types: data.types,
	    atrules: data.atrules,
	    properties: data.properties,
	    node: node
	};

	var cmpChar = tokenizer$3.cmpChar;
	var cmpStr = tokenizer$3.cmpStr;
	var TYPE$4 = tokenizer$3.TYPE;

	var IDENT$3 = TYPE$4.Ident;
	var STRING$1 = TYPE$4.String;
	var NUMBER$1 = TYPE$4.Number;
	var FUNCTION$2 = TYPE$4.Function;
	var URL$2 = TYPE$4.Url;
	var HASH$1 = TYPE$4.Hash;
	var DIMENSION$1 = TYPE$4.Dimension;
	var PERCENTAGE$1 = TYPE$4.Percentage;
	var LEFTPARENTHESIS$2 = TYPE$4.LeftParenthesis;
	var LEFTSQUAREBRACKET$1 = TYPE$4.LeftSquareBracket;
	var COMMA$1 = TYPE$4.Comma;
	var DELIM$1 = TYPE$4.Delim;
	var NUMBERSIGN$1 = 0x0023;  // U+0023 NUMBER SIGN (#)
	var ASTERISK$1 = 0x002A;    // U+002A ASTERISK (*)
	var PLUSSIGN$1 = 0x002B;    // U+002B PLUS SIGN (+)
	var HYPHENMINUS = 0x002D; // U+002D HYPHEN-MINUS (-)
	var SOLIDUS$1 = 0x002F;     // U+002F SOLIDUS (/)
	var U = 0x0075;           // U+0075 LATIN SMALL LETTER U (u)

	var _default = function defaultRecognizer(context) {
	    switch (this.scanner.tokenType) {
	        case HASH$1:
	            return this.Hash();

	        case COMMA$1:
	            context.space = null;
	            context.ignoreWSAfter = true;
	            return this.Operator();

	        case LEFTPARENTHESIS$2:
	            return this.Parentheses(this.readSequence, context.recognizer);

	        case LEFTSQUAREBRACKET$1:
	            return this.Brackets(this.readSequence, context.recognizer);

	        case STRING$1:
	            return this.String();

	        case DIMENSION$1:
	            return this.Dimension();

	        case PERCENTAGE$1:
	            return this.Percentage();

	        case NUMBER$1:
	            return this.Number();

	        case FUNCTION$2:
	            return cmpStr(this.scanner.source, this.scanner.tokenStart, this.scanner.tokenEnd, 'url(')
	                ? this.Url()
	                : this.Function(this.readSequence, context.recognizer);

	        case URL$2:
	            return this.Url();

	        case IDENT$3:
	            // check for unicode range, it should start with u+ or U+
	            if (cmpChar(this.scanner.source, this.scanner.tokenStart, U) &&
	                cmpChar(this.scanner.source, this.scanner.tokenStart + 1, PLUSSIGN$1)) {
	                return this.UnicodeRange();
	            } else {
	                return this.Identifier();
	            }

	        case DELIM$1:
	            var code = this.scanner.source.charCodeAt(this.scanner.tokenStart);

	            if (code === SOLIDUS$1 ||
	                code === ASTERISK$1 ||
	                code === PLUSSIGN$1 ||
	                code === HYPHENMINUS) {
	                return this.Operator(); // TODO: replace with Delim
	            }

	            // TODO: produce a node with Delim node type

	            if (code === NUMBERSIGN$1) {
	                this.error('Hex or identifier is expected', this.scanner.tokenStart + 1);
	            }

	            break;
	    }
	};

	var atrulePrelude = {
	    getNode: _default
	};

	var TYPE$3 = tokenizer$3.TYPE;

	var DELIM = TYPE$3.Delim;
	var IDENT$2 = TYPE$3.Ident;
	var DIMENSION = TYPE$3.Dimension;
	var PERCENTAGE = TYPE$3.Percentage;
	var NUMBER = TYPE$3.Number;
	var HASH = TYPE$3.Hash;
	var COLON$1 = TYPE$3.Colon;
	var LEFTSQUAREBRACKET = TYPE$3.LeftSquareBracket;
	var NUMBERSIGN = 0x0023;      // U+0023 NUMBER SIGN (#)
	var ASTERISK = 0x002A;        // U+002A ASTERISK (*)
	var PLUSSIGN = 0x002B;        // U+002B PLUS SIGN (+)
	var SOLIDUS = 0x002F;         // U+002F SOLIDUS (/)
	var FULLSTOP = 0x002E;        // U+002E FULL STOP (.)
	var GREATERTHANSIGN = 0x003E; // U+003E GREATER-THAN SIGN (>)
	var VERTICALLINE = 0x007C;    // U+007C VERTICAL LINE (|)
	var TILDE = 0x007E;           // U+007E TILDE (~)

	function getNode(context) {
	    switch (this.scanner.tokenType) {
	        case LEFTSQUAREBRACKET:
	            return this.AttributeSelector();

	        case HASH:
	            return this.IdSelector();

	        case COLON$1:
	            if (this.scanner.lookupType(1) === COLON$1) {
	                return this.PseudoElementSelector();
	            } else {
	                return this.PseudoClassSelector();
	            }

	        case IDENT$2:
	            return this.TypeSelector();

	        case NUMBER:
	        case PERCENTAGE:
	            return this.Percentage();

	        case DIMENSION:
	            // throws when .123ident
	            if (this.scanner.source.charCodeAt(this.scanner.tokenStart) === FULLSTOP) {
	                this.error('Identifier is expected', this.scanner.tokenStart + 1);
	            }
	            break;

	        case DELIM:
	            var code = this.scanner.source.charCodeAt(this.scanner.tokenStart);

	            switch (code) {
	                case PLUSSIGN:
	                case GREATERTHANSIGN:
	                case TILDE:
	                    context.space = null;
	                    context.ignoreWSAfter = true;
	                    return this.Combinator();

	                case SOLIDUS:  // /deep/
	                    return this.Combinator();

	                case FULLSTOP:
	                    return this.ClassSelector();

	                case ASTERISK:
	                case VERTICALLINE:
	                    return this.TypeSelector();

	                case NUMBERSIGN:
	                    return this.IdSelector();
	            }

	            break;
	    }
	}
	var selector = {
	    getNode: getNode
	};

	// legacy IE function
	// expression( <any-value> )
	var expression = function() {
	    return this.createSingleNodeList(
	        this.Raw(this.scanner.tokenIndex, null, false)
	    );
	};

	var TYPE$2 = tokenizer$3.TYPE;
	var rawMode = Raw.mode;

	var COMMA = TYPE$2.Comma;
	var WHITESPACE$1 = TYPE$2.WhiteSpace;

	// var( <ident> , <value>? )
	var _var = function() {
	    var children = this.createList();

	    this.scanner.skipSC();

	    // NOTE: Don't check more than a first argument is an ident, rest checks are for lexer
	    children.push(this.Identifier());

	    this.scanner.skipSC();

	    if (this.scanner.tokenType === COMMA) {
	        children.push(this.Operator());

	        const startIndex = this.scanner.tokenIndex;
	        const value = this.parseCustomProperty
	            ? this.Value(null)
	            : this.Raw(this.scanner.tokenIndex, rawMode.exclamationMarkOrSemicolon, false);

	        if (value.type === 'Value' && value.children.isEmpty()) {
	            for (let offset = startIndex - this.scanner.tokenIndex; offset <= 0; offset++) {
	                if (this.scanner.lookupType(offset) === WHITESPACE$1) {
	                    value.children.appendData({
	                        type: 'WhiteSpace',
	                        loc: null,
	                        value: ' '
	                    });
	                    break;
	                }
	            }
	        }

	        children.push(value);
	    }

	    return children;
	};

	var value$2 = {
	    getNode: _default,
	    'expression': expression,
	    'var': _var
	};

	var scope = {
	    AtrulePrelude: atrulePrelude,
	    Selector: selector,
	    Value: value$2
	};

	var fontFace = {
	    parse: {
	        prelude: null,
	        block: function() {
	            return this.Block(true);
	        }
	    }
	};

	var TYPE$1 = tokenizer$3.TYPE;

	var STRING = TYPE$1.String;
	var IDENT$1 = TYPE$1.Ident;
	var URL$1 = TYPE$1.Url;
	var FUNCTION$1 = TYPE$1.Function;
	var LEFTPARENTHESIS$1 = TYPE$1.LeftParenthesis;

	var _import = {
	    parse: {
	        prelude: function() {
	            var children = this.createList();

	            this.scanner.skipSC();

	            switch (this.scanner.tokenType) {
	                case STRING:
	                    children.push(this.String());
	                    break;

	                case URL$1:
	                case FUNCTION$1:
	                    children.push(this.Url());
	                    break;

	                default:
	                    this.error('String or url() is expected');
	            }

	            if (this.lookupNonWSType(0) === IDENT$1 ||
	                this.lookupNonWSType(0) === LEFTPARENTHESIS$1) {
	                children.push(this.WhiteSpace());
	                children.push(this.MediaQueryList());
	            }

	            return children;
	        },
	        block: null
	    }
	};

	var media = {
	    parse: {
	        prelude: function() {
	            return this.createSingleNodeList(
	                this.MediaQueryList()
	            );
	        },
	        block: function() {
	            return this.Block(false);
	        }
	    }
	};

	var page = {
	    parse: {
	        prelude: function() {
	            return this.createSingleNodeList(
	                this.SelectorList()
	            );
	        },
	        block: function() {
	            return this.Block(true);
	        }
	    }
	};

	var TYPE = tokenizer$3.TYPE;

	var WHITESPACE = TYPE.WhiteSpace;
	var COMMENT = TYPE.Comment;
	var IDENT = TYPE.Ident;
	var FUNCTION = TYPE.Function;
	var COLON = TYPE.Colon;
	var LEFTPARENTHESIS = TYPE.LeftParenthesis;

	function consumeRaw() {
	    return this.createSingleNodeList(
	        this.Raw(this.scanner.tokenIndex, null, false)
	    );
	}

	function parentheses() {
	    this.scanner.skipSC();

	    if (this.scanner.tokenType === IDENT &&
	        this.lookupNonWSType(1) === COLON) {
	        return this.createSingleNodeList(
	            this.Declaration()
	        );
	    }

	    return readSequence.call(this);
	}

	function readSequence() {
	    var children = this.createList();
	    var space = null;
	    var child;

	    this.scanner.skipSC();

	    scan:
	    while (!this.scanner.eof) {
	        switch (this.scanner.tokenType) {
	            case WHITESPACE:
	                space = this.WhiteSpace();
	                continue;

	            case COMMENT:
	                this.scanner.next();
	                continue;

	            case FUNCTION:
	                child = this.Function(consumeRaw, this.scope.AtrulePrelude);
	                break;

	            case IDENT:
	                child = this.Identifier();
	                break;

	            case LEFTPARENTHESIS:
	                child = this.Parentheses(parentheses, this.scope.AtrulePrelude);
	                break;

	            default:
	                break scan;
	        }

	        if (space !== null) {
	            children.push(space);
	            space = null;
	        }

	        children.push(child);
	    }

	    return children;
	}

	var supports = {
	    parse: {
	        prelude: function() {
	            var children = readSequence.call(this);

	            if (this.getFirstListNode(children) === null) {
	                this.error('Condition is expected');
	            }

	            return children;
	        },
	        block: function() {
	            return this.Block(false);
	        }
	    }
	};

	var atrule = {
	    'font-face': fontFace,
	    'import': _import,
	    'media': media,
	    'page': page,
	    'supports': supports
	};

	var dir = {
	    parse: function() {
	        return this.createSingleNodeList(
	            this.Identifier()
	        );
	    }
	};

	var has = {
	    parse: function() {
	        return this.createSingleNodeList(
	            this.SelectorList()
	        );
	    }
	};

	var lang = {
	    parse: function() {
	        return this.createSingleNodeList(
	            this.Identifier()
	        );
	    }
	};

	var selectorList = {
	    parse: function selectorList() {
	        return this.createSingleNodeList(
	            this.SelectorList()
	        );
	    }
	};

	var matches = selectorList;

	var not = selectorList;

	var ALLOW_OF_CLAUSE = true;

	var nthWithOfClause = {
	    parse: function nthWithOfClause() {
	        return this.createSingleNodeList(
	            this.Nth(ALLOW_OF_CLAUSE)
	        );
	    }
	};

	var nthChild = nthWithOfClause;

	var nthLastChild = nthWithOfClause;

	var DISALLOW_OF_CLAUSE = false;

	var nth = {
	    parse: function nth() {
	        return this.createSingleNodeList(
	            this.Nth(DISALLOW_OF_CLAUSE)
	        );
	    }
	};

	var nthLastOfType = nth;

	var nthOfType = nth;

	var slotted = {
	    parse: function compoundSelector() {
	        return this.createSingleNodeList(
	            this.Selector()
	        );
	    }
	};

	var pseudo = {
	    'dir': dir,
	    'has': has,
	    'lang': lang,
	    'matches': matches,
	    'not': not,
	    'nth-child': nthChild,
	    'nth-last-child': nthLastChild,
	    'nth-last-of-type': nthLastOfType,
	    'nth-of-type': nthOfType,
	    'slotted': slotted
	};

	var parser = {
	    parseContext: {
	        default: 'StyleSheet',
	        stylesheet: 'StyleSheet',
	        atrule: 'Atrule',
	        atrulePrelude: function(options) {
	            return this.AtrulePrelude(options.atrule ? String(options.atrule) : null);
	        },
	        mediaQueryList: 'MediaQueryList',
	        mediaQuery: 'MediaQuery',
	        rule: 'Rule',
	        selectorList: 'SelectorList',
	        selector: 'Selector',
	        block: function() {
	            return this.Block(true);
	        },
	        declarationList: 'DeclarationList',
	        declaration: 'Declaration',
	        value: 'Value'
	    },
	    scope: scope,
	    atrule: atrule,
	    pseudo: pseudo,
	    node: node
	};

	var walker = {
	    node: node
	};

	var name = "css-tree";
	var version = "1.1.3";
	var description = "A tool set for CSS: fast detailed parser (CSS → AST), walker (AST traversal), generator (AST → CSS) and lexer (validation and matching) based on specs and browser implementations";
	var author = "Roman Dvornov <rdvornov@gmail.com> (https://github.com/lahmatiy)";
	var license = "MIT";
	var repository = "csstree/csstree";
	var keywords = [
		"css",
		"ast",
		"tokenizer",
		"parser",
		"walker",
		"lexer",
		"generator",
		"utils",
		"syntax",
		"validation"
	];
	var main = "lib/index.js";
	var unpkg = "dist/csstree.min.js";
	var jsdelivr = "dist/csstree.min.js";
	var scripts = {
		build: "rollup --config",
		lint: "eslint data lib scripts test && node scripts/review-syntax-patch --lint && node scripts/update-docs --lint",
		"lint-and-test": "npm run lint && npm test",
		"update:docs": "node scripts/update-docs",
		"review:syntax-patch": "node scripts/review-syntax-patch",
		test: "mocha --reporter progress",
		coverage: "nyc npm test",
		travis: "nyc npm run lint-and-test && npm run coveralls",
		coveralls: "nyc report --reporter=text-lcov | coveralls",
		prepublishOnly: "npm run build",
		hydrogen: "node --trace-hydrogen --trace-phase=Z --trace-deopt --code-comments --hydrogen-track-positions --redirect-code-traces --redirect-code-traces-to=code.asm --trace_hydrogen_file=code.cfg --print-opt-code bin/parse --stat -o /dev/null"
	};
	var dependencies = {
		"mdn-data": "2.0.14",
		"source-map": "^0.6.1"
	};
	var devDependencies = {
		"@rollup/plugin-commonjs": "^11.0.2",
		"@rollup/plugin-json": "^4.0.2",
		"@rollup/plugin-node-resolve": "^7.1.1",
		coveralls: "^3.0.9",
		eslint: "^6.8.0",
		"json-to-ast": "^2.1.0",
		mocha: "^6.2.3",
		nyc: "^14.1.1",
		rollup: "^1.32.1",
		"rollup-plugin-terser": "^5.3.0"
	};
	var engines = {
		node: ">=8.0.0"
	};
	var files = [
		"data",
		"dist",
		"lib"
	];
	var require$$4 = {
		name: name,
		version: version,
		description: description,
		author: author,
		license: license,
		repository: repository,
		keywords: keywords,
		main: main,
		unpkg: unpkg,
		jsdelivr: jsdelivr,
		scripts: scripts,
		dependencies: dependencies,
		devDependencies: devDependencies,
		engines: engines,
		files: files
	};

	function merge() {
	    var dest = {};

	    for (var i = 0; i < arguments.length; i++) {
	        var src = arguments[i];
	        for (var key in src) {
	            dest[key] = src[key];
	        }
	    }

	    return dest;
	}

	syntax.exports = create$4.create(
	    merge(
	        lexer,
	        parser,
	        walker
	    )
	);
	syntax.exports.version = require$$4.version;

	var syntaxExports = syntax.exports;

	var lib = syntaxExports;

	var csstree = /*@__PURE__*/getDefaultExportFromCjs(lib);

	class Sheet {
		constructor(url, hooks) {

			if (hooks) {
				this.hooks = hooks;
			} else {
				this.hooks = {};
				this.hooks.onUrl = new Hook(this);
				this.hooks.onAtPage = new Hook(this);
				this.hooks.onAtMedia = new Hook(this);
				this.hooks.onRule = new Hook(this);
				this.hooks.onDeclaration = new Hook(this);
				this.hooks.onSelector = new Hook(this);
				this.hooks.onPseudoSelector = new Hook(this);

				this.hooks.onContent = new Hook(this);
				this.hooks.onImport = new Hook(this);

				this.hooks.beforeTreeParse = new Hook(this);
				this.hooks.beforeTreeWalk = new Hook(this);
				this.hooks.afterTreeWalk = new Hook(this);
			}

			try {
				this.url = new URL(url, window.location.href);
			} catch (e) {
				this.url = new URL(window.location.href);
			}
		}



		// parse
		async parse(text) {
			this.text = text;

			await this.hooks.beforeTreeParse.trigger(this.text, this);

			// send to csstree
			this.ast = csstree.parse(this._text);

			await this.hooks.beforeTreeWalk.trigger(this.ast);

			// Replace urls
			this.replaceUrls(this.ast);

			// Scope
			this.id = UUID();
			// this.addScope(this.ast, this.uuid);

			// Replace IDs with data-id
			this.replaceIds(this.ast);

			this.imported = [];

			// Trigger Hooks
			this.urls(this.ast);
			this.rules(this.ast);
			this.atrules(this.ast);

			await this.hooks.afterTreeWalk.trigger(this.ast, this);

			// return ast
			return this.ast;
		}



		insertRule(rule) {
			let inserted = this.ast.children.appendData(rule);

			this.declarations(rule);

			return inserted;
		}

		urls(ast) {
			csstree.walk(ast, {
				visit: "Url",
				enter: (node, item, list) => {
					this.hooks.onUrl.trigger(node, item, list);
				}
			});
		}

		atrules(ast) {
			csstree.walk(ast, {
				visit: "Atrule",
				enter: (node, item, list) => {
					const basename = csstree.keyword(node.name).basename;

					if (basename === "page") {
						this.hooks.onAtPage.trigger(node, item, list);
						this.declarations(node, item, list);
					}

					if (basename === "media") {
						this.hooks.onAtMedia.trigger(node, item, list);
						this.declarations(node, item, list);
					}

					if (basename === "import") {
						this.hooks.onImport.trigger(node, item, list);
						this.imports(node, item, list);
					}
				}
			});
		}


		rules(ast) {
			csstree.walk(ast, {
				visit: "Rule",
				enter: (ruleNode, ruleItem, rulelist) => {

					this.hooks.onRule.trigger(ruleNode, ruleItem, rulelist);
					this.declarations(ruleNode, ruleItem, rulelist);
					this.onSelector(ruleNode, ruleItem, rulelist);

				}
			});
		}

		declarations(ruleNode, ruleItem, rulelist) {
			csstree.walk(ruleNode, {
				visit: "Declaration",
				enter: (declarationNode, dItem, dList) => {

					this.hooks.onDeclaration.trigger(declarationNode, dItem, dList, {ruleNode, ruleItem, rulelist});

					if (declarationNode.property === "content") {
						csstree.walk(declarationNode, {
							visit: "Function",
							enter: (funcNode, fItem, fList) => {
								this.hooks.onContent.trigger(funcNode, fItem, fList, {declarationNode, dItem, dList}, {ruleNode, ruleItem, rulelist});
							}
						});
					}

				}
			});
		}

		// add pseudo elements to parser
		onSelector(ruleNode, ruleItem, rulelist) {
			csstree.walk(ruleNode, {
				visit: "Selector",
				enter: (selectNode, selectItem, selectList) => {
					this.hooks.onSelector.trigger(selectNode, selectItem, selectList, {ruleNode, ruleItem, rulelist});

					if (selectNode.children.forEach(node => {if (node.type === "PseudoElementSelector") {
						csstree.walk(node, {
							visit: "PseudoElementSelector",
							enter: (pseudoNode, pItem, pList) => {
								this.hooks.onPseudoSelector.trigger(pseudoNode, pItem, pList, {selectNode, selectItem, selectList}, {ruleNode, ruleItem, rulelist});
							}
						});
					}}));
				}
			});
		}

		replaceUrls(ast) {
			csstree.walk(ast, {
				visit: "Url",
				enter: (node, item, list) => {
					let content = node.value.value;
					if ((node.value.type === "Raw" && content.startsWith("data:")) || (node.value.type === "String" && (content.startsWith("\"data:") || content.startsWith("'data:")))) ; else {
						let href = content.replace(/["']/g, "");
						let url = new URL(href, this.url);
						node.value.value = url.toString();
					}
				}
			});
		}

		addScope(ast, id) {
			// Get all selector lists
			// add an id
			csstree.walk(ast, {
				visit: "Selector",
				enter: (node, item, list) => {
					let children = node.children;
					children.prepend(children.createItem({
						type: "WhiteSpace",
						value: " "
					}));
					children.prepend(children.createItem({
						type: "IdSelector",
						name: id,
						loc: null,
						children: null
					}));
				}
			});
		}

		getNamedPageSelectors(ast) {
			let namedPageSelectors = {};
			csstree.walk(ast, {
				visit: "Rule",
				enter: (node, item, list) => {
					csstree.walk(node, {
						visit: "Declaration",
						enter: (declaration, dItem, dList) => {
							if (declaration.property === "page") {
								let value = declaration.value.children.first();
								let name = value.name;
								let selector = csstree.generate(node.prelude);
								namedPageSelectors[name] = {
									name: name,
									selector: selector
								};

								// dList.remove(dItem);

								// Add in page break
								declaration.property = "break-before";
								value.type = "Identifier";
								value.name = "always";

							}
						}
					});
				}
			});
			return namedPageSelectors;
		}

		replaceIds(ast) {
			csstree.walk(ast, {
				visit: "Rule",
				enter: (node, item, list) => {

					csstree.walk(node, {
						visit: "IdSelector",
						enter: (idNode, idItem, idList) => {
							let name = idNode.name;
							idNode.flags = null;
							idNode.matcher = "=";
							idNode.name = {type: "Identifier", loc: null, name: "data-id"};
							idNode.type = "AttributeSelector";
							idNode.value = {type: "String", loc: null, value: `"${name}"`};
						}
					});
				}
			});
		}

		imports(node, item, list) {
			// console.log("import", node, item, list);
			let queries = [];
			csstree.walk(node, {
				visit: "MediaQuery",
				enter: (mqNode, mqItem, mqList) => {
					csstree.walk(mqNode, {
						visit: "Identifier",
						enter: (identNode, identItem, identList) => {
							queries.push(identNode.name);
						}
					});
				}
			});

			// Just basic media query support for now
			let shouldNotApply = queries.some((query, index) => {
				let q = query;
				if (q === "not") {
					q = queries[index + 1];
					return !(q === "screen" || q === "speech");
				} else {
					return (q === "screen" || q === "speech");
				}
			});

			if (shouldNotApply) {
				return;
			}

			csstree.walk(node, {
				visit: "String",
				enter: (urlNode, urlItem, urlList) => {
					let href = urlNode.value.replace(/["']/g, "");
					let url = new URL(href, this.url);
					let value = url.toString();

					this.imported.push(value);

					// Remove the original
					list.remove(item);
				}
			});
		}

		set text(t) {
			this._text = t;
		}

		get text() {
			return this._text;
		}

		// generate string
		toString(ast) {
			return csstree.generate(ast || this.ast);
		}
	}

	var baseStyles = `
:root {
	--pagedjs-width: 8.5in;
	--pagedjs-height: 11in;
	--pagedjs-width-right: 8.5in;
	--pagedjs-height-right: 11in;
	--pagedjs-width-left: 8.5in;
	--pagedjs-height-left: 11in;
	--pagedjs-pagebox-width: 8.5in;
	--pagedjs-pagebox-height: 11in;
	--pagedjs-footnotes-height: 0mm;
	--pagedjs-margin-top: 1in;
	--pagedjs-margin-right: 1in;
	--pagedjs-margin-bottom: 1in;
	--pagedjs-margin-left: 1in;
	--pagedjs-padding-top: 0mm;
	--pagedjs-padding-right: 0mm;
	--pagedjs-padding-bottom: 0mm;
	--pagedjs-padding-left: 0mm;
	--pagedjs-border-top: 0mm;
	--pagedjs-border-right: 0mm;
	--pagedjs-border-bottom: 0mm;
	--pagedjs-border-left: 0mm;
	--pagedjs-bleed-top: 0mm;
	--pagedjs-bleed-right: 0mm;
	--pagedjs-bleed-bottom: 0mm;
	--pagedjs-bleed-left: 0mm;
	--pagedjs-bleed-right-top: 0mm;
	--pagedjs-bleed-right-right: 0mm;
	--pagedjs-bleed-right-bottom: 0mm;
	--pagedjs-bleed-right-left: 0mm;
	--pagedjs-bleed-left-top: 0mm;
	--pagedjs-bleed-left-right: 0mm;
	--pagedjs-bleed-left-bottom: 0mm;
	--pagedjs-bleed-left-left: 0mm;
	--pagedjs-crop-color: black;
	--pagedjs-crop-shadow: white;
	--pagedjs-crop-offset: 2mm;
	--pagedjs-crop-stroke: 1px;
	--pagedjs-cross-size: 5mm;
	--pagedjs-mark-cross-display: none;
	--pagedjs-mark-crop-display: none;
	--pagedjs-page-count: 0;
	--pagedjs-page-counter-increment: 1;
	--pagedjs-footnotes-count: 0;
	--pagedjs-column-gap-offset: 1000px;
}

@page {
	size: letter;
	margin: 0;
}

.pagedjs_sheet {
	box-sizing: border-box;
	width: var(--pagedjs-width);
	height: var(--pagedjs-height);
	overflow: hidden;
	position: relative;
	display: grid;
	grid-template-columns: [bleed-left] var(--pagedjs-bleed-left) [sheet-center] calc(var(--pagedjs-width) - var(--pagedjs-bleed-left) - var(--pagedjs-bleed-right)) [bleed-right] var(--pagedjs-bleed-right);
	grid-template-rows: [bleed-top] var(--pagedjs-bleed-top) [sheet-middle] calc(var(--pagedjs-height) - var(--pagedjs-bleed-top) - var(--pagedjs-bleed-bottom)) [bleed-bottom] var(--pagedjs-bleed-bottom);
}

.pagedjs_right_page .pagedjs_sheet {
	width: var(--pagedjs-width-right);
	height: var(--pagedjs-height-right);
	grid-template-columns: [bleed-left] var(--pagedjs-bleed-right-left) [sheet-center] calc(var(--pagedjs-width) - var(--pagedjs-bleed-right-left) - var(--pagedjs-bleed-right-right)) [bleed-right] var(--pagedjs-bleed-right-right);
	grid-template-rows: [bleed-top] var(--pagedjs-bleed-right-top) [sheet-middle] calc(var(--pagedjs-height) - var(--pagedjs-bleed-right-top) - var(--pagedjs-bleed-right-bottom)) [bleed-bottom] var(--pagedjs-bleed-right-bottom);
}

.pagedjs_left_page .pagedjs_sheet {
	width: var(--pagedjs-width-left);
	height: var(--pagedjs-height-left);
	grid-template-columns: [bleed-left] var(--pagedjs-bleed-left-left) [sheet-center] calc(var(--pagedjs-width) - var(--pagedjs-bleed-left-left) - var(--pagedjs-bleed-left-right)) [bleed-right] var(--pagedjs-bleed-left-right);
	grid-template-rows: [bleed-top] var(--pagedjs-bleed-left-top) [sheet-middle] calc(var(--pagedjs-height) - var(--pagedjs-bleed-left-top) - var(--pagedjs-bleed-left-bottom)) [bleed-bottom] var(--pagedjs-bleed-left-bottom);
}

.pagedjs_bleed {
	display: flex;
	align-items: center;
	justify-content: center;
	flex-wrap: nowrap;
	overflow: hidden;
}

.pagedjs_bleed-top {
	grid-column: bleed-left / -1;
	grid-row: bleed-top;
	flex-direction: row;
}

.pagedjs_bleed-bottom {
	grid-column: bleed-left / -1;
	grid-row: bleed-bottom;
	flex-direction: row;
}

.pagedjs_bleed-left {
	grid-column: bleed-left;
	grid-row: bleed-top / -1;
	flex-direction: column;
}

.pagedjs_bleed-right {
	grid-column: bleed-right;
	grid-row: bleed-top / -1;
	flex-direction: column;
}

.pagedjs_marks-crop {
	display: var(--pagedjs-mark-crop-display);
	flex-grow: 0;
	flex-shrink: 0;
	z-index: 9999999999;
}

.pagedjs_bleed-top .pagedjs_marks-crop:nth-child(1),
.pagedjs_bleed-bottom .pagedjs_marks-crop:nth-child(1) {
	width: calc(var(--pagedjs-bleed-left) - var(--pagedjs-crop-stroke));
	border-right: var(--pagedjs-crop-stroke) solid var(--pagedjs-crop-color);
	box-shadow: 1px 0px 0px 0px var(--pagedjs-crop-shadow);
}

.pagedjs_right_page .pagedjs_bleed-top .pagedjs_marks-crop:nth-child(1),
.pagedjs_right_page .pagedjs_bleed-bottom .pagedjs_marks-crop:nth-child(1) {
	width: calc(var(--pagedjs-bleed-right-left) - var(--pagedjs-crop-stroke));
}

.pagedjs_left_page .pagedjs_bleed-top .pagedjs_marks-crop:nth-child(1),
.pagedjs_left_page .pagedjs_bleed-bottom .pagedjs_marks-crop:nth-child(1) {
	width: calc(var(--pagedjs-bleed-left-left) - var(--pagedjs-crop-stroke));
}

.pagedjs_bleed-top .pagedjs_marks-crop:nth-child(3),
.pagedjs_bleed-bottom .pagedjs_marks-crop:nth-child(3) {
	width: calc(var(--pagedjs-bleed-right) - var(--pagedjs-crop-stroke));
	border-left: var(--pagedjs-crop-stroke) solid var(--pagedjs-crop-color);
	box-shadow: -1px 0px 0px 0px var(--pagedjs-crop-shadow);
}

.pagedjs_right_page .pagedjs_bleed-top .pagedjs_marks-crop:nth-child(3),
.pagedjs_right_page .pagedjs_bleed-bottom .pagedjs_marks-crop:nth-child(3) {
	width: calc(var(--pagedjs-bleed-right-right) - var(--pagedjs-crop-stroke));
}

.pagedjs_left_page .pagedjs_bleed-top .pagedjs_marks-crop:nth-child(3),
.pagedjs_left_page .pagedjs_bleed-bottom .pagedjs_marks-crop:nth-child(3) {
	width: calc(var(--pagedjs-bleed-left-right) - var(--pagedjs-crop-stroke));
}

.pagedjs_bleed-top .pagedjs_marks-crop {
	align-self: flex-start;
	height: calc(var(--pagedjs-bleed-top) - var(--pagedjs-crop-offset));
}

.pagedjs_right_page .pagedjs_bleed-top .pagedjs_marks-crop {
	height: calc(var(--pagedjs-bleed-right-top) - var(--pagedjs-crop-offset));
}

.pagedjs_left_page .pagedjs_bleed-top .pagedjs_marks-crop {
	height: calc(var(--pagedjs-bleed-left-top) - var(--pagedjs-crop-offset));
}

.pagedjs_bleed-bottom .pagedjs_marks-crop {
	align-self: flex-end;
	height: calc(var(--pagedjs-bleed-bottom) - var(--pagedjs-crop-offset));
}

.pagedjs_right_page .pagedjs_bleed-bottom .pagedjs_marks-crop {
	height: calc(var(--pagedjs-bleed-right-bottom) - var(--pagedjs-crop-offset));
}

.pagedjs_left_page .pagedjs_bleed-bottom .pagedjs_marks-crop {
	height: calc(var(--pagedjs-bleed-left-bottom) - var(--pagedjs-crop-offset));
}

.pagedjs_bleed-left .pagedjs_marks-crop:nth-child(1),
.pagedjs_bleed-right .pagedjs_marks-crop:nth-child(1) {
	height: calc(var(--pagedjs-bleed-top) - var(--pagedjs-crop-stroke));
	border-bottom: var(--pagedjs-crop-stroke) solid var(--pagedjs-crop-color);
	box-shadow: 0px 1px 0px 0px var(--pagedjs-crop-shadow);
}

.pagedjs_right_page .pagedjs_bleed-left .pagedjs_marks-crop:nth-child(1),
.pagedjs_right_page .pagedjs_bleed-right .pagedjs_marks-crop:nth-child(1) {
	height: calc(var(--pagedjs-bleed-right-top) - var(--pagedjs-crop-stroke));
}

.pagedjs_left_page .pagedjs_bleed-left .pagedjs_marks-crop:nth-child(1),
.pagedjs_left_page .pagedjs_bleed-right .pagedjs_marks-crop:nth-child(1) {
	height: calc(var(--pagedjs-bleed-left-top) - var(--pagedjs-crop-stroke));
}

.pagedjs_bleed-left .pagedjs_marks-crop:nth-child(3),
.pagedjs_bleed-right .pagedjs_marks-crop:nth-child(3) {
	height: calc(var(--pagedjs-bleed-bottom) - var(--pagedjs-crop-stroke));
	border-top: var(--pagedjs-crop-stroke) solid var(--pagedjs-crop-color);
	box-shadow: 0px -1px 0px 0px var(--pagedjs-crop-shadow);
}

.pagedjs_right_page .pagedjs_bleed-left .pagedjs_marks-crop:nth-child(3),
.pagedjs_right_page .pagedjs_bleed-right .pagedjs_marks-crop:nth-child(3) {
	height: calc(var(--pagedjs-bleed-right-bottom) - var(--pagedjs-crop-stroke));
}

.pagedjs_left_page .pagedjs_bleed-left .pagedjs_marks-crop:nth-child(3),
.pagedjs_left_page .pagedjs_bleed-right .pagedjs_marks-crop:nth-child(3) {
	height: calc(var(--pagedjs-bleed-left-bottom) - var(--pagedjs-crop-stroke));
}

.pagedjs_bleed-left .pagedjs_marks-crop {
	width: calc(var(--pagedjs-bleed-left) - var(--pagedjs-crop-offset));
	align-self: flex-start;
}

.pagedjs_right_page .pagedjs_bleed-left .pagedjs_marks-crop {
	width: calc(var(--pagedjs-bleed-right-left) - var(--pagedjs-crop-offset));
}

.pagedjs_left_page .pagedjs_bleed-left .pagedjs_marks-crop {
	width: calc(var(--pagedjs-bleed-left-left) - var(--pagedjs-crop-offset));
}

.pagedjs_bleed-right .pagedjs_marks-crop {
	width: calc(var(--pagedjs-bleed-right) - var(--pagedjs-crop-offset));
	align-self: flex-end;
}

.pagedjs_right_page .pagedjs_bleed-right .pagedjs_marks-crop {
	width: calc(var(--pagedjs-bleed-right-right) - var(--pagedjs-crop-offset));
}

.pagedjs_left_page .pagedjs_bleed-right .pagedjs_marks-crop {
	width: calc(var(--pagedjs-bleed-left-right) - var(--pagedjs-crop-offset));
}

.pagedjs_marks-middle {
	display: flex;
	flex-grow: 1;
	flex-shrink: 0;
	align-items: center;
	justify-content: center;
}

.pagedjs_marks-cross {
	display: var(--pagedjs-mark-cross-display);
	background-image: url(data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0idXRmLTgiPz48IURPQ1RZUEUgc3ZnIFBVQkxJQyAiLS8vVzNDLy9EVEQgU1ZHIDEuMS8vRU4iICJodHRwOi8vd3d3LnczLm9yZy9HcmFwaGljcy9TVkcvMS4xL0RURC9zdmcxMS5kdGQiPjxzdmcgdmVyc2lvbj0iMS4xIiBpZD0iTGF5ZXJfMSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayIgeD0iMHB4IiB5PSIwcHgiIHdpZHRoPSIzMi41MzdweCIgaGVpZ2h0PSIzMi41MzdweCIgdmlld0JveD0iMC4xMDQgMC4xMDQgMzIuNTM3IDMyLjUzNyIgZW5hYmxlLWJhY2tncm91bmQ9Im5ldyAwLjEwNCAwLjEwNCAzMi41MzcgMzIuNTM3IiB4bWw6c3BhY2U9InByZXNlcnZlIj48cGF0aCBmaWxsPSJub25lIiBzdHJva2U9IiNGRkZGRkYiIHN0cm9rZS13aWR0aD0iMy4zODkzIiBzdHJva2UtbWl0ZXJsaW1pdD0iMTAiIGQ9Ik0yOS45MzEsMTYuMzczYzAsNy40ODktNi4wNjgsMTMuNTYtMTMuNTU4LDEzLjU2Yy03LjQ4MywwLTEzLjU1Ny02LjA3Mi0xMy41NTctMTMuNTZjMC03LjQ4Niw2LjA3NC0xMy41NTQsMTMuNTU3LTEzLjU1NEMyMy44NjIsMi44MTksMjkuOTMxLDguODg3LDI5LjkzMSwxNi4zNzN6Ii8+PGxpbmUgZmlsbD0ibm9uZSIgc3Ryb2tlPSIjRkZGRkZGIiBzdHJva2Utd2lkdGg9IjMuMzg5MyIgc3Ryb2tlLW1pdGVybGltaXQ9IjEwIiB4MT0iMC4xMDQiIHkxPSIxNi4zNzMiIHgyPSIzMi42NDIiIHkyPSIxNi4zNzMiLz48bGluZSBmaWxsPSJub25lIiBzdHJva2U9IiNGRkZGRkYiIHN0cm9rZS13aWR0aD0iMy4zODkzIiBzdHJva2UtbWl0ZXJsaW1pdD0iMTAiIHgxPSIxNi4zNzMiIHkxPSIwLjEwNCIgeDI9IjE2LjM3MyIgeTI9IjMyLjY0MiIvPjxwYXRoIGZpbGw9Im5vbmUiIHN0cm9rZT0iI0ZGRkZGRiIgc3Ryb2tlLXdpZHRoPSIzLjM4OTMiIHN0cm9rZS1taXRlcmxpbWl0PSIxMCIgZD0iTTI0LjUwOCwxNi4zNzNjMCw0LjQ5Ni0zLjYzOCw4LjEzNS04LjEzNSw4LjEzNWMtNC40OTEsMC04LjEzNS0zLjYzOC04LjEzNS04LjEzNWMwLTQuNDg5LDMuNjQ0LTguMTM1LDguMTM1LTguMTM1QzIwLjg2OSw4LjIzOSwyNC41MDgsMTEuODg0LDI0LjUwOCwxNi4zNzN6Ii8+PHBhdGggZmlsbD0ibm9uZSIgc3Ryb2tlPSIjMDAwMDAwIiBzdHJva2Utd2lkdGg9IjAuNjc3OCIgc3Ryb2tlLW1pdGVybGltaXQ9IjEwIiBkPSJNMjkuOTMxLDE2LjM3M2MwLDcuNDg5LTYuMDY4LDEzLjU2LTEzLjU1OCwxMy41NmMtNy40ODMsMC0xMy41NTctNi4wNzItMTMuNTU3LTEzLjU2YzAtNy40ODYsNi4wNzQtMTMuNTU0LDEzLjU1Ny0xMy41NTRDMjMuODYyLDIuODE5LDI5LjkzMSw4Ljg4NywyOS45MzEsMTYuMzczeiIvPjxsaW5lIGZpbGw9Im5vbmUiIHN0cm9rZT0iIzAwMDAwMCIgc3Ryb2tlLXdpZHRoPSIwLjY3NzgiIHN0cm9rZS1taXRlcmxpbWl0PSIxMCIgeDE9IjAuMTA0IiB5MT0iMTYuMzczIiB4Mj0iMzIuNjQyIiB5Mj0iMTYuMzczIi8+PGxpbmUgZmlsbD0ibm9uZSIgc3Ryb2tlPSIjMDAwMDAwIiBzdHJva2Utd2lkdGg9IjAuNjc3OCIgc3Ryb2tlLW1pdGVybGltaXQ9IjEwIiB4MT0iMTYuMzczIiB5MT0iMC4xMDQiIHgyPSIxNi4zNzMiIHkyPSIzMi42NDIiLz48cGF0aCBkPSJNMjQuNTA4LDE2LjM3M2MwLDQuNDk2LTMuNjM4LDguMTM1LTguMTM1LDguMTM1Yy00LjQ5MSwwLTguMTM1LTMuNjM4LTguMTM1LTguMTM1YzAtNC40ODksMy42NDQtOC4xMzUsOC4xMzUtOC4xMzVDMjAuODY5LDguMjM5LDI0LjUwOCwxMS44ODQsMjQuNTA4LDE2LjM3MyIvPjxsaW5lIGZpbGw9Im5vbmUiIHN0cm9rZT0iI0ZGRkZGRiIgc3Ryb2tlLXdpZHRoPSIwLjY3NzgiIHN0cm9rZS1taXRlcmxpbWl0PSIxMCIgeDE9IjguMjM5IiB5MT0iMTYuMzczIiB4Mj0iMjQuNTA4IiB5Mj0iMTYuMzczIi8+PGxpbmUgZmlsbD0ibm9uZSIgc3Ryb2tlPSIjRkZGRkZGIiBzdHJva2Utd2lkdGg9IjAuNjc3OCIgc3Ryb2tlLW1pdGVybGltaXQ9IjEwIiB4MT0iMTYuMzczIiB5MT0iOC4yMzkiIHgyPSIxNi4zNzMiIHkyPSIyNC41MDgiLz48L3N2Zz4=);
  background-repeat: no-repeat;
  background-position: 50% 50%;
  background-size: var(--pagedjs-cross-size);

  z-index: 2147483647;
	width: var(--pagedjs-cross-size);
	height: var(--pagedjs-cross-size);
}

.pagedjs_pagebox {
	box-sizing: border-box;
	width: var(--pagedjs-pagebox-width);
	height: var(--pagedjs-pagebox-height);
	position: relative;
	display: grid;
	grid-template-columns: [left] var(--pagedjs-margin-left) [center] calc(var(--pagedjs-pagebox-width) - var(--pagedjs-margin-left) - var(--pagedjs-margin-right)) [right] var(--pagedjs-margin-right);
	grid-template-rows: [header] var(--pagedjs-margin-top) [page] calc(var(--pagedjs-pagebox-height) - var(--pagedjs-margin-top) - var(--pagedjs-margin-bottom)) [footer] var(--pagedjs-margin-bottom);
	grid-column: sheet-center;
	grid-row: sheet-middle;
}

.pagedjs_pagebox * {
	box-sizing: border-box;
}

.pagedjs_margin-top {
	width: calc(var(--pagedjs-pagebox-width) - var(--pagedjs-margin-left) - var(--pagedjs-margin-right));
	height: var(--pagedjs-margin-top);
	grid-column: center;
	grid-row: header;
	flex-wrap: nowrap;
	display: grid;
	grid-template-columns: repeat(3, 1fr);
	grid-template-rows: 100%;
}

.pagedjs_margin-top-left-corner-holder {
	width: var(--pagedjs-margin-left);
	height: var(--pagedjs-margin-top);
	display: flex;
	grid-column: left;
	grid-row: header;
}

.pagedjs_margin-top-right-corner-holder {
	width: var(--pagedjs-margin-right);
	height: var(--pagedjs-margin-top);
	display: flex;
	grid-column: right;
	grid-row: header;
}

.pagedjs_margin-top-left-corner {
	width: var(--pagedjs-margin-left);
}

.pagedjs_margin-top-right-corner {
	width: var(--pagedjs-margin-right);
}

.pagedjs_margin-right {
	height: calc(var(--pagedjs-pagebox-height) - var(--pagedjs-margin-top) - var(--pagedjs-margin-bottom));
	width: var(--pagedjs-margin-right);
	right: 0;
	grid-column: right;
	grid-row: page;
	display: grid;
	grid-template-rows: repeat(3, 33.3333%);
	grid-template-columns: 100%;
}

.pagedjs_margin-bottom {
	width: calc(var(--pagedjs-pagebox-width) - var(--pagedjs-margin-left) - var(--pagedjs-margin-right));
	height: var(--pagedjs-margin-bottom);
	grid-column: center;
	grid-row: footer;
	display: grid;
	grid-template-columns: repeat(3, 1fr);
	grid-template-rows: 100%;
}

.pagedjs_margin-bottom-left-corner-holder {
	width: var(--pagedjs-margin-left);
	height: var(--pagedjs-margin-bottom);
	display: flex;
	grid-column: left;
	grid-row: footer;
}

.pagedjs_margin-bottom-right-corner-holder {
	width: var(--pagedjs-margin-right);
	height: var(--pagedjs-margin-bottom);
	display: flex;
	grid-column: right;
	grid-row: footer;
}

.pagedjs_margin-bottom-left-corner {
	width: var(--pagedjs-margin-left);
}

.pagedjs_margin-bottom-right-corner {
	width: var(--pagedjs-margin-right);
}



.pagedjs_margin-left {
	height: calc(var(--pagedjs-pagebox-height) - var(--pagedjs-margin-top) - var(--pagedjs-margin-bottom));
	width: var(--pagedjs-margin-left);
	grid-column: left;
	grid-row: page;
	display: grid;
	grid-template-rows: repeat(3, 33.33333%);
	grid-template-columns: 100%;
}

.pagedjs_pages .pagedjs_pagebox .pagedjs_margin:not(.hasContent) {
	visibility: hidden;
}

.pagedjs_pagebox > .pagedjs_area {
	grid-column: center;
	grid-row: page;
	width: 100%;
	height: 100%;
	padding: var(--pagedjs-padding-top) var(--pagedjs-padding-right) var(--pagedjs-padding-bottom) var(--pagedjs-padding-left);
	border-top: var(--pagedjs-border-top);
	border-right: var(--pagedjs-border-right);
	border-bottom: var(--pagedjs-border-bottom);
	border-left: var(--pagedjs-border-left);
}

.pagedjs_pagebox > .pagedjs_area > .pagedjs_page_content {
	width: 100%;
	height: calc(100% - var(--pagedjs-footnotes-height));
	position: relative;
	column-fill: auto;
}

.pagedjs_pagebox > .pagedjs_area > .pagedjs_page_content > div {
	height: inherit;
}

.pagedjs_pagebox > .pagedjs_area > .pagedjs_footnote_area {
	position: relative;
	overflow: hidden;
	height: var(--pagedjs-footnotes-height);
	display: flex;
    justify-content: flex-end;
    flex-flow: column;
}

.pagedjs_pagebox > .pagedjs_area > .pagedjs_footnote_area > .pagedjs_footnote_content {
	overflow: hidden;
}

.pagedjs_pagebox > .pagedjs_area > .pagedjs_footnote_area > .pagedjs_footnote_inner_content {
	overflow: hidden;
}

.pagedjs_area [data-footnote-call] {
	all: unset;
	counter-increment: footnote;
}

.pagedjs_area [data-split-from] {
	counter-increment: unset;
	counter-reset: unset;
}

[data-footnote-call]::after {
	vertical-align: super;
	font-size: 65%;
	line-height: normal;
	content: counter(footnote);
}

@supports ( font-variant-position: super ) {
	[data-footnote-call]::after {
		vertical-align: baseline;
		font-size: 100%;
		line-height: inherit;
		font-variant-position: super;
	}
}

.pagedjs_footnote_empty {
	display: none;
}

.pagedjs_area [data-split-from] {
	counter-increment: unset;
	counter-reset: unset;
}

[data-footnote-marker] {
	text-indent: 0;
	display: list-item;
	list-style-position: inside;
}

[data-footnote-marker][data-split-from] {
	list-style: none;
}

[data-footnote-marker]:not([data-split-from]) {
	counter-increment: footnote-marker;
}

[data-footnote-marker]::marker {
	content: counter(footnote-marker) ". ";
}

[data-footnote-marker][data-split-from]::marker {
	content: unset;
}

.pagedjs_area .pagedjs_footnote_inner_content [data-note-display="inline"] {
 	display: inline;
}

.pagedjs_page {
	counter-increment: page var(--pagedjs-page-counter-increment);
	width: var(--pagedjs-width);
	height: var(--pagedjs-height);
}

.pagedjs_page.pagedjs_right_page {
	width: var(--pagedjs-width-right);
	height: var(--pagedjs-height-right);
}

.pagedjs_page.pagedjs_left_page {
	width: var(--pagedjs-width-left);
	height: var(--pagedjs-height-left);
}

.pagedjs_pages {
	counter-reset: pages var(--pagedjs-page-count) footnote var(--pagedjs-footnotes-count) footnote-marker var(--pagedjs-footnotes-count);
}

.pagedjs_pagebox .pagedjs_margin-top-left-corner,
.pagedjs_pagebox .pagedjs_margin-top-right-corner,
.pagedjs_pagebox .pagedjs_margin-bottom-left-corner,
.pagedjs_pagebox .pagedjs_margin-bottom-right-corner,
.pagedjs_pagebox .pagedjs_margin-top-left,
.pagedjs_pagebox .pagedjs_margin-top-right,
.pagedjs_pagebox .pagedjs_margin-bottom-left,
.pagedjs_pagebox .pagedjs_margin-bottom-right,
.pagedjs_pagebox .pagedjs_margin-top-center,
.pagedjs_pagebox .pagedjs_margin-bottom-center,
.pagedjs_pagebox .pagedjs_margin-top-center,
.pagedjs_pagebox .pagedjs_margin-bottom-center,
.pagedjs_margin-right-middle,
.pagedjs_margin-left-middle  {
	display: flex;
	align-items: center;
}

.pagedjs_margin-right-top,
.pagedjs_margin-left-top  {
	display: flex;
	align-items: flex-top;
}


.pagedjs_margin-right-bottom,
.pagedjs_margin-left-bottom  {
	display: flex;
	align-items: flex-end;
}



/*
.pagedjs_pagebox .pagedjs_margin-top-center,
.pagedjs_pagebox .pagedjs_margin-bottom-center {
	height: 100%;
	display: none;
	align-items: center;
	flex: 1 0 33%;
	margin: 0 auto;
}

.pagedjs_pagebox .pagedjs_margin-top-left-corner,
.pagedjs_pagebox .pagedjs_margin-top-right-corner,
.pagedjs_pagebox .pagedjs_margin-bottom-right-corner,
.pagedjs_pagebox .pagedjs_margin-bottom-left-corner {
	display: none;
	align-items: center;
}

.pagedjs_pagebox .pagedjs_margin-left-top,
.pagedjs_pagebox .pagedjs_margin-right-top {
	display: none;
	align-items: flex-start;
}

.pagedjs_pagebox .pagedjs_margin-right-middle,
.pagedjs_pagebox .pagedjs_margin-left-middle {
	display: none;
	align-items: center;
}

.pagedjs_pagebox .pagedjs_margin-left-bottom,
.pagedjs_pagebox .pagedjs_margin-right-bottom {
	display: none;
	align-items: flex-end;
}
*/

.pagedjs_pagebox .pagedjs_margin-top-left,
.pagedjs_pagebox .pagedjs_margin-top-right-corner,
.pagedjs_pagebox .pagedjs_margin-bottom-left,
.pagedjs_pagebox .pagedjs_margin-bottom-right-corner { text-align: left; }

.pagedjs_pagebox .pagedjs_margin-top-left-corner,
.pagedjs_pagebox .pagedjs_margin-top-right,
.pagedjs_pagebox .pagedjs_margin-bottom-left-corner,
.pagedjs_pagebox .pagedjs_margin-bottom-right { text-align: right; }

.pagedjs_pagebox .pagedjs_margin-top-center,
.pagedjs_pagebox .pagedjs_margin-bottom-center,
.pagedjs_pagebox .pagedjs_margin-left-top,
.pagedjs_pagebox .pagedjs_margin-left-middle,
.pagedjs_pagebox .pagedjs_margin-left-bottom,
.pagedjs_pagebox .pagedjs_margin-right-top,
.pagedjs_pagebox .pagedjs_margin-right-middle,
.pagedjs_pagebox .pagedjs_margin-right-bottom { text-align: center; }

.pagedjs_pages .pagedjs_margin .pagedjs_margin-content {
	width: 100%;
}

.pagedjs_pages .pagedjs_margin-left .pagedjs_margin-content::after,
.pagedjs_pages .pagedjs_margin-top .pagedjs_margin-content::after,
.pagedjs_pages .pagedjs_margin-right .pagedjs_margin-content::after,
.pagedjs_pages .pagedjs_margin-bottom .pagedjs_margin-content::after {
	display: block;
}

.pagedjs_pages > .pagedjs_page > .pagedjs_sheet > .pagedjs_pagebox > .pagedjs_area > div [data-split-to] {
	margin-bottom: unset;
	padding-bottom: unset;
}

.pagedjs_pages > .pagedjs_page > .pagedjs_sheet > .pagedjs_pagebox > .pagedjs_area > div [data-split-from] {
	text-indent: unset;
	margin-top: unset;
	padding-top: unset;
	initial-letter: unset;
}

.pagedjs_pages > .pagedjs_page > .pagedjs_sheet > .pagedjs_pagebox > .pagedjs_area > div [data-split-from] > *::first-letter,
.pagedjs_pages > .pagedjs_page > .pagedjs_sheet > .pagedjs_pagebox > .pagedjs_area > div [data-split-from]::first-letter {
	color: unset;
	font-size: unset;
	font-weight: unset;
	font-family: unset;
	color: unset;
	line-height: unset;
	float: unset;
	padding: unset;
	margin: unset;
}

.pagedjs_pages > .pagedjs_page > .pagedjs_sheet > .pagedjs_pagebox > .pagedjs_area > div [data-split-to]:not([data-footnote-call]):after,
.pagedjs_pages > .pagedjs_page > .pagedjs_sheet > .pagedjs_pagebox > .pagedjs_area > div [data-split-to]:not([data-footnote-call])::after {
	content: unset;
}

.pagedjs_pages > .pagedjs_page > .pagedjs_sheet > .pagedjs_pagebox > .pagedjs_area > div [data-split-from]:not([data-footnote-call]):before,
.pagedjs_pages > .pagedjs_page > .pagedjs_sheet > .pagedjs_pagebox > .pagedjs_area > div [data-split-from]:not([data-footnote-call])::before {
	content: unset;
}

.pagedjs_pages > .pagedjs_page > .pagedjs_sheet > .pagedjs_pagebox > .pagedjs_area > div li[data-split-from]:first-of-type {
	list-style: none;
}

/*
[data-page]:not([data-split-from]),
[data-break-before="page"]:not([data-split-from]),
[data-break-before="always"]:not([data-split-from]),
[data-break-before="left"]:not([data-split-from]),
[data-break-before="right"]:not([data-split-from]),
[data-break-before="recto"]:not([data-split-from]),
[data-break-before="verso"]:not([data-split-from])
{
	break-before: column;
}

[data-page]:not([data-split-to]),
[data-break-after="page"]:not([data-split-to]),
[data-break-after="always"]:not([data-split-to]),
[data-break-after="left"]:not([data-split-to]),
[data-break-after="right"]:not([data-split-to]),
[data-break-after="recto"]:not([data-split-to]),
[data-break-after="verso"]:not([data-split-to])
{
	break-after: column;
}
*/

.pagedjs_clear-after::after {
	content: none !important;
}

[data-align-last-split-element='justify'] {
	text-align-last: justify;
}


@media print {
	html {
		width: 100%;
		height: 100%;
		-webkit-print-color-adjust: exact;
		print-color-adjust: exact;
	}
	body {
		margin: 0;
		padding: 0;
		width: 100% !important;
		height: 100% !important;
		min-width: 100%;
		max-width: 100%;
		min-height: 100%;
		max-height: 100%;
	}
	.pagedjs_pages {
		width: auto;
		display: block !important;
		transform: none !important;
		height: 100% !important;
		min-height: 100%;
		max-height: 100%;
		overflow: visible;
	}
	.pagedjs_page {
		margin: 0;
		padding: 0;
		max-height: 100%;
		min-height: 100%;
		height: 100% !important;
		page-break-after: always;
		break-after: page;
	}
	.pagedjs_sheet {
		margin: 0;
		padding: 0;
		max-height: 100%;
		min-height: 100%;
		height: 100% !important;
	}
}
`;

	async function request(url, options={}) {
		return new Promise(function(resolve, reject) {
			let request = new XMLHttpRequest();

			request.open(options.method || "get", url, true);

			for (let i in options.headers) {
				request.setRequestHeader(i, options.headers[i]);
			}

			request.withCredentials = options.credentials === "include";

			request.onload = () => {
				// Chrome returns a status code of 0 for local files
				const status = request.status === 0 && url.startsWith("file://") ? 200 : request.status;
				resolve(new Response(request.responseText, {status}));
			};

			request.onerror = reject;

			request.send(options.body || null);
		});
	}

	class Polisher {
		constructor(setup) {
			this.sheets = [];
			this.inserted = [];

			this.hooks = {};
			this.hooks.onUrl = new Hook(this);
			this.hooks.onAtPage = new Hook(this);
			this.hooks.onAtMedia = new Hook(this);
			this.hooks.onRule = new Hook(this);
			this.hooks.onDeclaration = new Hook(this);
			this.hooks.onContent = new Hook(this);
			this.hooks.onSelector = new Hook(this);
			this.hooks.onPseudoSelector = new Hook(this);

			this.hooks.onImport = new Hook(this);

			this.hooks.beforeTreeParse = new Hook(this);
			this.hooks.beforeTreeWalk = new Hook(this);
			this.hooks.afterTreeWalk = new Hook(this);

			if (setup !== false) {
				this.setup();
			}
		}

		setup() {
			this.base = this.insert(baseStyles);
			this.styleEl = document.createElement("style");
			document.head.appendChild(this.styleEl);
			this.styleSheet = this.styleEl.sheet;
			return this.styleSheet;
		}

		async add() {
			let fetched = [];
			let urls = [];

			for (var i = 0; i < arguments.length; i++) {
				let f;

				if (typeof arguments[i] === "object") {
					for (let url in arguments[i]) {
						let obj = arguments[i];
						f = new Promise(function(resolve, reject) {
							urls.push(url);
							resolve(obj[url]);
						});
					}
				} else {
					urls.push(arguments[i]);
					f = request(arguments[i]).then((response) => {
						return response.text();
					});
				}


				fetched.push(f);
			}

			return await Promise.all(fetched)
				.then(async (originals) => {
					let text = "";
					for (let index = 0; index < originals.length; index++) {
						text = await this.convertViaSheet(originals[index], urls[index]);
						this.insert(text);
					}
					return text;
				});
		}

		async convertViaSheet(cssStr, href) {
			let sheet = new Sheet(href, this.hooks);
			await sheet.parse(cssStr);

			// Insert the imported sheets first
			for (let url of sheet.imported) {
				let str = await request(url).then((response) => {
					return response.text();
				});
				let text = await this.convertViaSheet(str, url);
				this.insert(text);
			}

			this.sheets.push(sheet);

			if (typeof sheet.width !== "undefined") {
				this.width = sheet.width;
			}
			if (typeof sheet.height !== "undefined") {
				this.height = sheet.height;
			}
			if (typeof sheet.orientation !== "undefined") {
				this.orientation = sheet.orientation;
			}
			return sheet.toString();
		}

		insert(text){
			let head = document.querySelector("head");
			let style = document.createElement("style");
			style.setAttribute("data-pagedjs-inserted-styles", "true");

			style.appendChild(document.createTextNode(text));

			head.appendChild(style);

			this.inserted.push(style);
			return style;
		}

		destroy() {
			this.styleEl.remove();
			this.inserted.forEach((s) => {
				s.remove();
			});
			this.sheets = [];
		}
	}

	class Handler {
		constructor(chunker, polisher, caller) {
			let hooks = Object.assign({}, chunker && chunker.hooks, polisher && polisher.hooks, caller && caller.hooks);
			this.chunker = chunker;
			this.polisher = polisher;
			this.caller = caller;

			for (let name in hooks) {
				if (name in this) {
					let hook = hooks[name];
					hook.register(this[name].bind(this));
				}
			}
		}
	}

	EventEmitter(Handler.prototype);

	// https://www.w3.org/TR/css3-page/#page-size-prop

	var pageSizes = {
		"A0": {
			width: {
				value: 841,
				unit: "mm"
			},
			height: {
				value: 1189,
				unit: "mm"
			}
		},
		"A1": {
			width: {
				value: 594,
				unit: "mm"
			},
			height: {
				value: 841,
				unit: "mm"
			}
		},
		"A2": {
			width: {
				value: 420,
				unit: "mm"
			},
			height: {
				value: 594,
				unit: "mm"
			}
		},
		"A3": {
			width: {
				value: 297,
				unit: "mm"
			},
			height: {
				value: 420,
				unit: "mm"
			}
		},
		"A4": {
			width: {
				value: 210,
				unit: "mm"
			},
			height: {
				value: 297,
				unit: "mm"
			}
		},
		"A5": {
			width: {
				value: 148,
				unit: "mm"
			},
			height: {
				value: 210,
				unit: "mm"
			}
		},
		"A6": {
			width: {
				value: 105,
				unit: "mm"
			},
			height: {
				value: 148,
				unit: "mm"
			}
		},
		"A7": {
			width: {
				value: 74,
				unit: "mm"
			},
			height: {
				value: 105,
				unit: "mm"
			}
		},
		"A8": {
			width: {
				value: 52,
				unit: "mm"
			},
			height: {
				value: 74,
				unit: "mm"
			}
		},
		"A9": {
			width: {
				value: 37,
				unit: "mm"
			},
			height: {
				value: 52,
				unit: "mm"
			}
		},
		"A10": {
			width: {
				value: 26,
				unit: "mm"
			},
			height: {
				value: 37,
				unit: "mm"
			}
		},
		"B4": {
			width: {
				value: 250,
				unit: "mm"
			},
			height: {
				value: 353,
				unit: "mm"
			}
		},
		"B5": {
			width: {
				value: 176,
				unit: "mm"
			},
			height: {
				value: 250,
				unit: "mm"
			}
		},
		"letter": {
			width: {
				value: 8.5,
				unit: "in"
			},
			height: {
				value: 11,
				unit: "in"
			}
		},
		"legal": {
			width: {
				value: 8.5,
				unit: "in"
			},
			height: {
				value: 14,
				unit: "in"
			}
		},
		"ledger": {
			width: {
				value: 11,
				unit: "in"
			},
			height: {
				value: 17,
				unit: "in"
			}
		}
	};

	class AtPage extends Handler {
		constructor(chunker, polisher, caller) {
			super(chunker, polisher, caller);

			this.pages = {};

			this.width = undefined;
			this.height = undefined;
			this.orientation = undefined;
			this.marginalia = {};
		}

		pageModel(selector) {
			return {
				selector: selector,
				name: undefined,
				psuedo: undefined,
				nth: undefined,
				marginalia: {},
				width: undefined,
				height: undefined,
				orientation: undefined,
				margin: {
					top: {},
					right: {},
					left: {},
					bottom: {}
				},
				padding: {
					top: {},
					right: {},
					left: {},
					bottom: {}
				},
				border: {
					top: {},
					right: {},
					left: {},
					bottom: {}
				},
				backgroundOrigin: undefined,
				block: {},
				marks: undefined,
				notes: undefined,
				added: false
			};
		}

		// Find and Remove @page rules
		onAtPage(node, item, list) {
			let page, marginalia;
			let selector = "";
			let named, psuedo, nth;
			let needsMerge = false;

			if (node.prelude) {
				named = this.getTypeSelector(node);
				psuedo = this.getPsuedoSelector(node);
				nth = this.getNthSelector(node);
				selector = csstree.generate(node.prelude);
			} else {
				selector = "*";
			}

			if (selector in this.pages) {
				// this.pages[selector] = Object.assign(this.pages[selector], page);
				// console.log("after", selector, this.pages[selector]);

				// this.pages[selector].added = false;
				page = this.pages[selector];
				marginalia = this.replaceMarginalia(node);
				needsMerge = true;
				// Mark page for getting classes added again
				page.added = false;
			} else {
				page = this.pageModel(selector);
				marginalia = this.replaceMarginalia(node);
				this.pages[selector] = page;
			}

			page.name = named;
			page.psuedo = psuedo;
			page.nth = nth;

			if (needsMerge) {
				page.marginalia = Object.assign(page.marginalia, marginalia);
			} else {
				page.marginalia = marginalia;
			}

			let notes = this.replaceNotes(node);
			page.notes = notes;

			let declarations = this.replaceDeclarations(node);

			if (declarations.size) {
				page.size = declarations.size;
				page.width = declarations.size.width;
				page.height = declarations.size.height;
				page.orientation = declarations.size.orientation;
				page.format = declarations.size.format;
			}

			if (declarations.bleed && declarations.bleed[0] != "auto") {
				switch (declarations.bleed.length) {
					case 4: // top right bottom left
						page.bleed = {
							top: declarations.bleed[0],
							right: declarations.bleed[1],
							bottom: declarations.bleed[2],
							left: declarations.bleed[3]
						};
						break;
					case 3: // top right bottom right
						page.bleed = {
							top: declarations.bleed[0],
							right: declarations.bleed[1],
							bottom: declarations.bleed[2],
							left: declarations.bleed[1]
						};
						break;
					case 2: // top right top right
						page.bleed = {
							top: declarations.bleed[0],
							right: declarations.bleed[1],
							bottom: declarations.bleed[0],
							left: declarations.bleed[1]
						};
						break;
					default:
						page.bleed = {
							top: declarations.bleed[0],
							right: declarations.bleed[0],
							bottom: declarations.bleed[0],
							left: declarations.bleed[0]
						};
				}
			}

			if (declarations.marks) {
				if (!declarations.bleed || declarations.bleed && declarations.bleed[0] === "auto") {
					// Spec say 6pt, but needs more space for marks
					page.bleed = {
						top: { value: 6, unit: "mm" },
						right: { value: 6, unit: "mm" },
						bottom: { value: 6, unit: "mm" },
						left: { value: 6, unit: "mm" }
					};
				}

				page.marks = declarations.marks;
			}

			if (declarations.margin) {
				page.margin = declarations.margin;
			}
			if (declarations.padding) {
				page.padding = declarations.padding;
			}

			if (declarations.border) {
				page.border = declarations.border;
			}

			if (declarations.marks) {
				page.marks = declarations.marks;
			}

			if (needsMerge) {
				page.block.children.appendList(node.block.children);
			} else {
				page.block = node.block;
			}

			// Remove the rule
			list.remove(item);
		}

		/* Handled in breaks */
		/*
		afterParsed(parsed) {
			for (let b in this.named) {
				// Find elements
				let elements = parsed.querySelectorAll(b);
				// Add break data
				for (var i = 0; i < elements.length; i++) {
					elements[i].setAttribute("data-page", this.named[b]);
				}
			}
		}
		*/

		afterTreeWalk(ast, sheet) {
			let dirtyPage = "*" in this.pages && this.pages["*"].added === false;

			this.addPageClasses(this.pages, ast, sheet);

			if (dirtyPage) {
				let width = this.pages["*"].width;
				let height = this.pages["*"].height;
				let format = this.pages["*"].format;
				let orientation = this.pages["*"].orientation;
				let bleed = this.pages["*"].bleed;
				let marks = this.pages["*"].marks;
				let bleedverso = undefined;
				let bleedrecto = undefined;

				if (":left" in this.pages) {
					bleedverso = this.pages[":left"].bleed;
				}

				if (":right" in this.pages) {
					bleedrecto = this.pages[":right"].bleed;
				}

				if ((width && height) &&
					(this.width !== width || this.height !== height)) {
					this.width = width;
					this.height = height;
					this.format = format;
					this.orientation = orientation;

					this.addRootVars(ast, width, height, orientation, bleed, bleedrecto, bleedverso, marks);
					this.addRootPage(ast, this.pages["*"].size, bleed, bleedrecto, bleedverso);

					this.emit("size", { width, height, orientation, format, bleed });
					this.emit("atpages", this.pages);
				}

			}
		}

		getTypeSelector(ast) {
			// Find page name
			let name;

			csstree.walk(ast, {
				visit: "TypeSelector",
				enter: (node, item, list) => {
					name = node.name;
				}
			});

			return name;
		}

		getPsuedoSelector(ast) {
			// Find if it has :left & :right & :black & :first
			let name;
			csstree.walk(ast, {
				visit: "PseudoClassSelector",
				enter: (node, item, list) => {
					if (node.name !== "nth") {
						name = node.name;
					}
				}
			});

			return name;
		}

		getNthSelector(ast) {
			// Find if it has :nth
			let nth;
			csstree.walk(ast, {
				visit: "PseudoClassSelector",
				enter: (node, item, list) => {
					if (node.name === "nth" && node.children) {
						let raw = node.children.first();
						nth = raw.value;
					}
				}
			});

			return nth;
		}

		replaceMarginalia(ast) {
			let parsed = {};
			const MARGINS = [
				"top-left-corner", "top-left", "top", "top-center", "top-right", "top-right-corner",
				"bottom-left-corner", "bottom-left", "bottom", "bottom-center", "bottom-right", "bottom-right-corner",
				"left-top", "left-middle", "left", "left-bottom", "top-right-corner",
				"right-top", "right-middle", "right", "right-bottom", "right-right-corner"
			];
			csstree.walk(ast.block, {
				visit: "Atrule",
				enter: (node, item, list) => {
					let name = node.name;
					if (MARGINS.includes(name)) {
						if (name === "top") {
							name = "top-center";
						}
						if (name === "right") {
							name = "right-middle";
						}
						if (name === "left") {
							name = "left-middle";
						}
						if (name === "bottom") {
							name = "bottom-center";
						}
						parsed[name] = node.block;
						list.remove(item);
					}
				}
			});

			return parsed;
		}

		replaceNotes(ast) {
			let parsed = {};

			csstree.walk(ast.block, {
				visit: "Atrule",
				enter: (node, item, list) => {
					let name = node.name;
					if (name === "footnote") {
						parsed[name] = node.block;
						list.remove(item);
					}
				}
			});

			return parsed;
		}

		replaceDeclarations(ast) {
			let parsed = {};

			csstree.walk(ast.block, {
				visit: "Declaration",
				enter: (declaration, dItem, dList) => {
					let prop = csstree.property(declaration.property).name;
					// let value = declaration.value;

					if (prop === "marks") {
						parsed.marks = [];
						csstree.walk(declaration, {
							visit: "Identifier",
							enter: (ident) => {
								parsed.marks.push(ident.name);
							}
						});
						dList.remove(dItem);
					} else if (prop === "margin") {
						parsed.margin = this.getMargins(declaration);
						dList.remove(dItem);

					} else if (prop.indexOf("margin-") === 0) {
						let m = prop.substring("margin-".length);
						if (!parsed.margin) {
							parsed.margin = {
								top: {},
								right: {},
								left: {},
								bottom: {}
							};
						}
						parsed.margin[m] = declaration.value.children.first();
						dList.remove(dItem);

					} else if (prop === "padding") {
						parsed.padding = this.getPaddings(declaration.value);
						dList.remove(dItem);

					} else if (prop.indexOf("padding-") === 0) {
						let p = prop.substring("padding-".length);
						if (!parsed.padding) {
							parsed.padding = {
								top: {},
								right: {},
								left: {},
								bottom: {}
							};
						}
						parsed.padding[p] = declaration.value.children.first();
						dList.remove(dItem);
					}

					else if (prop === "border") {
						if (!parsed.border) {
							parsed.border = {
								top: {},
								right: {},
								left: {},
								bottom: {}
							};
						}
						parsed.border.top = csstree.generate(declaration.value);
						parsed.border.right = csstree.generate(declaration.value);
						parsed.border.left = csstree.generate(declaration.value);
						parsed.border.bottom = csstree.generate(declaration.value);

						dList.remove(dItem);

					}

					else if (prop.indexOf("border-") === 0) {
						if (!parsed.border) {
							parsed.border = {
								top: {},
								right: {},
								left: {},
								bottom: {}
							};
						}
						let p = prop.substring("border-".length);

						parsed.border[p] = csstree.generate(declaration.value);
						dList.remove(dItem);

					}

					else if (prop === "size") {
						parsed.size = this.getSize(declaration);
						dList.remove(dItem);
					} else if (prop === "bleed") {
						parsed.bleed = [];

						csstree.walk(declaration, {
							enter: (subNode) => {
								switch (subNode.type) {
									case "String": // bleed: "auto"
										if (subNode.value.indexOf("auto") > -1) {
											parsed.bleed.push("auto");
										}
										break;
									case "Dimension": // bleed: 1in 2in, bleed: 20px ect.
										parsed.bleed.push({
											value: subNode.value,
											unit: subNode.unit
										});
										break;
									case "Number":
										parsed.bleed.push({
											value: subNode.value,
											unit: "px"
										});
										break;
									// ignore
								}

							}
						});

						dList.remove(dItem);
					}

				}
			});

			return parsed;

		}
		getSize(declaration) {
			let width, height, orientation, format;

			// Get size: Xmm Ymm
			csstree.walk(declaration, {
				visit: "Dimension",
				enter: (node, item, list) => {
					let { value, unit } = node;
					if (typeof width === "undefined") {
						width = { value, unit };
					} else if (typeof height === "undefined") {
						height = { value, unit };
					}
				}
			});

			// Get size: "A4"
			csstree.walk(declaration, {
				visit: "String",
				enter: (node, item, list) => {
					let name = node.value.replace(/["|']/g, "");
					let s = pageSizes[name];
					if (s) {
						width = s.width;
						height = s.height;
					}
				}
			});

			// Get Format or Landscape or Portrait
			csstree.walk(declaration, {
				visit: "Identifier",
				enter: (node, item, list) => {
					let name = node.name;
					if (name === "landscape" || name === "portrait") {
						orientation = node.name;
					} else if (name !== "auto") {
						let s = pageSizes[name];
						if (s) {
							width = s.width;
							height = s.height;
						}
						format = name;
					}
				}
			});

			return {
				width,
				height,
				orientation,
				format
			};
		}

		getMargins(declaration) {
			let margins = [];
			let margin = {
				top: {},
				right: {},
				left: {},
				bottom: {}
			};

			csstree.walk(declaration, {
				enter: (node) => {
					switch (node.type) {
						case "Dimension": // margin: 1in 2in, margin: 20px, etc...
							margins.push(node);
							break;
						case "Number": // margin: 0
							margins.push({value: node.value, unit: "px"});
							break;
						// ignore
					}
				}
			});

			if (margins.length === 1) {
				for (let m in margin) {
					margin[m] = margins[0];
				}
			} else if (margins.length === 2) {
				margin.top = margins[0];
				margin.right = margins[1];
				margin.bottom = margins[0];
				margin.left = margins[1];
			} else if (margins.length === 3) {
				margin.top = margins[0];
				margin.right = margins[1];
				margin.bottom = margins[2];
				margin.left = margins[1];
			} else if (margins.length === 4) {
				margin.top = margins[0];
				margin.right = margins[1];
				margin.bottom = margins[2];
				margin.left = margins[3];
			}

			return margin;
		}

		getPaddings(declaration) {
			let paddings = [];
			let padding = {
				top: {},
				right: {},
				left: {},
				bottom: {}
			};

			csstree.walk(declaration, {
				enter: (node) => {
					switch (node.type) {
						case "Dimension": // padding: 1in 2in, padding: 20px, etc...
							paddings.push(node);
							break;
						case "Number": // padding: 0
							paddings.push({value: node.value, unit: "px"});
							break;
						// ignore
					}
				}
			});
			if (paddings.length === 1) {
				for (let p in padding) {
					padding[p] = paddings[0];
				}
			} else if (paddings.length === 2) {

				padding.top = paddings[0];
				padding.right = paddings[1];
				padding.bottom = paddings[0];
				padding.left = paddings[1];
			} else if (paddings.length === 3) {

				padding.top = paddings[0];
				padding.right = paddings[1];
				padding.bottom = paddings[2];
				padding.left = paddings[1];
			} else if (paddings.length === 4) {

				padding.top = paddings[0];
				padding.right = paddings[1];
				padding.bottom = paddings[2];
				padding.left = paddings[3];
			}
			return padding;
		}

		// get values for the border on the @page to pass them to the element with the .pagedjs_area class
		getBorders(declaration) {
			let border = {
				top: {},
				right: {},
				left: {},
				bottom: {}
			};

			if (declaration.prop == "border") {
				border.top = csstree.generate(declaration.value);
				border.right = csstree.generate(declaration.value);
				border.bottom = csstree.generate(declaration.value);
				border.left = csstree.generate(declaration.value);

			}
			else if (declaration.prop == "border-top") {
				border.top = csstree.generate(declaration.value);
			}
			else if (declaration.prop == "border-right") {
				border.right = csstree.generate(declaration.value);

			}
			else if (declaration.prop == "border-bottom") {
				border.bottom = csstree.generate(declaration.value);

			}
			else if (declaration.prop == "border-left") {
				border.left = csstree.generate(declaration.value);
			}

			return border;
		}


		addPageClasses(pages, ast, sheet) {
			// First add * page
			if ("*" in pages && pages["*"].added === false) {
				let p = this.createPage(pages["*"], ast.children, sheet);
				sheet.insertRule(p);
				pages["*"].added = true;
			}
			// Add :left & :right
			if (":left" in pages && pages[":left"].added === false) {
				let left = this.createPage(pages[":left"], ast.children, sheet);
				sheet.insertRule(left);
				pages[":left"].added = true;
			}
			if (":right" in pages && pages[":right"].added === false) {
				let right = this.createPage(pages[":right"], ast.children, sheet);
				sheet.insertRule(right);
				pages[":right"].added = true;
			}
			// Add :first & :blank
			if (":first" in pages && pages[":first"].added === false) {
				let first = this.createPage(pages[":first"], ast.children, sheet);
				sheet.insertRule(first);
				pages[":first"].added = true;
			}
			if (":blank" in pages && pages[":blank"].added === false) {
				let blank = this.createPage(pages[":blank"], ast.children, sheet);
				sheet.insertRule(blank);
				pages[":blank"].added = true;
			}
			// Add nth pages
			for (let pg in pages) {
				if (pages[pg].nth && pages[pg].added === false) {
					let nth = this.createPage(pages[pg], ast.children, sheet);
					sheet.insertRule(nth);
					pages[pg].added = true;
				}
			}

			// Add named pages
			for (let pg in pages) {
				if (pages[pg].name && pages[pg].added === false) {
					let named = this.createPage(pages[pg], ast.children, sheet);
					sheet.insertRule(named);
					pages[pg].added = true;
				}
			}

		}

		createPage(page, ruleList, sheet) {

			let selectors = this.selectorsForPage(page);
			let children = page.block.children.copy();
			let block = {
				type: "Block",
				loc: 0,
				children: children
			};


			let rule = this.createRule(selectors, block);

			this.addMarginVars(page.margin, children, children.first());
			this.addPaddingVars(page.padding, children, children.first());
			this.addBorderVars(page.border, children, children.first());


			if (page.width) {
				this.addDimensions(page.width, page.height, page.orientation, children, children.first());
			}

			if (page.marginalia) {
				this.addMarginaliaStyles(page, ruleList, rule, sheet);
				this.addMarginaliaContent(page, ruleList, rule, sheet);
			}

			if(page.notes) {
				this.addNotesStyles(page.notes, page, ruleList, rule, sheet);
			}

			return rule;
		}

		addMarginVars(margin, list, item) {
			// variables for margins
			for (let m in margin) {
				if (typeof margin[m].value !== "undefined") {
					let value = margin[m].value + (margin[m].unit || "");
					let mVar = list.createItem({
						type: "Declaration",
						property: "--pagedjs-margin-" + m,
						value: {
							type: "Raw",
							value: value
						}
					});
					list.append(mVar, item);

				}
			}
		}

		addPaddingVars(padding, list, item) {
			// variables for padding
			for (let p in padding) {

				if (typeof padding[p].value !== "undefined") {
					let value = padding[p].value + (padding[p].unit || "");
					let pVar = list.createItem({
						type: "Declaration",
						property: "--pagedjs-padding-" + p,
						value: {
							type: "Raw",
							value: value
						}
					});

					list.append(pVar, item);
				}

			}
		}

		addBorderVars(border, list, item) {
			// variables for borders
			for (const name of Object.keys(border)) {
				const value = border[name];
				// value is an empty object when undefined
				if (typeof value === "string") {
					const borderItem = list.createItem({
						type: "Declaration",
						property: "--pagedjs-border-" + name,
						value: {
							type: "Raw",
							value: value
						}
					});
					list.append(borderItem, item);
				}
			}
		}

		addDimensions(width, height, orientation, list, item) {
			let widthString, heightString;

			widthString = CSSValueToString(width);
			heightString = CSSValueToString(height);

			if (orientation && orientation !== "portrait") {
				// reverse for orientation
				[widthString, heightString] = [heightString, widthString];
			}

			// width variable
			let wVar = this.createVariable("--pagedjs-pagebox-width", widthString);
			list.appendData(wVar);

			// height variable
			let hVar = this.createVariable("--pagedjs-pagebox-height", heightString);
			list.appendData(hVar);

			// let w = this.createDimension("width", width);
			// let h = this.createDimension("height", height);
			// list.appendData(w);
			// list.appendData(h);
		}

		addMarginaliaStyles(page, list, item, sheet) {
			for (let loc in page.marginalia) {
				let block = csstree.clone(page.marginalia[loc]);
				let hasContent = false;

				if (block.children.isEmpty()) {
					continue;
				}

				csstree.walk(block, {
					visit: "Declaration",
					enter: (node, item, list) => {
						if (node.property === "content") {
							if (node.value.children && node.value.children.first().name === "none") {
								hasContent = false;
							} else {
								hasContent = true;
							}
							list.remove(item);
						}
						if (node.property === "vertical-align") {
							csstree.walk(node, {
								visit: "Identifier",
								enter: (identNode, identItem, identlist) => {
									let name = identNode.name;
									if (name === "top") {
										identNode.name = "flex-start";
									} else if (name === "middle") {
										identNode.name = "center";
									} else if (name === "bottom") {
										identNode.name = "flex-end";
									}
								}
							});
							node.property = "align-items";
						}

						if (node.property === "width" &&
							(loc === "top-left" ||
								loc === "top-center" ||
								loc === "top-right" ||
								loc === "bottom-left" ||
								loc === "bottom-center" ||
								loc === "bottom-right")) {
							let c = csstree.clone(node);
							c.property = "max-width";
							list.appendData(c);
						}

						if (node.property === "height" &&
							(loc === "left-top" ||
								loc === "left-middle" ||
								loc === "left-bottom" ||
								loc === "right-top" ||
								loc === "right-middle" ||
								loc === "right-bottom")) {
							let c = csstree.clone(node);
							c.property = "max-height";
							list.appendData(c);
						}
					}
				});

				let marginSelectors = this.selectorsForPageMargin(page, loc);
				let marginRule = this.createRule(marginSelectors, block);

				list.appendData(marginRule);

				let sel = csstree.generate({
					type: "Selector",
					children: marginSelectors
				});

				this.marginalia[sel] = {
					page: page,
					selector: sel,
					block: page.marginalia[loc],
					hasContent: hasContent
				};

			}
		}

		addMarginaliaContent(page, list, item, sheet) {
			let displayNone;
			// Just content
			for (let loc in page.marginalia) {
				let content = csstree.clone(page.marginalia[loc]);
				csstree.walk(content, {
					visit: "Declaration",
					enter: (node, item, list) => {
						if (node.property !== "content") {
							list.remove(item);
						}

						if (node.value.children && node.value.children.first().name === "none") {
							displayNone = true;
						}
					}
				});

				if (content.children.isEmpty()) {
					continue;
				}

				let displaySelectors = this.selectorsForPageMargin(page, loc);
				let displayDeclaration;

				displaySelectors.insertData({
					type: "Combinator",
					name: ">"
				});

				displaySelectors.insertData({
					type: "ClassSelector",
					name: "pagedjs_margin-content"
				});

				displaySelectors.insertData({
					type: "Combinator",
					name: ">"
				});

				displaySelectors.insertData({
					type: "TypeSelector",
					name: "*"
				});

				if (displayNone) {
					displayDeclaration = this.createDeclaration("display", "none");
				} else {
					displayDeclaration = this.createDeclaration("display", "block");
				}

				let displayRule = this.createRule(displaySelectors, [displayDeclaration]);
				sheet.insertRule(displayRule);

				// insert content rule
				let contentSelectors = this.selectorsForPageMargin(page, loc);

				contentSelectors.insertData({
					type: "Combinator",
					name: ">"
				});

				contentSelectors.insertData({
					type: "ClassSelector",
					name: "pagedjs_margin-content"
				});

				contentSelectors.insertData({
					type: "PseudoElementSelector",
					name: "after",
					children: null
				});

				let contentRule = this.createRule(contentSelectors, content);
				sheet.insertRule(contentRule);
			}
		}

		addRootVars(ast, width, height, orientation, bleed, bleedrecto, bleedverso, marks) {
			let rules = [];
			let selectors = new csstree.List();
			selectors.insertData({
				type: "PseudoClassSelector",
				name: "root",
				children: null
			});

			let widthString, heightString;
			let widthStringRight, heightStringRight;
			let widthStringLeft, heightStringLeft;

			if (!bleed) {
				widthString = CSSValueToString(width);
				heightString = CSSValueToString(height);
				widthStringRight = CSSValueToString(width);
				heightStringRight = CSSValueToString(height);
				widthStringLeft = CSSValueToString(width);
				heightStringLeft = CSSValueToString(height);
			} else {
				widthString = `calc( ${CSSValueToString(width)} + ${CSSValueToString(bleed.left)} + ${CSSValueToString(bleed.right)} )`;
				heightString = `calc( ${CSSValueToString(height)} + ${CSSValueToString(bleed.top)} + ${CSSValueToString(bleed.bottom)} )`;

				widthStringRight = `calc( ${CSSValueToString(width)} + ${CSSValueToString(bleed.left)} + ${CSSValueToString(bleed.right)} )`;
				heightStringRight = `calc( ${CSSValueToString(height)} + ${CSSValueToString(bleed.top)} + ${CSSValueToString(bleed.bottom)} )`;

				widthStringLeft = `calc( ${CSSValueToString(width)} + ${CSSValueToString(bleed.left)} + ${CSSValueToString(bleed.right)} )`;
				heightStringLeft = `calc( ${CSSValueToString(height)} + ${CSSValueToString(bleed.top)} + ${CSSValueToString(bleed.bottom)} )`;

				let bleedTop = this.createVariable("--pagedjs-bleed-top", CSSValueToString(bleed.top));
				let bleedRight = this.createVariable("--pagedjs-bleed-right", CSSValueToString(bleed.right));
				let bleedBottom = this.createVariable("--pagedjs-bleed-bottom", CSSValueToString(bleed.bottom));
				let bleedLeft = this.createVariable("--pagedjs-bleed-left", CSSValueToString(bleed.left));

				let bleedTopRecto = this.createVariable("--pagedjs-bleed-right-top", CSSValueToString(bleed.top));
				let bleedRightRecto = this.createVariable("--pagedjs-bleed-right-right", CSSValueToString(bleed.right));
				let bleedBottomRecto = this.createVariable("--pagedjs-bleed-right-bottom", CSSValueToString(bleed.bottom));
				let bleedLeftRecto = this.createVariable("--pagedjs-bleed-right-left", CSSValueToString(bleed.left));

				let bleedTopVerso = this.createVariable("--pagedjs-bleed-left-top", CSSValueToString(bleed.top));
				let bleedRightVerso = this.createVariable("--pagedjs-bleed-left-right", CSSValueToString(bleed.right));
				let bleedBottomVerso = this.createVariable("--pagedjs-bleed-left-bottom", CSSValueToString(bleed.bottom));
				let bleedLeftVerso = this.createVariable("--pagedjs-bleed-left-left", CSSValueToString(bleed.left));

				if (bleedrecto) {
					bleedTopRecto = this.createVariable("--pagedjs-bleed-right-top", CSSValueToString(bleedrecto.top));
					bleedRightRecto = this.createVariable("--pagedjs-bleed-right-right", CSSValueToString(bleedrecto.right));
					bleedBottomRecto = this.createVariable("--pagedjs-bleed-right-bottom", CSSValueToString(bleedrecto.bottom));
					bleedLeftRecto = this.createVariable("--pagedjs-bleed-right-left", CSSValueToString(bleedrecto.left));

					widthStringRight = `calc( ${CSSValueToString(width)} + ${CSSValueToString(bleedrecto.left)} + ${CSSValueToString(bleedrecto.right)} )`;
					heightStringRight = `calc( ${CSSValueToString(height)} + ${CSSValueToString(bleedrecto.top)} + ${CSSValueToString(bleedrecto.bottom)} )`;
				}
				if (bleedverso) {
					bleedTopVerso = this.createVariable("--pagedjs-bleed-left-top", CSSValueToString(bleedverso.top));
					bleedRightVerso = this.createVariable("--pagedjs-bleed-left-right", CSSValueToString(bleedverso.right));
					bleedBottomVerso = this.createVariable("--pagedjs-bleed-left-bottom", CSSValueToString(bleedverso.bottom));
					bleedLeftVerso = this.createVariable("--pagedjs-bleed-left-left", CSSValueToString(bleedverso.left));

					widthStringLeft = `calc( ${CSSValueToString(width)} + ${CSSValueToString(bleedverso.left)} + ${CSSValueToString(bleedverso.right)} )`;
					heightStringLeft = `calc( ${CSSValueToString(height)} + ${CSSValueToString(bleedverso.top)} + ${CSSValueToString(bleedverso.bottom)} )`;
				}

				let pageWidthVar = this.createVariable("--pagedjs-width", CSSValueToString(width));
				let pageHeightVar = this.createVariable("--pagedjs-height", CSSValueToString(height));

				rules.push(
					bleedTop,
					bleedRight,
					bleedBottom,
					bleedLeft,
					bleedTopRecto,
					bleedRightRecto,
					bleedBottomRecto,
					bleedLeftRecto,
					bleedTopVerso,
					bleedRightVerso,
					bleedBottomVerso,
					bleedLeftVerso,
					pageWidthVar,
					pageHeightVar
				);
			}

			if (marks) {
				marks.forEach((mark) => {
					let markDisplay = this.createVariable("--pagedjs-mark-" + mark + "-display", "block");
					rules.push(markDisplay);
				});
			}

			// orientation variable
			if (orientation) {
				let oVar = this.createVariable("--pagedjs-orientation", orientation);
				rules.push(oVar);

				if (orientation !== "portrait") {
					// reverse for orientation
					[widthString, heightString] = [heightString, widthString];
					[widthStringRight, heightStringRight] = [heightStringRight, widthStringRight];
					[widthStringLeft, heightStringLeft] = [heightStringLeft, widthStringLeft];
				}
			}

			let wVar = this.createVariable("--pagedjs-width", widthString);
			let hVar = this.createVariable("--pagedjs-height", heightString);

			let wVarR = this.createVariable("--pagedjs-width-right", widthStringRight);
			let hVarR = this.createVariable("--pagedjs-height-right", heightStringRight);

			let wVarL = this.createVariable("--pagedjs-width-left", widthStringLeft);
			let hVarL = this.createVariable("--pagedjs-height-left", heightStringLeft);

			rules.push(wVar, hVar, wVarR, hVarR, wVarL, hVarL);

			let rule = this.createRule(selectors, rules);

			ast.children.appendData(rule);
		}


		addNotesStyles(notes, page, list, item, sheet) {

			for (const note in notes) {
				let selectors = this.selectorsForPage(page);

				selectors.insertData({
					type: "Combinator",
					name: " "
				});

				selectors.insertData({
					type: "ClassSelector",
					name: "pagedjs_" + note + "_content"
				});

				let notesRule = this.createRule(selectors, notes[note]);

				list.appendData(notesRule);
			}

		}

		/*
		@page {
			size: var(--pagedjs-width) var(--pagedjs-height);
			margin: 0;
			padding: 0;
		}
		*/
		addRootPage(ast, size, bleed, bleedrecto, bleedverso) {
			let { width, height, orientation, format } = size;
			let children = new csstree.List();
			let childrenLeft = new csstree.List();
			let childrenRight = new csstree.List();
			let dimensions = new csstree.List();
			let dimensionsLeft = new csstree.List();
			let dimensionsRight = new csstree.List();

			if (bleed) {
				let widthCalculations = new csstree.List();
				let heightCalculations = new csstree.List();

				// width
				widthCalculations.appendData({
					type: "Dimension",
					unit: width.unit,
					value: width.value
				});

				widthCalculations.appendData({
					type: "WhiteSpace",
					value: " "
				});

				widthCalculations.appendData({
					type: "Operator",
					value: "+"
				});

				widthCalculations.appendData({
					type: "WhiteSpace",
					value: " "
				});

				widthCalculations.appendData({
					type: "Dimension",
					unit: bleed.left.unit,
					value: bleed.left.value
				});

				widthCalculations.appendData({
					type: "WhiteSpace",
					value: " "
				});

				widthCalculations.appendData({
					type: "Operator",
					value: "+"
				});

				widthCalculations.appendData({
					type: "WhiteSpace",
					value: " "
				});

				widthCalculations.appendData({
					type: "Dimension",
					unit: bleed.right.unit,
					value: bleed.right.value
				});

				// height
				heightCalculations.appendData({
					type: "Dimension",
					unit: height.unit,
					value: height.value
				});

				heightCalculations.appendData({
					type: "WhiteSpace",
					value: " "
				});

				heightCalculations.appendData({
					type: "Operator",
					value: "+"
				});

				heightCalculations.appendData({
					type: "WhiteSpace",
					value: " "
				});

				heightCalculations.appendData({
					type: "Dimension",
					unit: bleed.top.unit,
					value: bleed.top.value
				});

				heightCalculations.appendData({
					type: "WhiteSpace",
					value: " "
				});

				heightCalculations.appendData({
					type: "Operator",
					value: "+"
				});

				heightCalculations.appendData({
					type: "WhiteSpace",
					value: " "
				});

				heightCalculations.appendData({
					type: "Dimension",
					unit: bleed.bottom.unit,
					value: bleed.bottom.value
				});

				dimensions.appendData({
					type: "Function",
					name: "calc",
					children: widthCalculations
				});

				dimensions.appendData({
					type: "WhiteSpace",
					value: " "
				});

				dimensions.appendData({
					type: "Function",
					name: "calc",
					children: heightCalculations
				});

			} else if (format) {
				dimensions.appendData({
					type: "Identifier",
					name: format
				});

				if (orientation) {
					dimensions.appendData({
						type: "WhiteSpace",
						value: " "
					});

					dimensions.appendData({
						type: "Identifier",
						name: orientation
					});
				}
			} else {
				dimensions.appendData({
					type: "Dimension",
					unit: width.unit,
					value: width.value
				});

				dimensions.appendData({
					type: "WhiteSpace",
					value: " "
				});

				dimensions.appendData({
					type: "Dimension",
					unit: height.unit,
					value: height.value
				});
			}

			children.appendData({
				type: "Declaration",
				property: "size",
				loc: null,
				value: {
					type: "Value",
					children: dimensions
				}
			});

			children.appendData({
				type: "Declaration",
				property: "margin",
				loc: null,
				value: {
					type: "Value",
					children: [{
						type: "Dimension",
						unit: "px",
						value: 0
					}]
				}
			});

			children.appendData({
				type: "Declaration",
				property: "padding",
				loc: null,
				value: {
					type: "Value",
					children: [{
						type: "Dimension",
						unit: "px",
						value: 0
					}]
				}
			});

			children.appendData({
				type: "Declaration",
				property: "padding",
				loc: null,
				value: {
					type: "Value",
					children: [{
						type: "Dimension",
						unit: "px",
						value: 0
					}]
				}
			});

			let rule = ast.children.createItem({
				type: "Atrule",
				prelude: null,
				name: "page",
				block: {
					type: "Block",
					loc: null,
					children: children
				}
			});

			ast.children.append(rule);

			if (bleedverso) {
				let widthCalculationsLeft = new csstree.List();
				let heightCalculationsLeft = new csstree.List();

				// width
				widthCalculationsLeft.appendData({
					type: "Dimension",
					unit: width.unit,
					value: width.value
				});

				widthCalculationsLeft.appendData({
					type: "WhiteSpace",
					value: " "
				});

				widthCalculationsLeft.appendData({
					type: "Operator",
					value: "+"
				});

				widthCalculationsLeft.appendData({
					type: "WhiteSpace",
					value: " "
				});

				widthCalculationsLeft.appendData({
					type: "Dimension",
					unit: bleedverso.left.unit,
					value: bleedverso.left.value
				});

				widthCalculationsLeft.appendData({
					type: "WhiteSpace",
					value: " "
				});

				widthCalculationsLeft.appendData({
					type: "Operator",
					value: "+"
				});

				widthCalculationsLeft.appendData({
					type: "WhiteSpace",
					value: " "
				});

				widthCalculationsLeft.appendData({
					type: "Dimension",
					unit: bleedverso.right.unit,
					value: bleedverso.right.value
				});

				// height
				heightCalculationsLeft.appendData({
					type: "Dimension",
					unit: height.unit,
					value: height.value
				});

				heightCalculationsLeft.appendData({
					type: "WhiteSpace",
					value: " "
				});

				heightCalculationsLeft.appendData({
					type: "Operator",
					value: "+"
				});

				heightCalculationsLeft.appendData({
					type: "WhiteSpace",
					value: " "
				});

				heightCalculationsLeft.appendData({
					type: "Dimension",
					unit: bleedverso.top.unit,
					value: bleedverso.top.value
				});

				heightCalculationsLeft.appendData({
					type: "WhiteSpace",
					value: " "
				});

				heightCalculationsLeft.appendData({
					type: "Operator",
					value: "+"
				});

				heightCalculationsLeft.appendData({
					type: "WhiteSpace",
					value: " "
				});

				heightCalculationsLeft.appendData({
					type: "Dimension",
					unit: bleedverso.bottom.unit,
					value: bleedverso.bottom.value
				});

				dimensionsLeft.appendData({
					type: "Function",
					name: "calc",
					children: widthCalculationsLeft
				});

				dimensionsLeft.appendData({
					type: "WhiteSpace",
					value: " "
				});

				dimensionsLeft.appendData({
					type: "Function",
					name: "calc",
					children: heightCalculationsLeft
				});

				childrenLeft.appendData({
					type: "Declaration",
					property: "size",
					loc: null,
					value: {
						type: "Value",
						children: dimensionsLeft
					}
				});

				let ruleLeft = ast.children.createItem({
					type: "Atrule",
					prelude: null,
					name: "page :left",
					block: {
						type: "Block",
						loc: null,
						children: childrenLeft
					}
				});

				ast.children.append(ruleLeft);

			}

			if (bleedrecto) {
				let widthCalculationsRight = new csstree.List();
				let heightCalculationsRight = new csstree.List();

				// width
				widthCalculationsRight.appendData({
					type: "Dimension",
					unit: width.unit,
					value: width.value
				});

				widthCalculationsRight.appendData({
					type: "WhiteSpace",
					value: " "
				});

				widthCalculationsRight.appendData({
					type: "Operator",
					value: "+"
				});

				widthCalculationsRight.appendData({
					type: "WhiteSpace",
					value: " "
				});

				widthCalculationsRight.appendData({
					type: "Dimension",
					unit: bleedrecto.left.unit,
					value: bleedrecto.left.value
				});

				widthCalculationsRight.appendData({
					type: "WhiteSpace",
					value: " "
				});

				widthCalculationsRight.appendData({
					type: "Operator",
					value: "+"
				});

				widthCalculationsRight.appendData({
					type: "WhiteSpace",
					value: " "
				});

				widthCalculationsRight.appendData({
					type: "Dimension",
					unit: bleedrecto.right.unit,
					value: bleedrecto.right.value
				});

				// height
				heightCalculationsRight.appendData({
					type: "Dimension",
					unit: height.unit,
					value: height.value
				});

				heightCalculationsRight.appendData({
					type: "WhiteSpace",
					value: " "
				});

				heightCalculationsRight.appendData({
					type: "Operator",
					value: "+"
				});

				heightCalculationsRight.appendData({
					type: "WhiteSpace",
					value: " "
				});

				heightCalculationsRight.appendData({
					type: "Dimension",
					unit: bleedrecto.top.unit,
					value: bleedrecto.top.value
				});

				heightCalculationsRight.appendData({
					type: "WhiteSpace",
					value: " "
				});

				heightCalculationsRight.appendData({
					type: "Operator",
					value: "+"
				});

				heightCalculationsRight.appendData({
					type: "WhiteSpace",
					value: " "
				});

				heightCalculationsRight.appendData({
					type: "Dimension",
					unit: bleedrecto.bottom.unit,
					value: bleedrecto.bottom.value
				});

				dimensionsRight.appendData({
					type: "Function",
					name: "calc",
					children: widthCalculationsRight
				});

				dimensionsRight.appendData({
					type: "WhiteSpace",
					value: " "
				});

				dimensionsRight.appendData({
					type: "Function",
					name: "calc",
					children: heightCalculationsRight
				});

				childrenRight.appendData({
					type: "Declaration",
					property: "size",
					loc: null,
					value: {
						type: "Value",
						children: dimensionsRight
					}
				});

				let ruleRight = ast.children.createItem({
					type: "Atrule",
					prelude: null,
					name: "page :right",
					block: {
						type: "Block",
						loc: null,
						children: childrenRight
					}
				});

				ast.children.append(ruleRight);

			}
		}

		getNth(nth) {
			let n = nth.indexOf("n");
			let plus = nth.indexOf("+");
			let splitN = nth.split("n");
			let splitP = nth.split("+");
			let a = null;
			let b = null;
			if (n > -1) {
				a = splitN[0];
				if (plus > -1) {
					b = splitP[1];
				}
			} else {
				b = nth;
			}

			return {
				type: "Nth",
				loc: null,
				selector: null,
				nth: {
					type: "AnPlusB",
					loc: null,
					a: a,
					b: b
				}
			};
		}

		addPageAttributes(page, start, pages) {
			let namedPages = [start.dataset.page];

			if (namedPages && namedPages.length) {
				for (const named of namedPages) {
					if (!named) {
						continue;
					}
					page.name = named;
					page.element.classList.add("pagedjs_named_page");
					page.element.classList.add("pagedjs_" + named + "_page");

					if (!start.dataset.splitFrom) {
						page.element.classList.add("pagedjs_" + named + "_first_page");
					}
				}
			}
		}

		getStartElement(content, breakToken) {
			let node = breakToken && breakToken.node;

			if (!content && !breakToken) {
				return;
			}

			// No break
			if (!node) {
				return content.children[0];
			}

			// Top level element
			if (node.nodeType === 1 && node.parentNode.nodeType === 11) {
				return node;
			}

			// Named page
			if (node.nodeType === 1 && node.dataset.page) {
				return node;
			}

			// Get top level Named parent
			let fragment = rebuildAncestors(node);
			let pages = fragment.querySelectorAll("[data-page]");

			if (pages.length) {
				return pages[pages.length - 1];
			} else {
				return fragment.children[0];
			}
		}

		beforePageLayout(page, contents, breakToken, chunker) {
			let start = this.getStartElement(contents, breakToken);
			if (start) {
				this.addPageAttributes(page, start, chunker.pages);
			}
			// page.element.querySelector('.paged_area').style.color = red;
		}

		finalizePage(fragment, page, breakToken, chunker) {
			for (let m in this.marginalia) {
				let margin = this.marginalia[m];
				let sels = m.split(" ");

				let content;
				if (page.element.matches(sels[0]) && margin.hasContent) {
					content = page.element.querySelector(sels[1]);
					content.classList.add("hasContent");
				}
			}

			// check center
			["top", "bottom"].forEach((loc) => {
				let marginGroup = page.element.querySelector(".pagedjs_margin-" + loc);
				let center = page.element.querySelector(".pagedjs_margin-" + loc + "-center");
				let left = page.element.querySelector(".pagedjs_margin-" + loc + "-left");
				let right = page.element.querySelector(".pagedjs_margin-" + loc + "-right");

				let centerContent = center.classList.contains("hasContent");
				let leftContent = left.classList.contains("hasContent");
				let rightContent = right.classList.contains("hasContent");
				let centerWidth, leftWidth, rightWidth;

				if (leftContent) {
					leftWidth = window.getComputedStyle(left)["max-width"];
				}

				if (rightContent) {
					rightWidth = window.getComputedStyle(right)["max-width"];
				}


				if (centerContent) {
					centerWidth = window.getComputedStyle(center)["max-width"];

					if (centerWidth === "none" || centerWidth === "auto") {
						if (!leftContent && !rightContent) {
							marginGroup.style["grid-template-columns"] = "0 1fr 0";
						} else if (leftContent) {
							if (!rightContent) {
								if (leftWidth !== "none" && leftWidth !== "auto") {
									marginGroup.style["grid-template-columns"] = leftWidth + " 1fr " + leftWidth;
								} else {
									marginGroup.style["grid-template-columns"] = "auto auto 1fr";
									left.style["white-space"] = "nowrap";
									center.style["white-space"] = "nowrap";
									let leftOuterWidth = left.offsetWidth;
									let centerOuterWidth = center.offsetWidth;
									let outerwidths = leftOuterWidth + centerOuterWidth;
									let newcenterWidth = centerOuterWidth * 100 / outerwidths;
									marginGroup.style["grid-template-columns"] = "minmax(16.66%, 1fr) minmax(33%, " + newcenterWidth + "%) minmax(16.66%, 1fr)";
									left.style["white-space"] = "normal";
									center.style["white-space"] = "normal";
								}
							} else {
								if (leftWidth !== "none" && leftWidth !== "auto") {
									if (rightWidth !== "none" && rightWidth !== "auto") {
										marginGroup.style["grid-template-columns"] = leftWidth + " 1fr " + rightWidth;
									} else {
										marginGroup.style["grid-template-columns"] = leftWidth + " 1fr " + leftWidth;
									}
								} else {
									if (rightWidth !== "none" && rightWidth !== "auto") {
										marginGroup.style["grid-template-columns"] = rightWidth + " 1fr " + rightWidth;
									} else {
										marginGroup.style["grid-template-columns"] = "auto auto 1fr";
										left.style["white-space"] = "nowrap";
										center.style["white-space"] = "nowrap";
										right.style["white-space"] = "nowrap";
										let leftOuterWidth = left.offsetWidth;
										let centerOuterWidth = center.offsetWidth;
										let rightOuterWidth = right.offsetWidth;
										let outerwidths = leftOuterWidth + centerOuterWidth + rightOuterWidth;
										let newcenterWidth = centerOuterWidth * 100 / outerwidths;
										if (newcenterWidth > 40) {
											marginGroup.style["grid-template-columns"] = "minmax(16.66%, 1fr) minmax(33%, " + newcenterWidth + "%) minmax(16.66%, 1fr)";
										} else {
											marginGroup.style["grid-template-columns"] = "repeat(3, 1fr)";
										}
										left.style["white-space"] = "normal";
										center.style["white-space"] = "normal";
										right.style["white-space"] = "normal";
									}
								}
							}
						} else {
							if (rightWidth !== "none" && rightWidth !== "auto") {
								marginGroup.style["grid-template-columns"] = rightWidth + " 1fr " + rightWidth;
							} else {
								marginGroup.style["grid-template-columns"] = "auto auto 1fr";
								right.style["white-space"] = "nowrap";
								center.style["white-space"] = "nowrap";
								let rightOuterWidth = right.offsetWidth;
								let centerOuterWidth = center.offsetWidth;
								let outerwidths = rightOuterWidth + centerOuterWidth;
								let newcenterWidth = centerOuterWidth * 100 / outerwidths;
								marginGroup.style["grid-template-columns"] = "minmax(16.66%, 1fr) minmax(33%, " + newcenterWidth + "%) minmax(16.66%, 1fr)";
								right.style["white-space"] = "normal";
								center.style["white-space"] = "normal";
							}
						}
					} else if (centerWidth !== "none" && centerWidth !== "auto") {
						if (leftContent && leftWidth !== "none" && leftWidth !== "auto") {
							marginGroup.style["grid-template-columns"] = leftWidth + " " + centerWidth + " 1fr";
						} else if (rightContent && rightWidth !== "none" && rightWidth !== "auto") {
							marginGroup.style["grid-template-columns"] = "1fr " + centerWidth + " " + rightWidth;
						} else {
							marginGroup.style["grid-template-columns"] = "1fr " + centerWidth + " 1fr";
						}

					}

				} else {
					if (leftContent) {
						if (!rightContent) {
							marginGroup.style["grid-template-columns"] = "1fr 0 0";
						} else {
							if (leftWidth !== "none" && leftWidth !== "auto") {
								if (rightWidth !== "none" && rightWidth !== "auto") {
									marginGroup.style["grid-template-columns"] = leftWidth + " 1fr " + rightWidth;
								} else {
									marginGroup.style["grid-template-columns"] = leftWidth + " 0 1fr";
								}
							} else {
								if (rightWidth !== "none" && rightWidth !== "auto") {
									marginGroup.style["grid-template-columns"] = "1fr 0 " + rightWidth;
								} else {
									marginGroup.style["grid-template-columns"] = "auto 1fr auto";
									left.style["white-space"] = "nowrap";
									right.style["white-space"] = "nowrap";
									let leftOuterWidth = left.offsetWidth;
									let rightOuterWidth = right.offsetWidth;
									let outerwidths = leftOuterWidth + rightOuterWidth;
									let newLeftWidth = leftOuterWidth * 100 / outerwidths;
									marginGroup.style["grid-template-columns"] = "minmax(16.66%, " + newLeftWidth + "%) 0 1fr";
									left.style["white-space"] = "normal";
									right.style["white-space"] = "normal";
								}
							}
						}
					} else {
						if (rightWidth !== "none" && rightWidth !== "auto") {
							marginGroup.style["grid-template-columns"] = "1fr 0 " + rightWidth;
						} else {
							marginGroup.style["grid-template-columns"] = "0 0 1fr";
						}
					}
				}
			});

			// check middle
			["left", "right"].forEach((loc) => {
				let middle = page.element.querySelector(".pagedjs_margin-" + loc + "-middle.hasContent");
				let marginGroup = page.element.querySelector(".pagedjs_margin-" + loc);
				let top = page.element.querySelector(".pagedjs_margin-" + loc + "-top");
				let bottom = page.element.querySelector(".pagedjs_margin-" + loc + "-bottom");
				let topContent = top.classList.contains("hasContent");
				let bottomContent = bottom.classList.contains("hasContent");
				let middleHeight, topHeight, bottomHeight;

				if (topContent) {
					topHeight = window.getComputedStyle(top)["max-height"];
				}

				if (bottomContent) {
					bottomHeight = window.getComputedStyle(bottom)["max-height"];
				}

				if (middle) {
					middleHeight = window.getComputedStyle(middle)["max-height"];

					if (middleHeight === "none" || middleHeight === "auto") {
						if (!topContent && !bottomContent) {
							marginGroup.style["grid-template-rows"] = "0 1fr 0";
						} else if (topContent) {
							if (!bottomContent) {
								if (topHeight !== "none" && topHeight !== "auto") {
									marginGroup.style["grid-template-rows"] = topHeight + " calc(100% - " + topHeight + "*2) " + topHeight;
								}
							} else {
								if (topHeight !== "none" && topHeight !== "auto") {
									if (bottomHeight !== "none" && bottomHeight !== "auto") {
										marginGroup.style["grid-template-rows"] = topHeight + " calc(100% - " + topHeight + " - " + bottomHeight + ") " + bottomHeight;
									} else {
										marginGroup.style["grid-template-rows"] = topHeight + " calc(100% - " + topHeight + "*2) " + topHeight;
									}
								} else {
									if (bottomHeight !== "none" && bottomHeight !== "auto") {
										marginGroup.style["grid-template-rows"] = bottomHeight + " calc(100% - " + bottomHeight + "*2) " + bottomHeight;
									}
								}
							}
						} else {
							if (bottomHeight !== "none" && bottomHeight !== "auto") {
								marginGroup.style["grid-template-rows"] = bottomHeight + " calc(100% - " + bottomHeight + "*2) " + bottomHeight;
							}
						}
					} else {
						if (topContent && topHeight !== "none" && topHeight !== "auto") {
							marginGroup.style["grid-template-rows"] = topHeight + " " + middleHeight + " calc(100% - (" + topHeight + " + " + middleHeight + "))";
						} else if (bottomContent && bottomHeight !== "none" && bottomHeight !== "auto") {
							marginGroup.style["grid-template-rows"] = "1fr " + middleHeight + " " + bottomHeight;
						} else {
							marginGroup.style["grid-template-rows"] = "calc((100% - " + middleHeight + ")/2) " + middleHeight + " calc((100% - " + middleHeight + ")/2)";
						}

					}

				} else {
					if (topContent) {
						if (!bottomContent) {
							marginGroup.style["grid-template-rows"] = "1fr 0 0";
						} else {
							if (topHeight !== "none" && topHeight !== "auto") {
								if (bottomHeight !== "none" && bottomHeight !== "auto") {
									marginGroup.style["grid-template-rows"] = topHeight + " 1fr " + bottomHeight;
								} else {
									marginGroup.style["grid-template-rows"] = topHeight + " 0 1fr";
								}
							} else {
								if (bottomHeight !== "none" && bottomHeight !== "auto") {
									marginGroup.style["grid-template-rows"] = "1fr 0 " + bottomHeight;
								} else {
									marginGroup.style["grid-template-rows"] = "1fr 0 1fr";
								}
							}
						}
					} else {
						if (bottomHeight !== "none" && bottomHeight !== "auto") {
							marginGroup.style["grid-template-rows"] = "1fr 0 " + bottomHeight;
						} else {
							marginGroup.style["grid-template-rows"] = "0 0 1fr";
						}
					}
				}



			});

		}

		// CSS Tree Helpers

		selectorsForPage(page) {
			let nthlist;
			let nth;

			let selectors = new csstree.List();

			selectors.insertData({
				type: "ClassSelector",
				name: "pagedjs_page"
			});

			// Named page
			if (page.name) {
				selectors.insertData({
					type: "ClassSelector",
					name: "pagedjs_named_page"
				});

				selectors.insertData({
					type: "ClassSelector",
					name: "pagedjs_" + page.name + "_page"
				});
			}

			// PsuedoSelector
			if (page.psuedo && !(page.name && page.psuedo === "first")) {
				selectors.insertData({
					type: "ClassSelector",
					name: "pagedjs_" + page.psuedo + "_page"
				});
			}

			if (page.name && page.psuedo === "first") {
				selectors.insertData({
					type: "ClassSelector",
					name: "pagedjs_" + page.name + "_" + page.psuedo + "_page"
				});
			}

			// Nth
			if (page.nth) {
				nthlist = new csstree.List();
				nth = this.getNth(page.nth);

				nthlist.insertData(nth);

				selectors.insertData({
					type: "PseudoClassSelector",
					name: "nth-of-type",
					children: nthlist
				});
			}

			return selectors;
		}

		selectorsForPageMargin(page, margin) {
			let selectors = this.selectorsForPage(page);

			selectors.insertData({
				type: "Combinator",
				name: " "
			});

			selectors.insertData({
				type: "ClassSelector",
				name: "pagedjs_margin-" + margin
			});

			return selectors;
		}

		createDeclaration(property, value, important) {
			let children = new csstree.List();

			children.insertData({
				type: "Identifier",
				loc: null,
				name: value
			});

			return {
				type: "Declaration",
				loc: null,
				important: important,
				property: property,
				value: {
					type: "Value",
					loc: null,
					children: children
				}
			};
		}

		createVariable(property, value) {
			return {
				type: "Declaration",
				loc: null,
				property: property,
				value: {
					type: "Raw",
					value: value
				}
			};
		}

		createCalculatedDimension(property, items, important, operator = "+") {
			let children = new csstree.List();
			let calculations = new csstree.List();

			items.forEach((item, index) => {
				calculations.appendData({
					type: "Dimension",
					unit: item.unit,
					value: item.value
				});

				calculations.appendData({
					type: "WhiteSpace",
					value: " "
				});

				if (index + 1 < items.length) {
					calculations.appendData({
						type: "Operator",
						value: operator
					});

					calculations.appendData({
						type: "WhiteSpace",
						value: " "
					});
				}
			});

			children.insertData({
				type: "Function",
				loc: null,
				name: "calc",
				children: calculations
			});

			return {
				type: "Declaration",
				loc: null,
				important: important,
				property: property,
				value: {
					type: "Value",
					loc: null,
					children: children
				}
			};
		}

		createDimension(property, cssValue, important) {
			let children = new csstree.List();

			children.insertData({
				type: "Dimension",
				loc: null,
				value: cssValue.value,
				unit: cssValue.unit
			});

			return {
				type: "Declaration",
				loc: null,
				important: important,
				property: property,
				value: {
					type: "Value",
					loc: null,
					children: children
				}
			};
		}

		createBlock(declarations) {
			let block = new csstree.List();

			declarations.forEach((declaration) => {
				block.insertData(declaration);
			});

			return {
				type: "Block",
				loc: null,
				children: block
			};
		}

		createRule(selectors, block) {
			let selectorList = new csstree.List();
			selectorList.insertData({
				type: "Selector",
				children: selectors
			});

			if (Array.isArray(block)) {
				block = this.createBlock(block);
			}

			return {
				type: "Rule",
				prelude: {
					type: "SelectorList",
					children: selectorList
				},
				block: block
			};
		}

	}

	class Breaks extends Handler {
		constructor(chunker, polisher, caller) {
			super(chunker, polisher, caller);

			this.breaks = {};
		}

		onDeclaration(declaration, dItem, dList, rule) {
			let property = declaration.property;

			if (property === "page") {
				let children = declaration.value.children.first();
				let value = children.name;
				let selector = csstree.generate(rule.ruleNode.prelude);
				let name = value;

				let breaker = {
					property: property,
					value: value,
					selector: selector,
					name: name
				};

				selector.split(",").forEach((s) => {
					if (!this.breaks[s]) {
						this.breaks[s] = [breaker];
					} else {
						this.breaks[s].push(breaker);
					}
				});

				dList.remove(dItem);
			}

			if (property === "break-before" ||
					property === "break-after" ||
					property === "page-break-before" ||
					property === "page-break-after"
			) {
				let child = declaration.value.children.first();
				let value = child.name;
				let selector = csstree.generate(rule.ruleNode.prelude);

				if (property === "page-break-before") {
					property = "break-before";
				} else if (property === "page-break-after") {
					property = "break-after";
				}

				let breaker = {
					property: property,
					value: value,
					selector: selector
				};

				selector.split(",").forEach((s) => {
					if (!this.breaks[s]) {
						this.breaks[s] = [breaker];
					} else {
						this.breaks[s].push(breaker);
					}
				});

				// Remove from CSS -- handle right / left in module
				dList.remove(dItem);
			}
		}

		afterParsed(parsed) {
			this.processBreaks(parsed, this.breaks);
		}

		processBreaks(parsed, breaks) {
			for (let b in breaks) {
				// Find elements
				let elements = parsed.querySelectorAll(b);
				// Add break data
				for (var i = 0; i < elements.length; i++) {
					for (let prop of breaks[b]) {

						if (prop.property === "break-after") {
							let nodeAfter = displayedElementAfter(elements[i], parsed);

							elements[i].setAttribute("data-break-after", prop.value);

							if (nodeAfter) {
								nodeAfter.setAttribute("data-previous-break-after", prop.value);
							}
						} else if (prop.property === "break-before") {
							let nodeBefore = displayedElementBefore(elements[i], parsed);

							// Breaks are only allowed between siblings, not between a box and its container.
							// If we cannot find a node before we should not break!
							// https://drafts.csswg.org/css-break-3/#break-propagation
							if (nodeBefore) {
								if (prop.value === "page" && needsPageBreak(elements[i], nodeBefore)) {
									// we ignore this explicit page break because an implicit page break is already needed
									continue;
								}
								elements[i].setAttribute("data-break-before", prop.value);
								nodeBefore.setAttribute("data-next-break-before", prop.value);
							}
						} else if (prop.property === "page") {
							elements[i].setAttribute("data-page", prop.value);

							let nodeAfter = displayedElementAfter(elements[i], parsed);

							if (nodeAfter) {
								nodeAfter.setAttribute("data-after-page", prop.value);
							}
						} else {
							elements[i].setAttribute("data-" + prop.property, prop.value);
						}
					}
				}
			}
		}

		mergeBreaks(pageBreaks, newBreaks) {
			for (let b in newBreaks) {
				if (b in pageBreaks) {
					pageBreaks[b] = pageBreaks[b].concat(newBreaks[b]);
				} else {
					pageBreaks[b] = newBreaks[b];
				}
			}
			return pageBreaks;
		}

		addBreakAttributes(pageElement, page) {
			let before = pageElement.querySelector("[data-break-before]");
			let after = pageElement.querySelector("[data-break-after]");
			let previousBreakAfter = pageElement.querySelector("[data-previous-break-after]");

			if (before) {
				if (before.dataset.splitFrom) {
					page.splitFrom = before.dataset.splitFrom;
					pageElement.setAttribute("data-split-from", before.dataset.splitFrom);
				} else if (before.dataset.breakBefore && before.dataset.breakBefore !== "avoid") {
					page.breakBefore = before.dataset.breakBefore;
					pageElement.setAttribute("data-break-before", before.dataset.breakBefore);
				}
			}

			if (after && after.dataset) {
				if (after.dataset.splitTo) {
					page.splitTo = after.dataset.splitTo;
					pageElement.setAttribute("data-split-to", after.dataset.splitTo);
				} else if (after.dataset.breakAfter && after.dataset.breakAfter !== "avoid") {
					page.breakAfter = after.dataset.breakAfter;
					pageElement.setAttribute("data-break-after", after.dataset.breakAfter);
				}
			}

			if (previousBreakAfter && previousBreakAfter.dataset) {
				if (previousBreakAfter.dataset.previousBreakAfter && previousBreakAfter.dataset.previousBreakAfter !== "avoid") {
					page.previousBreakAfter = previousBreakAfter.dataset.previousBreakAfter;
				}
			}
		}

		afterPageLayout(pageElement, page) {
			this.addBreakAttributes(pageElement, page);
		}
	}

	class PrintMedia extends Handler {
		constructor(chunker, polisher, caller) {
			super(chunker, polisher, caller);
		}

		onAtMedia(node, item, list) {
			let media = this.getMediaName(node);
			let rules;
			if (media.includes("print")) {
				rules = node.block.children;

				// Append rules to the end of main rules list
				// TODO: this isn't working right, needs to check what is in the prelude
				/*
				rules.forEach((selectList) => {
					if (selectList.prelude) {
						selectList.prelude.children.forEach((rule) => {

							rule.children.prependData({
								type: "Combinator",
								name: " "
							});
		
							rule.children.prependData({
								type: "ClassSelector",
								name: "pagedjs_page"
							});
						});	
					}
				});

				list.insertList(rules, item);
				*/

				// Append rules to the end of main rules list
				list.appendList(rules);

				// Remove rules from the @media block
				list.remove(item);
			} else if (!media.includes("all") && !media.includes("pagedjs-ignore")) {
				list.remove(item);
			}

		}

		getMediaName(node) {
			let media = [];

			if (typeof node.prelude === "undefined" ||
					node.prelude.type !== "AtrulePrelude" ) {
				return;
			}

			csstree.walk(node.prelude, {
				visit: "Identifier",
				enter: (identNode, iItem, iList) => {
					media.push(identNode.name);
				}
			});
			return media;
		}


	}

	class Splits extends Handler {
		constructor(chunker, polisher, caller) {
			super(chunker, polisher, caller);
		}

		afterPageLayout(pageElement, page, breakToken, chunker) {
			let splits = Array.from(pageElement.querySelectorAll("[data-split-from]"));
			let pages = pageElement.parentNode;
			let index = Array.prototype.indexOf.call(pages.children, pageElement);
			let prevPage;

			if (index === 0) {
				return;
			}

			prevPage = pages.children[index - 1];

			let from; // Capture the last from element
			splits.forEach((split) => {
				let ref = split.dataset.ref;
				from = prevPage.querySelector("[data-ref='"+ ref +"']:not([data-split-to])");

				if (from) {
					from.dataset.splitTo = ref;

					if (!from.dataset.splitFrom) {
						from.dataset.splitOriginal = true;
					}
				}
			});

			// Fix alignment on the deepest split element
			if (from) {
				this.handleAlignment(from);
			}
		}

		handleAlignment(node) {
			let styles = window.getComputedStyle(node);
			let align = styles["text-align"];
			let alignLast = styles["text-align-last"];
			node.dataset.lastSplitElement = "true";
			if (align === "justify" && alignLast === "auto") {
				node.dataset.alignLastSplitElement = "justify";
			} else {
				node.dataset.alignLastSplitElement = alignLast;
			}
		}

	}

	class Counters extends Handler {
		constructor(chunker, polisher, caller) {
			super(chunker, polisher, caller);

			this.styleSheet = polisher.styleSheet;
			this.counters = {};
			this.resetCountersMap = new Map();
		}

		onDeclaration(declaration, dItem, dList, rule) {
			let property = declaration.property;

			if (property === "counter-increment") {
				this.handleIncrement(declaration, rule);
				// clean up empty declaration
				let hasProperities = false;
				declaration.value.children.forEach((data) => {
					if (data.type && data.type !== "WhiteSpace") {
						hasProperities = true;
					}
				});
				if (!hasProperities) {
					dList.remove(dItem);
				}
			} else if (property === "counter-reset") {
				this.handleReset(declaration, rule);
				// clean up empty declaration
				let hasProperities = false;
				declaration.value.children.forEach((data) => {
					if (data.type && data.type !== "WhiteSpace") {
						hasProperities = true;
					}
				});
				if (!hasProperities) {
					dList.remove(dItem);
				}
			}
		}

		afterParsed(parsed) {
			this.processCounters(parsed, this.counters);
			this.scopeCounters(this.counters);
		}

		addCounter(name) {
			if (name in this.counters) {
				return this.counters[name];
			}

			this.counters[name] = {
				name: name,
				increments: {},
				resets: {}
			};

			return this.counters[name];
		}

		handleIncrement(declaration, rule) {
			let increments = [];
			let children = declaration.value.children;

			children.forEach((data, item) => {
				if (data.type && data.type === "Identifier") {
					let name = data.name;

					if (name === "page" || name.indexOf("target-counter-") === 0) {
						return;
					}

					let whitespace, number, value;
					if (item.next && item.next.data.type === "WhiteSpace") {
						whitespace = item.next;
					}
					if (whitespace && whitespace.next && whitespace.next.data.type === "Number") {
						number = whitespace.next;
						value = parseInt(number.data.value);
					}

					let selector = csstree.generate(rule.ruleNode.prelude);

					let counter;
					if (!(name in this.counters)) {
						counter = this.addCounter(name);
					} else {
						counter = this.counters[name];
					}
					let increment = {
						selector: selector,
						number: value || 1
					};
					counter.increments[selector] = increment;
					increments.push(increment);

					// Remove the parsed resets
					children.remove(item);
					if (whitespace) {
						children.remove(whitespace);
					}
					if (number) {
						children.remove(number);
					}
				}
			});
			
			return increments;
		}

		handleReset(declaration, rule) {
			let children = declaration.value.children;

			children.forEach((data, item) => {
				if (data.type && data.type === "Identifier") {
					let name = data.name;
					let whitespace, number, value;
					if (item.next && item.next.data.type === "WhiteSpace") {
						whitespace = item.next;
					}
					if (whitespace && whitespace.next) {
						if (whitespace.next.data.type === "Number") {
							// The counter reset value is specified using a number. E.g. counter-reset: c2 5;
							number = whitespace.next;
							value = parseInt(number.data.value);
						} else if (whitespace.next.data.type === "Function" && whitespace.next.data.name === "var") {
							// The counter reset value is specified using a CSS variable (custom property).
							// E.g. counter-reset: c2 var(--my-variable);
							// See https://developer.mozilla.org/en-US/docs/Web/CSS/var
							number = whitespace.next;
							// Use the variable name (e.g. '--my-variable') as value for now. The actual value is resolved later by the
							// processCounterResets function.
							value = whitespace.next.data.children.head.data.name;
						}
					}

					let counter;
					let selector;
					let prelude = rule.ruleNode.prelude;

					if (rule.ruleNode.type === "Atrule" && rule.ruleNode.name === "page") {
						selector = ".pagedjs_page";
					} else {
						selector = csstree.generate(prelude || rule.ruleNode);
					}

					if (name === "footnote") {
						this.addFootnoteMarkerCounter(declaration.value.children);
					}

					if (!(name in this.counters)) {
						counter = this.addCounter(name);
					} else {
						counter = this.counters[name];
					}

					let reset = {
						selector: selector,
						number: value || 0
					};

					counter.resets[selector] = reset;

					if (selector !== ".pagedjs_page") {
						// Remove the parsed resets
						children.remove(item);
						if (whitespace) {
							children.remove(whitespace);
						}
						if (number) {
							children.remove(number);
						}
					}
				}
			});
		}

		processCounters(parsed, counters) {
			let counter;
			for (let c in counters) {
				counter = this.counters[c];
				this.processCounterIncrements(parsed, counter);
				this.processCounterResets(parsed, counter);
				if (c !== "page") {
					this.addCounterValues(parsed, counter);
				}
			}
		}

		scopeCounters(counters) {
			let countersArray = [];
			for (let c in counters) {
				if(c !== "page") {
					countersArray.push(`${counters[c].name} 0`);
				}
			}
			// Add to pages to allow cross page scope
			this.insertRule(`.pagedjs_pages { counter-reset: ${countersArray.join(" ")} page 0 pages var(--pagedjs-page-count) footnote var(--pagedjs-footnotes-count) footnote-marker var(--pagedjs-footnotes-count)}`);
		}

		insertRule(rule) {
			this.styleSheet.insertRule(rule, this.styleSheet.cssRules.length);
		}

		processCounterIncrements(parsed, counter) {
			let increment;
			for (let inc in counter.increments) {
				increment = counter.increments[inc];
				// Find elements for increments
				let incrementElements = parsed.querySelectorAll(increment.selector);
				// Add counter data
				for (let i = 0; i < incrementElements.length; i++) {
					incrementElements[i].setAttribute("data-counter-"+ counter.name +"-increment", increment.number);
					if (incrementElements[i].getAttribute("data-counter-increment")) {
						incrementElements[i].setAttribute("data-counter-increment", incrementElements[i].getAttribute("data-counter-increment") + " " + counter.name);
					} else {
						incrementElements[i].setAttribute("data-counter-increment", counter.name);
					}
				}
			}
		}

		processCounterResets(parsed, counter) {
			let reset;
			for (let r in counter.resets) {
				reset = counter.resets[r];
				// Find elements for resets
				let resetElements = parsed.querySelectorAll(reset.selector);
				// Add counter data
				for (var i = 0; i < resetElements.length; i++) {
					let value = reset.number;
					if (typeof value === "string" && value.startsWith("--")) {
						// The value is specified using a CSS variable (custom property).
						// FIXME: We get the variable value only from the inline style of the element because at this point the
						// element is detached and thus using:
						//
						//		getComputedStyle(resetElements[i]).getPropertyValue(value)
						//
						// always returns an empty string. We could try to temporarily attach the element to get its computed style,
						// but for now using the inline style is enough for us.
						value = resetElements[i].style.getPropertyValue(value) || 0;
					}
					resetElements[i].setAttribute("data-counter-"+ counter.name +"-reset", value);
					if (resetElements[i].getAttribute("data-counter-reset")) {
						resetElements[i].setAttribute("data-counter-reset", resetElements[i].getAttribute("data-counter-reset") + " " + counter.name);
					} else {
						resetElements[i].setAttribute("data-counter-reset", counter.name);
					}
				}
			}
		}

		addCounterValues(parsed, counter) {
			let counterName = counter.name;

			if (counterName === "page" || counterName === "footnote") {
				return;
			}

			let elements = parsed.querySelectorAll("[data-counter-"+ counterName +"-reset], [data-counter-"+ counterName +"-increment]");

			let count = 0;
			let element;
			let increment, reset;
			let resetValue, incrementValue, resetDelta;
			let incrementArray;

			for (let i = 0; i < elements.length; i++) {
				element = elements[i];
				resetDelta = 0;
				incrementArray = [];

				if (element.hasAttribute("data-counter-"+ counterName +"-reset")) {
					reset = element.getAttribute("data-counter-"+ counterName +"-reset");
					resetValue = parseInt(reset);

					// Use negative increment value inplace of reset
					resetDelta = resetValue - count;
					incrementArray.push(`${counterName} ${resetDelta}`);

					count = resetValue;
				}

				if (element.hasAttribute("data-counter-"+ counterName +"-increment")) {

					increment = element.getAttribute("data-counter-"+ counterName +"-increment");
					incrementValue = parseInt(increment);

					count += incrementValue;

					element.setAttribute("data-counter-"+counterName+"-value", count);

					incrementArray.push(`${counterName} ${incrementValue}`);
				}

				if (incrementArray.length > 0) {
					this.incrementCounterForElement(element, incrementArray);
				}

			}
		}

		addFootnoteMarkerCounter(list) {
			let markers = [];
			csstree.walk(list, {
				visit: "Identifier",
				enter: (identNode, iItem, iList) => {
					markers.push(identNode.name);
				}
			});

			// Already added
			if (markers.includes("footnote-maker")) {
				return;
			}

			list.insertData({
				type: "WhiteSpace",
				value: " "
			});

			list.insertData({
				type: "Identifier",
				name: "footnote-marker"
			});

			list.insertData({
				type: "WhiteSpace",
				value: " "
			});

			list.insertData({
				type: "Number",
				value: 0
			});
		}

		incrementCounterForElement(element, incrementArray) {
			if (!element || !incrementArray || incrementArray.length === 0) return;

			const ref = element.dataset.ref;
			const increments = Array.from(this.styleSheet.cssRules).filter((rule) => {
				return rule.selectorText === `[data-ref="${element.dataset.ref}"]:not([data-split-from])`
							 && rule.style[0] === "counter-increment";
			}).map(rule => rule.style.counterIncrement);

			// Merge the current increments by summing the values because we generate both a decrement and an increment when the
			// element resets and increments the counter at the same time. E.g. ['c1 -7', 'c1 1'] should lead to 'c1 -6'.
			increments.push(this.mergeIncrements(incrementArray,
				(prev, next) => (parseInt(prev) || 0) + (parseInt(next) || 0)));

			// Keep the last value for each counter when merging with the previous increments. E.g. ['c1 -7 c2 3', 'c1 1']
			// should lead to 'c1 1 c2 3'.
			const counterIncrement = this.mergeIncrements(increments, (prev, next) => next);
			this.insertRule(`[data-ref="${ref}"]:not([data-split-from]) { counter-increment: ${counterIncrement} }`);
		}

		/**
		 * Merge multiple values of a counter-increment CSS rule, using the specified operator.
		 *
		 * @param {Array} incrementArray the values to merge, e.g. ['c1 1', 'c1 -7 c2 1']
		 * @param {Function} operator the function used to merge counter values (e.g. keep the last value of a counter or sum
		 *					the counter values)
		 * @return {string} the merged value of the counter-increment CSS rule
		 */
		mergeIncrements(incrementArray, operator) {
			const increments = {};
			incrementArray.forEach(increment => {
				let values = increment.split(" ");
				for (let i = 0; i < values.length; i+=2) {
					increments[values[i]] = operator(increments[values[i]], values[i + 1]);
				}
			});

			return Object.entries(increments).map(([key, value]) => `${key} ${value}`).join(" ");
		}

		afterPageLayout(pageElement, page) {
			let resets = [];

			let pgreset = pageElement.querySelectorAll("[data-counter-page-reset]:not([data-split-from])");
			pgreset.forEach((reset) => {
				const ref = reset.dataset && reset.dataset.ref;
				if (ref && this.resetCountersMap.has(ref)) ; else {
					if (ref) {
						this.resetCountersMap.set(ref, "");
					}
					let value = reset.dataset.counterPageReset;
					resets.push(`page ${value}`);
				}
			});

			let notereset = pageElement.querySelectorAll("[data-counter-footnote-reset]:not([data-split-from])");
			notereset.forEach((reset) => {
				let value = reset.dataset.counterFootnoteReset;
				resets.push(`footnote ${value}`);
				resets.push(`footnote-marker ${value}`);
			});

			if (resets.length) {
				this.styleSheet.insertRule(`[data-page-number="${pageElement.dataset.pageNumber}"] { counter-increment: none; counter-reset: ${resets.join(" ")} }`, this.styleSheet.cssRules.length);
			}
		}

	}

	class Lists extends Handler {
		constructor(chunker, polisher, caller) {
			super(chunker, polisher, caller);
		}
		afterParsed(content) {
			const orderedLists = content.querySelectorAll("ol");

			for (var list of orderedLists) {
				this.addDataNumbers(list);
			}
		}

		afterPageLayout(pageElement, page, breakToken, chunker) {
			var orderedLists = pageElement.getElementsByTagName("ol");
			for (var list of orderedLists) {
				if (list.firstElementChild) {
					list.start = list.firstElementChild.dataset.itemNum;
				}
			}
		}

		addDataNumbers(list) {
			let start = 1;
			if (list.hasAttribute("start")) {
				start = parseInt(list.getAttribute("start"), 10);
				if (isNaN(start)) {
					start = 1;
				}
			}
			let items = list.children;
			for (var i = 0; i < items.length; i++) {
				items[i].setAttribute("data-item-num", i + start);
			}
		}

	}

	class PositionFixed extends Handler {
		constructor(chunker, polisher, caller) {
			super(chunker, polisher, caller);
			this.styleSheet = polisher.styleSheet;
			this.fixedElementsSelector = [];
			this.fixedElements = [];
		}

		onDeclaration(declaration, dItem, dList, rule) {
			if (declaration.property === "position" && declaration.value.children.first().name === "fixed") {
				let selector = csstree.generate(rule.ruleNode.prelude);
				this.fixedElementsSelector.push(selector);
				dList.remove(dItem);
			}
		}

		afterParsed(fragment) {
			this.fixedElementsSelector.forEach(fixedEl => {
				fragment.querySelectorAll(`${fixedEl}`).forEach(el => {
					el.style.setProperty("position", "absolute");
					this.fixedElements.push(el);
					el.remove();
				});
			});
		}

		afterPageLayout(pageElement, page, breakToken) {
			this.fixedElements.forEach(el => {
				const clone = el.cloneNode(true);
				pageElement.querySelector(".pagedjs_pagebox").insertAdjacentElement("afterbegin", clone);
			});
		}
	}

	class PageCounterIncrement extends Handler {
		constructor(chunker, polisher, caller) {
			super(chunker, polisher, caller);

			this.styleSheet = polisher.styleSheet;
			this.pageCounter = {
				name: "page",
				increments: {},
				resets: {}
			};
		}

		onDeclaration(declaration, dItem, dList, rule) {
			const property = declaration.property;

			if (property === "counter-increment") {
				let inc = this.handleIncrement(declaration, rule);
				if (inc) {
					dList.remove(dItem);
				}
			}
		}

		afterParsed(_) {
			for (const inc in this.pageCounter.increments) {
				const increment = this.pageCounter.increments[inc];
				this.insertRule(`${increment.selector} { --pagedjs-page-counter-increment: ${increment.number} }`);
			}
		}

		handleIncrement(declaration, rule) {
			const identifier = declaration.value.children.first();
			const number = declaration.value.children.getSize() > 1 ? declaration.value.children.last().value : 1;
			const name = identifier && identifier.name;

			if (name && name.indexOf("target-counter-") === 0) {
				return;
			}
			// A counter named page is automatically created and incremented by 1 on every page of the document,
			// unless the counter-increment property in the page context explicitly specifies a different increment for the page counter.
			// https://www.w3.org/TR/css-page-3/#page-based-counters
			if (name !== "page") {
				return;
			}
			// the counter-increment property is not defined on the page context (i.e. @page rule), ignoring...
			if (rule.ruleNode.name === "page" && rule.ruleNode.type === "Atrule") {
				return;
			}
			const selector = csstree.generate(rule.ruleNode.prelude);
			return this.pageCounter.increments[selector] = {
				selector: selector,
				number
			};
		}

		insertRule(rule) {
			this.styleSheet.insertRule(rule, this.styleSheet.cssRules.length);
		}
	}

	class NthOfType extends Handler {
		constructor(chunker, polisher, caller) {
			super(chunker, polisher, caller);

			this.styleSheet = polisher.styleSheet;
			this.selectors = {};
		}

		onRule(ruleNode, ruleItem, rulelist) {
			let selector = csstree.generate(ruleNode.prelude);
			if (selector.match(/:(first|last|nth)-of-type/)) {
				
				let declarations = csstree.generate(ruleNode.block);
				declarations = declarations.replace(/[{}]/g,"");

				let uuid = "nth-of-type-" + UUID();

				selector.split(",").forEach((s) => {
					if (!this.selectors[s]) {
						this.selectors[s] = [uuid, declarations];
					} else {
						this.selectors[s][1] = `${this.selectors[s][1]};${declarations}` ;
					}
				});

				rulelist.remove(ruleItem);
			}
		}

		afterParsed(parsed) {
			this.processSelectors(parsed, this.selectors);
		}

		processSelectors(parsed, selectors) {
			// add the new attributes to matching elements
			for (let s in selectors) {
				let elements = parsed.querySelectorAll(s);

				for (var i = 0; i < elements.length; i++) {
					let dataNthOfType = elements[i].getAttribute("data-nth-of-type");

					if (dataNthOfType && dataNthOfType != "") {
						dataNthOfType = `${dataNthOfType},${selectors[s][0]}`;
						elements[i].setAttribute("data-nth-of-type", dataNthOfType);
					} else {
						elements[i].setAttribute("data-nth-of-type", selectors[s][0]);
					}
				}

				let rule = `*[data-nth-of-type*='${selectors[s][0]}'] { ${selectors[s][1]}; }`;
				this.styleSheet.insertRule(rule, this.styleSheet.cssRules.length);
			}
		}
	}

	class Following extends Handler {
		constructor(chunker, polisher, caller) {
			super(chunker, polisher, caller);

			this.styleSheet = polisher.styleSheet;
			this.selectors = {};
		}

		onRule(ruleNode, ruleItem, rulelist) {
			let selector = csstree.generate(ruleNode.prelude);
			if (selector.match(/\+/)) {
				
				let declarations = csstree.generate(ruleNode.block);
				declarations = declarations.replace(/[{}]/g,"");

				let uuid = "following-" + UUID();

				selector.split(",").forEach((s) => {
					if (!this.selectors[s]) {
						this.selectors[s] = [uuid, declarations];
					} else {
						this.selectors[s][1] = `${this.selectors[s][1]};${declarations}` ;
					}
				});

				rulelist.remove(ruleItem);
			}
		}

		afterParsed(parsed) {
			this.processSelectors(parsed, this.selectors);
		}

		processSelectors(parsed, selectors) {
			// add the new attributes to matching elements
			for (let s in selectors) {
				let elements = parsed.querySelectorAll(s);

				for (var i = 0; i < elements.length; i++) {
					let dataFollowing = elements[i].getAttribute("data-following");

					if (dataFollowing && dataFollowing != "") {
						dataFollowing = `${dataFollowing},${selectors[s][0]}`;
						elements[i].setAttribute("data-following", dataFollowing);
					} else {
						elements[i].setAttribute("data-following", selectors[s][0]);
					}
				}

				let rule = `*[data-following*='${selectors[s][0]}'] { ${selectors[s][1]}; }`;
				this.styleSheet.insertRule(rule, this.styleSheet.cssRules.length);
			}
		}
	}

	class Footnotes extends Handler {
		constructor(chunker, polisher, caller) {
			super(chunker, polisher, caller);

			this.footnotes = {};
			this.needsLayout = [];
		}

		onDeclaration(declaration, dItem, dList, rule) {
			let property = declaration.property;
			if (property === "float") {
				let identifier = declaration.value.children && declaration.value.children.first();
				let location = identifier && identifier.name;
				if (location === "footnote") {
					let selector = csstree.generate(rule.ruleNode.prelude);
					this.footnotes[selector] = {
						selector: selector,
						policy: "auto",
						display: "block"
					};
					dList.remove(dItem);
				}
			}
			if (property === "footnote-policy") {
				let identifier = declaration.value.children && declaration.value.children.first();
				let policy = identifier && identifier.name;
				if (policy) {
					let selector = csstree.generate(rule.ruleNode.prelude);
					let note = this.footnotes[selector];
					if (note) {
						note.policy = policy;
					}
				}
			}
			if (property === "footnote-display") {
				let identifier = declaration.value.children && declaration.value.children.first();
				let display = identifier && identifier.name;
				let selector = csstree.generate(rule.ruleNode.prelude);
				if (display && this.footnotes[selector]) {
					let note = this.footnotes[selector];
					if (note) {
						note.display = display;
					}
				}
			}
		}

		onPseudoSelector(pseudoNode, pItem, pList, selector, rule) {
			let name = pseudoNode.name;
			if (name === "footnote-marker") {
				// switch ::footnote-marker to [data-footnote-marker]::before
				let prelude = rule.ruleNode.prelude;
				let newPrelude = new csstree.List();

				// Can't get remove to work, so just copying everything else
				prelude.children.first().children.each((node) => {
					if (node.type !== "PseudoElementSelector") {
						newPrelude.appendData(node);
					}
				});

				// Add our data call
				newPrelude.appendData({
					type: "AttributeSelector",
					name: {
						type: "Identifier",
						name: "data-footnote-marker",
					},
					flags: null,
					loc: null,
					matcher: null,
					value: null
				});

				// Add new pseudo element
				newPrelude.appendData({
					type: "PseudoElementSelector",
					name: "marker",
					loc: null,
					children: null
				});

				prelude.children.first().children = newPrelude;
			}

			if (name === "footnote-call") {
				// switch ::footnote-call to [data-footnote-call]::after

				let prelude = rule.ruleNode.prelude;
				let newPrelude = new csstree.List();

				// Can't get remove to work, so just copying everything else
				prelude.children.first().children.each((node) => {
					if (node.type !== "PseudoElementSelector") {
						newPrelude.appendData(node);
					}
				});

				// Add our data call
				newPrelude.appendData({
					type: "AttributeSelector",
					name: {
						type: "Identifier",
						name: "data-footnote-call",
					},
					flags: null,
					loc: null,
					matcher: null,
					value: null
				});

				// Add new pseudo element
				newPrelude.appendData({
					type: "PseudoElementSelector",
					name: "after",
					loc: null,
					children: null
				});

				prelude.children.first().children = newPrelude;
			}
		}

		afterParsed(parsed) {
			this.processFootnotes(parsed, this.footnotes);
		}

		processFootnotes(parsed, notes) {
			for (let n in notes) {
				// Find elements
				let elements = parsed.querySelectorAll(n);
				let element;
				let note = notes[n];
				for (var i = 0; i < elements.length; i++) {
					element = elements[i];
					// Add note type
					element.setAttribute("data-note", "footnote");
					element.setAttribute("data-break-before", "avoid");
					element.setAttribute("data-note-policy", note.policy || "auto");
					element.setAttribute("data-note-display", note.display || "block");
					// Mark all parents
					this.processFootnoteContainer(element);
				}
			}
		}

		processFootnoteContainer(node) {
			// Find the container
			let element = node.parentElement;
			let prevElement = element;
			// Walk up the dom until we find a container element
			while (element) {
				if (isContainer(element)) {
					// Add flag to the previous non-container element that will render with children
					prevElement.setAttribute("data-has-notes", "true");
					break;
				}

				prevElement = element;
				element = element.parentElement;

				// If no containers were found and there are no further parents flag the last element
				if (!element) {
					prevElement.setAttribute("data-has-notes", "true");
				}
			}
		}

		renderNode(node) {
			if (node.nodeType == 1) {
				// Get all notes
				let notes;

				// Ingnore html element nodes, like mathml
				if (!node.dataset) {
					return;
				}

				if (node.dataset.note === "footnote") {
					notes = [node];
				} else if (node.dataset.hasNotes || node.querySelectorAll("[data-note='footnote']")) {
					notes = node.querySelectorAll("[data-note='footnote']");
				}

				if (notes && notes.length) {
					this.findVisibleFootnotes(notes, node);
				}
			}
		}

		findVisibleFootnotes(notes, node) {
			let area, size, right;
			area = node.closest(".pagedjs_page_content");
			size = area.getBoundingClientRect();
			right = size.left + size.width;

			for (let i = 0; i < notes.length; ++i) {
				let currentNote = notes[i];
				let bounds = currentNote.getBoundingClientRect();
				let left = bounds.left;

				if (left < right) {
					// Add call for the note
					this.moveFootnote(currentNote, node.closest(".pagedjs_area"), true);
				}
			}
		}

		moveFootnote(node, pageArea, needsNoteCall) {
			// let pageArea = node.closest(".pagedjs_area");
			let noteArea = pageArea.querySelector(".pagedjs_footnote_area");
			let noteContent = noteArea.querySelector(".pagedjs_footnote_content");
			let noteInnerContent = noteContent.querySelector(".pagedjs_footnote_inner_content");

			if (!isElement(node)) {
				return;
			}

			// Add call for the note
			let noteCall;
			if (needsNoteCall) {
				noteCall = this.createFootnoteCall(node);
			}

			// Remove the break before attribute for future layout
			node.removeAttribute("data-break-before");

			// Check if note already exists for overflow
			let existing = noteInnerContent.querySelector(`[data-ref="${node.dataset.ref}"]`);
			if (existing) {
				// Remove the note from the flow but no need to render it again
				node.remove();
				return;
			}

			// Add the note node
			noteInnerContent.appendChild(node);

			// Remove empty class
			if (noteContent.classList.contains("pagedjs_footnote_empty")) {
				noteContent.classList.remove("pagedjs_footnote_empty");
			}

			// Add marker
			node.dataset.footnoteMarker = node.dataset.ref;

			// Add Id
			node.id = `note-${node.dataset.ref}`;

			// Get note content size
			let height = noteContent.scrollHeight;

			// Check the noteCall is still on screen
			let area = pageArea.querySelector(".pagedjs_page_content");
			let size = area.getBoundingClientRect();
			let right = size.left + size.width;

			// TODO: add a max height in CSS

			// Check element sizes
			let noteCallBounds = noteCall && noteCall.getBoundingClientRect();
			let noteAreaBounds = noteArea.getBoundingClientRect();

			// Get the @footnote margins
			let noteContentMargins = this.marginsHeight(noteContent);
			let noteContentPadding = this.paddingHeight(noteContent);
			let noteContentBorders = this.borderHeight(noteContent);
			let total = noteContentMargins + noteContentPadding + noteContentBorders;

			// Get the top of the @footnote area
			let notAreaTop = Math.floor(noteAreaBounds.top);
			// If the height isn't set yet, remove the margins from the top
			if (noteAreaBounds.height === 0) {
				notAreaTop -= this.marginsHeight(noteContent, false);
				notAreaTop -= this.paddingHeight(noteContent, false);
				notAreaTop -= this.borderHeight(noteContent, false);
			}
			// Determine the note call position and offset per policy
			let notePolicy = node.dataset.notePolicy;
			let noteCallPosition = 0;
			let noteCallOffset = 0;
			if (noteCall) {
				// Get the correct line bottom for super or sub styled callouts
				let prevSibling = noteCall.previousSibling;
				let range = new Range();
				if (prevSibling) {
					range.setStartBefore(prevSibling);
				} else {
					range.setStartBefore(noteCall);
				}
				range.setEndAfter(noteCall);
				let rangeBounds = range.getBoundingClientRect();
				noteCallPosition = rangeBounds.bottom;
				if (!notePolicy || notePolicy === "auto") {
					noteCallOffset = Math.ceil(rangeBounds.bottom);
				} else if (notePolicy === "line") {
					noteCallOffset = Math.ceil(rangeBounds.top);
				} else if (notePolicy === "block") {
					// Check that there is a previous element on the page
					let parentParagraph = noteCall.closest("p").previousElementSibling;
					if (parentParagraph) {
						noteCallOffset = Math.ceil(
							parentParagraph.getBoundingClientRect().bottom
						);
					} else {
						noteCallOffset = Math.ceil(rangeBounds.bottom);
					}
				}
			}

			let contentDelta = height + total - noteAreaBounds.height;
			// Space between the top of the footnotes area and the bottom of the footnote call
			let noteDelta = noteCallPosition ? notAreaTop - noteCallPosition : 0;
			// Space needed for the force a break for the policy of the footnote
			let notePolicyDelta = noteCallPosition ? Math.floor(noteAreaBounds.top) - noteCallOffset : 0;
			let hasNotes = noteArea.querySelector("[data-note='footnote']");
			if (needsNoteCall && noteCallBounds.left > right) {
				// Note is offscreen and will be chunked to the next page on overflow
				node.remove();
			} else if (!hasNotes && needsNoteCall && total > noteDelta) {
				// No space to add even the footnote area
				pageArea.style.setProperty("--pagedjs-footnotes-height", "0px");
				// Add a wrapper as this div is removed later
				let wrapperDiv = document.createElement("div");
				wrapperDiv.appendChild(node);
				// Push to the layout queue for the next page
				this.needsLayout.push(wrapperDiv);
			} else if (!needsNoteCall) {
				// Call was previously added, force adding footnote
				pageArea.style.setProperty(
					"--pagedjs-footnotes-height",
					`${height + total}px`
				);
			} else if (noteCallPosition < noteAreaBounds.top - contentDelta) {
				// the current note content will fit without pushing the call to the next page
				pageArea.style.setProperty(
					"--pagedjs-footnotes-height",
					`${height + noteContentMargins + noteContentBorders}px`
				);
			} else {
				// set height to just before note call
				pageArea.style.setProperty(
					"--pagedjs-footnotes-height",
					`${noteAreaBounds.height + notePolicyDelta}px`
				);
				noteInnerContent.style.height =
					noteAreaBounds.height + notePolicyDelta - total + "px";
			}
		}

		createFootnoteCall(node) {
			let parentElement = node.parentElement;
			let footnoteCall = document.createElement("a");
			for (const className of node.classList) {
				footnoteCall.classList.add(`${className}`);
			}

			footnoteCall.dataset.footnoteCall = node.dataset.ref;
			footnoteCall.dataset.ref = node.dataset.ref;

			// Increment for counters
			footnoteCall.dataset.dataCounterFootnoteIncrement = 1;

			// Add link
			footnoteCall.href = `#note-${node.dataset.ref}`;

			parentElement.insertBefore(footnoteCall, node);

			return footnoteCall;
		}

		afterPageLayout(pageElement, page, breakToken, chunker) {
			let pageArea = pageElement.querySelector(".pagedjs_area");
			let noteArea = page.footnotesArea;
			let noteContent = noteArea.querySelector(".pagedjs_footnote_content");
			let noteInnerContent = noteArea.querySelector(".pagedjs_footnote_inner_content");

			let noteContentBounds = noteContent.getBoundingClientRect();
			let { width } = noteContentBounds;

			noteInnerContent.style.columnWidth = Math.round(width) + "px";
			noteInnerContent.style.columnGap = "calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left))";

			// Get overflow
			let layout = new Layout(noteArea, undefined, chunker.settings);
			let overflow = layout.findOverflow(noteInnerContent, noteContentBounds);

			if (overflow) {
				let { startContainer, startOffset } = overflow;
				let startIsNode;
				if (isElement(startContainer)) {
					let start = startContainer.childNodes[startOffset];
					startIsNode = isElement(start) && start.hasAttribute("data-footnote-marker");
				}

				let extracted = overflow.extractContents();

				if (!startIsNode) {
					let splitChild = extracted.firstElementChild;
					splitChild.dataset.splitFrom = splitChild.dataset.ref;

					this.handleAlignment(noteInnerContent.lastElementChild);
				}

				this.needsLayout.push(extracted);

				noteContent.style.removeProperty("height");
				noteInnerContent.style.removeProperty("height");

				let noteInnerContentBounds = noteInnerContent.getBoundingClientRect();
				let { height } = noteInnerContentBounds;

				// Get the @footnote margins
				let noteContentMargins = this.marginsHeight(noteContent);
				let noteContentPadding = this.paddingHeight(noteContent);
				let noteContentBorders = this.borderHeight(noteContent);
				pageArea.style.setProperty(
					"--pagedjs-footnotes-height",
					`${height + noteContentMargins + noteContentBorders + noteContentPadding}px`
				);

				// Hide footnote content if empty
				if (noteInnerContent.childNodes.length === 0) {
					noteContent.classList.add("pagedjs_footnote_empty");
				}

				if (!breakToken) {
					chunker.clonePage(page);
				} else {
					let breakBefore, previousBreakAfter;
					if (
						breakToken.node &&
						typeof breakToken.node.dataset !== "undefined" &&
						typeof breakToken.node.dataset.previousBreakAfter !== "undefined"
					) {
						previousBreakAfter = breakToken.node.dataset.previousBreakAfter;
					}

					if (
						breakToken.node &&
						typeof breakToken.node.dataset !== "undefined" &&
						typeof breakToken.node.dataset.breakBefore !== "undefined"
					) {
						breakBefore = breakToken.node.dataset.breakBefore;
					}

					if (breakBefore || previousBreakAfter) {
						chunker.clonePage(page);
					}
				}
			}
			noteInnerContent.style.height = "auto";
		}

		handleAlignment(node) {
			let styles = window.getComputedStyle(node);
			let alignLast = styles["text-align-last"];
			node.dataset.lastSplitElement = "true";
			if (alignLast === "auto") {
				node.dataset.alignLastSplitElement = "justify";
			} else {
				node.dataset.alignLastSplitElement = alignLast;
			}
		}

		beforePageLayout(page) {
			while (this.needsLayout.length) {
				let fragment = this.needsLayout.shift();

				Array.from(fragment.childNodes).forEach((node) => {
					this.moveFootnote(
						node,
						page.element.querySelector(".pagedjs_area"),
						false
					);
				});
			}
		}

		afterOverflowRemoved(removed, rendered) {
			// Find the page area
			let area = rendered.closest(".pagedjs_area");
			// Get any rendered footnotes
			let notes = area.querySelectorAll(".pagedjs_footnote_area [data-note='footnote']");
			for (let n = 0; n < notes.length; n++) {
				const note = notes[n];
				// Check if the call for that footnote has been removed with the overflow
				let call = removed.querySelector(`[data-footnote-call="${note.dataset.ref}"]`);
				if (call) {
					note.remove();
				}
			}
			// Hide footnote content if empty
			let noteInnerContent = area.querySelector(".pagedjs_footnote_inner_content");
			if (noteInnerContent && noteInnerContent.childNodes.length === 0) {
				noteInnerContent.parentElement.classList.add("pagedjs_footnote_empty");
			}
		}

		marginsHeight(element, total=true) {
			let styles = window.getComputedStyle(element);
			let marginTop = parseInt(styles.marginTop);
			let marginBottom = parseInt(styles.marginBottom);
			let margin = 0;
			if (marginTop) {
				margin += marginTop;
			}
			if (marginBottom && total) {
				margin += marginBottom;
			}
			return margin;
		}

		paddingHeight(element, total=true) {
			let styles = window.getComputedStyle(element);
			let paddingTop = parseInt(styles.paddingTop);
			let paddingBottom = parseInt(styles.paddingBottom);
			let padding = 0;
			if (paddingTop) {
				padding += paddingTop;
			}
			if (paddingBottom && total) {
				padding += paddingBottom;
			}
			return padding;
		}

		borderHeight(element, total=true) {
			let styles = window.getComputedStyle(element);
			let borderTop = parseInt(styles.borderTop);
			let borderBottom = parseInt(styles.borderBottom);
			let borders = 0;
			if (borderTop) {
				borders += borderTop;
			}
			if (borderBottom && total) {
				borders += borderBottom;
			}
			return borders;
		}
	}

	var pagedMediaHandlers = [
		PrintMedia,
		AtPage,
		Breaks,
		Splits,
		Counters,
		Lists,
		PositionFixed,
		PageCounterIncrement,
		NthOfType,
		Following,
		Footnotes
	];

	class RunningHeaders extends Handler {
		constructor(chunker, polisher, caller) {
			super(chunker, polisher, caller);

			this.runningSelectors = {};
			this.elements = {};
		}

		onDeclaration(declaration, dItem, dList, rule) {
			if (declaration.property === "position") {
				let selector = csstree.generate(rule.ruleNode.prelude);
				let identifier = declaration.value.children.first().name;

				if (identifier === "running") {
					let value;
					csstree.walk(declaration, {
						visit: "Function",
						enter: (node, item, list) => {
							value = node.children.first().name;
						}
					});

					this.runningSelectors[value] = {
						identifier: identifier,
						value: value,
						selector: selector
					};
				}
			}

			if (declaration.property === "content") {

				csstree.walk(declaration, {
					visit: "Function",
					enter: (funcNode, fItem, fList) => {

						if (funcNode.name.indexOf("element") > -1) {

							let selector = csstree.generate(rule.ruleNode.prelude);

							let func = funcNode.name;

							let value = funcNode.children.first().name;

							let args = [value];

							// we only handle first for now
							let style = "first";

							selector.split(",").forEach((s) => {
								// remove before / after
								s = s.replace(/::after|::before/, "");

								this.elements[s] = {
									func: func,
									args: args,
									value: value,
									style: style ,
									selector: s,
									fullSelector: selector
								};
							});
						}

					}
				});
			}
		}

		afterParsed(fragment) {
			for (let name of Object.keys(this.runningSelectors)) {
				let set = this.runningSelectors[name];
				let selected = Array.from(fragment.querySelectorAll(set.selector));

				if (set.identifier === "running") {
					for (let header of selected) {
						header.style.display = "none";
					}
				}

			}
		}

		afterPageLayout(fragment) {
			for (let name of Object.keys(this.runningSelectors)) {
				let set = this.runningSelectors[name];
				let selected = fragment.querySelector(set.selector);
				if (selected) {
					// let cssVar;
					if (set.identifier === "running") {
						// cssVar = selected.textContent.replace(/\\([\s\S])|(["|'])/g,"\\$1$2");
						// this.styleSheet.insertRule(`:root { --string-${name}: "${cssVar}"; }`, this.styleSheet.cssRules.length);
						// fragment.style.setProperty(`--string-${name}`, `"${cssVar}"`);
						set.first = selected;
					} else {
						console.warn(set.value + "needs css replacement");
					}
				}
			}

			// move elements
			if (!this.orderedSelectors) {
				this.orderedSelectors = this.orderSelectors(this.elements);
			}

			for (let selector of this.orderedSelectors) {
				if (selector) {

					let el = this.elements[selector];
					let selected = fragment.querySelector(selector);
					if (selected) {
						let running = this.runningSelectors[el.args[0]];
						if (running && running.first) {
							selected.innerHTML = ""; // Clear node
							// selected.classList.add("pagedjs_clear-after"); // Clear ::after
							let clone = running.first.cloneNode(true);
							clone.style.display = null;
							selected.appendChild(clone);
						}
					}
				}
			}
		}

		/**
		* Assign a weight to @page selector classes
		* 1) page
		* 2) left & right
		* 3) blank
		* 4) first & nth
		* 5) named page
		* 6) named left & right
		* 7) named first & nth
		* @param {string} [s] selector string
		* @return {int} weight
		*/
		pageWeight(s) {
			let weight = 1;
			let selector = s.split(" ");
			let parts = selector.length && selector[0].split(".");

			parts.shift(); // remove empty first part

			switch (parts.length) {
				case 4:
					if (/^pagedjs_[\w-]+_first_page$/.test(parts[3])) {
						weight = 7;
					} else if (parts[3] === "pagedjs_left_page" || parts[3] === "pagedjs_right_page") {
						weight = 6;
					}
					break;
				case 3:
					if (parts[1] === "pagedjs_named_page") {
						if (parts[2].indexOf(":nth-of-type") > -1) {
							weight = 7;
						} else {
							weight = 5;
						}
					}
					break;
				case 2:
					if (parts[1] === "pagedjs_first_page") {
						weight = 4;
					} else if (parts[1] === "pagedjs_blank_page") {
						weight = 3;
					} else if (parts[1] === "pagedjs_left_page" || parts[1] === "pagedjs_right_page") {
						weight = 2;
					}
					break;
				default:
					if (parts[0].indexOf(":nth-of-type") > -1) {
						weight = 4;
					} else {
						weight = 1;
					}
			}

			return weight;
		}

		/**
		* Orders the selectors based on weight
		*
		* Does not try to deduplicate base on specifity of the selector
		* Previous matched selector will just be overwritten
		* @param {obj} [obj] selectors object
		* @return {Array} orderedSelectors
		*/
		orderSelectors(obj) {
			let selectors = Object.keys(obj);
			let weighted = {
				1: [],
				2: [],
				3: [],
				4: [],
				5: [],
				6: [],
				7: []
			};

			let orderedSelectors = [];

			for (let s of selectors) {
				let w = this.pageWeight(s);
				weighted[w].unshift(s);
			}

			for (var i = 1; i <= 7; i++) {
				orderedSelectors = orderedSelectors.concat(weighted[i]);
			}

			return orderedSelectors;
		}

		beforeTreeParse(text, sheet) {
			// element(x) is parsed as image element selector, so update element to element-ident
			sheet.text = text.replace(/element[\s]*\(([^|^#)]*)\)/g, "element-ident($1)");
		}
	}

	function cleanPseudoContent(el, trim = "\"' ") {
		if(el == null) return;
		return el
			.replace(new RegExp(`^[${trim}]+`), "")
			.replace(new RegExp(`[${trim}]+$`), "")
			.replace(/["']/g, match => {
				return "\\" + match;
			})
			.replace(/[\n]/g, match => {
				return "\\00000A";
			});
	}

	function cleanSelector(el) {
		if(el == null) return;
		return el
			.replace(new RegExp("::footnote-call", "g"), "")
			.replace(new RegExp("::footnote-marker", "g"), "");
	}

	class StringSets extends Handler {
		constructor(chunker, polisher, caller) {
			super(chunker, polisher, caller);

			this.stringSetSelectors = {};
			this.type;
			// pageLastString = last string variable defined on the page
			this.pageLastString;

		}
		
		onDeclaration(declaration, dItem, dList, rule) {
			if (declaration.property === "string-set") {
				let selector = csstree.generate(rule.ruleNode.prelude);

				let identifiers = [];
				let functions = [];
				let values = [];

				declaration.value.children.forEach((child) => {
					if (child.type === "Identifier") {
						identifiers.push(child.name);
					}
					if (child.type === "Function") {
						functions.push(child.name);
						child.children.forEach((subchild) => {
							if (subchild.type === "Identifier") {
								values.push(subchild.name);
							}
						});
					}
				});

				identifiers.forEach((identifier, index) => {
					let func = functions[index];
					let value = values[index];
					this.stringSetSelectors[identifier] = {
						identifier,
						func,
						value,
						selector
					};
				});

			}
		}

		onContent(funcNode, fItem, fList, declaration, rule) {

			if (funcNode.name === "string") {
				let identifier = funcNode.children && funcNode.children.first().name;
				this.type = funcNode.children.last().name;
				funcNode.name = "var";
				funcNode.children = new csstree.List();

	 
				if(this.type === "first" || this.type === "last" || this.type === "start" || this.type === "first-except"){
					funcNode.children.append(
						funcNode.children.createItem({
							type: "Identifier",
							loc: null,
							name: "--pagedjs-string-" + this.type + "-" + identifier
						})
					);
				}else {
					funcNode.children.append(
						funcNode.children.createItem({
							type: "Identifier",
							loc: null,
							name: "--pagedjs-string-first-" + identifier
						})
					);
				}
			}
		}

		afterPageLayout(fragment) {

		
			if ( this.pageLastString === undefined )
			{
				this.pageLastString = {};
			}

			
			for (let name of Object.keys(this.stringSetSelectors)) {
		
				let set = this.stringSetSelectors[name];
				let value = set.value;
				let func = set.func;
				let selected = fragment.querySelectorAll(set.selector);

				// Get the last found string for the current identifier
				let stringPrevPage = ( name in this.pageLastString ) ? this.pageLastString[name] : "";

				let varFirst, varLast, varStart, varFirstExcept;

				if(selected.length == 0){
					// if there is no sel. on the page
					varFirst = stringPrevPage;
					varLast = stringPrevPage;
					varStart = stringPrevPage;
					varFirstExcept = stringPrevPage;
				}else {

					selected.forEach((sel) => {
						// push each content into the array to define in the variable the first and the last element of the page.
						if (func === "content") {
							this.pageLastString[name] = selected[selected.length - 1].textContent;
						}

						if (func === "attr") {
							this.pageLastString[name] = selected[selected.length - 1].getAttribute(value) || "";
						}

					});	

					/* FIRST */
		
					if (func === "content") {
						varFirst = selected[0].textContent;
					}

					if (func === "attr") {
						varFirst = selected[0].getAttribute(value) || "";
					}


					/* LAST */

					if (func === "content") {
						varLast = selected[selected.length - 1].textContent;
					}

					if (func === "attr") {
						varLast = selected[selected.length - 1].getAttribute(value) || "";
					}


					/* START */

					// Hack to find if the sel. is the first elem of the page / find a better way 
					let selTop = selected[0].getBoundingClientRect().top;
					let pageContent = selected[0].closest(".pagedjs_page_content");
					let pageContentTop = pageContent.getBoundingClientRect().top;

					if(selTop == pageContentTop){
						varStart = varFirst;
					}else {
						varStart = stringPrevPage;
					}

					/* FIRST EXCEPT */

					varFirstExcept = "";
					
				}

				fragment.style.setProperty(`--pagedjs-string-first-${name}`, `"${cleanPseudoContent(varFirst)}`);
				fragment.style.setProperty(`--pagedjs-string-last-${name}`, `"${cleanPseudoContent(varLast)}`);
				fragment.style.setProperty(`--pagedjs-string-start-${name}`, `"${cleanPseudoContent(varStart)}`);
				fragment.style.setProperty(`--pagedjs-string-first-except-${name}`, `"${cleanPseudoContent(varFirstExcept)}`);
				
		
			}
		}
		

	}

	class TargetCounters extends Handler {
		constructor(chunker, polisher, caller) {
			super(chunker, polisher, caller);

			this.styleSheet = polisher.styleSheet;

			this.counterTargets = {};
		}

		onContent(funcNode, fItem, fList, declaration, rule) {
			if (funcNode.name === "target-counter") {
				let selector = csstree.generate(rule.ruleNode.prelude);

				let first = funcNode.children.first();
				let func = first.name;

				let value = csstree.generate(funcNode);

				let args = [];

				first.children.forEach((child) => {
					if (child.type === "Identifier") {

						args.push(child.name);
					}
				});

				let counter;
				let style;
				let styleIdentifier;

				funcNode.children.forEach((child) => {
					if (child.type === "Identifier") {
						if (!counter) {
							counter = child.name;
						} else if (!style) {
							styleIdentifier = csstree.clone(child);
							style = child.name;
						}
					}
				});

				let variable = "target-counter-" + UUID();

				selector.split(",").forEach((s) => {
					this.counterTargets[s] = {
						func: func,
						args: args,
						value: value,
						counter: counter,
						style: style,
						selector: s,
						fullSelector: selector,
						variable: variable
					};
				});

				// Replace with counter
				funcNode.name = "counter";
				funcNode.children = new csstree.List();
				funcNode.children.appendData({
					type: "Identifier",
					loc: 0,
					name: variable
				});

				if (styleIdentifier) {
					funcNode.children.appendData({type: "Operator", loc: null, value: ","});
					funcNode.children.appendData(styleIdentifier);
				}
			}
		}

		afterPageLayout(fragment, page, breakToken, chunker) {
			Object.keys(this.counterTargets).forEach((name) => {
				let target = this.counterTargets[name];
				let split = target.selector.split(/::?/g);
				let query = split[0];

				let queried = chunker.pagesArea.querySelectorAll(query + ":not([data-" + target.variable + "])");

				queried.forEach((selected, index) => {
					// TODO: handle func other than attr
					if (target.func !== "attr") {
						return;
					}
					let val = attr(selected, target.args);
					let element = chunker.pagesArea.querySelector(querySelectorEscape(val));

					if (element) {
						let selector = UUID();
						selected.setAttribute("data-" + target.variable, selector);
						// TODO: handle other counter types (by query)
						let pseudo = "";
						if (split.length > 1) {
							pseudo += "::" + split[1];
						}
						if (target.counter === "page") {
							let pages = chunker.pagesArea.querySelectorAll(".pagedjs_page");
							let pg = 0;
							for (let i = 0; i < pages.length; i++) {
								let page = pages[i];
								let styles = window.getComputedStyle(page);
								let reset = styles["counter-reset"].replace("page", "").trim();
								let increment = styles["counter-increment"].replace("page", "").trim();

								if (reset !== "none") {
									pg = parseInt(reset);
								}
								if (increment !== "none") {
									pg += parseInt(increment);
								}

								if (page.contains(element)){
									break;
								}
							}
							this.styleSheet.insertRule(`[data-${target.variable}="${selector}"]${pseudo} { counter-reset: ${target.variable} ${pg}; }`, this.styleSheet.cssRules.length);
						} else {
							let value = element.getAttribute(`data-counter-${target.counter}-value`);
							if (value) {
								this.styleSheet.insertRule(`[data-${target.variable}="${selector}"]${pseudo} { counter-reset: ${target.variable} ${target.variable} ${parseInt(value)}; }`, this.styleSheet.cssRules.length);
							}
						}

						// force redraw
						let el = document.querySelector(`[data-${target.variable}="${selector}"]`);
						if (el) {
							el.style.display = "none";
							el.clientHeight;
							el.style.removeProperty("display");
						}
					}
				});
			});
		}
	}

	// import { nodeAfter } from "../../utils/dom";

	class TargetText extends Handler {
		constructor(chunker, polisher, caller) {
			super(chunker, polisher, caller);

			this.styleSheet = polisher.styleSheet;
			this.textTargets = {};
			this.beforeContent = "";
			this.afterContent = "";
			this.selector = {};
		}

		onContent(funcNode, fItem, fList, declaration, rule) {
			if (funcNode.name === "target-text") {
				this.selector = csstree.generate(rule.ruleNode.prelude);
				let first = funcNode.children.first();
				let last = funcNode.children.last();
				let func = first.name;

				let value = csstree.generate(funcNode);

				let args = [];

				first.children.forEach(child => {
					if (child.type === "Identifier") {
						args.push(child.name);
					}
				});

				let style;
				if (last !== first) {
					style = last.name;
				}

				let variable = "--pagedjs-" + UUID();

				this.selector.split(",").forEach(s => {
					this.textTargets[s] = {
						func: func,
						args: args,
						value: value,
						style: style || "content",
						selector: s,
						fullSelector: this.selector,
						variable: variable
					};
				});

				// Replace with variable
				funcNode.name = "var";
				funcNode.children = new csstree.List();
				funcNode.children.appendData({
					type: "Identifier",
					loc: 0,
					name: variable
				});
			}
		}

		//   parse this on the ONCONTENT : get all before and after and replace the value with a variable
		onPseudoSelector(pseudoNode, pItem, pList, selector, rule) {
			// console.log(pseudoNode);
			// console.log(rule);

			rule.ruleNode.block.children.forEach(properties => {
				if (pseudoNode.name === "before" && properties.property === "content") {
					// let beforeVariable = "--pagedjs-" + UUID();

					let contenu = properties.value.children;
					contenu.forEach(prop => {
						if (prop.type === "String") {
							this.beforeContent = prop.value;
						}
					});
				} else if (pseudoNode.name === "after" && properties.property === "content") {
					properties.value.children.forEach(prop => {
						if (prop.type === "String") {
							this.afterContent = prop.value;
						}
					});
				}
			});
		}

		afterParsed(fragment) {
			Object.keys(this.textTargets).forEach(name => {
				let target = this.textTargets[name];
				let split = target.selector.split("::");
				let query = split[0];
				let queried = fragment.querySelectorAll(query);
				let textContent;
				queried.forEach((selected, index) => {
					let val = attr(selected, target.args);
					let element = fragment.querySelector(querySelectorEscape(val));
					if (element) {
						// content & first-letter & before & after refactorized
						if (target.style) {
							this.selector = UUID();
							selected.setAttribute("data-target-text", this.selector);

							let psuedo = "";
							if (split.length > 1) {
								psuedo += "::" + split[1];
							}
							
							if (target.style === "before" || target.style === "after") {
								const pseudoType = `${target.style}Content`;
								textContent = cleanPseudoContent(this[pseudoType]);
							} else {
								textContent = cleanPseudoContent(element.textContent, " ");
							}
							textContent = target.style === "first-letter" ? textContent.charAt(0) : textContent;
							this.styleSheet.insertRule(`[data-target-text="${this.selector}"]${psuedo} { ${target.variable}: "${textContent}" }`);
						} else {
							console.warn("missed target", val);
						}
					}
				});
			});
		}
	}

	var generatedContentHandlers = [
		RunningHeaders,
		StringSets,
		TargetCounters,
		TargetText
	];

	class WhiteSpaceFilter extends Handler {
		constructor(chunker, polisher, caller) {
			super(chunker, polisher, caller);
		}

		filter(content) {

			filterTree(content, (node) => {
				return this.filterEmpty(node);
			}, NodeFilter.SHOW_TEXT);

		}

		filterEmpty(node) {
			if (node.textContent.length > 1 && isIgnorable(node)) {

				// Do not touch the content if text is pre-formatted
				let parent = node.parentNode;
				let pre = isElement(parent) && parent.closest("pre");
				if (pre) {
					return NodeFilter.FILTER_REJECT;
				}

				const previousSibling = previousSignificantNode(node);
				const nextSibling = nextSignificantNode(node);

				if (nextSibling === null && previousSibling === null) {
					// we should not remove a Node that does not have any siblings.
					node.textContent = " ";
					return NodeFilter.FILTER_REJECT;
				}
				if (nextSibling === null) {
					// we can safely remove this node
					return NodeFilter.FILTER_ACCEPT;
				}
				if (previousSibling === null) {
					// we can safely remove this node
					return NodeFilter.FILTER_ACCEPT;
				}

				// replace the content with a single space
				node.textContent = " ";

				// TODO: we also need to preserve sequences of white spaces when the parent has "white-space" rule:
				// pre
				// Sequences of white space are preserved. Lines are only broken at newline characters in the source and at <br> elements.
				//
				// pre-wrap
				// Sequences of white space are preserved. Lines are broken at newline characters, at <br>, and as necessary to fill line boxes.
				//
				// pre-line
				// Sequences of white space are collapsed. Lines are broken at newline characters, at <br>, and as necessary to fill line boxes.
				//
				// break-spaces
				// The behavior is identical to that of pre-wrap, except that:
				// - Any sequence of preserved white space always takes up space, including at the end of the line.
				// - A line breaking opportunity exists after every preserved white space character, including between white space characters.
				// - Such preserved spaces take up space and do not hang, and thus affect the box’s intrinsic sizes (min-content size and max-content size).
				//
				// See: https://developer.mozilla.org/en-US/docs/Web/CSS/white-space#Values

				return NodeFilter.FILTER_REJECT;
			} else {
				return NodeFilter.FILTER_REJECT;
			}
		}

	}

	class CommentsFilter extends Handler {
		constructor(chunker, polisher, caller) {
			super(chunker, polisher, caller);
		}

		filter(content) {
			filterTree(content, null, NodeFilter.SHOW_COMMENT);
		}

	}

	class ScriptsFilter extends Handler {
		constructor(chunker, polisher, caller) {
			super(chunker, polisher, caller);
		}

		filter(content) {
			content.querySelectorAll("script").forEach( script => { script.remove(); });
		}

	}

	var clearCut = {};

	/**
	 * Originally ported from https://github.com/keeganstreet/specificity/blob/866bf7ab4e7f62a7179c15b13a95af4e1c7b1afa/specificity.js
	 *
	 * Calculates the specificity of CSS selectors
	 * http://www.w3.org/TR/css3-selectors/#specificity
	 *
	 * Returns a selector integer value
	 */

	(function (exports) {
		// The following regular expressions assume that selectors matching the preceding regular expressions have been removed
		var attributeRegex = /(\[[^\]]+\])/g;
		var idRegex = /(#[^\s\+>~\.\[:]+)/g;
		var classRegex = /(\.[^\s\+>~\.\[:]+)/g;
		var pseudoElementRegex = /(::[^\s\+>~\.\[:]+|:first-line|:first-letter|:before|:after)/g;
		var pseudoClassRegex = /(:[^\s\+>~\.\[:]+)/g;
		var elementRegex = /([^\s\+>~\.\[:]+)/g;
		var notRegex = /:not\(([^\)]*)\)/g;
		var ruleRegex = /\{[^]*/gm;
		var separatorRegex = /[\*\s\+>~]/g;
		var straysRegex = /[#\.]/g;

		// Find matches for a regular expression in a string and push their details to parts
		// Type is "a" for IDs, "b" for classes, attributes and pseudo-classes and "c" for elements and pseudo-elements
		var findMatch = function(regex, type, types, selector) {
		  var matches = selector.match(regex);
		  if (matches) {
		    for (var i = 0; i < matches.length; i++) {
		      types[type]++;
		      // Replace this simple selector with whitespace so it won't be counted in further simple selectors
		      selector = selector.replace(matches[i], ' ');
		    }
		  }

		  return selector;
		};

		// Calculate the specificity for a selector by dividing it into simple selectors and counting them
		var calculate = function(selector) {
		  var commaIndex = selector.indexOf(',');
		  if (commaIndex !== -1) {
		    selector = selector.substring(0, commaIndex);
		  }

		  var  types = {
		    a: 0,
		    b: 0,
		    c: 0
		  };

		  // Remove the negation psuedo-class (:not) but leave its argument because specificity is calculated on its argument
		  selector = selector.replace(notRegex, ' $1 ');

		  // Remove anything after a left brace in case a user has pasted in a rule, not just a selector
		  selector = selector.replace(ruleRegex, ' ');

		  // Add attribute selectors to parts collection (type b)
		  selector = findMatch(attributeRegex, 'b', types, selector);

		  // Add ID selectors to parts collection (type a)
		  selector = findMatch(idRegex, 'a', types, selector);

		  // Add class selectors to parts collection (type b)
		  selector = findMatch(classRegex, 'b', types, selector);

		  // Add pseudo-element selectors to parts collection (type c)
		  selector = findMatch(pseudoElementRegex, 'c', types, selector);

		  // Add pseudo-class selectors to parts collection (type b)
		  selector = findMatch(pseudoClassRegex, 'b', types, selector);

		  // Remove universal selector and separator characters
		  selector = selector.replace(separatorRegex, ' ');

		  // Remove any stray dots or hashes which aren't attached to words
		  // These may be present if the user is live-editing this selector
		  selector = selector.replace(straysRegex, ' ');

		  // The only things left should be element selectors (type c)
		  findMatch(elementRegex, 'c', types, selector);

		  return (types.a * 100) + (types.b * 10) + (types.c * 1);
		};

		var specificityCache = {};

		exports.calculateSpecificity = function(selector) {
		  var specificity = specificityCache[selector];
		  if (specificity === undefined) {
		    specificity = calculate(selector);
		    specificityCache[selector] = specificity;
		  }
		  return specificity;
		};

		var validSelectorCache = {};
		var testSelectorElement = null;

		exports.isSelectorValid = function(selector) {
		  var valid = validSelectorCache[selector];
		  if (valid === undefined) {
		    if (testSelectorElement == null) {
		      testSelectorElement = document.createElement('div');
		    }

		    try {
		      testSelectorElement.querySelector(selector);
		      valid = true;
		    } catch (error) {
		      valid = false;
		    }
		    validSelectorCache[selector] = valid;
		  }
		  return valid;
		};

		exports.validateSelector = function(selector) {
		  if (!exports.isSelectorValid(selector)) {
		    var error = new SyntaxError(selector + ' is not a valid selector');
		    error.code = 'EBADSELECTOR';
		    throw error;
		  }
		}; 
	} (clearCut));

	class UndisplayedFilter extends Handler {
		constructor(chunker, polisher, caller) {
			super(chunker, polisher, caller);
			this.displayRules = {};
		}

		onDeclaration(declaration, dItem, dList, rule) {
			if (declaration.property === "display") {
				let selector = csstree.generate(rule.ruleNode.prelude);
				let value = declaration.value.children.first().name;

				selector.split(",").forEach((s) => {
					this.displayRules[s] = {
						value: value,
						selector: s,
						specificity: clearCut.calculateSpecificity(s),
						important: declaration.important
					};
				});
			}
		}

		filter(content) {
			let { matches, selectors } = this.sortDisplayedSelectors(content, this.displayRules);

			// Find matching elements that have display styles
			for (let i = 0; i < matches.length; i++) {
				let element = matches[i];
				let selector = selectors[i];
				let displayValue = selector[selector.length-1].value;
				if(this.removable(element) && displayValue === "none") {
					element.dataset.undisplayed = "undisplayed";
				}
			}

			// Find elements that have inline styles
			let styledElements = content.querySelectorAll("[style]");
			for (let i = 0; i < styledElements.length; i++) {
				let element = styledElements[i];
				if (this.removable(element)) {
					element.dataset.undisplayed = "undisplayed";
				}
			}
		}

		sorter(a, b) {
			if (a.important && !b.important) {
				return 1;
			}

			if (b.important && !a.important) {
				return -1;
			}

			return a.specificity - b.specificity;
		}

		sortDisplayedSelectors(content, displayRules=[]) {
			let matches = [];
			let selectors = [];
			for (let d in displayRules) {
				let displayItem = displayRules[d];
				let selector = displayItem.selector;
				let query = [];
				try {
					try {
						query = content.querySelectorAll(selector);
					} catch (e) {
						query = content.querySelectorAll(cleanSelector(selector));
					}
				} catch (e) {
					query = [];
				}
				let elements = Array.from(query);
				for (let e of elements) {
					if (matches.includes(e)) {
						let index = matches.indexOf(e);
						selectors[index].push(displayItem);
						selectors[index] = selectors[index].sort(this.sorter);
					} else {
						matches.push(e);
						selectors.push([displayItem]);
					}
				}
			}

			return { matches, selectors };
		}

		removable(element) {
			if (element.style &&
					element.style.display !== "" &&
					element.style.display !== "none") {
				return false;
			}

			return true;
		}
	}

	var filters = [
		WhiteSpaceFilter,
		CommentsFilter,
		ScriptsFilter,
		UndisplayedFilter
	];

	var isImplemented$4 = function () {
		var from = Array.from, arr, result;
		if (typeof from !== "function") return false;
		arr = ["raz", "dwa"];
		result = from(arr);
		return Boolean(result && result !== arr && result[1] === "dwa");
	};

	var isImplemented$3;
	var hasRequiredIsImplemented$1;

	function requireIsImplemented$1 () {
		if (hasRequiredIsImplemented$1) return isImplemented$3;
		hasRequiredIsImplemented$1 = 1;

		isImplemented$3 = function () {
			if (typeof globalThis !== "object") return false;
			if (!globalThis) return false;
			return globalThis.Array === Array;
		};
		return isImplemented$3;
	}

	var implementation;
	var hasRequiredImplementation;

	function requireImplementation () {
		if (hasRequiredImplementation) return implementation;
		hasRequiredImplementation = 1;
		var naiveFallback = function () {
			if (typeof self === "object" && self) return self;
			if (typeof window === "object" && window) return window;
			throw new Error("Unable to resolve global `this`");
		};

		implementation = (function () {
			if (this) return this;

			// Unexpected strict mode (may happen if e.g. bundled into ESM module)

			// Thanks @mathiasbynens -> https://mathiasbynens.be/notes/globalthis
			// In all ES5+ engines global object inherits from Object.prototype
			// (if you approached one that doesn't please report)
			try {
				Object.defineProperty(Object.prototype, "__global__", {
					get: function () { return this; },
					configurable: true
				});
			} catch (error) {
				// Unfortunate case of Object.prototype being sealed (via preventExtensions, seal or freeze)
				return naiveFallback();
			}
			try {
				// Safari case (window.__global__ is resolved with global context, but __global__ does not)
				if (!__global__) return naiveFallback();
				return __global__;
			} finally {
				delete Object.prototype.__global__;
			}
		})();
		return implementation;
	}

	var globalThis_1;
	var hasRequiredGlobalThis;

	function requireGlobalThis () {
		if (hasRequiredGlobalThis) return globalThis_1;
		hasRequiredGlobalThis = 1;

		globalThis_1 = requireIsImplemented$1()() ? globalThis : requireImplementation();
		return globalThis_1;
	}

	var isImplemented$2;
	var hasRequiredIsImplemented;

	function requireIsImplemented () {
		if (hasRequiredIsImplemented) return isImplemented$2;
		hasRequiredIsImplemented = 1;

		var global     = requireGlobalThis()
		  , validTypes = { object: true, symbol: true };

		isImplemented$2 = function () {
			var Symbol = global.Symbol;
			var symbol;
			if (typeof Symbol !== "function") return false;
			symbol = Symbol("test symbol");
			try { String(symbol); }
			catch (e) { return false; }

			// Return 'true' also for polyfills
			if (!validTypes[typeof Symbol.iterator]) return false;
			if (!validTypes[typeof Symbol.toPrimitive]) return false;
			if (!validTypes[typeof Symbol.toStringTag]) return false;

			return true;
		};
		return isImplemented$2;
	}

	var isSymbol;
	var hasRequiredIsSymbol;

	function requireIsSymbol () {
		if (hasRequiredIsSymbol) return isSymbol;
		hasRequiredIsSymbol = 1;

		isSymbol = function (value) {
			if (!value) return false;
			if (typeof value === "symbol") return true;
			if (!value.constructor) return false;
			if (value.constructor.name !== "Symbol") return false;
			return value[value.constructor.toStringTag] === "Symbol";
		};
		return isSymbol;
	}

	var validateSymbol;
	var hasRequiredValidateSymbol;

	function requireValidateSymbol () {
		if (hasRequiredValidateSymbol) return validateSymbol;
		hasRequiredValidateSymbol = 1;

		var isSymbol = requireIsSymbol();

		validateSymbol = function (value) {
			if (!isSymbol(value)) throw new TypeError(value + " is not a symbol");
			return value;
		};
		return validateSymbol;
	}

	var generateName;
	var hasRequiredGenerateName;

	function requireGenerateName () {
		if (hasRequiredGenerateName) return generateName;
		hasRequiredGenerateName = 1;

		var d = dExports;

		var create = Object.create, defineProperty = Object.defineProperty, objPrototype = Object.prototype;

		var created = create(null);
		generateName = function (desc) {
			var postfix = 0, name, ie11BugWorkaround;
			while (created[desc + (postfix || "")]) ++postfix;
			desc += postfix || "";
			created[desc] = true;
			name = "@@" + desc;
			defineProperty(
				objPrototype,
				name,
				d.gs(null, function (value) {
					// For IE11 issue see:
					// https://connect.microsoft.com/IE/feedbackdetail/view/1928508/
					//    ie11-broken-getters-on-dom-objects
					// https://github.com/medikoo/es6-symbol/issues/12
					if (ie11BugWorkaround) return;
					ie11BugWorkaround = true;
					defineProperty(this, name, d(value));
					ie11BugWorkaround = false;
				})
			);
			return name;
		};
		return generateName;
	}

	var standardSymbols;
	var hasRequiredStandardSymbols;

	function requireStandardSymbols () {
		if (hasRequiredStandardSymbols) return standardSymbols;
		hasRequiredStandardSymbols = 1;

		var d            = dExports
		  , NativeSymbol = requireGlobalThis().Symbol;

		standardSymbols = function (SymbolPolyfill) {
			return Object.defineProperties(SymbolPolyfill, {
				// To ensure proper interoperability with other native functions (e.g. Array.from)
				// fallback to eventual native implementation of given symbol
				hasInstance: d(
					"", (NativeSymbol && NativeSymbol.hasInstance) || SymbolPolyfill("hasInstance")
				),
				isConcatSpreadable: d(
					"",
					(NativeSymbol && NativeSymbol.isConcatSpreadable) ||
						SymbolPolyfill("isConcatSpreadable")
				),
				iterator: d("", (NativeSymbol && NativeSymbol.iterator) || SymbolPolyfill("iterator")),
				match: d("", (NativeSymbol && NativeSymbol.match) || SymbolPolyfill("match")),
				replace: d("", (NativeSymbol && NativeSymbol.replace) || SymbolPolyfill("replace")),
				search: d("", (NativeSymbol && NativeSymbol.search) || SymbolPolyfill("search")),
				species: d("", (NativeSymbol && NativeSymbol.species) || SymbolPolyfill("species")),
				split: d("", (NativeSymbol && NativeSymbol.split) || SymbolPolyfill("split")),
				toPrimitive: d(
					"", (NativeSymbol && NativeSymbol.toPrimitive) || SymbolPolyfill("toPrimitive")
				),
				toStringTag: d(
					"", (NativeSymbol && NativeSymbol.toStringTag) || SymbolPolyfill("toStringTag")
				),
				unscopables: d(
					"", (NativeSymbol && NativeSymbol.unscopables) || SymbolPolyfill("unscopables")
				)
			});
		};
		return standardSymbols;
	}

	var symbolRegistry;
	var hasRequiredSymbolRegistry;

	function requireSymbolRegistry () {
		if (hasRequiredSymbolRegistry) return symbolRegistry;
		hasRequiredSymbolRegistry = 1;

		var d              = dExports
		  , validateSymbol = requireValidateSymbol();

		var registry = Object.create(null);

		symbolRegistry = function (SymbolPolyfill) {
			return Object.defineProperties(SymbolPolyfill, {
				for: d(function (key) {
					if (registry[key]) return registry[key];
					return (registry[key] = SymbolPolyfill(String(key)));
				}),
				keyFor: d(function (symbol) {
					var key;
					validateSymbol(symbol);
					for (key in registry) {
						if (registry[key] === symbol) return key;
					}
					return undefined;
				})
			});
		};
		return symbolRegistry;
	}

	var polyfill;
	var hasRequiredPolyfill;

	function requirePolyfill () {
		if (hasRequiredPolyfill) return polyfill;
		hasRequiredPolyfill = 1;

		var d                    = dExports
		  , validateSymbol       = requireValidateSymbol()
		  , NativeSymbol         = requireGlobalThis().Symbol
		  , generateName         = requireGenerateName()
		  , setupStandardSymbols = requireStandardSymbols()
		  , setupSymbolRegistry  = requireSymbolRegistry();

		var create = Object.create
		  , defineProperties = Object.defineProperties
		  , defineProperty = Object.defineProperty;

		var SymbolPolyfill, HiddenSymbol, isNativeSafe;

		if (typeof NativeSymbol === "function") {
			try {
				String(NativeSymbol());
				isNativeSafe = true;
			} catch (ignore) {}
		} else {
			NativeSymbol = null;
		}

		// Internal constructor (not one exposed) for creating Symbol instances.
		// This one is used to ensure that `someSymbol instanceof Symbol` always return false
		HiddenSymbol = function Symbol(description) {
			if (this instanceof HiddenSymbol) throw new TypeError("Symbol is not a constructor");
			return SymbolPolyfill(description);
		};

		// Exposed `Symbol` constructor
		// (returns instances of HiddenSymbol)
		polyfill = SymbolPolyfill = function Symbol(description) {
			var symbol;
			if (this instanceof Symbol) throw new TypeError("Symbol is not a constructor");
			if (isNativeSafe) return NativeSymbol(description);
			symbol = create(HiddenSymbol.prototype);
			description = description === undefined ? "" : String(description);
			return defineProperties(symbol, {
				__description__: d("", description),
				__name__: d("", generateName(description))
			});
		};

		setupStandardSymbols(SymbolPolyfill);
		setupSymbolRegistry(SymbolPolyfill);

		// Internal tweaks for real symbol producer
		defineProperties(HiddenSymbol.prototype, {
			constructor: d(SymbolPolyfill),
			toString: d("", function () { return this.__name__; })
		});

		// Proper implementation of methods exposed on Symbol.prototype
		// They won't be accessible on produced symbol instances as they derive from HiddenSymbol.prototype
		defineProperties(SymbolPolyfill.prototype, {
			toString: d(function () { return "Symbol (" + validateSymbol(this).__description__ + ")"; }),
			valueOf: d(function () { return validateSymbol(this); })
		});
		defineProperty(
			SymbolPolyfill.prototype,
			SymbolPolyfill.toPrimitive,
			d("", function () {
				var symbol = validateSymbol(this);
				if (typeof symbol === "symbol") return symbol;
				return symbol.toString();
			})
		);
		defineProperty(SymbolPolyfill.prototype, SymbolPolyfill.toStringTag, d("c", "Symbol"));

		// Proper implementaton of toPrimitive and toStringTag for returned symbol instances
		defineProperty(
			HiddenSymbol.prototype, SymbolPolyfill.toStringTag,
			d("c", SymbolPolyfill.prototype[SymbolPolyfill.toStringTag])
		);

		// Note: It's important to define `toPrimitive` as last one, as some implementations
		// implement `toPrimitive` natively without implementing `toStringTag` (or other specified symbols)
		// And that may invoke error in definition flow:
		// See: https://github.com/medikoo/es6-symbol/issues/13#issuecomment-164146149
		defineProperty(
			HiddenSymbol.prototype, SymbolPolyfill.toPrimitive,
			d("c", SymbolPolyfill.prototype[SymbolPolyfill.toPrimitive])
		);
		return polyfill;
	}

	var es6Symbol;
	var hasRequiredEs6Symbol;

	function requireEs6Symbol () {
		if (hasRequiredEs6Symbol) return es6Symbol;
		hasRequiredEs6Symbol = 1;

		es6Symbol = requireIsImplemented()()
			? requireGlobalThis().Symbol
			: requirePolyfill();
		return es6Symbol;
	}

	var isArguments;
	var hasRequiredIsArguments;

	function requireIsArguments () {
		if (hasRequiredIsArguments) return isArguments;
		hasRequiredIsArguments = 1;

		var objToString = Object.prototype.toString
		  , id = objToString.call((function () { return arguments; })());

		isArguments = function (value) { return objToString.call(value) === id; };
		return isArguments;
	}

	var isFunction;
	var hasRequiredIsFunction;

	function requireIsFunction () {
		if (hasRequiredIsFunction) return isFunction;
		hasRequiredIsFunction = 1;

		var objToString = Object.prototype.toString
		  , isFunctionStringTag = RegExp.prototype.test.bind(/^[object [A-Za-z0-9]*Function]$/);

		isFunction = function (value) {
			return typeof value === "function" && isFunctionStringTag(objToString.call(value));
		};
		return isFunction;
	}

	var isImplemented$1 = function () {
		var sign = Math.sign;
		if (typeof sign !== "function") return false;
		return sign(10) === 1 && sign(-20) === -1;
	};

	var shim$2;
	var hasRequiredShim$2;

	function requireShim$2 () {
		if (hasRequiredShim$2) return shim$2;
		hasRequiredShim$2 = 1;

		shim$2 = function (value) {
			value = Number(value);
			if (isNaN(value) || value === 0) return value;
			return value > 0 ? 1 : -1;
		};
		return shim$2;
	}

	var sign$1 = isImplemented$1() ? Math.sign : requireShim$2();

	var sign  = sign$1
	  , abs$1   = Math.abs
	  , floor$1 = Math.floor;

	var toInteger$1 = function (value) {
		if (isNaN(value)) return 0;
		value = Number(value);
		if (value === 0 || !isFinite(value)) return value;
		return sign(value) * floor$1(abs$1(value));
	};

	var toInteger = toInteger$1
	  , max       = Math.max;

	var toPosInteger = function (value) { return max(0, toInteger(value)); };

	var isString;
	var hasRequiredIsString;

	function requireIsString () {
		if (hasRequiredIsString) return isString;
		hasRequiredIsString = 1;

		var objToString = Object.prototype.toString, id = objToString.call("");

		isString = function (value) {
			return (
				typeof value === "string" ||
				(value &&
					typeof value === "object" &&
					(value instanceof String || objToString.call(value) === id)) ||
				false
			);
		};
		return isString;
	}

	var shim$1;
	var hasRequiredShim$1;

	function requireShim$1 () {
		if (hasRequiredShim$1) return shim$1;
		hasRequiredShim$1 = 1;

		var iteratorSymbol = requireEs6Symbol().iterator
		  , isArguments    = requireIsArguments()
		  , isFunction     = requireIsFunction()
		  , toPosInt       = toPosInteger
		  , callable       = validCallable
		  , validValue$1     = validValue
		  , isValue        = isValue$4
		  , isString       = requireIsString()
		  , isArray        = Array.isArray
		  , call           = Function.prototype.call
		  , desc           = { configurable: true, enumerable: true, writable: true, value: null }
		  , defineProperty = Object.defineProperty;

		// eslint-disable-next-line complexity, max-lines-per-function
		shim$1 = function (arrayLike /*, mapFn, thisArg*/) {
			var mapFn = arguments[1]
			  , thisArg = arguments[2]
			  , Context
			  , i
			  , j
			  , arr
			  , length
			  , code
			  , iterator
			  , result
			  , getIterator
			  , value;

			arrayLike = Object(validValue$1(arrayLike));

			if (isValue(mapFn)) callable(mapFn);
			if (!this || this === Array || !isFunction(this)) {
				// Result: Plain array
				if (!mapFn) {
					if (isArguments(arrayLike)) {
						// Source: Arguments
						length = arrayLike.length;
						if (length !== 1) return Array.apply(null, arrayLike);
						arr = new Array(1);
						arr[0] = arrayLike[0];
						return arr;
					}
					if (isArray(arrayLike)) {
						// Source: Array
						arr = new Array((length = arrayLike.length));
						for (i = 0; i < length; ++i) arr[i] = arrayLike[i];
						return arr;
					}
				}
				arr = [];
			} else {
				// Result: Non plain array
				Context = this;
			}

			if (!isArray(arrayLike)) {
				if ((getIterator = arrayLike[iteratorSymbol]) !== undefined) {
					// Source: Iterator
					iterator = callable(getIterator).call(arrayLike);
					if (Context) arr = new Context();
					result = iterator.next();
					i = 0;
					while (!result.done) {
						value = mapFn ? call.call(mapFn, thisArg, result.value, i) : result.value;
						if (Context) {
							desc.value = value;
							defineProperty(arr, i, desc);
						} else {
							arr[i] = value;
						}
						result = iterator.next();
						++i;
					}
					length = i;
				} else if (isString(arrayLike)) {
					// Source: String
					length = arrayLike.length;
					if (Context) arr = new Context();
					for (i = 0, j = 0; i < length; ++i) {
						value = arrayLike[i];
						if (i + 1 < length) {
							code = value.charCodeAt(0);
							// eslint-disable-next-line max-depth
							if (code >= 0xd800 && code <= 0xdbff) value += arrayLike[++i];
						}
						value = mapFn ? call.call(mapFn, thisArg, value, j) : value;
						if (Context) {
							desc.value = value;
							defineProperty(arr, j, desc);
						} else {
							arr[j] = value;
						}
						++j;
					}
					length = j;
				}
			}
			if (length === undefined) {
				// Source: array or array-like
				length = toPosInt(arrayLike.length);
				if (Context) arr = new Context(length);
				for (i = 0; i < length; ++i) {
					value = mapFn ? call.call(mapFn, thisArg, arrayLike[i], i) : arrayLike[i];
					if (Context) {
						desc.value = value;
						defineProperty(arr, i, desc);
					} else {
						arr[i] = value;
					}
				}
			}
			if (Context) {
				desc.value = null;
				arr.length = length;
			}
			return arr;
		};
		return shim$1;
	}

	var from = isImplemented$4() ? Array.from : requireShim$1();

	var isImplemented = function () {
		var numberIsNaN = Number.isNaN;
		if (typeof numberIsNaN !== "function") return false;
		return !numberIsNaN({}) && numberIsNaN(NaN) && !numberIsNaN(34);
	};

	var shim;
	var hasRequiredShim;

	function requireShim () {
		if (hasRequiredShim) return shim;
		hasRequiredShim = 1;

		shim = function (value) {
			// eslint-disable-next-line no-self-compare
			return value !== value;
		};
		return shim;
	}

	var isNan = isImplemented() ? Number.isNaN : requireShim();

	var numberIsNaN       = isNan
	  , toPosInt          = toPosInteger
	  , value$1             = validValue
	  , indexOf$1           = Array.prototype.indexOf
	  , objHasOwnProperty = Object.prototype.hasOwnProperty
	  , abs               = Math.abs
	  , floor             = Math.floor;

	var eIndexOf = function (searchElement /*, fromIndex*/) {
		var i, length, fromIndex, val;
		if (!numberIsNaN(searchElement)) return indexOf$1.apply(this, arguments);

		length = toPosInt(value$1(this).length);
		fromIndex = arguments[1];
		if (isNaN(fromIndex)) fromIndex = 0;
		else if (fromIndex >= 0) fromIndex = floor(fromIndex);
		else fromIndex = toPosInt(this.length) - floor(abs(fromIndex));

		for (i = fromIndex; i < length; ++i) {
			if (objHasOwnProperty.call(this, i)) {
				val = this[i];
				if (numberIsNaN(val)) return i; // Jslint: ignore
			}
		}
		return -1;
	};

	var indexOf = eIndexOf
	  , forEach = Array.prototype.forEach
	  , splice  = Array.prototype.splice;

	// eslint-disable-next-line no-unused-vars
	var remove$1 = function (itemToRemove /*, …item*/) {
		forEach.call(
			arguments,
			function (item) {
				var index = indexOf.call(this, item);
				if (index !== -1) splice.call(this, index, 1);
			},
			this
		);
	};

	var isValue = isValue$4;

	var map = { function: true, object: true };

	var isObject$1 = function (value) { return (isValue(value) && map[typeof value]) || false; };

	var isObject = isObject$1;

	var validObject = function (value) {
		if (!isObject(value)) throw new TypeError(value + " is not an Object");
		return value;
	};

	var aFrom          = from
	  , remove         = remove$1
	  , value          = validObject
	  , d              = dExports
	  , emit           = eventEmitterExports.methods.emit

	  , defineProperty = Object.defineProperty
	  , hasOwnProperty$1 = Object.prototype.hasOwnProperty
	  , getOwnPropertyDescriptor = Object.getOwnPropertyDescriptor;

	var pipe = function (e1, e2/*, name*/) {
		var pipes, pipe, desc, name;

		(value(e1) && value(e2));
		name = arguments[2];
		if (name === undefined) name = 'emit';

		pipe = {
			close: function () { remove.call(pipes, e2); }
		};
		if (hasOwnProperty$1.call(e1, '__eePipes__')) {
			(pipes = e1.__eePipes__).push(e2);
			return pipe;
		}
		defineProperty(e1, '__eePipes__', d('c', pipes = [e2]));
		desc = getOwnPropertyDescriptor(e1, name);
		if (!desc) {
			desc = d('c', undefined);
		} else {
			delete desc.get;
			delete desc.set;
		}
		desc.value = function () {
			var i, emitter, data = aFrom(pipes);
			emit.apply(this, arguments);
			for (i = 0; (emitter = data[i]); ++i) emit.apply(emitter, arguments);
		};
		defineProperty(e1, name, desc);
		return pipe;
	};

	var pipe$1 = /*@__PURE__*/getDefaultExportFromCjs(pipe);

	let registeredHandlers = [...pagedMediaHandlers, ...generatedContentHandlers, ...filters];

	class Handlers {
		constructor(chunker, polisher, caller) {

			registeredHandlers.forEach((Handler) => {
				let handler = new Handler(chunker, polisher, caller);
				pipe$1(handler, this);
			});
		}
	}

	EventEmitter(Handlers.prototype);

	function registerHandlers() {
		for (var i = 0; i < arguments.length; i++) {
			registeredHandlers.push(arguments[i]);
		}
	}

	function initializeHandlers(chunker, polisher, caller) {
		let handlers = new Handlers(chunker, polisher, caller);
		return handlers;
	}

	class Previewer {
		constructor(options) {
			// this.preview = this.getParams("preview") !== "false";

			this.settings = options || {};

			// Process styles
			this.polisher = new Polisher(false);

			// Chunk contents
			this.chunker = new Chunker(undefined, undefined, this.settings);

			// Hooks
			this.hooks = {};
			this.hooks.beforePreview = new Hook(this);
			this.hooks.afterPreview = new Hook(this);

			// default size
			this.size = {
				width: {
					value: 8.5,
					unit: "in"
				},
				height: {
					value: 11,
					unit: "in"
				},
				format: undefined,
				orientation: undefined
			};

			this.chunker.on("page", (page) => {
				this.emit("page", page);
			});

			this.chunker.on("rendering", () => {
				this.emit("rendering", this.chunker);
			});
		}

		initializeHandlers() {
			let handlers = initializeHandlers(this.chunker, this.polisher, this);

			handlers.on("size", (size) => {
				this.size = size;
				this.emit("size", size);
			});

			handlers.on("atpages", (pages) => {
				this.atpages = pages;
				this.emit("atpages", pages);
			});

			return handlers;
		}

		registerHandlers() {
			return registerHandlers.apply(registerHandlers, arguments);
		}

		getParams(name) {
			let param;
			let url = new URL(window.location);
			let params = new URLSearchParams(url.search);
			for(var pair of params.entries()) {
				if(pair[0] === name) {
					param = pair[1];
				}
			}

			return param;
		}

		wrapContent() {
			// Wrap body in template tag
			let body = document.querySelector("body");

			// Check if a template exists
			let template;
			template = body.querySelector(":scope > template[data-ref='pagedjs-content']");

			if (!template) {
				// Otherwise create one
				template = document.createElement("template");
				template.dataset.ref = "pagedjs-content";
				template.innerHTML = body.innerHTML;
				body.innerHTML = "";
				body.appendChild(template);
			}

			return template.content;
		}

		removeStyles(doc=document) {
			// Get all stylesheets
			const stylesheets = Array.from(doc.querySelectorAll("link[rel='stylesheet']:not([data-pagedjs-ignore], [media~='screen'])"));
			// Get inline styles
			const inlineStyles = Array.from(doc.querySelectorAll("style:not([data-pagedjs-inserted-styles], [data-pagedjs-ignore], [media~='screen'])"));
			const elements = [...stylesheets, ...inlineStyles];
			return elements
				// preserve order
				.sort(function (element1, element2) {
					const position = element1.compareDocumentPosition(element2);
					if (position === Node.DOCUMENT_POSITION_PRECEDING) {
						return 1;
					} else if (position === Node.DOCUMENT_POSITION_FOLLOWING) {
						return -1;
					}
					return 0;
				})
				// extract the href
				.map((element) => {
					if (element.nodeName === "STYLE") {
						const obj = {};
						obj[window.location.href] = element.textContent;
						element.remove();
						return obj;
					}
					if (element.nodeName === "LINK") {
						element.remove();
						return element.href;
					}
					// ignore
					console.warn(`Unable to process: ${element}, ignoring.`);
				});
		}

		async preview(content, stylesheets, renderTo) {

			await this.hooks.beforePreview.trigger(content, renderTo);

			if (!content) {
				content = this.wrapContent();
			}

			if (!stylesheets) {
				stylesheets = this.removeStyles();
			}

			this.polisher.setup();

			this.handlers = this.initializeHandlers();

			await this.polisher.add(...stylesheets);

			let startTime = performance.now();

			// Render flow
			let flow = await this.chunker.flow(content, renderTo);

			let endTime = performance.now();

			flow.performance = (endTime - startTime);
			flow.size = this.size;

			this.emit("rendered", flow);

			await this.hooks.afterPreview.trigger(flow.pages);

			return flow;
		}
	}

	EventEmitter(Previewer.prototype);

	var Paged = /*#__PURE__*/Object.freeze({
		__proto__: null,
		Chunker: Chunker,
		Handler: Handler,
		Polisher: Polisher,
		Previewer: Previewer,
		initializeHandlers: initializeHandlers,
		registerHandlers: registerHandlers,
		registeredHandlers: registeredHandlers
	});

	window.Paged = Paged;

	let ready = new Promise(function(resolve, reject){
		if (document.readyState === "interactive" || document.readyState === "complete") {
			resolve(document.readyState);
			return;
		}

		document.onreadystatechange = function ($) {
			if (document.readyState === "interactive") {
				resolve(document.readyState);
			}
		};
	});

	let config = window.PagedConfig || {
		auto: true,
		before: undefined,
		after: undefined,
		content: undefined,
		stylesheets: undefined,
		renderTo: undefined,
		settings: undefined
	};

	let previewer = new Previewer(config.settings);

	ready.then(async function () {
		let done;
		if (config.before) {
			await config.before();
		}

		if(config.auto !== false) {
			done = await previewer.preview(config.content, config.stylesheets, config.renderTo);
		}


		if (config.after) {
			await config.after(done);
		}
	});

	return previewer;

}));
//# sourceURL=/Users/fredatherden/Documents/GitHub/jats-to-pdf/node_modules/pagedjs-cli/dist/browser.js</script><script type="text/javascript">var csstree = (() => {
  var ns = Object.create;
  var kt = Object.defineProperty;
  var os = Object.getOwnPropertyDescriptor;
  var is = Object.getOwnPropertyNames;
  var as = Object.getPrototypeOf,
    ss = Object.prototype.hasOwnProperty;
  var ze = (e, t) => () => (
      t || e((t = { exports: {} }).exports, t), t.exports
    ),
    b = (e, t) => {
      for (var r in t) kt(e, r, { get: t[r], enumerable: !0 });
    },
    jo = (e, t, r, n) => {
      if ((t && typeof t == "object") || typeof t == "function")
        for (let o of is(t))
          !ss.call(e, o) &&
            o !== r &&
            kt(e, o, {
              get: () => t[o],
              enumerable: !(n = os(t, o)) || n.enumerable,
            });
      return e;
    };
  var ls = (e, t, r) => (
      (r = e != null ? ns(as(e)) : {}),
      jo(
        t || !e || !e.__esModule
          ? kt(r, "default", { value: e, enumerable: !0 })
          : r,
        e
      )
    ),
    cs = (e) => jo(kt({}, "__esModule", { value: !0 }), e);
  var ri = ze((dr) => {
    var ti =
      "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/".split(
        ""
      );
    dr.encode = function (e) {
      if (0 <= e && e < ti.length) return ti[e];
      throw new TypeError("Must be between 0 and 63: " + e);
    };
    dr.decode = function (e) {
      var t = 65,
        r = 90,
        n = 97,
        o = 122,
        i = 48,
        s = 57,
        u = 43,
        c = 47,
        a = 26,
        l = 52;
      return t <= e && e <= r
        ? e - t
        : n <= e && e <= o
        ? e - n + a
        : i <= e && e <= s
        ? e - i + l
        : e == u
        ? 62
        : e == c
        ? 63
        : -1;
    };
  });
  var si = ze((br) => {
    var ni = ri(),
      gr = 5,
      oi = 1 << gr,
      ii = oi - 1,
      ai = oi;
    function ws(e) {
      return e < 0 ? (-e << 1) + 1 : (e << 1) + 0;
    }
    function vs(e) {
      var t = (e & 1) === 1,
        r = e >> 1;
      return t ? -r : r;
    }
    br.encode = function (t) {
      var r = "",
        n,
        o = ws(t);
      do (n = o & ii), (o >>>= gr), o > 0 && (n |= ai), (r += ni.encode(n));
      while (o > 0);
      return r;
    };
    br.decode = function (t, r, n) {
      var o = t.length,
        i = 0,
        s = 0,
        u,
        c;
      do {
        if (r >= o)
          throw new Error("Expected more digits in base 64 VLQ value.");
        if (((c = ni.decode(t.charCodeAt(r++))), c === -1))
          throw new Error("Invalid base64 digit: " + t.charAt(r - 1));
        (u = !!(c & ai)), (c &= ii), (i = i + (c << s)), (s += gr);
      } while (u);
      (n.value = vs(i)), (n.rest = r);
    };
  });
  var It = ze((K) => {
    function Ss(e, t, r) {
      if (t in e) return e[t];
      if (arguments.length === 3) return r;
      throw new Error('"' + t + '" is a required argument.');
    }
    K.getArg = Ss;
    var li = /^(?:([\w+\-.]+):)?\/\/(?:(\w+:\w+)@)?([\w.-]*)(?::(\d+))?(.*)$/,
      Cs = /^data:.+\,.+$/;
    function nt(e) {
      var t = e.match(li);
      return t
        ? { scheme: t[1], auth: t[2], host: t[3], port: t[4], path: t[5] }
        : null;
    }
    K.urlParse = nt;
    function We(e) {
      var t = "";
      return (
        e.scheme && (t += e.scheme + ":"),
        (t += "//"),
        e.auth && (t += e.auth + "@"),
        e.host && (t += e.host),
        e.port && (t += ":" + e.port),
        e.path && (t += e.path),
        t
      );
    }
    K.urlGenerate = We;
    var As = 32;
    function Ts(e) {
      var t = [];
      return function (r) {
        for (var n = 0; n < t.length; n++)
          if (t[n].input === r) {
            var o = t[0];
            return (t[0] = t[n]), (t[n] = o), t[0].result;
          }
        var i = e(r);
        return t.unshift({ input: r, result: i }), t.length > As && t.pop(), i;
      };
    }
    var xr = Ts(function (t) {
      var r = t,
        n = nt(t);
      if (n) {
        if (!n.path) return t;
        r = n.path;
      }
      for (var o = K.isAbsolute(r), i = [], s = 0, u = 0; ; )
        if (((s = u), (u = r.indexOf("/", s)), u === -1)) {
          i.push(r.slice(s));
          break;
        } else for (i.push(r.slice(s, u)); u < r.length && r[u] === "/"; ) u++;
      for (var c, a = 0, u = i.length - 1; u >= 0; u--)
        (c = i[u]),
          c === "."
            ? i.splice(u, 1)
            : c === ".."
            ? a++
            : a > 0 &&
              (c === ""
                ? (i.splice(u + 1, a), (a = 0))
                : (i.splice(u, 2), a--));
      return (
        (r = i.join("/")),
        r === "" && (r = o ? "/" : "."),
        n ? ((n.path = r), We(n)) : r
      );
    });
    K.normalize = xr;
    function ci(e, t) {
      e === "" && (e = "."), t === "" && (t = ".");
      var r = nt(t),
        n = nt(e);
      if ((n && (e = n.path || "/"), r && !r.scheme))
        return n && (r.scheme = n.scheme), We(r);
      if (r || t.match(Cs)) return t;
      if (n && !n.host && !n.path) return (n.host = t), We(n);
      var o = t.charAt(0) === "/" ? t : xr(e.replace(/\/+$/, "") + "/" + t);
      return n ? ((n.path = o), We(n)) : o;
    }
    K.join = ci;
    K.isAbsolute = function (e) {
      return e.charAt(0) === "/" || li.test(e);
    };
    function Es(e, t) {
      e === "" && (e = "."), (e = e.replace(/\/$/, ""));
      for (var r = 0; t.indexOf(e + "/") !== 0; ) {
        var n = e.lastIndexOf("/");
        if (n < 0 || ((e = e.slice(0, n)), e.match(/^([^\/]+:\/)?\/*$/)))
          return t;
        ++r;
      }
      return Array(r + 1).join("../") + t.substr(e.length + 1);
    }
    K.relative = Es;
    var ui = (function () {
      var e = Object.create(null);
      return !("__proto__" in e);
    })();
    function pi(e) {
      return e;
    }
    function Ls(e) {
      return hi(e) ? "$" + e : e;
    }
    K.toSetString = ui ? pi : Ls;
    function Ps(e) {
      return hi(e) ? e.slice(1) : e;
    }
    K.fromSetString = ui ? pi : Ps;
    function hi(e) {
      if (!e) return !1;
      var t = e.length;
      if (
        t < 9 ||
        e.charCodeAt(t - 1) !== 95 ||
        e.charCodeAt(t - 2) !== 95 ||
        e.charCodeAt(t - 3) !== 111 ||
        e.charCodeAt(t - 4) !== 116 ||
        e.charCodeAt(t - 5) !== 111 ||
        e.charCodeAt(t - 6) !== 114 ||
        e.charCodeAt(t - 7) !== 112 ||
        e.charCodeAt(t - 8) !== 95 ||
        e.charCodeAt(t - 9) !== 95
      )
        return !1;
      for (var r = t - 10; r >= 0; r--) if (e.charCodeAt(r) !== 36) return !1;
      return !0;
    }
    function Is(e, t, r) {
      var n = be(e.source, t.source);
      return n !== 0 ||
        ((n = e.originalLine - t.originalLine), n !== 0) ||
        ((n = e.originalColumn - t.originalColumn), n !== 0 || r) ||
        ((n = e.generatedColumn - t.generatedColumn), n !== 0) ||
        ((n = e.generatedLine - t.generatedLine), n !== 0)
        ? n
        : be(e.name, t.name);
    }
    K.compareByOriginalPositions = Is;
    function Ds(e, t, r) {
      var n;
      return (
        (n = e.originalLine - t.originalLine),
        n !== 0 ||
        ((n = e.originalColumn - t.originalColumn), n !== 0 || r) ||
        ((n = e.generatedColumn - t.generatedColumn), n !== 0) ||
        ((n = e.generatedLine - t.generatedLine), n !== 0)
          ? n
          : be(e.name, t.name)
      );
    }
    K.compareByOriginalPositionsNoSource = Ds;
    function Os(e, t, r) {
      var n = e.generatedLine - t.generatedLine;
      return n !== 0 ||
        ((n = e.generatedColumn - t.generatedColumn), n !== 0 || r) ||
        ((n = be(e.source, t.source)), n !== 0) ||
        ((n = e.originalLine - t.originalLine), n !== 0) ||
        ((n = e.originalColumn - t.originalColumn), n !== 0)
        ? n
        : be(e.name, t.name);
    }
    K.compareByGeneratedPositionsDeflated = Os;
    function Ns(e, t, r) {
      var n = e.generatedColumn - t.generatedColumn;
      return n !== 0 ||
        r ||
        ((n = be(e.source, t.source)), n !== 0) ||
        ((n = e.originalLine - t.originalLine), n !== 0) ||
        ((n = e.originalColumn - t.originalColumn), n !== 0)
        ? n
        : be(e.name, t.name);
    }
    K.compareByGeneratedPositionsDeflatedNoLine = Ns;
    function be(e, t) {
      return e === t ? 0 : e === null ? 1 : t === null ? -1 : e > t ? 1 : -1;
    }
    function zs(e, t) {
      var r = e.generatedLine - t.generatedLine;
      return r !== 0 ||
        ((r = e.generatedColumn - t.generatedColumn), r !== 0) ||
        ((r = be(e.source, t.source)), r !== 0) ||
        ((r = e.originalLine - t.originalLine), r !== 0) ||
        ((r = e.originalColumn - t.originalColumn), r !== 0)
        ? r
        : be(e.name, t.name);
    }
    K.compareByGeneratedPositionsInflated = zs;
    function Ms(e) {
      return JSON.parse(e.replace(/^\)]}'[^\n]*\n/, ""));
    }
    K.parseSourceMapInput = Ms;
    function Rs(e, t, r) {
      if (
        ((t = t || ""),
        e &&
          (e[e.length - 1] !== "/" && t[0] !== "/" && (e += "/"), (t = e + t)),
        r)
      ) {
        var n = nt(r);
        if (!n) throw new Error("sourceMapURL could not be parsed");
        if (n.path) {
          var o = n.path.lastIndexOf("/");
          o >= 0 && (n.path = n.path.substring(0, o + 1));
        }
        t = ci(We(n), t);
      }
      return xr(t);
    }
    K.computeSourceURL = Rs;
  });
  var fi = ze((mi) => {
    var yr = It(),
      kr = Object.prototype.hasOwnProperty,
      Pe = typeof Map < "u";
    function xe() {
      (this._array = []), (this._set = Pe ? new Map() : Object.create(null));
    }
    xe.fromArray = function (t, r) {
      for (var n = new xe(), o = 0, i = t.length; o < i; o++) n.add(t[o], r);
      return n;
    };
    xe.prototype.size = function () {
      return Pe ? this._set.size : Object.getOwnPropertyNames(this._set).length;
    };
    xe.prototype.add = function (t, r) {
      var n = Pe ? t : yr.toSetString(t),
        o = Pe ? this.has(t) : kr.call(this._set, n),
        i = this._array.length;
      (!o || r) && this._array.push(t),
        o || (Pe ? this._set.set(t, i) : (this._set[n] = i));
    };
    xe.prototype.has = function (t) {
      if (Pe) return this._set.has(t);
      var r = yr.toSetString(t);
      return kr.call(this._set, r);
    };
    xe.prototype.indexOf = function (t) {
      if (Pe) {
        var r = this._set.get(t);
        if (r >= 0) return r;
      } else {
        var n = yr.toSetString(t);
        if (kr.call(this._set, n)) return this._set[n];
      }
      throw new Error('"' + t + '" is not in the set.');
    };
    xe.prototype.at = function (t) {
      if (t >= 0 && t < this._array.length) return this._array[t];
      throw new Error("No element indexed by " + t);
    };
    xe.prototype.toArray = function () {
      return this._array.slice();
    };
    mi.ArraySet = xe;
  });
  var bi = ze((gi) => {
    var di = It();
    function Fs(e, t) {
      var r = e.generatedLine,
        n = t.generatedLine,
        o = e.generatedColumn,
        i = t.generatedColumn;
      return (
        n > r ||
        (n == r && i >= o) ||
        di.compareByGeneratedPositionsInflated(e, t) <= 0
      );
    }
    function Dt() {
      (this._array = []),
        (this._sorted = !0),
        (this._last = { generatedLine: -1, generatedColumn: 0 });
    }
    Dt.prototype.unsortedForEach = function (t, r) {
      this._array?.forEach(t, r);
    };
    Dt.prototype.add = function (t) {
      Fs(this._last, t)
        ? ((this._last = t), this._array.push(t))
        : ((this._sorted = !1), this._array.push(t));
    };
    Dt.prototype.toArray = function () {
      return (
        this._sorted ||
          (this._array.sort(di.compareByGeneratedPositionsInflated),
          (this._sorted = !0)),
        this._array
      );
    };
    gi.MappingList = Dt;
  });
  var yi = ze((xi) => {
    var ot = si(),
      j = It(),
      Ot = fi().ArraySet,
      Bs = bi().MappingList;
    function oe(e) {
      e || (e = {}),
        (this._file = j.getArg(e, "file", null)),
        (this._sourceRoot = j.getArg(e, "sourceRoot", null)),
        (this._skipValidation = j.getArg(e, "skipValidation", !1)),
        (this._sources = new Ot()),
        (this._names = new Ot()),
        (this._mappings = new Bs()),
        (this._sourcesContents = null);
    }
    oe.prototype._version = 3;
    oe.fromSourceMap = function (t) {
      var r = t.sourceRoot,
        n = new oe({ file: t.file, sourceRoot: r });
      return (
        t.eachMapping(function (o) {
          var i = {
            generated: { line: o.generatedLine, column: o.generatedColumn },
          };
          o.source != null &&
            ((i.source = o.source),
            r != null && (i.source = j.relative(r, i.source)),
            (i.original = { line: o.originalLine, column: o.originalColumn }),
            o.name != null && (i.name = o.name)),
            n.addMapping(i);
        }),
        t.sources?.forEach(function (o) {
          var i = o;
          r !== null && (i = j.relative(r, o)),
            n._sources.has(i) || n._sources.add(i);
          var s = t.sourceContentFor(o);
          s != null && n.setSourceContent(o, s);
        }),
        n
      );
    };
    oe.prototype.addMapping = function (t) {
      var r = j.getArg(t, "generated"),
        n = j.getArg(t, "original", null),
        o = j.getArg(t, "source", null),
        i = j.getArg(t, "name", null);
      this._skipValidation || this._validateMapping(r, n, o, i),
        o != null &&
          ((o = String(o)), this._sources.has(o) || this._sources.add(o)),
        i != null &&
          ((i = String(i)), this._names.has(i) || this._names.add(i)),
        this._mappings.add({
          generatedLine: r.line,
          generatedColumn: r.column,
          originalLine: n != null && n.line,
          originalColumn: n != null && n.column,
          source: o,
          name: i,
        });
    };
    oe.prototype.setSourceContent = function (t, r) {
      var n = t;
      this._sourceRoot != null && (n = j.relative(this._sourceRoot, n)),
        r != null
          ? (this._sourcesContents ||
              (this._sourcesContents = Object.create(null)),
            (this._sourcesContents[j.toSetString(n)] = r))
          : this._sourcesContents &&
            (delete this._sourcesContents[j.toSetString(n)],
            Object.keys(this._sourcesContents).length === 0 &&
              (this._sourcesContents = null));
    };
    oe.prototype.applySourceMap = function (t, r, n) {
      var o = r;
      if (r == null) {
        if (t.file == null)
          throw new Error(
            `SourceMapGenerator.prototype.applySourceMap requires either an explicit source file, or the source map's "file" property. Both were omitted.`
          );
        o = t.file;
      }
      var i = this._sourceRoot;
      i != null && (o = j.relative(i, o));
      var s = new Ot(),
        u = new Ot();
      this._mappings.unsortedForEach(function (c) {
        if (c.source === o && c.originalLine != null) {
          var a = t.originalPositionFor({
            line: c.originalLine,
            column: c.originalColumn,
          });
          a.source != null &&
            ((c.source = a.source),
            n != null && (c.source = j.join(n, c.source)),
            i != null && (c.source = j.relative(i, c.source)),
            (c.originalLine = a.line),
            (c.originalColumn = a.column),
            a.name != null && (c.name = a.name));
        }
        var l = c.source;
        l != null && !s.has(l) && s.add(l);
        var p = c.name;
        p != null && !u.has(p) && u.add(p);
      }, this),
        (this._sources = s),
        (this._names = u),
        t.sources?.forEach(function (c) {
          var a = t.sourceContentFor(c);
          a != null &&
            (n != null && (c = j.join(n, c)),
            i != null && (c = j.relative(i, c)),
            this.setSourceContent(c, a));
        }, this);
    };
    oe.prototype._validateMapping = function (t, r, n, o) {
      if (r && typeof r.line != "number" && typeof r.column != "number")
        throw new Error(
          "original.line and original.column are not numbers -- you probably meant to omit the original mapping entirely and only map the generated position. If so, pass null for the original mapping instead of an object with empty or null values."
        );
      if (
        !(
          t &&
          "line" in t &&
          "column" in t &&
          t.line > 0 &&
          t.column >= 0 &&
          !r &&
          !n &&
          !o
        )
      ) {
        if (
          t &&
          "line" in t &&
          "column" in t &&
          r &&
          "line" in r &&
          "column" in r &&
          t.line > 0 &&
          t.column >= 0 &&
          r.line > 0 &&
          r.column >= 0 &&
          n
        )
          return;
        throw new Error(
          "Invalid mapping: " +
            JSON.stringify({ generated: t, source: n, original: r, name: o })
        );
      }
    };
    oe.prototype._serializeMappings = function () {
      for (
        var t = 0,
          r = 1,
          n = 0,
          o = 0,
          i = 0,
          s = 0,
          u = "",
          c,
          a,
          l,
          p,
          m = this._mappings.toArray(),
          f = 0,
          P = m.length;
        f < P;
        f++
      ) {
        if (((a = m[f]), (c = ""), a.generatedLine !== r))
          for (t = 0; a.generatedLine !== r; ) (c += ";"), r++;
        else if (f > 0) {
          if (!j.compareByGeneratedPositionsInflated(a, m[f - 1])) continue;
          c += ",";
        }
        (c += ot.encode(a.generatedColumn - t)),
          (t = a.generatedColumn),
          a.source != null &&
            ((p = this._sources.indexOf(a.source)),
            (c += ot.encode(p - s)),
            (s = p),
            (c += ot.encode(a.originalLine - 1 - o)),
            (o = a.originalLine - 1),
            (c += ot.encode(a.originalColumn - n)),
            (n = a.originalColumn),
            a.name != null &&
              ((l = this._names.indexOf(a.name)),
              (c += ot.encode(l - i)),
              (i = l))),
          (u += c);
      }
      return u;
    };
    oe.prototype._generateSourcesContent = function (t, r) {
      return t.map(function (n) {
        if (!this._sourcesContents) return null;
        r != null && (n = j.relative(r, n));
        var o = j.toSetString(n);
        return Object.prototype.hasOwnProperty.call(this._sourcesContents, o)
          ? this._sourcesContents[o]
          : null;
      }, this);
    };
    oe.prototype.toJSON = function () {
      var t = {
        version: this._version,
        sources: this._sources.toArray(),
        names: this._names.toArray(),
        mappings: this._serializeMappings(),
      };
      return (
        this._file != null && (t.file = this._file),
        this._sourceRoot != null && (t.sourceRoot = this._sourceRoot),
        this._sourcesContents &&
          (t.sourcesContent = this._generateSourcesContent(
            t.sources,
            t.sourceRoot
          )),
        t
      );
    };
    oe.prototype.toString = function () {
      return JSON.stringify(this.toJSON());
    };
    xi.SourceMapGenerator = oe;
  });
  var sm = {};
  b(sm, {
    Lexer: () => De,
    List: () => D,
    TokenStream: () => Ue,
    clone: () => ar,
    createLexer: () => Jh,
    createSyntax: () => Zt,
    definitionSyntax: () => qr,
    find: () => tm,
    findAll: () => nm,
    findLast: () => rm,
    fork: () => am,
    fromPlainObject: () => im,
    generate: () => $h,
    ident: () => Uo,
    isCustomProperty: () => lt,
    keyword: () => st,
    lexer: () => Zh,
    parse: () => Xh,
    property: () => Ft,
    string: () => rr,
    toPlainObject: () => om,
    tokenNames: () => Ee,
    tokenTypes: () => Me,
    tokenize: () => Qh,
    url: () => or,
    vendorPrefix: () => Js,
    version: () => Gh,
    walk: () => em,
  });
  var Me = {};
  b(Me, {
    AtKeyword: () => I,
    BadString: () => Ae,
    BadUrl: () => Y,
    CDC: () => U,
    CDO: () => ue,
    Colon: () => O,
    Comma: () => G,
    Comment: () => E,
    Delim: () => g,
    Dimension: () => y,
    EOF: () => Ze,
    Function: () => x,
    Hash: () => v,
    Ident: () => h,
    LeftCurlyBracket: () => M,
    LeftParenthesis: () => T,
    LeftSquareBracket: () => q,
    Number: () => d,
    Percentage: () => A,
    RightCurlyBracket: () => H,
    RightParenthesis: () => w,
    RightSquareBracket: () => V,
    Semicolon: () => _,
    String: () => W,
    Url: () => F,
    WhiteSpace: () => k,
  });
  var Ze = 0,
    h = 1,
    x = 2,
    I = 3,
    v = 4,
    W = 5,
    Ae = 6,
    F = 7,
    Y = 8,
    g = 9,
    d = 10,
    A = 11,
    y = 12,
    k = 13,
    ue = 14,
    U = 15,
    O = 16,
    _ = 17,
    G = 18,
    q = 19,
    V = 20,
    T = 21,
    w = 22,
    M = 23,
    H = 24,
    E = 25;
  function B(e) {
    return e >= 48 && e <= 57;
  }
  function ee(e) {
    return B(e) || (e >= 65 && e <= 70) || (e >= 97 && e <= 102);
  }
  function vt(e) {
    return e >= 65 && e <= 90;
  }
  function us(e) {
    return e >= 97 && e <= 122;
  }
  function ps(e) {
    return vt(e) || us(e);
  }
  function hs(e) {
    return e >= 128;
  }
  function wt(e) {
    return ps(e) || hs(e) || e === 95;
  }
  function Re(e) {
    return wt(e) || B(e) || e === 45;
  }
  function ms(e) {
    return (e >= 0 && e <= 8) || e === 11 || (e >= 14 && e <= 31) || e === 127;
  }
  function Je(e) {
    return e === 10 || e === 13 || e === 12;
  }
  function pe(e) {
    return Je(e) || e === 32 || e === 9;
  }
  function $(e, t) {
    return !(e !== 92 || Je(t) || t === 0);
  }
  function Fe(e, t, r) {
    return e === 45
      ? wt(t) || t === 45 || $(t, r)
      : wt(e)
      ? !0
      : e === 92
      ? $(e, t)
      : !1;
  }
  function St(e, t, r) {
    return e === 43 || e === 45
      ? B(t)
        ? 2
        : t === 46 && B(r)
        ? 3
        : 0
      : e === 46
      ? B(t)
        ? 2
        : 0
      : B(e)
      ? 1
      : 0;
  }
  function Ct(e) {
    return e === 65279 || e === 65534 ? 1 : 0;
  }
  var sr = new Array(128),
    fs = 128,
    et = 130,
    lr = 131,
    At = 132,
    cr = 133;
  for (let e = 0; e < sr.length; e++)
    sr[e] =
      (pe(e) && et) ||
      (B(e) && lr) ||
      (wt(e) && At) ||
      (ms(e) && cr) ||
      e ||
      fs;
  function Tt(e) {
    return e < 128 ? sr[e] : At;
  }
  function Be(e, t) {
    return t < e.length ? e.charCodeAt(t) : 0;
  }
  function Et(e, t, r) {
    return r === 13 && Be(e, t + 1) === 10 ? 2 : 1;
  }
  function de(e, t, r) {
    let n = e.charCodeAt(t);
    return vt(n) && (n = n | 32), n === r;
  }
  function ge(e, t, r, n) {
    if (r - t !== n.length || t < 0 || r > e.length) return !1;
    for (let o = t; o < r; o++) {
      let i = n.charCodeAt(o - t),
        s = e.charCodeAt(o);
      if ((vt(s) && (s = s | 32), s !== i)) return !1;
    }
    return !0;
  }
  function Wo(e, t) {
    for (; t >= 0 && pe(e.charCodeAt(t)); t--);
    return t + 1;
  }
  function tt(e, t) {
    for (; t < e.length && pe(e.charCodeAt(t)); t++);
    return t;
  }
  function ur(e, t) {
    for (; t < e.length && B(e.charCodeAt(t)); t++);
    return t;
  }
  function se(e, t) {
    if (((t += 2), ee(Be(e, t - 1)))) {
      for (let n = Math.min(e.length, t + 5); t < n && ee(Be(e, t)); t++);
      let r = Be(e, t);
      pe(r) && (t += Et(e, t, r));
    }
    return t;
  }
  function rt(e, t) {
    for (; t < e.length; t++) {
      let r = e.charCodeAt(t);
      if (!Re(r)) {
        if ($(r, Be(e, t + 1))) {
          t = se(e, t) - 1;
          continue;
        }
        break;
      }
    }
    return t;
  }
  function Te(e, t) {
    let r = e.charCodeAt(t);
    if (
      ((r === 43 || r === 45) && (r = e.charCodeAt((t += 1))),
      B(r) && ((t = ur(e, t + 1)), (r = e.charCodeAt(t))),
      r === 46 && B(e.charCodeAt(t + 1)) && ((t += 2), (t = ur(e, t))),
      de(e, t, 101))
    ) {
      let n = 0;
      (r = e.charCodeAt(t + 1)),
        (r === 45 || r === 43) && ((n = 1), (r = e.charCodeAt(t + 2))),
        B(r) && (t = ur(e, t + 1 + n + 1));
    }
    return t;
  }
  function Lt(e, t) {
    for (; t < e.length; t++) {
      let r = e.charCodeAt(t);
      if (r === 41) {
        t++;
        break;
      }
      $(r, Be(e, t + 1)) && (t = se(e, t));
    }
    return t;
  }
  function _e(e) {
    if (e.length === 1 && !ee(e.charCodeAt(0))) return e[0];
    let t = parseInt(e, 16);
    return (
      (t === 0 || (t >= 55296 && t <= 57343) || t > 1114111) && (t = 65533),
      String.fromCodePoint(t)
    );
  }
  var Ee = [
    "EOF-token",
    "ident-token",
    "function-token",
    "at-keyword-token",
    "hash-token",
    "string-token",
    "bad-string-token",
    "url-token",
    "bad-url-token",
    "delim-token",
    "number-token",
    "percentage-token",
    "dimension-token",
    "whitespace-token",
    "CDO-token",
    "CDC-token",
    "colon-token",
    "semicolon-token",
    "comma-token",
    "[-token",
    "]-token",
    "(-token",
    ")-token",
    "{-token",
    "}-token",
  ];
  function qe(e = null, t) {
    return e === null || e.length < t
      ? new Uint32Array(Math.max(t + 1024, 16384))
      : e;
  }
  var Ho = 10,
    ds = 12,
    Yo = 13;
  function Go(e) {
    let t = e.source,
      r = t.length,
      n = t.length > 0 ? Ct(t.charCodeAt(0)) : 0,
      o = qe(e.lines, r),
      i = qe(e.columns, r),
      s = e.startLine,
      u = e.startColumn;
    for (let c = n; c < r; c++) {
      let a = t.charCodeAt(c);
      (o[c] = s),
        (i[c] = u++),
        (a === Ho || a === Yo || a === ds) &&
          (a === Yo &&
            c + 1 < r &&
            t.charCodeAt(c + 1) === Ho &&
            (c++, (o[c] = s), (i[c] = u)),
          s++,
          (u = 1));
    }
    (o[r] = s), (i[r] = u), (e.lines = o), (e.columns = i), (e.computed = !0);
  }
  var Pt = class {
    constructor() {
      (this.lines = null), (this.columns = null), (this.computed = !1);
    }
    setSource(t, r = 0, n = 1, o = 1) {
      (this.source = t),
        (this.startOffset = r),
        (this.startLine = n),
        (this.startColumn = o),
        (this.computed = !1);
    }
    getLocation(t, r) {
      return (
        this.computed || Go(this),
        {
          source: r,
          offset: this.startOffset + t,
          line: this.lines[t],
          column: this.columns[t],
        }
      );
    }
    getLocationRange(t, r, n) {
      return (
        this.computed || Go(this),
        {
          source: n,
          start: {
            offset: this.startOffset + t,
            line: this.lines[t],
            column: this.columns[t],
          },
          end: {
            offset: this.startOffset + r,
            line: this.lines[r],
            column: this.columns[r],
          },
        }
      );
    }
  };
  var ne = 16777215,
    we = 24,
    gs = new Map([
      [2, 22],
      [21, 22],
      [19, 20],
      [23, 24],
    ]),
    Ue = class {
      constructor(t, r) {
        this.setSource(t, r);
      }
      reset() {
        (this.eof = !1),
          (this.tokenIndex = -1),
          (this.tokenType = 0),
          (this.tokenStart = this.firstCharOffset),
          (this.tokenEnd = this.firstCharOffset);
      }
      setSource(t = "", r = () => {}) {
        t = String(t || "");
        let n = t.length,
          o = qe(this.offsetAndType, t.length + 1),
          i = qe(this.balance, t.length + 1),
          s = 0,
          u = 0,
          c = 0,
          a = -1;
        for (
          this.offsetAndType = null,
            this.balance = null,
            r(t, (l, p, m) => {
              switch (l) {
                default:
                  i[s] = n;
                  break;
                case u: {
                  let f = c & ne;
                  for (c = i[f], u = c >> we, i[s] = f, i[f++] = s; f < s; f++)
                    i[f] === n && (i[f] = s);
                  break;
                }
                case 21:
                case 2:
                case 19:
                case 23:
                  (i[s] = c), (u = gs.get(l)), (c = (u << we) | s);
                  break;
              }
              (o[s++] = (l << we) | m), a === -1 && (a = p);
            }),
            o[s] = (0 << we) | n,
            i[s] = n,
            i[n] = n;
          c !== 0;

        ) {
          let l = c & ne;
          (c = i[l]), (i[l] = n);
        }
        (this.source = t),
          (this.firstCharOffset = a === -1 ? 0 : a),
          (this.tokenCount = s),
          (this.offsetAndType = o),
          (this.balance = i),
          this.reset(),
          this.next();
      }
      lookupType(t) {
        return (
          (t += this.tokenIndex),
          t < this.tokenCount ? this.offsetAndType[t] >> we : 0
        );
      }
      lookupOffset(t) {
        return (
          (t += this.tokenIndex),
          t < this.tokenCount
            ? this.offsetAndType[t - 1] & ne
            : this.source.length
        );
      }
      lookupValue(t, r) {
        return (
          (t += this.tokenIndex),
          t < this.tokenCount
            ? ge(
                this.source,
                this.offsetAndType[t - 1] & ne,
                this.offsetAndType[t] & ne,
                r
              )
            : !1
        );
      }
      getTokenStart(t) {
        return t === this.tokenIndex
          ? this.tokenStart
          : t > 0
          ? t < this.tokenCount
            ? this.offsetAndType[t - 1] & ne
            : this.offsetAndType[this.tokenCount] & ne
          : this.firstCharOffset;
      }
      substrToCursor(t) {
        return this.source.substring(t, this.tokenStart);
      }
      isBalanceEdge(t) {
        return this.balance[this.tokenIndex] < t;
      }
      isDelim(t, r) {
        return r
          ? this.lookupType(r) === 9 &&
              this.source.charCodeAt(this.lookupOffset(r)) === t
          : this.tokenType === 9 &&
              this.source.charCodeAt(this.tokenStart) === t;
      }
      skip(t) {
        let r = this.tokenIndex + t;
        r < this.tokenCount
          ? ((this.tokenIndex = r),
            (this.tokenStart = this.offsetAndType[r - 1] & ne),
            (r = this.offsetAndType[r]),
            (this.tokenType = r >> we),
            (this.tokenEnd = r & ne))
          : ((this.tokenIndex = this.tokenCount), this.next());
      }
      next() {
        let t = this.tokenIndex + 1;
        t < this.tokenCount
          ? ((this.tokenIndex = t),
            (this.tokenStart = this.tokenEnd),
            (t = this.offsetAndType[t]),
            (this.tokenType = t >> we),
            (this.tokenEnd = t & ne))
          : ((this.eof = !0),
            (this.tokenIndex = this.tokenCount),
            (this.tokenType = 0),
            (this.tokenStart = this.tokenEnd = this.source.length));
      }
      skipSC() {
        for (; this.tokenType === 13 || this.tokenType === 25; ) this.next();
      }
      skipUntilBalanced(t, r) {
        let n = t,
          o,
          i;
        e: for (; n < this.tokenCount; n++) {
          if (((o = this.balance[n]), o < t)) break e;
          switch (
            ((i =
              n > 0 ? this.offsetAndType[n - 1] & ne : this.firstCharOffset),
            r(this.source.charCodeAt(i)))
          ) {
            case 1:
              break e;
            case 2:
              n++;
              break e;
            default:
              this.balance[o] === n && (n = o);
          }
        }
        this.skip(n - this.tokenIndex);
      }
      forEachToken(t) {
        for (let r = 0, n = this.firstCharOffset; r < this.tokenCount; r++) {
          let o = n,
            i = this.offsetAndType[r],
            s = i & ne,
            u = i >> we;
          (n = s), t(u, o, s, r);
        }
      }
      dump() {
        let t = new Array(this.tokenCount);
        return (
          this?.forEachToken((r, n, o, i) => {
            t[i] = {
              idx: i,
              type: Ee[r],
              chunk: this.source.substring(n, o),
              balance: this.balance[i],
            };
          }),
          t
        );
      }
    };
  function ve(e, t) {
    function r(p) {
      return p < u ? e.charCodeAt(p) : 0;
    }
    function n() {
      if (((a = Te(e, a)), Fe(r(a), r(a + 1), r(a + 2)))) {
        (l = 12), (a = rt(e, a));
        return;
      }
      if (r(a) === 37) {
        (l = 11), a++;
        return;
      }
      l = 10;
    }
    function o() {
      let p = a;
      if (((a = rt(e, a)), ge(e, p, a, "url") && r(a) === 40)) {
        if (((a = tt(e, a + 1)), r(a) === 34 || r(a) === 39)) {
          (l = 2), (a = p + 4);
          return;
        }
        s();
        return;
      }
      if (r(a) === 40) {
        (l = 2), a++;
        return;
      }
      l = 1;
    }
    function i(p) {
      for (p || (p = r(a++)), l = 5; a < e.length; a++) {
        let m = e.charCodeAt(a);
        switch (Tt(m)) {
          case p:
            a++;
            return;
          case et:
            if (Je(m)) {
              (a += Et(e, a, m)), (l = 6);
              return;
            }
            break;
          case 92:
            if (a === e.length - 1) break;
            let f = r(a + 1);
            Je(f) ? (a += Et(e, a + 1, f)) : $(m, f) && (a = se(e, a) - 1);
            break;
        }
      }
    }
    function s() {
      for (l = 7, a = tt(e, a); a < e.length; a++) {
        let p = e.charCodeAt(a);
        switch (Tt(p)) {
          case 41:
            a++;
            return;
          case et:
            if (((a = tt(e, a)), r(a) === 41 || a >= e.length)) {
              a < e.length && a++;
              return;
            }
            (a = Lt(e, a)), (l = 8);
            return;
          case 34:
          case 39:
          case 40:
          case cr:
            (a = Lt(e, a)), (l = 8);
            return;
          case 92:
            if ($(p, r(a + 1))) {
              a = se(e, a) - 1;
              break;
            }
            (a = Lt(e, a)), (l = 8);
            return;
        }
      }
    }
    e = String(e || "");
    let u = e.length,
      c = Ct(r(0)),
      a = c,
      l;
    for (; a < u; ) {
      let p = e.charCodeAt(a);
      switch (Tt(p)) {
        case et:
          (l = 13), (a = tt(e, a + 1));
          break;
        case 34:
          i();
          break;
        case 35:
          Re(r(a + 1)) || $(r(a + 1), r(a + 2))
            ? ((l = 4), (a = rt(e, a + 1)))
            : ((l = 9), a++);
          break;
        case 39:
          i();
          break;
        case 40:
          (l = 21), a++;
          break;
        case 41:
          (l = 22), a++;
          break;
        case 43:
          St(p, r(a + 1), r(a + 2)) ? n() : ((l = 9), a++);
          break;
        case 44:
          (l = 18), a++;
          break;
        case 45:
          St(p, r(a + 1), r(a + 2))
            ? n()
            : r(a + 1) === 45 && r(a + 2) === 62
            ? ((l = 15), (a = a + 3))
            : Fe(p, r(a + 1), r(a + 2))
            ? o()
            : ((l = 9), a++);
          break;
        case 46:
          St(p, r(a + 1), r(a + 2)) ? n() : ((l = 9), a++);
          break;
        case 47:
          r(a + 1) === 42
            ? ((l = 25),
              (a = e.indexOf("*/", a + 2)),
              (a = a === -1 ? e.length : a + 2))
            : ((l = 9), a++);
          break;
        case 58:
          (l = 16), a++;
          break;
        case 59:
          (l = 17), a++;
          break;
        case 60:
          r(a + 1) === 33 && r(a + 2) === 45 && r(a + 3) === 45
            ? ((l = 14), (a = a + 4))
            : ((l = 9), a++);
          break;
        case 64:
          Fe(r(a + 1), r(a + 2), r(a + 3))
            ? ((l = 3), (a = rt(e, a + 1)))
            : ((l = 9), a++);
          break;
        case 91:
          (l = 19), a++;
          break;
        case 92:
          $(p, r(a + 1)) ? o() : ((l = 9), a++);
          break;
        case 93:
          (l = 20), a++;
          break;
        case 123:
          (l = 23), a++;
          break;
        case 125:
          (l = 24), a++;
          break;
        case lr:
          n();
          break;
        case At:
          o();
          break;
        default:
          (l = 9), a++;
      }
      t(l, c, (c = a));
    }
  }
  var je = null,
    D = class {
      static createItem(t) {
        return { prev: null, next: null, data: t };
      }
      constructor() {
        (this.head = null), (this.tail = null), (this.cursor = null);
      }
      createItem(t) {
        return D.createItem(t);
      }
      allocateCursor(t, r) {
        let n;
        return (
          je !== null
            ? ((n = je),
              (je = je.cursor),
              (n.prev = t),
              (n.next = r),
              (n.cursor = this.cursor))
            : (n = { prev: t, next: r, cursor: this.cursor }),
          (this.cursor = n),
          n
        );
      }
      releaseCursor() {
        let { cursor: t } = this;
        (this.cursor = t.cursor),
          (t.prev = null),
          (t.next = null),
          (t.cursor = je),
          (je = t);
      }
      updateCursors(t, r, n, o) {
        let { cursor: i } = this;
        for (; i !== null; )
          i.prev === t && (i.prev = r),
            i.next === n && (i.next = o),
            (i = i.cursor);
      }
      *[Symbol.iterator]() {
        for (let t = this.head; t !== null; t = t.next) yield t.data;
      }
      get size() {
        let t = 0;
        for (let r = this.head; r !== null; r = r.next) t++;
        return t;
      }
      get isEmpty() {
        return this.head === null;
      }
      get first() {
        return this.head && this.head.data;
      }
      get last() {
        return this.tail && this.tail.data;
      }
      fromArray(t) {
        let r = null;
        this.head = null;
        for (let n of t) {
          let o = D.createItem(n);
          r !== null ? (r.next = o) : (this.head = o), (o.prev = r), (r = o);
        }
        return (this.tail = r), this;
      }
      toArray() {
        return [...this];
      }
      toJSON() {
        return [...this];
      }
      forEach(t, r = this) {
        let n = this.allocateCursor(null, this.head);
        for (; n.next !== null; ) {
          let o = n.next;
          (n.next = o.next), t.call(r, o.data, o, this);
        }
        this.releaseCursor();
      }
      forEachRight(t, r = this) {
        let n = this.allocateCursor(this.tail, null);
        for (; n.prev !== null; ) {
          let o = n.prev;
          (n.prev = o.prev), t.call(r, o.data, o, this);
        }
        this.releaseCursor();
      }
      reduce(t, r, n = this) {
        let o = this.allocateCursor(null, this.head),
          i = r,
          s;
        for (; o.next !== null; )
          (s = o.next), (o.next = s.next), (i = t.call(n, i, s.data, s, this));
        return this.releaseCursor(), i;
      }
      reduceRight(t, r, n = this) {
        let o = this.allocateCursor(this.tail, null),
          i = r,
          s;
        for (; o.prev !== null; )
          (s = o.prev), (o.prev = s.prev), (i = t.call(n, i, s.data, s, this));
        return this.releaseCursor(), i;
      }
      some(t, r = this) {
        for (let n = this.head; n !== null; n = n.next)
          if (t.call(r, n.data, n, this)) return !0;
        return !1;
      }
      map(t, r = this) {
        let n = new D();
        for (let o = this.head; o !== null; o = o.next)
          n.appendData(t.call(r, o.data, o, this));
        return n;
      }
      filter(t, r = this) {
        let n = new D();
        for (let o = this.head; o !== null; o = o.next)
          t.call(r, o.data, o, this) && n.appendData(o.data);
        return n;
      }
      nextUntil(t, r, n = this) {
        if (t === null) return;
        let o = this.allocateCursor(null, t);
        for (; o.next !== null; ) {
          let i = o.next;
          if (((o.next = i.next), r.call(n, i.data, i, this))) break;
        }
        this.releaseCursor();
      }
      prevUntil(t, r, n = this) {
        if (t === null) return;
        let o = this.allocateCursor(t, null);
        for (; o.prev !== null; ) {
          let i = o.prev;
          if (((o.prev = i.prev), r.call(n, i.data, i, this))) break;
        }
        this.releaseCursor();
      }
      clear() {
        (this.head = null), (this.tail = null);
      }
      copy() {
        let t = new D();
        for (let r of this) t.appendData(r);
        return t;
      }
      prepend(t) {
        return (
          this.updateCursors(null, t, this.head, t),
          this.head !== null
            ? ((this.head.prev = t), (t.next = this.head))
            : (this.tail = t),
          (this.head = t),
          this
        );
      }
      prependData(t) {
        return this.prepend(D.createItem(t));
      }
      append(t) {
        return this.insert(t);
      }
      appendData(t) {
        return this.insert(D.createItem(t));
      }
      insert(t, r = null) {
        if (r !== null)
          if ((this.updateCursors(r.prev, t, r, t), r.prev === null)) {
            if (this.head !== r)
              throw new Error("before doesn't belong to list");
            (this.head = t),
              (r.prev = t),
              (t.next = r),
              this.updateCursors(null, t);
          } else
            (r.prev.next = t), (t.prev = r.prev), (r.prev = t), (t.next = r);
        else
          this.updateCursors(this.tail, t, null, t),
            this.tail !== null
              ? ((this.tail.next = t), (t.prev = this.tail))
              : (this.head = t),
            (this.tail = t);
        return this;
      }
      insertData(t, r) {
        return this.insert(D.createItem(t), r);
      }
      remove(t) {
        if ((this.updateCursors(t, t.prev, t, t.next), t.prev !== null))
          t.prev.next = t.next;
        else {
          if (this.head !== t) throw new Error("item doesn't belong to list");
          this.head = t.next;
        }
        if (t.next !== null) t.next.prev = t.prev;
        else {
          if (this.tail !== t) throw new Error("item doesn't belong to list");
          this.tail = t.prev;
        }
        return (t.prev = null), (t.next = null), t;
      }
      push(t) {
        this.insert(D.createItem(t));
      }
      pop() {
        return this.tail !== null ? this.remove(this.tail) : null;
      }
      unshift(t) {
        this.prepend(D.createItem(t));
      }
      shift() {
        return this.head !== null ? this.remove(this.head) : null;
      }
      prependList(t) {
        return this.insertList(t, this.head);
      }
      appendList(t) {
        return this.insertList(t);
      }
      insertList(t, r) {
        return t.head === null
          ? this
          : (r != null
              ? (this.updateCursors(r.prev, t.tail, r, t.head),
                r.prev !== null
                  ? ((r.prev.next = t.head), (t.head.prev = r.prev))
                  : (this.head = t.head),
                (r.prev = t.tail),
                (t.tail.next = r))
              : (this.updateCursors(this.tail, t.tail, null, t.head),
                this.tail !== null
                  ? ((this.tail.next = t.head), (t.head.prev = this.tail))
                  : (this.head = t.head),
                (this.tail = t.tail)),
            (t.head = null),
            (t.tail = null),
            this);
      }
      replace(t, r) {
        "head" in r ? this.insertList(r, t) : this.insert(r, t), this.remove(t);
      }
    };
  function Le(e, t) {
    let r = Object.create(SyntaxError.prototype),
      n = new Error();
    return Object.assign(r, {
      name: e,
      message: t,
      get stack() {
        return (n.stack || "").replace(
          /^(.+\n){1,3}/,
          `${e}: ${t}
`
        );
      },
    });
  }
  var pr = 100,
    Vo = 60,
    Ko = "    ";
  function Qo({ source: e, line: t, column: r }, n) {
    function o(l, p) {
      return i
        .slice(l, p)
        .map((m, f) => String(l + f + 1).padStart(c) + " |" + m).join(`
`);
    }
    let i = e.split(/\r\n?|\n|\f/),
      s = Math.max(1, t - n) - 1,
      u = Math.min(t + n, i.length + 1),
      c = Math.max(4, String(u).length) + 1,
      a = 0;
    (r +=
      (Ko.length - 1) * (i[t - 1].substr(0, r - 1).match(/\t/g) || []).length),
      r > pr && ((a = r - Vo + 3), (r = Vo - 2));
    for (let l = s; l <= u; l++)
      l >= 0 &&
        l < i.length &&
        ((i[l] = i[l].replace(/\t/g, Ko)),
        (i[l] =
          (a > 0 && i[l].length > a ? "\u2026" : "") +
          i[l].substr(a, pr - 2) +
          (i[l].length > a + pr - 1 ? "\u2026" : "")));
    return [o(s, t), new Array(r + c + 2).join("-") + "^", o(t, u)].filter(
      Boolean
    ).join(`
`);
  }
  function hr(e, t, r, n, o) {
    return Object.assign(Le("SyntaxError", e), {
      source: t,
      offset: r,
      line: n,
      column: o,
      sourceFragment(s) {
        return Qo({ source: t, line: n, column: o }, isNaN(s) ? 0 : s);
      },
      get formattedMessage() {
        return (
          `Parse error: ${e}
` + Qo({ source: t, line: n, column: o }, 2)
        );
      },
    });
  }
  function Xo(e) {
    let t = this.createList(),
      r = !1,
      n = { recognizer: e };
    for (; !this.eof; ) {
      switch (this.tokenType) {
        case 25:
          this.next();
          continue;
        case 13:
          (r = !0), this.next();
          continue;
      }
      let o = e.getNode.call(this, n);
      if (o === void 0) break;
      r && (e.onWhiteSpace && e.onWhiteSpace.call(this, o, t, n), (r = !1)),
        t.push(o);
    }
    return r && e.onWhiteSpace && e.onWhiteSpace.call(this, null, t, n), t;
  }
  var $o = () => {},
    bs = 33,
    xs = 35,
    mr = 59,
    Zo = 123,
    Jo = 0;
  function ys(e) {
    return function () {
      return this[e]();
    };
  }
  function fr(e) {
    let t = Object.create(null);
    for (let r in e) {
      let n = e[r],
        o = n.parse || n;
      o && (t[r] = o);
    }
    return t;
  }
  function ks(e) {
    let t = {
      context: Object.create(null),
      scope: Object.assign(Object.create(null), e.scope),
      atrule: fr(e.atrule),
      pseudo: fr(e.pseudo),
      node: fr(e.node),
    };
    for (let r in e.parseContext)
      switch (typeof e.parseContext[r]) {
        case "function":
          t.context[r] = e.parseContext[r];
          break;
        case "string":
          t.context[r] = ys(e.parseContext[r]);
          break;
      }
    return { config: t, ...t, ...t.node };
  }
  function ei(e) {
    let t = "",
      r = "<unknown>",
      n = !1,
      o = $o,
      i = !1,
      s = new Pt(),
      u = Object.assign(new Ue(), ks(e || {}), {
        parseAtrulePrelude: !0,
        parseRulePrelude: !0,
        parseValue: !0,
        parseCustomProperty: !1,
        readSequence: Xo,
        consumeUntilBalanceEnd: () => 0,
        consumeUntilLeftCurlyBracket(a) {
          return a === Zo ? 1 : 0;
        },
        consumeUntilLeftCurlyBracketOrSemicolon(a) {
          return a === Zo || a === mr ? 1 : 0;
        },
        consumeUntilExclamationMarkOrSemicolon(a) {
          return a === bs || a === mr ? 1 : 0;
        },
        consumeUntilSemicolonIncluded(a) {
          return a === mr ? 2 : 0;
        },
        createList() {
          return new D();
        },
        createSingleNodeList(a) {
          return new D().appendData(a);
        },
        getFirstListNode(a) {
          return a && a.first;
        },
        getLastListNode(a) {
          return a && a.last;
        },
        parseWithFallback(a, l) {
          let p = this.tokenIndex;
          try {
            return a.call(this);
          } catch (m) {
            if (i) throw m;
            let f = l.call(this, p);
            return (i = !0), o(m, f), (i = !1), f;
          }
        },
        lookupNonWSType(a) {
          let l;
          do if (((l = this.lookupType(a++)), l !== 13)) return l;
          while (l !== Jo);
          return Jo;
        },
        charCodeAt(a) {
          return a >= 0 && a < t.length ? t.charCodeAt(a) : 0;
        },
        substring(a, l) {
          return t.substring(a, l);
        },
        substrToCursor(a) {
          return this.source.substring(a, this.tokenStart);
        },
        cmpChar(a, l) {
          return de(t, a, l);
        },
        cmpStr(a, l, p) {
          return ge(t, a, l, p);
        },
        consume(a) {
          let l = this.tokenStart;
          return this.eat(a), this.substrToCursor(l);
        },
        consumeFunctionName() {
          let a = t.substring(this.tokenStart, this.tokenEnd - 1);
          return this.eat(2), a;
        },
        consumeNumber(a) {
          let l = t.substring(this.tokenStart, Te(t, this.tokenStart));
          return this.eat(a), l;
        },
        eat(a) {
          if (this.tokenType !== a) {
            let l = Ee[a]
                .slice(0, -6)
                .replace(/-/g, " ")
                .replace(/^./, (f) => f.toUpperCase()),
              p = `${/[[\](){}]/.test(l) ? `"${l}"` : l} is expected`,
              m = this.tokenStart;
            switch (a) {
              case 1:
                this.tokenType === 2 || this.tokenType === 7
                  ? ((m = this.tokenEnd - 1),
                    (p = "Identifier is expected but function found"))
                  : (p = "Identifier is expected");
                break;
              case 4:
                this.isDelim(xs) &&
                  (this.next(), m++, (p = "Name is expected"));
                break;
              case 11:
                this.tokenType === 10 &&
                  ((m = this.tokenEnd), (p = "Percent sign is expected"));
                break;
            }
            this.error(p, m);
          }
          this.next();
        },
        eatIdent(a) {
          (this.tokenType !== 1 || this.lookupValue(0, a) === !1) &&
            this.error(`Identifier "${a}" is expected`),
            this.next();
        },
        eatDelim(a) {
          this.isDelim(a) ||
            this.error(`Delim "${String.fromCharCode(a)}" is expected`),
            this.next();
        },
        getLocation(a, l) {
          return n ? s.getLocationRange(a, l, r) : null;
        },
        getLocationFromList(a) {
          if (n) {
            let l = this.getFirstListNode(a),
              p = this.getLastListNode(a);
            return s.getLocationRange(
              l !== null ? l.loc.start.offset - s.startOffset : this.tokenStart,
              p !== null ? p.loc.end.offset - s.startOffset : this.tokenStart,
              r
            );
          }
          return null;
        },
        error(a, l) {
          let p =
            typeof l < "u" && l < t.length
              ? s.getLocation(l)
              : this.eof
              ? s.getLocation(Wo(t, t.length - 1))
              : s.getLocation(this.tokenStart);
          throw new hr(a || "Unexpected input", t, p.offset, p.line, p.column);
        },
      });
    return Object.assign(
      function (a, l) {
        (t = a),
          (l = l || {}),
          u.setSource(t, ve),
          s.setSource(t, l.offset, l.line, l.column),
          (r = l.filename || "<unknown>"),
          (n = Boolean(l.positions)),
          (o = typeof l.onParseError == "function" ? l.onParseError : $o),
          (i = !1),
          (u.parseAtrulePrelude =
            "parseAtrulePrelude" in l ? Boolean(l.parseAtrulePrelude) : !0),
          (u.parseRulePrelude =
            "parseRulePrelude" in l ? Boolean(l.parseRulePrelude) : !0),
          (u.parseValue = "parseValue" in l ? Boolean(l.parseValue) : !0),
          (u.parseCustomProperty =
            "parseCustomProperty" in l ? Boolean(l.parseCustomProperty) : !1);
        let { context: p = "default", onComment: m } = l;
        if (!(p in u.context)) throw new Error("Unknown context `" + p + "`");
        typeof m == "function" &&
          u?.forEachToken((P, te, X) => {
            if (P === 25) {
              let S = u.getLocation(te, X),
                R = ge(t, X - 2, X, "*/")
                  ? t.slice(te + 2, X - 2)
                  : t.slice(te + 2, X);
              m(R, S);
            }
          });
        let f = u.context[p].call(u, l);
        return u.eof || u.error(), f;
      },
      { SyntaxError: hr, config: u.config }
    );
  }
  var wi = ls(yi(), 1),
    ki = new Set(["Atrule", "Selector", "Declaration"]);
  function vi(e) {
    let t = new wi.SourceMapGenerator(),
      r = { line: 1, column: 0 },
      n = { line: 0, column: 0 },
      o = { line: 1, column: 0 },
      i = { generated: o },
      s = 1,
      u = 0,
      c = !1,
      a = e.node;
    e.node = function (m) {
      if (m.loc && m.loc.start && ki.has(m.type)) {
        let f = m.loc.start.line,
          P = m.loc.start.column - 1;
        (n.line !== f || n.column !== P) &&
          ((n.line = f),
          (n.column = P),
          (r.line = s),
          (r.column = u),
          c &&
            ((c = !1),
            (r.line !== o.line || r.column !== o.column) && t.addMapping(i)),
          (c = !0),
          t.addMapping({ source: m.loc.source, original: n, generated: r }));
      }
      a.call(this, m), c && ki.has(m.type) && ((o.line = s), (o.column = u));
    };
    let l = e.emit;
    e.emit = function (m, f, P) {
      for (let te = 0; te < m.length; te++)
        m.charCodeAt(te) === 10 ? (s++, (u = 0)) : u++;
      l(m, f, P);
    };
    let p = e.result;
    return (
      (e.result = function () {
        return c && t.addMapping(i), { css: p(), map: t };
      }),
      e
    );
  }
  var Nt = {};
  b(Nt, { safe: () => vr, spec: () => js });
  var _s = 43,
    qs = 45,
    wr = (e, t) => {
      if ((e === 9 && (e = t), typeof e == "string")) {
        let r = e.charCodeAt(0);
        return r > 127 ? 32768 : r << 8;
      }
      return e;
    },
    Si = [
      [1, 1],
      [1, 2],
      [1, 7],
      [1, 8],
      [1, "-"],
      [1, 10],
      [1, 11],
      [1, 12],
      [1, 15],
      [1, 21],
      [3, 1],
      [3, 2],
      [3, 7],
      [3, 8],
      [3, "-"],
      [3, 10],
      [3, 11],
      [3, 12],
      [3, 15],
      [4, 1],
      [4, 2],
      [4, 7],
      [4, 8],
      [4, "-"],
      [4, 10],
      [4, 11],
      [4, 12],
      [4, 15],
      [12, 1],
      [12, 2],
      [12, 7],
      [12, 8],
      [12, "-"],
      [12, 10],
      [12, 11],
      [12, 12],
      [12, 15],
      ["#", 1],
      ["#", 2],
      ["#", 7],
      ["#", 8],
      ["#", "-"],
      ["#", 10],
      ["#", 11],
      ["#", 12],
      ["#", 15],
      ["-", 1],
      ["-", 2],
      ["-", 7],
      ["-", 8],
      ["-", "-"],
      ["-", 10],
      ["-", 11],
      ["-", 12],
      ["-", 15],
      [10, 1],
      [10, 2],
      [10, 7],
      [10, 8],
      [10, 10],
      [10, 11],
      [10, 12],
      [10, "%"],
      [10, 15],
      ["@", 1],
      ["@", 2],
      ["@", 7],
      ["@", 8],
      ["@", "-"],
      ["@", 15],
      [".", 10],
      [".", 11],
      [".", 12],
      ["+", 10],
      ["+", 11],
      ["+", 12],
      ["/", "*"],
    ],
    Us = Si.concat([
      [1, 4],
      [12, 4],
      [4, 4],
      [3, 21],
      [3, 5],
      [3, 16],
      [11, 11],
      [11, 12],
      [11, 2],
      [11, "-"],
      [22, 1],
      [22, 2],
      [22, 11],
      [22, 12],
      [22, 4],
      [22, "-"],
    ]);
  function Ci(e) {
    let t = new Set(e.map(([r, n]) => (wr(r) << 16) | wr(n)));
    return function (r, n, o) {
      let i = wr(n, o),
        s = o.charCodeAt(0);
      return (
        ((s === qs && n !== 1 && n !== 2 && n !== 15) || s === _s
          ? t.has((r << 16) | (s << 8))
          : t.has((r << 16) | i)) && this.emit(" ", 13, !0),
        i
      );
    };
  }
  var js = Ci(Si),
    vr = Ci(Us);
  var Ws = 92;
  function Hs(e, t) {
    if (typeof t == "function") {
      let r = null;
      e.children?.forEach((n) => {
        r !== null && t.call(this, r), this.node(n), (r = n);
      });
      return;
    }
    e.children?.forEach(this.node, this);
  }
  function Ys(e) {
    ve(e, (t, r, n) => {
      this.token(t, e.slice(r, n));
    });
  }
  function Ai(e) {
    let t = new Map();
    for (let r in e.node) {
      let n = e.node[r];
      typeof (n.generate || n) == "function" && t.set(r, n.generate || n);
    }
    return function (r, n) {
      let o = "",
        i = 0,
        s = {
          node(c) {
            if (t.has(c.type)) t.get(c.type).call(u, c);
            else throw new Error("Unknown node type: " + c.type);
          },
          tokenBefore: vr,
          token(c, a) {
            (i = this.tokenBefore(i, c, a)),
              this.emit(a, c, !1),
              c === 9 &&
                a.charCodeAt(0) === Ws &&
                this.emit(
                  `
`,
                  13,
                  !0
                );
          },
          emit(c) {
            o += c;
          },
          result() {
            return o;
          },
        };
      n &&
        (typeof n.decorator == "function" && (s = n.decorator(s)),
        n.sourceMap && (s = vi(s)),
        n.mode in Nt && (s.tokenBefore = Nt[n.mode]));
      let u = {
        node: (c) => s.node(c),
        children: Hs,
        token: (c, a) => s.token(c, a),
        tokenize: Ys,
      };
      return s.node(r), s.result();
    };
  }
  function Ti(e) {
    return {
      fromPlainObject(t) {
        return (
          e(t, {
            enter(r) {
              r.children &&
                !(r.children instanceof D) &&
                (r.children = new D().fromArray(r.children));
            },
          }),
          t
        );
      },
      toPlainObject(t) {
        return (
          e(t, {
            leave(r) {
              r.children &&
                r.children instanceof D &&
                (r.children = r.children.toArray());
            },
          }),
          t
        );
      },
    };
  }
  var { hasOwnProperty: Sr } = Object.prototype,
    it = function () {};
  function Ei(e) {
    return typeof e == "function" ? e : it;
  }
  function Li(e, t) {
    return function (r, n, o) {
      r.type === t && e.call(this, r, n, o);
    };
  }
  function Gs(e, t) {
    let r = t.structure,
      n = [];
    for (let o in r) {
      if (Sr.call(r, o) === !1) continue;
      let i = r[o],
        s = { name: o, type: !1, nullable: !1 };
      Array.isArray(i) || (i = [i]);
      for (let u of i)
        u === null
          ? (s.nullable = !0)
          : typeof u == "string"
          ? (s.type = "node")
          : Array.isArray(u) && (s.type = "list");
      s.type && n.push(s);
    }
    return n.length ? { context: t.walkContext, fields: n } : null;
  }
  function Vs(e) {
    let t = {};
    for (let r in e.node)
      if (Sr.call(e.node, r)) {
        let n = e.node[r];
        if (!n.structure)
          throw new Error(
            "Missed `structure` field in `" + r + "` node type definition"
          );
        t[r] = Gs(r, n);
      }
    return t;
  }
  function Pi(e, t) {
    let r = e.fields.slice(),
      n = e.context,
      o = typeof n == "string";
    return (
      t && r.reverse(),
      function (i, s, u, c) {
        let a;
        o && ((a = s[n]), (s[n] = i));
        for (let l of r) {
          let p = i[l.name];
          if (!l.nullable || p) {
            if (l.type === "list") {
              if (t ? p.reduceRight(c, !1) : p.reduce(c, !1)) return !0;
            } else if (u(p)) return !0;
          }
        }
        o && (s[n] = a);
      }
    );
  }
  function Ii({
    StyleSheet: e,
    Atrule: t,
    Rule: r,
    Block: n,
    DeclarationList: o,
  }) {
    return {
      Atrule: { StyleSheet: e, Atrule: t, Rule: r, Block: n },
      Rule: { StyleSheet: e, Atrule: t, Rule: r, Block: n },
      Declaration: {
        StyleSheet: e,
        Atrule: t,
        Rule: r,
        Block: n,
        DeclarationList: o,
      },
    };
  }
  function Di(e) {
    let t = Vs(e),
      r = {},
      n = {},
      o = Symbol("break-walk"),
      i = Symbol("skip-node");
    for (let a in t)
      Sr.call(t, a) &&
        t[a] !== null &&
        ((r[a] = Pi(t[a], !1)), (n[a] = Pi(t[a], !0)));
    let s = Ii(r),
      u = Ii(n),
      c = function (a, l) {
        function p(S, R, ke) {
          let z = m.call(X, S, R, ke);
          return z === o
            ? !0
            : z === i
            ? !1
            : !!(
                (P.hasOwnProperty(S.type) && P[S.type](S, X, p, te)) ||
                f.call(X, S, R, ke) === o
              );
        }
        let m = it,
          f = it,
          P = r,
          te = (S, R, ke, z) => S || p(R, ke, z),
          X = {
            break: o,
            skip: i,
            root: a,
            stylesheet: null,
            atrule: null,
            atrulePrelude: null,
            rule: null,
            selector: null,
            block: null,
            declaration: null,
            function: null,
          };
        if (typeof l == "function") m = l;
        else if (
          l &&
          ((m = Ei(l.enter)), (f = Ei(l.leave)), l.reverse && (P = n), l.visit)
        ) {
          if (s.hasOwnProperty(l.visit))
            P = l.reverse ? u[l.visit] : s[l.visit];
          else if (!t.hasOwnProperty(l.visit))
            throw new Error(
              "Bad value `" +
                l.visit +
                "` for `visit` option (should be: " +
                Object.keys(t).sort().join(", ") +
                ")"
            );
          (m = Li(m, l.visit)), (f = Li(f, l.visit));
        }
        if (m === it && f === it)
          throw new Error(
            "Neither `enter` nor `leave` walker handler is set or both aren't a function"
          );
        p(a);
      };
    return (
      (c.break = o),
      (c.skip = i),
      (c.find = function (a, l) {
        let p = null;
        return (
          c(a, function (m, f, P) {
            if (l.call(this, m, f, P)) return (p = m), o;
          }),
          p
        );
      }),
      (c.findLast = function (a, l) {
        let p = null;
        return (
          c(a, {
            reverse: !0,
            enter(m, f, P) {
              if (l.call(this, m, f, P)) return (p = m), o;
            },
          }),
          p
        );
      }),
      (c.findAll = function (a, l) {
        let p = [];
        return (
          c(a, function (m, f, P) {
            l.call(this, m, f, P) && p.push(m);
          }),
          p
        );
      }),
      c
    );
  }
  function Ks(e) {
    return e;
  }
  function Qs(e) {
    let { min: t, max: r, comma: n } = e;
    return t === 0 && r === 0
      ? n
        ? "#?"
        : "*"
      : t === 0 && r === 1
      ? "?"
      : t === 1 && r === 0
      ? n
        ? "#"
        : "+"
      : t === 1 && r === 1
      ? ""
      : (n ? "#" : "") +
        (t === r ? "{" + t + "}" : "{" + t + "," + (r !== 0 ? r : "") + "}");
  }
  function Xs(e) {
    switch (e.type) {
      case "Range":
        return (
          " [" +
          (e.min === null ? "-\u221E" : e.min) +
          "," +
          (e.max === null ? "\u221E" : e.max) +
          "]"
        );
      default:
        throw new Error("Unknown node type `" + e.type + "`");
    }
  }
  function $s(e, t, r, n) {
    let o = e.combinator === " " || n ? e.combinator : " " + e.combinator + " ",
      i = e.terms.map((s) => Cr(s, t, r, n)).join(o);
    return e.explicit || r
      ? (n || i[0] === "," ? "[" : "[ ") + i + (n ? "]" : " ]")
      : i;
  }
  function Cr(e, t, r, n) {
    let o;
    switch (e.type) {
      case "Group":
        o = $s(e, t, r, n) + (e.disallowEmpty ? "!" : "");
        break;
      case "Multiplier":
        return Cr(e.term, t, r, n) + t(Qs(e), e);
      case "Type":
        o = "<" + e.name + (e.opts ? t(Xs(e.opts), e.opts) : "") + ">";
        break;
      case "Property":
        o = "<'" + e.name + "'>";
        break;
      case "Keyword":
        o = e.name;
        break;
      case "AtKeyword":
        o = "@" + e.name;
        break;
      case "Function":
        o = e.name + "(";
        break;
      case "String":
      case "Token":
        o = e.value;
        break;
      case "Comma":
        o = ",";
        break;
      default:
        throw new Error("Unknown node type `" + e.type + "`");
    }
    return t(o, e);
  }
  function Ie(e, t) {
    let r = Ks,
      n = !1,
      o = !1;
    return (
      typeof t == "function"
        ? (r = t)
        : t &&
          ((n = Boolean(t.forceBraces)),
          (o = Boolean(t.compact)),
          typeof t.decorate == "function" && (r = t.decorate)),
      Cr(e, r, n, o)
    );
  }
  var Oi = { offset: 0, line: 1, column: 1 };
  function Zs(e, t) {
    let r = e.tokens,
      n = e.longestMatch,
      o = (n < r.length && r[n].node) || null,
      i = o !== t ? o : null,
      s = 0,
      u = 0,
      c = 0,
      a = "",
      l,
      p;
    for (let m = 0; m < r.length; m++) {
      let f = r[m].value;
      m === n && ((u = f.length), (s = a.length)),
        i !== null && r[m].node === i && (m <= n ? c++ : (c = 0)),
        (a += f);
    }
    return (
      n === r.length || c > 1
        ? ((l = zt(i || t, "end") || at(Oi, a)), (p = at(l)))
        : ((l = zt(i, "start") || at(zt(t, "start") || Oi, a.slice(0, s))),
          (p = zt(i, "end") || at(l, a.substr(s, u)))),
      { css: a, mismatchOffset: s, mismatchLength: u, start: l, end: p }
    );
  }
  function zt(e, t) {
    let r = e && e.loc && e.loc[t];
    return r ? ("line" in r ? at(r) : r) : null;
  }
  function at({ offset: e, line: t, column: r }, n) {
    let o = { offset: e, line: t, column: r };
    if (n) {
      let i = n.split(/\n|\r\n?|\f/);
      (o.offset += n.length),
        (o.line += i.length - 1),
        (o.column = i.length === 1 ? o.column + n.length : i.pop().length + 1);
    }
    return o;
  }
  var He = function (e, t) {
      let r = Le("SyntaxReferenceError", e + (t ? " `" + t + "`" : ""));
      return (r.reference = t), r;
    },
    Ni = function (e, t, r, n) {
      let o = Le("SyntaxMatchError", e),
        {
          css: i,
          mismatchOffset: s,
          mismatchLength: u,
          start: c,
          end: a,
        } = Zs(n, r);
      return (
        (o.rawMessage = e),
        (o.syntax = t ? Ie(t) : "<generic>"),
        (o.css = i),
        (o.mismatchOffset = s),
        (o.mismatchLength = u),
        (o.message =
          e +
          `
  syntax: ` +
          o.syntax +
          `
   value: ` +
          (i || "<empty string>") +
          `
  --------` +
          new Array(o.mismatchOffset + 1).join("-") +
          "^"),
        Object.assign(o, c),
        (o.loc = {
          source: (r && r.loc && r.loc.source) || "<unknown>",
          start: c,
          end: a,
        }),
        o
      );
    };
  var Mt = new Map(),
    Ye = new Map(),
    Rt = 45,
    st = el,
    Ft = tl,
    Js = Ar;
  function lt(e, t) {
    return (
      (t = t || 0),
      e.length - t >= 2 && e.charCodeAt(t) === Rt && e.charCodeAt(t + 1) === Rt
    );
  }
  function Ar(e, t) {
    if (
      ((t = t || 0),
      e.length - t >= 3 && e.charCodeAt(t) === Rt && e.charCodeAt(t + 1) !== Rt)
    ) {
      let r = e.indexOf("-", t + 2);
      if (r !== -1) return e.substring(t, r + 1);
    }
    return "";
  }
  function el(e) {
    if (Mt.has(e)) return Mt.get(e);
    let t = e.toLowerCase(),
      r = Mt.get(t);
    if (r === void 0) {
      let n = lt(t, 0),
        o = n ? "" : Ar(t, 0);
      r = Object.freeze({
        basename: t.substr(o.length),
        name: t,
        prefix: o,
        vendor: o,
        custom: n,
      });
    }
    return Mt.set(e, r), r;
  }
  function tl(e) {
    if (Ye.has(e)) return Ye.get(e);
    let t = e,
      r = e[0];
    r === "/"
      ? (r = e[1] === "/" ? "//" : "/")
      : r !== "_" &&
        r !== "*" &&
        r !== "$" &&
        r !== "#" &&
        r !== "+" &&
        r !== "&" &&
        (r = "");
    let n = lt(t, r.length);
    if (!n && ((t = t.toLowerCase()), Ye.has(t))) {
      let u = Ye.get(t);
      return Ye.set(e, u), u;
    }
    let o = n ? "" : Ar(t, r.length),
      i = t.substr(0, r.length + o.length),
      s = Object.freeze({
        basename: t.substr(i.length),
        name: t.substr(r.length),
        hack: r,
        vendor: o,
        prefix: i,
        custom: n,
      });
    return Ye.set(e, s), s;
  }
  var Bt = ["initial", "inherit", "unset", "revert", "revert-layer"];
  var ut = 43,
    he = 45,
    Tr = 110,
    Ge = !0,
    nl = !1;
  function Lr(e, t) {
    return e !== null && e.type === 9 && e.value.charCodeAt(0) === t;
  }
  function ct(e, t, r) {
    for (; e !== null && (e.type === 13 || e.type === 25); ) e = r(++t);
    return t;
  }
  function Se(e, t, r, n) {
    if (!e) return 0;
    let o = e.value.charCodeAt(t);
    if (o === ut || o === he) {
      if (r) return 0;
      t++;
    }
    for (; t < e.value.length; t++) if (!B(e.value.charCodeAt(t))) return 0;
    return n + 1;
  }
  function Er(e, t, r) {
    let n = !1,
      o = ct(e, t, r);
    if (((e = r(o)), e === null)) return t;
    if (e.type !== 10)
      if (Lr(e, ut) || Lr(e, he)) {
        if (
          ((n = !0),
          (o = ct(r(++o), o, r)),
          (e = r(o)),
          e === null || e.type !== 10)
        )
          return 0;
      } else return t;
    if (!n) {
      let i = e.value.charCodeAt(0);
      if (i !== ut && i !== he) return 0;
    }
    return Se(e, n ? 0 : 1, n, o);
  }
  function Pr(e, t) {
    let r = 0;
    if (!e) return 0;
    if (e.type === 10) return Se(e, 0, nl, r);
    if (e.type === 1 && e.value.charCodeAt(0) === he) {
      if (!de(e.value, 1, Tr)) return 0;
      switch (e.value.length) {
        case 2:
          return Er(t(++r), r, t);
        case 3:
          return e.value.charCodeAt(2) !== he
            ? 0
            : ((r = ct(t(++r), r, t)), (e = t(r)), Se(e, 0, Ge, r));
        default:
          return e.value.charCodeAt(2) !== he ? 0 : Se(e, 3, Ge, r);
      }
    } else if (e.type === 1 || (Lr(e, ut) && t(r + 1).type === 1)) {
      if ((e.type !== 1 && (e = t(++r)), e === null || !de(e.value, 0, Tr)))
        return 0;
      switch (e.value.length) {
        case 1:
          return Er(t(++r), r, t);
        case 2:
          return e.value.charCodeAt(1) !== he
            ? 0
            : ((r = ct(t(++r), r, t)), (e = t(r)), Se(e, 0, Ge, r));
        default:
          return e.value.charCodeAt(1) !== he ? 0 : Se(e, 2, Ge, r);
      }
    } else if (e.type === 12) {
      let n = e.value.charCodeAt(0),
        o = n === ut || n === he ? 1 : 0,
        i = o;
      for (; i < e.value.length && B(e.value.charCodeAt(i)); i++);
      return i === o || !de(e.value, i, Tr)
        ? 0
        : i + 1 === e.value.length
        ? Er(t(++r), r, t)
        : e.value.charCodeAt(i + 1) !== he
        ? 0
        : i + 2 === e.value.length
        ? ((r = ct(t(++r), r, t)), (e = t(r)), Se(e, 0, Ge, r))
        : Se(e, i + 2, Ge, r);
    }
    return 0;
  }
  var ol = 43,
    zi = 45,
    Mi = 63,
    il = 117;
  function Ir(e, t) {
    return e !== null && e.type === 9 && e.value.charCodeAt(0) === t;
  }
  function al(e, t) {
    return e.value.charCodeAt(0) === t;
  }
  function pt(e, t, r) {
    let n = 0;
    for (let o = t; o < e.value.length; o++) {
      let i = e.value.charCodeAt(o);
      if (i === zi && r && n !== 0) return pt(e, t + n + 1, !1), 6;
      if (!ee(i) || ++n > 6) return 0;
    }
    return n;
  }
  function _t(e, t, r) {
    if (!e) return 0;
    for (; Ir(r(t), Mi); ) {
      if (++e > 6) return 0;
      t++;
    }
    return t;
  }
  function Dr(e, t) {
    let r = 0;
    if (
      e === null ||
      e.type !== 1 ||
      !de(e.value, 0, il) ||
      ((e = t(++r)), e === null)
    )
      return 0;
    if (Ir(e, ol))
      return (
        (e = t(++r)),
        e === null
          ? 0
          : e.type === 1
          ? _t(pt(e, 0, !0), ++r, t)
          : Ir(e, Mi)
          ? _t(1, ++r, t)
          : 0
      );
    if (e.type === 10) {
      let n = pt(e, 1, !0);
      return n === 0
        ? 0
        : ((e = t(++r)),
          e === null
            ? r
            : e.type === 12 || e.type === 10
            ? !al(e, zi) || !pt(e, 1, !1)
              ? 0
              : r + 1
            : _t(n, r, t));
    }
    return e.type === 12 ? _t(pt(e, 1, !0), ++r, t) : 0;
  }
  var sl = ["calc(", "-moz-calc(", "-webkit-calc("],
    Or = new Map([
      [2, 22],
      [21, 22],
      [19, 20],
      [23, 24],
    ]);
  function le(e, t) {
    return t < e.length ? e.charCodeAt(t) : 0;
  }
  function Ri(e, t) {
    return ge(e, 0, e.length, t);
  }
  function Fi(e, t) {
    for (let r = 0; r < t.length; r++) if (Ri(e, t[r])) return !0;
    return !1;
  }
  function Bi(e, t) {
    return t !== e.length - 2 ? !1 : le(e, t) === 92 && B(le(e, t + 1));
  }
  function qt(e, t, r) {
    if (e && e.type === "Range") {
      let n = Number(r !== void 0 && r !== t.length ? t.substr(0, r) : t);
      if (
        isNaN(n) ||
        (e.min !== null && n < e.min && typeof e.min != "string") ||
        (e.max !== null && n > e.max && typeof e.max != "string")
      )
        return !0;
    }
    return !1;
  }
  function ll(e, t) {
    let r = 0,
      n = [],
      o = 0;
    e: do {
      switch (e.type) {
        case 24:
        case 22:
        case 20:
          if (e.type !== r) break e;
          if (((r = n.pop()), n.length === 0)) {
            o++;
            break e;
          }
          break;
        case 2:
        case 21:
        case 19:
        case 23:
          n.push(r), (r = Or.get(e.type));
          break;
      }
      o++;
    } while ((e = t(o)));
    return o;
  }
  function ie(e) {
    return function (t, r, n) {
      return t === null
        ? 0
        : t.type === 2 && Fi(t.value, sl)
        ? ll(t, r)
        : e(t, r, n);
    };
  }
  function N(e) {
    return function (t) {
      return t === null || t.type !== e ? 0 : 1;
    };
  }
  function cl(e) {
    if (e === null || e.type !== 1) return 0;
    let t = e.value.toLowerCase();
    return Fi(t, Bt) || Ri(t, "default") ? 0 : 1;
  }
  function ul(e) {
    return e === null ||
      e.type !== 1 ||
      le(e.value, 0) !== 45 ||
      le(e.value, 1) !== 45
      ? 0
      : 1;
  }
  function pl(e) {
    if (e === null || e.type !== 4) return 0;
    let t = e.value.length;
    if (t !== 4 && t !== 5 && t !== 7 && t !== 9) return 0;
    for (let r = 1; r < t; r++) if (!ee(le(e.value, r))) return 0;
    return 1;
  }
  function hl(e) {
    return e === null ||
      e.type !== 4 ||
      !Fe(le(e.value, 1), le(e.value, 2), le(e.value, 3))
      ? 0
      : 1;
  }
  function ml(e, t) {
    if (!e) return 0;
    let r = 0,
      n = [],
      o = 0;
    e: do {
      switch (e.type) {
        case 6:
        case 8:
          break e;
        case 24:
        case 22:
        case 20:
          if (e.type !== r) break e;
          r = n.pop();
          break;
        case 17:
          if (r === 0) break e;
          break;
        case 9:
          if (r === 0 && e.value === "!") break e;
          break;
        case 2:
        case 21:
        case 19:
        case 23:
          n.push(r), (r = Or.get(e.type));
          break;
      }
      o++;
    } while ((e = t(o)));
    return o;
  }
  function fl(e, t) {
    if (!e) return 0;
    let r = 0,
      n = [],
      o = 0;
    e: do {
      switch (e.type) {
        case 6:
        case 8:
          break e;
        case 24:
        case 22:
        case 20:
          if (e.type !== r) break e;
          r = n.pop();
          break;
        case 2:
        case 21:
        case 19:
        case 23:
          n.push(r), (r = Or.get(e.type));
          break;
      }
      o++;
    } while ((e = t(o)));
    return o;
  }
  function ye(e) {
    return (
      e && (e = new Set(e)),
      function (t, r, n) {
        if (t === null || t.type !== 12) return 0;
        let o = Te(t.value, 0);
        if (e !== null) {
          let i = t.value.indexOf("\\", o),
            s =
              i === -1 || !Bi(t.value, i)
                ? t.value.substr(o)
                : t.value.substring(o, i);
          if (e.has(s.toLowerCase()) === !1) return 0;
        }
        return qt(n, t.value, o) ? 0 : 1;
      }
    );
  }
  function dl(e, t, r) {
    return e === null || e.type !== 11 || qt(r, e.value, e.value.length - 1)
      ? 0
      : 1;
  }
  function _i(e) {
    return (
      typeof e != "function" &&
        (e = function () {
          return 0;
        }),
      function (t, r, n) {
        return t !== null && t.type === 10 && Number(t.value) === 0
          ? 1
          : e(t, r, n);
      }
    );
  }
  function gl(e, t, r) {
    if (e === null) return 0;
    let n = Te(e.value, 0);
    return (!(n === e.value.length) && !Bi(e.value, n)) || qt(r, e.value, n)
      ? 0
      : 1;
  }
  function bl(e, t, r) {
    if (e === null || e.type !== 10) return 0;
    let n = le(e.value, 0) === 43 || le(e.value, 0) === 45 ? 1 : 0;
    for (; n < e.value.length; n++) if (!B(le(e.value, n))) return 0;
    return qt(r, e.value, n) ? 0 : 1;
  }
  var xl = {
      "ident-token": N(1),
      "function-token": N(2),
      "at-keyword-token": N(3),
      "hash-token": N(4),
      "string-token": N(5),
      "bad-string-token": N(6),
      "url-token": N(7),
      "bad-url-token": N(8),
      "delim-token": N(9),
      "number-token": N(10),
      "percentage-token": N(11),
      "dimension-token": N(12),
      "whitespace-token": N(13),
      "CDO-token": N(14),
      "CDC-token": N(15),
      "colon-token": N(16),
      "semicolon-token": N(17),
      "comma-token": N(18),
      "[-token": N(19),
      "]-token": N(20),
      "(-token": N(21),
      ")-token": N(22),
      "{-token": N(23),
      "}-token": N(24),
    },
    yl = {
      string: N(5),
      ident: N(1),
      percentage: ie(dl),
      zero: _i(),
      number: ie(gl),
      integer: ie(bl),
      "custom-ident": cl,
      "custom-property-name": ul,
      "hex-color": pl,
      "id-selector": hl,
      "an-plus-b": Pr,
      urange: Dr,
      "declaration-value": ml,
      "any-value": fl,
    };
  function kl(e) {
    let {
      angle: t,
      decibel: r,
      frequency: n,
      flex: o,
      length: i,
      resolution: s,
      semitones: u,
      time: c,
    } = e || {};
    return {
      dimension: ie(ye(null)),
      angle: ie(ye(t)),
      decibel: ie(ye(r)),
      frequency: ie(ye(n)),
      flex: ie(ye(o)),
      length: ie(_i(ye(i))),
      resolution: ie(ye(s)),
      semitones: ie(ye(u)),
      time: ie(ye(c)),
    };
  }
  function qi(e) {
    return { ...xl, ...yl, ...kl(e) };
  }
  var Ut = {};
  b(Ut, {
    angle: () => vl,
    decibel: () => El,
    flex: () => Tl,
    frequency: () => Cl,
    length: () => wl,
    resolution: () => Al,
    semitones: () => Ll,
    time: () => Sl,
  });
  var wl = [
      "cm",
      "mm",
      "q",
      "in",
      "pt",
      "pc",
      "px",
      "em",
      "rem",
      "ex",
      "rex",
      "cap",
      "rcap",
      "ch",
      "rch",
      "ic",
      "ric",
      "lh",
      "rlh",
      "vw",
      "svw",
      "lvw",
      "dvw",
      "vh",
      "svh",
      "lvh",
      "dvh",
      "vi",
      "svi",
      "lvi",
      "dvi",
      "vb",
      "svb",
      "lvb",
      "dvb",
      "vmin",
      "svmin",
      "lvmin",
      "dvmin",
      "vmax",
      "svmax",
      "lvmax",
      "dvmax",
      "cqw",
      "cqh",
      "cqi",
      "cqb",
      "cqmin",
      "cqmax",
    ],
    vl = ["deg", "grad", "rad", "turn"],
    Sl = ["s", "ms"],
    Cl = ["hz", "khz"],
    Al = ["dpi", "dpcm", "dppx", "x"],
    Tl = ["fr"],
    El = ["db"],
    Ll = ["st"];
  var qr = {};
  b(qr, {
    SyntaxError: () => jt,
    generate: () => Ie,
    parse: () => Qe,
    walk: () => Qt,
  });
  function jt(e, t, r) {
    return Object.assign(Le("SyntaxError", e), {
      input: t,
      offset: r,
      rawMessage: e,
      message:
        e +
        `
  ` +
        t +
        `
--` +
        new Array((r || t.length) + 1).join("-") +
        "^",
    });
  }
  var Pl = 9,
    Il = 10,
    Dl = 12,
    Ol = 13,
    Nl = 32,
    Wt = class {
      constructor(t) {
        (this.str = t), (this.pos = 0);
      }
      charCodeAt(t) {
        return t < this.str.length ? this.str.charCodeAt(t) : 0;
      }
      charCode() {
        return this.charCodeAt(this.pos);
      }
      nextCharCode() {
        return this.charCodeAt(this.pos + 1);
      }
      nextNonWsCode(t) {
        return this.charCodeAt(this.findWsEnd(t));
      }
      findWsEnd(t) {
        for (; t < this.str.length; t++) {
          let r = this.str.charCodeAt(t);
          if (r !== Ol && r !== Il && r !== Dl && r !== Nl && r !== Pl) break;
        }
        return t;
      }
      substringToPos(t) {
        return this.str.substring(this.pos, (this.pos = t));
      }
      eat(t) {
        this.charCode() !== t &&
          this.error("Expect `" + String.fromCharCode(t) + "`"),
          this.pos++;
      }
      peek() {
        return this.pos < this.str.length ? this.str.charAt(this.pos++) : "";
      }
      error(t) {
        throw new jt(t, this.str, this.pos);
      }
    };
  var zl = 9,
    Ml = 10,
    Rl = 12,
    Fl = 13,
    Bl = 32,
    Ki = 33,
    Mr = 35,
    Ui = 38,
    Ht = 39,
    Qi = 40,
    _l = 41,
    Xi = 42,
    Rr = 43,
    Fr = 44,
    ji = 45,
    Br = 60,
    $i = 62,
    zr = 63,
    ql = 64,
    Kt = 91,
    _r = 93,
    Yt = 123,
    Wi = 124,
    Hi = 125,
    Yi = 8734,
    ht = new Uint8Array(128).map((e, t) =>
      /[a-zA-Z0-9\-]/.test(String.fromCharCode(t)) ? 1 : 0
    ),
    Gi = { " ": 1, "&&": 2, "||": 3, "|": 4 };
  function Gt(e) {
    return e.substringToPos(e.findWsEnd(e.pos));
  }
  function Ve(e) {
    let t = e.pos;
    for (; t < e.str.length; t++) {
      let r = e.str.charCodeAt(t);
      if (r >= 128 || ht[r] === 0) break;
    }
    return e.pos === t && e.error("Expect a keyword"), e.substringToPos(t);
  }
  function Vt(e) {
    let t = e.pos;
    for (; t < e.str.length; t++) {
      let r = e.str.charCodeAt(t);
      if (r < 48 || r > 57) break;
    }
    return e.pos === t && e.error("Expect a number"), e.substringToPos(t);
  }
  function Ul(e) {
    let t = e.str.indexOf("'", e.pos + 1);
    return (
      t === -1 && ((e.pos = e.str.length), e.error("Expect an apostrophe")),
      e.substringToPos(t + 1)
    );
  }
  function Vi(e) {
    let t = null,
      r = null;
    return (
      e.eat(Yt),
      (t = Vt(e)),
      e.charCode() === Fr
        ? (e.pos++, e.charCode() !== Hi && (r = Vt(e)))
        : (r = t),
      e.eat(Hi),
      { min: Number(t), max: r ? Number(r) : 0 }
    );
  }
  function jl(e) {
    let t = null,
      r = !1;
    switch (e.charCode()) {
      case Xi:
        e.pos++, (t = { min: 0, max: 0 });
        break;
      case Rr:
        e.pos++, (t = { min: 1, max: 0 });
        break;
      case zr:
        e.pos++, (t = { min: 0, max: 1 });
        break;
      case Mr:
        e.pos++,
          (r = !0),
          e.charCode() === Yt
            ? (t = Vi(e))
            : e.charCode() === zr
            ? (e.pos++, (t = { min: 0, max: 0 }))
            : (t = { min: 1, max: 0 });
        break;
      case Yt:
        t = Vi(e);
        break;
      default:
        return null;
    }
    return { type: "Multiplier", comma: r, min: t.min, max: t.max, term: null };
  }
  function Ke(e, t) {
    let r = jl(e);
    return r !== null
      ? ((r.term = t),
        e.charCode() === Mr && e.charCodeAt(e.pos - 1) === Rr ? Ke(e, r) : r)
      : t;
  }
  function Nr(e) {
    let t = e.peek();
    return t === "" ? null : { type: "Token", value: t };
  }
  function Wl(e) {
    let t;
    return (
      e.eat(Br),
      e.eat(Ht),
      (t = Ve(e)),
      e.eat(Ht),
      e.eat($i),
      Ke(e, { type: "Property", name: t })
    );
  }
  function Hl(e) {
    let t = null,
      r = null,
      n = 1;
    return (
      e.eat(Kt),
      e.charCode() === ji && (e.peek(), (n = -1)),
      n == -1 && e.charCode() === Yi
        ? e.peek()
        : ((t = n * Number(Vt(e))), ht[e.charCode()] !== 0 && (t += Ve(e))),
      Gt(e),
      e.eat(Fr),
      Gt(e),
      e.charCode() === Yi
        ? e.peek()
        : ((n = 1),
          e.charCode() === ji && (e.peek(), (n = -1)),
          (r = n * Number(Vt(e))),
          ht[e.charCode()] !== 0 && (r += Ve(e))),
      e.eat(_r),
      { type: "Range", min: t, max: r }
    );
  }
  function Yl(e) {
    let t,
      r = null;
    return (
      e.eat(Br),
      (t = Ve(e)),
      e.charCode() === Qi &&
        e.nextCharCode() === _l &&
        ((e.pos += 2), (t += "()")),
      e.charCodeAt(e.findWsEnd(e.pos)) === Kt && (Gt(e), (r = Hl(e))),
      e.eat($i),
      Ke(e, { type: "Type", name: t, opts: r })
    );
  }
  function Gl(e) {
    let t = Ve(e);
    return e.charCode() === Qi
      ? (e.pos++, { type: "Function", name: t })
      : Ke(e, { type: "Keyword", name: t });
  }
  function Vl(e, t) {
    function r(o, i) {
      return {
        type: "Group",
        terms: o,
        combinator: i,
        disallowEmpty: !1,
        explicit: !1,
      };
    }
    let n;
    for (t = Object.keys(t).sort((o, i) => Gi[o] - Gi[i]); t.length > 0; ) {
      n = t.shift();
      let o = 0,
        i = 0;
      for (; o < e.length; o++) {
        let s = e[o];
        s.type === "Combinator" &&
          (s.value === n
            ? (i === -1 && (i = o - 1), e.splice(o, 1), o--)
            : (i !== -1 &&
                o - i > 1 &&
                (e.splice(i, o - i, r(e.slice(i, o), n)), (o = i + 1)),
              (i = -1)));
      }
      i !== -1 && t.length && e.splice(i, o - i, r(e.slice(i, o), n));
    }
    return n;
  }
  function Zi(e) {
    let t = [],
      r = {},
      n,
      o = null,
      i = e.pos;
    for (; (n = Ql(e)); )
      n.type !== "Spaces" &&
        (n.type === "Combinator"
          ? ((o === null || o.type === "Combinator") &&
              ((e.pos = i), e.error("Unexpected combinator")),
            (r[n.value] = !0))
          : o !== null &&
            o.type !== "Combinator" &&
            ((r[" "] = !0), t.push({ type: "Combinator", value: " " })),
        t.push(n),
        (o = n),
        (i = e.pos));
    return (
      o !== null &&
        o.type === "Combinator" &&
        ((e.pos -= i), e.error("Unexpected combinator")),
      {
        type: "Group",
        terms: t,
        combinator: Vl(t, r) || " ",
        disallowEmpty: !1,
        explicit: !1,
      }
    );
  }
  function Kl(e) {
    let t;
    return (
      e.eat(Kt),
      (t = Zi(e)),
      e.eat(_r),
      (t.explicit = !0),
      e.charCode() === Ki && (e.pos++, (t.disallowEmpty = !0)),
      t
    );
  }
  function Ql(e) {
    let t = e.charCode();
    if (t < 128 && ht[t] === 1) return Gl(e);
    switch (t) {
      case _r:
        break;
      case Kt:
        return Ke(e, Kl(e));
      case Br:
        return e.nextCharCode() === Ht ? Wl(e) : Yl(e);
      case Wi:
        return {
          type: "Combinator",
          value: e.substringToPos(e.pos + (e.nextCharCode() === Wi ? 2 : 1)),
        };
      case Ui:
        return e.pos++, e.eat(Ui), { type: "Combinator", value: "&&" };
      case Fr:
        return e.pos++, { type: "Comma" };
      case Ht:
        return Ke(e, { type: "String", value: Ul(e) });
      case Bl:
      case zl:
      case Ml:
      case Fl:
      case Rl:
        return { type: "Spaces", value: Gt(e) };
      case ql:
        return (
          (t = e.nextCharCode()),
          t < 128 && ht[t] === 1
            ? (e.pos++, { type: "AtKeyword", name: Ve(e) })
            : Nr(e)
        );
      case Xi:
      case Rr:
      case zr:
      case Mr:
      case Ki:
        break;
      case Yt:
        if (((t = e.nextCharCode()), t < 48 || t > 57)) return Nr(e);
        break;
      default:
        return Nr(e);
    }
  }
  function Qe(e) {
    let t = new Wt(e),
      r = Zi(t);
    return (
      t.pos !== e.length && t.error("Unexpected input"),
      r.terms.length === 1 && r.terms[0].type === "Group" ? r.terms[0] : r
    );
  }
  var mt = function () {};
  function Ji(e) {
    return typeof e == "function" ? e : mt;
  }
  function Qt(e, t, r) {
    function n(s) {
      switch ((o.call(r, s), s.type)) {
        case "Group":
          s.terms?.forEach(n);
          break;
        case "Multiplier":
          n(s.term);
          break;
        case "Type":
        case "Property":
        case "Keyword":
        case "AtKeyword":
        case "Function":
        case "String":
        case "Token":
        case "Comma":
          break;
        default:
          throw new Error("Unknown type: " + s.type);
      }
      i.call(r, s);
    }
    let o = mt,
      i = mt;
    if (
      (typeof t == "function"
        ? (o = t)
        : t && ((o = Ji(t.enter)), (i = Ji(t.leave))),
      o === mt && i === mt)
    )
      throw new Error(
        "Neither `enter` nor `leave` walker handler is set or both aren't a function"
      );
    n(e, r);
  }
  var Xl = {
    decorator(e) {
      let t = [],
        r = null;
      return {
        ...e,
        node(n) {
          let o = r;
          (r = n), e.node.call(this, n), (r = o);
        },
        emit(n, o, i) {
          t.push({ type: o, value: n, node: i ? null : r });
        },
        result() {
          return t;
        },
      };
    },
  };
  function $l(e) {
    let t = [];
    return (
      ve(e, (r, n, o) => t.push({ type: r, value: e.slice(n, o), node: null })),
      t
    );
  }
  function ea(e, t) {
    return typeof e == "string" ? $l(e) : t.generate(e, Xl);
  }
  var C = { type: "Match" },
    L = { type: "Mismatch" },
    Xt = { type: "DisallowEmpty" },
    Zl = 40,
    Jl = 41;
  function Z(e, t, r) {
    return (t === C && r === L) || (e === C && t === C && r === C)
      ? e
      : (e.type === "If" &&
          e.else === L &&
          t === C &&
          ((t = e.then), (e = e.match)),
        { type: "If", match: e, then: t, else: r });
  }
  function ra(e) {
    return (
      e.length > 2 &&
      e.charCodeAt(e.length - 2) === Zl &&
      e.charCodeAt(e.length - 1) === Jl
    );
  }
  function ta(e) {
    return (
      e.type === "Keyword" ||
      e.type === "AtKeyword" ||
      e.type === "Function" ||
      (e.type === "Type" && ra(e.name))
    );
  }
  function Ur(e, t, r) {
    switch (e) {
      case " ": {
        let n = C;
        for (let o = t.length - 1; o >= 0; o--) {
          let i = t[o];
          n = Z(i, n, L);
        }
        return n;
      }
      case "|": {
        let n = L,
          o = null;
        for (let i = t.length - 1; i >= 0; i--) {
          let s = t[i];
          if (
            ta(s) &&
            (o === null &&
              i > 0 &&
              ta(t[i - 1]) &&
              ((o = Object.create(null)),
              (n = Z({ type: "Enum", map: o }, C, n))),
            o !== null)
          ) {
            let u = (ra(s.name) ? s.name.slice(0, -1) : s.name).toLowerCase();
            if (!(u in o)) {
              o[u] = s;
              continue;
            }
          }
          (o = null), (n = Z(s, C, n));
        }
        return n;
      }
      case "&&": {
        if (t.length > 5) return { type: "MatchOnce", terms: t, all: !0 };
        let n = L;
        for (let o = t.length - 1; o >= 0; o--) {
          let i = t[o],
            s;
          t.length > 1
            ? (s = Ur(
                e,
                t.filter(function (u) {
                  return u !== i;
                }),
                !1
              ))
            : (s = C),
            (n = Z(i, s, n));
        }
        return n;
      }
      case "||": {
        if (t.length > 5) return { type: "MatchOnce", terms: t, all: !1 };
        let n = r ? C : L;
        for (let o = t.length - 1; o >= 0; o--) {
          let i = t[o],
            s;
          t.length > 1
            ? (s = Ur(
                e,
                t.filter(function (u) {
                  return u !== i;
                }),
                !0
              ))
            : (s = C),
            (n = Z(i, s, n));
        }
        return n;
      }
    }
  }
  function ec(e) {
    let t = C,
      r = jr(e.term);
    if (e.max === 0)
      (r = Z(r, Xt, L)),
        (t = Z(r, null, L)),
        (t.then = Z(C, C, t)),
        e.comma && (t.then.else = Z({ type: "Comma", syntax: e }, t, L));
    else
      for (let n = e.min || 1; n <= e.max; n++)
        e.comma && t !== C && (t = Z({ type: "Comma", syntax: e }, t, L)),
          (t = Z(r, Z(C, C, t), L));
    if (e.min === 0) t = Z(C, C, t);
    else
      for (let n = 0; n < e.min - 1; n++)
        e.comma && t !== C && (t = Z({ type: "Comma", syntax: e }, t, L)),
          (t = Z(r, t, L));
    return t;
  }
  function jr(e) {
    if (typeof e == "function") return { type: "Generic", fn: e };
    switch (e.type) {
      case "Group": {
        let t = Ur(e.combinator, e.terms.map(jr), !1);
        return e.disallowEmpty && (t = Z(t, Xt, L)), t;
      }
      case "Multiplier":
        return ec(e);
      case "Type":
      case "Property":
        return { type: e.type, name: e.name, syntax: e };
      case "Keyword":
        return { type: e.type, name: e.name.toLowerCase(), syntax: e };
      case "AtKeyword":
        return { type: e.type, name: "@" + e.name.toLowerCase(), syntax: e };
      case "Function":
        return { type: e.type, name: e.name.toLowerCase() + "(", syntax: e };
      case "String":
        return e.value.length === 3
          ? { type: "Token", value: e.value.charAt(1), syntax: e }
          : {
              type: e.type,
              value: e.value.substr(1, e.value.length - 2).replace(/\\'/g, "'"),
              syntax: e,
            };
      case "Token":
        return { type: e.type, value: e.value, syntax: e };
      case "Comma":
        return { type: e.type, syntax: e };
      default:
        throw new Error("Unknown node type:", e.type);
    }
  }
  function $t(e, t) {
    return (
      typeof e == "string" && (e = Qe(e)),
      { type: "MatchGraph", match: jr(e), syntax: t || null, source: e }
    );
  }
  var { hasOwnProperty: na } = Object.prototype,
    tc = 0,
    rc = 1,
    Hr = 2,
    la = 3,
    oa = "Match",
    nc = "Mismatch",
    oc =
      "Maximum iteration number exceeded (please fill an issue on https://github.com/csstree/csstree/issues)",
    ia = 15e3,
    ic = 0;
  function ac(e) {
    let t = null,
      r = null,
      n = e;
    for (; n !== null; ) (r = n.prev), (n.prev = t), (t = n), (n = r);
    return t;
  }
  function Wr(e, t) {
    if (e.length !== t.length) return !1;
    for (let r = 0; r < e.length; r++) {
      let n = t.charCodeAt(r),
        o = e.charCodeAt(r);
      if ((o >= 65 && o <= 90 && (o = o | 32), o !== n)) return !1;
    }
    return !0;
  }
  function sc(e) {
    return e.type !== 9 ? !1 : e.value !== "?";
  }
  function aa(e) {
    return e === null
      ? !0
      : e.type === 18 ||
          e.type === 2 ||
          e.type === 21 ||
          e.type === 19 ||
          e.type === 23 ||
          sc(e);
  }
  function sa(e) {
    return e === null
      ? !0
      : e.type === 22 ||
          e.type === 20 ||
          e.type === 24 ||
          (e.type === 9 && e.value === "/");
  }
  function lc(e, t, r) {
    function n() {
      do R++, (S = R < e.length ? e[R] : null);
      while (S !== null && (S.type === 13 || S.type === 25));
    }
    function o(ae) {
      let fe = R + ae;
      return fe < e.length ? e[fe] : null;
    }
    function i(ae, fe) {
      return {
        nextState: ae,
        matchStack: z,
        syntaxStack: p,
        thenStack: m,
        tokenIndex: R,
        prev: fe,
      };
    }
    function s(ae) {
      m = { nextState: ae, matchStack: z, syntaxStack: p, prev: m };
    }
    function u(ae) {
      f = i(ae, f);
    }
    function c() {
      (z = { type: rc, syntax: t.syntax, token: S, prev: z }),
        n(),
        (P = null),
        R > ke && (ke = R);
    }
    function a() {
      (p = {
        syntax: t.syntax,
        opts: t.syntax.opts || (p !== null && p.opts) || null,
        prev: p,
      }),
        (z = { type: Hr, syntax: t.syntax, token: z.token, prev: z });
    }
    function l() {
      z.type === Hr
        ? (z = z.prev)
        : (z = { type: la, syntax: p.syntax, token: z.token, prev: z }),
        (p = p.prev);
    }
    let p = null,
      m = null,
      f = null,
      P = null,
      te = 0,
      X = null,
      S = null,
      R = -1,
      ke = 0,
      z = { type: tc, syntax: null, token: null, prev: null };
    for (n(); X === null && ++te < ia; )
      switch (t.type) {
        case "Match":
          if (m === null) {
            if (
              S !== null &&
              (R !== e.length - 1 || (S.value !== "\\0" && S.value !== "\\9"))
            ) {
              t = L;
              break;
            }
            X = oa;
            break;
          }
          if (((t = m.nextState), t === Xt))
            if (m.matchStack === z) {
              t = L;
              break;
            } else t = C;
          for (; m.syntaxStack !== p; ) l();
          m = m.prev;
          break;
        case "Mismatch":
          if (P !== null && P !== !1)
            (f === null || R > f.tokenIndex) && ((f = P), (P = !1));
          else if (f === null) {
            X = nc;
            break;
          }
          (t = f.nextState),
            (m = f.thenStack),
            (p = f.syntaxStack),
            (z = f.matchStack),
            (R = f.tokenIndex),
            (S = R < e.length ? e[R] : null),
            (f = f.prev);
          break;
        case "MatchGraph":
          t = t.match;
          break;
        case "If":
          t.else !== L && u(t.else), t.then !== C && s(t.then), (t = t.match);
          break;
        case "MatchOnce":
          t = { type: "MatchOnceBuffer", syntax: t, index: 0, mask: 0 };
          break;
        case "MatchOnceBuffer": {
          let Q = t.syntax.terms;
          if (t.index === Q.length) {
            if (t.mask === 0 || t.syntax.all) {
              t = L;
              break;
            }
            t = C;
            break;
          }
          if (t.mask === (1 << Q.length) - 1) {
            t = C;
            break;
          }
          for (; t.index < Q.length; t.index++) {
            let J = 1 << t.index;
            if ((t.mask & J) === 0) {
              u(t),
                s({ type: "AddMatchOnce", syntax: t.syntax, mask: t.mask | J }),
                (t = Q[t.index++]);
              break;
            }
          }
          break;
        }
        case "AddMatchOnce":
          t = {
            type: "MatchOnceBuffer",
            syntax: t.syntax,
            index: 0,
            mask: t.mask,
          };
          break;
        case "Enum":
          if (S !== null) {
            let Q = S.value.toLowerCase();
            if (
              (Q.indexOf("\\") !== -1 && (Q = Q.replace(/\\[09].*$/, "")),
              na.call(t.map, Q))
            ) {
              t = t.map[Q];
              break;
            }
          }
          t = L;
          break;
        case "Generic": {
          let Q = p !== null ? p.opts : null,
            J = R + Math.floor(t.fn(S, o, Q));
          if (!isNaN(J) && J > R) {
            for (; R < J; ) c();
            t = C;
          } else t = L;
          break;
        }
        case "Type":
        case "Property": {
          let Q = t.type === "Type" ? "types" : "properties",
            J = na.call(r, Q) ? r[Q][t.name] : null;
          if (!J || !J.match)
            throw new Error(
              "Bad syntax reference: " +
                (t.type === "Type" ? "<" + t.name + ">" : "<'" + t.name + "'>")
            );
          if (
            P !== !1 &&
            S !== null &&
            t.type === "Type" &&
            ((t.name === "custom-ident" && S.type === 1) ||
              (t.name === "length" && S.value === "0"))
          ) {
            P === null && (P = i(t, f)), (t = L);
            break;
          }
          a(), (t = J.match);
          break;
        }
        case "Keyword": {
          let Q = t.name;
          if (S !== null) {
            let J = S.value;
            if (
              (J.indexOf("\\") !== -1 && (J = J.replace(/\\[09].*$/, "")),
              Wr(J, Q))
            ) {
              c(), (t = C);
              break;
            }
          }
          t = L;
          break;
        }
        case "AtKeyword":
        case "Function":
          if (S !== null && Wr(S.value, t.name)) {
            c(), (t = C);
            break;
          }
          t = L;
          break;
        case "Token":
          if (S !== null && S.value === t.value) {
            c(), (t = C);
            break;
          }
          t = L;
          break;
        case "Comma":
          S !== null && S.type === 18
            ? aa(z.token)
              ? (t = L)
              : (c(), (t = sa(S) ? L : C))
            : (t = aa(z.token) || sa(S) ? C : L);
          break;
        case "String":
          let ae = "",
            fe = R;
          for (; fe < e.length && ae.length < t.value.length; fe++)
            ae += e[fe].value;
          if (Wr(ae, t.value)) {
            for (; R < fe; ) c();
            t = C;
          } else t = L;
          break;
        default:
          throw new Error("Unknown node type: " + t.type);
      }
    switch (((ic += te), X)) {
      case null:
        console.warn("[csstree-match] BREAK after " + ia + " iterations"),
          (X = oc),
          (z = null);
        break;
      case oa:
        for (; p !== null; ) l();
        break;
      default:
        z = null;
    }
    return { tokens: e, reason: X, iterations: te, match: z, longestMatch: ke };
  }
  function Yr(e, t, r) {
    let n = lc(e, t, r || {});
    if (n.match === null) return n;
    let o = n.match,
      i = (n.match = { syntax: t.syntax || null, match: [] }),
      s = [i];
    for (o = ac(o).prev; o !== null; ) {
      switch (o.type) {
        case Hr:
          i.match.push((i = { syntax: o.syntax, match: [] })), s.push(i);
          break;
        case la:
          s.pop(), (i = s[s.length - 1]);
          break;
        default:
          i.match.push({
            syntax: o.syntax || null,
            token: o.token.value,
            node: o.token.node,
          });
      }
      o = o.prev;
    }
    return n;
  }
  var Vr = {};
  b(Vr, {
    getTrace: () => ca,
    isKeyword: () => pc,
    isProperty: () => uc,
    isType: () => cc,
  });
  function ca(e) {
    function t(o) {
      return o === null
        ? !1
        : o.type === "Type" || o.type === "Property" || o.type === "Keyword";
    }
    function r(o) {
      if (Array.isArray(o.match)) {
        for (let i = 0; i < o.match.length; i++)
          if (r(o.match[i])) return t(o.syntax) && n.unshift(o.syntax), !0;
      } else if (o.node === e) return (n = t(o.syntax) ? [o.syntax] : []), !0;
      return !1;
    }
    let n = null;
    return this.matched !== null && r(this.matched), n;
  }
  function cc(e, t) {
    return Gr(this, e, (r) => r.type === "Type" && r.name === t);
  }
  function uc(e, t) {
    return Gr(this, e, (r) => r.type === "Property" && r.name === t);
  }
  function pc(e) {
    return Gr(this, e, (t) => t.type === "Keyword");
  }
  function Gr(e, t, r) {
    let n = ca.call(e, t);
    return n === null ? !1 : n.some(r);
  }
  function ua(e) {
    return "node" in e ? e.node : ua(e.match[0]);
  }
  function pa(e) {
    return "node" in e ? e.node : pa(e.match[e.match.length - 1]);
  }
  function Kr(e, t, r, n, o) {
    function i(u) {
      if (u.syntax !== null && u.syntax.type === n && u.syntax.name === o) {
        let c = ua(u),
          a = pa(u);
        e.syntax.walk(t, function (l, p, m) {
          if (l === c) {
            let f = new D();
            do {
              if ((f.appendData(p.data), p.data === a)) break;
              p = p.next;
            } while (p !== null);
            s.push({ parent: m, nodes: f });
          }
        });
      }
      Array.isArray(u.match) && u.match?.forEach(i);
    }
    let s = [];
    return r.matched !== null && i(r.matched), s;
  }
  var { hasOwnProperty: ft } = Object.prototype;
  function Qr(e) {
    return typeof e == "number" && isFinite(e) && Math.floor(e) === e && e >= 0;
  }
  function ha(e) {
    return Boolean(e) && Qr(e.offset) && Qr(e.line) && Qr(e.column);
  }
  function hc(e, t) {
    return function (n, o) {
      if (!n || n.constructor !== Object)
        return o(n, "Type of node should be an Object");
      for (let i in n) {
        let s = !0;
        if (ft.call(n, i) !== !1) {
          if (i === "type")
            n.type !== e &&
              o(n, "Wrong node type `" + n.type + "`, expected `" + e + "`");
          else if (i === "loc") {
            if (n.loc === null) continue;
            if (n.loc && n.loc.constructor === Object)
              if (typeof n.loc.source != "string") i += ".source";
              else if (!ha(n.loc.start)) i += ".start";
              else if (!ha(n.loc.end)) i += ".end";
              else continue;
            s = !1;
          } else if (t.hasOwnProperty(i)) {
            s = !1;
            for (let u = 0; !s && u < t[i].length; u++) {
              let c = t[i][u];
              switch (c) {
                case String:
                  s = typeof n[i] == "string";
                  break;
                case Boolean:
                  s = typeof n[i] == "boolean";
                  break;
                case null:
                  s = n[i] === null;
                  break;
                default:
                  typeof c == "string"
                    ? (s = n[i] && n[i].type === c)
                    : Array.isArray(c) && (s = n[i] instanceof D);
              }
            }
          } else o(n, "Unknown field `" + i + "` for " + e + " node type");
          s || o(n, "Bad value for `" + e + "." + i + "`");
        }
      }
      for (let i in t)
        ft.call(t, i) &&
          ft.call(n, i) === !1 &&
          o(n, "Field `" + e + "." + i + "` is missed");
    };
  }
  function mc(e, t) {
    let r = t.structure,
      n = { type: String, loc: !0 },
      o = { type: '"' + e + '"' };
    for (let i in r) {
      if (ft.call(r, i) === !1) continue;
      let s = [],
        u = (n[i] = Array.isArray(r[i]) ? r[i].slice() : [r[i]]);
      for (let c = 0; c < u.length; c++) {
        let a = u[c];
        if (a === String || a === Boolean) s.push(a.name);
        else if (a === null) s.push("null");
        else if (typeof a == "string") s.push("<" + a + ">");
        else if (Array.isArray(a)) s.push("List");
        else
          throw new Error(
            "Wrong value `" +
              a +
              "` in `" +
              e +
              "." +
              i +
              "` structure definition"
          );
      }
      o[i] = s.join(" | ");
    }
    return { docs: o, check: hc(e, n) };
  }
  function ma(e) {
    let t = {};
    if (e.node) {
      for (let r in e.node)
        if (ft.call(e.node, r)) {
          let n = e.node[r];
          if (n.structure) t[r] = mc(r, n);
          else
            throw new Error(
              "Missed `structure` field in `" + r + "` node type definition"
            );
        }
    }
    return t;
  }
  var fc = $t(Bt.join(" | "));
  function Xr(e, t, r) {
    let n = {};
    for (let o in e)
      e[o].syntax && (n[o] = r ? e[o].syntax : Ie(e[o].syntax, { compact: t }));
    return n;
  }
  function dc(e, t, r) {
    let n = {};
    for (let [o, i] of Object.entries(e))
      n[o] = {
        prelude:
          i.prelude &&
          (r ? i.prelude.syntax : Ie(i.prelude.syntax, { compact: t })),
        descriptors: i.descriptors && Xr(i.descriptors, t, r),
      };
    return n;
  }
  function gc(e) {
    for (let t = 0; t < e.length; t++)
      if (e[t].value.toLowerCase() === "var(") return !0;
    return !1;
  }
  function ce(e, t, r) {
    return { matched: e, iterations: r, error: t, ...Vr };
  }
  function Xe(e, t, r, n) {
    let o = ea(r, e.syntax),
      i;
    return gc(o)
      ? ce(null, new Error("Matching for a tree with var() is not supported"))
      : (n && (i = Yr(o, e.cssWideKeywordsSyntax, e)),
        (!n || !i.match) && ((i = Yr(o, t.match, e)), !i.match)
          ? ce(null, new Ni(i.reason, t.syntax, r, i), i.iterations)
          : ce(i.match, null, i.iterations));
  }
  var De = class {
    constructor(t, r, n) {
      if (
        ((this.cssWideKeywordsSyntax = fc),
        (this.syntax = r),
        (this.generic = !1),
        (this.units = { ...Ut }),
        (this.atrules = Object.create(null)),
        (this.properties = Object.create(null)),
        (this.types = Object.create(null)),
        (this.structure = n || ma(t)),
        t)
      ) {
        if (t.units)
          for (let o of Object.keys(Ut))
            Array.isArray(t.units[o]) && (this.units[o] = t.units[o]);
        if (t.types) for (let o in t.types) this.addType_(o, t.types[o]);
        if (t.generic) {
          this.generic = !0;
          for (let [o, i] of Object.entries(qi(this.units)))
            this.addType_(o, i);
        }
        if (t.atrules)
          for (let o in t.atrules) this.addAtrule_(o, t.atrules[o]);
        if (t.properties)
          for (let o in t.properties) this.addProperty_(o, t.properties[o]);
      }
    }
    checkStructure(t) {
      function r(i, s) {
        o.push({ node: i, message: s });
      }
      let n = this.structure,
        o = [];
      return (
        this.syntax.walk(t, function (i) {
          n.hasOwnProperty(i.type)
            ? n[i.type].check(i, r)
            : r(i, "Unknown node type `" + i.type + "`");
        }),
        o.length ? o : !1
      );
    }
    createDescriptor(t, r, n, o = null) {
      let i = { type: r, name: n },
        s = {
          type: r,
          name: n,
          parent: o,
          serializable:
            typeof t == "string" || (t && typeof t.type == "string"),
          syntax: null,
          match: null,
        };
      return (
        typeof t == "function"
          ? (s.match = $t(t, i))
          : (typeof t == "string"
              ? Object.defineProperty(s, "syntax", {
                  get() {
                    return (
                      Object.defineProperty(s, "syntax", { value: Qe(t) }),
                      s.syntax
                    );
                  },
                })
              : (s.syntax = t),
            Object.defineProperty(s, "match", {
              get() {
                return (
                  Object.defineProperty(s, "match", { value: $t(s.syntax, i) }),
                  s.match
                );
              },
            })),
        s
      );
    }
    addAtrule_(t, r) {
      !r ||
        (this.atrules[t] = {
          type: "Atrule",
          name: t,
          prelude: r.prelude
            ? this.createDescriptor(r.prelude, "AtrulePrelude", t)
            : null,
          descriptors: r.descriptors
            ? Object.keys(r.descriptors).reduce(
                (n, o) => (
                  (n[o] = this.createDescriptor(
                    r.descriptors[o],
                    "AtruleDescriptor",
                    o,
                    t
                  )),
                  n
                ),
                Object.create(null)
              )
            : null,
        });
    }
    addProperty_(t, r) {
      !r || (this.properties[t] = this.createDescriptor(r, "Property", t));
    }
    addType_(t, r) {
      !r || (this.types[t] = this.createDescriptor(r, "Type", t));
    }
    checkAtruleName(t) {
      if (!this.getAtrule(t)) return new He("Unknown at-rule", "@" + t);
    }
    checkAtrulePrelude(t, r) {
      let n = this.checkAtruleName(t);
      if (n) return n;
      let o = this.getAtrule(t);
      if (!o.prelude && r)
        return new SyntaxError(
          "At-rule `@" + t + "` should not contain a prelude"
        );
      if (o.prelude && !r && !Xe(this, o.prelude, "", !1).matched)
        return new SyntaxError("At-rule `@" + t + "` should contain a prelude");
    }
    checkAtruleDescriptorName(t, r) {
      let n = this.checkAtruleName(t);
      if (n) return n;
      let o = this.getAtrule(t),
        i = st(r);
      if (!o.descriptors)
        return new SyntaxError("At-rule `@" + t + "` has no known descriptors");
      if (!o.descriptors[i.name] && !o.descriptors[i.basename])
        return new He("Unknown at-rule descriptor", r);
    }
    checkPropertyName(t) {
      if (!this.getProperty(t)) return new He("Unknown property", t);
    }
    matchAtrulePrelude(t, r) {
      let n = this.checkAtrulePrelude(t, r);
      if (n) return ce(null, n);
      let o = this.getAtrule(t);
      return o.prelude ? Xe(this, o.prelude, r || "", !1) : ce(null, null);
    }
    matchAtruleDescriptor(t, r, n) {
      let o = this.checkAtruleDescriptorName(t, r);
      if (o) return ce(null, o);
      let i = this.getAtrule(t),
        s = st(r);
      return Xe(
        this,
        i.descriptors[s.name] || i.descriptors[s.basename],
        n,
        !1
      );
    }
    matchDeclaration(t) {
      return t.type !== "Declaration"
        ? ce(null, new Error("Not a Declaration node"))
        : this.matchProperty(t.property, t.value);
    }
    matchProperty(t, r) {
      if (Ft(t).custom)
        return ce(
          null,
          new Error("Lexer matching doesn't applicable for custom properties")
        );
      let n = this.checkPropertyName(t);
      return n ? ce(null, n) : Xe(this, this.getProperty(t), r, !0);
    }
    matchType(t, r) {
      let n = this.getType(t);
      return n ? Xe(this, n, r, !1) : ce(null, new He("Unknown type", t));
    }
    match(t, r) {
      return typeof t != "string" && (!t || !t.type)
        ? ce(null, new He("Bad syntax"))
        : ((typeof t == "string" || !t.match) &&
            (t = this.createDescriptor(t, "Type", "anonymous")),
          Xe(this, t, r, !1));
    }
    findValueFragments(t, r, n, o) {
      return Kr(this, r, this.matchProperty(t, r), n, o);
    }
    findDeclarationValueFragments(t, r, n) {
      return Kr(this, t.value, this.matchDeclaration(t), r, n);
    }
    findAllFragments(t, r, n) {
      let o = [];
      return (
        this.syntax.walk(t, {
          visit: "Declaration",
          enter: (i) => {
            o.push.apply(o, this.findDeclarationValueFragments(i, r, n));
          },
        }),
        o
      );
    }
    getAtrule(t, r = !0) {
      let n = st(t);
      return (
        (n.vendor && r
          ? this.atrules[n.name] || this.atrules[n.basename]
          : this.atrules[n.name]) || null
      );
    }
    getAtrulePrelude(t, r = !0) {
      let n = this.getAtrule(t, r);
      return (n && n.prelude) || null;
    }
    getAtruleDescriptor(t, r) {
      return (
        (this.atrules.hasOwnProperty(t) &&
          this.atrules.declarators &&
          this.atrules[t].declarators[r]) ||
        null
      );
    }
    getProperty(t, r = !0) {
      let n = Ft(t);
      return (
        (n.vendor && r
          ? this.properties[n.name] || this.properties[n.basename]
          : this.properties[n.name]) || null
      );
    }
    getType(t) {
      return hasOwnProperty.call(this.types, t) ? this.types[t] : null;
    }
    validate() {
      function t(o, i, s, u) {
        if (s.has(i)) return s.get(i);
        s.set(i, !1),
          u.syntax !== null &&
            Qt(
              u.syntax,
              function (c) {
                if (c.type !== "Type" && c.type !== "Property") return;
                let a = c.type === "Type" ? o.types : o.properties,
                  l = c.type === "Type" ? r : n;
                (!hasOwnProperty.call(a, c.name) ||
                  t(o, c.name, l, a[c.name])) &&
                  s.set(i, !0);
              },
              this
            );
      }
      let r = new Map(),
        n = new Map();
      for (let o in this.types) t(this, o, r, this.types[o]);
      for (let o in this.properties) t(this, o, n, this.properties[o]);
      return (
        (r = [...r.keys()].filter((o) => r.get(o))),
        (n = [...n.keys()].filter((o) => n.get(o))),
        r.length || n.length ? { types: r, properties: n } : null
      );
    }
    dump(t, r) {
      return {
        generic: this.generic,
        units: this.units,
        types: Xr(this.types, !r, t),
        properties: Xr(this.properties, !r, t),
        atrules: dc(this.atrules, !r, t),
      };
    }
    toString() {
      return JSON.stringify(this.dump());
    }
  };
  function $r(e, t) {
    return typeof t == "string" && /^\s*\|/.test(t)
      ? typeof e == "string"
        ? e + t
        : t.replace(/^\s*\|\s*/, "")
      : t || null;
  }
  function fa(e, t) {
    let r = Object.create(null);
    for (let [n, o] of Object.entries(e))
      if (o) {
        r[n] = {};
        for (let i of Object.keys(o)) t.includes(i) && (r[n][i] = o[i]);
      }
    return r;
  }
  function dt(e, t) {
    let r = { ...e };
    for (let [n, o] of Object.entries(t))
      switch (n) {
        case "generic":
          r[n] = Boolean(o);
          break;
        case "units":
          r[n] = { ...e[n] };
          for (let [i, s] of Object.entries(o))
            r[n][i] = Array.isArray(s) ? s : [];
          break;
        case "atrules":
          r[n] = { ...e[n] };
          for (let [i, s] of Object.entries(o)) {
            let u = r[n][i] || {},
              c = (r[n][i] = {
                prelude: u.prelude || null,
                descriptors: { ...u.descriptors },
              });
            if (!!s) {
              c.prelude = s.prelude
                ? $r(c.prelude, s.prelude)
                : c.prelude || null;
              for (let [a, l] of Object.entries(s.descriptors || {}))
                c.descriptors[a] = l ? $r(c.descriptors[a], l) : null;
              Object.keys(c.descriptors).length || (c.descriptors = null);
            }
          }
          break;
        case "types":
        case "properties":
          r[n] = { ...e[n] };
          for (let [i, s] of Object.entries(o)) r[n][i] = $r(r[n][i], s);
          break;
        case "scope":
          r[n] = { ...e[n] };
          for (let [i, s] of Object.entries(o)) r[n][i] = { ...r[n][i], ...s };
          break;
        case "parseContext":
          r[n] = { ...e[n], ...o };
          break;
        case "atrule":
        case "pseudo":
          r[n] = { ...e[n], ...fa(o, ["parse"]) };
          break;
        case "node":
          r[n] = {
            ...e[n],
            ...fa(o, ["name", "structure", "parse", "generate", "walkContext"]),
          };
          break;
      }
    return r;
  }
  function da(e) {
    let t = ei(e),
      r = Di(e),
      n = Ai(e),
      { fromPlainObject: o, toPlainObject: i } = Ti(r),
      s = {
        lexer: null,
        createLexer: (u) => new De(u, s, s.lexer.structure),
        tokenize: ve,
        parse: t,
        generate: n,
        walk: r,
        find: r.find,
        findLast: r.findLast,
        findAll: r.findAll,
        fromPlainObject: o,
        toPlainObject: i,
        fork(u) {
          let c = dt({}, e);
          return da(typeof u == "function" ? u(c, Object.assign) : dt(c, u));
        },
      };
    return (
      (s.lexer = new De(
        {
          generic: !0,
          units: e.units,
          types: e.types,
          atrules: e.atrules,
          properties: e.properties,
          node: e.node,
        },
        s
      )),
      s
    );
  }
  var Zt = (e) => da(dt({}, e));
  var ga = {
    generic: !0,
    units: {
      angle: ["deg", "grad", "rad", "turn"],
      decibel: ["db"],
      flex: ["fr"],
      frequency: ["hz", "khz"],
      length: [
        "cm",
        "mm",
        "q",
        "in",
        "pt",
        "pc",
        "px",
        "em",
        "rem",
        "ex",
        "rex",
        "cap",
        "rcap",
        "ch",
        "rch",
        "ic",
        "ric",
        "lh",
        "rlh",
        "vw",
        "svw",
        "lvw",
        "dvw",
        "vh",
        "svh",
        "lvh",
        "dvh",
        "vi",
        "svi",
        "lvi",
        "dvi",
        "vb",
        "svb",
        "lvb",
        "dvb",
        "vmin",
        "svmin",
        "lvmin",
        "dvmin",
        "vmax",
        "svmax",
        "lvmax",
        "dvmax",
        "cqw",
        "cqh",
        "cqi",
        "cqb",
        "cqmin",
        "cqmax",
      ],
      resolution: ["dpi", "dpcm", "dppx", "x"],
      semitones: ["st"],
      time: ["s", "ms"],
    },
    types: {
      "abs()": "abs( <calc-sum> )",
      "absolute-size":
        "xx-small|x-small|small|medium|large|x-large|xx-large|xxx-large",
      "acos()": "acos( <calc-sum> )",
      "alpha-value": "<number>|<percentage>",
      "angle-percentage": "<angle>|<percentage>",
      "angular-color-hint": "<angle-percentage>",
      "angular-color-stop": "<color>&&<color-stop-angle>?",
      "angular-color-stop-list":
        "[<angular-color-stop> [, <angular-color-hint>]?]# , <angular-color-stop>",
      "animateable-feature": "scroll-position|contents|<custom-ident>",
      "asin()": "asin( <calc-sum> )",
      "atan()": "atan( <calc-sum> )",
      "atan2()": "atan2( <calc-sum> , <calc-sum> )",
      attachment: "scroll|fixed|local",
      "attr()": "attr( <attr-name> <type-or-unit>? [, <attr-fallback>]? )",
      "attr-matcher": "['~'|'|'|'^'|'$'|'*']? '='",
      "attr-modifier": "i|s",
      "attribute-selector":
        "'[' <wq-name> ']'|'[' <wq-name> <attr-matcher> [<string-token>|<ident-token>] <attr-modifier>? ']'",
      "auto-repeat":
        "repeat( [auto-fill|auto-fit] , [<line-names>? <fixed-size>]+ <line-names>? )",
      "auto-track-list":
        "[<line-names>? [<fixed-size>|<fixed-repeat>]]* <line-names>? <auto-repeat> [<line-names>? [<fixed-size>|<fixed-repeat>]]* <line-names>?",
      axis: "block|inline|vertical|horizontal",
      "baseline-position": "[first|last]? baseline",
      "basic-shape": "<inset()>|<circle()>|<ellipse()>|<polygon()>|<path()>",
      "bg-image": "none|<image>",
      "bg-layer":
        "<bg-image>||<bg-position> [/ <bg-size>]?||<repeat-style>||<attachment>||<box>||<box>",
      "bg-position":
        "[[left|center|right|top|bottom|<length-percentage>]|[left|center|right|<length-percentage>] [top|center|bottom|<length-percentage>]|[center|[left|right] <length-percentage>?]&&[center|[top|bottom] <length-percentage>?]]",
      "bg-size": "[<length-percentage>|auto]{1,2}|cover|contain",
      "blur()": "blur( <length> )",
      "blend-mode":
        "normal|multiply|screen|overlay|darken|lighten|color-dodge|color-burn|hard-light|soft-light|difference|exclusion|hue|saturation|color|luminosity",
      box: "border-box|padding-box|content-box",
      "brightness()": "brightness( <number-percentage> )",
      "calc()": "calc( <calc-sum> )",
      "calc-sum": "<calc-product> [['+'|'-'] <calc-product>]*",
      "calc-product": "<calc-value> ['*' <calc-value>|'/' <number>]*",
      "calc-value":
        "<number>|<dimension>|<percentage>|<calc-constant>|( <calc-sum> )",
      "calc-constant": "e|pi|infinity|-infinity|NaN",
      "cf-final-image": "<image>|<color>",
      "cf-mixing-image": "<percentage>?&&<image>",
      "circle()": "circle( [<shape-radius>]? [at <position>]? )",
      "clamp()": "clamp( <calc-sum>#{3} )",
      "class-selector": "'.' <ident-token>",
      "clip-source": "<url>",
      color:
        "<rgb()>|<rgba()>|<hsl()>|<hsla()>|<hwb()>|<lab()>|<lch()>|<hex-color>|<named-color>|currentcolor|<deprecated-system-color>",
      "color-stop": "<color-stop-length>|<color-stop-angle>",
      "color-stop-angle": "<angle-percentage>{1,2}",
      "color-stop-length": "<length-percentage>{1,2}",
      "color-stop-list":
        "[<linear-color-stop> [, <linear-color-hint>]?]# , <linear-color-stop>",
      combinator: "'>'|'+'|'~'|['||']",
      "common-lig-values": "[common-ligatures|no-common-ligatures]",
      "compat-auto":
        "searchfield|textarea|push-button|slider-horizontal|checkbox|radio|square-button|menulist|listbox|meter|progress-bar|button",
      "composite-style":
        "clear|copy|source-over|source-in|source-out|source-atop|destination-over|destination-in|destination-out|destination-atop|xor",
      "compositing-operator": "add|subtract|intersect|exclude",
      "compound-selector":
        "[<type-selector>? <subclass-selector>* [<pseudo-element-selector> <pseudo-class-selector>*]*]!",
      "compound-selector-list": "<compound-selector>#",
      "complex-selector":
        "<compound-selector> [<combinator>? <compound-selector>]*",
      "complex-selector-list": "<complex-selector>#",
      "conic-gradient()":
        "conic-gradient( [from <angle>]? [at <position>]? , <angular-color-stop-list> )",
      "contextual-alt-values": "[contextual|no-contextual]",
      "content-distribution": "space-between|space-around|space-evenly|stretch",
      "content-list":
        "[<string>|contents|<image>|<counter>|<quote>|<target>|<leader()>|<attr()>]+",
      "content-position": "center|start|end|flex-start|flex-end",
      "content-replacement": "<image>",
      "contrast()": "contrast( [<number-percentage>] )",
      "cos()": "cos( <calc-sum> )",
      counter: "<counter()>|<counters()>",
      "counter()": "counter( <counter-name> , <counter-style>? )",
      "counter-name": "<custom-ident>",
      "counter-style": "<counter-style-name>|symbols( )",
      "counter-style-name": "<custom-ident>",
      "counters()": "counters( <counter-name> , <string> , <counter-style>? )",
      "cross-fade()": "cross-fade( <cf-mixing-image> , <cf-final-image>? )",
      "cubic-bezier-timing-function":
        "ease|ease-in|ease-out|ease-in-out|cubic-bezier( <number [0,1]> , <number> , <number [0,1]> , <number> )",
      "deprecated-system-color":
        "ActiveBorder|ActiveCaption|AppWorkspace|Background|ButtonFace|ButtonHighlight|ButtonShadow|ButtonText|CaptionText|GrayText|Highlight|HighlightText|InactiveBorder|InactiveCaption|InactiveCaptionText|InfoBackground|InfoText|Menu|MenuText|Scrollbar|ThreeDDarkShadow|ThreeDFace|ThreeDHighlight|ThreeDLightShadow|ThreeDShadow|Window|WindowFrame|WindowText",
      "discretionary-lig-values":
        "[discretionary-ligatures|no-discretionary-ligatures]",
      "display-box": "contents|none",
      "display-inside": "flow|flow-root|table|flex|grid|ruby",
      "display-internal":
        "table-row-group|table-header-group|table-footer-group|table-row|table-cell|table-column-group|table-column|table-caption|ruby-base|ruby-text|ruby-base-container|ruby-text-container",
      "display-legacy":
        "inline-block|inline-list-item|inline-table|inline-flex|inline-grid",
      "display-listitem": "<display-outside>?&&[flow|flow-root]?&&list-item",
      "display-outside": "block|inline|run-in",
      "drop-shadow()": "drop-shadow( <length>{2,3} <color>? )",
      "east-asian-variant-values":
        "[jis78|jis83|jis90|jis04|simplified|traditional]",
      "east-asian-width-values": "[full-width|proportional-width]",
      "element()":
        "element( <custom-ident> , [first|start|last|first-except]? )|element( <id-selector> )",
      "ellipse()": "ellipse( [<shape-radius>{2}]? [at <position>]? )",
      "ending-shape": "circle|ellipse",
      "env()": "env( <custom-ident> , <declaration-value>? )",
      "exp()": "exp( <calc-sum> )",
      "explicit-track-list": "[<line-names>? <track-size>]+ <line-names>?",
      "family-name": "<string>|<custom-ident>+",
      "feature-tag-value": "<string> [<integer>|on|off]?",
      "feature-type":
        "@stylistic|@historical-forms|@styleset|@character-variant|@swash|@ornaments|@annotation",
      "feature-value-block":
        "<feature-type> '{' <feature-value-declaration-list> '}'",
      "feature-value-block-list": "<feature-value-block>+",
      "feature-value-declaration": "<custom-ident> : <integer>+ ;",
      "feature-value-declaration-list": "<feature-value-declaration>",
      "feature-value-name": "<custom-ident>",
      "fill-rule": "nonzero|evenodd",
      "filter-function":
        "<blur()>|<brightness()>|<contrast()>|<drop-shadow()>|<grayscale()>|<hue-rotate()>|<invert()>|<opacity()>|<saturate()>|<sepia()>",
      "filter-function-list": "[<filter-function>|<url>]+",
      "final-bg-layer":
        "<'background-color'>||<bg-image>||<bg-position> [/ <bg-size>]?||<repeat-style>||<attachment>||<box>||<box>",
      "fixed-breadth": "<length-percentage>",
      "fixed-repeat":
        "repeat( [<integer [1,\u221E]>] , [<line-names>? <fixed-size>]+ <line-names>? )",
      "fixed-size":
        "<fixed-breadth>|minmax( <fixed-breadth> , <track-breadth> )|minmax( <inflexible-breadth> , <fixed-breadth> )",
      "font-stretch-absolute":
        "normal|ultra-condensed|extra-condensed|condensed|semi-condensed|semi-expanded|expanded|extra-expanded|ultra-expanded|<percentage>",
      "font-variant-css21": "[normal|small-caps]",
      "font-weight-absolute": "normal|bold|<number [1,1000]>",
      "frequency-percentage": "<frequency>|<percentage>",
      "general-enclosed":
        "[<function-token> <any-value> )]|( <ident> <any-value> )",
      "generic-family":
        "serif|sans-serif|cursive|fantasy|monospace|-apple-system",
      "generic-name": "serif|sans-serif|cursive|fantasy|monospace",
      "geometry-box": "<shape-box>|fill-box|stroke-box|view-box",
      gradient:
        "<linear-gradient()>|<repeating-linear-gradient()>|<radial-gradient()>|<repeating-radial-gradient()>|<conic-gradient()>|<repeating-conic-gradient()>|<-legacy-gradient>",
      "grayscale()": "grayscale( <number-percentage> )",
      "grid-line":
        "auto|<custom-ident>|[<integer>&&<custom-ident>?]|[span&&[<integer>||<custom-ident>]]",
      "historical-lig-values": "[historical-ligatures|no-historical-ligatures]",
      "hsl()":
        "hsl( <hue> <percentage> <percentage> [/ <alpha-value>]? )|hsl( <hue> , <percentage> , <percentage> , <alpha-value>? )",
      "hsla()":
        "hsla( <hue> <percentage> <percentage> [/ <alpha-value>]? )|hsla( <hue> , <percentage> , <percentage> , <alpha-value>? )",
      hue: "<number>|<angle>",
      "hue-rotate()": "hue-rotate( <angle> )",
      "hwb()":
        "hwb( [<hue>|none] [<percentage>|none] [<percentage>|none] [/ [<alpha-value>|none]]? )",
      "hypot()": "hypot( <calc-sum># )",
      image:
        "<url>|<image()>|<image-set()>|<element()>|<paint()>|<cross-fade()>|<gradient>",
      "image()": "image( <image-tags>? [<image-src>? , <color>?]! )",
      "image-set()": "image-set( <image-set-option># )",
      "image-set-option": "[<image>|<string>] [<resolution>||type( <string> )]",
      "image-src": "<url>|<string>",
      "image-tags": "ltr|rtl",
      "inflexible-breadth": "<length-percentage>|min-content|max-content|auto",
      "inset()": "inset( <length-percentage>{1,4} [round <'border-radius'>]? )",
      "invert()": "invert( <number-percentage> )",
      "keyframes-name": "<custom-ident>|<string>",
      "keyframe-block": "<keyframe-selector># { <declaration-list> }",
      "keyframe-block-list": "<keyframe-block>+",
      "keyframe-selector": "from|to|<percentage>",
      "lab()":
        "lab( [<percentage>|<number>|none] [<percentage>|<number>|none] [<percentage>|<number>|none] [/ [<alpha-value>|none]]? )",
      "layer()": "layer( <layer-name> )",
      "layer-name": "<ident> ['.' <ident>]*",
      "lch()":
        "lch( [<percentage>|<number>|none] [<percentage>|<number>|none] [<hue>|none] [/ [<alpha-value>|none]]? )",
      "leader()": "leader( <leader-type> )",
      "leader-type": "dotted|solid|space|<string>",
      "length-percentage": "<length>|<percentage>",
      "line-names": "'[' <custom-ident>* ']'",
      "line-name-list": "[<line-names>|<name-repeat>]+",
      "line-style":
        "none|hidden|dotted|dashed|solid|double|groove|ridge|inset|outset",
      "line-width": "<length>|thin|medium|thick",
      "linear-color-hint": "<length-percentage>",
      "linear-color-stop": "<color> <color-stop-length>?",
      "linear-gradient()":
        "linear-gradient( [<angle>|to <side-or-corner>]? , <color-stop-list> )",
      "log()": "log( <calc-sum> , <calc-sum>? )",
      "mask-layer":
        "<mask-reference>||<position> [/ <bg-size>]?||<repeat-style>||<geometry-box>||[<geometry-box>|no-clip]||<compositing-operator>||<masking-mode>",
      "mask-position":
        "[<length-percentage>|left|center|right] [<length-percentage>|top|center|bottom]?",
      "mask-reference": "none|<image>|<mask-source>",
      "mask-source": "<url>",
      "masking-mode": "alpha|luminance|match-source",
      "matrix()": "matrix( <number>#{6} )",
      "matrix3d()": "matrix3d( <number>#{16} )",
      "max()": "max( <calc-sum># )",
      "media-and": "<media-in-parens> [and <media-in-parens>]+",
      "media-condition": "<media-not>|<media-and>|<media-or>|<media-in-parens>",
      "media-condition-without-or": "<media-not>|<media-and>|<media-in-parens>",
      "media-feature": "( [<mf-plain>|<mf-boolean>|<mf-range>] )",
      "media-in-parens":
        "( <media-condition> )|<media-feature>|<general-enclosed>",
      "media-not": "not <media-in-parens>",
      "media-or": "<media-in-parens> [or <media-in-parens>]+",
      "media-query":
        "<media-condition>|[not|only]? <media-type> [and <media-condition-without-or>]?",
      "media-query-list": "<media-query>#",
      "media-type": "<ident>",
      "mf-boolean": "<mf-name>",
      "mf-name": "<ident>",
      "mf-plain": "<mf-name> : <mf-value>",
      "mf-range":
        "<mf-name> ['<'|'>']? '='? <mf-value>|<mf-value> ['<'|'>']? '='? <mf-name>|<mf-value> '<' '='? <mf-name> '<' '='? <mf-value>|<mf-value> '>' '='? <mf-name> '>' '='? <mf-value>",
      "mf-value": "<number>|<dimension>|<ident>|<ratio>",
      "min()": "min( <calc-sum># )",
      "minmax()":
        "minmax( [<length-percentage>|min-content|max-content|auto] , [<length-percentage>|<flex>|min-content|max-content|auto] )",
      "mod()": "mod( <calc-sum> , <calc-sum> )",
      "name-repeat":
        "repeat( [<integer [1,\u221E]>|auto-fill] , <line-names>+ )",
      "named-color":
        "transparent|aliceblue|antiquewhite|aqua|aquamarine|azure|beige|bisque|black|blanchedalmond|blue|blueviolet|brown|burlywood|cadetblue|chartreuse|chocolate|coral|cornflowerblue|cornsilk|crimson|cyan|darkblue|darkcyan|darkgoldenrod|darkgray|darkgreen|darkgrey|darkkhaki|darkmagenta|darkolivegreen|darkorange|darkorchid|darkred|darksalmon|darkseagreen|darkslateblue|darkslategray|darkslategrey|darkturquoise|darkviolet|deeppink|deepskyblue|dimgray|dimgrey|dodgerblue|firebrick|floralwhite|forestgreen|fuchsia|gainsboro|ghostwhite|gold|goldenrod|gray|green|greenyellow|grey|honeydew|hotpink|indianred|indigo|ivory|khaki|lavender|lavenderblush|lawngreen|lemonchiffon|lightblue|lightcoral|lightcyan|lightgoldenrodyellow|lightgray|lightgreen|lightgrey|lightpink|lightsalmon|lightseagreen|lightskyblue|lightslategray|lightslategrey|lightsteelblue|lightyellow|lime|limegreen|linen|magenta|maroon|mediumaquamarine|mediumblue|mediumorchid|mediumpurple|mediumseagreen|mediumslateblue|mediumspringgreen|mediumturquoise|mediumvioletred|midnightblue|mintcream|mistyrose|moccasin|navajowhite|navy|oldlace|olive|olivedrab|orange|orangered|orchid|palegoldenrod|palegreen|paleturquoise|palevioletred|papayawhip|peachpuff|peru|pink|plum|powderblue|purple|rebeccapurple|red|rosybrown|royalblue|saddlebrown|salmon|sandybrown|seagreen|seashell|sienna|silver|skyblue|slateblue|slategray|slategrey|snow|springgreen|steelblue|tan|teal|thistle|tomato|turquoise|violet|wheat|white|whitesmoke|yellow|yellowgreen|<-non-standard-color>",
      "namespace-prefix": "<ident>",
      "ns-prefix": "[<ident-token>|'*']? '|'",
      "number-percentage": "<number>|<percentage>",
      "numeric-figure-values": "[lining-nums|oldstyle-nums]",
      "numeric-fraction-values": "[diagonal-fractions|stacked-fractions]",
      "numeric-spacing-values": "[proportional-nums|tabular-nums]",
      nth: "<an-plus-b>|even|odd",
      "opacity()": "opacity( [<number-percentage>] )",
      "overflow-position": "unsafe|safe",
      "outline-radius": "<length>|<percentage>",
      "page-body":
        "<declaration>? [; <page-body>]?|<page-margin-box> <page-body>",
      "page-margin-box": "<page-margin-box-type> '{' <declaration-list> '}'",
      "page-margin-box-type":
        "@top-left-corner|@top-left|@top-center|@top-right|@top-right-corner|@bottom-left-corner|@bottom-left|@bottom-center|@bottom-right|@bottom-right-corner|@left-top|@left-middle|@left-bottom|@right-top|@right-middle|@right-bottom",
      "page-selector-list": "[<page-selector>#]?",
      "page-selector": "<pseudo-page>+|<ident> <pseudo-page>*",
      "page-size": "A5|A4|A3|B5|B4|JIS-B5|JIS-B4|letter|legal|ledger",
      "path()": "path( [<fill-rule> ,]? <string> )",
      "paint()": "paint( <ident> , <declaration-value>? )",
      "perspective()": "perspective( [<length [0,\u221E]>|none] )",
      "polygon()":
        "polygon( <fill-rule>? , [<length-percentage> <length-percentage>]# )",
      position:
        "[[left|center|right]||[top|center|bottom]|[left|center|right|<length-percentage>] [top|center|bottom|<length-percentage>]?|[[left|right] <length-percentage>]&&[[top|bottom] <length-percentage>]]",
      "pow()": "pow( <calc-sum> , <calc-sum> )",
      "pseudo-class-selector":
        "':' <ident-token>|':' <function-token> <any-value> ')'",
      "pseudo-element-selector": "':' <pseudo-class-selector>",
      "pseudo-page": ": [left|right|first|blank]",
      quote: "open-quote|close-quote|no-open-quote|no-close-quote",
      "radial-gradient()":
        "radial-gradient( [<ending-shape>||<size>]? [at <position>]? , <color-stop-list> )",
      ratio: "<number [0,\u221E]> [/ <number [0,\u221E]>]?",
      "relative-selector": "<combinator>? <complex-selector>",
      "relative-selector-list": "<relative-selector>#",
      "relative-size": "larger|smaller",
      "rem()": "rem( <calc-sum> , <calc-sum> )",
      "repeat-style": "repeat-x|repeat-y|[repeat|space|round|no-repeat]{1,2}",
      "repeating-conic-gradient()":
        "repeating-conic-gradient( [from <angle>]? [at <position>]? , <angular-color-stop-list> )",
      "repeating-linear-gradient()":
        "repeating-linear-gradient( [<angle>|to <side-or-corner>]? , <color-stop-list> )",
      "repeating-radial-gradient()":
        "repeating-radial-gradient( [<ending-shape>||<size>]? [at <position>]? , <color-stop-list> )",
      "reversed-counter-name": "reversed( <counter-name> )",
      "rgb()":
        "rgb( <percentage>{3} [/ <alpha-value>]? )|rgb( <number>{3} [/ <alpha-value>]? )|rgb( <percentage>#{3} , <alpha-value>? )|rgb( <number>#{3} , <alpha-value>? )",
      "rgba()":
        "rgba( <percentage>{3} [/ <alpha-value>]? )|rgba( <number>{3} [/ <alpha-value>]? )|rgba( <percentage>#{3} , <alpha-value>? )|rgba( <number>#{3} , <alpha-value>? )",
      "rotate()": "rotate( [<angle>|<zero>] )",
      "rotate3d()":
        "rotate3d( <number> , <number> , <number> , [<angle>|<zero>] )",
      "rotateX()": "rotateX( [<angle>|<zero>] )",
      "rotateY()": "rotateY( [<angle>|<zero>] )",
      "rotateZ()": "rotateZ( [<angle>|<zero>] )",
      "round()": "round( <rounding-strategy>? , <calc-sum> , <calc-sum> )",
      "rounding-strategy": "nearest|up|down|to-zero",
      "saturate()": "saturate( <number-percentage> )",
      "scale()": "scale( [<number>|<percentage>]#{1,2} )",
      "scale3d()": "scale3d( [<number>|<percentage>]#{3} )",
      "scaleX()": "scaleX( [<number>|<percentage>] )",
      "scaleY()": "scaleY( [<number>|<percentage>] )",
      "scaleZ()": "scaleZ( [<number>|<percentage>] )",
      scroller: "root|nearest",
      "self-position":
        "center|start|end|self-start|self-end|flex-start|flex-end",
      "shape-radius": "<length-percentage>|closest-side|farthest-side",
      "sign()": "sign( <calc-sum> )",
      "skew()": "skew( [<angle>|<zero>] , [<angle>|<zero>]? )",
      "skewX()": "skewX( [<angle>|<zero>] )",
      "skewY()": "skewY( [<angle>|<zero>] )",
      "sepia()": "sepia( <number-percentage> )",
      shadow: "inset?&&<length>{2,4}&&<color>?",
      "shadow-t": "[<length>{2,3}&&<color>?]",
      shape:
        "rect( <top> , <right> , <bottom> , <left> )|rect( <top> <right> <bottom> <left> )",
      "shape-box": "<box>|margin-box",
      "side-or-corner": "[left|right]||[top|bottom]",
      "sin()": "sin( <calc-sum> )",
      "single-animation":
        "<time>||<easing-function>||<time>||<single-animation-iteration-count>||<single-animation-direction>||<single-animation-fill-mode>||<single-animation-play-state>||[none|<keyframes-name>]",
      "single-animation-direction":
        "normal|reverse|alternate|alternate-reverse",
      "single-animation-fill-mode": "none|forwards|backwards|both",
      "single-animation-iteration-count": "infinite|<number>",
      "single-animation-play-state": "running|paused",
      "single-animation-timeline":
        "auto|none|<timeline-name>|scroll( <axis>? <scroller>? )",
      "single-transition":
        "[none|<single-transition-property>]||<time>||<easing-function>||<time>",
      "single-transition-property": "all|<custom-ident>",
      size: "closest-side|farthest-side|closest-corner|farthest-corner|<length>|<length-percentage>{2}",
      "sqrt()": "sqrt( <calc-sum> )",
      "step-position": "jump-start|jump-end|jump-none|jump-both|start|end",
      "step-timing-function":
        "step-start|step-end|steps( <integer> [, <step-position>]? )",
      "subclass-selector":
        "<id-selector>|<class-selector>|<attribute-selector>|<pseudo-class-selector>",
      "supports-condition":
        "not <supports-in-parens>|<supports-in-parens> [and <supports-in-parens>]*|<supports-in-parens> [or <supports-in-parens>]*",
      "supports-in-parens":
        "( <supports-condition> )|<supports-feature>|<general-enclosed>",
      "supports-feature": "<supports-decl>|<supports-selector-fn>",
      "supports-decl": "( <declaration> )",
      "supports-selector-fn": "selector( <complex-selector> )",
      symbol: "<string>|<image>|<custom-ident>",
      "tan()": "tan( <calc-sum> )",
      target: "<target-counter()>|<target-counters()>|<target-text()>",
      "target-counter()":
        "target-counter( [<string>|<url>] , <custom-ident> , <counter-style>? )",
      "target-counters()":
        "target-counters( [<string>|<url>] , <custom-ident> , <string> , <counter-style>? )",
      "target-text()":
        "target-text( [<string>|<url>] , [content|before|after|first-letter]? )",
      "time-percentage": "<time>|<percentage>",
      "timeline-name": "<custom-ident>|<string>",
      "easing-function":
        "linear|<cubic-bezier-timing-function>|<step-timing-function>",
      "track-breadth":
        "<length-percentage>|<flex>|min-content|max-content|auto",
      "track-list":
        "[<line-names>? [<track-size>|<track-repeat>]]+ <line-names>?",
      "track-repeat":
        "repeat( [<integer [1,\u221E]>] , [<line-names>? <track-size>]+ <line-names>? )",
      "track-size":
        "<track-breadth>|minmax( <inflexible-breadth> , <track-breadth> )|fit-content( <length-percentage> )",
      "transform-function":
        "<matrix()>|<translate()>|<translateX()>|<translateY()>|<scale()>|<scaleX()>|<scaleY()>|<rotate()>|<skew()>|<skewX()>|<skewY()>|<matrix3d()>|<translate3d()>|<translateZ()>|<scale3d()>|<scaleZ()>|<rotate3d()>|<rotateX()>|<rotateY()>|<rotateZ()>|<perspective()>",
      "transform-list": "<transform-function>+",
      "translate()": "translate( <length-percentage> , <length-percentage>? )",
      "translate3d()":
        "translate3d( <length-percentage> , <length-percentage> , <length> )",
      "translateX()": "translateX( <length-percentage> )",
      "translateY()": "translateY( <length-percentage> )",
      "translateZ()": "translateZ( <length> )",
      "type-or-unit":
        "string|color|url|integer|number|length|angle|time|frequency|cap|ch|em|ex|ic|lh|rlh|rem|vb|vi|vw|vh|vmin|vmax|mm|Q|cm|in|pt|pc|px|deg|grad|rad|turn|ms|s|Hz|kHz|%",
      "type-selector": "<wq-name>|<ns-prefix>? '*'",
      "var()": "var( <custom-property-name> , <declaration-value>? )",
      "viewport-length": "auto|<length-percentage>",
      "visual-box": "content-box|padding-box|border-box",
      "wq-name": "<ns-prefix>? <ident-token>",
      "-legacy-gradient":
        "<-webkit-gradient()>|<-legacy-linear-gradient>|<-legacy-repeating-linear-gradient>|<-legacy-radial-gradient>|<-legacy-repeating-radial-gradient>",
      "-legacy-linear-gradient":
        "-moz-linear-gradient( <-legacy-linear-gradient-arguments> )|-webkit-linear-gradient( <-legacy-linear-gradient-arguments> )|-o-linear-gradient( <-legacy-linear-gradient-arguments> )",
      "-legacy-repeating-linear-gradient":
        "-moz-repeating-linear-gradient( <-legacy-linear-gradient-arguments> )|-webkit-repeating-linear-gradient( <-legacy-linear-gradient-arguments> )|-o-repeating-linear-gradient( <-legacy-linear-gradient-arguments> )",
      "-legacy-linear-gradient-arguments":
        "[<angle>|<side-or-corner>]? , <color-stop-list>",
      "-legacy-radial-gradient":
        "-moz-radial-gradient( <-legacy-radial-gradient-arguments> )|-webkit-radial-gradient( <-legacy-radial-gradient-arguments> )|-o-radial-gradient( <-legacy-radial-gradient-arguments> )",
      "-legacy-repeating-radial-gradient":
        "-moz-repeating-radial-gradient( <-legacy-radial-gradient-arguments> )|-webkit-repeating-radial-gradient( <-legacy-radial-gradient-arguments> )|-o-repeating-radial-gradient( <-legacy-radial-gradient-arguments> )",
      "-legacy-radial-gradient-arguments":
        "[<position> ,]? [[[<-legacy-radial-gradient-shape>||<-legacy-radial-gradient-size>]|[<length>|<percentage>]{2}] ,]? <color-stop-list>",
      "-legacy-radial-gradient-size":
        "closest-side|closest-corner|farthest-side|farthest-corner|contain|cover",
      "-legacy-radial-gradient-shape": "circle|ellipse",
      "-non-standard-font":
        "-apple-system-body|-apple-system-headline|-apple-system-subheadline|-apple-system-caption1|-apple-system-caption2|-apple-system-footnote|-apple-system-short-body|-apple-system-short-headline|-apple-system-short-subheadline|-apple-system-short-caption1|-apple-system-short-footnote|-apple-system-tall-body",
      "-non-standard-color":
        "-moz-ButtonDefault|-moz-ButtonHoverFace|-moz-ButtonHoverText|-moz-CellHighlight|-moz-CellHighlightText|-moz-Combobox|-moz-ComboboxText|-moz-Dialog|-moz-DialogText|-moz-dragtargetzone|-moz-EvenTreeRow|-moz-Field|-moz-FieldText|-moz-html-CellHighlight|-moz-html-CellHighlightText|-moz-mac-accentdarkestshadow|-moz-mac-accentdarkshadow|-moz-mac-accentface|-moz-mac-accentlightesthighlight|-moz-mac-accentlightshadow|-moz-mac-accentregularhighlight|-moz-mac-accentregularshadow|-moz-mac-chrome-active|-moz-mac-chrome-inactive|-moz-mac-focusring|-moz-mac-menuselect|-moz-mac-menushadow|-moz-mac-menutextselect|-moz-MenuHover|-moz-MenuHoverText|-moz-MenuBarText|-moz-MenuBarHoverText|-moz-nativehyperlinktext|-moz-OddTreeRow|-moz-win-communicationstext|-moz-win-mediatext|-moz-activehyperlinktext|-moz-default-background-color|-moz-default-color|-moz-hyperlinktext|-moz-visitedhyperlinktext|-webkit-activelink|-webkit-focus-ring-color|-webkit-link|-webkit-text",
      "-non-standard-image-rendering":
        "optimize-contrast|-moz-crisp-edges|-o-crisp-edges|-webkit-optimize-contrast",
      "-non-standard-overflow":
        "-moz-scrollbars-none|-moz-scrollbars-horizontal|-moz-scrollbars-vertical|-moz-hidden-unscrollable",
      "-non-standard-width":
        "fill-available|min-intrinsic|intrinsic|-moz-available|-moz-fit-content|-moz-min-content|-moz-max-content|-webkit-min-content|-webkit-max-content",
      "-webkit-gradient()":
        "-webkit-gradient( <-webkit-gradient-type> , <-webkit-gradient-point> [, <-webkit-gradient-point>|, <-webkit-gradient-radius> , <-webkit-gradient-point>] [, <-webkit-gradient-radius>]? [, <-webkit-gradient-color-stop>]* )",
      "-webkit-gradient-color-stop":
        "from( <color> )|color-stop( [<number-zero-one>|<percentage>] , <color> )|to( <color> )",
      "-webkit-gradient-point":
        "[left|center|right|<length-percentage>] [top|center|bottom|<length-percentage>]",
      "-webkit-gradient-radius": "<length>|<percentage>",
      "-webkit-gradient-type": "linear|radial",
      "-webkit-mask-box-repeat": "repeat|stretch|round",
      "-webkit-mask-clip-style":
        "border|border-box|padding|padding-box|content|content-box|text",
      "-ms-filter-function-list": "<-ms-filter-function>+",
      "-ms-filter-function":
        "<-ms-filter-function-progid>|<-ms-filter-function-legacy>",
      "-ms-filter-function-progid":
        "'progid:' [<ident-token> '.']* [<ident-token>|<function-token> <any-value>? )]",
      "-ms-filter-function-legacy":
        "<ident-token>|<function-token> <any-value>? )",
      "-ms-filter": "<string>",
      age: "child|young|old",
      "attr-name": "<wq-name>",
      "attr-fallback": "<any-value>",
      "bg-clip": "<box>|border|text",
      bottom: "<length>|auto",
      "container-name": "<custom-ident>",
      "container-condition":
        "not <query-in-parens>|<query-in-parens> [[and <query-in-parens>]*|[or <query-in-parens>]*]",
      "generic-voice": "[<age>? <gender> <integer>?]",
      gender: "male|female|neutral",
      left: "<length>|auto",
      "mask-image": "<mask-reference>#",
      paint: "none|<color>|<url> [none|<color>]?|context-fill|context-stroke",
      right: "<length>|auto",
      "scroll-timeline-axis": "block|inline|vertical|horizontal",
      "scroll-timeline-name": "none|<custom-ident>",
      "single-animation-composition": "replace|add|accumulate",
      "svg-length": "<percentage>|<length>|<number>",
      "svg-writing-mode": "lr-tb|rl-tb|tb-rl|lr|rl|tb",
      top: "<length>|auto",
      x: "<number>",
      y: "<number>",
      declaration: "<ident-token> : <declaration-value>? ['!' important]?",
      "declaration-list": "[<declaration>? ';']* <declaration>?",
      url: "url( <string> <url-modifier>* )|<url-token>",
      "url-modifier": "<ident>|<function-token> <any-value> )",
      "number-zero-one": "<number [0,1]>",
      "number-one-or-greater": "<number [1,\u221E]>",
      "query-in-parens":
        "( <container-condition> )|( <size-feature> )|style( <style-query> )|<general-enclosed>",
      "size-feature": "<mf-plain>|<mf-boolean>|<mf-range>",
      "style-feature": "<declaration>",
      "style-query": "<style-condition>|<style-feature>",
      "style-condition":
        "not <style-in-parens>|<style-in-parens> [[and <style-in-parens>]*|[or <style-in-parens>]*]",
      "style-in-parens":
        "( <style-condition> )|( <style-feature> )|<general-enclosed>",
      "-non-standard-display":
        "-ms-inline-flexbox|-ms-grid|-ms-inline-grid|-webkit-flex|-webkit-inline-flex|-webkit-box|-webkit-inline-box|-moz-inline-stack|-moz-box|-moz-inline-box",
    },
    properties: {
      "--*": "<declaration-value>",
      "-ms-accelerator": "false|true",
      "-ms-block-progression": "tb|rl|bt|lr",
      "-ms-content-zoom-chaining": "none|chained",
      "-ms-content-zooming": "none|zoom",
      "-ms-content-zoom-limit":
        "<'-ms-content-zoom-limit-min'> <'-ms-content-zoom-limit-max'>",
      "-ms-content-zoom-limit-max": "<percentage>",
      "-ms-content-zoom-limit-min": "<percentage>",
      "-ms-content-zoom-snap":
        "<'-ms-content-zoom-snap-type'>||<'-ms-content-zoom-snap-points'>",
      "-ms-content-zoom-snap-points":
        "snapInterval( <percentage> , <percentage> )|snapList( <percentage># )",
      "-ms-content-zoom-snap-type": "none|proximity|mandatory",
      "-ms-filter": "<string>",
      "-ms-flow-from": "[none|<custom-ident>]#",
      "-ms-flow-into": "[none|<custom-ident>]#",
      "-ms-grid-columns": "none|<track-list>|<auto-track-list>",
      "-ms-grid-rows": "none|<track-list>|<auto-track-list>",
      "-ms-high-contrast-adjust": "auto|none",
      "-ms-hyphenate-limit-chars": "auto|<integer>{1,3}",
      "-ms-hyphenate-limit-lines": "no-limit|<integer>",
      "-ms-hyphenate-limit-zone": "<percentage>|<length>",
      "-ms-ime-align": "auto|after",
      "-ms-overflow-style": "auto|none|scrollbar|-ms-autohiding-scrollbar",
      "-ms-scrollbar-3dlight-color": "<color>",
      "-ms-scrollbar-arrow-color": "<color>",
      "-ms-scrollbar-base-color": "<color>",
      "-ms-scrollbar-darkshadow-color": "<color>",
      "-ms-scrollbar-face-color": "<color>",
      "-ms-scrollbar-highlight-color": "<color>",
      "-ms-scrollbar-shadow-color": "<color>",
      "-ms-scrollbar-track-color": "<color>",
      "-ms-scroll-chaining": "chained|none",
      "-ms-scroll-limit":
        "<'-ms-scroll-limit-x-min'> <'-ms-scroll-limit-y-min'> <'-ms-scroll-limit-x-max'> <'-ms-scroll-limit-y-max'>",
      "-ms-scroll-limit-x-max": "auto|<length>",
      "-ms-scroll-limit-x-min": "<length>",
      "-ms-scroll-limit-y-max": "auto|<length>",
      "-ms-scroll-limit-y-min": "<length>",
      "-ms-scroll-rails": "none|railed",
      "-ms-scroll-snap-points-x":
        "snapInterval( <length-percentage> , <length-percentage> )|snapList( <length-percentage># )",
      "-ms-scroll-snap-points-y":
        "snapInterval( <length-percentage> , <length-percentage> )|snapList( <length-percentage># )",
      "-ms-scroll-snap-type": "none|proximity|mandatory",
      "-ms-scroll-snap-x":
        "<'-ms-scroll-snap-type'> <'-ms-scroll-snap-points-x'>",
      "-ms-scroll-snap-y":
        "<'-ms-scroll-snap-type'> <'-ms-scroll-snap-points-y'>",
      "-ms-scroll-translation": "none|vertical-to-horizontal",
      "-ms-text-autospace":
        "none|ideograph-alpha|ideograph-numeric|ideograph-parenthesis|ideograph-space",
      "-ms-touch-select": "grippers|none",
      "-ms-user-select": "none|element|text",
      "-ms-wrap-flow": "auto|both|start|end|maximum|clear",
      "-ms-wrap-margin": "<length>",
      "-ms-wrap-through": "wrap|none",
      "-moz-appearance":
        "none|button|button-arrow-down|button-arrow-next|button-arrow-previous|button-arrow-up|button-bevel|button-focus|caret|checkbox|checkbox-container|checkbox-label|checkmenuitem|dualbutton|groupbox|listbox|listitem|menuarrow|menubar|menucheckbox|menuimage|menuitem|menuitemtext|menulist|menulist-button|menulist-text|menulist-textfield|menupopup|menuradio|menuseparator|meterbar|meterchunk|progressbar|progressbar-vertical|progresschunk|progresschunk-vertical|radio|radio-container|radio-label|radiomenuitem|range|range-thumb|resizer|resizerpanel|scale-horizontal|scalethumbend|scalethumb-horizontal|scalethumbstart|scalethumbtick|scalethumb-vertical|scale-vertical|scrollbarbutton-down|scrollbarbutton-left|scrollbarbutton-right|scrollbarbutton-up|scrollbarthumb-horizontal|scrollbarthumb-vertical|scrollbartrack-horizontal|scrollbartrack-vertical|searchfield|separator|sheet|spinner|spinner-downbutton|spinner-textfield|spinner-upbutton|splitter|statusbar|statusbarpanel|tab|tabpanel|tabpanels|tab-scroll-arrow-back|tab-scroll-arrow-forward|textfield|textfield-multiline|toolbar|toolbarbutton|toolbarbutton-dropdown|toolbargripper|toolbox|tooltip|treeheader|treeheadercell|treeheadersortarrow|treeitem|treeline|treetwisty|treetwistyopen|treeview|-moz-mac-unified-toolbar|-moz-win-borderless-glass|-moz-win-browsertabbar-toolbox|-moz-win-communicationstext|-moz-win-communications-toolbox|-moz-win-exclude-glass|-moz-win-glass|-moz-win-mediatext|-moz-win-media-toolbox|-moz-window-button-box|-moz-window-button-box-maximized|-moz-window-button-close|-moz-window-button-maximize|-moz-window-button-minimize|-moz-window-button-restore|-moz-window-frame-bottom|-moz-window-frame-left|-moz-window-frame-right|-moz-window-titlebar|-moz-window-titlebar-maximized",
      "-moz-binding": "<url>|none",
      "-moz-border-bottom-colors": "<color>+|none",
      "-moz-border-left-colors": "<color>+|none",
      "-moz-border-right-colors": "<color>+|none",
      "-moz-border-top-colors": "<color>+|none",
      "-moz-context-properties":
        "none|[fill|fill-opacity|stroke|stroke-opacity]#",
      "-moz-float-edge": "border-box|content-box|margin-box|padding-box",
      "-moz-force-broken-image-icon": "0|1",
      "-moz-image-region": "<shape>|auto",
      "-moz-orient": "inline|block|horizontal|vertical",
      "-moz-outline-radius": "<outline-radius>{1,4} [/ <outline-radius>{1,4}]?",
      "-moz-outline-radius-bottomleft": "<outline-radius>",
      "-moz-outline-radius-bottomright": "<outline-radius>",
      "-moz-outline-radius-topleft": "<outline-radius>",
      "-moz-outline-radius-topright": "<outline-radius>",
      "-moz-stack-sizing": "ignore|stretch-to-fit",
      "-moz-text-blink": "none|blink",
      "-moz-user-focus":
        "ignore|normal|select-after|select-before|select-menu|select-same|select-all|none",
      "-moz-user-input": "auto|none|enabled|disabled",
      "-moz-user-modify": "read-only|read-write|write-only",
      "-moz-window-dragging": "drag|no-drag",
      "-moz-window-shadow": "default|menu|tooltip|sheet|none",
      "-webkit-appearance":
        "none|button|button-bevel|caps-lock-indicator|caret|checkbox|default-button|inner-spin-button|listbox|listitem|media-controls-background|media-controls-fullscreen-background|media-current-time-display|media-enter-fullscreen-button|media-exit-fullscreen-button|media-fullscreen-button|media-mute-button|media-overlay-play-button|media-play-button|media-seek-back-button|media-seek-forward-button|media-slider|media-sliderthumb|media-time-remaining-display|media-toggle-closed-captions-button|media-volume-slider|media-volume-slider-container|media-volume-sliderthumb|menulist|menulist-button|menulist-text|menulist-textfield|meter|progress-bar|progress-bar-value|push-button|radio|scrollbarbutton-down|scrollbarbutton-left|scrollbarbutton-right|scrollbarbutton-up|scrollbargripper-horizontal|scrollbargripper-vertical|scrollbarthumb-horizontal|scrollbarthumb-vertical|scrollbartrack-horizontal|scrollbartrack-vertical|searchfield|searchfield-cancel-button|searchfield-decoration|searchfield-results-button|searchfield-results-decoration|slider-horizontal|slider-vertical|sliderthumb-horizontal|sliderthumb-vertical|square-button|textarea|textfield|-apple-pay-button",
      "-webkit-border-before": "<'border-width'>||<'border-style'>||<color>",
      "-webkit-border-before-color": "<color>",
      "-webkit-border-before-style": "<'border-style'>",
      "-webkit-border-before-width": "<'border-width'>",
      "-webkit-box-reflect": "[above|below|right|left]? <length>? <image>?",
      "-webkit-line-clamp": "none|<integer>",
      "-webkit-mask":
        "[<mask-reference>||<position> [/ <bg-size>]?||<repeat-style>||[<box>|border|padding|content|text]||[<box>|border|padding|content]]#",
      "-webkit-mask-attachment": "<attachment>#",
      "-webkit-mask-clip": "[<box>|border|padding|content|text]#",
      "-webkit-mask-composite": "<composite-style>#",
      "-webkit-mask-image": "<mask-reference>#",
      "-webkit-mask-origin": "[<box>|border|padding|content]#",
      "-webkit-mask-position": "<position>#",
      "-webkit-mask-position-x": "[<length-percentage>|left|center|right]#",
      "-webkit-mask-position-y": "[<length-percentage>|top|center|bottom]#",
      "-webkit-mask-repeat": "<repeat-style>#",
      "-webkit-mask-repeat-x": "repeat|no-repeat|space|round",
      "-webkit-mask-repeat-y": "repeat|no-repeat|space|round",
      "-webkit-mask-size": "<bg-size>#",
      "-webkit-overflow-scrolling": "auto|touch",
      "-webkit-tap-highlight-color": "<color>",
      "-webkit-text-fill-color": "<color>",
      "-webkit-text-stroke": "<length>||<color>",
      "-webkit-text-stroke-color": "<color>",
      "-webkit-text-stroke-width": "<length>",
      "-webkit-touch-callout": "default|none",
      "-webkit-user-modify": "read-only|read-write|read-write-plaintext-only",
      "accent-color": "auto|<color>",
      "align-content":
        "normal|<baseline-position>|<content-distribution>|<overflow-position>? <content-position>",
      "align-items":
        "normal|stretch|<baseline-position>|[<overflow-position>? <self-position>]",
      "align-self":
        "auto|normal|stretch|<baseline-position>|<overflow-position>? <self-position>",
      "align-tracks":
        "[normal|<baseline-position>|<content-distribution>|<overflow-position>? <content-position>]#",
      all: "initial|inherit|unset|revert|revert-layer",
      animation: "<single-animation>#",
      "animation-composition": "<single-animation-composition>#",
      "animation-delay": "<time>#",
      "animation-direction": "<single-animation-direction>#",
      "animation-duration": "<time>#",
      "animation-fill-mode": "<single-animation-fill-mode>#",
      "animation-iteration-count": "<single-animation-iteration-count>#",
      "animation-name": "[none|<keyframes-name>]#",
      "animation-play-state": "<single-animation-play-state>#",
      "animation-timing-function": "<easing-function>#",
      "animation-timeline": "<single-animation-timeline>#",
      appearance: "none|auto|textfield|menulist-button|<compat-auto>",
      "aspect-ratio": "auto|<ratio>",
      azimuth:
        "<angle>|[[left-side|far-left|left|center-left|center|center-right|right|far-right|right-side]||behind]|leftwards|rightwards",
      "backdrop-filter": "none|<filter-function-list>",
      "backface-visibility": "visible|hidden",
      background: "[<bg-layer> ,]* <final-bg-layer>",
      "background-attachment": "<attachment>#",
      "background-blend-mode": "<blend-mode>#",
      "background-clip": "<bg-clip>#",
      "background-color": "<color>",
      "background-image": "<bg-image>#",
      "background-origin": "<box>#",
      "background-position": "<bg-position>#",
      "background-position-x":
        "[center|[[left|right|x-start|x-end]? <length-percentage>?]!]#",
      "background-position-y":
        "[center|[[top|bottom|y-start|y-end]? <length-percentage>?]!]#",
      "background-repeat": "<repeat-style>#",
      "background-size": "<bg-size>#",
      "block-overflow": "clip|ellipsis|<string>",
      "block-size": "<'width'>",
      border: "<line-width>||<line-style>||<color>",
      "border-block": "<'border-top-width'>||<'border-top-style'>||<color>",
      "border-block-color": "<'border-top-color'>{1,2}",
      "border-block-style": "<'border-top-style'>",
      "border-block-width": "<'border-top-width'>",
      "border-block-end": "<'border-top-width'>||<'border-top-style'>||<color>",
      "border-block-end-color": "<'border-top-color'>",
      "border-block-end-style": "<'border-top-style'>",
      "border-block-end-width": "<'border-top-width'>",
      "border-block-start":
        "<'border-top-width'>||<'border-top-style'>||<color>",
      "border-block-start-color": "<'border-top-color'>",
      "border-block-start-style": "<'border-top-style'>",
      "border-block-start-width": "<'border-top-width'>",
      "border-bottom": "<line-width>||<line-style>||<color>",
      "border-bottom-color": "<'border-top-color'>",
      "border-bottom-left-radius": "<length-percentage>{1,2}",
      "border-bottom-right-radius": "<length-percentage>{1,2}",
      "border-bottom-style": "<line-style>",
      "border-bottom-width": "<line-width>",
      "border-collapse": "collapse|separate",
      "border-color": "<color>{1,4}",
      "border-end-end-radius": "<length-percentage>{1,2}",
      "border-end-start-radius": "<length-percentage>{1,2}",
      "border-image":
        "<'border-image-source'>||<'border-image-slice'> [/ <'border-image-width'>|/ <'border-image-width'>? / <'border-image-outset'>]?||<'border-image-repeat'>",
      "border-image-outset": "[<length>|<number>]{1,4}",
      "border-image-repeat": "[stretch|repeat|round|space]{1,2}",
      "border-image-slice": "<number-percentage>{1,4}&&fill?",
      "border-image-source": "none|<image>",
      "border-image-width": "[<length-percentage>|<number>|auto]{1,4}",
      "border-inline": "<'border-top-width'>||<'border-top-style'>||<color>",
      "border-inline-end":
        "<'border-top-width'>||<'border-top-style'>||<color>",
      "border-inline-color": "<'border-top-color'>{1,2}",
      "border-inline-style": "<'border-top-style'>",
      "border-inline-width": "<'border-top-width'>",
      "border-inline-end-color": "<'border-top-color'>",
      "border-inline-end-style": "<'border-top-style'>",
      "border-inline-end-width": "<'border-top-width'>",
      "border-inline-start":
        "<'border-top-width'>||<'border-top-style'>||<color>",
      "border-inline-start-color": "<'border-top-color'>",
      "border-inline-start-style": "<'border-top-style'>",
      "border-inline-start-width": "<'border-top-width'>",
      "border-left": "<line-width>||<line-style>||<color>",
      "border-left-color": "<color>",
      "border-left-style": "<line-style>",
      "border-left-width": "<line-width>",
      "border-radius": "<length-percentage>{1,4} [/ <length-percentage>{1,4}]?",
      "border-right": "<line-width>||<line-style>||<color>",
      "border-right-color": "<color>",
      "border-right-style": "<line-style>",
      "border-right-width": "<line-width>",
      "border-spacing": "<length> <length>?",
      "border-start-end-radius": "<length-percentage>{1,2}",
      "border-start-start-radius": "<length-percentage>{1,2}",
      "border-style": "<line-style>{1,4}",
      "border-top": "<line-width>||<line-style>||<color>",
      "border-top-color": "<color>",
      "border-top-left-radius": "<length-percentage>{1,2}",
      "border-top-right-radius": "<length-percentage>{1,2}",
      "border-top-style": "<line-style>",
      "border-top-width": "<line-width>",
      "border-width": "<line-width>{1,4}",
      bottom: "<length>|<percentage>|auto",
      "box-align": "start|center|end|baseline|stretch",
      "box-decoration-break": "slice|clone",
      "box-direction": "normal|reverse|inherit",
      "box-flex": "<number>",
      "box-flex-group": "<integer>",
      "box-lines": "single|multiple",
      "box-ordinal-group": "<integer>",
      "box-orient": "horizontal|vertical|inline-axis|block-axis|inherit",
      "box-pack": "start|center|end|justify",
      "box-shadow": "none|<shadow>#",
      "box-sizing": "content-box|border-box",
      "break-after":
        "auto|avoid|always|all|avoid-page|page|left|right|recto|verso|avoid-column|column|avoid-region|region",
      "break-before":
        "auto|avoid|always|all|avoid-page|page|left|right|recto|verso|avoid-column|column|avoid-region|region",
      "break-inside": "auto|avoid|avoid-page|avoid-column|avoid-region",
      "caption-side":
        "top|bottom|block-start|block-end|inline-start|inline-end",
      caret: "<'caret-color'>||<'caret-shape'>",
      "caret-color": "auto|<color>",
      "caret-shape": "auto|bar|block|underscore",
      clear: "none|left|right|both|inline-start|inline-end",
      clip: "<shape>|auto",
      "clip-path": "<clip-source>|[<basic-shape>||<geometry-box>]|none",
      color: "<color>",
      "print-color-adjust": "economy|exact",
      "color-scheme": "normal|[light|dark|<custom-ident>]+&&only?",
      "column-count": "<integer>|auto",
      "column-fill": "auto|balance|balance-all",
      "column-gap": "normal|<length-percentage>",
      "column-rule":
        "<'column-rule-width'>||<'column-rule-style'>||<'column-rule-color'>",
      "column-rule-color": "<color>",
      "column-rule-style": "<'border-style'>",
      "column-rule-width": "<'border-width'>",
      "column-span": "none|all",
      "column-width": "<length>|auto",
      columns: "<'column-width'>||<'column-count'>",
      contain:
        "none|strict|content|[[size||inline-size]||layout||style||paint]",
      "contain-intrinsic-size": "[none|<length>|auto <length>]{1,2}",
      "contain-intrinsic-block-size": "none|<length>|auto <length>",
      "contain-intrinsic-height": "none|<length>|auto <length>",
      "contain-intrinsic-inline-size": "none|<length>|auto <length>",
      "contain-intrinsic-width": "none|<length>|auto <length>",
      content:
        "normal|none|[<content-replacement>|<content-list>] [/ [<string>|<counter>]+]?",
      "content-visibility": "visible|auto|hidden",
      "counter-increment": "[<counter-name> <integer>?]+|none",
      "counter-reset":
        "[<counter-name> <integer>?|<reversed-counter-name> <integer>?]+|none",
      "counter-set": "[<counter-name> <integer>?]+|none",
      cursor:
        "[[<url> [<x> <y>]? ,]* [auto|default|none|context-menu|help|pointer|progress|wait|cell|crosshair|text|vertical-text|alias|copy|move|no-drop|not-allowed|e-resize|n-resize|ne-resize|nw-resize|s-resize|se-resize|sw-resize|w-resize|ew-resize|ns-resize|nesw-resize|nwse-resize|col-resize|row-resize|all-scroll|zoom-in|zoom-out|grab|grabbing|hand|-webkit-grab|-webkit-grabbing|-webkit-zoom-in|-webkit-zoom-out|-moz-grab|-moz-grabbing|-moz-zoom-in|-moz-zoom-out]]",
      direction: "ltr|rtl",
      display:
        "[<display-outside>||<display-inside>]|<display-listitem>|<display-internal>|<display-box>|<display-legacy>|<-non-standard-display>",
      "empty-cells": "show|hide",
      filter: "none|<filter-function-list>|<-ms-filter-function-list>",
      flex: "none|[<'flex-grow'> <'flex-shrink'>?||<'flex-basis'>]",
      "flex-basis": "content|<'width'>",
      "flex-direction": "row|row-reverse|column|column-reverse",
      "flex-flow": "<'flex-direction'>||<'flex-wrap'>",
      "flex-grow": "<number>",
      "flex-shrink": "<number>",
      "flex-wrap": "nowrap|wrap|wrap-reverse",
      float: "left|right|none|inline-start|inline-end",
      font: "[[<'font-style'>||<font-variant-css21>||<'font-weight'>||<'font-stretch'>]? <'font-size'> [/ <'line-height'>]? <'font-family'>]|caption|icon|menu|message-box|small-caption|status-bar",
      "font-family": "[<family-name>|<generic-family>]#",
      "font-feature-settings": "normal|<feature-tag-value>#",
      "font-kerning": "auto|normal|none",
      "font-language-override": "normal|<string>",
      "font-optical-sizing": "auto|none",
      "font-variation-settings": "normal|[<string> <number>]#",
      "font-size": "<absolute-size>|<relative-size>|<length-percentage>",
      "font-size-adjust":
        "none|[ex-height|cap-height|ch-width|ic-width|ic-height]? [from-font|<number>]",
      "font-smooth": "auto|never|always|<absolute-size>|<length>",
      "font-stretch": "<font-stretch-absolute>",
      "font-style": "normal|italic|oblique <angle>?",
      "font-synthesis": "none|[weight||style||small-caps]",
      "font-variant":
        "normal|none|[<common-lig-values>||<discretionary-lig-values>||<historical-lig-values>||<contextual-alt-values>||stylistic( <feature-value-name> )||historical-forms||styleset( <feature-value-name># )||character-variant( <feature-value-name># )||swash( <feature-value-name> )||ornaments( <feature-value-name> )||annotation( <feature-value-name> )||[small-caps|all-small-caps|petite-caps|all-petite-caps|unicase|titling-caps]||<numeric-figure-values>||<numeric-spacing-values>||<numeric-fraction-values>||ordinal||slashed-zero||<east-asian-variant-values>||<east-asian-width-values>||ruby]",
      "font-variant-alternates":
        "normal|[stylistic( <feature-value-name> )||historical-forms||styleset( <feature-value-name># )||character-variant( <feature-value-name># )||swash( <feature-value-name> )||ornaments( <feature-value-name> )||annotation( <feature-value-name> )]",
      "font-variant-caps":
        "normal|small-caps|all-small-caps|petite-caps|all-petite-caps|unicase|titling-caps",
      "font-variant-east-asian":
        "normal|[<east-asian-variant-values>||<east-asian-width-values>||ruby]",
      "font-variant-ligatures":
        "normal|none|[<common-lig-values>||<discretionary-lig-values>||<historical-lig-values>||<contextual-alt-values>]",
      "font-variant-numeric":
        "normal|[<numeric-figure-values>||<numeric-spacing-values>||<numeric-fraction-values>||ordinal||slashed-zero]",
      "font-variant-position": "normal|sub|super",
      "font-weight": "<font-weight-absolute>|bolder|lighter",
      "forced-color-adjust": "auto|none",
      gap: "<'row-gap'> <'column-gap'>?",
      grid: "<'grid-template'>|<'grid-template-rows'> / [auto-flow&&dense?] <'grid-auto-columns'>?|[auto-flow&&dense?] <'grid-auto-rows'>? / <'grid-template-columns'>",
      "grid-area": "<grid-line> [/ <grid-line>]{0,3}",
      "grid-auto-columns": "<track-size>+",
      "grid-auto-flow": "[row|column]||dense",
      "grid-auto-rows": "<track-size>+",
      "grid-column": "<grid-line> [/ <grid-line>]?",
      "grid-column-end": "<grid-line>",
      "grid-column-gap": "<length-percentage>",
      "grid-column-start": "<grid-line>",
      "grid-gap": "<'grid-row-gap'> <'grid-column-gap'>?",
      "grid-row": "<grid-line> [/ <grid-line>]?",
      "grid-row-end": "<grid-line>",
      "grid-row-gap": "<length-percentage>",
      "grid-row-start": "<grid-line>",
      "grid-template":
        "none|[<'grid-template-rows'> / <'grid-template-columns'>]|[<line-names>? <string> <track-size>? <line-names>?]+ [/ <explicit-track-list>]?",
      "grid-template-areas": "none|<string>+",
      "grid-template-columns":
        "none|<track-list>|<auto-track-list>|subgrid <line-name-list>?",
      "grid-template-rows":
        "none|<track-list>|<auto-track-list>|subgrid <line-name-list>?",
      "hanging-punctuation": "none|[first||[force-end|allow-end]||last]",
      height:
        "auto|<length>|<percentage>|min-content|max-content|fit-content|fit-content( <length-percentage> )",
      "hyphenate-character": "auto|<string>",
      hyphens: "none|manual|auto",
      "image-orientation": "from-image|<angle>|[<angle>? flip]",
      "image-rendering":
        "auto|crisp-edges|pixelated|optimizeSpeed|optimizeQuality|<-non-standard-image-rendering>",
      "image-resolution": "[from-image||<resolution>]&&snap?",
      "ime-mode": "auto|normal|active|inactive|disabled",
      "initial-letter": "normal|[<number> <integer>?]",
      "initial-letter-align": "[auto|alphabetic|hanging|ideographic]",
      "inline-size": "<'width'>",
      "input-security": "auto|none",
      inset: "<'top'>{1,4}",
      "inset-block": "<'top'>{1,2}",
      "inset-block-end": "<'top'>",
      "inset-block-start": "<'top'>",
      "inset-inline": "<'top'>{1,2}",
      "inset-inline-end": "<'top'>",
      "inset-inline-start": "<'top'>",
      isolation: "auto|isolate",
      "justify-content":
        "normal|<content-distribution>|<overflow-position>? [<content-position>|left|right]",
      "justify-items":
        "normal|stretch|<baseline-position>|<overflow-position>? [<self-position>|left|right]|legacy|legacy&&[left|right|center]",
      "justify-self":
        "auto|normal|stretch|<baseline-position>|<overflow-position>? [<self-position>|left|right]",
      "justify-tracks":
        "[normal|<content-distribution>|<overflow-position>? [<content-position>|left|right]]#",
      left: "<length>|<percentage>|auto",
      "letter-spacing": "normal|<length-percentage>",
      "line-break": "auto|loose|normal|strict|anywhere",
      "line-clamp": "none|<integer>",
      "line-height": "normal|<number>|<length>|<percentage>",
      "line-height-step": "<length>",
      "list-style":
        "<'list-style-type'>||<'list-style-position'>||<'list-style-image'>",
      "list-style-image": "<image>|none",
      "list-style-position": "inside|outside",
      "list-style-type": "<counter-style>|<string>|none",
      margin: "[<length>|<percentage>|auto]{1,4}",
      "margin-block": "<'margin-left'>{1,2}",
      "margin-block-end": "<'margin-left'>",
      "margin-block-start": "<'margin-left'>",
      "margin-bottom": "<length>|<percentage>|auto",
      "margin-inline": "<'margin-left'>{1,2}",
      "margin-inline-end": "<'margin-left'>",
      "margin-inline-start": "<'margin-left'>",
      "margin-left": "<length>|<percentage>|auto",
      "margin-right": "<length>|<percentage>|auto",
      "margin-top": "<length>|<percentage>|auto",
      "margin-trim": "none|in-flow|all",
      mask: "<mask-layer>#",
      "mask-border":
        "<'mask-border-source'>||<'mask-border-slice'> [/ <'mask-border-width'>? [/ <'mask-border-outset'>]?]?||<'mask-border-repeat'>||<'mask-border-mode'>",
      "mask-border-mode": "luminance|alpha",
      "mask-border-outset": "[<length>|<number>]{1,4}",
      "mask-border-repeat": "[stretch|repeat|round|space]{1,2}",
      "mask-border-slice": "<number-percentage>{1,4} fill?",
      "mask-border-source": "none|<image>",
      "mask-border-width": "[<length-percentage>|<number>|auto]{1,4}",
      "mask-clip": "[<geometry-box>|no-clip]#",
      "mask-composite": "<compositing-operator>#",
      "mask-image": "<mask-reference>#",
      "mask-mode": "<masking-mode>#",
      "mask-origin": "<geometry-box>#",
      "mask-position": "<position>#",
      "mask-repeat": "<repeat-style>#",
      "mask-size": "<bg-size>#",
      "mask-type": "luminance|alpha",
      "masonry-auto-flow": "[pack|next]||[definite-first|ordered]",
      "math-depth": "auto-add|add( <integer> )|<integer>",
      "math-shift": "normal|compact",
      "math-style": "normal|compact",
      "max-block-size": "<'max-width'>",
      "max-height":
        "none|<length-percentage>|min-content|max-content|fit-content|fit-content( <length-percentage> )",
      "max-inline-size": "<'max-width'>",
      "max-lines": "none|<integer>",
      "max-width":
        "none|<length-percentage>|min-content|max-content|fit-content|fit-content( <length-percentage> )|<-non-standard-width>",
      "min-block-size": "<'min-width'>",
      "min-height":
        "auto|<length>|<percentage>|min-content|max-content|fit-content|fit-content( <length-percentage> )",
      "min-inline-size": "<'min-width'>",
      "min-width":
        "auto|<length>|<percentage>|min-content|max-content|fit-content|fit-content( <length-percentage> )|<-non-standard-width>",
      "mix-blend-mode": "<blend-mode>|plus-lighter",
      "object-fit": "fill|contain|cover|none|scale-down",
      "object-position": "<position>",
      offset:
        "[<'offset-position'>? [<'offset-path'> [<'offset-distance'>||<'offset-rotate'>]?]?]! [/ <'offset-anchor'>]?",
      "offset-anchor": "auto|<position>",
      "offset-distance": "<length-percentage>",
      "offset-path":
        "none|ray( [<angle>&&<size>&&contain?] )|<path()>|<url>|[<basic-shape>||<geometry-box>]",
      "offset-position": "auto|<position>",
      "offset-rotate": "[auto|reverse]||<angle>",
      opacity: "<alpha-value>",
      order: "<integer>",
      orphans: "<integer>",
      outline: "[<'outline-color'>||<'outline-style'>||<'outline-width'>]",
      "outline-color": "<color>|invert",
      "outline-offset": "<length>",
      "outline-style": "auto|<'border-style'>",
      "outline-width": "<line-width>",
      overflow:
        "[visible|hidden|clip|scroll|auto]{1,2}|<-non-standard-overflow>",
      "overflow-anchor": "auto|none",
      "overflow-block": "visible|hidden|clip|scroll|auto",
      "overflow-clip-box": "padding-box|content-box",
      "overflow-clip-margin": "<visual-box>||<length [0,\u221E]>",
      "overflow-inline": "visible|hidden|clip|scroll|auto",
      "overflow-wrap": "normal|break-word|anywhere",
      "overflow-x": "visible|hidden|clip|scroll|auto",
      "overflow-y": "visible|hidden|clip|scroll|auto",
      "overscroll-behavior": "[contain|none|auto]{1,2}",
      "overscroll-behavior-block": "contain|none|auto",
      "overscroll-behavior-inline": "contain|none|auto",
      "overscroll-behavior-x": "contain|none|auto",
      "overscroll-behavior-y": "contain|none|auto",
      padding: "[<length>|<percentage>]{1,4}",
      "padding-block": "<'padding-left'>{1,2}",
      "padding-block-end": "<'padding-left'>",
      "padding-block-start": "<'padding-left'>",
      "padding-bottom": "<length>|<percentage>",
      "padding-inline": "<'padding-left'>{1,2}",
      "padding-inline-end": "<'padding-left'>",
      "padding-inline-start": "<'padding-left'>",
      "padding-left": "<length>|<percentage>",
      "padding-right": "<length>|<percentage>",
      "padding-top": "<length>|<percentage>",
      "page-break-after": "auto|always|avoid|left|right|recto|verso",
      "page-break-before": "auto|always|avoid|left|right|recto|verso",
      "page-break-inside": "auto|avoid",
      "paint-order": "normal|[fill||stroke||markers]",
      perspective: "none|<length>",
      "perspective-origin": "<position>",
      "place-content": "<'align-content'> <'justify-content'>?",
      "place-items": "<'align-items'> <'justify-items'>?",
      "place-self": "<'align-self'> <'justify-self'>?",
      "pointer-events":
        "auto|none|visiblePainted|visibleFill|visibleStroke|visible|painted|fill|stroke|all|inherit",
      position: "static|relative|absolute|sticky|fixed|-webkit-sticky",
      quotes: "none|auto|[<string> <string>]+",
      resize: "none|both|horizontal|vertical|block|inline",
      right: "<length>|<percentage>|auto",
      rotate: "none|<angle>|[x|y|z|<number>{3}]&&<angle>",
      "row-gap": "normal|<length-percentage>",
      "ruby-align": "start|center|space-between|space-around",
      "ruby-merge": "separate|collapse|auto",
      "ruby-position": "[alternate||[over|under]]|inter-character",
      scale: "none|<number>{1,3}",
      "scrollbar-color": "auto|<color>{2}",
      "scrollbar-gutter": "auto|stable&&both-edges?",
      "scrollbar-width": "auto|thin|none",
      "scroll-behavior": "auto|smooth",
      "scroll-margin": "<length>{1,4}",
      "scroll-margin-block": "<length>{1,2}",
      "scroll-margin-block-start": "<length>",
      "scroll-margin-block-end": "<length>",
      "scroll-margin-bottom": "<length>",
      "scroll-margin-inline": "<length>{1,2}",
      "scroll-margin-inline-start": "<length>",
      "scroll-margin-inline-end": "<length>",
      "scroll-margin-left": "<length>",
      "scroll-margin-right": "<length>",
      "scroll-margin-top": "<length>",
      "scroll-padding": "[auto|<length-percentage>]{1,4}",
      "scroll-padding-block": "[auto|<length-percentage>]{1,2}",
      "scroll-padding-block-start": "auto|<length-percentage>",
      "scroll-padding-block-end": "auto|<length-percentage>",
      "scroll-padding-bottom": "auto|<length-percentage>",
      "scroll-padding-inline": "[auto|<length-percentage>]{1,2}",
      "scroll-padding-inline-start": "auto|<length-percentage>",
      "scroll-padding-inline-end": "auto|<length-percentage>",
      "scroll-padding-left": "auto|<length-percentage>",
      "scroll-padding-right": "auto|<length-percentage>",
      "scroll-padding-top": "auto|<length-percentage>",
      "scroll-snap-align": "[none|start|end|center]{1,2}",
      "scroll-snap-coordinate": "none|<position>#",
      "scroll-snap-destination": "<position>",
      "scroll-snap-points-x": "none|repeat( <length-percentage> )",
      "scroll-snap-points-y": "none|repeat( <length-percentage> )",
      "scroll-snap-stop": "normal|always",
      "scroll-snap-type": "none|[x|y|block|inline|both] [mandatory|proximity]?",
      "scroll-snap-type-x": "none|mandatory|proximity",
      "scroll-snap-type-y": "none|mandatory|proximity",
      "scroll-timeline": "<scroll-timeline-name>||<scroll-timeline-axis>",
      "scroll-timeline-axis": "block|inline|vertical|horizontal",
      "scroll-timeline-name": "none|<custom-ident>",
      "shape-image-threshold": "<alpha-value>",
      "shape-margin": "<length-percentage>",
      "shape-outside": "none|[<shape-box>||<basic-shape>]|<image>",
      "tab-size": "<integer>|<length>",
      "table-layout": "auto|fixed",
      "text-align": "start|end|left|right|center|justify|match-parent",
      "text-align-last": "auto|start|end|left|right|center|justify",
      "text-combine-upright": "none|all|[digits <integer>?]",
      "text-decoration":
        "<'text-decoration-line'>||<'text-decoration-style'>||<'text-decoration-color'>||<'text-decoration-thickness'>",
      "text-decoration-color": "<color>",
      "text-decoration-line":
        "none|[underline||overline||line-through||blink]|spelling-error|grammar-error",
      "text-decoration-skip":
        "none|[objects||[spaces|[leading-spaces||trailing-spaces]]||edges||box-decoration]",
      "text-decoration-skip-ink": "auto|all|none",
      "text-decoration-style": "solid|double|dotted|dashed|wavy",
      "text-decoration-thickness": "auto|from-font|<length>|<percentage>",
      "text-emphasis": "<'text-emphasis-style'>||<'text-emphasis-color'>",
      "text-emphasis-color": "<color>",
      "text-emphasis-position": "[over|under]&&[right|left]",
      "text-emphasis-style":
        "none|[[filled|open]||[dot|circle|double-circle|triangle|sesame]]|<string>",
      "text-indent": "<length-percentage>&&hanging?&&each-line?",
      "text-justify": "auto|inter-character|inter-word|none",
      "text-orientation": "mixed|upright|sideways",
      "text-overflow": "[clip|ellipsis|<string>]{1,2}",
      "text-rendering":
        "auto|optimizeSpeed|optimizeLegibility|geometricPrecision",
      "text-shadow": "none|<shadow-t>#",
      "text-size-adjust": "none|auto|<percentage>",
      "text-transform":
        "none|capitalize|uppercase|lowercase|full-width|full-size-kana",
      "text-underline-offset": "auto|<length>|<percentage>",
      "text-underline-position": "auto|from-font|[under||[left|right]]",
      top: "<length>|<percentage>|auto",
      "touch-action":
        "auto|none|[[pan-x|pan-left|pan-right]||[pan-y|pan-up|pan-down]||pinch-zoom]|manipulation",
      transform: "none|<transform-list>",
      "transform-box": "content-box|border-box|fill-box|stroke-box|view-box",
      "transform-origin":
        "[<length-percentage>|left|center|right|top|bottom]|[[<length-percentage>|left|center|right]&&[<length-percentage>|top|center|bottom]] <length>?",
      "transform-style": "flat|preserve-3d",
      transition: "<single-transition>#",
      "transition-delay": "<time>#",
      "transition-duration": "<time>#",
      "transition-property": "none|<single-transition-property>#",
      "transition-timing-function": "<easing-function>#",
      translate: "none|<length-percentage> [<length-percentage> <length>?]?",
      "unicode-bidi":
        "normal|embed|isolate|bidi-override|isolate-override|plaintext|-moz-isolate|-moz-isolate-override|-moz-plaintext|-webkit-isolate|-webkit-isolate-override|-webkit-plaintext",
      "user-select": "auto|text|none|contain|all",
      "vertical-align":
        "baseline|sub|super|text-top|text-bottom|middle|top|bottom|<percentage>|<length>",
      visibility: "visible|hidden|collapse",
      "white-space": "normal|pre|nowrap|pre-wrap|pre-line|break-spaces",
      widows: "<integer>",
      width:
        "auto|<length>|<percentage>|min-content|max-content|fit-content|fit-content( <length-percentage> )|fill|stretch|intrinsic|-moz-max-content|-webkit-max-content|-moz-fit-content|-webkit-fit-content",
      "will-change": "auto|<animateable-feature>#",
      "word-break": "normal|break-all|keep-all|break-word",
      "word-spacing": "normal|<length>",
      "word-wrap": "normal|break-word",
      "writing-mode":
        "horizontal-tb|vertical-rl|vertical-lr|sideways-rl|sideways-lr|<svg-writing-mode>",
      "z-index": "auto|<integer>",
      zoom: "normal|reset|<number>|<percentage>",
      "-moz-background-clip": "padding|border",
      "-moz-border-radius-bottomleft": "<'border-bottom-left-radius'>",
      "-moz-border-radius-bottomright": "<'border-bottom-right-radius'>",
      "-moz-border-radius-topleft": "<'border-top-left-radius'>",
      "-moz-border-radius-topright": "<'border-bottom-right-radius'>",
      "-moz-control-character-visibility": "visible|hidden",
      "-moz-osx-font-smoothing": "auto|grayscale",
      "-moz-user-select": "none|text|all|-moz-none",
      "-ms-flex-align": "start|end|center|baseline|stretch",
      "-ms-flex-item-align": "auto|start|end|center|baseline|stretch",
      "-ms-flex-line-pack": "start|end|center|justify|distribute|stretch",
      "-ms-flex-negative": "<'flex-shrink'>",
      "-ms-flex-pack": "start|end|center|justify|distribute",
      "-ms-flex-order": "<integer>",
      "-ms-flex-positive": "<'flex-grow'>",
      "-ms-flex-preferred-size": "<'flex-basis'>",
      "-ms-interpolation-mode": "nearest-neighbor|bicubic",
      "-ms-grid-column-align": "start|end|center|stretch",
      "-ms-grid-row-align": "start|end|center|stretch",
      "-ms-hyphenate-limit-last": "none|always|column|page|spread",
      "-webkit-background-clip": "[<box>|border|padding|content|text]#",
      "-webkit-column-break-after": "always|auto|avoid",
      "-webkit-column-break-before": "always|auto|avoid",
      "-webkit-column-break-inside": "always|auto|avoid",
      "-webkit-font-smoothing": "auto|none|antialiased|subpixel-antialiased",
      "-webkit-mask-box-image":
        "[<url>|<gradient>|none] [<length-percentage>{4} <-webkit-mask-box-repeat>{2}]?",
      "-webkit-print-color-adjust": "economy|exact",
      "-webkit-text-security": "none|circle|disc|square",
      "-webkit-user-drag": "none|element|auto",
      "-webkit-user-select": "auto|none|text|all",
      "alignment-baseline":
        "auto|baseline|before-edge|text-before-edge|middle|central|after-edge|text-after-edge|ideographic|alphabetic|hanging|mathematical",
      "baseline-shift": "baseline|sub|super|<svg-length>",
      behavior: "<url>+",
      "clip-rule": "nonzero|evenodd",
      container: "<'container-name'> [/ <'container-type'>]?",
      "container-name": "none|<custom-ident>+",
      "container-type": "normal||[size|inline-size]",
      cue: "<'cue-before'> <'cue-after'>?",
      "cue-after": "<url> <decibel>?|none",
      "cue-before": "<url> <decibel>?|none",
      "dominant-baseline":
        "auto|use-script|no-change|reset-size|ideographic|alphabetic|hanging|mathematical|central|middle|text-after-edge|text-before-edge",
      fill: "<paint>",
      "fill-opacity": "<number-zero-one>",
      "fill-rule": "nonzero|evenodd",
      "glyph-orientation-horizontal": "<angle>",
      "glyph-orientation-vertical": "<angle>",
      kerning: "auto|<svg-length>",
      marker: "none|<url>",
      "marker-end": "none|<url>",
      "marker-mid": "none|<url>",
      "marker-start": "none|<url>",
      pause: "<'pause-before'> <'pause-after'>?",
      "pause-after": "<time>|none|x-weak|weak|medium|strong|x-strong",
      "pause-before": "<time>|none|x-weak|weak|medium|strong|x-strong",
      rest: "<'rest-before'> <'rest-after'>?",
      "rest-after": "<time>|none|x-weak|weak|medium|strong|x-strong",
      "rest-before": "<time>|none|x-weak|weak|medium|strong|x-strong",
      "shape-rendering": "auto|optimizeSpeed|crispEdges|geometricPrecision",
      src: "[<url> [format( <string># )]?|local( <family-name> )]#",
      speak: "auto|none|normal",
      "speak-as":
        "normal|spell-out||digits||[literal-punctuation|no-punctuation]",
      stroke: "<paint>",
      "stroke-dasharray": "none|[<svg-length>+]#",
      "stroke-dashoffset": "<svg-length>",
      "stroke-linecap": "butt|round|square",
      "stroke-linejoin": "miter|round|bevel",
      "stroke-miterlimit": "<number-one-or-greater>",
      "stroke-opacity": "<number-zero-one>",
      "stroke-width": "<svg-length>",
      "text-anchor": "start|middle|end",
      "unicode-range": "<urange>#",
      "voice-balance": "<number>|left|center|right|leftwards|rightwards",
      "voice-duration": "auto|<time>",
      "voice-family":
        "[[<family-name>|<generic-voice>] ,]* [<family-name>|<generic-voice>]|preserve",
      "voice-pitch":
        "<frequency>&&absolute|[[x-low|low|medium|high|x-high]||[<frequency>|<semitones>|<percentage>]]",
      "voice-range":
        "<frequency>&&absolute|[[x-low|low|medium|high|x-high]||[<frequency>|<semitones>|<percentage>]]",
      "voice-rate": "[normal|x-slow|slow|medium|fast|x-fast]||<percentage>",
      "voice-stress": "normal|strong|moderate|none|reduced",
      "voice-volume": "silent|[[x-soft|soft|medium|loud|x-loud]||<decibel>]",
    },
    atrules: {
      charset: { prelude: "<string>", descriptors: null },
      "counter-style": {
        prelude: "<counter-style-name>",
        descriptors: {
          "additive-symbols": "[<integer>&&<symbol>]#",
          fallback: "<counter-style-name>",
          negative: "<symbol> <symbol>?",
          pad: "<integer>&&<symbol>",
          prefix: "<symbol>",
          range: "[[<integer>|infinite]{2}]#|auto",
          "speak-as":
            "auto|bullets|numbers|words|spell-out|<counter-style-name>",
          suffix: "<symbol>",
          symbols: "<symbol>+",
          system:
            "cyclic|numeric|alphabetic|symbolic|additive|[fixed <integer>?]|[extends <counter-style-name>]",
        },
      },
      document: {
        prelude:
          "[<url>|url-prefix( <string> )|domain( <string> )|media-document( <string> )|regexp( <string> )]#",
        descriptors: null,
      },
      "font-face": {
        prelude: null,
        descriptors: {
          "ascent-override": "normal|<percentage>",
          "descent-override": "normal|<percentage>",
          "font-display": "[auto|block|swap|fallback|optional]",
          "font-family": "<family-name>",
          "font-feature-settings": "normal|<feature-tag-value>#",
          "font-variation-settings": "normal|[<string> <number>]#",
          "font-stretch": "<font-stretch-absolute>{1,2}",
          "font-style": "normal|italic|oblique <angle>{0,2}",
          "font-weight": "<font-weight-absolute>{1,2}",
          "font-variant":
            "normal|none|[<common-lig-values>||<discretionary-lig-values>||<historical-lig-values>||<contextual-alt-values>||stylistic( <feature-value-name> )||historical-forms||styleset( <feature-value-name># )||character-variant( <feature-value-name># )||swash( <feature-value-name> )||ornaments( <feature-value-name> )||annotation( <feature-value-name> )||[small-caps|all-small-caps|petite-caps|all-petite-caps|unicase|titling-caps]||<numeric-figure-values>||<numeric-spacing-values>||<numeric-fraction-values>||ordinal||slashed-zero||<east-asian-variant-values>||<east-asian-width-values>||ruby]",
          "line-gap-override": "normal|<percentage>",
          "size-adjust": "<percentage>",
          src: "[<url> [format( <string># )]?|local( <family-name> )]#",
          "unicode-range": "<urange>#",
        },
      },
      "font-feature-values": { prelude: "<family-name>#", descriptors: null },
      import: {
        prelude:
          "[<string>|<url>] [layer|layer( <layer-name> )]? [supports( [<supports-condition>|<declaration>] )]? <media-query-list>?",
        descriptors: null,
      },
      keyframes: { prelude: "<keyframes-name>", descriptors: null },
      layer: { prelude: "[<layer-name>#|<layer-name>?]", descriptors: null },
      media: { prelude: "<media-query-list>", descriptors: null },
      namespace: {
        prelude: "<namespace-prefix>? [<string>|<url>]",
        descriptors: null,
      },
      page: {
        prelude: "<page-selector-list>",
        descriptors: {
          bleed: "auto|<length>",
          marks: "none|[crop||cross]",
          size: "<length>{1,2}|auto|[<page-size>||[portrait|landscape]]",
        },
      },
      property: {
        prelude: "<custom-property-name>",
        descriptors: {
          syntax: "<string>",
          inherits: "true|false",
          "initial-value": "<string>",
        },
      },
      "scroll-timeline": { prelude: "<timeline-name>", descriptors: null },
      supports: { prelude: "<supports-condition>", descriptors: null },
      viewport: {
        prelude: null,
        descriptors: {
          height: "<viewport-length>{1,2}",
          "max-height": "<viewport-length>",
          "max-width": "<viewport-length>",
          "max-zoom": "auto|<number>|<percentage>",
          "min-height": "<viewport-length>",
          "min-width": "<viewport-length>",
          "min-zoom": "auto|<number>|<percentage>",
          orientation: "auto|portrait|landscape",
          "user-zoom": "zoom|fixed",
          "viewport-fit": "auto|contain|cover",
          width: "<viewport-length>{1,2}",
          zoom: "auto|<number>|<percentage>",
        },
      },
      container: {
        prelude: "[<container-name>]? <container-condition>",
        descriptors: null,
      },
      nest: { prelude: "<complex-selector-list>", descriptors: null },
    },
  };
  var xt = {};
  b(xt, {
    AnPlusB: () => en,
    Atrule: () => rn,
    AtrulePrelude: () => on,
    AttributeSelector: () => ln,
    Block: () => un,
    Brackets: () => hn,
    CDC: () => fn,
    CDO: () => gn,
    ClassSelector: () => xn,
    Combinator: () => kn,
    Comment: () => vn,
    Declaration: () => Cn,
    DeclarationList: () => En,
    Dimension: () => Pn,
    Function: () => Dn,
    Hash: () => Nn,
    IdSelector: () => Fn,
    Identifier: () => Mn,
    MediaFeature: () => _n,
    MediaQuery: () => Un,
    MediaQueryList: () => Wn,
    NestingSelector: () => Yn,
    Nth: () => Vn,
    Number: () => Qn,
    Operator: () => $n,
    Parentheses: () => Jn,
    Percentage: () => to,
    PseudoClassSelector: () => no,
    PseudoElementSelector: () => io,
    Ratio: () => so,
    Raw: () => co,
    Rule: () => po,
    Selector: () => mo,
    SelectorList: () => go,
    String: () => ko,
    StyleSheet: () => vo,
    TypeSelector: () => Ao,
    UnicodeRange: () => Lo,
    Url: () => No,
    Value: () => Mo,
    WhiteSpace: () => Fo,
  });
  var en = {};
  b(en, {
    generate: () => kc,
    name: () => xc,
    parse: () => Jr,
    structure: () => yc,
  });
  var me = 43,
    re = 45,
    Jt = 110,
    Oe = !0,
    bc = !1;
  function er(e, t) {
    let r = this.tokenStart + e,
      n = this.charCodeAt(r);
    for (
      (n === me || n === re) &&
      (t && this.error("Number sign is not allowed"), r++);
      r < this.tokenEnd;
      r++
    )
      B(this.charCodeAt(r)) || this.error("Integer is expected", r);
  }
  function $e(e) {
    return er.call(this, 0, e);
  }
  function Ce(e, t) {
    if (!this.cmpChar(this.tokenStart + e, t)) {
      let r = "";
      switch (t) {
        case Jt:
          r = "N is expected";
          break;
        case re:
          r = "HyphenMinus is expected";
          break;
      }
      this.error(r, this.tokenStart + e);
    }
  }
  function Zr() {
    let e = 0,
      t = 0,
      r = this.tokenType;
    for (; r === 13 || r === 25; ) r = this.lookupType(++e);
    if (r !== 10)
      if (this.isDelim(me, e) || this.isDelim(re, e)) {
        t = this.isDelim(me, e) ? me : re;
        do r = this.lookupType(++e);
        while (r === 13 || r === 25);
        r !== 10 && (this.skip(e), $e.call(this, Oe));
      } else return null;
    return (
      e > 0 && this.skip(e),
      t === 0 &&
        ((r = this.charCodeAt(this.tokenStart)),
        r !== me && r !== re && this.error("Number sign is expected")),
      $e.call(this, t !== 0),
      t === re ? "-" + this.consume(10) : this.consume(10)
    );
  }
  var xc = "AnPlusB",
    yc = { a: [String, null], b: [String, null] };
  function Jr() {
    let e = this.tokenStart,
      t = null,
      r = null;
    if (this.tokenType === 10) $e.call(this, bc), (r = this.consume(10));
    else if (this.tokenType === 1 && this.cmpChar(this.tokenStart, re))
      switch (
        ((t = "-1"), Ce.call(this, 1, Jt), this.tokenEnd - this.tokenStart)
      ) {
        case 2:
          this.next(), (r = Zr.call(this));
          break;
        case 3:
          Ce.call(this, 2, re),
            this.next(),
            this.skipSC(),
            $e.call(this, Oe),
            (r = "-" + this.consume(10));
          break;
        default:
          Ce.call(this, 2, re),
            er.call(this, 3, Oe),
            this.next(),
            (r = this.substrToCursor(e + 2));
      }
    else if (
      this.tokenType === 1 ||
      (this.isDelim(me) && this.lookupType(1) === 1)
    ) {
      let n = 0;
      switch (
        ((t = "1"),
        this.isDelim(me) && ((n = 1), this.next()),
        Ce.call(this, 0, Jt),
        this.tokenEnd - this.tokenStart)
      ) {
        case 1:
          this.next(), (r = Zr.call(this));
          break;
        case 2:
          Ce.call(this, 1, re),
            this.next(),
            this.skipSC(),
            $e.call(this, Oe),
            (r = "-" + this.consume(10));
          break;
        default:
          Ce.call(this, 1, re),
            er.call(this, 2, Oe),
            this.next(),
            (r = this.substrToCursor(e + n + 1));
      }
    } else if (this.tokenType === 12) {
      let n = this.charCodeAt(this.tokenStart),
        o = n === me || n === re,
        i = this.tokenStart + o;
      for (; i < this.tokenEnd && B(this.charCodeAt(i)); i++);
      i === this.tokenStart + o &&
        this.error("Integer is expected", this.tokenStart + o),
        Ce.call(this, i - this.tokenStart, Jt),
        (t = this.substring(e, i)),
        i + 1 === this.tokenEnd
          ? (this.next(), (r = Zr.call(this)))
          : (Ce.call(this, i - this.tokenStart + 1, re),
            i + 2 === this.tokenEnd
              ? (this.next(),
                this.skipSC(),
                $e.call(this, Oe),
                (r = "-" + this.consume(10)))
              : (er.call(this, i - this.tokenStart + 2, Oe),
                this.next(),
                (r = this.substrToCursor(i + 1))));
    } else this.error();
    return (
      t !== null && t.charCodeAt(0) === me && (t = t.substr(1)),
      r !== null && r.charCodeAt(0) === me && (r = r.substr(1)),
      { type: "AnPlusB", loc: this.getLocation(e, this.tokenStart), a: t, b: r }
    );
  }
  function kc(e) {
    if (e.a) {
      let t =
        (e.a === "+1" && "n") ||
        (e.a === "1" && "n") ||
        (e.a === "-1" && "-n") ||
        e.a + "n";
      if (e.b) {
        let r = e.b[0] === "-" || e.b[0] === "+" ? e.b : "+" + e.b;
        this.tokenize(t + r);
      } else this.tokenize(t);
    } else this.tokenize(e.b);
  }
  var rn = {};
  b(rn, {
    generate: () => Ac,
    name: () => vc,
    parse: () => tn,
    structure: () => Cc,
    walkContext: () => Sc,
  });
  function ba(e) {
    return this.Raw(e, this.consumeUntilLeftCurlyBracketOrSemicolon, !0);
  }
  function wc() {
    for (let e = 1, t; (t = this.lookupType(e)); e++) {
      if (t === 24) return !0;
      if (t === 23 || t === 3) return !1;
    }
    return !1;
  }
  var vc = "Atrule",
    Sc = "atrule",
    Cc = {
      name: String,
      prelude: ["AtrulePrelude", "Raw", null],
      block: ["Block", null],
    };
  function tn(e = !1) {
    let t = this.tokenStart,
      r,
      n,
      o = null,
      i = null;
    switch (
      (this.eat(3),
      (r = this.substrToCursor(t + 1)),
      (n = r.toLowerCase()),
      this.skipSC(),
      this.eof === !1 &&
        this.tokenType !== 23 &&
        this.tokenType !== 17 &&
        (this.parseAtrulePrelude
          ? (o = this.parseWithFallback(
              this.AtrulePrelude.bind(this, r, e),
              ba
            ))
          : (o = ba.call(this, this.tokenIndex)),
        this.skipSC()),
      this.tokenType)
    ) {
      case 17:
        this.next();
        break;
      case 23:
        hasOwnProperty.call(this.atrule, n) &&
        typeof this.atrule[n].block == "function"
          ? (i = this.atrule[n].block.call(this, e))
          : (i = this.Block(wc.call(this)));
        break;
    }
    return {
      type: "Atrule",
      loc: this.getLocation(t, this.tokenStart),
      name: r,
      prelude: o,
      block: i,
    };
  }
  function Ac(e) {
    this.token(3, "@" + e.name),
      e.prelude !== null && this.node(e.prelude),
      e.block ? this.node(e.block) : this.token(17, ";");
  }
  var on = {};
  b(on, {
    generate: () => Pc,
    name: () => Tc,
    parse: () => nn,
    structure: () => Lc,
    walkContext: () => Ec,
  });
  var Tc = "AtrulePrelude",
    Ec = "atrulePrelude",
    Lc = { children: [[]] };
  function nn(e) {
    let t = null;
    return (
      e !== null && (e = e.toLowerCase()),
      this.skipSC(),
      hasOwnProperty.call(this.atrule, e) &&
      typeof this.atrule[e].prelude == "function"
        ? (t = this.atrule[e].prelude.call(this))
        : (t = this.readSequence(this.scope.AtrulePrelude)),
      this.skipSC(),
      this.eof !== !0 &&
        this.tokenType !== 23 &&
        this.tokenType !== 17 &&
        this.error("Semicolon or block is expected"),
      { type: "AtrulePrelude", loc: this.getLocationFromList(t), children: t }
    );
  }
  function Pc(e) {
    this.children(e);
  }
  var ln = {};
  b(ln, {
    generate: () => Fc,
    name: () => Mc,
    parse: () => sn,
    structure: () => Rc,
  });
  var Ic = 36,
    xa = 42,
    tr = 61,
    Dc = 94,
    an = 124,
    Oc = 126;
  function Nc() {
    this.eof && this.error("Unexpected end of input");
    let e = this.tokenStart,
      t = !1;
    return (
      this.isDelim(xa)
        ? ((t = !0), this.next())
        : this.isDelim(an) || this.eat(1),
      this.isDelim(an)
        ? this.charCodeAt(this.tokenStart + 1) !== tr
          ? (this.next(), this.eat(1))
          : t && this.error("Identifier is expected", this.tokenEnd)
        : t && this.error("Vertical line is expected"),
      {
        type: "Identifier",
        loc: this.getLocation(e, this.tokenStart),
        name: this.substrToCursor(e),
      }
    );
  }
  function zc() {
    let e = this.tokenStart,
      t = this.charCodeAt(e);
    return (
      t !== tr &&
        t !== Oc &&
        t !== Dc &&
        t !== Ic &&
        t !== xa &&
        t !== an &&
        this.error("Attribute selector (=, ~=, ^=, $=, *=, |=) is expected"),
      this.next(),
      t !== tr &&
        (this.isDelim(tr) || this.error("Equal sign is expected"), this.next()),
      this.substrToCursor(e)
    );
  }
  var Mc = "AttributeSelector",
    Rc = {
      name: "Identifier",
      matcher: [String, null],
      value: ["String", "Identifier", null],
      flags: [String, null],
    };
  function sn() {
    let e = this.tokenStart,
      t,
      r = null,
      n = null,
      o = null;
    return (
      this.eat(19),
      this.skipSC(),
      (t = Nc.call(this)),
      this.skipSC(),
      this.tokenType !== 20 &&
        (this.tokenType !== 1 &&
          ((r = zc.call(this)),
          this.skipSC(),
          (n = this.tokenType === 5 ? this.String() : this.Identifier()),
          this.skipSC()),
        this.tokenType === 1 && ((o = this.consume(1)), this.skipSC())),
      this.eat(20),
      {
        type: "AttributeSelector",
        loc: this.getLocation(e, this.tokenStart),
        name: t,
        matcher: r,
        value: n,
        flags: o,
      }
    );
  }
  function Fc(e) {
    this.token(9, "["),
      this.node(e.name),
      e.matcher !== null && (this.tokenize(e.matcher), this.node(e.value)),
      e.flags !== null && this.token(1, e.flags),
      this.token(9, "]");
  }
  var un = {};
  b(un, {
    generate: () => Wc,
    name: () => qc,
    parse: () => cn,
    structure: () => jc,
    walkContext: () => Uc,
  });
  var Bc = 38;
  function wa(e) {
    return this.Raw(e, null, !0);
  }
  function ya() {
    return this.parseWithFallback(this.Rule, wa);
  }
  function ka(e) {
    return this.Raw(e, this.consumeUntilSemicolonIncluded, !0);
  }
  function _c() {
    if (this.tokenType === 17) return ka.call(this, this.tokenIndex);
    let e = this.parseWithFallback(this.Declaration, ka);
    return this.tokenType === 17 && this.next(), e;
  }
  var qc = "Block",
    Uc = "block",
    jc = { children: [["Atrule", "Rule", "Declaration"]] };
  function cn(e) {
    let t = e ? _c : ya,
      r = this.tokenStart,
      n = this.createList();
    this.eat(23);
    e: for (; !this.eof; )
      switch (this.tokenType) {
        case 24:
          break e;
        case 13:
        case 25:
          this.next();
          break;
        case 3:
          n.push(this.parseWithFallback(this.Atrule.bind(this, e), wa));
          break;
        default:
          e && this.isDelim(Bc) ? n.push(ya.call(this)) : n.push(t.call(this));
      }
    return (
      this.eof || this.eat(24),
      { type: "Block", loc: this.getLocation(r, this.tokenStart), children: n }
    );
  }
  function Wc(e) {
    this.token(23, "{"),
      this.children(e, (t) => {
        t.type === "Declaration" && this.token(17, ";");
      }),
      this.token(24, "}");
  }
  var hn = {};
  b(hn, {
    generate: () => Gc,
    name: () => Hc,
    parse: () => pn,
    structure: () => Yc,
  });
  var Hc = "Brackets",
    Yc = { children: [[]] };
  function pn(e, t) {
    let r = this.tokenStart,
      n = null;
    return (
      this.eat(19),
      (n = e.call(this, t)),
      this.eof || this.eat(20),
      {
        type: "Brackets",
        loc: this.getLocation(r, this.tokenStart),
        children: n,
      }
    );
  }
  function Gc(e) {
    this.token(9, "["), this.children(e), this.token(9, "]");
  }
  var fn = {};
  b(fn, {
    generate: () => Qc,
    name: () => Vc,
    parse: () => mn,
    structure: () => Kc,
  });
  var Vc = "CDC",
    Kc = [];
  function mn() {
    let e = this.tokenStart;
    return (
      this.eat(15), { type: "CDC", loc: this.getLocation(e, this.tokenStart) }
    );
  }
  function Qc() {
    this.token(15, "-->");
  }
  var gn = {};
  b(gn, {
    generate: () => Zc,
    name: () => Xc,
    parse: () => dn,
    structure: () => $c,
  });
  var Xc = "CDO",
    $c = [];
  function dn() {
    let e = this.tokenStart;
    return (
      this.eat(14), { type: "CDO", loc: this.getLocation(e, this.tokenStart) }
    );
  }
  function Zc() {
    this.token(14, "<!--");
  }
  var xn = {};
  b(xn, {
    generate: () => ru,
    name: () => eu,
    parse: () => bn,
    structure: () => tu,
  });
  var Jc = 46,
    eu = "ClassSelector",
    tu = { name: String };
  function bn() {
    return (
      this.eatDelim(Jc),
      {
        type: "ClassSelector",
        loc: this.getLocation(this.tokenStart - 1, this.tokenEnd),
        name: this.consume(1),
      }
    );
  }
  function ru(e) {
    this.token(9, "."), this.token(1, e.name);
  }
  var kn = {};
  b(kn, {
    generate: () => lu,
    name: () => au,
    parse: () => yn,
    structure: () => su,
  });
  var nu = 43,
    va = 47,
    ou = 62,
    iu = 126,
    au = "Combinator",
    su = { name: String };
  function yn() {
    let e = this.tokenStart,
      t;
    switch (this.tokenType) {
      case 13:
        t = " ";
        break;
      case 9:
        switch (this.charCodeAt(this.tokenStart)) {
          case ou:
          case nu:
          case iu:
            this.next();
            break;
          case va:
            this.next(), this.eatIdent("deep"), this.eatDelim(va);
            break;
          default:
            this.error("Combinator is expected");
        }
        t = this.substrToCursor(e);
        break;
    }
    return {
      type: "Combinator",
      loc: this.getLocation(e, this.tokenStart),
      name: t,
    };
  }
  function lu(e) {
    this.tokenize(e.name);
  }
  var vn = {};
  b(vn, {
    generate: () => mu,
    name: () => pu,
    parse: () => wn,
    structure: () => hu,
  });
  var cu = 42,
    uu = 47,
    pu = "Comment",
    hu = { value: String };
  function wn() {
    let e = this.tokenStart,
      t = this.tokenEnd;
    return (
      this.eat(25),
      t - e + 2 >= 2 &&
        this.charCodeAt(t - 2) === cu &&
        this.charCodeAt(t - 1) === uu &&
        (t -= 2),
      {
        type: "Comment",
        loc: this.getLocation(e, this.tokenStart),
        value: this.substring(e + 2, t),
      }
    );
  }
  function mu(e) {
    this.token(25, "/*" + e.value + "*/");
  }
  var Cn = {};
  b(Cn, {
    generate: () => Au,
    name: () => vu,
    parse: () => Sn,
    structure: () => Cu,
    walkContext: () => Su,
  });
  var Ca = 33,
    fu = 35,
    du = 36,
    gu = 38,
    bu = 42,
    xu = 43,
    Sa = 47;
  function yu(e) {
    return this.Raw(e, this.consumeUntilExclamationMarkOrSemicolon, !0);
  }
  function ku(e) {
    return this.Raw(e, this.consumeUntilExclamationMarkOrSemicolon, !1);
  }
  function wu() {
    let e = this.tokenIndex,
      t = this.Value();
    return (
      t.type !== "Raw" &&
        this.eof === !1 &&
        this.tokenType !== 17 &&
        this.isDelim(Ca) === !1 &&
        this.isBalanceEdge(e) === !1 &&
        this.error(),
      t
    );
  }
  var vu = "Declaration",
    Su = "declaration",
    Cu = {
      important: [Boolean, String],
      property: String,
      value: ["Value", "Raw"],
    };
  function Sn() {
    let e = this.tokenStart,
      t = this.tokenIndex,
      r = Tu.call(this),
      n = lt(r),
      o = n ? this.parseCustomProperty : this.parseValue,
      i = n ? ku : yu,
      s = !1,
      u;
    this.skipSC(), this.eat(16);
    let c = this.tokenIndex;
    if (
      (n || this.skipSC(),
      o
        ? (u = this.parseWithFallback(wu, i))
        : (u = i.call(this, this.tokenIndex)),
      n && u.type === "Value" && u.children.isEmpty)
    ) {
      for (let a = c - this.tokenIndex; a <= 0; a++)
        if (this.lookupType(a) === 13) {
          u.children.appendData({ type: "WhiteSpace", loc: null, value: " " });
          break;
        }
    }
    return (
      this.isDelim(Ca) && ((s = Eu.call(this)), this.skipSC()),
      this.eof === !1 &&
        this.tokenType !== 17 &&
        this.isBalanceEdge(t) === !1 &&
        this.error(),
      {
        type: "Declaration",
        loc: this.getLocation(e, this.tokenStart),
        important: s,
        property: r,
        value: u,
      }
    );
  }
  function Au(e) {
    this.token(1, e.property),
      this.token(16, ":"),
      this.node(e.value),
      e.important &&
        (this.token(9, "!"),
        this.token(1, e.important === !0 ? "important" : e.important));
  }
  function Tu() {
    let e = this.tokenStart;
    if (this.tokenType === 9)
      switch (this.charCodeAt(this.tokenStart)) {
        case bu:
        case du:
        case xu:
        case fu:
        case gu:
          this.next();
          break;
        case Sa:
          this.next(), this.isDelim(Sa) && this.next();
          break;
      }
    return (
      this.tokenType === 4 ? this.eat(4) : this.eat(1), this.substrToCursor(e)
    );
  }
  function Eu() {
    this.eat(9), this.skipSC();
    let e = this.consume(1);
    return e === "important" ? !0 : e;
  }
  var En = {};
  b(En, {
    generate: () => Du,
    name: () => Pu,
    parse: () => Tn,
    structure: () => Iu,
  });
  var Lu = 38;
  function An(e) {
    return this.Raw(e, this.consumeUntilSemicolonIncluded, !0);
  }
  var Pu = "DeclarationList",
    Iu = { children: [["Declaration", "Atrule", "Rule"]] };
  function Tn() {
    let e = this.createList();
    e: for (; !this.eof; )
      switch (this.tokenType) {
        case 13:
        case 25:
        case 17:
          this.next();
          break;
        case 3:
          e.push(this.parseWithFallback(this.Atrule.bind(this, !0), An));
          break;
        default:
          this.isDelim(Lu)
            ? e.push(this.parseWithFallback(this.Rule, An))
            : e.push(this.parseWithFallback(this.Declaration, An));
      }
    return {
      type: "DeclarationList",
      loc: this.getLocationFromList(e),
      children: e,
    };
  }
  function Du(e) {
    this.children(e, (t) => {
      t.type === "Declaration" && this.token(17, ";");
    });
  }
  var Pn = {};
  b(Pn, {
    generate: () => zu,
    name: () => Ou,
    parse: () => Ln,
    structure: () => Nu,
  });
  var Ou = "Dimension",
    Nu = { value: String, unit: String };
  function Ln() {
    let e = this.tokenStart,
      t = this.consumeNumber(12);
    return {
      type: "Dimension",
      loc: this.getLocation(e, this.tokenStart),
      value: t,
      unit: this.substring(e + t.length, this.tokenStart),
    };
  }
  function zu(e) {
    this.token(12, e.value + e.unit);
  }
  var Dn = {};
  b(Dn, {
    generate: () => Bu,
    name: () => Mu,
    parse: () => In,
    structure: () => Fu,
    walkContext: () => Ru,
  });
  var Mu = "Function",
    Ru = "function",
    Fu = { name: String, children: [[]] };
  function In(e, t) {
    let r = this.tokenStart,
      n = this.consumeFunctionName(),
      o = n.toLowerCase(),
      i;
    return (
      (i = t.hasOwnProperty(o) ? t[o].call(this, t) : e.call(this, t)),
      this.eof || this.eat(22),
      {
        type: "Function",
        loc: this.getLocation(r, this.tokenStart),
        name: n,
        children: i,
      }
    );
  }
  function Bu(e) {
    this.token(2, e.name + "("), this.children(e), this.token(22, ")");
  }
  var Nn = {};
  b(Nn, {
    generate: () => ju,
    name: () => qu,
    parse: () => On,
    structure: () => Uu,
    xxx: () => _u,
  });
  var _u = "XXX",
    qu = "Hash",
    Uu = { value: String };
  function On() {
    let e = this.tokenStart;
    return (
      this.eat(4),
      {
        type: "Hash",
        loc: this.getLocation(e, this.tokenStart),
        value: this.substrToCursor(e + 1),
      }
    );
  }
  function ju(e) {
    this.token(4, "#" + e.value);
  }
  var Mn = {};
  b(Mn, {
    generate: () => Yu,
    name: () => Wu,
    parse: () => zn,
    structure: () => Hu,
  });
  var Wu = "Identifier",
    Hu = { name: String };
  function zn() {
    return {
      type: "Identifier",
      loc: this.getLocation(this.tokenStart, this.tokenEnd),
      name: this.consume(1),
    };
  }
  function Yu(e) {
    this.token(1, e.name);
  }
  var Fn = {};
  b(Fn, {
    generate: () => Ku,
    name: () => Gu,
    parse: () => Rn,
    structure: () => Vu,
  });
  var Gu = "IdSelector",
    Vu = { name: String };
  function Rn() {
    let e = this.tokenStart;
    return (
      this.eat(4),
      {
        type: "IdSelector",
        loc: this.getLocation(e, this.tokenStart),
        name: this.substrToCursor(e + 1),
      }
    );
  }
  function Ku(e) {
    this.token(9, "#" + e.name);
  }
  var _n = {};
  b(_n, {
    generate: () => $u,
    name: () => Qu,
    parse: () => Bn,
    structure: () => Xu,
  });
  var Qu = "MediaFeature",
    Xu = {
      name: String,
      value: ["Identifier", "Number", "Dimension", "Ratio", null],
    };
  function Bn() {
    let e = this.tokenStart,
      t,
      r = null;
    if (
      (this.eat(21),
      this.skipSC(),
      (t = this.consume(1)),
      this.skipSC(),
      this.tokenType !== 22)
    ) {
      switch ((this.eat(16), this.skipSC(), this.tokenType)) {
        case 10:
          this.lookupNonWSType(1) === 9
            ? (r = this.Ratio())
            : (r = this.Number());
          break;
        case 12:
          r = this.Dimension();
          break;
        case 1:
          r = this.Identifier();
          break;
        default:
          this.error("Number, dimension, ratio or identifier is expected");
      }
      this.skipSC();
    }
    return (
      this.eat(22),
      {
        type: "MediaFeature",
        loc: this.getLocation(e, this.tokenStart),
        name: t,
        value: r,
      }
    );
  }
  function $u(e) {
    this.token(21, "("),
      this.token(1, e.name),
      e.value !== null && (this.token(16, ":"), this.node(e.value)),
      this.token(22, ")");
  }
  var Un = {};
  b(Un, {
    generate: () => ep,
    name: () => Zu,
    parse: () => qn,
    structure: () => Ju,
  });
  var Zu = "MediaQuery",
    Ju = { children: [["Identifier", "MediaFeature", "WhiteSpace"]] };
  function qn() {
    let e = this.createList(),
      t = null;
    this.skipSC();
    e: for (; !this.eof; ) {
      switch (this.tokenType) {
        case 25:
        case 13:
          this.next();
          continue;
        case 1:
          t = this.Identifier();
          break;
        case 21:
          t = this.MediaFeature();
          break;
        default:
          break e;
      }
      e.push(t);
    }
    return (
      t === null && this.error("Identifier or parenthesis is expected"),
      { type: "MediaQuery", loc: this.getLocationFromList(e), children: e }
    );
  }
  function ep(e) {
    this.children(e);
  }
  var Wn = {};
  b(Wn, {
    generate: () => np,
    name: () => tp,
    parse: () => jn,
    structure: () => rp,
  });
  var tp = "MediaQueryList",
    rp = { children: [["MediaQuery"]] };
  function jn() {
    let e = this.createList();
    for (
      this.skipSC();
      !this.eof && (e.push(this.MediaQuery()), this.tokenType === 18);

    )
      this.next();
    return {
      type: "MediaQueryList",
      loc: this.getLocationFromList(e),
      children: e,
    };
  }
  function np(e) {
    this.children(e, () => this.token(18, ","));
  }
  var Yn = {};
  b(Yn, {
    generate: () => sp,
    name: () => ip,
    parse: () => Hn,
    structure: () => ap,
  });
  var op = 38,
    ip = "NestingSelector",
    ap = {};
  function Hn() {
    let e = this.tokenStart;
    return (
      this.eatDelim(op),
      { type: "NestingSelector", loc: this.getLocation(e, this.tokenStart) }
    );
  }
  function sp() {
    this.token(9, "&");
  }
  var Vn = {};
  b(Vn, {
    generate: () => up,
    name: () => lp,
    parse: () => Gn,
    structure: () => cp,
  });
  var lp = "Nth",
    cp = { nth: ["AnPlusB", "Identifier"], selector: ["SelectorList", null] };
  function Gn() {
    this.skipSC();
    let e = this.tokenStart,
      t = e,
      r = null,
      n;
    return (
      this.lookupValue(0, "odd") || this.lookupValue(0, "even")
        ? (n = this.Identifier())
        : (n = this.AnPlusB()),
      (t = this.tokenStart),
      this.skipSC(),
      this.lookupValue(0, "of") &&
        (this.next(), (r = this.SelectorList()), (t = this.tokenStart)),
      { type: "Nth", loc: this.getLocation(e, t), nth: n, selector: r }
    );
  }
  function up(e) {
    this.node(e.nth),
      e.selector !== null && (this.token(1, "of"), this.node(e.selector));
  }
  var Qn = {};
  b(Qn, {
    generate: () => mp,
    name: () => pp,
    parse: () => Kn,
    structure: () => hp,
  });
  var pp = "Number",
    hp = { value: String };
  function Kn() {
    return {
      type: "Number",
      loc: this.getLocation(this.tokenStart, this.tokenEnd),
      value: this.consume(10),
    };
  }
  function mp(e) {
    this.token(10, e.value);
  }
  var $n = {};
  b($n, {
    generate: () => gp,
    name: () => fp,
    parse: () => Xn,
    structure: () => dp,
  });
  var fp = "Operator",
    dp = { value: String };
  function Xn() {
    let e = this.tokenStart;
    return (
      this.next(),
      {
        type: "Operator",
        loc: this.getLocation(e, this.tokenStart),
        value: this.substrToCursor(e),
      }
    );
  }
  function gp(e) {
    this.tokenize(e.value);
  }
  var Jn = {};
  b(Jn, {
    generate: () => yp,
    name: () => bp,
    parse: () => Zn,
    structure: () => xp,
  });
  var bp = "Parentheses",
    xp = { children: [[]] };
  function Zn(e, t) {
    let r = this.tokenStart,
      n = null;
    return (
      this.eat(21),
      (n = e.call(this, t)),
      this.eof || this.eat(22),
      {
        type: "Parentheses",
        loc: this.getLocation(r, this.tokenStart),
        children: n,
      }
    );
  }
  function yp(e) {
    this.token(21, "("), this.children(e), this.token(22, ")");
  }
  var to = {};
  b(to, {
    generate: () => vp,
    name: () => kp,
    parse: () => eo,
    structure: () => wp,
  });
  var kp = "Percentage",
    wp = { value: String };
  function eo() {
    return {
      type: "Percentage",
      loc: this.getLocation(this.tokenStart, this.tokenEnd),
      value: this.consumeNumber(11),
    };
  }
  function vp(e) {
    this.token(11, e.value + "%");
  }
  var no = {};
  b(no, {
    generate: () => Tp,
    name: () => Sp,
    parse: () => ro,
    structure: () => Ap,
    walkContext: () => Cp,
  });
  var Sp = "PseudoClassSelector",
    Cp = "function",
    Ap = { name: String, children: [["Raw"], null] };
  function ro() {
    let e = this.tokenStart,
      t = null,
      r,
      n;
    return (
      this.eat(16),
      this.tokenType === 2
        ? ((r = this.consumeFunctionName()),
          (n = r.toLowerCase()),
          hasOwnProperty.call(this.pseudo, n)
            ? (this.skipSC(), (t = this.pseudo[n].call(this)), this.skipSC())
            : ((t = this.createList()),
              t.push(this.Raw(this.tokenIndex, null, !1))),
          this.eat(22))
        : (r = this.consume(1)),
      {
        type: "PseudoClassSelector",
        loc: this.getLocation(e, this.tokenStart),
        name: r,
        children: t,
      }
    );
  }
  function Tp(e) {
    this.token(16, ":"),
      e.children === null
        ? this.token(1, e.name)
        : (this.token(2, e.name + "("), this.children(e), this.token(22, ")"));
  }
  var io = {};
  b(io, {
    generate: () => Ip,
    name: () => Ep,
    parse: () => oo,
    structure: () => Pp,
    walkContext: () => Lp,
  });
  var Ep = "PseudoElementSelector",
    Lp = "function",
    Pp = { name: String, children: [["Raw"], null] };
  function oo() {
    let e = this.tokenStart,
      t = null,
      r,
      n;
    return (
      this.eat(16),
      this.eat(16),
      this.tokenType === 2
        ? ((r = this.consumeFunctionName()),
          (n = r.toLowerCase()),
          hasOwnProperty.call(this.pseudo, n)
            ? (this.skipSC(), (t = this.pseudo[n].call(this)), this.skipSC())
            : ((t = this.createList()),
              t.push(this.Raw(this.tokenIndex, null, !1))),
          this.eat(22))
        : (r = this.consume(1)),
      {
        type: "PseudoElementSelector",
        loc: this.getLocation(e, this.tokenStart),
        name: r,
        children: t,
      }
    );
  }
  function Ip(e) {
    this.token(16, ":"),
      this.token(16, ":"),
      e.children === null
        ? this.token(1, e.name)
        : (this.token(2, e.name + "("), this.children(e), this.token(22, ")"));
  }
  var so = {};
  b(so, {
    generate: () => Mp,
    name: () => Np,
    parse: () => ao,
    structure: () => zp,
  });
  var Dp = 47,
    Op = 46;
  function Aa() {
    this.skipSC();
    let e = this.consume(10);
    for (let t = 0; t < e.length; t++) {
      let r = e.charCodeAt(t);
      !B(r) &&
        r !== Op &&
        this.error(
          "Unsigned number is expected",
          this.tokenStart - e.length + t
        );
    }
    return (
      Number(e) === 0 &&
        this.error("Zero number is not allowed", this.tokenStart - e.length),
      e
    );
  }
  var Np = "Ratio",
    zp = { left: String, right: String };
  function ao() {
    let e = this.tokenStart,
      t = Aa.call(this),
      r;
    return (
      this.skipSC(),
      this.eatDelim(Dp),
      (r = Aa.call(this)),
      {
        type: "Ratio",
        loc: this.getLocation(e, this.tokenStart),
        left: t,
        right: r,
      }
    );
  }
  function Mp(e) {
    this.token(10, e.left), this.token(9, "/"), this.token(10, e.right);
  }
  var co = {};
  b(co, {
    generate: () => _p,
    name: () => Fp,
    parse: () => lo,
    structure: () => Bp,
  });
  function Rp() {
    return this.tokenIndex > 0 && this.lookupType(-1) === 13
      ? this.tokenIndex > 1
        ? this.getTokenStart(this.tokenIndex - 1)
        : this.firstCharOffset
      : this.tokenStart;
  }
  var Fp = "Raw",
    Bp = { value: String };
  function lo(e, t, r) {
    let n = this.getTokenStart(e),
      o;
    return (
      this.skipUntilBalanced(e, t || this.consumeUntilBalanceEnd),
      r && this.tokenStart > n ? (o = Rp.call(this)) : (o = this.tokenStart),
      { type: "Raw", loc: this.getLocation(n, o), value: this.substring(n, o) }
    );
  }
  function _p(e) {
    this.tokenize(e.value);
  }
  var po = {};
  b(po, {
    generate: () => Hp,
    name: () => Up,
    parse: () => uo,
    structure: () => Wp,
    walkContext: () => jp,
  });
  function Ta(e) {
    return this.Raw(e, this.consumeUntilLeftCurlyBracket, !0);
  }
  function qp() {
    let e = this.SelectorList();
    return (
      e.type !== "Raw" &&
        this.eof === !1 &&
        this.tokenType !== 23 &&
        this.error(),
      e
    );
  }
  var Up = "Rule",
    jp = "rule",
    Wp = { prelude: ["SelectorList", "Raw"], block: ["Block"] };
  function uo() {
    let e = this.tokenIndex,
      t = this.tokenStart,
      r,
      n;
    return (
      this.parseRulePrelude
        ? (r = this.parseWithFallback(qp, Ta))
        : (r = Ta.call(this, e)),
      (n = this.Block(!0)),
      {
        type: "Rule",
        loc: this.getLocation(t, this.tokenStart),
        prelude: r,
        block: n,
      }
    );
  }
  function Hp(e) {
    this.node(e.prelude), this.node(e.block);
  }
  var mo = {};
  b(mo, {
    generate: () => Vp,
    name: () => Yp,
    parse: () => ho,
    structure: () => Gp,
  });
  var Yp = "Selector",
    Gp = {
      children: [
        [
          "TypeSelector",
          "IdSelector",
          "ClassSelector",
          "AttributeSelector",
          "PseudoClassSelector",
          "PseudoElementSelector",
          "Combinator",
          "WhiteSpace",
        ],
      ],
    };
  function ho() {
    let e = this.readSequence(this.scope.Selector);
    return (
      this.getFirstListNode(e) === null && this.error("Selector is expected"),
      { type: "Selector", loc: this.getLocationFromList(e), children: e }
    );
  }
  function Vp(e) {
    this.children(e);
  }
  var go = {};
  b(go, {
    generate: () => $p,
    name: () => Kp,
    parse: () => fo,
    structure: () => Xp,
    walkContext: () => Qp,
  });
  var Kp = "SelectorList",
    Qp = "selector",
    Xp = { children: [["Selector", "Raw"]] };
  function fo() {
    let e = this.createList();
    for (; !this.eof; ) {
      if ((e.push(this.Selector()), this.tokenType === 18)) {
        this.next();
        continue;
      }
      break;
    }
    return {
      type: "SelectorList",
      loc: this.getLocationFromList(e),
      children: e,
    };
  }
  function $p(e) {
    this.children(e, () => this.token(18, ","));
  }
  var ko = {};
  b(ko, {
    generate: () => eh,
    name: () => Zp,
    parse: () => yo,
    structure: () => Jp,
  });
  var rr = {};
  b(rr, { decode: () => gt, encode: () => xo });
  var bo = 92,
    Ea = 34,
    La = 39;
  function gt(e) {
    let t = e.length,
      r = e.charCodeAt(0),
      n = r === Ea || r === La ? 1 : 0,
      o = n === 1 && t > 1 && e.charCodeAt(t - 1) === r ? t - 2 : t - 1,
      i = "";
    for (let s = n; s <= o; s++) {
      let u = e.charCodeAt(s);
      if (u === bo) {
        if (s === o) {
          s !== t - 1 && (i = e.substr(s + 1));
          break;
        }
        if (((u = e.charCodeAt(++s)), $(bo, u))) {
          let c = s - 1,
            a = se(e, c);
          (s = a - 1), (i += _e(e.substring(c + 1, a)));
        } else u === 13 && e.charCodeAt(s + 1) === 10 && s++;
      } else i += e[s];
    }
    return i;
  }
  function xo(e, t) {
    let r = t ? "'" : '"',
      n = t ? La : Ea,
      o = "",
      i = !1;
    for (let s = 0; s < e.length; s++) {
      let u = e.charCodeAt(s);
      if (u === 0) {
        o += "\uFFFD";
        continue;
      }
      if (u <= 31 || u === 127) {
        (o += "\\" + u.toString(16)), (i = !0);
        continue;
      }
      u === n || u === bo
        ? ((o += "\\" + e.charAt(s)), (i = !1))
        : (i && (ee(u) || pe(u)) && (o += " "), (o += e.charAt(s)), (i = !1));
    }
    return r + o + r;
  }
  var Zp = "String",
    Jp = { value: String };
  function yo() {
    return {
      type: "String",
      loc: this.getLocation(this.tokenStart, this.tokenEnd),
      value: gt(this.consume(5)),
    };
  }
  function eh(e) {
    this.token(5, xo(e.value));
  }
  var vo = {};
  b(vo, {
    generate: () => ih,
    name: () => rh,
    parse: () => wo,
    structure: () => oh,
    walkContext: () => nh,
  });
  var th = 33;
  function Pa(e) {
    return this.Raw(e, null, !1);
  }
  var rh = "StyleSheet",
    nh = "stylesheet",
    oh = { children: [["Comment", "CDO", "CDC", "Atrule", "Rule", "Raw"]] };
  function wo() {
    let e = this.tokenStart,
      t = this.createList(),
      r;
    e: for (; !this.eof; ) {
      switch (this.tokenType) {
        case 13:
          this.next();
          continue;
        case 25:
          if (this.charCodeAt(this.tokenStart + 2) !== th) {
            this.next();
            continue;
          }
          r = this.Comment();
          break;
        case 14:
          r = this.CDO();
          break;
        case 15:
          r = this.CDC();
          break;
        case 3:
          r = this.parseWithFallback(this.Atrule, Pa);
          break;
        default:
          r = this.parseWithFallback(this.Rule, Pa);
      }
      t.push(r);
    }
    return {
      type: "StyleSheet",
      loc: this.getLocation(e, this.tokenStart),
      children: t,
    };
  }
  function ih(e) {
    this.children(e);
  }
  var Ao = {};
  b(Ao, {
    generate: () => ch,
    name: () => sh,
    parse: () => Co,
    structure: () => lh,
  });
  var ah = 42,
    Ia = 124;
  function So() {
    this.tokenType !== 1 &&
      this.isDelim(ah) === !1 &&
      this.error("Identifier or asterisk is expected"),
      this.next();
  }
  var sh = "TypeSelector",
    lh = { name: String };
  function Co() {
    let e = this.tokenStart;
    return (
      this.isDelim(Ia)
        ? (this.next(), So.call(this))
        : (So.call(this), this.isDelim(Ia) && (this.next(), So.call(this))),
      {
        type: "TypeSelector",
        loc: this.getLocation(e, this.tokenStart),
        name: this.substrToCursor(e),
      }
    );
  }
  function ch(e) {
    this.tokenize(e.name);
  }
  var Lo = {};
  b(Lo, {
    generate: () => fh,
    name: () => hh,
    parse: () => Eo,
    structure: () => mh,
  });
  var Da = 43,
    Oa = 45,
    To = 63;
  function bt(e, t) {
    let r = 0;
    for (let n = this.tokenStart + e; n < this.tokenEnd; n++) {
      let o = this.charCodeAt(n);
      if (o === Oa && t && r !== 0) return bt.call(this, e + r + 1, !1), -1;
      ee(o) ||
        this.error(
          t && r !== 0
            ? "Hyphen minus" + (r < 6 ? " or hex digit" : "") + " is expected"
            : r < 6
            ? "Hex digit is expected"
            : "Unexpected input",
          n
        ),
        ++r > 6 && this.error("Too many hex digits", n);
    }
    return this.next(), r;
  }
  function nr(e) {
    let t = 0;
    for (; this.isDelim(To); )
      ++t > e && this.error("Too many question marks"), this.next();
  }
  function uh(e) {
    this.charCodeAt(this.tokenStart) !== e &&
      this.error((e === Da ? "Plus sign" : "Hyphen minus") + " is expected");
  }
  function ph() {
    let e = 0;
    switch (this.tokenType) {
      case 10:
        if (((e = bt.call(this, 1, !0)), this.isDelim(To))) {
          nr.call(this, 6 - e);
          break;
        }
        if (this.tokenType === 12 || this.tokenType === 10) {
          uh.call(this, Oa), bt.call(this, 1, !1);
          break;
        }
        break;
      case 12:
        (e = bt.call(this, 1, !0)), e > 0 && nr.call(this, 6 - e);
        break;
      default:
        if ((this.eatDelim(Da), this.tokenType === 1)) {
          (e = bt.call(this, 0, !0)), e > 0 && nr.call(this, 6 - e);
          break;
        }
        if (this.isDelim(To)) {
          this.next(), nr.call(this, 5);
          break;
        }
        this.error("Hex digit or question mark is expected");
    }
  }
  var hh = "UnicodeRange",
    mh = { value: String };
  function Eo() {
    let e = this.tokenStart;
    return (
      this.eatIdent("u"),
      ph.call(this),
      {
        type: "UnicodeRange",
        loc: this.getLocation(e, this.tokenStart),
        value: this.substrToCursor(e),
      }
    );
  }
  function fh(e) {
    this.tokenize(e.value);
  }
  var No = {};
  b(No, {
    generate: () => wh,
    name: () => yh,
    parse: () => Oo,
    structure: () => kh,
  });
  var or = {};
  b(or, { decode: () => Io, encode: () => Do });
  var dh = 32,
    Po = 92,
    gh = 34,
    bh = 39,
    xh = 40,
    Na = 41;
  function Io(e) {
    let t = e.length,
      r = 4,
      n = e.charCodeAt(t - 1) === Na ? t - 2 : t - 1,
      o = "";
    for (; r < n && pe(e.charCodeAt(r)); ) r++;
    for (; r < n && pe(e.charCodeAt(n)); ) n--;
    for (let i = r; i <= n; i++) {
      let s = e.charCodeAt(i);
      if (s === Po) {
        if (i === n) {
          i !== t - 1 && (o = e.substr(i + 1));
          break;
        }
        if (((s = e.charCodeAt(++i)), $(Po, s))) {
          let u = i - 1,
            c = se(e, u);
          (i = c - 1), (o += _e(e.substring(u + 1, c)));
        } else s === 13 && e.charCodeAt(i + 1) === 10 && i++;
      } else o += e[i];
    }
    return o;
  }
  function Do(e) {
    let t = "",
      r = !1;
    for (let n = 0; n < e.length; n++) {
      let o = e.charCodeAt(n);
      if (o === 0) {
        t += "\uFFFD";
        continue;
      }
      if (o <= 31 || o === 127) {
        (t += "\\" + o.toString(16)), (r = !0);
        continue;
      }
      o === dh || o === Po || o === gh || o === bh || o === xh || o === Na
        ? ((t += "\\" + e.charAt(n)), (r = !1))
        : (r && ee(o) && (t += " "), (t += e.charAt(n)), (r = !1));
    }
    return "url(" + t + ")";
  }
  var yh = "Url",
    kh = { value: String };
  function Oo() {
    let e = this.tokenStart,
      t;
    switch (this.tokenType) {
      case 7:
        t = Io(this.consume(7));
        break;
      case 2:
        this.cmpStr(this.tokenStart, this.tokenEnd, "url(") ||
          this.error("Function name must be `url`"),
          this.eat(2),
          this.skipSC(),
          (t = gt(this.consume(5))),
          this.skipSC(),
          this.eof || this.eat(22);
        break;
      default:
        this.error("Url or Function is expected");
    }
    return { type: "Url", loc: this.getLocation(e, this.tokenStart), value: t };
  }
  function wh(e) {
    this.token(7, Do(e.value));
  }
  var Mo = {};
  b(Mo, {
    generate: () => Ch,
    name: () => vh,
    parse: () => zo,
    structure: () => Sh,
  });
  var vh = "Value",
    Sh = { children: [[]] };
  function zo() {
    let e = this.tokenStart,
      t = this.readSequence(this.scope.Value);
    return {
      type: "Value",
      loc: this.getLocation(e, this.tokenStart),
      children: t,
    };
  }
  function Ch(e) {
    this.children(e);
  }
  var Fo = {};
  b(Fo, {
    generate: () => Lh,
    name: () => Th,
    parse: () => Ro,
    structure: () => Eh,
  });
  var Ah = Object.freeze({ type: "WhiteSpace", loc: null, value: " " }),
    Th = "WhiteSpace",
    Eh = { value: String };
  function Ro() {
    return this.eat(13), Ah;
  }
  function Lh(e) {
    this.token(13, e.value);
  }
  var za = { generic: !0, ...ga, node: xt };
  var Bo = {};
  b(Bo, { AtrulePrelude: () => Ra, Selector: () => Ba, Value: () => ja });
  var Ph = 35,
    Ih = 42,
    Ma = 43,
    Dh = 45,
    Oh = 47,
    Nh = 117;
  function yt(e) {
    switch (this.tokenType) {
      case 4:
        return this.Hash();
      case 18:
        return this.Operator();
      case 21:
        return this.Parentheses(this.readSequence, e.recognizer);
      case 19:
        return this.Brackets(this.readSequence, e.recognizer);
      case 5:
        return this.String();
      case 12:
        return this.Dimension();
      case 11:
        return this.Percentage();
      case 10:
        return this.Number();
      case 2:
        return this.cmpStr(this.tokenStart, this.tokenEnd, "url(")
          ? this.Url()
          : this.Function(this.readSequence, e.recognizer);
      case 7:
        return this.Url();
      case 1:
        return this.cmpChar(this.tokenStart, Nh) &&
          this.cmpChar(this.tokenStart + 1, Ma)
          ? this.UnicodeRange()
          : this.Identifier();
      case 9: {
        let t = this.charCodeAt(this.tokenStart);
        if (t === Oh || t === Ih || t === Ma || t === Dh)
          return this.Operator();
        t === Ph &&
          this.error("Hex or identifier is expected", this.tokenStart + 1);
        break;
      }
    }
  }
  var Ra = { getNode: yt };
  var zh = 35,
    Mh = 38,
    Rh = 42,
    Fh = 43,
    Bh = 47,
    Fa = 46,
    _h = 62,
    qh = 124,
    Uh = 126;
  function jh(e, t) {
    t.last !== null &&
      t.last.type !== "Combinator" &&
      e !== null &&
      e.type !== "Combinator" &&
      t.push({ type: "Combinator", loc: null, name: " " });
  }
  function Wh() {
    switch (this.tokenType) {
      case 19:
        return this.AttributeSelector();
      case 4:
        return this.IdSelector();
      case 16:
        return this.lookupType(1) === 16
          ? this.PseudoElementSelector()
          : this.PseudoClassSelector();
      case 1:
        return this.TypeSelector();
      case 10:
      case 11:
        return this.Percentage();
      case 12:
        this.charCodeAt(this.tokenStart) === Fa &&
          this.error("Identifier is expected", this.tokenStart + 1);
        break;
      case 9: {
        switch (this.charCodeAt(this.tokenStart)) {
          case Fh:
          case _h:
          case Uh:
          case Bh:
            return this.Combinator();
          case Fa:
            return this.ClassSelector();
          case Rh:
          case qh:
            return this.TypeSelector();
          case zh:
            return this.IdSelector();
          case Mh:
            return this.NestingSelector();
        }
        break;
      }
    }
  }
  var Ba = { onWhiteSpace: jh, getNode: Wh };
  function _a() {
    return this.createSingleNodeList(this.Raw(this.tokenIndex, null, !1));
  }
  function qa() {
    let e = this.createList();
    if (
      (this.skipSC(),
      e.push(this.Identifier()),
      this.skipSC(),
      this.tokenType === 18)
    ) {
      e.push(this.Operator());
      let t = this.tokenIndex,
        r = this.parseCustomProperty
          ? this.Value(null)
          : this.Raw(
              this.tokenIndex,
              this.consumeUntilExclamationMarkOrSemicolon,
              !1
            );
      if (r.type === "Value" && r.children.isEmpty) {
        for (let n = t - this.tokenIndex; n <= 0; n++)
          if (this.lookupType(n) === 13) {
            r.children.appendData({
              type: "WhiteSpace",
              loc: null,
              value: " ",
            });
            break;
          }
      }
      e.push(r);
    }
    return e;
  }
  function Ua(e) {
    return (
      e !== null &&
      e.type === "Operator" &&
      (e.value[e.value.length - 1] === "-" ||
        e.value[e.value.length - 1] === "+")
    );
  }
  var ja = {
    getNode: yt,
    onWhiteSpace(e, t) {
      Ua(e) && (e.value = " " + e.value), Ua(t.last) && (t.last.value += " ");
    },
    expression: _a,
    var: qa,
  };
  var Wa = {
    parse: {
      prelude: null,
      block() {
        return this.Block(!0);
      },
    },
  };
  var Ha = {
    parse: {
      prelude() {
        let e = this.createList();
        switch ((this.skipSC(), this.tokenType)) {
          case 5:
            e.push(this.String());
            break;
          case 7:
          case 2:
            e.push(this.Url());
            break;
          default:
            this.error("String or url() is expected");
        }
        return (
          (this.lookupNonWSType(0) === 1 || this.lookupNonWSType(0) === 21) &&
            e.push(this.MediaQueryList()),
          e
        );
      },
      block: null,
    },
  };
  var Ya = {
    parse: {
      prelude() {
        return this.createSingleNodeList(this.MediaQueryList());
      },
      block(e = !1) {
        return this.Block(e);
      },
    },
  };
  var Ga = {
    parse: {
      prelude() {
        return this.createSingleNodeList(this.SelectorList());
      },
      block() {
        return this.Block(!0);
      },
    },
  };
  var Va = {
    parse: {
      prelude() {
        return this.createSingleNodeList(this.SelectorList());
      },
      block() {
        return this.Block(!0);
      },
    },
  };
  function Hh() {
    return this.createSingleNodeList(this.Raw(this.tokenIndex, null, !1));
  }
  function Yh() {
    return (
      this.skipSC(),
      this.tokenType === 1 && this.lookupNonWSType(1) === 16
        ? this.createSingleNodeList(this.Declaration())
        : Ka.call(this)
    );
  }
  function Ka() {
    let e = this.createList(),
      t;
    this.skipSC();
    e: for (; !this.eof; ) {
      switch (this.tokenType) {
        case 25:
        case 13:
          this.next();
          continue;
        case 2:
          t = this.Function(Hh, this.scope.AtrulePrelude);
          break;
        case 1:
          t = this.Identifier();
          break;
        case 21:
          t = this.Parentheses(Yh, this.scope.AtrulePrelude);
          break;
        default:
          break e;
      }
      e.push(t);
    }
    return e;
  }
  var Qa = {
    parse: {
      prelude() {
        let e = Ka.call(this);
        return (
          this.getFirstListNode(e) === null &&
            this.error("Condition is expected"),
          e
        );
      },
      block(e = !1) {
        return this.Block(e);
      },
    },
  };
  var Xa = {
    "font-face": Wa,
    import: Ha,
    media: Ya,
    nest: Ga,
    page: Va,
    supports: Qa,
  };
  var Ne = {
      parse() {
        return this.createSingleNodeList(this.SelectorList());
      },
    },
    _o = {
      parse() {
        return this.createSingleNodeList(this.Selector());
      },
    },
    $a = {
      parse() {
        return this.createSingleNodeList(this.Identifier());
      },
    },
    ir = {
      parse() {
        return this.createSingleNodeList(this.Nth());
      },
    },
    Za = {
      dir: $a,
      has: Ne,
      lang: $a,
      matches: Ne,
      is: Ne,
      "-moz-any": Ne,
      "-webkit-any": Ne,
      where: Ne,
      not: Ne,
      "nth-child": ir,
      "nth-last-child": ir,
      "nth-last-of-type": ir,
      "nth-of-type": ir,
      slotted: _o,
      host: _o,
      "host-context": _o,
    };
  var qo = {};
  b(qo, {
    AnPlusB: () => Jr,
    Atrule: () => tn,
    AtrulePrelude: () => nn,
    AttributeSelector: () => sn,
    Block: () => cn,
    Brackets: () => pn,
    CDC: () => mn,
    CDO: () => dn,
    ClassSelector: () => bn,
    Combinator: () => yn,
    Comment: () => wn,
    Declaration: () => Sn,
    DeclarationList: () => Tn,
    Dimension: () => Ln,
    Function: () => In,
    Hash: () => On,
    IdSelector: () => Rn,
    Identifier: () => zn,
    MediaFeature: () => Bn,
    MediaQuery: () => qn,
    MediaQueryList: () => jn,
    NestingSelector: () => Hn,
    Nth: () => Gn,
    Number: () => Kn,
    Operator: () => Xn,
    Parentheses: () => Zn,
    Percentage: () => eo,
    PseudoClassSelector: () => ro,
    PseudoElementSelector: () => oo,
    Ratio: () => ao,
    Raw: () => lo,
    Rule: () => uo,
    Selector: () => ho,
    SelectorList: () => fo,
    String: () => yo,
    StyleSheet: () => wo,
    TypeSelector: () => Co,
    UnicodeRange: () => Eo,
    Url: () => Oo,
    Value: () => zo,
    WhiteSpace: () => Ro,
  });
  var Ja = {
    parseContext: {
      default: "StyleSheet",
      stylesheet: "StyleSheet",
      atrule: "Atrule",
      atrulePrelude(e) {
        return this.AtrulePrelude(e.atrule ? String(e.atrule) : null);
      },
      mediaQueryList: "MediaQueryList",
      mediaQuery: "MediaQuery",
      rule: "Rule",
      selectorList: "SelectorList",
      selector: "Selector",
      block() {
        return this.Block(!0);
      },
      declarationList: "DeclarationList",
      declaration: "Declaration",
      value: "Value",
    },
    scope: Bo,
    atrule: Xa,
    pseudo: Za,
    node: qo,
  };
  var es = { node: xt };
  var ts = Zt({ ...za, ...Ja, ...es });
  var Gh = "2.3.0";
  function ar(e) {
    let t = {};
    for (let r in e) {
      let n = e[r];
      n &&
        (Array.isArray(n) || n instanceof D
          ? (n = n.map(ar))
          : n.constructor === Object && (n = ar(n))),
        (t[r] = n);
    }
    return t;
  }
  var Uo = {};
  b(Uo, { decode: () => Vh, encode: () => Kh });
  var rs = 92;
  function Vh(e) {
    let t = e.length - 1,
      r = "";
    for (let n = 0; n < e.length; n++) {
      let o = e.charCodeAt(n);
      if (o === rs) {
        if (n === t) break;
        if (((o = e.charCodeAt(++n)), $(rs, o))) {
          let i = n - 1,
            s = se(e, i);
          (n = s - 1), (r += _e(e.substring(i + 1, s)));
        } else o === 13 && e.charCodeAt(n + 1) === 10 && n++;
      } else r += e[n];
    }
    return r;
  }
  function Kh(e) {
    let t = "";
    if (e.length === 1 && e.charCodeAt(0) === 45) return "\\-";
    for (let r = 0; r < e.length; r++) {
      let n = e.charCodeAt(r);
      if (n === 0) {
        t += "\uFFFD";
        continue;
      }
      if (
        n <= 31 ||
        n === 127 ||
        (n >= 48 && n <= 57 && (r === 0 || (r === 1 && e.charCodeAt(0) === 45)))
      ) {
        t += "\\" + n.toString(16) + " ";
        continue;
      }
      Re(n) ? (t += e.charAt(r)) : (t += "\\" + e.charAt(r));
    }
    return t;
  }
  var {
    tokenize: Qh,
    parse: Xh,
    generate: $h,
    lexer: Zh,
    createLexer: Jh,
    walk: em,
    find: tm,
    findLast: rm,
    findAll: nm,
    toPlainObject: om,
    fromPlainObject: im,
    fork: am,
  } = ts;
  return cs(sm);
})();
//# sourceURL=/Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/js/csstree.js</script><script type="text/javascript">// Set the handler
class elifeBuild extends Paged.Handler {
  constructor(chunker, polisher, caller) {
    super(chunker, polisher, caller);
  }

  beforeParsed(content) {

    // to divide figure in block
    dispatchFigure(content);

    //tag figure
    tagImgInFigures(content);

    // add id to anything to fix things
    addIDtoEachElement(content);
  }

  afterRendered(pages) {
    // add the running head
    let runninghead = document.querySelector(".runninghead");
    // add the MSAs
    let msas = document.querySelector(".article-flag-list");
    // add the logo
    let logo = document.getElementById("logo");
    document
      .querySelectorAll(".pagedjs_pagedjs-filler_page")
      .forEach((page) => {
        if (runninghead) {
          page
            .querySelector(".pagedjs_pagebox")
            .insertAdjacentElement("afterbegin", runninghead.cloneNode(true));
        }
        if (msas) {
          page
            .querySelector(".pagedjs_pagebox")
            .insertAdjacentElement("afterbegin", msas.cloneNode(true));
        }
        let marginForLogo = page.querySelector(".pagedjs_margin-top-left > .pagedjs_margin-content");
        let marginBox = page.querySelector(".pagedjs_margin-top-left");
        if (logo && marginForLogo) {
          marginForLogo.insertAdjacentElement("afterbegin", logo.cloneNode(true));
          if (marginBox) {
            marginBox.classList.add("hasContent");
            }  
        }
      });
  }
}

Paged.registerHandlers(elifeBuild);

// To use to remove hyphens between pages
function getFinalWord(words) {
  var n = words.split(" ");
  return n[n.length - 1];
}


/*========================== 
         addIDtoEachElement
    ========================== */

// Define here the tags you want to give id
let tags = [
  "figure",
  "figcaption",
  "img",
  "ol",
  "ul",
  "li",
  "p",
  "table",
  "h1",
  "h2",
  "h3",
  "h4",
  "div",
  "aside",
];

function addIDtoEachElement(content) {
  let total = 0;
  tags.forEach((tag) => {
    content.querySelectorAll(tag).forEach((el, index) => {
      if (!el.id) {
        if (el.tagName == "p") {
          if (el.closest("figcaption")) {
            return;
          }
        }

        el.id = `el-${el.tagName.toLowerCase()}-${index}`;
        total++;
      }
    });
  });
}

// no hyphens between page
class noHyphenBetweenPage extends Paged.Handler {
  constructor(chunker, polisher, caller) {
    super(chunker, polisher, caller);
    this.hyphenToken;
  }

  afterPageLayout(pageFragment, page, breakToken) {
    if (pageFragment.querySelector(".pagedjs_hyphen")) {
      // find the hyphenated word
      let block = pageFragment.querySelector(".pagedjs_hyphen");

      // i dont know what that line was for :thinking: i removed it
      // block.dataset.ref = this.prevHyphen;

      // move the breakToken
      let offsetMove = getFinalWord(block.innerHTML).length;

      // move the token accordingly
      page.breakToken = page.endToken.offset - offsetMove;

      // remove the last word
      block.innerHTML = block.innerHTML.replace(
        getFinalWord(block.innerHTML),
        ""
      );

      breakToken.offset = page.endToken.offset - offsetMove;
    }
  }
}

Paged.registerHandlers(noHyphenBetweenPage);

class pushThings extends Paged.Handler {
  constructor(chunker, polisher, caller) {
    super(chunker, polisher, caller);
    this.pushblock = [];
  }
  onDeclaration(declaration, dItem, dList, rule) {
    // move the element to the next bit
    if (declaration.property == "--experimental-push") {
      let sel = csstree.generate(rule.ruleNode.prelude);
      sel = sel.replace('[data-id="', "#");
      sel = sel.replace('"]', "");
      let itemsList = sel.split(",");
      itemsList.forEach((elId) => {
        this.pushblock.push([elId, declaration.value.value]);
      });
    }
  }
  afterParsed(parsed) {
    //reset the page break for the figures titles
    // let titles1 = parsed.querySelectorAll("h1,h2");
    // titles1.forEach((title, index) => {
    //   if (
    //     title.classList?.contains("probably-figures-title") &&
    //     !title.classList?.contains("skipBreakAfter")
    //   ) {
    //     title.dataset.breakAfter = "page";
    //   }
    //   if (
    //     title.classList?.contains("probably-figures-title-after") &&
    //     !title.classList?.contains("skipBreakBefore")
    //   ) {
    //     title.dataset.breakBefore = "page";
    //     delete title.dataset.breakAfter;
    //   }
    // });

    if (this.pushblock.length > 0) {
      this.pushblock.forEach((elToPush) => {
        const elem = parsed.querySelector(elToPush[0]);
        if (!elem) {
          return;
        }

        elem.dataset.pushBlock = elToPush[1];
        let direction = "";
        if (elToPush[1] < 0) {
          direction = "back";
        }
        if (direction == "back") {
          for (let index = 0; index < Math.abs(elToPush[1]); index++) {
            if (elem.previousElementSibling) {
              elem.previousElementSibling.insertAdjacentElement(
                "beforebegin",
                elem
              );
            }
          }
        } else {
          for (let index = 0; index < Math.abs(elToPush[1]); index++) {
            if (elem.nextElementSibling) {
              elem.nextElementSibling.insertAdjacentElement(
                "beforebegin",
                elem
              );
            }
          }
        }
      });
    }
  }
}

Paged.registerHandlers(pushThings);

class CSStoClass extends Paged.Handler {
  constructor(chunker, polisher, caller) {
    super(chunker, polisher, caller);
    this.floatSameTop = [];
    this.floatSameBottom = [];
    this.floatNextTop = [];
    this.floatNextBottom = [];
    this.experimentalImageEdit = [];
    this.spacing = [];
    this.pushblock = [];
    this.experimentalMerged = [];
    this.fullPageBackground = [];
  }
  onDeclaration(declaration, dItem, dList, rule) {
    // alter the image
    if (declaration.property == "--experimental-image-edit") {
      let sel = csstree.generate(rule.ruleNode.prelude);
      sel = sel.replace('[data-id="', "#");
      sel = sel.replace('"]', "");
      let itemsList = sel.split(",");
      itemsList.forEach((elId) => {
        this.experimentalImageEdit.push(elId);
      });
    }
    // move the element to the next bit
    else if (declaration.property == "--experimental-push") {
      let sel = csstree.generate(rule.ruleNode.prelude);
      sel = sel.replace('[data-id="', "#");
      sel = sel.replace('"]', "");
      let itemsList = sel.split(",");
      itemsList.forEach((elId) => {
        this.pushblock.push([elId, declaration.value.value]);
      });
    } else if (declaration.property == "--experimental-fullpage") {
      let sel = csstree.generate(rule.ruleNode.prelude);
      sel = sel.replace('[data-id="', "#");
      sel = sel.replace('"]', "");
      let itemsList = sel.split(",");
      itemsList.forEach((elId) => {
        this.fullPageBackground.push([elId, declaration.value.value]);
      });
    }
    //experimental merge
    else if (declaration.property == "--experimental-merge") {
      let sel = csstree.generate(rule.ruleNode.prelude);
      sel = sel.replace('[data-id="', "#");
      sel = sel.replace('"]', "");
      let itemsList = sel.split(",");
      itemsList.forEach((elId) => {
        this.experimentalMerged.push([elId, declaration.value.value]);
      });
    }
    // page floats
    else if (declaration.property == "--experimental-page-float") {
      if (declaration.value.value.includes("same-top")) {
        let sel = csstree.generate(rule.ruleNode.prelude);
        sel = sel.replace('[data-id="', "#");
        sel = sel.replace('"]', "");
        this.floatSameTop.push(sel.split(","));
        // console.log("floatSameTop: ", this.floatSameTop);
      } else if (declaration.value.value.includes("same-bottom")) {
        let sel = csstree.generate(rule.ruleNode.prelude);
        sel = sel.replace('[data-id="', "#");
        sel = sel.replace('"]', "");
        this.floatSameBottom.push(sel.split(","));
        //console.log("floatSameBottom: ", this.floatSameBottom);
      } else if (declaration.value.value.includes("next-top")) {
        let sel = csstree.generate(rule.ruleNode.prelude);
        sel = sel.replace('[data-id="', "#");
        sel = sel.replace('"]', "");
        this.floatNextTop.push(sel.split(","));
        //console.log('floatNextTop: ', this.floatNextTop);
      } else if (declaration.value.value.includes("next-bottom")) {
        let sel = csstree.generate(rule.ruleNode.prelude);
        sel = sel.replace('[data-id="', "#");
        sel = sel.replace('"]', "");
        this.floatNextBottom.push(sel.split(","));
        //console.log("floatNextBottom: ", this.floatNextBottom);
      }
    }
    // spacing
    else if (declaration.property == "--experimental-spacing") {
      var spacingValue = declaration.value.value;
      spacingValue = spacingValue.replace(/\s/g, "");
      spacingValue = parseInt(spacingValue);
      let sel = csstree.generate(rule.ruleNode.prelude);
      sel = sel.replace('[data-id="', "#");
      sel = sel.replace('"]', "");
      var thisSpacing = [sel.split(","), spacingValue];
      this.spacing.push(thisSpacing);
    }
  }

  afterParsed(parsed) {
    if (this.pushblock.length > 0) {
      this.pushblock.forEach((elToPush) => {
        const elem = parsed.querySelector(elToPush[0]);
        if (!elem) {
          return;
        }

        elem.dataset.pushBlock = elToPush[1];
        let direction = "";
        if (elToPush[1] < 0) {
          direction = "back";
        }
        if (direction == "back") {
          for (let index = 0; index < Math.abs(elToPush[1]); index++) {
            if (elem.previousElementSibling) {
              elem.previousElementSibling.insertAdjacentElement(
                "beforebegin",
                elem
              );
            }
          }
        } else {
          for (let index = 0; index < Math.abs(elToPush[1]); index++) {
            if (elem.nextElementSibling) {
              elem.nextElementSibling.insertAdjacentElement("afterend", elem);
            }
          }
        }
      });
    }
    if (this.experimentalMerged.length > 0) {
      this.experimentalMerged.forEach((couple) => {
        const host = parsed.querySelector(couple[0]);
        const guest = parsed.querySelector(couple[1]);
        if (!host || !guest) {
          return;
        }
        guest.style.display = "none";
        host.classList.add("merged!");
        host.dataset.mergedGuest = guest.id;
        host.insertAdjacentHTML("beforeend", guest.innerHTML);
      });
    }
    if (this.experimentalImageEdit.length > 0) {
      this.experimentalImageEdit.forEach((elNBlist) => {
        parsed.querySelectorAll(elNBlist).forEach((img) => {
          img.classList.add("imageMover");

          // console.log("#" + img.id + ": image Mover");
        });
      });
    }
    if (this.floatNextBottom) {
      this.floatNextBottom.forEach((elNBlist) => {
        parsed.querySelectorAll(elNBlist).forEach((el) => {
          el.classList.add("page-float-next-bottom");
          // console.log("#" + el.id + " moved to next-bottom");
        });
      });
    }
    if (this.floatNextTop) {
      this.floatNextTop.forEach((elNBlist) => {
        parsed.querySelectorAll(elNBlist).forEach((el) => {
          el.classList.add("page-float-next-top");
          // console.log("#" + el.id + " moved to next-top");
        });
      });
    }
    if (this.floatSameTop) {
      this.floatSameTop.forEach((elNBlist) => {
        parsed.querySelectorAll(elNBlist).forEach((el) => {
          el.classList.add("page-float-same-top");
          // console.log("#" + el.id + " moved to same-top");
        });
      });
    }
    if (this.floatSameBottom) {
      this.floatSameBottom.forEach((elNBlist) => {
        parsed.querySelectorAll(elNBlist).forEach((el) => {
          el.classList.add("page-float-same-bottom");
          // console.log("#" + el.id + " moved to same-bottom");
        });
      });
    }
    if (this.spacing) {
      this.spacing.forEach((elNBlist) => {
        parsed.querySelectorAll(elNBlist[0]).forEach((el) => {
          var spacingValue = elNBlist[1];
          var spacingClass = "spacing-" + spacingValue;
          // console.log(spacingClass);
          el.classList.add(spacingClass);
          // console.log("#" + el.id + " spaced " + spacingValue);
        });
      });
    }
    if (this.fullPageBackground) {
      this.fullPageBackground.forEach((background) => {
        parsed.querySelectorAll(background[0]).forEach((el) => {
          el.classList.add("moveToBackgroundImage");
        });
      });
    }
  }
}

Paged.registerHandlers(CSStoClass);

//float top

// lets you manualy add classes to some pages elements
// to simulate page floats.
// works only for elements that are not across two pages

let classElemFloatSameTop = "page-float-same-top"; // â† class of floated elements on same page
let classElemFloatSameBottom = "page-float-same-bottom"; // â† class of floated elements bottom on same page

let classElemFloatNextTop = "page-float-next-top"; // â† class of floated elements on next page
let classElemFloatNextBottom = "page-float-next-bottom"; // â† class of floated elements bottom on next page

class elemFloatTop extends Paged.Handler {
  constructor(chunker, polisher, caller) {
    super(chunker, polisher, caller);
    this.experimentalFloatNextTop = [];
    this.baseline = 22;
    this.experimentalFloatNextBottom = [];
    this.token;
  }

  layoutNode(node) {
    // If you find a float page element, move it in the array,
    if (node.nodeType == 1 && node.classList.contains(classElemFloatNextTop)) {
      let clone = node.cloneNode(true);
      this.experimentalFloatNextTop.push(clone);
      // Remove the element from the flow by hiding it.
      node.style.display = "none";
    }
    if (
      node.nodeType == 1 &&
      node.classList.contains(classElemFloatNextBottom)
    ) {
      let clone = node.cloneNode(true);
      this.experimentalFloatNextBottom.push(clone);
      // Remove the element from the flow by hiding it.
      node.style.display = "none";
    }

    if (
      node.nodeType == 1 &&
      node.classList.contains(classElemFloatSameBottom)
    ) {
      let clone = node.cloneNode(true);
      // this.experimentalFloatNextBottom.push(clone);
      // Remove the element from the flow by hiding it.
      node.style.display = "none";
    }
  }

  beforePageLayout(page, content, breakToken) {
    //console.log(breakToken);
    // If there is an element in the floatPageEls array,
    if (this.experimentalFloatNextTop.length >= 1) {
      // Put the first element on the page.
      page.element
        .querySelector(".pagedjs_page_content")
        .insertAdjacentElement("afterbegin", this.experimentalFloatNextTop[0]);
      this.experimentalFloatNextTop.shift();
    }
    if (this.experimentalFloatNextBottom.length >= 1) {
      // Put the first element on the page.
      page.element
        .querySelector(".pagedjs_page_content")
        .insertAdjacentElement(
          "afterbegin",
          this.experimentalFloatNextBottom[0]
        );
      this.experimentalFloatNextBottom.shift();
    }
  }

  layoutNode(node) {
    if (node.nodeType == 1) {
      if (node.classList.contains(classElemFloatSameTop)) {
        let clone = node.cloneNode(true);
        clone.classList.add("figadded");
        document
          .querySelector(".pagedjs_pages")
          .lastElementChild.querySelector("article")
          .insertAdjacentElement("afterbegin", clone);
        node.style.display = "none";
        node.classList.add("hide");
      }

      if (
        node.previousElementSibling?.classList.contains(classElemFloatSameTop)
      ) {
        let img = document
          .querySelector(".pagedjs_pages")
          .lastElementChild.querySelector(`.${classElemFloatSameTop}`);
        // console.log(img)
        // console.log(node)
        // if (img?.clientHeight) {
        //   // count the number of line for the image
        //   let imgHeightLine = Math.floor(img.clientHeight / this.baseline)
        //   // add one light and get the height in pixeol
        //   img.dataset.lineOffset = imgHeightLine + 0
        //   img.style.height = `${(imgHeightLine + 0) * this.baseline}px`
        // }
      }
    }
  }
  // works only with non breaked elements
  afterPageLayout(page, content, breakToken) {
    // try fixed bottom on same if requested
    if (page.querySelector("." + classElemFloatSameBottom)) {
      var bloc = page.querySelector("." + classElemFloatSameBottom);
      bloc.classList.add("absolute-bottom");
      bloc.classList.add("figadded");
    }

    // try fixed bottom if requested
    if (page.querySelector("." + classElemFloatNextBottom)) {
      var bloc = page.querySelector("." + classElemFloatNextBottom);
      bloc.classList.add("absolute-bottom");
      bloc.classList.add("figadded");
    }
  }
}
Paged.registerHandlers(elemFloatTop);

/* url cut*/

class expMerge extends Paged.Handler {
  constructor(chunker, polisher, caller) {
    super(chunker, polisher, caller);
    this.experimentalMerged = [];
  }

  onDeclaration(declaration, dItem, dList, rule) {
    // alter the image
    //experimental merge
    if (declaration.property == "--experimental-merge") {
      let sel = csstree.generate(rule.ruleNode.prelude);
      sel = sel.replace('[data-id="', "#");
      sel = sel.replace('"]', "");
      let itemsList = sel.split(",");
      itemsList.forEach((elId) => {
        this.experimentalMerged.push([elId, declaration.value.value]);
      });
    }
  }

  beforeParsed(parsed) {
    if (this.experimentalMerged.length > 0) {
      this.experimentalMerged.forEach((couple) => {
        const host = parsed.querySelector(couple[0]);
        const guest = parsed.querySelector(couple[1]);
        if (!host || !guest) {
          return;
        }
        guest.style.display = "none";
        host.classList.add("merged!");
        host.dataset.mergedGuest = guest.id;
        host.insertAdjacentHTML("beforeend", guest.innerHTML);
      });
    }
  }
}

Paged.registerHandlers(expMerge);

class moveToParentFig extends Paged.Handler {
  constructor(chunker, polisher, caller) {
    super(chunker, polisher, caller);
    this.moveToParentFig = [];
  }
  onDeclaration(declaration, dItem, dList, rule) {
    // move the element to the next bit
    if (declaration.property == "--experimental-moveToOutsideFigure") {
      let sel = csstree.generate(rule.ruleNode.prelude);
      sel = sel.replace('[data-id="', "#");
      sel = sel.replace('"]', "");
      let itemsList = sel.split(",");
      itemsList.forEach((elId) => {
        this.moveToParentFig.push([elId, declaration.value.value]);
      });
    }
  }
  beforeParsed(content) {
    if (this.moveToParentFig.length > 0) {
      this.moveToParentFig.forEach((elToPush) => {
        const elem = content.querySelector(elToPush[0]);
        if (!elem) {
          return;
        }
        let fighead = elem.querySelector("label").cloneNode(true);
        console.log("label", fighead);
        elem.insertAdjacentElement("beforeend", fighead);
        elem.closest("figure").insertAdjacentElement("afterend", elem);
      });
    }
  }
}
Paged.registerHandlers(moveToParentFig);

/** This is a rough draft 
async function formulaeTest(parsed) {
  const imagePromises = [];
  const paraImages = parsed.querySelectorAll("p img");
  paraImages.forEach((image) => {
    const img = new Image();
    let resolve, reject;
    const imageLoaded = new Promise(function (r, x) {
      resolve = r;
      reject = x;
    });

    img.onload = function () {
      const { widthClass, heightClass, ratioClass } = getSizeRatioClass(
        img.naturalWidth,
        img.naturalHeight
      );

      image.classList.add(widthClass, heightClass, ratioClass);

      const para = image.closest("p");

      for (child of para?.childNodes) {
        if (child.nodeType === Node.ELEMENT_NODE)
          para.classList.add(
            child.tagName === "IMG" ? "hasMultipleImages" : "hasOtherElem"
          );
        if (child.nodeType === Node.TEXT_NODE) para.classList.add("hasContent");
        if (para.matches(`p.hasMultipleImages.hasOtherElem.hasContent`)) break;
      }
      // console.log("loaded-------");
      para.classList.add("hasImage");
      resolve();
    };
    img.onerror = function () {
      reject();
    };

    img.src = image.src;

    imagePromises.push(imageLoaded);
  });
  try {
    return await Promise.all(imagePromises);
  } catch (err) {
    console.warn(err);
  }
}*/

function getSizeRatioClass(width, height) {
  return {
    widthClass: getSizeClass(width, "width"),
    heightClass: getSizeClass(height, "height"),
    ratioClass: getRatioClass(width / height),
  };
}
function getSizeClass(size, paramStr) {
  if (size <= 48) return `${paramStr}-40`;
  if (size <= 80) return `${paramStr}-80`;
  if (size <= 160) return `${paramStr}-160`;
  if (size <= 240) return `${paramStr}-240`;
  if (size <= 360) return `${paramStr}-360`;
  if (size <= 480) return `${paramStr}-480`;
  return `${paramStr}-480-above`;
}

function getRatioClass(ratio) {
  if (ratio >= 1.7) return "landscape";
  if (ratio <= 0.95) return "portrait";
  if (ratio < 1.39 || ratio > 0.95) return "square";
}

class fixMarginTop extends Paged.Handler {
  constructor(chunker, polisher, caller) {
    super(chunker, polisher, caller);
    this.pagedone = false;
    let hasRenderedContent = false;
  }

  onPageLayout() {
    this.hasRenderedContent = false;
    this.pagedone = false;
  }

  renderNode(node) {
    if (this.pagedone == true) return;
    if (node.nodeType != 1) return;

    if (!this.hasRenderedContent) {
      this.hasRenderedContent = hasContent(node);
    }

    if (this.hasRenderedContent == true) {
      // check if the starting page is really at 0
      const wrapper = node
        .closest(".pagedjs_page_content")
        .querySelector("div");
      let wrapperOffsetTop = wrapper.offsetTop;
      if (wrapperOffsetTop != 0) {
        // console.log(wrapperOffsetTop);
        // console.log(wrapper.style.marginTop);
        // wrapper.style.position = `relative`
        // wrapper.style.marginTop = `-${wrapperOffsetTop}px`;
      }

      // console.log(wrapperOffsetTop);
    }
    this.pagedone = true;
  }

  // check if node is the first element
}

function isElement(node) {
  return node && node.nodeType === 1;
}

function isText(node) {
  return node && node.nodeType === 3;
}

function hasContent(node) {
  if (isElement(node)) {
    return true;
  } else if (isText(node) && node.textContent.trim().length) {
    return true;
  }
  return false;
}

Paged.registerHandlers(fixMarginTop);

function dispatchFigure(content) {
  content.querySelectorAll("figure").forEach((fig) => {
    // if there is no label keep the image together
    if (fig.classList.contains("no-label")) {
      return;
    }

    if (fig.querySelectorAll("img").length > 1) {
      // not to fill is not yet added.
      // we need to do that before anythin
      if (fig.classList.contains("nottofill")) {
        return;
      }

      // get the last figure
      let imageToMove = [...fig.querySelectorAll("img")];

      // Filter out images that are children of <figcaption> elements
      imageToMove = imageToMove.filter((img) => !img.closest("figcaption"));

      // create figures for sub images block
      for (let index = imageToMove.length; index > 1; index--) {
        const element = imageToMove[index - 1];

        // link figure
        fig.classList.add("has-continued");
        fig.dataset.continuedMain = `ctn-${fig.id}`;
        fig.insertAdjacentHTML(
          "afterend",
          `<figure data-continued-from="ctn-${
            fig.id
          }" class="added-figure-tofill tofill continuedContent" id="${
            fig.id
          }-cont-${index}"><figcaption><p data-brea-after="unset" class="continued"><span class="label figure-name">${
            fig?.querySelector(".label")
              ? fig.querySelector(".label").innerHTML
              : ""
          }</span> <span class="ctn">(continued)</span></p></figcaption>${
            element.outerHTML
          }</figure>`
        );
        element.remove();
      }
    }
  });
}

// if figure have multiple images, just put the image into following figures
// so <fig > img1 + img2
// becomes
// <fig > img1 > + fig img2 etc.
// add a class to those figure so they’re following the main process instead of being rendered as image outisde of figures
//
//

async function addWidthHeightToImg(parsed) {
  let imagePromises = [];
  let images = parsed.querySelectorAll("img");
  images.forEach((image) => {
    if (image.width && image.height) return;
    let img = new Image();
    let resolve, reject;
    let imageLoaded = new Promise(function (r, x) {
      resolve = r;
      reject = x;
    });

    img.onload = function () {
      let height = img.naturalHeight;
      let width = img.naturalWidth;
      image.setAttribute("height", height);
      image.setAttribute("width", width);

      let ratio = width / height;
      if (ratio >= 1.7) {
        image.classList.add("landscape");
        image.parentNode.classList.add("fig-landscape");
      } else if (ratio <= 0.95) {
        image.classList.add("portrait");
        image.parentNode.classList.add("fig-portrait");
      } else if (ratio < 1.39 || ratio > 0.95) {
        image.classList.add("square");
        image.parentNode.classList.add("fig-square");
      }
      resolve();
    };
    img.onerror = function () {
      reject();
    };

    img.src = image.src;

    imagePromises.push(imageLoaded);
  });
  return Promise.all(imagePromises).catch((err) => {
    console.warn("err", err);
  });
}

/*break after*/
// check if the element has a break after avoid and move it on next page
class avoidBreakAfter extends Paged.Handler {
  constructor(chunker, polisher, caller) {
    super(chunker, polisher, caller);
  }

  finalizePage(pageFragment, page) {
    // remove any final emtpy element

    pageFragment.querySelectorAll("ol").forEach((ol) => {
      // remove any empty ol at the end of any page: if ol is the last element and is empty
      if (ol.textContent.length < 1 && ol.parentElement.lastChild == ol) {
        ol.remove();
      }
    });

    let possibleBreakAvoid = pageFragment.querySelectorAll(
      `[data-break-after="avoid"]`
    );

    let goodEl;

    possibleBreakAvoid.forEach((el) => {
      // if element hs been marked as having nothing after to stick to.

      if (el.classList.contains("no-el-next")) return;
      if (el.classList.contains("done")) return;
      // if (el.nextElementSibling) return;
      if (el.closest(".movedfig")) return;
      if (el.closest("figure")) return;
      if (el.parentElement.tagName == "FIGURE") {
        // .... there shouldnt be any figures anymore
      }
      // el.classList.add("touched")
      //

      if (
        !el.nextElementSibling &&
        !el.parentElement.nextElementSibling &&
        !el.tagName == "SECTION"
      ) {
        goodEl = el;

        // console.log(el.nextElementSibling);

        while (
          goodEl.previousElementSibling &&
          goodEl.previousElementSibling.dataset.breakAfter == "avoid"
        ) {
          goodEl.classList.add("toremove");
          goodEl = goodEl.previousElementSibling;
          debugger;
        }
      }
    });

    //if there is no element at the end keep going
    if (!goodEl) return;

    console.log("goodel", goodEl.dataset.ref);

    // console.log(page);
    const elementFromSource = this.chunker.source.querySelector(
      `[data-ref="${goodEl.dataset.ref}"]`
    );

    elementFromSource.dataset.breakAfter = "none";

    // elementFromSource.data

    while (goodEl.nextElementSibling) {
      goodEl.nextElementSibling.remove();
    }
    // goodEl.remove();

    // switch the breaktoken
    if (page.endToken) {
      page.breakToken = page.endToken.node = elementFromSource;
      page.breakToken = page.endToken.offset = 0;
    }
  }
}
Paged.registerHandlers(avoidBreakAfter);

function tagImgInFigures(content) {
  const figures = content.querySelectorAll("figure");
  figures.forEach((figure) => {
    //number of images
    figure.classList.add(`imgs-${figure.querySelectorAll("img").length}`);
    //
    // is there a title
    // TODO
    // figure.classList.add(`has-title`);
    // is there a caption

    if (
      !figure.querySelector("figcaption p") &&
      !figure.querySelector("label")
    ) {
      figure.classList.add("font");
      figure.classList.add("no-caption");
    }
    // how long is the caption
    // when the caption is less than 550, makes it one column

    let captionLength = 0;
    figure
      .querySelector("figcaption")
      ?.querySelectorAll("p, li")
      .forEach((el) => {
        captionLength = captionLength + el.textContent.length;
      });

    // console.log(figure.id, captionLength);
    if (captionLength < 800 && captionLength >= 500) {
      figure.classList.add("shortcaption");
    } else if (captionLength < 500) {
      figure.classList.add("shortercaption");
    } else {
      figure.classList.add("longcaption");
    }

    // if it’s a continued element, don’t show its name
    if (figure.querySelector(".ctn")) return;
  });
}//# sourceURL=/Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/js/elife-custom-scripts.js</script><script type="text/javascript">// this script wil put any element with the property
// `position: fill page` on its own page while keeping the rest of the content flowing.
//
// use:
//
// ---===---===---
//
//   .elementToFill {
//     position: fill-page;
//   }
//
// ---===---===---
//
// the pagedjs-fillpage template is created by the script to manage the layout of the fullpage layout
//
//this will try to fill up the page with any image coming up from the content
//
//

const fillPageClass = "pagedjs-fill-next-page";

class fullpage extends Paged.Handler {
  constructor(chunker, polisher, caller) {
    super(chunker, polisher, caller);
    this.fillPageElements = [];
    this.pushToNextPage = [];
    this.putThatFirst = [];
    this.needPage = [];
    this.notToFillEls = [];
  }

  async onPageLayout(page, Token, layout) {
    if (this.putThatFirst.length > 1) {
      while (this.putThatFirst.length > 1) {
        // get the first element while removing it from the array
        const elem = this.putThatFirst.shift();

        // check if there is enough room to put the element, and make sure that it’s not a continuted figure,
        console.log("elem", elem);

        // if its an element that has continued figure, move it on the next page by default
        if (elem.dataset.continuedMain) {
          this.needPage.push(elem);
          return;
        } else if (elem.classList.includes("continuedContent")) {
          console.log(elem);
          this.needPage.push(elem);
          return;
        } else if (
          (await getHeightOfHiddenElement(elem)) <
          (await getRemainingSpaceOnPage(page))
        ) {
          // check if there is enough room, otherwise add a page

          elem.className = "added-fig-post";
          elem.style.marginBottom = "17px";
          page.insertAdjacentHTML(
            "afterbegin",
            `<figure class="internalfig">${elem.innerHTML}</figure>`,
          );
        } else {
          this.needPage.push(elem);
        } // then check if there is enough room on the page, other wise add a new one
      }
    }
  }

  //find from the css the element you wanna have  full page
  onDeclaration(declaration, dItem, dList, rule) {
    if (declaration.property == "position") {
      if (declaration.value.children.head.data.name.includes("fill-page")) {
        let sel = csstree.generate(rule.ruleNode.prelude);
        // sel = sel.replace(/\[data-id="\\"(.*)\\""]/g, "#$1");
        this.fillPageElements.push(sel.split(","));
      }
    }
    if (declaration.property == "--not-to-fill") {
      let sel = csstree.generate(rule.ruleNode.prelude);
      console.log(sel);
      sel = sel.replace(/\[data-id="\\"(.*)\\""]/g, "#$1");
      this.notToFillEls.push(sel.split(","));
    }
  }

  async afterParsed(parsed) {
    // // get the size for the image
    // add the width and height to the image as attribute to get more data about what they are
    await addWidthHeightToImg(parsed);
    await calculateDesiredWidthPercentages(parsed.querySelectorAll("img"));

    /* This is redundant now that we know which images are maths
    try to see if an image is less than 50 px ans assume it’s math
    parsed.querySelectorAll("img").forEach((img) => {
      if (
        img.height &&
        img.height < 400 &&
        !img.closest("figure")?.id.includes("tbl")
      ) {
        img.classList.add("probablyMath");
      }
      if (img.height && img.height < 200) {
        if (img.closest("figure")) {
          if (
            img.closest("figure").classList.contains("continuedContent") ||
            img.closest("figure").dataset.continuedMain
          )
            return;
          img.closest("figure")?.classList.remove(fillPageClass);
          // img.closest("figure").remove();
        }
      }
    });*/

    parsed.querySelectorAll(".disp-formula, .inline-formula").forEach((img) => {
      // if (img.closest("figure")) {
      //   return;
      // }

      // console.log(img);

      // for those image, we should remove the max-width
      // if (img.height < 20) {
      //   img.classList.add("formula-max-20")
      // }
      // if (img.height < 40) {
      //   img.classList.add("formula-max-40")
      // }
      // else if (img.height >= 40 && img.height < 80) {
      //   img.classList.add("formula-max-80")
      // }
      // else if (img.height >= 80 && img.height < 100) {
      //   img.classList.add("formula-max-100")
      // }
      // else if (img.height >= 100 && img.height < 140) {
      //   img.classList.add("formula-max-140")
      // }
      // else if (img.height >= 140 && img.height < 180) {
      //   img.classList.add("formula-max-180")
      // }
      // else if (img.height >= 180 && img.height < 240) {
      //   img.classList.add("formula-max-240")
      // }
      // else if (img.height >= 240 && img.height < 300) {
      //   img.classList.add("formula-max-300")
      // }

      for (let i = 0; i <= 400; i += 20) {
        if (img.height > i - 10 && img.height <= i + 10) {
          img.classList.add(`formula-max-${i}`);
          break;
        }
      }
    });

    //find from the css the element you wanna have  full page
    //
    //
    //
    // override tofill
    if (this.notToFillEls.length > 0) {
      this.notToFillEls.forEach((selector) => {
        console.log(selector);
        parsed.querySelectorAll(selector).forEach((el) => {
          console.log(el);
          el.classList.add("nottofill");
          el.classList.remove("tofill");
          //
          el.insertAdjacentElement("beforeend", el.querySelector("figcaption"));
        });
      });
    }

    if (("this", this.fillPageElements)) {
      this.fillPageElements.forEach((selector) => {
        parsed.querySelectorAll(selector).forEach((el) => {
          //   if (
          //     !el.querySelector("img").closest("figcaption") &&
          //     el.querySelector("img").height < 150 &&
          //     !el
          //       .querySelector("img")
          //       .closest("figure")
          //       .classList.contains("continuedContent")
          //   )
          //   return el.closest("figure").classList.remove("tofill");

          // check when a new element is added, remove it if it has dataset.continuedMain

          if (!el.classList.contains("nottofill")) {
            // else add fill page class
            el.classList.add(fillPageClass);
            el.style.display = "none";
          }
        });
      });
    }

    parsed.querySelectorAll(".nottofill").forEach((el) => {
      console.log(el);
      const continued = parsed.querySelectorAll(
        `[data-continued-from*="ctn-${el.id}"`,
      );

      // if the fig is dived we bring it together
      if (continued.length > 0) {
        continued.forEach((continuedEl) => {
          el.insertAdjacentElement(
            "beforeend",
            continuedEl.querySelector("img"),
          );
          continuedEl.remove();
        });
      }
      el.insertAdjacentElement("beforeend", el.querySelector("figcaption"));
      // continued.insertAdjacentElement("beforeend", continued.querySelector("figcaption"))
    });
  }

  // for column safety
  async finalizePage(pageFragment, oldpage) {
    // if (pageFragment.classList.contains("pagedjs_pagedjs-filler_page"))
    //   return;

    // if (this.needPage.length > 1) {
    // while (this.needPage.length > 1) {
    //   const elem = this.needPage.shift()
    //   elem.style.display = "block"
    //   let newPage = this.chunker.addPage();
    //   newPage.element.querySelector(".pagedjs_page_content").insertAdjacentElement("afterbegin", elem);
    // }
    // } else {

    // if the page has no element with the fillpageclass
    if (!oldpage.element.querySelector(`.${fillPageClass}`)) return;

    // console.log("page without anything to push to the nextpage");
    // else push that element to next full page
    // console.log("pushing");
    // push it to the next page
    pushItToNextFullPage(
      pageFragment,
      oldpage,
      this.chunker,
      this.putThatFirst,
    );
    // }
  }

  afterRendered(pages) {
    document
      .querySelectorAll(".pagedjs_pagedjs-filler_page")
      .forEach((page) => {
        // console.log(page.dataset.pageNumber);
        // check if the page has only one figure
        if (page.querySelectorAll("figure").length > 1) {
          // check if there is a continued element, link them together

          let addedElements = page.querySelectorAll(".continuedContent");

          addedElements.forEach((addedEl) => {
            // only do it if there is a figure before
            console.log(addedEl);
            console.log(addedEl.previousElementSibling);
            if (addedEl.previousElementSibling) {
              addedEl.previousElementSibling.insertAdjacentElement(
                "beforeend",
                addedEl.querySelector("img"),
              );
              addedEl.remove();
            }
          });

          let figures = page.querySelectorAll("figure");

          figures.forEach((fig) => {
            // reorder images and captions
            // fig.classList.add("finishes2");
            // fig.insertAdjacentElement(
            //   "afterbegin",
            //   fig.querySelector("figcaption"),
            // );
          });
        } else if (page.querySelectorAll("figure")?.length < 2) {
          // try to show everything so the cal is good
          // page.querySelector(".pagedjs_page_content").style.overflow =
          //   "visible";
          //

          page.querySelector("figure")?.classList.add("finishes");

          // fix the height of elements
          //get page height,
          const pageHeight = page.querySelector(
            ".pagedjs_page_content",
          ).offsetHeight;

          const figureHeight = page.querySelector("figure")?.offsetHeight;

          //make image bigger if the page has more roome
          if (figureHeight > pageHeight) {
            let img = page.querySelector(" figure > img");

            let captionHeight = page.querySelector("figcaption")
              ? page.querySelector("figcaption").offsetHeight
              : 0;

            let finalHeight = Math.abs(pageHeight - captionHeight);
            img.style.height = `${finalHeight - 34}px`;
            img.style.maxWidth = "100%";
            img.style.width = "auto";
            img.style.objectFit = "contain";
            // img.style.objectPosition= "left bottom"
          } else if (figureHeight > pageHeight) {
            //reduce img.
            let img = page.querySelector("img");

            // img.style.height = `${img.offsetHeight}px`;
            let captionHeight = page.querySelector("figcaption")
              ? page.querySelector("figcaption").offsetHeight
              : 0;

            let finalHeight = Math.abs(pageHeight - captionHeight);

            img.style.height = `${finalHeight - 34}px`;
            // img.style.width = `auto`;
            img.style.margin = " 0 auto";
            img.style.display = "block";
            //check if the image is out of the page
            //reduce the image size
          }
        }

        //set all page filler as grid
        page.classList.add("page-grid");

        // put the title at the beginning on a single figure with multiple imges
        // page.querySelectorAll("figure").forEach((el) => {
        //   el.querySelectorAll("figcaption").forEach((el) => {
        //     el.closest("figure").insertAdjacentElement("afterbegin", el)
        //   });
        //
        //   if (
        //     el.querySelectorAll("img").length > 1 &&
        //     el.querySelector("figcaption")
        //   ) {
        //     el.insertAdjacentElement(
        //       "afterbegin",
        //       el.querySelector("figcaption"),
        //     );
        //   }
        // });
      });
  }
}

Paged.registerHandlers(fullpage);

async function pushItToNextFullPage(
  pageFragment,
  oldpage,
  chunker,
  // putthatfirst
) {
  //make sure you’re not checkintg the added page
  // script to run in the async finalize page
  if (oldpage.element.classList.contains("addedpage")) {
    return console.log("dont run the page twice");
  }

  // try to put to the next page, if there is not enough room, create a new page until the element is empty

  // the spread operator to get the list of all the element to push to the next page
  let elementsList = [...pageFragment.querySelectorAll(`.${fillPageClass}`)];

  while (elementsList.length > 0) {
    let element = elementsList[0];
    const elementHeight = await getHeightOfHiddenElement(element);
    const pageHeight = window.getComputedStyle(getLastPage()).height;

    if (elementHeight < pageHeight) {
      //then let the page continue and add it add the top of the next page, and dont do anything with the rest of the array
      // putthatfirst.push(element);
      // elementsList.shift();
      // // return;
      // continue;
    }

    // if (element.querySelectorAll("img").length > 1) {
    // do something if there is more than 1 image in there
    //
    // elementsList.shift();
    // continue;
    // }

    // check if the last page has a class of
    // pagedjs_pagedjs-filler_page

    // check if the last page
    const lastpage = getLastPage();

    //if the last page contains the classsName of an added page
    if (lastpage.classList.contains("pagedjs_pagedjs-filler_page")) {
      // console.log(lastpage);
      // a page exist with a filler class
      //show the element hidden before
      element.style.display = "block";
      // elementsList[0].style.position = "unset";

      // let see how much space is left
      const remainingSpace = await getRemainingSpaceOnPage(
        lastpage.querySelector(".pagedjs_page_content"),
      );
      const elementHeight = await getHeightOfHiddenElement(element);

      // console.log(remainingSpace, elementHeight, element);

      // if there is enough space, put the block
      if (remainingSpace > elementHeight && !element.dataset.continuedMain) {
        lastpage
          .querySelector(".pagedjs_page_content")
          .insertAdjacentElement("beforeend", element);
      } else {
        // otherwise create a new page
        const newpage = await chunker.addPage();

        // emulate the beforepage lyout to add the page to the flow pagedjs is waiting for
        await chunker.hooks.beforePageLayout.trigger(
          newpage,
          undefined,
          undefined,
          chunker,
        );

        // tell pagedjs that a new page has been set
        chunker.emit("page", newpage);

        newpage.element.classList.add("pagedjs_pagedjs-filler_page");
        newpage.element
          .querySelector(".pagedjs_page_content")
          .insertAdjacentElement("beforeend", element);
        // create a new page and fill it up
      }

      // create a new page and fill it up
    } else if (!lastpage.classList.contains("pagedjs_pagedjs-filler_page")) {
      //prepare the elements
      element.style.display = "block";
      element.style.position = "unset";

      const newpage = await chunker.addPage();
      newpage.element.classList.add("pagedjs_pagedjs-filler_page");
      newpage.element
        .querySelector(".pagedjs_page_content")
        .insertAdjacentElement("afterbegin", element);
      // create a new page and fill it up
    } else {
      console.log("there is no new page damn");
    }

    // do the next figure
    elementsList.shift();
  }
}

async function addWidthHeightToImg(content) {
  let imagePromises = [];
  let images = content.querySelectorAll("img");
  images.forEach((image) => {
    // you win a bit of time with this.
    if (image.width && image.height) return;
    let img = new Image();
    let resolve, reject;
    let imageLoaded = new Promise(function(r, x) {
      resolve = r;
      reject = x;
    });

    img.onload = function() {
      let height = img.naturalHeight;
      let width = img.naturalWidth;
      image.setAttribute("height", height);
      image.setAttribute("width", width);

      let ratio = width / height;
      if (ratio >= 1.7) {
        image.classList.add("landscape");
        image.parentNode.classList.add("fig-landscape");
      } else if (ratio <= 0.95) {
        image.classList.add("portrait");
        image.parentNode.classList.add("fig-portrait");
      } else if (ratio < 1.39 || ratio > 0.95) {
        image.classList.add("square");
        image.parentNode.classList.add("fig-square");
      }
      resolve();
    };
    img.onerror = function() {
      reject();
    };

    img.src = image.src;

    imagePromises.push(imageLoaded);
  });
  return Promise.all(imagePromises).catch((err) => {
    console.warn("err", err);
  });
}

// get the homothetic reduce height when reducing the width to see if the image can get in.
// with a baseline
function getHeight(originalWidth, originalHeight, reducedWidth, baseline) {
  const homotheticHeight = (reducedWidth * originalHeight) / originalWidth;
  const baselineHeight = Math.floor(homotheticHeight / baseline) * baseline;
  return Math.floor(baselineHeight);
}

function checkImageHeightRatio(imageHeight, availableSpace) {
  return imageHeight / availableSpace;
}

function resizeToBaseline(originalWidth, originalHeight, baselineGrid) {
  // Calculate the new height relative to the baseline grid
  const baselineHeight =
    Math.floor(originalHeight / baselineGrid) * baselineGrid;

  // Calculate the width reduction ratio based on the new height
  const reductionRatio = baselineHeight / originalHeight;

  // Calculate the new width proportionally
  const reducedWidth = originalWidth * reductionRatio;

  // Return the reduced dimensions as an object
  return {
    width: reducedWidth,
    height: baselineHeight,
  };
}

//check if the element is the the nessted first child of the page
function isNestedFirstChild(childElement, parentElement) {
  const firstChild = parentElement.firstElementChild;
  return firstChild === childElement;
}

// check if element is empty
function isElementEmpty(element) {
  return element.textContent.trim() === "";
}

// function to move thing to a new page
//
//
//

function getLastBlockOffsetBottom(element) {
  const children = element.children;
  let lastBlockOffsetBottom = 0;

  for (let i = children.length - 1; i >= 0; i--) {
    const child = children[i];

    // Check if the child is a block-level element (by checking it’s display)
    const display = window.getComputedStyle(child).display;
    // if (display === "block") {
    // Calculate the cumulative offset bottom
    const childOffsetBottom = child.offsetTop + child.offsetHeight;

    // Update the last block offset bottom if it's greater
    if (childOffsetBottom > lastBlockOffsetBottom) {
      lastBlockOffsetBottom = childOffsetBottom;
    }

    // Recursively check nested elements
    const nestedBlockOffsetBottom = getLastBlockOffsetBottom(child);
    if (nestedBlockOffsetBottom > lastBlockOffsetBottom) {
      lastBlockOffsetBottom = nestedBlockOffsetBottom;
    }
    // }
  }

  // return the offset bottom in px (number)
  return lastBlockOffsetBottom;
}
async function getRemainingSpaceOnPage(page, security = 40) {
  // console.log(page);
  const parentHeight = parseInt(window.getComputedStyle(page).height);
  let result = parentHeight - getLastBlockOffsetBottom(page) - security;
  // console.log("result", result);
  return result;
}

function tagFigure(content) {
  const figures = content.querySelectorAll("figure");
  figures.forEach((figure) => {
    //number of images
    figure.classList.add(`imgs-${figure.querySelectorAll("img").length}`);
    //
    // is there a title
    // TODO
    // figure.classList.add(`has-title`);
    // is there a caption

    if (
      !figure.querySelector("figcaption p") &&
      !figure.querySelector("label")
    ) {
      figure.classList.add("font");
      figure.classList.add("no-caption");
    }
    // how long is the caption
    // when the caption is less than 550, makes it one column

    let captionLength = 0;
    figure
      .querySelector("figcaption")
      ?.querySelectorAll("p, li")
      .forEach((el) => {
        captionLength = captionLength + el.textContent.length;
      });

    // console.log(figure.id, captionLength);
    if (captionLength < 800 && captionLength >= 500) {
      figure.classList.add("shortcaption");
    } else if (captionLength < 500) {
      figure.classList.add("shortercaption");
    } else {
      figure.classList.add("longcaption");
    }

    // if it’s a continued element, don’t show its name
    //
    if (!figure.querySelector("h3")) return;
    if (figure.querySelector(".ctn")) return;

    // figure.insertAdjacentHTML(
    //   "afterend",
    //   `<p class="sendtolink">See ${figure
    //     .querySelector("h3").innerHTML}</p>`
    // );
  });
}

function getLastPage() {
  // create a page variable which is the one pagedjs is working on.
  // use latest page to get the latest page
  let pages = document.querySelectorAll(".pagedjs_page");
  const latestPage = pages[pages.length - 1];
  return latestPage;
}

// WIDTH
// current image width / biggest img * 100%;

// this let us find a correspondances of width for the image we get.
// really experimental

async function calculateDesiredWidthPercentages(images) {
  // Find the width of the largest image
  let largestWidth = 0;
  images.forEach((image) => {
    const width = image.width;
    if (width > largestWidth) {
      largestWidth = width;
    }
  });

  // Calculate the desired width percentage for each image and set as data attribute
  // and set the width style
  images.forEach((image) => {
    if (image.closest("figcaption")) return;
    const widthPercentage = (image.width / largestWidth) * 100;
    image.setAttribute("data-desired-width", widthPercentage);
    // image.style.maxWidth = `${Math.round(widthPercentage)}%`;

    // console.log(widthPercentage);
    let maxwidth = "";
    if (widthPercentage <= 100) {
      maxwidth = "max-100";
    }
    if (widthPercentage <= 95) {
      maxwidth = "max-90";
    }
    if (widthPercentage <= 85) {
      maxwidth = "max-80";
    }
    if (widthPercentage <= 75) {
      maxwidth = "max-70";
    }
    if (widthPercentage <= 65) {
      maxwidth = "max-60";
    }
    if (widthPercentage <= 55) {
      maxwidth = "max-50";
    }
    if (widthPercentage <= 45) {
      maxwidth = "max-40";
    }
    if (widthPercentage <= 35) {
      maxwidth = "max-30";
    }
    if (widthPercentage <= 25) {
      maxwidth = "max-20";
    }
    if (widthPercentage <= 15) {
      maxwidth = "max-10";
    }

    if (maxwidth) image.classList.add(maxwidth);

    if (image.parentElement.tagName == "P") {
      //dont add max-width to image that are inside paragraphs. they should stick for 100% max
      // otherwise they will get hacked by the list
      // image.style.maxWidth = `${Math.round(widthPercentage) + 20}%`;
    }
  });

  // Return the modified array of image elements
  return images;
}

async function getHeightOfHiddenElement(element) {
  // find a page to render on
  let pageToRenderOn = getLastPage();

  let clone = element.cloneNode(true);
  clone.style.position = "absolute";
  clone.style.top = "0";
  clone.style.left = "0";
  clone.style.display = "block";

  // insert a clone of that element
  pageToRenderOn
    .querySelector(".pagedjs_page_content")
    .insertAdjacentElement("afterbegin", clone);

  // get margin in case there are somex
  let margins = clone.style.marginTop + clone.style.marginBottom;

  let elementHeight = clone.offsetHeight + margins;
  clone.remove();
  return elementHeight;
}

async function putOnPage(element, page) {
  // check if there is enough room to put an element on that page
  let availableSpace = getRemainingSpaceOnPage(page);

  // you can’t know the right width and hwgiht of the element because it’s never been rendered.
  // option1 : set a position: absolute; top: 0, leftz. 0, check its size
  // render a clone of the element to get the height of the element
  let elHeight = await getHeightOfHiddenElement(element);

  // if there is enough room on the page
  if (availableSpace > elHeight) {
    const clone = element.cloneNode(true);
    lastpage
      .querySelector(".pagedjs_page_content")
      .insertAdjacentElement("afterbegin", clone);
    return true;
  }
}
//# sourceURL=/Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/js/pagedjs-fill-page.js</script><script type="text/javascript">// a plugin to work with a baseline
//
// set --pagedjs-baseline-offset-color to see where things are used
// collapsing margin makes things almost impossible to work with as we’re trying to move things around, which add margins to element by default

// next step is finding hte first offset for any object to move the baseline too

// next step put an offset to the base line to define the starting point.
// allow multiple baselines based on the parent.
// check this d,*-['&% '] collaping margins that make working with added object a pain. so this now is a padding to our element. it’s easier to manage but on the long term we should definitely change that to have a proper pushing objectd.

class baselineProcess extends Paged.Handler {
  constructor(chunker, polisher, caller) {
    super(chunker, polisher, caller);
    this.snapToBaselineClass = "pagedjs-snap-to-baseline";
    this.await;
    this.snapToBaseline = [];
  }

  //find from the css the element you wanna have on the grid
  onDeclaration(declaration, dItem, dList, rule) {
    if (declaration.property == "--pagedjs-baseline-snap") {
      let sel = csstree.generate(rule.ruleNode.prelude);
      sel = sel.replace('[data-id="', "#");
      sel = sel.replace('"]', "");
      this.snapToBaseline.push([sel.split(","), declaration.value]);
    }
  }

  //find from the css the element you wanna have on the baseline
  afterParsed(content) {
    if (("this", this.snapToBaseline)) {
      this.snapToBaseline.forEach((entry) => {
        content.querySelectorAll(entry[0]).forEach((el) => {
          // console.log(entry[0])
          // console.log(entry[1].value)
          el.classList.add(this.snapToBaselineClass);
          el.dataset.snapBaseline = entry[1].value;
          el.style.lineHeight = entry[1].value
        });
      });
    }
  }

  // check the offset value
  renderNode(node) {
    // put all node on base line
    if (
      node &&
      node.nodeType == "1" &&
      node.classList.contains(this.snapToBaselineClass)
    ) {
    node.parentElement.classList.add("snapparent")
      startBaseline(node, parseInt(node.dataset.snapBaseline));
    }
  }
}
function startBaseline(element, baseline) {
  // snap element after specific element on the baseline grid.

  if (element && element.offsetTop) {
    // le nb de pixel entre l’élément et son parent
    const elementOffsetTop = element.offsetTop; // element à x pixel de haut

    // get the current line number
    const currentLine = Math.floor(element.offsetTop / baseline); // element à x pixel de haut

    // check the difference between the baseline place and the top offset
    const difference = elementOffsetTop % baseline; // element donc sur la ligne y

    // next line
    const nextLine = Math.ceil(element.offsetTop / baseline);

    // nextLine
    const nextLineOffset = nextLine * baseline;

    if (difference > 0) {
      element.style.paddingTop = baseline - difference + "px";
    }
  }
}

Paged.registerHandlers(baselineProcess);
//# sourceURL=/Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/js/pagedjs-plugin-baseline.js</script><style data-pagedjs-inserted-styles="true">
:root {
	--pagedjs-width: 8.5in;
	--pagedjs-height: 11in;
	--pagedjs-width-right: 8.5in;
	--pagedjs-height-right: 11in;
	--pagedjs-width-left: 8.5in;
	--pagedjs-height-left: 11in;
	--pagedjs-pagebox-width: 8.5in;
	--pagedjs-pagebox-height: 11in;
	--pagedjs-footnotes-height: 0mm;
	--pagedjs-margin-top: 1in;
	--pagedjs-margin-right: 1in;
	--pagedjs-margin-bottom: 1in;
	--pagedjs-margin-left: 1in;
	--pagedjs-padding-top: 0mm;
	--pagedjs-padding-right: 0mm;
	--pagedjs-padding-bottom: 0mm;
	--pagedjs-padding-left: 0mm;
	--pagedjs-border-top: 0mm;
	--pagedjs-border-right: 0mm;
	--pagedjs-border-bottom: 0mm;
	--pagedjs-border-left: 0mm;
	--pagedjs-bleed-top: 0mm;
	--pagedjs-bleed-right: 0mm;
	--pagedjs-bleed-bottom: 0mm;
	--pagedjs-bleed-left: 0mm;
	--pagedjs-bleed-right-top: 0mm;
	--pagedjs-bleed-right-right: 0mm;
	--pagedjs-bleed-right-bottom: 0mm;
	--pagedjs-bleed-right-left: 0mm;
	--pagedjs-bleed-left-top: 0mm;
	--pagedjs-bleed-left-right: 0mm;
	--pagedjs-bleed-left-bottom: 0mm;
	--pagedjs-bleed-left-left: 0mm;
	--pagedjs-crop-color: black;
	--pagedjs-crop-shadow: white;
	--pagedjs-crop-offset: 2mm;
	--pagedjs-crop-stroke: 1px;
	--pagedjs-cross-size: 5mm;
	--pagedjs-mark-cross-display: none;
	--pagedjs-mark-crop-display: none;
	--pagedjs-page-count: 0;
	--pagedjs-page-counter-increment: 1;
	--pagedjs-footnotes-count: 0;
	--pagedjs-column-gap-offset: 1000px;
}

@page {
	size: letter;
	margin: 0;
}

.pagedjs_sheet {
	box-sizing: border-box;
	width: var(--pagedjs-width);
	height: var(--pagedjs-height);
	overflow: hidden;
	position: relative;
	display: grid;
	grid-template-columns: [bleed-left] var(--pagedjs-bleed-left) [sheet-center] calc(var(--pagedjs-width) - var(--pagedjs-bleed-left) - var(--pagedjs-bleed-right)) [bleed-right] var(--pagedjs-bleed-right);
	grid-template-rows: [bleed-top] var(--pagedjs-bleed-top) [sheet-middle] calc(var(--pagedjs-height) - var(--pagedjs-bleed-top) - var(--pagedjs-bleed-bottom)) [bleed-bottom] var(--pagedjs-bleed-bottom);
}

.pagedjs_right_page .pagedjs_sheet {
	width: var(--pagedjs-width-right);
	height: var(--pagedjs-height-right);
	grid-template-columns: [bleed-left] var(--pagedjs-bleed-right-left) [sheet-center] calc(var(--pagedjs-width) - var(--pagedjs-bleed-right-left) - var(--pagedjs-bleed-right-right)) [bleed-right] var(--pagedjs-bleed-right-right);
	grid-template-rows: [bleed-top] var(--pagedjs-bleed-right-top) [sheet-middle] calc(var(--pagedjs-height) - var(--pagedjs-bleed-right-top) - var(--pagedjs-bleed-right-bottom)) [bleed-bottom] var(--pagedjs-bleed-right-bottom);
}

.pagedjs_left_page .pagedjs_sheet {
	width: var(--pagedjs-width-left);
	height: var(--pagedjs-height-left);
	grid-template-columns: [bleed-left] var(--pagedjs-bleed-left-left) [sheet-center] calc(var(--pagedjs-width) - var(--pagedjs-bleed-left-left) - var(--pagedjs-bleed-left-right)) [bleed-right] var(--pagedjs-bleed-left-right);
	grid-template-rows: [bleed-top] var(--pagedjs-bleed-left-top) [sheet-middle] calc(var(--pagedjs-height) - var(--pagedjs-bleed-left-top) - var(--pagedjs-bleed-left-bottom)) [bleed-bottom] var(--pagedjs-bleed-left-bottom);
}

.pagedjs_bleed {
	display: flex;
	align-items: center;
	justify-content: center;
	flex-wrap: nowrap;
	overflow: hidden;
}

.pagedjs_bleed-top {
	grid-column: bleed-left / -1;
	grid-row: bleed-top;
	flex-direction: row;
}

.pagedjs_bleed-bottom {
	grid-column: bleed-left / -1;
	grid-row: bleed-bottom;
	flex-direction: row;
}

.pagedjs_bleed-left {
	grid-column: bleed-left;
	grid-row: bleed-top / -1;
	flex-direction: column;
}

.pagedjs_bleed-right {
	grid-column: bleed-right;
	grid-row: bleed-top / -1;
	flex-direction: column;
}

.pagedjs_marks-crop {
	display: var(--pagedjs-mark-crop-display);
	flex-grow: 0;
	flex-shrink: 0;
	z-index: 9999999999;
}

.pagedjs_bleed-top .pagedjs_marks-crop:nth-child(1),
.pagedjs_bleed-bottom .pagedjs_marks-crop:nth-child(1) {
	width: calc(var(--pagedjs-bleed-left) - var(--pagedjs-crop-stroke));
	border-right: var(--pagedjs-crop-stroke) solid var(--pagedjs-crop-color);
	box-shadow: 1px 0px 0px 0px var(--pagedjs-crop-shadow);
}

.pagedjs_right_page .pagedjs_bleed-top .pagedjs_marks-crop:nth-child(1),
.pagedjs_right_page .pagedjs_bleed-bottom .pagedjs_marks-crop:nth-child(1) {
	width: calc(var(--pagedjs-bleed-right-left) - var(--pagedjs-crop-stroke));
}

.pagedjs_left_page .pagedjs_bleed-top .pagedjs_marks-crop:nth-child(1),
.pagedjs_left_page .pagedjs_bleed-bottom .pagedjs_marks-crop:nth-child(1) {
	width: calc(var(--pagedjs-bleed-left-left) - var(--pagedjs-crop-stroke));
}

.pagedjs_bleed-top .pagedjs_marks-crop:nth-child(3),
.pagedjs_bleed-bottom .pagedjs_marks-crop:nth-child(3) {
	width: calc(var(--pagedjs-bleed-right) - var(--pagedjs-crop-stroke));
	border-left: var(--pagedjs-crop-stroke) solid var(--pagedjs-crop-color);
	box-shadow: -1px 0px 0px 0px var(--pagedjs-crop-shadow);
}

.pagedjs_right_page .pagedjs_bleed-top .pagedjs_marks-crop:nth-child(3),
.pagedjs_right_page .pagedjs_bleed-bottom .pagedjs_marks-crop:nth-child(3) {
	width: calc(var(--pagedjs-bleed-right-right) - var(--pagedjs-crop-stroke));
}

.pagedjs_left_page .pagedjs_bleed-top .pagedjs_marks-crop:nth-child(3),
.pagedjs_left_page .pagedjs_bleed-bottom .pagedjs_marks-crop:nth-child(3) {
	width: calc(var(--pagedjs-bleed-left-right) - var(--pagedjs-crop-stroke));
}

.pagedjs_bleed-top .pagedjs_marks-crop {
	align-self: flex-start;
	height: calc(var(--pagedjs-bleed-top) - var(--pagedjs-crop-offset));
}

.pagedjs_right_page .pagedjs_bleed-top .pagedjs_marks-crop {
	height: calc(var(--pagedjs-bleed-right-top) - var(--pagedjs-crop-offset));
}

.pagedjs_left_page .pagedjs_bleed-top .pagedjs_marks-crop {
	height: calc(var(--pagedjs-bleed-left-top) - var(--pagedjs-crop-offset));
}

.pagedjs_bleed-bottom .pagedjs_marks-crop {
	align-self: flex-end;
	height: calc(var(--pagedjs-bleed-bottom) - var(--pagedjs-crop-offset));
}

.pagedjs_right_page .pagedjs_bleed-bottom .pagedjs_marks-crop {
	height: calc(var(--pagedjs-bleed-right-bottom) - var(--pagedjs-crop-offset));
}

.pagedjs_left_page .pagedjs_bleed-bottom .pagedjs_marks-crop {
	height: calc(var(--pagedjs-bleed-left-bottom) - var(--pagedjs-crop-offset));
}

.pagedjs_bleed-left .pagedjs_marks-crop:nth-child(1),
.pagedjs_bleed-right .pagedjs_marks-crop:nth-child(1) {
	height: calc(var(--pagedjs-bleed-top) - var(--pagedjs-crop-stroke));
	border-bottom: var(--pagedjs-crop-stroke) solid var(--pagedjs-crop-color);
	box-shadow: 0px 1px 0px 0px var(--pagedjs-crop-shadow);
}

.pagedjs_right_page .pagedjs_bleed-left .pagedjs_marks-crop:nth-child(1),
.pagedjs_right_page .pagedjs_bleed-right .pagedjs_marks-crop:nth-child(1) {
	height: calc(var(--pagedjs-bleed-right-top) - var(--pagedjs-crop-stroke));
}

.pagedjs_left_page .pagedjs_bleed-left .pagedjs_marks-crop:nth-child(1),
.pagedjs_left_page .pagedjs_bleed-right .pagedjs_marks-crop:nth-child(1) {
	height: calc(var(--pagedjs-bleed-left-top) - var(--pagedjs-crop-stroke));
}

.pagedjs_bleed-left .pagedjs_marks-crop:nth-child(3),
.pagedjs_bleed-right .pagedjs_marks-crop:nth-child(3) {
	height: calc(var(--pagedjs-bleed-bottom) - var(--pagedjs-crop-stroke));
	border-top: var(--pagedjs-crop-stroke) solid var(--pagedjs-crop-color);
	box-shadow: 0px -1px 0px 0px var(--pagedjs-crop-shadow);
}

.pagedjs_right_page .pagedjs_bleed-left .pagedjs_marks-crop:nth-child(3),
.pagedjs_right_page .pagedjs_bleed-right .pagedjs_marks-crop:nth-child(3) {
	height: calc(var(--pagedjs-bleed-right-bottom) - var(--pagedjs-crop-stroke));
}

.pagedjs_left_page .pagedjs_bleed-left .pagedjs_marks-crop:nth-child(3),
.pagedjs_left_page .pagedjs_bleed-right .pagedjs_marks-crop:nth-child(3) {
	height: calc(var(--pagedjs-bleed-left-bottom) - var(--pagedjs-crop-stroke));
}

.pagedjs_bleed-left .pagedjs_marks-crop {
	width: calc(var(--pagedjs-bleed-left) - var(--pagedjs-crop-offset));
	align-self: flex-start;
}

.pagedjs_right_page .pagedjs_bleed-left .pagedjs_marks-crop {
	width: calc(var(--pagedjs-bleed-right-left) - var(--pagedjs-crop-offset));
}

.pagedjs_left_page .pagedjs_bleed-left .pagedjs_marks-crop {
	width: calc(var(--pagedjs-bleed-left-left) - var(--pagedjs-crop-offset));
}

.pagedjs_bleed-right .pagedjs_marks-crop {
	width: calc(var(--pagedjs-bleed-right) - var(--pagedjs-crop-offset));
	align-self: flex-end;
}

.pagedjs_right_page .pagedjs_bleed-right .pagedjs_marks-crop {
	width: calc(var(--pagedjs-bleed-right-right) - var(--pagedjs-crop-offset));
}

.pagedjs_left_page .pagedjs_bleed-right .pagedjs_marks-crop {
	width: calc(var(--pagedjs-bleed-left-right) - var(--pagedjs-crop-offset));
}

.pagedjs_marks-middle {
	display: flex;
	flex-grow: 1;
	flex-shrink: 0;
	align-items: center;
	justify-content: center;
}

.pagedjs_marks-cross {
	display: var(--pagedjs-mark-cross-display);
	background-image: url(data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0idXRmLTgiPz48IURPQ1RZUEUgc3ZnIFBVQkxJQyAiLS8vVzNDLy9EVEQgU1ZHIDEuMS8vRU4iICJodHRwOi8vd3d3LnczLm9yZy9HcmFwaGljcy9TVkcvMS4xL0RURC9zdmcxMS5kdGQiPjxzdmcgdmVyc2lvbj0iMS4xIiBpZD0iTGF5ZXJfMSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayIgeD0iMHB4IiB5PSIwcHgiIHdpZHRoPSIzMi41MzdweCIgaGVpZ2h0PSIzMi41MzdweCIgdmlld0JveD0iMC4xMDQgMC4xMDQgMzIuNTM3IDMyLjUzNyIgZW5hYmxlLWJhY2tncm91bmQ9Im5ldyAwLjEwNCAwLjEwNCAzMi41MzcgMzIuNTM3IiB4bWw6c3BhY2U9InByZXNlcnZlIj48cGF0aCBmaWxsPSJub25lIiBzdHJva2U9IiNGRkZGRkYiIHN0cm9rZS13aWR0aD0iMy4zODkzIiBzdHJva2UtbWl0ZXJsaW1pdD0iMTAiIGQ9Ik0yOS45MzEsMTYuMzczYzAsNy40ODktNi4wNjgsMTMuNTYtMTMuNTU4LDEzLjU2Yy03LjQ4MywwLTEzLjU1Ny02LjA3Mi0xMy41NTctMTMuNTZjMC03LjQ4Niw2LjA3NC0xMy41NTQsMTMuNTU3LTEzLjU1NEMyMy44NjIsMi44MTksMjkuOTMxLDguODg3LDI5LjkzMSwxNi4zNzN6Ii8+PGxpbmUgZmlsbD0ibm9uZSIgc3Ryb2tlPSIjRkZGRkZGIiBzdHJva2Utd2lkdGg9IjMuMzg5MyIgc3Ryb2tlLW1pdGVybGltaXQ9IjEwIiB4MT0iMC4xMDQiIHkxPSIxNi4zNzMiIHgyPSIzMi42NDIiIHkyPSIxNi4zNzMiLz48bGluZSBmaWxsPSJub25lIiBzdHJva2U9IiNGRkZGRkYiIHN0cm9rZS13aWR0aD0iMy4zODkzIiBzdHJva2UtbWl0ZXJsaW1pdD0iMTAiIHgxPSIxNi4zNzMiIHkxPSIwLjEwNCIgeDI9IjE2LjM3MyIgeTI9IjMyLjY0MiIvPjxwYXRoIGZpbGw9Im5vbmUiIHN0cm9rZT0iI0ZGRkZGRiIgc3Ryb2tlLXdpZHRoPSIzLjM4OTMiIHN0cm9rZS1taXRlcmxpbWl0PSIxMCIgZD0iTTI0LjUwOCwxNi4zNzNjMCw0LjQ5Ni0zLjYzOCw4LjEzNS04LjEzNSw4LjEzNWMtNC40OTEsMC04LjEzNS0zLjYzOC04LjEzNS04LjEzNWMwLTQuNDg5LDMuNjQ0LTguMTM1LDguMTM1LTguMTM1QzIwLjg2OSw4LjIzOSwyNC41MDgsMTEuODg0LDI0LjUwOCwxNi4zNzN6Ii8+PHBhdGggZmlsbD0ibm9uZSIgc3Ryb2tlPSIjMDAwMDAwIiBzdHJva2Utd2lkdGg9IjAuNjc3OCIgc3Ryb2tlLW1pdGVybGltaXQ9IjEwIiBkPSJNMjkuOTMxLDE2LjM3M2MwLDcuNDg5LTYuMDY4LDEzLjU2LTEzLjU1OCwxMy41NmMtNy40ODMsMC0xMy41NTctNi4wNzItMTMuNTU3LTEzLjU2YzAtNy40ODYsNi4wNzQtMTMuNTU0LDEzLjU1Ny0xMy41NTRDMjMuODYyLDIuODE5LDI5LjkzMSw4Ljg4NywyOS45MzEsMTYuMzczeiIvPjxsaW5lIGZpbGw9Im5vbmUiIHN0cm9rZT0iIzAwMDAwMCIgc3Ryb2tlLXdpZHRoPSIwLjY3NzgiIHN0cm9rZS1taXRlcmxpbWl0PSIxMCIgeDE9IjAuMTA0IiB5MT0iMTYuMzczIiB4Mj0iMzIuNjQyIiB5Mj0iMTYuMzczIi8+PGxpbmUgZmlsbD0ibm9uZSIgc3Ryb2tlPSIjMDAwMDAwIiBzdHJva2Utd2lkdGg9IjAuNjc3OCIgc3Ryb2tlLW1pdGVybGltaXQ9IjEwIiB4MT0iMTYuMzczIiB5MT0iMC4xMDQiIHgyPSIxNi4zNzMiIHkyPSIzMi42NDIiLz48cGF0aCBkPSJNMjQuNTA4LDE2LjM3M2MwLDQuNDk2LTMuNjM4LDguMTM1LTguMTM1LDguMTM1Yy00LjQ5MSwwLTguMTM1LTMuNjM4LTguMTM1LTguMTM1YzAtNC40ODksMy42NDQtOC4xMzUsOC4xMzUtOC4xMzVDMjAuODY5LDguMjM5LDI0LjUwOCwxMS44ODQsMjQuNTA4LDE2LjM3MyIvPjxsaW5lIGZpbGw9Im5vbmUiIHN0cm9rZT0iI0ZGRkZGRiIgc3Ryb2tlLXdpZHRoPSIwLjY3NzgiIHN0cm9rZS1taXRlcmxpbWl0PSIxMCIgeDE9IjguMjM5IiB5MT0iMTYuMzczIiB4Mj0iMjQuNTA4IiB5Mj0iMTYuMzczIi8+PGxpbmUgZmlsbD0ibm9uZSIgc3Ryb2tlPSIjRkZGRkZGIiBzdHJva2Utd2lkdGg9IjAuNjc3OCIgc3Ryb2tlLW1pdGVybGltaXQ9IjEwIiB4MT0iMTYuMzczIiB5MT0iOC4yMzkiIHgyPSIxNi4zNzMiIHkyPSIyNC41MDgiLz48L3N2Zz4=);
  background-repeat: no-repeat;
  background-position: 50% 50%;
  background-size: var(--pagedjs-cross-size);

  z-index: 2147483647;
	width: var(--pagedjs-cross-size);
	height: var(--pagedjs-cross-size);
}

.pagedjs_pagebox {
	box-sizing: border-box;
	width: var(--pagedjs-pagebox-width);
	height: var(--pagedjs-pagebox-height);
	position: relative;
	display: grid;
	grid-template-columns: [left] var(--pagedjs-margin-left) [center] calc(var(--pagedjs-pagebox-width) - var(--pagedjs-margin-left) - var(--pagedjs-margin-right)) [right] var(--pagedjs-margin-right);
	grid-template-rows: [header] var(--pagedjs-margin-top) [page] calc(var(--pagedjs-pagebox-height) - var(--pagedjs-margin-top) - var(--pagedjs-margin-bottom)) [footer] var(--pagedjs-margin-bottom);
	grid-column: sheet-center;
	grid-row: sheet-middle;
}

.pagedjs_pagebox * {
	box-sizing: border-box;
}

.pagedjs_margin-top {
	width: calc(var(--pagedjs-pagebox-width) - var(--pagedjs-margin-left) - var(--pagedjs-margin-right));
	height: var(--pagedjs-margin-top);
	grid-column: center;
	grid-row: header;
	flex-wrap: nowrap;
	display: grid;
	grid-template-columns: repeat(3, 1fr);
	grid-template-rows: 100%;
}

.pagedjs_margin-top-left-corner-holder {
	width: var(--pagedjs-margin-left);
	height: var(--pagedjs-margin-top);
	display: flex;
	grid-column: left;
	grid-row: header;
}

.pagedjs_margin-top-right-corner-holder {
	width: var(--pagedjs-margin-right);
	height: var(--pagedjs-margin-top);
	display: flex;
	grid-column: right;
	grid-row: header;
}

.pagedjs_margin-top-left-corner {
	width: var(--pagedjs-margin-left);
}

.pagedjs_margin-top-right-corner {
	width: var(--pagedjs-margin-right);
}

.pagedjs_margin-right {
	height: calc(var(--pagedjs-pagebox-height) - var(--pagedjs-margin-top) - var(--pagedjs-margin-bottom));
	width: var(--pagedjs-margin-right);
	right: 0;
	grid-column: right;
	grid-row: page;
	display: grid;
	grid-template-rows: repeat(3, 33.3333%);
	grid-template-columns: 100%;
}

.pagedjs_margin-bottom {
	width: calc(var(--pagedjs-pagebox-width) - var(--pagedjs-margin-left) - var(--pagedjs-margin-right));
	height: var(--pagedjs-margin-bottom);
	grid-column: center;
	grid-row: footer;
	display: grid;
	grid-template-columns: repeat(3, 1fr);
	grid-template-rows: 100%;
}

.pagedjs_margin-bottom-left-corner-holder {
	width: var(--pagedjs-margin-left);
	height: var(--pagedjs-margin-bottom);
	display: flex;
	grid-column: left;
	grid-row: footer;
}

.pagedjs_margin-bottom-right-corner-holder {
	width: var(--pagedjs-margin-right);
	height: var(--pagedjs-margin-bottom);
	display: flex;
	grid-column: right;
	grid-row: footer;
}

.pagedjs_margin-bottom-left-corner {
	width: var(--pagedjs-margin-left);
}

.pagedjs_margin-bottom-right-corner {
	width: var(--pagedjs-margin-right);
}



.pagedjs_margin-left {
	height: calc(var(--pagedjs-pagebox-height) - var(--pagedjs-margin-top) - var(--pagedjs-margin-bottom));
	width: var(--pagedjs-margin-left);
	grid-column: left;
	grid-row: page;
	display: grid;
	grid-template-rows: repeat(3, 33.33333%);
	grid-template-columns: 100%;
}

.pagedjs_pages .pagedjs_pagebox .pagedjs_margin:not(.hasContent) {
	visibility: hidden;
}

.pagedjs_pagebox > .pagedjs_area {
	grid-column: center;
	grid-row: page;
	width: 100%;
	height: 100%;
	padding: var(--pagedjs-padding-top) var(--pagedjs-padding-right) var(--pagedjs-padding-bottom) var(--pagedjs-padding-left);
	border-top: var(--pagedjs-border-top);
	border-right: var(--pagedjs-border-right);
	border-bottom: var(--pagedjs-border-bottom);
	border-left: var(--pagedjs-border-left);
}

.pagedjs_pagebox > .pagedjs_area > .pagedjs_page_content {
	width: 100%;
	height: calc(100% - var(--pagedjs-footnotes-height));
	position: relative;
	column-fill: auto;
}

.pagedjs_pagebox > .pagedjs_area > .pagedjs_page_content > div {
	height: inherit;
}

.pagedjs_pagebox > .pagedjs_area > .pagedjs_footnote_area {
	position: relative;
	overflow: hidden;
	height: var(--pagedjs-footnotes-height);
	display: flex;
    justify-content: flex-end;
    flex-flow: column;
}

.pagedjs_pagebox > .pagedjs_area > .pagedjs_footnote_area > .pagedjs_footnote_content {
	overflow: hidden;
}

.pagedjs_pagebox > .pagedjs_area > .pagedjs_footnote_area > .pagedjs_footnote_inner_content {
	overflow: hidden;
}

.pagedjs_area [data-footnote-call] {
	all: unset;
	counter-increment: footnote;
}

.pagedjs_area [data-split-from] {
	counter-increment: unset;
	counter-reset: unset;
}

[data-footnote-call]::after {
	vertical-align: super;
	font-size: 65%;
	line-height: normal;
	content: counter(footnote);
}

@supports ( font-variant-position: super ) {
	[data-footnote-call]::after {
		vertical-align: baseline;
		font-size: 100%;
		line-height: inherit;
		font-variant-position: super;
	}
}

.pagedjs_footnote_empty {
	display: none;
}

.pagedjs_area [data-split-from] {
	counter-increment: unset;
	counter-reset: unset;
}

[data-footnote-marker] {
	text-indent: 0;
	display: list-item;
	list-style-position: inside;
}

[data-footnote-marker][data-split-from] {
	list-style: none;
}

[data-footnote-marker]:not([data-split-from]) {
	counter-increment: footnote-marker;
}

[data-footnote-marker]::marker {
	content: counter(footnote-marker) ". ";
}

[data-footnote-marker][data-split-from]::marker {
	content: unset;
}

.pagedjs_area .pagedjs_footnote_inner_content [data-note-display="inline"] {
 	display: inline;
}

.pagedjs_page {
	counter-increment: page var(--pagedjs-page-counter-increment);
	width: var(--pagedjs-width);
	height: var(--pagedjs-height);
}

.pagedjs_page.pagedjs_right_page {
	width: var(--pagedjs-width-right);
	height: var(--pagedjs-height-right);
}

.pagedjs_page.pagedjs_left_page {
	width: var(--pagedjs-width-left);
	height: var(--pagedjs-height-left);
}

.pagedjs_pages {
	counter-reset: pages var(--pagedjs-page-count) footnote var(--pagedjs-footnotes-count) footnote-marker var(--pagedjs-footnotes-count);
}

.pagedjs_pagebox .pagedjs_margin-top-left-corner,
.pagedjs_pagebox .pagedjs_margin-top-right-corner,
.pagedjs_pagebox .pagedjs_margin-bottom-left-corner,
.pagedjs_pagebox .pagedjs_margin-bottom-right-corner,
.pagedjs_pagebox .pagedjs_margin-top-left,
.pagedjs_pagebox .pagedjs_margin-top-right,
.pagedjs_pagebox .pagedjs_margin-bottom-left,
.pagedjs_pagebox .pagedjs_margin-bottom-right,
.pagedjs_pagebox .pagedjs_margin-top-center,
.pagedjs_pagebox .pagedjs_margin-bottom-center,
.pagedjs_pagebox .pagedjs_margin-top-center,
.pagedjs_pagebox .pagedjs_margin-bottom-center,
.pagedjs_margin-right-middle,
.pagedjs_margin-left-middle  {
	display: flex;
	align-items: center;
}

.pagedjs_margin-right-top,
.pagedjs_margin-left-top  {
	display: flex;
	align-items: flex-top;
}


.pagedjs_margin-right-bottom,
.pagedjs_margin-left-bottom  {
	display: flex;
	align-items: flex-end;
}



/*
.pagedjs_pagebox .pagedjs_margin-top-center,
.pagedjs_pagebox .pagedjs_margin-bottom-center {
	height: 100%;
	display: none;
	align-items: center;
	flex: 1 0 33%;
	margin: 0 auto;
}

.pagedjs_pagebox .pagedjs_margin-top-left-corner,
.pagedjs_pagebox .pagedjs_margin-top-right-corner,
.pagedjs_pagebox .pagedjs_margin-bottom-right-corner,
.pagedjs_pagebox .pagedjs_margin-bottom-left-corner {
	display: none;
	align-items: center;
}

.pagedjs_pagebox .pagedjs_margin-left-top,
.pagedjs_pagebox .pagedjs_margin-right-top {
	display: none;
	align-items: flex-start;
}

.pagedjs_pagebox .pagedjs_margin-right-middle,
.pagedjs_pagebox .pagedjs_margin-left-middle {
	display: none;
	align-items: center;
}

.pagedjs_pagebox .pagedjs_margin-left-bottom,
.pagedjs_pagebox .pagedjs_margin-right-bottom {
	display: none;
	align-items: flex-end;
}
*/

.pagedjs_pagebox .pagedjs_margin-top-left,
.pagedjs_pagebox .pagedjs_margin-top-right-corner,
.pagedjs_pagebox .pagedjs_margin-bottom-left,
.pagedjs_pagebox .pagedjs_margin-bottom-right-corner { text-align: left; }

.pagedjs_pagebox .pagedjs_margin-top-left-corner,
.pagedjs_pagebox .pagedjs_margin-top-right,
.pagedjs_pagebox .pagedjs_margin-bottom-left-corner,
.pagedjs_pagebox .pagedjs_margin-bottom-right { text-align: right; }

.pagedjs_pagebox .pagedjs_margin-top-center,
.pagedjs_pagebox .pagedjs_margin-bottom-center,
.pagedjs_pagebox .pagedjs_margin-left-top,
.pagedjs_pagebox .pagedjs_margin-left-middle,
.pagedjs_pagebox .pagedjs_margin-left-bottom,
.pagedjs_pagebox .pagedjs_margin-right-top,
.pagedjs_pagebox .pagedjs_margin-right-middle,
.pagedjs_pagebox .pagedjs_margin-right-bottom { text-align: center; }

.pagedjs_pages .pagedjs_margin .pagedjs_margin-content {
	width: 100%;
}

.pagedjs_pages .pagedjs_margin-left .pagedjs_margin-content::after,
.pagedjs_pages .pagedjs_margin-top .pagedjs_margin-content::after,
.pagedjs_pages .pagedjs_margin-right .pagedjs_margin-content::after,
.pagedjs_pages .pagedjs_margin-bottom .pagedjs_margin-content::after {
	display: block;
}

.pagedjs_pages > .pagedjs_page > .pagedjs_sheet > .pagedjs_pagebox > .pagedjs_area > div [data-split-to] {
	margin-bottom: unset;
	padding-bottom: unset;
}

.pagedjs_pages > .pagedjs_page > .pagedjs_sheet > .pagedjs_pagebox > .pagedjs_area > div [data-split-from] {
	text-indent: unset;
	margin-top: unset;
	padding-top: unset;
	initial-letter: unset;
}

.pagedjs_pages > .pagedjs_page > .pagedjs_sheet > .pagedjs_pagebox > .pagedjs_area > div [data-split-from] > *::first-letter,
.pagedjs_pages > .pagedjs_page > .pagedjs_sheet > .pagedjs_pagebox > .pagedjs_area > div [data-split-from]::first-letter {
	color: unset;
	font-size: unset;
	font-weight: unset;
	font-family: unset;
	color: unset;
	line-height: unset;
	float: unset;
	padding: unset;
	margin: unset;
}

.pagedjs_pages > .pagedjs_page > .pagedjs_sheet > .pagedjs_pagebox > .pagedjs_area > div [data-split-to]:not([data-footnote-call]):after,
.pagedjs_pages > .pagedjs_page > .pagedjs_sheet > .pagedjs_pagebox > .pagedjs_area > div [data-split-to]:not([data-footnote-call])::after {
	content: unset;
}

.pagedjs_pages > .pagedjs_page > .pagedjs_sheet > .pagedjs_pagebox > .pagedjs_area > div [data-split-from]:not([data-footnote-call]):before,
.pagedjs_pages > .pagedjs_page > .pagedjs_sheet > .pagedjs_pagebox > .pagedjs_area > div [data-split-from]:not([data-footnote-call])::before {
	content: unset;
}

.pagedjs_pages > .pagedjs_page > .pagedjs_sheet > .pagedjs_pagebox > .pagedjs_area > div li[data-split-from]:first-of-type {
	list-style: none;
}

/*
[data-page]:not([data-split-from]),
[data-break-before="page"]:not([data-split-from]),
[data-break-before="always"]:not([data-split-from]),
[data-break-before="left"]:not([data-split-from]),
[data-break-before="right"]:not([data-split-from]),
[data-break-before="recto"]:not([data-split-from]),
[data-break-before="verso"]:not([data-split-from])
{
	break-before: column;
}

[data-page]:not([data-split-to]),
[data-break-after="page"]:not([data-split-to]),
[data-break-after="always"]:not([data-split-to]),
[data-break-after="left"]:not([data-split-to]),
[data-break-after="right"]:not([data-split-to]),
[data-break-after="recto"]:not([data-split-to]),
[data-break-after="verso"]:not([data-split-to])
{
	break-after: column;
}
*/

.pagedjs_clear-after::after {
	content: none !important;
}

[data-align-last-split-element='justify'] {
	text-align-last: justify;
}


@media print {
	html {
		width: 100%;
		height: 100%;
		-webkit-print-color-adjust: exact;
		print-color-adjust: exact;
	}
	body {
		margin: 0;
		padding: 0;
		width: 100% !important;
		height: 100% !important;
		min-width: 100%;
		max-width: 100%;
		min-height: 100%;
		max-height: 100%;
	}
	.pagedjs_pages {
		width: auto;
		display: block !important;
		transform: none !important;
		height: 100% !important;
		min-height: 100%;
		max-height: 100%;
		overflow: visible;
	}
	.pagedjs_page {
		margin: 0;
		padding: 0;
		max-height: 100%;
		min-height: 100%;
		height: 100% !important;
		page-break-after: always;
		break-after: page;
	}
	.pagedjs_sheet {
		margin: 0;
		padding: 0;
		max-height: 100%;
		min-height: 100%;
		height: 100% !important;
	}
}
</style><style></style><style data-pagedjs-inserted-styles="true">@font-face{font-family:"Noto Serif";src:url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/fonts/NotoSerif-Regular.ttf);font-style:normal;font-weight:400}@font-face{font-family:"Noto Serif";src:url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/fonts/NotoSerif-Italic.ttf);font-style:italic;font-weight:400}@font-face{font-family:"Noto Serif";src:url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/fonts/NotoSerif-Bold.ttf);font-style:normal;font-weight:600}@font-face{font-family:"Noto Serif";src:url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/fonts/NotoSerif-BoldItalic.ttf);font-style:italic;font-weight:600}@font-face{font-family:"Noto Sans";src:url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/fonts/NotoSans-Regular.ttf);font-style:normal;font-weight:400}@font-face{font-family:"Noto Sans";src:url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/fonts/NotoSans-Italic.ttf);font-style:italic;font-weight:400}@font-face{font-family:"Noto Sans";src:url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/fonts/NotoSans-Bold.ttf);font-style:normal;font-weight:600}@font-face{font-family:"Noto Sans";src:url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/fonts/NotoSans-BoldItalic.ttf);font-style:italic;font-weight:600}</style><style data-pagedjs-inserted-styles="true">:root{--color-main: var(--color-elife-blue);--color-main-light: var(--color-elife-blue-light);--color-sub: var(--color-elife-grey);--color-sub-light: var(--color-elife-grey-light);--color-elife-blue: rgb(8, 122, 204);--color-elife-blue-light: rgb(159, 215, 255);--color-elife-grey: rgb(118, 118, 118);--color-elife-grey-light: rgb(240, 240, 240);--color-elife-black: rgb(37, 42, 54);--font-sans: "Noto Sans", sans-serif;--font-serif: "Noto Serif"}</style><style data-pagedjs-inserted-styles="true">@media screen,pagedjs-ignore{.pagedjs_area{--pagedjs-baseline-offset-color: red;--pagedjs-baseline-color: transparent;--pagedjs-baseline-lineheight: 17px;--pagedjs-baseline-position: 13px;background:linear-gradient(transparent 0%,transparent calc(var(--pagedjs-baseline-lineheight) - 1px),var(--pagedjs-baseline-color) calc(var(--pagedjs-baseline-lineheight) - 1px),var(--pagedjs-baseline-color) var(--pagedjs-baseline-lineheight)),transparent;background-size:100% var(--pagedjs-baseline-lineheight);background-repeat:repeat-y;background-position-y:var(--pagedjs-baseline-position)}}article{--offset: 180px}article>section{margin-left:var(--offset)}[data-id="logo"]{position:running(logo);display:block;width:106px;height:40px;text-align:left;position:static;top:auto;left:auto;margin-left:30px}[data-id="logo"] img{width:100%!important;height:100%!important;display:block}.pagedjs_margin-top-left .pagedjs_margin-content [data-id="logo"]{position:absolute!important;top:12px;left:15px}[data-id="right-header-links"]{position:running(right-header-links);display:block;width:100%;text-align:right;position:static;top:auto;right:auto}[data-id="right-header-links"]>a,[data-id="right-header-links"]>span{display:inline-block;margin-left:5px;vertical-align:middle}.runninghead{font-family:var(--font-sans);color:var(--color-elife-grey);border-top:1px solid var(--color-elife-grey);bottom:0mm;line-height:1;left:50px;right:50px;font-size:12px;margin-top:0;padding-top:12px;display:flex;align-items:space-between}.runninghead a{color:var(--color-elife-blue)}.runninghead .counter{margin-left:auto}.runninghead .counter::after{content:counter(page) " of " counter(pages)}.article-flag-list{top:32px;right:50px;left:auto;display:flex;align-items:center;gap:5px;list-style:none;padding:0;margin:0;z-index:1000;font-size:10px}.article-flag-list__item{margin-left:0!important;padding-right:0!important}.article-flag-list__item:not(:last-child)::after{content:"|";display:inline-block;padding:0 10px;margin:0;color:var(--color-elife-grey)}.article-flag__link{color:var(--color-elife-blue);text-decoration:none;white-space:nowrap}.pagedjs_page .pagedjs_margin-top-left{text-align:left}.pagedjs_page .pagedjs_margin-top-right{}.pagedjs_page .pagedjs_margin-top-left>.pagedjs_margin-content>*{display:block}.pagedjs_page .pagedjs_margin-top-left>.pagedjs_margin-content::after{content:element-ident(logo)}.pagedjs_page .pagedjs_margin-top-right>.pagedjs_margin-content>*{display:block}.pagedjs_page .pagedjs_margin-top-right>.pagedjs_margin-content::after{content:""}.pagedjs_page{--pagedjs-margin-top:75px;--pagedjs-margin-right:50px;--pagedjs-margin-left:50px;--pagedjs-margin-bottom:52px;--pagedjs-pagebox-width:8.5in;--pagedjs-pagebox-height:11in}.pagedjs_page.pagedjs_first_page .pagedjs_margin-top-right{text-align:right}.pagedjs_page.pagedjs_first_page .pagedjs_margin-top-right>.pagedjs_margin-content>*{display:block}.pagedjs_page.pagedjs_first_page .pagedjs_margin-top-right>.pagedjs_margin-content::after{content:element-ident(right-header-links)}.pagedjs_page.pagedjs_first_page{}:root{--pagedjs-width:8.5in;--pagedjs-height:11in;--pagedjs-width-right:8.5in;--pagedjs-height-right:11in;--pagedjs-width-left:8.5in;--pagedjs-height-left:11in}@page{size:letter;margin:0px;padding:0px;padding:0px}</style><style data-pagedjs-inserted-styles="true">body{font-family:var(--font-serif);font-size:12px;line-height:17px}[data-id="article-content"] :is(p, li){--pagedjs-baseline-snap: 17px;line-height:17px;font-size:0.95em;margin:0;widows:3;orphans:3}[data-id="article-content"] li p{font-size:1em}[data-id="article-content"] p,[data-id="peer-reviews"] p{margin-bottom:5px}blockquote{margin-top:8px;margin-bottom:8px;border-left:1px solid var(--color-sub);margin-left:0;padding-left:3ch}.author-response__body blockquote{font-style:italic;font-family:var(--font-sans)}dl,dd,dt{font-size:0.95em;line-height:17px}:is(p) :is(span, a, em, sup, sub){line-height:0}a{color:inherit;text-decoration:none}section a{color:var(--color-main)}h1,h2,h3,h4,h5,h6{font-family:var(--font-sans);break-inside:avoid;text-wrap:wrap;font-weight:600}.pagedjs_page_content article>section:first-child>section:first-child>h1:first-child,.pagedjs_page_content article>section:first-child>section:first-child>section:first-child>h2:first-child,.pagedjs_page_content article>section:first-child>section:first-child>section:first-child>section:first-child>h3:first-child{margin-top:0!important}h1{font-size:20px;margin-top:10px;margin-bottom:6px;line-height:28px;color:var(--color-main)}h1.title{color:var(--color-main)}h2{font-size:16px;margin-top:8px;margin-bottom:6px;line-height:24px}h3{margin-top:8px;margin-bottom:6px;font-size:14px;color:var(--color-body)}h4{font-size:12px;margin-top:6px;margin-bottom:6px}h5{font-size:11px;margin-top:6px;margin-bottom:6px}[data-id="article-content"]>ol{list-style-position:oustide;padding:0}[data-id="article-content"]>ol li{margin-bottom:1em}[data-id="article-content"] ol ol{list-style-position:oustide;margin:0;padding:0 3ch}[data-id="article-content"] li p{margin:0;text-align:left}ol.list-roman-lower{list-style-type:lower-roman}ol.list-roman-upper{list-style-type:upper-roman}ol.list-alpha-lower{list-style-type:lower-alpha}ol.list-alpha-upper{list-style-type:upper-alpha}code,pre{max-width:530px;word-wrap:break-word;white-space:pre-wrap}.statement,.boxed-text{margin-top:20px;margin-bottom:15px;border-left:5px solid var(--color-elife-blue-light);padding-left:20px}</style><style data-pagedjs-inserted-styles="true">.article-flag-list{display:flex;justify-content:flex-end;list-style:none;padding:0;margin:0}.article-flag-list__item{margin:0;padding:0}.article-flag-list__item::after{content:" | ";color:var(--color-secondary)}.article-flag-list__item:last-child::after{content:none}.article-flag__link{text-decoration:none}[data-id="header"]{margin-left:var(--offset);max-width:90%}[data-id="header"] .title{font-size:2.3em;line-height:1.2;padding-top:0}.descriptors__icon a{padding-left:3ch}.descriptors__icon a::before{position:absolute;width:2ch;margin-left:-3ch;margin-right:2ch;height:auto;display:inline-block}.descriptors__icon .hidden{display:none}.pagedjs_first_page .article-flag-list{display:none}.header-icon.oa-icon::before{content:url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/assets/oa.svg);display:block;width:20px;height:20px}.header-icon.cc-icon::before{content:url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/assets/cc.svg);display:block;width:20px;height:20px}[data-id="sideSection"]{font-family:var(--font-sans);width:calc(var(--offset) - 5ch);position:absolute;left:calc(var(--offset) * -1);font-size:0.9em}[data-id="sideSection"] h2{font-size:inherit;line-height:inherit;font-family:inherit;color:var(--color-main)}[data-id="sideSection"] p{hyphens:none}[data-id="margin-notes"]{font-size:0.8em}[data-id="margin-notes"] h3{font-size:inherit;line-height:inherit;font-family:inherit;color:var(--color-secondary);margin-bottom:0.5em}[data-id="margin-notes"] p{hyphens:none;margin-top:5px;margin-bottom:5px}[data-id="margin-notes"] p:has(.email-icon){margin-bottom:0}[data-id="margin-notes"] .copyright{color:grey}[data-id="margin-notes"] .email-icon::before{content:"";display:inline-block;height:9px;width:9px;margin-right:0.2em;background-image:url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/assets/email.svg);background-size:contain;background-repeat:no-repeat}.author-notes ul{margin-top:0}.author-notes li a{overflow-wrap:anywhere}.article-status__heading{margin-bottom:0}.article-status__link{margin-top:0;display:block}.modal-container,.review-timeline__link{display:none}[data-id="sideSection"] dd{padding-left:0;margin-left:0;margin-bottom:17px}[data-id="sideSection"] dt{font-weight:800}</style><style data-pagedjs-inserted-styles="true">@import url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/modules/authors-and-institutions.css);@import url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/modules/article-content.css);@import url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/modules/figures.css);@import url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/modules/assesment.css);@import url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/modules/references.css);@import url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/modules/timeline.css);body{--font-family-primary: '__Noto_Sans_d69442', '__Noto_Sans_Fallback_d69442', arial, helvetica, sans-serif;--font-family-secondary: '__Noto_Serif_288e51', '__Noto_Serif_Fallback_288e51', serif}.fix-phi{font-family:'NotoSerifForce';font-variant-ligatures:none;text-transform:uppercase;font-style:italic}.onlynumbers{display:none}.figure:not(.tofill){max-width:100%}.figure:not(.tofill) img .imageonly{width:100%!important;height:100%!important;object-fit:contain}.img-in-link{float:left;margin-left:calc(var(--offset) * -1)}.pagedjs_pages{--pagedjs-baseline-color: transparent !important}[data-id="abstract"] .widowfix{display:none}ol{display:block}ol>li{break-inside:avoid}.insideurl a{color:inherit;text-decoration:none;border-bottom:none;border-bottom:0}.insideurl a::after{}.insideurl .hidden{display:inline}.movedfig img{display:block}.nottofill{margin:1em 0}.nottofill img{display:block;margin:0.8em auto}:is(.prettified-link, .prettified-symbol){display:none}p{widows:2;orphans:2}.tofill,figure{position:fill-page}.figure[data-split-from]{border-top:0!important}.descriptors__icon--cc a{border-bottom:0;color:inherit}.descriptors__icon--cc a .infourl{display:none}.descriptors__icon--cc a .hidden{display:inline}[data-id="reviews"] .descriptors__identifiers{list-style-type:none;padding-left:0;margin-top:0;font-family:var(--font-sans);margin-bottom:2em}.descriptors__identifiers{list-style-type:none;padding:0;font-family:var(--font-sans);margin-bottom:0}.pagebreak{}[data-id="authors"] ul{list-style-type:none;padding-left:0}:root{--color-primary: #087acc;--color-primary-dark: #0769b0;--color-text: #212121;--color-text-secondary: #757575;--color-button: #edeff4;--color-background: #f7f7f7;--divider: #e0e0e0;--link-decoration: #212121;--link-decoration-secondary: #6d6d6d;--border: 1px solid var(--divider);--font-family-primary: noto sans, arial, helvetica, sans-serif;--font-family-secondary: noto serif, serif;--grid-gutter: 24px;--site-spacing: 24px;--site-spacing-large: 48px;--base-font-size: 16
}:after,:before{box-sizing:border-box}.visuallyhidden{border:0;clip:rect(0 0 0 0);height:1px;margin:-1px;overflow:hidden;padding:0;position:absolute;-webkit-user-select:none;user-select:none;width:1px}</style><style data-pagedjs-inserted-styles="true">@font-face{font-family:"Noto Serif";src:url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/fonts/NotoSerif-Regular.ttf);font-style:normal;font-weight:400}@font-face{font-family:"Noto Serif";src:url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/fonts/NotoSerif-Italic.ttf);font-style:italic;font-weight:400}@font-face{font-family:"Noto Serif";src:url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/fonts/NotoSerif-Bold.ttf);font-style:normal;font-weight:600}@font-face{font-family:"Noto Serif";src:url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/fonts/NotoSerif-BoldItalic.ttf);font-style:italic;font-weight:600}@font-face{font-family:"Noto Sans";src:url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/fonts/NotoSans-Regular.ttf);font-style:normal;font-weight:400}@font-face{font-family:"Noto Sans";src:url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/fonts/NotoSans-Italic.ttf);font-style:italic;font-weight:400}@font-face{font-family:"Noto Sans";src:url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/fonts/NotoSans-Bold.ttf);font-style:normal;font-weight:600}@font-face{font-family:"Noto Sans";src:url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/fonts/NotoSans-BoldItalic.ttf);font-style:italic;font-weight:600}</style><style data-pagedjs-inserted-styles="true">:root{--color-main: var(--color-elife-blue);--color-main-light: var(--color-elife-blue-light);--color-sub: var(--color-elife-grey);--color-sub-light: var(--color-elife-grey-light);--color-elife-blue: rgb(8, 122, 204);--color-elife-blue-light: rgb(159, 215, 255);--color-elife-grey: rgb(118, 118, 118);--color-elife-grey-light: rgb(240, 240, 240);--color-elife-black: rgb(37, 42, 54);--font-sans: "Noto Sans", sans-serif;--font-serif: "Noto Serif"}</style><style data-pagedjs-inserted-styles="true">@media screen,pagedjs-ignore{.pagedjs_area{--pagedjs-baseline-offset-color: red;--pagedjs-baseline-color: transparent;--pagedjs-baseline-lineheight: 17px;--pagedjs-baseline-position: 13px;background:linear-gradient(transparent 0%,transparent calc(var(--pagedjs-baseline-lineheight) - 1px),var(--pagedjs-baseline-color) calc(var(--pagedjs-baseline-lineheight) - 1px),var(--pagedjs-baseline-color) var(--pagedjs-baseline-lineheight)),transparent;background-size:100% var(--pagedjs-baseline-lineheight);background-repeat:repeat-y;background-position-y:var(--pagedjs-baseline-position)}}article{--offset: 180px}article>section{margin-left:var(--offset)}[data-id="logo"]{position:running(logo);display:block;width:106px;height:40px;text-align:left;position:static;top:auto;left:auto;margin-left:30px}[data-id="logo"] img{width:100%!important;height:100%!important;display:block}.pagedjs_margin-top-left .pagedjs_margin-content [data-id="logo"]{position:absolute!important;top:12px;left:15px}[data-id="right-header-links"]{position:running(right-header-links);display:block;width:100%;text-align:right;position:static;top:auto;right:auto}[data-id="right-header-links"]>a,[data-id="right-header-links"]>span{display:inline-block;margin-left:5px;vertical-align:middle}.runninghead{font-family:var(--font-sans);color:var(--color-elife-grey);border-top:1px solid var(--color-elife-grey);bottom:0mm;line-height:1;left:50px;right:50px;font-size:12px;margin-top:0;padding-top:12px;display:flex;align-items:space-between}.runninghead a{color:var(--color-elife-blue)}.runninghead .counter{margin-left:auto}.runninghead .counter::after{content:counter(page) " of " counter(pages)}.article-flag-list{top:32px;right:50px;left:auto;display:flex;align-items:center;gap:5px;list-style:none;padding:0;margin:0;z-index:1000;font-size:10px}.article-flag-list__item{margin-left:0!important;padding-right:0!important}.article-flag-list__item:not(:last-child)::after{content:"|";display:inline-block;padding:0 10px;margin:0;color:var(--color-elife-grey)}.article-flag__link{color:var(--color-elife-blue);text-decoration:none;white-space:nowrap}.pagedjs_page .pagedjs_margin-top-left{text-align:left}.pagedjs_page .pagedjs_margin-top-right{}.pagedjs_page .pagedjs_margin-top-left>.pagedjs_margin-content>*{display:block}.pagedjs_page .pagedjs_margin-top-left>.pagedjs_margin-content::after{content:element-ident(logo)}.pagedjs_page .pagedjs_margin-top-right>.pagedjs_margin-content>*{display:block}.pagedjs_page .pagedjs_margin-top-right>.pagedjs_margin-content::after{content:""}.pagedjs_page{--pagedjs-margin-top:75px;--pagedjs-margin-right:50px;--pagedjs-margin-left:50px;--pagedjs-margin-bottom:52px;--pagedjs-pagebox-width:8.5in;--pagedjs-pagebox-height:11in}.pagedjs_page.pagedjs_first_page .pagedjs_margin-top-right{text-align:right}.pagedjs_page.pagedjs_first_page .pagedjs_margin-top-right>.pagedjs_margin-content>*{display:block}.pagedjs_page.pagedjs_first_page .pagedjs_margin-top-right>.pagedjs_margin-content::after{content:element-ident(right-header-links)}.pagedjs_page.pagedjs_first_page{}</style><style data-pagedjs-inserted-styles="true">body{font-family:var(--font-serif);font-size:12px;line-height:17px}[data-id="article-content"] :is(p, li){--pagedjs-baseline-snap: 17px;line-height:17px;font-size:0.95em;margin:0;widows:3;orphans:3}[data-id="article-content"] li p{font-size:1em}[data-id="article-content"] p,[data-id="peer-reviews"] p{margin-bottom:5px}blockquote{margin-top:8px;margin-bottom:8px;border-left:1px solid var(--color-sub);margin-left:0;padding-left:3ch}.author-response__body blockquote{font-style:italic;font-family:var(--font-sans)}dl,dd,dt{font-size:0.95em;line-height:17px}:is(p) :is(span, a, em, sup, sub){line-height:0}a{color:inherit;text-decoration:none}section a{color:var(--color-main)}h1,h2,h3,h4,h5,h6{font-family:var(--font-sans);break-inside:avoid;text-wrap:wrap;font-weight:600}.pagedjs_page_content article>section:first-child>section:first-child>h1:first-child,.pagedjs_page_content article>section:first-child>section:first-child>section:first-child>h2:first-child,.pagedjs_page_content article>section:first-child>section:first-child>section:first-child>section:first-child>h3:first-child{margin-top:0!important}h1{font-size:20px;margin-top:10px;margin-bottom:6px;line-height:28px;color:var(--color-main)}h1.title{color:var(--color-main)}h2{font-size:16px;margin-top:8px;margin-bottom:6px;line-height:24px}h3{margin-top:8px;margin-bottom:6px;font-size:14px;color:var(--color-body)}h4{font-size:12px;margin-top:6px;margin-bottom:6px}h5{font-size:11px;margin-top:6px;margin-bottom:6px}[data-id="article-content"]>ol{list-style-position:oustide;padding:0}[data-id="article-content"]>ol li{margin-bottom:1em}[data-id="article-content"] ol ol{list-style-position:oustide;margin:0;padding:0 3ch}[data-id="article-content"] li p{margin:0;text-align:left}ol.list-roman-lower{list-style-type:lower-roman}ol.list-roman-upper{list-style-type:upper-roman}ol.list-alpha-lower{list-style-type:lower-alpha}ol.list-alpha-upper{list-style-type:upper-alpha}code,pre{max-width:530px;word-wrap:break-word;white-space:pre-wrap}.statement,.boxed-text{margin-top:20px;margin-bottom:15px;border-left:5px solid var(--color-elife-blue-light);padding-left:20px}</style><style data-pagedjs-inserted-styles="true">.article-flag-list{display:flex;justify-content:flex-end;list-style:none;padding:0;margin:0}.article-flag-list__item{margin:0;padding:0}.article-flag-list__item::after{content:" | ";color:var(--color-secondary)}.article-flag-list__item:last-child::after{content:none}.article-flag__link{text-decoration:none}[data-id="header"]{margin-left:var(--offset);max-width:90%}[data-id="header"] .title{font-size:2.3em;line-height:1.2;padding-top:0}.descriptors__icon a{padding-left:3ch}.descriptors__icon a::before{position:absolute;width:2ch;margin-left:-3ch;margin-right:2ch;height:auto;display:inline-block}.descriptors__icon .hidden{display:none}.pagedjs_first_page .article-flag-list{display:none}.header-icon.oa-icon::before{content:url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/assets/oa.svg);display:block;width:20px;height:20px}.header-icon.cc-icon::before{content:url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/assets/cc.svg);display:block;width:20px;height:20px}[data-id="sideSection"]{font-family:var(--font-sans);width:calc(var(--offset) - 5ch);position:absolute;left:calc(var(--offset) * -1);font-size:0.9em}[data-id="sideSection"] h2{font-size:inherit;line-height:inherit;font-family:inherit;color:var(--color-main)}[data-id="sideSection"] p{hyphens:none}[data-id="margin-notes"]{font-size:0.8em}[data-id="margin-notes"] h3{font-size:inherit;line-height:inherit;font-family:inherit;color:var(--color-secondary);margin-bottom:0.5em}[data-id="margin-notes"] p{hyphens:none;margin-top:5px;margin-bottom:5px}[data-id="margin-notes"] p:has(.email-icon){margin-bottom:0}[data-id="margin-notes"] .copyright{color:grey}[data-id="margin-notes"] .email-icon::before{content:"";display:inline-block;height:9px;width:9px;margin-right:0.2em;background-image:url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/assets/email.svg);background-size:contain;background-repeat:no-repeat}.author-notes ul{margin-top:0}.author-notes li a{overflow-wrap:anywhere}.article-status__heading{margin-bottom:0}.article-status__link{margin-top:0;display:block}.modal-container,.review-timeline__link{display:none}[data-id="sideSection"] dd{padding-left:0;margin-left:0;margin-bottom:17px}[data-id="sideSection"] dt{font-weight:800}</style><style data-pagedjs-inserted-styles="true">@import url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/modules/authors-and-institutions.css);@import url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/modules/article-content.css);@import url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/modules/figures.css);@import url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/modules/assesment.css);@import url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/modules/references.css);@import url(file:///Users/fredatherden/Documents/GitHub/jats-to-pdf/paged-js/modules/timeline.css);body{--font-family-primary: '__Noto_Sans_d69442', '__Noto_Sans_Fallback_d69442', arial, helvetica, sans-serif;--font-family-secondary: '__Noto_Serif_288e51', '__Noto_Serif_Fallback_288e51', serif}.fix-phi{font-family:'NotoSerifForce';font-variant-ligatures:none;text-transform:uppercase;font-style:italic}.onlynumbers{display:none}.figure:not(.tofill){max-width:100%}.figure:not(.tofill) img .imageonly{width:100%!important;height:100%!important;object-fit:contain}.img-in-link{float:left;margin-left:calc(var(--offset) * -1)}.pagedjs_pages{--pagedjs-baseline-color: transparent !important}[data-id="abstract"] .widowfix{display:none}ol{display:block}ol>li{break-inside:avoid}.insideurl a{color:inherit;text-decoration:none;border-bottom:none;border-bottom:0}.insideurl a::after{}.insideurl .hidden{display:inline}.movedfig img{display:block}.nottofill{margin:1em 0}.nottofill img{display:block;margin:0.8em auto}:is(.prettified-link, .prettified-symbol){display:none}p{widows:2;orphans:2}.tofill,figure{position:fill-page}.figure[data-split-from]{border-top:0!important}.descriptors__icon--cc a{border-bottom:0;color:inherit}.descriptors__icon--cc a .infourl{display:none}.descriptors__icon--cc a .hidden{display:inline}[data-id="reviews"] .descriptors__identifiers{list-style-type:none;padding-left:0;margin-top:0;font-family:var(--font-sans);margin-bottom:2em}.descriptors__identifiers{list-style-type:none;padding:0;font-family:var(--font-sans);margin-bottom:0}.pagebreak{}[data-id="authors"] ul{list-style-type:none;padding-left:0}:root{--color-primary: #087acc;--color-primary-dark: #0769b0;--color-text: #212121;--color-text-secondary: #757575;--color-button: #edeff4;--color-background: #f7f7f7;--divider: #e0e0e0;--link-decoration: #212121;--link-decoration-secondary: #6d6d6d;--border: 1px solid var(--divider);--font-family-primary: noto sans, arial, helvetica, sans-serif;--font-family-secondary: noto serif, serif;--grid-gutter: 24px;--site-spacing: 24px;--site-spacing-large: 48px;--base-font-size: 16
}:after,:before{box-sizing:border-box}.visuallyhidden{border:0;clip:rect(0 0 0 0);height:1px;margin:-1px;overflow:hidden;padding:0;position:absolute;-webkit-user-select:none;user-select:none;width:1px}</style></head><body><template data-ref="pagedjs-content"><div id="page-templates" data-ref="98c1cf40-8bda-40b5-aabc-2d6df0b7d792" data-id="page-templates"><div id="first-page-links" class="elife-intro" data-ref="e06f9f30-5f47-40d0-ae1e-f2658d8a55f4" data-id="first-page-links"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style="display: none;"><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a><div id="right-header-links" data-ref="3c7afed3-0b3e-40d4-9870-9f68daa5c26d" data-id="right-header-links" style="display: none;"><a href="https://en.wikipedia.org/wiki/Open_access" target="_blank" class="header-icon oa-icon" data-ref="e8cc0315-0223-4b9e-89f5-f72b617f8ec9"></a><span class="separator" data-ref="fc3f7d6a-6e26-42fd-8a8f-cf50f6e37e53">|</span><a href="https://creativecommons.org/" target="_blank" class="header-icon cc-icon" data-ref="fd273eb0-acfc-499d-8310-14a992663ed9"></a></div></div></div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e"><section id="sideSection" class="side-section" data-ref="d29e5412-f79b-4013-8e4a-0c1f74571ad1" data-id="sideSection"><div class="article-status" id="el-div-3" data-ref="d728c3de-e2f2-4ca0-8303-1de5cf9f9087" data-id="el-div-3"><div class="review-timeline-container" id="el-div-4" data-ref="63975305-dba9-44ad-a1ee-34b8255f471c" data-id="el-div-4"><dl class="review-timeline review-timeline--expanded" id="review-timeline" aria-label="Version history" data-ref="945d6a41-f13b-47fe-a468-c8f0db6969aa" data-id="review-timeline"><dt class="review-timeline__event review-timeline__event--title review-timeline__event--active review-timeline__event--with-evaluation-summary review-timeline__event--reviewed" data-ref="04603fa3-a54d-4382-a9d3-97731b1b9642"><a href="https://doi.org/10.7554/eLife.101224.1" class="review-timeline__event-link" data-ref="27d287d4-7e81-4aaa-8c71-97d2003d0615">Reviewed Preprint</a></dt><dd class="review-timeline__event review-timeline__event--detail review-timeline__event--active review-timeline__event--with-evaluation-summary review-timeline__event--reviewed" data-ref="a96f25cc-31e2-470f-b706-26c35778576c"><span class="review-timeline__version" data-ref="bf0bd9af-1559-4813-bd69-f4cd0b787f10">v1</span><time class="review-timeline__date" datetime="2024-11-08T00:00:00.000Z" data-ref="90d80755-ded6-450d-b81a-118fa06b13eb">November 8, 2024</time><span class="review-timeline__link" data-ref="cf1c9173-23ad-4481-94c9-f4c7e35abf5b" data-undisplayed="undisplayed">Not revised</span></dd><dt class="review-timeline__event review-timeline__event--title review-timeline__event--active review-timeline__event--with-evaluation-summary review-timeline__event--revised" data-ref="a0fc9fe9-9eb5-49f3-8cf4-ae5e73a14a10">Reviewed Preprint</dt><dd class="review-timeline__event review-timeline__event--detail review-timeline__event--active review-timeline__event--with-evaluation-summary review-timeline__event--revised" data-ref="ae600ed9-b4d3-4ce4-8703-39eabca4bb99"><span class="review-timeline__version" data-ref="2bdad4b3-724d-4a1c-817a-114429b8160e"> v2</span><time class="review-timeline__date" datetime="2025-10-29T00:00:00.000Z" data-ref="079832fa-c378-4e4d-a53e-412dc8977542">October 29, 2025</time><span class="review-timeline__link" data-ref="308d901c-9055-40f7-9d48-67f1044a1654" data-undisplayed="undisplayed">Revised by authors</span></dd></dl></div></div><section id="margin-notes" data-ref="f843c510-d79d-4993-98d8-d07dcc58ec46" data-id="margin-notes"><div class="author-notes" id="el-div-5" data-ref="cbffb5a8-034b-422e-8ecf-07a26ed5ba14" data-id="el-div-5"><p id="el-p-1" data-ref="e49cf352-f465-47a9-927f-492dd60e221b" data-id="el-p-1"><strong class="email-icon" data-ref="07b68d3b-54b5-4c7f-9b54-ac48abb32726">For correspondence:</strong></p><ul class="list-simple" id="el-ul-1" data-ref="542eb2d6-a007-444c-a063-892105efa9cb" data-id="el-ul-1"><li id="el-li-2" data-ref="c867c34f-70e2-4b01-abbd-6fbbdc7e504f" data-id="el-li-2"><a href="mailto:test@umich.edu" data-ref="6fe0c5dd-0a56-4f98-9ec4-280ea9ffd66f">test@umich.edu</a></li><li id="el-li-3" data-ref="edef9f88-7f51-4e9b-a500-aba6405bb699" data-id="el-li-3"><a href="mailto:iheemske@umich.edu" data-ref="6b3f1a3a-78e3-475a-a36c-9d921d2d2f0f">iheemske@umich.edu</a></li></ul><p id="el-p-2" data-ref="d2de8c1b-78d1-4aa8-b11d-6118f653dd12" data-id="el-p-2"></p><p class="author-notes__list_item" id="el-p-3" data-ref="bec24e2d-0975-4e10-a982-8653c461f6b0" data-id="el-p-3"><sup aria-hidden="true" data-ref="e27061af-0089-41e7-b942-2a2963ccac13"><strong data-ref="a5778af5-6c51-4328-b7fd-0cc66a8bdc2f">*</strong></sup>&nbsp;Equal contributions</p><p class="author-notes__list_item" id="el-p-4" data-ref="d240c063-b0f3-4ab8-b94f-76d6718da1be" data-id="el-p-4"><strong data-ref="f960dc6b-193e-4266-8b1a-4af46413f0af">Competing interests:</strong> No competing interests declared</p></div><div id="editors" data-ref="82fc0bec-7cff-4e2f-a0ee-5b4ab776eb69" data-id="editors"><p class="editors-and-reviewers__person" id="el-p-5" data-ref="5a438732-7c24-4e0a-84aa-55be09c0dea4" data-id="el-p-5"><strong data-ref="4c1e3658-70f3-4ae5-932f-2cc56fba11fb">Reviewing editor: </strong>Sergio Ruiz Macias, National Cancer Institute, NIH, United States</p></div><div id="copyright" class="copyright" data-ref="9706c82a-0c35-418d-a4ac-d43de4bd7768" data-id="copyright"><p id="el-p-6" data-ref="2d745f9c-bcaa-4523-a885-ff7bdd36c447" data-id="el-p-6">© 2024, Jo et al. This article is distributed under the terms of the <a href="https://creativecommons.org/licenses/by/4.0/" data-ref="1055e913-737c-4bc8-b092-ac46ae57b5a6">Creative Commons Attribution License</a>, which permits unrestricted use and redistribution provided that the original author and source are credited.</p></div></section></section><header id="header" class="content-header" data-ref="b92868f7-c408-4834-8d56-25be65880738" data-id="header"><h1 class="title" id="el-h1-0" data-ref="f04801a6-b526-463b-8afb-f2376e4167bc" data-id="el-h1-0" data-break-after="avoid">Endogenous FGFs drive ERK-dependent cell fate patterning in 2D human gastruloids</h1><div class="authors" id="el-div-8" data-ref="cc7758c7-b6ac-4ca5-9ead-8f0b8c41826d" data-id="el-div-8" data-previous-break-after="avoid"><ol class="authors-list authors-list--expanded" aria-label="Authors of this article" id="el-ol-0" data-ref="9f18dc52-7ab7-45a4-a2d1-d8d0531dabdb" data-id="el-ol-0"><li class="authors-list__item" id="el-li-4" data-ref="1a2882d9-e976-4cd4-99bb-9dcab2911c6d" data-id="el-li-4" data-item-num="1"><span class="authors-link" data-ref="cec7a04d-60a6-4039-9dc9-c9f6d48977df">Kyoung Jo<sup data-ref="e43b275a-4c0e-4528-804f-d7030858fedf">1,</sup><sup data-ref="a44ab537-b68b-4a26-80a3-7a13b5e55ae4">*</sup></span></li><li class="authors-list__item" id="el-li-5" data-ref="e8f83055-de72-4849-9ab9-5b8971e861ca" data-id="el-li-5" data-item-num="2"><span class="authors-link" data-ref="0ed3a9c4-8eb0-4bb8-ba0b-378583819bf9">Zong-Yuan Liu<sup data-ref="514c5786-1e95-43ad-a460-fe12f892a578">1,</sup><sup data-ref="ceae9108-1891-4794-aa33-86df45d78355">*</sup></span></li><li class="authors-list__item" id="el-li-6" data-ref="69f2f072-8d5a-4a20-9216-9333d79d145e" data-id="el-li-6" data-item-num="3"><span class="authors-link" data-ref="dc9acce3-05b2-497f-9463-41836b83ea92">Gauri Patel<sup data-ref="5e10f5ef-24a7-408e-9c25-fb2283930154">4</sup></span></li><li class="authors-list__item" id="el-li-7" data-ref="5ece50f5-f90e-4587-8441-74cb896ce496" data-id="el-li-7" data-item-num="4"><span class="authors-link" data-ref="0cdd4c42-1eab-4051-8b15-303fc94741b5">Zhiyuan Yu<sup data-ref="1a54094c-4861-4b62-adcc-3d27b9c53e78">2</sup></span></li><li class="authors-list__item" id="el-li-8" data-ref="e1f75c7c-8b69-462e-a02f-fbb6e61d064c" data-id="el-li-8" data-item-num="5"><span class="authors-link" data-ref="0b80f5e2-2fd1-4a27-8f28-f003d271fa3f">LiAng Yao<sup data-ref="4e5515b4-9cd7-438b-9756-d21f3081c0ef">1</sup></span></li><li class="authors-list__item" id="el-li-9" data-ref="93f2f43c-7d57-4640-b8b2-6c6bee32e0f9" data-id="el-li-9" data-item-num="6"><span class="authors-link" data-ref="1e26f6e6-8ac1-402d-b042-73fa2651125c">Seth Teague<sup data-ref="a7bf62a2-7ecb-4582-907e-be67a9f27a3e">4</sup></span></li><li class="authors-list__item" id="el-li-10" data-ref="089a7951-c3d3-4582-a1d0-f6e716de5b82" data-id="el-li-10" data-item-num="7"><span class="authors-link authors-email__link" data-ref="56465565-7f7f-490e-9075-73f32088fb05">Craig Johnson<sup data-ref="082dc54f-3c1e-4422-91fe-979998eeb5fc">1</sup><a href="mailto:test@umich.edu" data-ref="94342959-b471-4fb1-a41f-09546632ba72"><span class="email-icon" data-ref="3c69cae0-72f7-4125-8b0b-821e3aebd90f"></span></a></span></li><li class="authors-list__item" id="el-li-11" data-ref="abeec578-ce81-44b2-8442-6db2a33ff7db" data-id="el-li-11" data-item-num="8"><span class="authors-link" data-ref="468d45d3-a705-436e-be61-832b1b1ea099">Jason Spence<sup data-ref="79b92302-9ae0-40aa-b853-6f75483bfc54">1,</sup><sup data-ref="372b4e2c-012e-4962-9a79-4ca726caa914">3,</sup><sup data-ref="aa6e554f-614b-4f3d-8e44-3774adb35665">4,</sup><sup data-ref="549908f4-5890-465e-b554-1cd803565001">5</sup></span></li><li class="authors-list__item" id="el-li-12" data-ref="15afb4c1-f322-4b8c-99f1-f5c40428cbc7" data-id="el-li-12" data-item-num="9"><span class="authors-link authors-email__link" data-ref="a06f0e26-d012-439d-b821-84e580715ee5">Idse Heemskerk<sup data-ref="a580d364-eb7d-4b83-b14f-79cc939c3bcb">1,</sup><sup data-ref="594acd17-5386-40d8-b737-9b73c134aabd">2,</sup><sup data-ref="e75fe4eb-bb23-45bf-809b-effdaddf2223">3,</sup><sup data-ref="3aae3716-bbfe-4474-baf7-47055cf15553">4,</sup><sup data-ref="3056d230-ffde-4f32-8374-1e4ed7b0b1ad">6</sup><a href="mailto:iheemske@umich.edu" data-ref="97fd5283-845b-48dc-9bac-c00cb7a343fb"><span class="email-icon" data-ref="e24fbe88-f9db-4489-8dfc-ba88c3e7bc92"></span></a></span></li></ol></div><div class="institutions" id="el-div-9" data-ref="07708a74-5968-4d2d-9c8f-60e80e1bf03d" data-id="el-div-9"><ol class="institutions-list" aria-label="Author institutions" id="el-ol-1" data-ref="c7b9ed9c-2607-4fdd-ab90-31bcb7c58aa0" data-id="el-ol-1"><li class="institutions-list__item" id="el-li-13" data-ref="1c96bbd6-f05e-47e2-b78c-93846b21d93e" data-id="el-li-13" data-item-num="1"><sup data-ref="dda6b018-7f80-40fc-8c9e-dc3575a11558">1</sup>Department of Cell and Developmental Biology, University of Michigan Medical School, Ann Arbor, United States</li><li class="institutions-list__item" id="el-li-14" data-ref="bd9c9265-23dd-4345-96e2-451cb60f5252" data-id="el-li-14" data-item-num="2"><sup data-ref="6cac471e-14df-41c6-ad06-a48f2ad11c46">2</sup>Department of Computational Medicine and Bioinformatics, University of Michigan Medical School, Ann Arbor, United States</li><li class="institutions-list__item" id="el-li-15" data-ref="60c79fd5-9216-4db6-a4f6-a0fda72e4ec1" data-id="el-li-15" data-item-num="3"><sup data-ref="1636ae8e-245c-4840-af6a-ed761bb8d801">3</sup>Center for Cell Plasticity and Organ Design, University of Michigan Medical School, Ann Arbor, United States</li><li class="institutions-list__item" id="el-li-16" data-ref="653b419a-702b-43c7-90c9-3b4e88b3b355" data-id="el-li-16" data-item-num="4"><sup data-ref="19e38072-18d1-409c-80f5-2f6ee102b668">4</sup>Department of Biomedical Engineering, University of Michigan, Ann Arbor, United States</li><li class="institutions-list__item" id="el-li-17" data-ref="a9c5f6e6-ca77-421c-8260-9f706c8b783a" data-id="el-li-17" data-item-num="5"><sup data-ref="4d520166-fbd1-4c92-8b6e-9bbe504824c0">5</sup>Department of Internal Medicine, Gastroenterology, University of Michigan Medical School, Ann Arbor, United States</li><li class="institutions-list__item" id="el-li-18" data-ref="7c91b326-f615-4936-9677-debb4585a9ae" data-id="el-li-18" data-item-num="6"><sup data-ref="c1095db7-d73d-4f59-b871-bf9e17f4ebf6">6</sup>Department of Physics, University of Michigan, Ann Arbor, United States</li></ol></div></header><section id="article-content" data-ref="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-id="article-content"><section id="assessment" class="assessment" data-ref="4a99554b-fc89-4aa5-9e54-358cdbc3b2e8" data-id="assessment"><h2 id="el-h2-0" data-ref="11d2d409-dfe3-471e-a1e7-3976e20cd6ed" data-id="el-h2-0" data-break-after="avoid">eLife Assessment</h2><div class="assessment__body" id="el-div-10" data-ref="e5137e0e-5343-462c-a0c5-58f2a98c9813" data-id="el-div-10" data-previous-break-after="avoid"><p id="el-p-7" data-ref="30f4438c-86a7-45d9-9763-d45860ca4688" data-id="el-p-7">This work is an <strong class="highlighted-term" aria-label="Highlighted" data-ref="741035f0-c050-4027-877c-096f05c22d25">important</strong> contribution to understanding the role of FGF signaling in the induction of primitive-like cells in a 2D system of human gastrulation. The authors provide <strong class="highlighted-term" aria-label="Highlighted" data-ref="17d6941e-6d42-4633-b7de-a6759ad0454a">compelling</strong> evidence showing that endogenous FGF ligands, acting through FGF receptors localized basolaterally, are determinant in the acquisition of specific cell fates. These observations will be of broad relevance to the FGF field.</p></div><div class="descriptors" id="el-div-11" data-ref="89d51c0a-0b44-4d3a-bbe8-72542f0b761a" data-id="el-div-11"><ul class="descriptors__identifiers" id="el-ul-2" data-ref="48a458aa-919b-4f4c-953a-20b0fd8bbee5" data-id="el-ul-2"><li class="descriptors__identifier" id="el-li-19" data-ref="c338fb42-6d62-4c41-8a89-9aa8c181a055" data-id="el-li-19"><a href="https://doi.org/10.7554/eLife.101224.2.sa4" data-ref="455f62a4-a33a-4e9a-9c32-b7132b226881">https://doi.org/10.7554/eLife.101224.2.sa4</a></li></ul></div></section><section id="abstract" class="abstract" data-ref="021e9f8f-59d3-4c08-bf24-ed1d3b5689f4" data-id="abstract"><h1 id="el-h1-1" data-ref="760fdf5d-c352-44c3-86bd-399e922cc81e" data-id="el-h1-1" data-break-after="avoid">Abstract</h1><p id="el-p-8" data-ref="2d113aff-cf16-4885-8a8d-be57b20dd02f" data-id="el-p-8" data-previous-break-after="avoid">The role of FGF is the least understood of the morphogens driving mammalian gastrulation. Here we investigated the function of FGF in a stem cell model for human gastrulation known as a 2D gastruloid. We found a ring of FGF-dependent ERK activity that closely follows the emergence of primitive streak (PS)-like cells but expands further inward. We showed that this ERK activity pattern is required for PS-like differentiation and that loss of PS-like cells upon FGF receptor inhibition can be rescued by directly activating ERK. We further demonstrated that the ERK-ring depends on localized activation of basolaterally positioned FGF receptors (FGFR) by endogenous FGF gradients. We confirmed and extended previous studies in analyzing expression of FGF pathway components, showing FGFR1 is the main receptor, FGF2 is highly expressed across several cell types, and FGF4/17 are the main FGF ligands expressed in the PS-like cells, similar to the human and monkey embryo but different from the mouse. We found that knockdown of FGF4 greatly reduced PS-like differentiation while FGF17 knockdown primarily affected subsequent mesoderm differentiation. FGF8 expression was spatially displaced from PS-markers and FGF4 expression and peaked earlier, while knockdown led to an expansion in PS-like cells, suggesting FGF8 may counteract FGF4 to limit PS-like differentiation. Thus, we have identified a previously unknown role for FGF-dependent ERK signaling in 2D gastruloids and possibly the human embryo, driven by a mechanism where FGF4 and FGF17 signal through basally localized FGFR1 to induce PS-like cells and their derivatives, potentially restricted by FGF8.</p></section><section id="s1" data-ref="790b4fb7-01cd-4a65-93f2-98b59b2c0a72" data-id="s1"><h1 class="heading-1" id="el-h1-2" data-ref="e417c872-9646-4929-80c9-73546a79ba0c" data-id="el-h1-2" data-break-after="avoid">Introduction</h1><p id="el-p-9" data-ref="3a3e36c5-1a53-42ac-87e9-0bdfb7ef9921" data-id="el-p-9" data-previous-break-after="avoid">During gastrulation the three germ layers are established, and the body plan is laid out. In amniotes this involves the formation of the primitive streak on the posterior side of the embryo from which cells ingress beneath the epiblast and migrate to form the mesodermal and endodermal layers. An important role in gastrulation is played by evolutionarily conserved morphogens from the BMP, Wnt, Nodal, and FGF families<sup data-ref="d04a5c12-9e71-42e8-aa34-73bfd47efd34"><a href="#c1" data-ref="2291fee7-ce6a-4008-aa64-459da4f2e081">1</a></sup>. The function of the first three signaling pathways is comparatively well understood: in both mouse and human a transcriptional hierarchy between BMP4, Wnt3, and Nodal plays a central role in inducing dynamic signaling gradients along the anterior posterior axis that control cell fate patterning<sup data-ref="9ca92e2e-4e4a-4c17-98d4-95af15839217"><a href="#c1" data-ref="29695663-3d39-41c2-92e0-cb70fda103e4">1</a>–<a href="#c3" data-ref="a062b2bb-3e9e-43b2-b803-9ab060aa9718">3</a></sup>. For BMP and Nodal, basal receptor localization in the epiblast is crucial to form signaling activity gradients<sup data-ref="270c28f2-296f-49e6-97dd-83d389e955e4"><a href="#c3" data-ref="c1e0e162-b837-4462-b1ab-af56f9d43c6f">3</a>–<a href="#c7" data-ref="c4492cb5-5554-46c1-b10c-9591b620b17d">7</a></sup>. Wnt and Nodal signaling are directly required for mesoderm and endoderm differentiation, while the balance between BMP and Nodal signaling is important for patterning the germ layers along the body axes<sup data-ref="24cf7f11-04e5-4e29-a207-cfe2fa3abc12"><a href="#c8" data-ref="c621c8cd-ac61-4e63-9ea8-d354f4ea759c">8</a></sup>.</p><p id="el-p-10" data-ref="92d932c1-7237-418f-97b4-d182c8d138a9" data-id="el-p-10">In contrast, the role of FGF in mammalian and especially human gastrulation has been more elusive. Secreted FGFs signal through FGF receptor tyrosine kinases (FGFRs) in a heparan sulfate proteoglycan (HSPG)-dependent manner to activate several signaling pathways including MAPK/ERK, PI3K, and PLCγ<sup data-ref="3dff0c57-e88d-4dc5-8e50-375bf9293eef"><a href="#c9" data-ref="43d4ca26-b824-4558-8559-da78ffe4bce4">9</a></sup>. There is evidence for a conserved requirement for FGF/ERK signaling in mesoderm formation across vertebrate model organisms<sup data-ref="257cd079-1ab1-496c-9944-f1c16a65f1d5"><a href="#c10" data-ref="ae11d120-4571-433b-ba8b-7892e8c8ed53">10</a>–<a href="#c13" data-ref="993fbd92-6877-4b0e-86da-8ac0561519cd">13</a></sup>. However, it is unknown if there are FGF- and phosphorylated ERK gradients across the mammalian primitive streak<sup data-ref="8304d601-b07b-4993-8a1d-084ac4460d0a"><a href="#c14" data-ref="244e305d-7f2e-40c6-9870-266b586f65f4">14</a>,<a href="#c15" data-ref="6b6039a0-7338-436d-9fc5-1561429b5bc8">15</a></sup>, whether there are functionally relevant ERK signaling dynamics like in other contexts, or whether receptor localization is important for establishing signaling gradients<sup data-ref="5dfdd301-0684-46ba-954b-6027b18988f5"><a href="#c16" data-ref="a387ead4-3297-4943-9e87-2e584ac19f13">16</a>–<a href="#c19" data-ref="316bd30e-7e65-404e-a365-b7775093dd27">19</a></sup>. It is also unclear to what extent the requirement in mesoderm induction is direct, or indirect by modulating other signals<sup data-ref="17d11cc4-a799-4a4f-896b-49a2737cda9e"><a href="#c14" data-ref="4a4a12fd-37a2-41b0-823b-24465aed2102">14</a>,<a href="#c15" data-ref="742f7318-324d-43e1-8029-b6d79ab84495">15</a>,<a href="#c20" data-ref="a764eda8-5a8a-4d3c-be47-8623d858a254">20</a></sup>. In addition, several FGFs are co-expressed and the roles of specific FGFs are uncertain. Moreover, there may be significant interspecies differences in this respect. Notably, FGF8 is the only FGF which has been shown to be required for gastrulation in the mouse<sup data-ref="d03383f0-4092-4234-8d26-1f45a046b8d7"><a href="#c21" data-ref="794c68e0-2799-4097-897a-0f09d4d15b6e">21</a></sup>, and its requirement in gastrulation appears to be conserved across vertebrate model organisms<sup data-ref="f8b71f14-2972-42ba-9df0-a64f5feca27b"><a href="#c22" data-ref="86cf46c3-d11e-4f09-8c6b-689333920eaf">22</a>–<a href="#c24" data-ref="587e9f07-5519-4b98-962b-30ade0868304">24</a></sup>, yet FGF8 is not highly expressed in primate embryos<sup data-ref="56939c8a-2c90-4c46-834f-2bf3c42269bc"><a href="#c25" data-ref="0675581d-1047-4f31-9d4b-382fd07d0666">25</a>,<a href="#c26" data-ref="4dd40abf-fd32-4bdb-882c-1c522a0a4a26">26</a></sup>.</p><p id="el-p-11" data-ref="646d7491-c386-4657-a21d-9cc40b427bae" data-id="el-p-11">Mutants for FGF8, FGFR1, HSPG synthesis, and ERK2, as well as pharmacological inhibition of MEK and FGFR give rise to similar phenotypes in mouse gastrulation, suggesting that FGFR1 and FGF8 are the only FGF ligand-receptor pair required for gastrulation and that these act through the MAPK/ERK pathway<sup data-ref="1dee02c6-5b59-426b-a398-d6913d969710"><a href="#c12" data-ref="a268984c-08e9-45a7-bc3a-d8bdacf98fd0">12</a>,<a href="#c20" data-ref="9a4c2f32-72ec-43fe-8b31-c0c6a8e2fb03">20</a>,<a href="#c21" data-ref="e8483df4-158f-4512-aa2a-ad29b4921579">21</a>,<a href="#c27" data-ref="1d1ed612-6d55-4b04-ae51-dfef9b472dc0">27</a>–<a href="#c31" data-ref="62e32e5b-d724-493d-9c76-37b0d66ab81c">31</a></sup>. FGF4 expression during gastrulation is also lost in FGF8 mutants, leaving the possibility of a role for FGF4, but this remains unknown as FGF4 is also required in the blastocyst and mutants therefore do not reach gastrulation<sup data-ref="20c1c6ef-b96e-4b3f-ac31-fdeea6ebb109"><a href="#c32" data-ref="53f8327a-7e7a-4877-bbfc-51b2ab452a87">32</a></sup>. In all mutants and for pharmacological inhibition of FGF/ERK signaling, expression of the primitive streak marker brachyury (TBXT, also BRA, or T in mouse) is lost or severely diminished in the posterior streak and mesoderm fails to migrate away, piling up in this region. In addition, TBX6 expression is lost completely, and no lateral or paraxial mesoderm are formed. In contrast, TBXT expression in the anterior streak and extraembryonic mesoderm is not dependent on FGF/ERK signaling; extraembryonic mesoderm development is normal, and the axial mesoderm is expanded<sup data-ref="b7e05bff-8041-454f-9785-1eb73d74a9d8"><a href="#c20" data-ref="3ee1baa5-b267-42b9-8df8-71bb5c54a68d">20</a>,<a href="#c21" data-ref="8bee4257-a8b1-45dc-9af8-cf4e7b021ad5">21</a>,<a href="#c27" data-ref="9b2704a8-bd66-44c6-9dbb-0aa95e19113e">27</a>,<a href="#c28" data-ref="455bd1c8-1f2c-4f35-ad1f-80291333c311">28</a></sup>.</p><p id="el-p-12" data-ref="9b2edefd-4399-423e-97e9-ab31265d1c81" data-id="el-p-12">In vitro models offer an opportunity to disentangle the complexity of FGF signaling in human development. Here, we explore the role of FGF/ERK signaling in micropatterned human pluripotent stem cells treated with BMP4, also known as 2D human gastruloids. These give rise to concentric rings of different cell types associated with gastrulation after 2 days of differentiation. Pluripotent cells in the center are surrounded by a primitive streak-like ring, which itself is surrounded by primordial germ cells and amnion-like cells on the colony edge<sup data-ref="445e86b2-9d89-4d86-b57e-734c0ee4a38b"><a href="#c33" data-ref="8214e7ca-8ffa-4d44-bc85-88451a27924d">33</a>–<a href="#c35" data-ref="99ba116f-9848-4f12-8c5c-17ca68a570c4">35</a></sup>. Due to its simplicity and reproducibility, this system has proven powerful in deciphering the mechanisms underlying aspects of human gastrulation. It was used to show that BMP-Wnt-Nodal hierarchy that controls primitive streak formation during mouse gastrulation is preserved in human while providing a range of new insights into how these signals function<sup data-ref="a4e015f5-cfb3-4974-bb65-057ddffd1243"><a href="#c2" data-ref="23b37e7a-bc73-4ffa-ab8a-1507ee118d65">2</a>,<a href="#c3" data-ref="28ccf6c3-1db3-4e8d-9001-8c3d963d3b49">3</a></sup>. For example, the crucial role for BMP and Nodal receptor localization in patterning was discovered in 2D gastruloids and later confirmed in the mouse embryo. This system also revealed that BMP, Wnt, and Nodal signaling patterns are highly dynamic<sup data-ref="84ffa1bd-c764-4245-9a07-e103f0fb1faf"><a href="#c3" data-ref="9b144001-ccc9-4510-8057-e31972719e82">3</a>–<a href="#c7" data-ref="4362078e-e9f9-4dbf-bf38-2604af0fb967">7</a></sup>. It therefore also provides a promising approach to determine how FGF may function in human gastrulation.</p><p id="el-p-13" data-ref="10b2ef82-b265-474f-a826-f805fed45f2d" data-id="el-p-13">In the 2D human gastruloid model, we show that a ring of phosphorylated ERK forms in an FGF-dependent manner and expands largely in lockstep with the formation of primitive streak-like cells but extends further inward. We demonstrate that this pattern of ERK activity is due to localized FGF receptor (FGFR) activation that requires basal FGFR localization. We then show that exogenous FGF2 in the media is required for pattern formation to occur but can be removed well before the phosphorylated (active) ERK (pERK) and PS-like rings appear, suggesting the pERK ring instead depends on endogenous FGF gradients. Using single cell RNA-sequencing, we determine the expression of FGF pathway components including FGFR1 as the main receptor and FGF2, FGF4, and FGF17 as the most highly expressed ligands. Our analysis corroborates previous studies describing expression of FGF2 and FGF17 in human embryos and gastruloids but for the first time recognizes a possible role for FGF4 and includes a broad analysis of genes involved in FGF signaling modulation<sup data-ref="694f4a52-c8a3-4983-b128-29c6967659e5"><a href="#c25" data-ref="7541f572-3690-404d-a2f5-c3b39eb19223">25</a>,<a href="#c36" data-ref="92516521-5cbb-4c0d-ab3f-bef33542ebe1">36</a></sup>. We find that the FGF4 and FGF17 expression are restricted to the PS-like ring and that knockdown of FGF4 significantly reduces PS-like differentiation, while FGF17 knockdown primarily affects downstream mesoderm differentiation. Given its importance in the mouse, we also analyze FGF8 and find its expression is spatially and temporally displaced from FGF4 and PS-markers, while FGF8 knockdown leads to an increase in PS-like differentiation, suggesting it may counteract FGF4. In summary, we have greatly advanced our understanding of how FGFs function in a model for human gastrulation.</p></section><section id="s2" data-ref="c2323182-eb7e-45cc-b2f0-c40d361a838e" data-id="s2"><h1 class="heading-1" id="el-h1-3" data-ref="94622f3c-5666-42e4-adfb-974b4bb17390" data-id="el-h1-3" data-break-after="avoid">Results</h1><section id="s2a" data-ref="866c7dfe-80ac-446d-810f-f971eaa64035" data-id="s2a" data-previous-break-after="avoid"><h2 class="heading-2" id="el-h2-1" data-ref="11b35ab6-5e7a-44e3-af6c-52fb4bd2304e" data-id="el-h2-1" data-break-after="avoid" data-following="following-ff2c00ba-51a1-482d-9afa-f5a370dcfa50">FGF/ERK signaling is required for primitive streak-like differentiation</h2><p id="el-p-14" data-ref="fa059b58-a419-4158-a326-870d8f87cb33" data-id="el-p-14" data-previous-break-after="avoid">We previously observed that pharmacological inhibition of MEK (the kinase upstream of ERK) or FGF receptors leads to a loss of differentiation to primitive streak-like and primordial germ cell-like cells, suggesting a crucial role for FGF/ERK signaling<sup data-ref="21bae75d-dcea-4513-9a3c-e887b98d6b28"><a href="#c34" data-ref="1a145780-94f9-4abc-90f5-c6ffc1492848">34</a></sup>. Moreover, a ring of active ERK (pERK) has been observed in 2D gastruloids but its connection to FGF signaling or differentiation remains unclear<sup data-ref="f7f4645c-4e6a-400b-95b8-2c6e9865a63c"><a href="#c33" data-ref="040c207b-e31f-4517-b401-24c3af5e4f81">33</a></sup>. Therefore, we wanted to investigate the role of FGF/ERK signaling further.</p><p id="el-p-15" data-ref="01722b51-6ae5-48ac-b9c9-e36104eadf9f" data-id="el-p-15">To determine the dynamics of pERK and relate it to PS-like differentiation, we first performed time series immunofluorescence co-staining for pERK and the PS marker TBXT after BMP4 treatment in 700um diameter colonies. This revealed the pERK ring first emerges around 24h, around the same time as TBXT expression, and gradually gets brighter and wider (<span class="fakelink linktofig" data-ref="34e3f548-6291-4309-86fd-a2c172b6eb31"><span class="fakelink-text" data-ref="1b9e0cbd-5819-42d0-a341-9c90ff9d1d1b">Fig. 1a</span><a class="fake-icon" href="#fig1" data-ref="ffbb2b00-6fbf-4f40-a0ba-13a2438fcc33"><span class="fakelink-icon" data-ref="2d89ce29-3414-4c69-8a5f-83dcd5e02dbd">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). By 42h the outer edge of the ERK ring coincided with TBXT expression, while the inner edge was less sharp and extended further towards the center than TBXT, reminiscent of the domains of WNT and NODAL signaling that extend further in than the PS-like region<sup data-ref="b1a91842-a51c-474f-8937-9d6a9dae3117"><a href="#c3" data-ref="09dcf322-7225-47bb-b7e7-2f19ab46eda9">3</a>,<a href="#c6" data-ref="dd53c413-b2bd-4002-b4dc-5f87008db985">6</a></sup> (<span class="fakelink linktofig" data-ref="971444a6-fee1-4438-99ff-396b232a5717"><span class="fakelink-text" data-ref="9ad4f28e-62f1-4f19-914e-0780383f8105">Fig. 1a</span><a class="fake-icon" href="#fig1" data-ref="86ccaa48-b0bb-49ff-888e-7eb5f5f10dd3"><span class="fakelink-icon" data-ref="b3737295-39db-4fe2-9b3a-98cf15c51c2e">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Cell fate patterning in 2D gastruloids has a fixed length scale from the edge, with the central part absent in smaller colonies due to the fixed length scale of the BMP signaling gradient from the edge. Consistently, we found that ERK signaling also has a fixed pattern from the edge regardless of colony size, so that in smaller colonies the ERK ring becomes a domain of high ERK signaling extending through the colony center (<span class="fakelink linktofig" data-ref="51408b7d-a35e-4f10-a888-e1839bc19045"><span class="fakelink-text" data-ref="a281df27-3398-45bf-92fd-a33fefea2626">Supp. Fig. 1a</span><a class="fake-icon" href="#figS1" data-ref="582062fb-40a7-402f-b3f6-02fbe88350ce"><span class="fakelink-icon" data-ref="8d6e4ef0-5a9f-4364-a3e6-1a2a47f532c4">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). MEK or FGFR inhibition for 30 minutes after 41.5h eliminated ERK activity, suggesting ERK activity depends entirely on FGF and that continuous FGF signaling is necessary to maintain the ERK signaling pattern (<span class="fakelink linktofig" data-ref="cd9e2e9d-123e-401a-b1a8-8ec0b7be250e"><span class="fakelink-text" data-ref="799e01c2-cb4d-48a9-a073-d4c1d2fa26a5">Fig. 1bc</span><a class="fake-icon" href="#fig1" data-ref="7f318b09-a692-4221-ab6e-400a23d7c5b8"><span class="fakelink-icon" data-ref="bf80dcac-b013-4918-89c6-6b5846aa3a0b">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Inhibition for the full duration of differentiation eliminated TBXT expression as well as other PS markers (<span class="fakelink linktofig" data-ref="d1f4ef9c-41d2-4c0f-8dea-71e40e428a1e"><span class="fakelink-text" data-ref="f13341ea-02c7-46ca-8b9a-b50bf6292261">Fig. 1bc</span><a class="fake-icon" href="#fig1" data-ref="ee5e22e5-c742-461c-9bf5-61324ecb69c4"><span class="fakelink-icon" data-ref="74ff6ff5-7950-4e6f-8ec3-6647db654c3d">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="68ddb468-cdc3-4294-b046-de8832c1e4f2"><span class="fakelink-text" data-ref="dd4b45fc-6c49-48fb-88c7-8036934cb8cb">Supp. Fig. 1b</span><a class="fake-icon" href="#figS1" data-ref="9be9fed2-d3e6-4bda-a297-8a02d789a76b"><span class="fakelink-icon" data-ref="e0d0099b-f80a-4613-919b-113b72f3369c">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). However, we were able to rescue TBXT expression by directly and uniformly activating ERK (<span class="fakelink linktofig" data-ref="60e6e364-498b-4348-b0ad-b6a09a15900b"><span class="fakelink-text" data-ref="c61eae97-1813-491b-9718-a73d7e318538">Fig. 1d-f</span><a class="fake-icon" href="#fig1" data-ref="475d1146-06ef-4e7e-acba-9f4aa34df9f3"><span class="fakelink-icon" data-ref="98f854df-de90-4ac6-8a2e-b8fd2bba8e24">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="5489b7ac-4280-425f-af90-659fd6108d29"><span class="fakelink-text" data-ref="3ecba870-9713-41d0-aa89-b2fd5a773481">Supp. Fig. 1cd</span><a class="fake-icon" href="#figS1" data-ref="ffc990a0-50a9-44df-b486-5e11a71efd1c"><span class="fakelink-icon" data-ref="9aa38862-0182-4c0b-93f5-7cbefa136ae0">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>) in cells stably expressing the doxycycline-inducible membrane-targeted catalytic domain of SOS (dox-SOS<sup data-ref="943309b2-a8ad-44bb-966a-05b0a2b01b6e">cat</sup>)<sup data-ref="6c75bcc1-20dc-4acd-8826-aa9c4653f2f9"><a href="#c37" data-ref="3175deed-86dd-4d39-acc2-adb0ff2a51a5">37</a></sup>. Despite uniform induced ERK activity, TBXT expression remained excluded from the center, showing that ERK activation alone is not sufficient for TBXT induction. However, TBXT levels were much higher in the rescue (<span class="fakelink linktofig" data-ref="b1942e8d-ade7-4c2f-bd72-28de6fd0c6a7"><span class="fakelink-text" data-ref="16f5efa3-20a4-4170-902e-a5e1017035ba">Supp. Fig. 1cd</span><a class="fake-icon" href="#figS1" data-ref="be2b4b5c-acad-4c49-ac42-38d6e20e9313"><span class="fakelink-icon" data-ref="6b4a85b5-70a2-49c2-a2e9-6c13a0425733">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>), possibly reflecting the much higher level of ERK activity induced by dox-SOS<sup data-ref="6aedf548-f427-4c17-b2a9-569e84b81352">cat</sup>. MEK and FGFR inhibition also led to a reduction in cell number which staining for BrdU and cleaved Cas3 suggested was primarily due to reduced proliferation (<span class="fakelink linktofig" data-ref="69ce898a-5e48-4269-8d1e-c96c1a82e130"><span class="fakelink-text" data-ref="ed1fecd3-680e-4523-a859-69a256bb102f">Supp. Fig. 1ef</span><a class="fake-icon" href="#figS1" data-ref="1157bfd5-d653-4067-a4f1-6eeafba3df08"><span class="fakelink-icon" data-ref="33d5200c-6988-4811-9d6b-b49baa645a4f">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). In contrast to TBXT expression, reduced cell number caused by FGFR inhibition was only partially rescued by dox-SOS<sup data-ref="6c59f1fe-987e-46f0-89de-0f5800ef801c">cat</sup>, indicating additional downstream pathways regulate cell number (<span class="fakelink linktofig" data-ref="5343b0b5-f08c-4cf7-b2ef-2f4213593db3"><span class="fakelink-text" data-ref="7f4c4aa1-a09b-403d-ab5b-3afce2400ab0">Fig. 1e</span><a class="fake-icon" href="#fig1" data-ref="c4360973-6028-48f4-bd7c-10d907f29e2c"><span class="fakelink-icon" data-ref="4d419025-7416-4776-9e27-0323d2888371">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Altogether, these data show the formation of a PS-like ring is dependent on FGF signaling through ERK.</p><figure class="figure tofill imgs-1 longcaption fig-portrait pagedjs-fill-next-page" id="fig1" data-ref="6b4b05b4-0bd8-489c-8b02-4f729eb11503" data-id="fig1" style="display: none;"><figcaption class="figure__caption" id="el-figcaption-0" data-ref="ff235660-d0b2-496e-8134-ce02057ec1c9" data-id="el-figcaption-0"><h3 id="el-h3-0" data-ref="f27cee53-6f4e-401f-8983-4605449ef255" data-id="el-h3-0" data-break-after="avoid"><span class="label figure-name" data-ref="0e27e2e6-ce1c-4ba0-bb58-50abc64fd442"><label class="figure__label" data-ref="5d980441-9a6b-47b7-b49a-8fa3d40e2126">Figure 1.</label></span></h3><p id="el-p-16" data-ref="4d0bdaaf-61fe-46f0-a036-38521e673e43" data-id="el-p-16" data-previous-break-after="avoid"><strong data-ref="6144b037-de80-4538-8f27-8886fa176be3">a)</strong> Immunofluorescence stainings for pERK and TBXT and their radial intensity profiles in 2D gastruloids at different times. Error bars in graphs are standard deviations over N=4 colonies. <strong data-ref="9c7c4b0e-6f20-4831-ad3b-588b1c127553">b,c)</strong> Effect FGF receptor inhibition (FGFRi, b) or MEK inhibition (MEKi, c) on pERK and TBXT in colonies fixed at 42h after BMP treatment. Inhibition for either 30 minutes at 41.5h (rounded to 42 in figure) or throughout differentiation. <strong data-ref="b90c60c3-5e15-4925-8b5a-9a10abe7887d">d)</strong> BMP only control and FGFRi inhibition in cells expressing doxycycline-inducible SOS<sup data-ref="98499325-0a07-47c9-9177-3b3da6355fbd">cat</sup> with and without addition of doxycycline. <strong data-ref="1e07ebf4-2307-4739-b16a-5cbba2494768">e-f)</strong> Cell numbers (e) and radial profiles of TBXT positive cells (f) for conditions in (d). Error bars represent standard deviation. Statistical significance was assessed using one-way ANOVA followed by Tukey’s HSD test for pairwise comparisons. <strong data-ref="b068f09c-9a60-4848-88e1-7f92457e9d20">g)</strong> PS marker TBXT, pERK and pluripotency marker SOX2 in Nodal knockout cells with inhibitors of Wnt secretion (WntSeci) and BMP receptors (BMPRi) with or without MEKi or FGFRi. All scale bars 50um.</p></figcaption><img class="child-of-figure has-boundary imageonly portrait max-100" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_fig1.tif/full/max/0/default.jpg" alt="" id="el-img-1" data-ref="4760eca6-0cd0-4033-b262-f501a6160eea" data-id="el-img-1" height="2800" width="2112" data-desired-width="100"></figure><p id="el-p-17" data-ref="8e0cbdfd-59f9-48e7-b877-8e173ae419c2" data-id="el-p-17">This raised several questions. First, is FGF/ERK signaling required directly for PS-like differentiation, or does it act indirectly? These possibilities are not mutually exclusive. For example, FGF/ERK could be required directly but also act indirectly by controlling Wnt or Nodal expression, as both Wnt and Nodal signaling are required for PS-like differentiation. Second, which are the FGFs activating the ERK pathway? The pathway could be activated by exogenous FGF2 present in the pluripotency medium mTeSR1 or by endogenously expressed ligands. Third, how is ERK activity restricted to a ring?</p><p id="el-p-18" data-ref="50069754-d2c4-4ca2-bdd4-ce5ed59aaaa4" data-id="el-p-18">To determine if there is a direct requirement of FGF/ERK signaling and whether this depends on exogenous FGF, we performed differentiation to PS-like cells in standard culture by exogenous activation of Wnt and Nodal signaling, bypassing the need to form endogenous Wnt and Nodal gradient as in 2D gastruloids<sup data-ref="a60517d4-259b-4cbb-8a21-208a85841985"><a href="#c38" data-ref="5e2eea70-9977-4cd0-9b76-c9dd9ec7cb2c">38</a></sup>. We used a minimal base medium (E6) without exogenous FGF or other growth factors. To rule out any additional requirement for endogenous Nodal and Wnt which may be FGF/ERK-dependent, we used Nodal knockout cells<sup data-ref="ab069c0e-d535-493d-b82a-ecd8a2c12f1c"><a href="#c3" data-ref="6e289ae1-0e73-4bc5-bb10-61ccd8838f8a">3</a></sup> treated with the Wnt secretion inhibitor IWP2, thereby eliminating any endogenous activity of these pathways. We also inhibited BMP receptors using LDN193189 to rule out any indirect effect through endogenous BMP. Under these conditions, with exogenous Wnt and Nodal activation for 24h in the absence of exogenous FGF, cells efficiently differentiated to TBXT+ PS-like cells (<span class="fakelink linktofig" data-ref="28099cb2-5e7a-48ba-ba0b-88fc27f1b789"><span class="fakelink-text" data-ref="4b80783e-2ea7-41b4-b78d-17fed4a9726a">Fig. 1g</span><a class="fake-icon" href="#fig1" data-ref="890976f1-b5b0-4843-a1fb-2dcf26001ead"><span class="fakelink-icon" data-ref="239c3ff8-dbc9-4da2-8755-978d74d8c03b">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="3b2cd05f-24a3-407c-b0ec-307bc4188581"><span class="fakelink-text" data-ref="7103957e-87fe-4974-85a6-5434ffb04cbf">Supp. Fig. 1g</span><a class="fake-icon" href="#figS1" data-ref="6eeb2700-f362-4ee9-8d09-e2ee532710e7"><span class="fakelink-icon" data-ref="5e1c4ea4-3109-4070-afd3-48a81cb6200d">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). These PS-like cells had high active ERK levels just as in the 2D gastruloids. Inhibition of either FGFR or MEK abrogated ERK activity and TBXT expression while FGFR inhibition also severely reduced proliferation. In both cases significant expression of the pluripotency and ectoderm marker SOX2 was maintained (<span class="fakelink linktofig" data-ref="7a076250-829b-4ea6-9b90-fbcdb34c8515"><span class="fakelink-text" data-ref="6eb09e3f-512a-4934-ba91-0017b9a89de6">Supp. Fig. 1g</span><a class="fake-icon" href="#figS1" data-ref="ba3ad64d-0be4-464d-96eb-4e3b5725f0f7"><span class="fakelink-icon" data-ref="f75d391f-8a41-4fba-ad66-d08b2eb33677">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Given that FGF/ERK is known to be involved in pluripotency maintenance, this suggests inhibition may not be complete and that much lower ERK activity is required for pluripotency maintenance than PS-like differentiation, consistent with lower ERK activity in the pluripotent center of gastruloids.</p><p id="el-p-19" data-ref="f274fe20-7125-40b5-8a7b-45cebac52aa7" data-id="el-p-19">MEK and FGFR inhibition reduce proliferation and thereby cell density, raising the possibility that their lack of differentiation is an indirect effect that depends on cell density: a critical density may be required for differentiation due to a community effect. To rule this out we performed the experiment across a range of densities, which confirmed that PS differentiation in the control was efficient even at the lowest densities while FGFR and MEK inhibition effectively blocked differentiation when compared to controls with similar final cell densities (<span class="fakelink linktofig" data-ref="de17bbbd-3de7-45ea-bb7d-e445e2e73445"><span class="fakelink-text" data-ref="ddc13506-75ed-4468-8046-ecc5e19028e2">Supp. Fig. 1h</span><a class="fake-icon" href="#figS1" data-ref="1ae5ae31-728b-46d8-9568-d10052429f74"><span class="fakelink-icon" data-ref="17c3b0c2-e1ee-4fb9-8b11-7b2d6877582c">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). We also tested whether proliferation and differentiation require similar levels of ERK signaling by determining the MEKi dose response of gastruloids. This revealed that lower doses of MEKi effectively block differentiation without strongly affecting cell number, further supporting that the effects of ERK signaling on proliferation and differentiation are independent (<span class="fakelink linktofig" data-ref="cf4ecd67-48d6-4e36-b760-15ad4ee465ca"><span class="fakelink-text" data-ref="cf0a44af-2a07-4406-98cd-aae4a56ce5d9">Supp. Fig. 1i</span><a class="fake-icon" href="#figS1" data-ref="519db83a-05de-423c-b6b9-e3a6b127b8b3"><span class="fakelink-icon" data-ref="12e4b407-6789-484e-b071-3ba6080b3980">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Altogether, our results show that endogenous FGF expression is both necessary and sufficient for PS-like differentiation by exogenous Wnt and Nodal stimulation and demonstrate a direct requirement for ERK signaling in PS-like differentiation.</p></section><section id="s2b" data-ref="921dc64d-5d0b-4e06-b347-13a7f32896c1" data-id="s2b"><h2 class="heading-2" id="el-h2-2" data-ref="cfe8fc0a-9735-42c4-b292-4d978be4e396" data-id="el-h2-2" data-break-after="avoid">Single cell RNA-sequence reveals differentially expressed FGF pathway components</h2><p id="el-p-20" data-ref="9e5f2bdb-0ad0-4c6f-8a50-1f5b390ec298" data-id="el-p-20" data-previous-break-after="avoid">The above results suggest endogenous FGF expression may contribute to elevated ERK signaling in the PS-like ring. However, FGF/ERK signaling may be modulated at many levels (<span class="fakelink linktofig" data-ref="0e27eb01-7dfa-4e54-a79d-b4429f90affd"><span class="fakelink-text" data-ref="0b30c038-3fa0-4775-a934-00d147663cc1">Fig. 2a</span><a class="fake-icon" href="#fig2" data-ref="d4d9eefc-4e37-4f68-b0a2-a548ccc30353"><span class="fakelink-icon" data-ref="58248a51-f47f-47e4-8704-d1ee99a25636">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). A spatial pattern of ERK activation could reflect a concentration gradient of FGF. Alternatively, differential expression of receptors and HSPG-modifying enzymes could lead to spatially patterned receptor activation and downstream ERK activity even in the absence of an FGF gradient. Even if receptor activation is uniform, downstream ERK activity could be spatially patterned by downstream negative regulators of ERK activity such as Sprouty-family proteins (SPRY) and dual-specificity phosphatases (DUSPs)<sup data-ref="05639086-d987-4771-9351-e54b0f61af95"><a href="#c40" data-ref="f3311c74-81b1-4278-8d57-d86bd2c64821">40</a></sup>.</p><figure class="figure tofill imgs-1 longcaption fig-landscape pagedjs-fill-next-page" id="fig2" data-ref="0b4d6bd6-67d8-4594-bea8-6adc45883ff8" data-id="fig2" style="display: none;"><figcaption class="figure__caption" id="el-figcaption-1" data-ref="a9c01dfb-a630-49f3-90bb-2c20b921dfb6" data-id="el-figcaption-1"><h3 id="el-h3-1" data-ref="60ea95c4-0e04-4cd7-a150-28d2755b49a1" data-id="el-h3-1" data-break-after="avoid"><span class="label figure-name" data-ref="489a2676-92f8-457b-93e3-d32307ebb7a1"><label class="figure__label" data-ref="de96ebfa-2e73-4699-87f1-3616be0f1175">Figure 2.</label></span></h3><p id="el-p-21" data-ref="91381679-f54a-4ab7-a0eb-91b6c694bc1b" data-id="el-p-21" data-previous-break-after="avoid"><strong data-ref="e8267bb2-a876-4c75-b8c3-9eecf3affb51">a)</strong> Schematic of the FGF/ERK signaling pathway. <strong data-ref="1cb2104d-db7e-4608-b544-06d7f1885c0e">b)</strong> Distribution of pERK levels in TBXT+ cells versus ISL1+ and other cells based on data and thresholds in <span class="fakelink linktofig" data-ref="e3c96de8-b0e8-402c-93ba-f39879c9d498"><span class="fakelink-text" data-ref="916ebffb-8a29-4feb-a647-199cdb2aa07c">Supp. Fig. 2b</span><a class="fake-icon" href="#figS2" data-ref="703b1469-fcd0-4a06-9f5b-85ef260ce77a"><span class="fakelink-icon" data-ref="e9e25641-c0a5-4034-bc3f-01c80300b664">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. <strong data-ref="96fdf513-d2c0-461f-a383-475fec7390d5">c)</strong> Expression of key markers in PHATE<sup data-ref="416786f6-38ad-4446-a83f-74797f92b158"><a href="#c39" data-ref="97786a28-60f7-4849-89ef-ce24531b71ee">39</a></sup> projection of scRNA-seq data for 42h colony. <strong data-ref="a36422e1-af55-4d7f-9bcc-257d14fc44be">d)</strong> Cell type annotation. PS-LC: primitive streak-like cells, NasMe: nascent mesoderm, AE-LC: amniotic ectoderm-like cells: PGC-LC: primordial germ cell-like cells, endo: definitive endoderm. <strong data-ref="e7105e8f-1320-4013-9103-e87adc3fe8c4">e)</strong> Expression of canonical FGF ligands (log transformed and smoothened). <strong data-ref="549ba5e7-c91b-483a-a75d-a91c61901a4a">f)</strong> Comparison of FGF ligand expression in 2D human gastruloids to human, monkey, and mouse embryo, normalized within each type of sample (see methods). <strong data-ref="c7040d38-2a4d-4d3f-9bb7-2024f809d92f">g-i)</strong> Expression of several genes involved in FGF signaling with strong differential expression between PS-like cells or nascent mesoderm and pluripotent cells, split by gene category.</p></figcaption><img class="child-of-figure has-boundary imageonly landscape max-100" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_fig2.tif/full/max/0/default.jpg" alt="" id="el-img-2" data-ref="b70215fb-b6d6-4fd7-8a41-856762a4f96a" data-id="el-img-2" height="931" width="2102" data-desired-width="99.52651515151516"></figure><p id="el-p-22" data-ref="984ec47b-4290-452e-b0fc-9a095861c0a9" data-id="el-p-22">To understand which genes modulating FGF/ERK signaling could be responsible for the elevated ERK signaling in TBXT positive cells and required for TBXT expression (<span class="fakelink linktofig" data-ref="f3b7123b-99f0-441d-858d-8aa14f589f0e"><span class="fakelink-text" data-ref="c2a58514-d1a9-4c4a-ad01-102b413cf787">Fig.1</span><a class="fake-icon" href="#fig1" data-ref="8dd2dcb4-0fa5-4055-ae39-e82db2588c60"><span class="fakelink-icon" data-ref="dc4f7c10-1463-4083-b718-38ef9ec97407">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="c9a0cbff-58ea-4caa-8cc5-ff66674248ac"><span class="fakelink-text" data-ref="9bd0f181-fddb-4d41-b9c5-cf25474ae05d">Fig. 2b</span><a class="fake-icon" href="#fig2" data-ref="fef66da1-056f-4e76-bdea-1787e0f8185e"><span class="fakelink-icon" data-ref="84286886-a3a4-4831-972e-b80b5502ed7b">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>), we used single cell RNA-sequencing data from one previously published sample<sup data-ref="26247d56-23c1-471f-ab10-dcf3d5154c77"><a href="#c34" data-ref="a862e1eb-53be-4f65-b2ea-86e981ce98d6">34</a></sup> and one newly collected replicate (<span class="fakelink linktofig" data-ref="6e2ec2f7-e6d4-4356-90e2-fda61911049d"><span class="fakelink-text" data-ref="beb34b40-e5e7-4cec-8815-884730eb1c73">Supp. Fig. 2a</span><a class="fake-icon" href="#figS2" data-ref="19dc5b38-b92f-49af-9443-0de96d3d18bb"><span class="fakelink-icon" data-ref="2ce64080-0d6b-4371-b3f6-6f71fcb72474">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>) to determine expression of FGF pathway components in human 2D gastruloids, extending earlier work describing FGF expression<sup data-ref="c4bec304-7368-48a4-b169-1ff2e5e46037"><a href="#c25" data-ref="453ef03d-d05a-480c-932f-da91dd51e12c">25</a>,<a href="#c35" data-ref="9720bf41-955b-4721-82dc-365af3255562">35</a>,<a href="#c36" data-ref="01293238-8c0c-46ca-9d98-384df553d091">36</a>,<a href="#c41" data-ref="2d7a84d4-e284-4d03-8c0a-31dfbda6a14b">41</a></sup>. TBXT-positive PS-like cells give rise to nascent mesoderm expressing TBX6 as well as definitive endoderm expressing SOX17 and FOXA2, both of which may still express TBXT at 42h<sup data-ref="b6f9847f-6c73-4933-bdb3-119e2ff4598a"><a href="#c42" data-ref="ac631ac9-9d4a-4aed-af53-2216c5ea5c2a">42</a></sup>. To understand how <em data-ref="6834592c-ba3e-42da-ba6f-e654bc633b9c">TBXT</em> expression is initiated in an FGF-dependent manner, we focused on differential expression in the PS-LCs relative to pluripotent cells, reasoning that FGF modulation specific to nascent mesoderm or endoderm is a consequence rather than a cause of PS-LC differentiation. We identified different cell types based on marker genes with the PS-like cells defined as <em data-ref="1be067db-c701-40a8-8093-cbcf79509e59">TBXT+MIXL1+POU5F1+TBX6-SOX17-FOXA2-</em>(<span class="fakelink linktofig" data-ref="f872b55c-aabd-46d3-b00c-fbaec0186164"><span class="fakelink-text" data-ref="f861132e-5a63-4afd-8185-7a8e743b1c66">Fig. 2cd</span><a class="fake-icon" href="#fig2" data-ref="4358d432-3df8-4926-86af-b25776b84398"><span class="fakelink-icon" data-ref="47e681b1-0119-4a00-90a9-ac019811e447">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="9924934f-13b9-4e06-96c6-c3fd8ac841ec"><span class="fakelink-text" data-ref="229504a2-26e6-48e8-8e1f-a7ff1bb2cde3">Supp. Table 4</span><a class="fake-icon" href="#tblS4" data-ref="2bfce3ea-7146-4d31-b20d-8fb333889636"><span class="fakelink-icon" data-ref="40666c1d-97a8-4221-931a-42c4123e184e">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). We then analyzed expression of different categories of FGF/ERK regulators.</p><p id="el-p-23" data-ref="aee6f288-ecbc-4633-adc4-9d0f611de2ab" data-id="el-p-23">We found four FGF ligands with significant expression: <em data-ref="3f71d0e0-454c-48b8-a560-06a625b30aaf">FGF2, 4, 8,</em> and <em data-ref="d790d46c-e6d6-42eb-99c1-ee6b0b2b327e">17</em> (<span class="fakelink linktofig" data-ref="13cee253-3256-4c84-b632-f83255ab3294"><span class="fakelink-text" data-ref="9769c11f-ad76-4367-a1c2-b397eaf8e684">Fig. 2e</span><a class="fake-icon" href="#fig2" data-ref="c9593933-3f66-4360-b1b5-06f81e69d946"><span class="fakelink-icon" data-ref="23a2de28-5244-4c1d-b995-05670f46bab7">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="fc09b5fe-1790-4a9b-8141-0d948e06e93c"><span class="fakelink-text" data-ref="0fe52171-0f32-433c-8098-db228aaa752d">Supp. Fig. 2b</span><a class="fake-icon" href="#figS2" data-ref="9ad8e2db-4521-450b-8c08-dd5c77862a06"><span class="fakelink-icon" data-ref="44f7eb85-eed6-47df-932d-6edc1fc62e9e">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). We then compared the expression of these ligands in PS-like cells to reference data of CS7 human, CS8 monkey, and E6.5-E7.5 mouse gastrulation (<span class="fakelink linktofig" data-ref="6c756a29-9d15-4072-8e17-4ae8640164d1"><span class="fakelink-text" data-ref="336df8ab-fb14-4573-82b8-209de76f840f">Fig. 2f</span><a class="fake-icon" href="#fig2" data-ref="379fa510-2b51-41fe-92c7-9c847e483d0d"><span class="fakelink-icon" data-ref="cc8493b7-3bb5-4b3a-99b2-2b3ab61f9488">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="44d6ab9e-e1ca-4eb9-ad46-386bf9efdd9d"><span class="fakelink-text" data-ref="c289f0ff-b3b2-42f6-ba86-f26d41c01a70">Supp. Fig. 2cd</span><a class="fake-icon" href="#figS2" data-ref="9f69650e-7241-4101-a217-00054d3514e7"><span class="fakelink-icon" data-ref="9efbfc61-0b0e-4893-a660-a0a88c6c8993">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). We also included <em data-ref="cf3f14b9-3670-4c1f-b93f-a49c4b880ae4">FGF3</em> and <em data-ref="8147d87d-218b-4c80-9547-50fd299e9bad">FGF5</em>, which are expressed during mouse gastrulation<sup data-ref="8622ddb1-9d3c-478a-b69d-86b1b35b365e"><a href="#c32" data-ref="c2f6e2da-d019-47a6-9b8b-c7c9c34636ba">32</a>,<a href="#c43" data-ref="0500ad32-e022-466d-84c5-52fee0721e60">43</a>,<a href="#c44" data-ref="6b0b3e48-2330-48d7-8fbd-ffbae9217253">44</a></sup>. High expression of <em data-ref="24976637-27ec-4772-b2c0-00a389ce3e13">FGF4</em> and <em data-ref="fdaaf24d-d141-499f-81ab-5701f128f489">FGF17</em> and low expression of FGF8 was consistent between human, monkey, and 2D gastruloid. <em data-ref="e48de8b2-0ee5-48e6-bf08-1f648235eed2">FGF2</em> expression was also high in the PS-like populations in 2D gastruloids and the CS8 monkey embryo, but much lower in the human embryo data. Furthermore, the monkey and human embryos expressed <em data-ref="e87d1f3e-e936-4ae9-a61a-fe763f4988df">FGF3</em>, which was minimally expressed in gastruloids. A similar comparison in nascent mesoderm showed much lower <em data-ref="700fd0ff-8f77-4ffd-99c4-9ad9a880dfcc">FGF2/4</em> and much higher <em data-ref="cc174fd2-b371-4a2a-b035-e55b6bcae83b">FGF17</em> compared to PS across all primate samples (<span class="fakelink linktofig" data-ref="0e967b29-afac-4106-8225-dd6ff36c6bc7"><span class="fakelink-text" data-ref="cfebc816-e75e-4a3b-95d5-cb2326da3ae0">Supp. Fig. 2e</span><a class="fake-icon" href="#figS2" data-ref="c2fbb7aa-e1a1-4698-93b1-2ab6bed2df05"><span class="fakelink-icon" data-ref="8b5dc7c4-19a2-412f-8c33-ad01bd56e382">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Expression in 2D gastruloids was consistent with published RNA-seq data for hPSC-derived PS-like cells, which expressed high <em data-ref="d5988da5-8ca0-41d0-9895-4665c39bda46">FGF2</em> and <em data-ref="5f16546d-8d49-4e69-8697-c4f99c7ddc1a">FGF4</em>, some <em data-ref="5ee1132f-9d4b-4e49-a947-9795264a27e0">FGF8</em> and <em data-ref="827238ef-4e85-418e-8379-8eaeea473c16">FGF17</em>, and no <em data-ref="7455a114-de4f-408b-a455-e80270ac2d4c">FGF3</em> (<span class="fakelink linktofig" data-ref="e82e38f6-98bb-4d06-aeba-8abd9404ca21"><span class="fakelink-text" data-ref="3afbc49d-6524-4df1-a522-d987f9bbf501">Supp. Fig. 2f</span><a class="fake-icon" href="#figS2" data-ref="fa368af4-af21-4853-a7f7-55efe0609f94"><span class="fakelink-icon" data-ref="cb7648be-d330-4d8c-b0e9-22e410ca6d06">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Comparison with mouse confirmed significant differences from primates, with high expression of <em data-ref="88672081-4ed9-49d2-a25e-cab06a69a5d0">FGF5, 8</em>, much lower expression of <em data-ref="4f28de84-869a-4051-ac56-401b4e0e31a3">FGF4,17</em>, and no expression of <em data-ref="cba5c2db-4740-4605-b8a8-6e0af3909dd1">FGF2</em> in the mouse<sup data-ref="9aa15ca5-d634-4fb0-b40a-60d22a963bae"><a href="#c21" data-ref="bbc3080a-90d1-450e-8bd4-1f579972c975">21</a>,<a href="#c43" data-ref="9fe28a10-293f-4d8c-b33d-2b2eaf9de739">43</a>,<a href="#c44" data-ref="e7aed878-d62f-4e72-96bb-3c4fc6f6790e">44</a></sup>.</p><p id="el-p-24" data-ref="3c5f87af-6823-4cbc-9dca-78868e6f08a3" data-id="el-p-24">Of the FGF receptors, only <em data-ref="815eb87e-82d5-4cb9-aa70-d79abf07b418">FGFR1</em> and <em data-ref="262d785c-acbe-410e-9f1d-bd81f3f0427b">FGFR4</em> were strongly expressed, with <em data-ref="4af28172-f257-4a50-b182-3638d0be755f">FGFR1</em> expression about 5-fold higher (<span class="fakelink linktofig" data-ref="f40a2b0e-ad50-416c-8570-5f582f6526a1"><span class="fakelink-text" data-ref="5ec2a0de-5325-4892-99cc-178a8c9d1084">Fig. 2g</span><a class="fake-icon" href="#fig2" data-ref="559d051e-2360-4b58-a7fb-c902370d75d1"><span class="fakelink-icon" data-ref="fc0f3061-30f7-41d6-9a54-3cc3706b3432">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="864fda46-bfb9-4f4f-8618-1e74e0d7a769"><span class="fakelink-text" data-ref="6d18eaf5-f445-456f-ad70-e0c1e7a1918c">Supp. Fig. 2g</span><a class="fake-icon" href="#figS2" data-ref="135027dd-7d2c-4d2b-8ed1-3a50a7cfe046"><span class="fakelink-icon" data-ref="88f8419f-ff83-4a21-a5a7-5a2c160d8354">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Both receptors showed graded expression that decreased from pluripotent to PS-like cells and much lower expression in amnion-like cells (<span class="fakelink linktofig" data-ref="70784367-e297-4de0-a836-664a27f64e1e"><span class="fakelink-text" data-ref="7e794d4d-1a48-4c0c-a94f-63d0dd084dab">Fig. 2g</span><a class="fake-icon" href="#fig2" data-ref="74c8f798-f58b-43c5-ae1b-50c8ecd92e25"><span class="fakelink-icon" data-ref="7e54b63a-52aa-4706-a1b7-9443ec6165c5">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Therefore, elevated ERK activity in PS-like cells cannot be accounted for by overall receptor expression. Differential expression of genes regulating heparin-sulfate proteoglycans which modulate FGF receptor binding was mild except for glypicans 3 and 4 which were downregulated in PS-like cells relative to pluripotent cells (<span class="fakelink linktofig" data-ref="226ddddf-e3c1-4eba-b104-88fee4de25f1"><span class="fakelink-text" data-ref="07bc4223-24b1-4b65-a5eb-9ec0f082ff88">Fig. 2h</span><a class="fake-icon" href="#fig2" data-ref="75738fe9-06e0-4c80-9fa0-dc6449a6ce2c"><span class="fakelink-icon" data-ref="5bce0fee-bf83-46c6-8aad-45c859f3c75a">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="454eb9cb-d425-41b4-8ecb-07390c3106f2"><span class="fakelink-text" data-ref="b71b494d-56c2-46d7-99cc-8020481c65e5">Supp. Fig. 2g</span><a class="fake-icon" href="#figS2" data-ref="8c1eae34-5a91-441e-a5c7-6e278675088d"><span class="fakelink-icon" data-ref="9fe7eb5d-0f08-4d27-becc-3a63e6700e5b">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). In contrast, several intracellular negative regulators of ERK signaling were strongly differentially expressed between cell clusters, including <em data-ref="5a151048-54ea-4a46-813a-bc9008c2be25">SPRY1</em> and <em data-ref="90d8f61d-3ab6-4445-a8cf-0ccdb8228a9c">PEBP1</em> which were significantly higher in pluripotent cells (<span class="fakelink linktofig" data-ref="a27e20ab-48a1-4ce3-bbab-bc0c3008bfa1"><span class="fakelink-text" data-ref="ce98f383-5a01-4a20-8607-3e0a19bcb835">Fig. 2i</span><a class="fake-icon" href="#fig2" data-ref="b01e9d93-4af3-4499-8652-bd23825cf163"><span class="fakelink-icon" data-ref="5d628ec1-8284-4a71-8ea8-5d80575732ad">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="778587a2-a95d-4489-b9cd-578e1c28a292"><span class="fakelink-text" data-ref="7185d9f8-5e6a-45c6-b36e-4e072b65bf0a">Supp. Fig. 2g</span><a class="fake-icon" href="#figS2" data-ref="d9c87dea-8c1d-4e1c-8879-d1b1e2fb0399"><span class="fakelink-icon" data-ref="c9c83e07-e1ce-401b-b3e3-25c755ef1135">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Surprisingly, however, expression of <em data-ref="01b18b09-766e-452f-9578-2294d0b2466a">SPRY4</em> was not elevated in PS-like cells where ERK signaling is high, suggesting that transcription of <em data-ref="1f61f3dc-8e41-4c99-b58c-d9db21fb629d">SPRY4</em> may not be a good readout of ERK activity in human hPSCs like it is in the mouse<sup data-ref="29f4b7bf-4fc1-44c0-acbc-8a4dacd61a48"><a href="#c45" data-ref="b898f7d3-f69a-4df7-8b07-11ef0b9291e1">45</a></sup>. On the other hand, expression of the ERK negative feedback regulator <em data-ref="6731c8a7-000a-48ee-af5d-1973245c14f0">IL17RD</em> (<em data-ref="aa262715-cc27-4c7e-9ad4-0b0041642c6d">SEF</em>) does seem to approximately reflect expected ERK activity in each population. Components of the core ERK pathway including ERK itself were highly expressed across all clusters (<span class="fakelink linktofig" data-ref="fbc50fa5-fa8e-4ade-b790-241f0b9c7fba"><span class="fakelink-text" data-ref="a11fb569-22af-4d22-8acb-529f6a8072c0">Supp. Fig. 2g</span><a class="fake-icon" href="#figS2" data-ref="1afbfe5c-e3df-4c5d-afcd-90a965a3ac2f"><span class="fakelink-icon" data-ref="9d974454-bc4b-466b-9e40-da68c9b18288">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>).</p><p id="el-p-25" data-ref="78333755-a12f-402b-b2b3-ad3d25b6a020" data-id="el-p-25">Thus, while scRNA-seq identified specific genes involved, it is consistent with several non-exclusive mechanisms for generating an ERK pattern. For example, uniform FGF2 could form an ERK ring by equally activating receptors across the colony but with ERK response reduced by elevated expression of negative regulators in the pluripotent center. Alternatively, based on gene expression we cannot rule out HSPGs increasing receptor binding specifically in the ring. Finally, localized expression of FGF4 or FGF17 in the PS-like ring could lead to a concentration gradient responsible for increased receptor activation.</p></section><section id="s2c" data-ref="9618f35a-9f07-4fad-a0c7-087ba1366e4f" data-id="s2c"><h2 class="heading-2" id="el-h2-3" data-ref="51438d56-88f0-4f96-a158-a8e5f348da2b" data-id="el-h2-3" data-break-after="avoid">The phosphorylated ERK pattern is due to differential activation of FGF receptors</h2><p id="el-p-26" data-ref="731d8532-00c8-49a3-a923-b5d6d411edfe" data-id="el-p-26" data-previous-break-after="avoid">To distinguish between the different possibilities, we asked whether the spatial pattern of pERK signaling is controlled primarily below, at, or above the level of the receptors. To address this, we first visualized phosphorylated FGFR1 (pFGFR1) and found its spatial pattern strongly resembles that of pERK (<span class="fakelink linktofig" data-ref="d22a7e98-0e0f-4dc5-9741-a1dd5b14b1db"><span class="fakelink-text" data-ref="e8b423a7-3780-4571-8fce-547f88f59949">Fig. 3a</span><a class="fake-icon" href="#fig3" data-ref="cf650579-6e06-4dc5-8805-d5875201b44f"><span class="fakelink-icon" data-ref="0e3bdf79-94d3-4bea-8a7f-34bb8a11e22e">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="5be3cbc2-4c68-4420-a53d-acd4baa3bf52"><span class="fakelink-text" data-ref="2a774437-da1b-4ce9-8620-2cf34d201113">Supp. Fig. 3a</span><a class="fake-icon" href="#figS3" data-ref="b62567e7-44af-469c-a2fc-5495e987186f"><span class="fakelink-icon" data-ref="8d70df6c-0904-4a91-97df-ecd84c48830f">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). As expected, pFGFR1 was lost upon brief FGFR inhibition (<span class="fakelink linktofig" data-ref="11478da1-aa89-410c-a184-aecd2e2dcc13"><span class="fakelink-text" data-ref="f8305254-d8ca-4ef5-9624-1c4af7c1dab4">Fig. 3a</span><a class="fake-icon" href="#fig3" data-ref="bfd5939d-7fdd-4697-840a-5276a9170181"><span class="fakelink-icon" data-ref="c09178be-ab41-4c03-adbc-a937fdf23bba">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="6c1601fb-50d4-45d7-91b4-73ff2c7a8d8a"><span class="fakelink-text" data-ref="b9728edb-bf61-49ad-8cf4-91f298972763">Supp. Fig. 3a</span><a class="fake-icon" href="#figS3" data-ref="95a315a7-136a-4482-8e09-a08b418d6e8a"><span class="fakelink-icon" data-ref="87b2a1cb-a82f-4923-becf-b983b023e685">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). In contrast, after MEK inhibition, high pFGFR1 remained while pERK was greatly reduced (<span class="fakelink linktofig" data-ref="530bd253-8928-4670-952d-c90abaea538c"><span class="fakelink-text" data-ref="bb581854-8466-49a7-9b7a-567a34dd6335">Supp. Fig. 3bc</span><a class="fake-icon" href="#figS3" data-ref="2f6fc573-f65b-48f3-9b45-c34365f25fcd"><span class="fakelink-icon" data-ref="87db1848-26a4-442e-a2a7-51b743bc9cdf">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). This suggests the spatial pattern of ERK reflects a spatial pattern of receptor activity.</p><p id="el-p-27" data-ref="552f4150-217e-4f13-9a78-9f4ee3c47b9e" data-id="el-p-27">The spatial pattern in receptor activity could either be due to differential receptor expression or differential receptor activation. There was no significant upregulation of FGF receptors in PS-like cells according to our single cell RNA-sequencing data (<span class="fakelink linktofig" data-ref="b46f6232-b1fb-496c-bb85-615930b2ed98"><span class="fakelink-text" data-ref="6f08b960-e526-479e-a422-200c5dcbcd85">Fig 2e</span><a class="fake-icon" href="#fig2" data-ref="7aa9f0c9-7af7-4410-ac8a-e2f993264315"><span class="fakelink-icon" data-ref="65fe31d8-eba2-42fb-aa35-e991b65ff809">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). In fact, the most highly expressed receptor, FGFR1, was elevated in pluripotent cells. This suggests the ERK activity pattern is not due to a receptor expression pattern with two caveats. First, mRNA expression may not reflect protein expression. Second, multiple isoforms exist for FGF receptors 1-3 (but not 4) that have different affinity for FGF ligands, leaving open the possibility that PS-like cells could express a different FGFR1 isoform than pluripotent cells, despite overall FGFR1 expression being approximately uniform.</p><p id="el-p-28" data-ref="394bc537-3333-4535-9899-7d37c3d6d152" data-id="el-p-28">To determine if the spatial pattern in pFGFR1 could be explained by a spatial pattern of FGFR1 protein, we performed immunofluorescence staining of FGFR1. Consistent with the scRNA-seq data, and in sharp contrast to the pFGFR1 stain, FGFR1 protein expression was graded from center to edge with its highest levels in the pluripotent center and therefore could not explain the signaling pattern (<span class="fakelink linktofig" data-ref="fcd5c0f1-135c-4ad5-8ade-5a7209f37b9f"><span class="fakelink-text" data-ref="56d10286-358e-4a0e-923e-04ad2207c4f2">Fig. 3b</span><a class="fake-icon" href="#fig3" data-ref="f23f30ee-b72c-4187-b88c-ae67949067be"><span class="fakelink-icon" data-ref="12a91ccd-00b9-413e-bd27-82ebf40cd8f7">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="0f1e0d14-20a2-4488-a1ab-23003449fd71"><span class="fakelink-text" data-ref="505d7add-f5c4-49c7-9a69-05ee82fc5654">Supp. Fig. 3d</span><a class="fake-icon" href="#figS3" data-ref="e668207d-9693-46ee-8f71-1ba41a22db9c"><span class="fakelink-icon" data-ref="d9c96f01-93ed-479f-93a9-ae640769ddcc">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). To determine if different isoforms of FGFR1 are differentially expressed in PS-like vs. pluripotent cells, we designed qPCR primers to detect the IIIb and IIIc isoforms of <em data-ref="0c3f1f1c-0d7a-4ca4-a23b-75dfd6a432cc">FGFR1</em> (<span class="fakelink linktofig" data-ref="cb170996-20b0-4dbf-8aa1-03e049d328cd"><span class="fakelink-text" data-ref="f407fad6-7a1f-4f22-91da-db28db641a9d">Supp. Fig. 3e-g</span><a class="fake-icon" href="#figS3" data-ref="032d2e51-ef8c-4a1b-b54e-aa080333f912"><span class="fakelink-icon" data-ref="ce710721-ebd2-4b59-9f5e-e044802e98b0">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). These are the most common isoforms, also known as the epithelial and mesenchymal isoforms, and therefore appeared the mostly likely candidates for differential expression. We then analyzed expression of these isoforms in 42h micropatterns versus pluripotent cells and PS- like cells differentiated directly as in <span class="fakelink linktofig" data-ref="82bbdded-4f66-4388-b68c-861f963589db"><span class="fakelink-text" data-ref="a59777a9-9bd5-443a-84d1-eff7db269ba9">Fig. 1d</span><a class="fake-icon" href="#fig1" data-ref="4a76a405-892c-4ef7-9e90-95b44174f730"><span class="fakelink-icon" data-ref="25c06b4d-fec3-485f-88d7-8dcedf1e0ab0">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. However, the change in expression between samples was similar for each isoform, with all isoforms decreasing in 2D gastruloids and directly differentiated PS-like cells relative to pluripotent cells (<span class="fakelink linktofig" data-ref="8cad0582-7b8f-4543-917c-fbad3cec4f1d"><span class="fakelink-text" data-ref="a1d5fb23-5823-4894-b9a4-415a421c23dd">Fig. 3c</span><a class="fake-icon" href="#fig3" data-ref="b6b50f6e-35d0-4c07-8f78-67e32cf12e9a"><span class="fakelink-icon" data-ref="60ad412c-33a9-4a51-8935-3ad14e3b3945">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). To corroborate this and also determine absolute FGFR isoform expression in pluripotent and PS-like cells, we analyzed our single cell RNA-sequencing data and previously published bulk RNA-sequencing data for isoform expression in these two cell types. This revealed that for both pluripotent and PS-like cells FGFR IIIc is the dominant isoform with on average almost 900-fold higher expression than IIIb, while IIIa was not expressed at all (<span class="fakelink linktofig" data-ref="021da908-bb5f-4584-b0c8-32db14e8a3ca"><span class="fakelink-text" data-ref="0841a726-6b9a-47fe-b3c5-bf66236b998b">Fig. 3d</span><a class="fake-icon" href="#fig3" data-ref="d7372c47-5195-41b3-b085-496ef836e53f"><span class="fakelink-icon" data-ref="8e55c468-727e-43b1-89e5-459e405ede4c">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Altogether, these data suggest that spatial patterning in ERK is due to differential activation of FGFR receptors, either due to FGF concentration gradients or possibly spatially patterned HSPGs. Immunostaining for HS showed no clear correlation with TBXT or the expected ERK signaling pattern which is similar but extends further inward (<span class="fakelink linktofig" data-ref="302056bb-e17f-4576-b901-94b1aac6e6bc"><span class="fakelink-text" data-ref="82240c75-452b-4270-8e40-a3c0f18f7103">Supp. Fig. 4</span><a class="fake-icon" href="#figS4" data-ref="8220761e-e496-454c-9fc2-e0988323cf83"><span class="fakelink-icon" data-ref="65e23d18-123f-4fbd-b4f3-d6531ee80511">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). This does not rule out a role for HSPGs in spatial modulation, since FGF signaling depends on specific sulfation patterns rather than overall HS levels<sup data-ref="7b5957f9-31d8-45d7-bce4-37a9f2fb1ca8"><a href="#c46" data-ref="ce71ff93-cb60-423f-957b-17c3ffeb8518">46</a>,<a href="#c47" data-ref="718d8013-1e53-46b0-b17e-c02991d76f32">47</a></sup>. Nevertheless, it led us to focus on FGF gradients as the most likely cause for the ERK signaling pattern.</p><figure class="figure tofill imgs-1 shortcaption fig-square pagedjs-fill-next-page" id="fig3" data-ref="456b6ea0-27b5-46b0-b56e-b67e32cb9095" data-id="fig3" style="display: none;"><figcaption class="figure__caption" id="el-figcaption-2" data-ref="5d2d6b17-8b67-401c-8b51-0475d0333cd3" data-id="el-figcaption-2"><h3 id="el-h3-2" data-ref="f91170f6-599e-4dd3-998c-cbefebea1955" data-id="el-h3-2" data-break-after="avoid"><span class="label figure-name" data-ref="66c23f75-b0d3-4937-acc0-6b92ef5a0828"><label class="figure__label" data-ref="1eb95cf5-e1ba-415d-bf9a-35336eec0bd4">Figure 3.</label></span></h3><p id="el-p-29" data-ref="1a04647b-90db-4982-a008-f4d23936e284" data-id="el-p-29" data-previous-break-after="avoid"><strong data-ref="b2bdc082-b547-4386-ac49-562fb0b15727">a)</strong> phospho-FGFR1 stain with and without 1h FGFRi treatment at 41h, radial intensity profile on the right. <strong data-ref="a701e40f-d7b1-452a-8f9c-782f046fd3dd">b)</strong> FGFR1 with and without 1h FGFRi treatment at 41h, radial intensity profile on the right. Error bars in (b,c) represent standard deviation over N=4 colonies. <strong data-ref="4cbcafb7-768a-490a-9b2a-8262df75ace8">c)</strong> qPCR data for relative expression of FGF receptor 1 isoforms IIIb, IIIc and total FGFR1 in pluripotent cells (pluri), gastruloids at 42h (MP), and directly differentiated PS-like cells (dPS). Error bars represent standard deviation of technical triplicates. <strong data-ref="1cedec5b-4de0-48d8-8b5f-514a36728e57">d)</strong> Absolute FGFR1 isoform expression from RNA-seq for anterior/mid primitive streak and pluripotent cells from Loh 2016<sup data-ref="0356259d-2b89-4e23-b49d-0697eeecd834"><a href="#c38" data-ref="a4f405da-913b-43e5-9b94-e69906f8d623">38</a></sup> (Loh_ APS/MPS/HESC) as well as the pluripotent and PS-LC clusters from <span class="fakelink linktofig" data-ref="7f8425a5-cd6f-4b48-81fe-0194d33fd6da"><span class="fakelink-text" data-ref="7290aea0-0615-479d-9c1f-368d0133b5cb">Fig.2</span><a class="fake-icon" href="#fig2" data-ref="31f2beaa-3167-4d48-9465-78ac81cb28d2"><span class="fakelink-icon" data-ref="0f6c8b21-e09a-4e25-8c93-f58d6009c77a">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. All scale bars 50 um.</p></figcaption><img class="child-of-figure has-boundary imageonly square max-70" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_fig3.tif/full/max/0/default.jpg" alt="" id="el-img-3" data-ref="c97cc055-7c1f-486f-a173-31119dbff22d" data-id="el-img-3" height="1287" width="1403" data-desired-width="66.42992424242425"></figure></section><section id="s2d" data-ref="bbe7e9a6-dfa8-4a80-84ff-a98329995d18" data-id="s2d"><h2 class="heading-2" id="el-h2-4" data-ref="f38e40e7-4199-4e81-b68d-56e912f9c0fb" data-id="el-h2-4" data-break-after="avoid">FGF receptors are localized basolaterally</h2><p id="el-p-30" data-ref="9c39607e-9439-4c78-87ea-ba9c239d7d60" data-id="el-p-30" data-previous-break-after="avoid">To maintain an endogenous FGF concentration gradient on the apical side of the colony which faces the medium, would require ligands to remain tethered to the surface. However, FGF has been reported to diffuse freely in the intercellular space<sup data-ref="3aa4e1a6-b83f-4e40-ab9c-7f4216b4e7d1"><a href="#c48" data-ref="8cca9aca-8de3-4d40-a44e-36a02ec1bedb">48</a></sup>. Moreover, there is FGF2 present in the medium that would be expected to uniformly activate FGF receptors. The situation is similar for BMP4, which is present uniformly in the medium for 2D gastruloids but does not cause uniform signaling activity<sup data-ref="a34af4af-758d-4952-b353-b7772b224277"><a href="#c4" data-ref="a2f4bc97-2340-4a85-bbc1-66b8692adb57">4</a></sup>. In the case of BMP, that is because receptors are localized basolaterally<sup data-ref="29e2d429-c0e6-4cbe-bebe-9b3aba8e9416"><a href="#c4" data-ref="b9715a54-1440-44ca-b59b-f5e3a3c943ed">4</a></sup> and tight junctions seal the basolateral side from the medium. Consequently, response to exogenous ligands in the medium is restricted to the colony edge, while endogenous ligands are confined to the basal side and prevented from diffusing away into the medium everywhere but the edge. This was found to be critical for patterning in both 2D gastruloids and the mouse embryo<sup data-ref="f3f9e3b7-fbc1-4085-9d34-94caa2c98f0a"><a href="#c4" data-ref="0fa8867a-01f2-46bf-9f57-7ae68c677cc0">4</a>,<a href="#c5" data-ref="efdac636-9cc9-4a63-9eb6-055d1a5ddfd3">5</a></sup>. To explain the pERK pattern, we therefore hypothesized FGF receptors were similarly restricted to the basolateral side.</p><p id="el-p-31" data-ref="d082bbf0-c6c6-4ea3-afdc-eec2e48de97b" data-id="el-p-31">To test our hypothesis, we first starved pluripotent colonies of FGF by maintenance in FGF-free E6 medium for 12h and then treated with FGF2. Consistent with our hypothesis, phospho-FGFR and phosphor-ERK were both restricted to the colony edge (<span class="fakelink linktofig" data-ref="d9c29fde-a5e7-45ef-ad55-711347905e79"><span class="fakelink-text" data-ref="ba559ba5-5604-4ba2-8cb3-05cf6f811880">Fig. 4a</span><a class="fake-icon" href="#fig4" data-ref="8c7e2841-0611-45dc-917c-5e74637624b6"><span class="fakelink-icon" data-ref="885e0026-9efc-496d-8797-36e56a02dce3">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). To determine if this could be explained by diffusion of exogenous FGF from the medium, we repeated the experiment with micropattern pluripotent cells and added fluorescent dextran with a similar molecular weight as FGF. We observed fluorescent dextran diffusing into the intercellular space from the edge inward over the same distance that ERK signaling was observed (<span class="fakelink linktofig" data-ref="2091226d-1919-404b-8738-3c1b5942ac9e"><span class="fakelink-text" data-ref="76958e5e-f778-4ebf-8f42-1d8c9c775ccb">Fig. 4b-d</span><a class="fake-icon" href="#fig4" data-ref="b1915d59-5f0b-4b87-ac2e-0df9226cb231"><span class="fakelink-icon" data-ref="13b94ec8-a1e4-4182-a556-5110e2984fcb">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). We then acquired high resolution z-stacks of FGFR1 co-stained with the tight junction protein TJP1 (ZO-1) which revealed the majority of FGFR1 was below the tight junctions (<span class="fakelink linktofig" data-ref="faecd37c-2f40-47b9-bfbb-455f8af6b275"><span class="fakelink-text" data-ref="7a16a26e-8085-4d47-a6c3-d73504e608f8">Fig. 4e</span><a class="fake-icon" href="#fig4" data-ref="fce395b7-4f63-4037-8362-6e3d3651ed00"><span class="fakelink-icon" data-ref="ba395ec8-d739-45e1-8b6d-f245894dc64b">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). To directly test that cells are more responsive to basal FGF than apical FGF, we then performed a transwell assay, where pluripotent cells were grown on filters to enable separate apical and basal stimulation. We found that only basal stimulation with FGF led to a significant increase in pERK (<span class="fakelink linktofig" data-ref="8bbd80af-e86d-4e0e-909e-ddec3a724092"><span class="fakelink-text" data-ref="f7dae777-e3a2-47fb-8da1-ba636e777142">Fig. 4f</span><a class="fake-icon" href="#fig4" data-ref="2675d5e9-a3d9-4d78-8cae-d0768328701c"><span class="fakelink-icon" data-ref="eac43751-cc82-4f3f-ae54-194cd1ad1c56">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Finally, to determine if the center of the micropatterned colony is similarly able to respond to basal FGF stimulation and pERK levels there can be attributed to reduced FGF ligand concentration, we performed a scratch assay. Strong ERK response to FGF2 in the media was seen in the colony center around a scratch, but this response was blocked by FGFR inhibitor (<span class="fakelink linktofig" data-ref="ae6ad87a-be38-4381-9944-2aaad71b3fbf"><span class="fakelink-text" data-ref="a9ab921a-08e9-4bb8-bf96-79f896accee7">Fig. 4gh</span><a class="fake-icon" href="#fig4" data-ref="aa9e1772-732b-4575-8047-0d0c313acdfe"><span class="fakelink-icon" data-ref="ed8300bb-f202-4101-b15d-5e5612659b8e">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). If HSPGs rather than FGF ligand were limiting ERK response in the center, we would not have expected strong response around a scratch. The initial ERK levels around the scratch greatly exceeded those anywhere in the control. However, we found this was due to adaptive ERK response to FGF. After six hours ERK levels around the scratch were similar to those in the ERK ring around the PS-like cells (<span class="fakelink linktofig" data-ref="8edd178e-15cd-4226-a7d4-94b15c8bb3ad"><span class="fakelink-text" data-ref="fabe95b0-c992-4ab8-94c3-d257331a3397">Fig. 4ij</span><a class="fake-icon" href="#fig4" data-ref="7ffeeec2-e46c-450e-abed-551c6ed0594b"><span class="fakelink-icon" data-ref="a80f0a5e-71fc-433b-906d-62da5d646151">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). We therefore concluded that the pERK pattern is most likely due to a basolateral FGF ligand gradient which has its maximum in the PS-like ring.</p><figure class="figure tofill imgs-1 longcaption fig-portrait pagedjs-fill-next-page" id="fig4" data-ref="dd8f4a22-7f91-41bf-8097-0e12281a5472" data-id="fig4" style="display: none;"><figcaption class="figure__caption" id="el-figcaption-3" data-ref="fe369079-6656-45a3-a3d1-cdd99fed9889" data-id="el-figcaption-3"><h3 id="el-h3-3" data-ref="4b570ddc-9b0b-4772-8e57-dcd72890d911" data-id="el-h3-3" data-break-after="avoid"><span class="label figure-name" data-ref="b01c5e87-6c14-4753-8a62-59d7058946b3"><label class="figure__label" data-ref="6fff133d-1083-490d-978f-cd207285b7c3">Figure 4.</label></span></h3><p id="el-p-32" data-ref="ebe57380-bf09-4e3e-8b42-558d9fcf80f9" data-id="el-p-32" data-previous-break-after="avoid"><strong data-ref="4b1dd281-6c12-4f5a-83e7-1ef6d0767d80">a)</strong> pERK and pFGFR1 response of pluripotent colonies in standard culture grown in FGF-free (E6) medium for 24 hrs and then treated with FGF2 for 30 minutes. <strong data-ref="38b77e94-5880-4239-90ff-0b1a6fc190e0">b,c)</strong> Fluorescent dextran (b) and pERK (c) in micropatterned pluripotent colony 24h after media change containing both FGF2 and dextran. <strong data-ref="1ebfaf32-f6a4-4de1-b992-a728e347e20e">d)</strong> Radial intensity profile of dextran and pERK for conditions in b,c), averaged over 2 colonies. <strong data-ref="a04f5824-0911-4685-b067-6054b79d8768">e)</strong> Cross-section of 2D gastruloid shows FGF receptor 1 predominantly below the tight junction marked by ZO-1. White box marks are magnified in inset e’. <strong data-ref="cc3960ee-4ff8-42a6-854f-34c280133129">f)</strong> pERK and pFGFR1 response in transwell experiment with cell treated with FGF either apically or basally. The illustrations in this panel were created using <span class="fakelink linktoext" data-ref="988d15b4-2451-46fb-a5e0-ab50292bfb41"><span class="fakelink-text" data-ref="9da28a70-024b-419b-b0e2-4200fc38128a">BioRender.com/v73i147</span><a class="fake-icon" href="https://BioRender.com/v73i147" data-ref="020ec7a2-0be6-44a7-a0b7-370fa3d24666"><span class="fakelink-icon" data-ref="8fa3cfea-a393-4070-a57b-ea6456a5d2b3">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. <strong data-ref="5c567872-424d-43a6-b0b8-111788f74548">g)</strong> ERK signaling in scratched colonies with or without FGFRi. <strong data-ref="bbab6ab1-d72d-4565-85d0-b4a2ce1bf87b">h)</strong> Quantification of pERK intensity as a function of distance from the scratch. Error bars represent standard deviation over colonies. <strong data-ref="20e16321-948b-4b00-8d69-0e5e09a727b3">i)</strong> ERK signaling stains at different times after scratching. <strong data-ref="5d191984-8a9d-46f3-8b6b-94f5c9e296f7">j)</strong> Quantification of pERK intensity as a function of distance from the scratch at different times (left) and peak pERK level over time (right). Error bars represent standard deviation over N=4 colonies. Scale bars 50 micron.</p></figcaption><img class="child-of-figure has-boundary imageonly portrait max-100" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_fig4.tif/full/max/0/default.jpg" alt="" id="el-img-4" data-ref="78b0e615-9587-4871-a089-1bbd3dbcc906" data-id="el-img-4" height="2350" width="2101" data-desired-width="99.47916666666666"></figure></section><section id="s2e" data-ref="3b1d2b09-70bd-496f-bbde-99ad4f10fea3" data-id="s2e"><h2 class="heading-2" id="el-h2-5" data-ref="78f5b4f2-40f2-4d85-b506-fd0d4ce7e64c" data-id="el-h2-5" data-break-after="avoid">Endogenous FGF gradients underly the ERK activity pattern</h2><p id="el-p-33" data-ref="4e657750-a037-4dfe-a3ef-55e91c111a75" data-id="el-p-33" data-previous-break-after="avoid">To identify which FGF ligands are required for PS-like differentiation, we first determined the requirement for exogenous FGF2 in the media. Differentiation of 2D gastruloids in FGF-free medium was severely reduced but restored by addition of FGF2 (<span class="fakelink linktofig" data-ref="2eaca8ff-9108-4263-9ef7-0dac2d345f1e"><span class="fakelink-text" data-ref="1ab206c0-49df-4c10-b4fc-99a89a0a0c68">Fig. 5a</span><a class="fake-icon" href="#fig5" data-ref="b60d8e07-2fda-46f5-95ed-06295fddf6b3"><span class="fakelink-icon" data-ref="45eecdba-9d01-4863-9373-340a195fb44d">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="c6b9d293-ce3f-4e61-86a6-5cec1189cb3d"><span class="fakelink-text" data-ref="9c2b8d51-9f7a-410b-bb67-fccfbbf4099b">Supp. Fig. 5a</span><a class="fake-icon" href="#figS5" data-ref="7ac133a3-7b31-40d3-b4d8-38ab671cac33"><span class="fakelink-icon" data-ref="218af401-5c48-4103-832e-8e322bc65554">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). However, if FGF2 was removed 24h after BMP treatment, differentiation was normal. The fact that removal 2h after BMP treatment significantly reduced differentiation suggests this was not a failure to wash out the FGF2. Furthermore, MEK inhibition at 24h severely reduces TBXT expression<sup data-ref="5d93cbb9-a02a-467c-8695-291517abe0d4"><a href="#c34" data-ref="ce220d8c-3e44-4c20-a367-53451d47eeee">34</a></sup> while FGF2 is known to have a half-life on the order of hours<sup data-ref="439705df-8cdc-4163-bcbc-92c1def5c8c0"><a href="#c49" data-ref="d53f74d8-a4ed-4fc8-8d5e-6e0477f1dcc8">49</a></sup> and no media changes are performed during standard differentiation. Altogether this suggests that exogenous FGF2 is required at the time of BMP treatment but that endogenous FGFs are sufficient for the ERK activity ring and PS-like differentiation that start a day later.</p><figure class="figure tofill imgs-1 shortcaption fig-square pagedjs-fill-next-page" id="fig5" data-ref="5023f542-1bc1-4d19-a6ea-9de547f84932" data-id="fig5" style="display: none;"><figcaption class="figure__caption" id="el-figcaption-4" data-ref="bbd9eba3-3d31-427a-81b5-fe6135b63ab0" data-id="el-figcaption-4"><h3 id="el-h3-4" data-ref="397d7c56-853d-4bde-99d5-0c3d90ba3cc9" data-id="el-h3-4" data-break-after="avoid"><span class="label figure-name" data-ref="9781a66c-0c41-4041-bd3e-eb73f417acfb"><label class="figure__label" data-ref="e84c7b4a-9a83-4870-b410-d861b3a261bb">Figure 5.</label></span></h3><p id="el-p-34" data-ref="813f2e32-134e-48ab-bcb3-8bdf851f28a4" data-id="el-p-34" data-previous-break-after="avoid"><strong data-ref="d9660fc0-02dc-4e5e-a18a-a80548016d91">a)</strong> Pattern at 42h with exogenous FGF2 in the culture medium for different amounts of time. <strong data-ref="c79bb67e-70b3-4950-90bc-eadab87fe62a">b)</strong> Pseudotime analysis of FGFs and TBXT, TBX6 expression. <strong data-ref="1da309e5-ec31-4139-9296-12bae1f77c12">c-e)</strong> FGF4 (c), FGF8 (d), and FGF17 (e) FISH co-stained for TBXT and pERK at 30h or 32h and 40h or 42h after BMP4 treatment. <strong data-ref="7badf667-3b8a-46cb-b244-f586d7924f31">f)</strong> Radial intensity corresponding to (c-e). <strong data-ref="984fcc98-5440-4e66-9df8-0b7d7c65ba86">g-i)</strong> FGF4 (g), FGF8 (h), and FGF17 (i) FISH co-stained for TBXT and pERK in control siRNA versus FGF4 siRNA (g), FGF8 siRNA (h) and FGF17 siRNA (i) with corresponding radial intensity profiles. <strong data-ref="d74e304c-4046-496a-90e9-42f1e40dc850">j)</strong> TBX6 and SOX17 expression in control shRNA versus FGF4 and FGF17 shRNA. Error bars in all panels except (b) represent standard deviation over N=4 colonies. Error bars in (b) represent the 95% confidence interval of the generalized additive model (see methods).</p></figcaption><img class="child-of-figure has-boundary imageonly square max-100" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_fig5.tif/full/max/0/default.jpg" alt="" id="el-img-5" data-ref="605132c9-dc9f-44ad-a845-67c1f6623dc6" data-id="el-img-5" height="2175" width="2106" data-desired-width="99.7159090909091"></figure><p id="el-p-35" data-ref="f1a11b28-1b53-47ba-b894-39872b32a826" data-id="el-p-35">We reasoned that FGFs whose expression correlates with the presence of PS-like cells are more likely to be responsible for the observed ERK ring and PS-like differentiation. Our analysis in <span class="fakelink linktofig" data-ref="a26c0f4b-badf-40bf-8823-c9ebe601cebc"><span class="fakelink-text" data-ref="d582e02e-1bc0-4130-86ed-b101b19ca47d">figure 2</span><a class="fake-icon" href="#fig2" data-ref="c68c763d-fffe-4771-a3d1-a1be57271efb"><span class="fakelink-icon" data-ref="90754ae8-fa5c-499d-8fd7-b5c6f56e4f7a">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span> had implicated FGF4 and FGF17 as most strongly expressed in the PS-like cells and nascent mesoderm, respectively (<span class="fakelink linktofig" data-ref="c2d68c72-7079-4592-8458-d701e416cc4b"><span class="fakelink-text" data-ref="0874a9f2-b63f-4885-9b59-8a0f2170a080">Fig. 2c</span><a class="fake-icon" href="#fig2" data-ref="ca630bce-bf9e-45ad-b0d4-5f726ca199d3"><span class="fakelink-icon" data-ref="84901529-7c4b-4a46-875d-62fde8aa6bdf">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). However, it is unclear whether expression of these ligands is a cause or consequence of PS-like differentiation, or both, in the case of a feedback loop. To get more information about the relationship between FGF ligands and PS versus mesoderm differentiation, we performed pseudotime analysis of our scRNA-seq data (<span class="fakelink linktofig" data-ref="a1ffcde5-5598-40c3-9561-05b4f461d757"><span class="fakelink-text" data-ref="5255c174-a44c-4948-b436-c05e650395cb">Fig. 5b</span><a class="fake-icon" href="#fig5" data-ref="cc6daf67-18ef-4ebd-bc18-027671ae3d11"><span class="fakelink-icon" data-ref="6d239c40-adb3-4f49-a6e2-e4ad8bab4a54">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="71e3dd24-f0d5-4ef1-be29-f6acd07c0c02"><span class="fakelink-text" data-ref="7e53ed1a-4493-4398-8aeb-0741f619e2f8">Supp. Fig 5b</span><a class="fake-icon" href="#figS5" data-ref="2039ec33-f05f-4c94-969e-0a6ff6026358"><span class="fakelink-icon" data-ref="86eea061-b801-4f45-8b97-5f4528bb578f">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Along a pseudotime trajectory from pluripotent cells to nascent mesoderm, <em data-ref="b03f6cb5-d768-4efc-a529-d49bc4ddb1ad">FGF2</em> expression was nearly constant. <em data-ref="3428a895-d2ba-4023-9261-92b2c686efd1">FGF8</em> was upregulated at lower levels prior to <em data-ref="ae294e78-5efe-41c1-99cc-918697bf3968">TBXT</em> and declined as <em data-ref="1321f8c3-f20e-4c58-9b7b-566311b9805e">TBXT</em> was upregulated. <em data-ref="94ee8fcb-4ac2-445a-a1dd-e29cb1762631">FGF4</em> upregulation approximately coincided with <em data-ref="1b180064-10b1-4d64-94a1-6cbbc9537643">TBXT</em>, but <em data-ref="175aa7a7-258f-4fa2-8527-e815bcb1512b">FGF4</em> began decreasing as <em data-ref="fbd53fc5-fb02-49bf-8d6c-b3444dc51f9f">TBX6</em> expression increased. In contrast, <em data-ref="8b299ba2-5c27-4857-8963-48f0deb5534e">FGF17</em> expression started after <em data-ref="718d8e89-a382-47dc-8387-b4501a800775">FGF4</em> but continued to increase after <em data-ref="ea15ff57-5960-41f0-8e42-57b1f05c0d4f">FGF4</em> started decreasing. Direct comparison of <em data-ref="bd54b686-5c3e-41ae-a9d6-08d6a4a094d5">FGF</em> and <em data-ref="af92269f-4aae-453a-90f7-cad785c8f4cb">TBXT</em> expression also showed strong positive correlation between <em data-ref="fc226523-71e6-4713-b704-8b7b2f84c77d">TBXT</em> and both <em data-ref="7cda5874-1e34-488b-b79a-509b6de56520">FGF4</em> and <em data-ref="5e636cfc-ee9b-4803-a9db-32fabb13030c">FGF17</em> but opposing relations with <em data-ref="ff3dc3f3-d8f9-4d82-a995-56a147618661">TBX6</em> (<span class="fakelink linktofig" data-ref="86aa4fad-a82a-4678-84fd-941b5d25e72f"><span class="fakelink-text" data-ref="35801380-2a63-486e-ab5e-3bd75ce0df05">Supp. Fig. 5c</span><a class="fake-icon" href="#figS5" data-ref="3f95a14c-89c1-4ca7-8701-ef19f044d17f"><span class="fakelink-icon" data-ref="91a05cc0-08fc-4e5d-8f69-217889bf100c">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Together this suggested to us initial TBXT expression may depend more on FGF8 or FGF4, while FGF17 could play a later role in maintaining or expanding the PS-like population or facilitating subsequent mesoderm differentiation.</p><p id="el-p-36" data-ref="d8c98ea5-a74a-464f-b2cc-24563bfa6c30" data-id="el-p-36">Pseudotime may not accurately predict true dynamics. Therefore, we verified expression at two different times using RNA FISH (fluorescent in situ hybridization). We found that <em data-ref="88526701-54cf-4a38-ba05-f2c6090a3fc2">FGF4</em> expression was similar at 32h and 40h (<span class="fakelink linktofig" data-ref="4d9970bc-3f62-4a40-b080-77b77629a193"><span class="fakelink-text" data-ref="9cdfb0fe-3bca-4f75-8dd2-049f7960edd2">Fig. 5cf</span><a class="fake-icon" href="#fig5" data-ref="eab756d4-33d3-4842-84e2-9478cb5a71c6"><span class="fakelink-icon" data-ref="d0509728-c649-4d9b-a630-02116af10ebd">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="ecb338af-9281-45ac-9cd8-e80d35fb2460"><span class="fakelink-text" data-ref="b31061ec-4a5c-4320-9299-535819abf57e">Supp. Fig. 5d</span><a class="fake-icon" href="#figS5" data-ref="81952f9d-c13c-4427-86b8-886ad296d081"><span class="fakelink-icon" data-ref="4492629b-bc32-438a-a02c-3b436c0f76b4">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). This is consistent with its decrease in pseudotime as cell differentiate to nascent mesoderm, since in real time differentiation is asynchronous and new PS-like cells form while older ones differentiate to nascent mesoderm, which would yield a flatter temporal profile. In contrast, peak <em data-ref="d06a46bd-184c-411b-806b-31cea2a858c2">FGF8</em> levels decreased while <em data-ref="a1b66f01-248b-4504-a1ae-020e05a66c46">FGF17</em> strongly increased over time, both matching their pseudotime trends (<span class="fakelink linktofig" data-ref="c0686b18-f40e-41c1-9ab8-b96dca1bd555"><span class="fakelink-text" data-ref="933c4d4e-3db4-4bf7-bed2-591ac1b9b057">Fig. 5d-f</span><a class="fake-icon" href="#fig5" data-ref="75ab9422-7590-43b7-92b7-fdca82280d3b"><span class="fakelink-icon" data-ref="a50a03ae-0812-4370-836a-d57b2d6d0ea9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="5234fc93-df19-4d4e-9b23-b813663c4285"><span class="fakelink-text" data-ref="a6e7ee06-4694-41c2-b2ba-dc9670051e9f">Supp. Fig. 5ef</span><a class="fake-icon" href="#figS5" data-ref="0cb2343c-a679-433c-a08b-d3cf7ba3c3d2"><span class="fakelink-icon" data-ref="0f683120-e906-4a46-8528-0b5a8b5ed2de">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). As expected, the spatial profiles of both <em data-ref="e3d399c6-242a-4edf-b94f-ed335d27860d">FGF4</em> and <em data-ref="fed11889-bb7a-4ea8-9360-80c523ca3112">FGF17</em> resembled TBXT, although the peaks were displaced slightly inward, while <em data-ref="ea4b084d-9241-4ff3-98c9-7a18ab4f9aff">FGF8</em> expression was highest in the inner part of the ERK ring where TBXT expression was low (<span class="fakelink linktofig" data-ref="08c45a27-56f7-4002-bdfc-44a5de4a84ff"><span class="fakelink-text" data-ref="e431226a-f9ad-4e1a-aaf0-2eab91a10b5e">Fig. 5c-f</span><a class="fake-icon" href="#fig5" data-ref="9fbed97b-0499-43f0-b860-24da82689c6d"><span class="fakelink-icon" data-ref="f4b1dbcf-27e8-4374-9df7-330223516d12">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Co-staining showed <em data-ref="23bc72d8-6438-42e1-a80a-6b8cc23be92b">FGF4</em> and <em data-ref="1c926d24-4aee-4b57-b0c9-1960fa40dcb7">FGF8</em> expressed in concentric rings with little overlap at 42h (<span class="fakelink linktofig" data-ref="ee2ad17a-76a4-46d0-9c31-964d47651b1d"><span class="fakelink-text" data-ref="ec17e484-a717-4bd8-af7e-de98e5b07837">Supp. Fig. 5g</span><a class="fake-icon" href="#figS5" data-ref="99e1a621-683a-41be-967b-b310415d526d"><span class="fakelink-icon" data-ref="7a2b8ec0-5459-4773-9bd8-479d19eeb0b6">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Scatterplots further confirmed that high expression of both <em data-ref="33c81ba6-6f42-4cfb-acb9-1e57495407f4">FGF4</em> and <em data-ref="ba2d4be4-0db8-4aa1-b4a5-ff76993c320b">FGF17</em> but not <em data-ref="236d1c6b-bc93-4a2e-8e23-ebf307d2a8e5">FGF8</em> was restricted to TBXT positive cells (<span class="fakelink linktofig" data-ref="21481d21-1b93-46c0-8cfd-eff289621208"><span class="fakelink-text" data-ref="e2dbf9f8-9f60-4177-aa98-372f4f4d4df2">Supp. Fig. 5h</span><a class="fake-icon" href="#figS5" data-ref="bc5e4c0c-9ca5-4fcb-a0cc-8690b3f38c0a"><span class="fakelink-icon" data-ref="4cf97f01-ec98-446e-ad47-4845bcd62047">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>).</p><p id="el-p-37" data-ref="81e7095b-fce1-4e01-a8c0-ad79d57a9ef7" data-id="el-p-37">To test the functions of these FGFs we decreased their expression. Knockdown (KD) of <em data-ref="d511915a-6849-4412-9323-82c59f7d8fd0">FGF4</em> with small interfering RNAs (siRNA) led to a strong decrease in TBXT and pERK (<span class="fakelink linktofig" data-ref="28bed296-6a78-4347-aaea-545fb857e7ad"><span class="fakelink-text" data-ref="2149ff54-0fe3-4821-a76a-2203d74f2e02">Fig. 5g</span><a class="fake-icon" href="#fig5" data-ref="21f97442-d54c-463a-8b04-1d36ec00fcf3"><span class="fakelink-icon" data-ref="af9b1880-449b-4ba2-8423-110b818f7ed0">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="bbb546a4-c845-462c-8148-664d6670a0f6"><span class="fakelink-text" data-ref="8cab7276-e906-429b-b6d4-82403edfffb5">Supp. Fig. 6a</span><a class="fake-icon" href="#figS6" data-ref="6e429e3f-7391-49ce-be32-6c86afd8f997"><span class="fakelink-icon" data-ref="9162aa2e-5f58-4beb-9b3a-26f47dd1cb22">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). We repeated this in a different cell line and also created stable cell lines expressing short hairpin RNA (shRNA) to knock down <em data-ref="7da8f689-0472-4c87-8661-ca8d40c25015">FGF4</em> in both cell lines with similar effect (<span class="fakelink linktofig" data-ref="cb721fa8-c53a-4548-93ca-8141556fcd4d"><span class="fakelink-text" data-ref="8056e239-89dd-45e2-a1a1-1f8ac7346748">Supp. Fig. 6b-d</span><a class="fake-icon" href="#figS6" data-ref="31fc9103-1a4b-4224-b1e5-b91d0f8c152d"><span class="fakelink-icon" data-ref="ec724b9b-625e-45f9-87ae-e9f31ba6a379">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Strikingly, <em data-ref="816b54a2-b88c-4760-b9e6-805fd48e6af4">FGF8</em> KD had the opposite effect and led to an increase in TBXT expression without a strong effect on pERK visible at 42h (<span class="fakelink linktofig" data-ref="ed8f00f0-1cbf-416c-a213-da4c3fda28c8"><span class="fakelink-text" data-ref="0f102d7f-5059-4909-9cff-3c3778d2547a">Fig. 5h</span><a class="fake-icon" href="#fig5" data-ref="6e78ce8c-7ed6-4281-98bb-abc06536f4cb"><span class="fakelink-icon" data-ref="351ecb31-8b67-48a3-a259-be2f9a69366d">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="82a0f3ca-14b3-47b7-8633-cfb7571b93fd"><span class="fakelink-text" data-ref="706c75a1-0abe-42f9-a379-cb8cf4926566">Supp. Fig. 6e</span><a class="fake-icon" href="#figS6" data-ref="daf00f98-06da-4fb2-ac12-35000f651bb6"><span class="fakelink-icon" data-ref="e113ad11-d93e-4de4-8576-004f8bd1a01d">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Finally, <em data-ref="ca1224f0-52b6-42c1-8b50-dfdc3821ebe0">FGF17</em> KD caused only a small decrease in TBXT or pERK, again consistent between different cell lines (<span class="fakelink linktofig" data-ref="443acdc6-7d19-4613-bc50-80578eecb0d7"><span class="fakelink-text" data-ref="28697663-c572-42ee-a1f1-b6fea5b9a3c1">Fig. 5i</span><a class="fake-icon" href="#fig5" data-ref="373c4f3c-9a02-4ee1-96c3-2dd9a84f6603"><span class="fakelink-icon" data-ref="4ea07e52-4db6-4739-a887-ce2d269f29a3">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="08d55bf9-2119-4ace-afd2-281487ebb4d5"><span class="fakelink-text" data-ref="25cb7a81-2387-4561-9b9f-17bc99a15749">Supp. Fig. 6f-i</span><a class="fake-icon" href="#figS6" data-ref="114f3f90-7807-432c-b874-59721a150bfa"><span class="fakelink-icon" data-ref="ba91240b-9b5e-4495-9ce0-9acf2582c667">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). We further interrogated the endoderm and PGC marker SOX17 since both cell types require ERK signaling to differentiate and transiently express TBXT, as well as the mesoderm marker TBX6. TBX6 expression at 42h was near completely lost upon either <em data-ref="b246be42-4b30-4551-a139-a002d7a09794">FGF4</em> KD or <em data-ref="1caf86d8-6555-456e-81f6-92a2ca34fb66">FGF17</em> KD (<span class="fakelink linktofig" data-ref="ea773917-9121-4509-bac7-7e39a95032c6"><span class="fakelink-text" data-ref="20e3e33d-48af-450b-81b8-342538ac6141">Fig. 5j</span><a class="fake-icon" href="#fig5" data-ref="9a1ae61f-ed08-482d-b561-5a8ead84cfb1"><span class="fakelink-icon" data-ref="0e290544-cdc5-436a-a622-18fbe424f41c">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="e980130a-b93c-4800-b75f-9891854348d1"><span class="fakelink-text" data-ref="964156a3-d174-48c0-90a2-ba1b606cc99c">Supp. Fig. 6j</span><a class="fake-icon" href="#figS6" data-ref="b95772e5-2b10-40af-9ed8-75055d5de757"><span class="fakelink-icon" data-ref="e50317b1-2f83-4424-b75e-7f67df74e31a">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). This suggests that while FGF17 is not required for PS induction (<span class="fakelink linktofig" data-ref="1a2a7937-4d81-4836-91b9-ea919baa5dcb"><span class="fakelink-text" data-ref="9b58fa7c-46b8-4cf9-9a20-f99db8c1fe57">Fig. 5i</span><a class="fake-icon" href="#fig5" data-ref="3a463ccd-7911-4080-98f9-331bfa7a1c4b"><span class="fakelink-icon" data-ref="83a817ff-4619-45cc-a0ab-50617d0fe9e6">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>), it is needed for subsequent mesoderm differentiation. In contrast, there was only a small reduction of SOX17 (<span class="fakelink linktofig" data-ref="5f159eaf-6bd7-48fc-b18b-0e2e75943dc2"><span class="fakelink-text" data-ref="2291478d-366c-4b6c-8881-85293ead4e6a">Fig. 5j</span><a class="fake-icon" href="#fig5" data-ref="6fdc0b15-b919-412c-9d2e-a072f1d1ec91"><span class="fakelink-icon" data-ref="c8d10314-bee4-4ac3-a934-df9abfa76a60">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Since we previously found with our protocol most SOX17 positive cells at 42h represent PGC-LCs, while most endoderm differentiates later<sup data-ref="7f72d7a7-efcf-4d0b-a403-83c296fe8465"><a href="#c34" data-ref="46bd82bc-c1f5-47d5-be39-01cfd09ffb20">34</a></sup>, this suggests FGF4/17 are not required for PGC-LC induction but does not rule out a requirement for these FGFs in endoderm differentiation. A double KD of <em data-ref="5ea9d9e1-525a-47f1-bbbf-8f2c9b41ac3f">FGF4</em> and <em data-ref="53891934-019e-4c10-827c-32871c6bff96">FGF17</em> had a similar phenotype as <em data-ref="9c7c188d-3f61-4201-90d0-3f67aa6a29d6">FGF4</em> KD alone, suggesting FGF17 may be downstream of FGF4 (<span class="fakelink linktofig" data-ref="9daf1110-5508-41b0-81a9-2c708e10b7aa"><span class="fakelink-text" data-ref="16d5f6cc-b61d-4c10-bf1f-d24ea27489f9">Supp. Fig. 6k</span><a class="fake-icon" href="#figS6" data-ref="5d347968-9e66-48e1-b53a-96c905f17fc7"><span class="fakelink-icon" data-ref="0395fc28-fe1a-4ece-b08c-a68f3b52a742">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Combined, these data support distinct functions for FGF4, FGF8, and FGF17 in our human gastrulation model, where FGF4 and FGF8 may play opposing roles in PS induction while FGF17 is important during continued differentiation to mesoderm.</p></section></section><section id="s3" data-ref="e9c1a23a-846d-4e47-b528-1786831c8b70" data-id="s3"><h1 class="heading-1" id="el-h1-4" data-ref="2e7f60b6-6e0f-46d2-8beb-882e39db97af" data-id="el-h1-4" data-break-after="avoid">Discussion</h1><p id="el-p-38" data-ref="f8790baa-b4ee-479e-aaa6-6a9117d84a8c" data-id="el-p-38" data-previous-break-after="avoid">We have shown that in a stem cell model for early human gastrulation, FGF-dependent ERK signaling is directly required for differentiation to PS-like cells and their derivatives, and that loss of differentiation upon FGF inhibition can be rescued by directly activating ERK. We also showed that FGF receptors are polarized basolaterally, preventing the colony center from responding to exogenous FGF in the medium. Furthermore, FGF receptor activation is increased in the PS-like ring, matching the pattern of active ERK despite overall receptor expression being lower in the PS-like ring than in the pluripotent center. This suggests elevated ERK activity is due to increased binding of FGFs to the receptors in this region, possibly due to increased FGF concentration. We found that both FGF4 and FGF17 are highly expressed in PS-like cells, while FGF8 is expressed earlier and displaced inward (<span class="fakelink linktofig" data-ref="10f8964c-6682-421c-b771-cd220e80a068"><span class="fakelink-text" data-ref="94c76ab6-5e93-4409-bdf2-dd55787cf80d">Fig.6</span><a class="fake-icon" href="#fig6" data-ref="f81d7b3e-160e-4e81-8f44-657520f1d6d9"><span class="fakelink-icon" data-ref="c5431c20-37df-47ba-921b-c28df7f7c3c3">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Knockdown of FGF4 leads to a strong reduction in ERK activity and PS-like cells, suggesting it may be responsible for increased ERK signaling in the PS-like region. In contrast, FGF8 knockdown led to an expansion of ERK activity and PS-like differentiation. Finally, FGF17 knockdown leds to a loss of mesoderm differentiation.</p><figure class="figure tofill imgs-1 shortercaption fig-landscape pagedjs-fill-next-page" id="fig6" data-ref="b31cdd3c-2665-4afa-b708-ae5d2f3873f3" data-id="fig6" style="display: none;"><figcaption class="figure__caption" id="el-figcaption-5" data-ref="ac61a943-0c68-45a9-a390-85b4917dba19" data-id="el-figcaption-5"><h3 id="el-h3-5" data-ref="47f78fd9-5be4-4b7b-9121-4b6400d4ea93" data-id="el-h3-5" data-break-after="avoid"><span class="label figure-name" data-ref="94f102f7-4be1-4ec8-ab5a-79a0cff4c4d8"><label class="figure__label" data-ref="6dbf21e6-bb20-4830-8745-bfdf342bfdeb">Figure 6.</label></span>Graphical summary.</h3><p id="el-p-39" data-ref="fa1b9af4-cac0-40e2-aae4-aaf19726e5fe" data-id="el-p-39" data-previous-break-after="avoid">Exogenous FGF2 acts on the colony edge and is prevented from reaching the basal FGFR1 receptors elsewhere by tight junctions. Primitive streak cells express FGF4 and FGF17 and these FGFs are associated with high ERK activity. FGF8 is expressed further inside the colony.</p></figcaption><img class="child-of-figure has-boundary imageonly landscape max-70" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_fig6.tif/full/max/0/default.jpg" alt="" id="el-img-6" data-ref="6bbafbe7-b0ae-4665-81df-87287b6bfcf2" data-id="el-img-6" height="465" width="1436" data-desired-width="67.99242424242425"></figure><p id="el-p-40" data-ref="649d6aa6-9817-4a4e-b180-99d3ac852b4f" data-id="el-p-40">Many questions remain. Although we showed that FGF gradients are likely the dominant cause for the spatial pattern of ERK signaling, our scRNA-seq data suggest other levels of regulation may contribute, which remain to be explored. Furthermore, although we showed FGF/ERK signaling is directly required for PS-like differentiation, it may also act indirectly by affecting the expression of Wnt and Nodal. We will explore this and the more general question of how FGF integrates with the hierarchy of BMP, Wnt, and Nodal in a separate manuscript. Related is the indirect effect FGF may have on patterning by its control of cell growth and thereby cell density. Cell density affects the response to exogenous BMP and downstream expression of Wnt, Nodal, and various inhibitors<sup data-ref="8b404a11-d23e-413c-9992-964981442096"><a href="#c4" data-ref="2f8a35e7-34cd-4810-9bb3-713d5867d1af">4</a>,<a href="#c50" data-ref="206975e5-b461-49a5-8d9b-6b6feaa9691b">50</a>,<a href="#c51" data-ref="8da28019-5639-4b5b-b6af-6dd48e0f60eb">51</a></sup>, as well as FGF itself, possibly causing a “community effect”<sup data-ref="2949ade8-6110-48a6-b847-49c7adcc31cf"><a href="#c52" data-ref="08f15c34-2872-4f89-8fc8-026bebfb1756">52</a>–<a href="#c54" data-ref="4b8e02f3-66c4-4a2c-a038-ca70d6771d3f">54</a></sup>. It likely has other effects as well. However, the rescue of TBXT expression but not growth by ERK activation when FGF receptors are blocked suggests the FGF-dependent cell growth is not required to generate a primitive streak-like ring. The ability of exogenous Wnt and Nodal to induce PS-LCs in very sparse culture while MEK and FGFR inhibition block differentiation at the same density similarly argues against a requirement for FGF-dependent cell growth in PS-LC differentiation (<span class="fakelink linktofig" data-ref="1b856859-37e8-49d5-b846-65b4d445db6a"><span class="fakelink-text" data-ref="438fe7b9-e8f2-487d-9598-319c70a409ca">Supp. Fig. 1h</span><a class="fake-icon" href="#figS1" data-ref="696faf93-12de-48d4-bdc7-db7f68401ec8"><span class="fakelink-icon" data-ref="29599e7b-47ff-4a89-8916-dea757e22e96">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>).</p><p id="el-p-41" data-ref="dbb3afb2-fd15-49e3-99ca-ce3e6f87b779" data-id="el-p-41">Another important question is to what extent different FGFs act redundantly or perform different functions. Our data suggest a picture where different FGFs are required consecutively, starting with exogenous FGF2, then endogenous FGF4, to support initial differentiation of PS-like cells, and finally FGF17 to maintain and expand the PS-like ring and support mesoderm and possibly endoderm differentiation. Whether these FGFs could substitute for each other if they were expressed at similar times and levels is unclear. It also remains to be determined whether the nearly constant high expression of FGF2 we observed is required for patterning. Most puzzling is the role of FGF8, whose knockdown led to an increase in PS-like differentiation. Given that FGF8 expression is displaced inward, one possibility is that it induces inhibitors of other PS-inducing signals there in order to limit PS differentiation, e.g., the Nodal inhibitor Lefty, which is expressed in a similar domain<sup data-ref="bbbd1cf1-16fc-42a5-8936-9b72da47b6f3"><a href="#c55" data-ref="e70913a0-027d-4009-bd88-c4d0210af56f">55</a></sup>. The rescue of PS-like differentiation by uniform activation of ERK to a supraphysiological level suggests that the role of the different FGFs is simply to sufficiently activate ERK signaling. However, future work will have to investigate the dox-SOS<sup data-ref="6cace541-b957-4726-a3fc-6e5b6ec63c13">cat</sup> rescue phenotype in much greater detail to see how differentiation and other cell behaviors such as migration are affected.</p><p id="el-p-42" data-ref="1b305b32-d187-4b3e-a1e0-06164842a6ca" data-id="el-p-42">It is interesting to ask if there is a simple explanation for the seemingly large differences in FGF expression between primates and other species. FGF8 is the only FGF shown to be required for mouse gastrulation and is upstream of FGF4 in both chick and mouse, but it is barely expressed in human and monkey primitive streak and appears to play a different role in our stem cell model. If FGFs are interchangeable in simply activating ERK for primitive streak induction, variation in FGF expression is not surprising. However, FGF4 and FGF8 are the main FGFs in mouse, chick and frog gastrulation<sup data-ref="5fda6491-6cf9-4da8-89e5-0fcad7d03f3b"><a href="#c56" data-ref="ed79d7e2-e179-458e-92f1-17dfed17efe6">56</a>–<a href="#c58" data-ref="c14afeba-03e1-4633-8497-d8b2b82dddbe">58</a></sup>, suggesting a high degree of conservation. It is possible FGF8 has simply been substituted by FGF17, which is in the same subfamily<sup data-ref="81c41042-525f-4c89-a5a8-24c71bc403f4"><a href="#c9" data-ref="7f27ddee-79a9-4e76-a8f0-3bf3fc75b4fe">9</a></sup>, but this does not explain why the role of FGF8 in gastrulation otherwise appears strongly conserved, or why FGF8 is upstream of FGF4 in the mouse while FGF4 is expressed before FGF17 in 2D gastruloids. Similarly, it is possible that FGF2 in primates substitutes for FGF5 in the mouse as the FGF that is highly expressed in the epiblast. Although these are not in the same subfamily and generally have different receptor affinities, they both have highest affinity for FGFR1<sup data-ref="cc0d53d3-dd7e-49e0-bceb-cfc47b9740ca"><a href="#c9" data-ref="595f8644-6489-4960-bcd5-1cebca512b06">9</a></sup>. That leaves FGF4 as the only FGF that may be conserved across vertebrate gastrulation, and it is tempting to speculate on why. In Xenopus, FGF4 and TBXT (Xbra) function in a positive feedback loop<sup data-ref="f1f47e8d-bbaa-456f-a820-76281f988e4d"><a href="#c22" data-ref="9ade2534-d3a1-44f2-819d-91407c382c54">22</a>,<a href="#c59" data-ref="59da9f18-1939-4137-984b-32c2974f320d">59</a></sup>. The strongly correlated initial transcriptional dynamics of TBXT and FGF4 in 2D gastruloids are consistent with a similar feedback loop. However, an FGF8 feedback loop has also been proposed and recent work suggests TBXT is not required for FGF4 or FGF17 expression<sup data-ref="7aab2a53-7240-4988-8d83-72e2c4d63432"><a href="#c54" data-ref="f1141bf8-42d6-4591-8e34-cf44966c114a">54</a>,<a href="#c60" data-ref="88040d36-6614-4674-85fe-0a08fd48e25d">60</a>,<a href="#c61" data-ref="21336814-d806-4ab5-b283-c769804b05cc">61</a></sup>. Future work will therefore have to investigate further if FGF4 plays a unique role.</p><p id="el-p-43" data-ref="c8d9165b-34ee-4aaa-bf85-b146368617be" data-id="el-p-43">A final related question is the origin of the discrepancies we and others<sup data-ref="f49ca5f2-04ad-4107-8e8a-4d8af9d29011"><a href="#c36" data-ref="dc3eeaac-64aa-4bfc-aea7-73a12f9c1f57">36</a></sup> found between the primate (i.e., non-mouse) samples in <span class="fakelink linktofig" data-ref="210d8a4a-e632-485c-a89c-d0e06d5ce027"><span class="fakelink-text" data-ref="9318d535-e5bf-4e42-a4ab-5b39e6176756">Fig. 2</span><a class="fake-icon" href="#fig2" data-ref="6586d6fb-a971-4418-8a62-e1977760e9d3"><span class="fakelink-icon" data-ref="c4cb6036-96f0-4a70-af50-ec3a1cc8a1ae">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span> and <span class="fakelink linktofig" data-ref="300d04eb-461e-4868-962a-2a8d21eea702"><span class="fakelink-text" data-ref="d74a5ab0-2044-4607-aadc-cf45d22f94d0">Supp. Fig. 2</span><a class="fake-icon" href="#figS2" data-ref="479d1a3f-f88a-481b-b876-107bea21cf8a"><span class="fakelink-icon" data-ref="035f8ab8-d557-435d-8589-19cdff9f0d76">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>: FGF3 appears only significantly expressed in the human embryo, but not in the monkey embryo and human stem cell model. This could also reflect genetic variation: if FGFs function redundantly, their relative expression levels could be relatively unconstrained between individuals. On the other hand, in the nascent mesoderm, FGF2 is only high in human gastruloids. This may reflect developmental stage: the in vivo data is developmentally more advanced and nascent mesoderm may lose FGF2 expression later. It is also possible the maintenance conditions of hPSCs lead to in vitro artefacts in FGF expression.</p><p id="el-p-44" data-ref="570e22da-0774-4bbf-bcb4-5c9ebbb8d892" data-id="el-p-44">In conclusion, the complexity of FGF signaling, with its many ligands, modulators, and functions, has prevented a clear understanding of its role in mammalian gastrulation in general and in human gastrulation specifically. Here we showed that 2D human gastruloids provide a powerful approach to make headway and although many open questions remain, we have taken significant steps towards this goal.</p></section><section id="s7" data-ref="f25cc01d-3ded-4c57-9468-d579e87fa17a" data-id="s7"><h1 class="heading-1" id="el-h1-5" data-ref="6ad604e2-9286-4562-a537-b1b46441d3a2" data-id="el-h1-5" data-break-after="avoid">Methods</h1><section id="s7a" data-ref="0b712780-4be5-4066-b3f5-989cd95398cb" data-id="s7a" data-previous-break-after="avoid"><h2 class="heading-2" id="el-h2-6" data-ref="61675e78-3e50-46b2-a41c-a40246f3aac4" data-id="el-h2-6" data-break-after="avoid" data-following="following-ff2c00ba-51a1-482d-9afa-f5a370dcfa50">Cell lines</h2><p id="el-p-45" data-ref="57bbfd77-599e-4c3e-99e8-98ad0fc4a87a" data-id="el-p-45" data-previous-break-after="avoid">The cell lines used were the embryonic stem cell line ESI017 (XX), and the induced pluripotent stem cell line PGP1 (XY). The pluripotency of these cells was confirmed by immunostaining of pluripotency markers OCT3/4, SOX2, NANOG. All cells were routinely tested for mycoplasma contamination, and negative results were recorded. ESI017 cells stably expressing doxycycline inducible SOS<sup data-ref="f1642262-f169-4af3-8072-b3bcaee5ee00">cat</sup> were made using a published Piggybac plasmid gifted by Jared Toettcher<sup data-ref="930d44ac-a38d-4696-9053-c4fc61a899d9"><a href="#c37" data-ref="565a5866-abeb-48d2-a36f-d8cdc8961fc0">37</a></sup>.</p></section><section id="s7b" data-ref="150926f1-9429-498e-882e-e1a643873856" data-id="s7b"><h2 class="heading-2" id="el-h2-7" data-ref="5cd8b556-412c-4bcc-9f45-e9f342d3652f" data-id="el-h2-7" data-break-after="avoid">Cell culture and differentiation</h2><p id="el-p-46" data-ref="9ab6021f-8d25-4b0b-b8d0-215c91d45769" data-id="el-p-46" data-previous-break-after="avoid">Human pluripotent stem cells were cultured in the commercially defined pluripotency maintaining media mTeSR1 (STEMCELL Technologies #85850) on Cultrex (R&amp;D Systems)-coated tissue culture plates. For FGF2 starvation we used Essential 6 Medium (Thermofisher, A1516401), or mTeSR1 without bFGF, TGF-beta, LiCl, GABA and pipecolic acid (STEMCELL Technologies #05896). For routine cell maintenance, cell passaging was performed every 3 days. For whole-colony passaging, L7<sup data-ref="000a1f48-2aea-4388-8b44-4c2c8aeff949"><a href="#c62" data-ref="f4c13c4c-6a3c-43ec-9ec5-2fda4b83d1b0">62</a></sup> was used, while single-cell passaging was performed with Accutase or TrypLE (Gibco). Single-cell suspensions were used to seed for all the differentiation experiments to control initial cell number.</p><p id="el-p-47" data-ref="ec85afef-3ce6-47c1-beec-d0ff2c3cc9f3" data-id="el-p-47">To directly differentiate to primitive streak-like cells, cells were resuspended in a single-cell suspension and seeded in cultrex coated wells in mTeSR with ROCK inhibitor (RI) Y-27632 (MeChemExpress, cat# HY-10583). Cells were then maintained in mTeSR with RI for 24h before adding CHIR (3-5uM) and Activin (100ng/ml) treatment for 24-30h. Experiments were performed in 18-well Ibidi slides (cat# 81818).</p><p id="el-p-48" data-ref="839eff24-515f-4ced-a6b2-6868c058e15b" data-id="el-p-48">To differentiate cells for 2D gastruloids by micropatterning, we followed the protocol in<sup data-ref="2caca2f0-0f8c-4bcc-94e1-5cc4b007bee2"><a href="#c42" data-ref="5fa68afb-2d7c-4737-9526-2d08f171a91d">42</a></sup>. Briefly, cells were resuspended in a single-cell suspension and seeded in laminin-coated micropatterned wells in mTeSR with ROCK inhibitor (RI) (MeChemExpress, cat# HY-10583). Micropatterned wells were washed with PBS-/- 30 minutes after the initial seeding to wash off the non-attached cells. Cells were then maintained for 24 hours in mTeSR before adding the BMP4 treatment by a full medium change. For time series micropatterned differentiation, full media changes with designated treatment(s) were performed every 24 hours. Experiments were performed in micropatterned 18-well Ibidi slides (cat# 81818) prepared as previously described<sup data-ref="4c2c6418-1fc7-4297-9120-ff2fbe663a90"><a href="#c63" data-ref="480d2a57-3eca-4a63-914c-5411b565f239">63</a></sup>. Cell signaling reagents and doses used are listed in <span class="fakelink linktofig" data-ref="f5ae2f6b-f752-4664-975e-b88e87d2ebe9"><span class="fakelink-text" data-ref="afd575c4-0e49-4723-a046-758c5cac2108">Supplementary Table 1</span><a class="fake-icon" href="#tblS1" data-ref="b1120fe2-987f-4621-9027-40640c16bd84"><span class="fakelink-icon" data-ref="ea33a979-2c89-4057-999f-2652b4b3cc31">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>.</p></section><section id="s7c" data-ref="a21fc0f9-279a-45f5-952b-8e1a9aed017f" data-id="s7c"><h2 class="heading-2" id="el-h2-8" data-ref="6165b282-f2d7-4f45-b2f2-2053dd3f051d" data-id="el-h2-8" data-break-after="avoid">Immunofluorescence staining</h2><p id="el-p-49" data-ref="6e6b3db4-0f65-4d31-901a-020b2bdd52a3" data-id="el-p-49" data-previous-break-after="avoid">Antibodies can be found in Supplementary Tables 2 and 3. For stains including pERK, we used methanol fixation, for all other stains we used PFA fixation. PFA fixation: Samples from the 18-well Ibidi slides were rinsed with PBS, fixed for 20 min in 4% paraformaldehyde, rinsed twice with PBS, and blocked for 30 min at room temperature with 3% donkey serum and 0.1% Triton X-100 in 1× PBS. After blocking, cells were incubated with primary antibodies at 4°C overnight, followed by three washes in PBST (PBS with 0.1% Tween 20). They were then incubated with secondary antibodies and DAPI for 30 min at room temperature and washed twice in PBST at room temperature. Methanol Fixation: Samples from the 18-well Ibidi slides were rinsed with PBS, fixed for 20 mins in cold methanol, then rinsed with PBS for 5 min two times, after which 10% phosphate buffered formalin was added for 20 min at room temperature. Cells were then rinsed with TBS twice followed by 5 mins of incubation in cold methanol and rehydrate in TBS for 20 mins. Blocking was done for 30 mins at room temperature with 5% donkey serum, 1% BSA, 0.2% Triton-X-100 in 1x TBS after which staining proceeded the same way as with PFA fixation.</p></section><section id="s7d" data-ref="f6841dce-8542-4b85-8936-690726287081" data-id="s7d"><h2 class="heading-2" id="el-h2-9" data-ref="6b8756ab-99dd-49a7-9fa0-952437b46d8d" data-id="el-h2-9" data-break-after="avoid">Microscopy</h2><p id="el-p-50" data-ref="b1cf77dd-7555-42dd-8e53-135243e48a9c" data-id="el-p-50" data-previous-break-after="avoid">Fixed sample imaging was performed with an Andor Dragonfly/Leica DMI8 spinning disk confocal microscope with a 40x, 1.1NA water and 20x 0.8NA air objectives using Andor Fusion software version 2.3.0.31 and a Nikon/Yokogawa spinning dish confocal microscope with a 40x silicon oil objective using NIS Elements AR software version 5.41.02. Live-cell imaging was performed with an Andor Dragonfly/Leica DMI8 spinning dish confocal microscope under the controlled temperature (37°C), CO<sub data-ref="9f158469-51c3-4fa6-8c86-b3644bd51afb">2</sub> concentration (5%), and humidity (&gt;60%).</p></section><section id="s7e" data-ref="9b72425b-4768-4a1b-9cd6-348219338ede" data-id="s7e"><h2 class="heading-2" id="el-h2-10" data-ref="34b5e65f-d55d-4686-a3d4-76121e2c0188" data-id="el-h2-10" data-break-after="avoid">General image analysis</h2><p id="el-p-51" data-ref="0faa21c5-3627-460a-862d-c9996859e397" data-id="el-p-51" data-previous-break-after="avoid">We segmented nuclei in individual z-slices based on nuclear fluorescence stained with DAPI in fixed cells using a pipeline we previously described, which integrates two machine learning approaches: Ilastik pixel classification<sup data-ref="01c8f6f6-9bfd-4248-a2ed-62eb64ac203b"><a href="#c64" data-ref="437ec40f-95a3-4125-b384-74de7d0218d2">64</a></sup> and Cellpose<sup data-ref="2f00378c-68ed-42be-aa35-59f7bcee2d14"><a href="#c65" data-ref="1ae52269-7686-4d57-b104-2509fea57ee6">65</a></sup> (v1). Fluorescence intensities were then calculated per nucleus as mean intensities in the 3D nuclear again with our established custom image-processing pipeline. To calculate the radial fluorescence distributions in micropatterned colonies, colonies were subdivided into radial bins with equal numbers of cells, mean or median and variance were then calculated within these bins. To calculate positive fractions, markers were first thresholded based on their intensity distribution, after which the fraction of cells positive for each marker was calculated within different bins. Standard deviations were then calculated over multiple replicate colonies. For visualization purposes only, background subtraction using grayscale opening on a 100 micron scale and 5 pixel wide median filter was applied slice by slice to z-stacks of micropatterned colonies before making maximal intensity projections.</p></section><section id="s7f" data-ref="0f956348-533b-43b8-b2b2-2f7526ea3401" data-id="s7f"><h2 class="heading-2" id="el-h2-11" data-ref="8c36027d-88b0-4702-b7b3-3c92e2bd1f28" data-id="el-h2-11" data-break-after="avoid">(Single-cell) RNA sequencing and analysis</h2><p id="el-p-52" data-ref="e021b4e3-cc58-4ad1-93fa-db993eb5c0d6" data-id="el-p-52" data-previous-break-after="avoid">Cells were collected using accutase and resuspended in ice-cold PBS. Single-cell RNA-sequencing was performed by the University of Michigan Advanced Genomics Core. Cells were barcoded using the 10X Genomics Chromium system (part numbers 1000268, 1000120, 1000215). For quality control, cDNA was quantified by Qubit High Sensitivity DNA assay and Agilent TapeStation. Sequencing was performed on the Illumina NovaSeq 6000 with NovaSeq S4 flowcell and Control Software version 1.7.0. Reads were aligned using cellranger-4.0.0 with the GRCh38 reference. Further processing was mostly done in Python, primarily using the Scanpy<sup data-ref="ac4a3837-34c7-4217-bd4d-a991a1cf3cb6"><a href="#c66" data-ref="2af401c4-d3a2-42e0-b484-371d20412855">66</a></sup> and SCVI<sup data-ref="de7ab197-11b5-41e6-8398-35984586e492"><a href="#c67" data-ref="9a9e6101-8cb1-49e8-814f-02f3b799aa99">67</a></sup> packages. Analysis was performed on log transformed data. Library size normalization was performed on the intersection of the genes present in our data and the reference CS7 human embryo data to facilitate comparison of expression levels. We integrated the two scRNA-seq datasets for 42h 2D human gastruloids using SCVI based on the top 2000 highly variable genes, which were selected using the default function in Scanpy using the seurat_v3 flavor. Cell cycle genes were regressed out during integration. Dimensional reduction for visualization was performed using PHATE<sup data-ref="851b30ac-f8d2-40a3-a9a0-e267c7e32a7e"><a href="#c39" data-ref="95e6a9b4-0173-4111-a20d-017016cddc72">39</a></sup> on the SCVI latent space. Gene expression was smoothened with MAGIC<sup data-ref="db18db62-4910-4a19-a3ff-4f9f347714f1"><a href="#c68" data-ref="83c74556-8da6-4c23-9d44-cdd916c0ce69">68</a></sup> for visualization on PHATE plots in <span class="fakelink linktofig" data-ref="ddc20c16-3e99-4894-a2d0-1c88215057f0"><span class="fakelink-text" data-ref="94963edc-0534-4b8a-9a35-ef10c4f77c97">Fig. 2e,g-i</span><a class="fake-icon" href="#fig2" data-ref="8304059c-421d-4083-a0a6-d717825da2e0"><span class="fakelink-icon" data-ref="dcc01910-e18d-485b-bf5d-30cd2abb7c77">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. To annotate different cell types, we used MAGIC to impute that data and get smooth distributions of gene expression. We then thresholded marker genes to assign cell fate (Table 4), which produced better results than the more common unsupervised clustering using Leiden, and in particular more consistent results between datasets from different species. For interspecies comparison of FGF ligands we normalized the to the maximal FGF expression within each species, where for the mouse data, the three different time points were first integrated using SCVI. Thus we compare between species the relative expression of FGF ligands within each species, e.g., FGF8 expression is much higher than FGF4 at any time in the mouse, but much lower than FGF4 in human.</p><p id="el-p-53" data-ref="83064a85-15d8-4ba1-869d-9dfcc335ea69" data-id="el-p-53">For pseudotime analysis (<span class="fakelink linktofig" data-ref="43f039a6-28a3-4273-9887-ad7467fa99ec"><span class="fakelink-text" data-ref="03421c4d-1050-4ab9-a6e2-af6bd62838de">Fig. 5</span><a class="fake-icon" href="#fig5" data-ref="7aae573d-03c2-466e-bcd8-5490c2ec93f5"><span class="fakelink-icon" data-ref="53e50d73-60bc-41e6-af6f-7d7bf0c9fad8">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>), we applied the trajectory inference tool Slingshot<sup data-ref="7191f9a0-4d88-4145-84e7-4d5911b96e0b"><a href="#c69" data-ref="18871494-b73d-4943-9f31-7e860f88c098">69</a></sup> in R to compute the pseudotime trajectory in the SCVI latent representation. Slingshot determines the membership of a cell to a lineage at each branching point by its projection distance to the fitted principal curve of that lineage. We projected the mesoderm developmental trajectory onto the PHATE map for visualization. We further used TradeSeq<sup data-ref="a19b8906-3ef6-4191-800c-97b44b990099"><a href="#c70" data-ref="3e812994-0933-48c2-8d1e-04d7f3106e76">70</a></sup>, which fits a generalized additive model (GAM), to infer the pseudotime gene expression trends along the mesoderm lineage trajectory from Slingshot. For visualizing the correlation between the expression of FGF ligands and marker genes (<span class="fakelink linktofig" data-ref="e475cd28-56fe-435a-a7d5-a57f72d3558b"><span class="fakelink-text" data-ref="090e91f1-d8e6-42ae-b1c7-7c4a23a55e53">Supp. Fig. 5</span><a class="fake-icon" href="#figS5" data-ref="467dd987-a3bf-4d3e-ba81-371ba8ca67e2"><span class="fakelink-icon" data-ref="debeee53-f251-4962-8e03-46ec71abe05e">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>), we used our SCVI model to denoise the expression and mutual nearest neighbor to further harmonize the difference between samples.</p><p id="el-p-54" data-ref="082cf72d-7a42-47a5-a2e3-54e392b38837" data-id="el-p-54">To quantify isoforms from RNA sequencing salmon v1.10.0 was used<sup data-ref="e393c055-8991-49fd-9c8e-d6b7fa538d56"><a href="#c71" data-ref="0f9a24aa-2b85-42c4-9e24-595b01e13ff5">71</a></sup>. First, gencode.v45.transcripts.fa was downloaded from <span class="fakelink linktoext" data-ref="57595889-c601-4de2-9ff9-9a679b201553"><span class="fakelink-text" data-ref="a54384a1-b9f4-49c5-98eb-0d9f6431bbe6">gencodegenes.org</span><a class="fake-icon" href="https://gencodegenes.org" data-ref="ecfca714-e6be-4db4-9264-0efd97b0954f"><span class="fakelink-icon" data-ref="d55d00e0-1720-46b7-8610-43dfc6ec12c7">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. The reference was customized by removing lines for FGFR1 and replacing them with the transcript sequence for FGFRIIIa, FGFRIIIb, FGFRIIIc (sequences included as supplementary file). Three samples from GEO dataset GSM2257301 were downloaded through SRA (SRX1725562 (Pluripotent), SRX1725577 (APS), SRX1725643 (MPS))<sup data-ref="4beb6ff6-f2de-453a-92f8-c6a0fc08b9e9"><a href="#c38" data-ref="5f591f21-df67-49c7-a327-135a2b5e78e9">38</a></sup>. Adapters and the first 10 bases were trimmed from the FASTQ files using trim_galore. Finally, the Salmon “quant” command with the --validateMappings option was used to quantify transcripts from trimmed FASTQ files. For single cell datasets, <span class="fakelink linktoext" data-ref="b9e68d9d-6c41-4f4e-9168-d1175663dc6e"><span class="fakelink-text" data-ref="322e9f7d-b9a2-4037-bb51-1d959a29feec">https://timoast.github.io/sinto/</span><a class="fake-icon" href="https://timoast.github.io/sinto/" data-ref="7fbcbd8f-33e0-4e0c-878b-5815e09d777c"><span class="fakelink-icon" data-ref="511f6b7f-1d44-450d-8ad6-112aad1518a6">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span> was used to subset the 10x aligned file by cluster. Reads for clusters of interest were converted back to raw FASTQ files and run through the same alignment-based mode of salmon but as single end reads.</p></section><section id="s7g" data-ref="f4ff849e-52a8-40cc-9dd6-34719eced446" data-id="s7g"><h2 class="heading-2" id="el-h2-12" data-ref="91800f1b-e2d8-43b6-b73e-f79016758a5a" data-id="el-h2-12" data-break-after="avoid">Quantitative real-time PCR</h2><p id="el-p-55" data-ref="42c7e9e3-93f5-49e0-90cf-37a3c2a23b7e" data-id="el-p-55" data-previous-break-after="avoid">RNA were extracted using RNAqueous™-Micro Total RNA Isolation Kit (CAT# AM1931, Thermo Fisher Scientific) and then cDNA synthesis was prepared from it using SuperScript™ VILO™ cDNA Synthesis Kit (CAT# 11754250, Thermo Fisher Scientific) following the manufacturer‘s protocol with adjustment for optimization. Measurements were performed with SYBR green and the primers in the Table 5. GAPDH was used for normalization in all experiments.</p></section><section id="s7h" data-ref="32bd58a5-180a-41a2-b54c-4050df7b69ba" data-id="s7h"><h2 class="heading-2" id="el-h2-13" data-ref="34e76d36-a9b7-4e79-a7ab-27d68ec145c3" data-id="el-h2-13" data-break-after="avoid">Fluorescent dextran assays</h2><p id="el-p-56" data-ref="0da23cad-788e-42e1-9015-c95ab5f538dd" data-id="el-p-56" data-previous-break-after="avoid">Fluorescent dextran essays were performed by adding 10 ug/ml fluorescein isothiocyanate-dextran (10KD, Sigma Aldrich) to the media. To calculate fluorescence intensity as a function of distance from the colony edge, Otsu thresholding followed by several morphological operations was used to create a mask for the colony in each z-slice, thereby excluding bright dextran fluorescence outside the colony. Background subtraction on each z-slice of the image was performed to further remove background signal from dextran outside the colony while maintaining signal in the intercellular space. The radial intensity profile was then calculated from the maximal intensity projection of the masked, background subtracted image.</p></section><section id="s7i" data-ref="04d38b56-a0c1-4a22-bfbe-8dd33f2e07c0" data-id="s7i"><h2 class="heading-2" id="el-h2-14" data-ref="03876d94-c8fe-482e-97df-444d169f8c43" data-id="el-h2-14" data-break-after="avoid">Scratch assays</h2><p id="el-p-57" data-ref="02c0a806-715b-4e60-bd85-151a487387f8" data-id="el-p-57" data-previous-break-after="avoid">Human embryonic or induced pluripotent stem cells were differentiated on micropattern wells for 41h, then each micropattern colony was manually scratched with a serological needle. After 60mins, they were fixed with PFA or methanol followed by immunostaining with appropriate antibodies. To calculate the intensity profile as a function of distance from a scratch, we created a mask for the scratch based on the difference between an overall mask for the colony and the convex hull, calculated the distance transform of the scratch mask, then binned pixels by distance from the scratch and calculated their mean ERK intensity.</p></section><section id="s7j" data-ref="fc78b528-aae1-40fa-bdfe-e056d1b3574d" data-id="s7j"><h2 class="heading-2" id="el-h2-15" data-ref="5b7ef439-2aa2-49f5-8bce-15dae308d04f" data-id="el-h2-15" data-break-after="avoid">Transmembrane assays</h2><p id="el-p-58" data-ref="073be456-8953-4ecf-8457-93b4fd843592" data-id="el-p-58" data-previous-break-after="avoid">Human embryonic or induced pluripotent stem cells were seeded into Corning 6.5 mm transwell with 0.4 µm pore polyester membrane insert (cat# 3470) with E6 supplemented with TGF1b, FGF2, and Y-27632 for 16h, then media was changed to E6 with TGF1b. After 24h, the media was changed to E6 supplemented with FGF2 from the top or the bottom side of the transwell. 30 mins later, cells were fixed followed by immunofluorescence staining. The transwell cartoon in <span class="fakelink linktofig" data-ref="814b8c31-ceda-4935-a316-fed03aa18699"><span class="fakelink-text" data-ref="5f415d68-16d6-4ac9-bfe5-bcc4a157c7e0">Fig 4</span><a class="fake-icon" href="#fig4" data-ref="c58f9235-b9b8-4d89-8a9f-3b85d4e2f32a"><span class="fakelink-icon" data-ref="3a1651ef-d2c2-4540-bcca-f9300871bc62">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span> was generated with Biorender.</p></section><section id="s7k" data-ref="ed05afaf-1f04-47f9-9f62-e531faf702f5" data-id="s7k"><h2 class="heading-2" id="el-h2-16" data-ref="53ef41a2-5e0a-473e-b037-947e6bddc23f" data-id="el-h2-16" data-break-after="avoid">Fluorescence In Situ Hybridization</h2><p id="el-p-59" data-ref="3c27d096-f088-4827-ab65-4d498e7b42d7" data-id="el-p-59" data-previous-break-after="avoid">The fluorescence in situ hybridization protocol (FISH) was performed according to the manufacturer’s instructions (ACDbio, RNAscope multiplex fluorescent manual). A list of probes and reagents can be found in <span class="fakelink linktofig" data-ref="4d142ec1-e6d5-493e-a5ca-018688ae14b3"><span class="fakelink-text" data-ref="c01f2b3f-bad0-4466-8bf9-62af85f85683">Supplementary Table 6</span><a class="fake-icon" href="#tblS6" data-ref="25d664e1-bb11-439e-8394-7dfb0c0dbdca"><span class="fakelink-icon" data-ref="36505a63-adf5-411c-a9af-f13c715c32c0">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>.</p></section><section id="s7l" data-ref="d307c65d-81ea-4996-8c64-d010e3c7e2bb" data-id="s7l"><h2 class="heading-2" id="el-h2-17" data-ref="25de984b-002a-4a99-8353-1a7d891c47a3" data-id="el-h2-17" data-break-after="avoid">Gene targeting by shRNA</h2><p id="el-p-60" data-ref="31025fd3-a04f-4214-8e31-d32ff55d7453" data-id="el-p-60" data-previous-break-after="avoid">We used the constructs shRNA F4: pPB[shRNA]-EGFP-U6&gt;hFGF4[shRNA#1] and shRNA F17: pPB[3shRNA]-U6&gt;hFGF17[shRNA#1]-U6&gt;hFGF17[shRNA#2]-U6&gt;{hFGF17_shRNA3}-hPGK&gt;Bsd both purchased from VectorBuilder in a PiggyBac Transposon Vectors with puromycin resistance which was stably integrated into ESI17 hESCs and PGP1 hiPSCs. Transfections were performed using Lipofectamine™ Stem Transfection Reagent (CAT# STEM00001, Thermo Fisher Scientific), following the manufacturer‘s protocol.</p></section><section id="s7m" data-ref="54094027-9439-442f-a6b7-a98f6f8c4349" data-id="s7m"><h2 class="heading-2" id="el-h2-18" data-ref="ebe8d40f-6484-419b-9d59-f93c9d09718b" data-id="el-h2-18" data-break-after="avoid">Gene targeting by siRNA</h2><p id="el-p-61" data-ref="cd852a38-25c5-4e22-b062-38aeb8952bdf" data-id="el-p-61" data-previous-break-after="avoid">We purchased predesigned siRNA from Millipore Sigma; NM_002007 (FGF4), NM_006119 (FGF8), NM_008004 (FGF17). The siRNA transfections were performed using Lipofectamine™ RNAiMAX Transfection Reagent (CAT# 13778075, Thermo Fisher Scientific), following the manufacturer‘s protocol with adjustment for optimization.</p></section><section id="s7n" data-ref="d46ccec0-da44-4b25-b83b-5aabbae49bc5" data-id="s7n"><h2 class="heading-2" id="el-h2-19" data-ref="0de825ad-732f-43a5-906e-81417efcd151" data-id="el-h2-19" data-break-after="avoid">Statistics and reproducibility</h2><p id="el-p-62" data-ref="abc720a0-8cf9-47c9-843b-00684ba397dd" data-id="el-p-62" data-previous-break-after="avoid">All experiments were performed twice or more. All the attempts in replicating experiments yielded consistent results. Unless specifically noted, all the quantifications were performed with N=4 colonies within the same experimental condition.</p></section><section id="s7o" data-ref="427bca71-fa8c-4445-b2da-c76519730cf2" data-id="s7o"><h2 class="heading-2" id="el-h2-20" data-ref="163cc390-29cb-4355-8e6b-a6b39bc08698" data-id="el-h2-20" data-break-after="avoid">Ethics statement</h2><p id="el-p-63" data-ref="05de05ab-8d2d-477e-b8ef-2f8593c3842a" data-id="el-p-63" data-previous-break-after="avoid">We comply with all the ethical regulations in our research. We were granted approval by the Human Pluripotent Stem Cell Research Oversight (HPSCRO) Committee at the University of Michigan to work with human embryonic stem cells and human induced pluripotent stem cells.</p><ul class="list-bullet" id="el-ul-3" data-ref="c33faaaa-41b9-4f86-83b5-64b5615091f8" data-id="el-ul-3"><li id="el-li-20" data-ref="f798bb57-79c7-490b-96a9-1e95e7e1bf03" data-id="el-li-20"><p id="el-p-64" data-ref="e292462d-0d3c-4213-b8ef-9f7ed9383bb3" data-id="el-p-64">Read orthogroup alignment</p></li><li id="el-li-21" data-ref="fb0829cc-a4e3-438a-b57c-d7255e556e9e" data-id="el-li-21"><p id="el-p-65" data-ref="46376225-f4f1-4d9e-b94c-e24d21290869" data-id="el-p-65">Pre-process alignment</p><ol class="list-alpha-lower" id="el-ol-2" data-ref="3697a35c-e3a0-48bf-98a6-e4e0347b0149" data-id="el-ol-2"><li id="el-li-22" data-ref="77f9634f-b1c3-46d0-b8aa-dd07b1d64473" data-id="el-li-22" data-item-num="1"><p id="el-p-66" data-ref="dbbd5faa-2b85-45c1-b1a9-5ffc6608f6ad" data-id="el-p-66">Trim alignment edges and remove columns with high fraction of gap codons</p></li><li id="el-li-23" data-ref="a5898ff5-63ee-4a46-b620-7721290f05d6" data-id="el-li-23" data-item-num="2"><p id="el-p-67" data-ref="87a74e1b-9d35-4b4d-8336-cc0e7971f21e" data-id="el-p-67">For each species, infer consensus sequence and remove sequences outside a predetermined diameter around the consensus</p></li><li id="el-li-24" data-ref="ba549f86-4135-4242-93e0-299140c83f24" data-id="el-li-24" data-item-num="3"><p id="el-p-68" data-ref="17c0fa27-6083-4ed3-86bb-aeb45c982df2" data-id="el-p-68">Replace all SNPs with absolute frequencies <em data-ref="6a86a7e4-558d-48f7-ac68-19ea4f19026e">n</em> &lt; <em data-ref="49d72133-1efc-41fa-accb-ec6695c08714">n</em><sub data-ref="596959ec-5dfd-46bc-967b-f3a0c6a6559f"><em data-ref="18a32d90-f88f-4ead-a7d6-020d703d129f">c</em></sub> = 4 with the consensus for their respective columns</p></li></ol></li><li id="el-li-25" data-ref="d9a8a80b-841d-4c69-9b36-6ae9c53d7c18" data-id="el-li-25"><p id="el-p-69" data-ref="5631eb74-a13b-474b-a8ed-1cb36063822e" data-id="el-p-69">For each species <em data-ref="42f8e491-4749-4773-a599-e73c1549d7bb">s</em>, subsample sequences from cells assigned to <em data-ref="46736ce1-0cde-44a7-99c0-4d38d95c5ce0">s</em></p></li><li id="el-li-26" data-ref="074a6fd9-63e8-4f46-bd19-797105747324" data-id="el-li-26"><p id="el-p-70" data-ref="e8868823-4fc1-47ab-847c-38e59cf329fe" data-id="el-p-70">Calculate the linkage matrix <img class="child-of-p insidetext inline-formula max-10 formula-max-0" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F543983v2_inline26c.gif/full/max/0/default.jpg" alt="" id="el-img-7" data-ref="6d29f59e-3a9d-4f0a-81c9-1cf7574933b8" data-id="el-img-7" data-desired-width="0"> for all pairs of SNPs</p></li><li id="el-li-27" data-ref="0c9735f1-97ce-4dd6-8b88-2ea2bc205906" data-id="el-li-27"><p id="el-p-71" data-ref="50d37af0-6abf-448b-b0a5-65c918267af6" data-id="el-p-71">Identify groups of consecutive SNPs with <img class="child-of-p insidetext inline-formula max-10 formula-max-0" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F543983v2_inline26d.gif/full/max/0/default.jpg" alt="" id="el-img-8" data-ref="042d86cb-e5ac-40e3-8827-9619898fa9e6" data-id="el-img-8" data-desired-width="0">, up to a rounding error of <em data-ref="07657945-11ee-45fd-a725-b339611e5f5c">ϵ</em> = 10<sup data-ref="00f1f843-3410-4d01-b78f-0b0e79d67e2a">−5</sup>, containing at least <em data-ref="10980dc5-4076-4056-aa92-94d583ae831f">l</em><sub data-ref="f7be15f9-36e9-47b2-9f30-24553e943234"><em data-ref="10d20b36-93e4-4edc-8ddd-b7c4d9acf316">min</em></sub> = 5 SNPs</p></li></ul><pre data-ref="ded3ce88-af93-4fb5-a0a9-ed231d5e9c3a">for $x in y
dnskdnkfnfn
 nklfnk
    nklfnklf
      nklfnklfnklfnklfnkln-snk snklf msnk s njkf
 dnsklankldfa
ndjdb ekj
</pre></section></section><section id="s8" data-ref="10c18e3d-cedf-4dae-8ccd-63baf18431ea" data-id="s8"><h1 class="heading-1" id="el-h1-6" data-ref="592b7bbe-ca2c-4187-b9a4-ecb8ce1e86e9" data-id="el-h1-6" data-break-after="avoid">Supplementary tables</h1><div class="fig-group" id="el-div-12" data-ref="3ff23050-d50b-4bd5-8471-09f73402ddb3" data-id="el-div-12" data-previous-break-after="avoid"><p class="figure movedfig fig-square" id="tblS1" data-ref="ffe4f0ba-1e85-491c-ab29-5697fe2b3eba" data-id="tblS1"><img class="child-of-figure imageonly square max-90" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_tblS1.tif/full/max/0/default.jpg" alt="" id="el-img-9" data-ref="61712371-196c-4834-814a-8cd7c6e79dad" data-id="el-img-9" height="1201" width="1977" data-desired-width="93.60795454545455"></p><p class="figure__caption" id="el-p-73" data-ref="143e2839-c34c-4840-a96a-9cf2bbc47921" data-id="el-p-73"><span class="label figure-name" data-ref="6da19412-6e9f-442b-ba60-e7cadd9d6e6d"><label class="figure__label" data-ref="060f070b-5daa-44d0-8967-2fa2d1365afc">Supplementary Table 1.</label></span><strong data-ref="6423eaea-2daa-4458-a7c5-181e6ce1c29c">Cell signaling reagents.</strong> <strong data-ref="96e44c6a-dca8-4dbb-98d8-0a7eb7112563">a</strong>) Stains and quantification for pERK and TBXT in smaller colonies, diameter from left to right: 500um, 350um, 250um. <strong data-ref="a916d59b-d0f5-44a4-8011-6bc2dc719e06">b)</strong> Staining for three primitive streak markers with or with FGFRi or MEKi.</p></div><figure class="figure tofill has-continued imgs-1 shortercaption fig-landscape pagedjs-fill-next-page" id="tblS2" data-continued-main="ctn-tblS2" data-ref="dc731de0-bca6-4d5e-8534-1a536093bdd3" data-id="tblS2" style="display: none;"><figcaption class="figure__caption" id="el-figcaption-6" data-ref="e43fcfee-187e-40e5-9395-b20a5fb6512b" data-id="el-figcaption-6"><h3 id="el-h3-6" data-ref="84866565-4edc-4de5-9788-b4de711f3740" data-id="el-h3-6" data-break-after="avoid"><span class="label figure-name" data-ref="533aa9cf-39cf-428e-af3f-7446b800dc66"><label class="figure__label" data-ref="c80e2bd9-c92a-448f-a10d-ae42dae4f06c">Supplementary Table 2.</label></span>Primary antibodies used for immunofluorescence.</h3></figcaption><img class="child-of-figure imageonly landscape max-90" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_tblS2.tif/full/max/0/default.jpg" alt="" id="el-img-10" data-ref="237ffeb6-864d-4750-b234-a0b84e9a4108" data-id="el-img-10" data-previous-break-after="avoid" height="879" width="1977" data-desired-width="93.60795454545455"></figure><figure data-continued-from="ctn-tblS2" class="added-figure-tofill tofill continuedContent imgs-1 shortercaption fig-landscape pagedjs-fill-next-page" id="tblS2-cont-2" data-ref="e0cc2744-030f-4f30-af73-173135a84aea" data-id="tblS2-cont-2" data-following="following-3a4bc264-9217-479e-8230-fe32b422b13b" style="display: none;"><figcaption id="el-figcaption-7" data-ref="1826d21c-2c14-4443-879e-9e9b22b3d2b1" data-id="el-figcaption-7"><p data-brea-after="unset" class="continued" id="el-p-74" data-ref="982205ed-5eee-4ce5-ab67-4cb4788b6620" data-id="el-p-74"><span class="label figure-name" data-ref="9dea018e-a625-461f-b660-3ed7d7241963"><label class="figure__label" data-ref="954bea08-9511-45d0-96d2-e5b536111571">Supplementary Table 2.</label></span> <span class="ctn" data-ref="244a3643-a536-4a5f-8442-5904ba4bec1b">(continued)</span></p></figcaption><img class="child-of-figure imageonly landscape max-90" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_tblS2a.tif/full/max/0/default.jpg" alt="" id="el-img-11" data-ref="34c96e51-080a-4a21-bb03-f31c2283717c" data-id="el-img-11" height="229" width="1977" data-desired-width="93.60795454545455"></figure><figure class="figure tofill imgs-1 shortercaption fig-landscape pagedjs-fill-next-page" id="tblS3" data-ref="23f88b08-4ba7-426f-90d3-e78948723c17" data-id="tblS3" data-following="following-3a4bc264-9217-479e-8230-fe32b422b13b" style="display: none;"><figcaption class="figure__caption" id="el-figcaption-8" data-ref="6c45a311-fee2-4c10-8722-901be0e211ad" data-id="el-figcaption-8"><h3 id="el-h3-7" data-ref="fd9888fe-b7b6-4d5f-b1b6-85964d2b012d" data-id="el-h3-7" data-break-after="avoid"><span class="label figure-name" data-ref="8f8a1746-df6b-4ebb-bcca-e6d7cb219153"><label class="figure__label" data-ref="2b9596f3-e8d4-40fd-b8ab-026005a3861b">Supplementary Table 3.</label></span>Secondary antibodies.</h3></figcaption><img class="child-of-figure imageonly landscape max-90" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_tblS3.tif/full/max/0/default.jpg" alt="" id="el-img-12" data-ref="73802c59-f964-47c3-aaa5-8e48403a7e09" data-id="el-img-12" data-previous-break-after="avoid" height="643" width="1977" data-desired-width="93.60795454545455"></figure><figure class="figure tofill imgs-1 shortercaption fig-landscape pagedjs-fill-next-page" id="tblS4" data-ref="d31ac509-877f-410d-b169-7eec99ee7d96" data-id="tblS4" data-following="following-3a4bc264-9217-479e-8230-fe32b422b13b" style="display: none;"><figcaption class="figure__caption" id="el-figcaption-9" data-ref="cd8ccfb7-b6fa-443d-8505-96ac80d79bfe" data-id="el-figcaption-9"><h3 id="el-h3-8" data-ref="82e1c0c0-8439-4465-bae6-9790aacc116b" data-id="el-h3-8" data-break-after="avoid"><span class="label figure-name" data-ref="25685ffa-17ab-4c0f-85c4-d9320cc9f9fd"><label class="figure__label" data-ref="cb3e0b4f-392d-4fc7-a416-c31fc5b578d5">Supplementary Table 4.</label></span>Marker genes for cell fate annotation</h3></figcaption><img class="child-of-figure imageonly landscape max-90" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_tblS4.tif/full/max/0/default.jpg" alt="" id="el-img-13" data-ref="0f7c9dcd-169f-4d89-9fd0-318e29a0eb05" data-id="el-img-13" data-previous-break-after="avoid" height="854" width="1977" data-desired-width="93.60795454545455"></figure><figure class="figure tofill imgs-1 shortercaption fig-landscape pagedjs-fill-next-page" id="tblS5" data-ref="8b126539-c6df-45fe-9328-ab39855ada74" data-id="tblS5" data-following="following-3a4bc264-9217-479e-8230-fe32b422b13b" style="display: none;"><figcaption class="figure__caption" id="el-figcaption-10" data-ref="0294e6fa-f054-49a9-b3ca-2f6bf02e4129" data-id="el-figcaption-10"><h3 id="el-h3-9" data-ref="7668bfc2-119d-44a6-a022-b67d9e52f0d4" data-id="el-h3-9" data-break-after="avoid"><span class="label figure-name" data-ref="00dcd388-7c93-40ba-bfec-13c67fbe93f7"><label class="figure__label" data-ref="2d90edfe-c16b-419a-b4b3-fde2e34569f6">Supplementary Table 5.</label></span>qPCR primers</h3></figcaption><img class="child-of-figure imageonly landscape max-90" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_tblS5.tif/full/max/0/default.jpg" alt="" id="el-img-14" data-ref="2eacd477-79bd-4831-9a40-7a33906652f3" data-id="el-img-14" data-previous-break-after="avoid" height="434" width="1977" data-desired-width="93.60795454545455"></figure><figure class="figure tofill imgs-1 shortercaption fig-landscape pagedjs-fill-next-page" id="tblS6" data-ref="31ea43c3-94de-405e-8477-63f2e04d0d15" data-id="tblS6" data-next-break-before="page" data-following="following-3a4bc264-9217-479e-8230-fe32b422b13b" style="display: none;"><figcaption class="figure__caption" id="el-figcaption-11" data-ref="a156aaf4-2337-4f15-b1d2-e0beee0c1aaf" data-id="el-figcaption-11"><h3 id="el-h3-10" data-ref="9fef3ef9-7aab-4f75-acb6-a72c6091e997" data-id="el-h3-10" data-break-after="avoid"><span class="label figure-name" data-ref="b68877ac-1964-43e4-95e7-4601c23d818d"><label class="figure__label" data-ref="3a54d7b7-08c3-4f94-a19b-ed292dfe1e92">Supplementary Table 6.</label></span>Probes for Fluorescent In Situ Hybridization</h3></figcaption><img class="child-of-figure imageonly landscape max-90" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_tblS6.tif/full/max/0/default.jpg" alt="" id="el-img-15" data-ref="66d6754b-f4c1-4af5-b45a-d87fb38cd5e1" data-id="el-img-15" data-previous-break-after="avoid" height="288" width="1977" data-desired-width="93.60795454545455"></figure><div class="pagebreak" id="el-div-13" data-ref="6041e681-64fd-4b5f-977a-b2fad5da7042" data-id="el-div-13" data-break-before="page"></div></section><section id="s9" data-ref="4f67efb4-0c23-4f85-8633-c11ba90c3901" data-id="s9"><h1 class="heading-1" id="el-h1-7" data-ref="ae7e5410-2092-4922-bd70-5348b97e2d84" data-id="el-h1-7" data-break-after="avoid">Supplementary Figures</h1><div class="fig-group" id="el-div-14" data-ref="cbb4a61f-7117-4782-91fb-10bb576cfd97" data-id="el-div-14" data-previous-break-after="avoid"><p class="figure movedfig fig-portrait" id="figS1" data-ref="4effe846-3120-4498-a4d4-fa93f3356c64" data-id="figS1"><img class="child-of-figure has-boundary imageonly portrait max-100" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_figS1.tif/full/max/0/default.jpg" alt="" id="el-img-16" data-ref="8efee1c4-7ed6-4899-818a-616a6092ac1a" data-id="el-img-16" height="2390" width="2082" data-desired-width="98.57954545454545"></p><p class="figure__caption" id="el-p-76" data-ref="5955e46c-57ca-4675-b1e5-88e05910f5cb" data-id="el-p-76"><span class="label figure-name" data-ref="7e92aca6-52ac-44ff-a6cb-b1d0ceb8d033"><label class="figure__label" data-ref="4a9363d6-2426-45b0-81b2-6a20b437b94f">Supplementary Figure 1.</label></span><strong data-ref="6a461ecc-b04f-4c6e-9fa9-22150d317456">a</strong>) Stains and quantification for pERK and TBXT in smaller colonies, diameter from left to right: 500um, 350um, 250um. <strong data-ref="829ad814-9a0b-459d-b1a5-5719c62afdb3">b)</strong> Staining for three primitive streak markers with or with FGFRi or MEKi. <strong data-ref="de9e7c9f-bbc9-441f-8f86-d76a4643806e">c)</strong> DAPI stain corresponding to <span class="fakelink linktofig" data-ref="90a78405-4734-4f1c-94a1-838b350f27eb"><span class="fakelink-text" data-ref="43d35f1e-61fb-4edf-aef6-e4897cf2b076">Fig. 1d</span><a class="fake-icon" href="#fig1" data-ref="f70e7f9c-2b1b-43bb-ad76-d2ea1b748ed3"><span class="fakelink-icon" data-ref="90a3f6ca-e7ac-455d-b2ac-d0881ce5b1f2">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. <strong data-ref="a905ceef-ac08-4318-a6d3-5f79cd106c56">d)</strong> Radial intensity profile for TBXT corresponding to <span class="fakelink linktofig" data-ref="fd2bc05d-a858-404a-bb32-1620ca92d9b7"><span class="fakelink-text" data-ref="d38962c1-98d2-4263-94a6-cf14c74db500">Fig. 1d</span><a class="fake-icon" href="#fig1" data-ref="b93e98a2-2eb6-462f-9461-3b92bfe6186b"><span class="fakelink-icon" data-ref="b1d7c57f-386a-427d-bf60-1b284fe7da1b">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. <strong data-ref="b9ba7609-ae47-4d4f-9559-1bb93c7ef153">e</strong>) BrdU and cleaved Cas3 stainings after MEK or FGFR inhibition. <strong data-ref="64313dad-4774-437e-8685-0d808bc99fcd">f)</strong> Quantification of conditions in e. <strong data-ref="5ef244e1-ac79-4fb9-9ddf-74e2e7640de7">g)</strong> Quantification of conditions in <span class="fakelink linktofig" data-ref="36cf3660-4816-4cc1-abb5-192465bf2bbc"><span class="fakelink-text" data-ref="45c22ba4-1531-4f96-847f-c913911c3f10">Fig. 1g</span><a class="fake-icon" href="#fig1" data-ref="73fcc984-cf56-477e-8a2e-4b8db20fdcaf"><span class="fakelink-icon" data-ref="869a9c52-1751-4ac0-b9c5-3b850cf682f2">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span> for five images each, N indicates total number of cells. <strong data-ref="fab3b8fb-8201-47af-8d0c-3f2ec82ab51a">h</strong>) Low density differentiation with and without MEK and FGFR inhibition. <strong data-ref="ba1a28fb-a4ad-491a-a3e0-58ab96a5ed68">i)</strong> pERK and TBXT stains and quantification for different doses of MEK inhibitor shows low doses effectively block differentiation with minimal impact on cell number.</p></div><figure class="figure tofill imgs-1 longcaption fig-portrait pagedjs-fill-next-page" id="figS2" data-ref="80dcc250-1945-4d0c-a883-a6436eb63ed3" data-id="figS2" style="display: none;"><figcaption class="figure__caption" id="el-figcaption-12" data-ref="6da9d486-115c-400c-8da3-629bfff45923" data-id="el-figcaption-12"><h3 id="el-h3-11" data-ref="b8d11858-4263-4419-8077-32b007f32087" data-id="el-h3-11" data-break-after="avoid"><span class="label figure-name" data-ref="1454869f-906f-46c8-ba15-bab6e0694358"><label class="figure__label" data-ref="15641529-2a79-4a51-bfd0-686ee5c97519">Supplementary Figure 2.</label></span></h3><p id="el-p-77" data-ref="c4f71d91-918a-47f4-bb49-2dc20a8c16c9" data-id="el-p-77" data-previous-break-after="avoid"><strong data-ref="8fc267ce-228d-4e64-a0a6-7795ebd4dc75">a)</strong> PHATE projection of scRNA-seq data colored for sample. <strong data-ref="e344831a-c7df-424d-b30e-09355e62db13">b,c)</strong> Expression of canonical FGF ligands across different cell types in 2D gastruloids (b) and human CS7 embryo (c). Expression across clusters is normalized by row, while total expression is shown in a separate column to the directly to left. The leftmost column indicates differential expression between PS-like and pluripotent cells. The color scale of (differential) expression is cut off for better contrast. <strong data-ref="1107a637-d6df-4d82-8601-c79fb6675057">d)</strong> Threshold-based annotation (left) for human, monkey, and mouse embryos used in interspecies comparison (<span class="fakelink linktofig" data-ref="3e4311cc-aec3-42c1-a105-56260b697571"><span class="fakelink-text" data-ref="c61a06a0-226d-4922-82b6-28a1568de00e">Fig. 2f</span><a class="fake-icon" href="#fig2" data-ref="2727c34e-5105-4d62-9b20-040dce81bf0a"><span class="fakelink-icon" data-ref="7931918f-e42c-4026-a59c-c064b4e2165f">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>), compared to original annotation (right). <strong data-ref="fedb8e83-eab5-4a70-a85e-a8b3f33cb817">e)</strong> Interspecies comparison as in <span class="fakelink linktofig" data-ref="cf2c217f-21b4-4b30-8708-8455bc09c0b7"><span class="fakelink-text" data-ref="161fc497-156e-49d0-a21f-f30c1e56ed5c">Fig. 2f</span><a class="fake-icon" href="#fig2" data-ref="3b107273-af1c-4cba-8aa1-6a51e670fb9a"><span class="fakelink-icon" data-ref="70556db7-37b2-495e-baca-afe969b8d3bf">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span> but for nascent mesoderm and PS-LC + nascent mesoderm. <strong data-ref="08487d79-5af5-46ee-a637-3b6217578fbf">f)</strong> Bulk RNA-seq data by Loh et al<sup data-ref="7ba03a10-e289-4944-9257-f046fb964bb0"><a href="#c38" data-ref="0e5e1365-e0f9-4dfa-a8f2-bf43bd347915">38</a></sup> for expression of different FGFs in directed differentiation to mid primitive streak (MPS) or anterior primitive streak (APS) relative the human embryonic stem cells (hESCs). <strong data-ref="0085c411-abe1-4a84-b33e-dfd9cdbef33d">g)</strong> Expression of genes involved in FGF signaling.</p></figcaption><img class="child-of-figure has-boundary imageonly portrait max-100" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_figS2.tif/full/max/0/default.jpg" alt="" id="el-img-17" data-ref="75b87741-b8df-4cab-9948-916c17db0f53" data-id="el-img-17" height="2357" width="2026" data-desired-width="95.9280303030303"></figure><figure class="figure tofill imgs-1 shortercaption fig-square pagedjs-fill-next-page" id="figS3" data-ref="dd27229a-6db1-47f7-990e-ca5781051db9" data-id="figS3" data-following="following-3a4bc264-9217-479e-8230-fe32b422b13b" style="display: none;"><figcaption class="figure__caption" id="el-figcaption-13" data-ref="17373d6c-4300-4e40-9d6f-5d1e76ae0a2f" data-id="el-figcaption-13"><h3 id="el-h3-12" data-ref="1a5ce017-ccf9-4c0e-bd90-3f5f8aede3ec" data-id="el-h3-12" data-break-after="avoid"><span class="label figure-name" data-ref="cb9ffe82-9bf4-4da1-86ae-9e74e79ebb27"><label class="figure__label" data-ref="42ca2310-f229-4feb-86e9-fb5e6581c09b">Supplementary Figure 3.</label></span></h3><p id="el-p-78" data-ref="33ef876e-5756-4516-8bab-f36ee8371ef5" data-id="el-p-78" data-previous-break-after="avoid"><strong data-ref="f32341db-2e8b-4ed5-ad50-b41b6c506a9d">a)</strong> DAPI image corresponding to <span class="fakelink linktofig" data-ref="0a1e6920-398e-4b04-a23f-cd611f5457ac"><span class="fakelink-text" data-ref="25c26521-905c-4210-86a3-27fa95e71306">Fig. 3a</span><a class="fake-icon" href="#fig3" data-ref="eebee298-a7a4-4019-bff3-6684c7577c13"><span class="fakelink-icon" data-ref="cacb6265-f61e-4c13-b766-43459cc38171">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. <strong data-ref="bb9a9aa4-40cb-4edd-be72-e890930cd36b">b-c)</strong> pFGFR1 (b) or pERK (c) and TBXT stains with and without continuous MEKi treatment. <strong data-ref="4376ea97-f8f7-4aff-a047-692a3eb0669a">d)</strong> DAPI image corresponding to <span class="fakelink linktofig" data-ref="d057049b-55e7-4d96-a918-ce3d00cfcccb"><span class="fakelink-text" data-ref="ca04992a-c41f-4af0-a885-1dc5905584bb">Fig. 3b</span><a class="fake-icon" href="#fig3" data-ref="bbebb22b-deb3-4ad7-8f6c-62651449a513"><span class="fakelink-icon" data-ref="7cd35db8-89fd-407d-aa2b-d68078f61a87">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. <strong data-ref="bbaf5d97-1d94-451a-adb4-3139e01fdff7">e)</strong> Design of qPCR primers to detect FGFR1 isoforms, diagram adapted from Eswarakumar et al<sup data-ref="a4581912-1200-49c4-97da-f334c48623aa"><a href="#c72" data-ref="6da93a0c-d1a4-46c2-86ea-c37a1efc0f1d">72</a></sup>. <strong data-ref="6c281367-7ea7-4682-b517-ecbaa0fcebbb">f)</strong> Primer sequence and properties. <strong data-ref="64b88dda-c53f-4de3-a8dc-296b00dd13c1">g)</strong> Amplicon sizes match prediction. Scale bars 50um.</p></figcaption><img class="child-of-figure has-boundary imageonly square max-100" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_figS3.tif/full/max/0/default.jpg" alt="" id="el-img-18" data-ref="8f974b59-0136-4f25-8b66-9cfe11960524" data-id="el-img-18" height="1706" width="2103" data-desired-width="99.57386363636364"></figure><figure class="figure tofill imgs-1 shortercaption fig-landscape pagedjs-fill-next-page" id="figS4" data-ref="805e78b8-f6b9-4003-b5e6-38d139a66c59" data-id="figS4" data-following="following-3a4bc264-9217-479e-8230-fe32b422b13b" style="display: none;"><figcaption class="figure__caption" id="el-figcaption-14" data-ref="819dbce5-9796-4f99-9a2a-98be58c49a47" data-id="el-figcaption-14"><h3 id="el-h3-13" data-ref="1fb529ce-255d-4135-8a72-581758ffb871" data-id="el-h3-13" data-break-after="avoid"><span class="label figure-name" data-ref="ee7eefa0-5855-4e5c-a533-f60119b74ea1"><label class="figure__label" data-ref="af3f4075-bf90-441a-bad0-9ca742ddf6f8">Supplementary Figure 4.</label></span></h3><p id="el-p-79" data-ref="fe61e90d-20df-4e1a-ad9b-b1090309ee25" data-id="el-p-79" data-previous-break-after="avoid"><strong data-ref="561be02c-87f0-4157-b779-f62779701171">a,b)</strong> Heparan sulfate (HS) stain (a) and radial intensity profile (b). Scale bar 50um. Error bands represent standard deviation over N=4 colonies.</p></figcaption><img class="child-of-figure has-boundary imageonly landscape max-90" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_figS4.tif/full/max/0/default.jpg" alt="" id="el-img-19" data-ref="90d336cf-1438-4b68-a404-98d1a9a3cfd5" data-id="el-img-19" height="413" width="1872" data-desired-width="88.63636363636364"></figure><figure class="figure tofill imgs-1 shortercaption fig-square pagedjs-fill-next-page" id="figS5" data-ref="85118bbe-b148-4932-847c-06c99462d9f3" data-id="figS5" data-following="following-3a4bc264-9217-479e-8230-fe32b422b13b" style="display: none;"><figcaption class="figure__caption" id="el-figcaption-15" data-ref="0a281842-892d-4229-a4d4-7051f5e5fd29" data-id="el-figcaption-15"><h3 id="el-h3-14" data-ref="cb1b7e89-8ce3-4bca-bfea-830fd0f7aeb5" data-id="el-h3-14" data-break-after="avoid"><span class="label figure-name" data-ref="7ab79152-43b7-42a4-b9b4-eb90e95f9519"><label class="figure__label" data-ref="9781a2f7-be4e-4904-9861-cc6d853b4251">Supplementary Figure 5.</label></span></h3><p id="el-p-80" data-ref="c1dae0bc-c6ee-43ea-b794-3d1006352370" data-id="el-p-80" data-previous-break-after="avoid"><strong data-ref="b672ca16-3a62-48ac-a0f6-380e437a13b6">a)</strong> DAPI image corresponding to <span class="fakelink linktofig" data-ref="8d30fd8f-3273-4154-b71b-8705011aab6d"><span class="fakelink-text" data-ref="227662bf-1278-47c0-bba5-8b6ea720d07a">Fig. 5a</span><a class="fake-icon" href="#fig5" data-ref="8ed358b6-a9d2-4006-b965-0a45d5a402c1"><span class="fakelink-icon" data-ref="3ce25dc9-9f12-4c40-9734-f591df76a48e">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. <strong data-ref="ccb34b6c-d91a-4c11-a7b9-00f8fc61c8b1">b)</strong> Pseudotime dynamics from <span class="fakelink linktofig" data-ref="81fec67a-04a4-4e9d-ba6c-b277c6e7abad"><span class="fakelink-text" data-ref="1d785b2c-bdea-4416-a4e2-832073e275e2">Fig. 5b</span><a class="fake-icon" href="#fig5" data-ref="f9f3a81d-2d6c-4cde-91b9-2d5f611c59e3"><span class="fakelink-icon" data-ref="7c569a95-9652-4683-8bbd-e40542d3e810">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span> with each gene normalized individually. <strong data-ref="6d982933-290a-447e-b721-9b73f1f09e70">c)</strong> Correlations between FGFs and TBXT, TBX6 colored for clusters, with cluster colors matching <span class="fakelink linktofig" data-ref="fb8d01aa-c9ba-4520-98a8-172bdb3339d7"><span class="fakelink-text" data-ref="5af03978-a83b-47f6-b49c-6370e87d4b16">Fig. 2d</span><a class="fake-icon" href="#fig2" data-ref="e38f9c39-b52a-429c-9158-1432f18ba91f"><span class="fakelink-icon" data-ref="12316162-f009-465c-a4d7-0bbfff25b865">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. <strong data-ref="084b4364-0b00-4095-828b-0215351c5343">d-f)</strong> DAPI images and overlays including DAPI corresponding to <span class="fakelink linktofig" data-ref="04079d2b-870a-42ee-ba81-349cbe316122"><span class="fakelink-text" data-ref="619d5585-35ab-45fb-869e-e9dcb1b75bd3">Fig. 5c-e</span><a class="fake-icon" href="#fig5" data-ref="2b5d3db3-1791-407d-a784-73ab04fb31bd"><span class="fakelink-icon" data-ref="47f9e537-bd6e-4ad4-963e-ef5bd4b56e02">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, respectively. <strong data-ref="fdf134fa-d4ca-4c42-acd1-ad5a64d35637">g)</strong> Combined FISH for FGF4 and FGF8. <strong data-ref="cef6d464-d191-4290-a839-d590e110f8d9">h)</strong> Scatterplots of FGF4, FGF8, and FGF17 mRNA versus TBXT colored for cell density.</p></figcaption><img class="child-of-figure has-boundary imageonly square max-100" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_figS5.tif/full/max/0/default.jpg" alt="" id="el-img-20" data-ref="a59c8fab-6081-4447-bae7-cc47a0c8eed4" data-id="el-img-20" height="2023" width="2103" data-desired-width="99.57386363636364"></figure><figure class="figure tofill imgs-1 shortercaption fig-portrait pagedjs-fill-next-page" id="figS6" data-ref="6fd30c2f-85d8-4123-a99a-57279450f764" data-id="figS6" data-next-break-before="page" data-following="following-3a4bc264-9217-479e-8230-fe32b422b13b" style="display: none;"><figcaption class="figure__caption" id="el-figcaption-16" data-ref="d9427009-dd2b-442e-9f87-22c05494aa91" data-id="el-figcaption-16"><h3 id="el-h3-15" data-ref="7dc0379f-ffe5-4df7-8fa5-980cf7387f68" data-id="el-h3-15" data-break-after="avoid"><span class="label figure-name" data-ref="3ccec7e5-f1c7-4f33-b04f-0dddc129a4f9"><label class="figure__label" data-ref="462bc3d1-c9af-4e6a-acf7-5d9c8760c610">Supplementary Figure 6.</label></span></h3><p id="el-p-81" data-ref="4d1b8f2a-fe63-430f-bd92-7d82ca8223e6" data-id="el-p-81" data-previous-break-after="avoid"><strong data-ref="1436dcb7-5986-4081-bfda-36725da0707e">a)</strong> Additional channels for the colony shown in <span class="fakelink linktofig" data-ref="2ea67597-2b42-4de9-aa7f-c5889337bdf4"><span class="fakelink-text" data-ref="91be5ac9-ed83-470a-8de2-e0f1a2e044cd">Fig. 5g</span><a class="fake-icon" href="#fig5" data-ref="9394d97b-1ccc-48ab-8027-112fbf4dd8ce"><span class="fakelink-icon" data-ref="2b434111-83a8-424c-9776-813bb6678d46">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. <strong data-ref="aab940d0-9883-4893-ad96-f48f38f6029d">b)</strong> FGF4 shRNA in ESI17 cells. <strong data-ref="b62b13d3-50cf-44a0-8299-7d9da8fe753f">c-d)</strong> FGF4 siRNA (c) and shRNA (d) in PGP1 cells. <strong data-ref="faf5ac10-92eb-4106-8c5e-21485be9a560">e-f)</strong> additional channels for <span class="fakelink linktofig" data-ref="ca2cb263-2dc3-4619-b64b-9cd8e42e657b"><span class="fakelink-text" data-ref="eaba70a7-5fd1-44d4-9ac9-9c8f38aa5e93">Fig. 5h-i</span><a class="fake-icon" href="#fig5" data-ref="85689851-7dc5-4465-b5e9-f16f51268da9"><span class="fakelink-icon" data-ref="dacf4e41-4e8e-4d93-9c6b-2d3f2ab55566">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. <strong data-ref="c13f64a5-e178-43e6-9a19-d7840632e19e">g</strong>) FGF17 shRNA in ESI17 cells. <strong data-ref="e07e1d45-6263-4948-a8b0-5dea3f588a6e">h-i)</strong> FGF17 siRNA (h) and shRNA (i) in PGP1 cells. <strong data-ref="086d162e-da1b-41ef-91de-704e15996cea">j)</strong> additional channels for the TBX6 ctrl and FGF17 shRNA in <span class="fakelink linktofig" data-ref="2c67f35b-aab8-45e5-8877-e71c3ffdf96a"><span class="fakelink-text" data-ref="6394d4d4-1bc4-4e5b-8d89-97eb43ecc2ff">Fig. 5j</span><a class="fake-icon" href="#fig5" data-ref="70dcbc07-ec10-47eb-b29c-76a316d7f599"><span class="fakelink-icon" data-ref="c264a217-d0e9-4ad4-8a45-22307e4ddef4">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. <strong data-ref="3b95b603-56ae-4bc7-b646-5eb8883ab447">k)</strong> FGF4 + FG17 shRNA double knockdown (stains from multiple colonies).</p></figcaption><img class="child-of-figure has-boundary imageonly portrait max-100" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_figS6.tif/full/max/0/default.jpg" alt="" id="el-img-21" data-ref="031583a7-93d1-4bcd-8961-9d484d2bc6e3" data-id="el-img-21" height="2592" width="2103" data-desired-width="99.57386363636364"></figure><div class="pagebreak" id="el-div-15" data-ref="b8e11b54-8ad6-416f-83f2-d010d174a5b6" data-id="el-div-15" data-break-before="page"></div></section><section id="acknowledgements" data-ref="b8b95a3c-1f5f-4725-b84f-0569e967272e" data-id="acknowledgements"><h1 class="heading-1" id="el-h1-8" data-ref="7bb5e5ce-d8fc-4f58-9fe6-89ca510d56ad" data-id="el-h1-8" data-break-after="avoid">Acknowledgements</h1><p id="el-p-82" data-ref="1cd211c4-9b7c-450c-8af0-309e84ff1731" data-id="el-p-82" data-previous-break-after="avoid">We thank Lila Solnica-Krezel, Blerta Stringa, Aryeh Warmflash, and Ben Allen for discussions. We thank Jared Toettcher for the dox-SOScat plasmid. Library prep and next-generation sequencing was carried out in the Advanced Genomics Core at the University of Michigan. This work was supported by the National Institute of General Medical Sciences (NIGMS R35GM138346), and the Branco Weiss Fellowship – Society in Science. KJ was partially supported by the Michigan Pioneer Postdoctoral Fellowship and the NIH F32 Ruth L. Kirschstein Postdoctoral National Research Service Award (5F32HD108980-02).</p></section><section id="additional-info" data-ref="ee68b80d-31d4-4ffa-abc6-803efa058c7f" data-id="additional-info"><h1 class="heading-1" id="el-h1-9" data-ref="12582c6c-50a6-470d-9cf7-57bbcdfe34e8" data-id="el-h1-9" data-break-after="avoid">Additional information</h1><section id="das" data-ref="48bf1056-a4e3-4b92-89fb-d0847c0ad9be" data-id="das" data-previous-break-after="avoid"><h2 class="heading-2" id="el-h2-21" data-ref="40d38778-870d-4b4e-85e3-a8dafe31b8be" data-id="el-h2-21" data-break-after="avoid" data-following="following-ff2c00ba-51a1-482d-9afa-f5a370dcfa50">Data availability</h2><p id="el-p-83" data-ref="f0bbf8b8-065a-4b4b-858d-72c69c443ff4" data-id="el-p-83" data-previous-break-after="avoid">Raw single cell RNA sequencing data generated in this study was deposited in GEO (GSE271604). Publicly available datasets used in this study can be accessed from GEO (GSE193007, GSE85066) and ArrayExpress (E-MTAB-6967, E-MTAB-9388). Raw image data are available upon request.</p></section><section id="s5" data-ref="c7348dab-bf46-4dcb-ba18-d621ad9b729b" data-id="s5"><h2 class="heading-2" id="el-h2-22" data-ref="87e19654-e373-4de1-a8d2-daa2b7f1dbc1" data-id="el-h2-22" data-break-after="avoid">Code availability</h2><p id="el-p-84" data-ref="e2f19cf7-2ad8-4f6f-95e6-5465a6116a54" data-id="el-p-84" data-previous-break-after="avoid">All code for data analysis and model simulations is available on <span class="fakelink linktoext" data-ref="b62827fb-6b54-42b1-8b49-358d322c2302"><span class="fakelink-text" data-ref="8de40699-5ce6-4b40-a214-00d3b2519166">https://github.com/idse/FGF</span><a class="fake-icon" href="https://github.com/idse/FGF" data-ref="71f99f19-14a4-491b-982d-59e934103ead"><span class="fakelink-icon" data-ref="18faa64b-3c8e-47a8-8647-58a49dd5fedd">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span></p></section><section id="funding" data-ref="612062ca-3802-4b2c-aa17-83eb6ba78246" data-id="funding"><h2 id="el-h2-23" data-ref="5683cbef-189c-4539-ac9b-782a6d38fa38" data-id="el-h2-23" data-break-after="avoid">Funding</h2><ul class="funding-list list-simple" id="el-ul-4" data-ref="6b164475-e553-4350-bd76-f832054694c4" data-id="el-ul-4" data-previous-break-after="avoid"><li class="funding-list-item" id="el-li-28" data-ref="6d82f4e0-c715-44d6-9641-fb67e23a8968" data-id="el-li-28"><p id="el-p-85" data-ref="5a29d322-682c-4860-9c46-b19784a1bd0d" data-id="el-p-85">National Institute of General Medical Sciences</p></li><li class="funding-list-item" id="el-li-29" data-ref="53a3853f-cb12-41a7-9e62-ebe65f124ac4" data-id="el-li-29"><p id="el-p-86" data-ref="969884bb-4e8b-4255-b87a-235110881e9d" data-id="el-p-86">the Branco Weiss Fellowship</p></li><li class="funding-list-item" id="el-li-30" data-ref="2d15795e-90d7-401b-b0be-6360d190888e" data-id="el-li-30"><p id="el-p-87" data-ref="5d40727b-807d-45ff-816a-6493df47309d" data-id="el-p-87">Michigan Pioneer Postdoctoral Fellowship</p></li><li class="funding-list-item" id="el-li-31" data-ref="7de9f23e-0350-4fb0-af75-091f68aa739d" data-id="el-li-31"><p id="el-p-88" data-ref="0f873584-bb2e-4336-af16-bcc33898d013" data-id="el-p-88">NIH F32 Ruth L. Kirschstein Postdoctoral National Research Service Award</p></li></ul></section><section id="orcids" data-ref="37ecf16f-8771-4e30-9dae-603bc6dbc8df" data-id="orcids"><h2 class="orcid-list__title" id="el-h2-24" data-ref="4e88da33-53be-4933-b741-9a09bd8b714f" data-id="el-h2-24" data-break-after="avoid">Author ORCID iDs</h2><ul class="orcid-list list-simple" id="el-ul-5" data-ref="84e8bd96-3341-4d7d-a99b-3d54b8a4864c" data-id="el-ul-5" data-previous-break-after="avoid"><li class="orcid-list-item" id="el-li-32" data-ref="1e9deba0-b230-4b55-b8b9-2af761dbfa81" data-id="el-li-32"><p id="el-p-89" data-ref="aa7a7a72-e2cb-445f-b6a8-b3a632faef74" data-id="el-p-89"><strong data-ref="1c6bf09a-f8ed-4919-aa63-99e3e39a8a2b">Jason Spence</strong>: <a href="http://orcid.org/0000-0001-7869-3992" data-ref="0029e75e-ca1e-4bc6-806d-2d9eae9a82b6">http://orcid.org/0000-0001-7869-3992</a></p></li></ul></section></section><section id="references" data-ref="7921c412-ef92-4490-b273-a1fbe0815a2b" data-id="references"><h1 class="heading-1" id="el-h1-10" data-ref="63cb8820-844f-497b-82ef-b2051268d3ca" data-id="el-h1-10" data-break-after="avoid">References</h1><ul class="reference-list" id="el-ul-6" data-ref="5d1d46dc-2209-4116-9f07-193d224fefb4" data-id="el-ul-6" data-previous-break-after="avoid"><li class="reference-list__item" id="c1" data-ref="7e4c950c-7904-4995-a2a8-7b5cdcd462c6" data-id="c1"><span class="reference__label" data-ref="7cd461f8-eb6e-458d-a106-b94ba048e313">1.</span><ol class="reference__authors_list" id="el-ol-3" data-ref="e715c6f3-2879-4e8d-a48d-eccfb16af2b8" data-id="el-ol-3"><li class="reference__author" id="el-li-34" data-ref="6e12b676-0ccb-450e-a431-5b2f238c5e49" data-id="el-li-34" data-item-num="1">Arnold S. J.</li><li class="reference__author" id="el-li-35" data-ref="b3da5191-a45d-4639-bf9c-7313cce17c4a" data-id="el-li-35" data-item-num="2">Robertson E. J</li></ol><span class="reference__authors_list_suffix" data-ref="989c011a-1765-4357-9367-49a92b92f15c">2009</span><span class="reference__title" data-ref="0900321d-92af-4913-ac78-d9565916788c">Making a commitment: cell lineage allocation and axis patterning in the early mouse embryo.</span><span class="reference__origin" data-ref="96c04b46-9e20-415d-b767-08a5042eb443"><i data-ref="e83fff9b-7aa4-4d10-bf8e-3cf3f279fd6c">Nat. Rev. Mol. Cell Biol</i> <strong data-ref="93ff7dce-810c-4030-b99e-61313b7867cf">10</strong>:91‑103</span><span class="reference__doi" data-ref="bbfff459-891c-47a8-ae0d-ef0d31c1aefc"></span></li><li class="reference-list__item" id="c2" data-ref="0c134f85-c115-4dab-916e-0e2d7e6e819d" data-id="c2"><span class="reference__label" data-ref="af9a7e61-b01a-42d0-8c79-292c20f51202">2.</span><ol class="reference__authors_list" id="el-ol-4" data-ref="fd1ce35f-fdcf-4e68-98ed-c36ae2e1149a" data-id="el-ol-4"><li class="reference__author" id="el-li-37" data-ref="cd3e1159-dd40-4487-9869-13ebb62a28d1" data-id="el-li-37" data-item-num="1">Martyn I.</li><li class="reference__author" id="el-li-38" data-ref="888e7172-9903-4288-a3e8-561f3bdbdc14" data-id="el-li-38" data-item-num="2">Kanno T. Y.</li><li class="reference__author" id="el-li-39" data-ref="033da909-3ecf-4096-bef4-630ddb57ac97" data-id="el-li-39" data-item-num="3">Ruzo A.</li><li class="reference__author" id="el-li-40" data-ref="06114454-bac3-4f53-b5af-900ae13454c3" data-id="el-li-40" data-item-num="4">Siggia E. D.</li><li class="reference__author" id="el-li-41" data-ref="a2c19721-11ff-4f32-b4b1-2891a77b5eb2" data-id="el-li-41" data-item-num="5">Brivanlou A. H</li></ol><span class="reference__authors_list_suffix" data-ref="d162dc71-f218-4818-a31e-5af5bf02a7d0">2018</span><span class="reference__title" data-ref="662c9e3f-e07c-4e36-92c0-2287ccda5b8b">Self-organization of a human organizer by combined Wnt and Nodal signalling.</span><span class="reference__origin" data-ref="1d66e2f8-4209-43bc-a831-b73254c0fb35"><i data-ref="42eec796-c959-4bf6-bc0e-377e7dd2419f">Nature</i> </span><span class="reference__doi" data-ref="cf22f177-bf60-48ef-9753-cd2b67d28348"><a href="https://doi.org/10.1038/s41586-018-0150-y" class="reference__doi_link" data-ref="cef7512f-4130-49b4-913a-32cf47087070">https://doi.org/10.1038/s41586-018-0150-y</a></span></li><li class="reference-list__item" id="c3" data-ref="ebf5d9f7-a019-4537-b026-4a8b5922c6b4" data-id="c3"><span class="reference__label" data-ref="018bfff3-0460-416c-bfce-fe6f36ad309c">3.</span><ol class="reference__authors_list" id="el-ol-5" data-ref="921e5cbe-b692-49d9-a112-509e531776ed" data-id="el-ol-5"><li class="reference__author" id="el-li-43" data-ref="113f81c1-ea48-4484-bb38-e941a96de7a6" data-id="el-li-43" data-item-num="1">Chhabra S.</li><li class="reference__author" id="el-li-44" data-ref="9edeb0d8-b8fa-4798-89b5-7909aca046da" data-id="el-li-44" data-item-num="2">Liu L.</li><li class="reference__author" id="el-li-45" data-ref="b15a0d7d-55ee-4587-8596-1e2aa237109c" data-id="el-li-45" data-item-num="3">Goh R.</li><li class="reference__author" id="el-li-46" data-ref="5682a207-7119-4572-9e5d-374d913b8320" data-id="el-li-46" data-item-num="4">Kong X.</li><li class="reference__author" id="el-li-47" data-ref="b5a9750e-b987-4800-a377-4ddf5609b9c1" data-id="el-li-47" data-item-num="5">Warmflash A</li></ol><span class="reference__authors_list_suffix" data-ref="209ffb2d-40d7-4c55-9abe-8b6c4a74d362">2019</span><span class="reference__title" data-ref="7d43c996-4a10-4216-9353-d25e56294d17">Dissecting the dynamics of signaling events in the BMP, WNT, and NODAL cascade during self-organized fate patterning in human gastruloids.</span><span class="reference__origin" data-ref="563207ec-3fa8-436d-8165-336adf08a71a"><i data-ref="2f71af90-d9ea-4c84-a69f-fc3a92497882">PLoS Biol</i> <strong data-ref="580844ba-1772-4e92-8604-5d16efde939c">17</strong>:e3000498</span><span class="reference__doi" data-ref="948a6e39-0998-48c3-bd3c-61a7fa540c52"></span></li><li class="reference-list__item" id="c4" data-ref="82afde16-b8bb-4fb8-b625-fc36b97c3502" data-id="c4"><span class="reference__label" data-ref="fa2530a9-4178-4e28-94cc-eabeb29cdf3a">4.</span><ol class="reference__authors_list" id="el-ol-6" data-ref="72df4fa4-f7b0-4160-96cc-aa9b8fde8d04" data-id="el-ol-6"><li class="reference__author" id="el-li-49" data-ref="7195ec6b-0aab-43fa-b971-9160a60e434e" data-id="el-li-49" data-item-num="1">Etoc F.</li><li class="reference__author" id="el-li-50" data-ref="7d7dc14f-ccab-42c6-80d0-6e0a9fe02468" data-id="el-li-50" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="5715be0a-ad36-4e04-8d6c-21cf9ec62948">2016</span><span class="reference__title" data-ref="c8e96529-c8a9-489f-a9f9-cf41d250441d">A Balance between Secreted Inhibitors and Edge Sensing Controls Gastruloid Self-Organization.</span><span class="reference__origin" data-ref="5829238d-0976-4a98-a622-d18664f34f36"><i data-ref="9b59913d-f042-46d6-8a09-24a50a06d434">Dev. Cell</i> <strong data-ref="dc404ac9-45c1-4483-8535-993a56c983d7">39</strong>:302‑315</span><span class="reference__doi" data-ref="2ebd1347-84a8-4b31-9aa1-5816b19d755c"></span></li><li class="reference-list__item" id="c5" data-ref="21478299-a5eb-489f-88df-769e4ff63847" data-id="c5"><span class="reference__label" data-ref="c7b78e89-4eed-4f9c-a155-102f64286809">5.</span><ol class="reference__authors_list" id="el-ol-7" data-ref="366036ad-03b0-4dbb-9cb9-ad20ad99ef61" data-id="el-ol-7"><li class="reference__author" id="el-li-52" data-ref="5c60a121-3d4d-433b-8fd0-7b15cd7a3669" data-id="el-li-52" data-item-num="1">Zhang Z.</li><li class="reference__author" id="el-li-53" data-ref="b8fdcb4c-85a2-4abb-b18d-b3927fb65ced" data-id="el-li-53" data-item-num="2">Zwick S.</li><li class="reference__author" id="el-li-54" data-ref="fd1eb8f5-eed8-4afe-879b-228986b1d0c5" data-id="el-li-54" data-item-num="3">Loew E.</li><li class="reference__author" id="el-li-55" data-ref="a292ffac-e688-42f1-974a-67ad7e3bf0bb" data-id="el-li-55" data-item-num="4">Grimley J. S.</li><li class="reference__author" id="el-li-56" data-ref="729d8909-f9a5-4df9-9fa6-770e50804867" data-id="el-li-56" data-item-num="5">Ramanathan S</li></ol><span class="reference__authors_list_suffix" data-ref="fb0aa1f5-0234-4bb9-9c1f-d0f98f07959e">2018</span><span class="reference__title" data-ref="9ad61b13-e581-482a-817c-e6b16be1f763">Embryo geometry drives formation of robust signaling gradients through receptor localization.</span><span class="reference__origin" data-ref="40650ba3-f241-4e69-81d6-e0f1824bed69"><i data-ref="6e420acf-cc1d-43a8-aae0-fb1ea2d87d64">bioRxiv</i> </span><span class="reference__doi" data-ref="6944fb09-8f3f-493d-93b1-60fd9ff6198d"><a href="https://doi.org/10.1101/491290" class="reference__doi_link" data-ref="3fe244a0-ddde-420b-8b92-6cf31873f9a1">https://doi.org/10.1101/491290</a></span></li><li class="reference-list__item" id="c6" data-ref="fc15e12a-730b-4b5f-9bdb-b3edaf63bc3c" data-id="c6"><span class="reference__label" data-ref="7c4794bb-0512-4986-8ecf-88699b7fc94d">6.</span><ol class="reference__authors_list" id="el-ol-8" data-ref="32d0cef5-4840-4773-a520-bde640649535" data-id="el-ol-8"><li class="reference__author" id="el-li-58" data-ref="4d61e830-082a-4a21-baa5-1489af9495e3" data-id="el-li-58" data-item-num="1">Heemskerk I.</li><li class="reference__author" id="el-li-59" data-ref="70109124-51a6-452b-a351-4c6faf0b39c9" data-id="el-li-59" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="53243e03-4906-4bdc-b386-f2b6f0895cad">2019</span><span class="reference__title" data-ref="263b5c6c-5fee-432a-9db5-bf1fdeb4664e">Rapid changes in morphogen concentration control self-organized patterning in human embryonic stem cells.</span><span class="reference__origin" data-ref="b35f70c3-aa7c-479a-ab78-711923e7a44e"><i data-ref="edcf2543-4a63-42f4-914e-c2b564b307ef">eLife</i> <strong data-ref="43bad3e7-a5a3-4c58-b19e-98b74d1235d2">8</strong></span><span class="reference__doi" data-ref="615c93d4-2e37-4d74-8ae5-04c2d0455c97"></span></li><li class="reference-list__item" id="c7" data-ref="c6384cdb-34da-4e20-918e-9ca292ed06b8" data-id="c7"><span class="reference__label" data-ref="2b82a30d-a09e-432a-b703-57012de6d65e">7.</span><ol class="reference__authors_list" id="el-ol-9" data-ref="c2c0a4e0-4b4e-4f71-91cb-666854e267f1" data-id="el-ol-9"><li class="reference__author" id="el-li-61" data-ref="560565fa-95dc-493a-ac63-6e70b06f9bf2" data-id="el-li-61" data-item-num="1">Massey J.</li><li class="reference__author" id="el-li-62" data-ref="7f675217-6dd6-4496-9c8c-dbb55ee77ed5" data-id="el-li-62" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="2a1f3f10-e566-458b-aac0-c70def92b104">2018</span><span class="reference__title" data-ref="2c457572-d8ae-434e-a89e-8439c906709c">WNT ligands stimulate transient signaling in human pluripotent cells and synergize with TGF-β ligands to stimulate sustained signaling during differentiation.</span><span class="reference__origin" data-ref="9009c567-9b1d-45b3-95c7-038552697f63"><i data-ref="fb551399-fbd5-4dd2-a0e0-bba0336136eb">bioRxiv</i> </span><span class="reference__doi" data-ref="14e28eb0-5058-4ed7-95e3-9a4467200ece"><a href="https://doi.org/10.1101/406306" class="reference__doi_link" data-ref="632a0d91-c8ad-4cab-8eb8-dbf04569c30e">https://doi.org/10.1101/406306</a></span></li><li class="reference-list__item" id="c8" data-ref="72794aea-bae3-4bbf-bd25-126298b65bbb" data-id="c8"><span class="reference__label" data-ref="68431f94-5d37-4908-876f-e5069d66c32e">8.</span><ol class="reference__authors_list" id="el-ol-10" data-ref="cb68c804-0a39-4905-bc9a-ca626c89927a" data-id="el-ol-10"><li class="reference__author" id="el-li-64" data-ref="96b277f3-5c09-4477-a400-e3b1fdad37bb" data-id="el-li-64" data-item-num="1">Yang R.</li><li class="reference__author" id="el-li-65" data-ref="d4444fc6-7a23-4e72-bb70-3c11a901478c" data-id="el-li-65" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="2e65d209-ef0f-4b99-b6b9-a893371cdf62">2021</span><span class="reference__title" data-ref="39adb9b3-a74c-4c5d-b6bc-e99317de0f51">Amnion signals are essential for mesoderm formation in primates.</span><span class="reference__origin" data-ref="4211e018-acc9-41d6-b892-81298f861f6f"><i data-ref="d235617f-604b-4dc1-aa39-88bc43f7d87f">Nat Commun</i> <strong data-ref="06ad1728-cbef-459d-9c9e-930adf5ca98a">12</strong>:5126</span><span class="reference__doi" data-ref="3fcdfa77-d7ee-4ffd-8d7f-1875b0a99f26"></span></li><li class="reference-list__item" id="c9" data-ref="6e912998-936b-4c84-b203-9272fc10620b" data-id="c9"><span class="reference__label" data-ref="2b8ba453-1549-4e7f-a515-179570ba0529">9.</span><ol class="reference__authors_list" id="el-ol-11" data-ref="42ec6e82-2f5a-45d6-b547-8ce533044d0f" data-id="el-ol-11"><li class="reference__author" id="el-li-67" data-ref="cff3e78e-89fc-4286-9ed9-82bb716c53aa" data-id="el-li-67" data-item-num="1">Ornitz D. M.</li><li class="reference__author" id="el-li-68" data-ref="c5f2a9ac-eb68-40a2-83d6-781810bd9bfe" data-id="el-li-68" data-item-num="2">Itoh N</li></ol><span class="reference__authors_list_suffix" data-ref="01b12630-166b-4478-8b8b-25e0d5561752">2015</span><span class="reference__title" data-ref="99f1ed7a-2aa9-4b36-baf9-535ad8948da4">The Fibroblast Growth Factor signaling pathway.</span><span class="reference__origin" data-ref="80e29382-ec6b-4952-87a3-dd574ef4c16f"><i data-ref="91f4a9a2-178d-419e-acd2-db621d1235df">Wiley Interdiscip Rev Dev Biol</i> <strong data-ref="edf86d37-9310-4801-aef8-0ba5e7b6820e">4</strong>:215‑266</span><span class="reference__doi" data-ref="58f9aefb-0e4a-47a0-91ec-e93a86f27f66"></span></li><li class="reference-list__item" id="c10" data-ref="4a67d542-b81a-4bfc-bf59-69d656968d8a" data-id="c10"><span class="reference__label" data-ref="c6731a58-034c-4916-9a6a-02fbe71dcfcb">10.</span><ol class="reference__authors_list" id="el-ol-12" data-ref="4eaa2556-05dd-4bd0-b293-6d19f077fe9d" data-id="el-ol-12"><li class="reference__author" id="el-li-70" data-ref="e6aa80b6-acc9-4bbe-9df6-90c75080c07b" data-id="el-li-70" data-item-num="1">LaBonne C.</li><li class="reference__author" id="el-li-71" data-ref="476c0c82-3346-4ffc-b8fb-eded39b11c0e" data-id="el-li-71" data-item-num="2">Whitman M</li></ol><span class="reference__authors_list_suffix" data-ref="6cd69e47-6350-4964-86bf-bbdec9106ab3">1994</span><span class="reference__title" data-ref="2af35f56-527d-4c95-847c-7c9e5d55edba">Mesoderm induction by activin requires FGF-mediated intracellular signals.</span><span class="reference__origin" data-ref="90ad3831-aaed-4e14-a3ed-359821ff6a5e"><i data-ref="f69706d7-3ff0-4d2f-880a-73a0160840cb">Development</i> <strong data-ref="b2cfbc8d-819b-42da-9ef5-849dcaf2aad7">120</strong>:463‑472</span><span class="reference__doi" data-ref="93e1405a-c518-4a57-b94c-3bac5c3bbfb9"></span></li><li class="reference-list__item" id="c11" data-ref="7efebb3e-d194-4aab-8c7c-76b466495a90" data-id="c11"><span class="reference__label" data-ref="073b028d-c3d0-4824-bed6-6f1d79f0e6f0">11.</span><ol class="reference__authors_list" id="el-ol-13" data-ref="24f7090a-eb56-4441-89a2-e33f3e13b092" data-id="el-ol-13"><li class="reference__author" id="el-li-73" data-ref="140181f6-d046-4ca2-a548-b3a82701e1ac" data-id="el-li-73" data-item-num="1">Krens S. F. G.</li><li class="reference__author" id="el-li-74" data-ref="e683a903-48cc-4872-a35e-0f1b33945860" data-id="el-li-74" data-item-num="2">Corredor-Adámez M.</li><li class="reference__author" id="el-li-75" data-ref="c8607196-f7e5-4f72-a67b-bee71ad9e3c9" data-id="el-li-75" data-item-num="3">He S.</li><li class="reference__author" id="el-li-76" data-ref="69e261ae-47f0-4d38-a0de-6518b08c34af" data-id="el-li-76" data-item-num="4">Snaar-Jagalska B. E.</li><li class="reference__author" id="el-li-77" data-ref="c8e9fc73-f5e2-48ad-9865-9a7b5d80eb23" data-id="el-li-77" data-item-num="5">Spaink H. P</li></ol><span class="reference__authors_list_suffix" data-ref="3db1dca0-f41f-4f73-94b7-ee37e40bd569">2008</span><span class="reference__title" data-ref="878cbfa2-6f9f-48d9-b505-0f4643a10eb8">ERK1 and ERK2 MAPK are key regulators of distinct gene sets in zebrafish embryogenesis.</span><span class="reference__origin" data-ref="307312a8-c037-4139-99e1-f89fc08cbd25"><i data-ref="da59ff61-c5f4-4718-ad91-2f2a2051575d">BMC Genomics</i> <strong data-ref="d70b3be4-3055-43b7-b49a-8780a8e69a8a">9</strong>:196</span><span class="reference__doi" data-ref="47733132-b168-4526-87bf-30c50ad32f73"></span></li><li class="reference-list__item" id="c12" data-ref="c3877ed8-893e-4976-9767-4c5795cf657c" data-id="c12"><span class="reference__label" data-ref="825652a9-510d-4c3b-82aa-8402806f4270">12.</span><ol class="reference__authors_list" id="el-ol-14" data-ref="b5dffd93-3394-4094-8fad-f44124572a78" data-id="el-ol-14"><li class="reference__author" id="el-li-79" data-ref="7f87270e-c8d1-4aac-8892-6060faa57e7c" data-id="el-li-79" data-item-num="1">Yao Y.</li><li class="reference__author" id="el-li-80" data-ref="ccf39de1-8538-4385-943c-11cad98401e5" data-id="el-li-80" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="d39fc22e-ca2b-42aa-aa08-f48e26a35dc9">2003</span><span class="reference__title" data-ref="8fd0fa21-ea6c-4a41-a2ae-6bc1a8d63a32">Extracellular signal-regulated kinase 2 is necessary for mesoderm differentiation.</span><span class="reference__origin" data-ref="4870699c-c883-4ab1-9897-338efb2ed0cf"><i data-ref="2fb9a737-950d-4ed1-aedb-1e3b3d692bab">Proc Natl Acad Sci U S A</i> <strong data-ref="a31f89a0-2b39-4d83-a145-3ae3ad31903e">100</strong>:12759‑12764</span><span class="reference__doi" data-ref="32310008-8501-48ad-9528-26b6e8966d24"></span></li><li class="reference-list__item" id="c13" data-ref="c802d671-61a3-4a1a-954f-ee308aae1200" data-id="c13"><span class="reference__label" data-ref="16aa5276-b477-4b8f-a04e-399d8ebfb1f9">13.</span><ol class="reference__authors_list" id="el-ol-15" data-ref="e83bc63a-07c8-484d-842f-d796dbf4fcba" data-id="el-ol-15"><li class="reference__author" id="el-li-82" data-ref="7898a237-1e8e-44a4-8354-579f9e8cac75" data-id="el-li-82" data-item-num="1">Mitrani E.</li><li class="reference__author" id="el-li-83" data-ref="e71dda50-dbd7-4550-8cda-7e2b8fee8c13" data-id="el-li-83" data-item-num="2">Gruenbaum Y.</li><li class="reference__author" id="el-li-84" data-ref="8859dc9e-cf9e-4970-a502-d5050d1877d8" data-id="el-li-84" data-item-num="3">Shohat H.</li><li class="reference__author" id="el-li-85" data-ref="b5ecf1fd-7cf6-4598-a18d-3d28ddcdec7f" data-id="el-li-85" data-item-num="4">Ziv T</li></ol><span class="reference__authors_list_suffix" data-ref="9c46c236-20a6-4b12-ab0e-2929b39e20e7">1990</span><span class="reference__title" data-ref="b8f84daa-9b70-4cb5-b5fb-c721a43b029d">Fibroblast growth factor during mesoderm induction in the early chick embryo.</span><span class="reference__origin" data-ref="db02e0d4-bc0d-4c9a-8a55-8a9bf8fa570c"><i data-ref="9b13affe-3f8f-40dc-82a5-191dfeb0450a">Development</i> <strong data-ref="45e55612-a13f-4e39-8d9f-8a23a905ba78">109</strong>:387‑393</span><span class="reference__doi" data-ref="eb362f73-3afb-47dd-a023-7cf760c2ee9e"></span></li><li class="reference-list__item" id="c14" data-ref="8b4c7bd3-0c6a-470d-9253-4a859e7c2f3d" data-id="c14"><span class="reference__label" data-ref="5bcc5bf3-e3be-4dc7-9526-cc5cb4fb5e32">14.</span><ol class="reference__authors_list" id="el-ol-16" data-ref="d1a4434d-50a0-41d1-80e4-80e7243a6c0f" data-id="el-ol-16"><li class="reference__author" id="el-li-87" data-ref="570a487f-3fb1-48ac-94f5-eed094438650" data-id="el-li-87" data-item-num="1">Corson L. B.</li><li class="reference__author" id="el-li-88" data-ref="5741c17d-d5d7-441f-b103-47a479298c64" data-id="el-li-88" data-item-num="2">Yamanaka Y.</li><li class="reference__author" id="el-li-89" data-ref="921fb801-4c48-4200-9311-fc6c82ee15e0" data-id="el-li-89" data-item-num="3">Lai K.-M. V.</li><li class="reference__author" id="el-li-90" data-ref="b5c40c53-ac40-451f-a1f4-c33bbf5ebd96" data-id="el-li-90" data-item-num="4">Rossant J</li></ol><span class="reference__authors_list_suffix" data-ref="91b64e5c-1679-48d5-a918-aece995a6b47">2003</span><span class="reference__title" data-ref="47cbc3e4-b2e9-4f7f-918b-d6fb13c04aad">Spatial and temporal patterns of ERK signaling during mouse embryogenesis.</span><span class="reference__origin" data-ref="2f96a720-febc-4231-a805-696ada823050"><i data-ref="9c3a2959-2e4d-4596-ad26-a43a707e3f4b">Development</i> <strong data-ref="be9259db-caf4-4157-8b26-2959dae1b0d7">130</strong>:4527‑4537</span><span class="reference__doi" data-ref="0241b369-252a-4ca5-a41c-fd5de5d308da"></span></li><li class="reference-list__item" id="c15" data-ref="eb148bfc-86cc-4497-a9bf-dc81298f00c6" data-id="c15"><span class="reference__label" data-ref="10573e53-b221-4b6d-949d-249acb8a46f3">15.</span><ol class="reference__authors_list" id="el-ol-17" data-ref="4dae8e72-a0a3-4f47-8b56-9770e33408c2" data-id="el-ol-17"><li class="reference__author" id="el-li-92" data-ref="4105fd57-5445-477a-a733-95b4c51c2fa4" data-id="el-li-92" data-item-num="1">Morgani S. M.</li><li class="reference__author" id="el-li-93" data-ref="b5265cdb-9ecf-45c8-9d00-ee6c636e0ae1" data-id="el-li-93" data-item-num="2">Hadjantonakis A.-K.</li></ol><span class="reference__authors_list_suffix" data-ref="de950937-9671-4b08-bf39-c0fb5fce5d11">2020</span><span class="reference__title" data-ref="4434a759-9e3e-46c7-93ac-e7c06af92ef6">Signaling regulation during gastrulation: Insights from mouse embryos and <em data-ref="007b0334-30a2-470a-9e47-9a9e51b57257">in vitro</em> systems.</span><span class="reference__origin" data-ref="5da663a0-2ca3-4b30-a1e2-298d4f13920c">In: <ol class="reference__editors_list" id="el-ol-18" data-ref="3e54da36-5af7-425a-b70e-69389a85e908" data-id="el-ol-18"><li class="reference__editor" id="el-li-94" data-ref="cfd6a036-29ba-4d1c-86da-54c39dc4ff93" data-id="el-li-94" data-item-num="1">Small S.</li><li class="reference__editor" id="el-li-95" data-ref="02a22912-baf5-4e4e-b62f-16643202117f" data-id="el-li-95" data-item-num="2">Briscoe J.</li></ol> (Eds). <i data-ref="e3b4dd65-7ad3-4d82-a821-63fdb0bd72f2">Current Topics in Developmental Biology</i> <strong data-ref="af03097e-0a28-4907-8fb2-a0a797d9023c">137</strong> Academic Press. pp.&nbsp;391‑431</span><span class="reference__doi" data-ref="c7666ac2-185d-4f97-9356-901464c9069f"></span></li><li class="reference-list__item" id="c16" data-ref="2e8f94c6-f1e3-44aa-ac10-c37c6fc3df30" data-id="c16"><span class="reference__label" data-ref="b428f69e-e06b-4986-9cb2-7fe68e5e4ae3">16.</span><ol class="reference__authors_list" id="el-ol-19" data-ref="32e48447-c49d-4729-a796-57f9b3c421fb" data-id="el-ol-19"><li class="reference__author" id="el-li-97" data-ref="264e2718-faf1-4ab2-b7d6-7e55299a19c0" data-id="el-li-97" data-item-num="1">De Simone A.</li><li class="reference__author" id="el-li-98" data-ref="4d985f5e-d349-47d1-93e7-a607b1e8358f" data-id="el-li-98" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="596a18bb-d073-4076-b6b2-5a50ab4e2dff">2021</span><span class="reference__title" data-ref="fa2d98f4-da01-40fb-b067-4246e84db701">Control of osteoblast regeneration by a train of Erk activity waves.</span><span class="reference__origin" data-ref="0409dee1-a7fd-43a4-99a5-2b97f6f6946f"><i data-ref="e1fc0418-08d4-4bed-a44c-318f1cfe46d6">Nature</i> <strong data-ref="b75c7b96-6b87-41a8-a88b-f981deff6f43">590</strong>:129‑133</span><span class="reference__doi" data-ref="783bcbac-1e84-4ec7-8a32-db637c0771ef"></span></li><li class="reference-list__item" id="c17" data-ref="3d96b9e9-165d-4131-b4f2-cd55895806d7" data-id="c17"><span class="reference__label" data-ref="8cec99a6-446b-486a-9ee5-0faf67321212">17.</span><ol class="reference__authors_list" id="el-ol-20" data-ref="c1b07be1-8b5e-4d0f-83ac-a6c48da8be2c" data-id="el-ol-20"><li class="reference__author" id="el-li-100" data-ref="a03cc5af-2393-4710-aa73-a9a37114f75d" data-id="el-li-100" data-item-num="1">Marshall C. J</li></ol><span class="reference__authors_list_suffix" data-ref="b5d7449e-851d-4c65-aef3-ff34d39cbae3">1995</span><span class="reference__title" data-ref="f518fa7f-8cad-427f-9e5d-89f003efbe29">Specificity of receptor tyrosine kinase signaling: transient versus sustained extracellular signal-regulated kinase activation.</span><span class="reference__origin" data-ref="e74adc20-c706-4a83-aa28-f1cc581cc4f8"><i data-ref="daa4bffa-1396-4538-b95c-797c818e527e">Cell</i> <strong data-ref="50116503-6e34-460a-9b87-c1c4d426da1a">80</strong>:179‑185</span><span class="reference__doi" data-ref="230461c6-70a2-417c-aa33-b8c49f89fe0f"></span></li><li class="reference-list__item" id="c18" data-ref="8dcc1e7d-f52e-4647-ab53-1190691e5e0e" data-id="c18"><span class="reference__label" data-ref="42599176-5e2e-450b-b1b8-afff626fe1b4">18.</span><ol class="reference__authors_list" id="el-ol-21" data-ref="1f7dffe5-8f1b-46b2-9832-83b0dc333319" data-id="el-ol-21"><li class="reference__author" id="el-li-102" data-ref="a0337997-3b46-4f74-8183-e2f58a0db1f2" data-id="el-li-102" data-item-num="1">Ender P.</li><li class="reference__author" id="el-li-103" data-ref="3a22857d-7e2f-4f0d-b2ba-47f7e0985e82" data-id="el-li-103" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="0f57590f-fa3d-4aae-856f-fd6d3a2819b2">2022</span><span class="reference__title" data-ref="c7f817fc-5aa2-4a1e-8ba3-8af1f9007657">Spatiotemporal control of ERK pulse frequency coordinates fate decisions during mammary acinar morphogenesis.</span><span class="reference__origin" data-ref="0df07ddb-71a1-4125-ab5d-eb9ca20837fb"><i data-ref="6b73a7a4-fd1f-479b-b5b3-7dff9dc1edae">Dev Cell</i> <strong data-ref="d80f87e9-af80-4e37-9f38-bdd0d5fb4b67">57</strong>:2153‑2167.e6</span><span class="reference__doi" data-ref="abc13b67-2814-433b-a76e-144c3cc8e84e"></span></li><li class="reference-list__item" id="c19" data-ref="51199e54-b1e3-4a84-82bc-8962cb79dc1a" data-id="c19"><span class="reference__label" data-ref="f07249e2-df4d-48c4-858a-b409d2f40fa7">19.</span><ol class="reference__authors_list" id="el-ol-22" data-ref="f73d32b8-1b47-47c0-b438-141c74d22ebe" data-id="el-ol-22"><li class="reference__author" id="el-li-105" data-ref="52c891d9-447e-40ce-ae72-0c6f33532eb0" data-id="el-li-105" data-item-num="1">Aoki K.</li><li class="reference__author" id="el-li-106" data-ref="8853dc6b-7f69-47de-9a91-c4744b67f7e5" data-id="el-li-106" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="2117c2dc-cd33-4c86-ad43-658c74cc06bf">2017</span><span class="reference__title" data-ref="2bd8f2f5-89fc-45d3-9123-b9b732bc6ae7">Propagating Wave of ERK Activation Orients Collective Cell Migration.</span><span class="reference__origin" data-ref="c2816bc1-6a4e-4f33-81ab-b91be4b77d6d"><i data-ref="ec116793-586e-4615-bb80-0c056a736878">Dev Cell</i> <strong data-ref="abbb17e3-97be-44fc-8592-0bfa17738480">43</strong>:305‑317.e5</span><span class="reference__doi" data-ref="79d3f184-9ead-4cc4-8caf-8711f398691c"></span></li><li class="reference-list__item" id="c20" data-ref="b81d72ea-9e7d-4fbe-bbaa-f209d45648aa" data-id="c20"><span class="reference__label" data-ref="34d78af0-0fb3-45f6-abcb-50833c7d4971">20.</span><ol class="reference__authors_list" id="el-ol-23" data-ref="bc36d13d-078f-417e-8666-a856941795ba" data-id="el-ol-23"><li class="reference__author" id="el-li-108" data-ref="8821067c-56a9-44e4-8290-a8ddd1ab5965" data-id="el-li-108" data-item-num="1">Ciruna B.</li><li class="reference__author" id="el-li-109" data-ref="749c2e8f-a945-4b83-ae0a-1738b366eda5" data-id="el-li-109" data-item-num="2">Rossant J</li></ol><span class="reference__authors_list_suffix" data-ref="2448a15a-8e1c-4fbe-b597-a3579f311743">2001</span><span class="reference__title" data-ref="2eec85b6-abc9-4216-9cd1-b23b1b702ce2">FGF Signaling Regulates Mesoderm Cell Fate Specification and Morphogenetic Movement at the Primitive Streak.</span><span class="reference__origin" data-ref="0ce44839-9b1b-4a8a-aade-0b08ac3c7118"><i data-ref="13b6a0b9-505c-4e18-a76b-478eeaa04379">Dev. Cell</i> <strong data-ref="40581f5f-6f6e-428e-8ce7-da39a45cf593">1</strong>:37‑49</span><span class="reference__doi" data-ref="fbc255cc-31f9-44e1-889a-f39d2f5d2ea3"></span></li><li class="reference-list__item" id="c21" data-ref="b1307823-cccf-4c37-9f3e-47d90d35aaf1" data-id="c21"><span class="reference__label" data-ref="5c3c3375-0971-4414-bea3-88854d148a7b">21.</span><ol class="reference__authors_list" id="el-ol-24" data-ref="4672c433-a43c-4dc2-8065-75e6547cf117" data-id="el-ol-24"><li class="reference__author" id="el-li-111" data-ref="3638dbc7-c519-4844-a107-6490a160b985" data-id="el-li-111" data-item-num="1">Sun X.</li><li class="reference__author" id="el-li-112" data-ref="27c99f2e-b209-4fdd-bb0e-49b1bb98890d" data-id="el-li-112" data-item-num="2">Meyers E. N.</li><li class="reference__author" id="el-li-113" data-ref="d71cc8c4-6ba0-4211-908c-6d1eae7c48de" data-id="el-li-113" data-item-num="3">Lewandoski M.</li><li class="reference__author" id="el-li-114" data-ref="3babe324-9964-43c2-8a5c-1fa7890d8fe8" data-id="el-li-114" data-item-num="4">Martin G. R</li></ol><span class="reference__authors_list_suffix" data-ref="a0662fee-96cc-479e-8771-ae3175c3f254">1999</span><span class="reference__title" data-ref="a3c7a52c-7220-4e55-b818-3fbd1a3f428e">Targeted disruption of Fgf8 causes failure of cell migration in the gastrulating mouse embryo.</span><span class="reference__origin" data-ref="75662bc9-4b5b-45df-81d2-39b4e0e13497"><i data-ref="852cd638-929a-4127-a897-9496f40bf543">Genes Dev</i> <strong data-ref="1227da17-af5c-45ca-b5e4-6c3e510944fc">13</strong>:1834‑1846</span><span class="reference__doi" data-ref="9fc88b93-d136-4a17-8361-27d915c36586"></span></li><li class="reference-list__item" id="c22" data-ref="119bf123-9f0c-4c95-90f1-3575403eccf5" data-id="c22"><span class="reference__label" data-ref="97d00574-ba3a-4771-8606-131d4fef0a02">22.</span><ol class="reference__authors_list" id="el-ol-25" data-ref="a8623a7a-4950-4a01-a4c8-e604416acbc5" data-id="el-ol-25"><li class="reference__author" id="el-li-116" data-ref="6abeda82-8c26-45dc-9bf7-21603b638adb" data-id="el-li-116" data-item-num="1">Fletcher R. B.</li><li class="reference__author" id="el-li-117" data-ref="77961058-b050-4d7b-8d11-75a9aa602221" data-id="el-li-117" data-item-num="2">Harland R. M</li></ol><span class="reference__authors_list_suffix" data-ref="e9b34eae-1ac3-46e3-afb7-97bb95ffe3c3">2008</span><span class="reference__title" data-ref="3194abc3-6213-4aa2-8d1d-a6d6bf63b440">The role of FGF signaling in the establishment and maintenance of mesodermal gene expression in Xenopus.</span><span class="reference__origin" data-ref="a1561ab0-cd2d-4c0f-a31d-61ede31dd3a8"><i data-ref="26a75d5f-ac7a-4475-bc15-d56cd8b1d65f">Dev Dyn</i> <strong data-ref="ab5c4318-c546-4b4c-ba68-1952d2b607ee">237</strong>:1243‑1254</span><span class="reference__doi" data-ref="bbdb15e1-ce58-4e2b-8238-34680b0e4727"></span></li><li class="reference-list__item" id="c23" data-ref="a46ee167-d875-4a4b-a100-3f1d5e121477" data-id="c23"><span class="reference__label" data-ref="f603e1fa-c0b8-4ba3-a027-a46c2309d8f0">23.</span><ol class="reference__authors_list" id="el-ol-26" data-ref="8e8015b8-a910-485e-9ba1-570fef497d8f" data-id="el-ol-26"><li class="reference__author" id="el-li-119" data-ref="ee3f776b-1e97-48f9-b9a2-b6afa2873e53" data-id="el-li-119" data-item-num="1">Yang X.</li><li class="reference__author" id="el-li-120" data-ref="c55b5f27-aef4-49f1-9ebd-e9917b2e9ae7" data-id="el-li-120" data-item-num="2">Dormann D.</li><li class="reference__author" id="el-li-121" data-ref="d3d54737-ce50-4f87-b321-60898411ad0d" data-id="el-li-121" data-item-num="3">Münsterberg A. E.</li><li class="reference__author" id="el-li-122" data-ref="7e80ed14-88b3-4a03-858e-de9ba320a371" data-id="el-li-122" data-item-num="4">Weijer C. J</li></ol><span class="reference__authors_list_suffix" data-ref="92d1a8ae-b26e-4dbd-ae14-68f53ab1329b">2002</span><span class="reference__title" data-ref="bd72a40c-57d5-4e74-9432-48e22608d152">Cell Movement Patterns during Gastrulation in the Chick Are Controlled by Positive and Negative Chemotaxis Mediated by FGF4 and FGF8.</span><span class="reference__origin" data-ref="d8d2e51b-5f52-4ee4-8df0-fc7ded8c8907"><i data-ref="eb43d00d-b51a-49d6-b75a-740cd9bb6061">Developmental Cell</i> <strong data-ref="359d6ba7-81ce-4f24-a820-b3045c6823db">3</strong>:425‑437</span><span class="reference__doi" data-ref="e838e0e2-41b2-4955-af36-1022b122f8ca"></span></li><li class="reference-list__item" id="c24" data-ref="4779889e-1fd7-4b7a-953c-df4d0c35301b" data-id="c24"><span class="reference__label" data-ref="1cf1298b-f7eb-4258-bd91-f69f37677bcb">24.</span><ol class="reference__authors_list" id="el-ol-27" data-ref="869c25e8-e55d-4957-8c9f-91afd5be7d56" data-id="el-ol-27"><li class="reference__author" id="el-li-124" data-ref="64419559-f770-4c12-9a9f-554627aac660" data-id="el-li-124" data-item-num="1">Fürthauer M.</li><li class="reference__author" id="el-li-125" data-ref="1782206c-7c6d-4f25-941a-3af23cc8eddb" data-id="el-li-125" data-item-num="2">Reifers F.</li><li class="reference__author" id="el-li-126" data-ref="1d0c5b95-8c6a-4175-a14f-052b6465d573" data-id="el-li-126" data-item-num="3">Brand M.</li><li class="reference__author" id="el-li-127" data-ref="9c00e033-f5e5-4b1d-96a2-eeae88cafcab" data-id="el-li-127" data-item-num="4">Thisse B.</li><li class="reference__author" id="el-li-128" data-ref="0e4c5fef-b741-49b7-afdc-54194ec1fbcb" data-id="el-li-128" data-item-num="5">Thisse C</li></ol><span class="reference__authors_list_suffix" data-ref="14445830-94c3-4873-86a1-79ed6aabe4b9">2001</span><span class="reference__title" data-ref="92378523-a06b-423d-a9fe-78c5ea0cf7d9">sprouty4 acts in vivo as a feedback-induced antagonist of FGF signaling in zebrafish.</span><span class="reference__origin" data-ref="fc00f0cf-6211-44d4-91b7-710fdc4d807e"><i data-ref="fb1f4110-b66f-4372-923f-05a7b585decc">Development</i> <strong data-ref="90817629-3a3c-4f3a-b6a1-003b73d685a4">128</strong>:2175‑2186</span><span class="reference__doi" data-ref="be272959-1f48-48b5-abdd-5bc41acd9a12"></span></li><li class="reference-list__item" id="c25" data-ref="059d8044-25f1-491e-a39f-c8828ab524b2" data-id="c25"><span class="reference__label" data-ref="a22d1740-1c68-488b-bfea-fee167c6c2ae">25.</span><ol class="reference__authors_list" id="el-ol-28" data-ref="6ddf4821-2adc-4dc9-a3c0-1cf5b57fca77" data-id="el-ol-28"><li class="reference__author" id="el-li-130" data-ref="2ebfe4b7-36dd-4ad2-a76a-ee2261c8f648" data-id="el-li-130" data-item-num="1">Tyser R. C. V.</li><li class="reference__author" id="el-li-131" data-ref="69337f31-9784-4910-9606-f1416041439d" data-id="el-li-131" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="39209166-7334-48d9-a344-5a58521c32c7">2021</span><span class="reference__title" data-ref="c5056dcf-0f3d-4773-98ee-5d9dd174be91">Single-cell transcriptomic characterization of a gastrulating human embryo.</span><span class="reference__origin" data-ref="8f933b35-ae63-43ed-95d4-b395300e9952"><i data-ref="4aaf10d6-72c8-404b-81cc-f5e507680c98">Nature</i> <strong data-ref="5a96da51-1a51-404b-8bce-db8c48f19275">600</strong>:285‑289</span><span class="reference__doi" data-ref="740f030d-375c-4bca-9187-8e8500e3033c"></span></li><li class="reference-list__item" id="c26" data-ref="2b04f35f-70d8-4b51-8f1f-c249a42b5862" data-id="c26"><span class="reference__label" data-ref="53e25898-2f7b-4e26-96f3-b73fec39f28e">26.</span><ol class="reference__authors_list" id="el-ol-29" data-ref="e514dffb-ccba-46e7-9060-25ff4186abdd" data-id="el-ol-29"><li class="reference__author" id="el-li-133" data-ref="53ce7101-7860-4e73-8a44-37f1b75eeaa2" data-id="el-li-133" data-item-num="1">Zhai J.</li><li class="reference__author" id="el-li-134" data-ref="458af462-17e6-4880-b215-68cb43c0b7c5" data-id="el-li-134" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="c60efb41-5efa-4b94-b7af-ca1d619f47d5">2022</span><span class="reference__title" data-ref="e14778eb-c44e-47ad-876f-03d4a2f2e4ed">Primate gastrulation and early organogenesis at single-cell resolution.</span><span class="reference__origin" data-ref="01ed2478-61a3-46ce-ab0b-d5a124254fe3"><i data-ref="7840e78a-4c05-4065-a4d8-3ec022220998">Nature</i> <strong data-ref="a109b31e-7784-4313-b003-7803d25d4d29">612</strong>:732‑738</span><span class="reference__doi" data-ref="cc256f67-246c-485d-aead-7f376321b418"></span></li><li class="reference-list__item" id="c27" data-ref="e57b9290-9ab2-4a01-85e2-b82252d7e8da" data-id="c27"><span class="reference__label" data-ref="e0f8d35a-577b-4f2a-a78e-b5dafde108b3">27.</span><ol class="reference__authors_list" id="el-ol-30" data-ref="ff72386c-e8fe-48d0-a00e-4ee34e5f178f" data-id="el-ol-30"><li class="reference__author" id="el-li-136" data-ref="d5a21d6b-689a-445d-96cf-a75cc7137628" data-id="el-li-136" data-item-num="1">Deng C. X.</li><li class="reference__author" id="el-li-137" data-ref="74685d55-7626-47dc-a87e-d647c0fe3156" data-id="el-li-137" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="23695077-6f13-4e5f-8cc8-7d79b9b3f742">1994</span><span class="reference__title" data-ref="4c6ecb83-6e0b-48c9-b1e5-8dde620cd8b5">Murine FGFR-1 is required for early postimplantation growth and axial organization.</span><span class="reference__origin" data-ref="f340fee1-e439-4800-a587-36f036d120dd"><i data-ref="40eadde2-f169-4574-a81f-9aa947957730">Genes Dev</i> <strong data-ref="e3d8bee1-121d-4dac-bfed-f34eb9e513cb">8</strong>:3045‑3057</span><span class="reference__doi" data-ref="b46f3fd8-a207-4441-909b-f426a1ed5c94"></span></li><li class="reference-list__item" id="c28" data-ref="45f4198b-5a53-4333-ade3-fc648a60934a" data-id="c28"><span class="reference__label" data-ref="75421208-887c-4244-84ce-f8aec57944d7">28.</span><ol class="reference__authors_list" id="el-ol-31" data-ref="5096c0b8-0a3a-43a0-a273-1ec06cf75d5f" data-id="el-ol-31"><li class="reference__author" id="el-li-139" data-ref="e9062ec4-87fe-417a-afd7-8976f3a10a02" data-id="el-li-139" data-item-num="1">Yamaguchi T. P.</li><li class="reference__author" id="el-li-140" data-ref="4242ad5e-ddc7-4152-87e1-961892af009d" data-id="el-li-140" data-item-num="2">Harpal K.</li><li class="reference__author" id="el-li-141" data-ref="55b0cff2-0045-43b2-a1f9-7b1360f328be" data-id="el-li-141" data-item-num="3">Henkemeyer M.</li><li class="reference__author" id="el-li-142" data-ref="fc1566e4-5b58-4bc7-935b-a47a4904ef04" data-id="el-li-142" data-item-num="4">Rossant J.</li></ol><span class="reference__authors_list_suffix" data-ref="47daeab7-bb63-436a-ae07-986d391bb545">1994</span><span class="reference__title" data-ref="14c9f475-8df9-4f70-a47f-27356ffbbcd4">fgfr-1 is required for embryonic growth and mesodermal patterning during mouse gastrulation.</span><span class="reference__origin" data-ref="998ef39b-c64e-4ca3-ac46-27dfaaaa1b69"><i data-ref="5e630b3f-f96b-4e69-b8a5-d3d363e02b44">Genes Dev</i> <strong data-ref="606d06d7-3d4f-4005-b514-01196fa0897e">8</strong>:3032‑3044</span><span class="reference__doi" data-ref="8fd477ac-e2ba-421c-999f-77e7bf1b69d2"></span></li><li class="reference-list__item" id="c29" data-ref="5df9000c-ecd7-47cb-a6d0-1238ee26f35c" data-id="c29"><span class="reference__label" data-ref="4f3a8a6b-afaa-4aa5-a28e-dda02db6bf61">29.</span><ol class="reference__authors_list" id="el-ol-32" data-ref="49cc2947-7490-4b92-8db1-2a0b07887c28" data-id="el-ol-32"><li class="reference__author" id="el-li-144" data-ref="02637b08-3442-41cb-b616-38ffdfb37acf" data-id="el-li-144" data-item-num="1">Ciruna B. G.</li><li class="reference__author" id="el-li-145" data-ref="52315462-2720-449a-9077-945d4399a207" data-id="el-li-145" data-item-num="2">Schwartz L.</li><li class="reference__author" id="el-li-146" data-ref="511b55ae-3728-4dd5-9c4b-d2b18f3b4c84" data-id="el-li-146" data-item-num="3">Harpal K.</li><li class="reference__author" id="el-li-147" data-ref="78fe05f7-41e7-469d-957b-6212adc0d407" data-id="el-li-147" data-item-num="4">Yamaguchi T. P.</li><li class="reference__author" id="el-li-148" data-ref="086322dd-dee9-41e2-846d-69a10655b605" data-id="el-li-148" data-item-num="5">Rossant J</li></ol><span class="reference__authors_list_suffix" data-ref="d7ededf8-7476-44a6-901e-ba5fe7e5b4a0">1997</span><span class="reference__title" data-ref="235700b1-b85b-4c3c-b74e-4052acc90f96">Chimeric analysis of fibroblast growth factor receptor-1 (Fgfr1) function: a role for FGFR1 in morphogenetic movement through the primitive streak.</span><span class="reference__origin" data-ref="dde4f8fe-d9a5-45c7-9b9e-44bd95bc1fb9"><i data-ref="fc447b62-448d-43c8-a56d-2d2be55602ff">Development</i> <strong data-ref="f3390fd9-a480-4f95-a7a0-4c533f70552e">124</strong>:2829‑2841</span><span class="reference__doi" data-ref="c28d12ea-47ea-4ae9-8092-0211eeba6f01"></span></li><li class="reference-list__item" id="c30" data-ref="9576dd2d-7b81-44a2-8eaf-f00bec7eb102" data-id="c30"><span class="reference__label" data-ref="096cc2cc-26d6-48e3-b319-d9651553c665">30.</span><ol class="reference__authors_list" id="el-ol-33" data-ref="40be3c9e-9813-4d89-a13b-f4ab8e6cbf2a" data-id="el-ol-33"><li class="reference__author" id="el-li-150" data-ref="a655626e-96c6-4324-8e49-6f3df51f352b" data-id="el-li-150" data-item-num="1">García-García M. J.</li><li class="reference__author" id="el-li-151" data-ref="99336973-b649-4e5f-8fbd-1c3472e05ecd" data-id="el-li-151" data-item-num="2">Anderson K. V</li></ol><span class="reference__authors_list_suffix" data-ref="33268b11-cd1b-4455-ad8d-40f2fcd2902f">2003</span><span class="reference__title" data-ref="245bd66b-a57c-40c6-af23-162d4d64f163">Essential role of glycosaminoglycans in Fgf signaling during mouse gastrulation.</span><span class="reference__origin" data-ref="5c16caf9-5b47-4789-b07e-7878d7ec8652"><i data-ref="7edd03c8-af12-4abb-ad62-32cd0ecce439">Cell</i> <strong data-ref="2fe6dbe6-3c85-45af-b830-e137a4685970">114</strong>:727‑737</span><span class="reference__doi" data-ref="04190581-3a17-4872-a840-8967b32fdeee"></span></li><li class="reference-list__item" id="c31" data-ref="5efcfb8a-296f-4753-bb6b-20b12738c6db" data-id="c31"><span class="reference__label" data-ref="4a630f9b-61c4-4da3-8ddc-3533349eaf2e">31.</span><ol class="reference__authors_list" id="el-ol-34" data-ref="470f1e10-fcf1-44f7-b9fc-d0e2e1943e0f" data-id="el-ol-34"><li class="reference__author" id="el-li-153" data-ref="74b6df40-5721-4991-a03c-dd2f39c1040d" data-id="el-li-153" data-item-num="1">Oki S.</li><li class="reference__author" id="el-li-154" data-ref="a03350de-4c33-4fa4-83b1-7a9f54463bf9" data-id="el-li-154" data-item-num="2">Kitajima K.</li><li class="reference__author" id="el-li-155" data-ref="c9b52bef-2904-4465-b4ac-9fbc4788a404" data-id="el-li-155" data-item-num="3">Meno C</li></ol><span class="reference__authors_list_suffix" data-ref="eb6cd471-d184-4ec0-b3b8-54bcc72bfeeb">2010</span><span class="reference__title" data-ref="d0a42abf-a8ae-46d4-a333-88b56fdcc255">Dissecting the role of Fgf signaling during gastrulation and left-right axis formation in mouse embryos using chemical inhibitors.</span><span class="reference__origin" data-ref="c06dbb70-81df-44c1-a3ce-7b841e85cd0e"><i data-ref="762ed09b-ef73-4d20-b943-719c99e9c71b">Dev Dyn</i> <strong data-ref="01d371ac-82bc-4ca8-8a27-4caf2408f3a7">239</strong>:1768‑1778</span><span class="reference__doi" data-ref="91175d62-aac5-44a6-a302-358c2b68d5b7"></span></li><li class="reference-list__item" id="c32" data-ref="3475504a-18f6-475a-a44d-aaf62252e491" data-id="c32"><span class="reference__label" data-ref="136c772c-3c08-441f-ab4d-cedb71779dc1">32.</span><ol class="reference__authors_list" id="el-ol-35" data-ref="aa06b77f-54df-4494-b318-074602c5fd12" data-id="el-ol-35"><li class="reference__author" id="el-li-157" data-ref="2f29b4d2-d7b2-4b9a-85db-0b1547d0669c" data-id="el-li-157" data-item-num="1">Feldman B.</li><li class="reference__author" id="el-li-158" data-ref="3a4a8eb6-4c7f-46bf-aac0-2431f512298f" data-id="el-li-158" data-item-num="2">Poueymirou W.</li><li class="reference__author" id="el-li-159" data-ref="126daef3-945c-4765-90a3-532f8e6cfdff" data-id="el-li-159" data-item-num="3">Papaioannou V. E.</li><li class="reference__author" id="el-li-160" data-ref="79b558ab-b12d-49b0-8c6c-7348aa43371b" data-id="el-li-160" data-item-num="4">DeChiara T. M.</li><li class="reference__author" id="el-li-161" data-ref="5421c43e-af83-4bbe-95fc-b75f09444b8a" data-id="el-li-161" data-item-num="5">Goldfarb M</li></ol><span class="reference__authors_list_suffix" data-ref="1a581dbe-ce1f-4be2-997e-69798bb70c76">1995</span><span class="reference__title" data-ref="fdfbbc8d-1617-4e25-9b9e-b1b0909cf34a">Requirement of FGF-4 for postimplantation mouse development.</span><span class="reference__origin" data-ref="af8ff349-1e97-4ce6-8bc4-fb87cba044f2"><i data-ref="02cfaee6-9775-4be7-8457-1180cb0c4da1">Science</i> <strong data-ref="38bf92fd-d875-445f-978f-b811dc9a36f8">267</strong>:246‑249</span><span class="reference__doi" data-ref="df9928f3-f050-4e71-ab93-39e2e5b326a8"></span></li><li class="reference-list__item" id="c33" data-ref="a7dfb8c4-fc93-4deb-ae6f-3096f4b75fd3" data-id="c33"><span class="reference__label" data-ref="cf941a4f-d0c5-4e3b-b0dd-815be5bc4c83">33.</span><ol class="reference__authors_list" id="el-ol-36" data-ref="4d2ec906-f2fd-4dd6-bfc4-1cc3ee5385a1" data-id="el-ol-36"><li class="reference__author" id="el-li-163" data-ref="1910676c-cf6a-45c5-8610-30c2236be84d" data-id="el-li-163" data-item-num="1">Warmflash A.</li><li class="reference__author" id="el-li-164" data-ref="c1ed6901-45cd-48b1-beca-a387e2b7d4d1" data-id="el-li-164" data-item-num="2">Sorre B.</li><li class="reference__author" id="el-li-165" data-ref="117e2e97-8e40-4b84-b3a9-7932d32fd96b" data-id="el-li-165" data-item-num="3">Etoc F.</li><li class="reference__author" id="el-li-166" data-ref="2cfa4be2-6342-4cc7-b188-7d6737e395c4" data-id="el-li-166" data-item-num="4">Siggia E. D.</li><li class="reference__author" id="el-li-167" data-ref="e33a967c-a2c9-4e41-a089-103a043c6bfe" data-id="el-li-167" data-item-num="5">Brivanlou A. H</li></ol><span class="reference__authors_list_suffix" data-ref="f4c69c37-5657-4a2c-9e4f-7f452c233d96">2014</span><span class="reference__title" data-ref="1f103d5e-3dc2-4e29-8420-4300a319124a">A method to recapitulate early embryonic spatial patterning in human embryonic stem cells.</span><span class="reference__origin" data-ref="8fc9a347-01b5-4086-aea2-41dcfbd97153"><i data-ref="6e61dab0-9e8c-4b75-b1e9-62e5bde530e0">Nat. Methods</i> <strong data-ref="4f664aae-3df3-43c3-87d1-bd793b9bef2a">11</strong>:847‑854</span><span class="reference__doi" data-ref="eb20efa3-02f3-439d-a85f-c06ff54610fa"></span></li><li class="reference-list__item" id="c34" data-ref="570e7726-715f-4abd-a022-a63a2470849a" data-id="c34"><span class="reference__label" data-ref="f1e48a74-dd4c-44b9-a432-52492a347d0b">34.</span><ol class="reference__authors_list" id="el-ol-37" data-ref="3cfb673d-deb8-481a-beb9-bea4d35a079d" data-id="el-ol-37"><li class="reference__author" id="el-li-169" data-ref="e4cf5786-b3fc-4c8f-b1bf-4faafbd2ac5c" data-id="el-li-169" data-item-num="1">Jo K.</li><li class="reference__author" id="el-li-170" data-ref="bd2ce6e4-35e1-408f-8750-7e527d5d3bbb" data-id="el-li-170" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="94b6850b-8b4e-409b-91b9-3dc6f1599dbc">2022</span><span class="reference__title" data-ref="9de3dba0-0885-45bc-854e-75d1ff80e573">Efficient differentiation of human primordial germ cells through geometric control reveals a key role for Nodal signaling.</span><span class="reference__origin" data-ref="b8d7c848-2120-4c21-b328-a6a928276186"><i data-ref="d932f30a-9128-4856-a0f0-7065f82bc880">eLife</i> <strong data-ref="140c143c-2b9e-4130-a887-7abe7d94cfa6">11</strong>:e72811</span><span class="reference__doi" data-ref="335c72f8-4689-43b6-8f9b-533d8464a33c"><a href="https://doi.org/10.7554/eLife.72811" class="reference__doi_link" data-ref="d41016e2-6ac2-4514-b9b9-346548e6f02c">https://doi.org/10.7554/eLife.72811</a></span></li><li class="reference-list__item" id="c35" data-ref="8356ad9a-a77d-4633-9d37-d2c89128b0cc" data-id="c35"><span class="reference__label" data-ref="1941af9f-861a-4604-bc1c-bfc28881d13b">35.</span><ol class="reference__authors_list" id="el-ol-38" data-ref="f045c82d-333d-4b0e-86a7-1ae7097c3820" data-id="el-ol-38"><li class="reference__author" id="el-li-172" data-ref="0240e283-e0de-4e14-96fb-9234bbf9a0bf" data-id="el-li-172" data-item-num="1">Minn K. T.</li><li class="reference__author" id="el-li-173" data-ref="d6921f5a-9b09-419d-b827-05e0622c6da4" data-id="el-li-173" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="ec1c7c12-d20e-4c6e-a55e-da283e57527b">2020</span><span class="reference__title" data-ref="7ff6f395-d2ce-44d9-be30-5e03440daf0a">High-resolution transcriptional and morphogenetic profiling of cells from micropatterned human ESC gastruloid cultures.</span><span class="reference__origin" data-ref="ef999619-b029-4331-867c-943f5166eee5"><i data-ref="d62e3629-6753-4d45-8ae3-e6bab6327ad9">eLife</i> <strong data-ref="3e075baf-df8e-4ac3-bda8-bc50266eb43e">9</strong></span><span class="reference__doi" data-ref="f3bfd6ff-2fec-4a5a-973d-a149c42de326"></span></li><li class="reference-list__item" id="c36" data-ref="c6b55b9c-2f85-4dff-ad1f-4b313ae698c8" data-id="c36"><span class="reference__label" data-ref="5248e8bc-8c99-4861-b24d-f8107163038b">36.</span><ol class="reference__authors_list" id="el-ol-39" data-ref="2f2583f3-d9f2-45eb-92e6-ef0225019519" data-id="el-ol-39"><li class="reference__author" id="el-li-175" data-ref="e79af441-77d3-4d6e-a8f0-9468d34eca1b" data-id="el-li-175" data-item-num="1">Minn K. T.</li><li class="reference__author" id="el-li-176" data-ref="e6d83b22-969f-43a9-80af-67f876a69ce5" data-id="el-li-176" data-item-num="2">Dietmann S.</li><li class="reference__author" id="el-li-177" data-ref="f01e8218-eeb6-4707-99cb-4027a96654b6" data-id="el-li-177" data-item-num="3">Waye S. E.</li><li class="reference__author" id="el-li-178" data-ref="02c4d3a6-bf66-4c93-9dc6-1e138aa26a5c" data-id="el-li-178" data-item-num="4">Morris S. A.</li><li class="reference__author" id="el-li-179" data-ref="c6a6b04a-a7a6-4de2-8e3e-0e99c06dce32" data-id="el-li-179" data-item-num="5">Solnica-Krezel L</li></ol><span class="reference__authors_list_suffix" data-ref="cce0b235-eb7e-464c-bf0e-58f7cecd67b3">2021</span><span class="reference__title" data-ref="2f6e73ec-6b78-4e2a-ba2e-ba2f87df2b51">Gene expression dynamics underlying cell fate emergence in 2D micropatterned human embryonic stem cell gastruloids.</span><span class="reference__origin" data-ref="f3738bc5-152e-4939-999e-62513b8abc1c"><i data-ref="326189b8-9b71-40f3-bf4d-65b863d3dae4">Stem Cell Reports</i> <strong data-ref="4fe0a5a3-a722-4667-8f4e-632ebe2077ae">16</strong>:1210‑1227</span><span class="reference__doi" data-ref="4c235129-6e55-4d61-a7bc-a8ca22cdada5"></span></li><li class="reference-list__item" id="c37" data-ref="11bba441-24f4-4739-9e5a-ef887c33c145" data-id="c37"><span class="reference__label" data-ref="baf32f48-a9a6-4043-b390-037042adba2e">37.</span><ol class="reference__authors_list" id="el-ol-40" data-ref="e77f6c0f-0b94-481e-aab5-5cc137d44703" data-id="el-ol-40"><li class="reference__author" id="el-li-181" data-ref="63243a50-a334-4974-a96a-b26493f0cbda" data-id="el-li-181" data-item-num="1">Underhill E. J.</li><li class="reference__author" id="el-li-182" data-ref="28b7013f-2e44-4d04-a982-63fe2e81fe03" data-id="el-li-182" data-item-num="2">Toettcher J. E</li></ol><span class="reference__authors_list_suffix" data-ref="62524045-5d1f-4699-92b1-b224acab2bd2">2023</span><span class="reference__title" data-ref="b39cdcce-4aee-425a-9636-c6a48fa45d0f">Control of gastruloid patterning and morphogenesis by the Erk and Akt signaling pathways.</span><span class="reference__origin" data-ref="76c56493-8e01-4698-922c-08486819277e"><i data-ref="de18bb90-df53-4f43-b10d-0fe5475ed81e">Development</i> <strong data-ref="6e085bc0-3180-4e42-aca7-85b6cff32c02">150</strong>:dev201663</span><span class="reference__doi" data-ref="ceaa957a-447a-4ae7-9f61-66cc245ce582"></span></li><li class="reference-list__item" id="c38" data-ref="0d41caf4-1948-4059-a3c1-940c3d42d157" data-id="c38"><span class="reference__label" data-ref="f5717658-b2ec-47d0-ab26-3c1eeb9aa6bf">38.</span><ol class="reference__authors_list" id="el-ol-41" data-ref="ccc29d11-35d4-49d6-9116-47ed92358fa4" data-id="el-ol-41"><li class="reference__author" id="el-li-184" data-ref="71f2da39-f064-4d2f-a287-60b68286c33b" data-id="el-li-184" data-item-num="1">Loh K. M.</li><li class="reference__author" id="el-li-185" data-ref="11f255b4-c8c8-494e-8182-ae3d3228eb66" data-id="el-li-185" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="8be1dbf1-83d8-4d28-b70f-3762c95b6a79">2016</span><span class="reference__title" data-ref="f0ce5d6d-2360-434e-a121-89516c1dbf48">Mapping the Pairwise Choices Leading from Pluripotency to Human Bone, Heart, and Other Mesoderm Cell Types.</span><span class="reference__origin" data-ref="ccf549f5-f5dd-4b42-9ddb-8b150dcb679f"><i data-ref="269d3ff2-22dc-4082-89f0-c81665fc3b54">Cell</i> <strong data-ref="e105abf9-947e-4ccd-87d8-c7a0245a6818">166</strong>:451‑467</span><span class="reference__doi" data-ref="5c1bbe70-0362-4467-a90d-0e60b199f4bd"></span></li><li class="reference-list__item" id="c39" data-ref="2d191805-f3ea-429f-b08f-6d9060c135c1" data-id="c39"><span class="reference__label" data-ref="51bbf997-daf5-40f5-9942-f7cd32e35d29">39.</span><ol class="reference__authors_list" id="el-ol-42" data-ref="55e07162-b85d-4bca-bfea-95896190ee5a" data-id="el-ol-42"><li class="reference__author" id="el-li-187" data-ref="7b722108-6bb7-402c-a117-9c51a4789f40" data-id="el-li-187" data-item-num="1">Moon K. R.</li><li class="reference__author" id="el-li-188" data-ref="7770089c-8505-4414-9586-e8677269440a" data-id="el-li-188" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="f45731c3-2c4a-48d7-8977-d33b0a791402">2019</span><span class="reference__title" data-ref="2ed5567d-5321-4faf-9af0-1f929e630403">Visualizing structure and transitions in high-dimensional biological data.</span><span class="reference__origin" data-ref="e112d0f9-d3c8-40ad-b123-ee87da53a785"><i data-ref="aca0cda4-2fe2-49ed-9a4a-a9bbcfdd08da">Nat. Biotechnol</i> <strong data-ref="d7908d15-35e3-4ecb-b509-aa65731c5869">37</strong>:1482‑1492</span><span class="reference__doi" data-ref="21bf7b1e-24da-4c31-9a5a-1a99cc5466c4"></span></li><li class="reference-list__item" id="c40" data-ref="19f6195a-fdd9-495a-84bc-bbb2ca011c9c" data-id="c40"><span class="reference__label" data-ref="713aa63f-8add-42ab-8dfc-add524c1ad19">40.</span><ol class="reference__authors_list" id="el-ol-43" data-ref="eef5dc64-d935-4730-94dc-083f705f8dd1" data-id="el-ol-43"><li class="reference__author" id="el-li-190" data-ref="a926c3ff-4af5-4567-a290-1c8371842f3a" data-id="el-li-190" data-item-num="1">Neben C. L.</li><li class="reference__author" id="el-li-191" data-ref="58d0c569-fd4a-4ab2-bc62-51427d21e8ea" data-id="el-li-191" data-item-num="2">Lo M.</li><li class="reference__author" id="el-li-192" data-ref="89d4bb0e-9df2-4942-9abe-40d8ce8a1759" data-id="el-li-192" data-item-num="3">Jura N.</li><li class="reference__author" id="el-li-193" data-ref="5e7ffd82-1a27-4b13-9260-0c6babd929a4" data-id="el-li-193" data-item-num="4">Klein O. D</li></ol><span class="reference__authors_list_suffix" data-ref="b9355595-4411-4bbb-af7a-8e8430e7bf67">2019</span><span class="reference__title" data-ref="a857dea6-1fcd-42e4-9b6a-4be231c529e5">Feedback regulation of RTK signaling in development.</span><span class="reference__origin" data-ref="6a6d93ac-0c57-4df8-a720-1e9832174538"><i data-ref="a73befa0-71f4-4ab3-a8a1-73bd1c997a41">Dev Biol</i> <strong data-ref="95392f59-9986-4e6c-85c0-cda672a66655">447</strong>:71‑89</span><span class="reference__doi" data-ref="3dc6df37-c5a0-44f0-a763-59d060c524e1"></span></li><li class="reference-list__item" id="c41" data-ref="72ab001b-fef6-4dce-bc33-c68f2d3dc431" data-id="c41"><span class="reference__label" data-ref="2b4d9ae1-173f-4ddf-9212-7d75685be866">41.</span><ol class="reference__authors_list" id="el-ol-44" data-ref="93a8262b-6d8c-4677-9c00-04cf9bb5bd4a" data-id="el-ol-44"><li class="reference__author" id="el-li-195" data-ref="b9af9d13-b447-4664-ba54-20274d967a09" data-id="el-li-195" data-item-num="1">Tyser R. C. V.</li><li class="reference__author" id="el-li-196" data-ref="4bea766d-6765-4adc-8760-536c24d26a07" data-id="el-li-196" data-item-num="2">Srinivas S</li></ol><span class="reference__authors_list_suffix" data-ref="9693f8a5-6622-45c6-ba20-174f3e10648a">2022</span><span class="reference__title" data-ref="e89cdf71-2548-47e7-890a-c565627a8527">Recent advances in understanding cell types during human gastrulation.</span><span class="reference__origin" data-ref="937edb0c-9343-4e50-959a-fd4025c5d0e9"><i data-ref="adb88c34-1873-48de-a039-e065a217a6a5">Seminars in Cell &amp; Developmental Biology</i> <strong data-ref="8fad5956-ef73-4c5b-817c-1e3db9e1db02">131</strong>:35‑43</span><span class="reference__doi" data-ref="8e15e8d2-149f-4392-b7d3-38797268d858"></span></li><li class="reference-list__item" id="c42" data-ref="3efe6edb-062b-4c6b-9deb-c2050a37fa32" data-id="c42"><span class="reference__label" data-ref="8c149b6d-99b1-4f9a-b9c0-119554bc52c6">42.</span><ol class="reference__authors_list" id="el-ol-45" data-ref="f7f061f7-56c3-4a21-9210-c838d1ea9c3f" data-id="el-ol-45"><li class="reference__author" id="el-li-198" data-ref="49df7eb0-6f97-449e-8710-1d32a6baa377" data-id="el-li-198" data-item-num="1">Chen B.</li><li class="reference__author" id="el-li-199" data-ref="4efe258c-91a8-4f5a-a145-79a920863d9d" data-id="el-li-199" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="64dc1472-c9d4-44a8-bee7-6708b5d21b6f">2024</span><span class="reference__title" data-ref="410fd4b4-1963-4106-a5df-5eeaae6ea4ec">Extended culture of 2D gastruloids to model human mesoderm development.</span><span class="reference__origin" data-ref="d26cf0b9-8214-477f-a15e-fa66bab45c95"><i data-ref="403705e8-914d-4553-a74b-4a9faa065285">bioRxiv</i> </span><span class="reference__doi" data-ref="0ad5535e-edda-42b7-af6d-49361a1b758d"><a href="https://doi.org/10.1101/2024.03.21.585753" class="reference__doi_link" data-ref="1610d047-9f29-45ef-8ef6-023642d7c9c2">https://doi.org/10.1101/2024.03.21.585753</a></span></li><li class="reference-list__item" id="c43" data-ref="54f285ad-1597-4440-b0da-c77b825d6102" data-id="c43"><span class="reference__label" data-ref="d37d1759-5146-4fbc-8c81-033da4e27932">43.</span><ol class="reference__authors_list" id="el-ol-46" data-ref="046ab2ff-782e-4e81-84dc-f052067dec84" data-id="el-ol-46"><li class="reference__author" id="el-li-201" data-ref="4b6b5aab-c9f9-47c4-8aba-100e8237bbdc" data-id="el-li-201" data-item-num="1">Khoa L. T. P.</li><li class="reference__author" id="el-li-202" data-ref="4e097b10-a7ab-411d-9e2c-b6543d96d0ff" data-id="el-li-202" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="861b93a6-91cd-4223-b6a4-ead9c67b8a25">2016</span><span class="reference__title" data-ref="a4266ab4-37f0-40f1-b348-75736eff3da5">Visualization of the Epiblast and Visceral Endodermal Cells Using Fgf5-P2A-Venus BAC Transgenic Mice and Epiblast Stem Cells.</span><span class="reference__origin" data-ref="3df91b44-c53c-4c66-a5f2-2550d8713af1"><i data-ref="7cd6fbb2-acb9-49a6-bceb-b2575e924e44">PLOS One</i> <strong data-ref="ddfbe828-3379-4929-b910-1ee3385947ff">11</strong>:e0159246</span><span class="reference__doi" data-ref="4912cadd-eed1-4c0f-9922-e5c5acccbb1a"></span></li><li class="reference-list__item" id="c44" data-ref="cd1cffef-3465-469d-bc3a-32ba2a7bba0a" data-id="c44"><span class="reference__label" data-ref="7ac18599-c754-4e45-9d3d-dd73b81e8a5e">44.</span><ol class="reference__authors_list" id="el-ol-47" data-ref="ea16c466-05cf-437b-8f30-f3cc65729fe3" data-id="el-ol-47"><li class="reference__author" id="el-li-204" data-ref="baaf3b26-8a74-4705-9a67-ccaa2542d700" data-id="el-li-204" data-item-num="1">Pijuan-Sala B.</li><li class="reference__author" id="el-li-205" data-ref="9b237799-a0fc-4154-975b-00dbbdd49fbc" data-id="el-li-205" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="2379be92-0d9c-412c-a11f-db889cd89b42">2019</span><span class="reference__title" data-ref="4173fddc-1565-49b2-8779-ecd34e254d2d">A single-cell molecular map of mouse gastrulation and early organogenesis.</span><span class="reference__origin" data-ref="78e7bab8-d10a-49b7-ab56-c12999dfe80b"><i data-ref="580555a2-ea30-4d63-8842-39b812445bff">Nature</i> <strong data-ref="3a48893e-a144-4d35-98dd-e9a1ffd83d6a">566</strong>:490‑495</span><span class="reference__doi" data-ref="74220029-6e50-4977-b67a-7ee1a1e89430"></span></li><li class="reference-list__item" id="c45" data-ref="2b867549-8731-4d3a-a056-36b7734df071" data-id="c45"><span class="reference__label" data-ref="58b99cb6-21a4-42d1-a3c2-18a4fc3146dc">45.</span><ol class="reference__authors_list" id="el-ol-48" data-ref="292d4ccc-978e-45dd-a503-43af62a0a58b" data-id="el-ol-48"><li class="reference__author" id="el-li-207" data-ref="dcc5844e-5238-4414-ae83-23520ef14254" data-id="el-li-207" data-item-num="1">Morgani S. M.</li><li class="reference__author" id="el-li-208" data-ref="5b2f2346-a51f-4130-a175-c0d199280c2c" data-id="el-li-208" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="885378de-078f-4eb1-a1f1-2c182012a8b5">2018</span><span class="reference__title" data-ref="4559ac85-1f0f-4194-b52d-8982bb86d244">A Sprouty4 reporter to monitor FGF/ERK signaling activity in ESCs and mice.</span><span class="reference__origin" data-ref="aac1d037-4e14-4bb5-80a7-a29575310114"><i data-ref="4803e476-0bc8-4096-a071-44871a1b428a">Developmental Biology</i> <strong data-ref="a91b4cc0-1cbf-41f8-b9b0-a0ac0ec0749f">441</strong>:104‑126</span><span class="reference__doi" data-ref="f22288bc-f1fa-4f28-afc9-0738f6f44102"></span></li><li class="reference-list__item" id="c46" data-ref="dadf73b5-e341-4f58-bbe8-b48d008950b8" data-id="c46"><span class="reference__label" data-ref="98498c9e-cef4-4dd1-9547-3393234391ec">46.</span><ol class="reference__authors_list" id="el-ol-49" data-ref="7353d557-4616-4a81-bac0-4bf44323a910" data-id="el-ol-49"><li class="reference__author" id="el-li-210" data-ref="a0b9fa66-0809-4419-8c52-f71a45cd23bb" data-id="el-li-210" data-item-num="1">Xu D.</li><li class="reference__author" id="el-li-211" data-ref="a89d8ff0-f87c-4022-988c-00e92910361f" data-id="el-li-211" data-item-num="2">Esko J. D</li></ol><span class="reference__authors_list_suffix" data-ref="6ec5bbb0-fef7-4043-a572-b19c9eba6e9c">2014</span><span class="reference__title" data-ref="0ec2468c-a770-4c84-a1b8-4fac10ff679d">Demystifying heparan sulfate-protein interactions.</span><span class="reference__origin" data-ref="3e36d7b9-1fc2-42e5-9ba9-2b7a8cb7254b"><i data-ref="85a3d7b5-770f-44f7-8246-e59d163bb69f">Annu Rev Biochem</i> <strong data-ref="19b4da0e-e2ae-4e8f-a5b9-a0d320c09172">83</strong>:129‑157</span><span class="reference__doi" data-ref="db20e079-18ca-469b-99ff-cd49470fa44e"></span></li><li class="reference-list__item" id="c47" data-ref="6a663f23-eac5-41db-9107-0815ed0cc3f5" data-id="c47"><span class="reference__label" data-ref="7895fdc3-5ff6-4d59-b5c0-6c448fafdbb2">47.</span><ol class="reference__authors_list" id="el-ol-50" data-ref="2a684471-51a0-46ab-bacb-47363c74ddcd" data-id="el-ol-50"><li class="reference__author" id="el-li-213" data-ref="f39dc18a-de8a-421f-9647-41c54969cde2" data-id="el-li-213" data-item-num="1">Habuchi H.</li><li class="reference__author" id="el-li-214" data-ref="3fee3674-66b8-4f0f-98dd-5efa22e04f59" data-id="el-li-214" data-item-num="2">Kimata K</li></ol><span class="reference__authors_list_suffix" data-ref="b6532cb2-7dfb-42e4-8218-9323f9114978">2008</span><span class="reference__title" data-ref="ee388bcc-5feb-42b7-9d88-205f24502666">Functions of Heparan Sulfate Proteoglycans in Morphogenesis.</span><span class="reference__origin" data-ref="d1d1d7ed-d1b3-437e-95ba-9e21cebf25b8">In: <ol class="reference__editors_list" id="el-ol-51" data-ref="fe779cc2-2a71-4d58-977a-bb11f146810d" data-id="el-ol-51"><li class="reference__editor" id="el-li-215" data-ref="0ff89d59-4565-40c2-9798-2b41746b861c" data-id="el-li-215" data-item-num="1">Taniguchi N.</li><li class="reference__editor" id="el-li-216" data-ref="3d1c31ef-25b6-44c2-83e6-ba39eb868193" data-id="el-li-216" data-item-num="2">Suzuki A</li><li class="reference__editor" id="el-li-217" data-ref="a4f3428f-fe2f-4d29-8791-2ac8675e79b9" data-id="el-li-217" data-item-num="3">Ito Y</li><li class="reference__editor" id="el-li-218" data-ref="96993f10-1c2a-4a50-9817-9d7a0aab3b0d" data-id="el-li-218" data-item-num="4">Narimatsu H</li><li class="reference__editor" id="el-li-219" data-ref="704662df-9e0d-4021-bb83-afbc12f9c85b" data-id="el-li-219" data-item-num="5">Kawasaki T</li><li class="reference__editor" id="el-li-220" data-ref="f1103afd-9d75-4b55-80d4-fdea68e885c3" data-id="el-li-220" data-item-num="6">Hase S</li></ol> (Eds). <i data-ref="4730504f-86c9-47ac-bbc5-0409e10413dc">Experimental Glycoscience</i> Tokyo: Springer. pp.&nbsp;303‑308</span><span class="reference__doi" data-ref="3de4356b-00d9-4fec-9a34-dc1d300f8e7e"><a href="https://doi.org/10.1007/978-4-431-77922-3_72" class="reference__doi_link" data-ref="0f5eeb97-1ead-48a6-a360-70014ead596f">https://doi.org/10.1007/978-4-431-77922-3_72</a></span></li><li class="reference-list__item" id="c48" data-ref="1dbd441c-1407-4515-a789-183cd324b7e3" data-id="c48"><span class="reference__label" data-ref="1c9aa9f0-16b7-47d0-8bae-92421f418ca2">48.</span><ol class="reference__authors_list" id="el-ol-52" data-ref="728ee662-8509-4d40-aa91-f9c510568ace" data-id="el-ol-52"><li class="reference__author" id="el-li-222" data-ref="50806837-1c73-4fdb-91aa-c2bd403f7430" data-id="el-li-222" data-item-num="1">Yu S. R.</li><li class="reference__author" id="el-li-223" data-ref="e26445ee-1ee0-4ebd-84c6-1cd3e97ba557" data-id="el-li-223" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="44ed28a1-1eda-4b31-9733-6eb2047c9dcf">2009</span><span class="reference__title" data-ref="92dac5e4-f985-4983-a5d7-f3182517459c">Fgf8 morphogen gradient forms by a source-sink mechanism with freely diffusing molecules.</span><span class="reference__origin" data-ref="54a929d2-4b8a-406e-921b-cefba63e89c4"><i data-ref="695acd2a-fb54-4bf5-9f9c-39afaa809fff">Nature</i> <strong data-ref="292f3762-da76-4520-ac21-e50b2d49be6f">461</strong>:533‑536</span><span class="reference__doi" data-ref="6ecf5029-cdfe-42a5-a392-7a2925538de5"></span></li><li class="reference-list__item" id="c49" data-ref="b31aa406-e64b-4625-b4c8-1999f2528f6c" data-id="c49"><span class="reference__label" data-ref="eaa918d2-2f68-4886-8172-abb31ac73e89">49.</span><ol class="reference__authors_list" id="el-ol-53" data-ref="23fa2508-81af-439a-9ad5-ea9da0c710c8" data-id="el-ol-53"><li class="reference__author" id="el-li-225" data-ref="d840b7c0-7726-4d71-a2fc-69f26f56aaf5" data-id="el-li-225" data-item-num="1">Benington L.</li><li class="reference__author" id="el-li-226" data-ref="8fe08a4c-9043-42d8-9c41-dbba58f47bdb" data-id="el-li-226" data-item-num="2">Rajan G.</li><li class="reference__author" id="el-li-227" data-ref="986d5ff7-9ec7-4b85-b133-fe665617c2fe" data-id="el-li-227" data-item-num="3">Locher C.</li><li class="reference__author" id="el-li-228" data-ref="cd7efee6-dd34-4153-ae31-bf6742ecee5c" data-id="el-li-228" data-item-num="4">Lim L. Y</li></ol><span class="reference__authors_list_suffix" data-ref="ceaba6dd-1e6d-43f1-bd48-9c7a72769806">2020</span><span class="reference__title" data-ref="84c3d822-675c-4428-a8af-0e7a957c5452">Fibroblast Growth Factor 2-A Review of Stabilisation Approaches for Clinical Applications.</span><span class="reference__origin" data-ref="2c8e3107-300e-4b54-a4be-1da9ce15aa82"><i data-ref="73ebf559-543f-4494-9927-6130e4c49143">Pharmaceutics</i> <strong data-ref="329d121e-4c76-4c2c-869b-a390dc318a76">12</strong>:508</span><span class="reference__doi" data-ref="32c77416-974a-4ebd-890b-5259db4c395d"></span></li><li class="reference-list__item" id="c50" data-ref="8a9d2a8f-ff49-4d19-bd43-9c311557f460" data-id="c50"><span class="reference__label" data-ref="a6b59077-8e2e-46ed-a28c-bdf1ed8a2201">50.</span><ol class="reference__authors_list" id="el-ol-54" data-ref="5429e575-aeb2-4dbd-9a36-455e0bad8098" data-id="el-ol-54"><li class="reference__author" id="el-li-230" data-ref="aaa5ba57-0b06-4ac9-bea9-49240940a7bf" data-id="el-li-230" data-item-num="1">Teague S.</li><li class="reference__author" id="el-li-231" data-ref="7eb2143c-ba1b-4218-a49c-e680aeb02fbb" data-id="el-li-231" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="bb55dc77-838a-412f-a10a-b25f1c5bb665">2024</span><span class="reference__title" data-ref="2c214e39-d944-4060-bc79-b8f20df145dd">Time-integrated BMP signaling determines fate in a stem cell model for early human development.</span><span class="reference__origin" data-ref="f730eba6-146a-49fa-b5bd-2d482a40caa3"><i data-ref="5621b7ab-f993-4b46-89bd-397ed4807ac4">Nat Commun</i> <strong data-ref="4803b541-8393-4556-82bd-b8b90a5d85f3">15</strong>:1471</span><span class="reference__doi" data-ref="d41bda24-fbe1-4009-9b75-1fbe76a812a3"></span></li><li class="reference-list__item" id="c51" data-ref="b4e6164c-9fa6-4c3f-b382-00e6a03f5f19" data-id="c51"><span class="reference__label" data-ref="de07cb49-30d4-40b3-9f7b-50e36d606083">51.</span><ol class="reference__authors_list" id="el-ol-55" data-ref="4e52311c-9cc8-4ad3-a3c9-db816c8c16b9" data-id="el-ol-55"><li class="reference__author" id="el-li-233" data-ref="e78cbdab-1acf-481c-bb5f-ab0808bdc515" data-id="el-li-233" data-item-num="1">Ortiz-Salazar M. A.</li><li class="reference__author" id="el-li-234" data-ref="df6ebb31-c8da-4a54-b29c-10268157096e" data-id="el-li-234" data-item-num="2">Camacho-Aguilar E.</li><li class="reference__author" id="el-li-235" data-ref="02e67fdd-a41f-4085-b115-430a91b2393b" data-id="el-li-235" data-item-num="3">Warmflash A.</li></ol><span class="reference__authors_list_suffix" data-ref="55ac6e65-b958-4d09-a861-2c38e114595c">2024</span><span class="reference__title" data-ref="fce24243-75a5-49ac-b9f6-dee3e52942bf">Endogenous Nodal switches Wnt interpretation from posteriorization to germ layer differentiation in geometrically constrained human pluripotent cells.</span><span class="reference__origin" data-ref="81bc376b-c687-448e-8ddb-4a61ec6287ff"><i data-ref="b4225665-52fd-4d7f-85f9-530c2af2ce67">bioRxiv</i> </span><span class="reference__doi" data-ref="e43257fe-5aa4-44ea-a786-78c9c5e32240"><a href="https://doi.org/10.1101/2024.03.13.584912" class="reference__doi_link" data-ref="cfd04c16-79fe-4c08-a236-325372702258">https://doi.org/10.1101/2024.03.13.584912</a></span></li><li class="reference-list__item" id="c52" data-ref="680033bb-d61c-4ccd-946d-9d9f9a1bdcb7" data-id="c52"><span class="reference__label" data-ref="312f434b-9f25-468a-a4d0-5a59582c6737">52.</span><ol class="reference__authors_list" id="el-ol-56" data-ref="0a70bea1-b0af-488a-b173-3d451fb81922" data-id="el-ol-56"><li class="reference__author" id="el-li-237" data-ref="8d9b415e-49e8-44cf-9524-f4a019d8892c" data-id="el-li-237" data-item-num="1">Gurdon J. B</li></ol><span class="reference__authors_list_suffix" data-ref="ff51423f-2c92-4627-b2a0-21c6ba2fb50b">1988</span><span class="reference__title" data-ref="bf3f5247-6a7c-41c1-8d89-ba778d3a445a">A community effect in animal development.</span><span class="reference__origin" data-ref="2ca140ba-8b2f-4049-a80c-0d1f61e17deb"><i data-ref="753187f4-4c2a-4542-99e0-89e2240dfada">Nature</i> <strong data-ref="ac912784-bcf3-4c4f-9b28-82dcf6413e7f">336</strong>:772‑774</span><span class="reference__doi" data-ref="81cfef9e-c382-43ca-ae78-fcd34e71d99f"></span></li><li class="reference-list__item" id="c53" data-ref="f62295b5-84dd-4acb-841f-edad92bb6922" data-id="c53"><span class="reference__label" data-ref="b4813a9b-3218-4362-bbfd-88bbed8b166f">53.</span><ol class="reference__authors_list" id="el-ol-57" data-ref="a74dfcc8-7fd6-4a17-8ca6-0b70693b608e" data-id="el-ol-57"><li class="reference__author" id="el-li-239" data-ref="03de9c15-69db-4269-b1b4-cc67ce5feefa" data-id="el-li-239" data-item-num="1">Nemashkalo A.</li><li class="reference__author" id="el-li-240" data-ref="473bff86-18e6-406f-96b9-28472fd70e61" data-id="el-li-240" data-item-num="2">Ruzo A.</li><li class="reference__author" id="el-li-241" data-ref="388e4ff1-4672-48fe-8a75-f7a73477f31a" data-id="el-li-241" data-item-num="3">Heemskerk I.</li><li class="reference__author" id="el-li-242" data-ref="2b727048-072c-403c-85a3-b6399ddd4317" data-id="el-li-242" data-item-num="4">Warmflash A</li></ol><span class="reference__authors_list_suffix" data-ref="28d13e14-a293-4d86-994f-ec1d2f065052">2017</span><span class="reference__title" data-ref="dc8d06f8-ff8b-41fb-bea8-08c980f306ca">Morphogen and community effects determine cell fates in response to BMP4 signaling in human embryonic stem cells.</span><span class="reference__origin" data-ref="c03e435d-0892-4017-8187-ed8a437d061b"><i data-ref="53be5179-c0f8-4049-92cb-b5d3864bcfa6">Development</i> <strong data-ref="2449423e-f271-40bb-bf88-aad8792de567">144</strong>:3042‑3053</span><span class="reference__doi" data-ref="1ec3ec1c-be5b-4d68-9eb8-b1c8c480e7a5"></span></li><li class="reference-list__item" id="c54" data-ref="65629743-e087-4bb2-9ac3-c11a3048a77a" data-id="c54"><span class="reference__label" data-ref="fc394fc0-f9a8-464d-a7ae-1720db28212a">54.</span><ol class="reference__authors_list" id="el-ol-58" data-ref="259556a8-c18f-4b2f-bff0-79478ce3f78f" data-id="el-ol-58"><li class="reference__author" id="el-li-244" data-ref="b8f967f0-fa68-48b0-ad94-55b9c8f2e719" data-id="el-li-244" data-item-num="1">Gattiglio M.</li><li class="reference__author" id="el-li-245" data-ref="266cec7e-8a37-46bc-a95f-9d8156b71c20" data-id="el-li-245" data-item-num="2">Protzek M.</li><li class="reference__author" id="el-li-246" data-ref="99e343b4-0d65-40e3-9f36-7317e0846781" data-id="el-li-246" data-item-num="3">Schröter C</li></ol><span class="reference__authors_list_suffix" data-ref="47638349-668b-4329-b862-e5e3d451dc67">2023</span><span class="reference__title" data-ref="31084244-5ce8-4d3f-9253-8ecbdc17f5e7">Population-level antagonism between FGF and BMP signaling steers mesoderm differentiation in embryonic stem cells.</span><span class="reference__origin" data-ref="01acaae5-81c6-4a18-83af-c5eff5cf0c0a"><i data-ref="d97d060d-6b88-49af-bd15-12e3c86d1c76">Biology Open</i> <strong data-ref="6abedc35-4273-4977-905f-4f9a1f460681">12</strong>:bio059941</span><span class="reference__doi" data-ref="c8c60c9e-a5b8-4165-ac0a-8ab7c884ffa0"></span></li><li class="reference-list__item" id="c55" data-ref="6eca9cbe-c63b-4d6c-9ce8-e88070789c07" data-id="c55"><span class="reference__label" data-ref="0858cf77-e5e5-46b5-a134-78f53173701d">55.</span><ol class="reference__authors_list" id="el-ol-59" data-ref="11466b86-6d4f-43f6-8285-16c73e42bdc0" data-id="el-ol-59"><li class="reference__author" id="el-li-248" data-ref="743a9018-8e3c-4373-b219-879267898631" data-id="el-li-248" data-item-num="1">Liu L.</li><li class="reference__author" id="el-li-249" data-ref="64b5e769-5a49-4fde-b831-7dfa2743a914" data-id="el-li-249" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="427cb68f-9f13-48d0-8aab-6deed9dc7c19">2022</span><span class="reference__title" data-ref="7d9b4b20-0265-4105-bb81-93532ba98433">Nodal is a short-range morphogen with activity that spreads through a relay mechanism in human gastruloids.</span><span class="reference__origin" data-ref="b5dfed1f-7c2c-40e1-beda-1fb87b57b071"><i data-ref="a191107c-459e-4479-9d90-e2195d578b80">Nat Commun</i> <strong data-ref="94a1971b-b63a-433c-aa2f-df70614cf540">13</strong>:497</span><span class="reference__doi" data-ref="33b7f532-bcc3-4a31-a97b-de72f8aa7b54"></span></li><li class="reference-list__item" id="c56" data-ref="ed8b8f8f-0cf0-4e7c-81e0-729153583162" data-id="c56"><span class="reference__label" data-ref="b4433a4b-3967-4064-b527-036703619fe2">56.</span><ol class="reference__authors_list" id="el-ol-60" data-ref="6cecf0b2-2dc7-44c8-8394-601a641226b1" data-id="el-ol-60"><li class="reference__author" id="el-li-251" data-ref="90168859-12af-4242-8fc3-232a2b59ff0d" data-id="el-li-251" data-item-num="1">Isaacs H. V.</li><li class="reference__author" id="el-li-252" data-ref="c35f54c0-e208-459a-84ab-5be447d4ca18" data-id="el-li-252" data-item-num="2">Pownall M. E.</li><li class="reference__author" id="el-li-253" data-ref="6f525a0a-53d6-4cec-b9c6-062160bc13f2" data-id="el-li-253" data-item-num="3">Slack J. M</li></ol><span class="reference__authors_list_suffix" data-ref="99538cdd-04a8-4ddd-9b29-f4407dfc42f8">1995</span><span class="reference__title" data-ref="3659dbc0-045c-4a57-9ea6-7f3bb146254a">eFGF is expressed in the dorsal midline of Xenopus laevis.</span><span class="reference__origin" data-ref="66f1d50a-916b-45b2-9ef0-db21d9df8205"><i data-ref="558384b2-af26-4633-b559-826e717b2c3f">Int J Dev Biol</i> <strong data-ref="d1a8317a-6de5-4206-a888-9b06272d409c">39</strong>:575‑579</span><span class="reference__doi" data-ref="35412053-5cc0-4f3f-8409-a2cdc8a912a8"></span></li><li class="reference-list__item" id="c57" data-ref="c1bc3ec9-de10-483b-944d-ae2b74902488" data-id="c57"><span class="reference__label" data-ref="c369f5f7-4934-4acf-bfd6-47928acb2003">57.</span><ol class="reference__authors_list" id="el-ol-61" data-ref="a7a94a3d-e69f-459a-abb3-17a7e7719197" data-id="el-ol-61"><li class="reference__author" id="el-li-255" data-ref="89cf23e0-2e89-4dad-a1a6-c27bf628a9e0" data-id="el-li-255" data-item-num="1">Christen B.</li><li class="reference__author" id="el-li-256" data-ref="0558b21e-7576-4bfa-b0c4-0a6b29f0ac3f" data-id="el-li-256" data-item-num="2">Slack J. M</li></ol><span class="reference__authors_list_suffix" data-ref="c905c7ea-9717-4c51-adec-2cfc01f29ef4">1997</span><span class="reference__title" data-ref="45be7a43-92f3-4c12-b489-f7fd0726c807">FGF-8 is associated with anteroposterior patterning and limb regeneration in Xenopus.</span><span class="reference__origin" data-ref="7f86557b-9121-462c-8eba-2a61a381b318"><i data-ref="4cfb7ea4-d62e-40b3-bc22-313efb8d8ff1">Dev Biol</i> <strong data-ref="cdb8a68c-037c-4469-a07c-1e7271041abf">192</strong>:455‑466</span><span class="reference__doi" data-ref="5b76c58f-47f1-4c81-932e-a6216ebb3bb9"></span></li><li class="reference-list__item" id="c58" data-ref="6faac146-9bc3-43ac-a40c-2c2359c7d701" data-id="c58"><span class="reference__label" data-ref="71f2d922-3319-4269-bef3-88642017767c">58.</span><ol class="reference__authors_list" id="el-ol-62" data-ref="9b7d2a59-e534-493a-a5ea-2ff9fe460dbd" data-id="el-ol-62"><li class="reference__author" id="el-li-258" data-ref="a6ae5db3-cc52-47d0-ac7b-f23c48010218" data-id="el-li-258" data-item-num="1">Hardy K. M.</li><li class="reference__author" id="el-li-259" data-ref="8c7f35dd-ee24-4044-9516-ee987e6c0d9e" data-id="el-li-259" data-item-num="2">Yatskievych T. A.</li><li class="reference__author" id="el-li-260" data-ref="766b8e5d-02ab-46b3-8d02-44b1e750d7e9" data-id="el-li-260" data-item-num="3">Konieczka J.</li><li class="reference__author" id="el-li-261" data-ref="b0d22ed8-7faa-4c6e-bc74-e58ebbdc7f9e" data-id="el-li-261" data-item-num="4">Bobbs A. S.</li><li class="reference__author" id="el-li-262" data-ref="066e81d0-5847-4209-9d6c-bbd18e842f86" data-id="el-li-262" data-item-num="5">Antin P. B</li></ol><span class="reference__authors_list_suffix" data-ref="42cb7a8a-8a11-4d88-b136-fc2390535322">2011</span><span class="reference__title" data-ref="686f9dc2-037c-40d3-810f-72fece9d446e">FGF signalling through RAS/MAPK and PI3K pathways regulates cell movement and gene expression in the chicken primitive streak without affecting E-cadherin expression.</span><span class="reference__origin" data-ref="9c90fc83-da92-45b6-9dc8-bac636b7cd05"><i data-ref="3969b195-16ab-44c3-8e72-5f4ecc52ef35">BMC Dev Biol</i> <strong data-ref="0b19341d-bd5a-4bf6-ae8a-99dbc73466d3">11</strong>:20‑17</span><span class="reference__doi" data-ref="478d7b4c-6d3e-4b08-aab6-7d32d04cb7a4"></span></li><li class="reference-list__item" id="c59" data-ref="7f40cb35-ffc8-4f3d-933e-adbc9b5887be" data-id="c59"><span class="reference__label" data-ref="5de97f2b-dbe5-4946-8ff6-faf4e61c88d0">59.</span><ol class="reference__authors_list" id="el-ol-63" data-ref="58a76943-5ed7-4237-9202-25b2431620ce" data-id="el-ol-63"><li class="reference__author" id="el-li-264" data-ref="a2b03847-c22b-4549-90fc-9ddeff6cc628" data-id="el-li-264" data-item-num="1">Isaacs H. V.</li><li class="reference__author" id="el-li-265" data-ref="2f229799-1af8-4e7e-b386-9a5a8f3a0642" data-id="el-li-265" data-item-num="2">Pownall M. E.</li><li class="reference__author" id="el-li-266" data-ref="2c9c466c-8d76-4011-a07c-adc81b03f4f1" data-id="el-li-266" data-item-num="3">Slack J. M</li></ol><span class="reference__authors_list_suffix" data-ref="b99d132a-02a5-4760-84e2-61e52283bccc">1994</span><span class="reference__title" data-ref="b0c617d9-0096-4ac3-941e-24e933efe401">eFGF regulates Xbra expression during Xenopus gastrulation.</span><span class="reference__origin" data-ref="72e919ee-e587-4334-8cb9-40f1133024c4"><i data-ref="a6e89f6a-83ac-4cce-8c99-ac16982cbecd">The EMBO Journal</i> <strong data-ref="0017c692-e203-4019-a7c6-a2201f591419">13</strong>:4469‑4481</span><span class="reference__doi" data-ref="d2c992e8-364e-4c5b-be4e-d68f0d8bf5c3"></span></li><li class="reference-list__item" id="c60" data-ref="8147b550-9de1-42b3-9397-60994b795490" data-id="c60"><span class="reference__label" data-ref="f7e9b0d0-a44a-49d6-ae90-e75399826055">60.</span><ol class="reference__authors_list" id="el-ol-64" data-ref="0fe8806f-4ca9-457e-a4f4-fef3e7cd45f8" data-id="el-ol-64"><li class="reference__author" id="el-li-268" data-ref="a47d908b-a6a6-4f42-ac4f-2d4c1d85a558" data-id="el-li-268" data-item-num="1">Bulger E. A.</li><li class="reference__author" id="el-li-269" data-ref="d99ed458-d744-4a6d-afa4-4c0224f54bab" data-id="el-li-269" data-item-num="2">Muncie-Vasic I.</li><li class="reference__author" id="el-li-270" data-ref="da28f9a4-d15b-49b1-9fe2-3fe510de360f" data-id="el-li-270" data-item-num="3">Libby A. R. G.</li><li class="reference__author" id="el-li-271" data-ref="74880a3b-9fe2-4f66-b158-1ff093487650" data-id="el-li-271" data-item-num="4">McDevitt T. C.</li><li class="reference__author" id="el-li-272" data-ref="f8615346-9543-42e6-a77b-882c11ed8ea5" data-id="el-li-272" data-item-num="5">Bruneau B. G</li></ol><span class="reference__authors_list_suffix" data-ref="72f01b27-9aae-4283-99ba-ed4823752c95">2024</span><span class="reference__title" data-ref="2392d4d1-ec44-4379-9fc9-131b3ef65b1f">TBXT dose sensitivity and the decoupling of nascent mesoderm specification from EMT progression in 2D human gastruloids.</span><span class="reference__origin" data-ref="3780c2de-b953-41d6-a389-b1c1aaa68efe"><i data-ref="91c68141-039b-4ec0-a2c3-84e72694096b">Development</i> <strong data-ref="672505b0-ecdb-4d56-818e-1c0a0f07e268">151</strong>:dev202516</span><span class="reference__doi" data-ref="cff2cf03-c45f-433e-8da8-0b0207ca0423"></span></li><li class="reference-list__item" id="c61" data-ref="1fd5a420-8d2b-48fd-8d03-7d4758e7d740" data-id="c61"><span class="reference__label" data-ref="abee5970-7902-4414-b854-2d551e629335">61.</span><ol class="reference__authors_list" id="el-ol-65" data-ref="c726937e-8fed-48fb-bd0a-4385fc659327" data-id="el-ol-65"><li class="reference__author" id="el-li-274" data-ref="51fc5af9-e634-4bb4-8fc8-0b02093473dc" data-id="el-li-274" data-item-num="1">Evans A. L.</li><li class="reference__author" id="el-li-275" data-ref="ffa8afe2-92c0-4020-97c5-ea3a1ed3fd5c" data-id="el-li-275" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="17904a7c-d9f8-4446-bbfc-65b274876f76">2012</span><span class="reference__title" data-ref="8aa664d7-709d-4d55-a752-7464ae0d02cb">Genomic Targets of Brachyury (T) in Differentiating Mouse Embryonic Stem Cells.</span><span class="reference__origin" data-ref="f291c13d-ea67-4601-b238-03bcc8d75f07"><i data-ref="977ae48a-233a-45c6-afdc-a47c57b5255a">PLOS One</i> <strong data-ref="65e2f9dc-1c48-4714-94be-2a4476fa5547">7</strong>:e33346</span><span class="reference__doi" data-ref="be0de8f5-71ab-412b-8b7c-16ab0c05e933"></span></li><li class="reference-list__item" id="c62" data-ref="bd2d60c5-a506-4586-86df-89f4d7e5c9f5" data-id="c62"><span class="reference__label" data-ref="60f7311c-8c92-4ca2-9b1d-457e30b92f9d">62.</span><ol class="reference__authors_list" id="el-ol-66" data-ref="2263d3b3-5249-43ca-b1c4-93fb45bea380" data-id="el-ol-66"><li class="reference__author" id="el-li-277" data-ref="dc6b3f36-4577-4460-ba8f-9ca2ef8c4bba" data-id="el-li-277" data-item-num="1">Nie Y.</li><li class="reference__author" id="el-li-278" data-ref="70ed5347-7a9b-46b1-a761-a85232632604" data-id="el-li-278" data-item-num="2">Walsh P.</li><li class="reference__author" id="el-li-279" data-ref="e5914d9c-6429-45dc-90a8-5bc9f8487699" data-id="el-li-279" data-item-num="3">Clarke D. L.</li><li class="reference__author" id="el-li-280" data-ref="79159ddc-61c5-4239-9e93-ce9dd4c7d492" data-id="el-li-280" data-item-num="4">Rowley J. A.</li><li class="reference__author" id="el-li-281" data-ref="359851c6-7414-4311-ab08-261f3244ff4d" data-id="el-li-281" data-item-num="5">Fellner T</li></ol><span class="reference__authors_list_suffix" data-ref="9eb1f09e-a2b6-4e9f-80ab-890b5723d50e">2014</span><span class="reference__title" data-ref="1f299ce6-023a-4236-ad76-8a032715f615">Scalable passaging of adherent human pluripotent stem cells.</span><span class="reference__origin" data-ref="3aceb9ac-bc00-47ee-a651-4edb2ef6c3ab"><i data-ref="3b76e53b-60c4-4927-b90c-e23442c53ff8">PLoS ONE</i> <strong data-ref="b9c8cf73-ce67-4a3e-9a63-fb1eee86015b">9</strong>:e88012</span><span class="reference__doi" data-ref="ce105922-5afb-4723-ad3f-54a26cb279d3"></span></li><li class="reference-list__item" id="c63" data-ref="bd40cef0-e241-4a79-b602-7c310ccc64f9" data-id="c63"><span class="reference__label" data-ref="6545b26a-3f23-4fbd-a195-be0d9b3c9a4f">63.</span><ol class="reference__authors_list" id="el-ol-67" data-ref="4d72085c-de0c-4c6b-9725-b462e8407401" data-id="el-ol-67"><li class="reference__author" id="el-li-283" data-ref="ac853476-ac37-4aec-8e09-ef07e1133189" data-id="el-li-283" data-item-num="1">Freeburne E.</li><li class="reference__author" id="el-li-284" data-ref="de71f5a3-a5f5-41cc-b39f-65e464c424b2" data-id="el-li-284" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="ab1e7ad7-5f9e-4393-a49b-052d7c50a192">2023</span><span class="reference__title" data-ref="6b44c700-0975-488e-ad8a-ea10d8f724a0">Spatial Single Cell Analysis of Proteins in 2D Human Gastruloids Using Iterative Immunofluorescence.</span><span class="reference__origin" data-ref="86766b77-6d55-47fb-bac6-7500df42af24"><i data-ref="bfa28c6e-6888-441f-b809-2faba33f5f8b">Curr Protoc</i> <strong data-ref="b16887e7-d5fa-4d64-bbf7-32b2d4b8a28e">3</strong>:e915</span><span class="reference__doi" data-ref="01deb76f-20f7-4c97-981a-6b931c6c19f3"></span></li><li class="reference-list__item" id="c64" data-ref="0c62b32c-3afa-40f1-b018-c76c4c6a9cac" data-id="c64"><span class="reference__label" data-ref="974e3324-09b2-4e39-ba8a-e39d49b98753">64.</span><ol class="reference__authors_list" id="el-ol-68" data-ref="872370bf-8524-49ff-8a52-231ebabb7706" data-id="el-ol-68"><li class="reference__author" id="el-li-286" data-ref="77853f44-1a70-482a-b324-5e6ed0685aed" data-id="el-li-286" data-item-num="1">Sommer C.</li><li class="reference__author" id="el-li-287" data-ref="8f2637a8-9238-4cf1-84f7-564801090da6" data-id="el-li-287" data-item-num="2">Straehle C.</li><li class="reference__author" id="el-li-288" data-ref="52e54db9-7299-4efe-b535-c4d3205b8969" data-id="el-li-288" data-item-num="3">Kothe U.</li><li class="reference__author" id="el-li-289" data-ref="efaa8bfc-1e5b-4b20-b294-a5c6023e9e28" data-id="el-li-289" data-item-num="4">Hamprecht F. A.</li></ol><span class="reference__authors_list_suffix" data-ref="2f7febf0-1125-4afa-97b3-8bcac2b1c0b6">2011</span><span class="reference__title" data-ref="8b3b96f6-2540-48a6-93e8-0715ab8f6930">Ilastik: Interactive learning and segmentation toolkit.</span><span class="reference__origin" data-ref="335760c1-57ad-412e-a472-6b3d58849d66">In: 2011 8th IEEE International Symposium on Biomedical Imaging (ISBI 2011). pp.&nbsp;230‑233</span><span class="reference__doi" data-ref="2ad9bdc1-c5dc-4891-bd00-6836ece38986"><a href="https://doi.org/10.1109/ISBI.2011.5872394" class="reference__doi_link" data-ref="59533caa-79af-43e1-9a8d-d36482f5f9c8">https://doi.org/10.1109/ISBI.2011.5872394</a></span></li><li class="reference-list__item" id="c65" data-ref="4803d647-6ae3-48c1-9c58-9982159579e5" data-id="c65"><span class="reference__label" data-ref="7884ebcc-a262-4c1c-971d-411c89d75276">65.</span><ol class="reference__authors_list" id="el-ol-69" data-ref="cedddf56-12fd-49df-ab51-21ce594412ff" data-id="el-ol-69"><li class="reference__author" id="el-li-291" data-ref="b52f9f17-9349-4a88-80a3-76f709b1f605" data-id="el-li-291" data-item-num="1">Stringer C.</li><li class="reference__author" id="el-li-292" data-ref="2aed4b2b-74f7-43fb-bc08-815df3f17eee" data-id="el-li-292" data-item-num="2">Wang T.</li><li class="reference__author" id="el-li-293" data-ref="974ef2ae-9f76-4414-b55b-127509e52230" data-id="el-li-293" data-item-num="3">Michaelos M.</li><li class="reference__author" id="el-li-294" data-ref="9f6fe8b5-648e-4e8a-b3a9-f3fec5d383ea" data-id="el-li-294" data-item-num="4">Pachitariu M</li></ol><span class="reference__authors_list_suffix" data-ref="7a61637f-bb05-4ecc-9d37-fb1a64f60cbd">2021</span><span class="reference__title" data-ref="62c356c5-400c-484d-a262-dd5bf2f51772">Cellpose: a generalist algorithm for cellular segmentation.</span><span class="reference__origin" data-ref="ff3129d4-fcfe-4f7c-8935-5c763f19af60"><i data-ref="4a852cff-51a6-4735-90bb-6350002a3108">Nat. Methods</i> <strong data-ref="772cb512-5a6e-42ad-b571-9ab16d6362b2">18</strong>:100‑106</span><span class="reference__doi" data-ref="e61f970a-618a-4c82-befa-91c4a296cdb4"></span></li><li class="reference-list__item" id="c66" data-ref="7e795b7d-9acb-4e58-89fd-cf705d80e954" data-id="c66"><span class="reference__label" data-ref="06ad0473-dfd0-4b0b-8ff4-81ae9a7a8279">66.</span><ol class="reference__authors_list" id="el-ol-70" data-ref="b5c89286-9913-4faa-bcce-14c27cb1503c" data-id="el-ol-70"><li class="reference__author" id="el-li-296" data-ref="aff7d9a4-7928-4be9-a3c0-8f40c3450c48" data-id="el-li-296" data-item-num="1">Wolf F. A.</li><li class="reference__author" id="el-li-297" data-ref="57c496f9-d38c-4b49-9860-771024bb420c" data-id="el-li-297" data-item-num="2">Angerer P.</li><li class="reference__author" id="el-li-298" data-ref="cda20429-c9d2-4840-917f-d1a5636905b4" data-id="el-li-298" data-item-num="3">Theis F. J</li></ol><span class="reference__authors_list_suffix" data-ref="fae678cd-8d3e-4f04-bbcd-67a7da23f387">2018</span><span class="reference__title" data-ref="7ebbd762-1fd4-43e8-8823-beda5ca6774e">SCANPY: large-scale single-cell gene expression data analysis.</span><span class="reference__origin" data-ref="807a14c7-6a79-4e80-ab03-146261b452e5"><i data-ref="d025f391-38b7-4ca9-b2c6-fb66f63669a9">Genome Biol</i> <strong data-ref="5b053b33-526a-48a3-abb0-a231a915e286">19</strong>:15</span><span class="reference__doi" data-ref="aaa4af65-82f9-497d-aa5e-6081d8205a5c"></span></li><li class="reference-list__item" id="c67" data-ref="3af83f65-07e6-49a6-953b-85b8313808bd" data-id="c67"><span class="reference__label" data-ref="f1a74e9a-dc3e-41ae-8475-80ba6a037c88">67.</span><ol class="reference__authors_list" id="el-ol-71" data-ref="6c3a9ee9-3e6c-4bb4-b601-89b345a9ed51" data-id="el-ol-71"><li class="reference__author" id="el-li-300" data-ref="6cabd969-66fa-451a-b7e8-90c3e7ab2be9" data-id="el-li-300" data-item-num="1">Gayoso A.</li><li class="reference__author" id="el-li-301" data-ref="61848943-b982-421a-82e9-4bdf86e149f5" data-id="el-li-301" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="cb286899-51ea-415a-ab97-1bc9d01955bc">2022</span><span class="reference__title" data-ref="6f50828e-d034-4b4e-bd56-19d9413f9cda">A Python library for probabilistic analysis of single-cell omics data.</span><span class="reference__origin" data-ref="d2c20056-1b03-497f-b2f8-1f5a3fb700d7"><i data-ref="6691ebfb-e31e-4025-8b3f-5e4e4063e2e6">Nat Biotechnol</i> <strong data-ref="67acb0c1-eec1-49d4-a468-b8bb7d02b985">40</strong>:163‑166</span><span class="reference__doi" data-ref="c874d5eb-e12d-44f3-9e3f-6251e69f2019"></span></li><li class="reference-list__item" id="c68" data-ref="60fc8186-6a09-437c-9886-709ffc6ca7bd" data-id="c68"><span class="reference__label" data-ref="85ca53c9-4d83-4cd7-9d93-2624fdf0f5d2">68.</span><ol class="reference__authors_list" id="el-ol-72" data-ref="0279ff91-1c5c-4bf3-8839-b653a20ebe68" data-id="el-ol-72"><li class="reference__author" id="el-li-303" data-ref="e53889cf-6c72-4556-9bd5-2e7390f7eda2" data-id="el-li-303" data-item-num="1">van Dijk D.</li><li class="reference__author" id="el-li-304" data-ref="3853d7cd-6b07-4539-8d2e-2afe5f33ad8d" data-id="el-li-304" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="0d894cae-c721-40fa-8a56-a38abccddd92">2018</span><span class="reference__title" data-ref="480fdfb3-1f45-4549-9bac-99d15766fcd4">Recovering Gene Interactions from Single-Cell Data Using Data Diffusion.</span><span class="reference__origin" data-ref="ba0b8f50-fa18-498e-8f12-90a5e34e3b1f"><i data-ref="ff8dde2e-b2a6-4ec6-8381-f5186c528819">Cell</i> <strong data-ref="671ffe19-bd6b-4aa8-9b24-5b48cf15d78c">174</strong>:716‑729.e27</span><span class="reference__doi" data-ref="337402bd-2350-411c-a99f-ff21a35a09ba"></span></li><li class="reference-list__item" id="c69" data-ref="f607a348-7ec8-4b38-b6fb-520ef16749c9" data-id="c69"><span class="reference__label" data-ref="677fa184-d8ec-4280-8f37-fa7abbd3535e">69.</span><ol class="reference__authors_list" id="el-ol-73" data-ref="b59beb30-4577-4870-b5ff-5f0f6997bdfa" data-id="el-ol-73"><li class="reference__author" id="el-li-306" data-ref="7d86b5e4-cfb3-4b21-8809-7ff9172c4d43" data-id="el-li-306" data-item-num="1">Street K.</li><li class="reference__author" id="el-li-307" data-ref="736c227e-248c-40a0-8646-b2c62bc2b1f5" data-id="el-li-307" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="c16926e0-9fce-4283-9967-94f872ec6da4">2018</span><span class="reference__title" data-ref="54ac0d9b-2a4f-48c1-a72c-6e15f19633d1">Slingshot: cell lineage and pseudotime inference for single-cell transcriptomics.</span><span class="reference__origin" data-ref="4cea2ca8-7e55-4cdd-ad67-9203a50a949c"><i data-ref="78fbe736-267e-458c-ace4-0d1e2bbf47b9">BMC Genomics</i> <strong data-ref="ae82213a-8917-48da-acf8-848f16079af4">19</strong>:477</span><span class="reference__doi" data-ref="80d1c04f-51c8-4841-afdb-c2517b6ec91e"></span></li><li class="reference-list__item" id="c70" data-ref="14e71c1c-fdbb-44c2-bac6-ac7282b2dcaa" data-id="c70"><span class="reference__label" data-ref="637ca270-26c7-46ea-a5b9-c3fba283e0eb">70.</span><ol class="reference__authors_list" id="el-ol-74" data-ref="e9cd7fd3-68d9-4232-8f50-02679ccf915e" data-id="el-ol-74"><li class="reference__author" id="el-li-309" data-ref="ae971ebb-0c45-43ca-820f-41b2ca605dc6" data-id="el-li-309" data-item-num="1">Van den Berge K</li><li class="reference__author" id="el-li-310" data-ref="de3e1672-ab40-4d33-acc0-da0a39087dc0" data-id="el-li-310" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="cae2a6b0-dcab-41c2-9f3f-47111bc33685">2020</span><span class="reference__title" data-ref="4680f1d8-d93e-45f6-b453-c15061813886">Trajectory-based differential expression analysis for single-cell sequencing data.</span><span class="reference__origin" data-ref="5a0bbf15-84b6-46e4-8942-09ee3c4f71dc"><i data-ref="812f80ea-5c5b-45be-bfef-f0bf544511bc">Nat Commun</i> <strong data-ref="d86f430f-72d2-4875-9be4-64ba2e933643">11</strong>:1201</span><span class="reference__doi" data-ref="c1950621-2569-4a83-bd17-af55c7ffc02b"></span></li><li class="reference-list__item" id="c71" data-ref="40be6ce0-ff47-44be-bd2f-4b3fc84188f5" data-id="c71"><span class="reference__label" data-ref="140d0bb0-1b56-4ab1-9f99-e93d74e1e206">71.</span><ol class="reference__authors_list" id="el-ol-75" data-ref="c6dfbabe-c994-4de5-8a36-bbc81f1563c1" data-id="el-ol-75"><li class="reference__author" id="el-li-312" data-ref="dd8cfaa8-bd48-46c5-b812-723eb2c6f285" data-id="el-li-312" data-item-num="1">Patro R.</li><li class="reference__author" id="el-li-313" data-ref="e4e622c5-d643-4429-a9a2-57330e8765aa" data-id="el-li-313" data-item-num="2">Duggal G.</li><li class="reference__author" id="el-li-314" data-ref="8ccee559-be16-4b55-b066-6cf23fe5f3dc" data-id="el-li-314" data-item-num="3">Love M. I.</li><li class="reference__author" id="el-li-315" data-ref="237e1b76-f56e-4f42-9b05-5d1776f54aca" data-id="el-li-315" data-item-num="4">Irizarry R. A.</li><li class="reference__author" id="el-li-316" data-ref="8da0f970-0f78-49e7-ab09-593a19d475a4" data-id="el-li-316" data-item-num="5">Kingsford C</li></ol><span class="reference__authors_list_suffix" data-ref="6080315f-9a9a-4491-b4bd-c22c998fe80b">2017</span><span class="reference__title" data-ref="7ac3f0f6-ff71-4af7-8e0e-c307b83c394d">Salmon provides fast and bias-aware quantification of transcript expression.</span><span class="reference__origin" data-ref="1891dee3-054d-4dc3-b6bf-0bb30d08711f"><i data-ref="d6e6317d-514d-4ed5-99c7-b70da3c03b5b">Nat Methods</i> <strong data-ref="e96a766b-8ef9-42ce-8eed-cc47ae8e5707">14</strong>:417‑419</span><span class="reference__doi" data-ref="45af79c3-ae7b-4241-a240-814cb7137738"></span></li><li class="reference-list__item" id="c72" data-ref="28302382-2551-4e4a-a579-58d76ecb26e0" data-id="c72"><span class="reference__label" data-ref="15379fff-af2f-41a6-bf9e-1240a261a97c">72.</span><ol class="reference__authors_list" id="el-ol-76" data-ref="268077d9-fe55-44a5-9140-b78154483ba8" data-id="el-ol-76"><li class="reference__author" id="el-li-318" data-ref="b6d3a45d-4743-4c5c-9c79-7d49cab974a4" data-id="el-li-318" data-item-num="1">Eswarakumar V. P.</li><li class="reference__author" id="el-li-319" data-ref="b27ff7a6-7600-4aab-b45b-6243e7b2cf18" data-id="el-li-319" data-item-num="2">Lax I.</li><li class="reference__author" id="el-li-320" data-ref="c0e12817-cea2-4748-b117-27fa5c8e5e72" data-id="el-li-320" data-item-num="3">Schlessinger J</li></ol><span class="reference__authors_list_suffix" data-ref="303be9fe-80ba-4af9-93f6-af9f632fae4e">2005</span><span class="reference__title" data-ref="9c56e27a-a719-40f7-8ebc-c04cf0b3f1f9">Cellular signaling by fibroblast growth factor receptors.</span><span class="reference__origin" data-ref="3c1a75be-bc90-420f-aba1-e27d49103206"><i data-ref="095c98ae-d416-40f2-bba6-f8e2f3e635dd">Cytokine Growth Factor Rev</i> <strong data-ref="9e202028-3510-46ed-891b-f05439204ebe">16</strong>:139‑149</span><span class="reference__doi" data-ref="a384f693-666e-423d-9842-7082be9403be"></span></li><li class="reference-list__item" id="dataref1" data-ref="34c121fd-0a7e-4401-8bd9-3f1481ea78b6" data-id="dataref1"><ol class="reference__authors_list" id="el-ol-77" data-ref="76ead427-f4ef-4d7a-bc8e-0b05ace87e90" data-id="el-ol-77"><li class="reference__author" id="el-li-322" data-ref="9bea0fb0-a1ee-4d30-98c6-7d46e2477eb3" data-id="el-li-322" data-item-num="1">Jo K</li><li class="reference__author" id="el-li-323" data-ref="b40a4f50-17c0-41a9-8a17-a6167cc8ec38" data-id="el-li-323" data-item-num="2">Liu ZY</li><li class="reference__author" id="el-li-324" data-ref="45f56528-e145-468c-b5f5-fdc2726f9cc8" data-id="el-li-324" data-item-num="3">Patel G</li><li class="reference__author" id="el-li-325" data-ref="60cec438-29f0-47e9-9b25-c7d2bf2d1551" data-id="el-li-325" data-item-num="4">Yu Z</li><li class="reference__author" id="el-li-326" data-ref="59ddab95-ed1f-42ee-ad40-d8dbe7b65196" data-id="el-li-326" data-item-num="5">Yao L</li><li class="reference__author" id="el-li-327" data-ref="f6678ddd-77dd-4e84-9c17-763e9b1b6623" data-id="el-li-327" data-item-num="6">Teague S</li><li class="reference__author" id="el-li-328" data-ref="c1e61027-8bd4-4dca-8a30-b427034a0f8f" data-id="el-li-328" data-item-num="7">Johnson C</li><li class="reference__author" id="el-li-329" data-ref="8b7f6408-c50e-468e-b653-9af8d59c842a" data-id="el-li-329" data-item-num="8">Spence J</li><li class="reference__author" id="el-li-330" data-ref="75ab3e92-1f12-4ea3-9bd1-669cde2a258a" data-id="el-li-330" data-item-num="9">Heemskerk I</li></ol><span class="reference__authors_list_suffix" data-ref="e0876ee1-e6c2-4ea9-b18f-d08975a7bf09">2024</span><span class="reference__title" data-ref="a2535415-dcec-4a02-9ad9-e9e887a43cea">Endogenous FGF gradients are required for cell fate patterning in 2D human gastruloids.</span><span class="reference__origin" data-ref="689613fc-8509-4962-9d04-9b511678a4e0">NCBI Gene Expression Omnibus. ID ç</span><span class="reference__doi" data-ref="be84647c-24e3-4768-b57b-a3e0055ed387"><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE271604" class="reference__external_link" data-ref="e7d078b3-b15c-4456-af81-055b30a6229c">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE271604</a></span></li><li class="reference-list__item" id="dataref2" data-ref="a8f498d2-d7cd-4deb-97a3-88322b296a32" data-id="dataref2"><ol class="reference__authors_list" id="el-ol-78" data-ref="4d77d9d9-9c5c-4822-bfdb-3d1baf80d914" data-id="el-ol-78"><li class="reference__author" id="el-li-332" data-ref="3070a868-7eac-4f2d-9892-5a57e96be9a1" data-id="el-li-332" data-item-num="1">Zhai J</li><li class="reference__author" id="el-li-333" data-ref="66bb6190-0bd7-410a-aadc-76beca6436c2" data-id="el-li-333" data-item-num="2">Guo J</li><li class="reference__author" id="el-li-334" data-ref="bc430a70-4e6e-4611-8e82-28251549efcd" data-id="el-li-334" data-item-num="3">Wan H</li><li class="reference__author" id="el-li-335" data-ref="1e6b5fe9-1efb-43d8-967f-fb213096fd6e" data-id="el-li-335" data-item-num="4">Qi L</li><li class="reference__author" id="el-li-336" data-ref="5ac0c1c0-28e6-4d3a-9fcd-aaedf5761db7" data-id="el-li-336" data-item-num="5">Liu L</li><li class="reference__author" id="el-li-337" data-ref="2b5ff099-2330-455b-984c-d2e2df52419e" data-id="el-li-337" data-item-num="6">Xiao Z</li><li class="reference__author" id="el-li-338" data-ref="f456dffd-57d2-474a-a2de-63917b31e4fb" data-id="el-li-338" data-item-num="7">Yan L</li><li class="reference__author" id="el-li-339" data-ref="7da05b64-cb34-4c21-b88e-55a62ed45723" data-id="el-li-339" data-item-num="8">Schmitz DA</li><li class="reference__author" id="el-li-340" data-ref="06335589-814e-4d89-9d72-08745f25618d" data-id="el-li-340" data-item-num="9">Xu Y</li><li class="reference__author" id="el-li-341" data-ref="cbfb0c3b-fc31-4212-9851-e78008c88e86" data-id="el-li-341" data-item-num="10">Yu D</li></ol><em data-ref="646ff06d-5cb7-4b02-93e0-0e09b71d3ae8"> et al.</em><span class="reference__authors_list_suffix" data-ref="c8141752-1d35-4dad-82ae-a6fdd0613a78">2022</span><span class="reference__title" data-ref="af2d9af8-cec3-417a-9550-55fb6175042f">Primate gastrulation and early organogenesis at single-cell resolution.</span><span class="reference__origin" data-ref="796e8a35-aae4-4770-a6ba-edeeb30531d4">NCBI Gene Expression Omnibus. ID GSE193007</span><span class="reference__doi" data-ref="a316451a-2f18-45dc-be76-b1e9e6d0b475"><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE193007" class="reference__external_link" data-ref="21981c34-769c-4180-8418-e2a1f2569748">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE193007</a></span></li><li class="reference-list__item" id="dataref3" data-ref="e59a9edc-c430-4288-a9e6-2b9d31e7ae19" data-id="dataref3"><ol class="reference__authors_list" id="el-ol-79" data-ref="22c7b628-8030-4d6f-99fe-5d168553d6d4" data-id="el-ol-79"><li class="reference__author" id="el-li-343" data-ref="7551b6c0-6fb2-43a6-8c8e-e1824cf7279c" data-id="el-li-343" data-item-num="1">Pijuan-Sala B</li><li class="reference__author" id="el-li-344" data-ref="c5bdc1c9-9698-4630-85c2-5c5a3586c1b4" data-id="el-li-344" data-item-num="2">Griffiths JA</li><li class="reference__author" id="el-li-345" data-ref="c49ff661-6ac8-4f0b-9f20-942dced11231" data-id="el-li-345" data-item-num="3">Guibentif C</li><li class="reference__author" id="el-li-346" data-ref="2cc22c4e-c5df-4edc-989d-ef4caa3ce2dc" data-id="el-li-346" data-item-num="4">Hiscock TW</li><li class="reference__author" id="el-li-347" data-ref="867202ae-1109-40d3-81ba-e5ae53a546b8" data-id="el-li-347" data-item-num="5">Jawaid W</li><li class="reference__author" id="el-li-348" data-ref="446b6c47-7360-499a-a51e-da2ae52c23e9" data-id="el-li-348" data-item-num="6">Calero-Nieto FJ</li><li class="reference__author" id="el-li-349" data-ref="3a6279f0-9927-4623-bf1f-e991516e0cff" data-id="el-li-349" data-item-num="7">Mulas C</li><li class="reference__author" id="el-li-350" data-ref="e683e3db-6914-4c0f-8553-c9a162888b9d" data-id="el-li-350" data-item-num="8">Ibarra-Soria X</li><li class="reference__author" id="el-li-351" data-ref="8030c878-95f6-44ee-9dc1-b1f05a46af23" data-id="el-li-351" data-item-num="9">Tyser RCV</li><li class="reference__author" id="el-li-352" data-ref="70d50f5c-d22d-4ee3-98a1-c4b61d5beb6b" data-id="el-li-352" data-item-num="10">Ho DLL</li></ol><em data-ref="04b7bffd-bd98-44f9-829a-84675c4145ad"> et al.</em><span class="reference__authors_list_suffix" data-ref="ba51caf1-9c2e-4782-93e2-c4f6829ba8b6">2018</span><span class="reference__title" data-ref="30f8a8e3-3cf7-4d50-b1f0-7623073b4d4e">Timecourse single-cell RNAseq of whole mouse embryos harvested between days 6.5 and 8.5 of development.</span><span class="reference__origin" data-ref="e9344f7d-b39d-45ec-8b62-9d8ed211da93">ArrayExpress. ID E-MTAB-6967</span><span class="reference__doi" data-ref="4c9f1131-7f70-4cb2-9929-11a7f878228f"><a href="https://www.ebi.ac.uk/biostudies/arrayexpress/studies/E-MTAB-6967" class="reference__external_link" data-ref="35d92767-fa62-4b09-95ba-c1232db6dba9">https://www.ebi.ac.uk/biostudies/arrayexpress/studies/E-MTAB-6967</a></span></li><li class="reference-list__item" id="dataref4" data-ref="f921e958-5cc0-4416-9213-a949ba5bfab2" data-id="dataref4"><ol class="reference__authors_list" id="el-ol-80" data-ref="bdba8e03-cb51-4cc8-b9f0-f2ae62dd1ee4" data-id="el-ol-80"><li class="reference__author" id="el-li-354" data-ref="3d745dc9-55d3-4415-bc7f-3bbf7ac72ce0" data-id="el-li-354" data-item-num="1">Tyser RCV</li><li class="reference__author" id="el-li-355" data-ref="b26070a2-7197-45f4-bd01-c6903a64cbc4" data-id="el-li-355" data-item-num="2">Mahammadov E</li><li class="reference__author" id="el-li-356" data-ref="1b506ce6-fb78-4b9a-89ae-acaa653ff230" data-id="el-li-356" data-item-num="3">Nakanoh S</li><li class="reference__author" id="el-li-357" data-ref="bb6e0f74-2b74-4f32-9afb-fbca53f5c459" data-id="el-li-357" data-item-num="4">Vallier L</li><li class="reference__author" id="el-li-358" data-ref="620f1ea1-7534-4f8d-9510-9369b3ea0016" data-id="el-li-358" data-item-num="5">Scialdone A</li><li class="reference__author" id="el-li-359" data-ref="8e0f1567-7ddc-4542-be30-212b85f423a2" data-id="el-li-359" data-item-num="6">Srinivas S</li></ol><span class="reference__authors_list_suffix" data-ref="67e526cd-f584-4af6-8d47-a82268c54964">2021</span><span class="reference__title" data-ref="a0162004-0947-4566-b745-102703395c9f">A spatially resolved single cell atlas of human gastrulation.</span><span class="reference__origin" data-ref="985f310b-cfb8-43c0-b417-6a81e4157d5d">ArrayExpress. ID E-MTAB-9388</span><span class="reference__doi" data-ref="0fc053c7-f2d4-4fdc-a8c3-7e49838dfa0b"><a href="https://www.ebi.ac.uk/biostudies/arrayexpress/studies/E-MTAB-9388" class="reference__external_link" data-ref="fa0d425b-7514-4f26-93aa-7c782a939dd0">https://www.ebi.ac.uk/biostudies/arrayexpress/studies/E-MTAB-9388</a></span></li><li class="reference-list__item" id="dataref5" data-ref="93234cd8-ff68-4266-8fc3-b326837aaf1d" data-id="dataref5"><ol class="reference__authors_list" id="el-ol-81" data-ref="ce379475-1b6e-45c2-a14c-105db149e8d4" data-id="el-ol-81"><li class="reference__author" id="el-li-361" data-ref="c8991010-05cb-4438-8873-056be3833616" data-id="el-li-361" data-item-num="1">Koh PW</li><li class="reference__author" id="el-li-362" data-ref="a87a76f7-bf2d-405d-9ab2-ecb3531ae41d" data-id="el-li-362" data-item-num="2">Sinha R</li><li class="reference__author" id="el-li-363" data-ref="76d8fbde-738d-4fbd-abf9-00873ff80242" data-id="el-li-363" data-item-num="3">Barkal AA</li><li class="reference__author" id="el-li-364" data-ref="0cb2ac59-14a8-4b13-98d4-56d2a62c27b6" data-id="el-li-364" data-item-num="4">Morganti RM</li><li class="reference__author" id="el-li-365" data-ref="d7a8576c-e7cd-4638-944b-c73a10dc125d" data-id="el-li-365" data-item-num="5">Chen A</li><li class="reference__author" id="el-li-366" data-ref="d8e32811-975c-4638-a492-b4dde93d0beb" data-id="el-li-366" data-item-num="6">Weissman IL</li><li class="reference__author" id="el-li-367" data-ref="96ee317b-4e4a-4c92-bf9f-8255a4269dc2" data-id="el-li-367" data-item-num="7">Ang LT</li><li class="reference__author" id="el-li-368" data-ref="72b2c546-03e0-45cf-855d-5d26e6b1eda1" data-id="el-li-368" data-item-num="8">Kundaje A</li><li class="reference__author" id="el-li-369" data-ref="347dd967-98ba-441e-80a4-5dce543f152a" data-id="el-li-369" data-item-num="9">Loh KM</li></ol><span class="reference__authors_list_suffix" data-ref="17fc94e4-7685-48c4-9955-4278e184800e">2016</span><span class="reference__title" data-ref="9af24f56-1c95-4868-b2a4-7edeb8db2b8c">In vitro cultured H7 human embryonic stem cells (WiCell) and H7-derived downstream early mesoderm progenitors.</span><span class="reference__origin" data-ref="9337dab1-201e-4988-9d64-ef56c9d59a5b">NCBI Gene Expression Omnibus. ID GSE85066</span><span class="reference__doi" data-ref="7bf5e856-e63a-43df-8ec6-df5b38fbde88"><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE85066" class="reference__external_link" data-ref="451ac053-802a-4a6d-ae00-a74003e7a7a9">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE85066</a></span></li></ul></section></section><section id="peer-reviews" data-ref="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2" data-id="peer-reviews"><h1 id="el-h1-11" data-ref="193c330f-0fe9-48f4-9670-4f5715461a45" data-id="el-h1-11" data-break-after="avoid">Peer reviews</h1><section id="reviews" class="review-content" data-ref="1d625bc5-02ad-4fce-91fd-d9baddac7f30" data-id="reviews" data-previous-break-after="avoid"><h2 id="el-h2-25" data-ref="14d8ad61-c16d-404d-b5e0-e55397de1c1c" data-id="el-h2-25" data-break-after="avoid" data-following="following-ff2c00ba-51a1-482d-9afa-f5a370dcfa50">Reviewer #1 (Public review):</h2><div class="review-content__body" id="el-div-16" data-ref="9e7f3e80-f94e-4b3e-84c6-91a98247b329" data-id="el-div-16" data-previous-break-after="avoid"><p id="el-p-90" data-ref="849b05a8-6ea2-4a55-b7be-a2523fd07053" data-id="el-p-90">Summary:</p><p id="el-p-91" data-ref="d6c2e0e5-5d01-4544-a4f3-88d3bd029330" data-id="el-p-91">This is an interesting study on the role of FGF signaling in the induction of primitive streak like-cells (PS-LC) in human 2D-gastruloids. The authors use a previously characterized standard culture that generates a ring of PS-LCs (TBXT+) and correlate this with pERK staining. A requirement for FGF signaling in TBXT induction is demonstrated via pharmacological inhibition of MEK and FGFR activity. A second set of culture conditions (with no exogenous FGFs) suggests that endogenous FGFs are required for pERK and TBXT induction. The authors then characterize, via scRNA-seq, various components of the FGF pathway (genes for ligand, receptors, ERK regulators, HSPG regulation). They go on to characterize the pFGFR1, receptor isoforms and polarized localization of this receptor. Finally, they perform FGF4 inhibition and use a cell line with a limited FGF17 inactivation (heterozygous null) and show that loss of these FGFs reduce PS-LC and derivative cell types.</p><p id="el-p-92" data-ref="8580f9e9-1ccc-4ce7-ad4d-18f26174e71e" data-id="el-p-92">Strengths:</p><p id="el-p-93" data-ref="f21ac7a3-d80c-4947-9195-3d163df0e858" data-id="el-p-93">(1) As the authors point out, the role of FGF signaling in gastrulation is less well understood than other signaling pathways. Hence this is a valuable contribution to that field.</p><p id="el-p-94" data-ref="2e1e0a41-bba1-4257-8c80-3d29ab479beb" data-id="el-p-94">(2) The FGF4 and FGF17 loss-of-function experiments in Figure 5 are very intriguing. This is especially so given the intriguing observation that these FGFs appear to be dominating in this model of human gastrulation, in contrast to what FGFs dominate in mice, chick and frogs.</p><p id="el-p-95" data-ref="8fe8f4f3-3469-4ac9-8417-0a3735f4cfd6" data-id="el-p-95">(3) In general this paper is valuable as a further development of the Human gastruloid system and the role of FGF signaling in the induction of PS-CLs. The wide net that the authors cast in characterizing FGF ligand gene, receptor isoforms, and downstream components provides a foundation for future work. As the authors write near the beginning of the Discussion "Many questions remain."</p><p id="el-p-96" data-ref="9286a75a-188e-4eda-b796-c9081228f7bf" data-id="el-p-96">Weaknesses:</p><p id="el-p-97" data-ref="2dfddacf-f85b-4da4-be9e-72c94e9f8aca" data-id="el-p-97">(1) FGFs are cell survival factors in various aspects of development. The authors fail to address cell death due to loss of FGF signaling in any of their experiments. For example, in Figure 1E (which requires statistical analysis) and 1G (the bottom FGFRi row), there appears to be a significant amount of cell loss. Is this due to cell death? The authors should address the question of whether the role of FGF/ERK signaling is to keep the cells alive.</p><p id="el-p-98" data-ref="fb087b52-a9fd-436d-b36d-826aca0fdcbd" data-id="el-p-98">(2) Regarding the sparse cells in 1G, is there a reduction in cell number only with FGFRi and not MEKi? Is this reproducible? Gattiglio et al (Development, 2023, PMID: 37530863) present data supporting a "community effect" in the FGF-induced mesoderm differentiation of mouse embryonic stem cells. Could a community effect be at play in this human system (especially given the images in the bottom row of 1G). If the authors don't address this experimentally they should at least address the ideas in Gattoglio et al.</p><p id="el-p-99" data-ref="b9647971-fe6d-471b-86d3-85171d0379a4" data-id="el-p-99">(3) Do the FGF4 and FGF17 LOF experiments in Figure 5 affect cell number like FGFRi in Figure 1? Why examine PS-LC induction only in FGF17 heterozygous cells and not homozygous FGF17 nulls?</p><p id="el-p-100" data-ref="579800b1-0ea7-4d2b-a856-f6700be05265" data-id="el-p-100">(4) The idea that FGF8 plays a dominant role during gastrulation of other species but not humans is so intriguing it warrants deeper testing. The authors dismiss FGF8 because its mRNA "...levels always remained low." (line 363) as well as the data published in Zhai et al (PMID: 36517595) and Tyser et al (PMID: 34789876). But there are cases in mouse development where a gene was expressed at levels so low, it might be dismissed, and yet LOF experiments revealed it played a role or even was required in a developmental process. The authors should consider FGF8 inhibition or inactivation to explore its potential role, despite its low levels of expression.</p><p id="el-p-101" data-ref="ad63ee8a-2dc1-49b7-93c7-ca01fc21578b" data-id="el-p-101">(5) Redundancy is a common feature in FGF genetics. What is the effect of inhibiting FGF4 in FGF17 LOF cells?</p><p id="el-p-102" data-ref="0c246a58-4920-47cc-a056-c4fc0a291558" data-id="el-p-102">(6) I suggest stating that the authors take more caution describing FGF gradients. For example, in one Results heading they write "Endogenous FGF4 and FGF17 gradients underly the ERK activity pattern.", implying an FGF protein gradient. However, they only present data for FGF mRNA , not protein. This issue would be clarified if they used proper nomenclature for gene, mRNA (italics) and protein (no italics) throughout the paper.</p><p id="el-p-103" data-ref="bc9c1eb7-b89e-403e-8726-5f80c0588b03" data-id="el-p-103">Comments on revisions:</p><p id="el-p-104" data-ref="9c978591-c7ee-445a-a5de-abcf436adc0f" data-id="el-p-104">The authors have addressed my concerns.</p></div><div class="descriptors" id="el-div-17" data-ref="2fbbec87-25b5-46a4-8029-0f6cdd37d4f5" data-id="el-div-17"><ul class="descriptors__identifiers" id="el-ul-7" data-ref="cb2fd904-29c1-4a57-975b-53e5104f98f5" data-id="el-ul-7"><li class="descriptors__identifier" id="el-li-370" data-ref="1d7094d2-3be6-4d9b-86ca-f71212a0c466" data-id="el-li-370"><a href="https://doi.org/10.7554/eLife.101224.2.sa3" data-ref="31113fb7-79fe-4e7e-8132-5b1289fd2e9a">https://doi.org/10.7554/eLife.101224.2.sa3</a></li></ul></div></section><section id="reviews" class="review-content" data-ref="482c4127-9b84-4bff-82d8-e461ca58e221" data-id="reviews"><h2 id="el-h2-26" data-ref="db76f787-2f9e-4cc0-b1ae-f0837f21cd2e" data-id="el-h2-26" data-break-after="avoid">Reviewer #2 (Public review):</h2><div class="review-content__body" id="el-div-18" data-ref="580a7f89-7649-489c-9c78-0415aa3a0547" data-id="el-div-18" data-previous-break-after="avoid"><p id="el-p-105" data-ref="9787a0dc-c4e9-4dcb-835b-4b05b22822f6" data-id="el-p-105">Summary:</p><p id="el-p-106" data-ref="1ab22ab0-6d1e-4910-a020-87edd808d72b" data-id="el-p-106">The role of FGFs in embryonic development and stem cell differentiation has remained unclear due to its complexity. In this study, the authors utilized a 2D human stem cell-based gastrulation model to investigate the functions of FGFs. They discovered that FGF-dependent ERK activity is closely linked to the emergence of primitive streak cells. Importantly, this 2D model effectively illustrates the spatial distribution of key signaling effectors and receptors by correlating these markers with cell fate markers, such as T and ISL1. Through inhibition and loss-of-function studies, they further corroborated the needs of FGF ligands. Their data shows that FGFR1 is the primary receptor, and FGF2/4/17 are the key ligands for primitive streak development, which aligns with observations in primate embryos. Additional experiments revealed that the reduction of FGF4 and FGF17 decreases ERK activity.</p><p id="el-p-107" data-ref="f1d8ccd9-4a25-4d4d-9865-0dc32b022d5c" data-id="el-p-107">Strengths:</p><p id="el-p-108" data-ref="66400d11-f5be-4b9d-8aea-f0b3b441af15" data-id="el-p-108">This study provides comprehensive data and improves our understanding of the role of FGF signaling in primate primitive streak formation. The authors provide new insights related to the spatial localization of the key components of FGF signaling and attempt to reveal the temporal dynamics of the signal propagation and cell fate decision, which has been challenging.</p></div><div class="descriptors" id="el-div-19" data-ref="b8dbf020-daeb-4f6a-81e7-996bf5bf927b" data-id="el-div-19"><ul class="descriptors__identifiers" id="el-ul-8" data-ref="7d0cb99c-b700-49b6-8e07-1ac0bd6b213c" data-id="el-ul-8"><li class="descriptors__identifier" id="el-li-371" data-ref="d120bd51-8703-4cfc-bc37-6b25b2f35bbb" data-id="el-li-371"><a href="https://doi.org/10.7554/eLife.101224.2.sa2" data-ref="579d6b1c-73a6-48f8-9210-e806f605c3e1">https://doi.org/10.7554/eLife.101224.2.sa2</a></li></ul></div></section><section id="reviews" class="review-content" data-ref="4c2dabe6-69c1-4cd4-bedc-e4d78caf0677" data-id="reviews"><h2 id="el-h2-27" data-ref="5db120f2-5cc8-4908-8387-68bbeb4bfb22" data-id="el-h2-27" data-break-after="avoid">Reviewer #3 (Public review):</h2><div class="review-content__body" id="el-div-20" data-ref="6ac993fd-bd96-46e8-ba1d-674910bed89a" data-id="el-div-20" data-previous-break-after="avoid"><p id="el-p-109" data-ref="d836ec45-09be-4b88-935e-aa03225ba4c9" data-id="el-p-109">Jo and colleagues set out to investigate the origins and functions of localized FGF/ERK signaling for the differentiation and spatial patterning of primitive streak fates of human embryonic stem cells in a well-established micropattern system. They demonstrate that endogenous FGF signaling is required for ERK activation in a ring-domain in the micropatterns, and that this localized signaling is directly required for differentiation and spatial patterning of specific cell types. Through high-resolution microscopy and transwell assays, they show that cells receive FGF signals through basally localized receptors. Finally, the authors find that there is a requirement for exogenous FGF2 to initiate primitive streak-like differentiation, but endogenous FGFs, especially FGF4 and FGF17, fully take over at later stages.</p><p id="el-p-110" data-ref="c44e7a7b-19ee-430f-8e66-abe12fd79e6e" data-id="el-p-110">Even though some of the authors' findings - such as the localized expression of FGF ligands during gastrulation and the importance of FGF/ERK signaling for cell differentiation in the primitive streak - have been reported in model organisms before, this is one of the first studies to investigate the role of FGF signaling during primitive streak-like differentiation of human cells. In doing so, the paper reports a number of interesting and valuable observations, namely the basal localization of FGF receptors which mirrors that of BMP and Nodal receptors, as well as the existence of a positive feedback loop centered on FGF signaling that drives primitive-streak differentiation. In the revised version of their work, the authors have furthermore dissected the role of different FGFs through knockdown approaches. These experiments reveal discrete functions for different FGF genes in their system, as well as interesting differences between the role of specific FGFs in human compared to model systems.</p><p id="el-p-111" data-ref="74627652-564a-4329-84f7-d041fa7d1747" data-id="el-p-111">Comments on revisions:</p><p id="el-p-112" data-ref="12c9acd2-92b1-4a06-92ff-5a0459d238e4" data-id="el-p-112">The authors have appropriately addressed all comments and suggestions from the previous round of review. The only textual change that I would still like to suggest is to write explicitly in the main text corresponding to Fig. 1 that the mTESR1 medium used for these initial experiments already contains FGF. This is something that is probably known to experts in the field, but not necessarily to a broader readership.</p></div><div class="descriptors" id="el-div-21" data-ref="3c4001b3-f977-4d43-84ed-ae8e87ec6ac4" data-id="el-div-21"><ul class="descriptors__identifiers" id="el-ul-9" data-ref="8dfb07d5-b518-4f06-af6b-7eabb28ce8fd" data-id="el-ul-9"><li class="descriptors__identifier" id="el-li-372" data-ref="d556bbcc-a88a-4632-b65a-5ef95effd543" data-id="el-li-372"><a href="https://doi.org/10.7554/eLife.101224.2.sa1" data-ref="278574ec-50d4-4e15-af20-4893fc0f3867">https://doi.org/10.7554/eLife.101224.2.sa1</a></li></ul></div></section><section id="authorresponse" class="author-response" data-ref="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4" data-id="authorresponse"><h2 id="el-h2-28" data-ref="46fd2982-0a28-487a-958d-367ca72047bf" data-id="el-h2-28" data-break-after="avoid">Author response:</h2><div class="author-response__body" id="el-div-22" data-ref="0f8a4845-8525-4632-882f-9a92142367e9" data-id="el-div-22" data-previous-break-after="avoid"><p id="el-p-113" data-ref="6429f6bb-02a2-4398-a7ad-0ddaa9ffee93" data-id="el-p-113">The following is the authors’ response to the original reviews.</p><blockquote data-ref="4d267609-9d09-4287-87d4-8cda462eb822"><p id="el-p-114" data-ref="681677fa-9a39-4120-a4ac-69e50de31119" data-id="el-p-114"><strong data-ref="f3dcb14d-4aa8-4cf5-9070-6bbf14430d9f">Reviewer #1 (Public review):</strong></p><p id="el-p-115" data-ref="38db78e4-9347-4b5b-bbcd-84d274b11746" data-id="el-p-115">Summary:</p><p id="el-p-116" data-ref="3613efc2-7021-4002-89c3-93ad7ae26f7d" data-id="el-p-116">This is an interesting study on the role of FGF signaling in the induction of primitive streak-like cells (PS-LC) in human 2D-gastruloids. The authors use a previously characterized standard culture that generates a ring of PSLCs (TBXT+) and correlate this with pERK staining. A requirement for FGF signaling in TBXT induction is demonstrated via pharmacological inhibition of MEK and FGFR activity. A second set of culture conditions (with no exogenous FGFs) suggests that endogenous FGFs are required for pERK and TBXT induction. The authors then characterize, via scRNA-seq, various components of the FGF pathway (genes for ligands, receptors, ERK regulators, and HSPG regulation). They go on to characterize the pFGFR1, receptor isoforms, and polarized localization of this receptor. Finally, they perform FGF4 inhibition and use a cell line with a limited FGF17 inactivation (heterozygous null) and show that loss of these FGFs reduces PS-LC and derivative cell types.</p><p id="el-p-117" data-ref="3e51a978-af21-47dd-8a8b-b6483bd4d6f9" data-id="el-p-117">Strengths:</p><p id="el-p-118" data-ref="1300ec0a-cfba-4076-94ac-91d07f56c5fa" data-id="el-p-118">(1) As the authors point out, the role of FGF signaling in gastrulation is less well understood than other signaling pathways. Hence this is a valuable contribution to that field.</p><p id="el-p-119" data-ref="f0e7669f-e60e-41b1-a0f3-5c97ad1646da" data-id="el-p-119">(2) The FGF4 and FGF17 loss-of-function experiments in Figure 5 are very intriguing. This is especially so given the intriguing observation that these FGFs appear to be dominating in this model of human gastrulation, in contrast to what FGFs dominate in mice, chicks, and frogs.</p><p id="el-p-120" data-ref="b2705b6a-ae30-4eeb-95ed-0fe14328454b" data-id="el-p-120">(3) In general this paper is valuable as a further development of the Human gastruloid system and the role of FGF signaling in the induction of PS-CLs. The wide net that the authors cast in characterizing the FGF ligand gene, receptor isoforms, and downstream components provides a foundation for future work. As the authors write near the beginning of the Discussion "Many questions remain."</p></blockquote><p id="el-p-121" data-ref="cd3cef89-01c1-416d-a00d-f6250520b0b3" data-id="el-p-121">We thank the reviewer for these positive comments.</p><blockquote data-ref="43781384-61ae-4875-b1c1-1638509f7143"><p id="el-p-122" data-ref="5d5a37ac-49ea-405d-926b-5ee35d47362a" data-id="el-p-122">Weaknesses:</p><p id="el-p-123" data-ref="1a6157e6-f5e2-43d4-a80e-11bb2b2c06dc" data-id="el-p-123">(1) FGFs are cell survival factors in various aspects of development. The authors fail to address cell death due to loss of FGF signaling in their experiments. For example, in Figure 1E (which requires statistical analysis) and 1G (the bottom FGFRi row), there appears to be a significant amount of cell loss. Is this due to cell death? The authors should address the question of whether the role of FGF/ERK signaling is to keep the cells alive.</p></blockquote><p id="el-p-124" data-ref="ed126906-0e5d-4cb4-83cf-f943737515d5" data-id="el-p-124">Indeed, FGF also strongly affects cell survival and it is an interesting question to what extent this depends on ERK. Our manuscript focuses instead on the role of FGF/ERK signaling in cell fate patterning. As mentioned in our discussion, figure 1de show that doxycycline induced pERK leads to more TBXT+ cells than the control without restoring cell number, suggesting the role of FGF in controlling cell number is independent of the requirement for FGF/ERK in PS-LC differrentiation. To further support this, we have added data showing low doses of MEKi are sufficient to inhibit differentiation without affecting cell number (Supp. Fig. 1i).</p><p id="el-p-125" data-ref="c4ca3e10-54b2-4225-b54e-4b93cbd25bf1" data-id="el-p-125">To address the reviewers question regarding the cause of cell loss, we now stained for BrdU and cleaved Cas3 to assess proliferation and apoptosis in the presence and absence of MEK and FGFR inhibition (new Supp. Fig.</p><p id="el-p-126" data-ref="c2b8f7fe-30d0-40cd-bbe6-fcc07304c71c" data-id="el-p-126">1ef). This shows that the effect of these inhibitors on cell number is primarily due to a reduction in proliferation. We have also included statistical analysis in Fig.1e.</p><blockquote data-ref="1725e14d-51d6-4a44-938c-ab055d27352b"><p id="el-p-127" data-ref="32092e33-2f64-41a7-a7d2-24ec6a471cbc" data-id="el-p-127">(2) Regarding the sparse cells in 1G, is there a reduction in cell number only with FGFRi and not MEKi? Is this reproducible? Gattiglio et al (Development, 2023, PMID: 37530863) present data supporting a "community effect" in the FGF-induced mesoderm differentiation of mouse embryonic stem cells. Could a community effect be at play in this human system (especially given the images in the bottom row of 1G)? If the authors don't address this experimentally they should at least address the ideas in Gattoglio et al.</p></blockquote><p id="el-p-128" data-ref="8a1eb3bb-bc3a-489c-a7cd-a95c3d3e4fa9" data-id="el-p-128">Indeed, FGFRi reproducibly affects cell number more than MEKi, in line with the fact that pathways other than MAPK/ERK downstream of FGF (e.g. PI3K) play important roles in cell survival and growth. However, we think the lack of differentiation in MEKi and FGFRi in Fig.1g cannot be attributed to a loss of cells combined with a community effect. This is because without FGFRi or MEKi cells efficiently differentiate to primitive streak at much lower densities than those originally shown, consistent with the data we discuss in response to (1) arguing against a primarily indirect effect of FGF on PS-LC differentiation through cell density. In the context of directed differentiation (rather than 2D gastruloids), we have now shown in a controlled manner that the effect of MEKi and FGFRi does not depend on a community effect by repeating the experiment in Fig.1g while adjusting cell seeding densities to obtain similar final cell densities in all three conditions (new Fig.1g, new Supp Fig.1g). Furthermore we have included new data showing extremely sparse cells without MEKi or FGFRi still differentiate without problems (new Supp Fig 1h). We have also include Gattoglio et al in our revised discussion.</p><blockquote data-ref="f7283141-df7e-4161-a510-6a07b9376749"><p id="el-p-129" data-ref="f9bb140d-4e1a-416b-b4dd-a740f0937b48" data-id="el-p-129">(3) Do the FGF4 and FGF17 LOF experiments in Figure 5 affect cell numbers like FGFRi in Figure 1?</p></blockquote><p id="el-p-130" data-ref="984bed0d-66d6-48e6-8f35-8efc8ea6a827" data-id="el-p-130">We did not observe major changes in cell number in the FGF4 and FGF17 loss of function experiments. This is in line with our observation that low levels of ERK signaling are sufficient to maintain proliferation (new Supp. Fig. 1i), and the fact that low levels of ERK signaling are maintained in the absence of FGF4 and FGF17 (Fig.5), likely by FGF2 (Fig. 2). In contrast, FGFRi treatment in Fig.1 leads to a nearly complete loss of FGF signaling (ERK and other pathways) that has a dramatic effect on cell number.</p><blockquote data-ref="21cc2fe7-6e71-4d10-bf10-31dcd8e1b6c7"><p id="el-p-131" data-ref="2b0db90f-9e9a-44ae-8c88-b220efd82bdb" data-id="el-p-131">Why examine PS-LC induction only in FGF17 heterozygous cells and not homozygous FGF17 nulls?</p></blockquote><p id="el-p-132" data-ref="5a867e4d-f7fe-48e5-8b79-35fd7a805e5c" data-id="el-p-132">We were unable to obtain homozygous FGF17 nulls, it is not clear if there is a reason for this. In the absence of homozygous nulls, we have now further corroborated our findings with additional knockdown data (described in response to other comments below).</p><blockquote data-ref="f804ee20-8b7b-410d-8bcd-1924eee98c36"><p id="el-p-133" data-ref="edc97fc2-6dc7-4341-b45a-a8988035eca9" data-id="el-p-133">(4) The idea that FGF8 plays a dominant role during gastrulation of other species but not humans is so intriguing it warrants deeper testing. The authors dismiss FGF8 because its mRNA "...levels always remained low." (line 363) as well as the data published in Zhai et al (PMID: 36517595) and Tyser et al (PMID: 34789876). But there are cases in mouse development where a gene was expressed at levels so low, that it might be dismissed, and yet LOF experiments revealed it played a role or even was required in a developmental process. The authors should consider FGF8 inhibition or inactivation to explore its potential role, despite its low levels of expression.</p></blockquote><p id="el-p-134" data-ref="13473dae-b259-47e0-b3d1-3ac4db4ac2e2" data-id="el-p-134">We thank the reviewer for this suggestion. We have now analyzed the role of FGF8 using FISH to visualize its expression and siRNA to understand its function (Fig.5d,f,h; Supp.Fig.5e,g,6e). We found that FGF8 expression is higher earlier in differentiation, preceding most expression of TBXT. Our scRNA-seq only analyzed samples at 42h so did not capture this. Furthermore, FGF8 expression localized inside the PS-like ring rather than coinciding with it like FGF4. Surprisingly, FGF8 knockdown led to an increase in primitive streak-like differentiation, suggesting it may counteract FGF4. The results are shown in the revised Fig. 5 and Supplemental Fig. 5. While this certainly merits further investigation, understanding the role of FGF8 in more detail is beyond the scope of the current work.</p><blockquote data-ref="8d3f92eb-dde5-4274-b1bc-7c19c935e17c"><p id="el-p-135" data-ref="17d2b5e2-72b3-43eb-b1e3-3b07fa9d2836" data-id="el-p-135">(5) Redundancy is a common feature in FGF genetics. What is the effect of inhibiting FGF4 in FGF17 LOF cells?</p></blockquote><p id="el-p-136" data-ref="65660fea-3813-4f24-998c-86c634503cf9" data-id="el-p-136">Further siRNA and shRNA experiments showed that FGF17 knockdown had a much smaller effect than FGF4 knockdown on expression of primitive streak markers (Fig.5i, Supp.Fig.6f-i) but that FGF17 knockdown did lead to a complete loss of the mesoderm marker TBX6 (Fig.5j, Supp.Fig.6j). A double knockdown of FGF4+FGF17 looked similar to FGF4 alone (Supp.Fig.6k). Thus, we now think the more likely scenario is that FGF17 is downstream of FGF4-dependent PS-differentiation and although this may have a positive feedback effect whereby this FGF17 can then enhance further PS-differentiation, which we previously interpreted as partial redundancy, the primary role of FGF17 may be later, in mesoderm differentiation.</p><blockquote data-ref="d7db3727-62f3-467f-a587-72ca9ab4f1a2"><p id="el-p-137" data-ref="6f6359d6-6055-4640-8dc9-75c8f52682ea" data-id="el-p-137">(6) I suggest stating that the authors take more caution in describing FGF gradients. For example, in one Results heading they write "Endogenous FGF4 and FGF17 gradients underly the ERK activity pattern.", implying an FGF protein gradient. However, they only present data for FGF mRNA , not protein. This issue would be clarified if they used proper nomenclature for gene, mRNA (italics), and protein (no italics) throughout the paper.</p></blockquote><p id="el-p-138" data-ref="d5f398bc-23e3-443b-aa22-d8550a9cdfa1" data-id="el-p-138">Thank you for the suggestion. We have edited the paper to more clearly distinguish protein and mRNA. We do think our data provide substantial indirect evidence for a protein gradient which is what the results heading is meant to convey. Receptor activation is high where ERK activity is high (Fig.3), and receptor activation is limited by ligands, since creating a scratch to let exogenous FGF reach the basal side of cells in the center leads to receptor activation (Fig.4). This strongly suggests ERK activity reflects an FGF protein gradient.</p><blockquote data-ref="803bd602-f7eb-454d-9e6f-7eef3344cd6a"><p id="el-p-139" data-ref="b6ad45f4-e340-4b28-be59-2f8b3a33cc4d" data-id="el-p-139"><strong data-ref="44727c45-c822-4854-bf36-c0a7f5513ffc">Reviewer #2 (Public review):</strong></p><p id="el-p-140" data-ref="ec93e6a1-9b06-4104-8563-5021220a5c7e" data-id="el-p-140">Summary:</p><p id="el-p-141" data-ref="1629ca85-50f2-423f-8ccc-3bff571911e0" data-id="el-p-141">The role of FGFs in embryonic development and stem cell differentiation has remained unclear due to its complexity. In this study, the authors utilized a 2D human stem cell-based gastrulation model to investigate the functions of FGFs. They discovered that FGF-dependent ERK activity is closely linked to the emergence of primitive streak cells. Importantly, this 2D model effectively illustrates the spatial distribution of key signaling effectors and receptors by correlating these markers with cell fate markers, such as T and ISL1. Through inhibition and loss-of-function studies, they further corroborated the needs of FGF ligands. Their data shows that FGFR1 is the primary receptor, and FGF2/4/17 are the key ligands for primitive streak development, which aligns with observations in primate embryos. Additional experiments revealed that the reduction of FGF4 and FGF17 decreases ERK activity.</p><p id="el-p-142" data-ref="84320283-6e0b-4397-a189-1e23b6e0c697" data-id="el-p-142">Strengths:</p><p id="el-p-143" data-ref="98b3d4fe-dd00-4aed-acba-2f68abb161c2" data-id="el-p-143">This study provides comprehensive data and improves our understanding of the role of FGF signaling in primate</p><p id="el-p-144" data-ref="dc9049aa-9249-48ec-bb89-0a417d6c134b" data-id="el-p-144">primitive streak formation. The authors provide new insights related to the spatial localization of the key components of FGF signaling and attempt to reveal the temporal dynamics of the signal propagation and cell fate decision, which has been challenging.</p><p id="el-p-145" data-ref="91a9e6a8-293d-44eb-a58c-0a5383ea1d55" data-id="el-p-145">Weaknesses:</p><p id="el-p-146" data-ref="1ff2b665-6657-43ab-9468-94bafb245fef" data-id="el-p-146">Given the solid data, the work only partially clarifies the complex picture of FGF signaling, so details remain somewhat elusive. The findings lack a strong punchline, which may limit their broader impact.</p></blockquote><p id="el-p-147" data-ref="874a877c-3b78-4e6c-8317-e4b43d7d76d0" data-id="el-p-147">We thank this reviewer for their valuable feedback and compliment on the solidity of our data. The punchline of our work is that FGF4 and FGF17-dependent ERK signaling plays a key role in differentiation of human PS-like cells and mesoderm, and that these are different FGFs than those thought to drive mouse gastrulation. A second key point is that like BMP and TGFβ signaling, FGF signaling is restricted to the basolateral sides of pluripotent stem cell colonies due to polarized receptor expression, which is crucial for understanding the response to exogenous ligands added to the cell medium. Indeed, many facets of FGF signaling remain to be investigated in the future, such as how FGF regulates and is regulated by other signals, which we will dedicate a different manuscript to.</p><blockquote data-ref="4b6cc45c-61e6-4b68-b9b8-9f6c9cc68586"><p id="el-p-148" data-ref="5b4e6afd-5fec-4b3e-ac65-17333ed51da6" data-id="el-p-148"><strong data-ref="61f3a7e5-1597-4879-97b5-c976cfec3571">Reviewer #3 (Public review):</strong></p><p id="el-p-149" data-ref="feb874db-fca4-4d0a-8248-40648868d5aa" data-id="el-p-149">Jo and colleagues set out to investigate the origins and functions of localized FGF/ERK signaling for the differentiation and spatial patterning of primitive streak fates of human embryonic stem cells in a well-established micropattern system. They demonstrate that endogenous FGF signaling is required for ERK activation in a ringdomain in the micropatterns, and that this localized signaling is directly required for differentiation and spatial patterning of specific cell types. Through high-resolution microscopy and transwell assays, they show that cells receive FGF signals through basally localized receptors. Finally, the authors find that there is a requirement for exogenous FGF2 to initiate primitive streak-like differentiation, but endogenous FGFs, especially FGF4 and FGF17, fully take over at later stages.</p><p id="el-p-150" data-ref="6b33f7b7-80f6-4d8d-8501-569b0f1ff3c8" data-id="el-p-150">Even though some of the authors' findings - such as the localized expression of FGF ligands during gastrulation and the importance of FGF/ERK signaling for cell differentiation in the primitive streak - have been reported in model organisms before, this is one of the first studies to investigate the role of FGF signaling during primitive streak-like differentiation of human cells. In doing so, the paper reports a number of interesting and valuable observations, namely the basal localization of FGF receptors which mirrors that of BMP and Nodal receptors, as well as the existence of a positive feedback loop centered on FGF signaling that drives primitive-streak differentiation. The authors also perform a comparison of the role of different FGFs across species and try to assign specific functions to individual FGFs. In the absence of clean genetic loss-of-function cell lines, this part of the work remains less strong.</p></blockquote><p id="el-p-151" data-ref="db6a81d7-9345-494f-b550-5ea60fd1c30f" data-id="el-p-151">We thank the reviewer for emphasizing the value of our findings in a human model for gastrulation. We agree more loss-of-function experiments would provide further insight into the role of different FGFs. While we did not manage to create knockout cell lines, we have now performed both siRNA and shRNA knock-down of all FGF4, and FGF17 in two different hPSC lines, performed siRNA knockdown of FGF8, and also made a FGF4+FGF17 shRNA double knockdown cell lines to more completely test the functions of the individual FGFs (Fig.5, Supp.Fig.5,6). Our data suggest FGF17 may be downstream of FGF4 and primarily required for mesoderm differentiation while FGF8 appears to counteract FGF4. In doing this we have added a large amount of new data to the manuscript and we have removed the heterozygous knockout data in the first version of the manuscript which we felt added little to the new data. Further experiments are still needed to solidify our interpretation but those are beyond the scope of the current work.</p><blockquote data-ref="4548a455-b5e0-490c-8005-6b1f0bdd20d1"><p id="el-p-152" data-ref="03d36b1a-cfed-4f17-9ca5-7854e8ac57ea" data-id="el-p-152"><strong data-ref="1b914778-829f-46e5-b75a-e0d895690ec8">Reviewer #1 (Recommendations for the authors):</strong></p><p id="el-p-153" data-ref="6ae9a476-28d9-43da-85f3-1c0ded8d9798" data-id="el-p-153">(1) FGF2 is added to culture experiments (e.g. Figure 4), but the commercial source is not mentioned in Methods. For example, it could be added to "Supplementary Table 1: Cell signaling reagents."</p></blockquote><p id="el-p-154" data-ref="09597eb2-8517-48e6-9ef5-ea1c8c252b93" data-id="el-p-154">We apologize for this oversight and have now added the information to Supplementary Table 1.</p><blockquote data-ref="236c7b38-413a-4674-9e1b-ae220d76bf27"><p id="el-p-155" data-ref="c885b867-8b87-4ad7-ae54-05b73225341b" data-id="el-p-155">(2) Line 117-118: "For example, by controlling the expression of Wnt or Nodal which are both required for PS-like differentiation". It is clear what the authors mean, but this is not a complete sentence.</p></blockquote><p id="el-p-156" data-ref="fdbcc30a-f67e-4242-828d-834d73d7ccf5" data-id="el-p-156">We edited this for clarity, it now reads: “First, is FGF/ERK signaling required directly for PS-like differentiation, or does it act indirectly? These possibilities are not mutually exclusive. For example, FGF/ERK could be required directly but also act indirectly by controlling Wnt or Nodal expression, as both Wnt and Nodal signaling are required for PS-like differentiation.”</p><blockquote data-ref="0b93fd8c-2615-401c-bfff-e33fc83e60b0"><p id="el-p-157" data-ref="556f3429-6fcd-4923-b18d-d2716514e4d0" data-id="el-p-157">(3) Line 246 "...found its spatial pattern to strongly resembles that of pERK..." either remove "to" or change "resembles" to "resemble"</p></blockquote><p id="el-p-158" data-ref="a9d0d2c5-abec-4c88-a826-cb377f58b619" data-id="el-p-158">Thank you for catching this. We removed “to”.</p><blockquote data-ref="166f5710-3d6c-4ce0-a0c0-4462f36cdb09"><p id="el-p-159" data-ref="3bc0d905-ad18-460c-b22a-df96cb09be2b" data-id="el-p-159">(4) Lines 391- 393 seem to be missing a word in the last phrase: "...with FGF17 more important continued differentiation to mesoderm and endoderm." Maybe "during" after the word "important"?</p></blockquote><p id="el-p-160" data-ref="8006f94b-800f-43e7-ac86-ca302f941322" data-id="el-p-160">Thank you for catching this, indeed the word “during” was missing and we have now added it.</p><blockquote data-ref="aceb50a3-a0a8-4ffb-8de6-1b5d12f76ead"><p id="el-p-161" data-ref="bc5a7ed5-18c5-402f-b638-5e2ee6033428" data-id="el-p-161">(5) Please define acronyms in Figure 3D (PS-LC was defined previously, but not others).</p></blockquote><p id="el-p-162" data-ref="b30896c0-4b7c-4bc9-9f20-a997db65c709" data-id="el-p-162">We apologize for the oversight, we have now defined the acronyms.</p><blockquote data-ref="de386c51-758a-44ee-be12-43c2cb31b100"><p id="el-p-163" data-ref="ac474532-b3a3-48ad-a713-a312335b28ca" data-id="el-p-163">(6) The three blue lines in Figure 5B (right) are hard to discern (and I'm not colorblind). I suggest also using a variety of dotted lines in a subset of these FGFs.</p></blockquote><p id="el-p-164" data-ref="b64f4dce-8b64-49e3-a890-880d10cafcb2" data-id="el-p-164">Thanks you for the suggestion. We have now given all the FGFs colors that are more clearly distinct and made the TBXT and TBX6 lines dashed.</p><blockquote data-ref="c3d1b126-db69-4d0e-bdef-3aaf616667a3"><p id="el-p-165" data-ref="c6d61b3b-6752-4bc4-9693-f097736a984b" data-id="el-p-165"><strong data-ref="a0c878bc-57f0-444c-a206-f330e87bbf34">Reviewer #2 (Recommendations for the authors):</strong></p><p id="el-p-166" data-ref="55638ac4-cef1-4134-b9fd-ae51103a22c2" data-id="el-p-166">(1) The reviewer acknowledges that FGF signaling is complex, particularly when dynamics and its correlation with cell fates are considered. To improve the clarity of the findings, the authors are encouraged to provide an additional schematic figure that clearly delineates the main findings of this study.</p></blockquote><p id="el-p-167" data-ref="9e6c4d62-837c-4bfc-a279-1e7b8ac12c42" data-id="el-p-167">Thank you for the suggestion. We have now added a summary figure (Fig.6) to our discussion, which we hope helps present our findings more clearly.</p><blockquote data-ref="df55a656-68d9-471c-af21-33db742d0536"><p id="el-p-168" data-ref="0b428f03-e2ec-4a2f-bd99-2b3eaad396ea" data-id="el-p-168">(2) The data suggest that FGF signaling may function differently in mice compared to primates, and their stem cell model aligns more closely with the latter. While the authors discuss this in the contents only based on sequencing data, it would be valuable to conduct some experiments with mouse embryos to validate the key differences.</p></blockquote><p id="el-p-169" data-ref="de62f7da-998e-499f-90e4-cc46b6f0cc97" data-id="el-p-169">It is unclear to us which experiments the reviewer has in mind. There is ample data on FGF expression in the mouse literature, as are many knockout phenotypes. Furthermore, verifying loss of function phenotypes (e.g. FGF17 knockout) in mouse is beyond our expertise.</p><blockquote data-ref="fe0dfccb-2fc6-4562-ac23-a67963c68cab"><p id="el-p-170" data-ref="e4b1c23f-4bd5-430f-9e94-1243604f27cd" data-id="el-p-170">(3) Heparan sulfate proteoglycan (HSPG) is mentioned as an important component of FGF signaling; however, the only data related to HSPG is single-cell sequencing results. The authors should consider performing immunostaining or other assays to validate HSPG expression and spatial distribution, similar to the approach they used for other signaling components.</p></blockquote><p id="el-p-171" data-ref="b341e462-efbe-4496-a031-64bc5672ed38" data-id="el-p-171">Our scratch experiments in Fig. 4 strongly argue against HSPGs as being responsible for the spatial pattern of FGF receptor activation: after a scratch across the colony the response is strong all along the scratch as expected if presence of FGF (an FGF gradient) controls the level of activity. If HSPGs were limiting, FGF flowing in from the media show not be able to uniformly activate receptors around the scratch.</p><p id="el-p-172" data-ref="3d100854-3ebc-43f9-a7cf-c6b7b33de8f4" data-id="el-p-172">In addtion, we have now included an immunostain for HS in a newly added Supp. Fig. 4 which does not explain the observed pattern of ERK signaling.</p><blockquote data-ref="ee42498b-1a1d-4bff-8ea4-38f2cee120cd"><p id="el-p-173" data-ref="fcbca6f8-9ea7-4661-b95e-c9ee911f7fac" data-id="el-p-173">(4) In the scratch experiment, particularly high PERK expression is observed at the edge of the scratch. The authors should provide an explanation for why this expression is significantly higher compared to the edges of the colony. Additionally, it would be interesting to investigate the fate of the cells with super high PERK expression.</p></blockquote><p id="el-p-174" data-ref="ae5ad073-6944-487a-9124-61a131a98c40" data-id="el-p-174">We have now determined that adaptive response to FGF is the reason that the response around the scratch is initially much higher than in the ERK activity ring that overlaps with the primitive streak-like cells. We have added figures showing that although the intial response to FGF exposure after scratching is very high, the response around the scratch adapts to levels similar in those in the ERK ring over the course of 6 hours (Fig.4ij).</p><blockquote data-ref="8d1de816-5b30-4ba7-baf1-01de141deed2"><p id="el-p-175" data-ref="5b00ff87-20fb-4adb-b2b2-23df0d14bfda" data-id="el-p-175">(5) For some of the key experiments, multiple cell lines should be used to ensure that the findings are reproducible and applicable across different human stem cell lines.</p></blockquote><p id="el-p-176" data-ref="4a28224f-9e57-4a0d-85f7-bf4c4774a3cf" data-id="el-p-176">We have now checked FISH stainings and knockdown phenotypes for different FGFs in two different cell lines: ESI17 (hESC, XX) and PGP1 (hiPSC, XY). These results are shown in Supplementary Figures 6. We found all results to be consistent.</p><blockquote data-ref="babb514e-6bba-41f5-8338-1fcf2c9f0cff"><p id="el-p-177" data-ref="e0b6db53-1e9b-4c2c-879f-0cea05cb7f5f" data-id="el-p-177">(6) Where applicable, the meaning of error bars needs to be more clearly presented, including details on the number of independent experiments or samples used.</p></blockquote><p id="el-p-178" data-ref="9ac2e39d-6328-4e8f-95bd-08f6a8c18009" data-id="el-p-178">Thank you for pointing this out. Where error bar definitions were missing we have now added them to the figure captions.</p><blockquote data-ref="fad3964d-2d83-4e1a-a410-a9fe38effb78"><p id="el-p-179" data-ref="9bcfd5e5-ea35-4c56-a586-b7074877fbb6" data-id="el-p-179"><strong data-ref="1152392c-af94-423f-90ca-9bc1090a0949">Reviewer #3 (Recommendations for the authors):</strong></p><p id="el-p-180" data-ref="64117695-8b46-4389-8f7d-72b4f2e94c79" data-id="el-p-180">(1) The authors only analyze the ppERK ring in micropatterns of a single size. What was the motivation for the choice of this size? Can the authors how the ppERK ring is expected to depend on colony size?</p></blockquote><p id="el-p-181" data-ref="37ae54cf-179d-47b1-86ac-c5fe6c8e6f02" data-id="el-p-181">Much smaller patterns lose the interior pluripotent regions while much larger patters have a much larger pluripotent region, which requires larger tilings to image without providing additional insight. The colony sizedependence of cell fate patterning was described in the paper that established the 2D gastruloids model (Warmflash Nat Methods 2014) and we later showed this due to a fixed length scale of the BMP and Nodal signaling gradients from the colony edge (Jo et al Elife 2022). We have now included data showing that the ERK patterns behaves similarly, with a fixed length scale of the pattern implying that in smaller colonies the ERK ring becomes a disc and the entire center of the colony has high ERK signaling (Supp Fig 1a).</p><blockquote data-ref="010f865e-4f5a-46c2-a426-14a83fc6df3a"><p id="el-p-182" data-ref="3d6dcb5e-fb91-4dba-8386-75f966265448" data-id="el-p-182">(2) The scRNAseq is somewhat confusing - why do the two datasets not overlap in the PHATE representation? This is unexpected, because the two samples have been treated similarly, and the authors have integrated their data to iron out possible batch effects. This discrepancy should be discussed. The authors should also specify from which reference exactly the first dataset comes from.</p></blockquote><p id="el-p-183" data-ref="4f54e9f3-165e-4258-9d31-5a3d7c7fa5c9" data-id="el-p-183">The two datasets do overlap nicely, the same fates are well mixed in the same place and the gene expresison profiles for the integrated data (e.g., Fig.2e) look smooth, so we believe the integration is good, but different cell fates are represented to different degrees. In particular, sample 2 shows much more mesoderm differentiation making the mesoderm branch mostly orange. Occassionally samples differentiate faster or slower than average which we see here, and these samples were collected far apart in time. We do not believe this affects our conclusions, if anything, we think performing the analysis on two samples that differ this much should make the conclusions more robust.</p><blockquote data-ref="25edf3c3-410a-406d-957b-fc0b5feebfae"><p id="el-p-184" data-ref="884be425-1408-4c96-9c2c-8b46a7a3e6e4" data-id="el-p-184">(3) If find it intriguing that exogenous FGF2 is important early on for primitive streak-like differentiation, although the authors show that it does not reach the center of the colony. The authors may want to discuss this conundrum. Does the FGF2 effect propagate from the outside to the inside, or does it act at an early stage when the cells have not yet formed a tight epithelium on the micropattern?</p></blockquote><p id="el-p-185" data-ref="77812c1c-a04c-478d-ade0-2c0825a8c5ec" data-id="el-p-185">The cells in the experiment in Fig. 5a were given 24h to epithelialize, so we we do believe it acts from the edge. We believe this may be due to FGF2 modulating the early BMP response on the edge and are working on a manuscript that further explores this pathway crosstalk.</p><blockquote data-ref="94705489-5ba3-4c47-8074-db918d8d9514"><p id="el-p-186" data-ref="3972e677-d70e-4812-93b5-60068a45d9a6" data-id="el-p-186">(4) The authors' statement that FGF4 and FGF17 have partially redundant functions is not very strong, mainly because the study lacks a full FGF17 loss-of-function cell line. If the authors wanted to improve on this point, they could knock down FGF4 in the FGF17 heterozygous line, or produce a homozygous FGF17 KO line. If there are specific reasons why FGF17 homozygous lines cannot be produced, this could be interesting to discuss, too. Finally, I noticed that the methods list experiments with an FGF17 siRNA, but these are not shown in the manuscript.</p></blockquote><p id="el-p-187" data-ref="f08ee8b3-33ea-478f-b81f-4b00260c24b1" data-id="el-p-187">We agree our evidence was previously not as strong as it could be. While there is no reason we know of why homozygous knockout lines cannot be produced, we failed to produce on. To strengthen our evidence we have therefore included substantial new knockdown data.  We have now performed both siRNA and shRNA knockdown of all FGF4, and FGF17 in two different hPSC lines, performed siRNA knockdown of FGF8, and also made a FGF4+FGF17 shRNA double knockdown cell lines to more completely test the functions of the individual FGFs (Fig.5, Supp.Fig.5,6). These experiments showed that FGF17 knockdown had a much smaller effect than FGF4 knockdown on expression of primitive streak markers (Fig.5i, Supp.Fig.6f-i) but that FGF17 knockdown did lead to a complete loss of the mesoderm marker TBX6 (Fig.5j, Supp.Fig.6j). A double knockdown of FGF4+FGF17 looked similar to FGF4 alone (Supp.Fig.6k). Thus, we now think the more likely scenario is that FGF17 is downstream of FGF4-dependent PS-differentiation and although this may have a positive feedback effect whereby this FGF17 can then enhance further PS-differentiation, which we previously interpreted as partial redundancy, the primary role of FGF17 may be later, in mesoderm differentiation. Furthermore, our new data suggests FGF8 may counteract FGF4 and limit PS-like differentiation.</p><blockquote data-ref="75d1540d-41a6-4ac2-a018-1081d4589ab0"><p id="el-p-188" data-ref="4ba1e09e-b7d6-4145-aa89-1dea7e4c549a" data-id="el-p-188">Minor</p><p id="el-p-189" data-ref="bbe3f4e1-1237-4d1e-87f9-58fbca07e856" data-id="el-p-189">(5) Line 63: Reference(s) appear to be missing.</p></blockquote><p id="el-p-190" data-ref="0a834485-713d-4359-839d-8a3e27b648c0" data-id="el-p-190">This whole paragraph summarizes the results of the references given on line 55, we have now repeated the relevant references where the reviewer indicated.</p><blockquote data-ref="fe1776c8-3836-438e-b7c2-db6160b7acd2"><p id="el-p-191" data-ref="aff9ab34-128c-4260-902e-728151cc1c7c" data-id="el-p-191">(6) Supplementary Figure 1a,b does not show ppERK, unlike stated in lines 102 - 104.</p></blockquote><p id="el-p-192" data-ref="8332b666-9474-44a0-b8f3-86c96dba235f" data-id="el-p-192">Indeed, the data described in lines 102-104 is shown in Fig.1a and we have removed the original Supplementary Figure 1ab since it did not provide relevant information.</p><blockquote data-ref="cb3ae97f-2548-4759-96c8-8cf35f027498"><p id="el-p-193" data-ref="dbeda532-7563-4224-bc52-4c331197061a" data-id="el-p-193">(7) Line 201: It is not clear whether this is a new sequencing dataset, or if existing datasets have been reanalyzed.</p></blockquote><p id="el-p-194" data-ref="d3fc1e99-c5c9-4621-9430-7549a5eee57b" data-id="el-p-194">We agree our description was unclear. We have edited the text, which now explicitly states that our analysis is based on one dataset we collected previously and a replicate that was newly collected and deposited on GEO for this manuscript.</p><blockquote data-ref="addd5004-2dd8-464b-8fba-d345d401dc25"><p id="el-p-195" data-ref="5afab035-4833-4317-9c28-53d7145238d6" data-id="el-p-195">(8) Figure 2f; Supplementary Figure 2b, c: The colors need to be explained in scale bars. How has this data been normalized to allow for comparison between very different sample types?</p></blockquote><p id="el-p-196" data-ref="755645eb-a66e-4bb8-8ad7-1447bc619358" data-id="el-p-196">We have now added color bars indicating the scale for each of these figure panels. As the caption stated, the interspecies comparison was normalized within each species, so the highest FGF level for any FGF at any time within each species is normalized to one. We are thus comparing between species the relative expression of different FGFs within each species. Indeed there is no good way to compare absolute expression between species. For extra clarity we have expanded our description of the interspecies comparison analysis and normalization in the methods section.</p><blockquote data-ref="49f19baa-439f-4682-86b5-bd85f5fa2b96"><p id="el-p-197" data-ref="afed945c-0dfa-4ade-bd74-28793ebbb1f9" data-id="el-p-197">(9) Line 232: Where is the expression of SEF shown?</p></blockquote><p id="el-p-198" data-ref="f4c8608f-8dd0-42a9-88f9-49d074851685" data-id="el-p-198">It is shown in Fig. 2i, under the official gene name IL17RD.</p><blockquote data-ref="33c1e790-f501-44a3-8fd8-37a66f6fb360"><p id="el-p-199" data-ref="7e4be313-7268-4ced-96f4-7a3ab2fdd07d" data-id="el-p-199">(10) Supplementary Figure 4 seems to be missing.</p></blockquote><p id="el-p-200" data-ref="f1527a65-24aa-406b-aeee-60422af9a89a" data-id="el-p-200">Thank you for pointing this out. We have now added a supplementary Fig.4.</p><blockquote data-ref="cf8e1f7c-51c2-43e8-8e17-1f70a33cf7af"><p id="el-p-201" data-ref="9a5a8618-8a80-4642-be16-bee1d331ca2e" data-id="el-p-201">(11) Line 437: Citation needed.</p></blockquote><p id="el-p-202" data-ref="cc506633-1897-4317-bd0d-126a0ddd3d2c" data-id="el-p-202">We have included citations now.</p><blockquote data-ref="0abd1e02-0f52-414b-b3a2-4e4842848dee"><p id="el-p-203" data-ref="28bfa186-2253-44ec-b2ce-73a9fffe78a7" data-id="el-p-203">(12) Line 439: A similar feedback loop has been proposed to operate during mesoderm differentiation in mouse ESC (pmid: 37530863 ). The authors may consider citing this work.</p></blockquote><p id="el-p-204" data-ref="1cfb774f-ef23-4a38-b51d-931470733b89" data-id="el-p-204">Thank you for the suggestion, we have now included this work in the discussion. The feedback loop proposed in that work involves FGF8, while we were trying to explain why FGF4 and not FGF8 appears to be conserved across species by invoking an FGF4 feedback loop. Thus, it becomes even harder to explain differences in FGF4 and FGF8 expression between human and mouse gastrulation.</p><blockquote data-ref="a412a7c4-fabd-45eb-abe7-322070a82a40"><p id="el-p-205" data-ref="93d285fd-3a22-4eaf-94c3-e72256d7d37f" data-id="el-p-205">(13) Supplementary Figure 6 is not described in the main text.</p></blockquote><p id="el-p-206" data-ref="bc49f904-c6e9-46a5-a68c-32fbc15bffea" data-id="el-p-206">We have removed the original Supplementary Figure 6 and corresponding heterozygous knockout data in the main figure which we felt added little to the extensive knockdown data we now present. We did create a new Supplementary Figure 6 showing additional knockdown data which is described in the main tekst.</p><blockquote data-ref="23f1f301-f725-4cee-9886-e4d00671257c"><p id="el-p-207" data-ref="d90c26a0-41b1-4982-b088-e4c5f0371d49" data-id="el-p-207">(14) Submission of sequencing data to GEO needs to be updated.</p></blockquote><p id="el-p-208" data-ref="2d1c6c48-f7da-4f6b-b793-52267b4fcbe3" data-id="el-p-208">We have now made the GEO data public.</p></div><div class="descriptors" id="el-div-23" data-ref="851a1df4-2470-4607-b82f-13cedb6a79bf" data-id="el-div-23"><ul class="descriptors__identifiers" id="el-ul-10" data-ref="97d17940-f71c-49d0-9766-6e3a070a3749" data-id="el-ul-10"><li class="descriptors__identifier" id="el-li-373" data-ref="b3123223-0a14-4990-99d1-d9436dff211d" data-id="el-li-373"><a href="https://doi.org/10.7554/eLife.101224.2.sa0" data-ref="744a5acb-cf2e-465a-81e7-2a1aebbfcbce">https://doi.org/10.7554/eLife.101224.2.sa0</a></li></ul></div></section></section></article></template><div class="pagedjs_pages" style="--pagedjs-page-count: 42;">
<div class="pagedjs_page pagedjs_first_page pagedjs_right_page" data-page-number="1" id="page-1">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: minmax(16.66%, 0%) 0px 1fr;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><div id="right-header-links" data-ref="3c7afed3-0b3e-40d4-9870-9f68daa5c26d" data-id="right-header-links" style=""><a href="https://en.wikipedia.org/wiki/Open_access" target="_blank" class="header-icon oa-icon" data-ref="e8cc0315-0223-4b9e-89f5-f72b617f8ec9"></a><span class="separator" data-ref="fc3f7d6a-6e26-42fd-8a8f-cf50f6e37e53">|</span><a href="https://creativecommons.org/" target="_blank" class="header-icon cc-icon" data-ref="fd273eb0-acfc-499d-8310-14a992663ed9"></a></div></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><div id="page-templates" data-ref="98c1cf40-8bda-40b5-aabc-2d6df0b7d792" data-id="page-templates"><div id="first-page-links" class="elife-intro" data-ref="e06f9f30-5f47-40d0-ae1e-f2658d8a55f4" data-id="first-page-links"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style="display: none;"><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a><div id="right-header-links" data-ref="3c7afed3-0b3e-40d4-9870-9f68daa5c26d" data-id="right-header-links" style="display: none;"><a href="https://en.wikipedia.org/wiki/Open_access" target="_blank" class="header-icon oa-icon" data-ref="e8cc0315-0223-4b9e-89f5-f72b617f8ec9"></a><span class="separator" data-ref="fc3f7d6a-6e26-42fd-8a8f-cf50f6e37e53">|</span><a href="https://creativecommons.org/" target="_blank" class="header-icon cc-icon" data-ref="fd273eb0-acfc-499d-8310-14a992663ed9"></a></div></div></div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-to="c36d516e-61de-4652-b357-b350359ef39e" data-split-original="true"><section id="sideSection" class="side-section" data-ref="d29e5412-f79b-4013-8e4a-0c1f74571ad1" data-id="sideSection"><div class="article-status" id="el-div-3" data-ref="d728c3de-e2f2-4ca0-8303-1de5cf9f9087" data-id="el-div-3"><div class="review-timeline-container" id="el-div-4" data-ref="63975305-dba9-44ad-a1ee-34b8255f471c" data-id="el-div-4"><dl class="review-timeline review-timeline--expanded" id="review-timeline" aria-label="Version history" data-ref="945d6a41-f13b-47fe-a468-c8f0db6969aa" data-id="review-timeline"><dt class="review-timeline__event review-timeline__event--title review-timeline__event--active review-timeline__event--with-evaluation-summary review-timeline__event--reviewed" data-ref="04603fa3-a54d-4382-a9d3-97731b1b9642"><a href="https://doi.org/10.7554/eLife.101224.1" class="review-timeline__event-link" data-ref="27d287d4-7e81-4aaa-8c71-97d2003d0615">Reviewed Preprint</a></dt><dd class="review-timeline__event review-timeline__event--detail review-timeline__event--active review-timeline__event--with-evaluation-summary review-timeline__event--reviewed" data-ref="a96f25cc-31e2-470f-b706-26c35778576c"><span class="review-timeline__version" data-ref="bf0bd9af-1559-4813-bd69-f4cd0b787f10">v1</span><time class="review-timeline__date" datetime="2024-11-08T00:00:00.000Z" data-ref="90d80755-ded6-450d-b81a-118fa06b13eb">November 8, 2024</time><span class="review-timeline__link" data-ref="cf1c9173-23ad-4481-94c9-f4c7e35abf5b" data-undisplayed="undisplayed">Not revised</span></dd><dt class="review-timeline__event review-timeline__event--title review-timeline__event--active review-timeline__event--with-evaluation-summary review-timeline__event--revised" data-ref="a0fc9fe9-9eb5-49f3-8cf4-ae5e73a14a10">Reviewed Preprint</dt><dd class="review-timeline__event review-timeline__event--detail review-timeline__event--active review-timeline__event--with-evaluation-summary review-timeline__event--revised" data-ref="ae600ed9-b4d3-4ce4-8703-39eabca4bb99"><span class="review-timeline__version" data-ref="2bdad4b3-724d-4a1c-817a-114429b8160e"> v2</span><time class="review-timeline__date" datetime="2025-10-29T00:00:00.000Z" data-ref="079832fa-c378-4e4d-a53e-412dc8977542">October 29, 2025</time><span class="review-timeline__link" data-ref="308d901c-9055-40f7-9d48-67f1044a1654" data-undisplayed="undisplayed">Revised by authors</span></dd></dl></div></div><section id="margin-notes" data-ref="f843c510-d79d-4993-98d8-d07dcc58ec46" data-id="margin-notes"><div class="author-notes" id="el-div-5" data-ref="cbffb5a8-034b-422e-8ecf-07a26ed5ba14" data-id="el-div-5"><p id="el-p-1" data-ref="e49cf352-f465-47a9-927f-492dd60e221b" data-id="el-p-1"><strong class="email-icon" data-ref="07b68d3b-54b5-4c7f-9b54-ac48abb32726">For correspondence:</strong></p><ul class="list-simple" id="el-ul-1" data-ref="542eb2d6-a007-444c-a063-892105efa9cb" data-id="el-ul-1"><li id="el-li-2" data-ref="c867c34f-70e2-4b01-abbd-6fbbdc7e504f" data-id="el-li-2"><a href="mailto:test@umich.edu" data-ref="6fe0c5dd-0a56-4f98-9ec4-280ea9ffd66f">test@umich.edu</a></li><li id="el-li-3" data-ref="edef9f88-7f51-4e9b-a500-aba6405bb699" data-id="el-li-3"><a href="mailto:iheemske@umich.edu" data-ref="6b3f1a3a-78e3-475a-a36c-9d921d2d2f0f">iheemske@umich.edu</a></li></ul><p id="el-p-2" data-ref="d2de8c1b-78d1-4aa8-b11d-6118f653dd12" data-id="el-p-2"></p><p class="author-notes__list_item" id="el-p-3" data-ref="bec24e2d-0975-4e10-a982-8653c461f6b0" data-id="el-p-3"><sup aria-hidden="true" data-ref="e27061af-0089-41e7-b942-2a2963ccac13"><strong data-ref="a5778af5-6c51-4328-b7fd-0cc66a8bdc2f">*</strong></sup>&nbsp;Equal contributions</p><p class="author-notes__list_item" id="el-p-4" data-ref="d240c063-b0f3-4ab8-b94f-76d6718da1be" data-id="el-p-4"><strong data-ref="f960dc6b-193e-4266-8b1a-4af46413f0af">Competing interests:</strong> No competing interests declared</p></div><div id="editors" data-ref="82fc0bec-7cff-4e2f-a0ee-5b4ab776eb69" data-id="editors"><p class="editors-and-reviewers__person" id="el-p-5" data-ref="5a438732-7c24-4e0a-84aa-55be09c0dea4" data-id="el-p-5"><strong data-ref="4c1e3658-70f3-4ae5-932f-2cc56fba11fb">Reviewing editor: </strong>Sergio Ruiz Macias, National Cancer Institute, NIH, United States</p></div><div id="copyright" class="copyright" data-ref="9706c82a-0c35-418d-a4ac-d43de4bd7768" data-id="copyright"><p id="el-p-6" data-ref="2d745f9c-bcaa-4523-a885-ff7bdd36c447" data-id="el-p-6">© 2024, Jo et al. This article is distributed under the terms of the <a href="https://creativecommons.org/licenses/by/4.0/" data-ref="1055e913-737c-4bc8-b092-ac46ae57b5a6">Creative Commons Attribution License</a>, which permits unrestricted use and redistribution provided that the original author and source are credited.</p></div></section></section><header id="header" class="content-header" data-ref="b92868f7-c408-4834-8d56-25be65880738" data-id="header"><h1 class="title" id="el-h1-0" data-ref="f04801a6-b526-463b-8afb-f2376e4167bc" data-id="el-h1-0" data-break-after="avoid">Endogenous FGFs drive ERK-dependent cell fate patterning in 2D human gastruloids</h1><div class="authors" id="el-div-8" data-ref="cc7758c7-b6ac-4ca5-9ead-8f0b8c41826d" data-id="el-div-8" data-previous-break-after="avoid"><ol class="authors-list authors-list--expanded" aria-label="Authors of this article" id="el-ol-0" data-ref="9f18dc52-7ab7-45a4-a2d1-d8d0531dabdb" data-id="el-ol-0" start="1"><li class="authors-list__item" id="el-li-4" data-ref="1a2882d9-e976-4cd4-99bb-9dcab2911c6d" data-id="el-li-4" data-item-num="1"><span class="authors-link" data-ref="cec7a04d-60a6-4039-9dc9-c9f6d48977df">Kyoung Jo<sup data-ref="e43b275a-4c0e-4528-804f-d7030858fedf">1,</sup><sup data-ref="a44ab537-b68b-4a26-80a3-7a13b5e55ae4">*</sup></span></li><li class="authors-list__item" id="el-li-5" data-ref="e8f83055-de72-4849-9ab9-5b8971e861ca" data-id="el-li-5" data-item-num="2"><span class="authors-link" data-ref="0ed3a9c4-8eb0-4bb8-ba0b-378583819bf9">Zong-Yuan Liu<sup data-ref="514c5786-1e95-43ad-a460-fe12f892a578">1,</sup><sup data-ref="ceae9108-1891-4794-aa33-86df45d78355">*</sup></span></li><li class="authors-list__item" id="el-li-6" data-ref="69f2f072-8d5a-4a20-9216-9333d79d145e" data-id="el-li-6" data-item-num="3"><span class="authors-link" data-ref="dc9acce3-05b2-497f-9463-41836b83ea92">Gauri Patel<sup data-ref="5e10f5ef-24a7-408e-9c25-fb2283930154">4</sup></span></li><li class="authors-list__item" id="el-li-7" data-ref="5ece50f5-f90e-4587-8441-74cb896ce496" data-id="el-li-7" data-item-num="4"><span class="authors-link" data-ref="0cdd4c42-1eab-4051-8b15-303fc94741b5">Zhiyuan Yu<sup data-ref="1a54094c-4861-4b62-adcc-3d27b9c53e78">2</sup></span></li><li class="authors-list__item" id="el-li-8" data-ref="e1f75c7c-8b69-462e-a02f-fbb6e61d064c" data-id="el-li-8" data-item-num="5"><span class="authors-link" data-ref="0b80f5e2-2fd1-4a27-8f28-f003d271fa3f">LiAng Yao<sup data-ref="4e5515b4-9cd7-438b-9756-d21f3081c0ef">1</sup></span></li><li class="authors-list__item" id="el-li-9" data-ref="93f2f43c-7d57-4640-b8b2-6c6bee32e0f9" data-id="el-li-9" data-item-num="6"><span class="authors-link" data-ref="1e26f6e6-8ac1-402d-b042-73fa2651125c">Seth Teague<sup data-ref="a7bf62a2-7ecb-4582-907e-be67a9f27a3e">4</sup></span></li><li class="authors-list__item" id="el-li-10" data-ref="089a7951-c3d3-4582-a1d0-f6e716de5b82" data-id="el-li-10" data-item-num="7"><span class="authors-link authors-email__link" data-ref="56465565-7f7f-490e-9075-73f32088fb05">Craig Johnson<sup data-ref="082dc54f-3c1e-4422-91fe-979998eeb5fc">1</sup><a href="mailto:test@umich.edu" data-ref="94342959-b471-4fb1-a41f-09546632ba72"><span class="email-icon" data-ref="3c69cae0-72f7-4125-8b0b-821e3aebd90f"></span></a></span></li><li class="authors-list__item" id="el-li-11" data-ref="abeec578-ce81-44b2-8442-6db2a33ff7db" data-id="el-li-11" data-item-num="8"><span class="authors-link" data-ref="468d45d3-a705-436e-be61-832b1b1ea099">Jason Spence<sup data-ref="79b92302-9ae0-40aa-b853-6f75483bfc54">1,</sup><sup data-ref="372b4e2c-012e-4962-9a79-4ca726caa914">3,</sup><sup data-ref="aa6e554f-614b-4f3d-8e44-3774adb35665">4,</sup><sup data-ref="549908f4-5890-465e-b554-1cd803565001">5</sup></span></li><li class="authors-list__item" id="el-li-12" data-ref="15afb4c1-f322-4b8c-99f1-f5c40428cbc7" data-id="el-li-12" data-item-num="9"><span class="authors-link authors-email__link" data-ref="a06f0e26-d012-439d-b821-84e580715ee5">Idse Heemskerk<sup data-ref="a580d364-eb7d-4b83-b14f-79cc939c3bcb">1,</sup><sup data-ref="594acd17-5386-40d8-b737-9b73c134aabd">2,</sup><sup data-ref="e75fe4eb-bb23-45bf-809b-effdaddf2223">3,</sup><sup data-ref="3aae3716-bbfe-4474-baf7-47055cf15553">4,</sup><sup data-ref="3056d230-ffde-4f32-8374-1e4ed7b0b1ad">6</sup><a href="mailto:iheemske@umich.edu" data-ref="97fd5283-845b-48dc-9bac-c00cb7a343fb"><span class="email-icon" data-ref="e24fbe88-f9db-4489-8dfc-ba88c3e7bc92"></span></a></span></li></ol></div><div class="institutions" id="el-div-9" data-ref="07708a74-5968-4d2d-9c8f-60e80e1bf03d" data-id="el-div-9"><ol class="institutions-list" aria-label="Author institutions" id="el-ol-1" data-ref="c7b9ed9c-2607-4fdd-ab90-31bcb7c58aa0" data-id="el-ol-1" start="1"><li class="institutions-list__item" id="el-li-13" data-ref="1c96bbd6-f05e-47e2-b78c-93846b21d93e" data-id="el-li-13" data-item-num="1"><sup data-ref="dda6b018-7f80-40fc-8c9e-dc3575a11558">1</sup>Department of Cell and Developmental Biology, University of Michigan Medical School, Ann Arbor, United States</li><li class="institutions-list__item" id="el-li-14" data-ref="bd9c9265-23dd-4345-96e2-451cb60f5252" data-id="el-li-14" data-item-num="2"><sup data-ref="6cac471e-14df-41c6-ad06-a48f2ad11c46">2</sup>Department of Computational Medicine and Bioinformatics, University of Michigan Medical School, Ann Arbor, United States</li><li class="institutions-list__item" id="el-li-15" data-ref="60c79fd5-9216-4db6-a4f6-a0fda72e4ec1" data-id="el-li-15" data-item-num="3"><sup data-ref="1636ae8e-245c-4840-af6a-ed761bb8d801">3</sup>Center for Cell Plasticity and Organ Design, University of Michigan Medical School, Ann Arbor, United States</li><li class="institutions-list__item" id="el-li-16" data-ref="653b419a-702b-43c7-90c9-3b4e88b3b355" data-id="el-li-16" data-item-num="4"><sup data-ref="19e38072-18d1-409c-80f5-2f6ee102b668">4</sup>Department of Biomedical Engineering, University of Michigan, Ann Arbor, United States</li><li class="institutions-list__item" id="el-li-17" data-ref="a9c5f6e6-ca77-421c-8260-9f706c8b783a" data-id="el-li-17" data-item-num="5"><sup data-ref="4d520166-fbd1-4c92-8b6e-9bbe504824c0">5</sup>Department of Internal Medicine, Gastroenterology, University of Michigan Medical School, Ann Arbor, United States</li><li class="institutions-list__item" id="el-li-18" data-ref="7c91b326-f615-4936-9677-debb4585a9ae" data-id="el-li-18" data-item-num="6"><sup data-ref="c1095db7-d73d-4f59-b871-bf9e17f4ebf6">6</sup>Department of Physics, University of Michigan, Ann Arbor, United States</li></ol></div></header><section id="article-content" data-ref="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-id="article-content" data-split-to="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-split-original="true"><section id="assessment" class="assessment" data-ref="4a99554b-fc89-4aa5-9e54-358cdbc3b2e8" data-id="assessment"><h2 id="el-h2-0" data-ref="11d2d409-dfe3-471e-a1e7-3976e20cd6ed" data-id="el-h2-0" data-break-after="avoid">eLife Assessment</h2><div class="assessment__body" id="el-div-10" data-ref="e5137e0e-5343-462c-a0c5-58f2a98c9813" data-id="el-div-10" data-previous-break-after="avoid"><p id="el-p-7" data-ref="30f4438c-86a7-45d9-9763-d45860ca4688" data-id="el-p-7">This work is an <strong class="highlighted-term" aria-label="Highlighted" data-ref="741035f0-c050-4027-877c-096f05c22d25">important</strong> contribution to understanding the role of FGF signaling in the induction of primitive-like cells in a 2D system of human gastrulation. The authors provide <strong class="highlighted-term" aria-label="Highlighted" data-ref="17d6941e-6d42-4633-b7de-a6759ad0454a">compelling</strong> evidence showing that endogenous FGF ligands, acting through FGF receptors localized basolaterally, are determinant in the acquisition of specific cell fates. These observations will be of broad relevance to the FGF field.</p></div><div class="descriptors" id="el-div-11" data-ref="89d51c0a-0b44-4d3a-bbe8-72542f0b761a" data-id="el-div-11"><ul class="descriptors__identifiers" id="el-ul-2" data-ref="48a458aa-919b-4f4c-953a-20b0fd8bbee5" data-id="el-ul-2"><li class="descriptors__identifier" id="el-li-19" data-ref="c338fb42-6d62-4c41-8a89-9aa8c181a055" data-id="el-li-19"><a href="https://doi.org/10.7554/eLife.101224.2.sa4" data-ref="455f62a4-a33a-4e9a-9c32-b7132b226881">https://doi.org/10.7554/eLife.101224.2.sa4</a></li></ul></div></section><section id="abstract" class="abstract" data-ref="021e9f8f-59d3-4c08-bf24-ed1d3b5689f4" data-id="abstract"><h1 id="el-h1-1" data-ref="760fdf5d-c352-44c3-86bd-399e922cc81e" data-id="el-h1-1" data-break-after="avoid">Abstract</h1><p id="el-p-8" data-ref="2d113aff-cf16-4885-8a8d-be57b20dd02f" data-id="el-p-8" data-previous-break-after="avoid">The role of FGF is the least understood of the morphogens driving mammalian gastrulation. Here we investigated the function of FGF in a stem cell model for human gastrulation known as a 2D gastruloid. We found a ring of FGF-dependent ERK activity that closely follows the emergence of primitive streak (PS)-like cells but expands further inward. We showed that this ERK activity pattern is required for PS-like differentiation and that loss of PS-like cells upon FGF receptor inhibition can be rescued by directly activating ERK. We further demonstrated that the ERK-ring depends on localized activation of basolaterally positioned FGF receptors (FGFR) by endogenous FGF gradients. We confirmed and extended previous studies in analyzing expression of FGF pathway components, showing FGFR1 is the main receptor, FGF2 is highly expressed across several cell types, and FGF4/17 are the main FGF ligands expressed in the PS-like cells, similar to the human and monkey embryo but different from the mouse. We found that knockdown of FGF4 greatly reduced PS-like differentiation while FGF17 knockdown primarily affected subsequent mesoderm differentiation. FGF8 expression was spatially displaced from PS-markers and FGF4 expression and peaked earlier, while knockdown led to an expansion in PS-like cells, suggesting FGF8 may counteract FGF4 to limit PS-like differentiation. Thus, we have identified a previously unknown role for FGF-dependent ERK signaling in 2D gastruloids and possibly the human embryo, driven by a mechanism where FGF4 and FGF17 signal through basally localized FGFR1 to induce PS-like cells and their derivatives, potentially restricted by FGF8.</p></section><section id="s1" data-ref="790b4fb7-01cd-4a65-93f2-98b59b2c0a72" data-id="s1" data-split-to="790b4fb7-01cd-4a65-93f2-98b59b2c0a72" data-split-original="true"><h1 class="heading-1" id="el-h1-2" data-ref="e417c872-9646-4929-80c9-73546a79ba0c" data-id="el-h1-2" data-break-after="avoid">Introduction</h1><p id="el-p-9" data-ref="3a3e36c5-1a53-42ac-87e9-0bdfb7ef9921" data-id="el-p-9" data-previous-break-after="avoid" data-split-to="3a3e36c5-1a53-42ac-87e9-0bdfb7ef9921" data-split-original="true" data-last-split-element="true" data-align-last-split-element="auto">During gastrulation the three germ layers are established, and the body plan is laid out. In amniotes this involves the formation of the primitive streak on the posterior side of the embryo from which cells ingress beneath the epiblast and migrate to form the mesodermal and endodermal layers. An important role in gastrulation is played by evolutionarily conserved </p></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_left_page" data-page-number="2" id="page-2">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e" data-split-to="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-id="article-content" data-split-from="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-split-to="726f25f1-fdd1-41b8-bf49-de736be3e3ae"><section data-ref="790b4fb7-01cd-4a65-93f2-98b59b2c0a72" data-id="s1" data-split-from="790b4fb7-01cd-4a65-93f2-98b59b2c0a72" data-split-to="790b4fb7-01cd-4a65-93f2-98b59b2c0a72"><p data-ref="3a3e36c5-1a53-42ac-87e9-0bdfb7ef9921" data-id="el-p-9" data-split-from="3a3e36c5-1a53-42ac-87e9-0bdfb7ef9921">morphogens from the BMP, Wnt, Nodal, and FGF families<sup data-ref="d04a5c12-9e71-42e8-aa34-73bfd47efd34"><a href="#c1" data-ref="2291fee7-ce6a-4008-aa64-459da4f2e081">1</a></sup>. The function of the first three signaling pathways is comparatively well understood: in both mouse and human a transcriptional hierarchy between BMP4, Wnt3, and Nodal plays a central role in inducing dynamic signaling gradients along the anterior posterior axis that control cell fate patterning<sup data-ref="9ca92e2e-4e4a-4c17-98d4-95af15839217"><a href="#c1" data-ref="29695663-3d39-41c2-92e0-cb70fda103e4">1</a>–<a href="#c3" data-ref="a062b2bb-3e9e-43b2-b803-9ab060aa9718">3</a></sup>. For BMP and Nodal, basal receptor localization in the epiblast is crucial to form signaling activity gradients<sup data-ref="270c28f2-296f-49e6-97dd-83d389e955e4"><a href="#c3" data-ref="c1e0e162-b837-4462-b1ab-af56f9d43c6f">3</a>–<a href="#c7" data-ref="c4492cb5-5554-46c1-b10c-9591b620b17d">7</a></sup>. Wnt and Nodal signaling are directly required for mesoderm and endoderm differentiation, while the balance between BMP and Nodal signaling is important for patterning the germ layers along the body axes<sup data-ref="24cf7f11-04e5-4e29-a207-cfe2fa3abc12"><a href="#c8" data-ref="c621c8cd-ac61-4e63-9ea8-d354f4ea759c">8</a></sup>.</p><p id="el-p-10" data-ref="92d932c1-7237-418f-97b4-d182c8d138a9" data-id="el-p-10">In contrast, the role of FGF in mammalian and especially human gastrulation has been more elusive. Secreted FGFs signal through FGF receptor tyrosine kinases (FGFRs) in a heparan sulfate proteoglycan (HSPG)-dependent manner to activate several signaling pathways including MAPK/ERK, PI3K, and PLCγ<sup data-ref="3dff0c57-e88d-4dc5-8e50-375bf9293eef"><a href="#c9" data-ref="43d4ca26-b824-4558-8559-da78ffe4bce4">9</a></sup>. There is evidence for a conserved requirement for FGF/ERK signaling in mesoderm formation across vertebrate model organisms<sup data-ref="257cd079-1ab1-496c-9944-f1c16a65f1d5"><a href="#c10" data-ref="ae11d120-4571-433b-ba8b-7892e8c8ed53">10</a>–<a href="#c13" data-ref="993fbd92-6877-4b0e-86da-8ac0561519cd">13</a></sup>. However, it is unknown if there are FGF- and phosphorylated ERK gradients across the mammalian primitive streak<sup data-ref="8304d601-b07b-4993-8a1d-084ac4460d0a"><a href="#c14" data-ref="244e305d-7f2e-40c6-9870-266b586f65f4">14</a>,<a href="#c15" data-ref="6b6039a0-7338-436d-9fc5-1561429b5bc8">15</a></sup>, whether there are functionally relevant ERK signaling dynamics like in other contexts, or whether receptor localization is important for establishing signaling gradients<sup data-ref="5dfdd301-0684-46ba-954b-6027b18988f5"><a href="#c16" data-ref="a387ead4-3297-4943-9e87-2e584ac19f13">16</a>–<a href="#c19" data-ref="316bd30e-7e65-404e-a365-b7775093dd27">19</a></sup>. It is also unclear to what extent the requirement in mesoderm induction is direct, or indirect by modulating other signals<sup data-ref="17d11cc4-a799-4a4f-896b-49a2737cda9e"><a href="#c14" data-ref="4a4a12fd-37a2-41b0-823b-24465aed2102">14</a>,<a href="#c15" data-ref="742f7318-324d-43e1-8029-b6d79ab84495">15</a>,<a href="#c20" data-ref="a764eda8-5a8a-4d3c-be47-8623d858a254">20</a></sup>. In addition, several FGFs are co-expressed and the roles of specific FGFs are uncertain. Moreover, there may be significant interspecies differences in this respect. Notably, FGF8 is the only FGF which has been shown to be required for gastrulation in the mouse<sup data-ref="d03383f0-4092-4234-8d26-1f45a046b8d7"><a href="#c21" data-ref="794c68e0-2799-4097-897a-0f09d4d15b6e">21</a></sup>, and its requirement in gastrulation appears to be conserved across vertebrate model organisms<sup data-ref="f8b71f14-2972-42ba-9df0-a64f5feca27b"><a href="#c22" data-ref="86cf46c3-d11e-4f09-8c6b-689333920eaf">22</a>–<a href="#c24" data-ref="587e9f07-5519-4b98-962b-30ade0868304">24</a></sup>, yet FGF8 is not highly expressed in primate embryos<sup data-ref="56939c8a-2c90-4c46-834f-2bf3c42269bc"><a href="#c25" data-ref="0675581d-1047-4f31-9d4b-382fd07d0666">25</a>,<a href="#c26" data-ref="4dd40abf-fd32-4bdb-882c-1c522a0a4a26">26</a></sup>.</p><p id="el-p-11" data-ref="646d7491-c386-4657-a21d-9cc40b427bae" data-id="el-p-11">Mutants for FGF8, FGFR1, HSPG synthesis, and ERK2, as well as pharmacological inhibition of MEK and FGFR give rise to similar phenotypes in mouse gastrulation, suggesting that FGFR1 and FGF8 are the only FGF ligand-receptor pair required for gastrulation and that these act through the MAPK/ERK pathway<sup data-ref="1dee02c6-5b59-426b-a398-d6913d969710"><a href="#c12" data-ref="a268984c-08e9-45a7-bc3a-d8bdacf98fd0">12</a>,<a href="#c20" data-ref="9a4c2f32-72ec-43fe-8b31-c0c6a8e2fb03">20</a>,<a href="#c21" data-ref="e8483df4-158f-4512-aa2a-ad29b4921579">21</a>,<a href="#c27" data-ref="1d1ed612-6d55-4b04-ae51-dfef9b472dc0">27</a>–<a href="#c31" data-ref="62e32e5b-d724-493d-9c76-37b0d66ab81c">31</a></sup>. FGF4 expression during gastrulation is also lost in FGF8 mutants, leaving the possibility of a role for FGF4, but this remains unknown as FGF4 is also required in the blastocyst and mutants therefore do not reach gastrulation<sup data-ref="20c1c6ef-b96e-4b3f-ac31-fdeea6ebb109"><a href="#c32" data-ref="53f8327a-7e7a-4877-bbfc-51b2ab452a87">32</a></sup>. In all mutants and for pharmacological inhibition of FGF/ERK signaling, expression of the primitive streak marker brachyury (TBXT, also BRA, or T in mouse) is lost or severely diminished in the posterior streak and mesoderm fails to migrate away, piling up in this region. In addition, TBX6 expression is lost completely, and no lateral or paraxial mesoderm are formed. In contrast, TBXT expression in the anterior streak and extraembryonic mesoderm is not dependent on FGF/ERK signaling; extraembryonic mesoderm development is normal, and the axial mesoderm is expanded<sup data-ref="b7e05bff-8041-454f-9785-1eb73d74a9d8"><a href="#c20" data-ref="3ee1baa5-b267-42b9-8df8-71bb5c54a68d">20</a>,<a href="#c21" data-ref="8bee4257-a8b1-45dc-9af8-cf4e7b021ad5">21</a>,<a href="#c27" data-ref="9b2704a8-bd66-44c6-9dbb-0aa95e19113e">27</a>,<a href="#c28" data-ref="455bd1c8-1f2c-4f35-ad1f-80291333c311">28</a></sup>.</p><p id="el-p-12" data-ref="9b2edefd-4399-423e-97e9-ab31265d1c81" data-id="el-p-12">In vitro models offer an opportunity to disentangle the complexity of FGF signaling in human development. Here, we explore the role of FGF/ERK signaling in micropatterned human pluripotent stem cells treated with BMP4, also known as 2D human gastruloids. These give rise to concentric rings of different cell types associated with gastrulation after 2 days of differentiation. Pluripotent cells in the center are surrounded by a primitive streak-like ring, which itself is surrounded by primordial germ cells and amnion-like cells on the colony edge<sup data-ref="445e86b2-9d89-4d86-b57e-734c0ee4a38b"><a href="#c33" data-ref="8214e7ca-8ffa-4d44-bc85-88451a27924d">33</a>–<a href="#c35" data-ref="99ba116f-9848-4f12-8c5c-17ca68a570c4">35</a></sup>. Due to its simplicity and reproducibility, this system has proven powerful in deciphering the mechanisms underlying aspects of human gastrulation. It was used to show that BMP-Wnt-Nodal hierarchy that controls primitive streak formation during mouse gastrulation is preserved in human while providing a range of new insights into how these signals function<sup data-ref="a4e015f5-cfb3-4974-bb65-057ddffd1243"><a href="#c2" data-ref="23b37e7a-bc73-4ffa-ab8a-1507ee118d65">2</a>,<a href="#c3" data-ref="28ccf6c3-1db3-4e8d-9001-8c3d963d3b49">3</a></sup>. For example, the crucial role for BMP and Nodal receptor localization in patterning was discovered in 2D gastruloids and later confirmed in the mouse embryo. This system also revealed that BMP, Wnt, and Nodal signaling patterns are highly dynamic<sup data-ref="84ffa1bd-c764-4245-9a07-e103f0fb1faf"><a href="#c3" data-ref="9b144001-ccc9-4510-8057-e31972719e82">3</a>–<a href="#c7" data-ref="4362078e-e9f9-4dbf-bf38-2604af0fb967">7</a></sup>. It therefore also provides a promising approach to determine how FGF may function in human gastrulation.</p><p id="el-p-13" data-ref="10b2ef82-b265-474f-a826-f805fed45f2d" data-id="el-p-13" data-split-to="10b2ef82-b265-474f-a826-f805fed45f2d" data-split-original="true" data-last-split-element="true" data-align-last-split-element="auto">In the 2D human gastruloid model, we show that a ring of phosphorylated ERK forms in an FGF-dependent manner and expands largely in lockstep with the formation of primitive streak-like cells but extends further inward. We demonstrate that this pattern of ERK activity is due to localized FGF receptor (FGFR) activation that requires basal FGFR localization. We then show that exogenous FGF2 in the media is required for pattern formation to occur but can be removed well </p></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_right_page" data-page-number="3" id="page-3">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-id="article-content" data-split-from="726f25f1-fdd1-41b8-bf49-de736be3e3ae"><section data-ref="790b4fb7-01cd-4a65-93f2-98b59b2c0a72" data-id="s1" data-split-from="790b4fb7-01cd-4a65-93f2-98b59b2c0a72"><p data-ref="10b2ef82-b265-474f-a826-f805fed45f2d" data-id="el-p-13" data-split-from="10b2ef82-b265-474f-a826-f805fed45f2d">before the phosphorylated (active) ERK (pERK) and PS-like rings appear, suggesting the pERK ring instead depends on endogenous FGF gradients. Using single cell RNA-sequencing, we determine the expression of FGF pathway components including FGFR1 as the main receptor and FGF2, FGF4, and FGF17 as the most highly expressed ligands. Our analysis corroborates previous studies describing expression of FGF2 and FGF17 in human embryos and gastruloids but for the first time recognizes a possible role for FGF4 and includes a broad analysis of genes involved in FGF signaling modulation<sup data-ref="694f4a52-c8a3-4983-b128-29c6967659e5"><a href="#c25" data-ref="7541f572-3690-404d-a2f5-c3b39eb19223">25</a>,<a href="#c36" data-ref="92516521-5cbb-4c0d-ab3f-bef33542ebe1">36</a></sup>. We find that the FGF4 and FGF17 expression are restricted to the PS-like ring and that knockdown of FGF4 significantly reduces PS-like differentiation, while FGF17 knockdown primarily affects downstream mesoderm differentiation. Given its importance in the mouse, we also analyze FGF8 and find its expression is spatially and temporally displaced from FGF4 and PS-markers, while FGF8 knockdown leads to an increase in PS-like differentiation, suggesting it may counteract FGF4. In summary, we have greatly advanced our understanding of how FGFs function in a model for human gastrulation.</p></section><section id="s2" data-ref="c2323182-eb7e-45cc-b2f0-c40d361a838e" data-id="s2"><h1 class="heading-1" id="el-h1-3" data-ref="94622f3c-5666-42e4-adfb-974b4bb17390" data-id="el-h1-3" data-break-after="avoid">Results</h1><section id="s2a" data-ref="866c7dfe-80ac-446d-810f-f971eaa64035" data-id="s2a" data-previous-break-after="avoid"><h2 class="heading-2" id="el-h2-1" data-ref="11b35ab6-5e7a-44e3-af6c-52fb4bd2304e" data-id="el-h2-1" data-break-after="avoid" data-following="following-ff2c00ba-51a1-482d-9afa-f5a370dcfa50">FGF/ERK signaling is required for primitive streak-like differentiation</h2><p id="el-p-14" data-ref="fa059b58-a419-4158-a326-870d8f87cb33" data-id="el-p-14" data-previous-break-after="avoid">We previously observed that pharmacological inhibition of MEK (the kinase upstream of ERK) or FGF receptors leads to a loss of differentiation to primitive streak-like and primordial germ cell-like cells, suggesting a crucial role for FGF/ERK signaling<sup data-ref="21bae75d-dcea-4513-9a3c-e887b98d6b28"><a href="#c34" data-ref="1a145780-94f9-4abc-90f5-c6ffc1492848">34</a></sup>. Moreover, a ring of active ERK (pERK) has been observed in 2D gastruloids but its connection to FGF signaling or differentiation remains unclear<sup data-ref="f7f4645c-4e6a-400b-95b8-2c6e9865a63c"><a href="#c33" data-ref="040c207b-e31f-4517-b401-24c3af5e4f81">33</a></sup>. Therefore, we wanted to investigate the role of FGF/ERK signaling further.</p><p id="el-p-15" data-ref="01722b51-6ae5-48ac-b9c9-e36104eadf9f" data-id="el-p-15">To determine the dynamics of pERK and relate it to PS-like differentiation, we first performed time series immunofluorescence co-staining for pERK and the PS marker TBXT after BMP4 treatment in 700um diameter colonies. This revealed the pERK ring first emerges around 24h, around the same time as TBXT expression, and gradually gets brighter and wider (<span class="fakelink linktofig" data-ref="34e3f548-6291-4309-86fd-a2c172b6eb31"><span class="fakelink-text" data-ref="1b9e0cbd-5819-42d0-a341-9c90ff9d1d1b">Fig. 1a</span><a class="fake-icon" href="#fig1" data-ref="ffbb2b00-6fbf-4f40-a0ba-13a2438fcc33"><span class="fakelink-icon" data-ref="2d89ce29-3414-4c69-8a5f-83dcd5e02dbd">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). By 42h the outer edge of the ERK ring coincided with TBXT expression, while the inner edge was less sharp and extended further towards the center than TBXT, reminiscent of the domains of WNT and NODAL signaling that extend further in than the PS-like region<sup data-ref="b1a91842-a51c-474f-8937-9d6a9dae3117"><a href="#c3" data-ref="09dcf322-7225-47bb-b7e7-2f19ab46eda9">3</a>,<a href="#c6" data-ref="dd53c413-b2bd-4002-b4dc-5f87008db985">6</a></sup> (<span class="fakelink linktofig" data-ref="971444a6-fee1-4438-99ff-396b232a5717"><span class="fakelink-text" data-ref="9ad4f28e-62f1-4f19-914e-0780383f8105">Fig. 1a</span><a class="fake-icon" href="#fig1" data-ref="86ccaa48-b0bb-49ff-888e-7eb5f5f10dd3"><span class="fakelink-icon" data-ref="b3737295-39db-4fe2-9b3a-98cf15c51c2e">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Cell fate patterning in 2D gastruloids has a fixed length scale from the edge, with the central part absent in smaller colonies due to the fixed length scale of the BMP signaling gradient from the edge. Consistently, we found that ERK signaling also has a fixed pattern from the edge regardless of colony size, so that in smaller colonies the ERK ring becomes a domain of high ERK signaling extending through the colony center (<span class="fakelink linktofig" data-ref="51408b7d-a35e-4f10-a888-e1839bc19045"><span class="fakelink-text" data-ref="a281df27-3398-45bf-92fd-a33fefea2626">Supp. Fig. 1a</span><a class="fake-icon" href="#figS1" data-ref="582062fb-40a7-402f-b3f6-02fbe88350ce"><span class="fakelink-icon" data-ref="8d6e4ef0-5a9f-4364-a3e6-1a2a47f532c4">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). MEK or FGFR inhibition for 30 minutes after 41.5h eliminated ERK activity, suggesting ERK activity depends entirely on FGF and that continuous FGF signaling is necessary to maintain the ERK signaling pattern (<span class="fakelink linktofig" data-ref="cd9e2e9d-123e-401a-b1a8-8ec0b7be250e"><span class="fakelink-text" data-ref="799e01c2-cb4d-48a9-a073-d4c1d2fa26a5">Fig. 1bc</span><a class="fake-icon" href="#fig1" data-ref="7f318b09-a692-4221-ab6e-400a23d7c5b8"><span class="fakelink-icon" data-ref="bf80dcac-b013-4918-89c6-6b5846aa3a0b">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Inhibition for the full duration of differentiation eliminated TBXT expression as well as other PS markers (<span class="fakelink linktofig" data-ref="d1f4ef9c-41d2-4c0f-8dea-71e40e428a1e"><span class="fakelink-text" data-ref="f13341ea-02c7-46ca-8b9a-b50bf6292261">Fig. 1bc</span><a class="fake-icon" href="#fig1" data-ref="ee5e22e5-c742-461c-9bf5-61324ecb69c4"><span class="fakelink-icon" data-ref="74ff6ff5-7950-4e6f-8ec3-6647db654c3d">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="68ddb468-cdc3-4294-b046-de8832c1e4f2"><span class="fakelink-text" data-ref="dd4b45fc-6c49-48fb-88c7-8036934cb8cb">Supp. Fig. 1b</span><a class="fake-icon" href="#figS1" data-ref="9be9fed2-d3e6-4bda-a297-8a02d789a76b"><span class="fakelink-icon" data-ref="e0d0099b-f80a-4613-919b-113b72f3369c">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). However, we were able to rescue TBXT expression by directly and uniformly activating ERK (<span class="fakelink linktofig" data-ref="60e6e364-498b-4348-b0ad-b6a09a15900b"><span class="fakelink-text" data-ref="c61eae97-1813-491b-9718-a73d7e318538">Fig. 1d-f</span><a class="fake-icon" href="#fig1" data-ref="475d1146-06ef-4e7e-acba-9f4aa34df9f3"><span class="fakelink-icon" data-ref="98f854df-de90-4ac6-8a2e-b8fd2bba8e24">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="5489b7ac-4280-425f-af90-659fd6108d29"><span class="fakelink-text" data-ref="3ecba870-9713-41d0-aa89-b2fd5a773481">Supp. Fig. 1cd</span><a class="fake-icon" href="#figS1" data-ref="ffc990a0-50a9-44df-b486-5e11a71efd1c"><span class="fakelink-icon" data-ref="9aa38862-0182-4c0b-93f5-7cbefa136ae0">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>) in cells stably expressing the doxycycline-inducible membrane-targeted catalytic domain of SOS (dox-SOS<sup data-ref="943309b2-a8ad-44bb-966a-05b0a2b01b6e">cat</sup>)<sup data-ref="6c75bcc1-20dc-4acd-8826-aa9c4653f2f9"><a href="#c37" data-ref="3175deed-86dd-4d39-acc2-adb0ff2a51a5">37</a></sup>. Despite uniform induced ERK activity, TBXT expression remained excluded from the center, showing that ERK activation alone is not sufficient for TBXT induction. However, TBXT levels were much higher in the rescue (<span class="fakelink linktofig" data-ref="b1942e8d-ade7-4c2f-bd72-28de6fd0c6a7"><span class="fakelink-text" data-ref="16f5efa3-20a4-4170-902e-a5e1017035ba">Supp. Fig. 1cd</span><a class="fake-icon" href="#figS1" data-ref="be2b4b5c-acad-4c49-ac42-38d6e20e9313"><span class="fakelink-icon" data-ref="6b4a85b5-70a2-49c2-a2e9-6c13a0425733">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>), possibly reflecting the much higher level of ERK activity induced by dox-SOS<sup data-ref="6aedf548-f427-4c17-b2a9-569e84b81352">cat</sup>. MEK and FGFR inhibition also led to a reduction in cell number which staining for BrdU and cleaved Cas3 suggested was primarily due to reduced proliferation (<span class="fakelink linktofig" data-ref="69ce898a-5e48-4269-8d1e-c96c1a82e130"><span class="fakelink-text" data-ref="ed1fecd3-680e-4523-a859-69a256bb102f">Supp. Fig. 1ef</span><a class="fake-icon" href="#figS1" data-ref="1157bfd5-d653-4067-a4f1-6eeafba3df08"><span class="fakelink-icon" data-ref="33d5200c-6988-4811-9d6b-b49baa645a4f">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). In contrast to TBXT expression, reduced cell number caused by FGFR inhibition was only partially rescued by dox-SOS<sup data-ref="6c59f1fe-987e-46f0-89de-0f5800ef801c">cat</sup>, indicating additional downstream pathways regulate cell number (<span class="fakelink linktofig" data-ref="5343b0b5-f08c-4cf7-b2ef-2f4213593db3"><span class="fakelink-text" data-ref="7f4c4aa1-a09b-403d-ab5b-3afce2400ab0">Fig. 1e</span><a class="fake-icon" href="#fig1" data-ref="c4360973-6028-48f4-bd7c-10d907f29e2c"><span class="fakelink-icon" data-ref="4d419025-7416-4776-9e27-0323d2888371">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Altogether, these data show the formation of a PS-like ring is dependent on FGF signaling through ERK.</p><p id="el-p-17" data-ref="8e0cbdfd-59f9-48e7-b877-8e173ae419c2" data-id="el-p-17">This raised several questions. First, is FGF/ERK signaling required directly for PS-like differentiation, or does it act indirectly? These possibilities are not mutually exclusive. For example, FGF/ERK could be required directly but also act indirectly by controlling Wnt or Nodal expression, as both Wnt and Nodal signaling are required for PS-like differentiation. Second, </p></section></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_left_page pagedjs_pagedjs-filler_page page-grid" data-page-number="4" id="page-4">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><figure class="figure tofill imgs-1 longcaption fig-portrait pagedjs-fill-next-page finishes" id="fig1" data-ref="6b4b05b4-0bd8-489c-8b02-4f729eb11503" data-id="fig1" style="display: block; position: unset;"><figcaption class="figure__caption" id="el-figcaption-0" data-ref="ff235660-d0b2-496e-8134-ce02057ec1c9" data-id="el-figcaption-0"><h3 id="el-h3-0" data-ref="f27cee53-6f4e-401f-8983-4605449ef255" data-id="el-h3-0" data-break-after="avoid"><span class="label figure-name" data-ref="0e27e2e6-ce1c-4ba0-bb58-50abc64fd442"><label class="figure__label" data-ref="5d980441-9a6b-47b7-b49a-8fa3d40e2126">Figure 1.</label></span></h3><p id="el-p-16" data-ref="4d0bdaaf-61fe-46f0-a036-38521e673e43" data-id="el-p-16" data-previous-break-after="avoid"><strong data-ref="6144b037-de80-4538-8f27-8886fa176be3">a)</strong> Immunofluorescence stainings for pERK and TBXT and their radial intensity profiles in 2D gastruloids at different times. Error bars in graphs are standard deviations over N=4 colonies. <strong data-ref="9c7c4b0e-6f20-4831-ad3b-588b1c127553">b,c)</strong> Effect FGF receptor inhibition (FGFRi, b) or MEK inhibition (MEKi, c) on pERK and TBXT in colonies fixed at 42h after BMP treatment. Inhibition for either 30 minutes at 41.5h (rounded to 42 in figure) or throughout differentiation. <strong data-ref="b90c60c3-5e15-4925-8b5a-9a10abe7887d">d)</strong> BMP only control and FGFRi inhibition in cells expressing doxycycline-inducible SOS<sup data-ref="98499325-0a07-47c9-9177-3b3da6355fbd">cat</sup> with and without addition of doxycycline. <strong data-ref="1e07ebf4-2307-4739-b16a-5cbba2494768">e-f)</strong> Cell numbers (e) and radial profiles of TBXT positive cells (f) for conditions in (d). Error bars represent standard deviation. Statistical significance was assessed using one-way ANOVA followed by Tukey’s HSD test for pairwise comparisons. <strong data-ref="b068f09c-9a60-4848-88e1-7f92457e9d20">g)</strong> PS marker TBXT, pERK and pluripotency marker SOX2 in Nodal knockout cells with inhibitors of Wnt secretion (WntSeci) and BMP receptors (BMPRi) with or without MEKi or FGFRi. All scale bars 50um.</p></figcaption><img class="child-of-figure has-boundary imageonly portrait max-100" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_fig1.tif/full/max/0/default.jpg" alt="" id="el-img-1" data-ref="4760eca6-0cd0-4033-b262-f501a6160eea" data-id="el-img-1" height="2800" width="2112" data-desired-width="100" style="height: 708px; max-width: 100%; width: auto; object-fit: contain;"></figure></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_right_page" data-page-number="5" id="page-5">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-id="article-content" data-split-from="726f25f1-fdd1-41b8-bf49-de736be3e3ae"><section data-ref="c2323182-eb7e-45cc-b2f0-c40d361a838e" data-id="s2" data-split-from="c2323182-eb7e-45cc-b2f0-c40d361a838e"><section data-ref="866c7dfe-80ac-446d-810f-f971eaa64035" data-id="s2a" data-split-from="866c7dfe-80ac-446d-810f-f971eaa64035"><p data-ref="8e0cbdfd-59f9-48e7-b877-8e173ae419c2" data-id="el-p-17" data-split-from="8e0cbdfd-59f9-48e7-b877-8e173ae419c2">which are the FGFs activating the ERK pathway? The pathway could be activated by exogenous FGF2 present in the pluripotency medium mTeSR1 or by endogenously expressed ligands. Third, how is ERK activity restricted to a ring?</p><p id="el-p-18" data-ref="50069754-d2c4-4ca2-bdd4-ce5ed59aaaa4" data-id="el-p-18">To determine if there is a direct requirement of FGF/ERK signaling and whether this depends on exogenous FGF, we performed differentiation to PS-like cells in standard culture by exogenous activation of Wnt and Nodal signaling, bypassing the need to form endogenous Wnt and Nodal gradient as in 2D gastruloids<sup data-ref="a60517d4-259b-4cbb-8a21-208a85841985"><a href="#c38" data-ref="5e2eea70-9977-4cd0-9b76-c9dd9ec7cb2c">38</a></sup>. We used a minimal base medium (E6) without exogenous FGF or other growth factors. To rule out any additional requirement for endogenous Nodal and Wnt which may be FGF/ERK-dependent, we used Nodal knockout cells<sup data-ref="ab069c0e-d535-493d-b82a-ecd8a2c12f1c"><a href="#c3" data-ref="6e289ae1-0e73-4bc5-bb10-61ccd8838f8a">3</a></sup> treated with the Wnt secretion inhibitor IWP2, thereby eliminating any endogenous activity of these pathways. We also inhibited BMP receptors using LDN193189 to rule out any indirect effect through endogenous BMP. Under these conditions, with exogenous Wnt and Nodal activation for 24h in the absence of exogenous FGF, cells efficiently differentiated to TBXT+ PS-like cells (<span class="fakelink linktofig" data-ref="28099cb2-5e7a-48ba-ba0b-88fc27f1b789"><span class="fakelink-text" data-ref="4b80783e-2ea7-41b4-b78d-17fed4a9726a">Fig. 1g</span><a class="fake-icon" href="#fig1" data-ref="890976f1-b5b0-4843-a1fb-2dcf26001ead"><span class="fakelink-icon" data-ref="239c3ff8-dbc9-4da2-8755-978d74d8c03b">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="3b2cd05f-24a3-407c-b0ec-307bc4188581"><span class="fakelink-text" data-ref="7103957e-87fe-4974-85a6-5434ffb04cbf">Supp. Fig. 1g</span><a class="fake-icon" href="#figS1" data-ref="6eeb2700-f362-4ee9-8d09-e2ee532710e7"><span class="fakelink-icon" data-ref="5e1c4ea4-3109-4070-afd3-48a81cb6200d">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). These PS-like cells had high active ERK levels just as in the 2D gastruloids. Inhibition of either FGFR or MEK abrogated ERK activity and TBXT expression while FGFR inhibition also severely reduced proliferation. In both cases significant expression of the pluripotency and ectoderm marker SOX2 was maintained (<span class="fakelink linktofig" data-ref="7a076250-829b-4ea6-9b90-fbcdb34c8515"><span class="fakelink-text" data-ref="6eb09e3f-512a-4934-ba91-0017b9a89de6">Supp. Fig. 1g</span><a class="fake-icon" href="#figS1" data-ref="ba3ad64d-0be4-464d-96eb-4e3b5725f0f7"><span class="fakelink-icon" data-ref="f75d391f-8a41-4fba-ad66-d08b2eb33677">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Given that FGF/ERK is known to be involved in pluripotency maintenance, this suggests inhibition may not be complete and that much lower ERK activity is required for pluripotency maintenance than PS-like differentiation, consistent with lower ERK activity in the pluripotent center of gastruloids.</p><p id="el-p-19" data-ref="f274fe20-7125-40b5-8a7b-45cebac52aa7" data-id="el-p-19">MEK and FGFR inhibition reduce proliferation and thereby cell density, raising the possibility that their lack of differentiation is an indirect effect that depends on cell density: a critical density may be required for differentiation due to a community effect. To rule this out we performed the experiment across a range of densities, which confirmed that PS differentiation in the control was efficient even at the lowest densities while FGFR and MEK inhibition effectively blocked differentiation when compared to controls with similar final cell densities (<span class="fakelink linktofig" data-ref="de17bbbd-3de7-45ea-bb7d-e445e2e73445"><span class="fakelink-text" data-ref="ddc13506-75ed-4468-8046-ecc5e19028e2">Supp. Fig. 1h</span><a class="fake-icon" href="#figS1" data-ref="1ae5ae31-728b-46d8-9568-d10052429f74"><span class="fakelink-icon" data-ref="17c3b0c2-e1ee-4fb9-8b11-7b2d6877582c">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). We also tested whether proliferation and differentiation require similar levels of ERK signaling by determining the MEKi dose response of gastruloids. This revealed that lower doses of MEKi effectively block differentiation without strongly affecting cell number, further supporting that the effects of ERK signaling on proliferation and differentiation are independent (<span class="fakelink linktofig" data-ref="cf4ecd67-48d6-4e36-b760-15ad4ee465ca"><span class="fakelink-text" data-ref="cf0a44af-2a07-4406-98cd-aae4a56ce5d9">Supp. Fig. 1i</span><a class="fake-icon" href="#figS1" data-ref="519db83a-05de-423c-b6b9-e3a6b127b8b3"><span class="fakelink-icon" data-ref="12e4b407-6789-484e-b071-3ba6080b3980">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Altogether, our results show that endogenous FGF expression is both necessary and sufficient for PS-like differentiation by exogenous Wnt and Nodal stimulation and demonstrate a direct requirement for ERK signaling in PS-like differentiation.</p></section><section id="s2b" data-ref="921dc64d-5d0b-4e06-b347-13a7f32896c1" data-id="s2b"><h2 class="heading-2" id="el-h2-2" data-ref="cfe8fc0a-9735-42c4-b292-4d978be4e396" data-id="el-h2-2" data-break-after="avoid">Single cell RNA-sequence reveals differentially expressed FGF pathway components</h2><p id="el-p-20" data-ref="9e5f2bdb-0ad0-4c6f-8a50-1f5b390ec298" data-id="el-p-20" data-previous-break-after="avoid">The above results suggest endogenous FGF expression may contribute to elevated ERK signaling in the PS-like ring. However, FGF/ERK signaling may be modulated at many levels (<span class="fakelink linktofig" data-ref="0e27eb01-7dfa-4e54-a79d-b4429f90affd"><span class="fakelink-text" data-ref="0b30c038-3fa0-4775-a934-00d147663cc1">Fig. 2a</span><a class="fake-icon" href="#fig2" data-ref="d4d9eefc-4e37-4f68-b0a2-a548ccc30353"><span class="fakelink-icon" data-ref="58248a51-f47f-47e4-8704-d1ee99a25636">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). A spatial pattern of ERK activation could reflect a concentration gradient of FGF. Alternatively, differential expression of receptors and HSPG-modifying enzymes could lead to spatially patterned receptor activation and downstream ERK activity even in the absence of an FGF gradient. Even if receptor activation is uniform, downstream ERK activity could be spatially patterned by downstream negative regulators of ERK activity such as Sprouty-family proteins (SPRY) and dual-specificity phosphatases (DUSPs)<sup data-ref="05639086-d987-4771-9351-e54b0f61af95"><a href="#c40" data-ref="f3311c74-81b1-4278-8d57-d86bd2c64821">40</a></sup>.</p><p id="el-p-22" data-ref="984ec47b-4290-452e-b0fc-9a095861c0a9" data-id="el-p-22">To understand which genes modulating FGF/ERK signaling could be responsible for the elevated ERK signaling in TBXT positive cells and required for TBXT expression (<span class="fakelink linktofig" data-ref="f3b7123b-99f0-441d-858d-8aa14f589f0e"><span class="fakelink-text" data-ref="c2a58514-d1a9-4c4a-ad01-102b413cf787">Fig.1</span><a class="fake-icon" href="#fig1" data-ref="8dd2dcb4-0fa5-4055-ae39-e82db2588c60"><span class="fakelink-icon" data-ref="dc4f7c10-1463-4083-b718-38ef9ec97407">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="c9a0cbff-58ea-4caa-8cc5-ff66674248ac"><span class="fakelink-text" data-ref="9bd0f181-fddb-4d41-b9c5-cf25474ae05d">Fig. 2b</span><a class="fake-icon" href="#fig2" data-ref="fef66da1-056f-4e76-bdea-1787e0f8185e"><span class="fakelink-icon" data-ref="84286886-a3a4-4831-972e-b80b5502ed7b">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>), we used single cell RNA-sequencing data from one previously published sample<sup data-ref="26247d56-23c1-471f-ab10-dcf3d5154c77"><a href="#c34" data-ref="a862e1eb-53be-4f65-b2ea-86e981ce98d6">34</a></sup> and one newly collected replicate (<span class="fakelink linktofig" data-ref="6e2ec2f7-e6d4-4356-90e2-fda61911049d"><span class="fakelink-text" data-ref="beb34b40-e5e7-4cec-8815-884730eb1c73">Supp. Fig. 2a</span><a class="fake-icon" href="#figS2" data-ref="19dc5b38-b92f-49af-9443-0de96d3d18bb"><span class="fakelink-icon" data-ref="2ce64080-0d6b-4371-b3f6-6f71fcb72474">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>) to determine expression of FGF pathway components in human 2D gastruloids, extending earlier work describing FGF expression<sup data-ref="c4bec304-7368-48a4-b169-1ff2e5e46037"><a href="#c25" data-ref="453ef03d-d05a-480c-932f-da91dd51e12c">25</a>,<a href="#c35" data-ref="9720bf41-955b-4721-82dc-365af3255562">35</a>,<a href="#c36" data-ref="01293238-8c0c-46ca-9d98-384df553d091">36</a>,<a href="#c41" data-ref="2d7a84d4-e284-4d03-8c0a-31dfbda6a14b">41</a></sup>. TBXT-positive PS-like cells give rise to nascent mesoderm expressing TBX6 as well as definitive endoderm expressing SOX17 and FOXA2, both of which may still express TBXT at 42h<sup data-ref="b6f9847f-6c73-4933-bdb3-119e2ff4598a"><a href="#c42" data-ref="ac631ac9-9d4a-4aed-af53-2216c5ea5c2a">42</a></sup>. To understand how <em data-ref="6834592c-ba3e-42da-ba6f-e654bc633b9c">TBXT</em> expression is initiated in an FGF-dependent manner, we focused on differential expression in the PS-LCs relative to pluripotent cells, reasoning that FGF modulation specific to nascent </p></section></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_left_page pagedjs_pagedjs-filler_page page-grid" data-page-number="6" id="page-6">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><figure class="figure tofill imgs-1 longcaption fig-landscape pagedjs-fill-next-page finishes" id="fig2" data-ref="0b4d6bd6-67d8-4594-bea8-6adc45883ff8" data-id="fig2" style="display: block; position: unset;"><figcaption class="figure__caption" id="el-figcaption-1" data-ref="a9c01dfb-a630-49f3-90bb-2c20b921dfb6" data-id="el-figcaption-1"><h3 id="el-h3-1" data-ref="60ea95c4-0e04-4cd7-a150-28d2755b49a1" data-id="el-h3-1" data-break-after="avoid"><span class="label figure-name" data-ref="489a2676-92f8-457b-93e3-d32307ebb7a1"><label class="figure__label" data-ref="de96ebfa-2e73-4699-87f1-3616be0f1175">Figure 2.</label></span></h3><p id="el-p-21" data-ref="91381679-f54a-4ab7-a0eb-91b6c694bc1b" data-id="el-p-21" data-previous-break-after="avoid"><strong data-ref="e8267bb2-a876-4c75-b8c3-9eecf3affb51">a)</strong> Schematic of the FGF/ERK signaling pathway. <strong data-ref="1cb2104d-db7e-4608-b544-06d7f1885c0e">b)</strong> Distribution of pERK levels in TBXT+ cells versus ISL1+ and other cells based on data and thresholds in <span class="fakelink linktofig" data-ref="e3c96de8-b0e8-402c-93ba-f39879c9d498"><span class="fakelink-text" data-ref="916ebffb-8a29-4feb-a647-199cdb2aa07c">Supp. Fig. 2b</span><a class="fake-icon" href="#figS2" data-ref="703b1469-fcd0-4a06-9f5b-85ef260ce77a"><span class="fakelink-icon" data-ref="e9e25641-c0a5-4034-bc3f-01c80300b664">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. <strong data-ref="96fdf513-d2c0-461f-a383-475fec7390d5">c)</strong> Expression of key markers in PHATE<sup data-ref="416786f6-38ad-4446-a83f-74797f92b158"><a href="#c39" data-ref="97786a28-60f7-4849-89ef-ce24531b71ee">39</a></sup> projection of scRNA-seq data for 42h colony. <strong data-ref="a36422e1-af55-4d7f-9bcc-257d14fc44be">d)</strong> Cell type annotation. PS-LC: primitive streak-like cells, NasMe: nascent mesoderm, AE-LC: amniotic ectoderm-like cells: PGC-LC: primordial germ cell-like cells, endo: definitive endoderm. <strong data-ref="e7105e8f-1320-4013-9103-e87adc3fe8c4">e)</strong> Expression of canonical FGF ligands (log transformed and smoothened). <strong data-ref="549ba5e7-c91b-483a-a75d-a91c61901a4a">f)</strong> Comparison of FGF ligand expression in 2D human gastruloids to human, monkey, and mouse embryo, normalized within each type of sample (see methods). <strong data-ref="c7040d38-2a4d-4d3f-9bb7-2024f809d92f">g-i)</strong> Expression of several genes involved in FGF signaling with strong differential expression between PS-like cells or nascent mesoderm and pluripotent cells, split by gene category.</p></figcaption><img class="child-of-figure has-boundary imageonly landscape max-100" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_fig2.tif/full/max/0/default.jpg" alt="" id="el-img-2" data-ref="b70215fb-b6d6-4fd7-8a41-856762a4f96a" data-id="el-img-2" height="931" width="2102" data-desired-width="99.52651515151516"></figure></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_right_page" data-page-number="7" id="page-7">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e" data-split-to="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-id="article-content" data-split-from="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-split-to="726f25f1-fdd1-41b8-bf49-de736be3e3ae"><section data-ref="c2323182-eb7e-45cc-b2f0-c40d361a838e" data-id="s2" data-split-from="c2323182-eb7e-45cc-b2f0-c40d361a838e" data-split-to="c2323182-eb7e-45cc-b2f0-c40d361a838e"><section data-ref="921dc64d-5d0b-4e06-b347-13a7f32896c1" data-id="s2b" data-split-from="921dc64d-5d0b-4e06-b347-13a7f32896c1"><p data-ref="984ec47b-4290-452e-b0fc-9a095861c0a9" data-id="el-p-22" data-split-from="984ec47b-4290-452e-b0fc-9a095861c0a9">mesoderm or endoderm is a consequence rather than a cause of PS-LC differentiation. We identified different cell types based on marker genes with the PS-like cells defined as <em data-ref="1be067db-c701-40a8-8093-cbcf79509e59">TBXT+MIXL1+POU5F1+TBX6-SOX17-FOXA2-</em>(<span class="fakelink linktofig" data-ref="f872b55c-aabd-46d3-b00c-fbaec0186164"><span class="fakelink-text" data-ref="f861132e-5a63-4afd-8185-7a8e743b1c66">Fig. 2cd</span><a class="fake-icon" href="#fig2" data-ref="4358d432-3df8-4926-86af-b25776b84398"><span class="fakelink-icon" data-ref="47e681b1-0119-4a00-90a9-ac019811e447">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="9924934f-13b9-4e06-96c6-c3fd8ac841ec"><span class="fakelink-text" data-ref="229504a2-26e6-48e8-8e1f-a7ff1bb2cde3">Supp. Table 4</span><a class="fake-icon" href="#tblS4" data-ref="2bfce3ea-7146-4d31-b20d-8fb333889636"><span class="fakelink-icon" data-ref="40666c1d-97a8-4221-931a-42c4123e184e">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). We then analyzed expression of different categories of FGF/ERK regulators.</p><p id="el-p-23" data-ref="aee6f288-ecbc-4633-adc4-9d0f611de2ab" data-id="el-p-23">We found four FGF ligands with significant expression: <em data-ref="3f71d0e0-454c-48b8-a560-06a625b30aaf">FGF2, 4, 8,</em> and <em data-ref="d790d46c-e6d6-42eb-99c1-ee6b0b2b327e">17</em> (<span class="fakelink linktofig" data-ref="13cee253-3256-4c84-b632-f83255ab3294"><span class="fakelink-text" data-ref="9769c11f-ad76-4367-a1c2-b397eaf8e684">Fig. 2e</span><a class="fake-icon" href="#fig2" data-ref="c9593933-3f66-4360-b1b5-06f81e69d946"><span class="fakelink-icon" data-ref="23a2de28-5244-4c1d-b995-05670f46bab7">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="fc09b5fe-1790-4a9b-8141-0d948e06e93c"><span class="fakelink-text" data-ref="0fe52171-0f32-433c-8098-db228aaa752d">Supp. Fig. 2b</span><a class="fake-icon" href="#figS2" data-ref="9ad8e2db-4521-450b-8c08-dd5c77862a06"><span class="fakelink-icon" data-ref="44f7eb85-eed6-47df-932d-6edc1fc62e9e">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). We then compared the expression of these ligands in PS-like cells to reference data of CS7 human, CS8 monkey, and E6.5-E7.5 mouse gastrulation (<span class="fakelink linktofig" data-ref="6c756a29-9d15-4072-8e17-4ae8640164d1"><span class="fakelink-text" data-ref="336df8ab-fb14-4573-82b8-209de76f840f">Fig. 2f</span><a class="fake-icon" href="#fig2" data-ref="379fa510-2b51-41fe-92c7-9c847e483d0d"><span class="fakelink-icon" data-ref="cc8493b7-3bb5-4b3a-99b2-2b3ab61f9488">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="44d6ab9e-e1ca-4eb9-ad46-386bf9efdd9d"><span class="fakelink-text" data-ref="c289f0ff-b3b2-42f6-ba86-f26d41c01a70">Supp. Fig. 2cd</span><a class="fake-icon" href="#figS2" data-ref="9f69650e-7241-4101-a217-00054d3514e7"><span class="fakelink-icon" data-ref="9efbfc61-0b0e-4893-a660-a0a88c6c8993">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). We also included <em data-ref="cf3f14b9-3670-4c1f-b93f-a49c4b880ae4">FGF3</em> and <em data-ref="8147d87d-218b-4c80-9547-50fd299e9bad">FGF5</em>, which are expressed during mouse gastrulation<sup data-ref="8622ddb1-9d3c-478a-b69d-86b1b35b365e"><a href="#c32" data-ref="c2f6e2da-d019-47a6-9b8b-c7c9c34636ba">32</a>,<a href="#c43" data-ref="0500ad32-e022-466d-84c5-52fee0721e60">43</a>,<a href="#c44" data-ref="6b0b3e48-2330-48d7-8fbd-ffbae9217253">44</a></sup>. High expression of <em data-ref="24976637-27ec-4772-b2c0-00a389ce3e13">FGF4</em> and <em data-ref="fdaaf24d-d141-499f-81ab-5701f128f489">FGF17</em> and low expression of FGF8 was consistent between human, monkey, and 2D gastruloid. <em data-ref="e48de8b2-0ee5-48e6-bf08-1f648235eed2">FGF2</em> expression was also high in the PS-like populations in 2D gastruloids and the CS8 monkey embryo, but much lower in the human embryo data. Furthermore, the monkey and human embryos expressed <em data-ref="e87d1f3e-e936-4ae9-a61a-fe763f4988df">FGF3</em>, which was minimally expressed in gastruloids. A similar comparison in nascent mesoderm showed much lower <em data-ref="700fd0ff-8f77-4ffd-99c4-9ad9a880dfcc">FGF2/4</em> and much higher <em data-ref="cc174fd2-b371-4a2a-b035-e55b6bcae83b">FGF17</em> compared to PS across all primate samples (<span class="fakelink linktofig" data-ref="0e967b29-afac-4106-8225-dd6ff36c6bc7"><span class="fakelink-text" data-ref="cfebc816-e75e-4a3b-95d5-cb2326da3ae0">Supp. Fig. 2e</span><a class="fake-icon" href="#figS2" data-ref="c2fbb7aa-e1a1-4698-93b1-2ab6bed2df05"><span class="fakelink-icon" data-ref="8b5dc7c4-19a2-412f-8c33-ad01bd56e382">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Expression in 2D gastruloids was consistent with published RNA-seq data for hPSC-derived PS-like cells, which expressed high <em data-ref="d5988da5-8ca0-41d0-9895-4665c39bda46">FGF2</em> and <em data-ref="5f16546d-8d49-4e69-8697-c4f99c7ddc1a">FGF4</em>, some <em data-ref="5ee1132f-9d4b-4e49-a947-9795264a27e0">FGF8</em> and <em data-ref="827238ef-4e85-418e-8379-8eaeea473c16">FGF17</em>, and no <em data-ref="7455a114-de4f-408b-a455-e80270ac2d4c">FGF3</em> (<span class="fakelink linktofig" data-ref="e82e38f6-98bb-4d06-aeba-8abd9404ca21"><span class="fakelink-text" data-ref="3afbc49d-6524-4df1-a522-d987f9bbf501">Supp. Fig. 2f</span><a class="fake-icon" href="#figS2" data-ref="fa368af4-af21-4853-a7f7-55efe0609f94"><span class="fakelink-icon" data-ref="cb7648be-d330-4d8c-b0e9-22e410ca6d06">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Comparison with mouse confirmed significant differences from primates, with high expression of <em data-ref="88672081-4ed9-49d2-a25e-cab06a69a5d0">FGF5, 8</em>, much lower expression of <em data-ref="4f28de84-869a-4051-ac56-401b4e0e31a3">FGF4,17</em>, and no expression of <em data-ref="cba5c2db-4740-4605-b8a8-6e0af3909dd1">FGF2</em> in the mouse<sup data-ref="9aa15ca5-d634-4fb0-b40a-60d22a963bae"><a href="#c21" data-ref="bbc3080a-90d1-450e-8bd4-1f579972c975">21</a>,<a href="#c43" data-ref="9fe28a10-293f-4d8c-b33d-2b2eaf9de739">43</a>,<a href="#c44" data-ref="e7aed878-d62f-4e72-96bb-3c4fc6f6790e">44</a></sup>.</p><p id="el-p-24" data-ref="3c5f87af-6823-4cbc-9dca-78868e6f08a3" data-id="el-p-24">Of the FGF receptors, only <em data-ref="815eb87e-82d5-4cb9-aa70-d79abf07b418">FGFR1</em> and <em data-ref="262d785c-acbe-410e-9f1d-bd81f3f0427b">FGFR4</em> were strongly expressed, with <em data-ref="4af28172-f257-4a50-b182-3638d0be755f">FGFR1</em> expression about 5-fold higher (<span class="fakelink linktofig" data-ref="f40a2b0e-ad50-416c-8570-5f582f6526a1"><span class="fakelink-text" data-ref="5ec2a0de-5325-4892-99cc-178a8c9d1084">Fig. 2g</span><a class="fake-icon" href="#fig2" data-ref="559d051e-2360-4b58-a7fb-c902370d75d1"><span class="fakelink-icon" data-ref="fc0f3061-30f7-41d6-9a54-3cc3706b3432">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="864fda46-bfb9-4f4f-8618-1e74e0d7a769"><span class="fakelink-text" data-ref="6d18eaf5-f445-456f-ad70-e0c1e7a1918c">Supp. Fig. 2g</span><a class="fake-icon" href="#figS2" data-ref="135027dd-7d2c-4d2b-8ed1-3a50a7cfe046"><span class="fakelink-icon" data-ref="88f8419f-ff83-4a21-a5a7-5a2c160d8354">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Both receptors showed graded expression that decreased from pluripotent to PS-like cells and much lower expression in amnion-like cells (<span class="fakelink linktofig" data-ref="70784367-e297-4de0-a836-664a27f64e1e"><span class="fakelink-text" data-ref="7e794d4d-1a48-4c0c-a94f-63d0dd084dab">Fig. 2g</span><a class="fake-icon" href="#fig2" data-ref="74c8f798-f58b-43c5-ae1b-50c8ecd92e25"><span class="fakelink-icon" data-ref="7e54b63a-52aa-4706-a1b7-9443ec6165c5">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Therefore, elevated ERK activity in PS-like cells cannot be accounted for by overall receptor expression. Differential expression of genes regulating heparin-sulfate proteoglycans which modulate FGF receptor binding was mild except for glypicans 3 and 4 which were downregulated in PS-like cells relative to pluripotent cells (<span class="fakelink linktofig" data-ref="226ddddf-e3c1-4eba-b104-88fee4de25f1"><span class="fakelink-text" data-ref="07bc4223-24b1-4b65-a5eb-9ec0f082ff88">Fig. 2h</span><a class="fake-icon" href="#fig2" data-ref="75738fe9-06e0-4c80-9fa0-dc6449a6ce2c"><span class="fakelink-icon" data-ref="5bce0fee-bf83-46c6-8aad-45c859f3c75a">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="454eb9cb-d425-41b4-8ecb-07390c3106f2"><span class="fakelink-text" data-ref="b71b494d-56c2-46d7-99cc-8020481c65e5">Supp. Fig. 2g</span><a class="fake-icon" href="#figS2" data-ref="8c1eae34-5a91-441e-a5c7-6e278675088d"><span class="fakelink-icon" data-ref="9fe7eb5d-0f08-4d27-becc-3a63e6700e5b">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). In contrast, several intracellular negative regulators of ERK signaling were strongly differentially expressed between cell clusters, including <em data-ref="5a151048-54ea-4a46-813a-bc9008c2be25">SPRY1</em> and <em data-ref="90d8f61d-3ab6-4445-a8cf-0ccdb8228a9c">PEBP1</em> which were significantly higher in pluripotent cells (<span class="fakelink linktofig" data-ref="a27e20ab-48a1-4ce3-bbab-bc0c3008bfa1"><span class="fakelink-text" data-ref="ce98f383-5a01-4a20-8607-3e0a19bcb835">Fig. 2i</span><a class="fake-icon" href="#fig2" data-ref="b01e9d93-4af3-4499-8652-bd23825cf163"><span class="fakelink-icon" data-ref="5d628ec1-8284-4a71-8ea8-5d80575732ad">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="778587a2-a95d-4489-b9cd-578e1c28a292"><span class="fakelink-text" data-ref="7185d9f8-5e6a-45c6-b36e-4e072b65bf0a">Supp. Fig. 2g</span><a class="fake-icon" href="#figS2" data-ref="d9c87dea-8c1d-4e1c-8879-d1b1e2fb0399"><span class="fakelink-icon" data-ref="c9c83e07-e1ce-401b-b3e3-25c755ef1135">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Surprisingly, however, expression of <em data-ref="01b18b09-766e-452f-9578-2294d0b2466a">SPRY4</em> was not elevated in PS-like cells where ERK signaling is high, suggesting that transcription of <em data-ref="1f61f3dc-8e41-4c99-b58c-d9db21fb629d">SPRY4</em> may not be a good readout of ERK activity in human hPSCs like it is in the mouse<sup data-ref="29f4b7bf-4fc1-44c0-acbc-8a4dacd61a48"><a href="#c45" data-ref="b898f7d3-f69a-4df7-8b07-11ef0b9291e1">45</a></sup>. On the other hand, expression of the ERK negative feedback regulator <em data-ref="6731c8a7-000a-48ee-af5d-1973245c14f0">IL17RD</em> (<em data-ref="aa262715-cc27-4c7e-9ad4-0b0041642c6d">SEF</em>) does seem to approximately reflect expected ERK activity in each population. Components of the core ERK pathway including ERK itself were highly expressed across all clusters (<span class="fakelink linktofig" data-ref="fbc50fa5-fa8e-4ade-b790-241f0b9c7fba"><span class="fakelink-text" data-ref="a11fb569-22af-4d22-8acb-529f6a8072c0">Supp. Fig. 2g</span><a class="fake-icon" href="#figS2" data-ref="1afbfe5c-e3df-4c5d-afcd-90a965a3ac2f"><span class="fakelink-icon" data-ref="9d974454-bc4b-466b-9e40-da68c9b18288">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>).</p><p id="el-p-25" data-ref="78333755-a12f-402b-b2b3-ad3d25b6a020" data-id="el-p-25">Thus, while scRNA-seq identified specific genes involved, it is consistent with several non-exclusive mechanisms for generating an ERK pattern. For example, uniform FGF2 could form an ERK ring by equally activating receptors across the colony but with ERK response reduced by elevated expression of negative regulators in the pluripotent center. Alternatively, based on gene expression we cannot rule out HSPGs increasing receptor binding specifically in the ring. Finally, localized expression of FGF4 or FGF17 in the PS-like ring could lead to a concentration gradient responsible for increased receptor activation.</p></section><section id="s2c" data-ref="9618f35a-9f07-4fad-a0c7-087ba1366e4f" data-id="s2c" data-split-to="9618f35a-9f07-4fad-a0c7-087ba1366e4f" data-split-original="true" data-last-split-element="true" data-align-last-split-element="auto"><h2 class="heading-2" id="el-h2-3" data-ref="51438d56-88f0-4f96-a158-a8e5f348da2b" data-id="el-h2-3" data-break-after="avoid">The phosphorylated ERK pattern is due to differential activation of FGF receptors</h2><p id="el-p-26" data-ref="731d8532-00c8-49a3-a923-b5d6d411edfe" data-id="el-p-26" data-previous-break-after="avoid">To distinguish between the different possibilities, we asked whether the spatial pattern of pERK signaling is controlled primarily below, at, or above the level of the receptors. To address this, we first visualized phosphorylated FGFR1 (pFGFR1) and found its spatial pattern strongly resembles that of pERK (<span class="fakelink linktofig" data-ref="d22a7e98-0e0f-4dc5-9741-a1dd5b14b1db"><span class="fakelink-text" data-ref="e8b423a7-3780-4571-8fce-547f88f59949">Fig. 3a</span><a class="fake-icon" href="#fig3" data-ref="cf650579-6e06-4dc5-8805-d5875201b44f"><span class="fakelink-icon" data-ref="0e3bdf79-94d3-4bea-8a7f-34bb8a11e22e">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="5be3cbc2-4c68-4420-a53d-acd4baa3bf52"><span class="fakelink-text" data-ref="2a774437-da1b-4ce9-8620-2cf34d201113">Supp. Fig. 3a</span><a class="fake-icon" href="#figS3" data-ref="b62567e7-44af-469c-a2fc-5495e987186f"><span class="fakelink-icon" data-ref="8d70df6c-0904-4a91-97df-ecd84c48830f">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). As expected, pFGFR1 was lost upon brief FGFR inhibition (<span class="fakelink linktofig" data-ref="11478da1-aa89-410c-a184-aecd2e2dcc13"><span class="fakelink-text" data-ref="f8305254-d8ca-4ef5-9624-1c4af7c1dab4">Fig. 3a</span><a class="fake-icon" href="#fig3" data-ref="bfd5939d-7fdd-4697-840a-5276a9170181"><span class="fakelink-icon" data-ref="c09178be-ab41-4c03-adbc-a937fdf23bba">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="6c1601fb-50d4-45d7-91b4-73ff2c7a8d8a"><span class="fakelink-text" data-ref="b9728edb-bf61-49ad-8cf4-91f298972763">Supp. Fig. 3a</span><a class="fake-icon" href="#figS3" data-ref="95a315a7-136a-4482-8e09-a08b418d6e8a"><span class="fakelink-icon" data-ref="87b2a1cb-a82f-4923-becf-b983b023e685">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). In contrast, after MEK inhibition, high pFGFR1 remained while pERK was greatly reduced (<span class="fakelink linktofig" data-ref="530bd253-8928-4670-952d-c90abaea538c"><span class="fakelink-text" data-ref="bb581854-8466-49a7-9b7a-567a34dd6335">Supp. Fig. 3bc</span><a class="fake-icon" href="#figS3" data-ref="2f6fc573-f65b-48f3-9b45-c34365f25fcd"><span class="fakelink-icon" data-ref="87db1848-26a4-442e-a2a7-51b743bc9cdf">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). This suggests the spatial pattern of ERK reflects a spatial pattern of receptor activity.</p></section></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_left_page" data-page-number="8" id="page-8">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-id="article-content" data-split-from="726f25f1-fdd1-41b8-bf49-de736be3e3ae"><section data-ref="c2323182-eb7e-45cc-b2f0-c40d361a838e" data-id="s2" data-split-from="c2323182-eb7e-45cc-b2f0-c40d361a838e"><section data-ref="9618f35a-9f07-4fad-a0c7-087ba1366e4f" data-id="s2c" data-split-from="9618f35a-9f07-4fad-a0c7-087ba1366e4f"><p id="el-p-27" data-ref="552f4150-217e-4f13-9a78-9f4ee3c47b9e" data-id="el-p-27">The spatial pattern in receptor activity could either be due to differential receptor expression or differential receptor activation. There was no significant upregulation of FGF receptors in PS-like cells according to our single cell RNA-sequencing data (<span class="fakelink linktofig" data-ref="b46f6232-b1fb-496c-bb85-615930b2ed98"><span class="fakelink-text" data-ref="6f08b960-e526-479e-a422-200c5dcbcd85">Fig 2e</span><a class="fake-icon" href="#fig2" data-ref="7aa9f0c9-7af7-4410-ac8a-e2f993264315"><span class="fakelink-icon" data-ref="65fe31d8-eba2-42fb-aa35-e991b65ff809">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). In fact, the most highly expressed receptor, FGFR1, was elevated in pluripotent cells. This suggests the ERK activity pattern is not due to a receptor expression pattern with two caveats. First, mRNA expression may not reflect protein expression. Second, multiple isoforms exist for FGF receptors 1-3 (but not 4) that have different affinity for FGF ligands, leaving open the possibility that PS-like cells could express a different FGFR1 isoform than pluripotent cells, despite overall FGFR1 expression being approximately uniform.</p><p id="el-p-28" data-ref="394bc537-3333-4535-9899-7d37c3d6d152" data-id="el-p-28">To determine if the spatial pattern in pFGFR1 could be explained by a spatial pattern of FGFR1 protein, we performed immunofluorescence staining of FGFR1. Consistent with the scRNA-seq data, and in sharp contrast to the pFGFR1 stain, FGFR1 protein expression was graded from center to edge with its highest levels in the pluripotent center and therefore could not explain the signaling pattern (<span class="fakelink linktofig" data-ref="fcd5c0f1-135c-4ad5-8ade-5a7209f37b9f"><span class="fakelink-text" data-ref="56d10286-358e-4a0e-923e-04ad2207c4f2">Fig. 3b</span><a class="fake-icon" href="#fig3" data-ref="f23f30ee-b72c-4187-b88c-ae67949067be"><span class="fakelink-icon" data-ref="12a91ccd-00b9-413e-bd27-82ebf40cd8f7">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="0f1e0d14-20a2-4488-a1ab-23003449fd71"><span class="fakelink-text" data-ref="505d7add-f5c4-49c7-9a69-05ee82fc5654">Supp. Fig. 3d</span><a class="fake-icon" href="#figS3" data-ref="e668207d-9693-46ee-8f71-1ba41a22db9c"><span class="fakelink-icon" data-ref="d9c96f01-93ed-479f-93a9-ae640769ddcc">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). To determine if different isoforms of FGFR1 are differentially expressed in PS-like vs. pluripotent cells, we designed qPCR primers to detect the IIIb and IIIc isoforms of <em data-ref="0c3f1f1c-0d7a-4ca4-a23b-75dfd6a432cc">FGFR1</em> (<span class="fakelink linktofig" data-ref="cb170996-20b0-4dbf-8aa1-03e049d328cd"><span class="fakelink-text" data-ref="f407fad6-7a1f-4f22-91da-db28db641a9d">Supp. Fig. 3e-g</span><a class="fake-icon" href="#figS3" data-ref="032d2e51-ef8c-4a1b-b54e-aa080333f912"><span class="fakelink-icon" data-ref="ce710721-ebd2-4b59-9f5e-e044802e98b0">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). These are the most common isoforms, also known as the epithelial and mesenchymal isoforms, and therefore appeared the mostly likely candidates for differential expression. We then analyzed expression of these isoforms in 42h micropatterns versus pluripotent cells and PS- like cells differentiated directly as in <span class="fakelink linktofig" data-ref="82bbdded-4f66-4388-b68c-861f963589db"><span class="fakelink-text" data-ref="a59777a9-9bd5-443a-84d1-eff7db269ba9">Fig. 1d</span><a class="fake-icon" href="#fig1" data-ref="4a76a405-892c-4ef7-9e90-95b44174f730"><span class="fakelink-icon" data-ref="25c06b4d-fec3-485f-88d7-8dcedf1e0ab0">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. However, the change in expression between samples was similar for each isoform, with all isoforms decreasing in 2D gastruloids and directly differentiated PS-like cells relative to pluripotent cells (<span class="fakelink linktofig" data-ref="8cad0582-7b8f-4543-917c-fbad3cec4f1d"><span class="fakelink-text" data-ref="a1d5fb23-5823-4894-b9a4-415a421c23dd">Fig. 3c</span><a class="fake-icon" href="#fig3" data-ref="b6b50f6e-35d0-4c07-8f78-67e32cf12e9a"><span class="fakelink-icon" data-ref="60ad412c-33a9-4a51-8935-3ad14e3b3945">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). To corroborate this and also determine absolute FGFR isoform expression in pluripotent and PS-like cells, we analyzed our single cell RNA-sequencing data and previously published bulk RNA-sequencing data for isoform expression in these two cell types. This revealed that for both pluripotent and PS-like cells FGFR IIIc is the dominant isoform with on average almost 900-fold higher expression than IIIb, while IIIa was not expressed at all (<span class="fakelink linktofig" data-ref="021da908-bb5f-4584-b0c8-32db14e8a3ca"><span class="fakelink-text" data-ref="0841a726-6b9a-47fe-b3c5-bf66236b998b">Fig. 3d</span><a class="fake-icon" href="#fig3" data-ref="d7372c47-5195-41b3-b085-496ef836e53f"><span class="fakelink-icon" data-ref="8e55c468-727e-43b1-89e5-459e405ede4c">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Altogether, these data suggest that spatial patterning in ERK is due to differential activation of FGFR receptors, either due to FGF concentration gradients or possibly spatially patterned HSPGs. Immunostaining for HS showed no clear correlation with TBXT or the expected ERK signaling pattern which is similar but extends further inward (<span class="fakelink linktofig" data-ref="302056bb-e17f-4576-b901-94b1aac6e6bc"><span class="fakelink-text" data-ref="82240c75-452b-4270-8e40-a3c0f18f7103">Supp. Fig. 4</span><a class="fake-icon" href="#figS4" data-ref="8220761e-e496-454c-9fc2-e0988323cf83"><span class="fakelink-icon" data-ref="65e23d18-123f-4fbd-b4f3-d6531ee80511">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). This does not rule out a role for HSPGs in spatial modulation, since FGF signaling depends on specific sulfation patterns rather than overall HS levels<sup data-ref="7b5957f9-31d8-45d7-bce4-37a9f2fb1ca8"><a href="#c46" data-ref="ce71ff93-cb60-423f-957b-17c3ffeb8518">46</a>,<a href="#c47" data-ref="718d8013-1e53-46b0-b17e-c02991d76f32">47</a></sup>. Nevertheless, it led us to focus on FGF gradients as the most likely cause for the ERK signaling pattern.</p></section><section id="s2d" data-ref="bbe7e9a6-dfa8-4a80-84ff-a98329995d18" data-id="s2d"><h2 class="heading-2" id="el-h2-4" data-ref="f38e40e7-4199-4e81-b68d-56e912f9c0fb" data-id="el-h2-4" data-break-after="avoid">FGF receptors are localized basolaterally</h2><p id="el-p-30" data-ref="9c39607e-9439-4c78-87ea-ba9c239d7d60" data-id="el-p-30" data-previous-break-after="avoid">To maintain an endogenous FGF concentration gradient on the apical side of the colony which faces the medium, would require ligands to remain tethered to the surface. However, FGF has been reported to diffuse freely in the intercellular space<sup data-ref="3aa4e1a6-b83f-4e40-ab9c-7f4216b4e7d1"><a href="#c48" data-ref="8cca9aca-8de3-4d40-a44e-36a02ec1bedb">48</a></sup>. Moreover, there is FGF2 present in the medium that would be expected to uniformly activate FGF receptors. The situation is similar for BMP4, which is present uniformly in the medium for 2D gastruloids but does not cause uniform signaling activity<sup data-ref="a34af4af-758d-4952-b353-b7772b224277"><a href="#c4" data-ref="a2f4bc97-2340-4a85-bbc1-66b8692adb57">4</a></sup>. In the case of BMP, that is because receptors are localized basolaterally<sup data-ref="29e2d429-c0e6-4cbe-bebe-9b3aba8e9416"><a href="#c4" data-ref="b9715a54-1440-44ca-b59b-f5e3a3c943ed">4</a></sup> and tight junctions seal the basolateral side from the medium. Consequently, response to exogenous ligands in the medium is restricted to the colony edge, while endogenous ligands are confined to the basal side and prevented from diffusing away into the medium everywhere but the edge. This was found to be critical for patterning in both 2D gastruloids and the mouse embryo<sup data-ref="f3f9e3b7-fbc1-4085-9d34-94caa2c98f0a"><a href="#c4" data-ref="0fa8867a-01f2-46bf-9f57-7ae68c677cc0">4</a>,<a href="#c5" data-ref="efdac636-9cc9-4a63-9eb6-055d1a5ddfd3">5</a></sup>. To explain the pERK pattern, we therefore hypothesized FGF receptors were similarly restricted to the basolateral side.</p><p id="el-p-31" data-ref="d082bbf0-c6c6-4ea3-afdc-eec2e48de97b" data-id="el-p-31">To test our hypothesis, we first starved pluripotent colonies of FGF by maintenance in FGF-free E6 medium for 12h and then treated with FGF2. Consistent with our hypothesis, phospho-FGFR and phosphor-ERK were both restricted to the colony edge (<span class="fakelink linktofig" data-ref="d9c29fde-a5e7-45ef-ad55-711347905e79"><span class="fakelink-text" data-ref="ba559ba5-5604-4ba2-8cb3-05cf6f811880">Fig. 4a</span><a class="fake-icon" href="#fig4" data-ref="8c7e2841-0611-45dc-917c-5e74637624b6"><span class="fakelink-icon" data-ref="885e0026-9efc-496d-8797-36e56a02dce3">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). To determine if this could be explained by diffusion of exogenous FGF from the medium, we repeated the experiment with micropattern pluripotent cells and added fluorescent dextran with a similar molecular weight as FGF. We observed fluorescent dextran diffusing into the intercellular space from the edge inward </p></section></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_right_page pagedjs_pagedjs-filler_page page-grid" data-page-number="9" id="page-9">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><figure class="figure tofill imgs-1 shortcaption fig-square pagedjs-fill-next-page finishes" id="fig3" data-ref="456b6ea0-27b5-46b0-b56e-b67e32cb9095" data-id="fig3" style="display: block; position: unset;"><figcaption class="figure__caption" id="el-figcaption-2" data-ref="5d2d6b17-8b67-401c-8b51-0475d0333cd3" data-id="el-figcaption-2"><h3 id="el-h3-2" data-ref="f91170f6-599e-4dd3-998c-cbefebea1955" data-id="el-h3-2" data-break-after="avoid"><span class="label figure-name" data-ref="66c23f75-b0d3-4937-acc0-6b92ef5a0828"><label class="figure__label" data-ref="1eb95cf5-e1ba-415d-bf9a-35336eec0bd4">Figure 3.</label></span></h3><p id="el-p-29" data-ref="1a04647b-90db-4982-a008-f4d23936e284" data-id="el-p-29" data-previous-break-after="avoid"><strong data-ref="b2bdc082-b547-4386-ac49-562fb0b15727">a)</strong> phospho-FGFR1 stain with and without 1h FGFRi treatment at 41h, radial intensity profile on the right. <strong data-ref="a701e40f-d7b1-452a-8f9c-782f046fd3dd">b)</strong> FGFR1 with and without 1h FGFRi treatment at 41h, radial intensity profile on the right. Error bars in (b,c) represent standard deviation over N=4 colonies. <strong data-ref="4cbcafb7-768a-490a-9b2a-8262df75ace8">c)</strong> qPCR data for relative expression of FGF receptor 1 isoforms IIIb, IIIc and total FGFR1 in pluripotent cells (pluri), gastruloids at 42h (MP), and directly differentiated PS-like cells (dPS). Error bars represent standard deviation of technical triplicates. <strong data-ref="1cedec5b-4de0-48d8-8b5f-514a36728e57">d)</strong> Absolute FGFR1 isoform expression from RNA-seq for anterior/mid primitive streak and pluripotent cells from Loh 2016<sup data-ref="0356259d-2b89-4e23-b49d-0697eeecd834"><a href="#c38" data-ref="a4f405da-913b-43e5-9b94-e69906f8d623">38</a></sup> (Loh_ APS/MPS/HESC) as well as the pluripotent and PS-LC clusters from <span class="fakelink linktofig" data-ref="7f8425a5-cd6f-4b48-81fe-0194d33fd6da"><span class="fakelink-text" data-ref="7290aea0-0615-479d-9c1f-368d0133b5cb">Fig.2</span><a class="fake-icon" href="#fig2" data-ref="31f2beaa-3167-4d48-9465-78ac81cb28d2"><span class="fakelink-icon" data-ref="0f6c8b21-e09a-4e25-8c93-f58d6009c77a">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. All scale bars 50 um.</p></figcaption><img class="child-of-figure has-boundary imageonly square max-70" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_fig3.tif/full/max/0/default.jpg" alt="" id="el-img-3" data-ref="c97cc055-7c1f-486f-a173-31119dbff22d" data-id="el-img-3" height="1287" width="1403" data-desired-width="66.42992424242425"></figure></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_left_page" data-page-number="10" id="page-10">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-id="article-content" data-split-from="726f25f1-fdd1-41b8-bf49-de736be3e3ae"><section data-ref="c2323182-eb7e-45cc-b2f0-c40d361a838e" data-id="s2" data-split-from="c2323182-eb7e-45cc-b2f0-c40d361a838e"><section data-ref="bbe7e9a6-dfa8-4a80-84ff-a98329995d18" data-id="s2d" data-split-from="bbe7e9a6-dfa8-4a80-84ff-a98329995d18"><p data-ref="d082bbf0-c6c6-4ea3-afdc-eec2e48de97b" data-id="el-p-31" data-split-from="d082bbf0-c6c6-4ea3-afdc-eec2e48de97b">over the same distance that ERK signaling was observed (<span class="fakelink linktofig" data-ref="2091226d-1919-404b-8738-3c1b5942ac9e"><span class="fakelink-text" data-ref="76958e5e-f778-4ebf-8f42-1d8c9c775ccb">Fig. 4b-d</span><a class="fake-icon" href="#fig4" data-ref="b1915d59-5f0b-4b87-ac2e-0df9226cb231"><span class="fakelink-icon" data-ref="13b94ec8-a1e4-4182-a556-5110e2984fcb">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). We then acquired high resolution z-stacks of FGFR1 co-stained with the tight junction protein TJP1 (ZO-1) which revealed the majority of FGFR1 was below the tight junctions (<span class="fakelink linktofig" data-ref="faecd37c-2f40-47b9-bfbb-455f8af6b275"><span class="fakelink-text" data-ref="7a16a26e-8085-4d47-a6c3-d73504e608f8">Fig. 4e</span><a class="fake-icon" href="#fig4" data-ref="fce395b7-4f63-4037-8362-6e3d3651ed00"><span class="fakelink-icon" data-ref="ba395ec8-d739-45e1-8b6d-f245894dc64b">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). To directly test that cells are more responsive to basal FGF than apical FGF, we then performed a transwell assay, where pluripotent cells were grown on filters to enable separate apical and basal stimulation. We found that only basal stimulation with FGF led to a significant increase in pERK (<span class="fakelink linktofig" data-ref="8bbd80af-e86d-4e0e-909e-ddec3a724092"><span class="fakelink-text" data-ref="f7dae777-e3a2-47fb-8da1-ba636e777142">Fig. 4f</span><a class="fake-icon" href="#fig4" data-ref="2675d5e9-a3d9-4d78-8cae-d0768328701c"><span class="fakelink-icon" data-ref="eac43751-cc82-4f3f-ae54-194cd1ad1c56">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Finally, to determine if the center of the micropatterned colony is similarly able to respond to basal FGF stimulation and pERK levels there can be attributed to reduced FGF ligand concentration, we performed a scratch assay. Strong ERK response to FGF2 in the media was seen in the colony center around a scratch, but this response was blocked by FGFR inhibitor (<span class="fakelink linktofig" data-ref="ae6ad87a-be38-4381-9944-2aaad71b3fbf"><span class="fakelink-text" data-ref="a9ab921a-08e9-4bb8-bf96-79f896accee7">Fig. 4gh</span><a class="fake-icon" href="#fig4" data-ref="aa9e1772-732b-4575-8047-0d0c313acdfe"><span class="fakelink-icon" data-ref="ed8300bb-f202-4101-b15d-5e5612659b8e">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). If HSPGs rather than FGF ligand were limiting ERK response in the center, we would not have expected strong response around a scratch. The initial ERK levels around the scratch greatly exceeded those anywhere in the control. However, we found this was due to adaptive ERK response to FGF. After six hours ERK levels around the scratch were similar to those in the ERK ring around the PS-like cells (<span class="fakelink linktofig" data-ref="8edd178e-15cd-4226-a7d4-94b15c8bb3ad"><span class="fakelink-text" data-ref="fabe95b0-c992-4ab8-94c3-d257331a3397">Fig. 4ij</span><a class="fake-icon" href="#fig4" data-ref="7ffeeec2-e46c-450e-abed-551c6ed0594b"><span class="fakelink-icon" data-ref="a80f0a5e-71fc-433b-906d-62da5d646151">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). We therefore concluded that the pERK pattern is most likely due to a basolateral FGF ligand gradient which has its maximum in the PS-like ring.</p></section><section id="s2e" data-ref="3b1d2b09-70bd-496f-bbde-99ad4f10fea3" data-id="s2e"><h2 class="heading-2" id="el-h2-5" data-ref="78f5b4f2-40f2-4d85-b506-fd0d4ce7e64c" data-id="el-h2-5" data-break-after="avoid">Endogenous FGF gradients underly the ERK activity pattern</h2><p id="el-p-33" data-ref="4e657750-a037-4dfe-a3ef-55e91c111a75" data-id="el-p-33" data-previous-break-after="avoid">To identify which FGF ligands are required for PS-like differentiation, we first determined the requirement for exogenous FGF2 in the media. Differentiation of 2D gastruloids in FGF-free medium was severely reduced but restored by addition of FGF2 (<span class="fakelink linktofig" data-ref="2eaca8ff-9108-4263-9ef7-0dac2d345f1e"><span class="fakelink-text" data-ref="1ab206c0-49df-4c10-b4fc-99a89a0a0c68">Fig. 5a</span><a class="fake-icon" href="#fig5" data-ref="b60d8e07-2fda-46f5-95ed-06295fddf6b3"><span class="fakelink-icon" data-ref="45eecdba-9d01-4863-9373-340a195fb44d">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="c6b9d293-ce3f-4e61-86a6-5cec1189cb3d"><span class="fakelink-text" data-ref="9c2b8d51-9f7a-410b-bb67-fccfbbf4099b">Supp. Fig. 5a</span><a class="fake-icon" href="#figS5" data-ref="7ac133a3-7b31-40d3-b4d8-38ab671cac33"><span class="fakelink-icon" data-ref="218af401-5c48-4103-832e-8e322bc65554">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). However, if FGF2 was removed 24h after BMP treatment, differentiation was normal. The fact that removal 2h after BMP treatment significantly reduced differentiation suggests this was not a failure to wash out the FGF2. Furthermore, MEK inhibition at 24h severely reduces TBXT expression<sup data-ref="5d93cbb9-a02a-467c-8695-291517abe0d4"><a href="#c34" data-ref="ce220d8c-3e44-4c20-a367-53451d47eeee">34</a></sup> while FGF2 is known to have a half-life on the order of hours<sup data-ref="439705df-8cdc-4163-bcbc-92c1def5c8c0"><a href="#c49" data-ref="d53f74d8-a4ed-4fc8-8d5e-6e0477f1dcc8">49</a></sup> and no media changes are performed during standard differentiation. Altogether this suggests that exogenous FGF2 is required at the time of BMP treatment but that endogenous FGFs are sufficient for the ERK activity ring and PS-like differentiation that start a day later.</p><p id="el-p-35" data-ref="f1a11b28-1b53-47ba-b894-39872b32a826" data-id="el-p-35">We reasoned that FGFs whose expression correlates with the presence of PS-like cells are more likely to be responsible for the observed ERK ring and PS-like differentiation. Our analysis in <span class="fakelink linktofig" data-ref="a26c0f4b-badf-40bf-8823-c9ebe601cebc"><span class="fakelink-text" data-ref="d582e02e-1bc0-4130-86ed-b101b19ca47d">figure 2</span><a class="fake-icon" href="#fig2" data-ref="c68c763d-fffe-4771-a3d1-a1be57271efb"><span class="fakelink-icon" data-ref="90754ae8-fa5c-499d-8fd7-b5c6f56e4f7a">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span> had implicated FGF4 and FGF17 as most strongly expressed in the PS-like cells and nascent mesoderm, respectively (<span class="fakelink linktofig" data-ref="c2d68c72-7079-4592-8458-d701e416cc4b"><span class="fakelink-text" data-ref="0874a9f2-b63f-4885-9b59-8a0f2170a080">Fig. 2c</span><a class="fake-icon" href="#fig2" data-ref="ca630bce-bf9e-45ad-b0d4-5f726ca199d3"><span class="fakelink-icon" data-ref="84901529-7c4b-4a46-875d-62fde8aa6bdf">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). However, it is unclear whether expression of these ligands is a cause or consequence of PS-like differentiation, or both, in the case of a feedback loop. To get more information about the relationship between FGF ligands and PS versus mesoderm differentiation, we performed pseudotime analysis of our scRNA-seq data (<span class="fakelink linktofig" data-ref="a1ffcde5-5598-40c3-9561-05b4f461d757"><span class="fakelink-text" data-ref="5255c174-a44c-4948-b436-c05e650395cb">Fig. 5b</span><a class="fake-icon" href="#fig5" data-ref="cc6daf67-18ef-4ebd-bc18-027671ae3d11"><span class="fakelink-icon" data-ref="6d239c40-adb3-4f49-a6e2-e4ad8bab4a54">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="71e3dd24-f0d5-4ef1-be29-f6acd07c0c02"><span class="fakelink-text" data-ref="7e53ed1a-4493-4398-8aeb-0741f619e2f8">Supp. Fig 5b</span><a class="fake-icon" href="#figS5" data-ref="2039ec33-f05f-4c94-969e-0a6ff6026358"><span class="fakelink-icon" data-ref="86eea061-b801-4f45-8b97-5f4528bb578f">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Along a pseudotime trajectory from pluripotent cells to nascent mesoderm, <em data-ref="b03f6cb5-d768-4efc-a529-d49bc4ddb1ad">FGF2</em> expression was nearly constant. <em data-ref="3428a895-d2ba-4023-9261-92b2c686efd1">FGF8</em> was upregulated at lower levels prior to <em data-ref="ae294e78-5efe-41c1-99cc-918697bf3968">TBXT</em> and declined as <em data-ref="1321f8c3-f20e-4c58-9b7b-566311b9805e">TBXT</em> was upregulated. <em data-ref="94ee8fcb-4ac2-445a-a1dd-e29cb1762631">FGF4</em> upregulation approximately coincided with <em data-ref="1b180064-10b1-4d64-94a1-6cbbc9537643">TBXT</em>, but <em data-ref="175aa7a7-258f-4fa2-8527-e815bcb1512b">FGF4</em> began decreasing as <em data-ref="fbd53fc5-fb02-49bf-8d6c-b3444dc51f9f">TBX6</em> expression increased. In contrast, <em data-ref="8b299ba2-5c27-4857-8963-48f0deb5534e">FGF17</em> expression started after <em data-ref="718d8e89-a382-47dc-8387-b4501a800775">FGF4</em> but continued to increase after <em data-ref="ea15ff57-5960-41f0-8e42-57b1f05c0d4f">FGF4</em> started decreasing. Direct comparison of <em data-ref="bd54b686-5c3e-41ae-a9d6-08d6a4a094d5">FGF</em> and <em data-ref="af92269f-4aae-453a-90f7-cad785c8f4cb">TBXT</em> expression also showed strong positive correlation between <em data-ref="fc226523-71e6-4713-b704-8b7b2f84c77d">TBXT</em> and both <em data-ref="7cda5874-1e34-488b-b79a-509b6de56520">FGF4</em> and <em data-ref="5e636cfc-ee9b-4803-a9db-32fabb13030c">FGF17</em> but opposing relations with <em data-ref="ff3dc3f3-d8f9-4d82-a995-56a147618661">TBX6</em> (<span class="fakelink linktofig" data-ref="86aa4fad-a82a-4678-84fd-941b5d25e72f"><span class="fakelink-text" data-ref="35801380-2a63-486e-ab5e-3bd75ce0df05">Supp. Fig. 5c</span><a class="fake-icon" href="#figS5" data-ref="3f95a14c-89c1-4ca7-8701-ef19f044d17f"><span class="fakelink-icon" data-ref="91a05cc0-08fc-4e5d-8f69-217889bf100c">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Together this suggested to us initial TBXT expression may depend more on FGF8 or FGF4, while FGF17 could play a later role in maintaining or expanding the PS-like population or facilitating subsequent mesoderm differentiation.</p><p id="el-p-36" data-ref="d8c98ea5-a74a-464f-b2cc-24563bfa6c30" data-id="el-p-36">Pseudotime may not accurately predict true dynamics. Therefore, we verified expression at two different times using RNA FISH (fluorescent in situ hybridization). We found that <em data-ref="88526701-54cf-4a38-ba05-f2c6090a3fc2">FGF4</em> expression was similar at 32h and 40h (<span class="fakelink linktofig" data-ref="4d9970bc-3f62-4a40-b080-77b77629a193"><span class="fakelink-text" data-ref="9cdfb0fe-3bca-4f75-8dd2-049f7960edd2">Fig. 5cf</span><a class="fake-icon" href="#fig5" data-ref="eab756d4-33d3-4842-84e2-9478cb5a71c6"><span class="fakelink-icon" data-ref="d0509728-c649-4d9b-a630-02116af10ebd">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="ecb338af-9281-45ac-9cd8-e80d35fb2460"><span class="fakelink-text" data-ref="b31061ec-4a5c-4320-9299-535819abf57e">Supp. Fig. 5d</span><a class="fake-icon" href="#figS5" data-ref="81952f9d-c13c-4427-86b8-886ad296d081"><span class="fakelink-icon" data-ref="4492629b-bc32-438a-a02c-3b436c0f76b4">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). This is consistent with its decrease in pseudotime as cell differentiate to nascent mesoderm, since in real time differentiation is asynchronous and new PS-like cells form while older ones differentiate to nascent mesoderm, which would yield a flatter temporal profile. In contrast, peak <em data-ref="d06a46bd-184c-411b-806b-31cea2a858c2">FGF8</em> levels decreased while <em data-ref="a1b66f01-248b-4504-a1ae-020e05a66c46">FGF17</em> strongly increased over time, both matching their pseudotime trends (<span class="fakelink linktofig" data-ref="c0686b18-f40e-41c1-9ab8-b96dca1bd555"><span class="fakelink-text" data-ref="933c4d4e-3db4-4bf7-bed2-591ac1b9b057">Fig. 5d-f</span><a class="fake-icon" href="#fig5" data-ref="75ab9422-7590-43b7-92b7-fdca82280d3b"><span class="fakelink-icon" data-ref="a50a03ae-0812-4370-836a-d57b2d6d0ea9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="5234fc93-df19-4d4e-9b23-b813663c4285"><span class="fakelink-text" data-ref="a6e7ee06-4694-41c2-b2ba-dc9670051e9f">Supp. Fig. 5ef</span><a class="fake-icon" href="#figS5" data-ref="0cb2343c-a679-433c-a08b-d3cf7ba3c3d2"><span class="fakelink-icon" data-ref="0f683120-e906-4a46-8528-0b5a8b5ed2de">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). As expected, the spatial profiles of both <em data-ref="e3d399c6-242a-4edf-b94f-ed335d27860d">FGF4</em> and <em data-ref="fed11889-bb7a-4ea8-9360-80c523ca3112">FGF17</em> resembled TBXT, although the peaks were displaced slightly inward, while <em data-ref="ea4b084d-9241-4ff3-98c9-7a18ab4f9aff">FGF8</em> expression was highest in the inner part of the </p></section></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_right_page pagedjs_pagedjs-filler_page page-grid" data-page-number="11" id="page-11">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><figure class="figure tofill imgs-1 longcaption fig-portrait pagedjs-fill-next-page finishes" id="fig4" data-ref="dd8f4a22-7f91-41bf-8097-0e12281a5472" data-id="fig4" style="display: block; position: unset;"><figcaption class="figure__caption" id="el-figcaption-3" data-ref="fe369079-6656-45a3-a3d1-cdd99fed9889" data-id="el-figcaption-3"><h3 id="el-h3-3" data-ref="4b570ddc-9b0b-4772-8e57-dcd72890d911" data-id="el-h3-3" data-break-after="avoid"><span class="label figure-name" data-ref="b01c5e87-6c14-4753-8a62-59d7058946b3"><label class="figure__label" data-ref="6fff133d-1083-490d-978f-cd207285b7c3">Figure 4.</label></span></h3><p id="el-p-32" data-ref="ebe57380-bf09-4e3e-8b42-558d9fcf80f9" data-id="el-p-32" data-previous-break-after="avoid"><strong data-ref="4b1dd281-6c12-4f5a-83e7-1ef6d0767d80">a)</strong> pERK and pFGFR1 response of pluripotent colonies in standard culture grown in FGF-free (E6) medium for 24 hrs and then treated with FGF2 for 30 minutes. <strong data-ref="38b77e94-5880-4239-90ff-0b1a6fc190e0">b,c)</strong> Fluorescent dextran (b) and pERK (c) in micropatterned pluripotent colony 24h after media change containing both FGF2 and dextran. <strong data-ref="1ebfaf32-f6a4-4de1-b992-a728e347e20e">d)</strong> Radial intensity profile of dextran and pERK for conditions in b,c), averaged over 2 colonies. <strong data-ref="a04f5824-0911-4685-b067-6054b79d8768">e)</strong> Cross-section of 2D gastruloid shows FGF receptor 1 predominantly below the tight junction marked by ZO-1. White box marks are magnified in inset e’. <strong data-ref="cc3960ee-4ff8-42a6-854f-34c280133129">f)</strong> pERK and pFGFR1 response in transwell experiment with cell treated with FGF either apically or basally. The illustrations in this panel were created using <span class="fakelink linktoext" data-ref="988d15b4-2451-46fb-a5e0-ab50292bfb41"><span class="fakelink-text" data-ref="9da28a70-024b-419b-b0e2-4200fc38128a">BioRender.com/v73i147</span><a class="fake-icon" href="https://BioRender.com/v73i147" data-ref="020ec7a2-0be6-44a7-a0b7-370fa3d24666"><span class="fakelink-icon" data-ref="8fa3cfea-a393-4070-a57b-ea6456a5d2b3">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. <strong data-ref="5c567872-424d-43a6-b0b8-111788f74548">g)</strong> ERK signaling in scratched colonies with or without FGFRi. <strong data-ref="bbab6ab1-d72d-4565-85d0-b4a2ce1bf87b">h)</strong> Quantification of pERK intensity as a function of distance from the scratch. Error bars represent standard deviation over colonies. <strong data-ref="20e16321-948b-4b00-8d69-0e5e09a727b3">i)</strong> ERK signaling stains at different times after scratching. <strong data-ref="5d191984-8a9d-46f3-8b6b-94f5c9e296f7">j)</strong> Quantification of pERK intensity as a function of distance from the scratch at different times (left) and peak pERK level over time (right). Error bars represent standard deviation over N=4 colonies. Scale bars 50 micron.</p></figcaption><img class="child-of-figure has-boundary imageonly portrait max-100" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_fig4.tif/full/max/0/default.jpg" alt="" id="el-img-4" data-ref="78b0e615-9587-4871-a089-1bbd3dbcc906" data-id="el-img-4" height="2350" width="2101" data-desired-width="99.47916666666666"></figure></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_left_page pagedjs_pagedjs-filler_page page-grid" data-page-number="12" id="page-12">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><figure class="figure tofill imgs-1 shortcaption fig-square pagedjs-fill-next-page finishes" id="fig5" data-ref="5023f542-1bc1-4d19-a6ea-9de547f84932" data-id="fig5" style="display: block;"><figcaption class="figure__caption" id="el-figcaption-4" data-ref="bbd9eba3-3d31-427a-81b5-fe6135b63ab0" data-id="el-figcaption-4"><h3 id="el-h3-4" data-ref="397d7c56-853d-4bde-99d5-0c3d90ba3cc9" data-id="el-h3-4" data-break-after="avoid"><span class="label figure-name" data-ref="9781a66c-0c41-4041-bd3e-eb73f417acfb"><label class="figure__label" data-ref="e84c7b4a-9a83-4870-b410-d861b3a261bb">Figure 5.</label></span></h3><p id="el-p-34" data-ref="813f2e32-134e-48ab-bcb3-8bdf851f28a4" data-id="el-p-34" data-previous-break-after="avoid"><strong data-ref="d9660fc0-02dc-4e5e-a18a-a80548016d91">a)</strong> Pattern at 42h with exogenous FGF2 in the culture medium for different amounts of time. <strong data-ref="c79bb67e-70b3-4950-90bc-eadab87fe62a">b)</strong> Pseudotime analysis of FGFs and TBXT, TBX6 expression. <strong data-ref="1da309e5-ec31-4139-9296-12bae1f77c12">c-e)</strong> FGF4 (c), FGF8 (d), and FGF17 (e) FISH co-stained for TBXT and pERK at 30h or 32h and 40h or 42h after BMP4 treatment. <strong data-ref="7badf667-3b8a-46cb-b244-f586d7924f31">f)</strong> Radial intensity corresponding to (c-e). <strong data-ref="984fcc98-5440-4e66-9df8-0b7d7c65ba86">g-i)</strong> FGF4 (g), FGF8 (h), and FGF17 (i) FISH co-stained for TBXT and pERK in control siRNA versus FGF4 siRNA (g), FGF8 siRNA (h) and FGF17 siRNA (i) with corresponding radial intensity profiles. <strong data-ref="d74e304c-4046-496a-90e9-42f1e40dc850">j)</strong> TBX6 and SOX17 expression in control shRNA versus FGF4 and FGF17 shRNA. Error bars in all panels except (b) represent standard deviation over N=4 colonies. Error bars in (b) represent the 95% confidence interval of the generalized additive model (see methods).</p></figcaption><img class="child-of-figure has-boundary imageonly square max-100" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_fig5.tif/full/max/0/default.jpg" alt="" id="el-img-5" data-ref="605132c9-dc9f-44ad-a845-67c1f6623dc6" data-id="el-img-5" height="2175" width="2106" data-desired-width="99.7159090909091"></figure></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_right_page" data-page-number="13" id="page-13">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-id="article-content" data-split-from="726f25f1-fdd1-41b8-bf49-de736be3e3ae"><section data-ref="c2323182-eb7e-45cc-b2f0-c40d361a838e" data-id="s2" data-split-from="c2323182-eb7e-45cc-b2f0-c40d361a838e"><section data-ref="3b1d2b09-70bd-496f-bbde-99ad4f10fea3" data-id="s2e" data-split-from="3b1d2b09-70bd-496f-bbde-99ad4f10fea3"><p data-ref="d8c98ea5-a74a-464f-b2cc-24563bfa6c30" data-id="el-p-36" data-split-from="d8c98ea5-a74a-464f-b2cc-24563bfa6c30">ERK ring where TBXT expression was low (<span class="fakelink linktofig" data-ref="08c45a27-56f7-4002-bdfc-44a5de4a84ff"><span class="fakelink-text" data-ref="e431226a-f9ad-4e1a-aaf0-2eab91a10b5e">Fig. 5c-f</span><a class="fake-icon" href="#fig5" data-ref="9fbed97b-0499-43f0-b860-24da82689c6d"><span class="fakelink-icon" data-ref="f4b1dbcf-27e8-4374-9df7-330223516d12">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Co-staining showed <em data-ref="23bc72d8-6438-42e1-a80a-6b8cc23be92b">FGF4</em> and <em data-ref="1c926d24-4aee-4b57-b0c9-1960fa40dcb7">FGF8</em> expressed in concentric rings with little overlap at 42h (<span class="fakelink linktofig" data-ref="ee2ad17a-76a4-46d0-9c31-964d47651b1d"><span class="fakelink-text" data-ref="ec17e484-a717-4bd8-af7e-de98e5b07837">Supp. Fig. 5g</span><a class="fake-icon" href="#figS5" data-ref="99e1a621-683a-41be-967b-b310415d526d"><span class="fakelink-icon" data-ref="7a2b8ec0-5459-4773-9bd8-479d19eeb0b6">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Scatterplots further confirmed that high expression of both <em data-ref="33c81ba6-6f42-4cfb-acb9-1e57495407f4">FGF4</em> and <em data-ref="ba2d4be4-0db8-4aa1-b4a5-ff76993c320b">FGF17</em> but not <em data-ref="236d1c6b-bc93-4a2e-8e23-ebf307d2a8e5">FGF8</em> was restricted to TBXT positive cells (<span class="fakelink linktofig" data-ref="21481d21-1b93-46c0-8cfd-eff289621208"><span class="fakelink-text" data-ref="e2dbf9f8-9f60-4177-aa98-372f4f4d4df2">Supp. Fig. 5h</span><a class="fake-icon" href="#figS5" data-ref="bc5e4c0c-9ca5-4fcb-a0cc-8690b3f38c0a"><span class="fakelink-icon" data-ref="4cf97f01-ec98-446e-ad47-4845bcd62047">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>).</p><p id="el-p-37" data-ref="81e7095b-fce1-4e01-a8c0-ad79d57a9ef7" data-id="el-p-37">To test the functions of these FGFs we decreased their expression. Knockdown (KD) of <em data-ref="d511915a-6849-4412-9323-82c59f7d8fd0">FGF4</em> with small interfering RNAs (siRNA) led to a strong decrease in TBXT and pERK (<span class="fakelink linktofig" data-ref="28bed296-6a78-4347-aaea-545fb857e7ad"><span class="fakelink-text" data-ref="2149ff54-0fe3-4821-a76a-2203d74f2e02">Fig. 5g</span><a class="fake-icon" href="#fig5" data-ref="21f97442-d54c-463a-8b04-1d36ec00fcf3"><span class="fakelink-icon" data-ref="af9b1880-449b-4ba2-8423-110b818f7ed0">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="bbb546a4-c845-462c-8148-664d6670a0f6"><span class="fakelink-text" data-ref="8cab7276-e906-429b-b6d4-82403edfffb5">Supp. Fig. 6a</span><a class="fake-icon" href="#figS6" data-ref="6e429e3f-7391-49ce-be32-6c86afd8f997"><span class="fakelink-icon" data-ref="9162aa2e-5f58-4beb-9b3a-26f47dd1cb22">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). We repeated this in a different cell line and also created stable cell lines expressing short hairpin RNA (shRNA) to knock down <em data-ref="7da8f689-0472-4c87-8661-ca8d40c25015">FGF4</em> in both cell lines with similar effect (<span class="fakelink linktofig" data-ref="cb721fa8-c53a-4548-93ca-8141556fcd4d"><span class="fakelink-text" data-ref="8056e239-89dd-45e2-a1a1-1f8ac7346748">Supp. Fig. 6b-d</span><a class="fake-icon" href="#figS6" data-ref="31fc9103-1a4b-4224-b1e5-b91d0f8c152d"><span class="fakelink-icon" data-ref="ec724b9b-625e-45f9-87ae-e9f31ba6a379">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Strikingly, <em data-ref="816b54a2-b88c-4760-b9e6-805fd48e6af4">FGF8</em> KD had the opposite effect and led to an increase in TBXT expression without a strong effect on pERK visible at 42h (<span class="fakelink linktofig" data-ref="ed8f00f0-1cbf-416c-a213-da4c3fda28c8"><span class="fakelink-text" data-ref="0f102d7f-5059-4909-9cff-3c3778d2547a">Fig. 5h</span><a class="fake-icon" href="#fig5" data-ref="6e78ce8c-7ed6-4281-98bb-abc06536f4cb"><span class="fakelink-icon" data-ref="351ecb31-8b67-48a3-a259-be2f9a69366d">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="82a0f3ca-14b3-47b7-8633-cfb7571b93fd"><span class="fakelink-text" data-ref="706c75a1-0abe-42f9-a379-cb8cf4926566">Supp. Fig. 6e</span><a class="fake-icon" href="#figS6" data-ref="daf00f98-06da-4fb2-ac12-35000f651bb6"><span class="fakelink-icon" data-ref="e113ad11-d93e-4de4-8576-004f8bd1a01d">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Finally, <em data-ref="ca1224f0-52b6-42c1-8b50-dfdc3821ebe0">FGF17</em> KD caused only a small decrease in TBXT or pERK, again consistent between different cell lines (<span class="fakelink linktofig" data-ref="443acdc6-7d19-4613-bc50-80578eecb0d7"><span class="fakelink-text" data-ref="28697663-c572-42ee-a1f1-b6fea5b9a3c1">Fig. 5i</span><a class="fake-icon" href="#fig5" data-ref="373c4f3c-9a02-4ee1-96c3-2dd9a84f6603"><span class="fakelink-icon" data-ref="4ea07e52-4db6-4739-a887-ce2d269f29a3">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="08d55bf9-2119-4ace-afd2-281487ebb4d5"><span class="fakelink-text" data-ref="25cb7a81-2387-4561-9b9f-17bc99a15749">Supp. Fig. 6f-i</span><a class="fake-icon" href="#figS6" data-ref="114f3f90-7807-432c-b874-59721a150bfa"><span class="fakelink-icon" data-ref="ba91240b-9b5e-4495-9ce0-9acf2582c667">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). We further interrogated the endoderm and PGC marker SOX17 since both cell types require ERK signaling to differentiate and transiently express TBXT, as well as the mesoderm marker TBX6. TBX6 expression at 42h was near completely lost upon either <em data-ref="b246be42-4b30-4551-a139-a002d7a09794">FGF4</em> KD or <em data-ref="1caf86d8-6555-456e-81f6-92a2ca34fb66">FGF17</em> KD (<span class="fakelink linktofig" data-ref="ea773917-9121-4509-bac7-7e39a95032c6"><span class="fakelink-text" data-ref="20e3e33d-48af-450b-81b8-342538ac6141">Fig. 5j</span><a class="fake-icon" href="#fig5" data-ref="9a1ae61f-ed08-482d-b561-5a8ead84cfb1"><span class="fakelink-icon" data-ref="0e290544-cdc5-436a-a622-18fbe424f41c">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, <span class="fakelink linktofig" data-ref="e980130a-b93c-4800-b75f-9891854348d1"><span class="fakelink-text" data-ref="964156a3-d174-48c0-90a2-ba1b606cc99c">Supp. Fig. 6j</span><a class="fake-icon" href="#figS6" data-ref="b95772e5-2b10-40af-9ed8-75055d5de757"><span class="fakelink-icon" data-ref="e50317b1-2f83-4424-b75e-7f67df74e31a">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). This suggests that while FGF17 is not required for PS induction (<span class="fakelink linktofig" data-ref="1a2a7937-4d81-4836-91b9-ea919baa5dcb"><span class="fakelink-text" data-ref="9b58fa7c-46b8-4cf9-9a20-f99db8c1fe57">Fig. 5i</span><a class="fake-icon" href="#fig5" data-ref="3a463ccd-7911-4080-98f9-331bfa7a1c4b"><span class="fakelink-icon" data-ref="83a817ff-4619-45cc-a0ab-50617d0fe9e6">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>), it is needed for subsequent mesoderm differentiation. In contrast, there was only a small reduction of SOX17 (<span class="fakelink linktofig" data-ref="5f159eaf-6bd7-48fc-b18b-0e2e75943dc2"><span class="fakelink-text" data-ref="2291478d-366c-4b6c-8881-85293ead4e6a">Fig. 5j</span><a class="fake-icon" href="#fig5" data-ref="6fdc0b15-b919-412c-9d2e-a072f1d1ec91"><span class="fakelink-icon" data-ref="c8d10314-bee4-4ac3-a934-df9abfa76a60">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Since we previously found with our protocol most SOX17 positive cells at 42h represent PGC-LCs, while most endoderm differentiates later<sup data-ref="7f72d7a7-efcf-4d0b-a403-83c296fe8465"><a href="#c34" data-ref="46bd82bc-c1f5-47d5-be39-01cfd09ffb20">34</a></sup>, this suggests FGF4/17 are not required for PGC-LC induction but does not rule out a requirement for these FGFs in endoderm differentiation. A double KD of <em data-ref="5ea9d9e1-525a-47f1-bbbf-8f2c9b41ac3f">FGF4</em> and <em data-ref="53891934-019e-4c10-827c-32871c6bff96">FGF17</em> had a similar phenotype as <em data-ref="9c7c188d-3f61-4201-90d0-3f67aa6a29d6">FGF4</em> KD alone, suggesting FGF17 may be downstream of FGF4 (<span class="fakelink linktofig" data-ref="9daf1110-5508-41b0-81a9-2c708e10b7aa"><span class="fakelink-text" data-ref="16d5f6cc-b61d-4c10-bf1f-d24ea27489f9">Supp. Fig. 6k</span><a class="fake-icon" href="#figS6" data-ref="5d347968-9e66-48e1-b53a-96c905f17fc7"><span class="fakelink-icon" data-ref="0395fc28-fe1a-4ece-b08c-a68f3b52a742">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Combined, these data support distinct functions for FGF4, FGF8, and FGF17 in our human gastrulation model, where FGF4 and FGF8 may play opposing roles in PS induction while FGF17 is important during continued differentiation to mesoderm.</p></section></section><section id="s3" data-ref="e9c1a23a-846d-4e47-b528-1786831c8b70" data-id="s3"><h1 class="heading-1" id="el-h1-4" data-ref="2e7f60b6-6e0f-46d2-8beb-882e39db97af" data-id="el-h1-4" data-break-after="avoid">Discussion</h1><p id="el-p-38" data-ref="f8790baa-b4ee-479e-aaa6-6a9117d84a8c" data-id="el-p-38" data-previous-break-after="avoid">We have shown that in a stem cell model for early human gastrulation, FGF-dependent ERK signaling is directly required for differentiation to PS-like cells and their derivatives, and that loss of differentiation upon FGF inhibition can be rescued by directly activating ERK. We also showed that FGF receptors are polarized basolaterally, preventing the colony center from responding to exogenous FGF in the medium. Furthermore, FGF receptor activation is increased in the PS-like ring, matching the pattern of active ERK despite overall receptor expression being lower in the PS-like ring than in the pluripotent center. This suggests elevated ERK activity is due to increased binding of FGFs to the receptors in this region, possibly due to increased FGF concentration. We found that both FGF4 and FGF17 are highly expressed in PS-like cells, while FGF8 is expressed earlier and displaced inward (<span class="fakelink linktofig" data-ref="10f8964c-6682-421c-b771-cd220e80a068"><span class="fakelink-text" data-ref="94c76ab6-5e93-4409-bdf2-dd55787cf80d">Fig.6</span><a class="fake-icon" href="#fig6" data-ref="f81d7b3e-160e-4e81-8f44-657520f1d6d9"><span class="fakelink-icon" data-ref="c5431c20-37df-47ba-921b-c28df7f7c3c3">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>). Knockdown of FGF4 leads to a strong reduction in ERK activity and PS-like cells, suggesting it may be responsible for increased ERK signaling in the PS-like region. In contrast, FGF8 knockdown led to an expansion of ERK activity and PS-like differentiation. Finally, FGF17 knockdown leds to a loss of mesoderm differentiation.</p><p id="el-p-40" data-ref="649d6aa6-9817-4a4e-b180-99d3ac852b4f" data-id="el-p-40">Many questions remain. Although we showed that FGF gradients are likely the dominant cause for the spatial pattern of ERK signaling, our scRNA-seq data suggest other levels of regulation may contribute, which remain to be explored. Furthermore, although we showed FGF/ERK signaling is directly required for PS-like differentiation, it may also act indirectly by affecting the expression of Wnt and Nodal. We will explore this and the more general question of how FGF integrates with the hierarchy of BMP, Wnt, and Nodal in a separate manuscript. Related is the indirect effect FGF may have on patterning by its control of cell growth and thereby cell density. Cell density affects the response to exogenous BMP and downstream expression of Wnt, Nodal, and various inhibitors<sup data-ref="8b404a11-d23e-413c-9992-964981442096"><a href="#c4" data-ref="2f8a35e7-34cd-4810-9bb3-713d5867d1af">4</a>,<a href="#c50" data-ref="206975e5-b461-49a5-8d9b-6b6feaa9691b">50</a>,<a href="#c51" data-ref="8da28019-5639-4b5b-b6af-6dd48e0f60eb">51</a></sup>, as well as FGF itself, possibly causing a “community effect”<sup data-ref="2949ade8-6110-48a6-b847-49c7adcc31cf"><a href="#c52" data-ref="08f15c34-2872-4f89-8fc8-026bebfb1756">52</a>–<a href="#c54" data-ref="4b8e02f3-66c4-4a2c-a038-ca70d6771d3f">54</a></sup>. It likely has other effects as well. However, the rescue of TBXT expression but not growth by ERK activation when FGF receptors are blocked suggests the FGF-dependent cell growth is not required to generate a primitive streak-like ring. The ability of exogenous Wnt and Nodal to induce PS-LCs in </p></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_left_page pagedjs_pagedjs-filler_page page-grid" data-page-number="14" id="page-14">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><figure class="figure tofill imgs-1 shortercaption fig-landscape pagedjs-fill-next-page finishes" id="fig6" data-ref="b31cdd3c-2665-4afa-b708-ae5d2f3873f3" data-id="fig6" style="display: block; position: unset;"><figcaption class="figure__caption" id="el-figcaption-5" data-ref="ac61a943-0c68-45a9-a390-85b4917dba19" data-id="el-figcaption-5"><h3 id="el-h3-5" data-ref="47f78fd9-5be4-4b7b-9121-4b6400d4ea93" data-id="el-h3-5" data-break-after="avoid"><span class="label figure-name" data-ref="94f102f7-4be1-4ec8-ab5a-79a0cff4c4d8"><label class="figure__label" data-ref="6dbf21e6-bb20-4830-8745-bfdf342bfdeb">Figure 6.</label></span>Graphical summary.</h3><p id="el-p-39" data-ref="fa1b9af4-cac0-40e2-aae4-aaf19726e5fe" data-id="el-p-39" data-previous-break-after="avoid">Exogenous FGF2 acts on the colony edge and is prevented from reaching the basal FGFR1 receptors elsewhere by tight junctions. Primitive streak cells express FGF4 and FGF17 and these FGFs are associated with high ERK activity. FGF8 is expressed further inside the colony.</p></figcaption><img class="child-of-figure has-boundary imageonly landscape max-70" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_fig6.tif/full/max/0/default.jpg" alt="" id="el-img-6" data-ref="6bbafbe7-b0ae-4665-81df-87287b6bfcf2" data-id="el-img-6" height="465" width="1436" data-desired-width="67.99242424242425"></figure></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_right_page" data-page-number="15" id="page-15">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e" data-split-to="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-id="article-content" data-split-from="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-split-to="726f25f1-fdd1-41b8-bf49-de736be3e3ae"><section data-ref="e9c1a23a-846d-4e47-b528-1786831c8b70" data-id="s3" data-split-from="e9c1a23a-846d-4e47-b528-1786831c8b70"><p data-ref="649d6aa6-9817-4a4e-b180-99d3ac852b4f" data-id="el-p-40" data-split-from="649d6aa6-9817-4a4e-b180-99d3ac852b4f">very sparse culture while MEK and FGFR inhibition block differentiation at the same density similarly argues against a requirement for FGF-dependent cell growth in PS-LC differentiation (<span class="fakelink linktofig" data-ref="1b856859-37e8-49d5-b846-65b4d445db6a"><span class="fakelink-text" data-ref="438fe7b9-e8f2-487d-9598-319c70a409ca">Supp. Fig. 1h</span><a class="fake-icon" href="#figS1" data-ref="696faf93-12de-48d4-bdc7-db7f68401ec8"><span class="fakelink-icon" data-ref="29599e7b-47ff-4a89-8916-dea757e22e96">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>).</p><p id="el-p-41" data-ref="dbb3afb2-fd15-49e3-99ca-ce3e6f87b779" data-id="el-p-41">Another important question is to what extent different FGFs act redundantly or perform different functions. Our data suggest a picture where different FGFs are required consecutively, starting with exogenous FGF2, then endogenous FGF4, to support initial differentiation of PS-like cells, and finally FGF17 to maintain and expand the PS-like ring and support mesoderm and possibly endoderm differentiation. Whether these FGFs could substitute for each other if they were expressed at similar times and levels is unclear. It also remains to be determined whether the nearly constant high expression of FGF2 we observed is required for patterning. Most puzzling is the role of FGF8, whose knockdown led to an increase in PS-like differentiation. Given that FGF8 expression is displaced inward, one possibility is that it induces inhibitors of other PS-inducing signals there in order to limit PS differentiation, e.g., the Nodal inhibitor Lefty, which is expressed in a similar domain<sup data-ref="bbbd1cf1-16fc-42a5-8936-9b72da47b6f3"><a href="#c55" data-ref="e70913a0-027d-4009-bd88-c4d0210af56f">55</a></sup>. The rescue of PS-like differentiation by uniform activation of ERK to a supraphysiological level suggests that the role of the different FGFs is simply to sufficiently activate ERK signaling. However, future work will have to investigate the dox-SOS<sup data-ref="6cace541-b957-4726-a3fc-6e5b6ec63c13">cat</sup> rescue phenotype in much greater detail to see how differentiation and other cell behaviors such as migration are affected.</p><p id="el-p-42" data-ref="1b305b32-d187-4b3e-a1e0-06164842a6ca" data-id="el-p-42">It is interesting to ask if there is a simple explanation for the seemingly large differences in FGF expression between primates and other species. FGF8 is the only FGF shown to be required for mouse gastrulation and is upstream of FGF4 in both chick and mouse, but it is barely expressed in human and monkey primitive streak and appears to play a different role in our stem cell model. If FGFs are interchangeable in simply activating ERK for primitive streak induction, variation in FGF expression is not surprising. However, FGF4 and FGF8 are the main FGFs in mouse, chick and frog gastrulation<sup data-ref="5fda6491-6cf9-4da8-89e5-0fcad7d03f3b"><a href="#c56" data-ref="ed79d7e2-e179-458e-92f1-17dfed17efe6">56</a>–<a href="#c58" data-ref="c14afeba-03e1-4633-8497-d8b2b82dddbe">58</a></sup>, suggesting a high degree of conservation. It is possible FGF8 has simply been substituted by FGF17, which is in the same subfamily<sup data-ref="81c41042-525f-4c89-a5a8-24c71bc403f4"><a href="#c9" data-ref="7f27ddee-79a9-4e76-a8f0-3bf3fc75b4fe">9</a></sup>, but this does not explain why the role of FGF8 in gastrulation otherwise appears strongly conserved, or why FGF8 is upstream of FGF4 in the mouse while FGF4 is expressed before FGF17 in 2D gastruloids. Similarly, it is possible that FGF2 in primates substitutes for FGF5 in the mouse as the FGF that is highly expressed in the epiblast. Although these are not in the same subfamily and generally have different receptor affinities, they both have highest affinity for FGFR1<sup data-ref="cc0d53d3-dd7e-49e0-bceb-cfc47b9740ca"><a href="#c9" data-ref="595f8644-6489-4960-bcd5-1cebca512b06">9</a></sup>. That leaves FGF4 as the only FGF that may be conserved across vertebrate gastrulation, and it is tempting to speculate on why. In Xenopus, FGF4 and TBXT (Xbra) function in a positive feedback loop<sup data-ref="f1f47e8d-bbaa-456f-a820-76281f988e4d"><a href="#c22" data-ref="9ade2534-d3a1-44f2-819d-91407c382c54">22</a>,<a href="#c59" data-ref="59da9f18-1939-4137-984b-32c2974f320d">59</a></sup>. The strongly correlated initial transcriptional dynamics of TBXT and FGF4 in 2D gastruloids are consistent with a similar feedback loop. However, an FGF8 feedback loop has also been proposed and recent work suggests TBXT is not required for FGF4 or FGF17 expression<sup data-ref="7aab2a53-7240-4988-8d83-72e2c4d63432"><a href="#c54" data-ref="f1141bf8-42d6-4591-8e34-cf44966c114a">54</a>,<a href="#c60" data-ref="88040d36-6614-4674-85fe-0a08fd48e25d">60</a>,<a href="#c61" data-ref="21336814-d806-4ab5-b283-c769804b05cc">61</a></sup>. Future work will therefore have to investigate further if FGF4 plays a unique role.</p><p id="el-p-43" data-ref="c8d9165b-34ee-4aaa-bf85-b146368617be" data-id="el-p-43">A final related question is the origin of the discrepancies we and others<sup data-ref="f49ca5f2-04ad-4107-8e8a-4d8af9d29011"><a href="#c36" data-ref="dc3eeaac-64aa-4bfc-aea7-73a12f9c1f57">36</a></sup> found between the primate (i.e., non-mouse) samples in <span class="fakelink linktofig" data-ref="210d8a4a-e632-485c-a89c-d0e06d5ce027"><span class="fakelink-text" data-ref="9318d535-e5bf-4e42-a4ab-5b39e6176756">Fig. 2</span><a class="fake-icon" href="#fig2" data-ref="6586d6fb-a971-4418-8a62-e1977760e9d3"><span class="fakelink-icon" data-ref="c4cb6036-96f0-4a70-af50-ec3a1cc8a1ae">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span> and <span class="fakelink linktofig" data-ref="300d04eb-461e-4868-962a-2a8d21eea702"><span class="fakelink-text" data-ref="d74a5ab0-2044-4607-aadc-cf45d22f94d0">Supp. Fig. 2</span><a class="fake-icon" href="#figS2" data-ref="479d1a3f-f88a-481b-b876-107bea21cf8a"><span class="fakelink-icon" data-ref="035f8ab8-d557-435d-8589-19cdff9f0d76">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>: FGF3 appears only significantly expressed in the human embryo, but not in the monkey embryo and human stem cell model. This could also reflect genetic variation: if FGFs function redundantly, their relative expression levels could be relatively unconstrained between individuals. On the other hand, in the nascent mesoderm, FGF2 is only high in human gastruloids. This may reflect developmental stage: the in vivo data is developmentally more advanced and nascent mesoderm may lose FGF2 expression later. It is also possible the maintenance conditions of hPSCs lead to in vitro artefacts in FGF expression.</p><p id="el-p-44" data-ref="570e22da-0774-4bbf-bcb4-5c9ebbb8d892" data-id="el-p-44">In conclusion, the complexity of FGF signaling, with its many ligands, modulators, and functions, has prevented a clear understanding of its role in mammalian gastrulation in general and in human gastrulation specifically. Here we showed that 2D human gastruloids provide a powerful approach to make headway and although many open questions remain, we have taken significant steps towards this goal.</p></section><section id="s7" data-ref="f25cc01d-3ded-4c57-9468-d579e87fa17a" data-id="s7" data-split-to="f25cc01d-3ded-4c57-9468-d579e87fa17a" data-split-original="true" data-last-split-element="true" data-align-last-split-element="auto"></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_left_page" data-page-number="16" id="page-16">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e" data-split-to="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-id="article-content" data-split-from="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-split-to="726f25f1-fdd1-41b8-bf49-de736be3e3ae"><section data-ref="f25cc01d-3ded-4c57-9468-d579e87fa17a" data-id="s7" data-split-from="f25cc01d-3ded-4c57-9468-d579e87fa17a" data-split-to="f25cc01d-3ded-4c57-9468-d579e87fa17a"><h1 class="heading-1" id="el-h1-5" data-ref="6ad604e2-9286-4562-a537-b1b46441d3a2" data-id="el-h1-5" data-break-after="avoid">Methods</h1><section id="s7a" data-ref="0b712780-4be5-4066-b3f5-989cd95398cb" data-id="s7a" data-previous-break-after="avoid"><h2 class="heading-2" id="el-h2-6" data-ref="61675e78-3e50-46b2-a41c-a40246f3aac4" data-id="el-h2-6" data-break-after="avoid" data-following="following-ff2c00ba-51a1-482d-9afa-f5a370dcfa50">Cell lines</h2><p id="el-p-45" data-ref="57bbfd77-599e-4c3e-99e8-98ad0fc4a87a" data-id="el-p-45" data-previous-break-after="avoid">The cell lines used were the embryonic stem cell line ESI017 (XX), and the induced pluripotent stem cell line PGP1 (XY). The pluripotency of these cells was confirmed by immunostaining of pluripotency markers OCT3/4, SOX2, NANOG. All cells were routinely tested for mycoplasma contamination, and negative results were recorded. ESI017 cells stably expressing doxycycline inducible SOS<sup data-ref="f1642262-f169-4af3-8072-b3bcaee5ee00">cat</sup> were made using a published Piggybac plasmid gifted by Jared Toettcher<sup data-ref="930d44ac-a38d-4696-9053-c4fc61a899d9"><a href="#c37" data-ref="565a5866-abeb-48d2-a36f-d8cdc8961fc0">37</a></sup>.</p></section><section id="s7b" data-ref="150926f1-9429-498e-882e-e1a643873856" data-id="s7b"><h2 class="heading-2" id="el-h2-7" data-ref="5cd8b556-412c-4bcc-9f45-e9f342d3652f" data-id="el-h2-7" data-break-after="avoid">Cell culture and differentiation</h2><p id="el-p-46" data-ref="9ab6021f-8d25-4b0b-b8d0-215c91d45769" data-id="el-p-46" data-previous-break-after="avoid">Human pluripotent stem cells were cultured in the commercially defined pluripotency maintaining media mTeSR1 (STEMCELL Technologies #85850) on Cultrex (R&amp;D Systems)-coated tissue culture plates. For FGF2 starvation we used Essential 6 Medium (Thermofisher, A1516401), or mTeSR1 without bFGF, TGF-beta, LiCl, GABA and pipecolic acid (STEMCELL Technologies #05896). For routine cell maintenance, cell passaging was performed every 3 days. For whole-colony passaging, L7<sup data-ref="000a1f48-2aea-4388-8b44-4c2c8aeff949"><a href="#c62" data-ref="f4c13c4c-6a3c-43ec-9ec5-2fda4b83d1b0">62</a></sup> was used, while single-cell passaging was performed with Accutase or TrypLE (Gibco). Single-cell suspensions were used to seed for all the differentiation experiments to control initial cell number.</p><p id="el-p-47" data-ref="ec85afef-3ce6-47c1-beec-d0ff2c3cc9f3" data-id="el-p-47">To directly differentiate to primitive streak-like cells, cells were resuspended in a single-cell suspension and seeded in cultrex coated wells in mTeSR with ROCK inhibitor (RI) Y-27632 (MeChemExpress, cat# HY-10583). Cells were then maintained in mTeSR with RI for 24h before adding CHIR (3-5uM) and Activin (100ng/ml) treatment for 24-30h. Experiments were performed in 18-well Ibidi slides (cat# 81818).</p><p id="el-p-48" data-ref="839eff24-515f-4ced-a6b2-6868c058e15b" data-id="el-p-48">To differentiate cells for 2D gastruloids by micropatterning, we followed the protocol in<sup data-ref="2caca2f0-0f8c-4bcc-94e1-5cc4b007bee2"><a href="#c42" data-ref="5fa68afb-2d7c-4737-9526-2d08f171a91d">42</a></sup>. Briefly, cells were resuspended in a single-cell suspension and seeded in laminin-coated micropatterned wells in mTeSR with ROCK inhibitor (RI) (MeChemExpress, cat# HY-10583). Micropatterned wells were washed with PBS-/- 30 minutes after the initial seeding to wash off the non-attached cells. Cells were then maintained for 24 hours in mTeSR before adding the BMP4 treatment by a full medium change. For time series micropatterned differentiation, full media changes with designated treatment(s) were performed every 24 hours. Experiments were performed in micropatterned 18-well Ibidi slides (cat# 81818) prepared as previously described<sup data-ref="4c2c6418-1fc7-4297-9120-ff2fbe663a90"><a href="#c63" data-ref="480d2a57-3eca-4a63-914c-5411b565f239">63</a></sup>. Cell signaling reagents and doses used are listed in <span class="fakelink linktofig" data-ref="f5ae2f6b-f752-4664-975e-b88e87d2ebe9"><span class="fakelink-text" data-ref="afd575c4-0e49-4723-a046-758c5cac2108">Supplementary Table 1</span><a class="fake-icon" href="#tblS1" data-ref="b1120fe2-987f-4621-9027-40640c16bd84"><span class="fakelink-icon" data-ref="ea33a979-2c89-4057-999f-2652b4b3cc31">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>.</p></section><section id="s7c" data-ref="a21fc0f9-279a-45f5-952b-8e1a9aed017f" data-id="s7c"><h2 class="heading-2" id="el-h2-8" data-ref="6165b282-f2d7-4f45-b2f2-2053dd3f051d" data-id="el-h2-8" data-break-after="avoid">Immunofluorescence staining</h2><p id="el-p-49" data-ref="6e6b3db4-0f65-4d31-901a-020b2bdd52a3" data-id="el-p-49" data-previous-break-after="avoid">Antibodies can be found in Supplementary Tables 2 and 3. For stains including pERK, we used methanol fixation, for all other stains we used PFA fixation. PFA fixation: Samples from the 18-well Ibidi slides were rinsed with PBS, fixed for 20 min in 4% paraformaldehyde, rinsed twice with PBS, and blocked for 30 min at room temperature with 3% donkey serum and 0.1% Triton X-100 in 1× PBS. After blocking, cells were incubated with primary antibodies at 4°C overnight, followed by three washes in PBST (PBS with 0.1% Tween 20). They were then incubated with secondary antibodies and DAPI for 30 min at room temperature and washed twice in PBST at room temperature. Methanol Fixation: Samples from the 18-well Ibidi slides were rinsed with PBS, fixed for 20 mins in cold methanol, then rinsed with PBS for 5 min two times, after which 10% phosphate buffered formalin was added for 20 min at room temperature. Cells were then rinsed with TBS twice followed by 5 mins of incubation in cold methanol and rehydrate in TBS for 20 mins. Blocking was done for 30 mins at room temperature with 5% donkey serum, 1% BSA, 0.2% Triton-X-100 in 1x TBS after which staining proceeded the same way as with PFA fixation.</p></section><section id="s7d" data-ref="f6841dce-8542-4b85-8936-690726287081" data-id="s7d" data-split-to="f6841dce-8542-4b85-8936-690726287081" data-split-original="true"><h2 class="heading-2" id="el-h2-9" data-ref="6b8756ab-99dd-49a7-9fa0-952437b46d8d" data-id="el-h2-9" data-break-after="avoid">Microscopy</h2><p id="el-p-50" data-ref="b1cf77dd-7555-42dd-8e53-135243e48a9c" data-id="el-p-50" data-previous-break-after="avoid" data-split-to="b1cf77dd-7555-42dd-8e53-135243e48a9c" data-split-original="true" data-last-split-element="true" data-align-last-split-element="auto">Fixed sample imaging was performed with an Andor Dragonfly/Leica DMI8 spinning disk confocal microscope with a 40x, 1.1NA water and 20x 0.8NA air objectives using Andor Fusion software version 2.3.0.31 and a Nikon/Yokogawa spinning dish confocal microscope with a 40x silicon oil </p></section></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_right_page" data-page-number="17" id="page-17">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e" data-split-to="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-id="article-content" data-split-from="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-split-to="726f25f1-fdd1-41b8-bf49-de736be3e3ae"><section data-ref="f25cc01d-3ded-4c57-9468-d579e87fa17a" data-id="s7" data-split-from="f25cc01d-3ded-4c57-9468-d579e87fa17a" data-split-to="f25cc01d-3ded-4c57-9468-d579e87fa17a"><section data-ref="f6841dce-8542-4b85-8936-690726287081" data-id="s7d" data-split-from="f6841dce-8542-4b85-8936-690726287081"><p data-ref="b1cf77dd-7555-42dd-8e53-135243e48a9c" data-id="el-p-50" data-split-from="b1cf77dd-7555-42dd-8e53-135243e48a9c">objective using NIS Elements AR software version 5.41.02. Live-cell imaging was performed with an Andor Dragonfly/Leica DMI8 spinning dish confocal microscope under the controlled temperature (37°C), CO<sub data-ref="9f158469-51c3-4fa6-8c86-b3644bd51afb">2</sub> concentration (5%), and humidity (&gt;60%).</p></section><section id="s7e" data-ref="9b72425b-4768-4a1b-9cd6-348219338ede" data-id="s7e"><h2 class="heading-2" id="el-h2-10" data-ref="34b5e65f-d55d-4686-a3d4-76121e2c0188" data-id="el-h2-10" data-break-after="avoid">General image analysis</h2><p id="el-p-51" data-ref="0faa21c5-3627-460a-862d-c9996859e397" data-id="el-p-51" data-previous-break-after="avoid">We segmented nuclei in individual z-slices based on nuclear fluorescence stained with DAPI in fixed cells using a pipeline we previously described, which integrates two machine learning approaches: Ilastik pixel classification<sup data-ref="01c8f6f6-9bfd-4248-a2ed-62eb64ac203b"><a href="#c64" data-ref="437ec40f-95a3-4125-b384-74de7d0218d2">64</a></sup> and Cellpose<sup data-ref="2f00378c-68ed-42be-aa35-59f7bcee2d14"><a href="#c65" data-ref="1ae52269-7686-4d57-b104-2509fea57ee6">65</a></sup> (v1). Fluorescence intensities were then calculated per nucleus as mean intensities in the 3D nuclear again with our established custom image-processing pipeline. To calculate the radial fluorescence distributions in micropatterned colonies, colonies were subdivided into radial bins with equal numbers of cells, mean or median and variance were then calculated within these bins. To calculate positive fractions, markers were first thresholded based on their intensity distribution, after which the fraction of cells positive for each marker was calculated within different bins. Standard deviations were then calculated over multiple replicate colonies. For visualization purposes only, background subtraction using grayscale opening on a 100 micron scale and 5 pixel wide median filter was applied slice by slice to z-stacks of micropatterned colonies before making maximal intensity projections.</p></section><section id="s7f" data-ref="0f956348-533b-43b8-b2b2-2f7526ea3401" data-id="s7f" data-split-to="0f956348-533b-43b8-b2b2-2f7526ea3401" data-split-original="true" data-last-split-element="true" data-align-last-split-element="auto"><h2 class="heading-2" id="el-h2-11" data-ref="8c36027d-88b0-4702-b7b3-3c92e2bd1f28" data-id="el-h2-11" data-break-after="avoid">(Single-cell) RNA sequencing and analysis</h2><p id="el-p-52" data-ref="e021b4e3-cc58-4ad1-93fa-db993eb5c0d6" data-id="el-p-52" data-previous-break-after="avoid">Cells were collected using accutase and resuspended in ice-cold PBS. Single-cell RNA-sequencing was performed by the University of Michigan Advanced Genomics Core. Cells were barcoded using the 10X Genomics Chromium system (part numbers 1000268, 1000120, 1000215). For quality control, cDNA was quantified by Qubit High Sensitivity DNA assay and Agilent TapeStation. Sequencing was performed on the Illumina NovaSeq 6000 with NovaSeq S4 flowcell and Control Software version 1.7.0. Reads were aligned using cellranger-4.0.0 with the GRCh38 reference. Further processing was mostly done in Python, primarily using the Scanpy<sup data-ref="ac4a3837-34c7-4217-bd4d-a991a1cf3cb6"><a href="#c66" data-ref="2af401c4-d3a2-42e0-b484-371d20412855">66</a></sup> and SCVI<sup data-ref="de7ab197-11b5-41e6-8398-35984586e492"><a href="#c67" data-ref="9a9e6101-8cb1-49e8-814f-02f3b799aa99">67</a></sup> packages. Analysis was performed on log transformed data. Library size normalization was performed on the intersection of the genes present in our data and the reference CS7 human embryo data to facilitate comparison of expression levels. We integrated the two scRNA-seq datasets for 42h 2D human gastruloids using SCVI based on the top 2000 highly variable genes, which were selected using the default function in Scanpy using the seurat_v3 flavor. Cell cycle genes were regressed out during integration. Dimensional reduction for visualization was performed using PHATE<sup data-ref="851b30ac-f8d2-40a3-a9a0-e267c7e32a7e"><a href="#c39" data-ref="95e6a9b4-0173-4111-a20d-017016cddc72">39</a></sup> on the SCVI latent space. Gene expression was smoothened with MAGIC<sup data-ref="db18db62-4910-4a19-a3ff-4f9f347714f1"><a href="#c68" data-ref="83c74556-8da6-4c23-9d44-cdd916c0ce69">68</a></sup> for visualization on PHATE plots in <span class="fakelink linktofig" data-ref="ddc20c16-3e99-4894-a2d0-1c88215057f0"><span class="fakelink-text" data-ref="94963edc-0534-4b8a-9a35-ef10c4f77c97">Fig. 2e,g-i</span><a class="fake-icon" href="#fig2" data-ref="8304059c-421d-4083-a0a6-d717825da2e0"><span class="fakelink-icon" data-ref="dcc01910-e18d-485b-bf5d-30cd2abb7c77">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. To annotate different cell types, we used MAGIC to impute that data and get smooth distributions of gene expression. We then thresholded marker genes to assign cell fate (Table 4), which produced better results than the more common unsupervised clustering using Leiden, and in particular more consistent results between datasets from different species. For interspecies comparison of FGF ligands we normalized the to the maximal FGF expression within each species, where for the mouse data, the three different time points were first integrated using SCVI. Thus we compare between species the relative expression of FGF ligands within each species, e.g., FGF8 expression is much higher than FGF4 at any time in the mouse, but much lower than FGF4 in human.</p><p id="el-p-53" data-ref="83064a85-15d8-4ba1-869d-9dfcc335ea69" data-id="el-p-53">For pseudotime analysis (<span class="fakelink linktofig" data-ref="43f039a6-28a3-4273-9887-ad7467fa99ec"><span class="fakelink-text" data-ref="03421c4d-1050-4ab9-a6e2-af6bd62838de">Fig. 5</span><a class="fake-icon" href="#fig5" data-ref="7aae573d-03c2-466e-bcd8-5490c2ec93f5"><span class="fakelink-icon" data-ref="53e50d73-60bc-41e6-af6f-7d7bf0c9fad8">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>), we applied the trajectory inference tool Slingshot<sup data-ref="7191f9a0-4d88-4145-84e7-4d5911b96e0b"><a href="#c69" data-ref="18871494-b73d-4943-9f31-7e860f88c098">69</a></sup> in R to compute the pseudotime trajectory in the SCVI latent representation. Slingshot determines the membership of a cell to a lineage at each branching point by its projection distance to the fitted principal curve of that lineage. We projected the mesoderm developmental trajectory onto the PHATE map for visualization. We further used TradeSeq<sup data-ref="a19b8906-3ef6-4191-800c-97b44b990099"><a href="#c70" data-ref="3e812994-0933-48c2-8d1e-04d7f3106e76">70</a></sup>, which fits a generalized additive model (GAM), to infer the pseudotime gene expression trends along the mesoderm lineage trajectory from Slingshot. For visualizing the correlation between the expression of FGF ligands and marker genes (<span class="fakelink linktofig" data-ref="e475cd28-56fe-435a-a7d5-a57f72d3558b"><span class="fakelink-text" data-ref="090e91f1-d8e6-42ae-b1c7-7c4a23a55e53">Supp. Fig. 5</span><a class="fake-icon" href="#figS5" data-ref="467dd987-a3bf-4d3e-ba81-371ba8ca67e2"><span class="fakelink-icon" data-ref="debeee53-f251-4962-8e03-46ec71abe05e">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>), we used our SCVI model to denoise the expression and mutual nearest neighbor to further harmonize the difference between samples.</p></section></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_left_page" data-page-number="18" id="page-18">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e" data-split-to="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-id="article-content" data-split-from="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-split-to="726f25f1-fdd1-41b8-bf49-de736be3e3ae"><section data-ref="f25cc01d-3ded-4c57-9468-d579e87fa17a" data-id="s7" data-split-from="f25cc01d-3ded-4c57-9468-d579e87fa17a" data-split-to="f25cc01d-3ded-4c57-9468-d579e87fa17a"><section data-ref="0f956348-533b-43b8-b2b2-2f7526ea3401" data-id="s7f" data-split-from="0f956348-533b-43b8-b2b2-2f7526ea3401"><p id="el-p-54" data-ref="082cf72d-7a42-47a5-a2e3-54e392b38837" data-id="el-p-54">To quantify isoforms from RNA sequencing salmon v1.10.0 was used<sup data-ref="e393c055-8991-49fd-9c8e-d6b7fa538d56"><a href="#c71" data-ref="0f9a24aa-2b85-42c4-9e24-595b01e13ff5">71</a></sup>. First, gencode.v45.transcripts.fa was downloaded from <span class="fakelink linktoext" data-ref="57595889-c601-4de2-9ff9-9a679b201553"><span class="fakelink-text" data-ref="a54384a1-b9f4-49c5-98eb-0d9f6431bbe6">gencodegenes.org</span><a class="fake-icon" href="https://gencodegenes.org" data-ref="ecfca714-e6be-4db4-9264-0efd97b0954f"><span class="fakelink-icon" data-ref="d55d00e0-1720-46b7-8610-43dfc6ec12c7">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. The reference was customized by removing lines for FGFR1 and replacing them with the transcript sequence for FGFRIIIa, FGFRIIIb, FGFRIIIc (sequences included as supplementary file). Three samples from GEO dataset GSM2257301 were downloaded through SRA (SRX1725562 (Pluripotent), SRX1725577 (APS), SRX1725643 (MPS))<sup data-ref="4beb6ff6-f2de-453a-92f8-c6a0fc08b9e9"><a href="#c38" data-ref="5f591f21-df67-49c7-a327-135a2b5e78e9">38</a></sup>. Adapters and the first 10 bases were trimmed from the FASTQ files using trim_galore. Finally, the Salmon “quant” command with the --validateMappings option was used to quantify transcripts from trimmed FASTQ files. For single cell datasets, <span class="fakelink linktoext" data-ref="b9e68d9d-6c41-4f4e-9168-d1175663dc6e"><span class="fakelink-text" data-ref="322e9f7d-b9a2-4037-bb51-1d959a29feec">https://timoast.github.io/sinto/</span><a class="fake-icon" href="https://timoast.github.io/sinto/" data-ref="7fbcbd8f-33e0-4e0c-878b-5815e09d777c"><span class="fakelink-icon" data-ref="511f6b7f-1d44-450d-8ad6-112aad1518a6">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span> was used to subset the 10x aligned file by cluster. Reads for clusters of interest were converted back to raw FASTQ files and run through the same alignment-based mode of salmon but as single end reads.</p></section><section id="s7g" data-ref="f4ff849e-52a8-40cc-9dd6-34719eced446" data-id="s7g"><h2 class="heading-2" id="el-h2-12" data-ref="91800f1b-e2d8-43b6-b73e-f79016758a5a" data-id="el-h2-12" data-break-after="avoid">Quantitative real-time PCR</h2><p id="el-p-55" data-ref="42c7e9e3-93f5-49e0-90cf-37a3c2a23b7e" data-id="el-p-55" data-previous-break-after="avoid">RNA were extracted using RNAqueous™-Micro Total RNA Isolation Kit (CAT# AM1931, Thermo Fisher Scientific) and then cDNA synthesis was prepared from it using SuperScript™ VILO™ cDNA Synthesis Kit (CAT# 11754250, Thermo Fisher Scientific) following the manufacturer‘s protocol with adjustment for optimization. Measurements were performed with SYBR green and the primers in the Table 5. GAPDH was used for normalization in all experiments.</p></section><section id="s7h" data-ref="32bd58a5-180a-41a2-b54c-4050df7b69ba" data-id="s7h"><h2 class="heading-2" id="el-h2-13" data-ref="34e76d36-a9b7-4e79-a7ab-27d68ec145c3" data-id="el-h2-13" data-break-after="avoid">Fluorescent dextran assays</h2><p id="el-p-56" data-ref="0da23cad-788e-42e1-9015-c95ab5f538dd" data-id="el-p-56" data-previous-break-after="avoid">Fluorescent dextran essays were performed by adding 10 ug/ml fluorescein isothiocyanate-dextran (10KD, Sigma Aldrich) to the media. To calculate fluorescence intensity as a function of distance from the colony edge, Otsu thresholding followed by several morphological operations was used to create a mask for the colony in each z-slice, thereby excluding bright dextran fluorescence outside the colony. Background subtraction on each z-slice of the image was performed to further remove background signal from dextran outside the colony while maintaining signal in the intercellular space. The radial intensity profile was then calculated from the maximal intensity projection of the masked, background subtracted image.</p></section><section id="s7i" data-ref="04d38b56-a0c1-4a22-bfbe-8dd33f2e07c0" data-id="s7i"><h2 class="heading-2" id="el-h2-14" data-ref="03876d94-c8fe-482e-97df-444d169f8c43" data-id="el-h2-14" data-break-after="avoid">Scratch assays</h2><p id="el-p-57" data-ref="02c0a806-715b-4e60-bd85-151a487387f8" data-id="el-p-57" data-previous-break-after="avoid">Human embryonic or induced pluripotent stem cells were differentiated on micropattern wells for 41h, then each micropattern colony was manually scratched with a serological needle. After 60mins, they were fixed with PFA or methanol followed by immunostaining with appropriate antibodies. To calculate the intensity profile as a function of distance from a scratch, we created a mask for the scratch based on the difference between an overall mask for the colony and the convex hull, calculated the distance transform of the scratch mask, then binned pixels by distance from the scratch and calculated their mean ERK intensity.</p></section><section id="s7j" data-ref="fc78b528-aae1-40fa-bdfe-e056d1b3574d" data-id="s7j"><h2 class="heading-2" id="el-h2-15" data-ref="5b7ef439-2aa2-49f5-8bce-15dae308d04f" data-id="el-h2-15" data-break-after="avoid">Transmembrane assays</h2><p id="el-p-58" data-ref="073be456-8953-4ecf-8457-93b4fd843592" data-id="el-p-58" data-previous-break-after="avoid">Human embryonic or induced pluripotent stem cells were seeded into Corning 6.5 mm transwell with 0.4 µm pore polyester membrane insert (cat# 3470) with E6 supplemented with TGF1b, FGF2, and Y-27632 for 16h, then media was changed to E6 with TGF1b. After 24h, the media was changed to E6 supplemented with FGF2 from the top or the bottom side of the transwell. 30 mins later, cells were fixed followed by immunofluorescence staining. The transwell cartoon in <span class="fakelink linktofig" data-ref="814b8c31-ceda-4935-a316-fed03aa18699"><span class="fakelink-text" data-ref="5f415d68-16d6-4ac9-bfe5-bcc4a157c7e0">Fig 4</span><a class="fake-icon" href="#fig4" data-ref="c58f9235-b9b8-4d89-8a9f-3b85d4e2f32a"><span class="fakelink-icon" data-ref="3a1651ef-d2c2-4540-bcca-f9300871bc62">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span> was generated with Biorender.</p></section><section id="s7k" data-ref="ed05afaf-1f04-47f9-9f62-e531faf702f5" data-id="s7k"><h2 class="heading-2" id="el-h2-16" data-ref="53ef41a2-5e0a-473e-b037-947e6bddc23f" data-id="el-h2-16" data-break-after="avoid">Fluorescence In Situ Hybridization</h2><p id="el-p-59" data-ref="3c27d096-f088-4827-ab65-4d498e7b42d7" data-id="el-p-59" data-previous-break-after="avoid">The fluorescence in situ hybridization protocol (FISH) was performed according to the manufacturer’s instructions (ACDbio, RNAscope multiplex fluorescent manual). A list of probes and reagents can be found in <span class="fakelink linktofig" data-ref="4d142ec1-e6d5-493e-a5ca-018688ae14b3"><span class="fakelink-text" data-ref="c01f2b3f-bad0-4466-8bf9-62af85f85683">Supplementary Table 6</span><a class="fake-icon" href="#tblS6" data-ref="25d664e1-bb11-439e-8394-7dfb0c0dbdca"><span class="fakelink-icon" data-ref="36505a63-adf5-411c-a9af-f13c715c32c0">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>.</p></section><section id="s7l" data-ref="d307c65d-81ea-4996-8c64-d010e3c7e2bb" data-id="s7l" data-split-to="d307c65d-81ea-4996-8c64-d010e3c7e2bb" data-split-original="true" data-last-split-element="true" data-align-last-split-element="auto"></section></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_right_page" data-page-number="19" id="page-19">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e" data-split-to="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-id="article-content" data-split-from="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-split-to="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-last-split-element="true" data-align-last-split-element="auto"><section data-ref="f25cc01d-3ded-4c57-9468-d579e87fa17a" data-id="s7" data-split-from="f25cc01d-3ded-4c57-9468-d579e87fa17a"><section data-ref="d307c65d-81ea-4996-8c64-d010e3c7e2bb" data-id="s7l" data-split-from="d307c65d-81ea-4996-8c64-d010e3c7e2bb"><h2 class="heading-2" id="el-h2-17" data-ref="25de984b-002a-4a99-8353-1a7d891c47a3" data-id="el-h2-17" data-break-after="avoid">Gene targeting by shRNA</h2><p id="el-p-60" data-ref="31025fd3-a04f-4214-8e31-d32ff55d7453" data-id="el-p-60" data-previous-break-after="avoid">We used the constructs shRNA F4: pPB[shRNA]-EGFP-U6&gt;hFGF4[shRNA#1] and shRNA F17: pPB[3shRNA]-U6&gt;hFGF17[shRNA#1]-U6&gt;hFGF17[shRNA#2]-U6&gt;{hFGF17_shRNA3}-hPGK&gt;Bsd both purchased from VectorBuilder in a PiggyBac Transposon Vectors with puromycin resistance which was stably integrated into ESI17 hESCs and PGP1 hiPSCs. Transfections were performed using Lipofectamine™ Stem Transfection Reagent (CAT# STEM00001, Thermo Fisher Scientific), following the manufacturer‘s protocol.</p></section><section id="s7m" data-ref="54094027-9439-442f-a6b7-a98f6f8c4349" data-id="s7m"><h2 class="heading-2" id="el-h2-18" data-ref="ebe8d40f-6484-419b-9d59-f93c9d09718b" data-id="el-h2-18" data-break-after="avoid">Gene targeting by siRNA</h2><p id="el-p-61" data-ref="cd852a38-25c5-4e22-b062-38aeb8952bdf" data-id="el-p-61" data-previous-break-after="avoid">We purchased predesigned siRNA from Millipore Sigma; NM_002007 (FGF4), NM_006119 (FGF8), NM_008004 (FGF17). The siRNA transfections were performed using Lipofectamine™ RNAiMAX Transfection Reagent (CAT# 13778075, Thermo Fisher Scientific), following the manufacturer‘s protocol with adjustment for optimization.</p></section><section id="s7n" data-ref="d46ccec0-da44-4b25-b83b-5aabbae49bc5" data-id="s7n"><h2 class="heading-2" id="el-h2-19" data-ref="0de825ad-732f-43a5-906e-81417efcd151" data-id="el-h2-19" data-break-after="avoid">Statistics and reproducibility</h2><p id="el-p-62" data-ref="abc720a0-8cf9-47c9-843b-00684ba397dd" data-id="el-p-62" data-previous-break-after="avoid">All experiments were performed twice or more. All the attempts in replicating experiments yielded consistent results. Unless specifically noted, all the quantifications were performed with N=4 colonies within the same experimental condition.</p></section><section id="s7o" data-ref="427bca71-fa8c-4445-b2da-c76519730cf2" data-id="s7o"><h2 class="heading-2" id="el-h2-20" data-ref="163cc390-29cb-4355-8e6b-a6b39bc08698" data-id="el-h2-20" data-break-after="avoid">Ethics statement</h2><p id="el-p-63" data-ref="05de05ab-8d2d-477e-b8ef-2f8593c3842a" data-id="el-p-63" data-previous-break-after="avoid">We comply with all the ethical regulations in our research. We were granted approval by the Human Pluripotent Stem Cell Research Oversight (HPSCRO) Committee at the University of Michigan to work with human embryonic stem cells and human induced pluripotent stem cells.</p><ul class="list-bullet" id="el-ul-3" data-ref="c33faaaa-41b9-4f86-83b5-64b5615091f8" data-id="el-ul-3"><li id="el-li-20" data-ref="f798bb57-79c7-490b-96a9-1e95e7e1bf03" data-id="el-li-20"><p id="el-p-64" data-ref="e292462d-0d3c-4213-b8ef-9f7ed9383bb3" data-id="el-p-64">Read orthogroup alignment</p></li><li id="el-li-21" data-ref="fb0829cc-a4e3-438a-b57c-d7255e556e9e" data-id="el-li-21"><p id="el-p-65" data-ref="46376225-f4f1-4d9e-b94c-e24d21290869" data-id="el-p-65">Pre-process alignment</p><ol class="list-alpha-lower" id="el-ol-2" data-ref="3697a35c-e3a0-48bf-98a6-e4e0347b0149" data-id="el-ol-2" start="1"><li id="el-li-22" data-ref="77f9634f-b1c3-46d0-b8aa-dd07b1d64473" data-id="el-li-22" data-item-num="1"><p id="el-p-66" data-ref="dbbd5faa-2b85-45c1-b1a9-5ffc6608f6ad" data-id="el-p-66">Trim alignment edges and remove columns with high fraction of gap codons</p></li><li id="el-li-23" data-ref="a5898ff5-63ee-4a46-b620-7721290f05d6" data-id="el-li-23" data-item-num="2"><p id="el-p-67" data-ref="87a74e1b-9d35-4b4d-8336-cc0e7971f21e" data-id="el-p-67">For each species, infer consensus sequence and remove sequences outside a predetermined diameter around the consensus</p></li><li id="el-li-24" data-ref="ba549f86-4135-4242-93e0-299140c83f24" data-id="el-li-24" data-item-num="3"><p id="el-p-68" data-ref="17c0fa27-6083-4ed3-86bb-aeb45c982df2" data-id="el-p-68">Replace all SNPs with absolute frequencies <em data-ref="6a86a7e4-558d-48f7-ac68-19ea4f19026e">n</em> &lt; <em data-ref="49d72133-1efc-41fa-accb-ec6695c08714">n</em><sub data-ref="596959ec-5dfd-46bc-967b-f3a0c6a6559f"><em data-ref="18a32d90-f88f-4ead-a7d6-020d703d129f">c</em></sub> = 4 with the consensus for their respective columns</p></li></ol></li><li id="el-li-25" data-ref="d9a8a80b-841d-4c69-9b36-6ae9c53d7c18" data-id="el-li-25"><p id="el-p-69" data-ref="5631eb74-a13b-474b-a8ed-1cb36063822e" data-id="el-p-69">For each species <em data-ref="42f8e491-4749-4773-a599-e73c1549d7bb">s</em>, subsample sequences from cells assigned to <em data-ref="46736ce1-0cde-44a7-99c0-4d38d95c5ce0">s</em></p></li><li id="el-li-26" data-ref="074a6fd9-63e8-4f46-bd19-797105747324" data-id="el-li-26"><p id="el-p-70" data-ref="e8868823-4fc1-47ab-847c-38e59cf329fe" data-id="el-p-70">Calculate the linkage matrix <img class="child-of-p insidetext inline-formula max-10 formula-max-0" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F543983v2_inline26c.gif/full/max/0/default.jpg" alt="" id="el-img-7" data-ref="6d29f59e-3a9d-4f0a-81c9-1cf7574933b8" data-id="el-img-7" data-desired-width="0"> for all pairs of SNPs</p></li><li id="el-li-27" data-ref="0c9735f1-97ce-4dd6-8b88-2ea2bc205906" data-id="el-li-27"><p id="el-p-71" data-ref="50d37af0-6abf-448b-b0a5-65c918267af6" data-id="el-p-71">Identify groups of consecutive SNPs with <img class="child-of-p insidetext inline-formula max-10 formula-max-0" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F543983v2_inline26d.gif/full/max/0/default.jpg" alt="" id="el-img-8" data-ref="042d86cb-e5ac-40e3-8827-9619898fa9e6" data-id="el-img-8" data-desired-width="0">, up to a rounding error of <em data-ref="07657945-11ee-45fd-a725-b339611e5f5c">ϵ</em> = 10<sup data-ref="00f1f843-3410-4d01-b78f-0b0e79d67e2a">−5</sup>, containing at least <em data-ref="10980dc5-4076-4056-aa92-94d583ae831f">l</em><sub data-ref="f7be15f9-36e9-47b2-9f30-24553e943234"><em data-ref="10d20b36-93e4-4edc-8ddd-b7c4d9acf316">min</em></sub> = 5 SNPs</p></li></ul><pre data-ref="ded3ce88-af93-4fb5-a0a9-ed231d5e9c3a">for $x in y
dnskdnkfnfn
 nklfnk
    nklfnklf
      nklfnklfnklfnklfnkln-snk snklf msnk s njkf
 dnsklankldfa
ndjdb ekj
</pre></section></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_left_page" data-page-number="20" id="page-20">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-id="article-content" data-split-from="726f25f1-fdd1-41b8-bf49-de736be3e3ae"><section id="s8" data-ref="10c18e3d-cedf-4dae-8ccd-63baf18431ea" data-id="s8"><h1 class="heading-1" id="el-h1-6" data-ref="592b7bbe-ca2c-4187-b9a4-ecb8ce1e86e9" data-id="el-h1-6" data-break-after="avoid">Supplementary tables</h1><div class="fig-group" id="el-div-12" data-ref="3ff23050-d50b-4bd5-8471-09f73402ddb3" data-id="el-div-12" data-previous-break-after="avoid"><p class="figure movedfig fig-square" id="tblS1" data-ref="ffe4f0ba-1e85-491c-ab29-5697fe2b3eba" data-id="tblS1"><img class="child-of-figure imageonly square max-90" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_tblS1.tif/full/max/0/default.jpg" alt="" id="el-img-9" data-ref="61712371-196c-4834-814a-8cd7c6e79dad" data-id="el-img-9" height="1201" width="1977" data-desired-width="93.60795454545455"></p><p class="figure__caption" id="el-p-73" data-ref="143e2839-c34c-4840-a96a-9cf2bbc47921" data-id="el-p-73"><span class="label figure-name" data-ref="6da19412-6e9f-442b-ba60-e7cadd9d6e6d"><label class="figure__label" data-ref="060f070b-5daa-44d0-8967-2fa2d1365afc">Supplementary Table 1.</label></span><strong data-ref="6423eaea-2daa-4458-a7c5-181e6ce1c29c">Cell signaling reagents.</strong> <strong data-ref="96e44c6a-dca8-4dbb-98d8-0a7eb7112563">a</strong>) Stains and quantification for pERK and TBXT in smaller colonies, diameter from left to right: 500um, 350um, 250um. <strong data-ref="a916d59b-d0f5-44a4-8011-6bc2dc719e06">b)</strong> Staining for three primitive streak markers with or with FGFRi or MEKi.</p></div></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_right_page pagedjs_pagedjs-filler_page page-grid" data-page-number="21" id="page-21">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><figure class="figure tofill has-continued imgs-1 shortercaption fig-landscape pagedjs-fill-next-page" id="tblS2" data-continued-main="ctn-tblS2" data-ref="dc731de0-bca6-4d5e-8534-1a536093bdd3" data-id="tblS2" style="display: block; position: unset;"><figcaption class="figure__caption" id="el-figcaption-6" data-ref="e43fcfee-187e-40e5-9395-b20a5fb6512b" data-id="el-figcaption-6"><h3 id="el-h3-6" data-ref="84866565-4edc-4de5-9788-b4de711f3740" data-id="el-h3-6" data-break-after="avoid"><span class="label figure-name" data-ref="533aa9cf-39cf-428e-af3f-7446b800dc66"><label class="figure__label" data-ref="c80e2bd9-c92a-448f-a10d-ae42dae4f06c">Supplementary Table 2.</label></span>Primary antibodies used for immunofluorescence.</h3></figcaption><img class="child-of-figure imageonly landscape max-90" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_tblS2.tif/full/max/0/default.jpg" alt="" id="el-img-10" data-ref="237ffeb6-864d-4750-b234-a0b84e9a4108" data-id="el-img-10" data-previous-break-after="avoid" height="879" width="1977" data-desired-width="93.60795454545455"><img class="child-of-figure imageonly landscape max-90" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_tblS2a.tif/full/max/0/default.jpg" alt="" id="el-img-11" data-ref="34c96e51-080a-4a21-bb03-f31c2283717c" data-id="el-img-11" height="229" width="1977" data-desired-width="93.60795454545455"></figure><figure class="figure tofill imgs-1 shortercaption fig-landscape pagedjs-fill-next-page" id="tblS3" data-ref="23f88b08-4ba7-426f-90d3-e78948723c17" data-id="tblS3" data-following="following-3a4bc264-9217-479e-8230-fe32b422b13b" style="display: block;"><figcaption class="figure__caption" id="el-figcaption-8" data-ref="6c45a311-fee2-4c10-8722-901be0e211ad" data-id="el-figcaption-8"><h3 id="el-h3-7" data-ref="fd9888fe-b7b6-4d5f-b1b6-85964d2b012d" data-id="el-h3-7" data-break-after="avoid"><span class="label figure-name" data-ref="8f8a1746-df6b-4ebb-bcca-e6d7cb219153"><label class="figure__label" data-ref="2b9596f3-e8d4-40fd-b8ab-026005a3861b">Supplementary Table 3.</label></span>Secondary antibodies.</h3></figcaption><img class="child-of-figure imageonly landscape max-90" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_tblS3.tif/full/max/0/default.jpg" alt="" id="el-img-12" data-ref="73802c59-f964-47c3-aaa5-8e48403a7e09" data-id="el-img-12" data-previous-break-after="avoid" height="643" width="1977" data-desired-width="93.60795454545455"></figure><figure class="figure tofill imgs-1 shortercaption fig-landscape pagedjs-fill-next-page" id="tblS4" data-ref="d31ac509-877f-410d-b169-7eec99ee7d96" data-id="tblS4" data-following="following-3a4bc264-9217-479e-8230-fe32b422b13b" style="display: block;"><figcaption class="figure__caption" id="el-figcaption-9" data-ref="cd8ccfb7-b6fa-443d-8505-96ac80d79bfe" data-id="el-figcaption-9"><h3 id="el-h3-8" data-ref="82e1c0c0-8439-4465-bae6-9790aacc116b" data-id="el-h3-8" data-break-after="avoid"><span class="label figure-name" data-ref="25685ffa-17ab-4c0f-85c4-d9320cc9f9fd"><label class="figure__label" data-ref="cb3e0b4f-392d-4fc7-a416-c31fc5b578d5">Supplementary Table 4.</label></span>Marker genes for cell fate annotation</h3></figcaption><img class="child-of-figure imageonly landscape max-90" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_tblS4.tif/full/max/0/default.jpg" alt="" id="el-img-13" data-ref="0f7c9dcd-169f-4d89-9fd0-318e29a0eb05" data-id="el-img-13" data-previous-break-after="avoid" height="854" width="1977" data-desired-width="93.60795454545455"></figure></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_left_page pagedjs_pagedjs-filler_page page-grid" data-page-number="22" id="page-22">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><figure class="figure tofill imgs-1 shortercaption fig-landscape pagedjs-fill-next-page" id="tblS5" data-ref="8b126539-c6df-45fe-9328-ab39855ada74" data-id="tblS5" data-following="following-3a4bc264-9217-479e-8230-fe32b422b13b" style="display: block;"><figcaption class="figure__caption" id="el-figcaption-10" data-ref="0294e6fa-f054-49a9-b3ca-2f6bf02e4129" data-id="el-figcaption-10"><h3 id="el-h3-9" data-ref="7668bfc2-119d-44a6-a022-b67d9e52f0d4" data-id="el-h3-9" data-break-after="avoid"><span class="label figure-name" data-ref="00dcd388-7c93-40ba-bfec-13c67fbe93f7"><label class="figure__label" data-ref="2d90edfe-c16b-419a-b4b3-fde2e34569f6">Supplementary Table 5.</label></span>qPCR primers</h3></figcaption><img class="child-of-figure imageonly landscape max-90" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_tblS5.tif/full/max/0/default.jpg" alt="" id="el-img-14" data-ref="2eacd477-79bd-4831-9a40-7a33906652f3" data-id="el-img-14" data-previous-break-after="avoid" height="434" width="1977" data-desired-width="93.60795454545455"></figure><figure class="figure tofill imgs-1 shortercaption fig-landscape pagedjs-fill-next-page" id="tblS6" data-ref="31ea43c3-94de-405e-8477-63f2e04d0d15" data-id="tblS6" data-next-break-before="page" data-following="following-3a4bc264-9217-479e-8230-fe32b422b13b" style="display: block;"><figcaption class="figure__caption" id="el-figcaption-11" data-ref="a156aaf4-2337-4f15-b1d2-e0beee0c1aaf" data-id="el-figcaption-11"><h3 id="el-h3-10" data-ref="9fef3ef9-7aab-4f75-acb6-a72c6091e997" data-id="el-h3-10" data-break-after="avoid"><span class="label figure-name" data-ref="b68877ac-1964-43e4-95e7-4601c23d818d"><label class="figure__label" data-ref="3a54d7b7-08c3-4f94-a19b-ed292dfe1e92">Supplementary Table 6.</label></span>Probes for Fluorescent In Situ Hybridization</h3></figcaption><img class="child-of-figure imageonly landscape max-90" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_tblS6.tif/full/max/0/default.jpg" alt="" id="el-img-15" data-ref="66d6754b-f4c1-4af5-b45a-d87fb38cd5e1" data-id="el-img-15" data-previous-break-after="avoid" height="288" width="1977" data-desired-width="93.60795454545455"></figure></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_right_page" data-page-number="23" id="page-23" data-break-before="page">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-id="article-content" data-split-from="726f25f1-fdd1-41b8-bf49-de736be3e3ae"><section data-ref="10c18e3d-cedf-4dae-8ccd-63baf18431ea" data-id="s8" data-split-from="10c18e3d-cedf-4dae-8ccd-63baf18431ea"><div class="pagebreak" id="el-div-13" data-ref="6041e681-64fd-4b5f-977a-b2fad5da7042" data-id="el-div-13" data-break-before="page"></div></section><section id="s9" data-ref="4f67efb4-0c23-4f85-8633-c11ba90c3901" data-id="s9"><h1 class="heading-1" id="el-h1-7" data-ref="ae7e5410-2092-4922-bd70-5348b97e2d84" data-id="el-h1-7" data-break-after="avoid">Supplementary Figures</h1><div class="fig-group" id="el-div-14" data-ref="cbb4a61f-7117-4782-91fb-10bb576cfd97" data-id="el-div-14" data-previous-break-after="avoid"><p class="figure movedfig fig-portrait" id="figS1" data-ref="4effe846-3120-4498-a4d4-fa93f3356c64" data-id="figS1"><img class="child-of-figure has-boundary imageonly portrait max-100" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_figS1.tif/full/max/0/default.jpg" alt="" id="el-img-16" data-ref="8efee1c4-7ed6-4899-818a-616a6092ac1a" data-id="el-img-16" height="2390" width="2082" data-desired-width="98.57954545454545"></p><p class="figure__caption" id="el-p-76" data-ref="5955e46c-57ca-4675-b1e5-88e05910f5cb" data-id="el-p-76"><span class="label figure-name" data-ref="7e92aca6-52ac-44ff-a6cb-b1d0ceb8d033"><label class="figure__label" data-ref="4a9363d6-2426-45b0-81b2-6a20b437b94f">Supplementary Figure 1.</label></span><strong data-ref="6a461ecc-b04f-4c6e-9fa9-22150d317456">a</strong>) Stains and quantification for pERK and TBXT in smaller colonies, diameter from left to right: 500um, 350um, 250um. <strong data-ref="829ad814-9a0b-459d-b1a5-5719c62afdb3">b)</strong> Staining for three primitive streak markers with or with FGFRi or MEKi. <strong data-ref="de9e7c9f-bbc9-441f-8f86-d76a4643806e">c)</strong> DAPI stain corresponding to <span class="fakelink linktofig" data-ref="90a78405-4734-4f1c-94a1-838b350f27eb"><span class="fakelink-text" data-ref="43d35f1e-61fb-4edf-aef6-e4897cf2b076">Fig. 1d</span><a class="fake-icon" href="#fig1" data-ref="f70e7f9c-2b1b-43bb-ad76-d2ea1b748ed3"><span class="fakelink-icon" data-ref="90a3f6ca-e7ac-455d-b2ac-d0881ce5b1f2">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. <strong data-ref="a905ceef-ac08-4318-a6d3-5f79cd106c56">d)</strong> Radial intensity profile for TBXT corresponding to <span class="fakelink linktofig" data-ref="fd2bc05d-a858-404a-bb32-1620ca92d9b7"><span class="fakelink-text" data-ref="d38962c1-98d2-4263-94a6-cf14c74db500">Fig. 1d</span><a class="fake-icon" href="#fig1" data-ref="b93e98a2-2eb6-462f-9461-3b92bfe6186b"><span class="fakelink-icon" data-ref="b1d7c57f-386a-427d-bf60-1b284fe7da1b">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. <strong data-ref="b9ba7609-ae47-4d4f-9559-1bb93c7ef153">e</strong>) BrdU and cleaved Cas3 stainings after MEK or FGFR inhibition. <strong data-ref="64313dad-4774-437e-8685-0d808bc99fcd">f)</strong> Quantification of conditions in e. <strong data-ref="5ef244e1-ac79-4fb9-9ddf-74e2e7640de7">g)</strong> Quantification of conditions in <span class="fakelink linktofig" data-ref="36cf3660-4816-4cc1-abb5-192465bf2bbc"><span class="fakelink-text" data-ref="45c22ba4-1531-4f96-847f-c913911c3f10">Fig. 1g</span><a class="fake-icon" href="#fig1" data-ref="73fcc984-cf56-477e-8a2e-4b8db20fdcaf"><span class="fakelink-icon" data-ref="869a9c52-1751-4ac0-b9c5-3b850cf682f2">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span> for five images each, N indicates total number of cells. <strong data-ref="fab3b8fb-8201-47af-8d0c-3f2ec82ab51a">h</strong>) Low density differentiation with and without MEK and FGFR inhibition. <strong data-ref="ba1a28fb-a4ad-491a-a3e0-58ab96a5ed68">i)</strong> pERK and TBXT stains and quantification for different doses of MEK inhibitor shows low doses effectively block differentiation with minimal impact on cell number.</p></div></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_left_page pagedjs_pagedjs-filler_page page-grid" data-page-number="24" id="page-24">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><figure class="figure tofill imgs-1 longcaption fig-portrait pagedjs-fill-next-page finishes" id="figS2" data-ref="80dcc250-1945-4d0c-a883-a6436eb63ed3" data-id="figS2" style="display: block; position: unset;"><figcaption class="figure__caption" id="el-figcaption-12" data-ref="6da9d486-115c-400c-8da3-629bfff45923" data-id="el-figcaption-12"><h3 id="el-h3-11" data-ref="b8d11858-4263-4419-8077-32b007f32087" data-id="el-h3-11" data-break-after="avoid"><span class="label figure-name" data-ref="1454869f-906f-46c8-ba15-bab6e0694358"><label class="figure__label" data-ref="15641529-2a79-4a51-bfd0-686ee5c97519">Supplementary Figure 2.</label></span></h3><p id="el-p-77" data-ref="c4f71d91-918a-47f4-bb49-2dc20a8c16c9" data-id="el-p-77" data-previous-break-after="avoid"><strong data-ref="8fc267ce-228d-4e64-a0a6-7795ebd4dc75">a)</strong> PHATE projection of scRNA-seq data colored for sample. <strong data-ref="e344831a-c7df-424d-b30e-09355e62db13">b,c)</strong> Expression of canonical FGF ligands across different cell types in 2D gastruloids (b) and human CS7 embryo (c). Expression across clusters is normalized by row, while total expression is shown in a separate column to the directly to left. The leftmost column indicates differential expression between PS-like and pluripotent cells. The color scale of (differential) expression is cut off for better contrast. <strong data-ref="1107a637-d6df-4d82-8601-c79fb6675057">d)</strong> Threshold-based annotation (left) for human, monkey, and mouse embryos used in interspecies comparison (<span class="fakelink linktofig" data-ref="3e4311cc-aec3-42c1-a105-56260b697571"><span class="fakelink-text" data-ref="c61a06a0-226d-4922-82b6-28a1568de00e">Fig. 2f</span><a class="fake-icon" href="#fig2" data-ref="2727c34e-5105-4d62-9b20-040dce81bf0a"><span class="fakelink-icon" data-ref="7931918f-e42c-4026-a59c-c064b4e2165f">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>), compared to original annotation (right). <strong data-ref="fedb8e83-eab5-4a70-a85e-a8b3f33cb817">e)</strong> Interspecies comparison as in <span class="fakelink linktofig" data-ref="cf2c217f-21b4-4b30-8708-8455bc09c0b7"><span class="fakelink-text" data-ref="161fc497-156e-49d0-a21f-f30c1e56ed5c">Fig. 2f</span><a class="fake-icon" href="#fig2" data-ref="3b107273-af1c-4cba-8aa1-6a51e670fb9a"><span class="fakelink-icon" data-ref="70556db7-37b2-495e-baca-afe969b8d3bf">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span> but for nascent mesoderm and PS-LC + nascent mesoderm. <strong data-ref="08487d79-5af5-46ee-a637-3b6217578fbf">f)</strong> Bulk RNA-seq data by Loh et al<sup data-ref="7ba03a10-e289-4944-9257-f046fb964bb0"><a href="#c38" data-ref="0e5e1365-e0f9-4dfa-a8f2-bf43bd347915">38</a></sup> for expression of different FGFs in directed differentiation to mid primitive streak (MPS) or anterior primitive streak (APS) relative the human embryonic stem cells (hESCs). <strong data-ref="0085c411-abe1-4a84-b33e-dfd9cdbef33d">g)</strong> Expression of genes involved in FGF signaling.</p></figcaption><img class="child-of-figure has-boundary imageonly portrait max-100" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_figS2.tif/full/max/0/default.jpg" alt="" id="el-img-17" data-ref="75b87741-b8df-4cab-9948-916c17db0f53" data-id="el-img-17" height="2357" width="2026" data-desired-width="95.9280303030303"></figure></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_right_page pagedjs_pagedjs-filler_page page-grid" data-page-number="25" id="page-25">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><figure class="figure tofill imgs-1 shortercaption fig-square pagedjs-fill-next-page" id="figS3" data-ref="dd27229a-6db1-47f7-990e-ca5781051db9" data-id="figS3" data-following="following-3a4bc264-9217-479e-8230-fe32b422b13b" style="display: block;"><figcaption class="figure__caption" id="el-figcaption-13" data-ref="17373d6c-4300-4e40-9d6f-5d1e76ae0a2f" data-id="el-figcaption-13"><h3 id="el-h3-12" data-ref="1a5ce017-ccf9-4c0e-bd90-3f5f8aede3ec" data-id="el-h3-12" data-break-after="avoid"><span class="label figure-name" data-ref="cb9ffe82-9bf4-4da1-86ae-9e74e79ebb27"><label class="figure__label" data-ref="42ca2310-f229-4feb-86e9-fb5e6581c09b">Supplementary Figure 3.</label></span></h3><p id="el-p-78" data-ref="33ef876e-5756-4516-8bab-f36ee8371ef5" data-id="el-p-78" data-previous-break-after="avoid"><strong data-ref="f32341db-2e8b-4ed5-ad50-b41b6c506a9d">a)</strong> DAPI image corresponding to <span class="fakelink linktofig" data-ref="0a1e6920-398e-4b04-a23f-cd611f5457ac"><span class="fakelink-text" data-ref="25c26521-905c-4210-86a3-27fa95e71306">Fig. 3a</span><a class="fake-icon" href="#fig3" data-ref="eebee298-a7a4-4019-bff3-6684c7577c13"><span class="fakelink-icon" data-ref="cacb6265-f61e-4c13-b766-43459cc38171">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. <strong data-ref="bb9a9aa4-40cb-4edd-be72-e890930cd36b">b-c)</strong> pFGFR1 (b) or pERK (c) and TBXT stains with and without continuous MEKi treatment. <strong data-ref="4376ea97-f8f7-4aff-a047-692a3eb0669a">d)</strong> DAPI image corresponding to <span class="fakelink linktofig" data-ref="d057049b-55e7-4d96-a918-ce3d00cfcccb"><span class="fakelink-text" data-ref="ca04992a-c41f-4af0-a885-1dc5905584bb">Fig. 3b</span><a class="fake-icon" href="#fig3" data-ref="bbebb22b-deb3-4ad7-8f6c-62651449a513"><span class="fakelink-icon" data-ref="7cd35db8-89fd-407d-aa2b-d68078f61a87">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. <strong data-ref="bbaf5d97-1d94-451a-adb4-3139e01fdff7">e)</strong> Design of qPCR primers to detect FGFR1 isoforms, diagram adapted from Eswarakumar et al<sup data-ref="a4581912-1200-49c4-97da-f334c48623aa"><a href="#c72" data-ref="6da93a0c-d1a4-46c2-86ea-c37a1efc0f1d">72</a></sup>. <strong data-ref="6c281367-7ea7-4682-b517-ecbaa0fcebbb">f)</strong> Primer sequence and properties. <strong data-ref="64b88dda-c53f-4de3-a8dc-296b00dd13c1">g)</strong> Amplicon sizes match prediction. Scale bars 50um.</p></figcaption><img class="child-of-figure has-boundary imageonly square max-100" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_figS3.tif/full/max/0/default.jpg" alt="" id="el-img-18" data-ref="8f974b59-0136-4f25-8b66-9cfe11960524" data-id="el-img-18" height="1706" width="2103" data-desired-width="99.57386363636364"></figure><figure class="figure tofill imgs-1 shortercaption fig-landscape pagedjs-fill-next-page" id="figS4" data-ref="805e78b8-f6b9-4003-b5e6-38d139a66c59" data-id="figS4" data-following="following-3a4bc264-9217-479e-8230-fe32b422b13b" style="display: block;"><figcaption class="figure__caption" id="el-figcaption-14" data-ref="819dbce5-9796-4f99-9a2a-98be58c49a47" data-id="el-figcaption-14"><h3 id="el-h3-13" data-ref="1fb529ce-255d-4135-8a72-581758ffb871" data-id="el-h3-13" data-break-after="avoid"><span class="label figure-name" data-ref="ee7eefa0-5855-4e5c-a533-f60119b74ea1"><label class="figure__label" data-ref="af3f4075-bf90-441a-bad0-9ca742ddf6f8">Supplementary Figure 4.</label></span></h3><p id="el-p-79" data-ref="fe61e90d-20df-4e1a-ad9b-b1090309ee25" data-id="el-p-79" data-previous-break-after="avoid"><strong data-ref="561be02c-87f0-4157-b779-f62779701171">a,b)</strong> Heparan sulfate (HS) stain (a) and radial intensity profile (b). Scale bar 50um. Error bands represent standard deviation over N=4 colonies.</p></figcaption><img class="child-of-figure has-boundary imageonly landscape max-90" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_figS4.tif/full/max/0/default.jpg" alt="" id="el-img-19" data-ref="90d336cf-1438-4b68-a404-98d1a9a3cfd5" data-id="el-img-19" height="413" width="1872" data-desired-width="88.63636363636364"></figure></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_left_page pagedjs_pagedjs-filler_page page-grid" data-page-number="26" id="page-26">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><figure class="figure tofill imgs-1 shortercaption fig-square pagedjs-fill-next-page finishes" id="figS5" data-ref="85118bbe-b148-4932-847c-06c99462d9f3" data-id="figS5" data-following="following-3a4bc264-9217-479e-8230-fe32b422b13b" style="display: block;"><figcaption class="figure__caption" id="el-figcaption-15" data-ref="0a281842-892d-4229-a4d4-7051f5e5fd29" data-id="el-figcaption-15"><h3 id="el-h3-14" data-ref="cb1b7e89-8ce3-4bca-bfea-830fd0f7aeb5" data-id="el-h3-14" data-break-after="avoid"><span class="label figure-name" data-ref="7ab79152-43b7-42a4-b9b4-eb90e95f9519"><label class="figure__label" data-ref="9781a2f7-be4e-4904-9861-cc6d853b4251">Supplementary Figure 5.</label></span></h3><p id="el-p-80" data-ref="c1dae0bc-c6ee-43ea-b794-3d1006352370" data-id="el-p-80" data-previous-break-after="avoid"><strong data-ref="b672ca16-3a62-48ac-a0f6-380e437a13b6">a)</strong> DAPI image corresponding to <span class="fakelink linktofig" data-ref="8d30fd8f-3273-4154-b71b-8705011aab6d"><span class="fakelink-text" data-ref="227662bf-1278-47c0-bba5-8b6ea720d07a">Fig. 5a</span><a class="fake-icon" href="#fig5" data-ref="8ed358b6-a9d2-4006-b965-0a45d5a402c1"><span class="fakelink-icon" data-ref="3ce25dc9-9f12-4c40-9734-f591df76a48e">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. <strong data-ref="ccb34b6c-d91a-4c11-a7b9-00f8fc61c8b1">b)</strong> Pseudotime dynamics from <span class="fakelink linktofig" data-ref="81fec67a-04a4-4e9d-ba6c-b277c6e7abad"><span class="fakelink-text" data-ref="1d785b2c-bdea-4416-a4e2-832073e275e2">Fig. 5b</span><a class="fake-icon" href="#fig5" data-ref="f9f3a81d-2d6c-4cde-91b9-2d5f611c59e3"><span class="fakelink-icon" data-ref="7c569a95-9652-4683-8bbd-e40542d3e810">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span> with each gene normalized individually. <strong data-ref="6d982933-290a-447e-b721-9b73f1f09e70">c)</strong> Correlations between FGFs and TBXT, TBX6 colored for clusters, with cluster colors matching <span class="fakelink linktofig" data-ref="fb8d01aa-c9ba-4520-98a8-172bdb3339d7"><span class="fakelink-text" data-ref="5af03978-a83b-47f6-b49c-6370e87d4b16">Fig. 2d</span><a class="fake-icon" href="#fig2" data-ref="e38f9c39-b52a-429c-9158-1432f18ba91f"><span class="fakelink-icon" data-ref="12316162-f009-465c-a4d7-0bbfff25b865">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. <strong data-ref="084b4364-0b00-4095-828b-0215351c5343">d-f)</strong> DAPI images and overlays including DAPI corresponding to <span class="fakelink linktofig" data-ref="04079d2b-870a-42ee-ba81-349cbe316122"><span class="fakelink-text" data-ref="619d5585-35ab-45fb-869e-e9dcb1b75bd3">Fig. 5c-e</span><a class="fake-icon" href="#fig5" data-ref="2b5d3db3-1791-407d-a784-73ab04fb31bd"><span class="fakelink-icon" data-ref="47f9e537-bd6e-4ad4-963e-ef5bd4b56e02">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>, respectively. <strong data-ref="fdf134fa-d4ca-4c42-acd1-ad5a64d35637">g)</strong> Combined FISH for FGF4 and FGF8. <strong data-ref="cef6d464-d191-4290-a839-d590e110f8d9">h)</strong> Scatterplots of FGF4, FGF8, and FGF17 mRNA versus TBXT colored for cell density.</p></figcaption><img class="child-of-figure has-boundary imageonly square max-100" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_figS5.tif/full/max/0/default.jpg" alt="" id="el-img-20" data-ref="a59c8fab-6081-4447-bae7-cc47a0c8eed4" data-id="el-img-20" height="2023" width="2103" data-desired-width="99.57386363636364"></figure></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_right_page pagedjs_pagedjs-filler_page page-grid" data-page-number="27" id="page-27">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><figure class="figure tofill imgs-1 shortercaption fig-portrait pagedjs-fill-next-page finishes" id="figS6" data-ref="6fd30c2f-85d8-4123-a99a-57279450f764" data-id="figS6" data-next-break-before="page" data-following="following-3a4bc264-9217-479e-8230-fe32b422b13b" style="display: block;"><figcaption class="figure__caption" id="el-figcaption-16" data-ref="d9427009-dd2b-442e-9f87-22c05494aa91" data-id="el-figcaption-16"><h3 id="el-h3-15" data-ref="7dc0379f-ffe5-4df7-8fa5-980cf7387f68" data-id="el-h3-15" data-break-after="avoid"><span class="label figure-name" data-ref="3ccec7e5-f1c7-4f33-b04f-0dddc129a4f9"><label class="figure__label" data-ref="462bc3d1-c9af-4e6a-acf7-5d9c8760c610">Supplementary Figure 6.</label></span></h3><p id="el-p-81" data-ref="4d1b8f2a-fe63-430f-bd92-7d82ca8223e6" data-id="el-p-81" data-previous-break-after="avoid"><strong data-ref="1436dcb7-5986-4081-bfda-36725da0707e">a)</strong> Additional channels for the colony shown in <span class="fakelink linktofig" data-ref="2ea67597-2b42-4de9-aa7f-c5889337bdf4"><span class="fakelink-text" data-ref="91be5ac9-ed83-470a-8de2-e0f1a2e044cd">Fig. 5g</span><a class="fake-icon" href="#fig5" data-ref="9394d97b-1ccc-48ab-8027-112fbf4dd8ce"><span class="fakelink-icon" data-ref="2b434111-83a8-424c-9776-813bb6678d46">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. <strong data-ref="aab940d0-9883-4893-ad96-f48f38f6029d">b)</strong> FGF4 shRNA in ESI17 cells. <strong data-ref="b62b13d3-50cf-44a0-8299-7d9da8fe753f">c-d)</strong> FGF4 siRNA (c) and shRNA (d) in PGP1 cells. <strong data-ref="faf5ac10-92eb-4106-8c5e-21485be9a560">e-f)</strong> additional channels for <span class="fakelink linktofig" data-ref="ca2cb263-2dc3-4619-b64b-9cd8e42e657b"><span class="fakelink-text" data-ref="eaba70a7-5fd1-44d4-9ac9-9c8f38aa5e93">Fig. 5h-i</span><a class="fake-icon" href="#fig5" data-ref="85689851-7dc5-4465-b5e9-f16f51268da9"><span class="fakelink-icon" data-ref="dacf4e41-4e8e-4d93-9c6b-2d3f2ab55566">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. <strong data-ref="c13f64a5-e178-43e6-9a19-d7840632e19e">g</strong>) FGF17 shRNA in ESI17 cells. <strong data-ref="e07e1d45-6263-4948-a8b0-5dea3f588a6e">h-i)</strong> FGF17 siRNA (h) and shRNA (i) in PGP1 cells. <strong data-ref="086d162e-da1b-41ef-91de-704e15996cea">j)</strong> additional channels for the TBX6 ctrl and FGF17 shRNA in <span class="fakelink linktofig" data-ref="2c67f35b-aab8-45e5-8877-e71c3ffdf96a"><span class="fakelink-text" data-ref="6394d4d4-1bc4-4e5b-8d89-97eb43ecc2ff">Fig. 5j</span><a class="fake-icon" href="#fig5" data-ref="70dcbc07-ec10-47eb-b29c-76a316d7f599"><span class="fakelink-icon" data-ref="c264a217-d0e9-4ad4-8a45-22307e4ddef4">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span>. <strong data-ref="3b95b603-56ae-4bc7-b646-5eb8883ab447">k)</strong> FGF4 + FG17 shRNA double knockdown (stains from multiple colonies).</p></figcaption><img class="child-of-figure has-boundary imageonly portrait max-100" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_figS6.tif/full/max/0/default.jpg" alt="" id="el-img-21" data-ref="031583a7-93d1-4bcd-8961-9d484d2bc6e3" data-id="el-img-21" height="2592" width="2103" data-desired-width="99.57386363636364"></figure></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_left_page" data-page-number="28" id="page-28" data-break-before="page">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e" data-split-to="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-id="article-content" data-split-from="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-split-to="726f25f1-fdd1-41b8-bf49-de736be3e3ae"><section data-ref="4f67efb4-0c23-4f85-8633-c11ba90c3901" data-id="s9" data-split-from="4f67efb4-0c23-4f85-8633-c11ba90c3901"><div class="pagebreak" id="el-div-15" data-ref="b8e11b54-8ad6-416f-83f2-d010d174a5b6" data-id="el-div-15" data-break-before="page"></div></section><section id="acknowledgements" data-ref="b8b95a3c-1f5f-4725-b84f-0569e967272e" data-id="acknowledgements"><h1 class="heading-1" id="el-h1-8" data-ref="7bb5e5ce-d8fc-4f58-9fe6-89ca510d56ad" data-id="el-h1-8" data-break-after="avoid">Acknowledgements</h1><p id="el-p-82" data-ref="1cd211c4-9b7c-450c-8af0-309e84ff1731" data-id="el-p-82" data-previous-break-after="avoid">We thank Lila Solnica-Krezel, Blerta Stringa, Aryeh Warmflash, and Ben Allen for discussions. We thank Jared Toettcher for the dox-SOScat plasmid. Library prep and next-generation sequencing was carried out in the Advanced Genomics Core at the University of Michigan. This work was supported by the National Institute of General Medical Sciences (NIGMS R35GM138346), and the Branco Weiss Fellowship – Society in Science. KJ was partially supported by the Michigan Pioneer Postdoctoral Fellowship and the NIH F32 Ruth L. Kirschstein Postdoctoral National Research Service Award (5F32HD108980-02).</p></section><section id="additional-info" data-ref="ee68b80d-31d4-4ffa-abc6-803efa058c7f" data-id="additional-info"><h1 class="heading-1" id="el-h1-9" data-ref="12582c6c-50a6-470d-9cf7-57bbcdfe34e8" data-id="el-h1-9" data-break-after="avoid">Additional information</h1><section id="das" data-ref="48bf1056-a4e3-4b92-89fb-d0847c0ad9be" data-id="das" data-previous-break-after="avoid"><h2 class="heading-2" id="el-h2-21" data-ref="40d38778-870d-4b4e-85e3-a8dafe31b8be" data-id="el-h2-21" data-break-after="avoid" data-following="following-ff2c00ba-51a1-482d-9afa-f5a370dcfa50">Data availability</h2><p id="el-p-83" data-ref="f0bbf8b8-065a-4b4b-858d-72c69c443ff4" data-id="el-p-83" data-previous-break-after="avoid">Raw single cell RNA sequencing data generated in this study was deposited in GEO (GSE271604). Publicly available datasets used in this study can be accessed from GEO (GSE193007, GSE85066) and ArrayExpress (E-MTAB-6967, E-MTAB-9388). Raw image data are available upon request.</p></section><section id="s5" data-ref="c7348dab-bf46-4dcb-ba18-d621ad9b729b" data-id="s5"><h2 class="heading-2" id="el-h2-22" data-ref="87e19654-e373-4de1-a8d2-daa2b7f1dbc1" data-id="el-h2-22" data-break-after="avoid">Code availability</h2><p id="el-p-84" data-ref="e2f19cf7-2ad8-4f6f-95e6-5465a6116a54" data-id="el-p-84" data-previous-break-after="avoid">All code for data analysis and model simulations is available on <span class="fakelink linktoext" data-ref="b62827fb-6b54-42b1-8b49-358d322c2302"><span class="fakelink-text" data-ref="8de40699-5ce6-4b40-a214-00d3b2519166">https://github.com/idse/FGF</span><a class="fake-icon" href="https://github.com/idse/FGF" data-ref="71f99f19-14a4-491b-982d-59e934103ead"><span class="fakelink-icon" data-ref="18faa64b-3c8e-47a8-8647-58a49dd5fedd">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></a></span></p></section><section id="funding" data-ref="612062ca-3802-4b2c-aa17-83eb6ba78246" data-id="funding"><h2 id="el-h2-23" data-ref="5683cbef-189c-4539-ac9b-782a6d38fa38" data-id="el-h2-23" data-break-after="avoid">Funding</h2><ul class="funding-list list-simple" id="el-ul-4" data-ref="6b164475-e553-4350-bd76-f832054694c4" data-id="el-ul-4" data-previous-break-after="avoid"><li class="funding-list-item" id="el-li-28" data-ref="6d82f4e0-c715-44d6-9641-fb67e23a8968" data-id="el-li-28"><p id="el-p-85" data-ref="5a29d322-682c-4860-9c46-b19784a1bd0d" data-id="el-p-85">National Institute of General Medical Sciences</p></li><li class="funding-list-item" id="el-li-29" data-ref="53a3853f-cb12-41a7-9e62-ebe65f124ac4" data-id="el-li-29"><p id="el-p-86" data-ref="969884bb-4e8b-4255-b87a-235110881e9d" data-id="el-p-86">the Branco Weiss Fellowship</p></li><li class="funding-list-item" id="el-li-30" data-ref="2d15795e-90d7-401b-b0be-6360d190888e" data-id="el-li-30"><p id="el-p-87" data-ref="5d40727b-807d-45ff-816a-6493df47309d" data-id="el-p-87">Michigan Pioneer Postdoctoral Fellowship</p></li><li class="funding-list-item" id="el-li-31" data-ref="7de9f23e-0350-4fb0-af75-091f68aa739d" data-id="el-li-31"><p id="el-p-88" data-ref="0f873584-bb2e-4336-af16-bcc33898d013" data-id="el-p-88">NIH F32 Ruth L. Kirschstein Postdoctoral National Research Service Award</p></li></ul></section><section id="orcids" data-ref="37ecf16f-8771-4e30-9dae-603bc6dbc8df" data-id="orcids"><h2 class="orcid-list__title" id="el-h2-24" data-ref="4e88da33-53be-4933-b741-9a09bd8b714f" data-id="el-h2-24" data-break-after="avoid">Author ORCID iDs</h2><ul class="orcid-list list-simple" id="el-ul-5" data-ref="84e8bd96-3341-4d7d-a99b-3d54b8a4864c" data-id="el-ul-5" data-previous-break-after="avoid"><li class="orcid-list-item" id="el-li-32" data-ref="1e9deba0-b230-4b55-b8b9-2af761dbfa81" data-id="el-li-32"><p id="el-p-89" data-ref="aa7a7a72-e2cb-445f-b6a8-b3a632faef74" data-id="el-p-89"><strong data-ref="1c6bf09a-f8ed-4919-aa63-99e3e39a8a2b">Jason Spence</strong>: <a href="http://orcid.org/0000-0001-7869-3992" data-ref="0029e75e-ca1e-4bc6-806d-2d9eae9a82b6">http://orcid.org/0000-0001-7869-3992</a></p></li></ul></section></section><section id="references" data-ref="7921c412-ef92-4490-b273-a1fbe0815a2b" data-id="references" data-split-to="7921c412-ef92-4490-b273-a1fbe0815a2b" data-split-original="true"><h1 class="heading-1" id="el-h1-10" data-ref="63cb8820-844f-497b-82ef-b2051268d3ca" data-id="el-h1-10" data-break-after="avoid">References</h1><ul class="reference-list" id="el-ul-6" data-ref="5d1d46dc-2209-4116-9f07-193d224fefb4" data-id="el-ul-6" data-previous-break-after="avoid" data-split-to="5d1d46dc-2209-4116-9f07-193d224fefb4" data-split-original="true" data-last-split-element="true" data-align-last-split-element="auto"><li class="reference-list__item" id="c1" data-ref="7e4c950c-7904-4995-a2a8-7b5cdcd462c6" data-id="c1"><span class="reference__label" data-ref="7cd461f8-eb6e-458d-a106-b94ba048e313">1.</span><ol class="reference__authors_list" id="el-ol-3" data-ref="e715c6f3-2879-4e8d-a48d-eccfb16af2b8" data-id="el-ol-3" start="1"><li class="reference__author" id="el-li-34" data-ref="6e12b676-0ccb-450e-a431-5b2f238c5e49" data-id="el-li-34" data-item-num="1">Arnold S. J.</li><li class="reference__author" id="el-li-35" data-ref="b3da5191-a45d-4639-bf9c-7313cce17c4a" data-id="el-li-35" data-item-num="2">Robertson E. J</li></ol><span class="reference__authors_list_suffix" data-ref="989c011a-1765-4357-9367-49a92b92f15c">2009</span><span class="reference__title" data-ref="0900321d-92af-4913-ac78-d9565916788c">Making a commitment: cell lineage allocation and axis patterning in the early mouse embryo.</span><span class="reference__origin" data-ref="96c04b46-9e20-415d-b767-08a5042eb443"><i data-ref="e83fff9b-7aa4-4d10-bf8e-3cf3f279fd6c">Nat. Rev. Mol. Cell Biol</i> <strong data-ref="93ff7dce-810c-4030-b99e-61313b7867cf">10</strong>:91‑103</span><span class="reference__doi" data-ref="bbfff459-891c-47a8-ae0d-ef0d31c1aefc"></span></li><li class="reference-list__item" id="c2" data-ref="0c134f85-c115-4dab-916e-0e2d7e6e819d" data-id="c2"><span class="reference__label" data-ref="af9a7e61-b01a-42d0-8c79-292c20f51202">2.</span><ol class="reference__authors_list" id="el-ol-4" data-ref="fd1ce35f-fdcf-4e68-98ed-c36ae2e1149a" data-id="el-ol-4" start="1"><li class="reference__author" id="el-li-37" data-ref="cd3e1159-dd40-4487-9869-13ebb62a28d1" data-id="el-li-37" data-item-num="1">Martyn I.</li><li class="reference__author" id="el-li-38" data-ref="888e7172-9903-4288-a3e8-561f3bdbdc14" data-id="el-li-38" data-item-num="2">Kanno T. Y.</li><li class="reference__author" id="el-li-39" data-ref="033da909-3ecf-4096-bef4-630ddb57ac97" data-id="el-li-39" data-item-num="3">Ruzo A.</li><li class="reference__author" id="el-li-40" data-ref="06114454-bac3-4f53-b5af-900ae13454c3" data-id="el-li-40" data-item-num="4">Siggia E. D.</li><li class="reference__author" id="el-li-41" data-ref="a2c19721-11ff-4f32-b4b1-2891a77b5eb2" data-id="el-li-41" data-item-num="5">Brivanlou A. H</li></ol><span class="reference__authors_list_suffix" data-ref="d162dc71-f218-4818-a31e-5af5bf02a7d0">2018</span><span class="reference__title" data-ref="662c9e3f-e07c-4e36-92c0-2287ccda5b8b">Self-organization of a human organizer by combined Wnt and Nodal signalling.</span><span class="reference__origin" data-ref="1d66e2f8-4209-43bc-a831-b73254c0fb35"><i data-ref="42eec796-c959-4bf6-bc0e-377e7dd2419f">Nature</i> </span><span class="reference__doi" data-ref="cf22f177-bf60-48ef-9753-cd2b67d28348"><a href="https://doi.org/10.1038/s41586-018-0150-y" class="reference__doi_link" data-ref="cef7512f-4130-49b4-913a-32cf47087070">https://doi.org/10.1038/s41586-018-0150-y</a></span></li><li class="reference-list__item" id="c3" data-ref="ebf5d9f7-a019-4537-b026-4a8b5922c6b4" data-id="c3"><span class="reference__label" data-ref="018bfff3-0460-416c-bfce-fe6f36ad309c">3.</span><ol class="reference__authors_list" id="el-ol-5" data-ref="921e5cbe-b692-49d9-a112-509e531776ed" data-id="el-ol-5" start="1"><li class="reference__author" id="el-li-43" data-ref="113f81c1-ea48-4484-bb38-e941a96de7a6" data-id="el-li-43" data-item-num="1">Chhabra S.</li><li class="reference__author" id="el-li-44" data-ref="9edeb0d8-b8fa-4798-89b5-7909aca046da" data-id="el-li-44" data-item-num="2">Liu L.</li><li class="reference__author" id="el-li-45" data-ref="b15a0d7d-55ee-4587-8596-1e2aa237109c" data-id="el-li-45" data-item-num="3">Goh R.</li><li class="reference__author" id="el-li-46" data-ref="5682a207-7119-4572-9e5d-374d913b8320" data-id="el-li-46" data-item-num="4">Kong X.</li><li class="reference__author" id="el-li-47" data-ref="b5a9750e-b987-4800-a377-4ddf5609b9c1" data-id="el-li-47" data-item-num="5">Warmflash A</li></ol><span class="reference__authors_list_suffix" data-ref="209ffb2d-40d7-4c55-9abe-8b6c4a74d362">2019</span><span class="reference__title" data-ref="7d43c996-4a10-4216-9353-d25e56294d17">Dissecting the dynamics of signaling events in the BMP, WNT, and NODAL cascade during self-organized fate patterning in human gastruloids.</span><span class="reference__origin" data-ref="563207ec-3fa8-436d-8165-336adf08a71a"><i data-ref="2f71af90-d9ea-4c84-a69f-fc3a92497882">PLoS Biol</i> <strong data-ref="580844ba-1772-4e92-8604-5d16efde939c">17</strong>:e3000498</span><span class="reference__doi" data-ref="948a6e39-0998-48c3-bd3c-61a7fa540c52"></span></li><li class="reference-list__item" id="c4" data-ref="82afde16-b8bb-4fb8-b625-fc36b97c3502" data-id="c4"><span class="reference__label" data-ref="fa2530a9-4178-4e28-94cc-eabeb29cdf3a">4.</span><ol class="reference__authors_list" id="el-ol-6" data-ref="72df4fa4-f7b0-4160-96cc-aa9b8fde8d04" data-id="el-ol-6" start="1"><li class="reference__author" id="el-li-49" data-ref="7195ec6b-0aab-43fa-b971-9160a60e434e" data-id="el-li-49" data-item-num="1">Etoc F.</li><li class="reference__author" id="el-li-50" data-ref="7d7dc14f-ccab-42c6-80d0-6e0a9fe02468" data-id="el-li-50" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="5715be0a-ad36-4e04-8d6c-21cf9ec62948">2016</span><span class="reference__title" data-ref="c8e96529-c8a9-489f-a9f9-cf41d250441d">A Balance between Secreted Inhibitors and Edge Sensing Controls Gastruloid Self-Organization.</span><span class="reference__origin" data-ref="5829238d-0976-4a98-a622-d18664f34f36"><i data-ref="9b59913d-f042-46d6-8a09-24a50a06d434">Dev. Cell</i> <strong data-ref="dc404ac9-45c1-4483-8535-993a56c983d7">39</strong>:302‑315</span><span class="reference__doi" data-ref="2ebd1347-84a8-4b31-9aa1-5816b19d755c"></span></li><li class="reference-list__item" id="c5" data-ref="21478299-a5eb-489f-88df-769e4ff63847" data-id="c5"><span class="reference__label" data-ref="c7b78e89-4eed-4f9c-a155-102f64286809">5.</span><ol class="reference__authors_list" id="el-ol-7" data-ref="366036ad-03b0-4dbb-9cb9-ad20ad99ef61" data-id="el-ol-7" start="1"><li class="reference__author" id="el-li-52" data-ref="5c60a121-3d4d-433b-8fd0-7b15cd7a3669" data-id="el-li-52" data-item-num="1">Zhang Z.</li><li class="reference__author" id="el-li-53" data-ref="b8fdcb4c-85a2-4abb-b18d-b3927fb65ced" data-id="el-li-53" data-item-num="2">Zwick S.</li><li class="reference__author" id="el-li-54" data-ref="fd1eb8f5-eed8-4afe-879b-228986b1d0c5" data-id="el-li-54" data-item-num="3">Loew E.</li><li class="reference__author" id="el-li-55" data-ref="a292ffac-e688-42f1-974a-67ad7e3bf0bb" data-id="el-li-55" data-item-num="4">Grimley J. S.</li><li class="reference__author" id="el-li-56" data-ref="729d8909-f9a5-4df9-9fa6-770e50804867" data-id="el-li-56" data-item-num="5">Ramanathan S</li></ol><span class="reference__authors_list_suffix" data-ref="fb0aa1f5-0234-4bb9-9c1f-d0f98f07959e">2018</span><span class="reference__title" data-ref="9ad61b13-e581-482a-817c-e6b16be1f763">Embryo geometry drives formation of robust signaling gradients through receptor localization.</span><span class="reference__origin" data-ref="40650ba3-f241-4e69-81d6-e0f1824bed69"><i data-ref="6e420acf-cc1d-43a8-aae0-fb1ea2d87d64">bioRxiv</i> </span><span class="reference__doi" data-ref="6944fb09-8f3f-493d-93b1-60fd9ff6198d"><a href="https://doi.org/10.1101/491290" class="reference__doi_link" data-ref="3fe244a0-ddde-420b-8b92-6cf31873f9a1">https://doi.org/10.1101/491290</a></span></li><li class="reference-list__item" id="c6" data-ref="fc15e12a-730b-4b5f-9bdb-b3edaf63bc3c" data-id="c6"><span class="reference__label" data-ref="7c4794bb-0512-4986-8ecf-88699b7fc94d">6.</span><ol class="reference__authors_list" id="el-ol-8" data-ref="32d0cef5-4840-4773-a520-bde640649535" data-id="el-ol-8" start="1"><li class="reference__author" id="el-li-58" data-ref="4d61e830-082a-4a21-baa5-1489af9495e3" data-id="el-li-58" data-item-num="1">Heemskerk I.</li><li class="reference__author" id="el-li-59" data-ref="70109124-51a6-452b-a351-4c6faf0b39c9" data-id="el-li-59" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="53243e03-4906-4bdc-b386-f2b6f0895cad">2019</span><span class="reference__title" data-ref="263b5c6c-5fee-432a-9db5-bf1fdeb4664e">Rapid changes in morphogen concentration control self-organized patterning in human embryonic stem cells.</span><span class="reference__origin" data-ref="b35f70c3-aa7c-479a-ab78-711923e7a44e"><i data-ref="edcf2543-4a63-42f4-914e-c2b564b307ef">eLife</i> <strong data-ref="43bad3e7-a5a3-4c58-b19e-98b74d1235d2">8</strong></span><span class="reference__doi" data-ref="615c93d4-2e37-4d74-8ae5-04c2d0455c97"></span></li><li class="reference-list__item" id="c7" data-ref="c6384cdb-34da-4e20-918e-9ca292ed06b8" data-id="c7"><span class="reference__label" data-ref="2b82a30d-a09e-432a-b703-57012de6d65e">7.</span><ol class="reference__authors_list" id="el-ol-9" data-ref="c2c0a4e0-4b4e-4f71-91cb-666854e267f1" data-id="el-ol-9" start="1"><li class="reference__author" id="el-li-61" data-ref="560565fa-95dc-493a-ac63-6e70b06f9bf2" data-id="el-li-61" data-item-num="1">Massey J.</li><li class="reference__author" id="el-li-62" data-ref="7f675217-6dd6-4496-9c8c-dbb55ee77ed5" data-id="el-li-62" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="2a1f3f10-e566-458b-aac0-c70def92b104">2018</span><span class="reference__title" data-ref="2c457572-d8ae-434e-a89e-8439c906709c">WNT ligands stimulate transient signaling in human pluripotent cells and synergize with TGF-β ligands to stimulate sustained signaling during differentiation.</span><span class="reference__origin" data-ref="9009c567-9b1d-45b3-95c7-038552697f63"><i data-ref="fb551399-fbd5-4dd2-a0e0-bba0336136eb">bioRxiv</i> </span><span class="reference__doi" data-ref="14e28eb0-5058-4ed7-95e3-9a4467200ece"><a href="https://doi.org/10.1101/406306" class="reference__doi_link" data-ref="632a0d91-c8ad-4cab-8eb8-dbf04569c30e">https://doi.org/10.1101/406306</a></span></li><li class="reference-list__item" id="c8" data-ref="72794aea-bae3-4bbf-bd25-126298b65bbb" data-id="c8"><span class="reference__label" data-ref="68431f94-5d37-4908-876f-e5069d66c32e">8.</span><ol class="reference__authors_list" id="el-ol-10" data-ref="cb68c804-0a39-4905-bc9a-ca626c89927a" data-id="el-ol-10" start="1"><li class="reference__author" id="el-li-64" data-ref="96b277f3-5c09-4477-a400-e3b1fdad37bb" data-id="el-li-64" data-item-num="1">Yang R.</li><li class="reference__author" id="el-li-65" data-ref="d4444fc6-7a23-4e72-bb70-3c11a901478c" data-id="el-li-65" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="2e65d209-ef0f-4b99-b6b9-a893371cdf62">2021</span><span class="reference__title" data-ref="39adb9b3-a74c-4c5d-b6bc-e99317de0f51">Amnion signals are essential for mesoderm formation in primates.</span><span class="reference__origin" data-ref="4211e018-acc9-41d6-b892-81298f861f6f"><i data-ref="d235617f-604b-4dc1-aa39-88bc43f7d87f">Nat Commun</i> <strong data-ref="06ad1728-cbef-459d-9c9e-930adf5ca98a">12</strong>:5126</span><span class="reference__doi" data-ref="3fcdfa77-d7ee-4ffd-8d7f-1875b0a99f26"></span></li></ul></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_right_page" data-page-number="29" id="page-29">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e" data-split-to="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-id="article-content" data-split-from="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-split-to="726f25f1-fdd1-41b8-bf49-de736be3e3ae"><section data-ref="7921c412-ef92-4490-b273-a1fbe0815a2b" data-id="references" data-split-from="7921c412-ef92-4490-b273-a1fbe0815a2b" data-split-to="7921c412-ef92-4490-b273-a1fbe0815a2b"><ul class="reference-list" data-ref="5d1d46dc-2209-4116-9f07-193d224fefb4" data-id="el-ul-6" data-split-from="5d1d46dc-2209-4116-9f07-193d224fefb4" data-split-to="5d1d46dc-2209-4116-9f07-193d224fefb4" data-last-split-element="true" data-align-last-split-element="auto"><li class="reference-list__item" id="c9" data-ref="6e912998-936b-4c84-b203-9272fc10620b" data-id="c9"><span class="reference__label" data-ref="2b8ba453-1549-4e7f-a515-179570ba0529">9.</span><ol class="reference__authors_list" id="el-ol-11" data-ref="42ec6e82-2f5a-45d6-b547-8ce533044d0f" data-id="el-ol-11" start="1"><li class="reference__author" id="el-li-67" data-ref="cff3e78e-89fc-4286-9ed9-82bb716c53aa" data-id="el-li-67" data-item-num="1">Ornitz D. M.</li><li class="reference__author" id="el-li-68" data-ref="c5f2a9ac-eb68-40a2-83d6-781810bd9bfe" data-id="el-li-68" data-item-num="2">Itoh N</li></ol><span class="reference__authors_list_suffix" data-ref="01b12630-166b-4478-8b8b-25e0d5561752">2015</span><span class="reference__title" data-ref="99f1ed7a-2aa9-4b36-baf9-535ad8948da4">The Fibroblast Growth Factor signaling pathway.</span><span class="reference__origin" data-ref="80e29382-ec6b-4952-87a3-dd574ef4c16f"><i data-ref="91f4a9a2-178d-419e-acd2-db621d1235df">Wiley Interdiscip Rev Dev Biol</i> <strong data-ref="edf86d37-9310-4801-aef8-0ba5e7b6820e">4</strong>:215‑266</span><span class="reference__doi" data-ref="58f9aefb-0e4a-47a0-91ec-e93a86f27f66"></span></li><li class="reference-list__item" id="c10" data-ref="4a67d542-b81a-4bfc-bf59-69d656968d8a" data-id="c10"><span class="reference__label" data-ref="c6731a58-034c-4916-9a6a-02fbe71dcfcb">10.</span><ol class="reference__authors_list" id="el-ol-12" data-ref="4eaa2556-05dd-4bd0-b293-6d19f077fe9d" data-id="el-ol-12" start="1"><li class="reference__author" id="el-li-70" data-ref="e6aa80b6-acc9-4bbe-9df6-90c75080c07b" data-id="el-li-70" data-item-num="1">LaBonne C.</li><li class="reference__author" id="el-li-71" data-ref="476c0c82-3346-4ffc-b8fb-eded39b11c0e" data-id="el-li-71" data-item-num="2">Whitman M</li></ol><span class="reference__authors_list_suffix" data-ref="6cd69e47-6350-4964-86bf-bbdec9106ab3">1994</span><span class="reference__title" data-ref="2af35f56-527d-4c95-847c-7c9e5d55edba">Mesoderm induction by activin requires FGF-mediated intracellular signals.</span><span class="reference__origin" data-ref="90ad3831-aaed-4e14-a3ed-359821ff6a5e"><i data-ref="f69706d7-3ff0-4d2f-880a-73a0160840cb">Development</i> <strong data-ref="b2cfbc8d-819b-42da-9ef5-849dcaf2aad7">120</strong>:463‑472</span><span class="reference__doi" data-ref="93e1405a-c518-4a57-b94c-3bac5c3bbfb9"></span></li><li class="reference-list__item" id="c11" data-ref="7efebb3e-d194-4aab-8c7c-76b466495a90" data-id="c11"><span class="reference__label" data-ref="073b028d-c3d0-4824-bed6-6f1d79f0e6f0">11.</span><ol class="reference__authors_list" id="el-ol-13" data-ref="24f7090a-eb56-4441-89a2-e33f3e13b092" data-id="el-ol-13" start="1"><li class="reference__author" id="el-li-73" data-ref="140181f6-d046-4ca2-a548-b3a82701e1ac" data-id="el-li-73" data-item-num="1">Krens S. F. G.</li><li class="reference__author" id="el-li-74" data-ref="e683a903-48cc-4872-a35e-0f1b33945860" data-id="el-li-74" data-item-num="2">Corredor-Adámez M.</li><li class="reference__author" id="el-li-75" data-ref="c8607196-f7e5-4f72-a67b-bee71ad9e3c9" data-id="el-li-75" data-item-num="3">He S.</li><li class="reference__author" id="el-li-76" data-ref="69e261ae-47f0-4d38-a0de-6518b08c34af" data-id="el-li-76" data-item-num="4">Snaar-Jagalska B. E.</li><li class="reference__author" id="el-li-77" data-ref="c8e9fc73-f5e2-48ad-9865-9a7b5d80eb23" data-id="el-li-77" data-item-num="5">Spaink H. P</li></ol><span class="reference__authors_list_suffix" data-ref="3db1dca0-f41f-4f73-94b7-ee37e40bd569">2008</span><span class="reference__title" data-ref="878cbfa2-6f9f-48d9-b505-0f4643a10eb8">ERK1 and ERK2 MAPK are key regulators of distinct gene sets in zebrafish embryogenesis.</span><span class="reference__origin" data-ref="307312a8-c037-4139-99e1-f89fc08cbd25"><i data-ref="da59ff61-c5f4-4718-ad91-2f2a2051575d">BMC Genomics</i> <strong data-ref="d70b3be4-3055-43b7-b49a-8780a8e69a8a">9</strong>:196</span><span class="reference__doi" data-ref="47733132-b168-4526-87bf-30c50ad32f73"></span></li><li class="reference-list__item" id="c12" data-ref="c3877ed8-893e-4976-9767-4c5795cf657c" data-id="c12"><span class="reference__label" data-ref="825652a9-510d-4c3b-82aa-8402806f4270">12.</span><ol class="reference__authors_list" id="el-ol-14" data-ref="b5dffd93-3394-4094-8fad-f44124572a78" data-id="el-ol-14" start="1"><li class="reference__author" id="el-li-79" data-ref="7f87270e-c8d1-4aac-8892-6060faa57e7c" data-id="el-li-79" data-item-num="1">Yao Y.</li><li class="reference__author" id="el-li-80" data-ref="ccf39de1-8538-4385-943c-11cad98401e5" data-id="el-li-80" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="d39fc22e-ca2b-42aa-aa08-f48e26a35dc9">2003</span><span class="reference__title" data-ref="8fd0fa21-ea6c-4a41-a2ae-6bc1a8d63a32">Extracellular signal-regulated kinase 2 is necessary for mesoderm differentiation.</span><span class="reference__origin" data-ref="4870699c-c883-4ab1-9897-338efb2ed0cf"><i data-ref="2fb9a737-950d-4ed1-aedb-1e3b3d692bab">Proc Natl Acad Sci U S A</i> <strong data-ref="a31f89a0-2b39-4d83-a145-3ae3ad31903e">100</strong>:12759‑12764</span><span class="reference__doi" data-ref="32310008-8501-48ad-9528-26b6e8966d24"></span></li><li class="reference-list__item" id="c13" data-ref="c802d671-61a3-4a1a-954f-ee308aae1200" data-id="c13"><span class="reference__label" data-ref="16aa5276-b477-4b8f-a04e-399d8ebfb1f9">13.</span><ol class="reference__authors_list" id="el-ol-15" data-ref="e83bc63a-07c8-484d-842f-d796dbf4fcba" data-id="el-ol-15" start="1"><li class="reference__author" id="el-li-82" data-ref="7898a237-1e8e-44a4-8354-579f9e8cac75" data-id="el-li-82" data-item-num="1">Mitrani E.</li><li class="reference__author" id="el-li-83" data-ref="e71dda50-dbd7-4550-8cda-7e2b8fee8c13" data-id="el-li-83" data-item-num="2">Gruenbaum Y.</li><li class="reference__author" id="el-li-84" data-ref="8859dc9e-cf9e-4970-a502-d5050d1877d8" data-id="el-li-84" data-item-num="3">Shohat H.</li><li class="reference__author" id="el-li-85" data-ref="b5ecf1fd-7cf6-4598-a18d-3d28ddcdec7f" data-id="el-li-85" data-item-num="4">Ziv T</li></ol><span class="reference__authors_list_suffix" data-ref="9c46c236-20a6-4b12-ab0e-2929b39e20e7">1990</span><span class="reference__title" data-ref="b8f84daa-9b70-4cb5-b5fb-c721a43b029d">Fibroblast growth factor during mesoderm induction in the early chick embryo.</span><span class="reference__origin" data-ref="db02e0d4-bc0d-4c9a-8a55-8a9bf8fa570c"><i data-ref="9b13affe-3f8f-40dc-82a5-191dfeb0450a">Development</i> <strong data-ref="45e55612-a13f-4e39-8d9f-8a23a905ba78">109</strong>:387‑393</span><span class="reference__doi" data-ref="eb362f73-3afb-47dd-a023-7cf760c2ee9e"></span></li><li class="reference-list__item" id="c14" data-ref="8b4c7bd3-0c6a-470d-9253-4a859e7c2f3d" data-id="c14"><span class="reference__label" data-ref="5bcc5bf3-e3be-4dc7-9526-cc5cb4fb5e32">14.</span><ol class="reference__authors_list" id="el-ol-16" data-ref="d1a4434d-50a0-41d1-80e4-80e7243a6c0f" data-id="el-ol-16" start="1"><li class="reference__author" id="el-li-87" data-ref="570a487f-3fb1-48ac-94f5-eed094438650" data-id="el-li-87" data-item-num="1">Corson L. B.</li><li class="reference__author" id="el-li-88" data-ref="5741c17d-d5d7-441f-b103-47a479298c64" data-id="el-li-88" data-item-num="2">Yamanaka Y.</li><li class="reference__author" id="el-li-89" data-ref="921fb801-4c48-4200-9311-fc6c82ee15e0" data-id="el-li-89" data-item-num="3">Lai K.-M. V.</li><li class="reference__author" id="el-li-90" data-ref="b5c40c53-ac40-451f-a1f4-c33bbf5ebd96" data-id="el-li-90" data-item-num="4">Rossant J</li></ol><span class="reference__authors_list_suffix" data-ref="91b64e5c-1679-48d5-a918-aece995a6b47">2003</span><span class="reference__title" data-ref="47cbc3e4-b2e9-4f7f-918b-d6fb13c04aad">Spatial and temporal patterns of ERK signaling during mouse embryogenesis.</span><span class="reference__origin" data-ref="2f96a720-febc-4231-a805-696ada823050"><i data-ref="9c3a2959-2e4d-4596-ad26-a43a707e3f4b">Development</i> <strong data-ref="be9259db-caf4-4157-8b26-2959dae1b0d7">130</strong>:4527‑4537</span><span class="reference__doi" data-ref="0241b369-252a-4ca5-a41c-fd5de5d308da"></span></li><li class="reference-list__item" id="c15" data-ref="eb148bfc-86cc-4497-a9bf-dc81298f00c6" data-id="c15"><span class="reference__label" data-ref="10573e53-b221-4b6d-949d-249acb8a46f3">15.</span><ol class="reference__authors_list" id="el-ol-17" data-ref="4dae8e72-a0a3-4f47-8b56-9770e33408c2" data-id="el-ol-17" start="1"><li class="reference__author" id="el-li-92" data-ref="4105fd57-5445-477a-a733-95b4c51c2fa4" data-id="el-li-92" data-item-num="1">Morgani S. M.</li><li class="reference__author" id="el-li-93" data-ref="b5265cdb-9ecf-45c8-9d00-ee6c636e0ae1" data-id="el-li-93" data-item-num="2">Hadjantonakis A.-K.</li></ol><span class="reference__authors_list_suffix" data-ref="de950937-9671-4b08-bf39-c0fb5fce5d11">2020</span><span class="reference__title" data-ref="4434a759-9e3e-46c7-93ac-e7c06af92ef6">Signaling regulation during gastrulation: Insights from mouse embryos and <em data-ref="007b0334-30a2-470a-9e47-9a9e51b57257">in vitro</em> systems.</span><span class="reference__origin" data-ref="5da663a0-2ca3-4b30-a1e2-298d4f13920c">In: <ol class="reference__editors_list" id="el-ol-18" data-ref="3e54da36-5af7-425a-b70e-69389a85e908" data-id="el-ol-18" start="1"><li class="reference__editor" id="el-li-94" data-ref="cfd6a036-29ba-4d1c-86da-54c39dc4ff93" data-id="el-li-94" data-item-num="1">Small S.</li><li class="reference__editor" id="el-li-95" data-ref="02a22912-baf5-4e4e-b62f-16643202117f" data-id="el-li-95" data-item-num="2">Briscoe J.</li></ol> (Eds). <i data-ref="e3b4dd65-7ad3-4d82-a821-63fdb0bd72f2">Current Topics in Developmental Biology</i> <strong data-ref="af03097e-0a28-4907-8fb2-a0a797d9023c">137</strong> Academic Press. pp.&nbsp;391‑431</span><span class="reference__doi" data-ref="c7666ac2-185d-4f97-9356-901464c9069f"></span></li><li class="reference-list__item" id="c16" data-ref="2e8f94c6-f1e3-44aa-ac10-c37c6fc3df30" data-id="c16"><span class="reference__label" data-ref="b428f69e-e06b-4986-9cb2-7fe68e5e4ae3">16.</span><ol class="reference__authors_list" id="el-ol-19" data-ref="32e48447-c49d-4729-a796-57f9b3c421fb" data-id="el-ol-19" start="1"><li class="reference__author" id="el-li-97" data-ref="264e2718-faf1-4ab2-b7d6-7e55299a19c0" data-id="el-li-97" data-item-num="1">De Simone A.</li><li class="reference__author" id="el-li-98" data-ref="4d985f5e-d349-47d1-93e7-a607b1e8358f" data-id="el-li-98" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="596a18bb-d073-4076-b6b2-5a50ab4e2dff">2021</span><span class="reference__title" data-ref="fa2d98f4-da01-40fb-b067-4246e84db701">Control of osteoblast regeneration by a train of Erk activity waves.</span><span class="reference__origin" data-ref="0409dee1-a7fd-43a4-99a5-2b97f6f6946f"><i data-ref="e1fc0418-08d4-4bed-a44c-318f1cfe46d6">Nature</i> <strong data-ref="b75c7b96-6b87-41a8-a88b-f981deff6f43">590</strong>:129‑133</span><span class="reference__doi" data-ref="783bcbac-1e84-4ec7-8a32-db637c0771ef"></span></li><li class="reference-list__item" id="c17" data-ref="3d96b9e9-165d-4131-b4f2-cd55895806d7" data-id="c17"><span class="reference__label" data-ref="8cec99a6-446b-486a-9ee5-0faf67321212">17.</span><ol class="reference__authors_list" id="el-ol-20" data-ref="c1b07be1-8b5e-4d0f-83ac-a6c48da8be2c" data-id="el-ol-20" start="1"><li class="reference__author" id="el-li-100" data-ref="a03cc5af-2393-4710-aa73-a9a37114f75d" data-id="el-li-100" data-item-num="1">Marshall C. J</li></ol><span class="reference__authors_list_suffix" data-ref="b5d7449e-851d-4c65-aef3-ff34d39cbae3">1995</span><span class="reference__title" data-ref="f518fa7f-8cad-427f-9e5d-89f003efbe29">Specificity of receptor tyrosine kinase signaling: transient versus sustained extracellular signal-regulated kinase activation.</span><span class="reference__origin" data-ref="e74adc20-c706-4a83-aa28-f1cc581cc4f8"><i data-ref="daa4bffa-1396-4538-b95c-797c818e527e">Cell</i> <strong data-ref="50116503-6e34-460a-9b87-c1c4d426da1a">80</strong>:179‑185</span><span class="reference__doi" data-ref="230461c6-70a2-417c-aa33-b8c49f89fe0f"></span></li><li class="reference-list__item" id="c18" data-ref="8dcc1e7d-f52e-4647-ab53-1190691e5e0e" data-id="c18"><span class="reference__label" data-ref="42599176-5e2e-450b-b1b8-afff626fe1b4">18.</span><ol class="reference__authors_list" id="el-ol-21" data-ref="1f7dffe5-8f1b-46b2-9832-83b0dc333319" data-id="el-ol-21" start="1"><li class="reference__author" id="el-li-102" data-ref="a0337997-3b46-4f74-8183-e2f58a0db1f2" data-id="el-li-102" data-item-num="1">Ender P.</li><li class="reference__author" id="el-li-103" data-ref="3a22857d-7e2f-4f0d-b2ba-47f7e0985e82" data-id="el-li-103" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="0f57590f-fa3d-4aae-856f-fd6d3a2819b2">2022</span><span class="reference__title" data-ref="c7f817fc-5aa2-4a1e-8ba3-8af1f9007657">Spatiotemporal control of ERK pulse frequency coordinates fate decisions during mammary acinar morphogenesis.</span><span class="reference__origin" data-ref="0df07ddb-71a1-4125-ab5d-eb9ca20837fb"><i data-ref="6b73a7a4-fd1f-479b-b5b3-7dff9dc1edae">Dev Cell</i> <strong data-ref="d80f87e9-af80-4e37-9f38-bdd0d5fb4b67">57</strong>:2153‑2167.e6</span><span class="reference__doi" data-ref="abc13b67-2814-433b-a76e-144c3cc8e84e"></span></li><li class="reference-list__item" id="c19" data-ref="51199e54-b1e3-4a84-82bc-8962cb79dc1a" data-id="c19"><span class="reference__label" data-ref="f07249e2-df4d-48c4-858a-b409d2f40fa7">19.</span><ol class="reference__authors_list" id="el-ol-22" data-ref="f73d32b8-1b47-47c0-b438-141c74d22ebe" data-id="el-ol-22" start="1"><li class="reference__author" id="el-li-105" data-ref="52c891d9-447e-40ce-ae72-0c6f33532eb0" data-id="el-li-105" data-item-num="1">Aoki K.</li><li class="reference__author" id="el-li-106" data-ref="8853dc6b-7f69-47de-9a91-c4744b67f7e5" data-id="el-li-106" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="2117c2dc-cd33-4c86-ad43-658c74cc06bf">2017</span><span class="reference__title" data-ref="2bd8f2f5-89fc-45d3-9123-b9b732bc6ae7">Propagating Wave of ERK Activation Orients Collective Cell Migration.</span><span class="reference__origin" data-ref="c2816bc1-6a4e-4f33-81ab-b91be4b77d6d"><i data-ref="ec116793-586e-4615-bb80-0c056a736878">Dev Cell</i> <strong data-ref="abbb17e3-97be-44fc-8592-0bfa17738480">43</strong>:305‑317.e5</span><span class="reference__doi" data-ref="79d3f184-9ead-4cc4-8caf-8711f398691c"></span></li><li class="reference-list__item" id="c20" data-ref="b81d72ea-9e7d-4fbe-bbaa-f209d45648aa" data-id="c20"><span class="reference__label" data-ref="34d78af0-0fb3-45f6-abcb-50833c7d4971">20.</span><ol class="reference__authors_list" id="el-ol-23" data-ref="bc36d13d-078f-417e-8666-a856941795ba" data-id="el-ol-23" start="1"><li class="reference__author" id="el-li-108" data-ref="8821067c-56a9-44e4-8290-a8ddd1ab5965" data-id="el-li-108" data-item-num="1">Ciruna B.</li><li class="reference__author" id="el-li-109" data-ref="749c2e8f-a945-4b83-ae0a-1738b366eda5" data-id="el-li-109" data-item-num="2">Rossant J</li></ol><span class="reference__authors_list_suffix" data-ref="2448a15a-8e1c-4fbe-b597-a3579f311743">2001</span><span class="reference__title" data-ref="2eec85b6-abc9-4216-9cd1-b23b1b702ce2">FGF Signaling Regulates Mesoderm Cell Fate Specification and Morphogenetic Movement at the Primitive Streak.</span><span class="reference__origin" data-ref="0ce44839-9b1b-4a8a-aade-0b08ac3c7118"><i data-ref="13b6a0b9-505c-4e18-a76b-478eeaa04379">Dev. Cell</i> <strong data-ref="40581f5f-6f6e-428e-8ce7-da39a45cf593">1</strong>:37‑49</span><span class="reference__doi" data-ref="fbc255cc-31f9-44e1-889a-f39d2f5d2ea3"></span></li><li class="reference-list__item" id="c21" data-ref="b1307823-cccf-4c37-9f3e-47d90d35aaf1" data-id="c21"><span class="reference__label" data-ref="5c3c3375-0971-4414-bea3-88854d148a7b">21.</span><ol class="reference__authors_list" id="el-ol-24" data-ref="4672c433-a43c-4dc2-8065-75e6547cf117" data-id="el-ol-24" start="1"><li class="reference__author" id="el-li-111" data-ref="3638dbc7-c519-4844-a107-6490a160b985" data-id="el-li-111" data-item-num="1">Sun X.</li><li class="reference__author" id="el-li-112" data-ref="27c99f2e-b209-4fdd-bb0e-49b1bb98890d" data-id="el-li-112" data-item-num="2">Meyers E. N.</li><li class="reference__author" id="el-li-113" data-ref="d71cc8c4-6ba0-4211-908c-6d1eae7c48de" data-id="el-li-113" data-item-num="3">Lewandoski M.</li><li class="reference__author" id="el-li-114" data-ref="3babe324-9964-43c2-8a5c-1fa7890d8fe8" data-id="el-li-114" data-item-num="4">Martin G. R</li></ol><span class="reference__authors_list_suffix" data-ref="a0662fee-96cc-479e-8771-ae3175c3f254">1999</span><span class="reference__title" data-ref="a3c7a52c-7220-4e55-b818-3fbd1a3f428e">Targeted disruption of Fgf8 causes failure of cell migration in the gastrulating mouse embryo.</span><span class="reference__origin" data-ref="75662bc9-4b5b-45df-81d2-39b4e0e13497"><i data-ref="852cd638-929a-4127-a897-9496f40bf543">Genes Dev</i> <strong data-ref="1227da17-af5c-45ca-b5e4-6c3e510944fc">13</strong>:1834‑1846</span><span class="reference__doi" data-ref="9fc88b93-d136-4a17-8361-27d915c36586"></span></li><li class="reference-list__item" id="c22" data-ref="119bf123-9f0c-4c95-90f1-3575403eccf5" data-id="c22"><span class="reference__label" data-ref="97d00574-ba3a-4771-8606-131d4fef0a02">22.</span><ol class="reference__authors_list" id="el-ol-25" data-ref="a8623a7a-4950-4a01-a4c8-e604416acbc5" data-id="el-ol-25" start="1"><li class="reference__author" id="el-li-116" data-ref="6abeda82-8c26-45dc-9bf7-21603b638adb" data-id="el-li-116" data-item-num="1">Fletcher R. B.</li><li class="reference__author" id="el-li-117" data-ref="77961058-b050-4d7b-8d11-75a9aa602221" data-id="el-li-117" data-item-num="2">Harland R. M</li></ol><span class="reference__authors_list_suffix" data-ref="e9b34eae-1ac3-46e3-afb7-97bb95ffe3c3">2008</span><span class="reference__title" data-ref="3194abc3-6213-4aa2-8d1d-a6d6bf63b440">The role of FGF signaling in the establishment and maintenance of mesodermal gene expression in Xenopus.</span><span class="reference__origin" data-ref="a1561ab0-cd2d-4c0f-a31d-61ede31dd3a8"><i data-ref="26a75d5f-ac7a-4475-bc15-d56cd8b1d65f">Dev Dyn</i> <strong data-ref="ab5c4318-c546-4b4c-ba68-1952d2b607ee">237</strong>:1243‑1254</span><span class="reference__doi" data-ref="bbdb15e1-ce58-4e2b-8238-34680b0e4727"></span></li><li class="reference-list__item" id="c23" data-ref="a46ee167-d875-4a4b-a100-3f1d5e121477" data-id="c23"><span class="reference__label" data-ref="f603e1fa-c0b8-4ba3-a027-a46c2309d8f0">23.</span><ol class="reference__authors_list" id="el-ol-26" data-ref="8e8015b8-a910-485e-9ba1-570fef497d8f" data-id="el-ol-26" start="1"><li class="reference__author" id="el-li-119" data-ref="ee3f776b-1e97-48f9-b9a2-b6afa2873e53" data-id="el-li-119" data-item-num="1">Yang X.</li><li class="reference__author" id="el-li-120" data-ref="c55b5f27-aef4-49f1-9ebd-e9917b2e9ae7" data-id="el-li-120" data-item-num="2">Dormann D.</li><li class="reference__author" id="el-li-121" data-ref="d3d54737-ce50-4f87-b321-60898411ad0d" data-id="el-li-121" data-item-num="3">Münsterberg A. E.</li><li class="reference__author" id="el-li-122" data-ref="7e80ed14-88b3-4a03-858e-de9ba320a371" data-id="el-li-122" data-item-num="4">Weijer C. J</li></ol><span class="reference__authors_list_suffix" data-ref="92d1a8ae-b26e-4dbd-ae14-68f53ab1329b">2002</span><span class="reference__title" data-ref="bd72a40c-57d5-4e74-9432-48e22608d152">Cell Movement Patterns during Gastrulation in the Chick Are Controlled by Positive and Negative Chemotaxis Mediated by FGF4 and FGF8.</span><span class="reference__origin" data-ref="d8d2e51b-5f52-4ee4-8df0-fc7ded8c8907"><i data-ref="eb43d00d-b51a-49d6-b75a-740cd9bb6061">Developmental Cell</i> <strong data-ref="359d6ba7-81ce-4f24-a820-b3045c6823db">3</strong>:425‑437</span><span class="reference__doi" data-ref="e838e0e2-41b2-4955-af36-1022b122f8ca"></span></li><li class="reference-list__item" id="c24" data-ref="4779889e-1fd7-4b7a-953c-df4d0c35301b" data-id="c24"><span class="reference__label" data-ref="1cf1298b-f7eb-4258-bd91-f69f37677bcb">24.</span><ol class="reference__authors_list" id="el-ol-27" data-ref="869c25e8-e55d-4957-8c9f-91afd5be7d56" data-id="el-ol-27" start="1"><li class="reference__author" id="el-li-124" data-ref="64419559-f770-4c12-9a9f-554627aac660" data-id="el-li-124" data-item-num="1">Fürthauer M.</li><li class="reference__author" id="el-li-125" data-ref="1782206c-7c6d-4f25-941a-3af23cc8eddb" data-id="el-li-125" data-item-num="2">Reifers F.</li><li class="reference__author" id="el-li-126" data-ref="1d0c5b95-8c6a-4175-a14f-052b6465d573" data-id="el-li-126" data-item-num="3">Brand M.</li><li class="reference__author" id="el-li-127" data-ref="9c00e033-f5e5-4b1d-96a2-eeae88cafcab" data-id="el-li-127" data-item-num="4">Thisse B.</li><li class="reference__author" id="el-li-128" data-ref="0e4c5fef-b741-49b7-afdc-54194ec1fbcb" data-id="el-li-128" data-item-num="5">Thisse C</li></ol><span class="reference__authors_list_suffix" data-ref="14445830-94c3-4873-86a1-79ed6aabe4b9">2001</span><span class="reference__title" data-ref="92378523-a06b-423d-a9fe-78c5ea0cf7d9">sprouty4 acts in vivo as a feedback-induced antagonist of FGF signaling in zebrafish.</span><span class="reference__origin" data-ref="fc00f0cf-6211-44d4-91b7-710fdc4d807e"><i data-ref="fb1f4110-b66f-4372-923f-05a7b585decc">Development</i> <strong data-ref="90817629-3a3c-4f3a-b6a1-003b73d685a4">128</strong>:2175‑2186</span><span class="reference__doi" data-ref="be272959-1f48-48b5-abdd-5bc41acd9a12"></span></li><li class="reference-list__item" id="c25" data-ref="059d8044-25f1-491e-a39f-c8828ab524b2" data-id="c25"><span class="reference__label" data-ref="a22d1740-1c68-488b-bfea-fee167c6c2ae">25.</span><ol class="reference__authors_list" id="el-ol-28" data-ref="6ddf4821-2adc-4dc9-a3c0-1cf5b57fca77" data-id="el-ol-28" start="1"><li class="reference__author" id="el-li-130" data-ref="2ebfe4b7-36dd-4ad2-a76a-ee2261c8f648" data-id="el-li-130" data-item-num="1">Tyser R. C. V.</li><li class="reference__author" id="el-li-131" data-ref="69337f31-9784-4910-9606-f1416041439d" data-id="el-li-131" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="39209166-7334-48d9-a344-5a58521c32c7">2021</span><span class="reference__title" data-ref="c5056dcf-0f3d-4773-98ee-5d9dd174be91">Single-cell transcriptomic characterization of a gastrulating human embryo.</span><span class="reference__origin" data-ref="8f933b35-ae63-43ed-95d4-b395300e9952"><i data-ref="4aaf10d6-72c8-404b-81cc-f5e507680c98">Nature</i> <strong data-ref="5a96da51-1a51-404b-8bce-db8c48f19275">600</strong>:285‑289</span><span class="reference__doi" data-ref="740f030d-375c-4bca-9187-8e8500e3033c"></span></li><li class="reference-list__item" id="c26" data-ref="2b04f35f-70d8-4b51-8f1f-c249a42b5862" data-id="c26"><span class="reference__label" data-ref="53e25898-2f7b-4e26-96f3-b73fec39f28e">26.</span><ol class="reference__authors_list" id="el-ol-29" data-ref="e514dffb-ccba-46e7-9060-25ff4186abdd" data-id="el-ol-29" start="1"><li class="reference__author" id="el-li-133" data-ref="53ce7101-7860-4e73-8a44-37f1b75eeaa2" data-id="el-li-133" data-item-num="1">Zhai J.</li><li class="reference__author" id="el-li-134" data-ref="458af462-17e6-4880-b215-68cb43c0b7c5" data-id="el-li-134" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="c60efb41-5efa-4b94-b7af-ca1d619f47d5">2022</span><span class="reference__title" data-ref="e14778eb-c44e-47ad-876f-03d4a2f2e4ed">Primate gastrulation and early organogenesis at single-cell resolution.</span><span class="reference__origin" data-ref="01ed2478-61a3-46ce-ab0b-d5a124254fe3"><i data-ref="7840e78a-4c05-4065-a4d8-3ec022220998">Nature</i> <strong data-ref="a109b31e-7784-4313-b003-7803d25d4d29">612</strong>:732‑738</span><span class="reference__doi" data-ref="cc256f67-246c-485d-aead-7f376321b418"></span></li><li class="reference-list__item" id="c27" data-ref="e57b9290-9ab2-4a01-85e2-b82252d7e8da" data-id="c27"><span class="reference__label" data-ref="e0f8d35a-577b-4f2a-a78e-b5dafde108b3">27.</span><ol class="reference__authors_list" id="el-ol-30" data-ref="ff72386c-e8fe-48d0-a00e-4ee34e5f178f" data-id="el-ol-30" start="1"><li class="reference__author" id="el-li-136" data-ref="d5a21d6b-689a-445d-96cf-a75cc7137628" data-id="el-li-136" data-item-num="1">Deng C. X.</li><li class="reference__author" id="el-li-137" data-ref="74685d55-7626-47dc-a87e-d647c0fe3156" data-id="el-li-137" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="23695077-6f13-4e5f-8cc8-7d79b9b3f742">1994</span><span class="reference__title" data-ref="4c6ecb83-6e0b-48c9-b1e5-8dde620cd8b5">Murine FGFR-1 is required for early postimplantation growth and axial organization.</span><span class="reference__origin" data-ref="f340fee1-e439-4800-a587-36f036d120dd"><i data-ref="40eadde2-f169-4574-a81f-9aa947957730">Genes Dev</i> <strong data-ref="e3d8bee1-121d-4dac-bfed-f34eb9e513cb">8</strong>:3045‑3057</span><span class="reference__doi" data-ref="b46f3fd8-a207-4441-909b-f426a1ed5c94"></span></li><li class="reference-list__item" id="c28" data-ref="45f4198b-5a53-4333-ade3-fc648a60934a" data-id="c28"><span class="reference__label" data-ref="75421208-887c-4244-84ce-f8aec57944d7">28.</span><ol class="reference__authors_list" id="el-ol-31" data-ref="5096c0b8-0a3a-43a0-a273-1ec06cf75d5f" data-id="el-ol-31" start="1"><li class="reference__author" id="el-li-139" data-ref="e9062ec4-87fe-417a-afd7-8976f3a10a02" data-id="el-li-139" data-item-num="1">Yamaguchi T. P.</li><li class="reference__author" id="el-li-140" data-ref="4242ad5e-ddc7-4152-87e1-961892af009d" data-id="el-li-140" data-item-num="2">Harpal K.</li><li class="reference__author" id="el-li-141" data-ref="55b0cff2-0045-43b2-a1f9-7b1360f328be" data-id="el-li-141" data-item-num="3">Henkemeyer M.</li><li class="reference__author" id="el-li-142" data-ref="fc1566e4-5b58-4bc7-935b-a47a4904ef04" data-id="el-li-142" data-item-num="4">Rossant J.</li></ol><span class="reference__authors_list_suffix" data-ref="47daeab7-bb63-436a-ae07-986d391bb545">1994</span><span class="reference__title" data-ref="14c9f475-8df9-4f70-a47f-27356ffbbcd4">fgfr-1 is required for embryonic growth and mesodermal patterning during mouse gastrulation.</span><span class="reference__origin" data-ref="998ef39b-c64e-4ca3-ac46-27dfaaaa1b69"><i data-ref="5e630b3f-f96b-4e69-b8a5-d3d363e02b44">Genes Dev</i> <strong data-ref="606d06d7-3d4f-4005-b514-01196fa0897e">8</strong>:3032‑3044</span><span class="reference__doi" data-ref="8fd477ac-e2ba-421c-999f-77e7bf1b69d2"></span></li><li class="reference-list__item" id="c29" data-ref="5df9000c-ecd7-47cb-a6d0-1238ee26f35c" data-id="c29"><span class="reference__label" data-ref="4f3a8a6b-afaa-4aa5-a28e-dda02db6bf61">29.</span><ol class="reference__authors_list" id="el-ol-32" data-ref="49cc2947-7490-4b92-8db1-2a0b07887c28" data-id="el-ol-32" start="1"><li class="reference__author" id="el-li-144" data-ref="02637b08-3442-41cb-b616-38ffdfb37acf" data-id="el-li-144" data-item-num="1">Ciruna B. G.</li><li class="reference__author" id="el-li-145" data-ref="52315462-2720-449a-9077-945d4399a207" data-id="el-li-145" data-item-num="2">Schwartz L.</li><li class="reference__author" id="el-li-146" data-ref="511b55ae-3728-4dd5-9c4b-d2b18f3b4c84" data-id="el-li-146" data-item-num="3">Harpal K.</li><li class="reference__author" id="el-li-147" data-ref="78fe05f7-41e7-469d-957b-6212adc0d407" data-id="el-li-147" data-item-num="4">Yamaguchi T. P.</li><li class="reference__author" id="el-li-148" data-ref="086322dd-dee9-41e2-846d-69a10655b605" data-id="el-li-148" data-item-num="5">Rossant J</li></ol><span class="reference__authors_list_suffix" data-ref="d7ededf8-7476-44a6-901e-ba5fe7e5b4a0">1997</span><span class="reference__title" data-ref="235700b1-b85b-4c3c-b74e-4052acc90f96">Chimeric analysis of fibroblast growth factor receptor-1 (Fgfr1) function: a role for FGFR1 in morphogenetic movement through the primitive streak.</span><span class="reference__origin" data-ref="dde4f8fe-d9a5-45c7-9b9e-44bd95bc1fb9"><i data-ref="fc447b62-448d-43c8-a56d-2d2be55602ff">Development</i> <strong data-ref="f3390fd9-a480-4f95-a7a0-4c533f70552e">124</strong>:2829‑2841</span><span class="reference__doi" data-ref="c28d12ea-47ea-4ae9-8092-0211eeba6f01"></span></li><li class="reference-list__item" id="c30" data-ref="9576dd2d-7b81-44a2-8eaf-f00bec7eb102" data-id="c30"><span class="reference__label" data-ref="096cc2cc-26d6-48e3-b319-d9651553c665">30.</span><ol class="reference__authors_list" id="el-ol-33" data-ref="40be3c9e-9813-4d89-a13b-f4ab8e6cbf2a" data-id="el-ol-33" start="1"><li class="reference__author" id="el-li-150" data-ref="a655626e-96c6-4324-8e49-6f3df51f352b" data-id="el-li-150" data-item-num="1">García-García M. J.</li><li class="reference__author" id="el-li-151" data-ref="99336973-b649-4e5f-8fbd-1c3472e05ecd" data-id="el-li-151" data-item-num="2">Anderson K. V</li></ol><span class="reference__authors_list_suffix" data-ref="33268b11-cd1b-4455-ad8d-40f2fcd2902f">2003</span><span class="reference__title" data-ref="245bd66b-a57c-40c6-af23-162d4d64f163">Essential role of glycosaminoglycans in Fgf signaling during mouse gastrulation.</span><span class="reference__origin" data-ref="5c16caf9-5b47-4789-b07e-7878d7ec8652"><i data-ref="7edd03c8-af12-4abb-ad62-32cd0ecce439">Cell</i> <strong data-ref="2fe6dbe6-3c85-45af-b830-e137a4685970">114</strong>:727‑737</span><span class="reference__doi" data-ref="04190581-3a17-4872-a840-8967b32fdeee"></span></li></ul></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_left_page" data-page-number="30" id="page-30">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e" data-split-to="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-id="article-content" data-split-from="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-split-to="726f25f1-fdd1-41b8-bf49-de736be3e3ae"><section data-ref="7921c412-ef92-4490-b273-a1fbe0815a2b" data-id="references" data-split-from="7921c412-ef92-4490-b273-a1fbe0815a2b" data-split-to="7921c412-ef92-4490-b273-a1fbe0815a2b"><ul class="reference-list" data-ref="5d1d46dc-2209-4116-9f07-193d224fefb4" data-id="el-ul-6" data-split-from="5d1d46dc-2209-4116-9f07-193d224fefb4" data-split-to="5d1d46dc-2209-4116-9f07-193d224fefb4" data-last-split-element="true" data-align-last-split-element="auto"><li class="reference-list__item" id="c31" data-ref="5efcfb8a-296f-4753-bb6b-20b12738c6db" data-id="c31"><span class="reference__label" data-ref="4a630f9b-61c4-4da3-8ddc-3533349eaf2e">31.</span><ol class="reference__authors_list" id="el-ol-34" data-ref="470f1e10-fcf1-44f7-b9fc-d0e2e1943e0f" data-id="el-ol-34" start="1"><li class="reference__author" id="el-li-153" data-ref="74b6df40-5721-4991-a03c-dd2f39c1040d" data-id="el-li-153" data-item-num="1">Oki S.</li><li class="reference__author" id="el-li-154" data-ref="a03350de-4c33-4fa4-83b1-7a9f54463bf9" data-id="el-li-154" data-item-num="2">Kitajima K.</li><li class="reference__author" id="el-li-155" data-ref="c9b52bef-2904-4465-b4ac-9fbc4788a404" data-id="el-li-155" data-item-num="3">Meno C</li></ol><span class="reference__authors_list_suffix" data-ref="eb6cd471-d184-4ec0-b3b8-54bcc72bfeeb">2010</span><span class="reference__title" data-ref="d0a42abf-a8ae-46d4-a333-88b56fdcc255">Dissecting the role of Fgf signaling during gastrulation and left-right axis formation in mouse embryos using chemical inhibitors.</span><span class="reference__origin" data-ref="c06dbb70-81df-44c1-a3ce-7b841e85cd0e"><i data-ref="762ed09b-ef73-4d20-b943-719c99e9c71b">Dev Dyn</i> <strong data-ref="01d371ac-82bc-4ca8-8a27-4caf2408f3a7">239</strong>:1768‑1778</span><span class="reference__doi" data-ref="91175d62-aac5-44a6-a302-358c2b68d5b7"></span></li><li class="reference-list__item" id="c32" data-ref="3475504a-18f6-475a-a44d-aaf62252e491" data-id="c32"><span class="reference__label" data-ref="136c772c-3c08-441f-ab4d-cedb71779dc1">32.</span><ol class="reference__authors_list" id="el-ol-35" data-ref="aa06b77f-54df-4494-b318-074602c5fd12" data-id="el-ol-35" start="1"><li class="reference__author" id="el-li-157" data-ref="2f29b4d2-d7b2-4b9a-85db-0b1547d0669c" data-id="el-li-157" data-item-num="1">Feldman B.</li><li class="reference__author" id="el-li-158" data-ref="3a4a8eb6-4c7f-46bf-aac0-2431f512298f" data-id="el-li-158" data-item-num="2">Poueymirou W.</li><li class="reference__author" id="el-li-159" data-ref="126daef3-945c-4765-90a3-532f8e6cfdff" data-id="el-li-159" data-item-num="3">Papaioannou V. E.</li><li class="reference__author" id="el-li-160" data-ref="79b558ab-b12d-49b0-8c6c-7348aa43371b" data-id="el-li-160" data-item-num="4">DeChiara T. M.</li><li class="reference__author" id="el-li-161" data-ref="5421c43e-af83-4bbe-95fc-b75f09444b8a" data-id="el-li-161" data-item-num="5">Goldfarb M</li></ol><span class="reference__authors_list_suffix" data-ref="1a581dbe-ce1f-4be2-997e-69798bb70c76">1995</span><span class="reference__title" data-ref="fdfbbc8d-1617-4e25-9b9e-b1b0909cf34a">Requirement of FGF-4 for postimplantation mouse development.</span><span class="reference__origin" data-ref="af8ff349-1e97-4ce6-8bc4-fb87cba044f2"><i data-ref="02cfaee6-9775-4be7-8457-1180cb0c4da1">Science</i> <strong data-ref="38bf92fd-d875-445f-978f-b811dc9a36f8">267</strong>:246‑249</span><span class="reference__doi" data-ref="df9928f3-f050-4e71-ab93-39e2e5b326a8"></span></li><li class="reference-list__item" id="c33" data-ref="a7dfb8c4-fc93-4deb-ae6f-3096f4b75fd3" data-id="c33"><span class="reference__label" data-ref="cf941a4f-d0c5-4e3b-b0dd-815be5bc4c83">33.</span><ol class="reference__authors_list" id="el-ol-36" data-ref="4d2ec906-f2fd-4dd6-bfc4-1cc3ee5385a1" data-id="el-ol-36" start="1"><li class="reference__author" id="el-li-163" data-ref="1910676c-cf6a-45c5-8610-30c2236be84d" data-id="el-li-163" data-item-num="1">Warmflash A.</li><li class="reference__author" id="el-li-164" data-ref="c1ed6901-45cd-48b1-beca-a387e2b7d4d1" data-id="el-li-164" data-item-num="2">Sorre B.</li><li class="reference__author" id="el-li-165" data-ref="117e2e97-8e40-4b84-b3a9-7932d32fd96b" data-id="el-li-165" data-item-num="3">Etoc F.</li><li class="reference__author" id="el-li-166" data-ref="2cfa4be2-6342-4cc7-b188-7d6737e395c4" data-id="el-li-166" data-item-num="4">Siggia E. D.</li><li class="reference__author" id="el-li-167" data-ref="e33a967c-a2c9-4e41-a089-103a043c6bfe" data-id="el-li-167" data-item-num="5">Brivanlou A. H</li></ol><span class="reference__authors_list_suffix" data-ref="f4c69c37-5657-4a2c-9e4f-7f452c233d96">2014</span><span class="reference__title" data-ref="1f103d5e-3dc2-4e29-8420-4300a319124a">A method to recapitulate early embryonic spatial patterning in human embryonic stem cells.</span><span class="reference__origin" data-ref="8fc9a347-01b5-4086-aea2-41dcfbd97153"><i data-ref="6e61dab0-9e8c-4b75-b1e9-62e5bde530e0">Nat. Methods</i> <strong data-ref="4f664aae-3df3-43c3-87d1-bd793b9bef2a">11</strong>:847‑854</span><span class="reference__doi" data-ref="eb20efa3-02f3-439d-a85f-c06ff54610fa"></span></li><li class="reference-list__item" id="c34" data-ref="570e7726-715f-4abd-a022-a63a2470849a" data-id="c34"><span class="reference__label" data-ref="f1e48a74-dd4c-44b9-a432-52492a347d0b">34.</span><ol class="reference__authors_list" id="el-ol-37" data-ref="3cfb673d-deb8-481a-beb9-bea4d35a079d" data-id="el-ol-37" start="1"><li class="reference__author" id="el-li-169" data-ref="e4cf5786-b3fc-4c8f-b1bf-4faafbd2ac5c" data-id="el-li-169" data-item-num="1">Jo K.</li><li class="reference__author" id="el-li-170" data-ref="bd2ce6e4-35e1-408f-8750-7e527d5d3bbb" data-id="el-li-170" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="94b6850b-8b4e-409b-91b9-3dc6f1599dbc">2022</span><span class="reference__title" data-ref="9de3dba0-0885-45bc-854e-75d1ff80e573">Efficient differentiation of human primordial germ cells through geometric control reveals a key role for Nodal signaling.</span><span class="reference__origin" data-ref="b8d7c848-2120-4c21-b328-a6a928276186"><i data-ref="d932f30a-9128-4856-a0f0-7065f82bc880">eLife</i> <strong data-ref="140c143c-2b9e-4130-a887-7abe7d94cfa6">11</strong>:e72811</span><span class="reference__doi" data-ref="335c72f8-4689-43b6-8f9b-533d8464a33c"><a href="https://doi.org/10.7554/eLife.72811" class="reference__doi_link" data-ref="d41016e2-6ac2-4514-b9b9-346548e6f02c">https://doi.org/10.7554/eLife.72811</a></span></li><li class="reference-list__item" id="c35" data-ref="8356ad9a-a77d-4633-9d37-d2c89128b0cc" data-id="c35"><span class="reference__label" data-ref="1941af9f-861a-4604-bc1c-bfc28881d13b">35.</span><ol class="reference__authors_list" id="el-ol-38" data-ref="f045c82d-333d-4b0e-86a7-1ae7097c3820" data-id="el-ol-38" start="1"><li class="reference__author" id="el-li-172" data-ref="0240e283-e0de-4e14-96fb-9234bbf9a0bf" data-id="el-li-172" data-item-num="1">Minn K. T.</li><li class="reference__author" id="el-li-173" data-ref="d6921f5a-9b09-419d-b827-05e0622c6da4" data-id="el-li-173" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="ec1c7c12-d20e-4c6e-a55e-da283e57527b">2020</span><span class="reference__title" data-ref="7ff6f395-d2ce-44d9-be30-5e03440daf0a">High-resolution transcriptional and morphogenetic profiling of cells from micropatterned human ESC gastruloid cultures.</span><span class="reference__origin" data-ref="ef999619-b029-4331-867c-943f5166eee5"><i data-ref="d62e3629-6753-4d45-8ae3-e6bab6327ad9">eLife</i> <strong data-ref="3e075baf-df8e-4ac3-bda8-bc50266eb43e">9</strong></span><span class="reference__doi" data-ref="f3bfd6ff-2fec-4a5a-973d-a149c42de326"></span></li><li class="reference-list__item" id="c36" data-ref="c6b55b9c-2f85-4dff-ad1f-4b313ae698c8" data-id="c36"><span class="reference__label" data-ref="5248e8bc-8c99-4861-b24d-f8107163038b">36.</span><ol class="reference__authors_list" id="el-ol-39" data-ref="2f2583f3-d9f2-45eb-92e6-ef0225019519" data-id="el-ol-39" start="1"><li class="reference__author" id="el-li-175" data-ref="e79af441-77d3-4d6e-a8f0-9468d34eca1b" data-id="el-li-175" data-item-num="1">Minn K. T.</li><li class="reference__author" id="el-li-176" data-ref="e6d83b22-969f-43a9-80af-67f876a69ce5" data-id="el-li-176" data-item-num="2">Dietmann S.</li><li class="reference__author" id="el-li-177" data-ref="f01e8218-eeb6-4707-99cb-4027a96654b6" data-id="el-li-177" data-item-num="3">Waye S. E.</li><li class="reference__author" id="el-li-178" data-ref="02c4d3a6-bf66-4c93-9dc6-1e138aa26a5c" data-id="el-li-178" data-item-num="4">Morris S. A.</li><li class="reference__author" id="el-li-179" data-ref="c6a6b04a-a7a6-4de2-8e3e-0e99c06dce32" data-id="el-li-179" data-item-num="5">Solnica-Krezel L</li></ol><span class="reference__authors_list_suffix" data-ref="cce0b235-eb7e-464c-bf0e-58f7cecd67b3">2021</span><span class="reference__title" data-ref="2f6e73ec-6b78-4e2a-ba2e-ba2f87df2b51">Gene expression dynamics underlying cell fate emergence in 2D micropatterned human embryonic stem cell gastruloids.</span><span class="reference__origin" data-ref="f3738bc5-152e-4939-999e-62513b8abc1c"><i data-ref="326189b8-9b71-40f3-bf4d-65b863d3dae4">Stem Cell Reports</i> <strong data-ref="4fe0a5a3-a722-4667-8f4e-632ebe2077ae">16</strong>:1210‑1227</span><span class="reference__doi" data-ref="4c235129-6e55-4d61-a7bc-a8ca22cdada5"></span></li><li class="reference-list__item" id="c37" data-ref="11bba441-24f4-4739-9e5a-ef887c33c145" data-id="c37"><span class="reference__label" data-ref="baf32f48-a9a6-4043-b390-037042adba2e">37.</span><ol class="reference__authors_list" id="el-ol-40" data-ref="e77f6c0f-0b94-481e-aab5-5cc137d44703" data-id="el-ol-40" start="1"><li class="reference__author" id="el-li-181" data-ref="63243a50-a334-4974-a96a-b26493f0cbda" data-id="el-li-181" data-item-num="1">Underhill E. J.</li><li class="reference__author" id="el-li-182" data-ref="28b7013f-2e44-4d04-a982-63fe2e81fe03" data-id="el-li-182" data-item-num="2">Toettcher J. E</li></ol><span class="reference__authors_list_suffix" data-ref="62524045-5d1f-4699-92b1-b224acab2bd2">2023</span><span class="reference__title" data-ref="b39cdcce-4aee-425a-9636-c6a48fa45d0f">Control of gastruloid patterning and morphogenesis by the Erk and Akt signaling pathways.</span><span class="reference__origin" data-ref="76c56493-8e01-4698-922c-08486819277e"><i data-ref="de18bb90-df53-4f43-b10d-0fe5475ed81e">Development</i> <strong data-ref="6e085bc0-3180-4e42-aca7-85b6cff32c02">150</strong>:dev201663</span><span class="reference__doi" data-ref="ceaa957a-447a-4ae7-9f61-66cc245ce582"></span></li><li class="reference-list__item" id="c38" data-ref="0d41caf4-1948-4059-a3c1-940c3d42d157" data-id="c38"><span class="reference__label" data-ref="f5717658-b2ec-47d0-ab26-3c1eeb9aa6bf">38.</span><ol class="reference__authors_list" id="el-ol-41" data-ref="ccc29d11-35d4-49d6-9116-47ed92358fa4" data-id="el-ol-41" start="1"><li class="reference__author" id="el-li-184" data-ref="71f2da39-f064-4d2f-a287-60b68286c33b" data-id="el-li-184" data-item-num="1">Loh K. M.</li><li class="reference__author" id="el-li-185" data-ref="11f255b4-c8c8-494e-8182-ae3d3228eb66" data-id="el-li-185" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="8be1dbf1-83d8-4d28-b70f-3762c95b6a79">2016</span><span class="reference__title" data-ref="f0ce5d6d-2360-434e-a121-89516c1dbf48">Mapping the Pairwise Choices Leading from Pluripotency to Human Bone, Heart, and Other Mesoderm Cell Types.</span><span class="reference__origin" data-ref="ccf549f5-f5dd-4b42-9ddb-8b150dcb679f"><i data-ref="269d3ff2-22dc-4082-89f0-c81665fc3b54">Cell</i> <strong data-ref="e105abf9-947e-4ccd-87d8-c7a0245a6818">166</strong>:451‑467</span><span class="reference__doi" data-ref="5c1bbe70-0362-4467-a90d-0e60b199f4bd"></span></li><li class="reference-list__item" id="c39" data-ref="2d191805-f3ea-429f-b08f-6d9060c135c1" data-id="c39"><span class="reference__label" data-ref="51bbf997-daf5-40f5-9942-f7cd32e35d29">39.</span><ol class="reference__authors_list" id="el-ol-42" data-ref="55e07162-b85d-4bca-bfea-95896190ee5a" data-id="el-ol-42" start="1"><li class="reference__author" id="el-li-187" data-ref="7b722108-6bb7-402c-a117-9c51a4789f40" data-id="el-li-187" data-item-num="1">Moon K. R.</li><li class="reference__author" id="el-li-188" data-ref="7770089c-8505-4414-9586-e8677269440a" data-id="el-li-188" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="f45731c3-2c4a-48d7-8977-d33b0a791402">2019</span><span class="reference__title" data-ref="2ed5567d-5321-4faf-9af0-1f929e630403">Visualizing structure and transitions in high-dimensional biological data.</span><span class="reference__origin" data-ref="e112d0f9-d3c8-40ad-b123-ee87da53a785"><i data-ref="aca0cda4-2fe2-49ed-9a4a-a9bbcfdd08da">Nat. Biotechnol</i> <strong data-ref="d7908d15-35e3-4ecb-b509-aa65731c5869">37</strong>:1482‑1492</span><span class="reference__doi" data-ref="21bf7b1e-24da-4c31-9a5a-1a99cc5466c4"></span></li><li class="reference-list__item" id="c40" data-ref="19f6195a-fdd9-495a-84bc-bbb2ca011c9c" data-id="c40"><span class="reference__label" data-ref="713aa63f-8add-42ab-8dfc-add524c1ad19">40.</span><ol class="reference__authors_list" id="el-ol-43" data-ref="eef5dc64-d935-4730-94dc-083f705f8dd1" data-id="el-ol-43" start="1"><li class="reference__author" id="el-li-190" data-ref="a926c3ff-4af5-4567-a290-1c8371842f3a" data-id="el-li-190" data-item-num="1">Neben C. L.</li><li class="reference__author" id="el-li-191" data-ref="58d0c569-fd4a-4ab2-bc62-51427d21e8ea" data-id="el-li-191" data-item-num="2">Lo M.</li><li class="reference__author" id="el-li-192" data-ref="89d4bb0e-9df2-4942-9abe-40d8ce8a1759" data-id="el-li-192" data-item-num="3">Jura N.</li><li class="reference__author" id="el-li-193" data-ref="5e7ffd82-1a27-4b13-9260-0c6babd929a4" data-id="el-li-193" data-item-num="4">Klein O. D</li></ol><span class="reference__authors_list_suffix" data-ref="b9355595-4411-4bbb-af7a-8e8430e7bf67">2019</span><span class="reference__title" data-ref="a857dea6-1fcd-42e4-9b6a-4be231c529e5">Feedback regulation of RTK signaling in development.</span><span class="reference__origin" data-ref="6a6d93ac-0c57-4df8-a720-1e9832174538"><i data-ref="a73befa0-71f4-4ab3-a8a1-73bd1c997a41">Dev Biol</i> <strong data-ref="95392f59-9986-4e6c-85c0-cda672a66655">447</strong>:71‑89</span><span class="reference__doi" data-ref="3dc6df37-c5a0-44f0-a763-59d060c524e1"></span></li><li class="reference-list__item" id="c41" data-ref="72ab001b-fef6-4dce-bc33-c68f2d3dc431" data-id="c41"><span class="reference__label" data-ref="2b4d9ae1-173f-4ddf-9212-7d75685be866">41.</span><ol class="reference__authors_list" id="el-ol-44" data-ref="93a8262b-6d8c-4677-9c00-04cf9bb5bd4a" data-id="el-ol-44" start="1"><li class="reference__author" id="el-li-195" data-ref="b9af9d13-b447-4664-ba54-20274d967a09" data-id="el-li-195" data-item-num="1">Tyser R. C. V.</li><li class="reference__author" id="el-li-196" data-ref="4bea766d-6765-4adc-8760-536c24d26a07" data-id="el-li-196" data-item-num="2">Srinivas S</li></ol><span class="reference__authors_list_suffix" data-ref="9693f8a5-6622-45c6-ba20-174f3e10648a">2022</span><span class="reference__title" data-ref="e89cdf71-2548-47e7-890a-c565627a8527">Recent advances in understanding cell types during human gastrulation.</span><span class="reference__origin" data-ref="937edb0c-9343-4e50-959a-fd4025c5d0e9"><i data-ref="adb88c34-1873-48de-a039-e065a217a6a5">Seminars in Cell &amp; Developmental Biology</i> <strong data-ref="8fad5956-ef73-4c5b-817c-1e3db9e1db02">131</strong>:35‑43</span><span class="reference__doi" data-ref="8e15e8d2-149f-4392-b7d3-38797268d858"></span></li><li class="reference-list__item" id="c42" data-ref="3efe6edb-062b-4c6b-9deb-c2050a37fa32" data-id="c42"><span class="reference__label" data-ref="8c149b6d-99b1-4f9a-b9c0-119554bc52c6">42.</span><ol class="reference__authors_list" id="el-ol-45" data-ref="f7f061f7-56c3-4a21-9210-c838d1ea9c3f" data-id="el-ol-45" start="1"><li class="reference__author" id="el-li-198" data-ref="49df7eb0-6f97-449e-8710-1d32a6baa377" data-id="el-li-198" data-item-num="1">Chen B.</li><li class="reference__author" id="el-li-199" data-ref="4efe258c-91a8-4f5a-a145-79a920863d9d" data-id="el-li-199" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="64dc1472-c9d4-44a8-bee7-6708b5d21b6f">2024</span><span class="reference__title" data-ref="410fd4b4-1963-4106-a5df-5eeaae6ea4ec">Extended culture of 2D gastruloids to model human mesoderm development.</span><span class="reference__origin" data-ref="d26cf0b9-8214-477f-a15e-fa66bab45c95"><i data-ref="403705e8-914d-4553-a74b-4a9faa065285">bioRxiv</i> </span><span class="reference__doi" data-ref="0ad5535e-edda-42b7-af6d-49361a1b758d"><a href="https://doi.org/10.1101/2024.03.21.585753" class="reference__doi_link" data-ref="1610d047-9f29-45ef-8ef6-023642d7c9c2">https://doi.org/10.1101/2024.03.21.585753</a></span></li><li class="reference-list__item" id="c43" data-ref="54f285ad-1597-4440-b0da-c77b825d6102" data-id="c43"><span class="reference__label" data-ref="d37d1759-5146-4fbc-8c81-033da4e27932">43.</span><ol class="reference__authors_list" id="el-ol-46" data-ref="046ab2ff-782e-4e81-84dc-f052067dec84" data-id="el-ol-46" start="1"><li class="reference__author" id="el-li-201" data-ref="4b6b5aab-c9f9-47c4-8aba-100e8237bbdc" data-id="el-li-201" data-item-num="1">Khoa L. T. P.</li><li class="reference__author" id="el-li-202" data-ref="4e097b10-a7ab-411d-9e2c-b6543d96d0ff" data-id="el-li-202" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="861b93a6-91cd-4223-b6a4-ead9c67b8a25">2016</span><span class="reference__title" data-ref="a4266ab4-37f0-40f1-b348-75736eff3da5">Visualization of the Epiblast and Visceral Endodermal Cells Using Fgf5-P2A-Venus BAC Transgenic Mice and Epiblast Stem Cells.</span><span class="reference__origin" data-ref="3df91b44-c53c-4c66-a5f2-2550d8713af1"><i data-ref="7cd6fbb2-acb9-49a6-bceb-b2575e924e44">PLOS One</i> <strong data-ref="ddfbe828-3379-4929-b910-1ee3385947ff">11</strong>:e0159246</span><span class="reference__doi" data-ref="4912cadd-eed1-4c0f-9922-e5c5acccbb1a"></span></li><li class="reference-list__item" id="c44" data-ref="cd1cffef-3465-469d-bc3a-32ba2a7bba0a" data-id="c44"><span class="reference__label" data-ref="7ac18599-c754-4e45-9d3d-dd73b81e8a5e">44.</span><ol class="reference__authors_list" id="el-ol-47" data-ref="ea16c466-05cf-437b-8f30-f3cc65729fe3" data-id="el-ol-47" start="1"><li class="reference__author" id="el-li-204" data-ref="baaf3b26-8a74-4705-9a67-ccaa2542d700" data-id="el-li-204" data-item-num="1">Pijuan-Sala B.</li><li class="reference__author" id="el-li-205" data-ref="9b237799-a0fc-4154-975b-00dbbdd49fbc" data-id="el-li-205" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="2379be92-0d9c-412c-a11f-db889cd89b42">2019</span><span class="reference__title" data-ref="4173fddc-1565-49b2-8779-ecd34e254d2d">A single-cell molecular map of mouse gastrulation and early organogenesis.</span><span class="reference__origin" data-ref="78e7bab8-d10a-49b7-ab56-c12999dfe80b"><i data-ref="580555a2-ea30-4d63-8842-39b812445bff">Nature</i> <strong data-ref="3a48893e-a144-4d35-98dd-e9a1ffd83d6a">566</strong>:490‑495</span><span class="reference__doi" data-ref="74220029-6e50-4977-b67a-7ee1a1e89430"></span></li><li class="reference-list__item" id="c45" data-ref="2b867549-8731-4d3a-a056-36b7734df071" data-id="c45"><span class="reference__label" data-ref="58b99cb6-21a4-42d1-a3c2-18a4fc3146dc">45.</span><ol class="reference__authors_list" id="el-ol-48" data-ref="292d4ccc-978e-45dd-a503-43af62a0a58b" data-id="el-ol-48" start="1"><li class="reference__author" id="el-li-207" data-ref="dcc5844e-5238-4414-ae83-23520ef14254" data-id="el-li-207" data-item-num="1">Morgani S. M.</li><li class="reference__author" id="el-li-208" data-ref="5b2f2346-a51f-4130-a175-c0d199280c2c" data-id="el-li-208" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="885378de-078f-4eb1-a1f1-2c182012a8b5">2018</span><span class="reference__title" data-ref="4559ac85-1f0f-4194-b52d-8982bb86d244">A Sprouty4 reporter to monitor FGF/ERK signaling activity in ESCs and mice.</span><span class="reference__origin" data-ref="aac1d037-4e14-4bb5-80a7-a29575310114"><i data-ref="4803e476-0bc8-4096-a071-44871a1b428a">Developmental Biology</i> <strong data-ref="a91b4cc0-1cbf-41f8-b9b0-a0ac0ec0749f">441</strong>:104‑126</span><span class="reference__doi" data-ref="f22288bc-f1fa-4f28-afc9-0738f6f44102"></span></li><li class="reference-list__item" id="c46" data-ref="dadf73b5-e341-4f58-bbe8-b48d008950b8" data-id="c46"><span class="reference__label" data-ref="98498c9e-cef4-4dd1-9547-3393234391ec">46.</span><ol class="reference__authors_list" id="el-ol-49" data-ref="7353d557-4616-4a81-bac0-4bf44323a910" data-id="el-ol-49" start="1"><li class="reference__author" id="el-li-210" data-ref="a0b9fa66-0809-4419-8c52-f71a45cd23bb" data-id="el-li-210" data-item-num="1">Xu D.</li><li class="reference__author" id="el-li-211" data-ref="a89d8ff0-f87c-4022-988c-00e92910361f" data-id="el-li-211" data-item-num="2">Esko J. D</li></ol><span class="reference__authors_list_suffix" data-ref="6ec5bbb0-fef7-4043-a572-b19c9eba6e9c">2014</span><span class="reference__title" data-ref="0ec2468c-a770-4c84-a1b8-4fac10ff679d">Demystifying heparan sulfate-protein interactions.</span><span class="reference__origin" data-ref="3e36d7b9-1fc2-42e5-9ba9-2b7a8cb7254b"><i data-ref="85a3d7b5-770f-44f7-8246-e59d163bb69f">Annu Rev Biochem</i> <strong data-ref="19b4da0e-e2ae-4e8f-a5b9-a0d320c09172">83</strong>:129‑157</span><span class="reference__doi" data-ref="db20e079-18ca-469b-99ff-cd49470fa44e"></span></li><li class="reference-list__item" id="c47" data-ref="6a663f23-eac5-41db-9107-0815ed0cc3f5" data-id="c47"><span class="reference__label" data-ref="7895fdc3-5ff6-4d59-b5c0-6c448fafdbb2">47.</span><ol class="reference__authors_list" id="el-ol-50" data-ref="2a684471-51a0-46ab-bacb-47363c74ddcd" data-id="el-ol-50" start="1"><li class="reference__author" id="el-li-213" data-ref="f39dc18a-de8a-421f-9647-41c54969cde2" data-id="el-li-213" data-item-num="1">Habuchi H.</li><li class="reference__author" id="el-li-214" data-ref="3fee3674-66b8-4f0f-98dd-5efa22e04f59" data-id="el-li-214" data-item-num="2">Kimata K</li></ol><span class="reference__authors_list_suffix" data-ref="b6532cb2-7dfb-42e4-8218-9323f9114978">2008</span><span class="reference__title" data-ref="ee388bcc-5feb-42b7-9d88-205f24502666">Functions of Heparan Sulfate Proteoglycans in Morphogenesis.</span><span class="reference__origin" data-ref="d1d1d7ed-d1b3-437e-95ba-9e21cebf25b8">In: <ol class="reference__editors_list" id="el-ol-51" data-ref="fe779cc2-2a71-4d58-977a-bb11f146810d" data-id="el-ol-51" start="1"><li class="reference__editor" id="el-li-215" data-ref="0ff89d59-4565-40c2-9798-2b41746b861c" data-id="el-li-215" data-item-num="1">Taniguchi N.</li><li class="reference__editor" id="el-li-216" data-ref="3d1c31ef-25b6-44c2-83e6-ba39eb868193" data-id="el-li-216" data-item-num="2">Suzuki A</li><li class="reference__editor" id="el-li-217" data-ref="a4f3428f-fe2f-4d29-8791-2ac8675e79b9" data-id="el-li-217" data-item-num="3">Ito Y</li><li class="reference__editor" id="el-li-218" data-ref="96993f10-1c2a-4a50-9817-9d7a0aab3b0d" data-id="el-li-218" data-item-num="4">Narimatsu H</li><li class="reference__editor" id="el-li-219" data-ref="704662df-9e0d-4021-bb83-afbc12f9c85b" data-id="el-li-219" data-item-num="5">Kawasaki T</li><li class="reference__editor" id="el-li-220" data-ref="f1103afd-9d75-4b55-80d4-fdea68e885c3" data-id="el-li-220" data-item-num="6">Hase S</li></ol> (Eds). <i data-ref="4730504f-86c9-47ac-bbc5-0409e10413dc">Experimental Glycoscience</i> Tokyo: Springer. pp.&nbsp;303‑308</span><span class="reference__doi" data-ref="3de4356b-00d9-4fec-9a34-dc1d300f8e7e"><a href="https://doi.org/10.1007/978-4-431-77922-3_72" class="reference__doi_link" data-ref="0f5eeb97-1ead-48a6-a360-70014ead596f">https://doi.org/10.1007/978-4-431-77922-3_72</a></span></li><li class="reference-list__item" id="c48" data-ref="1dbd441c-1407-4515-a789-183cd324b7e3" data-id="c48"><span class="reference__label" data-ref="1c9aa9f0-16b7-47d0-8bae-92421f418ca2">48.</span><ol class="reference__authors_list" id="el-ol-52" data-ref="728ee662-8509-4d40-aa91-f9c510568ace" data-id="el-ol-52" start="1"><li class="reference__author" id="el-li-222" data-ref="50806837-1c73-4fdb-91aa-c2bd403f7430" data-id="el-li-222" data-item-num="1">Yu S. R.</li><li class="reference__author" id="el-li-223" data-ref="e26445ee-1ee0-4ebd-84c6-1cd3e97ba557" data-id="el-li-223" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="44ed28a1-1eda-4b31-9733-6eb2047c9dcf">2009</span><span class="reference__title" data-ref="92dac5e4-f985-4983-a5d7-f3182517459c">Fgf8 morphogen gradient forms by a source-sink mechanism with freely diffusing molecules.</span><span class="reference__origin" data-ref="54a929d2-4b8a-406e-921b-cefba63e89c4"><i data-ref="695acd2a-fb54-4bf5-9f9c-39afaa809fff">Nature</i> <strong data-ref="292f3762-da76-4520-ac21-e50b2d49be6f">461</strong>:533‑536</span><span class="reference__doi" data-ref="6ecf5029-cdfe-42a5-a392-7a2925538de5"></span></li><li class="reference-list__item" id="c49" data-ref="b31aa406-e64b-4625-b4c8-1999f2528f6c" data-id="c49"><span class="reference__label" data-ref="eaa918d2-2f68-4886-8172-abb31ac73e89">49.</span><ol class="reference__authors_list" id="el-ol-53" data-ref="23fa2508-81af-439a-9ad5-ea9da0c710c8" data-id="el-ol-53" start="1"><li class="reference__author" id="el-li-225" data-ref="d840b7c0-7726-4d71-a2fc-69f26f56aaf5" data-id="el-li-225" data-item-num="1">Benington L.</li><li class="reference__author" id="el-li-226" data-ref="8fe08a4c-9043-42d8-9c41-dbba58f47bdb" data-id="el-li-226" data-item-num="2">Rajan G.</li><li class="reference__author" id="el-li-227" data-ref="986d5ff7-9ec7-4b85-b133-fe665617c2fe" data-id="el-li-227" data-item-num="3">Locher C.</li><li class="reference__author" id="el-li-228" data-ref="cd7efee6-dd34-4153-ae31-bf6742ecee5c" data-id="el-li-228" data-item-num="4">Lim L. Y</li></ol><span class="reference__authors_list_suffix" data-ref="ceaba6dd-1e6d-43f1-bd48-9c7a72769806">2020</span><span class="reference__title" data-ref="84c3d822-675c-4428-a8af-0e7a957c5452">Fibroblast Growth Factor 2-A Review of Stabilisation Approaches for Clinical Applications.</span><span class="reference__origin" data-ref="2c8e3107-300e-4b54-a4be-1da9ce15aa82"><i data-ref="73ebf559-543f-4494-9927-6130e4c49143">Pharmaceutics</i> <strong data-ref="329d121e-4c76-4c2c-869b-a390dc318a76">12</strong>:508</span><span class="reference__doi" data-ref="32c77416-974a-4ebd-890b-5259db4c395d"></span></li><li class="reference-list__item" id="c50" data-ref="8a9d2a8f-ff49-4d19-bd43-9c311557f460" data-id="c50"><span class="reference__label" data-ref="a6b59077-8e2e-46ed-a28c-bdf1ed8a2201">50.</span><ol class="reference__authors_list" id="el-ol-54" data-ref="5429e575-aeb2-4dbd-9a36-455e0bad8098" data-id="el-ol-54" start="1"><li class="reference__author" id="el-li-230" data-ref="aaa5ba57-0b06-4ac9-bea9-49240940a7bf" data-id="el-li-230" data-item-num="1">Teague S.</li><li class="reference__author" id="el-li-231" data-ref="7eb2143c-ba1b-4218-a49c-e680aeb02fbb" data-id="el-li-231" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="bb55dc77-838a-412f-a10a-b25f1c5bb665">2024</span><span class="reference__title" data-ref="2c214e39-d944-4060-bc79-b8f20df145dd">Time-integrated BMP signaling determines fate in a stem cell model for early human development.</span><span class="reference__origin" data-ref="f730eba6-146a-49fa-b5bd-2d482a40caa3"><i data-ref="5621b7ab-f993-4b46-89bd-397ed4807ac4">Nat Commun</i> <strong data-ref="4803b541-8393-4556-82bd-b8b90a5d85f3">15</strong>:1471</span><span class="reference__doi" data-ref="d41bda24-fbe1-4009-9b75-1fbe76a812a3"></span></li><li class="reference-list__item" id="c51" data-ref="b4e6164c-9fa6-4c3f-b382-00e6a03f5f19" data-id="c51"><span class="reference__label" data-ref="de07cb49-30d4-40b3-9f7b-50e36d606083">51.</span><ol class="reference__authors_list" id="el-ol-55" data-ref="4e52311c-9cc8-4ad3-a3c9-db816c8c16b9" data-id="el-ol-55" start="1"><li class="reference__author" id="el-li-233" data-ref="e78cbdab-1acf-481c-bb5f-ab0808bdc515" data-id="el-li-233" data-item-num="1">Ortiz-Salazar M. A.</li><li class="reference__author" id="el-li-234" data-ref="df6ebb31-c8da-4a54-b29c-10268157096e" data-id="el-li-234" data-item-num="2">Camacho-Aguilar E.</li><li class="reference__author" id="el-li-235" data-ref="02e67fdd-a41f-4085-b115-430a91b2393b" data-id="el-li-235" data-item-num="3">Warmflash A.</li></ol><span class="reference__authors_list_suffix" data-ref="55ac6e65-b958-4d09-a861-2c38e114595c">2024</span><span class="reference__title" data-ref="fce24243-75a5-49ac-b9f6-dee3e52942bf">Endogenous Nodal switches Wnt interpretation from posteriorization to germ layer differentiation in geometrically constrained human pluripotent cells.</span><span class="reference__origin" data-ref="81bc376b-c687-448e-8ddb-4a61ec6287ff"><i data-ref="b4225665-52fd-4d7f-85f9-530c2af2ce67">bioRxiv</i> </span><span class="reference__doi" data-ref="e43257fe-5aa4-44ea-a786-78c9c5e32240"><a href="https://doi.org/10.1101/2024.03.13.584912" class="reference__doi_link" data-ref="cfd04c16-79fe-4c08-a236-325372702258">https://doi.org/10.1101/2024.03.13.584912</a></span></li><li class="reference-list__item" id="c52" data-ref="680033bb-d61c-4ccd-946d-9d9f9a1bdcb7" data-id="c52"><span class="reference__label" data-ref="312f434b-9f25-468a-a4d0-5a59582c6737">52.</span><ol class="reference__authors_list" id="el-ol-56" data-ref="0a70bea1-b0af-488a-b173-3d451fb81922" data-id="el-ol-56" start="1"><li class="reference__author" id="el-li-237" data-ref="8d9b415e-49e8-44cf-9524-f4a019d8892c" data-id="el-li-237" data-item-num="1">Gurdon J. B</li></ol><span class="reference__authors_list_suffix" data-ref="ff51423f-2c92-4627-b2a0-21c6ba2fb50b">1988</span><span class="reference__title" data-ref="bf3f5247-6a7c-41c1-8d89-ba778d3a445a">A community effect in animal development.</span><span class="reference__origin" data-ref="2ca140ba-8b2f-4049-a80c-0d1f61e17deb"><i data-ref="753187f4-4c2a-4542-99e0-89e2240dfada">Nature</i> <strong data-ref="ac912784-bcf3-4c4f-9b28-82dcf6413e7f">336</strong>:772‑774</span><span class="reference__doi" data-ref="81cfef9e-c382-43ca-ae78-fcd34e71d99f"></span></li></ul></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_right_page" data-page-number="31" id="page-31">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e" data-split-to="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-id="article-content" data-split-from="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-split-to="726f25f1-fdd1-41b8-bf49-de736be3e3ae"><section data-ref="7921c412-ef92-4490-b273-a1fbe0815a2b" data-id="references" data-split-from="7921c412-ef92-4490-b273-a1fbe0815a2b" data-split-to="7921c412-ef92-4490-b273-a1fbe0815a2b"><ul class="reference-list" data-ref="5d1d46dc-2209-4116-9f07-193d224fefb4" data-id="el-ul-6" data-split-from="5d1d46dc-2209-4116-9f07-193d224fefb4" data-split-to="5d1d46dc-2209-4116-9f07-193d224fefb4" data-last-split-element="true" data-align-last-split-element="auto"><li class="reference-list__item" id="c53" data-ref="f62295b5-84dd-4acb-841f-edad92bb6922" data-id="c53"><span class="reference__label" data-ref="b4813a9b-3218-4362-bbfd-88bbed8b166f">53.</span><ol class="reference__authors_list" id="el-ol-57" data-ref="a74dfcc8-7fd6-4a17-8ca6-0b70693b608e" data-id="el-ol-57" start="1"><li class="reference__author" id="el-li-239" data-ref="03de9c15-69db-4269-b1b4-cc67ce5feefa" data-id="el-li-239" data-item-num="1">Nemashkalo A.</li><li class="reference__author" id="el-li-240" data-ref="473bff86-18e6-406f-96b9-28472fd70e61" data-id="el-li-240" data-item-num="2">Ruzo A.</li><li class="reference__author" id="el-li-241" data-ref="388e4ff1-4672-48fe-8a75-f7a73477f31a" data-id="el-li-241" data-item-num="3">Heemskerk I.</li><li class="reference__author" id="el-li-242" data-ref="2b727048-072c-403c-85a3-b6399ddd4317" data-id="el-li-242" data-item-num="4">Warmflash A</li></ol><span class="reference__authors_list_suffix" data-ref="28d13e14-a293-4d86-994f-ec1d2f065052">2017</span><span class="reference__title" data-ref="dc8d06f8-ff8b-41fb-bea8-08c980f306ca">Morphogen and community effects determine cell fates in response to BMP4 signaling in human embryonic stem cells.</span><span class="reference__origin" data-ref="c03e435d-0892-4017-8187-ed8a437d061b"><i data-ref="53be5179-c0f8-4049-92cb-b5d3864bcfa6">Development</i> <strong data-ref="2449423e-f271-40bb-bf88-aad8792de567">144</strong>:3042‑3053</span><span class="reference__doi" data-ref="1ec3ec1c-be5b-4d68-9eb8-b1c8c480e7a5"></span></li><li class="reference-list__item" id="c54" data-ref="65629743-e087-4bb2-9ac3-c11a3048a77a" data-id="c54"><span class="reference__label" data-ref="fc394fc0-f9a8-464d-a7ae-1720db28212a">54.</span><ol class="reference__authors_list" id="el-ol-58" data-ref="259556a8-c18f-4b2f-bff0-79478ce3f78f" data-id="el-ol-58" start="1"><li class="reference__author" id="el-li-244" data-ref="b8f967f0-fa68-48b0-ad94-55b9c8f2e719" data-id="el-li-244" data-item-num="1">Gattiglio M.</li><li class="reference__author" id="el-li-245" data-ref="266cec7e-8a37-46bc-a95f-9d8156b71c20" data-id="el-li-245" data-item-num="2">Protzek M.</li><li class="reference__author" id="el-li-246" data-ref="99e343b4-0d65-40e3-9f36-7317e0846781" data-id="el-li-246" data-item-num="3">Schröter C</li></ol><span class="reference__authors_list_suffix" data-ref="47638349-668b-4329-b862-e5e3d451dc67">2023</span><span class="reference__title" data-ref="31084244-5ce8-4d3f-9253-8ecbdc17f5e7">Population-level antagonism between FGF and BMP signaling steers mesoderm differentiation in embryonic stem cells.</span><span class="reference__origin" data-ref="01acaae5-81c6-4a18-83af-c5eff5cf0c0a"><i data-ref="d97d060d-6b88-49af-bd15-12e3c86d1c76">Biology Open</i> <strong data-ref="6abedc35-4273-4977-905f-4f9a1f460681">12</strong>:bio059941</span><span class="reference__doi" data-ref="c8c60c9e-a5b8-4165-ac0a-8ab7c884ffa0"></span></li><li class="reference-list__item" id="c55" data-ref="6eca9cbe-c63b-4d6c-9ce8-e88070789c07" data-id="c55"><span class="reference__label" data-ref="0858cf77-e5e5-46b5-a134-78f53173701d">55.</span><ol class="reference__authors_list" id="el-ol-59" data-ref="11466b86-6d4f-43f6-8285-16c73e42bdc0" data-id="el-ol-59" start="1"><li class="reference__author" id="el-li-248" data-ref="743a9018-8e3c-4373-b219-879267898631" data-id="el-li-248" data-item-num="1">Liu L.</li><li class="reference__author" id="el-li-249" data-ref="64b5e769-5a49-4fde-b831-7dfa2743a914" data-id="el-li-249" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="427cb68f-9f13-48d0-8aab-6deed9dc7c19">2022</span><span class="reference__title" data-ref="7d9b4b20-0265-4105-bb81-93532ba98433">Nodal is a short-range morphogen with activity that spreads through a relay mechanism in human gastruloids.</span><span class="reference__origin" data-ref="b5dfed1f-7c2c-40e1-beda-1fb87b57b071"><i data-ref="a191107c-459e-4479-9d90-e2195d578b80">Nat Commun</i> <strong data-ref="94a1971b-b63a-433c-aa2f-df70614cf540">13</strong>:497</span><span class="reference__doi" data-ref="33b7f532-bcc3-4a31-a97b-de72f8aa7b54"></span></li><li class="reference-list__item" id="c56" data-ref="ed8b8f8f-0cf0-4e7c-81e0-729153583162" data-id="c56"><span class="reference__label" data-ref="b4433a4b-3967-4064-b527-036703619fe2">56.</span><ol class="reference__authors_list" id="el-ol-60" data-ref="6cecf0b2-2dc7-44c8-8394-601a641226b1" data-id="el-ol-60" start="1"><li class="reference__author" id="el-li-251" data-ref="90168859-12af-4242-8fc3-232a2b59ff0d" data-id="el-li-251" data-item-num="1">Isaacs H. V.</li><li class="reference__author" id="el-li-252" data-ref="c35f54c0-e208-459a-84ab-5be447d4ca18" data-id="el-li-252" data-item-num="2">Pownall M. E.</li><li class="reference__author" id="el-li-253" data-ref="6f525a0a-53d6-4cec-b9c6-062160bc13f2" data-id="el-li-253" data-item-num="3">Slack J. M</li></ol><span class="reference__authors_list_suffix" data-ref="99538cdd-04a8-4ddd-9b29-f4407dfc42f8">1995</span><span class="reference__title" data-ref="3659dbc0-045c-4a57-9ea6-7f3bb146254a">eFGF is expressed in the dorsal midline of Xenopus laevis.</span><span class="reference__origin" data-ref="66f1d50a-916b-45b2-9ef0-db21d9df8205"><i data-ref="558384b2-af26-4633-b559-826e717b2c3f">Int J Dev Biol</i> <strong data-ref="d1a8317a-6de5-4206-a888-9b06272d409c">39</strong>:575‑579</span><span class="reference__doi" data-ref="35412053-5cc0-4f3f-8409-a2cdc8a912a8"></span></li><li class="reference-list__item" id="c57" data-ref="c1bc3ec9-de10-483b-944d-ae2b74902488" data-id="c57"><span class="reference__label" data-ref="c369f5f7-4934-4acf-bfd6-47928acb2003">57.</span><ol class="reference__authors_list" id="el-ol-61" data-ref="a7a94a3d-e69f-459a-abb3-17a7e7719197" data-id="el-ol-61" start="1"><li class="reference__author" id="el-li-255" data-ref="89cf23e0-2e89-4dad-a1a6-c27bf628a9e0" data-id="el-li-255" data-item-num="1">Christen B.</li><li class="reference__author" id="el-li-256" data-ref="0558b21e-7576-4bfa-b0c4-0a6b29f0ac3f" data-id="el-li-256" data-item-num="2">Slack J. M</li></ol><span class="reference__authors_list_suffix" data-ref="c905c7ea-9717-4c51-adec-2cfc01f29ef4">1997</span><span class="reference__title" data-ref="45be7a43-92f3-4c12-b489-f7fd0726c807">FGF-8 is associated with anteroposterior patterning and limb regeneration in Xenopus.</span><span class="reference__origin" data-ref="7f86557b-9121-462c-8eba-2a61a381b318"><i data-ref="4cfb7ea4-d62e-40b3-bc22-313efb8d8ff1">Dev Biol</i> <strong data-ref="cdb8a68c-037c-4469-a07c-1e7271041abf">192</strong>:455‑466</span><span class="reference__doi" data-ref="5b76c58f-47f1-4c81-932e-a6216ebb3bb9"></span></li><li class="reference-list__item" id="c58" data-ref="6faac146-9bc3-43ac-a40c-2c2359c7d701" data-id="c58"><span class="reference__label" data-ref="71f2d922-3319-4269-bef3-88642017767c">58.</span><ol class="reference__authors_list" id="el-ol-62" data-ref="9b7d2a59-e534-493a-a5ea-2ff9fe460dbd" data-id="el-ol-62" start="1"><li class="reference__author" id="el-li-258" data-ref="a6ae5db3-cc52-47d0-ac7b-f23c48010218" data-id="el-li-258" data-item-num="1">Hardy K. M.</li><li class="reference__author" id="el-li-259" data-ref="8c7f35dd-ee24-4044-9516-ee987e6c0d9e" data-id="el-li-259" data-item-num="2">Yatskievych T. A.</li><li class="reference__author" id="el-li-260" data-ref="766b8e5d-02ab-46b3-8d02-44b1e750d7e9" data-id="el-li-260" data-item-num="3">Konieczka J.</li><li class="reference__author" id="el-li-261" data-ref="b0d22ed8-7faa-4c6e-bc74-e58ebbdc7f9e" data-id="el-li-261" data-item-num="4">Bobbs A. S.</li><li class="reference__author" id="el-li-262" data-ref="066e81d0-5847-4209-9d6c-bbd18e842f86" data-id="el-li-262" data-item-num="5">Antin P. B</li></ol><span class="reference__authors_list_suffix" data-ref="42cb7a8a-8a11-4d88-b136-fc2390535322">2011</span><span class="reference__title" data-ref="686f9dc2-037c-40d3-810f-72fece9d446e">FGF signalling through RAS/MAPK and PI3K pathways regulates cell movement and gene expression in the chicken primitive streak without affecting E-cadherin expression.</span><span class="reference__origin" data-ref="9c90fc83-da92-45b6-9dc8-bac636b7cd05"><i data-ref="3969b195-16ab-44c3-8e72-5f4ecc52ef35">BMC Dev Biol</i> <strong data-ref="0b19341d-bd5a-4bf6-ae8a-99dbc73466d3">11</strong>:20‑17</span><span class="reference__doi" data-ref="478d7b4c-6d3e-4b08-aab6-7d32d04cb7a4"></span></li><li class="reference-list__item" id="c59" data-ref="7f40cb35-ffc8-4f3d-933e-adbc9b5887be" data-id="c59"><span class="reference__label" data-ref="5de97f2b-dbe5-4946-8ff6-faf4e61c88d0">59.</span><ol class="reference__authors_list" id="el-ol-63" data-ref="58a76943-5ed7-4237-9202-25b2431620ce" data-id="el-ol-63" start="1"><li class="reference__author" id="el-li-264" data-ref="a2b03847-c22b-4549-90fc-9ddeff6cc628" data-id="el-li-264" data-item-num="1">Isaacs H. V.</li><li class="reference__author" id="el-li-265" data-ref="2f229799-1af8-4e7e-b386-9a5a8f3a0642" data-id="el-li-265" data-item-num="2">Pownall M. E.</li><li class="reference__author" id="el-li-266" data-ref="2c9c466c-8d76-4011-a07c-adc81b03f4f1" data-id="el-li-266" data-item-num="3">Slack J. M</li></ol><span class="reference__authors_list_suffix" data-ref="b99d132a-02a5-4760-84e2-61e52283bccc">1994</span><span class="reference__title" data-ref="b0c617d9-0096-4ac3-941e-24e933efe401">eFGF regulates Xbra expression during Xenopus gastrulation.</span><span class="reference__origin" data-ref="72e919ee-e587-4334-8cb9-40f1133024c4"><i data-ref="a6e89f6a-83ac-4cce-8c99-ac16982cbecd">The EMBO Journal</i> <strong data-ref="0017c692-e203-4019-a7c6-a2201f591419">13</strong>:4469‑4481</span><span class="reference__doi" data-ref="d2c992e8-364e-4c5b-be4e-d68f0d8bf5c3"></span></li><li class="reference-list__item" id="c60" data-ref="8147b550-9de1-42b3-9397-60994b795490" data-id="c60"><span class="reference__label" data-ref="f7e9b0d0-a44a-49d6-ae90-e75399826055">60.</span><ol class="reference__authors_list" id="el-ol-64" data-ref="0fe8806f-4ca9-457e-a4f4-fef3e7cd45f8" data-id="el-ol-64" start="1"><li class="reference__author" id="el-li-268" data-ref="a47d908b-a6a6-4f42-ac4f-2d4c1d85a558" data-id="el-li-268" data-item-num="1">Bulger E. A.</li><li class="reference__author" id="el-li-269" data-ref="d99ed458-d744-4a6d-afa4-4c0224f54bab" data-id="el-li-269" data-item-num="2">Muncie-Vasic I.</li><li class="reference__author" id="el-li-270" data-ref="da28f9a4-d15b-49b1-9fe2-3fe510de360f" data-id="el-li-270" data-item-num="3">Libby A. R. G.</li><li class="reference__author" id="el-li-271" data-ref="74880a3b-9fe2-4f66-b158-1ff093487650" data-id="el-li-271" data-item-num="4">McDevitt T. C.</li><li class="reference__author" id="el-li-272" data-ref="f8615346-9543-42e6-a77b-882c11ed8ea5" data-id="el-li-272" data-item-num="5">Bruneau B. G</li></ol><span class="reference__authors_list_suffix" data-ref="72f01b27-9aae-4283-99ba-ed4823752c95">2024</span><span class="reference__title" data-ref="2392d4d1-ec44-4379-9fc9-131b3ef65b1f">TBXT dose sensitivity and the decoupling of nascent mesoderm specification from EMT progression in 2D human gastruloids.</span><span class="reference__origin" data-ref="3780c2de-b953-41d6-a389-b1c1aaa68efe"><i data-ref="91c68141-039b-4ec0-a2c3-84e72694096b">Development</i> <strong data-ref="672505b0-ecdb-4d56-818e-1c0a0f07e268">151</strong>:dev202516</span><span class="reference__doi" data-ref="cff2cf03-c45f-433e-8da8-0b0207ca0423"></span></li><li class="reference-list__item" id="c61" data-ref="1fd5a420-8d2b-48fd-8d03-7d4758e7d740" data-id="c61"><span class="reference__label" data-ref="abee5970-7902-4414-b854-2d551e629335">61.</span><ol class="reference__authors_list" id="el-ol-65" data-ref="c726937e-8fed-48fb-bd0a-4385fc659327" data-id="el-ol-65" start="1"><li class="reference__author" id="el-li-274" data-ref="51fc5af9-e634-4bb4-8fc8-0b02093473dc" data-id="el-li-274" data-item-num="1">Evans A. L.</li><li class="reference__author" id="el-li-275" data-ref="ffa8afe2-92c0-4020-97c5-ea3a1ed3fd5c" data-id="el-li-275" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="17904a7c-d9f8-4446-bbfc-65b274876f76">2012</span><span class="reference__title" data-ref="8aa664d7-709d-4d55-a752-7464ae0d02cb">Genomic Targets of Brachyury (T) in Differentiating Mouse Embryonic Stem Cells.</span><span class="reference__origin" data-ref="f291c13d-ea67-4601-b238-03bcc8d75f07"><i data-ref="977ae48a-233a-45c6-afdc-a47c57b5255a">PLOS One</i> <strong data-ref="65e2f9dc-1c48-4714-94be-2a4476fa5547">7</strong>:e33346</span><span class="reference__doi" data-ref="be0de8f5-71ab-412b-8b7c-16ab0c05e933"></span></li><li class="reference-list__item" id="c62" data-ref="bd2d60c5-a506-4586-86df-89f4d7e5c9f5" data-id="c62"><span class="reference__label" data-ref="60f7311c-8c92-4ca2-9b1d-457e30b92f9d">62.</span><ol class="reference__authors_list" id="el-ol-66" data-ref="2263d3b3-5249-43ca-b1c4-93fb45bea380" data-id="el-ol-66" start="1"><li class="reference__author" id="el-li-277" data-ref="dc6b3f36-4577-4460-ba8f-9ca2ef8c4bba" data-id="el-li-277" data-item-num="1">Nie Y.</li><li class="reference__author" id="el-li-278" data-ref="70ed5347-7a9b-46b1-a761-a85232632604" data-id="el-li-278" data-item-num="2">Walsh P.</li><li class="reference__author" id="el-li-279" data-ref="e5914d9c-6429-45dc-90a8-5bc9f8487699" data-id="el-li-279" data-item-num="3">Clarke D. L.</li><li class="reference__author" id="el-li-280" data-ref="79159ddc-61c5-4239-9e93-ce9dd4c7d492" data-id="el-li-280" data-item-num="4">Rowley J. A.</li><li class="reference__author" id="el-li-281" data-ref="359851c6-7414-4311-ab08-261f3244ff4d" data-id="el-li-281" data-item-num="5">Fellner T</li></ol><span class="reference__authors_list_suffix" data-ref="9eb1f09e-a2b6-4e9f-80ab-890b5723d50e">2014</span><span class="reference__title" data-ref="1f299ce6-023a-4236-ad76-8a032715f615">Scalable passaging of adherent human pluripotent stem cells.</span><span class="reference__origin" data-ref="3aceb9ac-bc00-47ee-a651-4edb2ef6c3ab"><i data-ref="3b76e53b-60c4-4927-b90c-e23442c53ff8">PLoS ONE</i> <strong data-ref="b9c8cf73-ce67-4a3e-9a63-fb1eee86015b">9</strong>:e88012</span><span class="reference__doi" data-ref="ce105922-5afb-4723-ad3f-54a26cb279d3"></span></li><li class="reference-list__item" id="c63" data-ref="bd40cef0-e241-4a79-b602-7c310ccc64f9" data-id="c63"><span class="reference__label" data-ref="6545b26a-3f23-4fbd-a195-be0d9b3c9a4f">63.</span><ol class="reference__authors_list" id="el-ol-67" data-ref="4d72085c-de0c-4c6b-9725-b462e8407401" data-id="el-ol-67" start="1"><li class="reference__author" id="el-li-283" data-ref="ac853476-ac37-4aec-8e09-ef07e1133189" data-id="el-li-283" data-item-num="1">Freeburne E.</li><li class="reference__author" id="el-li-284" data-ref="de71f5a3-a5f5-41cc-b39f-65e464c424b2" data-id="el-li-284" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="ab1e7ad7-5f9e-4393-a49b-052d7c50a192">2023</span><span class="reference__title" data-ref="6b44c700-0975-488e-ad8a-ea10d8f724a0">Spatial Single Cell Analysis of Proteins in 2D Human Gastruloids Using Iterative Immunofluorescence.</span><span class="reference__origin" data-ref="86766b77-6d55-47fb-bac6-7500df42af24"><i data-ref="bfa28c6e-6888-441f-b809-2faba33f5f8b">Curr Protoc</i> <strong data-ref="b16887e7-d5fa-4d64-bbf7-32b2d4b8a28e">3</strong>:e915</span><span class="reference__doi" data-ref="01deb76f-20f7-4c97-981a-6b931c6c19f3"></span></li><li class="reference-list__item" id="c64" data-ref="0c62b32c-3afa-40f1-b018-c76c4c6a9cac" data-id="c64"><span class="reference__label" data-ref="974e3324-09b2-4e39-ba8a-e39d49b98753">64.</span><ol class="reference__authors_list" id="el-ol-68" data-ref="872370bf-8524-49ff-8a52-231ebabb7706" data-id="el-ol-68" start="1"><li class="reference__author" id="el-li-286" data-ref="77853f44-1a70-482a-b324-5e6ed0685aed" data-id="el-li-286" data-item-num="1">Sommer C.</li><li class="reference__author" id="el-li-287" data-ref="8f2637a8-9238-4cf1-84f7-564801090da6" data-id="el-li-287" data-item-num="2">Straehle C.</li><li class="reference__author" id="el-li-288" data-ref="52e54db9-7299-4efe-b535-c4d3205b8969" data-id="el-li-288" data-item-num="3">Kothe U.</li><li class="reference__author" id="el-li-289" data-ref="efaa8bfc-1e5b-4b20-b294-a5c6023e9e28" data-id="el-li-289" data-item-num="4">Hamprecht F. A.</li></ol><span class="reference__authors_list_suffix" data-ref="2f7febf0-1125-4afa-97b3-8bcac2b1c0b6">2011</span><span class="reference__title" data-ref="8b3b96f6-2540-48a6-93e8-0715ab8f6930">Ilastik: Interactive learning and segmentation toolkit.</span><span class="reference__origin" data-ref="335760c1-57ad-412e-a472-6b3d58849d66">In: 2011 8th IEEE International Symposium on Biomedical Imaging (ISBI 2011). pp.&nbsp;230‑233</span><span class="reference__doi" data-ref="2ad9bdc1-c5dc-4891-bd00-6836ece38986"><a href="https://doi.org/10.1109/ISBI.2011.5872394" class="reference__doi_link" data-ref="59533caa-79af-43e1-9a8d-d36482f5f9c8">https://doi.org/10.1109/ISBI.2011.5872394</a></span></li><li class="reference-list__item" id="c65" data-ref="4803d647-6ae3-48c1-9c58-9982159579e5" data-id="c65"><span class="reference__label" data-ref="7884ebcc-a262-4c1c-971d-411c89d75276">65.</span><ol class="reference__authors_list" id="el-ol-69" data-ref="cedddf56-12fd-49df-ab51-21ce594412ff" data-id="el-ol-69" start="1"><li class="reference__author" id="el-li-291" data-ref="b52f9f17-9349-4a88-80a3-76f709b1f605" data-id="el-li-291" data-item-num="1">Stringer C.</li><li class="reference__author" id="el-li-292" data-ref="2aed4b2b-74f7-43fb-bc08-815df3f17eee" data-id="el-li-292" data-item-num="2">Wang T.</li><li class="reference__author" id="el-li-293" data-ref="974ef2ae-9f76-4414-b55b-127509e52230" data-id="el-li-293" data-item-num="3">Michaelos M.</li><li class="reference__author" id="el-li-294" data-ref="9f6fe8b5-648e-4e8a-b3a9-f3fec5d383ea" data-id="el-li-294" data-item-num="4">Pachitariu M</li></ol><span class="reference__authors_list_suffix" data-ref="7a61637f-bb05-4ecc-9d37-fb1a64f60cbd">2021</span><span class="reference__title" data-ref="62c356c5-400c-484d-a262-dd5bf2f51772">Cellpose: a generalist algorithm for cellular segmentation.</span><span class="reference__origin" data-ref="ff3129d4-fcfe-4f7c-8935-5c763f19af60"><i data-ref="4a852cff-51a6-4735-90bb-6350002a3108">Nat. Methods</i> <strong data-ref="772cb512-5a6e-42ad-b571-9ab16d6362b2">18</strong>:100‑106</span><span class="reference__doi" data-ref="e61f970a-618a-4c82-befa-91c4a296cdb4"></span></li><li class="reference-list__item" id="c66" data-ref="7e795b7d-9acb-4e58-89fd-cf705d80e954" data-id="c66"><span class="reference__label" data-ref="06ad0473-dfd0-4b0b-8ff4-81ae9a7a8279">66.</span><ol class="reference__authors_list" id="el-ol-70" data-ref="b5c89286-9913-4faa-bcce-14c27cb1503c" data-id="el-ol-70" start="1"><li class="reference__author" id="el-li-296" data-ref="aff7d9a4-7928-4be9-a3c0-8f40c3450c48" data-id="el-li-296" data-item-num="1">Wolf F. A.</li><li class="reference__author" id="el-li-297" data-ref="57c496f9-d38c-4b49-9860-771024bb420c" data-id="el-li-297" data-item-num="2">Angerer P.</li><li class="reference__author" id="el-li-298" data-ref="cda20429-c9d2-4840-917f-d1a5636905b4" data-id="el-li-298" data-item-num="3">Theis F. J</li></ol><span class="reference__authors_list_suffix" data-ref="fae678cd-8d3e-4f04-bbcd-67a7da23f387">2018</span><span class="reference__title" data-ref="7ebbd762-1fd4-43e8-8823-beda5ca6774e">SCANPY: large-scale single-cell gene expression data analysis.</span><span class="reference__origin" data-ref="807a14c7-6a79-4e80-ab03-146261b452e5"><i data-ref="d025f391-38b7-4ca9-b2c6-fb66f63669a9">Genome Biol</i> <strong data-ref="5b053b33-526a-48a3-abb0-a231a915e286">19</strong>:15</span><span class="reference__doi" data-ref="aaa4af65-82f9-497d-aa5e-6081d8205a5c"></span></li><li class="reference-list__item" id="c67" data-ref="3af83f65-07e6-49a6-953b-85b8313808bd" data-id="c67"><span class="reference__label" data-ref="f1a74e9a-dc3e-41ae-8475-80ba6a037c88">67.</span><ol class="reference__authors_list" id="el-ol-71" data-ref="6c3a9ee9-3e6c-4bb4-b601-89b345a9ed51" data-id="el-ol-71" start="1"><li class="reference__author" id="el-li-300" data-ref="6cabd969-66fa-451a-b7e8-90c3e7ab2be9" data-id="el-li-300" data-item-num="1">Gayoso A.</li><li class="reference__author" id="el-li-301" data-ref="61848943-b982-421a-82e9-4bdf86e149f5" data-id="el-li-301" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="cb286899-51ea-415a-ab97-1bc9d01955bc">2022</span><span class="reference__title" data-ref="6f50828e-d034-4b4e-bd56-19d9413f9cda">A Python library for probabilistic analysis of single-cell omics data.</span><span class="reference__origin" data-ref="d2c20056-1b03-497f-b2f8-1f5a3fb700d7"><i data-ref="6691ebfb-e31e-4025-8b3f-5e4e4063e2e6">Nat Biotechnol</i> <strong data-ref="67acb0c1-eec1-49d4-a468-b8bb7d02b985">40</strong>:163‑166</span><span class="reference__doi" data-ref="c874d5eb-e12d-44f3-9e3f-6251e69f2019"></span></li><li class="reference-list__item" id="c68" data-ref="60fc8186-6a09-437c-9886-709ffc6ca7bd" data-id="c68"><span class="reference__label" data-ref="85ca53c9-4d83-4cd7-9d93-2624fdf0f5d2">68.</span><ol class="reference__authors_list" id="el-ol-72" data-ref="0279ff91-1c5c-4bf3-8839-b653a20ebe68" data-id="el-ol-72" start="1"><li class="reference__author" id="el-li-303" data-ref="e53889cf-6c72-4556-9bd5-2e7390f7eda2" data-id="el-li-303" data-item-num="1">van Dijk D.</li><li class="reference__author" id="el-li-304" data-ref="3853d7cd-6b07-4539-8d2e-2afe5f33ad8d" data-id="el-li-304" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="0d894cae-c721-40fa-8a56-a38abccddd92">2018</span><span class="reference__title" data-ref="480fdfb3-1f45-4549-9bac-99d15766fcd4">Recovering Gene Interactions from Single-Cell Data Using Data Diffusion.</span><span class="reference__origin" data-ref="ba0b8f50-fa18-498e-8f12-90a5e34e3b1f"><i data-ref="ff8dde2e-b2a6-4ec6-8381-f5186c528819">Cell</i> <strong data-ref="671ffe19-bd6b-4aa8-9b24-5b48cf15d78c">174</strong>:716‑729.e27</span><span class="reference__doi" data-ref="337402bd-2350-411c-a99f-ff21a35a09ba"></span></li><li class="reference-list__item" id="c69" data-ref="f607a348-7ec8-4b38-b6fb-520ef16749c9" data-id="c69"><span class="reference__label" data-ref="677fa184-d8ec-4280-8f37-fa7abbd3535e">69.</span><ol class="reference__authors_list" id="el-ol-73" data-ref="b59beb30-4577-4870-b5ff-5f0f6997bdfa" data-id="el-ol-73" start="1"><li class="reference__author" id="el-li-306" data-ref="7d86b5e4-cfb3-4b21-8809-7ff9172c4d43" data-id="el-li-306" data-item-num="1">Street K.</li><li class="reference__author" id="el-li-307" data-ref="736c227e-248c-40a0-8646-b2c62bc2b1f5" data-id="el-li-307" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="c16926e0-9fce-4283-9967-94f872ec6da4">2018</span><span class="reference__title" data-ref="54ac0d9b-2a4f-48c1-a72c-6e15f19633d1">Slingshot: cell lineage and pseudotime inference for single-cell transcriptomics.</span><span class="reference__origin" data-ref="4cea2ca8-7e55-4cdd-ad67-9203a50a949c"><i data-ref="78fbe736-267e-458c-ace4-0d1e2bbf47b9">BMC Genomics</i> <strong data-ref="ae82213a-8917-48da-acf8-848f16079af4">19</strong>:477</span><span class="reference__doi" data-ref="80d1c04f-51c8-4841-afdb-c2517b6ec91e"></span></li><li class="reference-list__item" id="c70" data-ref="14e71c1c-fdbb-44c2-bac6-ac7282b2dcaa" data-id="c70"><span class="reference__label" data-ref="637ca270-26c7-46ea-a5b9-c3fba283e0eb">70.</span><ol class="reference__authors_list" id="el-ol-74" data-ref="e9cd7fd3-68d9-4232-8f50-02679ccf915e" data-id="el-ol-74" start="1"><li class="reference__author" id="el-li-309" data-ref="ae971ebb-0c45-43ca-820f-41b2ca605dc6" data-id="el-li-309" data-item-num="1">Van den Berge K</li><li class="reference__author" id="el-li-310" data-ref="de3e1672-ab40-4d33-acc0-da0a39087dc0" data-id="el-li-310" data-item-num="2">et al.</li></ol><span class="reference__authors_list_suffix" data-ref="cae2a6b0-dcab-41c2-9f3f-47111bc33685">2020</span><span class="reference__title" data-ref="4680f1d8-d93e-45f6-b453-c15061813886">Trajectory-based differential expression analysis for single-cell sequencing data.</span><span class="reference__origin" data-ref="5a0bbf15-84b6-46e4-8942-09ee3c4f71dc"><i data-ref="812f80ea-5c5b-45be-bfef-f0bf544511bc">Nat Commun</i> <strong data-ref="d86f430f-72d2-4875-9be4-64ba2e933643">11</strong>:1201</span><span class="reference__doi" data-ref="c1950621-2569-4a83-bd17-af55c7ffc02b"></span></li><li class="reference-list__item" id="c71" data-ref="40be6ce0-ff47-44be-bd2f-4b3fc84188f5" data-id="c71"><span class="reference__label" data-ref="140d0bb0-1b56-4ab1-9f99-e93d74e1e206">71.</span><ol class="reference__authors_list" id="el-ol-75" data-ref="c6dfbabe-c994-4de5-8a36-bbc81f1563c1" data-id="el-ol-75" start="1"><li class="reference__author" id="el-li-312" data-ref="dd8cfaa8-bd48-46c5-b812-723eb2c6f285" data-id="el-li-312" data-item-num="1">Patro R.</li><li class="reference__author" id="el-li-313" data-ref="e4e622c5-d643-4429-a9a2-57330e8765aa" data-id="el-li-313" data-item-num="2">Duggal G.</li><li class="reference__author" id="el-li-314" data-ref="8ccee559-be16-4b55-b066-6cf23fe5f3dc" data-id="el-li-314" data-item-num="3">Love M. I.</li><li class="reference__author" id="el-li-315" data-ref="237e1b76-f56e-4f42-9b05-5d1776f54aca" data-id="el-li-315" data-item-num="4">Irizarry R. A.</li><li class="reference__author" id="el-li-316" data-ref="8da0f970-0f78-49e7-ab09-593a19d475a4" data-id="el-li-316" data-item-num="5">Kingsford C</li></ol><span class="reference__authors_list_suffix" data-ref="6080315f-9a9a-4491-b4bd-c22c998fe80b">2017</span><span class="reference__title" data-ref="7ac3f0f6-ff71-4af7-8e0e-c307b83c394d">Salmon provides fast and bias-aware quantification of transcript expression.</span><span class="reference__origin" data-ref="1891dee3-054d-4dc3-b6bf-0bb30d08711f"><i data-ref="d6e6317d-514d-4ed5-99c7-b70da3c03b5b">Nat Methods</i> <strong data-ref="e96a766b-8ef9-42ce-8eed-cc47ae8e5707">14</strong>:417‑419</span><span class="reference__doi" data-ref="45af79c3-ae7b-4241-a240-814cb7137738"></span></li><li class="reference-list__item" id="c72" data-ref="28302382-2551-4e4a-a579-58d76ecb26e0" data-id="c72"><span class="reference__label" data-ref="15379fff-af2f-41a6-bf9e-1240a261a97c">72.</span><ol class="reference__authors_list" id="el-ol-76" data-ref="268077d9-fe55-44a5-9140-b78154483ba8" data-id="el-ol-76" start="1"><li class="reference__author" id="el-li-318" data-ref="b6d3a45d-4743-4c5c-9c79-7d49cab974a4" data-id="el-li-318" data-item-num="1">Eswarakumar V. P.</li><li class="reference__author" id="el-li-319" data-ref="b27ff7a6-7600-4aab-b45b-6243e7b2cf18" data-id="el-li-319" data-item-num="2">Lax I.</li><li class="reference__author" id="el-li-320" data-ref="c0e12817-cea2-4748-b117-27fa5c8e5e72" data-id="el-li-320" data-item-num="3">Schlessinger J</li></ol><span class="reference__authors_list_suffix" data-ref="303be9fe-80ba-4af9-93f6-af9f632fae4e">2005</span><span class="reference__title" data-ref="9c56e27a-a719-40f7-8ebc-c04cf0b3f1f9">Cellular signaling by fibroblast growth factor receptors.</span><span class="reference__origin" data-ref="3c1a75be-bc90-420f-aba1-e27d49103206"><i data-ref="095c98ae-d416-40f2-bba6-f8e2f3e635dd">Cytokine Growth Factor Rev</i> <strong data-ref="9e202028-3510-46ed-891b-f05439204ebe">16</strong>:139‑149</span><span class="reference__doi" data-ref="a384f693-666e-423d-9842-7082be9403be"></span></li><li class="reference-list__item" id="dataref1" data-ref="34c121fd-0a7e-4401-8bd9-3f1481ea78b6" data-id="dataref1"><ol class="reference__authors_list" id="el-ol-77" data-ref="76ead427-f4ef-4d7a-bc8e-0b05ace87e90" data-id="el-ol-77" start="1"><li class="reference__author" id="el-li-322" data-ref="9bea0fb0-a1ee-4d30-98c6-7d46e2477eb3" data-id="el-li-322" data-item-num="1">Jo K</li><li class="reference__author" id="el-li-323" data-ref="b40a4f50-17c0-41a9-8a17-a6167cc8ec38" data-id="el-li-323" data-item-num="2">Liu ZY</li><li class="reference__author" id="el-li-324" data-ref="45f56528-e145-468c-b5f5-fdc2726f9cc8" data-id="el-li-324" data-item-num="3">Patel G</li><li class="reference__author" id="el-li-325" data-ref="60cec438-29f0-47e9-9b25-c7d2bf2d1551" data-id="el-li-325" data-item-num="4">Yu Z</li><li class="reference__author" id="el-li-326" data-ref="59ddab95-ed1f-42ee-ad40-d8dbe7b65196" data-id="el-li-326" data-item-num="5">Yao L</li><li class="reference__author" id="el-li-327" data-ref="f6678ddd-77dd-4e84-9c17-763e9b1b6623" data-id="el-li-327" data-item-num="6">Teague S</li><li class="reference__author" id="el-li-328" data-ref="c1e61027-8bd4-4dca-8a30-b427034a0f8f" data-id="el-li-328" data-item-num="7">Johnson C</li><li class="reference__author" id="el-li-329" data-ref="8b7f6408-c50e-468e-b653-9af8d59c842a" data-id="el-li-329" data-item-num="8">Spence J</li><li class="reference__author" id="el-li-330" data-ref="75ab3e92-1f12-4ea3-9bd1-669cde2a258a" data-id="el-li-330" data-item-num="9">Heemskerk I</li></ol><span class="reference__authors_list_suffix" data-ref="e0876ee1-e6c2-4ea9-b18f-d08975a7bf09">2024</span><span class="reference__title" data-ref="a2535415-dcec-4a02-9ad9-e9e887a43cea">Endogenous FGF gradients are required for cell fate patterning in 2D human gastruloids.</span><span class="reference__origin" data-ref="689613fc-8509-4962-9d04-9b511678a4e0">NCBI Gene Expression Omnibus. ID ç</span><span class="reference__doi" data-ref="be84647c-24e3-4768-b57b-a3e0055ed387"><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE271604" class="reference__external_link" data-ref="e7d078b3-b15c-4456-af81-055b30a6229c">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE271604</a></span></li></ul></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_left_page" data-page-number="32" id="page-32">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e" data-split-to="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="726f25f1-fdd1-41b8-bf49-de736be3e3ae" data-id="article-content" data-split-from="726f25f1-fdd1-41b8-bf49-de736be3e3ae"><section data-ref="7921c412-ef92-4490-b273-a1fbe0815a2b" data-id="references" data-split-from="7921c412-ef92-4490-b273-a1fbe0815a2b"><ul class="reference-list" data-ref="5d1d46dc-2209-4116-9f07-193d224fefb4" data-id="el-ul-6" data-split-from="5d1d46dc-2209-4116-9f07-193d224fefb4"><li class="reference-list__item" id="dataref2" data-ref="a8f498d2-d7cd-4deb-97a3-88322b296a32" data-id="dataref2"><ol class="reference__authors_list" id="el-ol-78" data-ref="4d77d9d9-9c5c-4822-bfdb-3d1baf80d914" data-id="el-ol-78" start="1"><li class="reference__author" id="el-li-332" data-ref="3070a868-7eac-4f2d-9892-5a57e96be9a1" data-id="el-li-332" data-item-num="1">Zhai J</li><li class="reference__author" id="el-li-333" data-ref="66bb6190-0bd7-410a-aadc-76beca6436c2" data-id="el-li-333" data-item-num="2">Guo J</li><li class="reference__author" id="el-li-334" data-ref="bc430a70-4e6e-4611-8e82-28251549efcd" data-id="el-li-334" data-item-num="3">Wan H</li><li class="reference__author" id="el-li-335" data-ref="1e6b5fe9-1efb-43d8-967f-fb213096fd6e" data-id="el-li-335" data-item-num="4">Qi L</li><li class="reference__author" id="el-li-336" data-ref="5ac0c1c0-28e6-4d3a-9fcd-aaedf5761db7" data-id="el-li-336" data-item-num="5">Liu L</li><li class="reference__author" id="el-li-337" data-ref="2b5ff099-2330-455b-984c-d2e2df52419e" data-id="el-li-337" data-item-num="6">Xiao Z</li><li class="reference__author" id="el-li-338" data-ref="f456dffd-57d2-474a-a2de-63917b31e4fb" data-id="el-li-338" data-item-num="7">Yan L</li><li class="reference__author" id="el-li-339" data-ref="7da05b64-cb34-4c21-b88e-55a62ed45723" data-id="el-li-339" data-item-num="8">Schmitz DA</li><li class="reference__author" id="el-li-340" data-ref="06335589-814e-4d89-9d72-08745f25618d" data-id="el-li-340" data-item-num="9">Xu Y</li><li class="reference__author" id="el-li-341" data-ref="cbfb0c3b-fc31-4212-9851-e78008c88e86" data-id="el-li-341" data-item-num="10">Yu D</li></ol><em data-ref="646ff06d-5cb7-4b02-93e0-0e09b71d3ae8"> et al.</em><span class="reference__authors_list_suffix" data-ref="c8141752-1d35-4dad-82ae-a6fdd0613a78">2022</span><span class="reference__title" data-ref="af2d9af8-cec3-417a-9550-55fb6175042f">Primate gastrulation and early organogenesis at single-cell resolution.</span><span class="reference__origin" data-ref="796e8a35-aae4-4770-a6ba-edeeb30531d4">NCBI Gene Expression Omnibus. ID GSE193007</span><span class="reference__doi" data-ref="a316451a-2f18-45dc-be76-b1e9e6d0b475"><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE193007" class="reference__external_link" data-ref="21981c34-769c-4180-8418-e2a1f2569748">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE193007</a></span></li><li class="reference-list__item" id="dataref3" data-ref="e59a9edc-c430-4288-a9e6-2b9d31e7ae19" data-id="dataref3"><ol class="reference__authors_list" id="el-ol-79" data-ref="22c7b628-8030-4d6f-99fe-5d168553d6d4" data-id="el-ol-79" start="1"><li class="reference__author" id="el-li-343" data-ref="7551b6c0-6fb2-43a6-8c8e-e1824cf7279c" data-id="el-li-343" data-item-num="1">Pijuan-Sala B</li><li class="reference__author" id="el-li-344" data-ref="c5bdc1c9-9698-4630-85c2-5c5a3586c1b4" data-id="el-li-344" data-item-num="2">Griffiths JA</li><li class="reference__author" id="el-li-345" data-ref="c49ff661-6ac8-4f0b-9f20-942dced11231" data-id="el-li-345" data-item-num="3">Guibentif C</li><li class="reference__author" id="el-li-346" data-ref="2cc22c4e-c5df-4edc-989d-ef4caa3ce2dc" data-id="el-li-346" data-item-num="4">Hiscock TW</li><li class="reference__author" id="el-li-347" data-ref="867202ae-1109-40d3-81ba-e5ae53a546b8" data-id="el-li-347" data-item-num="5">Jawaid W</li><li class="reference__author" id="el-li-348" data-ref="446b6c47-7360-499a-a51e-da2ae52c23e9" data-id="el-li-348" data-item-num="6">Calero-Nieto FJ</li><li class="reference__author" id="el-li-349" data-ref="3a6279f0-9927-4623-bf1f-e991516e0cff" data-id="el-li-349" data-item-num="7">Mulas C</li><li class="reference__author" id="el-li-350" data-ref="e683e3db-6914-4c0f-8553-c9a162888b9d" data-id="el-li-350" data-item-num="8">Ibarra-Soria X</li><li class="reference__author" id="el-li-351" data-ref="8030c878-95f6-44ee-9dc1-b1f05a46af23" data-id="el-li-351" data-item-num="9">Tyser RCV</li><li class="reference__author" id="el-li-352" data-ref="70d50f5c-d22d-4ee3-98a1-c4b61d5beb6b" data-id="el-li-352" data-item-num="10">Ho DLL</li></ol><em data-ref="04b7bffd-bd98-44f9-829a-84675c4145ad"> et al.</em><span class="reference__authors_list_suffix" data-ref="ba51caf1-9c2e-4782-93e2-c4f6829ba8b6">2018</span><span class="reference__title" data-ref="30f8a8e3-3cf7-4d50-b1f0-7623073b4d4e">Timecourse single-cell RNAseq of whole mouse embryos harvested between days 6.5 and 8.5 of development.</span><span class="reference__origin" data-ref="e9344f7d-b39d-45ec-8b62-9d8ed211da93">ArrayExpress. ID E-MTAB-6967</span><span class="reference__doi" data-ref="4c9f1131-7f70-4cb2-9929-11a7f878228f"><a href="https://www.ebi.ac.uk/biostudies/arrayexpress/studies/E-MTAB-6967" class="reference__external_link" data-ref="35d92767-fa62-4b09-95ba-c1232db6dba9">https://www.ebi.ac.uk/biostudies/arrayexpress/studies/E-MTAB-6967</a></span></li><li class="reference-list__item" id="dataref4" data-ref="f921e958-5cc0-4416-9213-a949ba5bfab2" data-id="dataref4"><ol class="reference__authors_list" id="el-ol-80" data-ref="bdba8e03-cb51-4cc8-b9f0-f2ae62dd1ee4" data-id="el-ol-80" start="1"><li class="reference__author" id="el-li-354" data-ref="3d745dc9-55d3-4415-bc7f-3bbf7ac72ce0" data-id="el-li-354" data-item-num="1">Tyser RCV</li><li class="reference__author" id="el-li-355" data-ref="b26070a2-7197-45f4-bd01-c6903a64cbc4" data-id="el-li-355" data-item-num="2">Mahammadov E</li><li class="reference__author" id="el-li-356" data-ref="1b506ce6-fb78-4b9a-89ae-acaa653ff230" data-id="el-li-356" data-item-num="3">Nakanoh S</li><li class="reference__author" id="el-li-357" data-ref="bb6e0f74-2b74-4f32-9afb-fbca53f5c459" data-id="el-li-357" data-item-num="4">Vallier L</li><li class="reference__author" id="el-li-358" data-ref="620f1ea1-7534-4f8d-9510-9369b3ea0016" data-id="el-li-358" data-item-num="5">Scialdone A</li><li class="reference__author" id="el-li-359" data-ref="8e0f1567-7ddc-4542-be30-212b85f423a2" data-id="el-li-359" data-item-num="6">Srinivas S</li></ol><span class="reference__authors_list_suffix" data-ref="67e526cd-f584-4af6-8d47-a82268c54964">2021</span><span class="reference__title" data-ref="a0162004-0947-4566-b745-102703395c9f">A spatially resolved single cell atlas of human gastrulation.</span><span class="reference__origin" data-ref="985f310b-cfb8-43c0-b417-6a81e4157d5d">ArrayExpress. ID E-MTAB-9388</span><span class="reference__doi" data-ref="0fc053c7-f2d4-4fdc-a8c3-7e49838dfa0b"><a href="https://www.ebi.ac.uk/biostudies/arrayexpress/studies/E-MTAB-9388" class="reference__external_link" data-ref="fa0d425b-7514-4f26-93aa-7c782a939dd0">https://www.ebi.ac.uk/biostudies/arrayexpress/studies/E-MTAB-9388</a></span></li><li class="reference-list__item" id="dataref5" data-ref="93234cd8-ff68-4266-8fc3-b326837aaf1d" data-id="dataref5"><ol class="reference__authors_list" id="el-ol-81" data-ref="ce379475-1b6e-45c2-a14c-105db149e8d4" data-id="el-ol-81" start="1"><li class="reference__author" id="el-li-361" data-ref="c8991010-05cb-4438-8873-056be3833616" data-id="el-li-361" data-item-num="1">Koh PW</li><li class="reference__author" id="el-li-362" data-ref="a87a76f7-bf2d-405d-9ab2-ecb3531ae41d" data-id="el-li-362" data-item-num="2">Sinha R</li><li class="reference__author" id="el-li-363" data-ref="76d8fbde-738d-4fbd-abf9-00873ff80242" data-id="el-li-363" data-item-num="3">Barkal AA</li><li class="reference__author" id="el-li-364" data-ref="0cb2ac59-14a8-4b13-98d4-56d2a62c27b6" data-id="el-li-364" data-item-num="4">Morganti RM</li><li class="reference__author" id="el-li-365" data-ref="d7a8576c-e7cd-4638-944b-c73a10dc125d" data-id="el-li-365" data-item-num="5">Chen A</li><li class="reference__author" id="el-li-366" data-ref="d8e32811-975c-4638-a492-b4dde93d0beb" data-id="el-li-366" data-item-num="6">Weissman IL</li><li class="reference__author" id="el-li-367" data-ref="96ee317b-4e4a-4c92-bf9f-8255a4269dc2" data-id="el-li-367" data-item-num="7">Ang LT</li><li class="reference__author" id="el-li-368" data-ref="72b2c546-03e0-45cf-855d-5d26e6b1eda1" data-id="el-li-368" data-item-num="8">Kundaje A</li><li class="reference__author" id="el-li-369" data-ref="347dd967-98ba-441e-80a4-5dce543f152a" data-id="el-li-369" data-item-num="9">Loh KM</li></ol><span class="reference__authors_list_suffix" data-ref="17fc94e4-7685-48c4-9955-4278e184800e">2016</span><span class="reference__title" data-ref="9af24f56-1c95-4868-b2a4-7edeb8db2b8c">In vitro cultured H7 human embryonic stem cells (WiCell) and H7-derived downstream early mesoderm progenitors.</span><span class="reference__origin" data-ref="9337dab1-201e-4988-9d64-ef56c9d59a5b">NCBI Gene Expression Omnibus. ID GSE85066</span><span class="reference__doi" data-ref="7bf5e856-e63a-43df-8ec6-df5b38fbde88"><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE85066" class="reference__external_link" data-ref="451ac053-802a-4a6d-ae00-a74003e7a7a9">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE85066</a></span></li></ul></section></section><section id="peer-reviews" data-ref="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2" data-id="peer-reviews" data-split-to="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2" data-split-original="true"><h1 id="el-h1-11" data-ref="193c330f-0fe9-48f4-9670-4f5715461a45" data-id="el-h1-11" data-break-after="avoid">Peer reviews</h1><section id="reviews" class="review-content" data-ref="1d625bc5-02ad-4fce-91fd-d9baddac7f30" data-id="reviews" data-previous-break-after="avoid" data-split-to="1d625bc5-02ad-4fce-91fd-d9baddac7f30" data-split-original="true"><h2 id="el-h2-25" data-ref="14d8ad61-c16d-404d-b5e0-e55397de1c1c" data-id="el-h2-25" data-break-after="avoid" data-following="following-ff2c00ba-51a1-482d-9afa-f5a370dcfa50">Reviewer #1 (Public review):</h2><div class="review-content__body" id="el-div-16" data-ref="9e7f3e80-f94e-4b3e-84c6-91a98247b329" data-id="el-div-16" data-previous-break-after="avoid" data-split-to="9e7f3e80-f94e-4b3e-84c6-91a98247b329" data-split-original="true" data-last-split-element="true" data-align-last-split-element="auto"><p id="el-p-90" data-ref="849b05a8-6ea2-4a55-b7be-a2523fd07053" data-id="el-p-90">Summary:</p><p id="el-p-91" data-ref="d6c2e0e5-5d01-4544-a4f3-88d3bd029330" data-id="el-p-91">This is an interesting study on the role of FGF signaling in the induction of primitive streak like-cells (PS-LC) in human 2D-gastruloids. The authors use a previously characterized standard culture that generates a ring of PS-LCs (TBXT+) and correlate this with pERK staining. A requirement for FGF signaling in TBXT induction is demonstrated via pharmacological inhibition of MEK and FGFR activity. A second set of culture conditions (with no exogenous FGFs) suggests that endogenous FGFs are required for pERK and TBXT induction. The authors then characterize, via scRNA-seq, various components of the FGF pathway (genes for ligand, receptors, ERK regulators, HSPG regulation). They go on to characterize the pFGFR1, receptor isoforms and polarized localization of this receptor. Finally, they perform FGF4 inhibition and use a cell line with a limited FGF17 inactivation (heterozygous null) and show that loss of these FGFs reduce PS-LC and derivative cell types.</p><p id="el-p-92" data-ref="8580f9e9-1ccc-4ce7-ad4d-18f26174e71e" data-id="el-p-92">Strengths:</p><p id="el-p-93" data-ref="f21ac7a3-d80c-4947-9195-3d163df0e858" data-id="el-p-93">(1) As the authors point out, the role of FGF signaling in gastrulation is less well understood than other signaling pathways. Hence this is a valuable contribution to that field.</p><p id="el-p-94" data-ref="2e1e0a41-bba1-4257-8c80-3d29ab479beb" data-id="el-p-94">(2) The FGF4 and FGF17 loss-of-function experiments in Figure 5 are very intriguing. This is especially so given the intriguing observation that these FGFs appear to be dominating in this model of human gastrulation, in contrast to what FGFs dominate in mice, chick and frogs.</p><p id="el-p-95" data-ref="8fe8f4f3-3469-4ac9-8417-0a3735f4cfd6" data-id="el-p-95">(3) In general this paper is valuable as a further development of the Human gastruloid system and the role of FGF signaling in the induction of PS-CLs. The wide net that the authors cast in characterizing FGF ligand gene, receptor isoforms, and downstream components provides a foundation for future work. As the authors write near the beginning of the Discussion "Many questions remain."</p><p id="el-p-96" data-ref="9286a75a-188e-4eda-b796-c9081228f7bf" data-id="el-p-96">Weaknesses:</p><p id="el-p-97" data-ref="2dfddacf-f85b-4da4-be9e-72c94e9f8aca" data-id="el-p-97">(1) FGFs are cell survival factors in various aspects of development. The authors fail to address cell death due to loss of FGF signaling in any of their experiments. For example, in Figure 1E (which requires statistical analysis) and 1G (the bottom FGFRi row), there appears to be a significant amount of cell loss. Is this due to cell death? The authors should address the question of whether the role of FGF/ERK signaling is to keep the cells alive.</p></div></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_right_page" data-page-number="33" id="page-33">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e" data-split-to="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2" data-id="peer-reviews" data-split-from="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2" data-split-to="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2"><section class="review-content" data-ref="1d625bc5-02ad-4fce-91fd-d9baddac7f30" data-id="reviews" data-split-from="1d625bc5-02ad-4fce-91fd-d9baddac7f30"><div class="review-content__body" data-ref="9e7f3e80-f94e-4b3e-84c6-91a98247b329" data-id="el-div-16" data-split-from="9e7f3e80-f94e-4b3e-84c6-91a98247b329"><p id="el-p-98" data-ref="fb087b52-a9fd-436d-b36d-826aca0fdcbd" data-id="el-p-98">(2) Regarding the sparse cells in 1G, is there a reduction in cell number only with FGFRi and not MEKi? Is this reproducible? Gattiglio et al (Development, 2023, PMID: 37530863) present data supporting a "community effect" in the FGF-induced mesoderm differentiation of mouse embryonic stem cells. Could a community effect be at play in this human system (especially given the images in the bottom row of 1G). If the authors don't address this experimentally they should at least address the ideas in Gattoglio et al.</p><p id="el-p-99" data-ref="b9647971-fe6d-471b-86d3-85171d0379a4" data-id="el-p-99">(3) Do the FGF4 and FGF17 LOF experiments in Figure 5 affect cell number like FGFRi in Figure 1? Why examine PS-LC induction only in FGF17 heterozygous cells and not homozygous FGF17 nulls?</p><p id="el-p-100" data-ref="579800b1-0ea7-4d2b-a856-f6700be05265" data-id="el-p-100">(4) The idea that FGF8 plays a dominant role during gastrulation of other species but not humans is so intriguing it warrants deeper testing. The authors dismiss FGF8 because its mRNA "...levels always remained low." (line 363) as well as the data published in Zhai et al (PMID: 36517595) and Tyser et al (PMID: 34789876). But there are cases in mouse development where a gene was expressed at levels so low, it might be dismissed, and yet LOF experiments revealed it played a role or even was required in a developmental process. The authors should consider FGF8 inhibition or inactivation to explore its potential role, despite its low levels of expression.</p><p id="el-p-101" data-ref="ad63ee8a-2dc1-49b7-93c7-ca01fc21578b" data-id="el-p-101">(5) Redundancy is a common feature in FGF genetics. What is the effect of inhibiting FGF4 in FGF17 LOF cells?</p><p id="el-p-102" data-ref="0c246a58-4920-47cc-a056-c4fc0a291558" data-id="el-p-102">(6) I suggest stating that the authors take more caution describing FGF gradients. For example, in one Results heading they write "Endogenous FGF4 and FGF17 gradients underly the ERK activity pattern.", implying an FGF protein gradient. However, they only present data for FGF mRNA , not protein. This issue would be clarified if they used proper nomenclature for gene, mRNA (italics) and protein (no italics) throughout the paper.</p><p id="el-p-103" data-ref="bc9c1eb7-b89e-403e-8726-5f80c0588b03" data-id="el-p-103">Comments on revisions:</p><p id="el-p-104" data-ref="9c978591-c7ee-445a-a5de-abcf436adc0f" data-id="el-p-104">The authors have addressed my concerns.</p></div><div class="descriptors" id="el-div-17" data-ref="2fbbec87-25b5-46a4-8029-0f6cdd37d4f5" data-id="el-div-17"><ul class="descriptors__identifiers" id="el-ul-7" data-ref="cb2fd904-29c1-4a57-975b-53e5104f98f5" data-id="el-ul-7"><li class="descriptors__identifier" id="el-li-370" data-ref="1d7094d2-3be6-4d9b-86ca-f71212a0c466" data-id="el-li-370"><a href="https://doi.org/10.7554/eLife.101224.2.sa3" data-ref="31113fb7-79fe-4e7e-8132-5b1289fd2e9a">https://doi.org/10.7554/eLife.101224.2.sa3</a></li></ul></div></section><section id="reviews" class="review-content" data-ref="482c4127-9b84-4bff-82d8-e461ca58e221" data-id="reviews" data-split-to="482c4127-9b84-4bff-82d8-e461ca58e221" data-split-original="true"><h2 id="el-h2-26" data-ref="db76f787-2f9e-4cc0-b1ae-f0837f21cd2e" data-id="el-h2-26" data-break-after="avoid">Reviewer #2 (Public review):</h2><div class="review-content__body" id="el-div-18" data-ref="580a7f89-7649-489c-9c78-0415aa3a0547" data-id="el-div-18" data-previous-break-after="avoid" data-split-to="580a7f89-7649-489c-9c78-0415aa3a0547" data-split-original="true"><p id="el-p-105" data-ref="9787a0dc-c4e9-4dcb-835b-4b05b22822f6" data-id="el-p-105">Summary:</p><p id="el-p-106" data-ref="1ab22ab0-6d1e-4910-a020-87edd808d72b" data-id="el-p-106">The role of FGFs in embryonic development and stem cell differentiation has remained unclear due to its complexity. In this study, the authors utilized a 2D human stem cell-based gastrulation model to investigate the functions of FGFs. They discovered that FGF-dependent ERK activity is closely linked to the emergence of primitive streak cells. Importantly, this 2D model effectively illustrates the spatial distribution of key signaling effectors and receptors by correlating these markers with cell fate markers, such as T and ISL1. Through inhibition and loss-of-function studies, they further corroborated the needs of FGF ligands. Their data shows that FGFR1 is the primary receptor, and FGF2/4/17 are the key ligands for primitive streak development, which aligns with observations in primate embryos. Additional experiments revealed that the reduction of FGF4 and FGF17 decreases ERK activity.</p><p id="el-p-107" data-ref="f1d8ccd9-4a25-4d4d-9865-0dc32b022d5c" data-id="el-p-107">Strengths:</p><p id="el-p-108" data-ref="66400d11-f5be-4b9d-8aea-f0b3b441af15" data-id="el-p-108" data-split-to="66400d11-f5be-4b9d-8aea-f0b3b441af15" data-split-original="true" data-last-split-element="true" data-align-last-split-element="auto">This study provides comprehensive data and improves our understanding of the role of FGF signaling in primate primitive streak formation. The authors provide new insights related to the spatial localization of the key components of FGF signaling and attempt to reveal the </p></div></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_left_page" data-page-number="34" id="page-34">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e" data-split-to="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2" data-id="peer-reviews" data-split-from="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2" data-split-to="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2"><section class="review-content" data-ref="482c4127-9b84-4bff-82d8-e461ca58e221" data-id="reviews" data-split-from="482c4127-9b84-4bff-82d8-e461ca58e221"><div class="review-content__body" data-ref="580a7f89-7649-489c-9c78-0415aa3a0547" data-id="el-div-18" data-split-from="580a7f89-7649-489c-9c78-0415aa3a0547"><p data-ref="66400d11-f5be-4b9d-8aea-f0b3b441af15" data-id="el-p-108" data-split-from="66400d11-f5be-4b9d-8aea-f0b3b441af15">temporal dynamics of the signal propagation and cell fate decision, which has been challenging.</p></div><div class="descriptors" id="el-div-19" data-ref="b8dbf020-daeb-4f6a-81e7-996bf5bf927b" data-id="el-div-19"><ul class="descriptors__identifiers" id="el-ul-8" data-ref="7d0cb99c-b700-49b6-8e07-1ac0bd6b213c" data-id="el-ul-8"><li class="descriptors__identifier" id="el-li-371" data-ref="d120bd51-8703-4cfc-bc37-6b25b2f35bbb" data-id="el-li-371"><a href="https://doi.org/10.7554/eLife.101224.2.sa2" data-ref="579d6b1c-73a6-48f8-9210-e806f605c3e1">https://doi.org/10.7554/eLife.101224.2.sa2</a></li></ul></div></section><section id="reviews" class="review-content" data-ref="4c2dabe6-69c1-4cd4-bedc-e4d78caf0677" data-id="reviews"><h2 id="el-h2-27" data-ref="5db120f2-5cc8-4908-8387-68bbeb4bfb22" data-id="el-h2-27" data-break-after="avoid">Reviewer #3 (Public review):</h2><div class="review-content__body" id="el-div-20" data-ref="6ac993fd-bd96-46e8-ba1d-674910bed89a" data-id="el-div-20" data-previous-break-after="avoid"><p id="el-p-109" data-ref="d836ec45-09be-4b88-935e-aa03225ba4c9" data-id="el-p-109">Jo and colleagues set out to investigate the origins and functions of localized FGF/ERK signaling for the differentiation and spatial patterning of primitive streak fates of human embryonic stem cells in a well-established micropattern system. They demonstrate that endogenous FGF signaling is required for ERK activation in a ring-domain in the micropatterns, and that this localized signaling is directly required for differentiation and spatial patterning of specific cell types. Through high-resolution microscopy and transwell assays, they show that cells receive FGF signals through basally localized receptors. Finally, the authors find that there is a requirement for exogenous FGF2 to initiate primitive streak-like differentiation, but endogenous FGFs, especially FGF4 and FGF17, fully take over at later stages.</p><p id="el-p-110" data-ref="c44e7a7b-19ee-430f-8e66-abe12fd79e6e" data-id="el-p-110">Even though some of the authors' findings - such as the localized expression of FGF ligands during gastrulation and the importance of FGF/ERK signaling for cell differentiation in the primitive streak - have been reported in model organisms before, this is one of the first studies to investigate the role of FGF signaling during primitive streak-like differentiation of human cells. In doing so, the paper reports a number of interesting and valuable observations, namely the basal localization of FGF receptors which mirrors that of BMP and Nodal receptors, as well as the existence of a positive feedback loop centered on FGF signaling that drives primitive-streak differentiation. In the revised version of their work, the authors have furthermore dissected the role of different FGFs through knockdown approaches. These experiments reveal discrete functions for different FGF genes in their system, as well as interesting differences between the role of specific FGFs in human compared to model systems.</p><p id="el-p-111" data-ref="74627652-564a-4329-84f7-d041fa7d1747" data-id="el-p-111">Comments on revisions:</p><p id="el-p-112" data-ref="12c9acd2-92b1-4a06-92ff-5a0459d238e4" data-id="el-p-112">The authors have appropriately addressed all comments and suggestions from the previous round of review. The only textual change that I would still like to suggest is to write explicitly in the main text corresponding to Fig. 1 that the mTESR1 medium used for these initial experiments already contains FGF. This is something that is probably known to experts in the field, but not necessarily to a broader readership.</p></div><div class="descriptors" id="el-div-21" data-ref="3c4001b3-f977-4d43-84ed-ae8e87ec6ac4" data-id="el-div-21"><ul class="descriptors__identifiers" id="el-ul-9" data-ref="8dfb07d5-b518-4f06-af6b-7eabb28ce8fd" data-id="el-ul-9"><li class="descriptors__identifier" id="el-li-372" data-ref="d556bbcc-a88a-4632-b65a-5ef95effd543" data-id="el-li-372"><a href="https://doi.org/10.7554/eLife.101224.2.sa1" data-ref="278574ec-50d4-4e15-af20-4893fc0f3867">https://doi.org/10.7554/eLife.101224.2.sa1</a></li></ul></div></section><section id="authorresponse" class="author-response" data-ref="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4" data-id="authorresponse" data-split-to="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4" data-split-original="true"><h2 id="el-h2-28" data-ref="46fd2982-0a28-487a-958d-367ca72047bf" data-id="el-h2-28" data-break-after="avoid">Author response:</h2><div class="author-response__body" id="el-div-22" data-ref="0f8a4845-8525-4632-882f-9a92142367e9" data-id="el-div-22" data-previous-break-after="avoid" data-split-to="0f8a4845-8525-4632-882f-9a92142367e9" data-split-original="true"><p id="el-p-113" data-ref="6429f6bb-02a2-4398-a7ad-0ddaa9ffee93" data-id="el-p-113">The following is the authors’ response to the original reviews.</p><blockquote data-ref="4d267609-9d09-4287-87d4-8cda462eb822" data-split-to="4d267609-9d09-4287-87d4-8cda462eb822" data-split-original="true"><p id="el-p-114" data-ref="681677fa-9a39-4120-a4ac-69e50de31119" data-id="el-p-114"><strong data-ref="f3dcb14d-4aa8-4cf5-9070-6bbf14430d9f">Reviewer #1 (Public review):</strong></p><p id="el-p-115" data-ref="38db78e4-9347-4b5b-bbcd-84d274b11746" data-id="el-p-115">Summary:</p><p id="el-p-116" data-ref="3613efc2-7021-4002-89c3-93ad7ae26f7d" data-id="el-p-116" data-split-to="3613efc2-7021-4002-89c3-93ad7ae26f7d" data-split-original="true" data-last-split-element="true" data-align-last-split-element="auto">This is an interesting study on the role of FGF signaling in the induction of primitive streak-like cells (PS-LC) in human 2D-gastruloids. The authors use a previously characterized standard culture that generates a ring of PSLCs (TBXT+) and correlate this with pERK staining. A requirement for FGF signaling in TBXT induction is demonstrated via pharmacological inhibition of MEK and FGFR activity. A second set of culture conditions (with no exogenous FGFs) suggests that endogenous FGFs are required for pERK and TBXT induction. The authors then characterize, via scRNA-seq, various </p></blockquote></div></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_right_page" data-page-number="35" id="page-35">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e" data-split-to="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2" data-id="peer-reviews" data-split-from="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2" data-split-to="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2"><section class="author-response" data-ref="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4" data-id="authorresponse" data-split-from="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4" data-split-to="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4"><div class="author-response__body" data-ref="0f8a4845-8525-4632-882f-9a92142367e9" data-id="el-div-22" data-split-from="0f8a4845-8525-4632-882f-9a92142367e9" data-split-to="0f8a4845-8525-4632-882f-9a92142367e9"><blockquote data-ref="4d267609-9d09-4287-87d4-8cda462eb822" data-split-from="4d267609-9d09-4287-87d4-8cda462eb822"><p data-ref="3613efc2-7021-4002-89c3-93ad7ae26f7d" data-id="el-p-116" data-split-from="3613efc2-7021-4002-89c3-93ad7ae26f7d">components of the FGF pathway (genes for ligands, receptors, ERK regulators, and HSPG regulation). They go on to characterize the pFGFR1, receptor isoforms, and polarized localization of this receptor. Finally, they perform FGF4 inhibition and use a cell line with a limited FGF17 inactivation (heterozygous null) and show that loss of these FGFs reduces PS-LC and derivative cell types.</p><p id="el-p-117" data-ref="3e51a978-af21-47dd-8a8b-b6483bd4d6f9" data-id="el-p-117">Strengths:</p><p id="el-p-118" data-ref="1300ec0a-cfba-4076-94ac-91d07f56c5fa" data-id="el-p-118">(1) As the authors point out, the role of FGF signaling in gastrulation is less well understood than other signaling pathways. Hence this is a valuable contribution to that field.</p><p id="el-p-119" data-ref="f0e7669f-e60e-41b1-a0f3-5c97ad1646da" data-id="el-p-119">(2) The FGF4 and FGF17 loss-of-function experiments in Figure 5 are very intriguing. This is especially so given the intriguing observation that these FGFs appear to be dominating in this model of human gastrulation, in contrast to what FGFs dominate in mice, chicks, and frogs.</p><p id="el-p-120" data-ref="b2705b6a-ae30-4eeb-95ed-0fe14328454b" data-id="el-p-120">(3) In general this paper is valuable as a further development of the Human gastruloid system and the role of FGF signaling in the induction of PS-CLs. The wide net that the authors cast in characterizing the FGF ligand gene, receptor isoforms, and downstream components provides a foundation for future work. As the authors write near the beginning of the Discussion "Many questions remain."</p></blockquote><p id="el-p-121" data-ref="cd3cef89-01c1-416d-a00d-f6250520b0b3" data-id="el-p-121">We thank the reviewer for these positive comments.</p><blockquote data-ref="43781384-61ae-4875-b1c1-1638509f7143"><p id="el-p-122" data-ref="5d5a37ac-49ea-405d-926b-5ee35d47362a" data-id="el-p-122">Weaknesses:</p><p id="el-p-123" data-ref="1a6157e6-f5e2-43d4-a80e-11bb2b2c06dc" data-id="el-p-123">(1) FGFs are cell survival factors in various aspects of development. The authors fail to address cell death due to loss of FGF signaling in their experiments. For example, in Figure 1E (which requires statistical analysis) and 1G (the bottom FGFRi row), there appears to be a significant amount of cell loss. Is this due to cell death? The authors should address the question of whether the role of FGF/ERK signaling is to keep the cells alive.</p></blockquote><p id="el-p-124" data-ref="ed126906-0e5d-4cb4-83cf-f943737515d5" data-id="el-p-124">Indeed, FGF also strongly affects cell survival and it is an interesting question to what extent this depends on ERK. Our manuscript focuses instead on the role of FGF/ERK signaling in cell fate patterning. As mentioned in our discussion, figure 1de show that doxycycline induced pERK leads to more TBXT+ cells than the control without restoring cell number, suggesting the role of FGF in controlling cell number is independent of the requirement for FGF/ERK in PS-LC differrentiation. To further support this, we have added data showing low doses of MEKi are sufficient to inhibit differentiation without affecting cell number (Supp. Fig. 1i).</p><p id="el-p-125" data-ref="c4ca3e10-54b2-4225-b54e-4b93cbd25bf1" data-id="el-p-125">To address the reviewers question regarding the cause of cell loss, we now stained for BrdU and cleaved Cas3 to assess proliferation and apoptosis in the presence and absence of MEK and FGFR inhibition (new Supp. Fig.</p><p id="el-p-126" data-ref="c2b8f7fe-30d0-40cd-bbe6-fcc07304c71c" data-id="el-p-126">1ef). This shows that the effect of these inhibitors on cell number is primarily due to a reduction in proliferation. We have also included statistical analysis in Fig.1e.</p><blockquote data-ref="1725e14d-51d6-4a44-938c-ab055d27352b"><p id="el-p-127" data-ref="32092e33-2f64-41a7-a7d2-24ec6a471cbc" data-id="el-p-127">(2) Regarding the sparse cells in 1G, is there a reduction in cell number only with FGFRi and not MEKi? Is this reproducible? Gattiglio et al (Development, 2023, PMID: 37530863) present data supporting a "community effect" in the FGF-induced mesoderm differentiation of mouse embryonic stem cells. Could a community effect be at play in this human system (especially given the images in the bottom row of 1G)? If the authors don't address this experimentally they should at least address the ideas in Gattoglio et al.</p></blockquote><p id="el-p-128" data-ref="8a1eb3bb-bc3a-489c-a7cd-a95c3d3e4fa9" data-id="el-p-128" data-split-to="8a1eb3bb-bc3a-489c-a7cd-a95c3d3e4fa9" data-split-original="true" data-last-split-element="true" data-align-last-split-element="auto">Indeed, FGFRi reproducibly affects cell number more than MEKi, in line with the fact that pathways other than MAPK/ERK downstream of FGF (e.g. PI3K) play important roles in cell </p></div></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_left_page" data-page-number="36" id="page-36">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e" data-split-to="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2" data-id="peer-reviews" data-split-from="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2" data-split-to="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2"><section class="author-response" data-ref="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4" data-id="authorresponse" data-split-from="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4" data-split-to="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4"><div class="author-response__body" data-ref="0f8a4845-8525-4632-882f-9a92142367e9" data-id="el-div-22" data-split-from="0f8a4845-8525-4632-882f-9a92142367e9" data-split-to="0f8a4845-8525-4632-882f-9a92142367e9"><p data-ref="8a1eb3bb-bc3a-489c-a7cd-a95c3d3e4fa9" data-id="el-p-128" data-split-from="8a1eb3bb-bc3a-489c-a7cd-a95c3d3e4fa9">survival and growth. However, we think the lack of differentiation in MEKi and FGFRi in Fig.1g cannot be attributed to a loss of cells combined with a community effect. This is because without FGFRi or MEKi cells efficiently differentiate to primitive streak at much lower densities than those originally shown, consistent with the data we discuss in response to (1) arguing against a primarily indirect effect of FGF on PS-LC differentiation through cell density. In the context of directed differentiation (rather than 2D gastruloids), we have now shown in a controlled manner that the effect of MEKi and FGFRi does not depend on a community effect by repeating the experiment in Fig.1g while adjusting cell seeding densities to obtain similar final cell densities in all three conditions (new Fig.1g, new Supp Fig.1g). Furthermore we have included new data showing extremely sparse cells without MEKi or FGFRi still differentiate without problems (new Supp Fig 1h). We have also include Gattoglio et al in our revised discussion.</p><blockquote data-ref="f7283141-df7e-4161-a510-6a07b9376749"><p id="el-p-129" data-ref="f9bb140d-4e1a-416b-b4dd-a740f0937b48" data-id="el-p-129">(3) Do the FGF4 and FGF17 LOF experiments in Figure 5 affect cell numbers like FGFRi in Figure 1?</p></blockquote><p id="el-p-130" data-ref="984bed0d-66d6-48e6-8f35-8efc8ea6a827" data-id="el-p-130">We did not observe major changes in cell number in the FGF4 and FGF17 loss of function experiments. This is in line with our observation that low levels of ERK signaling are sufficient to maintain proliferation (new Supp. Fig. 1i), and the fact that low levels of ERK signaling are maintained in the absence of FGF4 and FGF17 (Fig.5), likely by FGF2 (Fig. 2). In contrast, FGFRi treatment in Fig.1 leads to a nearly complete loss of FGF signaling (ERK and other pathways) that has a dramatic effect on cell number.</p><blockquote data-ref="21cc2fe7-6e71-4d10-bf10-31dcd8e1b6c7"><p id="el-p-131" data-ref="2b0db90f-9e9a-44ae-8c88-b220efd82bdb" data-id="el-p-131">Why examine PS-LC induction only in FGF17 heterozygous cells and not homozygous FGF17 nulls?</p></blockquote><p id="el-p-132" data-ref="5a867e4d-f7fe-48e5-8b79-35fd7a805e5c" data-id="el-p-132">We were unable to obtain homozygous FGF17 nulls, it is not clear if there is a reason for this. In the absence of homozygous nulls, we have now further corroborated our findings with additional knockdown data (described in response to other comments below).</p><blockquote data-ref="f804ee20-8b7b-410d-8bcd-1924eee98c36"><p id="el-p-133" data-ref="edc97fc2-6dc7-4341-b45a-a8988035eca9" data-id="el-p-133">(4) The idea that FGF8 plays a dominant role during gastrulation of other species but not humans is so intriguing it warrants deeper testing. The authors dismiss FGF8 because its mRNA "...levels always remained low." (line 363) as well as the data published in Zhai et al (PMID: 36517595) and Tyser et al (PMID: 34789876). But there are cases in mouse development where a gene was expressed at levels so low, that it might be dismissed, and yet LOF experiments revealed it played a role or even was required in a developmental process. The authors should consider FGF8 inhibition or inactivation to explore its potential role, despite its low levels of expression.</p></blockquote><p id="el-p-134" data-ref="13473dae-b259-47e0-b3d1-3ac4db4ac2e2" data-id="el-p-134">We thank the reviewer for this suggestion. We have now analyzed the role of FGF8 using FISH to visualize its expression and siRNA to understand its function (Fig.5d,f,h; Supp.Fig.5e,g,6e). We found that FGF8 expression is higher earlier in differentiation, preceding most expression of TBXT. Our scRNA-seq only analyzed samples at 42h so did not capture this. Furthermore, FGF8 expression localized inside the PS-like ring rather than coinciding with it like FGF4. Surprisingly, FGF8 knockdown led to an increase in primitive streak-like differentiation, suggesting it may counteract FGF4. The results are shown in the revised Fig. 5 and Supplemental Fig. 5. While this certainly merits further investigation, understanding the role of FGF8 in more detail is beyond the scope of the current work.</p><blockquote data-ref="8d3f92eb-dde5-4274-b1bc-7c19c935e17c"><p id="el-p-135" data-ref="17d2b5e2-72b3-43eb-b1e3-3b07fa9d2836" data-id="el-p-135">(5) Redundancy is a common feature in FGF genetics. What is the effect of inhibiting FGF4 in FGF17 LOF cells?</p></blockquote><p id="el-p-136" data-ref="65660fea-3813-4f24-998c-86c634503cf9" data-id="el-p-136" data-split-to="65660fea-3813-4f24-998c-86c634503cf9" data-split-original="true" data-last-split-element="true" data-align-last-split-element="auto">Further siRNA and shRNA experiments showed that FGF17 knockdown had a much smaller effect than FGF4 knockdown on expression of primitive streak markers (Fig.5i, Supp.Fig.6f-i) but that FGF17 knockdown did lead to a complete loss of the mesoderm marker TBX6 (Fig.5j, Supp.Fig.6j). A double knockdown of FGF4+FGF17 looked similar to FGF4 alone (Supp.Fig.6k). Thus, we now think the more likely scenario is that FGF17 is downstream of FGF4-dependent </p></div></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_right_page" data-page-number="37" id="page-37">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e" data-split-to="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2" data-id="peer-reviews" data-split-from="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2" data-split-to="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2"><section class="author-response" data-ref="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4" data-id="authorresponse" data-split-from="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4" data-split-to="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4"><div class="author-response__body" data-ref="0f8a4845-8525-4632-882f-9a92142367e9" data-id="el-div-22" data-split-from="0f8a4845-8525-4632-882f-9a92142367e9" data-split-to="0f8a4845-8525-4632-882f-9a92142367e9"><p data-ref="65660fea-3813-4f24-998c-86c634503cf9" data-id="el-p-136" data-split-from="65660fea-3813-4f24-998c-86c634503cf9">PS-differentiation and although this may have a positive feedback effect whereby this FGF17 can then enhance further PS-differentiation, which we previously interpreted as partial redundancy, the primary role of FGF17 may be later, in mesoderm differentiation.</p><blockquote data-ref="d7db3727-62f3-467f-a587-72ca9ab4f1a2"><p id="el-p-137" data-ref="6f6359d6-6055-4640-8dc9-75c8f52682ea" data-id="el-p-137">(6) I suggest stating that the authors take more caution in describing FGF gradients. For example, in one Results heading they write "Endogenous FGF4 and FGF17 gradients underly the ERK activity pattern.", implying an FGF protein gradient. However, they only present data for FGF mRNA , not protein. This issue would be clarified if they used proper nomenclature for gene, mRNA (italics), and protein (no italics) throughout the paper.</p></blockquote><p id="el-p-138" data-ref="d5f398bc-23e3-443b-aa22-d8550a9cdfa1" data-id="el-p-138">Thank you for the suggestion. We have edited the paper to more clearly distinguish protein and mRNA. We do think our data provide substantial indirect evidence for a protein gradient which is what the results heading is meant to convey. Receptor activation is high where ERK activity is high (Fig.3), and receptor activation is limited by ligands, since creating a scratch to let exogenous FGF reach the basal side of cells in the center leads to receptor activation (Fig.4). This strongly suggests ERK activity reflects an FGF protein gradient.</p><blockquote data-ref="803bd602-f7eb-454d-9e6f-7eef3344cd6a"><p id="el-p-139" data-ref="b6ad45f4-e340-4b28-be59-2f8b3a33cc4d" data-id="el-p-139"><strong data-ref="44727c45-c822-4854-bf36-c0a7f5513ffc">Reviewer #2 (Public review):</strong></p><p id="el-p-140" data-ref="ec93e6a1-9b06-4104-8563-5021220a5c7e" data-id="el-p-140">Summary:</p><p id="el-p-141" data-ref="1629ca85-50f2-423f-8ccc-3bff571911e0" data-id="el-p-141">The role of FGFs in embryonic development and stem cell differentiation has remained unclear due to its complexity. In this study, the authors utilized a 2D human stem cell-based gastrulation model to investigate the functions of FGFs. They discovered that FGF-dependent ERK activity is closely linked to the emergence of primitive streak cells. Importantly, this 2D model effectively illustrates the spatial distribution of key signaling effectors and receptors by correlating these markers with cell fate markers, such as T and ISL1. Through inhibition and loss-of-function studies, they further corroborated the needs of FGF ligands. Their data shows that FGFR1 is the primary receptor, and FGF2/4/17 are the key ligands for primitive streak development, which aligns with observations in primate embryos. Additional experiments revealed that the reduction of FGF4 and FGF17 decreases ERK activity.</p><p id="el-p-142" data-ref="84320283-6e0b-4397-a189-1e23b6e0c697" data-id="el-p-142">Strengths:</p><p id="el-p-143" data-ref="98b3d4fe-dd00-4aed-acba-2f68abb161c2" data-id="el-p-143">This study provides comprehensive data and improves our understanding of the role of FGF signaling in primate</p><p id="el-p-144" data-ref="dc9049aa-9249-48ec-bb89-0a417d6c134b" data-id="el-p-144">primitive streak formation. The authors provide new insights related to the spatial localization of the key components of FGF signaling and attempt to reveal the temporal dynamics of the signal propagation and cell fate decision, which has been challenging.</p><p id="el-p-145" data-ref="91a9e6a8-293d-44eb-a58c-0a5383ea1d55" data-id="el-p-145">Weaknesses:</p><p id="el-p-146" data-ref="1ff2b665-6657-43ab-9468-94bafb245fef" data-id="el-p-146">Given the solid data, the work only partially clarifies the complex picture of FGF signaling, so details remain somewhat elusive. The findings lack a strong punchline, which may limit their broader impact.</p></blockquote><p id="el-p-147" data-ref="874a877c-3b78-4e6c-8317-e4b43d7d76d0" data-id="el-p-147">We thank this reviewer for their valuable feedback and compliment on the solidity of our data. The punchline of our work is that FGF4 and FGF17-dependent ERK signaling plays a key role in differentiation of human PS-like cells and mesoderm, and that these are different FGFs than those thought to drive mouse gastrulation. A second key point is that like BMP and TGFβ signaling, FGF signaling is restricted to the basolateral sides of pluripotent stem cell colonies due to polarized receptor expression, which is crucial for understanding the response to exogenous ligands added to the cell medium. Indeed, many facets of FGF signaling remain to be investigated in the future, such as how FGF regulates and is regulated by other signals, which we will dedicate a different manuscript to.</p><blockquote data-ref="4b6cc45c-61e6-4b68-b9b8-9f6c9cc68586" data-split-to="4b6cc45c-61e6-4b68-b9b8-9f6c9cc68586" data-split-original="true"><p id="el-p-148" data-ref="5b4e6afd-5fec-4b3e-ac65-17333ed51da6" data-id="el-p-148" data-split-to="5b4e6afd-5fec-4b3e-ac65-17333ed51da6" data-split-original="true" data-last-split-element="true" data-align-last-split-element="auto"></p></blockquote></div></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_left_page" data-page-number="38" id="page-38">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e" data-split-to="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2" data-id="peer-reviews" data-split-from="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2" data-split-to="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2"><section class="author-response" data-ref="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4" data-id="authorresponse" data-split-from="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4" data-split-to="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4"><div class="author-response__body" data-ref="0f8a4845-8525-4632-882f-9a92142367e9" data-id="el-div-22" data-split-from="0f8a4845-8525-4632-882f-9a92142367e9" data-split-to="0f8a4845-8525-4632-882f-9a92142367e9" data-last-split-element="true" data-align-last-split-element="auto"><blockquote data-ref="4b6cc45c-61e6-4b68-b9b8-9f6c9cc68586" data-split-from="4b6cc45c-61e6-4b68-b9b8-9f6c9cc68586"><p data-ref="5b4e6afd-5fec-4b3e-ac65-17333ed51da6" data-id="el-p-148" data-split-from="5b4e6afd-5fec-4b3e-ac65-17333ed51da6"><strong data-ref="61f3a7e5-1597-4879-97b5-c976cfec3571">Reviewer #3 (Public review):</strong></p><p id="el-p-149" data-ref="feb874db-fca4-4d0a-8248-40648868d5aa" data-id="el-p-149">Jo and colleagues set out to investigate the origins and functions of localized FGF/ERK signaling for the differentiation and spatial patterning of primitive streak fates of human embryonic stem cells in a well-established micropattern system. They demonstrate that endogenous FGF signaling is required for ERK activation in a ringdomain in the micropatterns, and that this localized signaling is directly required for differentiation and spatial patterning of specific cell types. Through high-resolution microscopy and transwell assays, they show that cells receive FGF signals through basally localized receptors. Finally, the authors find that there is a requirement for exogenous FGF2 to initiate primitive streak-like differentiation, but endogenous FGFs, especially FGF4 and FGF17, fully take over at later stages.</p><p id="el-p-150" data-ref="6b33f7b7-80f6-4d8d-8501-569b0f1ff3c8" data-id="el-p-150">Even though some of the authors' findings - such as the localized expression of FGF ligands during gastrulation and the importance of FGF/ERK signaling for cell differentiation in the primitive streak - have been reported in model organisms before, this is one of the first studies to investigate the role of FGF signaling during primitive streak-like differentiation of human cells. In doing so, the paper reports a number of interesting and valuable observations, namely the basal localization of FGF receptors which mirrors that of BMP and Nodal receptors, as well as the existence of a positive feedback loop centered on FGF signaling that drives primitive-streak differentiation. The authors also perform a comparison of the role of different FGFs across species and try to assign specific functions to individual FGFs. In the absence of clean genetic loss-of-function cell lines, this part of the work remains less strong.</p></blockquote><p id="el-p-151" data-ref="db6a81d7-9345-494f-b550-5ea60fd1c30f" data-id="el-p-151">We thank the reviewer for emphasizing the value of our findings in a human model for gastrulation. We agree more loss-of-function experiments would provide further insight into the role of different FGFs. While we did not manage to create knockout cell lines, we have now performed both siRNA and shRNA knock-down of all FGF4, and FGF17 in two different hPSC lines, performed siRNA knockdown of FGF8, and also made a FGF4+FGF17 shRNA double knockdown cell lines to more completely test the functions of the individual FGFs (Fig.5, Supp.Fig.5,6). Our data suggest FGF17 may be downstream of FGF4 and primarily required for mesoderm differentiation while FGF8 appears to counteract FGF4. In doing this we have added a large amount of new data to the manuscript and we have removed the heterozygous knockout data in the first version of the manuscript which we felt added little to the new data. Further experiments are still needed to solidify our interpretation but those are beyond the scope of the current work.</p><blockquote data-ref="4548a455-b5e0-490c-8005-6b1f0bdd20d1"><p id="el-p-152" data-ref="03d36b1a-cfed-4f17-9ca5-7854e8ac57ea" data-id="el-p-152"><strong data-ref="1b914778-829f-46e5-b75a-e0d895690ec8">Reviewer #1 (Recommendations for the authors):</strong></p><p id="el-p-153" data-ref="6ae9a476-28d9-43da-85f3-1c0ded8d9798" data-id="el-p-153">(1) FGF2 is added to culture experiments (e.g. Figure 4), but the commercial source is not mentioned in Methods. For example, it could be added to "Supplementary Table 1: Cell signaling reagents."</p></blockquote><p id="el-p-154" data-ref="09597eb2-8517-48e6-9ef5-ea1c8c252b93" data-id="el-p-154">We apologize for this oversight and have now added the information to Supplementary Table 1.</p><blockquote data-ref="236c7b38-413a-4674-9e1b-ae220d76bf27"><p id="el-p-155" data-ref="c885b867-8b87-4ad7-ae54-05b73225341b" data-id="el-p-155">(2) Line 117-118: "For example, by controlling the expression of Wnt or Nodal which are both required for PS-like differentiation". It is clear what the authors mean, but this is not a complete sentence.</p></blockquote><p id="el-p-156" data-ref="fdbcc30a-f67e-4242-828d-834d73d7ccf5" data-id="el-p-156">We edited this for clarity, it now reads: “First, is FGF/ERK signaling required directly for PS-like differentiation, or does it act indirectly? These possibilities are not mutually exclusive. For example, FGF/ERK could be required directly but also act indirectly by controlling Wnt or Nodal expression, as both Wnt and Nodal signaling are required for PS-like differentiation.”</p></div></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_right_page" data-page-number="39" id="page-39">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e" data-split-to="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2" data-id="peer-reviews" data-split-from="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2" data-split-to="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2"><section class="author-response" data-ref="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4" data-id="authorresponse" data-split-from="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4" data-split-to="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4"><div class="author-response__body" data-ref="0f8a4845-8525-4632-882f-9a92142367e9" data-id="el-div-22" data-split-from="0f8a4845-8525-4632-882f-9a92142367e9" data-split-to="0f8a4845-8525-4632-882f-9a92142367e9"><blockquote data-ref="0b93fd8c-2615-401c-bfff-e33fc83e60b0"><p id="el-p-157" data-ref="556f3429-6fcd-4923-b18d-d2716514e4d0" data-id="el-p-157">(3) Line 246 "...found its spatial pattern to strongly resembles that of pERK..." either remove "to" or change "resembles" to "resemble"</p></blockquote><p id="el-p-158" data-ref="a9d0d2c5-abec-4c88-a826-cb377f58b619" data-id="el-p-158">Thank you for catching this. We removed “to”.</p><blockquote data-ref="166f5710-3d6c-4ce0-a0c0-4462f36cdb09"><p id="el-p-159" data-ref="3bc0d905-ad18-460c-b22a-df96cb09be2b" data-id="el-p-159">(4) Lines 391- 393 seem to be missing a word in the last phrase: "...with FGF17 more important continued differentiation to mesoderm and endoderm." Maybe "during" after the word "important"?</p></blockquote><p id="el-p-160" data-ref="8006f94b-800f-43e7-ac86-ca302f941322" data-id="el-p-160">Thank you for catching this, indeed the word “during” was missing and we have now added it.</p><blockquote data-ref="aceb50a3-a0a8-4ffb-8de6-1b5d12f76ead"><p id="el-p-161" data-ref="bc5a7ed5-18c5-402f-b638-5e2ee6033428" data-id="el-p-161">(5) Please define acronyms in Figure 3D (PS-LC was defined previously, but not others).</p></blockquote><p id="el-p-162" data-ref="b30896c0-4b7c-4bc9-9f20-a997db65c709" data-id="el-p-162">We apologize for the oversight, we have now defined the acronyms.</p><blockquote data-ref="de386c51-758a-44ee-be12-43c2cb31b100"><p id="el-p-163" data-ref="ac474532-b3a3-48ad-a713-a312335b28ca" data-id="el-p-163">(6) The three blue lines in Figure 5B (right) are hard to discern (and I'm not colorblind). I suggest also using a variety of dotted lines in a subset of these FGFs.</p></blockquote><p id="el-p-164" data-ref="b64f4dce-8b64-49e3-a890-880d10cafcb2" data-id="el-p-164">Thanks you for the suggestion. We have now given all the FGFs colors that are more clearly distinct and made the TBXT and TBX6 lines dashed.</p><blockquote data-ref="c3d1b126-db69-4d0e-bdef-3aaf616667a3"><p id="el-p-165" data-ref="c6d61b3b-6752-4bc4-9693-f097736a984b" data-id="el-p-165"><strong data-ref="a0c878bc-57f0-444c-a206-f330e87bbf34">Reviewer #2 (Recommendations for the authors):</strong></p><p id="el-p-166" data-ref="55638ac4-cef1-4134-b9fd-ae51103a22c2" data-id="el-p-166">(1) The reviewer acknowledges that FGF signaling is complex, particularly when dynamics and its correlation with cell fates are considered. To improve the clarity of the findings, the authors are encouraged to provide an additional schematic figure that clearly delineates the main findings of this study.</p></blockquote><p id="el-p-167" data-ref="9e6c4d62-837c-4bfc-a279-1e7b8ac12c42" data-id="el-p-167">Thank you for the suggestion. We have now added a summary figure (Fig.6) to our discussion, which we hope helps present our findings more clearly.</p><blockquote data-ref="df55a656-68d9-471c-af21-33db742d0536"><p id="el-p-168" data-ref="0b428f03-e2ec-4a2f-bd99-2b3eaad396ea" data-id="el-p-168">(2) The data suggest that FGF signaling may function differently in mice compared to primates, and their stem cell model aligns more closely with the latter. While the authors discuss this in the contents only based on sequencing data, it would be valuable to conduct some experiments with mouse embryos to validate the key differences.</p></blockquote><p id="el-p-169" data-ref="de62f7da-998e-499f-90e4-cc46b6f0cc97" data-id="el-p-169">It is unclear to us which experiments the reviewer has in mind. There is ample data on FGF expression in the mouse literature, as are many knockout phenotypes. Furthermore, verifying loss of function phenotypes (e.g. FGF17 knockout) in mouse is beyond our expertise.</p><blockquote data-ref="fe0dfccb-2fc6-4562-ac23-a67963c68cab"><p id="el-p-170" data-ref="e4b1c23f-4bd5-430f-9e94-1243604f27cd" data-id="el-p-170">(3) Heparan sulfate proteoglycan (HSPG) is mentioned as an important component of FGF signaling; however, the only data related to HSPG is single-cell sequencing results. The authors should consider performing immunostaining or other assays to validate HSPG expression and spatial distribution, similar to the approach they used for other signaling components.</p></blockquote><p id="el-p-171" data-ref="b341e462-efbe-4496-a031-64bc5672ed38" data-id="el-p-171">Our scratch experiments in Fig. 4 strongly argue against HSPGs as being responsible for the spatial pattern of FGF receptor activation: after a scratch across the colony the response is strong all along the scratch as expected if presence of FGF (an FGF gradient) controls the level of activity. If HSPGs were limiting, FGF flowing in from the media show not be able to uniformly activate receptors around the scratch.</p><p id="el-p-172" data-ref="3d100854-3ebc-43f9-a7cf-c6b7b33de8f4" data-id="el-p-172">In addtion, we have now included an immunostain for HS in a newly added Supp. Fig. 4 which does not explain the observed pattern of ERK signaling.</p><blockquote data-ref="ee42498b-1a1d-4bff-8ea4-38f2cee120cd" data-split-to="ee42498b-1a1d-4bff-8ea4-38f2cee120cd" data-split-original="true"><p id="el-p-173" data-ref="fcbca6f8-9ea7-4661-b95e-c9ee911f7fac" data-id="el-p-173" data-split-to="fcbca6f8-9ea7-4661-b95e-c9ee911f7fac" data-split-original="true" data-last-split-element="true" data-align-last-split-element="auto">(4) In the scratch experiment, particularly high PERK expression is observed at the edge of the scratch. The authors should provide an explanation for why this expression is </p></blockquote></div></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_left_page" data-page-number="40" id="page-40">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e" data-split-to="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2" data-id="peer-reviews" data-split-from="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2" data-split-to="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2"><section class="author-response" data-ref="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4" data-id="authorresponse" data-split-from="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4" data-split-to="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4"><div class="author-response__body" data-ref="0f8a4845-8525-4632-882f-9a92142367e9" data-id="el-div-22" data-split-from="0f8a4845-8525-4632-882f-9a92142367e9" data-split-to="0f8a4845-8525-4632-882f-9a92142367e9"><blockquote data-ref="ee42498b-1a1d-4bff-8ea4-38f2cee120cd" data-split-from="ee42498b-1a1d-4bff-8ea4-38f2cee120cd"><p data-ref="fcbca6f8-9ea7-4661-b95e-c9ee911f7fac" data-id="el-p-173" data-split-from="fcbca6f8-9ea7-4661-b95e-c9ee911f7fac">significantly higher compared to the edges of the colony. Additionally, it would be interesting to investigate the fate of the cells with super high PERK expression.</p></blockquote><p id="el-p-174" data-ref="ae5ad073-6944-487a-9124-61a131a98c40" data-id="el-p-174">We have now determined that adaptive response to FGF is the reason that the response around the scratch is initially much higher than in the ERK activity ring that overlaps with the primitive streak-like cells. We have added figures showing that although the intial response to FGF exposure after scratching is very high, the response around the scratch adapts to levels similar in those in the ERK ring over the course of 6 hours (Fig.4ij).</p><blockquote data-ref="8d1de816-5b30-4ba7-baf1-01de141deed2"><p id="el-p-175" data-ref="5b00ff87-20fb-4adb-b2b2-23df0d14bfda" data-id="el-p-175">(5) For some of the key experiments, multiple cell lines should be used to ensure that the findings are reproducible and applicable across different human stem cell lines.</p></blockquote><p id="el-p-176" data-ref="4a28224f-9e57-4a0d-85f7-bf4c4774a3cf" data-id="el-p-176">We have now checked FISH stainings and knockdown phenotypes for different FGFs in two different cell lines: ESI17 (hESC, XX) and PGP1 (hiPSC, XY). These results are shown in Supplementary Figures 6. We found all results to be consistent.</p><blockquote data-ref="babb514e-6bba-41f5-8338-1fcf2c9f0cff"><p id="el-p-177" data-ref="e0b6db53-1e9b-4c2c-879f-0cea05cb7f5f" data-id="el-p-177">(6) Where applicable, the meaning of error bars needs to be more clearly presented, including details on the number of independent experiments or samples used.</p></blockquote><p id="el-p-178" data-ref="9ac2e39d-6328-4e8f-95bd-08f6a8c18009" data-id="el-p-178">Thank you for pointing this out. Where error bar definitions were missing we have now added them to the figure captions.</p><blockquote data-ref="fad3964d-2d83-4e1a-a410-a9fe38effb78"><p id="el-p-179" data-ref="9bcfd5e5-ea35-4c56-a586-b7074877fbb6" data-id="el-p-179"><strong data-ref="1152392c-af94-423f-90ca-9bc1090a0949">Reviewer #3 (Recommendations for the authors):</strong></p><p id="el-p-180" data-ref="64117695-8b46-4389-8f7d-72b4f2e94c79" data-id="el-p-180">(1) The authors only analyze the ppERK ring in micropatterns of a single size. What was the motivation for the choice of this size? Can the authors how the ppERK ring is expected to depend on colony size?</p></blockquote><p id="el-p-181" data-ref="37ae54cf-179d-47b1-86ac-c5fe6c8e6f02" data-id="el-p-181">Much smaller patterns lose the interior pluripotent regions while much larger patters have a much larger pluripotent region, which requires larger tilings to image without providing additional insight. The colony sizedependence of cell fate patterning was described in the paper that established the 2D gastruloids model (Warmflash Nat Methods 2014) and we later showed this due to a fixed length scale of the BMP and Nodal signaling gradients from the colony edge (Jo et al Elife 2022). We have now included data showing that the ERK patterns behaves similarly, with a fixed length scale of the pattern implying that in smaller colonies the ERK ring becomes a disc and the entire center of the colony has high ERK signaling (Supp Fig 1a).</p><blockquote data-ref="010f865e-4f5a-46c2-a426-14a83fc6df3a"><p id="el-p-182" data-ref="3d6dcb5e-fb91-4dba-8386-75f966265448" data-id="el-p-182">(2) The scRNAseq is somewhat confusing - why do the two datasets not overlap in the PHATE representation? This is unexpected, because the two samples have been treated similarly, and the authors have integrated their data to iron out possible batch effects. This discrepancy should be discussed. The authors should also specify from which reference exactly the first dataset comes from.</p></blockquote><p id="el-p-183" data-ref="4f54e9f3-165e-4258-9d31-5a3d7c7fa5c9" data-id="el-p-183">The two datasets do overlap nicely, the same fates are well mixed in the same place and the gene expresison profiles for the integrated data (e.g., Fig.2e) look smooth, so we believe the integration is good, but different cell fates are represented to different degrees. In particular, sample 2 shows much more mesoderm differentiation making the mesoderm branch mostly orange. Occassionally samples differentiate faster or slower than average which we see here, and these samples were collected far apart in time. We do not believe this affects our conclusions, if anything, we think performing the analysis on two samples that differ this much should make the conclusions more robust.</p><blockquote data-ref="25edf3c3-410a-406d-957b-fc0b5feebfae" data-split-to="25edf3c3-410a-406d-957b-fc0b5feebfae" data-split-original="true"><p id="el-p-184" data-ref="884be425-1408-4c96-9c2c-8b46a7a3e6e4" data-id="el-p-184" data-split-to="884be425-1408-4c96-9c2c-8b46a7a3e6e4" data-split-original="true" data-last-split-element="true" data-align-last-split-element="auto">(3) If find it intriguing that exogenous FGF2 is important early on for primitive streak-like differentiation, although the authors show that it does not reach the center of the colony. The authors may want to discuss this conundrum. Does the FGF2 effect propagate from </p></blockquote></div></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_right_page" data-page-number="41" id="page-41">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e" data-split-to="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2" data-id="peer-reviews" data-split-from="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2" data-split-to="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2"><section class="author-response" data-ref="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4" data-id="authorresponse" data-split-from="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4" data-split-to="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4"><div class="author-response__body" data-ref="0f8a4845-8525-4632-882f-9a92142367e9" data-id="el-div-22" data-split-from="0f8a4845-8525-4632-882f-9a92142367e9" data-split-to="0f8a4845-8525-4632-882f-9a92142367e9"><blockquote data-ref="25edf3c3-410a-406d-957b-fc0b5feebfae" data-split-from="25edf3c3-410a-406d-957b-fc0b5feebfae"><p data-ref="884be425-1408-4c96-9c2c-8b46a7a3e6e4" data-id="el-p-184" data-split-from="884be425-1408-4c96-9c2c-8b46a7a3e6e4">the outside to the inside, or does it act at an early stage when the cells have not yet formed a tight epithelium on the micropattern?</p></blockquote><p id="el-p-185" data-ref="77812c1c-a04c-478d-ade0-2c0825a8c5ec" data-id="el-p-185">The cells in the experiment in Fig. 5a were given 24h to epithelialize, so we we do believe it acts from the edge. We believe this may be due to FGF2 modulating the early BMP response on the edge and are working on a manuscript that further explores this pathway crosstalk.</p><blockquote data-ref="94705489-5ba3-4c47-8074-db918d8d9514"><p id="el-p-186" data-ref="3972e677-d70e-4812-93b5-60068a45d9a6" data-id="el-p-186">(4) The authors' statement that FGF4 and FGF17 have partially redundant functions is not very strong, mainly because the study lacks a full FGF17 loss-of-function cell line. If the authors wanted to improve on this point, they could knock down FGF4 in the FGF17 heterozygous line, or produce a homozygous FGF17 KO line. If there are specific reasons why FGF17 homozygous lines cannot be produced, this could be interesting to discuss, too. Finally, I noticed that the methods list experiments with an FGF17 siRNA, but these are not shown in the manuscript.</p></blockquote><p id="el-p-187" data-ref="f08ee8b3-33ea-478f-b81f-4b00260c24b1" data-id="el-p-187">We agree our evidence was previously not as strong as it could be. While there is no reason we know of why homozygous knockout lines cannot be produced, we failed to produce on. To strengthen our evidence we have therefore included substantial new knockdown data.  We have now performed both siRNA and shRNA knockdown of all FGF4, and FGF17 in two different hPSC lines, performed siRNA knockdown of FGF8, and also made a FGF4+FGF17 shRNA double knockdown cell lines to more completely test the functions of the individual FGFs (Fig.5, Supp.Fig.5,6). These experiments showed that FGF17 knockdown had a much smaller effect than FGF4 knockdown on expression of primitive streak markers (Fig.5i, Supp.Fig.6f-i) but that FGF17 knockdown did lead to a complete loss of the mesoderm marker TBX6 (Fig.5j, Supp.Fig.6j). A double knockdown of FGF4+FGF17 looked similar to FGF4 alone (Supp.Fig.6k). Thus, we now think the more likely scenario is that FGF17 is downstream of FGF4-dependent PS-differentiation and although this may have a positive feedback effect whereby this FGF17 can then enhance further PS-differentiation, which we previously interpreted as partial redundancy, the primary role of FGF17 may be later, in mesoderm differentiation. Furthermore, our new data suggests FGF8 may counteract FGF4 and limit PS-like differentiation.</p><blockquote data-ref="75d1540d-41a6-4ac2-a018-1081d4589ab0"><p id="el-p-188" data-ref="4ba1e09e-b7d6-4145-aa89-1dea7e4c549a" data-id="el-p-188">Minor</p><p id="el-p-189" data-ref="bbe3f4e1-1237-4d1e-87f9-58fbca07e856" data-id="el-p-189">(5) Line 63: Reference(s) appear to be missing.</p></blockquote><p id="el-p-190" data-ref="0a834485-713d-4359-839d-8a3e27b648c0" data-id="el-p-190">This whole paragraph summarizes the results of the references given on line 55, we have now repeated the relevant references where the reviewer indicated.</p><blockquote data-ref="fe1776c8-3836-438e-b7c2-db6160b7acd2"><p id="el-p-191" data-ref="aff9ab34-128c-4260-902e-728151cc1c7c" data-id="el-p-191">(6) Supplementary Figure 1a,b does not show ppERK, unlike stated in lines 102 - 104.</p></blockquote><p id="el-p-192" data-ref="8332b666-9474-44a0-b8f3-86c96dba235f" data-id="el-p-192">Indeed, the data described in lines 102-104 is shown in Fig.1a and we have removed the original Supplementary Figure 1ab since it did not provide relevant information.</p><blockquote data-ref="cb3ae97f-2548-4759-96c8-8cf35f027498"><p id="el-p-193" data-ref="dbeda532-7563-4224-bc52-4c331197061a" data-id="el-p-193">(7) Line 201: It is not clear whether this is a new sequencing dataset, or if existing datasets have been reanalyzed.</p></blockquote><p id="el-p-194" data-ref="d3fc1e99-c5c9-4621-9430-7549a5eee57b" data-id="el-p-194">We agree our description was unclear. We have edited the text, which now explicitly states that our analysis is based on one dataset we collected previously and a replicate that was newly collected and deposited on GEO for this manuscript.</p><blockquote data-ref="addd5004-2dd8-464b-8fba-d345d401dc25"><p id="el-p-195" data-ref="5afab035-4833-4317-9c28-53d7145238d6" data-id="el-p-195">(8) Figure 2f; Supplementary Figure 2b, c: The colors need to be explained in scale bars. How has this data been normalized to allow for comparison between very different sample types?</p></blockquote><p id="el-p-196" data-ref="755645eb-a66e-4bb8-8ad7-1447bc619358" data-id="el-p-196" data-split-to="755645eb-a66e-4bb8-8ad7-1447bc619358" data-split-original="true" data-last-split-element="true" data-align-last-split-element="auto">We have now added color bars indicating the scale for each of these figure panels. As the caption stated, the interspecies comparison was normalized within each species, so the highest FGF level for any FGF at any time within each species is normalized to one. We are </p></div></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div>
<div class="pagedjs_page pagedjs_left_page" data-page-number="42" id="page-42">
	<div class="pagedjs_sheet">
		<div class="pagedjs_bleed pagedjs_bleed-top">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-bottom">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-left">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>		<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_bleed pagedjs_bleed-right">
			<div class="pagedjs_marks-crop"></div>
			<div class="pagedjs_marks-middle">
				<div class="pagedjs_marks-cross"></div>
			</div>
			<div class="pagedjs_marks-crop"></div>
		</div>
		<div class="pagedjs_pagebox"><ul class="article-flag-list" id="el-ul-0" data-ref="e4a0f4c0-014d-4d9b-9a48-e6a72d6dcec6" data-id="el-ul-0" style="position: absolute;"><li class="article-flag-list__item" id="el-li-0" data-ref="11c430be-24df-4704-a186-9587a2f242a3" data-id="el-li-0"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology" data-ref="f9082a82-cfc2-4e1d-a1ac-fb3d66defcec">Developmental Biology</a></li><li class="article-flag-list__item" id="el-li-1" data-ref="deee70df-f9ba-4ce5-915b-a4fff3267927" data-id="el-li-1"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine" data-ref="7acdd927-7150-4177-b111-994dad909936">Stem Cells and Regenerative Medicine</a></li></ul><p class="runninghead" id="el-p-0" data-ref="691f660b-89fd-48b7-a97b-7adfb6b783ea" data-id="el-p-0" style="position: absolute;"><span class="author" data-ref="a4253b79-5599-4fd4-a81a-6ede1a2e53d0">Jo, Liu et al.,</span>&nbsp;<span class="date" data-ref="e88680bd-49c6-41a9-b4f3-758e79f32f97">2024&nbsp;<em data-ref="e4b6fe3d-bd61-4231-a836-69099ed3b934">eLife</em>&nbsp;<strong data-ref="251fe760-0a7c-4460-ba45-0bd14de0311c">13</strong>:RP101224.</span>&nbsp;<span class="doi" data-ref="60e6c976-76ed-4102-997e-6b977bf8863b">&nbsp;<a href="https://doi.org/10.7554/eLife.101224.2" data-ref="455fde8d-2406-4989-822f-72e89cf904c6">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter" data-ref="8b4f891b-33a3-401b-a7ab-4645d22fc761"></span></p>
			<div class="pagedjs_margin-top-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top" style="grid-template-columns: auto 1fr auto;">
				<div class="pagedjs_margin pagedjs_margin-top-left hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking" data-ref="37d25f19-f1b4-4433-9401-b4c1647726c9" data-id="logo" style=""><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo max-10" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg" id="el-img-0" data-ref="1945edb4-7b59-4c4d-835a-43667f4f6e5f" data-id="el-img-0" data-undisplayed="undisplayed" data-desired-width="3.787878787878788"></a></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-top-right hasContent" style="white-space: normal;"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-top-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-top-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-right">
				<div class="pagedjs_margin pagedjs_margin-right-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-right-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-left">
				<div class="pagedjs_margin pagedjs_margin-left-top"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-middle"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-left-bottom"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-left-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-left-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom">
				<div class="pagedjs_margin pagedjs_margin-bottom-left"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-center"><div class="pagedjs_margin-content"></div></div>
				<div class="pagedjs_margin pagedjs_margin-bottom-right"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_margin-bottom-right-corner-holder">
				<div class="pagedjs_margin pagedjs_margin-bottom-right-corner"><div class="pagedjs_margin-content"></div></div>
			</div>
			<div class="pagedjs_area">
				<div class="pagedjs_page_content" style="column-width: 716px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left) + var(--pagedjs-bleed-right) + var(--pagedjs-bleed-left) + var(--pagedjs-column-gap-offset));"><div><article data-ref="c36d516e-61de-4652-b357-b350359ef39e" data-split-from="c36d516e-61de-4652-b357-b350359ef39e"><section data-ref="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2" data-id="peer-reviews" data-split-from="2d3a091d-cc27-409e-a88b-4c5cf5c68cf2"><section class="author-response" data-ref="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4" data-id="authorresponse" data-split-from="8b5878aa-d9cc-4f3e-8a7a-3a8b214fbad4"><div class="author-response__body" data-ref="0f8a4845-8525-4632-882f-9a92142367e9" data-id="el-div-22" data-split-from="0f8a4845-8525-4632-882f-9a92142367e9"><p data-ref="755645eb-a66e-4bb8-8ad7-1447bc619358" data-id="el-p-196" data-split-from="755645eb-a66e-4bb8-8ad7-1447bc619358">thus comparing between species the relative expression of different FGFs within each species. Indeed there is no good way to compare absolute expression between species. For extra clarity we have expanded our description of the interspecies comparison analysis and normalization in the methods section.</p><blockquote data-ref="49f19baa-439f-4682-86b5-bd85f5fa2b96"><p id="el-p-197" data-ref="afed945c-0dfa-4ade-bd74-28793ebbb1f9" data-id="el-p-197">(9) Line 232: Where is the expression of SEF shown?</p></blockquote><p id="el-p-198" data-ref="f4c8608f-8dd0-42a9-88f9-49d074851685" data-id="el-p-198">It is shown in Fig. 2i, under the official gene name IL17RD.</p><blockquote data-ref="33c1e790-f501-44a3-8fd8-37a66f6fb360"><p id="el-p-199" data-ref="7e4be313-7268-4ced-96f4-7a3ab2fdd07d" data-id="el-p-199">(10) Supplementary Figure 4 seems to be missing.</p></blockquote><p id="el-p-200" data-ref="f1527a65-24aa-406b-aeee-60422af9a89a" data-id="el-p-200">Thank you for pointing this out. We have now added a supplementary Fig.4.</p><blockquote data-ref="cf8e1f7c-51c2-43e8-8e17-1f70a33cf7af"><p id="el-p-201" data-ref="9a5a8618-8a80-4642-be16-bee1d331ca2e" data-id="el-p-201">(11) Line 437: Citation needed.</p></blockquote><p id="el-p-202" data-ref="cc506633-1897-4317-bd0d-126a0ddd3d2c" data-id="el-p-202">We have included citations now.</p><blockquote data-ref="0abd1e02-0f52-414b-b3a2-4e4842848dee"><p id="el-p-203" data-ref="28bfa186-2253-44ec-b2ce-73a9fffe78a7" data-id="el-p-203">(12) Line 439: A similar feedback loop has been proposed to operate during mesoderm differentiation in mouse ESC (pmid: 37530863 ). The authors may consider citing this work.</p></blockquote><p id="el-p-204" data-ref="1cfb774f-ef23-4a38-b51d-931470733b89" data-id="el-p-204">Thank you for the suggestion, we have now included this work in the discussion. The feedback loop proposed in that work involves FGF8, while we were trying to explain why FGF4 and not FGF8 appears to be conserved across species by invoking an FGF4 feedback loop. Thus, it becomes even harder to explain differences in FGF4 and FGF8 expression between human and mouse gastrulation.</p><blockquote data-ref="a412a7c4-fabd-45eb-abe7-322070a82a40"><p id="el-p-205" data-ref="93d285fd-3a22-4eaf-94c3-e72256d7d37f" data-id="el-p-205">(13) Supplementary Figure 6 is not described in the main text.</p></blockquote><p id="el-p-206" data-ref="bc49f904-c6e9-46a5-a68c-32fbc15bffea" data-id="el-p-206">We have removed the original Supplementary Figure 6 and corresponding heterozygous knockout data in the main figure which we felt added little to the extensive knockdown data we now present. We did create a new Supplementary Figure 6 showing additional knockdown data which is described in the main tekst.</p><blockquote data-ref="23f1f301-f725-4cee-9886-e4d00671257c"><p id="el-p-207" data-ref="d90c26a0-41b1-4982-b088-e4c5f0371d49" data-id="el-p-207">(14) Submission of sequencing data to GEO needs to be updated.</p></blockquote><p id="el-p-208" data-ref="2d1c6c48-f7da-4f6b-b793-52267b4fcbe3" data-id="el-p-208">We have now made the GEO data public.</p></div><div class="descriptors" id="el-div-23" data-ref="851a1df4-2470-4607-b82f-13cedb6a79bf" data-id="el-div-23"><ul class="descriptors__identifiers" id="el-ul-10" data-ref="97d17940-f71c-49d0-9766-6e3a070a3749" data-id="el-ul-10"><li class="descriptors__identifier" id="el-li-373" data-ref="b3123223-0a14-4990-99d1-d9436dff211d" data-id="el-li-373"><a href="https://doi.org/10.7554/eLife.101224.2.sa0" data-ref="744a5acb-cf2e-465a-81e7-2a1aebbfcbce">https://doi.org/10.7554/eLife.101224.2.sa0</a></li></ul></div></section></section></article></div></div>
				<div class="pagedjs_footnote_area">
					<div class="pagedjs_footnote_content pagedjs_footnote_empty">
						<div class="pagedjs_footnote_inner_content" style="column-width: 0px; column-gap: calc(var(--pagedjs-margin-right) + var(--pagedjs-margin-left)); height: auto;"></div>
					</div>
				</div>
			</div>
		</div>
	</div>
</div></div></body></html>