<!DOCTYPE html>
<html><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charSet="utf-8"><meta name="viewport" content="width=device-width"><title>Endogenous FGFs drive ERK-dependent cell fate patterning in 2D human gastruloids</title><link rel="preload" href="print.css" as="style"><link rel="stylesheet" href="print.css"></head><body><div id="page-templates"><p class="runninghead"><span class="author">Jo, Liu et al.,</span> <span class="date">2024 <em>eLife</em> <strong>13</strong>:RP101224.</span> <span class="doi"> <a href="https://doi.org/10.7554/eLife.101224.2">https://doi.org/10.7554/eLife.101224.2</a></span><span class="counter"></span></p><ul class="article-flag-list"><li class="article-flag-list__item"><a class="article-flag__link" href="https://elifesciences.org/subjects/developmental-biology">Developmental Biology</a></li><li class="article-flag-list__item"><a class="article-flag__link" href="https://elifesciences.org/subjects/stem-cells-regenerative-medicine">Stem Cells and Regenerative Medicine</a></li></ul><div id="first-page-links" class="elife-intro"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking"><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo" style="color:transparent" src="https://elifesciences.org/assets/patterns/img/patterns/organisms/elife-logo-xs.fd623d00.svg"></a><div id="right-header-links"><a href="https://en.wikipedia.org/wiki/Open_access" target="_blank" class="header-icon oa-icon"></a><span class="separator">|</span><a href="https://creativecommons.org/" target="_blank" class="header-icon cc-icon"></a></div></div></div><article><section id="sideSection" class="side-section"><div class="article-status"><div class="review-timeline-container"><dl class="review-timeline review-timeline--expanded" id="review-timeline" aria-label="Version history"><dt class="review-timeline__event review-timeline__event--title review-timeline__event--active review-timeline__event--with-evaluation-summary review-timeline__event--reviewed"><a href="https://doi.org/10.7554/eLife.101224.1" class="review-timeline__event-link">Reviewed Preprint</a></dt><dd class="review-timeline__event review-timeline__event--detail review-timeline__event--active review-timeline__event--with-evaluation-summary review-timeline__event--reviewed"><span class="review-timeline__version">v1</span><time class="review-timeline__date" dateTime="2024-11-08T00:00:00.000Z">November 8, 2024</time><span class="review-timeline__link">Not revised</span></dd><dt class="review-timeline__event review-timeline__event--title review-timeline__event--active review-timeline__event--with-evaluation-summary review-timeline__event--revised">Reviewed Preprint</dt><dd class="review-timeline__event review-timeline__event--detail review-timeline__event--active review-timeline__event--with-evaluation-summary review-timeline__event--revised"><span class="review-timeline__version"> v2</span><time class="review-timeline__date" dateTime="2025-10-29T00:00:00.000Z">October 29, 2025</time><span class="review-timeline__link">Revised by authors</span></dd></dl></div></div><section id="margin-notes"><div class="author-notes"><p><strong class="email-icon">For correspondence:</strong><ul class="list-simple"><li><a href="mailto:test@umich.edu">test@umich.edu</a></li><li><a href="mailto:iheemske@umich.edu">iheemske@umich.edu</a></li></ul></p><p class="author-notes__list_item"><sup aria-hidden="true"><strong>*</strong></sup> Equal contributions</p><p class="author-notes__list_item"><strong>Competing interests:</strong> No competing interests declared</p><p class="author-notes__list_item"><strong>Funding:</strong> See <a class="page-ref" href="#funding"></a></p></div><div id="editors"><p class="editors-and-reviewers__person"><strong>Reviewing editor: </strong>Sergio Ruiz Macias, National Cancer Institute, NIH, United States</p></div><div id="copyright" class="copyright"><p>© 2024, Jo et al. This article is distributed under the terms of the <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution License</a>, which permits unrestricted use and redistribution provided that the original author and source are credited.</p></div></section></section><header id="header" class="content-header"><h1 class="title">Endogenous FGFs drive ERK-dependent cell fate patterning in 2D human gastruloids</h1><div class="authors"><ol class="authors-list authors-list--expanded" aria-label="Authors of this article"><li class="authors-list__item"><span class="authors-link">Kyoung Jo<sup>1,</sup><sup>*</sup></span></li><li class="authors-list__item"><span class="authors-link">Zong-Yuan Liu<sup>1,</sup><sup>*</sup></span></li><li class="authors-list__item"><span class="authors-link">Gauri Patel<sup>4</sup></span></li><li class="authors-list__item"><span class="authors-link">Zhiyuan Yu<sup>2</sup></span></li><li class="authors-list__item"><span class="authors-link">LiAng Yao<sup>1</sup></span></li><li class="authors-list__item"><span class="authors-link">Seth Teague<sup>4</sup></span></li><li class="authors-list__item"><span class="authors-link authors-email__link">Craig Johnson<sup>1</sup><a href="mailto:test@umich.edu"><span class="email-icon"></span></a></span></li><li class="authors-list__item"><span class="authors-link">Jason Spence<sup>1,</sup><sup>3,</sup><sup>4,</sup><sup>5</sup></span></li><li class="authors-list__item"><span class="authors-link authors-email__link">Idse Heemskerk<sup>1,</sup><sup>2,</sup><sup>3,</sup><sup>4,</sup><sup>6</sup><a href="mailto:iheemske@umich.edu"><span class="email-icon"></span></a></span></li></ol></div><div class="institutions"><ol class="institutions-list" aria-label="Author institutions"><li class="institutions-list__item"><sup>1</sup>Department of Cell and Developmental Biology, University of Michigan Medical School, Ann Arbor, United States</li><li class="institutions-list__item"><sup>2</sup>Department of Computational Medicine and Bioinformatics, University of Michigan Medical School, Ann Arbor, United States</li><li class="institutions-list__item"><sup>3</sup>Center for Cell Plasticity and Organ Design, University of Michigan Medical School, Ann Arbor, United States</li><li class="institutions-list__item"><sup>4</sup>Department of Biomedical Engineering, University of Michigan, Ann Arbor, United States</li><li class="institutions-list__item"><sup>5</sup>Department of Internal Medicine, Gastroenterology, University of Michigan Medical School, Ann Arbor, United States</li><li class="institutions-list__item"><sup>6</sup>Department of Physics, University of Michigan, Ann Arbor, United States</li></ol></div></header><section id="article-content"><section id="sa0" class="assessment"><h2>eLife Assessment</h2><div class="assessment__body"><p>This work is an <strong class="highlighted-term" aria-label="Highlighted">important</strong> contribution to understanding the role of FGF signaling in the induction of primitive-like cells in a 2D system of human gastrulation. The authors provide <strong class="highlighted-term" aria-label="Highlighted">compelling</strong> evidence showing that endogenous FGF ligands, acting through FGF receptors localized basolaterally, are determinant in the acquisition of specific cell fates. These observations will be of broad relevance to the FGF field.</p></div><div class="descriptors"><ul class="descriptors__identifiers"><li class="descriptors__identifier"><a href="https://doi.org/10.7554/eLife.101224.2.sa4">https://doi.org/10.7554/eLife.101224.2.sa4</a></li></ul></div></section><section class="abstract" id="abstract-1"><h1>Abstract</h1><p>The role of FGF is the least understood of the morphogens driving mammalian gastrulation. Here we investigated the function of FGF in a stem cell model for human gastrulation known as a 2D gastruloid. We found a ring of FGF-dependent ERK activity that closely follows the emergence of primitive streak (PS)-like cells but expands further inward. We showed that this ERK activity pattern is required for PS-like differentiation and that loss of PS-like cells upon FGF receptor inhibition can be rescued by directly activating ERK. We further demonstrated that the ERK-ring depends on localized activation of basolaterally positioned FGF receptors (FGFR) by endogenous FGF gradients. We confirmed and extended previous studies in analyzing expression of FGF pathway components, showing FGFR1 is the main receptor, FGF2 is highly expressed across several cell types, and FGF4/17 are the main FGF ligands expressed in the PS-like cells, similar to the human and monkey embryo but different from the mouse. We found that knockdown of FGF4 greatly reduced PS-like differentiation while FGF17 knockdown primarily affected subsequent mesoderm differentiation. FGF8 expression was spatially displaced from PS-markers and FGF4 expression and peaked earlier, while knockdown led to an expansion in PS-like cells, suggesting FGF8 may counteract FGF4 to limit PS-like differentiation. Thus, we have identified a previously unknown role for FGF-dependent ERK signaling in 2D gastruloids and possibly the human embryo, driven by a mechanism where FGF4 and FGF17 signal through basally localized FGFR1 to induce PS-like cells and their derivatives, potentially restricted by FGF8.</p></section><section id="s1"><h1 class="heading-1">Introduction</h1><p>During gastrulation the three germ layers are established, and the body plan is laid out. In amniotes this involves the formation of the primitive streak on the posterior side of the embryo from which cells ingress beneath the epiblast and migrate to form the mesodermal and endodermal layers. An important role in gastrulation is played by evolutionarily conserved morphogens from the BMP, Wnt, Nodal, and FGF families<sup><a href="#c1">1</a></sup>. The function of the first three signaling pathways is comparatively well understood: in both mouse and human a transcriptional hierarchy between BMP4, Wnt3, and Nodal plays a central role in inducing dynamic signaling gradients along the anterior posterior axis that control cell fate patterning<sup><a href="#c1">1</a>–<a href="#c3">3</a></sup>. For BMP and Nodal, basal receptor localization in the epiblast is crucial to form signaling activity gradients<sup><a href="#c3">3</a>–<a href="#c7">7</a></sup>. Wnt and Nodal signaling are directly required for mesoderm and endoderm differentiation, while the balance between BMP and Nodal signaling is important for patterning the germ layers along the body axes<sup><a href="#c8">8</a></sup>.</p><p>In contrast, the role of FGF in mammalian and especially human gastrulation has been more elusive. Secreted FGFs signal through FGF receptor tyrosine kinases (FGFRs) in a heparan sulfate proteoglycan (HSPG)-dependent manner to activate several signaling pathways including MAPK/ERK, PI3K, and PLCγ<sup><a href="#c9">9</a></sup>. There is evidence for a conserved requirement for FGF/ERK signaling in mesoderm formation across vertebrate model organisms<sup><a href="#c10">10</a>–<a href="#c13">13</a></sup>. However, it is unknown if there are FGF- and phosphorylated ERK gradients across the mammalian primitive streak<sup><a href="#c14">14</a>,<a href="#c15">15</a></sup>, whether there are functionally relevant ERK signaling dynamics like in other contexts, or whether receptor localization is important for establishing signaling gradients<sup><a href="#c16">16</a>–<a href="#c19">19</a></sup>. It is also unclear to what extent the requirement in mesoderm induction is direct, or indirect by modulating other signals<sup><a href="#c14">14</a>,<a href="#c15">15</a>,<a href="#c20">20</a></sup>. In addition, several FGFs are co-expressed and the roles of specific FGFs are uncertain. Moreover, there may be significant interspecies differences in this respect. Notably, FGF8 is the only FGF which has been shown to be required for gastrulation in the mouse<sup><a href="#c21">21</a></sup>, and its requirement in gastrulation appears to be conserved across vertebrate model organisms<sup><a href="#c22">22</a>–<a href="#c24">24</a></sup>, yet FGF8 is not highly expressed in primate embryos<sup><a href="#c25">25</a>,<a href="#c26">26</a></sup>.</p><p>Mutants for FGF8, FGFR1, HSPG synthesis, and ERK2, as well as pharmacological inhibition of MEK and FGFR give rise to similar phenotypes in mouse gastrulation, suggesting that FGFR1 and FGF8 are the only FGF ligand-receptor pair required for gastrulation and that these act through the MAPK/ERK pathway<sup><a href="#c12">12</a>,<a href="#c20">20</a>,<a href="#c21">21</a>,<a href="#c27">27</a>–<a href="#c31">31</a></sup>. FGF4 expression during gastrulation is also lost in FGF8 mutants, leaving the possibility of a role for FGF4, but this remains unknown as FGF4 is also required in the blastocyst and mutants therefore do not reach gastrulation<sup><a href="#c32">32</a></sup>. In all mutants and for pharmacological inhibition of FGF/ERK signaling, expression of the primitive streak marker brachyury (TBXT, also BRA, or T in mouse) is lost or severely diminished in the posterior streak and mesoderm fails to migrate away, piling up in this region. In addition, TBX6 expression is lost completely, and no lateral or paraxial mesoderm are formed. In contrast, TBXT expression in the anterior streak and extraembryonic mesoderm is not dependent on FGF/ERK signaling; extraembryonic mesoderm development is normal, and the axial mesoderm is expanded<sup><a href="#c20">20</a>,<a href="#c21">21</a>,<a href="#c27">27</a>,<a href="#c28">28</a></sup>.</p><p>In vitro models offer an opportunity to disentangle the complexity of FGF signaling in human development. Here, we explore the role of FGF/ERK signaling in micropatterned human pluripotent stem cells treated with BMP4, also known as 2D human gastruloids. These give rise to concentric rings of different cell types associated with gastrulation after 2 days of differentiation. Pluripotent cells in the center are surrounded by a primitive streak-like ring, which itself is surrounded by primordial germ cells and amnion-like cells on the colony edge<sup><a href="#c33">33</a>–<a href="#c35">35</a></sup>. Due to its simplicity and reproducibility, this system has proven powerful in deciphering the mechanisms underlying aspects of human gastrulation. It was used to show that BMP-Wnt-Nodal hierarchy that controls primitive streak formation during mouse gastrulation is preserved in human while providing a range of new insights into how these signals function<sup><a href="#c2">2</a>,<a href="#c3">3</a></sup>. For example, the crucial role for BMP and Nodal receptor localization in patterning was discovered in 2D gastruloids and later confirmed in the mouse embryo. This system also revealed that BMP, Wnt, and Nodal signaling patterns are highly dynamic<sup><a href="#c3">3</a>–<a href="#c7">7</a></sup>. It therefore also provides a promising approach to determine how FGF may function in human gastrulation.</p><p>In the 2D human gastruloid model, we show that a ring of phosphorylated ERK forms in an FGF-dependent manner and expands largely in lockstep with the formation of primitive streak-like cells but extends further inward. We demonstrate that this pattern of ERK activity is due to localized FGF receptor (FGFR) activation that requires basal FGFR localization. We then show that exogenous FGF2 in the media is required for pattern formation to occur but can be removed well before the phosphorylated (active) ERK (pERK) and PS-like rings appear, suggesting the pERK ring instead depends on endogenous FGF gradients. Using single cell RNA-sequencing, we determine the expression of FGF pathway components including FGFR1 as the main receptor and FGF2, FGF4, and FGF17 as the most highly expressed ligands. Our analysis corroborates previous studies describing expression of FGF2 and FGF17 in human embryos and gastruloids but for the first time recognizes a possible role for FGF4 and includes a broad analysis of genes involved in FGF signaling modulation<sup><a href="#c25">25</a>,<a href="#c36">36</a></sup>. We find that the FGF4 and FGF17 expression are restricted to the PS-like ring and that knockdown of FGF4 significantly reduces PS-like differentiation, while FGF17 knockdown primarily affects downstream mesoderm differentiation. Given its importance in the mouse, we also analyze FGF8 and find its expression is spatially and temporally displaced from FGF4 and PS-markers, while FGF8 knockdown leads to an increase in PS-like differentiation, suggesting it may counteract FGF4. In summary, we have greatly advanced our understanding of how FGFs function in a model for human gastrulation.</p></section><section id="s2"><h1 class="heading-1">Results</h1><section id="s2a"><h2 class="heading-2">FGF/ERK signaling is required for primitive streak-like differentiation</h2><p>We previously observed that pharmacological inhibition of MEK (the kinase upstream of ERK) or FGF receptors leads to a loss of differentiation to primitive streak-like and primordial germ cell-like cells, suggesting a crucial role for FGF/ERK signaling<sup><a href="#c34">34</a></sup>. Moreover, a ring of active ERK (pERK) has been observed in 2D gastruloids but its connection to FGF signaling or differentiation remains unclear<sup><a href="#c33">33</a></sup>. Therefore, we wanted to investigate the role of FGF/ERK signaling further.</p><p>To determine the dynamics of pERK and relate it to PS-like differentiation, we first performed time series immunofluorescence co-staining for pERK and the PS marker TBXT after BMP4 treatment in 700um diameter colonies. This revealed the pERK ring first emerges around 24h, around the same time as TBXT expression, and gradually gets brighter and wider (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 1a</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>). By 42h the outer edge of the ERK ring coincided with TBXT expression, while the inner edge was less sharp and extended further towards the center than TBXT, reminiscent of the domains of WNT and NODAL signaling that extend further in than the PS-like region<sup><a href="#c3">3</a>,<a href="#c6">6</a></sup> (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 1a</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>). Cell fate patterning in 2D gastruloids has a fixed length scale from the edge, with the central part absent in smaller colonies due to the fixed length scale of the BMP signaling gradient from the edge. Consistently, we found that ERK signaling also has a fixed pattern from the edge regardless of colony size, so that in smaller colonies the ERK ring becomes a domain of high ERK signaling extending through the colony center (<span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 1a</span><a class="fake-icon" href="#figS1"><span class="fakelink-icon">     </span></a></span>). MEK or FGFR inhibition for 30 minutes after 41.5h eliminated ERK activity, suggesting ERK activity depends entirely on FGF and that continuous FGF signaling is necessary to maintain the ERK signaling pattern (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 1bc</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>). Inhibition for the full duration of differentiation eliminated TBXT expression as well as other PS markers (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 1bc</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 1b</span><a class="fake-icon" href="#figS1"><span class="fakelink-icon">     </span></a></span>). However, we were able to rescue TBXT expression by directly and uniformly activating ERK (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 1d-f</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 1cd</span><a class="fake-icon" href="#figS1"><span class="fakelink-icon">     </span></a></span>) in cells stably expressing the doxycycline-inducible membrane-targeted catalytic domain of SOS (dox-SOS<sup>cat</sup>)<sup><a href="#c37">37</a></sup>. Despite uniform induced ERK activity, TBXT expression remained excluded from the center, showing that ERK activation alone is not sufficient for TBXT induction. However, TBXT levels were much higher in the rescue (<span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 1cd</span><a class="fake-icon" href="#figS1"><span class="fakelink-icon">     </span></a></span>), possibly reflecting the much higher level of ERK activity induced by dox-SOS<sup>cat</sup>. MEK and FGFR inhibition also led to a reduction in cell number which staining for BrdU and cleaved Cas3 suggested was primarily due to reduced proliferation (<span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 1ef</span><a class="fake-icon" href="#figS1"><span class="fakelink-icon">     </span></a></span>). In contrast to TBXT expression, reduced cell number caused by FGFR inhibition was only partially rescued by dox-SOS<sup>cat</sup>, indicating additional downstream pathways regulate cell number (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 1e</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>). Altogether, these data show the formation of a PS-like ring is dependent on FGF signaling through ERK.</p><figure class="figure tofill" id="fig1"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Figure 1.</label></span></h3><p><strong>a)</strong> Immunofluorescence stainings for pERK and TBXT and their radial intensity profiles in 2D gastruloids at different times. Error bars in graphs are standard deviations over N=4 colonies. <strong>b,c)</strong> Effect FGF receptor inhibition (FGFRi, b) or MEK inhibition (MEKi, c) on pERK and TBXT in colonies fixed at 42h after BMP treatment. Inhibition for either 30 minutes at 41.5h (rounded to 42 in figure) or throughout differentiation. <strong>d)</strong> BMP only control and FGFRi inhibition in cells expressing doxycycline-inducible SOS<sup>cat</sup> with and without addition of doxycycline. <strong>e-f)</strong> Cell numbers (e) and radial profiles of TBXT positive cells (f) for conditions in (d). Error bars represent standard deviation. Statistical significance was assessed using one-way ANOVA followed by Tukey’s HSD test for pairwise comparisons. <strong>g)</strong> PS marker TBXT, pERK and pluripotency marker SOX2 in Nodal knockout cells with inhibitors of Wnt secretion (WntSeci) and BMP receptors (BMPRi) with or without MEKi or FGFRi. All scale bars 50um.</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_fig1.tif/full/max/0/default.jpg" alt=""></figure><p>This raised several questions. First, is FGF/ERK signaling required directly for PS-like differentiation, or does it act indirectly? These possibilities are not mutually exclusive. For example, FGF/ERK could be required directly but also act indirectly by controlling Wnt or Nodal expression, as both Wnt and Nodal signaling are required for PS-like differentiation. Second, which are the FGFs activating the ERK pathway? The pathway could be activated by exogenous FGF2 present in the pluripotency medium mTeSR1 or by endogenously expressed ligands. Third, how is ERK activity restricted to a ring?</p><p>To determine if there is a direct requirement of FGF/ERK signaling and whether this depends on exogenous FGF, we performed differentiation to PS-like cells in standard culture by exogenous activation of Wnt and Nodal signaling, bypassing the need to form endogenous Wnt and Nodal gradient as in 2D gastruloids<sup><a href="#c38">38</a></sup>. We used a minimal base medium (E6) without exogenous FGF or other growth factors. To rule out any additional requirement for endogenous Nodal and Wnt which may be FGF/ERK-dependent, we used Nodal knockout cells<sup><a href="#c3">3</a></sup> treated with the Wnt secretion inhibitor IWP2, thereby eliminating any endogenous activity of these pathways. We also inhibited BMP receptors using LDN193189 to rule out any indirect effect through endogenous BMP. Under these conditions, with exogenous Wnt and Nodal activation for 24h in the absence of exogenous FGF, cells efficiently differentiated to TBXT+ PS-like cells (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 1g</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 1g</span><a class="fake-icon" href="#figS1"><span class="fakelink-icon">     </span></a></span>). These PS-like cells had high active ERK levels just as in the 2D gastruloids. Inhibition of either FGFR or MEK abrogated ERK activity and TBXT expression while FGFR inhibition also severely reduced proliferation. In both cases significant expression of the pluripotency and ectoderm marker SOX2 was maintained (<span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 1g</span><a class="fake-icon" href="#figS1"><span class="fakelink-icon">     </span></a></span>). Given that FGF/ERK is known to be involved in pluripotency maintenance, this suggests inhibition may not be complete and that much lower ERK activity is required for pluripotency maintenance than PS-like differentiation, consistent with lower ERK activity in the pluripotent center of gastruloids.</p><p>MEK and FGFR inhibition reduce proliferation and thereby cell density, raising the possibility that their lack of differentiation is an indirect effect that depends on cell density: a critical density may be required for differentiation due to a community effect. To rule this out we performed the experiment across a range of densities, which confirmed that PS differentiation in the control was efficient even at the lowest densities while FGFR and MEK inhibition effectively blocked differentiation when compared to controls with similar final cell densities (<span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 1h</span><a class="fake-icon" href="#figS1"><span class="fakelink-icon">     </span></a></span>). We also tested whether proliferation and differentiation require similar levels of ERK signaling by determining the MEKi dose response of gastruloids. This revealed that lower doses of MEKi effectively block differentiation without strongly affecting cell number, further supporting that the effects of ERK signaling on proliferation and differentiation are independent (<span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 1i</span><a class="fake-icon" href="#figS1"><span class="fakelink-icon">     </span></a></span>). Altogether, our results show that endogenous FGF expression is both necessary and sufficient for PS-like differentiation by exogenous Wnt and Nodal stimulation and demonstrate a direct requirement for ERK signaling in PS-like differentiation.</p></section><section id="s2b"><h2 class="heading-2">Single cell RNA-sequence reveals differentially expressed FGF pathway components</h2><p>The above results suggest endogenous FGF expression may contribute to elevated ERK signaling in the PS-like ring. However, FGF/ERK signaling may be modulated at many levels (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 2a</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span>). A spatial pattern of ERK activation could reflect a concentration gradient of FGF. Alternatively, differential expression of receptors and HSPG-modifying enzymes could lead to spatially patterned receptor activation and downstream ERK activity even in the absence of an FGF gradient. Even if receptor activation is uniform, downstream ERK activity could be spatially patterned by downstream negative regulators of ERK activity such as Sprouty-family proteins (SPRY) and dual-specificity phosphatases (DUSPs)<sup><a href="#c40">40</a></sup>.</p><figure class="figure tofill" id="fig2"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Figure 2.</label></span></h3><p><strong>a)</strong> Schematic of the FGF/ERK signaling pathway. <strong>b)</strong> Distribution of pERK levels in TBXT+ cells versus ISL1+ and other cells based on data and thresholds in <span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 2b</span><a class="fake-icon" href="#figS2"><span class="fakelink-icon">     </span></a></span>. <strong>c)</strong> Expression of key markers in PHATE<sup><a href="#c39">39</a></sup> projection of scRNA-seq data for 42h colony. <strong>d)</strong> Cell type annotation. PS-LC: primitive streak-like cells, NasMe: nascent mesoderm, AE-LC: amniotic ectoderm-like cells: PGC-LC: primordial germ cell-like cells, endo: definitive endoderm. <strong>e)</strong> Expression of canonical FGF ligands (log transformed and smoothened). <strong>f)</strong> Comparison of FGF ligand expression in 2D human gastruloids to human, monkey, and mouse embryo, normalized within each type of sample (see methods). <strong>g-i)</strong> Expression of several genes involved in FGF signaling with strong differential expression between PS-like cells or nascent mesoderm and pluripotent cells, split by gene category.</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_fig2.tif/full/max/0/default.jpg" alt=""></figure><p>To understand which genes modulating FGF/ERK signaling could be responsible for the elevated ERK signaling in TBXT positive cells and required for TBXT expression (<span class="fakelink linktofig"><span class="fakelink-text">Fig.1</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">Fig. 2b</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span>), we used single cell RNA-sequencing data from one previously published sample<sup><a href="#c34">34</a></sup> and one newly collected replicate (<span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 2a</span><a class="fake-icon" href="#figS2"><span class="fakelink-icon">     </span></a></span>) to determine expression of FGF pathway components in human 2D gastruloids, extending earlier work describing FGF expression<sup><a href="#c25">25</a>,<a href="#c35">35</a>,<a href="#c36">36</a>,<a href="#c41">41</a></sup>. TBXT-positive PS-like cells give rise to nascent mesoderm expressing TBX6 as well as definitive endoderm expressing SOX17 and FOXA2, both of which may still express TBXT at 42h<sup><a href="#c42">42</a></sup>. To understand how <em>TBXT</em> expression is initiated in an FGF-dependent manner, we focused on differential expression in the PS-LCs relative to pluripotent cells, reasoning that FGF modulation specific to nascent mesoderm or endoderm is a consequence rather than a cause of PS-LC differentiation. We identified different cell types based on marker genes with the PS-like cells defined as <em>TBXT+MIXL1+POU5F1+TBX6-SOX17-FOXA2-</em>(<span class="fakelink linktofig"><span class="fakelink-text">Fig. 2cd</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">Supp. Table 4</span><a class="fake-icon" href="#tblS4"><span class="fakelink-icon">     </span></a></span>). We then analyzed expression of different categories of FGF/ERK regulators.</p><p>We found four FGF ligands with significant expression: <em>FGF2, 4, 8,</em> and <em>17</em> (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 2e</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 2b</span><a class="fake-icon" href="#figS2"><span class="fakelink-icon">     </span></a></span>). We then compared the expression of these ligands in PS-like cells to reference data of CS7 human, CS8 monkey, and E6.5-E7.5 mouse gastrulation (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 2f</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 2cd</span><a class="fake-icon" href="#figS2"><span class="fakelink-icon">     </span></a></span>). We also included <em>FGF3</em> and <em>FGF5</em>, which are expressed during mouse gastrulation<sup><a href="#c32">32</a>,<a href="#c43">43</a>,<a href="#c44">44</a></sup>. High expression of <em>FGF4</em> and <em>FGF17</em> and low expression of FGF8 was consistent between human, monkey, and 2D gastruloid. <em>FGF2</em> expression was also high in the PS-like populations in 2D gastruloids and the CS8 monkey embryo, but much lower in the human embryo data. Furthermore, the monkey and human embryos expressed <em>FGF3</em>, which was minimally expressed in gastruloids. A similar comparison in nascent mesoderm showed much lower <em>FGF2/4</em> and much higher <em>FGF17</em> compared to PS across all primate samples (<span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 2e</span><a class="fake-icon" href="#figS2"><span class="fakelink-icon">     </span></a></span>). Expression in 2D gastruloids was consistent with published RNA-seq data for hPSC-derived PS-like cells, which expressed high <em>FGF2</em> and <em>FGF4</em>, some <em>FGF8</em> and <em>FGF17</em>, and no <em>FGF3</em> (<span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 2f</span><a class="fake-icon" href="#figS2"><span class="fakelink-icon">     </span></a></span>). Comparison with mouse confirmed significant differences from primates, with high expression of <em>FGF5, 8</em>, much lower expression of <em>FGF4,17</em>, and no expression of <em>FGF2</em> in the mouse<sup><a href="#c21">21</a>,<a href="#c43">43</a>,<a href="#c44">44</a></sup>.</p><p>Of the FGF receptors, only <em>FGFR1</em> and <em>FGFR4</em> were strongly expressed, with <em>FGFR1</em> expression about 5-fold higher (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 2g</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 2g</span><a class="fake-icon" href="#figS2"><span class="fakelink-icon">     </span></a></span>). Both receptors showed graded expression that decreased from pluripotent to PS-like cells and much lower expression in amnion-like cells (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 2g</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span>). Therefore, elevated ERK activity in PS-like cells cannot be accounted for by overall receptor expression. Differential expression of genes regulating heparin-sulfate proteoglycans which modulate FGF receptor binding was mild except for glypicans 3 and 4 which were downregulated in PS-like cells relative to pluripotent cells (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 2h</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 2g</span><a class="fake-icon" href="#figS2"><span class="fakelink-icon">     </span></a></span>). In contrast, several intracellular negative regulators of ERK signaling were strongly differentially expressed between cell clusters, including <em>SPRY1</em> and <em>PEBP1</em> which were significantly higher in pluripotent cells (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 2i</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 2g</span><a class="fake-icon" href="#figS2"><span class="fakelink-icon">     </span></a></span>). Surprisingly, however, expression of <em>SPRY4</em> was not elevated in PS-like cells where ERK signaling is high, suggesting that transcription of <em>SPRY4</em> may not be a good readout of ERK activity in human hPSCs like it is in the mouse<sup><a href="#c45">45</a></sup>. On the other hand, expression of the ERK negative feedback regulator <em>IL17RD</em> (<em>SEF</em>) does seem to approximately reflect expected ERK activity in each population. Components of the core ERK pathway including ERK itself were highly expressed across all clusters (<span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 2g</span><a class="fake-icon" href="#figS2"><span class="fakelink-icon">     </span></a></span>).</p><p>Thus, while scRNA-seq identified specific genes involved, it is consistent with several non-exclusive mechanisms for generating an ERK pattern. For example, uniform FGF2 could form an ERK ring by equally activating receptors across the colony but with ERK response reduced by elevated expression of negative regulators in the pluripotent center. Alternatively, based on gene expression we cannot rule out HSPGs increasing receptor binding specifically in the ring. Finally, localized expression of FGF4 or FGF17 in the PS-like ring could lead to a concentration gradient responsible for increased receptor activation.</p></section><section id="s2c"><h2 class="heading-2">The phosphorylated ERK pattern is due to differential activation of FGF receptors</h2><p>To distinguish between the different possibilities, we asked whether the spatial pattern of pERK signaling is controlled primarily below, at, or above the level of the receptors. To address this, we first visualized phosphorylated FGFR1 (pFGFR1) and found its spatial pattern strongly resembles that of pERK (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 3a</span><a class="fake-icon" href="#fig3"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 3a</span><a class="fake-icon" href="#figS3"><span class="fakelink-icon">     </span></a></span>). As expected, pFGFR1 was lost upon brief FGFR inhibition (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 3a</span><a class="fake-icon" href="#fig3"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 3a</span><a class="fake-icon" href="#figS3"><span class="fakelink-icon">     </span></a></span>). In contrast, after MEK inhibition, high pFGFR1 remained while pERK was greatly reduced (<span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 3bc</span><a class="fake-icon" href="#figS3"><span class="fakelink-icon">     </span></a></span>). This suggests the spatial pattern of ERK reflects a spatial pattern of receptor activity.</p><p>The spatial pattern in receptor activity could either be due to differential receptor expression or differential receptor activation. There was no significant upregulation of FGF receptors in PS-like cells according to our single cell RNA-sequencing data (<span class="fakelink linktofig"><span class="fakelink-text">Fig 2e</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span>). In fact, the most highly expressed receptor, FGFR1, was elevated in pluripotent cells. This suggests the ERK activity pattern is not due to a receptor expression pattern with two caveats. First, mRNA expression may not reflect protein expression. Second, multiple isoforms exist for FGF receptors 1-3 (but not 4) that have different affinity for FGF ligands, leaving open the possibility that PS-like cells could express a different FGFR1 isoform than pluripotent cells, despite overall FGFR1 expression being approximately uniform.</p><p>To determine if the spatial pattern in pFGFR1 could be explained by a spatial pattern of FGFR1 protein, we performed immunofluorescence staining of FGFR1. Consistent with the scRNA-seq data, and in sharp contrast to the pFGFR1 stain, FGFR1 protein expression was graded from center to edge with its highest levels in the pluripotent center and therefore could not explain the signaling pattern (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 3b</span><a class="fake-icon" href="#fig3"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 3d</span><a class="fake-icon" href="#figS3"><span class="fakelink-icon">     </span></a></span>). To determine if different isoforms of FGFR1 are differentially expressed in PS-like vs. pluripotent cells, we designed qPCR primers to detect the IIIb and IIIc isoforms of <em>FGFR1</em> (<span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 3e-g</span><a class="fake-icon" href="#figS3"><span class="fakelink-icon">     </span></a></span>). These are the most common isoforms, also known as the epithelial and mesenchymal isoforms, and therefore appeared the mostly likely candidates for differential expression. We then analyzed expression of these isoforms in 42h micropatterns versus pluripotent cells and PS- like cells differentiated directly as in <span class="fakelink linktofig"><span class="fakelink-text">Fig. 1d</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>. However, the change in expression between samples was similar for each isoform, with all isoforms decreasing in 2D gastruloids and directly differentiated PS-like cells relative to pluripotent cells (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 3c</span><a class="fake-icon" href="#fig3"><span class="fakelink-icon">     </span></a></span>). To corroborate this and also determine absolute FGFR isoform expression in pluripotent and PS-like cells, we analyzed our single cell RNA-sequencing data and previously published bulk RNA-sequencing data for isoform expression in these two cell types. This revealed that for both pluripotent and PS-like cells FGFR IIIc is the dominant isoform with on average almost 900-fold higher expression than IIIb, while IIIa was not expressed at all (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 3d</span><a class="fake-icon" href="#fig3"><span class="fakelink-icon">     </span></a></span>). Altogether, these data suggest that spatial patterning in ERK is due to differential activation of FGFR receptors, either due to FGF concentration gradients or possibly spatially patterned HSPGs. Immunostaining for HS showed no clear correlation with TBXT or the expected ERK signaling pattern which is similar but extends further inward (<span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 4</span><a class="fake-icon" href="#figS4"><span class="fakelink-icon">     </span></a></span>). This does not rule out a role for HSPGs in spatial modulation, since FGF signaling depends on specific sulfation patterns rather than overall HS levels<sup><a href="#c46">46</a>,<a href="#c47">47</a></sup>. Nevertheless, it led us to focus on FGF gradients as the most likely cause for the ERK signaling pattern.</p><figure class="figure tofill" id="fig3"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Figure 3.</label></span></h3><p><strong>a)</strong> phospho-FGFR1 stain with and without 1h FGFRi treatment at 41h, radial intensity profile on the right. <strong>b)</strong> FGFR1 with and without 1h FGFRi treatment at 41h, radial intensity profile on the right. Error bars in (b,c) represent standard deviation over N=4 colonies. <strong>c)</strong> qPCR data for relative expression of FGF receptor 1 isoforms IIIb, IIIc and total FGFR1 in pluripotent cells (pluri), gastruloids at 42h (MP), and directly differentiated PS-like cells (dPS). Error bars represent standard deviation of technical triplicates. <strong>d)</strong> Absolute FGFR1 isoform expression from RNA-seq for anterior/mid primitive streak and pluripotent cells from Loh 2016<sup><a href="#c38">38</a></sup> (Loh_ APS/MPS/HESC) as well as the pluripotent and PS-LC clusters from <span class="fakelink linktofig"><span class="fakelink-text">Fig.2</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span>. All scale bars 50 um.</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_fig3.tif/full/max/0/default.jpg" alt=""></figure></section><section id="s2d"><h2 class="heading-2">FGF receptors are localized basolaterally</h2><p>To maintain an endogenous FGF concentration gradient on the apical side of the colony which faces the medium, would require ligands to remain tethered to the surface. However, FGF has been reported to diffuse freely in the intercellular space<sup><a href="#c48">48</a></sup>. Moreover, there is FGF2 present in the medium that would be expected to uniformly activate FGF receptors. The situation is similar for BMP4, which is present uniformly in the medium for 2D gastruloids but does not cause uniform signaling activity<sup><a href="#c4">4</a></sup>. In the case of BMP, that is because receptors are localized basolaterally<sup><a href="#c4">4</a></sup> and tight junctions seal the basolateral side from the medium. Consequently, response to exogenous ligands in the medium is restricted to the colony edge, while endogenous ligands are confined to the basal side and prevented from diffusing away into the medium everywhere but the edge. This was found to be critical for patterning in both 2D gastruloids and the mouse embryo<sup><a href="#c4">4</a>,<a href="#c5">5</a></sup>. To explain the pERK pattern, we therefore hypothesized FGF receptors were similarly restricted to the basolateral side.</p><p>To test our hypothesis, we first starved pluripotent colonies of FGF by maintenance in FGF-free E6 medium for 12h and then treated with FGF2. Consistent with our hypothesis, phospho-FGFR and phosphor-ERK were both restricted to the colony edge (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 4a</span><a class="fake-icon" href="#fig4"><span class="fakelink-icon">     </span></a></span>). To determine if this could be explained by diffusion of exogenous FGF from the medium, we repeated the experiment with micropattern pluripotent cells and added fluorescent dextran with a similar molecular weight as FGF. We observed fluorescent dextran diffusing into the intercellular space from the edge inward over the same distance that ERK signaling was observed (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 4b-d</span><a class="fake-icon" href="#fig4"><span class="fakelink-icon">     </span></a></span>). We then acquired high resolution z-stacks of FGFR1 co-stained with the tight junction protein TJP1 (ZO-1) which revealed the majority of FGFR1 was below the tight junctions (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 4e</span><a class="fake-icon" href="#fig4"><span class="fakelink-icon">     </span></a></span>). To directly test that cells are more responsive to basal FGF than apical FGF, we then performed a transwell assay, where pluripotent cells were grown on filters to enable separate apical and basal stimulation. We found that only basal stimulation with FGF led to a significant increase in pERK (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 4f</span><a class="fake-icon" href="#fig4"><span class="fakelink-icon">     </span></a></span>). Finally, to determine if the center of the micropatterned colony is similarly able to respond to basal FGF stimulation and pERK levels there can be attributed to reduced FGF ligand concentration, we performed a scratch assay. Strong ERK response to FGF2 in the media was seen in the colony center around a scratch, but this response was blocked by FGFR inhibitor (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 4gh</span><a class="fake-icon" href="#fig4"><span class="fakelink-icon">     </span></a></span>). If HSPGs rather than FGF ligand were limiting ERK response in the center, we would not have expected strong response around a scratch. The initial ERK levels around the scratch greatly exceeded those anywhere in the control. However, we found this was due to adaptive ERK response to FGF. After six hours ERK levels around the scratch were similar to those in the ERK ring around the PS-like cells (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 4ij</span><a class="fake-icon" href="#fig4"><span class="fakelink-icon">     </span></a></span>). We therefore concluded that the pERK pattern is most likely due to a basolateral FGF ligand gradient which has its maximum in the PS-like ring.</p><figure class="figure tofill" id="fig4"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Figure 4.</label></span></h3><p><strong>a)</strong> pERK and pFGFR1 response of pluripotent colonies in standard culture grown in FGF-free (E6) medium for 24 hrs and then treated with FGF2 for 30 minutes. <strong>b,c)</strong> Fluorescent dextran (b) and pERK (c) in micropatterned pluripotent colony 24h after media change containing both FGF2 and dextran. <strong>d)</strong> Radial intensity profile of dextran and pERK for conditions in b,c), averaged over 2 colonies. <strong>e)</strong> Cross-section of 2D gastruloid shows FGF receptor 1 predominantly below the tight junction marked by ZO-1. White box marks are magnified in inset e’. <strong>f)</strong> pERK and pFGFR1 response in transwell experiment with cell treated with FGF either apically or basally. The illustrations in this panel were created using <span class="fakelink linktoext"><span class="fakelink-text">BioRender.com/v73i147</span><a class="fake-icon" href="https://BioRender.com/v73i147"><span class="fakelink-icon">     </span></a></span>. <strong>g)</strong> ERK signaling in scratched colonies with or without FGFRi. <strong>h)</strong> Quantification of pERK intensity as a function of distance from the scratch. Error bars represent standard deviation over colonies. <strong>i)</strong> ERK signaling stains at different times after scratching. <strong>j)</strong> Quantification of pERK intensity as a function of distance from the scratch at different times (left) and peak pERK level over time (right). Error bars represent standard deviation over N=4 colonies. Scale bars 50 micron.</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_fig4.tif/full/max/0/default.jpg" alt=""></figure></section><section id="s2e"><h2 class="heading-2">Endogenous FGF gradients underly the ERK activity pattern</h2><p>To identify which FGF ligands are required for PS-like differentiation, we first determined the requirement for exogenous FGF2 in the media. Differentiation of 2D gastruloids in FGF-free medium was severely reduced but restored by addition of FGF2 (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 5a</span><a class="fake-icon" href="#fig5"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 5a</span><a class="fake-icon" href="#figS5"><span class="fakelink-icon">     </span></a></span>). However, if FGF2 was removed 24h after BMP treatment, differentiation was normal. The fact that removal 2h after BMP treatment significantly reduced differentiation suggests this was not a failure to wash out the FGF2. Furthermore, MEK inhibition at 24h severely reduces TBXT expression<sup><a href="#c34">34</a></sup> while FGF2 is known to have a half-life on the order of hours<sup><a href="#c49">49</a></sup> and no media changes are performed during standard differentiation. Altogether this suggests that exogenous FGF2 is required at the time of BMP treatment but that endogenous FGFs are sufficient for the ERK activity ring and PS-like differentiation that start a day later.</p><figure class="figure tofill" id="fig5"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Figure 5.</label></span></h3><p><strong>a)</strong> Pattern at 42h with exogenous FGF2 in the culture medium for different amounts of time. <strong>b)</strong> Pseudotime analysis of FGFs and TBXT, TBX6 expression. <strong>c-e)</strong> FGF4 (c), FGF8 (d), and FGF17 (e) FISH co-stained for TBXT and pERK at 30h or 32h and 40h or 42h after BMP4 treatment. <strong>f)</strong> Radial intensity corresponding to (c-e). <strong>g-i)</strong> FGF4 (g), FGF8 (h), and FGF17 (i) FISH co-stained for TBXT and pERK in control siRNA versus FGF4 siRNA (g), FGF8 siRNA (h) and FGF17 siRNA (i) with corresponding radial intensity profiles. <strong>j)</strong> TBX6 and SOX17 expression in control shRNA versus FGF4 and FGF17 shRNA. Error bars in all panels except (b) represent standard deviation over N=4 colonies. Error bars in (b) represent the 95% confidence interval of the generalized additive model (see methods).</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_fig5.tif/full/max/0/default.jpg" alt=""></figure><p>We reasoned that FGFs whose expression correlates with the presence of PS-like cells are more likely to be responsible for the observed ERK ring and PS-like differentiation. Our analysis in <span class="fakelink linktofig"><span class="fakelink-text">figure 2</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span> had implicated FGF4 and FGF17 as most strongly expressed in the PS-like cells and nascent mesoderm, respectively (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 2c</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span>). However, it is unclear whether expression of these ligands is a cause or consequence of PS-like differentiation, or both, in the case of a feedback loop. To get more information about the relationship between FGF ligands and PS versus mesoderm differentiation, we performed pseudotime analysis of our scRNA-seq data (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 5b</span><a class="fake-icon" href="#fig5"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig 5b</span><a class="fake-icon" href="#figS5"><span class="fakelink-icon">     </span></a></span>). Along a pseudotime trajectory from pluripotent cells to nascent mesoderm, <em>FGF2</em> expression was nearly constant. <em>FGF8</em> was upregulated at lower levels prior to <em>TBXT</em> and declined as <em>TBXT</em> was upregulated. <em>FGF4</em> upregulation approximately coincided with <em>TBXT</em>, but <em>FGF4</em> began decreasing as <em>TBX6</em> expression increased. In contrast, <em>FGF17</em> expression started after <em>FGF4</em> but continued to increase after <em>FGF4</em> started decreasing. Direct comparison of <em>FGF</em> and <em>TBXT</em> expression also showed strong positive correlation between <em>TBXT</em> and both <em>FGF4</em> and <em>FGF17</em> but opposing relations with <em>TBX6</em> (<span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 5c</span><a class="fake-icon" href="#figS5"><span class="fakelink-icon">     </span></a></span>). Together this suggested to us initial TBXT expression may depend more on FGF8 or FGF4, while FGF17 could play a later role in maintaining or expanding the PS-like population or facilitating subsequent mesoderm differentiation.</p><p>Pseudotime may not accurately predict true dynamics. Therefore, we verified expression at two different times using RNA FISH (fluorescent in situ hybridization). We found that <em>FGF4</em> expression was similar at 32h and 40h (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 5cf</span><a class="fake-icon" href="#fig5"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 5d</span><a class="fake-icon" href="#figS5"><span class="fakelink-icon">     </span></a></span>). This is consistent with its decrease in pseudotime as cell differentiate to nascent mesoderm, since in real time differentiation is asynchronous and new PS-like cells form while older ones differentiate to nascent mesoderm, which would yield a flatter temporal profile. In contrast, peak <em>FGF8</em> levels decreased while <em>FGF17</em> strongly increased over time, both matching their pseudotime trends (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 5d-f</span><a class="fake-icon" href="#fig5"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 5ef</span><a class="fake-icon" href="#figS5"><span class="fakelink-icon">     </span></a></span>). As expected, the spatial profiles of both <em>FGF4</em> and <em>FGF17</em> resembled TBXT, although the peaks were displaced slightly inward, while <em>FGF8</em> expression was highest in the inner part of the ERK ring where TBXT expression was low (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 5c-f</span><a class="fake-icon" href="#fig5"><span class="fakelink-icon">     </span></a></span>). Co-staining showed <em>FGF4</em> and <em>FGF8</em> expressed in concentric rings with little overlap at 42h (<span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 5g</span><a class="fake-icon" href="#figS5"><span class="fakelink-icon">     </span></a></span>). Scatterplots further confirmed that high expression of both <em>FGF4</em> and <em>FGF17</em> but not <em>FGF8</em> was restricted to TBXT positive cells (<span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 5h</span><a class="fake-icon" href="#figS5"><span class="fakelink-icon">     </span></a></span>).</p><p>To test the functions of these FGFs we decreased their expression. Knockdown (KD) of <em>FGF4</em> with small interfering RNAs (siRNA) led to a strong decrease in TBXT and pERK (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 5g</span><a class="fake-icon" href="#fig5"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 6a</span><a class="fake-icon" href="#figS6"><span class="fakelink-icon">     </span></a></span>). We repeated this in a different cell line and also created stable cell lines expressing short hairpin RNA (shRNA) to knock down <em>FGF4</em> in both cell lines with similar effect (<span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 6b-d</span><a class="fake-icon" href="#figS6"><span class="fakelink-icon">     </span></a></span>). Strikingly, <em>FGF8</em> KD had the opposite effect and led to an increase in TBXT expression without a strong effect on pERK visible at 42h (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 5h</span><a class="fake-icon" href="#fig5"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 6e</span><a class="fake-icon" href="#figS6"><span class="fakelink-icon">     </span></a></span>). Finally, <em>FGF17</em> KD caused only a small decrease in TBXT or pERK, again consistent between different cell lines (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 5i</span><a class="fake-icon" href="#fig5"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 6f-i</span><a class="fake-icon" href="#figS6"><span class="fakelink-icon">     </span></a></span>). We further interrogated the endoderm and PGC marker SOX17 since both cell types require ERK signaling to differentiate and transiently express TBXT, as well as the mesoderm marker TBX6. TBX6 expression at 42h was near completely lost upon either <em>FGF4</em> KD or <em>FGF17</em> KD (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 5j</span><a class="fake-icon" href="#fig5"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 6j</span><a class="fake-icon" href="#figS6"><span class="fakelink-icon">     </span></a></span>). This suggests that while FGF17 is not required for PS induction (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 5i</span><a class="fake-icon" href="#fig5"><span class="fakelink-icon">     </span></a></span>), it is needed for subsequent mesoderm differentiation. In contrast, there was only a small reduction of SOX17 (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 5j</span><a class="fake-icon" href="#fig5"><span class="fakelink-icon">     </span></a></span>). Since we previously found with our protocol most SOX17 positive cells at 42h represent PGC-LCs, while most endoderm differentiates later<sup><a href="#c34">34</a></sup>, this suggests FGF4/17 are not required for PGC-LC induction but does not rule out a requirement for these FGFs in endoderm differentiation. A double KD of <em>FGF4</em> and <em>FGF17</em> had a similar phenotype as <em>FGF4</em> KD alone, suggesting FGF17 may be downstream of FGF4 (<span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 6k</span><a class="fake-icon" href="#figS6"><span class="fakelink-icon">     </span></a></span>). Combined, these data support distinct functions for FGF4, FGF8, and FGF17 in our human gastrulation model, where FGF4 and FGF8 may play opposing roles in PS induction while FGF17 is important during continued differentiation to mesoderm.</p></section></section><section id="s3"><h1 class="heading-1">Discussion</h1><p>We have shown that in a stem cell model for early human gastrulation, FGF-dependent ERK signaling is directly required for differentiation to PS-like cells and their derivatives, and that loss of differentiation upon FGF inhibition can be rescued by directly activating ERK. We also showed that FGF receptors are polarized basolaterally, preventing the colony center from responding to exogenous FGF in the medium. Furthermore, FGF receptor activation is increased in the PS-like ring, matching the pattern of active ERK despite overall receptor expression being lower in the PS-like ring than in the pluripotent center. This suggests elevated ERK activity is due to increased binding of FGFs to the receptors in this region, possibly due to increased FGF concentration. We found that both FGF4 and FGF17 are highly expressed in PS-like cells, while FGF8 is expressed earlier and displaced inward (<span class="fakelink linktofig"><span class="fakelink-text">Fig.6</span><a class="fake-icon" href="#fig6"><span class="fakelink-icon">     </span></a></span>). Knockdown of FGF4 leads to a strong reduction in ERK activity and PS-like cells, suggesting it may be responsible for increased ERK signaling in the PS-like region. In contrast, FGF8 knockdown led to an expansion of ERK activity and PS-like differentiation. Finally, FGF17 knockdown leds to a loss of mesoderm differentiation.</p><figure class="figure tofill" id="fig6"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Figure 6.</label></span>Graphical summary.</h3><p>Exogenous FGF2 acts on the colony edge and is prevented from reaching the basal FGFR1 receptors elsewhere by tight junctions. Primitive streak cells express FGF4 and FGF17 and these FGFs are associated with high ERK activity. FGF8 is expressed further inside the colony.</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_fig6.tif/full/max/0/default.jpg" alt=""></figure><p>Many questions remain. Although we showed that FGF gradients are likely the dominant cause for the spatial pattern of ERK signaling, our scRNA-seq data suggest other levels of regulation may contribute, which remain to be explored. Furthermore, although we showed FGF/ERK signaling is directly required for PS-like differentiation, it may also act indirectly by affecting the expression of Wnt and Nodal. We will explore this and the more general question of how FGF integrates with the hierarchy of BMP, Wnt, and Nodal in a separate manuscript. Related is the indirect effect FGF may have on patterning by its control of cell growth and thereby cell density. Cell density affects the response to exogenous BMP and downstream expression of Wnt, Nodal, and various inhibitors<sup><a href="#c4">4</a>,<a href="#c50">50</a>,<a href="#c51">51</a></sup>, as well as FGF itself, possibly causing a “community effect”<sup><a href="#c52">52</a>–<a href="#c54">54</a></sup>. It likely has other effects as well. However, the rescue of TBXT expression but not growth by ERK activation when FGF receptors are blocked suggests the FGF-dependent cell growth is not required to generate a primitive streak-like ring. The ability of exogenous Wnt and Nodal to induce PS-LCs in very sparse culture while MEK and FGFR inhibition block differentiation at the same density similarly argues against a requirement for FGF-dependent cell growth in PS-LC differentiation (<span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 1h</span><a class="fake-icon" href="#figS1"><span class="fakelink-icon">     </span></a></span>).</p><p>Another important question is to what extent different FGFs act redundantly or perform different functions. Our data suggest a picture where different FGFs are required consecutively, starting with exogenous FGF2, then endogenous FGF4, to support initial differentiation of PS-like cells, and finally FGF17 to maintain and expand the PS-like ring and support mesoderm and possibly endoderm differentiation. Whether these FGFs could substitute for each other if they were expressed at similar times and levels is unclear. It also remains to be determined whether the nearly constant high expression of FGF2 we observed is required for patterning. Most puzzling is the role of FGF8, whose knockdown led to an increase in PS-like differentiation. Given that FGF8 expression is displaced inward, one possibility is that it induces inhibitors of other PS-inducing signals there in order to limit PS differentiation, e.g., the Nodal inhibitor Lefty, which is expressed in a similar domain<sup><a href="#c55">55</a></sup>. The rescue of PS-like differentiation by uniform activation of ERK to a supraphysiological level suggests that the role of the different FGFs is simply to sufficiently activate ERK signaling. However, future work will have to investigate the dox-SOS<sup>cat</sup> rescue phenotype in much greater detail to see how differentiation and other cell behaviors such as migration are affected.</p><p>It is interesting to ask if there is a simple explanation for the seemingly large differences in FGF expression between primates and other species. FGF8 is the only FGF shown to be required for mouse gastrulation and is upstream of FGF4 in both chick and mouse, but it is barely expressed in human and monkey primitive streak and appears to play a different role in our stem cell model. If FGFs are interchangeable in simply activating ERK for primitive streak induction, variation in FGF expression is not surprising. However, FGF4 and FGF8 are the main FGFs in mouse, chick and frog gastrulation<sup><a href="#c56">56</a>–<a href="#c58">58</a></sup>, suggesting a high degree of conservation. It is possible FGF8 has simply been substituted by FGF17, which is in the same subfamily<sup><a href="#c9">9</a></sup>, but this does not explain why the role of FGF8 in gastrulation otherwise appears strongly conserved, or why FGF8 is upstream of FGF4 in the mouse while FGF4 is expressed before FGF17 in 2D gastruloids. Similarly, it is possible that FGF2 in primates substitutes for FGF5 in the mouse as the FGF that is highly expressed in the epiblast. Although these are not in the same subfamily and generally have different receptor affinities, they both have highest affinity for FGFR1<sup><a href="#c9">9</a></sup>. That leaves FGF4 as the only FGF that may be conserved across vertebrate gastrulation, and it is tempting to speculate on why. In Xenopus, FGF4 and TBXT (Xbra) function in a positive feedback loop<sup><a href="#c22">22</a>,<a href="#c59">59</a></sup>. The strongly correlated initial transcriptional dynamics of TBXT and FGF4 in 2D gastruloids are consistent with a similar feedback loop. However, an FGF8 feedback loop has also been proposed and recent work suggests TBXT is not required for FGF4 or FGF17 expression<sup><a href="#c54">54</a>,<a href="#c60">60</a>,<a href="#c61">61</a></sup>. Future work will therefore have to investigate further if FGF4 plays a unique role.</p><p>A final related question is the origin of the discrepancies we and others<sup><a href="#c36">36</a></sup> found between the primate (i.e., non-mouse) samples in <span class="fakelink linktofig"><span class="fakelink-text">Fig. 2</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span> and <span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 2</span><a class="fake-icon" href="#figS2"><span class="fakelink-icon">     </span></a></span>: FGF3 appears only significantly expressed in the human embryo, but not in the monkey embryo and human stem cell model. This could also reflect genetic variation: if FGFs function redundantly, their relative expression levels could be relatively unconstrained between individuals. On the other hand, in the nascent mesoderm, FGF2 is only high in human gastruloids. This may reflect developmental stage: the in vivo data is developmentally more advanced and nascent mesoderm may lose FGF2 expression later. It is also possible the maintenance conditions of hPSCs lead to in vitro artefacts in FGF expression.</p><p>In conclusion, the complexity of FGF signaling, with its many ligands, modulators, and functions, has prevented a clear understanding of its role in mammalian gastrulation in general and in human gastrulation specifically. Here we showed that 2D human gastruloids provide a powerful approach to make headway and although many open questions remain, we have taken significant steps towards this goal.</p></section><section id="s7"><h1 class="heading-1">Methods</h1><section id="s7a"><h2 class="heading-2">Cell lines</h2><p>The cell lines used were the embryonic stem cell line ESI017 (XX), and the induced pluripotent stem cell line PGP1 (XY). The pluripotency of these cells was confirmed by immunostaining of pluripotency markers OCT3/4, SOX2, NANOG. All cells were routinely tested for mycoplasma contamination, and negative results were recorded. ESI017 cells stably expressing doxycycline inducible SOS<sup>cat</sup> were made using a published Piggybac plasmid gifted by Jared Toettcher<sup><a href="#c37">37</a></sup>.</p></section><section id="s7b"><h2 class="heading-2">Cell culture and differentiation</h2><p>Human pluripotent stem cells were cultured in the commercially defined pluripotency maintaining media mTeSR1 (STEMCELL Technologies #85850) on Cultrex (R&amp;D Systems)-coated tissue culture plates. For FGF2 starvation we used Essential 6 Medium (Thermofisher, A1516401), or mTeSR1 without bFGF, TGF-beta, LiCl, GABA and pipecolic acid (STEMCELL Technologies #05896). For routine cell maintenance, cell passaging was performed every 3 days. For whole-colony passaging, L7<sup><a href="#c62">62</a></sup> was used, while single-cell passaging was performed with Accutase or TrypLE (Gibco). Single-cell suspensions were used to seed for all the differentiation experiments to control initial cell number.</p><p>To directly differentiate to primitive streak-like cells, cells were resuspended in a single-cell suspension and seeded in cultrex coated wells in mTeSR with ROCK inhibitor (RI) Y-27632 (MeChemExpress, cat# HY-10583). Cells were then maintained in mTeSR with RI for 24h before adding CHIR (3-5uM) and Activin (100ng/ml) treatment for 24-30h. Experiments were performed in 18-well Ibidi slides (cat# 81818).</p><p>To differentiate cells for 2D gastruloids by micropatterning, we followed the protocol in<sup><a href="#c42">42</a></sup>. Briefly, cells were resuspended in a single-cell suspension and seeded in laminin-coated micropatterned wells in mTeSR with ROCK inhibitor (RI) (MeChemExpress, cat# HY-10583). Micropatterned wells were washed with PBS-/- 30 minutes after the initial seeding to wash off the non-attached cells. Cells were then maintained for 24 hours in mTeSR before adding the BMP4 treatment by a full medium change. For time series micropatterned differentiation, full media changes with designated treatment(s) were performed every 24 hours. Experiments were performed in micropatterned 18-well Ibidi slides (cat# 81818) prepared as previously described<sup><a href="#c63">63</a></sup>. Cell signaling reagents and doses used are listed in <span class="fakelink linktofig"><span class="fakelink-text">Supplementary Table 1</span><a class="fake-icon" href="#tblS1"><span class="fakelink-icon">     </span></a></span>.</p></section><section id="s7c"><h2 class="heading-2">Immunofluorescence staining</h2><p>Antibodies can be found in Supplementary Tables 2 and 3. For stains including pERK, we used methanol fixation, for all other stains we used PFA fixation. PFA fixation: Samples from the 18-well Ibidi slides were rinsed with PBS, fixed for 20 min in 4% paraformaldehyde, rinsed twice with PBS, and blocked for 30 min at room temperature with 3% donkey serum and 0.1% Triton X-100 in 1× PBS. After blocking, cells were incubated with primary antibodies at 4°C overnight, followed by three washes in PBST (PBS with 0.1% Tween 20). They were then incubated with secondary antibodies and DAPI for 30 min at room temperature and washed twice in PBST at room temperature. Methanol Fixation: Samples from the 18-well Ibidi slides were rinsed with PBS, fixed for 20 mins in cold methanol, then rinsed with PBS for 5 min two times, after which 10% phosphate buffered formalin was added for 20 min at room temperature. Cells were then rinsed with TBS twice followed by 5 mins of incubation in cold methanol and rehydrate in TBS for 20 mins. Blocking was done for 30 mins at room temperature with 5% donkey serum, 1% BSA, 0.2% Triton-X-100 in 1x TBS after which staining proceeded the same way as with PFA fixation.</p></section><section id="s7d"><h2 class="heading-2">Microscopy</h2><p>Fixed sample imaging was performed with an Andor Dragonfly/Leica DMI8 spinning disk confocal microscope with a 40x, 1.1NA water and 20x 0.8NA air objectives using Andor Fusion software version 2.3.0.31 and a Nikon/Yokogawa spinning dish confocal microscope with a 40x silicon oil objective using NIS Elements AR software version 5.41.02. Live-cell imaging was performed with an Andor Dragonfly/Leica DMI8 spinning dish confocal microscope under the controlled temperature (37°C), CO<sub>2</sub> concentration (5%), and humidity (&gt;60%).</p></section><section id="s7e"><h2 class="heading-2">General image analysis</h2><p>We segmented nuclei in individual z-slices based on nuclear fluorescence stained with DAPI in fixed cells using a pipeline we previously described, which integrates two machine learning approaches: Ilastik pixel classification<sup><a href="#c64">64</a></sup> and Cellpose<sup><a href="#c65">65</a></sup> (v1). Fluorescence intensities were then calculated per nucleus as mean intensities in the 3D nuclear again with our established custom image-processing pipeline. To calculate the radial fluorescence distributions in micropatterned colonies, colonies were subdivided into radial bins with equal numbers of cells, mean or median and variance were then calculated within these bins. To calculate positive fractions, markers were first thresholded based on their intensity distribution, after which the fraction of cells positive for each marker was calculated within different bins. Standard deviations were then calculated over multiple replicate colonies. For visualization purposes only, background subtraction using grayscale opening on a 100 micron scale and 5 pixel wide median filter was applied slice by slice to z-stacks of micropatterned colonies before making maximal intensity projections.</p></section><section id="s7f"><h2 class="heading-2">(Single-cell) RNA sequencing and analysis</h2><p>Cells were collected using accutase and resuspended in ice-cold PBS. Single-cell RNA-sequencing was performed by the University of Michigan Advanced Genomics Core. Cells were barcoded using the 10X Genomics Chromium system (part numbers 1000268, 1000120, 1000215). For quality control, cDNA was quantified by Qubit High Sensitivity DNA assay and Agilent TapeStation. Sequencing was performed on the Illumina NovaSeq 6000 with NovaSeq S4 flowcell and Control Software version 1.7.0. Reads were aligned using cellranger-4.0.0 with the GRCh38 reference. Further processing was mostly done in Python, primarily using the Scanpy<sup><a href="#c66">66</a></sup> and SCVI<sup><a href="#c67">67</a></sup> packages. Analysis was performed on log transformed data. Library size normalization was performed on the intersection of the genes present in our data and the reference CS7 human embryo data to facilitate comparison of expression levels. We integrated the two scRNA-seq datasets for 42h 2D human gastruloids using SCVI based on the top 2000 highly variable genes, which were selected using the default function in Scanpy using the seurat_v3 flavor. Cell cycle genes were regressed out during integration. Dimensional reduction for visualization was performed using PHATE<sup><a href="#c39">39</a></sup> on the SCVI latent space. Gene expression was smoothened with MAGIC<sup><a href="#c68">68</a></sup> for visualization on PHATE plots in <span class="fakelink linktofig"><span class="fakelink-text">Fig. 2e,g-i</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span>. To annotate different cell types, we used MAGIC to impute that data and get smooth distributions of gene expression. We then thresholded marker genes to assign cell fate (Table 4), which produced better results than the more common unsupervised clustering using Leiden, and in particular more consistent results between datasets from different species. For interspecies comparison of FGF ligands we normalized the to the maximal FGF expression within each species, where for the mouse data, the three different time points were first integrated using SCVI. Thus we compare between species the relative expression of FGF ligands within each species, e.g., FGF8 expression is much higher than FGF4 at any time in the mouse, but much lower than FGF4 in human.</p><p>For pseudotime analysis (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 5</span><a class="fake-icon" href="#fig5"><span class="fakelink-icon">     </span></a></span>), we applied the trajectory inference tool Slingshot<sup><a href="#c69">69</a></sup> in R to compute the pseudotime trajectory in the SCVI latent representation. Slingshot determines the membership of a cell to a lineage at each branching point by its projection distance to the fitted principal curve of that lineage. We projected the mesoderm developmental trajectory onto the PHATE map for visualization. We further used TradeSeq<sup><a href="#c70">70</a></sup>, which fits a generalized additive model (GAM), to infer the pseudotime gene expression trends along the mesoderm lineage trajectory from Slingshot. For visualizing the correlation between the expression of FGF ligands and marker genes (<span class="fakelink linktofig"><span class="fakelink-text">Supp. Fig. 5</span><a class="fake-icon" href="#figS5"><span class="fakelink-icon">     </span></a></span>), we used our SCVI model to denoise the expression and mutual nearest neighbor to further harmonize the difference between samples.</p><p>To quantify isoforms from RNA sequencing salmon v1.10.0 was used<sup><a href="#c71">71</a></sup>. First, gencode.v45.transcripts.fa was downloaded from <span class="fakelink linktoext"><span class="fakelink-text">gencodegenes.org</span><a class="fake-icon" href="https://gencodegenes.org"><span class="fakelink-icon">     </span></a></span>. The reference was customized by removing lines for FGFR1 and replacing them with the transcript sequence for FGFRIIIa, FGFRIIIb, FGFRIIIc (sequences included as supplementary file). Three samples from GEO dataset GSM2257301 were downloaded through SRA (SRX1725562 (Pluripotent), SRX1725577 (APS), SRX1725643 (MPS))<sup><a href="#c38">38</a></sup>. Adapters and the first 10 bases were trimmed from the FASTQ files using trim_galore. Finally, the Salmon “quant” command with the --validateMappings option was used to quantify transcripts from trimmed FASTQ files. For single cell datasets, <span class="fakelink linktoext"><span class="fakelink-text">https://timoast.github.io/sinto/</span><a class="fake-icon" href="https://timoast.github.io/sinto/"><span class="fakelink-icon">     </span></a></span> was used to subset the 10x aligned file by cluster. Reads for clusters of interest were converted back to raw FASTQ files and run through the same alignment-based mode of salmon but as single end reads.</p></section><section id="s7g"><h2 class="heading-2">Quantitative real-time PCR</h2><p>RNA were extracted using RNAqueous™-Micro Total RNA Isolation Kit (CAT# AM1931, Thermo Fisher Scientific) and then cDNA synthesis was prepared from it using SuperScript™ VILO™ cDNA Synthesis Kit (CAT# 11754250, Thermo Fisher Scientific) following the manufacturer‘s protocol with adjustment for optimization. Measurements were performed with SYBR green and the primers in the Table 5. GAPDH was used for normalization in all experiments.</p></section><section id="s7h"><h2 class="heading-2">Fluorescent dextran assays</h2><p>Fluorescent dextran essays were performed by adding 10 ug/ml fluorescein isothiocyanate-dextran (10KD, Sigma Aldrich) to the media. To calculate fluorescence intensity as a function of distance from the colony edge, Otsu thresholding followed by several morphological operations was used to create a mask for the colony in each z-slice, thereby excluding bright dextran fluorescence outside the colony. Background subtraction on each z-slice of the image was performed to further remove background signal from dextran outside the colony while maintaining signal in the intercellular space. The radial intensity profile was then calculated from the maximal intensity projection of the masked, background subtracted image.</p></section><section id="s7i"><h2 class="heading-2">Scratch assays</h2><p>Human embryonic or induced pluripotent stem cells were differentiated on micropattern wells for 41h, then each micropattern colony was manually scratched with a serological needle. After 60mins, they were fixed with PFA or methanol followed by immunostaining with appropriate antibodies. To calculate the intensity profile as a function of distance from a scratch, we created a mask for the scratch based on the difference between an overall mask for the colony and the convex hull, calculated the distance transform of the scratch mask, then binned pixels by distance from the scratch and calculated their mean ERK intensity.</p></section><section id="s7j"><h2 class="heading-2">Transmembrane assays</h2><p>Human embryonic or induced pluripotent stem cells were seeded into Corning 6.5 mm transwell with 0.4 µm pore polyester membrane insert (cat# 3470) with E6 supplemented with TGF1b, FGF2, and Y-27632 for 16h, then media was changed to E6 with TGF1b. After 24h, the media was changed to E6 supplemented with FGF2 from the top or the bottom side of the transwell. 30 mins later, cells were fixed followed by immunofluorescence staining. The transwell cartoon in <span class="fakelink linktofig"><span class="fakelink-text">Fig 4</span><a class="fake-icon" href="#fig4"><span class="fakelink-icon">     </span></a></span> was generated with Biorender.</p></section><section id="s7k"><h2 class="heading-2">Fluorescence In Situ Hybridization</h2><p>The fluorescence in situ hybridization protocol (FISH) was performed according to the manufacturer’s instructions (ACDbio, RNAscope multiplex fluorescent manual). A list of probes and reagents can be found in <span class="fakelink linktofig"><span class="fakelink-text">Supplementary Table 6</span><a class="fake-icon" href="#tblS6"><span class="fakelink-icon">     </span></a></span>.</p></section><section id="s7l"><h2 class="heading-2">Gene targeting by shRNA</h2><p>We used the constructs shRNA F4: pPB[shRNA]-EGFP-U6&gt;hFGF4[shRNA#1] and shRNA F17: pPB[3shRNA]-U6&gt;hFGF17[shRNA#1]-U6&gt;hFGF17[shRNA#2]-U6&gt;{hFGF17_shRNA3}-hPGK&gt;Bsd both purchased from VectorBuilder in a PiggyBac Transposon Vectors with puromycin resistance which was stably integrated into ESI17 hESCs and PGP1 hiPSCs. Transfections were performed using Lipofectamine™ Stem Transfection Reagent (CAT# STEM00001, Thermo Fisher Scientific), following the manufacturer‘s protocol.</p></section><section id="s7m"><h2 class="heading-2">Gene targeting by siRNA</h2><p>We purchased predesigned siRNA from Millipore Sigma; NM_002007 (FGF4), NM_006119 (FGF8), NM_008004 (FGF17). The siRNA transfections were performed using Lipofectamine™ RNAiMAX Transfection Reagent (CAT# 13778075, Thermo Fisher Scientific), following the manufacturer‘s protocol with adjustment for optimization.</p></section><section id="s7n"><h2 class="heading-2">Statistics and reproducibility</h2><p>All experiments were performed twice or more. All the attempts in replicating experiments yielded consistent results. Unless specifically noted, all the quantifications were performed with N=4 colonies within the same experimental condition.</p></section><section id="s7o"><h2 class="heading-2">Ethics statement</h2><p>We comply with all the ethical regulations in our research. We were granted approval by the Human Pluripotent Stem Cell Research Oversight (HPSCRO) Committee at the University of Michigan to work with human embryonic stem cells and human induced pluripotent stem cells.</p><ul class="list-bullet"><li><p>Read orthogroup alignment</p></li><li><p>Pre-process alignment</p><ol class="list-alpha-lower"><li><p>Trim alignment edges and remove columns with high fraction of gap codons</p></li><li><p>For each species, infer consensus sequence and remove sequences outside a predetermined diameter around the consensus</p></li><li><p>Replace all SNPs with absolute frequencies <em>n</em> &lt; <em>n</em><sub><em>c</em></sub> = 4 with the consensus for their respective columns</p></li></ol></li><li><p>For each species <em>s</em>, subsample sequences from cells assigned to <em>s</em></p></li><li><p>Calculate the linkage matrix <img class="child-of-p insidetext inline-formula" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F543983v2_inline26c.gif/full/max/0/default.jpg" alt=""> for all pairs of SNPs</p></li><li><p>Identify groups of consecutive SNPs with <img class="child-of-p insidetext inline-formula" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F543983v2_inline26d.gif/full/max/0/default.jpg" alt="">, up to a rounding error of <em>ϵ</em> = 10<sup>−5</sup>, containing at least <em>l</em><sub><em>min</em></sub> = 5 SNPs</p></li></ul><pre>
for $x in y
dnskdnkfnfn
 nklfnk
    nklfnklf
      nklfnklfnklfnklfnkln-snk snklf msnk s njkf
 dnsklankldfa
ndjdb ekj
</pre></section></section><section id="s8"><h1 class="heading-1">Supplementary tables</h1><div class="fig-group"><p class="table movedfig" id="tblS1"><img class="child-of-figure imageonly" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_tblS1.tif/full/max/0/default.jpg" alt=""></p><p class="figure__caption"><span class="label figure-name"><label class="figure__label">Supplementary Table 1.</label></span><strong>Cell signaling reagents.</strong> <strong>a</strong>) Stains and quantification for pERK and TBXT in smaller colonies, diameter from left to right: 500um, 350um, 250um. <strong>b)</strong> Staining for three primitive streak markers with or with FGFRi or MEKi.</p></div><figure class="table tofill" id="tblS2"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Supplementary Table 2.</label></span>Primary antibodies used for immunofluorescence.</h3></figcaption><img class="child-of-figure imageonly" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_tblS2.tif/full/max/0/default.jpg" alt=""><img class="child-of-figure imageonly" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_tblS2a.tif/full/max/0/default.jpg" alt=""></figure><figure class="table tofill" id="tblS3"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Supplementary Table 3.</label></span>Secondary antibodies.</h3></figcaption><img class="child-of-figure imageonly" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_tblS3.tif/full/max/0/default.jpg" alt=""></figure><figure class="table tofill" id="tblS4"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Supplementary Table 4.</label></span>Marker genes for cell fate annotation</h3></figcaption><img class="child-of-figure imageonly" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_tblS4.tif/full/max/0/default.jpg" alt=""></figure><figure class="table tofill" id="tblS5"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Supplementary Table 5.</label></span>qPCR primers</h3></figcaption><img class="child-of-figure imageonly" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_tblS5.tif/full/max/0/default.jpg" alt=""></figure><figure class="table tofill" id="tblS6"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Supplementary Table 6.</label></span>Probes for Fluorescent In Situ Hybridization</h3></figcaption><img class="child-of-figure imageonly" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_tblS6.tif/full/max/0/default.jpg" alt=""></figure><div class="pagebreak"></div></section><section id="s9"><h1 class="heading-1">Supplementary Figures</h1><div class="fig-group"><p class="figure movedfig" id="figS1"><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_figS1.tif/full/max/0/default.jpg" alt=""></p><p class="figure__caption"><span class="label figure-name"><label class="figure__label">Supplementary Figure 1.</label></span><strong>a</strong>) Stains and quantification for pERK and TBXT in smaller colonies, diameter from left to right: 500um, 350um, 250um. <strong>b)</strong> Staining for three primitive streak markers with or with FGFRi or MEKi. <strong>c)</strong> DAPI stain corresponding to <span class="fakelink linktofig"><span class="fakelink-text">Fig. 1d</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>. <strong>d)</strong> Radial intensity profile for TBXT corresponding to <span class="fakelink linktofig"><span class="fakelink-text">Fig. 1d</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>. <strong>e</strong>) BrdU and cleaved Cas3 stainings after MEK or FGFR inhibition. <strong>f)</strong> Quantification of conditions in e. <strong>g)</strong> Quantification of conditions in <span class="fakelink linktofig"><span class="fakelink-text">Fig. 1g</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span> for five images each, N indicates total number of cells. <strong>h</strong>) Low density differentiation with and without MEK and FGFR inhibition. <strong>i)</strong> pERK and TBXT stains and quantification for different doses of MEK inhibitor shows low doses effectively block differentiation with minimal impact on cell number.</p></div><figure class="figure tofill" id="figS2"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Supplementary Figure 2.</label></span></h3><p><strong>a)</strong> PHATE projection of scRNA-seq data colored for sample. <strong>b,c)</strong> Expression of canonical FGF ligands across different cell types in 2D gastruloids (b) and human CS7 embryo (c). Expression across clusters is normalized by row, while total expression is shown in a separate column to the directly to left. The leftmost column indicates differential expression between PS-like and pluripotent cells. The color scale of (differential) expression is cut off for better contrast. <strong>d)</strong> Threshold-based annotation (left) for human, monkey, and mouse embryos used in interspecies comparison (<span class="fakelink linktofig"><span class="fakelink-text">Fig. 2f</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span>), compared to original annotation (right). <strong>e)</strong> Interspecies comparison as in <span class="fakelink linktofig"><span class="fakelink-text">Fig. 2f</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span> but for nascent mesoderm and PS-LC + nascent mesoderm. <strong>f)</strong> Bulk RNA-seq data by Loh et al<sup><a href="#c38">38</a></sup> for expression of different FGFs in directed differentiation to mid primitive streak (MPS) or anterior primitive streak (APS) relative the human embryonic stem cells (hESCs). <strong>g)</strong> Expression of genes involved in FGF signaling.</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_figS2.tif/full/max/0/default.jpg" alt=""></figure><figure class="figure tofill" id="figS3"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Supplementary Figure 3.</label></span></h3><p><strong>a)</strong> DAPI image corresponding to <span class="fakelink linktofig"><span class="fakelink-text">Fig. 3a</span><a class="fake-icon" href="#fig3"><span class="fakelink-icon">     </span></a></span>. <strong>b-c)</strong> pFGFR1 (b) or pERK (c) and TBXT stains with and without continuous MEKi treatment. <strong>d)</strong> DAPI image corresponding to <span class="fakelink linktofig"><span class="fakelink-text">Fig. 3b</span><a class="fake-icon" href="#fig3"><span class="fakelink-icon">     </span></a></span>. <strong>e)</strong> Design of qPCR primers to detect FGFR1 isoforms, diagram adapted from Eswarakumar et al<sup><a href="#c72">72</a></sup>. <strong>f)</strong> Primer sequence and properties. <strong>g)</strong> Amplicon sizes match prediction. Scale bars 50um.</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_figS3.tif/full/max/0/default.jpg" alt=""></figure><figure class="figure tofill" id="figS4"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Supplementary Figure 4.</label></span></h3><p><strong>a,b)</strong> Heparan sulfate (HS) stain (a) and radial intensity profile (b). Scale bar 50um. Error bands represent standard deviation over N=4 colonies.</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_figS4.tif/full/max/0/default.jpg" alt=""></figure><figure class="figure tofill" id="figS5"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Supplementary Figure 5.</label></span></h3><p><strong>a)</strong> DAPI image corresponding to <span class="fakelink linktofig"><span class="fakelink-text">Fig. 5a</span><a class="fake-icon" href="#fig5"><span class="fakelink-icon">     </span></a></span>. <strong>b)</strong> Pseudotime dynamics from <span class="fakelink linktofig"><span class="fakelink-text">Fig. 5b</span><a class="fake-icon" href="#fig5"><span class="fakelink-icon">     </span></a></span> with each gene normalized individually. <strong>c)</strong> Correlations between FGFs and TBXT, TBX6 colored for clusters, with cluster colors matching <span class="fakelink linktofig"><span class="fakelink-text">Fig. 2d</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span>. <strong>d-f)</strong> DAPI images and overlays including DAPI corresponding to <span class="fakelink linktofig"><span class="fakelink-text">Fig. 5c-e</span><a class="fake-icon" href="#fig5"><span class="fakelink-icon">     </span></a></span>, respectively. <strong>g)</strong> Combined FISH for FGF4 and FGF8. <strong>h)</strong> Scatterplots of FGF4, FGF8, and FGF17 mRNA versus TBXT colored for cell density.</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_figS5.tif/full/max/0/default.jpg" alt=""></figure><figure class="figure tofill" id="figS6"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Supplementary Figure 6.</label></span></h3><p><strong>a)</strong> Additional channels for the colony shown in <span class="fakelink linktofig"><span class="fakelink-text">Fig. 5g</span><a class="fake-icon" href="#fig5"><span class="fakelink-icon">     </span></a></span>. <strong>b)</strong> FGF4 shRNA in ESI17 cells. <strong>c-d)</strong> FGF4 siRNA (c) and shRNA (d) in PGP1 cells. <strong>e-f)</strong> additional channels for <span class="fakelink linktofig"><span class="fakelink-text">Fig. 5h-i</span><a class="fake-icon" href="#fig5"><span class="fakelink-icon">     </span></a></span>. <strong>g</strong>) FGF17 shRNA in ESI17 cells. <strong>h-i)</strong> FGF17 siRNA (h) and shRNA (i) in PGP1 cells. <strong>j)</strong> additional channels for the TBX6 ctrl and FGF17 shRNA in <span class="fakelink linktofig"><span class="fakelink-text">Fig. 5j</span><a class="fake-icon" href="#fig5"><span class="fakelink-icon">     </span></a></span>. <strong>k)</strong> FGF4 + FG17 shRNA double knockdown (stains from multiple colonies).</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://prod--epp.elifesciences.org/iiif/2/101224%2Fv2%2Fcontent%2F602611v2_figS6.tif/full/max/0/default.jpg" alt=""></figure><div class="pagebreak"></div></section><section id="acknowledgements"><h1 class="heading-1">Acknowledgements</h1><p>We thank Lila Solnica-Krezel, Blerta Stringa, Aryeh Warmflash, and Ben Allen for discussions. We thank Jared Toettcher for the dox-SOScat plasmid. Library prep and next-generation sequencing was carried out in the Advanced Genomics Core at the University of Michigan. This work was supported by the National Institute of General Medical Sciences (NIGMS R35GM138346), and the Branco Weiss Fellowship – Society in Science. KJ was partially supported by the Michigan Pioneer Postdoctoral Fellowship and the NIH F32 Ruth L. Kirschstein Postdoctoral National Research Service Award (5F32HD108980-02).</p></section><section id="additional-info"><h1 class="heading-1">Additional information</h1><section id="das"><h2 class="heading-2">Data availability</h2><p>Raw single cell RNA sequencing data generated in this study was deposited in GEO (GSE271604). Publicly available datasets used in this study can be accessed from GEO (GSE193007, GSE85066) and ArrayExpress (E-MTAB-6967, E-MTAB-9388). Raw image data are available upon request.</p></section><section id="s5"><h2 class="heading-2">Code availability</h2><p>All code for data analysis and model simulations is available on <span class="fakelink linktoext"><span class="fakelink-text">https://github.com/idse/FGF</span><a class="fake-icon" href="https://github.com/idse/FGF"><span class="fakelink-icon">     </span></a></span></p></section><section id="funding"><h2>Funding</h2><table id="funding-table"><thead><tr><th>Funder</th><th>Grant reference number</th></tr></thead><tbody><tr><td>National Institute of General Medical Sciences</td><td></td></tr><tr><td>the Branco Weiss Fellowship</td><td></td></tr><tr><td>Michigan Pioneer Postdoctoral Fellowship</td><td></td></tr><tr><td>NIH F32 Ruth L. Kirschstein Postdoctoral National Research Service Award</td><td></td></tr></tbody></table></section><section id="orcids"><h2 class="orcid-list__title">Author ORCID iDs</h2><ul class="orcid-list list-simple"><li class="orcid-list-item"><p><strong>Jason Spence</strong>: <a href="http://orcid.org/0000-0001-7869-3992">http://orcid.org/0000-0001-7869-3992</a></p></li></ul></section></section><section id="references"><h1 class="heading-1">References</h1><ul class="reference-list"><li class="reference-list__item" id="c1"><span class="reference__label">1.</span><ol class="reference__authors_list"><li class="reference__author">Arnold S. J.</li><li class="reference__author">Robertson E. J</li></ol><span class="reference__authors_list_suffix">2009</span><span class="reference__title">Making a commitment: cell lineage allocation and axis patterning in the early mouse embryo.</span><span class="reference__origin"><i>Nat. Rev. Mol. Cell Biol</i> <strong>10</strong>:91‑103</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c2"><span class="reference__label">2.</span><ol class="reference__authors_list"><li class="reference__author">Martyn I.</li><li class="reference__author">Kanno T. Y.</li><li class="reference__author">Ruzo A.</li><li class="reference__author">Siggia E. D.</li><li class="reference__author">Brivanlou A. H</li></ol><span class="reference__authors_list_suffix">2018</span><span class="reference__title">Self-organization of a human organizer by combined Wnt and Nodal signalling.</span><span class="reference__origin"><i>Nature</i> </span><span class="reference__doi"><a href="https://doi.org/10.1038/s41586-018-0150-y" class="reference__doi_link">https://doi.org/10.1038/s41586-018-0150-y</a></span></li><li class="reference-list__item" id="c3"><span class="reference__label">3.</span><ol class="reference__authors_list"><li class="reference__author">Chhabra S.</li><li class="reference__author">Liu L.</li><li class="reference__author">Goh R.</li><li class="reference__author">Kong X.</li><li class="reference__author">Warmflash A</li></ol><span class="reference__authors_list_suffix">2019</span><span class="reference__title">Dissecting the dynamics of signaling events in the BMP, WNT, and NODAL cascade during self-organized fate patterning in human gastruloids.</span><span class="reference__origin"><i>PLoS Biol</i> <strong>17</strong>:e3000498</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c4"><span class="reference__label">4.</span><ol class="reference__authors_list"><li class="reference__author">Etoc F.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2016</span><span class="reference__title">A Balance between Secreted Inhibitors and Edge Sensing Controls Gastruloid Self-Organization.</span><span class="reference__origin"><i>Dev. Cell</i> <strong>39</strong>:302‑315</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c5"><span class="reference__label">5.</span><ol class="reference__authors_list"><li class="reference__author">Zhang Z.</li><li class="reference__author">Zwick S.</li><li class="reference__author">Loew E.</li><li class="reference__author">Grimley J. S.</li><li class="reference__author">Ramanathan S</li></ol><span class="reference__authors_list_suffix">2018</span><span class="reference__title">Embryo geometry drives formation of robust signaling gradients through receptor localization.</span><span class="reference__origin"><i>bioRxiv</i> </span><span class="reference__doi"><a href="https://doi.org/10.1101/491290" class="reference__doi_link">https://doi.org/10.1101/491290</a></span></li><li class="reference-list__item" id="c6"><span class="reference__label">6.</span><ol class="reference__authors_list"><li class="reference__author">Heemskerk I.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2019</span><span class="reference__title">Rapid changes in morphogen concentration control self-organized patterning in human embryonic stem cells.</span><span class="reference__origin"><i>eLife</i> <strong>8</strong></span><span class="reference__doi"></span></li><li class="reference-list__item" id="c7"><span class="reference__label">7.</span><ol class="reference__authors_list"><li class="reference__author">Massey J.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2018</span><span class="reference__title">WNT ligands stimulate transient signaling in human pluripotent cells and synergize with TGF-β ligands to stimulate sustained signaling during differentiation.</span><span class="reference__origin"><i>bioRxiv</i> </span><span class="reference__doi"><a href="https://doi.org/10.1101/406306" class="reference__doi_link">https://doi.org/10.1101/406306</a></span></li><li class="reference-list__item" id="c8"><span class="reference__label">8.</span><ol class="reference__authors_list"><li class="reference__author">Yang R.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2021</span><span class="reference__title">Amnion signals are essential for mesoderm formation in primates.</span><span class="reference__origin"><i>Nat Commun</i> <strong>12</strong>:5126</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c9"><span class="reference__label">9.</span><ol class="reference__authors_list"><li class="reference__author">Ornitz D. M.</li><li class="reference__author">Itoh N</li></ol><span class="reference__authors_list_suffix">2015</span><span class="reference__title">The Fibroblast Growth Factor signaling pathway.</span><span class="reference__origin"><i>Wiley Interdiscip Rev Dev Biol</i> <strong>4</strong>:215‑266</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c10"><span class="reference__label">10.</span><ol class="reference__authors_list"><li class="reference__author">LaBonne C.</li><li class="reference__author">Whitman M</li></ol><span class="reference__authors_list_suffix">1994</span><span class="reference__title">Mesoderm induction by activin requires FGF-mediated intracellular signals.</span><span class="reference__origin"><i>Development</i> <strong>120</strong>:463‑472</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c11"><span class="reference__label">11.</span><ol class="reference__authors_list"><li class="reference__author">Krens S. F. G.</li><li class="reference__author">Corredor-Adámez M.</li><li class="reference__author">He S.</li><li class="reference__author">Snaar-Jagalska B. E.</li><li class="reference__author">Spaink H. P</li></ol><span class="reference__authors_list_suffix">2008</span><span class="reference__title">ERK1 and ERK2 MAPK are key regulators of distinct gene sets in zebrafish embryogenesis.</span><span class="reference__origin"><i>BMC Genomics</i> <strong>9</strong>:196</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c12"><span class="reference__label">12.</span><ol class="reference__authors_list"><li class="reference__author">Yao Y.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2003</span><span class="reference__title">Extracellular signal-regulated kinase 2 is necessary for mesoderm differentiation.</span><span class="reference__origin"><i>Proc Natl Acad Sci U S A</i> <strong>100</strong>:12759‑12764</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c13"><span class="reference__label">13.</span><ol class="reference__authors_list"><li class="reference__author">Mitrani E.</li><li class="reference__author">Gruenbaum Y.</li><li class="reference__author">Shohat H.</li><li class="reference__author">Ziv T</li></ol><span class="reference__authors_list_suffix">1990</span><span class="reference__title">Fibroblast growth factor during mesoderm induction in the early chick embryo.</span><span class="reference__origin"><i>Development</i> <strong>109</strong>:387‑393</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c14"><span class="reference__label">14.</span><ol class="reference__authors_list"><li class="reference__author">Corson L. B.</li><li class="reference__author">Yamanaka Y.</li><li class="reference__author">Lai K.-M. V.</li><li class="reference__author">Rossant J</li></ol><span class="reference__authors_list_suffix">2003</span><span class="reference__title">Spatial and temporal patterns of ERK signaling during mouse embryogenesis.</span><span class="reference__origin"><i>Development</i> <strong>130</strong>:4527‑4537</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c15"><span class="reference__label">15.</span><ol class="reference__authors_list"><li class="reference__author">Morgani S. M.</li><li class="reference__author">Hadjantonakis A.-K.</li></ol><span class="reference__authors_list_suffix">2020</span><span class="reference__title">Signaling regulation during gastrulation: Insights from mouse embryos and <em>in vitro</em> systems.</span><span class="reference__origin">In: <ol class="reference__editors_list"><li class="reference__editor">Small S.</li><li class="reference__editor">Briscoe J.</li></ol> (Eds). <i>Current Topics in Developmental Biology</i> <strong>137</strong> Academic Press. pp. 391‑431</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c16"><span class="reference__label">16.</span><ol class="reference__authors_list"><li class="reference__author">De Simone A.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2021</span><span class="reference__title">Control of osteoblast regeneration by a train of Erk activity waves.</span><span class="reference__origin"><i>Nature</i> <strong>590</strong>:129‑133</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c17"><span class="reference__label">17.</span><ol class="reference__authors_list"><li class="reference__author">Marshall C. J</li></ol><span class="reference__authors_list_suffix">1995</span><span class="reference__title">Specificity of receptor tyrosine kinase signaling: transient versus sustained extracellular signal-regulated kinase activation.</span><span class="reference__origin"><i>Cell</i> <strong>80</strong>:179‑185</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c18"><span class="reference__label">18.</span><ol class="reference__authors_list"><li class="reference__author">Ender P.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2022</span><span class="reference__title">Spatiotemporal control of ERK pulse frequency coordinates fate decisions during mammary acinar morphogenesis.</span><span class="reference__origin"><i>Dev Cell</i> <strong>57</strong>:2153‑2167.e6</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c19"><span class="reference__label">19.</span><ol class="reference__authors_list"><li class="reference__author">Aoki K.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2017</span><span class="reference__title">Propagating Wave of ERK Activation Orients Collective Cell Migration.</span><span class="reference__origin"><i>Dev Cell</i> <strong>43</strong>:305‑317.e5</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c20"><span class="reference__label">20.</span><ol class="reference__authors_list"><li class="reference__author">Ciruna B.</li><li class="reference__author">Rossant J</li></ol><span class="reference__authors_list_suffix">2001</span><span class="reference__title">FGF Signaling Regulates Mesoderm Cell Fate Specification and Morphogenetic Movement at the Primitive Streak.</span><span class="reference__origin"><i>Dev. Cell</i> <strong>1</strong>:37‑49</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c21"><span class="reference__label">21.</span><ol class="reference__authors_list"><li class="reference__author">Sun X.</li><li class="reference__author">Meyers E. N.</li><li class="reference__author">Lewandoski M.</li><li class="reference__author">Martin G. R</li></ol><span class="reference__authors_list_suffix">1999</span><span class="reference__title">Targeted disruption of Fgf8 causes failure of cell migration in the gastrulating mouse embryo.</span><span class="reference__origin"><i>Genes Dev</i> <strong>13</strong>:1834‑1846</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c22"><span class="reference__label">22.</span><ol class="reference__authors_list"><li class="reference__author">Fletcher R. B.</li><li class="reference__author">Harland R. M</li></ol><span class="reference__authors_list_suffix">2008</span><span class="reference__title">The role of FGF signaling in the establishment and maintenance of mesodermal gene expression in Xenopus.</span><span class="reference__origin"><i>Dev Dyn</i> <strong>237</strong>:1243‑1254</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c23"><span class="reference__label">23.</span><ol class="reference__authors_list"><li class="reference__author">Yang X.</li><li class="reference__author">Dormann D.</li><li class="reference__author">Münsterberg A. E.</li><li class="reference__author">Weijer C. J</li></ol><span class="reference__authors_list_suffix">2002</span><span class="reference__title">Cell Movement Patterns during Gastrulation in the Chick Are Controlled by Positive and Negative Chemotaxis Mediated by FGF4 and FGF8.</span><span class="reference__origin"><i>Developmental Cell</i> <strong>3</strong>:425‑437</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c24"><span class="reference__label">24.</span><ol class="reference__authors_list"><li class="reference__author">Fürthauer M.</li><li class="reference__author">Reifers F.</li><li class="reference__author">Brand M.</li><li class="reference__author">Thisse B.</li><li class="reference__author">Thisse C</li></ol><span class="reference__authors_list_suffix">2001</span><span class="reference__title">sprouty4 acts in vivo as a feedback-induced antagonist of FGF signaling in zebrafish.</span><span class="reference__origin"><i>Development</i> <strong>128</strong>:2175‑2186</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c25"><span class="reference__label">25.</span><ol class="reference__authors_list"><li class="reference__author">Tyser R. C. V.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2021</span><span class="reference__title">Single-cell transcriptomic characterization of a gastrulating human embryo.</span><span class="reference__origin"><i>Nature</i> <strong>600</strong>:285‑289</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c26"><span class="reference__label">26.</span><ol class="reference__authors_list"><li class="reference__author">Zhai J.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2022</span><span class="reference__title">Primate gastrulation and early organogenesis at single-cell resolution.</span><span class="reference__origin"><i>Nature</i> <strong>612</strong>:732‑738</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c27"><span class="reference__label">27.</span><ol class="reference__authors_list"><li class="reference__author">Deng C. X.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">1994</span><span class="reference__title">Murine FGFR-1 is required for early postimplantation growth and axial organization.</span><span class="reference__origin"><i>Genes Dev</i> <strong>8</strong>:3045‑3057</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c28"><span class="reference__label">28.</span><ol class="reference__authors_list"><li class="reference__author">Yamaguchi T. P.</li><li class="reference__author">Harpal K.</li><li class="reference__author">Henkemeyer M.</li><li class="reference__author">Rossant J.</li></ol><span class="reference__authors_list_suffix">1994</span><span class="reference__title">fgfr-1 is required for embryonic growth and mesodermal patterning during mouse gastrulation.</span><span class="reference__origin"><i>Genes Dev</i> <strong>8</strong>:3032‑3044</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c29"><span class="reference__label">29.</span><ol class="reference__authors_list"><li class="reference__author">Ciruna B. G.</li><li class="reference__author">Schwartz L.</li><li class="reference__author">Harpal K.</li><li class="reference__author">Yamaguchi T. P.</li><li class="reference__author">Rossant J</li></ol><span class="reference__authors_list_suffix">1997</span><span class="reference__title">Chimeric analysis of fibroblast growth factor receptor-1 (Fgfr1) function: a role for FGFR1 in morphogenetic movement through the primitive streak.</span><span class="reference__origin"><i>Development</i> <strong>124</strong>:2829‑2841</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c30"><span class="reference__label">30.</span><ol class="reference__authors_list"><li class="reference__author">García-García M. J.</li><li class="reference__author">Anderson K. V</li></ol><span class="reference__authors_list_suffix">2003</span><span class="reference__title">Essential role of glycosaminoglycans in Fgf signaling during mouse gastrulation.</span><span class="reference__origin"><i>Cell</i> <strong>114</strong>:727‑737</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c31"><span class="reference__label">31.</span><ol class="reference__authors_list"><li class="reference__author">Oki S.</li><li class="reference__author">Kitajima K.</li><li class="reference__author">Meno C</li></ol><span class="reference__authors_list_suffix">2010</span><span class="reference__title">Dissecting the role of Fgf signaling during gastrulation and left-right axis formation in mouse embryos using chemical inhibitors.</span><span class="reference__origin"><i>Dev Dyn</i> <strong>239</strong>:1768‑1778</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c32"><span class="reference__label">32.</span><ol class="reference__authors_list"><li class="reference__author">Feldman B.</li><li class="reference__author">Poueymirou W.</li><li class="reference__author">Papaioannou V. E.</li><li class="reference__author">DeChiara T. M.</li><li class="reference__author">Goldfarb M</li></ol><span class="reference__authors_list_suffix">1995</span><span class="reference__title">Requirement of FGF-4 for postimplantation mouse development.</span><span class="reference__origin"><i>Science</i> <strong>267</strong>:246‑249</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c33"><span class="reference__label">33.</span><ol class="reference__authors_list"><li class="reference__author">Warmflash A.</li><li class="reference__author">Sorre B.</li><li class="reference__author">Etoc F.</li><li class="reference__author">Siggia E. D.</li><li class="reference__author">Brivanlou A. H</li></ol><span class="reference__authors_list_suffix">2014</span><span class="reference__title">A method to recapitulate early embryonic spatial patterning in human embryonic stem cells.</span><span class="reference__origin"><i>Nat. Methods</i> <strong>11</strong>:847‑854</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c34"><span class="reference__label">34.</span><ol class="reference__authors_list"><li class="reference__author">Jo K.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2022</span><span class="reference__title">Efficient differentiation of human primordial germ cells through geometric control reveals a key role for Nodal signaling.</span><span class="reference__origin"><i>eLife</i> <strong>11</strong>:e72811</span><span class="reference__doi"><a href="https://doi.org/10.7554/eLife.72811" class="reference__doi_link">https://doi.org/10.7554/eLife.72811</a></span></li><li class="reference-list__item" id="c35"><span class="reference__label">35.</span><ol class="reference__authors_list"><li class="reference__author">Minn K. T.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2020</span><span class="reference__title">High-resolution transcriptional and morphogenetic profiling of cells from micropatterned human ESC gastruloid cultures.</span><span class="reference__origin"><i>eLife</i> <strong>9</strong></span><span class="reference__doi"></span></li><li class="reference-list__item" id="c36"><span class="reference__label">36.</span><ol class="reference__authors_list"><li class="reference__author">Minn K. T.</li><li class="reference__author">Dietmann S.</li><li class="reference__author">Waye S. E.</li><li class="reference__author">Morris S. A.</li><li class="reference__author">Solnica-Krezel L</li></ol><span class="reference__authors_list_suffix">2021</span><span class="reference__title">Gene expression dynamics underlying cell fate emergence in 2D micropatterned human embryonic stem cell gastruloids.</span><span class="reference__origin"><i>Stem Cell Reports</i> <strong>16</strong>:1210‑1227</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c37"><span class="reference__label">37.</span><ol class="reference__authors_list"><li class="reference__author">Underhill E. J.</li><li class="reference__author">Toettcher J. E</li></ol><span class="reference__authors_list_suffix">2023</span><span class="reference__title">Control of gastruloid patterning and morphogenesis by the Erk and Akt signaling pathways.</span><span class="reference__origin"><i>Development</i> <strong>150</strong>:dev201663</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c38"><span class="reference__label">38.</span><ol class="reference__authors_list"><li class="reference__author">Loh K. M.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2016</span><span class="reference__title">Mapping the Pairwise Choices Leading from Pluripotency to Human Bone, Heart, and Other Mesoderm Cell Types.</span><span class="reference__origin"><i>Cell</i> <strong>166</strong>:451‑467</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c39"><span class="reference__label">39.</span><ol class="reference__authors_list"><li class="reference__author">Moon K. R.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2019</span><span class="reference__title">Visualizing structure and transitions in high-dimensional biological data.</span><span class="reference__origin"><i>Nat. Biotechnol</i> <strong>37</strong>:1482‑1492</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c40"><span class="reference__label">40.</span><ol class="reference__authors_list"><li class="reference__author">Neben C. L.</li><li class="reference__author">Lo M.</li><li class="reference__author">Jura N.</li><li class="reference__author">Klein O. D</li></ol><span class="reference__authors_list_suffix">2019</span><span class="reference__title">Feedback regulation of RTK signaling in development.</span><span class="reference__origin"><i>Dev Biol</i> <strong>447</strong>:71‑89</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c41"><span class="reference__label">41.</span><ol class="reference__authors_list"><li class="reference__author">Tyser R. C. V.</li><li class="reference__author">Srinivas S</li></ol><span class="reference__authors_list_suffix">2022</span><span class="reference__title">Recent advances in understanding cell types during human gastrulation.</span><span class="reference__origin"><i>Seminars in Cell &amp; Developmental Biology</i> <strong>131</strong>:35‑43</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c42"><span class="reference__label">42.</span><ol class="reference__authors_list"><li class="reference__author">Chen B.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2024</span><span class="reference__title">Extended culture of 2D gastruloids to model human mesoderm development.</span><span class="reference__origin"><i>bioRxiv</i> </span><span class="reference__doi"><a href="https://doi.org/10.1101/2024.03.21.585753" class="reference__doi_link">https://doi.org/10.1101/2024.03.21.585753</a></span></li><li class="reference-list__item" id="c43"><span class="reference__label">43.</span><ol class="reference__authors_list"><li class="reference__author">Khoa L. T. P.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2016</span><span class="reference__title">Visualization of the Epiblast and Visceral Endodermal Cells Using Fgf5-P2A-Venus BAC Transgenic Mice and Epiblast Stem Cells.</span><span class="reference__origin"><i>PLOS One</i> <strong>11</strong>:e0159246</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c44"><span class="reference__label">44.</span><ol class="reference__authors_list"><li class="reference__author">Pijuan-Sala B.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2019</span><span class="reference__title">A single-cell molecular map of mouse gastrulation and early organogenesis.</span><span class="reference__origin"><i>Nature</i> <strong>566</strong>:490‑495</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c45"><span class="reference__label">45.</span><ol class="reference__authors_list"><li class="reference__author">Morgani S. M.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2018</span><span class="reference__title">A Sprouty4 reporter to monitor FGF/ERK signaling activity in ESCs and mice.</span><span class="reference__origin"><i>Developmental Biology</i> <strong>441</strong>:104‑126</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c46"><span class="reference__label">46.</span><ol class="reference__authors_list"><li class="reference__author">Xu D.</li><li class="reference__author">Esko J. D</li></ol><span class="reference__authors_list_suffix">2014</span><span class="reference__title">Demystifying heparan sulfate-protein interactions.</span><span class="reference__origin"><i>Annu Rev Biochem</i> <strong>83</strong>:129‑157</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c47"><span class="reference__label">47.</span><ol class="reference__authors_list"><li class="reference__author">Habuchi H.</li><li class="reference__author">Kimata K</li></ol><span class="reference__authors_list_suffix">2008</span><span class="reference__title">Functions of Heparan Sulfate Proteoglycans in Morphogenesis.</span><span class="reference__origin">In: <ol class="reference__editors_list"><li class="reference__editor">Taniguchi N.</li><li class="reference__editor">Suzuki A</li><li class="reference__editor">Ito Y</li><li class="reference__editor">Narimatsu H</li><li class="reference__editor">Kawasaki T</li><li class="reference__editor">Hase S</li></ol> (Eds). <i>Experimental Glycoscience</i> Tokyo: Springer. pp. 303‑308</span><span class="reference__doi"><a href="https://doi.org/10.1007/978-4-431-77922-3_72" class="reference__doi_link">https://doi.org/10.1007/978-4-431-77922-3_72</a></span></li><li class="reference-list__item" id="c48"><span class="reference__label">48.</span><ol class="reference__authors_list"><li class="reference__author">Yu S. R.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2009</span><span class="reference__title">Fgf8 morphogen gradient forms by a source-sink mechanism with freely diffusing molecules.</span><span class="reference__origin"><i>Nature</i> <strong>461</strong>:533‑536</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c49"><span class="reference__label">49.</span><ol class="reference__authors_list"><li class="reference__author">Benington L.</li><li class="reference__author">Rajan G.</li><li class="reference__author">Locher C.</li><li class="reference__author">Lim L. Y</li></ol><span class="reference__authors_list_suffix">2020</span><span class="reference__title">Fibroblast Growth Factor 2-A Review of Stabilisation Approaches for Clinical Applications.</span><span class="reference__origin"><i>Pharmaceutics</i> <strong>12</strong>:508</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c50"><span class="reference__label">50.</span><ol class="reference__authors_list"><li class="reference__author">Teague S.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2024</span><span class="reference__title">Time-integrated BMP signaling determines fate in a stem cell model for early human development.</span><span class="reference__origin"><i>Nat Commun</i> <strong>15</strong>:1471</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c51"><span class="reference__label">51.</span><ol class="reference__authors_list"><li class="reference__author">Ortiz-Salazar M. A.</li><li class="reference__author">Camacho-Aguilar E.</li><li class="reference__author">Warmflash A.</li></ol><span class="reference__authors_list_suffix">2024</span><span class="reference__title">Endogenous Nodal switches Wnt interpretation from posteriorization to germ layer differentiation in geometrically constrained human pluripotent cells.</span><span class="reference__origin"><i>bioRxiv</i> </span><span class="reference__doi"><a href="https://doi.org/10.1101/2024.03.13.584912" class="reference__doi_link">https://doi.org/10.1101/2024.03.13.584912</a></span></li><li class="reference-list__item" id="c52"><span class="reference__label">52.</span><ol class="reference__authors_list"><li class="reference__author">Gurdon J. B</li></ol><span class="reference__authors_list_suffix">1988</span><span class="reference__title">A community effect in animal development.</span><span class="reference__origin"><i>Nature</i> <strong>336</strong>:772‑774</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c53"><span class="reference__label">53.</span><ol class="reference__authors_list"><li class="reference__author">Nemashkalo A.</li><li class="reference__author">Ruzo A.</li><li class="reference__author">Heemskerk I.</li><li class="reference__author">Warmflash A</li></ol><span class="reference__authors_list_suffix">2017</span><span class="reference__title">Morphogen and community effects determine cell fates in response to BMP4 signaling in human embryonic stem cells.</span><span class="reference__origin"><i>Development</i> <strong>144</strong>:3042‑3053</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c54"><span class="reference__label">54.</span><ol class="reference__authors_list"><li class="reference__author">Gattiglio M.</li><li class="reference__author">Protzek M.</li><li class="reference__author">Schröter C</li></ol><span class="reference__authors_list_suffix">2023</span><span class="reference__title">Population-level antagonism between FGF and BMP signaling steers mesoderm differentiation in embryonic stem cells.</span><span class="reference__origin"><i>Biology Open</i> <strong>12</strong>:bio059941</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c55"><span class="reference__label">55.</span><ol class="reference__authors_list"><li class="reference__author">Liu L.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2022</span><span class="reference__title">Nodal is a short-range morphogen with activity that spreads through a relay mechanism in human gastruloids.</span><span class="reference__origin"><i>Nat Commun</i> <strong>13</strong>:497</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c56"><span class="reference__label">56.</span><ol class="reference__authors_list"><li class="reference__author">Isaacs H. V.</li><li class="reference__author">Pownall M. E.</li><li class="reference__author">Slack J. M</li></ol><span class="reference__authors_list_suffix">1995</span><span class="reference__title">eFGF is expressed in the dorsal midline of Xenopus laevis.</span><span class="reference__origin"><i>Int J Dev Biol</i> <strong>39</strong>:575‑579</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c57"><span class="reference__label">57.</span><ol class="reference__authors_list"><li class="reference__author">Christen B.</li><li class="reference__author">Slack J. M</li></ol><span class="reference__authors_list_suffix">1997</span><span class="reference__title">FGF-8 is associated with anteroposterior patterning and limb regeneration in Xenopus.</span><span class="reference__origin"><i>Dev Biol</i> <strong>192</strong>:455‑466</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c58"><span class="reference__label">58.</span><ol class="reference__authors_list"><li class="reference__author">Hardy K. M.</li><li class="reference__author">Yatskievych T. A.</li><li class="reference__author">Konieczka J.</li><li class="reference__author">Bobbs A. S.</li><li class="reference__author">Antin P. B</li></ol><span class="reference__authors_list_suffix">2011</span><span class="reference__title">FGF signalling through RAS/MAPK and PI3K pathways regulates cell movement and gene expression in the chicken primitive streak without affecting E-cadherin expression.</span><span class="reference__origin"><i>BMC Dev Biol</i> <strong>11</strong>:20‑17</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c59"><span class="reference__label">59.</span><ol class="reference__authors_list"><li class="reference__author">Isaacs H. V.</li><li class="reference__author">Pownall M. E.</li><li class="reference__author">Slack J. M</li></ol><span class="reference__authors_list_suffix">1994</span><span class="reference__title">eFGF regulates Xbra expression during Xenopus gastrulation.</span><span class="reference__origin"><i>The EMBO Journal</i> <strong>13</strong>:4469‑4481</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c60"><span class="reference__label">60.</span><ol class="reference__authors_list"><li class="reference__author">Bulger E. A.</li><li class="reference__author">Muncie-Vasic I.</li><li class="reference__author">Libby A. R. G.</li><li class="reference__author">McDevitt T. C.</li><li class="reference__author">Bruneau B. G</li></ol><span class="reference__authors_list_suffix">2024</span><span class="reference__title">TBXT dose sensitivity and the decoupling of nascent mesoderm specification from EMT progression in 2D human gastruloids.</span><span class="reference__origin"><i>Development</i> <strong>151</strong>:dev202516</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c61"><span class="reference__label">61.</span><ol class="reference__authors_list"><li class="reference__author">Evans A. L.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2012</span><span class="reference__title">Genomic Targets of Brachyury (T) in Differentiating Mouse Embryonic Stem Cells.</span><span class="reference__origin"><i>PLOS One</i> <strong>7</strong>:e33346</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c62"><span class="reference__label">62.</span><ol class="reference__authors_list"><li class="reference__author">Nie Y.</li><li class="reference__author">Walsh P.</li><li class="reference__author">Clarke D. L.</li><li class="reference__author">Rowley J. A.</li><li class="reference__author">Fellner T</li></ol><span class="reference__authors_list_suffix">2014</span><span class="reference__title">Scalable passaging of adherent human pluripotent stem cells.</span><span class="reference__origin"><i>PLoS ONE</i> <strong>9</strong>:e88012</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c63"><span class="reference__label">63.</span><ol class="reference__authors_list"><li class="reference__author">Freeburne E.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2023</span><span class="reference__title">Spatial Single Cell Analysis of Proteins in 2D Human Gastruloids Using Iterative Immunofluorescence.</span><span class="reference__origin"><i>Curr Protoc</i> <strong>3</strong>:e915</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c64"><span class="reference__label">64.</span><ol class="reference__authors_list"><li class="reference__author">Sommer C.</li><li class="reference__author">Straehle C.</li><li class="reference__author">Kothe U.</li><li class="reference__author">Hamprecht F. A.</li></ol><span class="reference__authors_list_suffix">2011</span><span class="reference__title">Ilastik: Interactive learning and segmentation toolkit.</span><span class="reference__origin">In: 2011 8th IEEE International Symposium on Biomedical Imaging (ISBI 2011). pp. 230‑233</span><span class="reference__doi"><a href="https://doi.org/10.1109/ISBI.2011.5872394" class="reference__doi_link">https://doi.org/10.1109/ISBI.2011.5872394</a></span></li><li class="reference-list__item" id="c65"><span class="reference__label">65.</span><ol class="reference__authors_list"><li class="reference__author">Stringer C.</li><li class="reference__author">Wang T.</li><li class="reference__author">Michaelos M.</li><li class="reference__author">Pachitariu M</li></ol><span class="reference__authors_list_suffix">2021</span><span class="reference__title">Cellpose: a generalist algorithm for cellular segmentation.</span><span class="reference__origin"><i>Nat. Methods</i> <strong>18</strong>:100‑106</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c66"><span class="reference__label">66.</span><ol class="reference__authors_list"><li class="reference__author">Wolf F. A.</li><li class="reference__author">Angerer P.</li><li class="reference__author">Theis F. J</li></ol><span class="reference__authors_list_suffix">2018</span><span class="reference__title">SCANPY: large-scale single-cell gene expression data analysis.</span><span class="reference__origin"><i>Genome Biol</i> <strong>19</strong>:15</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c67"><span class="reference__label">67.</span><ol class="reference__authors_list"><li class="reference__author">Gayoso A.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2022</span><span class="reference__title">A Python library for probabilistic analysis of single-cell omics data.</span><span class="reference__origin"><i>Nat Biotechnol</i> <strong>40</strong>:163‑166</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c68"><span class="reference__label">68.</span><ol class="reference__authors_list"><li class="reference__author">van Dijk D.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2018</span><span class="reference__title">Recovering Gene Interactions from Single-Cell Data Using Data Diffusion.</span><span class="reference__origin"><i>Cell</i> <strong>174</strong>:716‑729.e27</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c69"><span class="reference__label">69.</span><ol class="reference__authors_list"><li class="reference__author">Street K.</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2018</span><span class="reference__title">Slingshot: cell lineage and pseudotime inference for single-cell transcriptomics.</span><span class="reference__origin"><i>BMC Genomics</i> <strong>19</strong>:477</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c70"><span class="reference__label">70.</span><ol class="reference__authors_list"><li class="reference__author">Van den Berge K</li><li class="reference__author">et al.</li></ol><span class="reference__authors_list_suffix">2020</span><span class="reference__title">Trajectory-based differential expression analysis for single-cell sequencing data.</span><span class="reference__origin"><i>Nat Commun</i> <strong>11</strong>:1201</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c71"><span class="reference__label">71.</span><ol class="reference__authors_list"><li class="reference__author">Patro R.</li><li class="reference__author">Duggal G.</li><li class="reference__author">Love M. I.</li><li class="reference__author">Irizarry R. A.</li><li class="reference__author">Kingsford C</li></ol><span class="reference__authors_list_suffix">2017</span><span class="reference__title">Salmon provides fast and bias-aware quantification of transcript expression.</span><span class="reference__origin"><i>Nat Methods</i> <strong>14</strong>:417‑419</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c72"><span class="reference__label">72.</span><ol class="reference__authors_list"><li class="reference__author">Eswarakumar V. P.</li><li class="reference__author">Lax I.</li><li class="reference__author">Schlessinger J</li></ol><span class="reference__authors_list_suffix">2005</span><span class="reference__title">Cellular signaling by fibroblast growth factor receptors.</span><span class="reference__origin"><i>Cytokine Growth Factor Rev</i> <strong>16</strong>:139‑149</span><span class="reference__doi"></span></li><li class="reference-list__item" id="dataref1"><ol class="reference__authors_list"><li class="reference__author">Jo K</li><li class="reference__author">Liu ZY</li><li class="reference__author">Patel G</li><li class="reference__author">Yu Z</li><li class="reference__author">Yao L</li><li class="reference__author">Teague S</li><li class="reference__author">Johnson C</li><li class="reference__author">Spence J</li><li class="reference__author">Heemskerk I</li></ol><span class="reference__authors_list_suffix">2024</span><span class="reference__title">Endogenous FGF gradients are required for cell fate patterning in 2D human gastruloids.</span><span class="reference__origin">NCBI Gene Expression Omnibus. ID ç</span><span class="reference__doi"><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE271604" class="reference__external_link">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE271604</a></span></li><li class="reference-list__item" id="dataref2"><ol class="reference__authors_list"><li class="reference__author">Zhai J</li><li class="reference__author">Guo J</li><li class="reference__author">Wan H</li><li class="reference__author">Qi L</li><li class="reference__author">Liu L</li><li class="reference__author">Xiao Z</li><li class="reference__author">Yan L</li><li class="reference__author">Schmitz DA</li><li class="reference__author">Xu Y</li><li class="reference__author">Yu D</li></ol><em> et al.</em><span class="reference__authors_list_suffix">2022</span><span class="reference__title">Primate gastrulation and early organogenesis at single-cell resolution.</span><span class="reference__origin">NCBI Gene Expression Omnibus. ID GSE193007</span><span class="reference__doi"><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE193007" class="reference__external_link">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE193007</a></span></li><li class="reference-list__item" id="dataref3"><ol class="reference__authors_list"><li class="reference__author">Pijuan-Sala B</li><li class="reference__author">Griffiths JA</li><li class="reference__author">Guibentif C</li><li class="reference__author">Hiscock TW</li><li class="reference__author">Jawaid W</li><li class="reference__author">Calero-Nieto FJ</li><li class="reference__author">Mulas C</li><li class="reference__author">Ibarra-Soria X</li><li class="reference__author">Tyser RCV</li><li class="reference__author">Ho DLL</li></ol><em> et al.</em><span class="reference__authors_list_suffix">2018</span><span class="reference__title">Timecourse single-cell RNAseq of whole mouse embryos harvested between days 6.5 and 8.5 of development.</span><span class="reference__origin">ArrayExpress. ID E-MTAB-6967</span><span class="reference__doi"><a href="https://www.ebi.ac.uk/biostudies/arrayexpress/studies/E-MTAB-6967" class="reference__external_link">https://www.ebi.ac.uk/biostudies/arrayexpress/studies/E-MTAB-6967</a></span></li><li class="reference-list__item" id="dataref4"><ol class="reference__authors_list"><li class="reference__author">Tyser RCV</li><li class="reference__author">Mahammadov E</li><li class="reference__author">Nakanoh S</li><li class="reference__author">Vallier L</li><li class="reference__author">Scialdone A</li><li class="reference__author">Srinivas S</li></ol><span class="reference__authors_list_suffix">2021</span><span class="reference__title">A spatially resolved single cell atlas of human gastrulation.</span><span class="reference__origin">ArrayExpress. ID E-MTAB-9388</span><span class="reference__doi"><a href="https://www.ebi.ac.uk/biostudies/arrayexpress/studies/E-MTAB-9388" class="reference__external_link">https://www.ebi.ac.uk/biostudies/arrayexpress/studies/E-MTAB-9388</a></span></li><li class="reference-list__item" id="dataref5"><ol class="reference__authors_list"><li class="reference__author">Koh PW</li><li class="reference__author">Sinha R</li><li class="reference__author">Barkal AA</li><li class="reference__author">Morganti RM</li><li class="reference__author">Chen A</li><li class="reference__author">Weissman IL</li><li class="reference__author">Ang LT</li><li class="reference__author">Kundaje A</li><li class="reference__author">Loh KM</li></ol><span class="reference__authors_list_suffix">2016</span><span class="reference__title">In vitro cultured H7 human embryonic stem cells (WiCell) and H7-derived downstream early mesoderm progenitors.</span><span class="reference__origin">NCBI Gene Expression Omnibus. ID GSE85066</span><span class="reference__doi"><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE85066" class="reference__external_link">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE85066</a></span></li></ul></section></section><section id="peer-reviews"><h1>Peer reviews</h1><section id="sa1" class="review-content"><h2>Reviewer #1 (Public review):</h2><div class="review-content__body"><p>Summary:</p><p>This is an interesting study on the role of FGF signaling in the induction of primitive streak like-cells (PS-LC) in human 2D-gastruloids. The authors use a previously characterized standard culture that generates a ring of PS-LCs (TBXT+) and correlate this with pERK staining. A requirement for FGF signaling in TBXT induction is demonstrated via pharmacological inhibition of MEK and FGFR activity. A second set of culture conditions (with no exogenous FGFs) suggests that endogenous FGFs are required for pERK and TBXT induction. The authors then characterize, via scRNA-seq, various components of the FGF pathway (genes for ligand, receptors, ERK regulators, HSPG regulation). They go on to characterize the pFGFR1, receptor isoforms and polarized localization of this receptor. Finally, they perform FGF4 inhibition and use a cell line with a limited FGF17 inactivation (heterozygous null) and show that loss of these FGFs reduce PS-LC and derivative cell types.</p><p>Strengths:</p><p>(1) As the authors point out, the role of FGF signaling in gastrulation is less well understood than other signaling pathways. Hence this is a valuable contribution to that field.</p><p>(2) The FGF4 and FGF17 loss-of-function experiments in Figure 5 are very intriguing. This is especially so given the intriguing observation that these FGFs appear to be dominating in this model of human gastrulation, in contrast to what FGFs dominate in mice, chick and frogs.</p><p>(3) In general this paper is valuable as a further development of the Human gastruloid system and the role of FGF signaling in the induction of PS-CLs. The wide net that the authors cast in characterizing FGF ligand gene, receptor isoforms, and downstream components provides a foundation for future work. As the authors write near the beginning of the Discussion "Many questions remain."</p><p>Weaknesses:</p><p>(1) FGFs are cell survival factors in various aspects of development. The authors fail to address cell death due to loss of FGF signaling in any of their experiments. For example, in Figure 1E (which requires statistical analysis) and 1G (the bottom FGFRi row), there appears to be a significant amount of cell loss. Is this due to cell death? The authors should address the question of whether the role of FGF/ERK signaling is to keep the cells alive.</p><p>(2) Regarding the sparse cells in 1G, is there a reduction in cell number only with FGFRi and not MEKi? Is this reproducible? Gattiglio et al (Development, 2023, PMID: 37530863) present data supporting a "community effect" in the FGF-induced mesoderm differentiation of mouse embryonic stem cells. Could a community effect be at play in this human system (especially given the images in the bottom row of 1G). If the authors don't address this experimentally they should at least address the ideas in Gattoglio et al.</p><p>(3) Do the FGF4 and FGF17 LOF experiments in Figure 5 affect cell number like FGFRi in Figure 1? Why examine PS-LC induction only in FGF17 heterozygous cells and not homozygous FGF17 nulls?</p><p>(4) The idea that FGF8 plays a dominant role during gastrulation of other species but not humans is so intriguing it warrants deeper testing. The authors dismiss FGF8 because its mRNA "...levels always remained low." (line 363) as well as the data published in Zhai et al (PMID: 36517595) and Tyser et al (PMID: 34789876). But there are cases in mouse development where a gene was expressed at levels so low, it might be dismissed, and yet LOF experiments revealed it played a role or even was required in a developmental process. The authors should consider FGF8 inhibition or inactivation to explore its potential role, despite its low levels of expression.</p><p>(5) Redundancy is a common feature in FGF genetics. What is the effect of inhibiting FGF4 in FGF17 LOF cells?</p><p>(6) I suggest stating that the authors take more caution describing FGF gradients. For example, in one Results heading they write "Endogenous FGF4 and FGF17 gradients underly the ERK activity pattern.", implying an FGF protein gradient. However, they only present data for FGF mRNA , not protein. This issue would be clarified if they used proper nomenclature for gene, mRNA (italics) and protein (no italics) throughout the paper.</p><p>Comments on revisions:</p><p>The authors have addressed my concerns.</p></div><div class="descriptors"><ul class="descriptors__identifiers"><li class="descriptors__identifier"><a href="https://doi.org/10.7554/eLife.101224.2.sa3">https://doi.org/10.7554/eLife.101224.2.sa3</a></li></ul></div></section><section id="sa2" class="review-content"><h2>Reviewer #2 (Public review):</h2><div class="review-content__body"><p>Summary:</p><p>The role of FGFs in embryonic development and stem cell differentiation has remained unclear due to its complexity. In this study, the authors utilized a 2D human stem cell-based gastrulation model to investigate the functions of FGFs. They discovered that FGF-dependent ERK activity is closely linked to the emergence of primitive streak cells. Importantly, this 2D model effectively illustrates the spatial distribution of key signaling effectors and receptors by correlating these markers with cell fate markers, such as T and ISL1. Through inhibition and loss-of-function studies, they further corroborated the needs of FGF ligands. Their data shows that FGFR1 is the primary receptor, and FGF2/4/17 are the key ligands for primitive streak development, which aligns with observations in primate embryos. Additional experiments revealed that the reduction of FGF4 and FGF17 decreases ERK activity.</p><p>Strengths:</p><p>This study provides comprehensive data and improves our understanding of the role of FGF signaling in primate primitive streak formation. The authors provide new insights related to the spatial localization of the key components of FGF signaling and attempt to reveal the temporal dynamics of the signal propagation and cell fate decision, which has been challenging.</p></div><div class="descriptors"><ul class="descriptors__identifiers"><li class="descriptors__identifier"><a href="https://doi.org/10.7554/eLife.101224.2.sa2">https://doi.org/10.7554/eLife.101224.2.sa2</a></li></ul></div></section><section id="sa3" class="review-content"><h2>Reviewer #3 (Public review):</h2><div class="review-content__body"><p>Jo and colleagues set out to investigate the origins and functions of localized FGF/ERK signaling for the differentiation and spatial patterning of primitive streak fates of human embryonic stem cells in a well-established micropattern system. They demonstrate that endogenous FGF signaling is required for ERK activation in a ring-domain in the micropatterns, and that this localized signaling is directly required for differentiation and spatial patterning of specific cell types. Through high-resolution microscopy and transwell assays, they show that cells receive FGF signals through basally localized receptors. Finally, the authors find that there is a requirement for exogenous FGF2 to initiate primitive streak-like differentiation, but endogenous FGFs, especially FGF4 and FGF17, fully take over at later stages.</p><p>Even though some of the authors' findings - such as the localized expression of FGF ligands during gastrulation and the importance of FGF/ERK signaling for cell differentiation in the primitive streak - have been reported in model organisms before, this is one of the first studies to investigate the role of FGF signaling during primitive streak-like differentiation of human cells. In doing so, the paper reports a number of interesting and valuable observations, namely the basal localization of FGF receptors which mirrors that of BMP and Nodal receptors, as well as the existence of a positive feedback loop centered on FGF signaling that drives primitive-streak differentiation. In the revised version of their work, the authors have furthermore dissected the role of different FGFs through knockdown approaches. These experiments reveal discrete functions for different FGF genes in their system, as well as interesting differences between the role of specific FGFs in human compared to model systems.</p><p>Comments on revisions:</p><p>The authors have appropriately addressed all comments and suggestions from the previous round of review. The only textual change that I would still like to suggest is to write explicitly in the main text corresponding to Fig. 1 that the mTESR1 medium used for these initial experiments already contains FGF. This is something that is probably known to experts in the field, but not necessarily to a broader readership.</p></div><div class="descriptors"><ul class="descriptors__identifiers"><li class="descriptors__identifier"><a href="https://doi.org/10.7554/eLife.101224.2.sa1">https://doi.org/10.7554/eLife.101224.2.sa1</a></li></ul></div></section><section id="sa4" class="author-response"><h2>Author response:</h2><div class="author-response__body"><p>The following is the authors’ response to the original reviews.</p><blockquote><p><strong>Reviewer #1 (Public review):</strong></p><p>Summary:</p><p>This is an interesting study on the role of FGF signaling in the induction of primitive streak-like cells (PS-LC) in human 2D-gastruloids. The authors use a previously characterized standard culture that generates a ring of PSLCs (TBXT+) and correlate this with pERK staining. A requirement for FGF signaling in TBXT induction is demonstrated via pharmacological inhibition of MEK and FGFR activity. A second set of culture conditions (with no exogenous FGFs) suggests that endogenous FGFs are required for pERK and TBXT induction. The authors then characterize, via scRNA-seq, various components of the FGF pathway (genes for ligands, receptors, ERK regulators, and HSPG regulation). They go on to characterize the pFGFR1, receptor isoforms, and polarized localization of this receptor. Finally, they perform FGF4 inhibition and use a cell line with a limited FGF17 inactivation (heterozygous null) and show that loss of these FGFs reduces PS-LC and derivative cell types.</p><p>Strengths:</p><p>(1) As the authors point out, the role of FGF signaling in gastrulation is less well understood than other signaling pathways. Hence this is a valuable contribution to that field.</p><p>(2) The FGF4 and FGF17 loss-of-function experiments in Figure 5 are very intriguing. This is especially so given the intriguing observation that these FGFs appear to be dominating in this model of human gastrulation, in contrast to what FGFs dominate in mice, chicks, and frogs.</p><p>(3) In general this paper is valuable as a further development of the Human gastruloid system and the role of FGF signaling in the induction of PS-CLs. The wide net that the authors cast in characterizing the FGF ligand gene, receptor isoforms, and downstream components provides a foundation for future work. As the authors write near the beginning of the Discussion "Many questions remain."</p></blockquote><p>We thank the reviewer for these positive comments.</p><blockquote><p>Weaknesses:</p><p>(1) FGFs are cell survival factors in various aspects of development. The authors fail to address cell death due to loss of FGF signaling in their experiments. For example, in Figure 1E (which requires statistical analysis) and 1G (the bottom FGFRi row), there appears to be a significant amount of cell loss. Is this due to cell death? The authors should address the question of whether the role of FGF/ERK signaling is to keep the cells alive.</p></blockquote><p>Indeed, FGF also strongly affects cell survival and it is an interesting question to what extent this depends on ERK. Our manuscript focuses instead on the role of FGF/ERK signaling in cell fate patterning. As mentioned in our discussion, figure 1de show that doxycycline induced pERK leads to more TBXT+ cells than the control without restoring cell number, suggesting the role of FGF in controlling cell number is independent of the requirement for FGF/ERK in PS-LC differrentiation. To further support this, we have added data showing low doses of MEKi are sufficient to inhibit differentiation without affecting cell number (Supp. Fig. 1i).</p><p>To address the reviewers question regarding the cause of cell loss, we now stained for BrdU and cleaved Cas3 to assess proliferation and apoptosis in the presence and absence of MEK and FGFR inhibition (new Supp. Fig.</p><p>1ef). This shows that the effect of these inhibitors on cell number is primarily due to a reduction in proliferation. We have also included statistical analysis in Fig.1e.</p><blockquote><p>(2) Regarding the sparse cells in 1G, is there a reduction in cell number only with FGFRi and not MEKi? Is this reproducible? Gattiglio et al (Development, 2023, PMID: 37530863) present data supporting a "community effect" in the FGF-induced mesoderm differentiation of mouse embryonic stem cells. Could a community effect be at play in this human system (especially given the images in the bottom row of 1G)? If the authors don't address this experimentally they should at least address the ideas in Gattoglio et al.</p></blockquote><p>Indeed, FGFRi reproducibly affects cell number more than MEKi, in line with the fact that pathways other than MAPK/ERK downstream of FGF (e.g. PI3K) play important roles in cell survival and growth. However, we think the lack of differentiation in MEKi and FGFRi in Fig.1g cannot be attributed to a loss of cells combined with a community effect. This is because without FGFRi or MEKi cells efficiently differentiate to primitive streak at much lower densities than those originally shown, consistent with the data we discuss in response to (1) arguing against a primarily indirect effect of FGF on PS-LC differentiation through cell density. In the context of directed differentiation (rather than 2D gastruloids), we have now shown in a controlled manner that the effect of MEKi and FGFRi does not depend on a community effect by repeating the experiment in Fig.1g while adjusting cell seeding densities to obtain similar final cell densities in all three conditions (new Fig.1g, new Supp Fig.1g). Furthermore we have included new data showing extremely sparse cells without MEKi or FGFRi still differentiate without problems (new Supp Fig 1h). We have also include Gattoglio et al in our revised discussion.</p><blockquote><p>(3) Do the FGF4 and FGF17 LOF experiments in Figure 5 affect cell numbers like FGFRi in Figure 1?</p></blockquote><p>We did not observe major changes in cell number in the FGF4 and FGF17 loss of function experiments. This is in line with our observation that low levels of ERK signaling are sufficient to maintain proliferation (new Supp. Fig. 1i), and the fact that low levels of ERK signaling are maintained in the absence of FGF4 and FGF17 (Fig.5), likely by FGF2 (Fig. 2). In contrast, FGFRi treatment in Fig.1 leads to a nearly complete loss of FGF signaling (ERK and other pathways) that has a dramatic effect on cell number.</p><blockquote><p>Why examine PS-LC induction only in FGF17 heterozygous cells and not homozygous FGF17 nulls?</p></blockquote><p>We were unable to obtain homozygous FGF17 nulls, it is not clear if there is a reason for this. In the absence of homozygous nulls, we have now further corroborated our findings with additional knockdown data (described in response to other comments below).</p><blockquote><p>(4) The idea that FGF8 plays a dominant role during gastrulation of other species but not humans is so intriguing it warrants deeper testing. The authors dismiss FGF8 because its mRNA "...levels always remained low." (line 363) as well as the data published in Zhai et al (PMID: 36517595) and Tyser et al (PMID: 34789876). But there are cases in mouse development where a gene was expressed at levels so low, that it might be dismissed, and yet LOF experiments revealed it played a role or even was required in a developmental process. The authors should consider FGF8 inhibition or inactivation to explore its potential role, despite its low levels of expression.</p></blockquote><p>We thank the reviewer for this suggestion. We have now analyzed the role of FGF8 using FISH to visualize its expression and siRNA to understand its function (Fig.5d,f,h; Supp.Fig.5e,g,6e). We found that FGF8 expression is higher earlier in differentiation, preceding most expression of TBXT. Our scRNA-seq only analyzed samples at 42h so did not capture this. Furthermore, FGF8 expression localized inside the PS-like ring rather than coinciding with it like FGF4. Surprisingly, FGF8 knockdown led to an increase in primitive streak-like differentiation, suggesting it may counteract FGF4. The results are shown in the revised Fig. 5 and Supplemental Fig. 5. While this certainly merits further investigation, understanding the role of FGF8 in more detail is beyond the scope of the current work.</p><blockquote><p>(5) Redundancy is a common feature in FGF genetics. What is the effect of inhibiting FGF4 in FGF17 LOF cells?</p></blockquote><p>Further siRNA and shRNA experiments showed that FGF17 knockdown had a much smaller effect than FGF4 knockdown on expression of primitive streak markers (Fig.5i, Supp.Fig.6f-i) but that FGF17 knockdown did lead to a complete loss of the mesoderm marker TBX6 (Fig.5j, Supp.Fig.6j). A double knockdown of FGF4+FGF17 looked similar to FGF4 alone (Supp.Fig.6k). Thus, we now think the more likely scenario is that FGF17 is downstream of FGF4-dependent PS-differentiation and although this may have a positive feedback effect whereby this FGF17 can then enhance further PS-differentiation, which we previously interpreted as partial redundancy, the primary role of FGF17 may be later, in mesoderm differentiation.</p><blockquote><p>(6) I suggest stating that the authors take more caution in describing FGF gradients. For example, in one Results heading they write "Endogenous FGF4 and FGF17 gradients underly the ERK activity pattern.", implying an FGF protein gradient. However, they only present data for FGF mRNA , not protein. This issue would be clarified if they used proper nomenclature for gene, mRNA (italics), and protein (no italics) throughout the paper.</p></blockquote><p>Thank you for the suggestion. We have edited the paper to more clearly distinguish protein and mRNA. We do think our data provide substantial indirect evidence for a protein gradient which is what the results heading is meant to convey. Receptor activation is high where ERK activity is high (Fig.3), and receptor activation is limited by ligands, since creating a scratch to let exogenous FGF reach the basal side of cells in the center leads to receptor activation (Fig.4). This strongly suggests ERK activity reflects an FGF protein gradient.</p><blockquote><p><strong>Reviewer #2 (Public review):</strong></p><p>Summary:</p><p>The role of FGFs in embryonic development and stem cell differentiation has remained unclear due to its complexity. In this study, the authors utilized a 2D human stem cell-based gastrulation model to investigate the functions of FGFs. They discovered that FGF-dependent ERK activity is closely linked to the emergence of primitive streak cells. Importantly, this 2D model effectively illustrates the spatial distribution of key signaling effectors and receptors by correlating these markers with cell fate markers, such as T and ISL1. Through inhibition and loss-of-function studies, they further corroborated the needs of FGF ligands. Their data shows that FGFR1 is the primary receptor, and FGF2/4/17 are the key ligands for primitive streak development, which aligns with observations in primate embryos. Additional experiments revealed that the reduction of FGF4 and FGF17 decreases ERK activity.</p><p>Strengths:</p><p>This study provides comprehensive data and improves our understanding of the role of FGF signaling in primate</p><p>primitive streak formation. The authors provide new insights related to the spatial localization of the key components of FGF signaling and attempt to reveal the temporal dynamics of the signal propagation and cell fate decision, which has been challenging.</p><p>Weaknesses:</p><p>Given the solid data, the work only partially clarifies the complex picture of FGF signaling, so details remain somewhat elusive. The findings lack a strong punchline, which may limit their broader impact.</p></blockquote><p>We thank this reviewer for their valuable feedback and compliment on the solidity of our data. The punchline of our work is that FGF4 and FGF17-dependent ERK signaling plays a key role in differentiation of human PS-like cells and mesoderm, and that these are different FGFs than those thought to drive mouse gastrulation. A second key point is that like BMP and TGFβ signaling, FGF signaling is restricted to the basolateral sides of pluripotent stem cell colonies due to polarized receptor expression, which is crucial for understanding the response to exogenous ligands added to the cell medium. Indeed, many facets of FGF signaling remain to be investigated in the future, such as how FGF regulates and is regulated by other signals, which we will dedicate a different manuscript to.</p><blockquote><p><strong>Reviewer #3 (Public review):</strong></p><p>Jo and colleagues set out to investigate the origins and functions of localized FGF/ERK signaling for the differentiation and spatial patterning of primitive streak fates of human embryonic stem cells in a well-established micropattern system. They demonstrate that endogenous FGF signaling is required for ERK activation in a ringdomain in the micropatterns, and that this localized signaling is directly required for differentiation and spatial patterning of specific cell types. Through high-resolution microscopy and transwell assays, they show that cells receive FGF signals through basally localized receptors. Finally, the authors find that there is a requirement for exogenous FGF2 to initiate primitive streak-like differentiation, but endogenous FGFs, especially FGF4 and FGF17, fully take over at later stages.</p><p>Even though some of the authors' findings - such as the localized expression of FGF ligands during gastrulation and the importance of FGF/ERK signaling for cell differentiation in the primitive streak - have been reported in model organisms before, this is one of the first studies to investigate the role of FGF signaling during primitive streak-like differentiation of human cells. In doing so, the paper reports a number of interesting and valuable observations, namely the basal localization of FGF receptors which mirrors that of BMP and Nodal receptors, as well as the existence of a positive feedback loop centered on FGF signaling that drives primitive-streak differentiation. The authors also perform a comparison of the role of different FGFs across species and try to assign specific functions to individual FGFs. In the absence of clean genetic loss-of-function cell lines, this part of the work remains less strong.</p></blockquote><p>We thank the reviewer for emphasizing the value of our findings in a human model for gastrulation. We agree more loss-of-function experiments would provide further insight into the role of different FGFs. While we did not manage to create knockout cell lines, we have now performed both siRNA and shRNA knock-down of all FGF4, and FGF17 in two different hPSC lines, performed siRNA knockdown of FGF8, and also made a FGF4+FGF17 shRNA double knockdown cell lines to more completely test the functions of the individual FGFs (Fig.5, Supp.Fig.5,6). Our data suggest FGF17 may be downstream of FGF4 and primarily required for mesoderm differentiation while FGF8 appears to counteract FGF4. In doing this we have added a large amount of new data to the manuscript and we have removed the heterozygous knockout data in the first version of the manuscript which we felt added little to the new data. Further experiments are still needed to solidify our interpretation but those are beyond the scope of the current work.</p><blockquote><p><strong>Reviewer #1 (Recommendations for the authors):</strong></p><p>(1) FGF2 is added to culture experiments (e.g. Figure 4), but the commercial source is not mentioned in Methods. For example, it could be added to "Supplementary Table 1: Cell signaling reagents."</p></blockquote><p>We apologize for this oversight and have now added the information to Supplementary Table 1.</p><blockquote><p>(2) Line 117-118: "For example, by controlling the expression of Wnt or Nodal which are both required for PS-like differentiation". It is clear what the authors mean, but this is not a complete sentence.</p></blockquote><p>We edited this for clarity, it now reads: “First, is FGF/ERK signaling required directly for PS-like differentiation, or does it act indirectly? These possibilities are not mutually exclusive. For example, FGF/ERK could be required directly but also act indirectly by controlling Wnt or Nodal expression, as both Wnt and Nodal signaling are required for PS-like differentiation.”</p><blockquote><p>(3) Line 246 "...found its spatial pattern to strongly resembles that of pERK..." either remove "to" or change "resembles" to "resemble"</p></blockquote><p>Thank you for catching this. We removed “to”.</p><blockquote><p>(4) Lines 391- 393 seem to be missing a word in the last phrase: "...with FGF17 more important continued differentiation to mesoderm and endoderm." Maybe "during" after the word "important"?</p></blockquote><p>Thank you for catching this, indeed the word “during” was missing and we have now added it.</p><blockquote><p>(5) Please define acronyms in Figure 3D (PS-LC was defined previously, but not others).</p></blockquote><p>We apologize for the oversight, we have now defined the acronyms.</p><blockquote><p>(6) The three blue lines in Figure 5B (right) are hard to discern (and I'm not colorblind). I suggest also using a variety of dotted lines in a subset of these FGFs.</p></blockquote><p>Thanks you for the suggestion. We have now given all the FGFs colors that are more clearly distinct and made the TBXT and TBX6 lines dashed.</p><blockquote><p><strong>Reviewer #2 (Recommendations for the authors):</strong></p><p>(1) The reviewer acknowledges that FGF signaling is complex, particularly when dynamics and its correlation with cell fates are considered. To improve the clarity of the findings, the authors are encouraged to provide an additional schematic figure that clearly delineates the main findings of this study.</p></blockquote><p>Thank you for the suggestion. We have now added a summary figure (Fig.6) to our discussion, which we hope helps present our findings more clearly.</p><blockquote><p>(2) The data suggest that FGF signaling may function differently in mice compared to primates, and their stem cell model aligns more closely with the latter. While the authors discuss this in the contents only based on sequencing data, it would be valuable to conduct some experiments with mouse embryos to validate the key differences.</p></blockquote><p>It is unclear to us which experiments the reviewer has in mind. There is ample data on FGF expression in the mouse literature, as are many knockout phenotypes. Furthermore, verifying loss of function phenotypes (e.g. FGF17 knockout) in mouse is beyond our expertise.</p><blockquote><p>(3) Heparan sulfate proteoglycan (HSPG) is mentioned as an important component of FGF signaling; however, the only data related to HSPG is single-cell sequencing results. The authors should consider performing immunostaining or other assays to validate HSPG expression and spatial distribution, similar to the approach they used for other signaling components.</p></blockquote><p>Our scratch experiments in Fig. 4 strongly argue against HSPGs as being responsible for the spatial pattern of FGF receptor activation: after a scratch across the colony the response is strong all along the scratch as expected if presence of FGF (an FGF gradient) controls the level of activity. If HSPGs were limiting, FGF flowing in from the media show not be able to uniformly activate receptors around the scratch.</p><p>In addtion, we have now included an immunostain for HS in a newly added Supp. Fig. 4 which does not explain the observed pattern of ERK signaling.</p><blockquote><p>(4) In the scratch experiment, particularly high PERK expression is observed at the edge of the scratch. The authors should provide an explanation for why this expression is significantly higher compared to the edges of the colony. Additionally, it would be interesting to investigate the fate of the cells with super high PERK expression.</p></blockquote><p>We have now determined that adaptive response to FGF is the reason that the response around the scratch is initially much higher than in the ERK activity ring that overlaps with the primitive streak-like cells. We have added figures showing that although the intial response to FGF exposure after scratching is very high, the response around the scratch adapts to levels similar in those in the ERK ring over the course of 6 hours (Fig.4ij).</p><blockquote><p>(5) For some of the key experiments, multiple cell lines should be used to ensure that the findings are reproducible and applicable across different human stem cell lines.</p></blockquote><p>We have now checked FISH stainings and knockdown phenotypes for different FGFs in two different cell lines: ESI17 (hESC, XX) and PGP1 (hiPSC, XY). These results are shown in Supplementary Figures 6. We found all results to be consistent.</p><blockquote><p>(6) Where applicable, the meaning of error bars needs to be more clearly presented, including details on the number of independent experiments or samples used.</p></blockquote><p>Thank you for pointing this out. Where error bar definitions were missing we have now added them to the figure captions.</p><blockquote><p><strong>Reviewer #3 (Recommendations for the authors):</strong></p><p>(1) The authors only analyze the ppERK ring in micropatterns of a single size. What was the motivation for the choice of this size? Can the authors how the ppERK ring is expected to depend on colony size?</p></blockquote><p>Much smaller patterns lose the interior pluripotent regions while much larger patters have a much larger pluripotent region, which requires larger tilings to image without providing additional insight. The colony sizedependence of cell fate patterning was described in the paper that established the 2D gastruloids model (Warmflash Nat Methods 2014) and we later showed this due to a fixed length scale of the BMP and Nodal signaling gradients from the colony edge (Jo et al Elife 2022). We have now included data showing that the ERK patterns behaves similarly, with a fixed length scale of the pattern implying that in smaller colonies the ERK ring becomes a disc and the entire center of the colony has high ERK signaling (Supp Fig 1a).</p><blockquote><p>(2) The scRNAseq is somewhat confusing - why do the two datasets not overlap in the PHATE representation? This is unexpected, because the two samples have been treated similarly, and the authors have integrated their data to iron out possible batch effects. This discrepancy should be discussed. The authors should also specify from which reference exactly the first dataset comes from.</p></blockquote><p>The two datasets do overlap nicely, the same fates are well mixed in the same place and the gene expresison profiles for the integrated data (e.g., Fig.2e) look smooth, so we believe the integration is good, but different cell fates are represented to different degrees. In particular, sample 2 shows much more mesoderm differentiation making the mesoderm branch mostly orange. Occassionally samples differentiate faster or slower than average which we see here, and these samples were collected far apart in time. We do not believe this affects our conclusions, if anything, we think performing the analysis on two samples that differ this much should make the conclusions more robust.</p><blockquote><p>(3) If find it intriguing that exogenous FGF2 is important early on for primitive streak-like differentiation, although the authors show that it does not reach the center of the colony. The authors may want to discuss this conundrum. Does the FGF2 effect propagate from the outside to the inside, or does it act at an early stage when the cells have not yet formed a tight epithelium on the micropattern?</p></blockquote><p>The cells in the experiment in Fig. 5a were given 24h to epithelialize, so we we do believe it acts from the edge. We believe this may be due to FGF2 modulating the early BMP response on the edge and are working on a manuscript that further explores this pathway crosstalk.</p><blockquote><p>(4) The authors' statement that FGF4 and FGF17 have partially redundant functions is not very strong, mainly because the study lacks a full FGF17 loss-of-function cell line. If the authors wanted to improve on this point, they could knock down FGF4 in the FGF17 heterozygous line, or produce a homozygous FGF17 KO line. If there are specific reasons why FGF17 homozygous lines cannot be produced, this could be interesting to discuss, too. Finally, I noticed that the methods list experiments with an FGF17 siRNA, but these are not shown in the manuscript.</p></blockquote><p>We agree our evidence was previously not as strong as it could be. While there is no reason we know of why homozygous knockout lines cannot be produced, we failed to produce on. To strengthen our evidence we have therefore included substantial new knockdown data.  We have now performed both siRNA and shRNA knockdown of all FGF4, and FGF17 in two different hPSC lines, performed siRNA knockdown of FGF8, and also made a FGF4+FGF17 shRNA double knockdown cell lines to more completely test the functions of the individual FGFs (Fig.5, Supp.Fig.5,6). These experiments showed that FGF17 knockdown had a much smaller effect than FGF4 knockdown on expression of primitive streak markers (Fig.5i, Supp.Fig.6f-i) but that FGF17 knockdown did lead to a complete loss of the mesoderm marker TBX6 (Fig.5j, Supp.Fig.6j). A double knockdown of FGF4+FGF17 looked similar to FGF4 alone (Supp.Fig.6k). Thus, we now think the more likely scenario is that FGF17 is downstream of FGF4-dependent PS-differentiation and although this may have a positive feedback effect whereby this FGF17 can then enhance further PS-differentiation, which we previously interpreted as partial redundancy, the primary role of FGF17 may be later, in mesoderm differentiation. Furthermore, our new data suggests FGF8 may counteract FGF4 and limit PS-like differentiation.</p><blockquote><p>Minor</p><p>(5) Line 63: Reference(s) appear to be missing.</p></blockquote><p>This whole paragraph summarizes the results of the references given on line 55, we have now repeated the relevant references where the reviewer indicated.</p><blockquote><p>(6) Supplementary Figure 1a,b does not show ppERK, unlike stated in lines 102 - 104.</p></blockquote><p>Indeed, the data described in lines 102-104 is shown in Fig.1a and we have removed the original Supplementary Figure 1ab since it did not provide relevant information.</p><blockquote><p>(7) Line 201: It is not clear whether this is a new sequencing dataset, or if existing datasets have been reanalyzed.</p></blockquote><p>We agree our description was unclear. We have edited the text, which now explicitly states that our analysis is based on one dataset we collected previously and a replicate that was newly collected and deposited on GEO for this manuscript.</p><blockquote><p>(8) Figure 2f; Supplementary Figure 2b, c: The colors need to be explained in scale bars. How has this data been normalized to allow for comparison between very different sample types?</p></blockquote><p>We have now added color bars indicating the scale for each of these figure panels. As the caption stated, the interspecies comparison was normalized within each species, so the highest FGF level for any FGF at any time within each species is normalized to one. We are thus comparing between species the relative expression of different FGFs within each species. Indeed there is no good way to compare absolute expression between species. For extra clarity we have expanded our description of the interspecies comparison analysis and normalization in the methods section.</p><blockquote><p>(9) Line 232: Where is the expression of SEF shown?</p></blockquote><p>It is shown in Fig. 2i, under the official gene name IL17RD.</p><blockquote><p>(10) Supplementary Figure 4 seems to be missing.</p></blockquote><p>Thank you for pointing this out. We have now added a supplementary Fig.4.</p><blockquote><p>(11) Line 437: Citation needed.</p></blockquote><p>We have included citations now.</p><blockquote><p>(12) Line 439: A similar feedback loop has been proposed to operate during mesoderm differentiation in mouse ESC (pmid: 37530863 ). The authors may consider citing this work.</p></blockquote><p>Thank you for the suggestion, we have now included this work in the discussion. The feedback loop proposed in that work involves FGF8, while we were trying to explain why FGF4 and not FGF8 appears to be conserved across species by invoking an FGF4 feedback loop. Thus, it becomes even harder to explain differences in FGF4 and FGF8 expression between human and mouse gastrulation.</p><blockquote><p>(13) Supplementary Figure 6 is not described in the main text.</p></blockquote><p>We have removed the original Supplementary Figure 6 and corresponding heterozygous knockout data in the main figure which we felt added little to the extensive knockdown data we now present. We did create a new Supplementary Figure 6 showing additional knockdown data which is described in the main tekst.</p><blockquote><p>(14) Submission of sequencing data to GEO needs to be updated.</p></blockquote><p>We have now made the GEO data public.</p></div><div class="descriptors"><ul class="descriptors__identifiers"><li class="descriptors__identifier"><a href="https://doi.org/10.7554/eLife.101224.2.sa0">https://doi.org/10.7554/eLife.101224.2.sa0</a></li></ul></div></section></section></article></body></html>