<!DOCTYPE html>
<html><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charSet="utf-8"><meta name="viewport" content="width=device-width"><title>Two neuropeptides that promote blood-feeding in Anopheles stephensi mosquitoes</title><link rel="preload" href="print.css" as="style"><link rel="stylesheet" href="print.css"><style media="print">#fig1 {--not-to-fill: ok; break-before: page;}              #fig1 > img {max-width: 90% !important; max-height: unset !important; height: auto !important; text-align: center !important;}#fig4 > img, figure[data-continued-from="ctn-fig4"] {width: 100% !important; max-width: 100% !important;}</style></head><body><div id="page-templates"><p class="runninghead"><span class="author">Bansal et al.,</span> <span class="date">2025 <em>eLife</em> <strong>14</strong>:RP108625.</span> <span class="doi"> <a href="https://doi.org/10.7554/eLife.108625.1">https://doi.org/10.7554/eLife.108625.1</a></span><span class="counter"></span></p><ul class="article-flag-list"><li class="article-flag-list__item"><a class="article-flag__link" href="https://elifesciences.org/subjects/neuroscience">Neuroscience</a></li></ul><div id="first-page-links" class="elife-intro"><a id="logo" href="https://elifesciences.org/?utm_source=pdf&amp;utm_medium=article-pdf&amp;utm_campaign=PDF_tracking"><img alt="eLife logo" loading="eager" width="80" height="30" decoding="async" data-nimg="1" class="site-header__logo" style="color:transparent" src="assets/elife-logo-xs.fd623d00.svg"></a><div id="right-header-links"><a href="https://en.wikipedia.org/wiki/Open_access" target="_blank" class="header-icon oa-icon"></a><span class="separator">|</span><a href="https://creativecommons.org/" target="_blank" class="header-icon cc-icon"></a></div></div></div><article><section id="sideSection" class="side-section"><div class="article-status"><div class="review-timeline-container"><dl class="review-timeline review-timeline--expanded" id="review-timeline" aria-label="Version history"><dt class="review-timeline__event review-timeline__event--title review-timeline__event--active review-timeline__event--with-evaluation-summary review-timeline__event--reviewed">Reviewed Preprint</dt><dd class="review-timeline__event review-timeline__event--detail review-timeline__event--active review-timeline__event--with-evaluation-summary review-timeline__event--reviewed"><span class="review-timeline__version"> v1</span><time class="review-timeline__date" dateTime="2025-10-21T00:00:00.000Z">October 21, 2025</time><span class="review-timeline__link">Not revised</span></dd></dl></div></div><section id="margin-notes"><div class="author-notes"><p><strong class="email-icon">For correspondence:</strong><ul class="list-simple"><li><a href="mailto:prashali.bansal@tigs.res.in">prashali.bansal@tigs.res.in</a></li><li><a href="mailto:sonia.sen@tigs.res.in">sonia.sen@tigs.res.in</a></li></ul></p><p class="author-notes__list_item"><strong>Competing interests:</strong> No competing interests declared</p><p class="author-notes__list_item"><strong>Funding:</strong> See <a class="page-ref" href="#funding"></a></p></div><div id="editors"><p class="editors-and-reviewers__person"><strong>Reviewing editor: </strong>Meet Zandawala, University of Nevada, Reno, United States</p></div><div id="copyright" class="copyright"><p>© 2025, Bansal et al. This article is distributed under the terms of the <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution License</a>, which permits unrestricted use and redistribution provided that the original author and source are credited.</p></div></section></section><header id="header" class="content-header"><h1 class="title">Two neuropeptides that promote blood-feeding in <em>Anopheles stephensi</em> mosquitoes</h1><div class="authors"><ol class="authors-list authors-list--expanded" aria-label="Authors of this article"><li class="authors-list__item"><span class="authors-link authors-email__link">Prashali Bansal<sup>1</sup><a href="mailto:prashali.bansal@tigs.res.in"><span class="email-icon"></span></a></span></li><li class="authors-list__item"><span class="authors-link">Roshni Pillai<sup>1,</sup><sup>2</sup></span></li><li class="authors-list__item"><span class="authors-link">Pooja D Babu<sup>1</sup></span></li><li class="authors-list__item"><span class="authors-link authors-email__link">Sonia Q Sen<sup>1</sup><a href="mailto:sonia.sen@tigs.res.in"><span class="email-icon"></span></a></span></li></ol></div><div class="institutions"><ol class="institutions-list" aria-label="Author institutions"><li class="institutions-list__item"><sup>1</sup>Tata Institute for Genetics and Society, Bengaluru, India</li><li class="institutions-list__item"><sup>2</sup>Institute of Developmental Biology and Neurobiology, Johannes Gutenberg University, Mainz, Germany</li></ol></div></header><section id="article-content"><section id="sa0" class="assessment"><h2>eLife Assessment</h2><div class="assessment__body"><p>This <strong class="highlighted-term" aria-label="Highlighted">important</strong> study uses a combination of behavioral and molecular techniques to identify neuromodulators that influence blood-feeding behavior in the disease vector, Anopheles stephensi. Through a combination of gene expression analysis and RNA knockdown, the authors identify neuropeptides RYamide and sNPF as candidate regulators for blood-feeding, demonstrate behavioral changes upon co-knockdown, and anatomically characterize their expression patterns. While the evidence for behavioral characterization and expression mapping is <strong class="highlighted-term" aria-label="Highlighted">solid</strong>, the evidence supporting a direct causal role for these neuropeptides in promoting host-seeking remains unproven.</p></div><div class="descriptors"><ul class="descriptors__identifiers"><li class="descriptors__identifier"><a href="https://doi.org/10.7554/eLife.108625.1.sa4">https://doi.org/10.7554/eLife.108625.1.sa4</a></li></ul></div></section><section class="abstract" id="abstract-1"><h1>Abstract</h1><p>Animals routinely need to make decisions about what to eat and when. These decisions are influenced not only by the availability and quality of food but also by the internal state of the animal, which needs to compute and give weights to these different variables before making a choice. Feeding preferences of female mosquitoes exemplify this behavioural plasticity. Both male and female mosquitoes usually feed on carbohydrate-rich sources of nectar or sap, but the female also feeds on blood, which is essential for egg development. This blood-appetite is modulated across the female’s reproductive cycle, yet little is known about the factors that bring it about. We show that mated, but not virgin <em>Anopheles stephensi</em> females, a major vector of urban malaria in the Indian sub-continent and West-Africa, suppress blood-feeding between a blood-meal and oviposition. We identify several candidate genes through transcriptomics of blood-deprived and -sated <em>An. stephensi</em> central brains that could modulate this behaviour. We show that <em>short Neuropeptide-F (sNPF)</em> and <em>RYamide (RYa)</em> act together to promote blood-feeding and identify a cluster of cells in the subesophageal zone that expresses <em>sNPF</em> transcripts only in the blood-hungry state. Such females also have more <em>sNPF</em> transcripts in their midguts. Based on these data, we propose a model where increased <em>sNPF</em> levels in the brain and gut promotes a state of blood-hunger, which drives feeding behaviour either by <em>sNPF’s</em> action in the two tissues independently or via a communication between them.</p></section><section id="s1"><h1 class="heading-1">Introduction and Results</h1><p>An animal’s dietary choice is an outcome of its internal physiological state and external stimuli. Its changing nutritional demands, reproductive state, and developmental history are integrated with external cues before a food-choice is made. For example, when deprived of a particular nutrient, like proteins, animals show a specific increase in protein-appetite (<a class="linktoref" href="#c1">1</a>–<a class="linktoref" href="#c3">3</a>). Mating influences food choices of female flies by increasing their preference for proteins, sugars, and salt for egg development and oviposition (<a class="linktoref" href="#c4">4</a>–<a class="linktoref" href="#c8">8</a>).</p><p>Changes in both sensory perception and central brain circuits have been shown to underlie such behavioural changes. For example, sensory neurons of starved animals become more sensitive to attractive cues and less sensitive to aversive ones, both in vertebrates and invertebrates (<a class="linktoref" href="#c9">9</a>–<a class="linktoref" href="#c16">16</a>). These changes are often brought about by neuromodulators and neuropeptides (<a class="linktoref" href="#c9">9</a>, <a class="linktoref" href="#c10">10</a>, <a class="linktoref" href="#c17">17</a>–<a class="linktoref" href="#c22">22</a>). In the central brain, internal nutrient or hormone sensors can monitor the animal’s nutritional state.</p><p>In mice, neurons in the hypothalamus express receptors for hunger and satiety signals like ghrelin and leptin (<a class="linktoref" href="#c16">16</a>, <a class="linktoref" href="#c23">23</a>, <a class="linktoref" href="#c24">24</a>). In insects, several neurons in the mushroom body, central complex, and subesophageal zone (SEZ) have been shown to integrate hunger and satiety signals to regulate feeding on sugars or proteins, choose between exploration or exploitation of food sources, or initiate or terminate feeding (<a class="linktoref" href="#c25">25</a>–<a class="linktoref" href="#c33">33</a>).</p><p>Female mosquitoes offer a striking model for dissecting how internal state and external cues shape dietary choices. Many female mosquitoes have a dual appetite for carbohydrate-rich nectar or sap – to meet energetic requirements – and protein-rich blood – for egg development. Which meal a female chooses to take is influenced by nutritional demands, reproductive state, and developmental history. For example, mosquitoes with low teneral reserves will take multiple blood meals for egg development (<a class="linktoref" href="#c34">34</a>–<a class="linktoref" href="#c37">37</a>). Nutritional state also influences other internal state outputs on blood-feeding. For example, whether virgin females take blood-meals or not is dependent on prior sugar-feeding in most <em>Aedes</em> strains (<a class="linktoref" href="#c38">38</a>) (and references within). However, mating is dispensable in Anophelines even under conditions of nutritional satiety (<a class="linktoref" href="#c36">36</a>, <a class="linktoref" href="#c39">39</a>–<a class="linktoref" href="#c41">41</a>).</p><p>Despite these species-and context-dependent variations, a consistent finding is that female mosquitoes oscillate between states of heightened and suppressed host-seeking across successive reproductive cycles (<span class="fakelink linktofig"><span class="fakelink-text">figure 1A</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>). For example, in <em>Ae. aegypti</em> and <em>An. gambiae</em>, newly emerged mosquitoes display no or low interest in the host, which increases as they age (<a class="linktoref" href="#c34">34</a>, <a class="linktoref" href="#c42">42</a>–<a class="linktoref" href="#c44">44</a>) and once fed to repletion, host attraction is suppressed. The strength and timing of this suppression varies between species — being robust and well-characterised in mated <em>Ae. aegypti</em> (<a class="linktoref" href="#c45">45</a>–<a class="linktoref" href="#c47">47</a>), and more flexible in Anophelines (<a class="linktoref" href="#c36">36</a>, <a class="linktoref" href="#c48">48</a>, <a class="linktoref" href="#c49">49</a>). Sugar-feeding is not modulated in this way, in some cases it is preferred soon after emergence (<a class="linktoref" href="#c43">43</a>, <a class="linktoref" href="#c50">50</a>).</p><figure class="figure tofill" id="fig1"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Fig. 1.</label></span>Feeding behaviours of <em>An. stephensi</em></h3><p>A. Reproductive cycle of <em>An. stephensi</em>. Upon emergence [1, D0-D5] females have a dual appetite for protein-rich blood and carbohydrate-rich sources of sugar [2,3]. After a blood-meal [3,4,5, D1<sup>B</sup>-D4<sup>B</sup>], the eggs mature and are laid in water [6]. The next reproductive cycle can now begin. B. Blood-feeding assay. 0-7h old mosquitoes were collected in cups and aged appropriately on sugar. Prior to the test, mosquitoes were ‘activated’ for 5 minutes by presenting a human hand, followed by 3 exhalations. Blood-meals were presented through a Hemotek perfumed with human skin odours. Number of blood-fed mosquitoes were visually scored, irrespective of the quantity fed. C. Blood-feeding behaviour of <em>Ae. aegypti</em>. 8-days old virgin and mated females were assessed for their first blood-meal. Fully fed mated females were assessed for second blood-meal (D1<sup>B</sup>-D4<sup>B</sup>). Each dot represents a single trial with 18-21 females, n=10-12trials/group. Kruskal-Wallis with Dunn’s multiple comparisons test, p&lt;0.05. Data labelled with different letters are significantly different. D. Blood-feeding behaviour of <em>An. stephensi</em>. Co-housed and virgin females were tested on the day of emergence (D0) and each day for the next 5 days (D1-D5). D1<sup>B</sup>-D4<sup>B</sup> represents their blood-feeding 1-4 days after their first blood-meal and D1<sup>○</sup>-D2<sup>○</sup> represents their blood-feeding 1-2 days after oviposition (see methods for assay details). 18-21 females/trial, n=9-20trials/group. Generalized Linear Mixed Model with post-hoc pairwise comparisons using estimated marginal means and Bonferroni-corrected p-values; *p&lt;0.05; **p&lt;0.01; ****p&lt;0.0001. E. Sugar-feeding behaviour of <em>An. stephensi</em>. Females of similar conditions as in (D) were tested for sugar-feeding. 18-21females/trial, n=9-20trials/group. Generalized Linear Mixed Model with posthoc pairwise comparisons using estimated marginal means and Bonferroni-corrected p-values; **p&lt;0.01; ***p&lt;0.001; ****p&lt;0.0001. F. Choice assay for blood and sugar. Sugar-sated females or sugar-starved females, co-housed with males (indicated by the dashed abdomens) were given a choice between blood and sugar and assessed for their choice of food (bottom): blood and sugar; blood only; sugar only; none. While most sugar-sated females took blood (left of dashed line), most sugar-starved females fed on both (right of dashed line). As an internal control, co-housed males were accessed for sugar-feeding only and most were found to be fed in both conditions. 17-24 mosquitoes/trial; n=11-12 trials/condition. One-way ANOVA with Holm-Šídák multiple comparisons test, p&lt;0.05. Data labelled with different letters are significantly different. G. Mating and blood-feeding. The mating status of D0-D5 co-housed females assayed in (D) were determined post-hoc by dissecting their spermathecae. n=232-239 females were analysed for each day. Many blood-fed females were unmated. H. Mating after blood-feeding. Blood-fed virgins were allowed to mate and then offered a second blood-meal. A suppression in blood-appetite was observed. 15-20females/trial, n=10 trials. Unpaired t-test; ****p&lt;0.0001.</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://iiif.elifesciences.org/epp/108625%2Fv1%2Fcontent%2F594342v3_fig1.tif/full/max/0/default.jpg" alt=""></figure><p>The modulation of blood-feeding provides a useful frame-work for understanding how internal and external cues shape meal-choice. What underlying molecular cues represent internal physiological states to influence dietary choice is of interest not only because of its importance in public health, but also because such understanding has implications for other organisms. In <em>Aedes albopictus</em> anticipatory expression of the yolk protein precursor gene <em>vitellogenin 2</em> in the fat body promotes host-seeking behaviour (<a class="linktoref" href="#c51">51</a>). In <em>Aedes aegypti</em>, neuropeptide-Y receptor (NPYLR7), and more recently, short Neuropeptide F (sNPF) and neuropeptide RYamide (RYa) have been shown to suppresses hostattraction, while Neuropeptide F (NPF) promotes it (<a class="linktoref" href="#c47">47</a>, <a class="linktoref" href="#c52">52</a>– <a class="linktoref" href="#c55">55</a>). Factors from the host can also both promote (ATP) (<a class="linktoref" href="#c56">56</a>) and terminate feeding (Fibrinopeptide A)(<a class="linktoref" href="#c57">57</a>).</p><p>Here we describe the blood- and sugar-feeding patterns of <em>Anopheles stephensi</em>, which is endemic to the Indian sub-continent but has expanded its range into West Africa, putting millions more people at risk of malaria (<a class="linktoref" href="#c58">58</a>). Neurotranscriptomics across the different conditions of blood-deprivation and -satiety allowed us to shortlist several candidate molecules that might promote blood-feeding behaviour. Through dsRNA mediated knockdown of nine of them, we identified two neuropeptide genes – <em>sNPF</em> and <em>RYa</em> - tht together promote blood-feeding through its action in the brain and possibly the abdomen. We find that <em>sNPF</em> expression pattern reflects this requirement: a cluster of cells in the SEZ expresses the <em>sNPF</em> transcripts only in blood-hungry females and such females also have more <em>sNPF</em> transcripts in their midguts. <em>RYa</em> is not modulated similarly. Both their receptors are expressed in the brain and the <em>sNPF receptor</em> is also expressed in the midgut. Together, these data are consistent with a model where increase in levels of <em>sNPF</em> in the brain and the midgut promotes a state of blood-hunger in <em>An. stephensi</em>.</p><section id="s1a"><h2 class="heading-2">Blood-, but not sugar-feeding, is modulated across the reproductive cycle</h2><p>To assess the female <em>An. stephensi</em> appetite for blood through the reproductive cycle we designed a behavioural assay (<span class="fakelink linktofig"><span class="fakelink-text">figure 1B</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">S1A</span><a class="fake-icon" href="#figS1"><span class="fakelink-icon">     </span></a></span>; see methods for details), and tested it on <em>Ae. aegypti</em>, a well-studied model. We were able to reproduce the previously reported results that mating enhances blood-feeding in <em>Aedes</em> (<a class="linktoref" href="#c59">59</a>–<a class="linktoref" href="#c61">61</a>) and that once fed to repletion, females actively suppress blood-feeding until oviposition (<a class="linktoref" href="#c62">62</a>) (<span class="fakelink linktofig"><span class="fakelink-text">figure 1C</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>).</p><p>We used this assay on <em>An. stephensi</em>, and employed Generalised Linear Mixed Model (GLMM) to compare appetites across conditions (see supplementary sheet1). We found that these females were uninterested in blood when they emerge but that their appetite increased with age (magenta, D0-D5 in <span class="fakelink linktofig"><span class="fakelink-text">figure 1D</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>, supplementary sheet1: model 1). Once fed to repletion, like <em>Ae. aegypti, An. stephensi</em> also suppress blood-feeding for at least four days after a blood-meal (magenta, D1<sup>B</sup>-D4<sup>B</sup> in <span class="fakelink linktofig"><span class="fakelink-text">figure 1D</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>) and until oviposition, after which, the suppression was lifted to pre-blood-meal levels (magenta, D1<sup>○</sup>-D2<sup>○</sup> in <span class="fakelink linktofig"><span class="fakelink-text">figure 1D</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>). Unlike blood-feeding, sugar-feeding was not modulated similarly barring a brief suppression post-blood-meal, which was not dependent on oviposition (<span class="fakelink linktofig"><span class="fakelink-text">figure 1E</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>; methods and <span class="fakelink linktofig"><span class="fakelink-text">S1G</span><a class="fake-icon" href="#figS1"><span class="fakelink-icon">     </span></a></span> for details, supplementary sheet1: model 4, 5, 6).</p><p>To ascertain whether sugar and blood represent parallel appetites independent of each other, we presented sugar-sated co-housed females a choice between the two. Under these conditions, 70% of the females chose to feed on blood alone and only 21% took both meals (<span class="fakelink linktofig"><span class="fakelink-text">figure 1F</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>). When co-housed females were deprived of sugars for a day prior to testing, the pattern revered: 68% of the females now took both sugar and blood meals and only 24% fed on blood alone (<span class="fakelink linktofig"><span class="fakelink-text">figure 1F</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>). As expected, males fed robustly on sugar in both conditions (grey, <span class="fakelink linktofig"><span class="fakelink-text">figure 1F</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>).</p><p>Since females took blood-meals regardless of their prior sugar-feeding status and only sugar-feeding was selectively suppressed by prior sugar access, we infer that these two represent parallel appetites regulated independently. The fact that most females in the starved group took both sugar- and blood-meals, while most females in the sated condition ignored sugar, supports this interpretation. An alternate interpretation of these data are that two represent hierarchical appetites: that some sugar feeding is necessary to stimulate blood-feeding. Determining the order of feeding events in the choice assay would be necessary to comment on this. However, prior reports that females can survive on blood-meals alone (with no access to plant sugars (<a class="linktoref" href="#c36">36</a>)) argue against this model. Together, these data suggest that in newly-emerged females, blood and sugar represent parallel appetites. We note that their interdependence may emerge later in the reproductive cycle (D1<sup>B</sup>-D2<sup>B</sup> in <span class="fakelink linktofig"><span class="fakelink-text">figure 1E</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>).</p><p>As mating promotes the first blood-meal in <em>Ae. aegypti</em>, we tested if this was true for <em>An. stephensi</em>. A post-hoc dissection of each female’s spermatheca (where sperm is stored after mating) revealed that many of the blood-fed females in the above assay were unmated (<span class="fakelink linktofig"><span class="fakelink-text">figure 1G</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>). To confirm this, we assayed blood-feeding behaviour in virgin females and found that virgin <em>An. stephensi</em>, unlike <em>Ae. aegypti</em>, have a robust appetite for blood and that mating enhances it only marginally (cyan, D0-D5, <span class="fakelink linktofig"><span class="fakelink-text">figure 1D</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>, supplementary sheet1: model 1, 2). Interestingly, however, these virgins, unlike mated <em>An. stephensi</em> (or <em>Ae. aegypti)</em> females, failed to suppress their blood-appetite, even after engorging on blood. Instead, they continued to blood-feed for up to four days after their first blood-meal (cyan, D1<sup>B</sup> - D4<sup>B</sup> in <span class="fakelink linktofig"><span class="fakelink-text">figure 1D</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>, supplementary sheet1: model 3). A flipped order of the two behaviours - virgins were first given a blood-meal then allowed to mate – also resulted in suppression of the appetite (<span class="fakelink linktofig"><span class="fakelink-text">figure 1H</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>). This suggests that the post-blood-meal suppression of blood-appetite in <em>An. stephensi</em>, is dependent upon mating status.</p><p>Together, these data describe female <em>An. stephensi’s</em> feeding behaviours: 1) in newlyemerged females blood and sugar represent parallel appetites, 2) blood-, but not sugar-appetite (barring a brief effect), is suppressed between a blood-meal and oviposition, and 3) mating is not necessary for blood-feeding in newly-emerged females, but it is necessary for its suppression between a blood-meal and oviposition.</p></section><section id="s1b"><h2 class="heading-2">Host-seeking is modulated across the reproductive cycle</h2><p>The modulation of blood-feeding through the reproductive cycle could be due to 1) modulation in peripheral sensitivity to host kairomones, 2) modulation by central brain circuits, or 3) modulation in both peripheral and central circuit elements. Extensive data in other mosquito species suggest that sensitivity to host kairomones change as females age (<a class="linktoref" href="#c42">42</a>, <a class="linktoref" href="#c63">63</a>–<a class="linktoref" href="#c66">66</a>) and following a blood-meal (<a class="linktoref" href="#c49">49</a>, <a class="linktoref" href="#c67">67</a>, <a class="linktoref" href="#c68">68</a>). We tested this in <em>An. stephensi</em>.</p><p>Because of the multi-sensory nature of host-seeking, we used a two-choice behavioural assay where multiple hostcues were provided in one arm while the other had none. Mosquitoes had to make a choice between the two arms (<span class="fakelink linktofig"><span class="fakelink-text">figure 2A</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span>). When no host-cues are provided in either (blank) even blood-hungry females distribute equally between the two arms demonstrating the assay has no bias (black, <span class="fakelink linktofig"><span class="fakelink-text">figure 2B</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span>). Using this assay, we found that newly emerged females showed no preference for either the control or the host arm (<span class="fakelink linktofig"><span class="fakelink-text">figure 2B</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span>). Five days later, they were strongly attracted to the host-cues and this attraction was suppressed after their blood-meal and until oviposition (magenta, <span class="fakelink linktofig"><span class="fakelink-text">figure 2B</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span>). This general trend in attraction to host-cues tracks the trend in blood-feeding. Interestingly, the response of virgin females to host-cues in the Y-maze was highly variable under all conditions tested (cyan, <span class="fakelink linktofig"><span class="fakelink-text">figure 2B</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span>). Yet, like in the blood-feeding assay, blood-fed virgins were comparatively more attracted to human host-cues than were the mated blood-fed females (<span class="fakelink linktofig"><span class="fakelink-text">figure 2B</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span>).</p><figure class="figure tofill" id="fig2"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Fig. 2.</label></span>Host-seeking is modulated like blood-feeding is in <em>An. stephensi</em>.</h3><p>A. Schematic of the Y-maze used to assess host-seeking behaviour of female <em>An. stephensi</em>. Females were allowed to acclimatise in the chamber for 5 minutes while being exposed to host kairomones that were presented in a test arm. A fan sucked the air from both host and control arms at 0.3-0.6m/s. Mosquitoes were released after acclimatisation and allowed to choose between the two arms. B. Percent females attracted to either host-cues or control arm at the indicated time points in the Y-maze. 17-21 females/trial, n=10 trials/group. Unpaired t-test; n.s.: not significant, p&gt;0.05.; * p&lt;0.05; ****p&lt;0.0001. (D0: day of emergence; D5M: 5days post-emergence, mated; D5: 5days post-emergence, virgin; D3<sup>B</sup>M: 3days post-blood-meal, mated; D3<sup>B</sup>: 3days post-blood-meal, virgin; D2<sup>○</sup>: 2days post-oviposition)</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://iiif.elifesciences.org/epp/108625%2Fv1%2Fcontent%2F594342v3_fig2.tif/full/max/0/default.jpg" alt=""></figure><p>These data suggest that, like blood-feeding, host-seeking is also modulated through the female’s reproductive cycle, and mating influences this modulation by enhancing it initially and suppressing it post-blood-meal. The variability in response of virgin females suggests that mating enhances the female’s initial attraction to a host.</p></section><section id="s1c"><h2 class="heading-2">Neurotranscriptomics reveals several candidate genes that promote blood-feeding</h2><p>While sensitivity to host kairomones may modulate blood-feeding, it is also possible that central brain circuitry contributes to it. We reasoned that cues from mating, or signals from other tissues such as midgut could act on brain circuitry to modulate blood-feeding. To determine the molecular underpinnings of this behavioural modulation, we performed RNA sequencing from the central brains of mosquitoes at different stages of their reproductive cycle (consequently different states of blood-deprivation or -satiety). This included females at emergence (D0), 5-days after emergence (D5; sugar-fed, blood-deprived), virgins and mated females one day after a blood-meal (D1<sup>B</sup>, D1<sup>B</sup>M), and females one day after oviposition (D1<sup>○</sup>). Age-matched males were included for the earliest two time points (<span class="fakelink linktofig"><span class="fakelink-text">figure 3A</span><a class="fake-icon" href="#fig3"><span class="fakelink-icon">     </span></a></span>).</p><figure class="figure tofill" id="fig3"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Fig. 3.</label></span>Neurotranscriptome of <em>An. stephensi</em></h3><p>A. Central brains were sampled from males and females at different ages and feeding preferences for bulk RNA-seq. Uninterested in blood: D0 males, D0 females, D5 males. Blood-hungry: D5, sugar-fed virgin females, D1<sup>B</sup> virgin females, and D1<sup>○</sup> females. Blood-sated: D1<sup>B</sup> mated females. n=100 central brains per replicate; 3 biological replicates per sample, 7 samples. B. Principal Component Analysis (PCA) of the normalised counts-per-million from different brain RNA-seq samples. Replicates cluster closely, and age and sex contribute most to the differences between the samples. C. Genes (listed on the right) with known expression patterns plotted across different samples. These known expression patterns are reproduced in our data. Average of the triplicate data are represented as z-score normalised log10 (TPM+1) values. Grey cells represent no data. D. Identification of candidates potentially involved in promoting blood-feeding behaviour. Two types of comparisons were made: (left) genes commonly up-regulated in the three blood-hungry conditions (see <span class="fakelink linktofig"><span class="fakelink-text">figure 3A</span><a class="fake-icon" href="#fig3"><span class="fakelink-icon">     </span></a></span>) when compared against D0 females and excluded in males, and (right) genes up-regulated in D5 sugar-fed virgin females when compared against D1<sup>B</sup> mated females (blood-sated), and excluded in males. Number of candidate genes identified from two different sets of analyses are boxed in red. E. Nine candidate genes (listed on the right), shortlisted for further validation. RNA-seq data from central brain samples in triplicates are represented as z-score normalised log10 (TPM+1) values. (D0: day of emergence; D5: 5days post-emergence, virgin; D1<sup>B</sup>: 1day post-blood-meal, virgin; D1<sup>B</sup>M: 1day post-blood-meal, mated; D1<sup>○</sup>: 1day post-oviposition)</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://iiif.elifesciences.org/epp/108625%2Fv1%2Fcontent%2F594342v3_fig3.tif/full/max/0/default.jpg" alt=""></figure><p>For each of our samples, 81-87% of about 20-30million reads aligned to the <em>An. stephensi</em> genome (<a class="linktoref" href="#c69">69</a>). The variance between our replicates was low as represented in the Principal Component Analysis (PCA) (<span class="fakelink linktofig"><span class="fakelink-text">figure 3B</span><a class="fake-icon" href="#fig3"><span class="fakelink-icon">     </span></a></span>). In PCA, older female brains, regardless of mating-, blood-feeding-, or oviposition-status clustered together while the youngest female and male brains stood apart on PC1 and PC2 suggesting that age and sex contribute most significantly to the variance in our data (<span class="fakelink linktofig"><span class="fakelink-text">figure 3B</span><a class="fake-icon" href="#fig3"><span class="fakelink-icon">     </span></a></span>). The expression pattern of some hallmark genes showed the expected trends. Genes involved in eclosion behaviours – <em>eclosion hormone</em> (<a class="linktoref" href="#c70">70</a>) and <em>partner of bursicon</em> (<a class="linktoref" href="#c71">71</a>) – showed the highest expression in the D0 samples and were not expressed later. Male-specific genes, <em>HMG-B-13-like</em>, were expressed only in males (<a class="linktoref" href="#c72">72</a>), and genes involved in vitellogenic stages of egg-development – <em>vitellogenins, Ecdysone receptor</em>, and <em>E-20-monooxygenase</em> (<a class="linktoref" href="#c73">73</a>)– were expressed only in post-blood-meal samples (<span class="fakelink linktofig"><span class="fakelink-text">figure 3C</span><a class="fake-icon" href="#fig3"><span class="fakelink-icon">     </span></a></span>).</p><p>We next explored this data for insights into the molecular underpinnings of blood-feeding modulation. Reports suggest that anticipatory metabolic changes in female germline cells and a ‘vittelogenin wave’ in fat bodies can drive specific appetite in <em>Drosophila</em> (<a class="linktoref" href="#c74">74</a>) and <em>Ae. albopictus</em> (<a class="linktoref" href="#c51">51</a>) respectively. However, we did not find any evidence in the neurotranscriptomes for a similar anticipatory metabolic change that might drive blood-feeding in the mosquito brain (<span class="fakelink linktofig"><span class="fakelink-text">figure S2B</span><a class="fake-icon" href="#figS2"><span class="fakelink-icon">     </span></a></span>). Interestingly, we found that after a blood-meal, glucose is neither spent nor stored, and that the female brain goes into a state of metabolic ‘sugar rest’, while actively processing proteins (<span class="fakelink linktofig"><span class="fakelink-text">figure S2B</span><a class="fake-icon" href="#figS2"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">S3</span><a class="fake-icon" href="#figS3"><span class="fakelink-icon">     </span></a></span>). We similarly plotted genes involved in lipid metabolism, neurotransmitters, neuropeptides and their receptors and found no significant patterns (<span class="fakelink linktofig"><span class="fakelink-text">figure S4</span><a class="fake-icon" href="#figS4"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">S5</span><a class="fake-icon" href="#figS5"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">S6</span><a class="fake-icon" href="#figS6"><span class="fakelink-icon">     </span></a></span>). In summary, contrary to our expectations, we saw no evidence in the neurotranscriptomes for anticipatory metabolic changes in the brain that might in turn drive blood-feeding behaviour. It is possible, however, that other forms of regulation that do not involve transcriptional changes could still influence metabolism.</p><p>In an alternate approach to identifying genes that might modulate blood-feeding, we took the intersection of genes that were enriched in the three blood-hungry conditions (and absent in males). These comparisons were made against the two conditions that were not motivated to blood-feed: females at emergence and mated-blood-fed females (<span class="fakelink linktofig"><span class="fakelink-text">figure3D</span><a class="fake-icon" href="#fig3"><span class="fakelink-icon">     </span></a></span>). These shortlisted 38 and 34 genes respectively, gave us 68 unique candidates (excluding common genes and isoforms) that might promote blood-feeding in <em>An. stephensi</em> (<span class="fakelink linktofig"><span class="fakelink-text">figure S7</span><a class="fake-icon" href="#figS7"><span class="fakelink-icon">     </span></a></span>). Notably, we did not find any genes to be differentially expressed between the two blood-fed conditions (virgins and mated), despite their contrasting behaviours (figure S2A). The strongest signatures we observed in these females were those related to metabolism (<span class="fakelink linktofig"><span class="fakelink-text">figure S2B</span><a class="fake-icon" href="#figS2"><span class="fakelink-icon">     </span></a></span>). It is possible that the differences lie in small populations of cells, which get obscured in the averaged transcriptomes of the entire brain.</p></section><section id="s1d"><h2 class="heading-2">Neuropeptides <em>sNPF</em> and RYamide promote blood-feeding in <em>An. stephensi</em></h2><p>We employed multiple, independent criteria while further short-listing candidates from the above analysis: 1) genes that showed the expected expression pattern: low in newly-emerged males and females; high in blood-hungry females; low in blood-sated females, 2) all neuropeptides, since these are known to regulate feeding behaviours, and 3) all genes related to juvenile hormone (JH) and 20-hydroxyecdysone (20E) signalling pathway, since they regulate development and reproduction in mosquitoes. This gave us 9 candidates (<span class="fakelink linktofig"><span class="fakelink-text">figure 3E</span><a class="fake-icon" href="#fig3"><span class="fakelink-icon">     </span></a></span>): Two sodium-dependent amino acid transporters <em>118504075 and 118504077</em>. Transporters can sense nutrients and regulate feeding in <em>Drosophila</em> (<a class="linktoref" href="#c29">29</a>, <a class="linktoref" href="#c75">75</a>). The neuropeptides included <em>Orcokinin (Ork), Prothoracicostatic peptides (Prp), RYamide (RYa)</em> and <em>short Neuropeptide F (sNPF)</em>, and a metabolic hormone <em>Eiger</em>, which responds to low nutrient conditions in <em>Drosophila</em> (<a class="linktoref" href="#c76">76</a>). <em>Protein Hairy</em> and <em>Juvenile hormone epoxide hydrolase 1-like (JHEH)</em> are involved in the JH pathway (<a class="linktoref" href="#c77">77</a>, <a class="linktoref" href="#c78">78</a>)(<span class="fakelink linktofig"><span class="fakelink-text">figure 3E</span><a class="fake-icon" href="#fig3"><span class="fakelink-icon">     </span></a></span>). Using dsRNA, we knocked each of these genes down and tested their blood-feeding behaviour. In each case, we ensured efficiency of gene knockdown post-hoc (<span class="fakelink linktofig"><span class="fakelink-text">figure 4A</span><a class="fake-icon" href="#fig4"><span class="fakelink-icon">     </span></a></span>). We first confirmed that the injection itself did not cause any behavioural changes (<span class="fakelink linktofig"><span class="fakelink-text">figure 4B</span><a class="fake-icon" href="#fig4"><span class="fakelink-icon">     </span></a></span>) and in all following experiments either <em>dsGFP</em>-injected or uninjected females were included as controls.</p><figure class="figure tofill" id="fig4"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Fig. 4.</label></span><em>short Neuropeptide F (sNPF)</em> and <em>RYamide (RYa)</em> together promote blood-feeding in <em>An. stephensi</em></h3><p>A. Schematic representation of the pipeline used for functional validation. dsRNAs against candidate genes were injected in adult virgins for targeted RNAi. After recovery these were tested for blood-feeding. Knockdown efficiency was determined via qPCR in whole heads and abdomens of fed and/or unfed females after the behaviour. B. Comparison of uninjected females and <em>dsGFP</em>-injected females for blood-feeding behaviour revealed no difference between the two, indicating that the animals recovered from the injection procedure. 19-25 females/ replicate, n= 8-9 replicates/group. Unpaired t-test; n.s.: not significant, p&gt;0.05. C,D. Blood-feeding and sugar-feeding behaviour of females when both <em>RYa</em> and <em>sNPF</em> are knocked down in the head (C). Relative mRNA expressions of <em>RYa</em> and <em>sNPF</em> in the heads of <em>dsRYa+dssNPF</em> - injected blood-fed and unfed females, as compared to that in uninjected females, analysed via qPCR (D). Simultaneous knockdown of both neuropeptides suppressed the blood-feeding behaviour of females, without affecting the sugar-feeding behaviour. Note the more efficient knockdown in the heads of unfed females, as compared to blood-fed females. 14-25 females/replicate, n=8-9 replicates/group. Unpaired t-test; n.s.: not significant, p&gt;0.05; ****p&lt;0.0001 (C). One-way ANOVA with Holm-Šídák multiple comparisons test, p&lt;0.05 (D). Data labelled with different letters are significantly different. E,F,G: Blood-feeding and sugar-feeding behaviour of females when both <em>RYa</em> and <em>sNPF</em> are knocked down only in the abdomen (E). Relative mRNA expressions of <em>RYa</em> and <em>sNPF</em> in the heads (F) and abdomens (G) of <em>dsRYa+dssNPF</em> - injected fed females, as compared to that in dsGFP-injected females, analysed via qPCR. No change in the blood-feeding or sugar-feeding behaviour was observed, despite the strong knockdown of both the genes in the abdomens, while the levels in the heads remain unaltered. 17-21 females/replicate, n=3 replicates/group.Unpaired t-test; n.s.: not significant, p&gt;0.05. (E). One-way ANOVA with Holm-Šídák multiple comparisons test, p&lt;0.05 (F,G). Data labelled with different letters are significantly different. H, I. Whole mount mRNA <em>in situ</em> hybridisation of central brain of 5days old sugar-fed virgin female, with <em>RYa</em> and <em>sNPF</em> probes (H). Both neuropeptides are expressed in multiple clusters (cyan: <em>RYa</em>, magenta: <em>sNPF</em>, blue: nc82). Co-expression was observed only in the Kenyon cells of the mushroom body (I). Maximum intensity projections of the confocal stacks are shown. Scale bar, 50µm. J. <em>sNPF</em> expression in the SEZ region of the central brains of females uninterested in blood (D0), blood-hungry females (D5) and blood-sated females (D2<sup>B</sup>M). Note the appearance of a novel cluster only in the blood-hungry (D5) condition. Maximum intensity projections of selected confocal stacks are shown. Scale bar, 50µm. K. Whole mount mRNA <em>in situ</em> hybridisation of gut of 5days old sugar-fed virgin female with <em>sNPF</em> probes. <em>sNPF</em> expression was observed in the enteroendocrine cells (EECs) in the posterior midgut, shown in higher magnification in K’. Scale bar, 50µm. L,M. <em>sNPF</em> expression in the posterior midguts of females uninterested in blood (D0), blood-hungry females (D5) and blood-sated females (D2<sup>B</sup>M). In addition to the EECs in the posterior midgut, two cells in the anterior midgut also show strong <em>sNPF</em> expression (L, inset). Higher number of cells express <em>sNPF</em> in the blood-hungry condition (M). n=5-7 guts per condition. One-way ANOVA with Holm-Šídák multiple comparisons test, p&lt;0.05. N. Whole mount mRNA <em>in situ</em> hybridisation of central brain (left) and midgut (right) of 5days old sugar-fed virgin female, with <em>sNPF receptor (sNPFR)</em> and <em>RYa receptor(RYaR)</em> probes. While both receptors are expressed in the brain, only <em>sNPFR</em> is expressed in the gut. Scale bar, 50µm. O. Summary schematic of <em>sNPF</em> and <em>RYa</em> function in modulating blood-appetite in <em>An. stephensi. RYa</em> and <em>RYaR</em> (cyan) are expressed only in the brain, while <em>sNPF</em> and <em>sNPFR</em> (magenta) are expressed both in the brain and the gut. Increased <em>sNPF</em> levels in both the tissues (dashed circles) promote a state of blood-hunger, which may drive feeding behaviour either by <em>sNPF’s</em> action in the two tissues independently or via a communication between them. This happens in the context of <em>RYa</em> signalling in the brain. (D0: day of emergence; D5: 5days post-emergence, virgin; D2<sup>B</sup>M: 2days post-blood-meal, mated)</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://iiif.elifesciences.org/epp/108625%2Fv1%2Fcontent%2F594342v3_fig4.tif/full/max/0/default.jpg" alt=""></figure><p>We saw no change in behaviour upon loss of <em>Eiger, JHEH</em>, and <em>Ork</em> function (<span class="fakelink linktofig"><span class="fakelink-text">figure S8A-C</span><a class="fake-icon" href="#figS8"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">S8A’-C’</span><a class="fake-icon" href="#figS8"><span class="fakelink-icon">     </span></a></span>). We were able to downregulate <em>Hairy</em> and <em>Prp</em> transcript levels by only 3035%, hence cannot comment on their role in modulating blood-feeding (<span class="fakelink linktofig"><span class="fakelink-text">figure S8D</span><a class="fake-icon" href="#figS8"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">S8E</span><a class="fake-icon" href="#figS8"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">S8D’</span><a class="fake-icon" href="#figS8"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">S8E’</span><a class="fake-icon" href="#figS8"><span class="fakelink-icon">     </span></a></span>). Downregulation of the two transporters was efficient and resulted in a marginal reduction in blood-feeding (<span class="fakelink linktofig"><span class="fakelink-text">figure S8F</span><a class="fake-icon" href="#figS8"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">S8G</span><a class="fake-icon" href="#figS8"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">S8F’</span><a class="fake-icon" href="#figS8"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">S8G’</span><a class="fake-icon" href="#figS8"><span class="fakelink-icon">     </span></a></span>). However, their combined knockdowns, though effective, did not show any significant change in behaviour (<span class="fakelink linktofig"><span class="fakelink-text">figure S8H</span><a class="fake-icon" href="#figS8"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">S8H’</span><a class="fake-icon" href="#figS8"><span class="fakelink-icon">     </span></a></span>).</p><p>Two neuropeptides - <em>sNPF</em> and <em>RYa</em> - showed about 25% and 40% reduced mRNA levels in the heads and these animals showed no change in blood-feeding (<span class="fakelink linktofig"><span class="fakelink-text">figure S9B, S9C, S9E, S9F</span><a class="fake-icon" href="#figS9"><span class="fakelink-icon">     </span></a></span>). We noticed, however, that many of these animals did not engorge and instead took smaller blood-meals (<span class="fakelink linktofig"><span class="fakelink-text">figure S9D, S9G</span><a class="fake-icon" href="#figS9"><span class="fakelink-icon">     </span></a></span>). Given this, and since neuropeptides are known to act in concert (<a class="linktoref" href="#c79">79</a>), we tested the possibility that <em>sNPF</em> and <em>RYa</em> might act together to regulate blood-feeding behaviour. Indeed, their simultaneous dsRNA injections resulted in about 40% of the animals not taking blood-meals (<span class="fakelink linktofig"><span class="fakelink-text">figure 4C</span><a class="fake-icon" href="#fig4"><span class="fakelink-icon">     </span></a></span>). Since RNA interference (RNAi) can be variable, and each female is an independent experiment, we pooled the females that took blood-meals and those that didn’t separately before quantifying mRNA from them. We noted that knockdown efficiency was higher in the heads of females that did not take blood-meals suggesting that downregulating these genes caused the suppression of blood-feeding (<span class="fakelink linktofig"><span class="fakelink-text">figure 4D</span><a class="fake-icon" href="#fig4"><span class="fakelink-icon">     </span></a></span>). Importantly, since these females were offered both blood and sugar, we noted that their sugar-feeding was unaffected (<span class="fakelink linktofig"><span class="fakelink-text">figure 4C</span><a class="fake-icon" href="#fig4"><span class="fakelink-icon">     </span></a></span>).</p><p>Since RNAi results in systemic knockdowns and neuropep-tides are known to influence behaviour from their action in non-neuronal tissues such as the gut (<a class="linktoref" href="#c55">55</a>, <a class="linktoref" href="#c79">79</a>, <a class="linktoref" href="#c80">80</a>), we modified our dsRNA injection protocol to knock <em>sNPF</em> and <em>RYa</em> down in the abdomen while leaving their levels in the head unaltered (<span class="fakelink linktofig"><span class="fakelink-text">figure 4F</span><a class="fake-icon" href="#fig4"><span class="fakelink-icon">     </span></a></span>, 4G, S9J, S9L) (see methods for details). Abdomen-specific knockdown had no effect on blood-feeding (<span class="fakelink linktofig"><span class="fakelink-text">figure 4E</span><a class="fake-icon" href="#fig4"><span class="fakelink-icon">     </span></a></span>, S9I, S9K), indicating that abdominal <em>sNPF</em> and <em>RYa</em> alone cannot influence blood-feeding and their action in the brain is necessary. In contrast, since every head knockdown of these neuropeptides also abolished their abdominal expression (figure S9C, S9F), we cannot rule out the possibility that both brain-and gutderived neuropeptides modulate blood-feeding. Indeed, since such neuropeptide mediated gut-brain control of feeding is commonly seen (<a class="linktoref" href="#c79">79</a>), we propose a model, consistent with our data, where brain-derived <em>sNPF/RYamide</em> is essential to trigger blood-feeding, while gut-derived peptide acts as a modulatory feedback signal. This remains to be tested with cell-specific tools that are currently not available for this species.</p><p>To determine in which cells <em>sNPF</em> and <em>RYa</em> act, we used hybridisation chain reaction (HCR) (<a class="linktoref" href="#c81">81</a>, <a class="linktoref" href="#c82">82</a>) <em>in situ</em>. We found that their transcripts are expressed in several non-overlapping neuronal clusters distributed across many regions in the brain (<span class="fakelink linktofig"><span class="fakelink-text">figure 4H</span><a class="fake-icon" href="#fig4"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">S10A</span><a class="fake-icon" href="#figS10"><span class="fakelink-icon">     </span></a></span>) and only in the Kenyon cells of the mushroom body were they co-expressed (<span class="fakelink linktofig"><span class="fakelink-text">figure 4I</span><a class="fake-icon" href="#fig4"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">S10B</span><a class="fake-icon" href="#figS10"><span class="fakelink-icon">     </span></a></span>). This is reminiscent of their expression in <em>Drosophila</em> and <em>Bombyx</em> (<a class="linktoref" href="#c83">83</a>–<a class="linktoref" href="#c85">85</a>). Importantly, we found a cluster of cells in the SEZ that expresses the <em>sNPF</em> transcript only in the blood hungry-state (<span class="fakelink linktofig"><span class="fakelink-text">figure 4J</span><a class="fake-icon" href="#fig4"><span class="fakelink-icon">     </span></a></span>, supplementary movie). Its transcripts were either absent or diminished from this cluster in newly-emerged or sated females. Since the RNAi experiments suggested the abdominal involvement of these neuropeptides, we also looked for their expression in the midgut where we could detect only <em>sNPF’s</em> transcripts, not <em>RYa’s</em> (<span class="fakelink linktofig"><span class="fakelink-text">figure 4K,K</span><a class="fake-icon" href="#fig4"><span class="fakelink-icon">     </span></a></span>’, <span class="fakelink linktofig"><span class="fakelink-text">S11A, S11B, S11E</span><a class="fake-icon" href="#figS11"><span class="fakelink-icon">     </span></a></span>). Interestingly, <em>sNPF’s</em> transcript expression was also modulated in the midgut enteroendocrine cells (EECs) - hormone-producing cells of the gut. In these cells, <em>sNPF</em> transcripts were specifically increased in blood-hungry females when compared to newly-emerged or sated females (<span class="fakelink linktofig"><span class="fakelink-text">figure 4L,M</span><a class="fake-icon" href="#fig4"><span class="fakelink-icon">     </span></a></span>).</p><p>Neuropeptides work by acting on their receptors to trigger downstream signalling cascades in the cells that express the receptors. We identified putative receptors for <em>sNPF</em> and <em>RYa</em> (see methods) and found that both were expressed in several overlapping cell-clusters in the brain (<span class="fakelink linktofig"><span class="fakelink-text">figure 4N</span><a class="fake-icon" href="#fig4"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">S10C, S10D</span><a class="fake-icon" href="#figS10"><span class="fakelink-icon">     </span></a></span>). In the midgut, however, we could only detect <em>sNPFR</em> and not <em>RYaR</em> (<span class="fakelink linktofig"><span class="fakelink-text">figure 4N</span><a class="fake-icon" href="#fig4"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">S11C, S11D, S11F</span><a class="fake-icon" href="#figS11"><span class="fakelink-icon">     </span></a></span>). Since these receptors were identified based on homology, their functional validation will be necessary to implicate them in blood-feeding behaviour. However, neuropeptide receptors are often ligand-promiscuous and confirming their specificity would require pharmacological characterisation beyond the scope of this study.</p><p>Together, these data are consistent with a model where increased <em>sNPF</em> expression in both the brain and midgut might drive the blood-feeding behaviour, but that this occurs in the context of <em>RYa’s</em> action in the brain (<span class="fakelink linktofig"><span class="fakelink-text">figure 4O</span><a class="fake-icon" href="#fig4"><span class="fakelink-icon">     </span></a></span>).</p></section></section><section id="s2"><h1 class="heading-1">Conclusions</h1><p>Here, we report the blood- and sugar-feeding behaviours of <em>An. stephensi</em>. While sharing key features with other mosquito species, <em>An. stephensi</em> exhibits distinct regulatory patterns. Unlike <em>Ae. aegypti</em>, it does not require mating to initiate blood-feeding, though the presence of males is necessary to sustain the enhanced state. In both species, blood-feeding is suppressed after a blood-meal and until oviposition, but this suppression is mating-dependent in <em>An. stephensi</em>. Since unmated <em>Ae. aegypti</em> females did not blood-feed in our assays, we cannot comment on whether this suppression also requires mating.</p><p>Mosquito feeding behaviour is highly context-dependent, shaped by physiological state, environment, and experimental design (<a class="linktoref" href="#c36">36</a>). Our comparisons between <em>Ae. aegypti</em> and <em>An. stephensi</em> were conducted under standardised conditions. While direct comparisons across studies must be made cautiously, patterns observed under particular conditions can offer meaningful insights. For example, virgins of other <em>Anopheles</em> species readily blood-feed (<a class="linktoref" href="#c36">36</a>, <a class="linktoref" href="#c39">39</a>–<a class="linktoref" href="#c41">41</a>), suggesting that this is typical of the genus, whereas <em>Ae. aegypti</em> virgins do so only under specific conditions and these responses vary among strains (<a class="linktoref" href="#c38">38</a>) (and references within). Whether blood-fed <em>Aedes</em> virgins suppress subsequent feeding is likely also context dependent with reports suggesting both that they do (<a class="linktoref" href="#c86">86</a>, <a class="linktoref" href="#c87">87</a>) and do not (<a class="linktoref" href="#c45">45</a>). Post-blood-meal suppression of host-seeking - well characterised and robust in mated <em>Aedes</em> - appears weaker in mated <em>Anopheles</em> and can be lifted prior to oviposition (<a class="linktoref" href="#c88">88</a>). Indeed, we too observed a stronger post-blood-meal suppression of feeding in <em>Ae. aegypti</em> than in <em>An. stephensi</em> (compare <span class="fakelink linktofig"><span class="fakelink-text">Fig. 1C</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span> to <span class="fakelink linktofig"><span class="fakelink-text">Fig. 1D</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>). Few studies have investigated this suppression in <em>Anopheline</em> virgins, but when they do, they report a weak suppression (<a class="linktoref" href="#c89">89</a>). Taken together, <em>An. stephensi</em> displays characteristics of both <em>Aedes</em> and <em>Anopheline</em> feeding patterns - its virgin feeding patterns align more closely with other Anophelines, yet, like mated <em>Ae. aegypti</em>, it still exhibits clear suppression of blood-feeding until oviposition.</p><p>What mechanisms might underlie such mating-induced modulation of blood-feeding? In <em>Drosophila melanogaster</em>, mating induces a suite of behavioural changes - including increased protein and sugar intake - that are mediated by male-derived sex-peptide that acts on central brain circuits in the female brain (<a class="linktoref" href="#c4">4</a>–<a class="linktoref" href="#c7">7</a>, <a class="linktoref" href="#c90">90</a>–<a class="linktoref" href="#c92">92</a>). These changes are anticipatory in nature rather than a homeostatic control of feeding behaviour to meet reproductive demands (<a class="linktoref" href="#c7">7</a>, <a class="linktoref" href="#c8">8</a>). In mosquitoes, the molecular mechanisms driving post-mating behavioural changes are not likely to be the same (<a class="linktoref" href="#c93">93</a>). Instead, mating-derived modulators such as HP1 in <em>Aedes</em> (<a class="linktoref" href="#c62">62</a>, <a class="linktoref" href="#c94">94</a>) and male-deposited steroid hormone (20-hydroxyecdysone) in <em>Anopheles</em> (<a class="linktoref" href="#c95">95</a>, <a class="linktoref" href="#c96">96</a>) have been shown to induce sexual refractoriness. Whether they also shape feeding decisions remains unknown.</p><p>We have identified two neuropeptides - <em>RYa</em> and <em>sNPF</em> - that act synergistically to promote blood-feeding in <em>An. stephensi</em>. Our data are consistent with a model where an increase in <em>sNPF</em> levels in the brain - likely released by a cluster of neurons in the SEZ - as well as in the midgut, promotes a state of blood-hunger. Whether this reflects gut-to-brain communication or parallel, tissue-local computations to modulate behaviours remains unclear, especially since we detected the transcript of both <em>sNPF</em> and its receptor in both tissues. Inter-organ signalling is well established in feeding regulation (<a class="linktoref" href="#c79">79</a>, <a class="linktoref" href="#c97">97</a>). For example, gut-derived NPF in flies can influence brain circuits to increase protein intake (<a class="linktoref" href="#c80">80</a>). More broadly, feeding behaviour is likely regulated by a distributed network of hormonal and neuronal cues across multiple organs. Signals from the ovary, fatbody, and other nutrient-sensing tissues may also contribute, acting through both central and peripheral circuits. Many of these inputs remain to be identified.</p><p>It is possible that <em>sNPF</em> promotes blood-feeding independent of <em>RYa</em>. However, we were unable to knock <em>sNPF</em> down completely to test this. Its synergistic requirement with <em>RYa</em>, despite no apparent modulation of <em>RYa</em> or its receptor, could result from a general state of hunger that <em>RYa</em> promotes, in the context of which, we were able to uncover the blood-feeding role of the diminished <em>sNPF</em>. Since many neurons co-express these receptors, it is possible that neurons downstream of <em>sNPF</em> or <em>RYa</em> expressing neurons integrate the two inputs to modulate behaviour.</p><p>We note that our findings differ from those reported in <em>Aedes</em>. In <em>Aedes sNPF</em> and <em>RYamide</em> act as satiety brakes—peptide or receptor activation curbs host-seeking (<a class="linktoref" href="#c52">52</a>, <a class="linktoref" href="#c53">53</a>, <a class="linktoref" href="#c55">55</a>) —whereas in <em>Anopheles</em>, our knockdowns show that the same neuropeptides act as a hunger signal, stimulating blood-feeding. While these differences are interesting, in the context of their evolutionary histories, and the known differences in how neuropeptides can act differently across species, they are not entirely unexpected. <em>Aedes</em> and <em>Anopheles</em> lineages diverged <em>≈</em>150-200 million years ago (<a class="linktoref" href="#c98">98</a>), and the two genera have since diverged in almost every facet of biology—diel activity, host preference, feeding and oviposition sites, egg-desiccation tolerance, genome size, and chemosensory repertoires among other traits. Neuropeptide pathways are known to drive opposing behaviours in different species: <em>sNPF</em> is known to be orexic in <em>D. melanogaster, A. mellifera, L. decemlineata, B. mori</em>, and <em>P. americana</em>, while it has anorexic effects in <em>S. gregaria</em> (<a class="linktoref" href="#c99">99</a>). Against this backdrop, it is unsurprising that the same neuropeptide pathway can drive opposite behaviours in <em>Aedes</em> and <em>Anopheles</em>. These comparative studies underscore a key lesson for vector control strategies in general. A strategy that is effective in one mosquito species may inadvertently boost vectorial capacity in another. Behavioural interventions, chemical modulators, and even gene-drives therefore require rigorous cross-species validation because results from one mosquito species cannot be assumed to hold true for another.</p></section><div class="pagebreak"></div><section id="s3"><h1 class="heading-1">Methods</h1><section id="s3a"><h2 class="heading-2">Mosquito culture</h2><p><em>Anopheles stephensi</em> (Indian strain; Bangalore strain, TIGS-1) and <em>Aedes aegypti</em> (Bangalore collection) were reared in BugDorm cages and maintained at 26-28°C, 70-80% relative humidity with a photo-period of 12h light: 12h dark. All the behavioural assays were performed at these conditions of temperature and humidity. Larvae were fed on a mixture of yeast and dog biscuit (1:3 ratio) and cultured in trays in appropriate densities to prevent over-crowding. Adult mosquitoes were provided <em>ad libitum</em> access to 10% sugar solution (8% sucrose +2% glucose +0.05% methyl paraben +5% vitamin syrup (Polybion SF Complete, Merck Ltd.)) at all times, unless specified otherwise. Females were provided O+ve human blood, secured periodically from blood banks, both for general maintenance and feeding assays.</p></section><section id="s3b"><h2 class="heading-2">Regulatory permissions</h2><p>This project was approved by the Institutional Human Ethics Committee (Ref: inStem/IEC-22/02) and the Institutional Biosafety Committee (TIGS/M-7/8/2020-1).</p></section><section id="s3c"><h2 class="heading-2">Blood-feeding assay: first blood-meal, 0-120h post-emergence (D0-D5)</h2><p>All <em>An. stephensi</em> feeding experiments were performed from ZT12-ZT15 and <em>Ae. aegypti</em> experiments from ZT10-ZT12 (Zeitgeber time 0 is defined as the time of lights ON).</p><p>0-7h old mosquitoes were cold anaesthetised and collected during the day in 1100ml paper cups covered with a net and secured with a rubber band. These were maintained at the conditions of temperature and humidity described above. To collect virgins, males were separated at the time of collection. 20 females with males (1:1 ratio) or without males were collected for each trial and were aged appropriately with <em>ad libitum</em> access to 10% sugar solution (supplementary figure S1A).</p><p>Females were assayed for their blood-feeding behaviour from the day of emergence (D0, 0-7h old) to 5 days of age (D5), every 24h. They were provided constant access to sugar, except when they were offered blood. A detailed schematic has been shown in figure S1A. On the day of the test, appropriately aged females were offered a blood-meal via a Hemotek feeder (Hemotek Ltd), which was maintained at 37°C and perfumed with human skin odours. It has been reported that female mosquitoes need to be ‘activated’ to take blood-meals. This involves the presentation of CO<sub>2</sub> (and other host-cues) to the females, likely because CO<sub>2</sub> has been shown to gate temperature and visual cues (<a class="linktoref" href="#c100">100</a>, <a class="linktoref" href="#c101">101</a>). So, prior to offering the blood-meal, the mosquitoes were ‘activated’ by the presentation of a hand placed above the cup for 5 minutes, followed by 3 exhalations from the experimenter. Additionally, blood was supplemented with 1mM ATP (Sigma A6419) before loading the feeder (<span class="fakelink linktofig"><span class="fakelink-text">figure 1B</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>). Females were allowed to feed for 60-90mins in dark and abdomens were scored visually for presence of blood, irrespective of the amount ingested (inset, figure S1A). For D0, no sugar was provided at the time of collection. Females were allowed to acclimatise for at least 30 mins in the cups and then offered a choice of both sugar and blood simultaneously, to assess their preference for blood soon after emergence (<span class="fakelink linktofig"><span class="fakelink-text">figure S1A</span><a class="fake-icon" href="#figS1"><span class="fakelink-icon">     </span></a></span>).</p><p>In the case of co-housed females, males and females were housed together from the time of collection until the day of the experiment. Males were separated post-hoc and spermathecae of the females were dissected to score for the presence or absence of sperm (<span class="fakelink linktofig"><span class="fakelink-text">figure 1G</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>). All the assays with virgin and co-housed females were performed in parallel by a single experimenter. Mosquitoes from at least 4 independent batches were tested for each day of the experimental timeline. For <em>Ae. aegypti</em>, both virgin and mated females (mating status was confirmed via post-hoc dissection of spermathecae and all co-housed females were found to be mated) were tested for first blood-meal at only 8days post-emergence (D8, <span class="fakelink linktofig"><span class="fakelink-text">figure 1C</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>). Behavioural experiments were performed the same way as described for <em>An. stephensi</em>.</p></section><section id="s3d"><h2 class="heading-2">Blood-feeding assay: second blood-meal, 24-96h post-first blood-meal (D1<sup>B</sup>-D4<sup>B</sup>)</h2><p>To ensure that the second blood-meal is not driven due to an incomplete first blood-meal, we selected only fully-engorged females to test for subsequent feedings (figure S1A). 5days old <em>An. stephensi</em> (both virgins and co-hosued with males) and 7-9days old <em>Ae. aegypti</em> (mated, as previously determined) were fed on blood and fully fed females were separated and collected in paper cups as described above. As a female takes about 2 days to digest blood, the second blood-meal ingested 24h post-first blood-meal (D1<sup>B</sup>) was detected by the presence of Rhodamine B (Sigma R6626), which fluoresces in the red range (<span class="fakelink linktofig"><span class="fakelink-text">figure S1A</span><a class="fake-icon" href="#figS1"><span class="fakelink-icon">     </span></a></span>). Spiking the second blood-meal with Rhodamine B (0.04<em>µ</em>g/<em>µ</em>L of blood) allowed visualisation of the freshly ingested blood, without affecting the feeding efficiency (<span class="fakelink linktofig"><span class="fakelink-text">supplementary figure S1B-D</span><a class="fake-icon" href="#figS1"><span class="fakelink-icon">     </span></a></span>). To assess feeding at subsequent days (D2<sup>B</sup>-D4<sup>B</sup>), abdomens were dissected to differentiate between the bright red freshly ingested blood from the dark and clotted first blood-meal. In the case of co-housed <em>An. stephensi</em> females, spermathecae were not dissected to confirm mating as females were presumed to be mated when collected for this assay (5days old and co-housed with males for the entire duration from the time of emergence), as previously determined in <span class="fakelink linktofig"><span class="fakelink-text">figure 1G</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>.</p><p>To confirm whether mating suppresses subsequent feeding (<span class="fakelink linktofig"><span class="fakelink-text">figure 1H</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>), fully fed virgins were collected and allowed to mate for 3-4 days with aged-match males. No males were introduced in the virgin-only controls. Blood-feeding assay was performed as described above. Spermathecae were dissected post-hoc to determine the mating status and unmated females were not considered for the final analysis. Datasets with less than 80% total mating were discarded.</p></section><section id="s3e"><h2 class="heading-2">Blood-feeding assay: post-oviposition, 24-48h post-egg laying (D1<sup>○</sup>-D2<sup>○</sup>)</h2><p>For blood-feeding behaviour post-oviposition, fully fed females were separated and a water cup lined with moist paper (ovipositor) was provided for egg laying 2days post-blood-meal (determined in a separate assay described below, supplementary figure S1E; figure S1A). Females were allowed to lay eggs overnight. The following morning, non-gravid females were visually identified by their lean bellies and separated in cups. Blood-feeding assay was performed the same evening for 1day post-oviposition (D1<sup>○</sup>) and the following day (D2<sup>○</sup>), as described above. Females were dissected post-hoc and scored for absence of mature ovaries, as a sign of successful oviposition. Gravid females were not considered for the analysis.</p></section><section id="s3f"><h2 class="heading-2">Determination of oviposition timing for <em>An. stephensi</em></h2><p>To determine the day of maximum oviposition, 5-6days old females co-housed with males were fed on blood, as described above. Fully fed females were separated and collected in BugDorm cages in groups of 10 females/cage. An ovipositor was provided the same day and females were allowed to lay eggs overnight. Eggs laid were counted the next day and a fresh ovipositor was provided. This cycle was repeated for the next 5 days. Total eggs laid each day were divided by the number of females present at the time of counting (supplementary figure S1E).</p></section><section id="s3g"><h2 class="heading-2">Sugar-feeding assay for <em>An. stephensi</em></h2><p>As mosquitoes feed on sugar in small bouts, as opposed to a replete blood-meal, we assessed sugar-feeding behaviour of females in windows of 24h for each test day. 0-7h old mosquitoes were collected during the day, separated in cups and aged appropriately. They were given access to 10% sugar solution at all times, as described above, except for the day of emergence (D0) when no sugar was provided at the time of collection (similar to the blood-feeding assay performed at D0).</p><p>To assess their sugar-feeding behaviour on the day of emergence (D0), 0-7h old females were acclimatised in the cups for 1h and sugar spiked with Rhodamine B (1mg/25ml of sugar; Sigma R6626) was presented for 2.5-3h. The fluorescence of Rhodamine B in the red channel allowed visualisation of even the tiniest amounts of sugar ingested by a female (<span class="fakelink linktofig"><span class="fakelink-text">figure S1G</span><a class="fake-icon" href="#figS1"><span class="fakelink-icon">     </span></a></span>). For D1 feeding, 3-10h old females (with access to normal sugar so far) were then provided with Rhodamine B-sugar and allowed to feed on it for 24h. For D2 feeding, 27-34h old females (with access to normal sugar so far) were fed on Rhodamine B-sugar for the next 24h and scored as D2. This cycle was repeated until D5. In the case of co-housed females, males were separated post-hoc and spermathecae of the females were dissected to score for the presence or absence of sperm (<span class="fakelink linktofig"><span class="fakelink-text">figure S1H</span><a class="fake-icon" href="#figS1"><span class="fakelink-icon">     </span></a></span>).</p><p>To assess sugar-feeding post-blood-meal, females fully fed on blood were collected and separated in cups, as described above. They were allowed to feed on Rhodamine B-sugar for 24h following a blood-meal and scored as D1<sup>B</sup>. The next set of females (with access to normal sugar so far) were then provided with Rhodamine B-sugar from 24h-48h post-bloodmeal and scored as D2<sup>B</sup>. This cycle was repeated until 4 days post-blood-meal (D4<sup>B</sup>, <span class="fakelink linktofig"><span class="fakelink-text">figure 1E</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>).</p><p>For sugar-feeding post-oviposition, blood-fed females were provided an ovipositor 3 days later and were forced to oviposit during daytime by artificially creating dark conditions. Oviposited females (non-gravid) were then visually identified, separated and presented with Rhodamine B-sugar for the following 24h and scored as D1<sup>○</sup>. The next set (with access to normal sugar so far) was then provided with Rhodamine B-sugar, 24-48h post-oviposition and scored as D2<sup>○</sup>(<span class="fakelink linktofig"><span class="fakelink-text">figure 1E</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>). For each day tested, females were dissected post-hoc and scored for absence of mature ovaries, as a sign of successful oviposition. Gravid females were not considered for the analysis.</p></section><section id="s3h"><h2 class="heading-2">Dual choice assay of blood and sugar</h2><p>Mosquitoes were collected as described above. Males and females were cohoused for 4-6days with <em>ad libitum</em> access to sugar. To assess the choice between sugar and blood, mosquitoes were either given continuous access to normal sugar (sugar-sated) or starved with water 24h prior to the assay (sugar-starved). At the time of the assay, post-activation (as described above), blood and Rhodamine B-spiked sugar (1mg/25ml of sugar, Sigma R6626) were provided in parallel. After 60mins, abdomens were scored visually for ingestion of blood under white light, and under fluorescent microscope (red channel) for ingestion of sugar (<span class="fakelink linktofig"><span class="fakelink-text">figure S1F</span><a class="fake-icon" href="#figS1"><span class="fakelink-icon">     </span></a></span>). Co-housed males served as an internal control for sugar-feeding. As males are unable to feed on blood, they were only scored for presence or absence of Rhodamine B-spiked sugar. Results are presented in <span class="fakelink linktofig"><span class="fakelink-text">figure 1F</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>.</p></section><section id="s3i"><h2 class="heading-2">Statistical analysis of feeding behaviours</h2><p>Statistical analysis was performed using R (version 4.4.2) and Graph-Pad Prism (v10.2.0). Blood-feeding and sugar-feeding behaviours of virgins and co-housed females in <span class="fakelink linktofig"><span class="fakelink-text">figure 1D, E</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>, were analysed using Generalized Linear Mixed-effects Models (GLMMs) with a binomial distribution and logit link function. The models were fitted using the “glmer” function from the “lme4” package in R. The probabilities of blood-or sugar-feeding, both pre-and post-blood-meal were modeled as a function of Day (age) and Group (virgin vs. cohoused with males), including their interaction term to test whether group differences varied across days. Replicate identities (BioID) and larval batches (only for first blood-meal in sugar-fed females) were included as random factors to account for replicate-level and batch-level variations respectively. Post-hoc analyses were conducted using estimated marginal means (emmeans package) for age-effects within groups (p-value adjusted for multiple comparisons using Tukey’s method) and pairwise comparison between groups at each day (p-value adjusted using Bonferroni method). To test whether mating success altered the sugar- and blood-feeding propensity of females co-housed with males, the probabilities of feeding were modeled only the co-housed data, as a function of Day (age) and Mating (yes or no), including their interaction term and accounting for variation due to replicate IDs (BioID). Statistical significance was set at α = 0.05 for all analyses. Complete model outputs are provided in supplementary sheet1. Statistical analyses for data in <span class="fakelink linktofig"><span class="fakelink-text">Figures 1C, 1F</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>, and <span class="fakelink linktofig"><span class="fakelink-text">1H</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span> were performed using GraphPad Prism (v10.2.0) with appropriate tests as indicated in figure legends.</p></section><section id="s3j"><h2 class="heading-2">Y-maze olfactory assay for host-seeking behaviour in <em>An. stephensi</em></h2><p>A 30inch Y-maze was constructed as per the WHO guidelines (<span class="fakelink linktoext"><span class="fakelink-text">https://www.who.int/publications/i/item/9789241505024</span><a class="fake-icon" href="https://www.who.int/publications/i/item/9789241505024"><span class="fakelink-icon">     </span></a></span>). Each arm of the Y-maze had a trapping port, while the stem was attached to a holding port. Each port had a mesh screen at one end, while the other was fitted with a sliding door to trap or release mosquitoes. The holding port was attached to a fan which pulled air from the two arms, over the mosquitoes, and out at a velocity of 0.3-0.6m/s. Optic flow information (alternate black and white stripes) was placed underneath the Y-maze to act as visual guide for navigation.</p><p>Following conditions were tested: D0 virgins, D5 sugarfed virgins, D5 sugar-fed mated females (D5M), 3days post-blood-meal virgins (D3<sup>B</sup>), 3days post-blood-meal mated females (D3<sup>B</sup>M), and 2days post-oviposition females (D2<sup>○</sup>). For mated conditions, females were co-housed with males for 5days post-emergence and assumed to be mated based on our previous findings (<span class="fakelink linktofig"><span class="fakelink-text">figure 1G</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>). For each condition, mosquitoes were collected and provided <em>ad libitum</em> access to sugar as described above. Females were transferred to holding ports 7-8h prior to the assay and were starved during this time with only access to water. For each run, mosquitoes were acclimatised for 5 minutes: 2 min with clean air and 3 minutes with an experimenter presenting a hand at one of the arms and simultaneously exhaling at resting rates. To account for disruption in airflow, the hand of a mannequin was placed at the other arm (control). Post-acclimatisation, the door of the holding port was opened to release the mosquitoes and they were allowed to make a choice for 5 minutes in the presence of continued host-cues. The ports were then closed and mosquitoes in each arm counted. All the mosquitoes trapped in a port, sitting on the door or resting on the half-length of the Y-arm were considered as “attracted” to that stimuli (host or control, <span class="fakelink linktofig"><span class="fakelink-text">figure 2A</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span>). Participation percentage was also calculated as number of mosquitoes that passed halfway through the Y-stem/total number of mosquitoes released (<span class="fakelink linktofig"><span class="fakelink-text">figure S1I</span><a class="fake-icon" href="#figS1"><span class="fakelink-icon">     </span></a></span>).</p><p>Several controls were included to avoid experimental artefacts. Each experimental day started with a blank trial (where no stimuli were presented on either arm of the Y-maze) and the side of the host-cues presentation was alternated with each trial to eliminate side biases. Trials for blood-fed virgins and mated females were done both in parallel and on multiple days to eliminate the influence of external factors which might be present on a given day. When testing bloodfed females, aged-matched sugar-fed females (blood-hungry) were included as positive controls where ever possible, with satisfactory results. Lastly, we made sure that at least 50% of the mosquitoes had made either choice in all runs, except for D0 (newly emerged females) and blank runs. In the rare cases that this was not true, we did not consider those for the analysis. Statistical analysis was performed using GraphPad Prism (v10.2.0). To compare the attraction of females to hostcues and control in <span class="fakelink linktofig"><span class="fakelink-text">figure 2B</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span>, each time-point was treated as an independent experiment and p-values were computed for each pair-wise comparison.</p></section><section id="s3k"><h2 class="heading-2">Bulk RNA sequencing of central brains and data analysis</h2><p>For bulk RNA sequencing, central brains were dissected from the following conditions: D0 and D5 sugar-fed males; D0 and D5 sugar-fed virgin females; 1day post-blood-meal virgin (D1<sup>B</sup>) and mated females (D1<sup>B</sup>M) and 1day post-oviposition females (D1<sup>○</sup>). For post-blood-meal conditions, 5days old females were fed on blood and fully fed females were separated for sample preparation. Females co-housed with males for 5 days were assumed to be mated based on our previous findings (<span class="fakelink linktofig"><span class="fakelink-text">figure 1G</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>). Three replicates from independent batches were prepared for each condition.</p><p>Mosquitoes were cold anaesthetised before dissections. Central brains (ie. brains without optic lobes) were dissected in RNase-free 1X PBS and immediately placed into ACME solution (13:3:2:2 of water, methanol, acetic acid, and glycerol) on ice, which mildly fixes the tissues (<a class="linktoref" href="#c102">102</a>). For each condition, 100 brains were dissected per replicate. The mildly fixed brains were washed with cold 1X PBS and centrifuged at 13,000 rpm for 5 min at 4°C to remove ACME. Brains were then homogenised in TRIzol (Invitrogen), flash-frozen, and stored at −80°C until RNA extraction. RNA was extracted using Direct-zol RNA MicroPrep (Zymo Research) columns. RNA quantity and quality were assessed using the ThermoFisher Qubit 4 and RNA 6000 Nano kit Bioanalyser, respectively.</p><p>The cDNA library was prepared for each sample using the NEBNext Ultra II Directional RNA Library Prep kit (Catalog no-E7765L), with an average fragment size of 250 bp. Some of the library samples were subjected to size and quality checks using a bioanalyser. Paired-end sequencing was carried out on the Illumina Hiseq 2500 sequencing platform, generating 2×100 bp reads. The library preparation and sequencing were performed at the Next Generation Genomics Facility (NGGF) at Bangalore LifeScience Cluster.</p><p>Post-sequencing, samples were demultiplexed and delivered as raw Fastq files. After passing the quality check using the FastQC tool, the reads were mapped onto the reference <em>An. stephensi</em> transcriptome (NCBI RefSeq assembly GCF013141755.1) (<a class="linktoref" href="#c69">69</a>) using Kallisto v0.43.1 (<a class="linktoref" href="#c103">103</a>) with default parameters. 81-87% of reads aligned to the mosquito transcriptome.</p><p>Transcript-level abundance was quantified using Kallisto and gene-level summarisation was performed using the Tximport function in R, with transcript isoform information retained (supplementary sheet 2). Using the EgdeR package, the raw counts were converted to Counts Per Million (CPM) and log2 transformed before data filtering (genes with less than 1 CPM in at least 3 or more samples were filtered out) and normalisation, using the trimmed mean of M-values (TMM) method. Principal component analysis (PCA) was performed using the prcomp function in R and visualised using the ggplot function. Data was checked for batch effects using gPCA(<a class="linktoref" href="#c104">104</a>) in R and the test statistic “delta” was found to be 6.957686e-05, with a p-value of &lt;0.001, indicating no batch-effects.</p><section id="s3k1"><h3 class="heading-3">Differential gene expression analysis</h3><p>Differential expression analysis was performed using the Limma package in R. A design matrix was created to specify the sample conditions, and the Voom function was used to estimate the mean-variance relationship across genes. A linear model was then fitted to the data using the mean-variance relationship. Bayesian statistics were calculated for the data by creating a contrast matrix to extract the linear model fit. To identify differentially expressed genes (DEGs), the “decide-Tests” function was used by applying a fold change threshold of 2 (log2fc=1) and a FDR (false discovery rate) adjusted p-value cut-off of 0.05. The number of genes identified to be upregulated and downregulated in each pair-wise comparison are shown in figure S2A and the details are provided in supplementary sheet 3.</p><p>To plot Transcripts Per Million (TPM) values as heatmaps, the Fastq files for the central brain samples were mapped onto the reference <em>An. stephensi</em> transcriptome and abundance was quantified using Kallisto the same way as described above. Gene-level summarisation was done using import-Tx as described above, except all the isoforms of a given transcript were mapped to a single gene (supplementary sheet2). The resultant TPM values were first filtered for TPM&gt;0.1 in each sample. Next, the genes that were not quantified in at least 2 of the 3 replicates of each sample group were removed. The filtered dataset were then log10 transformed and heatmaps were plotted using the pheatmap function inR. To construct heatmaps of genes involved in carbohydrate and lipid metabolism (<span class="fakelink linktofig"><span class="fakelink-text">supplementary figure S3</span><a class="fake-icon" href="#figS3"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">S4</span><a class="fake-icon" href="#figS4"><span class="fakelink-icon">     </span></a></span>), key enzymes of the respective canonical pathways were manually curated based on the existing literature. Gene annotations for neuropeptides and neuropeptide receptors were curated from published sources (<a class="linktoref" href="#c105">105</a>–<a class="linktoref" href="#c107">107</a>). Fat body, midgut and ovary data (SRX620223, SRX618937 and SRX620224; bioproject SRP043489) for 2-4days old, sugar-fed <em>An. stephensi</em> females were taken from Prasad et al.(<a class="linktoref" href="#c108">108</a>). Fastq files were downloaded from the SRA explorer and processed similarly (supplementary sheet 2).</p></section><section id="s3k2"><h3 class="heading-3">Gene set enrichment analysis</h3><p>eggNOG mapper tool 2.1.9 (<a class="linktoref" href="#c109">109</a>) (parameters: 70% identity and minimum 70% query coverage) was used to find functionally annotated orthologs of the identified DEGs in <em>An. gambiae. An. gambiae</em>-specific orthologs were fed into the GOSt tool from gProfiler2 (<a class="linktoref" href="#c110">110</a>) toolset for functional enrichment analysis. A custom gene list comprising of all the genes identified in our RNA-seq experiment was provided as a background instead of the whole genome. Using this background, an ordered query was run with an FDR adjusted p-value threshold of 0.05 to determine which GO terms (Biological Processes) were over-represented. This was done for all sets of DEGs and the top 10 terms of each set were plotted using ggplot function in R (<span class="fakelink linktofig"><span class="fakelink-text">figure S2B</span><a class="fake-icon" href="#figS2"><span class="fakelink-icon">     </span></a></span>).</p></section></section><section id="s3l"><h2 class="heading-2">dsRNA-mediated gene knockdown</h2><p>To synthesise the double-stranded RNA (dsRNA), coding regions (500-600bp) of candidate genes were amplified from cDNA of <em>An. stephensi</em> females using forward and reverse primers with T7 promoter sequence at 5’ ends. The gel purified fragments were then used as a template for <em>in vitro</em> transcription reactions using Megascript RNAi kit (Invitrogen) or High-Yield T7 RNAi Kit (Jena Biosciences) as per the manufacturers’ protocols. The dsRNAs were purified using the phenol-chloroform method and pellet dissolved in nuclease-free water. The dsRNA for green fluorescent protein (GFP) was synthesised from plasmid pAC5.1B-EGFP (Addgene 21181) to be used as a negative control for off-target effects of dsRNA-mediated knockdown.</p><p>Virgins were collected for injections as described above. The day and concentration of dsRNA to be injected was determined for each target gene individually. <em>dsEiger</em>: injected in 24h old virgins at 5-6<em>µ</em>g/<em>µ</em>L; <em>dsHairy</em>: injected in 24h old virgins at 4-5<em>µ</em>g/<em>µ</em>L; <em>dsJHEH</em>: injected in 72h old virgins at 5<em>µ</em>g/<em>µ</em>L; <em>dsPrp</em>: injected in 72h old virgins at 3-5<em>µ</em>g/<em>µ</em>L; <em>dsOrk</em>: injected in 96h old virgins at 5<em>µ</em>g/<em>µ</em>L; <em>ds118504077</em> and <em>ds118504075</em>: 96h old virgins at 6<em>µ</em>g/<em>µ</em>L. To achieve tissue-specific knockdown of <em>sNPF</em> and <em>RYa</em>, injections were performed at different days: in 0-10h old virgins (4<em>µ</em>g/<em>µ</em>L or 7<em>µ</em>g/<em>µ</em>L) to achieve abdomen-specific knockdowns and in 96h old virgin females (7<em>µ</em>g/<em>µ</em>L) to achieve knockdowns in both head and abdominal tissues. For dual injections, purified dsRNAs were mixed in equal amounts to a total of 7<em>µ</em>g/<em>µ</em>L and injected on the day previously determined.</p><p>Injections were performed on a FemtoJet 4i system (Eppendorf) and each female was injected with 0.5-0.6<em>µ</em>L of the respective concentrated dsRNAs in the upper thorax. In some cases, dsRNAs were also injected in the abdomens to ensure that the site of injection does not alter the behaviour. Uninjected or <em>dsGFP</em>-injected (same concentration as that of the candidate gene) females were included as a control each time. 15-25 females were injected for each trial and at least 2 trials from independent batches of mosquitoes were performed for each target gene.</p><p>Injected females were allowed to recover in paper cups with <em>ad libitum</em> access to sugar and assessed for blood-feeding behaviour at 3 days post-injection, as previously described. Fed and unfed females were visually scored and whole heads and abdomens were collected separately for analysis. Total RNA was extracted using Direct-zol RNA MicroPrep (Zymo Research) columns, as per the manufacturer’s protocol. RNA was quantified using Qubit Broad range assay (Invitrogen). For each sample, 200-220ng RNA was used to prepare cDNA using the SuperScript™ IV Reverse Transcriptase (Invitrogen 18090050) or Maxima H Minus Reverse transcriptase (Thermo Scientific EP0752). Quantitative PCR (qPCR) was performed using PowerUp™ SYBR™ Green Master Mix (Applied Biosystems) to determine the knockdown efficiency. Relative mRNA abundances were calculated using the delta-delta Ct method (<a class="linktoref" href="#c111">111</a>), normalised to RpS7. Statistical analyses were performed using GraphPad Prism (v10.2.0). All the primers used in the study are listed in <span class="fakelink linktofig"><span class="fakelink-text">supplementary table 1</span><a class="fake-icon" href="#tblS1"><span class="fakelink-icon">     </span></a></span>.</p><p>To assess the sugar-feeding behaviour of dsRNA injected females (<em>dssNPF, dsRYa</em> and <em>dsRYa+sNPF</em>), injections were performed as described above. Injected females were allowed to recover with <em>ad libitum</em> access to normal sugar for the first two days, which was then replaced with Rhodamine B-spiked sugar (1mg/25ml of sugar, Sigma R6626) for the next 24h. Control females were treated similarly. The blood-feeding assay was performed 3 days post-injection as described above, with both blood and sugar provided in parallel. Abdomens were scored visually for ingestion of blood under white light, and under fluorescent microscope (red channel) for ingestion of sugar (figure S9A).</p></section><section id="s3m"><h2 class="heading-2">Hybridisation chain reaction (HCR)</h2><p>Previously published protocol (<a class="linktoref" href="#c112">112</a>) was modified to perform HCR <em>in situ</em> hybridisations on brain and gut tissues collected from females under three different conditions: D0 (not interested in blood), D5 virgins (blood-hungry) and D2<sup>B</sup>, mated (blood-sated). All the reagents, including the probes, amplifiers and buffers were procured from Molecular Instruments Inc. A set of custom-designed 20 probe pairs were used per target gene: <em>sNPF (118511038), RYa (118513478)</em> and their receptors. To find <em>sNPF</em> and <em>RYa</em> receptors in <em>An. stephensi</em>, putative protein sequences were blasted against orthologs in <em>Drosophila melanogaster, Ae. aegypti</em> and <em>An. gambiae</em>. Based on high identities, consistent in all three organisms, <em>118517381</em> and <em>118507167 (NPY receptor type 2)</em> were identified as <em>RYa receptor (RYaR)</em> and <em>sNPF receptor (sNPFR)</em> respectively.</p><p>5‐7 brains for each target, per condition were dissected in ice cold PBS and fixed with 4% PFA (in PBS) for 20 minutes at room temperature. Post-washing steps with PTw (0.1% Tween-20 in PBS), 4 times 10 mins each, tissues were permeabilised using detergent solution (50mM Tris-Cl, pH7.5, 150mM NaCl, 1mM EDTA, pH8, 0.5% Tween-20, 1% SDS) for 30 mins at room temperature. Brains were then pre-hybridised in probe hybridisation buffer for 30 mins at 37°C, followed by incubation with neuropeptide probes (0.8pmol each) at 37°C for 48h in a humid chamber. Samples were washed with pre-warmed wash buffer, 4 times 15 mins each, at 37°C to remove unbound probes. This was followed by 2 washes, 5 mins each with 5X SSCT (0.1% Tween-20 in 5X SSC) at room temperature. Samples were then pre-amplified using amplification buffer for 30 mins at room temperature. For each sample, 2<em>µ</em>L (3<em>µ</em>M stock) of each hairpins were mixed and prepared by heating at 95°C for 90secs for annealing. They were allowed to cool to room temperature for 30mins in a dark chamber and 100<em>µ</em>L amplification buffer was added to the mix. Brains were incubated in the hairpin solution for 12-16h at 37°C in dark and excess hairpins were washed off with multiple 5X SSCT washes: twice for 5 mins, twice for 30mins, once for 5 mins. For nc82 staining, all the steps were performed in 0.5% PTw buffer (0.5% Tween-20 in PBS). Brains were equilibrated in 0.5% PTw and incubated with anti-mouse nc82 (1:200; DSHB, RRID: AB2314866) for three overnights at 4°C. Post-washing, goat anti-mouse Alexa-fluor405 (1:400, Invitrogen) was added and samples were left in dark, for one-two overnights at 4°C. After washing off the unbound secondary antibody, samples were equilibrated for at least an hour in Vectashield and mounted in Vectashield for imaging. <em>In situs</em> against neuropeptide receptors were performed the same way, except the washing steps to remove the unbound probes were performed at 42°C and samples were incubated with the hairpin solution at room temperature (to minimise background). As controls for background signal, samples were prepared the same way, except no probes were added in the hybridisation buffer (<span class="fakelink linktofig"><span class="fakelink-text">figure S10E</span><a class="fake-icon" href="#figS10"><span class="fakelink-icon">     </span></a></span>). Samples for all three conditions were dissected, processed and imaged the same day.</p><p>Protocol for gut <em>in situs</em> was adapted from Slaidina et al. (<a class="linktoref" href="#c113">113</a>). 5-8 gut samples for each target, per condition were dissected in ice-cold PBS and fixed with 4% PFA (in PTw) for 20 minutes at room temperature. Post-washing steps with PTw, tissues were dehydrated on ice with graded methanol washes: 25% methanol (in PTw), 50% methanol (in PTw), 75% methanol (in PTw) and 100% methanol for 10 mins each. Samples were stored in 100% methanol for up to a month, at −20°C. For rehydration, samples were washed sequentially with graded methanol: 75% methanol (in PTw), 50% methanol (in PTw), 25% methanol (in PTw), followed by a wash in permeabilisation solution (1% Triton-X in PBS). Samples were permeabilised for 2h at room temperature and fixed again with 4% PFA (in PTw) for 20 minutes. Following 10mins washes: washed once with PTw, twice with 50%PTw/50% 5X SSCT and twice with 5X SSCT, samples were pre-hybridised in probe hybridisation buffer for 30 mins at 37°C, followed by incubation with probes (0.8pmol each) at 37°C for 48h in a humid chamber. Post washing steps with wash buffer at 37°C and 5X SSCT at room temperature, samples were pre-amplified in amplification buffer for 30 mins at room temperature. Hairpins were prepared the same way as described above and samples were incubated in hairpin solutions for 12-16h in dark at room temperature. Following 2 washes 5 mins each with 5X SSCT, samples were incubated with DAPI (1:1000 in 5X SSCT) for one hour at room temperature. Samples were washed with 5X SSCT, 4 times, 10mins each and equilibrated in Vectashield for at least an hour before mounting. As controls for background signal, samples were prepared the same way, except no probes were added in the hybridisation buffer (figure S11G, H). Samples for different conditions, prepared on different days and stored at −20°C, were processed and imaged together. Statistical analyses were performed using GraphPad Prism (v10.2.0). All samples were imaged at Central Imaging and Flow Cytometry Facility (CIFF) at NCBS. Images were acquired with a 40X/1.3NA or 10X/0.4NA (whole guts) objectives at a resolution of 512×512 pixels, on an Olympus FV3000 system. Images were viewed and manually analysed using ImageJ/FIJI (<a class="linktoref" href="#c114">114</a>). Maximum intensity projections of all z-stacks in an image are shown, except in <span class="fakelink linktofig"><span class="fakelink-text">figures 4H, 4I</span><a class="fake-icon" href="#fig4"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">S10B, S10D</span><a class="fake-icon" href="#figS10"><span class="fakelink-icon">     </span></a></span>, where stacks were manually selected to capture relevant brain structures.</p></section></section><section id="s6"><h1 class="heading-1">Supplementary figures</h1><div class="fig-group figure" id="figS1"><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://iiif.elifesciences.org/epp/108625%2Fv1%2Fcontent%2F594342v3_figS1.tif/full/max/0/default.jpg" alt=""><p class="figure__caption"><span class="label figure-name"><label class="figure__label">Supplementary Figure 1.</label></span><strong>Standardization of various behavioural assays to study feeding behaviours in <em>An. stephensi</em></strong> A. Detailed schematic of the assay designed to assess the blood-feeding behaviour of virgins and co-housed females across the reproductive cycle. 0-7h old mosquitoes were collected and about 20 females (with or without 20 males) were separated into cups. Each cup was aged appropriately and tested on the relevant day (see methods for more details). Number of blood-fed mosquitoes were visually scored, irrespective of the quantity fed. Representative images of differently fed females are shown in A’. Females were tested for first blood-meal from the day of emergence (D0) to five days of age, every 24h. Spermathecae of co-housed females were dissected post-hoc to determine the mating status. To assay for the second blood-meal, fully fed females were collected and aged appropriately in groups of 20. To assess feeding 1day post-first blood-meal (D1<sup>B</sup>), Rhodamine B was spiked in the second blood-meal offered to the females. The subsequent days (D2<sup>B</sup>-D4<sup>B</sup>) were tested without any such spiking (grey box; see methods for more details). To assay for blood-feeding behaviour in females after oviposition (D1<sup>○</sup>-D2<sup>○</sup>), freshly oviposited females were collected and aged in groups of 20 for the appropriate time in cups (blue box; see methods for more details). In all cases, females were presented with host-cues prior to the assay and testing was performed as shown in <span class="fakelink linktofig"><span class="fakelink-text">figure 1B</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span> (see methods for more details). B. Standardisation of spiking Rhodamine B dye in blood to assess feeding 24h post-first blood-meal (D1<sup>B</sup>). To assay for the second blood-meal, blood-fed females were collected, aged for 24h and offered Rhodamine B-spiked blood in a Hemotek. All other assay parameters remained the same as for the first blood-meal (see methods for more details). Widefield fluorescent images of <em>Ae. aegypti</em> and <em>An. stephensi</em> fed on second blood-meal spiked with Rhodamine B are shown. Images on the left (black box) show mosquitoes not fed and fed on un-spiked blood (without Rhodamine B), as controls for autofluorescence from body tissues and blood itself, respectively. Fluorescence from Rhodamine B added to the second blood-meal enabled identification of fresh blood intake in previously fed mosquitoes reliably. C,D. Comparison of feeding behaviours of <em>Ae. aegypti</em> females (C) and <em>An. stephensi</em> females (D) fed on blood with and without Rhodamine B at the indicated age. Females fed similarly on both indicating that Rhodamine B itself does not affect feeding behaviour. Unpaired t-test; n.s.: not significant, p&gt;0.05. 16-20 females/trial. n= 5 trials/group. D5M: 5days post-emergence, mated; D2<sup>B</sup>M: 2days post-blood-meal, mated; D8M: 8days post-emergence, mated). E. Determining optimal oviposition timing in <em>An. stephensi</em>. Females were tested in groups of 10 and average number of eggs laid per group are shown at the indicated day after the first blood-meal. Dashed lines connect the eggs laid by each group on the indicated day. Majority of the eggs were laid 2days after blood-meal. 10 females/group, n=10 replicates/group. D0<sup>B</sup>-D5<sup>B</sup>M: 0-5days post-blood-meal, mated. F. Standardisation of the dual choice assay, to determine the choice between blood and sugar in sugar-sated and sugar-starved <em>An. stephensi</em> females, co-housed with males. Widefield fluorescent images of females fed on either blood, sugar or both are shown. White arrowhead marks the fluorescence from sugar (spiked with Rhodamine B) in the abdomens of females engorged on blood, indicating that the female has fed on both. This is not observed in the females fed on blood alone. Females fed on sugar alone can be visualised by the fluorescent lean bellies. Unfed females exhibit no fluorescence or engorgement. G. Widefield fluorescent images of female <em>An. stephensi</em> fed on sugar spiked with Rhodamine B. This setup was used to study the sugar-feeding behaviour of virgins and co-housed females across the reproductive cycle. Addition of Rhodamine B to standard sucrose solution enabled visualisation of even the smallest amounts ingested, as indicated by the white arrowheads, under the fluorescent microscope. H. Mating status of sugar-fed and sugar-unfed co-housed females tested in <span class="fakelink linktofig"><span class="fakelink-text">figure 1E</span><a class="fake-icon" href="#fig1"><span class="fakelink-icon">     </span></a></span>, 24-120h post-emergence (D0-D5). n=175-179 females per time-point. I. Percent participation of virgin and co-housed females tested for host-seeking behaviour in the Y-maze olfactometer (<span class="fakelink linktofig"><span class="fakelink-text">figure 2B</span><a class="fake-icon" href="#fig2"><span class="fakelink-icon">     </span></a></span>), at different stages of the reproductive cycle. All females tested participated well. 17-21 females/trial, n=10 trials/group. Kruskal-Wallis, with Dunn’s multiple correction test, p&lt;0.05. Data labelled with different letters are significantly different.D0: day of emergence; D5M: 5days post-emergence, mated; D5: 5days post-emergence, virgin; D3<sup>B</sup>M: 3days post-blood-meal, mated; D3<sup>B</sup>: 3days post-blood-meal, virgin; D2<sup>○</sup>: 2days post-oviposition.</p></div><figure class="figure tofill" id="figS2"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Supplementary Figure 2.</label></span>Differential expression of genes across various states of blood-hunger and -satiety.</h3><p>A. Number of genes identified to be differentially up- and down-regulated in the central brain of females across the reproductive cycle (indicated at the bottom), compared to either newly emerged males or females (D0), or blood-fed sated females (D2<sup>B</sup>M). Note that no genes were found to be differentially expressed between mated and virgin blood-fed females, despite their contrasting blood- appetites. B. Top 10 Gene Ontology terms for Biological Processes (GOBP), overrepresented in the samples indicated at the bottom, compared to newly emerged males or females (D0). Size of the dot indicates the p-value significance. Processes involved in protein metabolism (highlighted in red) were found to be highly enriched in blood-fed females. (D0: day of emergence; D5: 5days post-emergence, virgin; D1<sup>B</sup>: 1day post-blood-meal, virgin; D1<sup>B</sup>M: 1day post-blood-meal, mated; D1<sup>○</sup>: 1day post-oviposition)</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://iiif.elifesciences.org/epp/108625%2Fv1%2Fcontent%2F594342v3_figS2.tif/full/max/0/default.jpg" alt=""></figure><figure class="figure tofill" id="figS3"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Supplementary Figure 3.</label></span>Differential expression of genes involved in carbohydrate metabolism across different tissues and states of blood-hunger and -satiety in <em>An. stephensi</em></h3><p>Expression patterns of genes (listed on the right) involved in sugar metabolism via different pathways, across the reproductive cycle in the central brain and in the fat body, midgut and ovary of blood-hungry females, are represented as heatmaps. Green box shows downregulation of genes involved in glucose utilisation and the magenta box represents genes involved in glucose storage. Note that both are low in the brains of blood-fed females indicating that the brain goes into a state of ‘sugar rest’ after a blood-meal. Analysing publicly available data from the midgut, ovary, and fat body of sugar-fed females suggest that energy expenditure via the non-oxidative branch of the pentose phosphate pathway of glucose metabolism is high in these tissues, likely to prime them for the blood-meal. RNA-seq data from all central brain samples in triplicates and single sample data for fat body, midgut and ovary (taken from Prasad et al., 2017) are represented as z-score normalised log10 (TPM+1) values. TPM values of zero were excluded from z-score transformation and are displayed as grey cells. (D0: day of emergence; D2-D4: 2-4days post emergence; D5: 5days post-emergence, virgin; D1<sup>B</sup>: 1day post-blood-meal, virgin; D1<sup>B</sup>M: 1day post-blood-meal, mated; D1<sup>○</sup>: 1day post-oviposition)</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://iiif.elifesciences.org/epp/108625%2Fv1%2Fcontent%2F594342v3_figS3.tif/full/max/0/default.jpg" alt=""></figure><figure class="figure tofill" id="figS4"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Supplementary Figure 4.</label></span>Differential expression of genes involved in lipid metabolism across different tissues and states of blood-hunger and -satiety in <em>An. stephensi</em></h3><p>Expression patterns of genes (listed on the right) involved in lipid metabolism: lipid breakdown (top) and lipid synthesis (bottom), across the reproductive cycle in the central brain and in the fat body, midgut and ovary of blood-hungry females, are represented as heatmaps. No specific patterns were observed in the central brain (pre- and post-blood-meal), midgut or ovaries. As expected, fat body showed high expression of genes involved in both the catabolic and anabolic branches of fatty-acid metabolism. RNA-seq data from all central brain samples in triplicates and single sample data for fat body, midgut and ovary (taken from Prasad et al., 2017) are represented as z-score normalised log10 (TPM+1) values. TPM values of zero were excluded from z-score transformation and are displayed as grey cells. (D0: day of emergence; D2-D4: 2-4days post emergence; D5: 5days post-emergence, virgin; D1<sup>B</sup>: 1day post-blood-meal, virgin; D1<sup>B</sup>M: 1day post-blood-meal, mated; D1<sup>○</sup>: 1day post-oviposition)</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://iiif.elifesciences.org/epp/108625%2Fv1%2Fcontent%2F594342v3_figS4.tif/full/max/0/default.jpg" alt=""></figure><figure class="figure tofill" id="figS5"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Supplementary Figure 5.</label></span>Expression patterns of neuropeptides and neuropeptide-receptor across different tissues and states of blood-hunger and -satiety in <em>An. stephensi</em></h3><p>Expression patterns of neuropeptides (top) and neuropeptide-receptors (bottom), across the reproductive cycle in the central brain and in the fat body, midgut and ovary of blood-hungry females, are represented as heatmaps. While neuropeptides were observed to be downregulated in the central brain of newly emerged males and females (D0), neuropeptide-receptors showed no such patterns. Additionally, both neuropeptides and their receptors were much more abundant in the brain, as compared to fat body, midgut or ovaries. Transcripts for the receptors highlighted in red were visualised <em>in situ</em> both in the midgut and brain samples via HCR (<span class="fakelink linktofig"><span class="fakelink-text">figure S10</span><a class="fake-icon" href="#figS10"><span class="fakelink-icon">     </span></a></span>, <span class="fakelink linktofig"><span class="fakelink-text">S11</span><a class="fake-icon" href="#figS11"><span class="fakelink-icon">     </span></a></span>). RNA-seq data from all central brain samples in triplicates and single sample data for fat body, midgut and ovary (taken from Prasad et al., 2017) are represented as z-score normalised log10 (TPM+1) values. TPM values of zero were excluded from z-score transformation and are displayed as grey cells. (D0: day of emergence; D2-D4: 2-4days post emergence; D5: 5days post-emergence, virgin; D1<sup>B</sup>: 1day post-blood-meal, virgin; D1<sup>B</sup>M: 1day post-blood-meal, mated; D1<sup>○</sup>: 1day post-oviposition)</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://iiif.elifesciences.org/epp/108625%2Fv1%2Fcontent%2F594342v3_figS5.tif/full/max/0/default.jpg" alt=""></figure><figure class="figure tofill" id="figS6"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Supplementary Figure 6.</label></span>Expression patterns of neurotransmitter-related genes in the central brain of <em>An. stephensi</em> across different states of blood-hunger and-satiety</h3><p>Expression patterns of genes (listed on the right) involved in neurotransmitter expression, processing and transport across the reproductive cycle in the central brain, are represented as heatmaps. Interestingly, many of these genes were observed to be up-regulated in the newly emerged (D0) mosquitoes as compared to the older and blood-fed mosquitoes. RNA-seq data from all central brain samples in triplicates are represented as z-score normalised log10 (TPM+1) values. (D0: day of emergence; D5: 5days post-emergence, virgin; D1<sup>B</sup>: 1day post-blood-meal, virgin; D1<sup>B</sup>M: 1day post-blood-meal, mated; D1<sup>○</sup>: 1day post-oviposition)</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://iiif.elifesciences.org/epp/108625%2Fv1%2Fcontent%2F594342v3_figS6.tif/full/max/0/default.jpg" alt=""></figure><figure class="figure tofill" id="figS7"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Supplementary Figure 7.</label></span>Candidates potentially involved in promoting blood-feeding behaviour in <em>An. stephensi</em></h3><p>71 candidate genes identified as potential promoters of blood-feeding behaviour are listed. These were identified to be commonly up-regulated in different conditions of blood-hunger (as compared to conditions of satiety or no-interest), and non-overlapping in males. Nine genes shortlisted for further validation are highlighted in red. RNA-seq data from all central brain samples in triplicates are represented as z-score normalised log10 (TPM+1) values. TPM values of zero were excluded from z-score transformation and are displayed as grey cells. (D0: day of emergence; D5: 5days post-emergence, virgin; D1<sup>B</sup>: 1day post-blood-meal, virgin; D1<sup>B</sup>M: 1day post-blood-meal, mated; D1<sup>○</sup>: 1day post-oviposition)</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://iiif.elifesciences.org/epp/108625%2Fv1%2Fcontent%2F594342v3_figS7.tif/full/max/0/default.jpg" alt=""></figure><figure class="figure tofill" id="figS8"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Supplementary Figure 8.</label></span>Screening of the shortlisted candidates, potentially involved in promoting blood-feeding behaviour in <em>An. stephensi</em> via dsRNAmediated gene knockdown</h3><p>A-H. Blood-feeding behaviour of virgin females injected with dsRNA against <em>Eiger</em> (A), <em>Juvenile hormone epoxide hydrolase-1 like (JHEH)</em> (B), <em>Orcokinin peptides type-A like (Ork)</em> (C), <em>Hairy</em> (D), <em>Prothoracicostatic peptides (Prp)</em> (E), <em>Na-dependent amino acid transporter-1 like 118504075</em> (F), <em>Na-dependent amino acid transporter-1 like 118504077</em> (G), <em>118504075+118504077</em> (H). <em>dsGFP</em> was injected as a control in each case. No change in the behaviour was observed for any of the candidates tested. 18-25 females/replicate, n=2-6 replicates/group. Unpaired t-test; n.s.: not significant, p&gt;0.05; *p&lt;0.05 (A, D, F-H). A’-H’. Relative mRNA expression in the heads of virgin <em>An. stephensi</em> females injected with the indicated dsRNA, as compared to that in <em>dsGFP</em> injected females, analysed via qPCR. Efficient knockdown could not be achieved for <em>Hairy</em> and <em>Prp</em>. Unpaired t-test; ****p&lt;0.0001 (A’, D’). One-way ANOVA with Holm-Šídák multiple comparisons test, p&lt;0.05 (F’-H’). For candidates <em>JHEH</em> (B), <em>Ork</em> (C) and <em>Prp</em> (E), significance could not assessed, as statistical testing requires a minimum of three replicates, which were not available for these experiments.</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://iiif.elifesciences.org/epp/108625%2Fv1%2Fcontent%2F594342v3_figS8.tif/full/max/0/default.jpg" alt=""></figure><figure class="figure tofill" id="figS9"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Supplementary Figure 9.</label></span>Abdominal knockdown of <em>RYa</em> or <em>sNPF</em> alone does not contribute to the blood-feeding behaviour in <em>An. stephensi</em></h3><p>A. Widefield fluorescent images of females assayed for both sugar and blood appetite, post-dsRNA injections. Females were injected as described earlier (see methods for details) and offered Rhodamine B-spiked sugar 24h prior to the assay. Blood-feeding behaviour was tested as described earlier. Representative images of females fed on both blood and sugar, blood only, sugar only or none, are shown. B-H. Blood-feeding and sugar-feeding behaviour of virgin females injected with <em>dsRYa</em> (B) and <em>dssNPF</em> (E). <em>dsGFP</em> injected or uninjected females were used as controls. Relative mRNA expression in the heads and abdomens of the injected females were analysed via qPCR (C, F) to assay for the knockdown efficiency in both tissues. Percent females engorged on blood upon injection with <em>dsRYa</em> and <em>dssNPF</em>, as compared to control females are shown in D and G respectively. Representative widefield images of females considered to be engorged or not, are shown in H. While the mRNA levels in the abdomens were significantly downregulated, efficient knockdown could not be achieved in the heads. Under these conditions, no change in the blood- or sugar-feeding behaviour was observed for either of the targets. However, fewer females were able to fully engorge on blood in both cases and instead took smaller meals. 16-26 females/replicate, n=5-10 replicates/group. Unpaired t-test; n.s.: not significant, p&gt;0.05; *p&lt;0.05, ***p&lt;0.001 (B,D,E,G). One-way ANOVA with Holm-Šídák multiple comparisons test, p&lt;0.05 (C,F). I-L. Blood-feeding and sugar-feeding behaviour of females when <em>RYa</em> and <em>sNPF</em> are knocked down only in the abdomen. Virgin females were injected with dsRNA against <em>RYa</em> (H) and <em>sNPF</em> (J) and assayed for both sugar- and blood-appetite, as described before. <em>dsGFP</em> injected females were used as controls. Relative mRNA expression in the heads and abdomens of <em>RYa</em> (I) and <em>sNPF</em> (K) injected females were analysed via qPCR to assay for the knockdown efficiency in both tissues. No change in the blood- or sugar-feeding behaviour was observed when <em>RYa</em> or <em>sNPF</em> levels were significantly downregulated in the abdomens without altering the levels in the heads, suggesting that neuronal expression is necessary to promote the blood-feeding behaviour. 19-24 females/replicate, n=4-5 replicates/group. Unpaired t-test; n.s.: not significant, p&gt;0.05 (I, K). One-way ANOVA with Holm-Šídák multiple comparisons test, p&lt;0.05 (J,L).</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://iiif.elifesciences.org/epp/108625%2Fv1%2Fcontent%2F594342v3_figS9.tif/full/max/0/default.jpg" alt=""></figure><figure class="figure tofill" id="figS10"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Supplementary Figure 10.</label></span>Transcripts of neuropeptides <em>sNPF, RYa</em> and their receptors are expressed in <em>An. stephensi</em> central brain</h3><p>A, B. <em>RYa</em> and <em>sNPF</em> mRNA expression in the central brain of females uninterested in blood (D0), blood-hungry females (D5) and blood-sated females (D2<sup>B</sup>M), analysed via HCR. Transcripts of both neuropeptides are expressed in several clusters across the central brain (A). While the clusters are largely non-overlapping, co-expression was observed in the Kenyon cells of the mushroom body, indicated by the marked area (B). Maximum intensity projections of confocal stacks are shown. Scale bar, 50µm. C. <em>RYa receptor (RYaR)</em> and <em>sNPF receptor (sNPFR)</em> mRNA expression in the central brain of females uninterested in blood (D0), blood-hungry females (D5) and blood-sated (D2<sup>B</sup>M), analysed via HCR. Transcripts of both receptors were found to be expressed in multiple cells. Maximum intensity projections of confocal stacks are shown. Scale bar, 50µm. D. <em>sNPFR</em> mRNA expression in the central complex. A magnified image is shown in D’. Maximum intensity projection of selected confocal stacks is shown. Scale bar, 50µm. E. central brain samples incubated with HCR-Amplifiers only, to serve as a control for background signal in the absence of HCR probes. No signal was observed for either of the amplifiers used (B1 labelled with 546 fluorophore (B1-546) and B2 labelled with 488 fluorophore (B2-488)). (D0: day of emergence; D5: 5days post-emergence, virgin; D2<sup>B</sup>M: 2days post-blood-meal, mated)</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://iiif.elifesciences.org/epp/108625%2Fv1%2Fcontent%2F594342v3_figS10.tif/full/max/0/default.jpg" alt=""></figure><figure class="figure tofill" id="figS11"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Supplementary Figure 11.</label></span>Transcripts of <em>sNPF</em> and <em>sNPFR</em> are expressed in <em>An. stephensi</em> posterior midgut</h3><p>A-F. <em>RYa</em> (A), <em>sNPF</em> (B), <em>RYaR</em> (C) and <em>sNPFR</em> (D) mRNA expression in the guts of females uninterested in blood (D0), blood-hungry females (D5) and blood-sated females (D2<sup>B</sup>M), analysed via HCR. Transcripts of both <em>sNPF</em> and its receptor <em>sNPFR</em> are expressed in the posterior midgut, at all stages analysed. No expression of <em>RYa</em> or <em>RYaR</em> was observed. Magnified images of transcript expression of neuropeptides and their receptors are shown in E and F, respectively. Scale bar, 50µm. G,H. Gut samples incubated with HCR-Amplifiers only, to serve as a control for background signal in the absence of HCR probes. No signal was observed for either of the amplifiers used (B1 labelled with 546 fluorophore (B1-546) and B2 labelled with 488 fluorophore (B2-488)). Scale bar, 50µm. (D0: day of emergence; D5: 5days post-emergence, virgin; D2<sup>B</sup>M: 2days post-blood-meal, mated)</p></figcaption><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://iiif.elifesciences.org/epp/108625%2Fv1%2Fcontent%2F594342v3_figS11.tif/full/max/0/default.jpg" alt=""></figure><figure class="table tofill" id="tblS1"><figcaption class="figure__caption"><h3><span class="label figure-name"><label class="figure__label">Table 1.</label></span>List of primers used in the study for <em>dsRNA</em>-mediated gene knockdown and quantitative PCR (qPCR).</h3></figcaption><img class="child-of-figure imageonly" loading="eager" src="https://iiif.elifesciences.org/epp/108625%2Fv1%2Fcontent%2F594342v3_tblS1.tif/full/max/0/default.jpg" alt=""></figure><div class="pagebreak"></div></section><section id="acknowledgements"><h1 class="heading-1">Acknowledgements</h1><p>This work was supported by the Tata Trust. We wish to acknowledge the insectary and its staff at TIGS. All sequencing was done at the Next Generation Genomics Facility (NGGF) at the Bangalore Life Science Cluster. All imaging was done at the Central Imaging and Flow Cytometry Facility (CIFF) at the Bangalore Life Science Cluster. We are grateful to Nitin Gupta, Mahul Chakraborty, Tussar Saha, Sriram Narayanan, Farah Ishtiaq, Kavita Isvaran, Sunil Laxman, Chris Q Doe, Claude Desplan, Krishna Melnattur, Suresh Subramani, K. VijayRaghavan, the Indian Neurobehaviour group, and the Vosshall lab for their invaluable input and discussion during the course of this project and comments on the manuscript.</p></section><section id="additional-info"><h1 class="heading-1">Additional information</h1><section id="s4"><h2 class="heading-2">Author contributions</h2><p>PB conceptualised the project, performed all experiments except RNA-seq and host-seeking assay, performed all analysis, including RNA-seq, wrote and reviewed the manuscript. RP performed the RNA-seq experiments and analysis, standardised HCR, and reviewed the manuscript. PDB performed the host-seeking assay, provided technical assistance during HCR standardisations and determination of mating status, and reviewed the manuscript. SQS conceptualised and supervised the project and wrote and reviewed the manuscript.</p></section><section id="funding"><h2>Funding</h2><table id="funding-table"><thead><tr><th>Funder</th><th>Grant reference number</th></tr></thead><tbody><tr><td>Tata Trusts</td><td></td></tr><tr class="funding-statement"><td colspan="2">All authors were supported by the funds from the Tata Trusts, India, to TIGS</td></tr></tbody></table></section><section id="orcids"><h2 class="orcid-list__title">Author ORCID iDs</h2><ul class="orcid-list list-simple"><li class="orcid-list-item"><p><strong>Prashali Bansal</strong>: <a href="http://orcid.org/0000-0003-3288-9855">http://orcid.org/0000-0003-3288-9855</a></p></li><li class="orcid-list-item"><p><strong>Sonia Q Sen</strong>: <a href="http://orcid.org/0000-0003-4693-3378">http://orcid.org/0000-0003-4693-3378</a></p></li></ul></section></section><section id="additional-files"><h1 class="heading-1">Additional files</h1><p id="supp1"><span class="fakelink linktoext"><span class="fakelink-text"><label>Supplementary sheet 1.</label></span><a class="fake-icon" href="https://prod--epp.elifesciences.org/api/files/108625/v1/content/supplements/594342_file02.pdf"><span class="fakelink-icon">     </span></a></span>Summary of statistical analysis of feeding behaviours using Generalised Linear Mixed Models (GLMM). </p><p id="supp2"><span class="fakelink linktoext"><span class="fakelink-text"><label>Supplementary sheet 2.</label></span><a class="fake-icon" href="https://prod--epp.elifesciences.org/api/files/108625/v1/content/supplements/594342_file03.xlsx"><span class="fakelink-icon">     </span></a></span>Table of differentially expressed genes (DEGs) identified across states of blood-hunger and -satiety. </p><p id="supp3"><span class="fakelink linktoext"><span class="fakelink-text"><label>Supplementary sheet 3.</label></span><a class="fake-icon" href="https://prod--epp.elifesciences.org/api/files/108625/v1/content/supplements/594342_file04.xlsx"><span class="fakelink-icon">     </span></a></span>Table of transcript abundance (in TPM values) for genes identified in the central brain RNAseq data across the gonotrophic cycle. </p><p id="supp4"><span class="fakelink linktoext"><span class="fakelink-text"><label>Supplementary movie</label></span><a class="fake-icon" href="https://prod--epp.elifesciences.org/api/files/108625/v1/content/supplements/594342_file05.avi"><span class="fakelink-icon">     </span></a></span><em>sNPF</em> expression in the central brain of females uninterested in blood (D0), blood-hungry females (D5) and blood-sated females (D2BM) Confocal stacks of whole brains probed for <em>sNPF</em> mRNA expression via HCR <em>in situ</em> hybridization, at three different conditions of blood-hunger and -satiety, are shown as movies. Note the appearance of novel clusters expressing <em>sNPF</em> transcripts (indicated by magenta arrows) in the subesophageal ganglionic region of the brain, only in the blood-hungry (D5) condition.</p></section><section id="references"><h1 class="heading-1">References</h1><ul class="reference-list"><li class="reference-list__item" id="c1"><span class="reference__label">1.</span><ol class="reference__authors_list"><li class="reference__author">Corrales-Carvajal Verónica María</li><li class="reference__author">Faisal Aldo A</li><li class="reference__author">Ribeiro Carlos</li></ol><span class="reference__authors_list_suffix">2016</span><span class="reference__title">Internal states drive nutrient homeostasis by modulating exploration-exploitation trade-off.</span><span class="reference__origin"><i>eLife</i> <strong>5</strong></span><span class="reference__doi"><a href="https://doi.org/10.7554/ELIFE.19920" class="reference__doi_link">https://doi.org/10.7554/ELIFE.19920</a></span></li><li class="reference-list__item" id="c2"><span class="reference__label">2.</span><ol class="reference__authors_list"><li class="reference__author">Khan Md Shahjalal</li><li class="reference__author">Spann Redin A</li><li class="reference__author">Münzberg Heike</li><li class="reference__author">Yu Sangho</li><li class="reference__author">Albaugh Vance L</li><li class="reference__author">He Yanlin</li><li class="reference__author">Berthoud Hans Rudolf</li><li class="reference__author">Morrison Christopher D</li></ol><span class="reference__authors_list_suffix">2021</span><span class="reference__title">Protein Appetite at the Interface between Nutrient Sensing and Physiological Homeostasis.</span><span class="reference__origin"><i>Nutrients</i> <strong>13</strong>:4103</span><span class="reference__doi"><a href="https://doi.org/10.3390/NU13114103" class="reference__doi_link">https://doi.org/10.3390/NU13114103</a></span></li><li class="reference-list__item" id="c3"><span class="reference__label">3.</span><ol class="reference__authors_list"><li class="reference__author">Heeley Nicholas</li><li class="reference__author">Blouet Clemence</li></ol><span class="reference__authors_list_suffix">2016</span><span class="reference__title">Central amino acid sensing in the control of feeding behavior.</span><span class="reference__origin"><i>Frontiers in Endocrinology</i> <strong>7</strong>:229840</span><span class="reference__doi"><a href="https://doi.org/10.3389/FENDO.2016.00148/BIBTEX" class="reference__doi_link">https://doi.org/10.3389/FENDO.2016.00148/BIBTEX</a></span></li><li class="reference-list__item" id="c4"><span class="reference__label">4.</span><ol class="reference__authors_list"><li class="reference__author">Carvalho Gil B</li><li class="reference__author">Kapahi Pankaj</li><li class="reference__author">Anderson David J</li><li class="reference__author">Benzer Seymour</li></ol><span class="reference__authors_list_suffix">2006</span><span class="reference__title">Allocrine Modulation of Feeding Behavior by the Sex Peptide of Drosophila.</span><span class="reference__origin"><i>Current Biology</i> <strong>16</strong>:692‑696</span><span class="reference__doi"><a href="https://doi.org/10.1016/j.cub.2006.02.064" class="reference__doi_link">https://doi.org/10.1016/j.cub.2006.02.064</a></span></li><li class="reference-list__item" id="c5"><span class="reference__label">5.</span><ol class="reference__authors_list"><li class="reference__author">Vargas Misha A</li><li class="reference__author">Luo Ningguang</li><li class="reference__author">Yamaguchi Atsushi</li><li class="reference__author">Kapahi Pankaj</li></ol><span class="reference__authors_list_suffix">2010</span><span class="reference__title">A Role for S6 Kinase and Serotonin in Postmating Dietary Switch and Balance of Nutrients in D. melanogaster.</span><span class="reference__origin"><i>Current Biology</i> <strong>20</strong>:1006‑1011</span><span class="reference__doi"><a href="https://doi.org/10.1016/j.cub.2010.04.009" class="reference__doi_link">https://doi.org/10.1016/j.cub.2010.04.009</a></span></li><li class="reference-list__item" id="c6"><span class="reference__label">6.</span><ol class="reference__authors_list"><li class="reference__author">Ribeiro Carlos</li><li class="reference__author">Dickson Barry J</li></ol><span class="reference__authors_list_suffix">2010</span><span class="reference__title">Sex peptide receptor and neuronal TOR/S6K signaling modulate nutrient balancing in Drosophila.</span><span class="reference__origin"><i>Current biology: CB</i> <strong>20</strong>:1000‑1005</span><span class="reference__doi"><a href="https://doi.org/10.1016/j.cub.2010.03.061" class="reference__doi_link">https://doi.org/10.1016/j.cub.2010.03.061</a></span></li><li class="reference-list__item" id="c7"><span class="reference__label">7.</span><ol class="reference__authors_list"><li class="reference__author">Walker Samuel James</li><li class="reference__author">Corrales-Carvajal Verónica María</li><li class="reference__author">Ribeiro Carlos</li></ol><span class="reference__authors_list_suffix">2015</span><span class="reference__title">Postmating Circuitry Modulates Salt Taste Processing to Increase Reproductive Output in Drosophila.</span><span class="reference__origin"><i>Current biology: CB</i> <strong>25</strong>:2621‑2630</span><span class="reference__doi"><a href="https://doi.org/10.1016/j.cub.2015.08.043" class="reference__doi_link">https://doi.org/10.1016/j.cub.2015.08.043</a></span></li><li class="reference-list__item" id="c8"><span class="reference__label">8.</span><ol class="reference__authors_list"><li class="reference__author">Laturney Meghan</li><li class="reference__author">Sterne Gabriella R.</li><li class="reference__author">Scott Kristin</li></ol><span class="reference__authors_list_suffix">2023</span><span class="reference__title">Mating activates neuroendocrine pathways signaling hunger in Drosophila females.</span><span class="reference__origin"><i>eLife</i> <strong>12</strong></span><span class="reference__doi"><a href="https://doi.org/10.7554/ELIFE.85117" class="reference__doi_link">https://doi.org/10.7554/ELIFE.85117</a></span></li><li class="reference-list__item" id="c9"><span class="reference__label">9.</span><ol class="reference__authors_list"><li class="reference__author">Inagaki Hidehiko K</li><li class="reference__author">De-Leon Shlomo Ben-Tabou</li><li class="reference__author">Wong Allan M</li><li class="reference__author">Jagadish Smitha</li><li class="reference__author">Ishimoto Hiroshi</li><li class="reference__author">Barnea Gilad</li><li class="reference__author">Kitamoto Toshihiro</li><li class="reference__author">Axel Richard</li><li class="reference__author">Anderson David J</li></ol><span class="reference__authors_list_suffix">2012</span><span class="reference__title">Visualizing neuromodulation in vivo: TANGO-mapping of dopamine signaling reveals appetite control of sugar sensing.</span><span class="reference__origin"><i>Cell</i> <strong>148</strong>:583‑595</span><span class="reference__doi"><a href="https://doi.org/10.1016/J.CELL.2011.12.022" class="reference__doi_link">https://doi.org/10.1016/J.CELL.2011.12.022</a></span></li><li class="reference-list__item" id="c10"><span class="reference__label">10.</span><ol class="reference__authors_list"><li class="reference__author">Inagaki Hidehiko K</li><li class="reference__author">Panse Ketaki M</li><li class="reference__author">Anderson David J</li></ol><span class="reference__authors_list_suffix">2014</span><span class="reference__title">Independent, reciprocal neuromodulatory control of sweet and bitter taste sensitivity during starvation in Drosophila.</span><span class="reference__origin"><i>Neuron</i> <strong>84</strong>:806‑820</span><span class="reference__doi"><a href="https://doi.org/10.1016/J.NEURON.2014.09.032" class="reference__doi_link">https://doi.org/10.1016/J.NEURON.2014.09.032</a></span></li><li class="reference-list__item" id="c11"><span class="reference__label">11.</span><ol class="reference__authors_list"><li class="reference__author">Marella Sunanda</li><li class="reference__author">Mann Kevin</li><li class="reference__author">Scott Kristin</li></ol><span class="reference__authors_list_suffix">2012</span><span class="reference__title">Dopaminergic Modulation of Sucrose Acceptance Behavior in Drosophila.</span><span class="reference__origin"><i>Neuron</i> <strong>73</strong>:941‑950</span><span class="reference__doi"><a href="https://doi.org/10.1016/J.NEURON.2011.12.032" class="reference__doi_link">https://doi.org/10.1016/J.NEURON.2011.12.032</a></span></li><li class="reference-list__item" id="c12"><span class="reference__label">12.</span><ol class="reference__authors_list"><li class="reference__author">Devineni Anita V</li><li class="reference__author">Scaplen Kristin M</li></ol><span class="reference__authors_list_suffix">2022</span><span class="reference__title">Neural Circuits Underlying Behavioral Flexibility: Insights From Drosophila.</span><span class="reference__origin"><i>Frontiers in Behavioral Neuroscience</i> <strong>15</strong></span><span class="reference__doi"><a href="https://doi.org/10.3389/FNBEH.2021.821680" class="reference__doi_link">https://doi.org/10.3389/FNBEH.2021.821680</a></span></li><li class="reference-list__item" id="c13"><span class="reference__label">13.</span><ol class="reference__authors_list"><li class="reference__author">Toshima Naoko</li><li class="reference__author">Tanimura Teiichi</li></ol><span class="reference__authors_list_suffix">2012</span><span class="reference__title">Taste preference for amino acids is dependent on internal nutritional state in Drosophila melanogaster.</span><span class="reference__origin"><i>The Journal of experimental biology</i> <strong>215</strong>:2827‑2832</span><span class="reference__doi"><a href="https://doi.org/10.1242/JEB.069146" class="reference__doi_link">https://doi.org/10.1242/JEB.069146</a></span></li><li class="reference-list__item" id="c14"><span class="reference__label">14.</span><ol class="reference__authors_list"><li class="reference__author">Ganguly Anindya</li><li class="reference__author">Dey Manali</li><li class="reference__author">Scott Christi</li><li class="reference__author">Duong Vi Khoi</li><li class="reference__author">Dahanukar Anupama Arun</li></ol><span class="reference__authors_list_suffix">2021</span><span class="reference__title">Dietary Macronutrient Imbalances Lead to Compensatory Changes in Peripheral Taste via Independent Signaling Pathways.</span><span class="reference__origin"><i>The Journal of neuroscience : the official journal of the Society for Neuroscience</i> <strong>41</strong>:10222‑10246</span><span class="reference__doi"><a href="https://doi.org/10.1523/JNEUROSCI.2154-20.2021" class="reference__doi_link">https://doi.org/10.1523/JNEUROSCI.2154-20.2021</a></span></li><li class="reference-list__item" id="c15"><span class="reference__label">15.</span><ol class="reference__authors_list"><li class="reference__author">Matty Molly A</li><li class="reference__author">Lau Hiu E</li><li class="reference__author">Haley Jessica A</li><li class="reference__author">Singh Anupama</li><li class="reference__author">Chakraborty Ahana</li><li class="reference__author">Kono Karina</li><li class="reference__author">Reddy Kirthi C</li><li class="reference__author">Hansen Malene</li><li class="reference__author">Chalasani Sreekanth H</li></ol><span class="reference__authors_list_suffix">2022</span><span class="reference__title">Intestine-to-neuronal signaling alters risk-taking behaviors in food-deprived Caenorhabditis elegans.</span><span class="reference__origin"><i>PLOS Genetics</i> <strong>18</strong>:e1010178</span><span class="reference__doi"><a href="https://doi.org/10.1371/JOURNAL.PGEN.1010178" class="reference__doi_link">https://doi.org/10.1371/JOURNAL.PGEN.1010178</a></span></li><li class="reference-list__item" id="c16"><span class="reference__label">16.</span><ol class="reference__authors_list"><li class="reference__author">Waterson Michael J</li><li class="reference__author">Horvath Tamas L</li></ol><span class="reference__authors_list_suffix">2015</span><span class="reference__title">Neuronal Regulation of Energy Homeostasis: Beyond the Hypothalamus and Feeding.</span><span class="reference__origin"><i>Cell Metabolism</i> <strong>22</strong>:962‑970</span><span class="reference__doi"><a href="https://doi.org/10.1016/J.CMET.2015.09.026" class="reference__doi_link">https://doi.org/10.1016/J.CMET.2015.09.026</a></span></li><li class="reference-list__item" id="c17"><span class="reference__label">17.</span><ol class="reference__authors_list"><li class="reference__author">LeDue Emily E</li><li class="reference__author">Mann Kevin</li><li class="reference__author">Koch Ellen</li><li class="reference__author">Chu Bonnie</li><li class="reference__author">Dakin Roslyn</li><li class="reference__author">Gordon Michael D</li></ol><span class="reference__authors_list_suffix">2016</span><span class="reference__title">Starvation-Induced Depotentiation of Bitter Taste in Drosophila.</span><span class="reference__origin"><i>Current biology : CB</i> <strong>26</strong>:2854‑2861</span><span class="reference__doi"><a href="https://doi.org/10.1016/J.CUB.2016.08.028" class="reference__doi_link">https://doi.org/10.1016/J.CUB.2016.08.028</a></span></li><li class="reference-list__item" id="c18"><span class="reference__label">18.</span><ol class="reference__authors_list"><li class="reference__author">Ko Kang I</li><li class="reference__author">Root Cory M</li><li class="reference__author">Lindsay Scott A</li><li class="reference__author">Zaninovich Orel A</li><li class="reference__author">Shepherd Andrew K</li><li class="reference__author">Wasserman Steven A</li><li class="reference__author">Kim Susy M</li><li class="reference__author">Wang Jing W</li></ol><span class="reference__authors_list_suffix">2015</span><span class="reference__title">Starvation promotes concerted modulation of appetitive olfactory behavior via parallel neuromodulatory circuits.</span><span class="reference__origin"><i>eLife</i> <strong>4</strong></span><span class="reference__doi"><a href="https://doi.org/10.7554/ELIFE.08298" class="reference__doi_link">https://doi.org/10.7554/ELIFE.08298</a></span></li><li class="reference-list__item" id="c19"><span class="reference__label">19.</span><ol class="reference__authors_list"><li class="reference__author">Root Cory M</li><li class="reference__author">Ko Kang I</li><li class="reference__author">Jafari Amir</li><li class="reference__author">Wang Jing W</li></ol><span class="reference__authors_list_suffix">2011</span><span class="reference__title">Presynaptic facilitation by neuropeptide signaling mediates odor-driven food search.</span><span class="reference__origin"><i>Cell</i> <strong>145</strong>:133‑144</span><span class="reference__doi"><a href="https://doi.org/10.1016/J.CELL.2011.02.008" class="reference__doi_link">https://doi.org/10.1016/J.CELL.2011.02.008</a></span></li><li class="reference-list__item" id="c20"><span class="reference__label">20.</span><ol class="reference__authors_list"><li class="reference__author">Horio Nao</li><li class="reference__author">Liberles Stephen D.</li></ol><span class="reference__authors_list_suffix">2021</span><span class="reference__title">Hunger enhances food-odour attraction through a neuropeptide Y spotlight.</span><span class="reference__origin"><i>Nature 2021 592:7853</i> <strong>592</strong>:262‑266</span><span class="reference__doi"><a href="https://doi.org/10.1038/s41586-021-03299-4" class="reference__doi_link">https://doi.org/10.1038/s41586-021-03299-4</a></span></li><li class="reference-list__item" id="c21"><span class="reference__label">21.</span><ol class="reference__authors_list"><li class="reference__author">Chen Yiming</li><li class="reference__author">Lin Yen Chu</li><li class="reference__author">Kuo Tzu Wei</li><li class="reference__author">Knight Zachary A.</li></ol><span class="reference__authors_list_suffix">2015</span><span class="reference__title">Sensory Detection of Food Rapidly Modulates Arcuate Feeding Circuits.</span><span class="reference__origin"><i>Cell</i> <strong>160</strong>:829‑841</span><span class="reference__doi"><a href="https://doi.org/10.1016/J.CELL.2015.01.033" class="reference__doi_link">https://doi.org/10.1016/J.CELL.2015.01.033</a></span></li><li class="reference-list__item" id="c22"><span class="reference__label">22.</span><ol class="reference__authors_list"><li class="reference__author">Fu Ou</li><li class="reference__author">Iwai Yuu</li><li class="reference__author">Narukawa Masataka</li><li class="reference__author">Ishikawa Ayako W.</li><li class="reference__author">Ishii Kentaro K.</li><li class="reference__author">Murata Ken</li><li class="reference__author">Yoshimura Yumiko</li><li class="reference__author">Touhara Kazushige</li><li class="reference__author">Misaka Takumi</li><li class="reference__author">Minokoshi Yasuhiko</li><li class="reference__author"><em>et al.</em></li></ol><span class="reference__authors_list_suffix">2019</span><span class="reference__title">Hypothalamic neuronal circuits regulating hunger-induced taste modification.</span><span class="reference__origin"><i>Nature Communications 2019 10:1</i> <strong>10</strong>:1‑14</span><span class="reference__doi"><a href="https://doi.org/10.1038/s41467-019-12478-x" class="reference__doi_link">https://doi.org/10.1038/s41467-019-12478-x</a></span></li><li class="reference-list__item" id="c23"><span class="reference__label">23.</span><ol class="reference__authors_list"><li class="reference__author">Andermann Mark L</li><li class="reference__author">Lowell Bradford B</li></ol><span class="reference__authors_list_suffix">2017</span><span class="reference__title">Toward a Wiring Diagram Understanding of Appetite Control.</span><span class="reference__origin"><i>Neuron</i> <strong>95</strong>:757‑778</span><span class="reference__doi"><a href="https://doi.org/10.1016/J.NEURON.2017.06.014" class="reference__doi_link">https://doi.org/10.1016/J.NEURON.2017.06.014</a></span></li><li class="reference-list__item" id="c24"><span class="reference__label">24.</span><ol class="reference__authors_list"><li class="reference__author">Wee Ryan W S</li><li class="reference__author">Mishchanchuk Karyna</li><li class="reference__author">AlSubaie Rawan</li><li class="reference__author">Church Timothy W</li><li class="reference__author">Gold Matthew G</li><li class="reference__author">MacAskill Andrew F</li></ol><span class="reference__authors_list_suffix">2024</span><span class="reference__title">Internal-state-dependent control of feeding behavior via hippocampal ghrelin signaling.</span><span class="reference__origin"><i>Neuron</i> <strong>112</strong>:288‑305.e7,</span><span class="reference__doi"><a href="https://doi.org/10.1016/J.NEURON.2023.10.016" class="reference__doi_link">https://doi.org/10.1016/J.NEURON.2023.10.016</a></span></li><li class="reference-list__item" id="c25"><span class="reference__label">25.</span><ol class="reference__authors_list"><li class="reference__author">Shiu Philip K.</li><li class="reference__author">Sterne Gabriella R.</li><li class="reference__author">Engert Stefanie</li><li class="reference__author">Dickson Barry J.</li><li class="reference__author">Scott Kristin</li></ol><span class="reference__authors_list_suffix">2022</span><span class="reference__title">Taste quality and hunger interactions in a feeding sensorimotor circuit.</span><span class="reference__origin"><i>eLife</i> <strong>11</strong></span><span class="reference__doi"><a href="https://doi.org/10.7554/ELIFE.79887" class="reference__doi_link">https://doi.org/10.7554/ELIFE.79887</a></span></li><li class="reference-list__item" id="c26"><span class="reference__label">26.</span><ol class="reference__authors_list"><li class="reference__author">Sayin Sercan</li><li class="reference__author">De Backer Jean Francois</li><li class="reference__author">Siju P P</li><li class="reference__author">Wosniack Marina E</li><li class="reference__author">Lewis Laurence P</li><li class="reference__author">Frisch Lisa Marie</li><li class="reference__author">Gansen Benedikt</li><li class="reference__author">Schlegel Philipp</li><li class="reference__author">Edmondson-Stait Amelia</li><li class="reference__author">Sharifi Nadiya</li><li class="reference__author"><em>et al.</em></li></ol><span class="reference__authors_list_suffix">2019</span><span class="reference__title">A Neural Circuit Arbitrates between Persistence and Withdrawal in Hungry Drosophila.</span><span class="reference__origin"><i>Neuron</i> <strong>104</strong>:544‑558.e6,</span><span class="reference__doi"><a href="https://doi.org/10.1016/J.NEURON.2019.07.028" class="reference__doi_link">https://doi.org/10.1016/J.NEURON.2019.07.028</a></span></li><li class="reference-list__item" id="c27"><span class="reference__label">27.</span><ol class="reference__authors_list"><li class="reference__author">Tsao Chang Hui</li><li class="reference__author">Chen Chien Chun</li><li class="reference__author">Lin Chen Han</li><li class="reference__author">Yang Hao Yu</li><li class="reference__author">Lin Suewei</li></ol><span class="reference__authors_list_suffix">2018</span><span class="reference__title">Drosophila mushroom bodies integrate hunger and satiety signals to control innate food-seeking behavior.</span><span class="reference__origin"><i>eLife</i> <strong>7</strong></span><span class="reference__doi"><a href="https://doi.org/10.7554/ELIFE.35264" class="reference__doi_link">https://doi.org/10.7554/ELIFE.35264</a></span></li><li class="reference-list__item" id="c28"><span class="reference__label">28.</span><ol class="reference__authors_list"><li class="reference__author">Sareen Preeti F.</li><li class="reference__author">McCurdy Li Yan</li><li class="reference__author">Nitabach Michael N.</li></ol><span class="reference__authors_list_suffix">2021</span><span class="reference__title">A neuronal ensemble encoding adaptive choice during sensory conflict in Drosophila.</span><span class="reference__origin"><i>Nature communications</i> <strong>12</strong></span><span class="reference__doi"><a href="https://doi.org/10.1038/S41467-021-24423-Y" class="reference__doi_link">https://doi.org/10.1038/S41467-021-24423-Y</a></span></li><li class="reference-list__item" id="c29"><span class="reference__label">29.</span><ol class="reference__authors_list"><li class="reference__author">Park Jin Yong</li><li class="reference__author">Dus Monica</li><li class="reference__author">Kim Seonil</li><li class="reference__author">Abu Farhan</li><li class="reference__author">Kanai Makoto I</li><li class="reference__author">Rudy Bernardo</li><li class="reference__author">Suh Greg S B</li></ol><span class="reference__authors_list_suffix">2016</span><span class="reference__title">Drosophila SLC5A11 Mediates Hunger by Regulating K(+) Channel Activity.</span><span class="reference__origin"><i>Current biology : CB</i> <strong>26</strong>:1965‑1974</span><span class="reference__doi"><a href="https://doi.org/10.1016/J.CUB.2016.05.076" class="reference__doi_link">https://doi.org/10.1016/J.CUB.2016.05.076</a></span></li><li class="reference-list__item" id="c30"><span class="reference__label">30.</span><ol class="reference__authors_list"><li class="reference__author">Musso Pierre Yves</li><li class="reference__author">Junca Pierre</li><li class="reference__author">Gordon Michael D.</li></ol><span class="reference__authors_list_suffix">2021</span><span class="reference__title">A neural circuit linking two sugar sensors regulates satiety-dependent fructose drive in Drosophila.</span><span class="reference__origin"><i>Science Advances</i> <strong>7</strong>:186</span><span class="reference__doi"><a href="https://doi.org/10.1126/SCIADV.ABJ0186" class="reference__doi_link">https://doi.org/10.1126/SCIADV.ABJ0186</a></span></li><li class="reference-list__item" id="c31"><span class="reference__label">31.</span><ol class="reference__authors_list"><li class="reference__author">Goldschmidt Dennis</li><li class="reference__author">Tastekin Ibrahim</li><li class="reference__author">Münch Daniel</li><li class="reference__author">Park Jin-Yong</li><li class="reference__author">Haberkern Hannah</li><li class="reference__author">Serra Lúcia</li><li class="reference__author">Baltazar Célia</li><li class="reference__author">Jayaraman Vivek</li><li class="reference__author">Rubin Gerald M.</li><li class="reference__author">Ribeiro Carlos</li></ol><span class="reference__authors_list_suffix">2023</span><span class="reference__title">A neuronal substrate for translating nutrient state and resource density estimations into foraging decisions.</span><span class="reference__origin"><i>bioRxiv</i> 2023.07.19.549514</span><span class="reference__doi"><a href="https://doi.org/10.1101/2023.07.19.549514" class="reference__doi_link">https://doi.org/10.1101/2023.07.19.549514</a></span></li><li class="reference-list__item" id="c32"><span class="reference__label">32.</span><ol class="reference__authors_list"><li class="reference__author">Liu Qili</li><li class="reference__author">Tabuchi Masashi</li><li class="reference__author">Liu Sha</li><li class="reference__author">Kodama Lay</li><li class="reference__author">Horiuchi Wakako</li><li class="reference__author">Daniels Jay</li><li class="reference__author">Chiu Lucinda</li><li class="reference__author">Baldoni Daniel</li><li class="reference__author">Wu Mark N.</li></ol><span class="reference__authors_list_suffix">2017</span><span class="reference__title">Branch-specific plasticity of a bifunctional dopamine circuit encodes protein hunger.</span><span class="reference__origin"><i>Science</i> <strong>356</strong>:534‑539</span><span class="reference__doi"><a href="https://doi.org/10.1126/SCIENCE.AAL3245" class="reference__doi_link">https://doi.org/10.1126/SCIENCE.AAL3245</a></span></li><li class="reference-list__item" id="c33"><span class="reference__label">33.</span><ol class="reference__authors_list"><li class="reference__author">Münch Daniel</li><li class="reference__author">Goldschmidt Dennis</li><li class="reference__author">Ribeiro Carlos</li></ol><span class="reference__authors_list_suffix">2022</span><span class="reference__title">The neuronal logic of how internal states control food choice.</span><span class="reference__origin"><i>Nature 2022 607:7920</i> <strong>607</strong>:747‑755</span><span class="reference__doi"><a href="https://doi.org/10.1038/s41586-022-04909-5" class="reference__doi_link">https://doi.org/10.1038/s41586-022-04909-5</a></span></li><li class="reference-list__item" id="c34"><span class="reference__label">34.</span><ol class="reference__authors_list"><li class="reference__author">Takken W.</li><li class="reference__author">Klowden M. J.</li><li class="reference__author">Chambers G. M.</li></ol><span class="reference__authors_list_suffix">1998</span><span class="reference__title">Effect of body size on host seeking and blood meal utilization in anopheles gambiae sensu stricto (diptera: Culicidae): The disadvantage of being small.</span><span class="reference__origin"><i>Journal of Medical Entomology</i> <strong>35</strong>:639‑645</span><span class="reference__doi"><a href="https://doi.org/10.1093/JMEDENT/35.5.639" class="reference__doi_link">https://doi.org/10.1093/JMEDENT/35.5.639</a></span></li><li class="reference-list__item" id="c35"><span class="reference__label">35.</span><ol class="reference__authors_list"><li class="reference__author">Farjana Thahsin</li><li class="reference__author">Tuno Nobuko</li></ol><span class="reference__authors_list_suffix">2013</span><span class="reference__title">Multiple blood feeding and host-seeking behavior in aedes aegypti and aedes albopictus (diptera: Culicidae).</span><span class="reference__origin"><i>Journal of Medical Entomology</i> <strong>50</strong>:838‑846</span><span class="reference__doi"><a href="https://doi.org/10.1603/ME12146" class="reference__doi_link">https://doi.org/10.1603/ME12146</a></span></li><li class="reference-list__item" id="c36"><span class="reference__label">36.</span><ol class="reference__authors_list"><li class="reference__author">Scott Thomas W.</li><li class="reference__author">Takken Willem</li></ol><span class="reference__authors_list_suffix">2012</span><span class="reference__title">Feeding strategies of anthropophilic mosquitoes result in increased risk of pathogen transmission.</span><span class="reference__origin"><i>Trends in Parasitology</i> <strong>28</strong>:114‑121</span><span class="reference__doi"><a href="https://doi.org/10.1016/j.pt.2012.01.001" class="reference__doi_link">https://doi.org/10.1016/j.pt.2012.01.001</a></span></li><li class="reference-list__item" id="c37"><span class="reference__label">37.</span><ol class="reference__authors_list"><li class="reference__author">Ramasamy M S</li><li class="reference__author">Srikrishnaraj K A</li><li class="reference__author">Hadjirin N</li><li class="reference__author">Perera S</li><li class="reference__author">Ramasamy R</li></ol><span class="reference__authors_list_suffix">2000</span><span class="reference__title">Physiological aspects of multiple blood feeding in the malaria vector anopheles tessellatus.</span><span class="reference__origin"><i>Journal of Insect Physiology</i> <strong>46</strong></span><span class="reference__doi"></span></li><li class="reference-list__item" id="c38"><span class="reference__label">38.</span><ol class="reference__authors_list"><li class="reference__author">League Garrett P.</li><li class="reference__author">Degner Ethan C.</li><li class="reference__author">Pitcher Sylvie A.</li><li class="reference__author">Hafezi Yassi</li><li class="reference__author">Tennant Erica</li><li class="reference__author">Cruz Priscilla C.</li><li class="reference__author">Krishnan Raksha S.</li><li class="reference__author">Castillo Stefano S.Garcia</li><li class="reference__author">Alfonso-Parra Catalina</li><li class="reference__author">Avila Frank W.</li><li class="reference__author"><em>et al.</em></li></ol><span class="reference__authors_list_suffix">2021</span><span class="reference__title">The impact of mating and sugar feeding on blood-feeding physiology and behavior in the arbovirus vector mosquito aedes aegypti.</span><span class="reference__origin"><i>PLoS Neglected Tropical Diseases</i> <strong>15</strong></span><span class="reference__doi"><a href="https://doi.org/10.1371/journal.pntd.0009815" class="reference__doi_link">https://doi.org/10.1371/journal.pntd.0009815</a></span></li><li class="reference-list__item" id="c39"><span class="reference__label">39.</span><ol class="reference__authors_list"><li class="reference__author">Charlwood J Derek</li><li class="reference__author">Pinto Joao</li><li class="reference__author">Sousa Carla A</li><li class="reference__author">Ferreira Conceicao</li><li class="reference__author">Petrarca Vincenso</li><li class="reference__author">Rosario Virgilio do E</li></ol><span class="reference__authors_list_suffix">2003</span><span class="reference__title">’A mate or a meal’ – Pre-gravid behaviour of female Anopheles gambiae from the islands of São Tomé and Príncipe, West Africa.</span><span class="reference__origin"><i>Malaria Journal</i> <strong>2</strong>:9</span><span class="reference__doi"><a href="https://doi.org/10.1186/1475-2875-2-9" class="reference__doi_link">https://doi.org/10.1186/1475-2875-2-9</a></span></li><li class="reference-list__item" id="c40"><span class="reference__label">40.</span><ol class="reference__authors_list"><li class="reference__author">Ye Zi</li><li class="reference__author">Liu Feng</li><li class="reference__author">Sun Huahua</li><li class="reference__author">Ferguson Stephen T</li><li class="reference__author">Baker Adam</li><li class="reference__author">Ochieng Samuel A</li><li class="reference__author">Zwiebel Laurence J</li></ol><span class="reference__authors_list_suffix">2022</span><span class="reference__title">Discrete roles of Ir76b ionotropic coreceptor impact olfaction, blood feeding, and mating in the malaria vector mosquito Anopheles coluzzii.</span><span class="reference__origin"><i>Proceedings of the National Academy of Sciences of the United States of America</i> <strong>119</strong>:e2112385119</span><span class="reference__doi"><a href="https://doi.org/10.1073/PNAS.2112385119" class="reference__doi_link">https://doi.org/10.1073/PNAS.2112385119</a></span></li><li class="reference-list__item" id="c41"><span class="reference__label">41.</span><ol class="reference__authors_list"><li class="reference__author">Stone C M</li><li class="reference__author">Hamilton I M</li><li class="reference__author">Foster W A</li></ol><span class="reference__authors_list_suffix">2011</span><span class="reference__title">A survival and reproduction trade-off is resolved in accordance with resource availability by virgin female mosquitoes.</span><span class="reference__origin"><i>Animal Behaviour</i> <strong>81</strong>:765‑774</span><span class="reference__doi"><a href="https://doi.org/10.1016/j.anbehav.2011.01.008" class="reference__doi_link">https://doi.org/10.1016/j.anbehav.2011.01.008</a></span></li><li class="reference-list__item" id="c42"><span class="reference__label">42.</span><ol class="reference__authors_list"><li class="reference__author">Davis Edward E</li></ol><span class="reference__authors_list_suffix">1984</span><span class="reference__title">Development of lactic acid-receptor sensitivity and host-seeking behaviour in newly emerged female Aedes aegypti mosquitoes.</span><span class="reference__origin"><i>Journal of Insect Physiology</i> <strong>30</strong>:211‑215</span><span class="reference__doi"><a href="https://doi.org/10.1016/0022-1910(84)90005-2" class="reference__doi_link">https://doi.org/10.1016/0022-1910(84)90005-2</a></span></li><li class="reference-list__item" id="c43"><span class="reference__label">43.</span><ol class="reference__authors_list"><li class="reference__author">Foster W A</li><li class="reference__author">Takken W</li></ol><span class="reference__authors_list_suffix">2004</span><span class="reference__title">Nectar-related vs. human-related volatiles: behavioural response and choice by female and male Anopheles gambiae (Diptera: Culicidae) between emergence and first feeding.</span><span class="reference__origin"><i>Bulletin of Entomological Research</i> <strong>94</strong>:145‑157</span><span class="reference__doi"><a href="https://doi.org/10.1079/ber2003288" class="reference__doi_link">https://doi.org/10.1079/ber2003288</a></span></li><li class="reference-list__item" id="c44"><span class="reference__label">44.</span><ol class="reference__authors_list"><li class="reference__author">Felipe Ramírez-Sánchez Luis</li><li class="reference__author">Juliana Brenda</li><li class="reference__author">Hernández Pablo Andrés Guzmán</li><li class="reference__author">Alfonso-Parra Catalina</li><li class="reference__author">Avila Frank W.</li></ol><span class="reference__authors_list_suffix">2023</span><span class="reference__title">The effects of female age on blood-feeding, insemination, sperm storage, and fertility in the dengue vector mosquito aedes aegypti (diptera: Culicidae).</span><span class="reference__origin"><i>Journal of Insect Physiology</i> <strong>150</strong></span><span class="reference__doi"><a href="https://doi.org/10.1016/j.jinsphys.2023.104570" class="reference__doi_link">https://doi.org/10.1016/j.jinsphys.2023.104570</a></span></li><li class="reference-list__item" id="c45"><span class="reference__label">45.</span><ol class="reference__authors_list"><li class="reference__author">Klowden M J</li><li class="reference__author">Lea A O</li></ol><span class="reference__authors_list_suffix">1979</span><span class="reference__title">Humoral inhibition of host-seeking in Aedes aegypti during oöcyte maturation.</span><span class="reference__origin"><i>Journal of Insect Physiology</i> <strong>25</strong>:231‑235</span><span class="reference__doi"><a href="https://doi.org/10.1016/0022-1910(79)90048-9" class="reference__doi_link">https://doi.org/10.1016/0022-1910(79)90048-9</a></span></li><li class="reference-list__item" id="c46"><span class="reference__label">46.</span><ol class="reference__authors_list"><li class="reference__author">Klowden M J</li><li class="reference__author">Lea A O</li></ol><span class="reference__authors_list_suffix">1979</span><span class="reference__title">Abdominal distention terminates subsequent host-seeking behaviour of Aedes aegypti following a blood meal.</span><span class="reference__origin"><i>Journal of Insect Physiology</i> <strong>25</strong>:583‑585</span><span class="reference__doi"><a href="https://doi.org/10.1016/0022-1910(79)90073-8" class="reference__doi_link">https://doi.org/10.1016/0022-1910(79)90073-8</a></span></li><li class="reference-list__item" id="c47"><span class="reference__label">47.</span><ol class="reference__authors_list"><li class="reference__author">Duvall Laura B</li><li class="reference__author">Ramos-Espiritu Lavoisier</li><li class="reference__author">Barsoum Kyrollos E</li><li class="reference__author">Glickman J Fraser</li><li class="reference__author">Vosshall Leslie B</li></ol><span class="reference__authors_list_suffix">2019</span><span class="reference__title">Small-Molecule Agonists of Ae. aegypti Neuropeptide Y Receptor Block Mosquito Biting.</span><span class="reference__origin"><i>Cell</i> <strong>176</strong>:687‑701.e5,</span><span class="reference__doi"><a href="https://doi.org/10.1016/J.CELL.2018.12.004" class="reference__doi_link">https://doi.org/10.1016/J.CELL.2018.12.004</a></span></li><li class="reference-list__item" id="c48"><span class="reference__label">48.</span><ol class="reference__authors_list"><li class="reference__author">Klowden Marc J</li><li class="reference__author">Briegel Hans</li></ol><span class="reference__authors_list_suffix">1994</span><span class="reference__title">Mosquito gonotrophic cycle and multiple feeding potential: Contrasts between anopheles and aedes (diptera: Culicidae).</span><span class="reference__origin"><i>Med Entomol</i> <strong>31</strong>:618‑22</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c49"><span class="reference__label">49.</span><ol class="reference__authors_list"><li class="reference__author">Takken W</li><li class="reference__author">null van Loon JJ</li><li class="reference__author">Adam W</li></ol><span class="reference__authors_list_suffix">2001</span><span class="reference__title">Inhibition of host-seeking response and olfactory responsiveness in Anopheles gambiae following blood feeding.</span><span class="reference__origin"><i>Journal of Insect Physiology</i> <strong>47</strong>:303‑310</span><span class="reference__doi"><a href="https://doi.org/10.1016/s0022-1910(00)00107-4" class="reference__doi_link">https://doi.org/10.1016/s0022-1910(00)00107-4</a></span></li><li class="reference-list__item" id="c50"><span class="reference__label">50.</span><ol class="reference__authors_list"><li class="reference__author">Foster W A</li></ol><span class="reference__authors_list_suffix">1995</span><span class="reference__title">Mosquito sugar feeding and reproductive energetics.</span><span class="reference__origin"><i>Annual Review of Entomology</i> <strong>40</strong>:443‑474</span><span class="reference__doi"><a href="https://doi.org/10.1146/annurev.en.40.010195.002303" class="reference__doi_link">https://doi.org/10.1146/annurev.en.40.010195.002303</a></span></li><li class="reference-list__item" id="c51"><span class="reference__label">51.</span><ol class="reference__authors_list"><li class="reference__author">Dittmer Jessica</li><li class="reference__author">Alafndi Ayad</li><li class="reference__author">Gabrieli Paolo</li></ol><span class="reference__authors_list_suffix">2019</span><span class="reference__title">Fat body-specific vitellogenin expression regulates host-seeking behaviour in the mosquito Aedes albopictus.</span><span class="reference__origin"><i>PLoS biology</i> <strong>17</strong></span><span class="reference__doi"><a href="https://doi.org/10.1371/JOURNAL.PBIO.3000238" class="reference__doi_link">https://doi.org/10.1371/JOURNAL.PBIO.3000238</a></span></li><li class="reference-list__item" id="c52"><span class="reference__label">52.</span><ol class="reference__authors_list"><li class="reference__author">Liesch Jeff</li><li class="reference__author">Bellani Lindsay L</li><li class="reference__author">Vosshall Leslie B</li></ol><span class="reference__authors_list_suffix">2013</span><span class="reference__title">Functional and Genetic Characterization of Neuropeptide Y-Like Receptors in Aedes aegypti.</span><span class="reference__origin"><i>PLOS Neglected Tropical Diseases</i> <strong>7</strong>:e2486</span><span class="reference__doi"><a href="https://doi.org/10.1371/journal.pntd.0002486" class="reference__doi_link">https://doi.org/10.1371/journal.pntd.0002486</a></span></li><li class="reference-list__item" id="c53"><span class="reference__label">53.</span><ol class="reference__authors_list"><li class="reference__author">Christ Peter</li><li class="reference__author">Reifenrath Anna</li><li class="reference__author">Kahnt Jörg</li><li class="reference__author">Hauser Frank</li><li class="reference__author">Hill Sharon Rose</li><li class="reference__author">Schachtner Joachim</li><li class="reference__author">Ignell Rickard</li></ol><span class="reference__authors_list_suffix">2017</span><span class="reference__title">Feeding-induced changes in allatostatin-a and short neuropeptide f in the antennal lobes affect odor-mediated host seeking in the yellow fever mosquito, aedes aegypti.</span><span class="reference__origin"><i>PLOS One</i> <strong>12</strong>:e0188243</span><span class="reference__doi"><a href="https://doi.org/10.1371/JOURNAL.PONE.0188243" class="reference__doi_link">https://doi.org/10.1371/JOURNAL.PONE.0188243</a></span></li><li class="reference-list__item" id="c54"><span class="reference__label">54.</span><ol class="reference__authors_list"><li class="reference__author">Zeledon Emely V.</li><li class="reference__author">Baxt Leigh A.</li><li class="reference__author">Khan Tanweer A.</li><li class="reference__author">Michino Mayako</li><li class="reference__author">Miller Michael</li><li class="reference__author">Huggins David J.</li><li class="reference__author">Jiang Caroline S.</li><li class="reference__author">Vosshall Leslie B.</li><li class="reference__author">Duvall Laura B.</li></ol><span class="reference__authors_list_suffix">2024</span><span class="reference__title">Next Generation Neuropeptide Y Receptor Small Molecule Agonists Inhibit Mosquito Biting Behavior.</span><span class="reference__origin"><i>bioRxiv</i> 2024.02.28.582529</span><span class="reference__doi"><a href="https://doi.org/10.1101/2024.02.28.582529" class="reference__doi_link">https://doi.org/10.1101/2024.02.28.582529</a></span></li><li class="reference-list__item" id="c55"><span class="reference__label">55.</span><ol class="reference__authors_list"><li class="reference__author">Dou Xiaoyi</li><li class="reference__author">Chen Kangkang</li><li class="reference__author">Brown Mark R.</li><li class="reference__author">Strand Michael R.</li></ol><span class="reference__authors_list_suffix">2024</span><span class="reference__title">Reciprocal interactions between neuropeptide f and ryamide regulate host attraction in the mosquito aedes aegypti.</span><span class="reference__origin"><i>Proceedings of the National Academy of Sciences of the United States of America</i> <strong>121</strong>:e2408072121</span><span class="reference__doi"><a href="https://doi.org/10.1073/PNAS.2408072121" class="reference__doi_link">https://doi.org/10.1073/PNAS.2408072121</a></span></li><li class="reference-list__item" id="c56"><span class="reference__label">56.</span><ol class="reference__authors_list"><li class="reference__author">Galun Rachel</li><li class="reference__author">Avi-Dor Y.</li><li class="reference__author">Bar-Zeev M.</li></ol><span class="reference__authors_list_suffix">1963</span><span class="reference__title">Feeding response in aedes aegypti: Stimulation by adenosine triphosphate.</span><span class="reference__origin"><i>Science</i> <strong>142</strong>:1674‑1675</span><span class="reference__doi"><a href="https://doi.org/10.1126/SCIENCE.142.3600.1674" class="reference__doi_link">https://doi.org/10.1126/SCIENCE.142.3600.1674</a></span></li><li class="reference-list__item" id="c57"><span class="reference__label">57.</span><ol class="reference__authors_list"><li class="reference__author">Sakuma Chisako</li><li class="reference__author">Iwamoto Takeo</li><li class="reference__author">Masuda Keiko</li><li class="reference__author">Shimizu Yoshihiro</li><li class="reference__author">Obata Fumiaki</li><li class="reference__author">Kanuka Hirotaka</li></ol><span class="reference__authors_list_suffix">2024</span><span class="reference__title">Fibrinopeptide a-induced blood-feeding arrest in the yellow fever mosquito aedes aegypti.</span><span class="reference__origin"><i>Cell Reports</i> <strong>43</strong></span><span class="reference__doi"><a href="https://doi.org/10.1016/J.CELREP.2024.114354" class="reference__doi_link">https://doi.org/10.1016/J.CELREP.2024.114354</a></span></li><li class="reference-list__item" id="c58"><span class="reference__label">58.</span><ol class="reference__authors_list"><li class="reference__author">Sinka M. E.</li><li class="reference__author">Pironon S.</li><li class="reference__author">Massey N. C.</li><li class="reference__author">Longbottom J.</li><li class="reference__author">Hemingway J.</li><li class="reference__author">Moyes C. L.</li><li class="reference__author">Willis K. J.</li></ol><span class="reference__authors_list_suffix">2020</span><span class="reference__title">A new malaria vector in africa: Predicting the expansion range of anopheles stephensi and identifying the urban populations at risk.</span><span class="reference__origin"><i>Proceedings of the National Academy of Sciences of the United States of America</i> <strong>117</strong>:24900‑24908</span><span class="reference__doi"><a href="https://doi.org/10.1073/PNAS.2003976117" class="reference__doi_link">https://doi.org/10.1073/PNAS.2003976117</a></span></li><li class="reference-list__item" id="c59"><span class="reference__label">59.</span><ol class="reference__authors_list"><li class="reference__author">Venkataraman Krithika</li><li class="reference__author">Shai Nadav</li><li class="reference__author">Lakhiani Priyanka</li><li class="reference__author">Zylka Sarah</li><li class="reference__author">Zhao Jieqing</li><li class="reference__author">Herre Margaret</li><li class="reference__author">Zeng Joshua</li><li class="reference__author">Neal Lauren A.</li><li class="reference__author">Molina Henrik</li><li class="reference__author">Zhao Li</li><li class="reference__author"><em>et al.</em></li></ol><span class="reference__authors_list_suffix">2023</span><span class="reference__title">Two novel, tightly linked, and rapidly evolving genes underlie aedes aegypti mosquito reproductive resilience during drought.</span><span class="reference__origin"><i>eLife</i> <strong>12</strong></span><span class="reference__doi"><a href="https://doi.org/10.7554/ELIFE.80489" class="reference__doi_link">https://doi.org/10.7554/ELIFE.80489</a></span></li><li class="reference-list__item" id="c60"><span class="reference__label">60.</span><ol class="reference__authors_list"><li class="reference__author">Villarreal Susan M.</li><li class="reference__author">Pitcher Sylvie</li><li class="reference__author">Helinski Michelle E.H.</li><li class="reference__author">Johnson Lynn</li><li class="reference__author">Wolfner Mariana F.</li><li class="reference__author">Harrington Laura C.</li></ol><span class="reference__authors_list_suffix">2018</span><span class="reference__title">Male contributions during mating increase female survival in the disease vector mosquito aedes aegypti.</span><span class="reference__origin"><i>Journal of Insect Physiology</i> <strong>108</strong>:1‑9</span><span class="reference__doi"><a href="https://doi.org/10.1016/J.JINSPHYS.2018.05.001" class="reference__doi_link">https://doi.org/10.1016/J.JINSPHYS.2018.05.001</a></span></li><li class="reference-list__item" id="c61"><span class="reference__label">61.</span><ol class="reference__authors_list"><li class="reference__author">Adlakha Vimla</li><li class="reference__author">Pillai M. K.K.</li></ol><span class="reference__authors_list_suffix">1976</span><span class="reference__title">Rôle of male accessory gland substance in the regulation of blood intake by mosquitoes.</span><span class="reference__origin"><i>Journal of Insect Physiology</i> <strong>22</strong>:1441‑1442</span><span class="reference__doi"><a href="https://doi.org/10.1016/0022-1910(76)90206-7" class="reference__doi_link">https://doi.org/10.1016/0022-1910(76)90206-7</a></span></li><li class="reference-list__item" id="c62"><span class="reference__label">62.</span><ol class="reference__authors_list"><li class="reference__author">Duvall Laura B</li><li class="reference__author">Basrur Nipun S</li><li class="reference__author">Molina Henrik</li><li class="reference__author">McMeniman Conor J</li><li class="reference__author">Vosshall Leslie B</li></ol><span class="reference__authors_list_suffix">2017</span><span class="reference__title">A Peptide Signaling System that Rapidly Enforces Paternity in the Aedes aegypti Mosquito.</span><span class="reference__origin"><i>Current biology : CB</i> <strong>27</strong>:3734‑3742.e5</span><span class="reference__doi"><a href="https://doi.org/10.1016/j.cub.2017.10.074" class="reference__doi_link">https://doi.org/10.1016/j.cub.2017.10.074</a></span></li><li class="reference-list__item" id="c63"><span class="reference__label">63.</span><ol class="reference__authors_list"><li class="reference__author">Grant A J</li><li class="reference__author">O’Connell R J</li></ol><span class="reference__authors_list_suffix">1996</span><span class="reference__title">Electrophysiological responses from receptor neurons in mosquito maxillary palp sensilla.</span><span class="reference__origin"><i>Ciba Foundation Symposium</i> <strong>200</strong>:233‑253</span><span class="reference__doi"><a href="https://doi.org/10.1002/9780470514948.ch17" class="reference__doi_link">https://doi.org/10.1002/9780470514948.ch17</a></span></li><li class="reference-list__item" id="c64"><span class="reference__label">64.</span><ol class="reference__authors_list"><li class="reference__author">Omondi A B</li><li class="reference__author">Ghaninia M</li><li class="reference__author">Dawit M</li><li class="reference__author">Svensson T</li><li class="reference__author">Ignell R</li></ol><span class="reference__authors_list_suffix">2019</span><span class="reference__title">Age-dependent regulation of host seeking in Anopheles coluzzii.</span><span class="reference__origin"><i>Scientific Reports</i> <strong>9</strong>:9699</span><span class="reference__doi"><a href="https://doi.org/10.1038/s41598-019-46220-w" class="reference__doi_link">https://doi.org/10.1038/s41598-019-46220-w</a></span></li><li class="reference-list__item" id="c65"><span class="reference__label">65.</span><ol class="reference__authors_list"><li class="reference__author">Hill Sharon Rose</li><li class="reference__author">Taparia Tanvi</li><li class="reference__author">Ignell Rickard</li></ol><span class="reference__authors_list_suffix">2021</span><span class="reference__title">Regulation of the antennal transcriptome of the dengue vector, Aedes aegypti, during the first gonotrophic cycle.</span><span class="reference__origin"><i>BMC genomics</i> <strong>22</strong>:71</span><span class="reference__doi"><a href="https://doi.org/10.1186/s12864-020-07336-w" class="reference__doi_link">https://doi.org/10.1186/s12864-020-07336-w</a></span></li><li class="reference-list__item" id="c66"><span class="reference__label">66.</span><ol class="reference__authors_list"><li class="reference__author">Tallon Anaïs Karine</li><li class="reference__author">Hill Sharon Rose</li><li class="reference__author">Ignell Rickard</li></ol><span class="reference__authors_list_suffix">2019</span><span class="reference__title">Sex and age modulate antennal chemosensory-related genes linked to the onset of host seeking in the yellow-fever mosquito, Aedes aegypti.</span><span class="reference__origin"><i>Scientific Reports</i> <strong>9</strong>:43</span><span class="reference__doi"><a href="https://doi.org/10.1038/s41598-018-36550-6" class="reference__doi_link">https://doi.org/10.1038/s41598-018-36550-6</a></span></li><li class="reference-list__item" id="c67"><span class="reference__label">67.</span><ol class="reference__authors_list"><li class="reference__author">Davis Edward E.</li></ol><span class="reference__authors_list_suffix">1984</span><span class="reference__title">Regulation of sensitivity in the peripheral chemoreceptor systems for host-seeking behaviour by a haemolymph-borne factor in Aedes aegypti.</span><span class="reference__origin"><i>Journal of Insect Physiology</i> <strong>30</strong>:179‑183</span><span class="reference__doi"><a href="https://doi.org/10.1016/0022-1910(84)90124-0" class="reference__doi_link">https://doi.org/10.1016/0022-1910(84)90124-0</a></span></li><li class="reference-list__item" id="c68"><span class="reference__label">68.</span><ol class="reference__authors_list"><li class="reference__author">Fox A.N.</li><li class="reference__author">Pitts R. J.</li><li class="reference__author">Robertson H. M.</li><li class="reference__author">Carlson J. R.</li><li class="reference__author">Zwiebel L. J.</li></ol><span class="reference__authors_list_suffix">2001</span><span class="reference__title">Candidate odorant receptors from the malaria vector mosquito anopheles gambiae and evidence of down-regulation in response to blood feeding.</span><span class="reference__origin"><i>Proceedings of the National Academy of Sciences of the United States of America</i> <strong>98</strong>:14693‑14697</span><span class="reference__doi"><a href="https://doi.org/10.1073/PNAS.261432998/ASSET/706A804C-7877-46A3-BE54-2858902574A7/ASSETS/GRAPHIC/PQ2614329004.JPEG" class="reference__doi_link">https://doi.org/10.1073/PNAS.261432998/ASSET/706A804C-7877-46A3-BE54-2858902574A7/ASSETS/GRAPHIC/PQ2614329004.JPEG</a></span></li><li class="reference-list__item" id="c69"><span class="reference__label">69.</span><ol class="reference__authors_list"><li class="reference__author">Chakraborty Mahul</li><li class="reference__author">Ramaiah Arunachalam</li><li class="reference__author">Adolfi Adriana</li><li class="reference__author">Halas Paige</li><li class="reference__author">Kaduskar Bhagyashree</li><li class="reference__author">Ngo Luna Thanh</li><li class="reference__author">Jayaprasad Suvratha</li><li class="reference__author">Paul Kiran</li><li class="reference__author">Whadgar Saurabh</li><li class="reference__author">Srinivasan Subhashini</li><li class="reference__author"><em>et al.</em></li></ol><span class="reference__authors_list_suffix">2021</span><span class="reference__title">Hidden genomic features of an invasive malaria vector, Anopheles stephensi, revealed by a chromosome-level genome assembly.</span><span class="reference__origin"><i>BMC biology</i> <strong>19</strong></span><span class="reference__doi"><a href="https://doi.org/10.1186/S12915-021-00963-Z" class="reference__doi_link">https://doi.org/10.1186/S12915-021-00963-Z</a></span></li><li class="reference-list__item" id="c70"><span class="reference__label">70.</span><ol class="reference__authors_list"><li class="reference__author">Krüger Eileen</li><li class="reference__author">Mena Wilson</li><li class="reference__author">Lahr Eleanor C</li><li class="reference__author">Johnson Erik C</li><li class="reference__author">Ewer John</li></ol><span class="reference__authors_list_suffix">2015</span><span class="reference__title">Genetic analysis of Eclosion hormone action during Drosophila larval ecdysis.</span><span class="reference__origin"><i>Development (Cambridge)</i> <strong>142</strong>:4279‑4287</span><span class="reference__doi"><a href="https://doi.org/10.1242/DEV.126995/256923/AM/GENETIC-ANALYSIS-OF-ECLOSION-HORMONE-ACTION-DURING" class="reference__doi_link">https://doi.org/10.1242/DEV.126995/256923/AM/GENETIC-ANALYSIS-OF-ECLOSION-HORMONE-ACTION-DURING</a></span></li><li class="reference-list__item" id="c71"><span class="reference__label">71.</span><ol class="reference__authors_list"><li class="reference__author">Luo Ching Wei</li><li class="reference__author">Dewey Elizabeth M.</li><li class="reference__author">Sudo Satoko</li><li class="reference__author">Ewer John</li><li class="reference__author">Hsu Sheau Yu</li><li class="reference__author">Honegger Hans Willi</li><li class="reference__author">Hsueh Aaron J.W.</li></ol><span class="reference__authors_list_suffix">2005</span><span class="reference__title">Bursicon, the insect cuticle-hardening hormone, is a heterodimeric cystine knot protein that activates g protein-coupled receptor lgr2.</span><span class="reference__origin"><i>Proceedings of the National Academy of Sciences of the United States of America</i> <strong>102</strong>:2820‑2825</span><span class="reference__doi"><a href="https://doi.org/10.1073/PNAS.0409916102" class="reference__doi_link">https://doi.org/10.1073/PNAS.0409916102</a></span></li><li class="reference-list__item" id="c72"><span class="reference__label">72.</span><ol class="reference__authors_list"><li class="reference__author">Hall Andrew Brantley</li><li class="reference__author">Qi Yumin</li><li class="reference__author">Timoshevskiy Vladimir</li><li class="reference__author">Sharakhova Maria V</li><li class="reference__author">Sharakhov Igor V</li><li class="reference__author">Tu Zhijian</li></ol><span class="reference__authors_list_suffix">2013</span><span class="reference__title">Six novel Y chromosome genes in Anopheles mosquitoes discovered by independently sequencing males and females.</span><span class="reference__origin"><i>BMC Genomics</i> <strong>14</strong>:1‑13</span><span class="reference__doi"><a href="https://doi.org/10.1186/1471-2164-14-273" class="reference__doi_link">https://doi.org/10.1186/1471-2164-14-273</a></span></li><li class="reference-list__item" id="c73"><span class="reference__label">73.</span><ol class="reference__authors_list"><li class="reference__author">Attardo Geoffrey M.</li><li class="reference__author">Hansen Immo A.</li><li class="reference__author">Raikhel Alexander S.</li></ol><span class="reference__authors_list_suffix">2005</span><span class="reference__title">Nutritional regulation of vitellogenesis in mosquitoes: Implications for anautogeny.</span><span class="reference__origin"><i>Insect Biochemistry and Molecular Biology</i> <strong>35</strong>:661‑675</span><span class="reference__doi"><a href="https://doi.org/10.1016/J.IBMB.2005.02.013" class="reference__doi_link">https://doi.org/10.1016/J.IBMB.2005.02.013</a></span></li><li class="reference-list__item" id="c74"><span class="reference__label">74.</span><ol class="reference__authors_list"><li class="reference__author">Carvalho-Santos Zita</li><li class="reference__author">Cardoso-Figueiredo Rita</li><li class="reference__author">Elias Ana Paula</li><li class="reference__author">Tastekin Ibrahim</li><li class="reference__author">Baltazar Célia</li><li class="reference__author">Ribeiro Carlos</li></ol><span class="reference__authors_list_suffix">2020</span><span class="reference__title">Cellular metabolic reprogramming controls sugar appetite in Drosophila.</span><span class="reference__origin"><i>Nature Metabolism 2020 2:9</i> <strong>2</strong>:958‑973</span><span class="reference__doi"><a href="https://doi.org/10.1038/s42255-020-0266-x" class="reference__doi_link">https://doi.org/10.1038/s42255-020-0266-x</a></span></li><li class="reference-list__item" id="c75"><span class="reference__label">75.</span><ol class="reference__authors_list"><li class="reference__author">Hundal Harinder S</li><li class="reference__author">Taylor Peter M</li></ol><span class="reference__authors_list_suffix">2009</span><span class="reference__title">Amino acid transceptors: gate keepers of nutrient exchange and regulators of nutrient signaling.</span><span class="reference__origin"><i>American Journal of Physiology-Endocrinology and Metabolism</i> <strong>296</strong>:E603</span><span class="reference__doi"><a href="https://doi.org/10.1152/AJPENDO.91002.2008" class="reference__doi_link">https://doi.org/10.1152/AJPENDO.91002.2008</a></span></li><li class="reference-list__item" id="c76"><span class="reference__label">76.</span><ol class="reference__authors_list"><li class="reference__author">Agrawal Neha</li><li class="reference__author">Delanoue Renald</li><li class="reference__author">Mauri Alessandra</li><li class="reference__author">Basco Davide</li><li class="reference__author">Pasco Matthieu</li><li class="reference__author">Thorens Bernard</li><li class="reference__author">Léopold Pierre</li></ol><span class="reference__authors_list_suffix">2016</span><span class="reference__title">The Drosophila TNF Eiger Is an Adipokine that Acts on Insulin-Producing Cells to Mediate Nutrient Response.</span><span class="reference__origin"><i>Cell Metabolism</i> <strong>23</strong>:675‑684</span><span class="reference__doi"><a href="https://doi.org/10.1016/j.cmet.2016.03.003" class="reference__doi_link">https://doi.org/10.1016/j.cmet.2016.03.003</a></span></li><li class="reference-list__item" id="c77"><span class="reference__label">77.</span><ol class="reference__authors_list"><li class="reference__author">Saha Tusar T</li><li class="reference__author">Shin Sang Woon</li><li class="reference__author">Dou Wei</li><li class="reference__author">Roy Sourav</li><li class="reference__author">Zhao Bo</li><li class="reference__author">Hou Yuan</li><li class="reference__author">Wang Xue Li</li><li class="reference__author">Zou Zhen</li><li class="reference__author">Girke Thomas</li><li class="reference__author">Raikhel Alexander S</li></ol><span class="reference__authors_list_suffix">2016</span><span class="reference__title">Hairy and Groucho mediate the action of juvenile hormone receptor Methoprene-tolerant in gene repression.</span><span class="reference__origin"><i>Proceedings of the National Academy of Sciences of the United States of America</i> <strong>113</strong>:E735‑E743</span><span class="reference__doi"><a href="https://doi.org/10.1073/PNAS.1523838113/-/DCSUPPLEMENTAL" class="reference__doi_link">https://doi.org/10.1073/PNAS.1523838113/-/DCSUPPLEMENTAL</a></span></li><li class="reference-list__item" id="c78"><span class="reference__label">78.</span><ol class="reference__authors_list"><li class="reference__author">Share Marla R.</li><li class="reference__author">Roe R. Michael</li></ol><span class="reference__authors_list_suffix">1988</span><span class="reference__title">A partition assay for the simultaneous determination of insect juvenile hormone esterase and epoxide hydrolase activity.</span><span class="reference__origin"><i>Analytical Biochemistry</i> <strong>169</strong>:81‑88</span><span class="reference__doi"><a href="https://doi.org/10.1016/0003-2697(88)90257-6" class="reference__doi_link">https://doi.org/10.1016/0003-2697(88)90257-6</a></span></li><li class="reference-list__item" id="c79"><span class="reference__label">79.</span><ol class="reference__authors_list"><li class="reference__author">Nässel Dick R.</li><li class="reference__author">Zandawala Meet</li></ol><span class="reference__authors_list_suffix">2020</span><span class="reference__title">Hormonal axes in drosophila: regulation of hormone release and multiplicity of actions.</span><span class="reference__origin"><i>Cell and Tissue Research</i> <strong>382</strong></span><span class="reference__doi"><a href="https://doi.org/10.1007/S00441-020-03264-Z" class="reference__doi_link">https://doi.org/10.1007/S00441-020-03264-Z</a></span></li><li class="reference-list__item" id="c80"><span class="reference__label">80.</span><ol class="reference__authors_list"><li class="reference__author">Malita Alina</li><li class="reference__author">Kubrak Olga</li><li class="reference__author">Koyama Takashi</li><li class="reference__author">Ahrentløv Nadja</li><li class="reference__author">Texada Michael J.</li><li class="reference__author">Nagy Stanislav</li><li class="reference__author">Halberg Kenneth V.</li><li class="reference__author">Rewitz Kim</li></ol><span class="reference__authors_list_suffix">2022</span><span class="reference__title">A gut-derived hormone suppresses sugar appetite and regulates food choice in drosophila.</span><span class="reference__origin"><i>Nature Metabolism 2022 4:11</i> <strong>4</strong>:1532‑1550</span><span class="reference__doi"><a href="https://doi.org/10.1038/s42255-022-00672-z" class="reference__doi_link">https://doi.org/10.1038/s42255-022-00672-z</a></span></li><li class="reference-list__item" id="c81"><span class="reference__label">81.</span><ol class="reference__authors_list"><li class="reference__author">Choi Harry M T</li><li class="reference__author">Beck Victor A</li><li class="reference__author">Pierce Niles A</li></ol><span class="reference__authors_list_suffix">2014</span><span class="reference__title">Next-generation in situ hybridization chain reaction: Higher gain, lower cost, greater durability.</span><span class="reference__origin"><i>ACS Nano</i> <strong>8</strong>:4284‑4294</span><span class="reference__doi"><a href="https://doi.org/10.1021/NN405717P" class="reference__doi_link">https://doi.org/10.1021/NN405717P</a></span></li><li class="reference-list__item" id="c82"><span class="reference__label">82.</span><ol class="reference__authors_list"><li class="reference__author">Choi Harry M.T.</li><li class="reference__author">Schwarzkopf Maayan</li><li class="reference__author">Fornace Mark E.</li><li class="reference__author">Acharya Aneesh</li><li class="reference__author">Artavanis Georgios</li><li class="reference__author">Stegmaier Johannes</li><li class="reference__author">Cunha Alexandre</li><li class="reference__author">Pierce Niles A.</li></ol><span class="reference__authors_list_suffix">2018</span><span class="reference__title">Third-generation in situ hybridization chain reaction: multiplexed, quantitative, sensitive, versatile, robust.</span><span class="reference__origin"><i>Development (Cambridge, England)</i> <strong>145</strong></span><span class="reference__doi"><a href="https://doi.org/10.1242/DEV.165753" class="reference__doi_link">https://doi.org/10.1242/DEV.165753</a></span></li><li class="reference-list__item" id="c83"><span class="reference__label">83.</span><ol class="reference__authors_list"><li class="reference__author">Nässel Dick R</li><li class="reference__author">Enell Lina E</li><li class="reference__author">Santos Jonathan G</li><li class="reference__author">Wegener Christian</li><li class="reference__author">Johard Helena A D</li></ol><span class="reference__authors_list_suffix">2008</span><span class="reference__title">A large population of diverse neurons in the Drosophila central nervous system expresses short neuropeptide F, suggesting multiple distributed peptide functions.</span><span class="reference__origin"><i>BMC Neuroscience</i> <strong>9</strong>:90</span><span class="reference__doi"><a href="https://doi.org/10.1186/1471-2202-9-90" class="reference__doi_link">https://doi.org/10.1186/1471-2202-9-90</a></span></li><li class="reference-list__item" id="c84"><span class="reference__label">84.</span><ol class="reference__authors_list"><li class="reference__author">Roller Ladislav</li><li class="reference__author">ižmár Daniel Č</li><li class="reference__author">Bednár Branislav</li><li class="reference__author">Žitňan Dušan</li></ol><span class="reference__authors_list_suffix">2016</span><span class="reference__title">Expression of RYamide in the nervous and endocrine system of Bombyx mori.</span><span class="reference__origin"><i>Peptides</i> <strong>80</strong>:72‑79</span><span class="reference__doi"><a href="https://doi.org/10.1016/J.PEPTIDES.2016.02.003" class="reference__doi_link">https://doi.org/10.1016/J.PEPTIDES.2016.02.003</a></span></li><li class="reference-list__item" id="c85"><span class="reference__label">85.</span><ol class="reference__authors_list"><li class="reference__author">Veenstra Jan A.</li><li class="reference__author">Khammassi Hela</li></ol><span class="reference__authors_list_suffix">2017</span><span class="reference__title">Rudimentary expression of RYamide in Drosophila melanogaster relative to other Drosophila species points to a functional decline of this neuropeptide gene.</span><span class="reference__origin"><i>Insect Biochemistry and Molecular Biology</i> <strong>83</strong>:68‑79</span><span class="reference__doi"><a href="https://doi.org/10.1016/J.IBMB.2017.03.001" class="reference__doi_link">https://doi.org/10.1016/J.IBMB.2017.03.001</a></span></li><li class="reference-list__item" id="c86"><span class="reference__label">86.</span><ol class="reference__authors_list"><li class="reference__author">Judson C L</li></ol><span class="reference__authors_list_suffix">1967</span><span class="reference__title">Feeding and oviposition behavior in the mosquito Aedes aegypti (L.). I. Preliminary studies of physiological control mechanisms.</span><span class="reference__origin"><i>The Biological Bulletin</i> <strong>133</strong>:369‑378</span><span class="reference__doi"><a href="https://doi.org/10.2307/1539832" class="reference__doi_link">https://doi.org/10.2307/1539832</a></span></li><li class="reference-list__item" id="c87"><span class="reference__label">87.</span><ol class="reference__authors_list"><li class="reference__author">Lavoipierre M.M.J.</li></ol><span class="reference__authors_list_suffix">1958</span><span class="reference__title">Presence of a factor inhibiting biting in aedes aegypti.</span><span class="reference__origin"><i>Nature</i> <strong>182</strong>:1567‑1568</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c88"><span class="reference__label">88.</span><ol class="reference__authors_list"><li class="reference__author">Takken Willem</li><li class="reference__author">Van Loon Joop J.A.</li><li class="reference__author">Adam Wim</li></ol><span class="reference__authors_list_suffix">2001</span><span class="reference__title">Inhibition of host-seeking response and olfactory responsiveness in Anopheles gambiae following blood feeding.</span><span class="reference__origin"><i>Journal of Insect Physiology</i> <strong>47</strong>:303‑310</span><span class="reference__doi"><a href="https://doi.org/10.1016/S0022-1910(00)00107-4" class="reference__doi_link">https://doi.org/10.1016/S0022-1910(00)00107-4</a></span></li><li class="reference-list__item" id="c89"><span class="reference__label">89.</span><ol class="reference__authors_list"><li class="reference__author">Rowland Mark</li></ol><span class="reference__authors_list_suffix">1989</span><span class="reference__title">Changes in the circadian flight activity of the mosquito anopheles stephensi associated with insemination, blood-feeding, oviposition and nocturnal light intensity.</span><span class="reference__origin"><i>Physiological Entomology</i> <strong>14</strong>:77‑84</span><span class="reference__doi"></span></li><li class="reference-list__item" id="c90"><span class="reference__label">90.</span><ol class="reference__authors_list"><li class="reference__author">Laturney Meghan</li><li class="reference__author">Sterne Gabriella R</li><li class="reference__author">Scott Kristin</li></ol><span class="reference__authors_list_suffix">2022</span><span class="reference__title">Mating activates neuroendocrine pathways signaling hunger in Drosophila females.</span><span class="reference__origin"><i>bioRxiv</i> </span><span class="reference__doi"></span></li><li class="reference-list__item" id="c91"><span class="reference__label">91.</span><ol class="reference__authors_list"><li class="reference__author">Wang Fei</li><li class="reference__author">Wang Kaiyu</li><li class="reference__author">Forknall Nora</li><li class="reference__author">Patrick Christopher</li><li class="reference__author">Yang Tansy</li><li class="reference__author">Parekh Ruchi</li><li class="reference__author">Bock Davi</li><li class="reference__author">Dickson Barry J</li></ol><span class="reference__authors_list_suffix">2020</span><span class="reference__title">Neural circuitry linking mating and egg laying in Drosophila females.</span><span class="reference__origin"><i>Nature</i> <strong>579</strong>:101‑105</span><span class="reference__doi"><a href="https://doi.org/10.1038/s41586-020-2055-9" class="reference__doi_link">https://doi.org/10.1038/s41586-020-2055-9</a></span></li><li class="reference-list__item" id="c92"><span class="reference__label">92.</span><ol class="reference__authors_list"><li class="reference__author">Wang Kaiyu</li><li class="reference__author">Wang Fei</li><li class="reference__author">Forknall Nora</li><li class="reference__author">Yang Tansy</li><li class="reference__author">Patrick Christopher</li><li class="reference__author">Parekh Ruchi</li><li class="reference__author">Dickson Barry J</li></ol><span class="reference__authors_list_suffix">2021</span><span class="reference__title">Neural circuit mechanisms of sexual receptivity in Drosophila females.</span><span class="reference__origin"><i>Nature</i> <strong>589</strong>:577‑581</span><span class="reference__doi"><a href="https://doi.org/10.1038/s41586-020-2972-7" class="reference__doi_link">https://doi.org/10.1038/s41586-020-2972-7</a></span></li><li class="reference-list__item" id="c93"><span class="reference__label">93.</span><ol class="reference__authors_list"><li class="reference__author">Amaro I Alexandra</li><li class="reference__author">Wohl Margot P</li><li class="reference__author">Pitcher Sylvie</li><li class="reference__author">Alfonso-Parra Catalina</li><li class="reference__author">Avila Frank W</li><li class="reference__author">Paige Andrew S</li><li class="reference__author">Helinski Michelle E H</li><li class="reference__author">Duvall Laura B</li><li class="reference__author">Harrington Laura C</li><li class="reference__author">FWolfner Mariana</li><li class="reference__author"><em>et al.</em></li></ol><span class="reference__authors_list_suffix">2024</span><span class="reference__title">Sex peptide receptor is not required for refractoriness to remating or induction of egg laying in Aedes aegypti.</span><span class="reference__origin"><i>Genetics</i> <strong>227</strong></span><span class="reference__doi"><a href="https://doi.org/10.1093/GENETICS/IYAE034" class="reference__doi_link">https://doi.org/10.1093/GENETICS/IYAE034</a></span></li><li class="reference-list__item" id="c94"><span class="reference__label">94.</span><ol class="reference__authors_list"><li class="reference__author">Naccarati Chiara</li><li class="reference__author">Audsley Neil</li><li class="reference__author">Keen Jeffrey N</li><li class="reference__author">Kim Jung Ha</li><li class="reference__author">Howell Gareth J</li><li class="reference__author">Kim Young Joon</li><li class="reference__author">Isaac Elwyn</li></ol><span class="reference__authors_list_suffix">2012</span><span class="reference__title">The host-seeking inhibitory peptide, Aea-HP-1, is made in the male accessory gland and transferred to the female during copulation.</span><span class="reference__origin"><i>Peptides</i> <strong>34</strong>:150</span><span class="reference__doi"><a href="https://doi.org/10.1016/J.PEPTIDES.2011.10.027" class="reference__doi_link">https://doi.org/10.1016/J.PEPTIDES.2011.10.027</a></span></li><li class="reference-list__item" id="c95"><span class="reference__label">95.</span><ol class="reference__authors_list"><li class="reference__author">Peng Duo</li><li class="reference__author">Kakani Evdoxia G</li><li class="reference__author">Mameli Enzo</li><li class="reference__author">Vidoudez Charles</li><li class="reference__author">Mitchell Sara N</li><li class="reference__author">Merrihew Gennifer E</li><li class="reference__author">MacCoss Michael J</li><li class="reference__author">Adams Kelsey</li><li class="reference__author">Rinvee Tasneem A</li><li class="reference__author">Shaw W Robert</li><li class="reference__author"><em>et al.</em></li></ol><span class="reference__authors_list_suffix">2022</span><span class="reference__title">A male steroid controls female sexual behaviour in the malaria mosquito.</span><span class="reference__origin"><i>Nature</i> <strong>608</strong>:93‑97</span><span class="reference__doi"><a href="https://doi.org/10.1038/s41586-022-04908-6" class="reference__doi_link">https://doi.org/10.1038/s41586-022-04908-6</a></span></li><li class="reference-list__item" id="c96"><span class="reference__label">96.</span><ol class="reference__authors_list"><li class="reference__author">Gabrieli Paolo</li><li class="reference__author">Kakani Evdoxia G.</li><li class="reference__author">Mitchell Sara N.</li><li class="reference__author">Mameli Enzo</li><li class="reference__author">Want Elizabeth J.</li><li class="reference__author">Anton Ainhoa Mariezcurrena</li><li class="reference__author">Serrao Aurelio</li><li class="reference__author">Baldini Francesco</li><li class="reference__author">Catteruccia Flaminia</li></ol><span class="reference__authors_list_suffix">2014</span><span class="reference__title">Sexual transfer of the steroid hormone 20e induces the postmating switch in anopheles gambiae.</span><span class="reference__origin"><i>Proceedings of the National Academy of Sciences of the United States of America</i> <strong>111</strong>:16353‑16358</span><span class="reference__doi"><a href="https://doi.org/10.1073/PNAS.1410488111/-/DCSUPPLEMENTAL" class="reference__doi_link">https://doi.org/10.1073/PNAS.1410488111/-/DCSUPPLEMENTAL</a></span></li><li class="reference-list__item" id="c97"><span class="reference__label">97.</span><ol class="reference__authors_list"><li class="reference__author">Zhao Jiajia</li><li class="reference__author">Yin Jiaming</li><li class="reference__author">Wang Zeng</li><li class="reference__author">Shen Jie</li><li class="reference__author">Dong Min</li><li class="reference__author">Yan Shuo</li></ol><span class="reference__authors_list_suffix">2025</span><span class="reference__title">Complicated gene network for regulating feeding behavior: novel efficient target for pest management.</span><span class="reference__origin"><i>Pest management science</i> <strong>81</strong></span><span class="reference__doi"><a href="https://doi.org/10.1002/PS.8459" class="reference__doi_link">https://doi.org/10.1002/PS.8459</a></span></li><li class="reference-list__item" id="c98"><span class="reference__label">98.</span><ol class="reference__authors_list"><li class="reference__author">Krzywinski Jaroslaw</li><li class="reference__author">Grushko Olga G.</li><li class="reference__author">Besansky Nora J.</li></ol><span class="reference__authors_list_suffix">2006</span><span class="reference__title">Analysis of the complete mitochondrial dna from anopheles funestus: An improved dipteran mitochondrial genome annotation and a temporal dimension of mosquito evolution.</span><span class="reference__origin"><i>Molecular Phylogenetics and Evolution</i> <strong>39</strong>:417‑423</span><span class="reference__doi"><a href="https://doi.org/10.1016/J.YMPEV.2006.01.006" class="reference__doi_link">https://doi.org/10.1016/J.YMPEV.2006.01.006</a></span></li><li class="reference-list__item" id="c99"><span class="reference__label">99.</span><ol class="reference__authors_list"><li class="reference__author">Fadda Melissa</li><li class="reference__author">Hasakiogullari Ilayda</li><li class="reference__author">Temmerman Liesbet</li><li class="reference__author">Beets Isabel</li><li class="reference__author">Zels Sven</li><li class="reference__author">Schoofs Liliane</li></ol><span class="reference__authors_list_suffix">2019</span><span class="reference__title">Regulation of feeding and metabolism by neuropeptide f and short neuropeptide f in invertebrates.</span><span class="reference__origin"><i>Frontiers in Endocrinology</i> <strong>10</strong>:1‑17</span><span class="reference__doi"><a href="https://doi.org/10.3389/fendo.2019.00064" class="reference__doi_link">https://doi.org/10.3389/fendo.2019.00064</a></span></li><li class="reference-list__item" id="c100"><span class="reference__label">100.</span><ol class="reference__authors_list"><li class="reference__author">McMeniman Conor J.</li><li class="reference__author">Corfas Román A.</li><li class="reference__author">Matthews Benjamin J.</li><li class="reference__author">Ritchie Scott A.</li><li class="reference__author">Vosshall Leslie B.</li></ol><span class="reference__authors_list_suffix">2014</span><span class="reference__title">Multimodal Integration of Carbon Dioxide and Other Sensory Cues Drives Mosquito Attraction to Humans.</span><span class="reference__origin"><i>Cell</i> <strong>156</strong>:1060</span><span class="reference__doi"><a href="https://doi.org/10.1016/J.CELL.2013.12.044" class="reference__doi_link">https://doi.org/10.1016/J.CELL.2013.12.044</a></span></li><li class="reference-list__item" id="c101"><span class="reference__label">101.</span><ol class="reference__authors_list"><li class="reference__author">Wolff Gabriella H.</li><li class="reference__author">Riffell Jeffrey A.</li></ol><span class="reference__authors_list_suffix">2018</span><span class="reference__title">Olfaction, experience and neural mechanisms underlying mosquito host preference.</span><span class="reference__origin"><i>The Journal of experimental biology</i> <strong>221</strong></span><span class="reference__doi"><a href="https://doi.org/10.1242/JEB.157131" class="reference__doi_link">https://doi.org/10.1242/JEB.157131</a></span></li><li class="reference-list__item" id="c102"><span class="reference__label">102.</span><ol class="reference__authors_list"><li class="reference__author">García-Castro Helena</li><li class="reference__author">Kenny Nathan J.</li><li class="reference__author">Iglesias Marta</li><li class="reference__author">Álvarez-Campos Patricia</li><li class="reference__author">Mason Vincent</li><li class="reference__author">Elek Anamaria</li><li class="reference__author">Schönauer Anna</li><li class="reference__author">Sleight Victoria A.</li><li class="reference__author">Neiro Jakke</li><li class="reference__author">Aboobaker Aziz</li><li class="reference__author"><em>et al.</em></li></ol><span class="reference__authors_list_suffix">2021</span><span class="reference__title">ACME dissociation: a versatile cell fixation-dissociation method for single-cell transcriptomics.</span><span class="reference__origin"><i>Genome Biology</i> <strong>22</strong>:1‑34</span><span class="reference__doi"><a href="https://doi.org/10.1186/S13059-021-02302-5" class="reference__doi_link">https://doi.org/10.1186/S13059-021-02302-5</a></span></li><li class="reference-list__item" id="c103"><span class="reference__label">103.</span><ol class="reference__authors_list"><li class="reference__author">Bray Nicolas L.</li><li class="reference__author">Pimentel Harold</li><li class="reference__author">Melsted Páll</li><li class="reference__author">Pachter Lior</li></ol><span class="reference__authors_list_suffix">2016</span><span class="reference__title">Near-optimal probabilistic RNA-seq quantification.</span><span class="reference__origin"><i>Nature Biotechnology 2016 34:5</i> <strong>34</strong>:525‑527</span><span class="reference__doi"><a href="https://doi.org/10.1038/nbt.3519" class="reference__doi_link">https://doi.org/10.1038/nbt.3519</a></span></li><li class="reference-list__item" id="c104"><span class="reference__label">104.</span><ol class="reference__authors_list"><li class="reference__author">Reese Sarah E.</li><li class="reference__author">Archer Kellie J.</li><li class="reference__author">Therneau Terry M.</li><li class="reference__author">Atkinson Elizabeth J.</li><li class="reference__author">Vachon Celine M.</li><li class="reference__author">De Andrade Mariza</li><li class="reference__author">Kocher Jean Pierre A.</li><li class="reference__author">Eckel-Passow Jeanette E.</li></ol><span class="reference__authors_list_suffix">2013</span><span class="reference__title">A new statistic for identifying batch effects in high-throughput genomic data that uses guided principal component analysis.</span><span class="reference__origin"><i>Bioinformatics</i> <strong>29</strong>:2877‑2883</span><span class="reference__doi"><a href="https://doi.org/10.1093/BIOINFORMATICS/BTT480" class="reference__doi_link">https://doi.org/10.1093/BIOINFORMATICS/BTT480</a></span></li><li class="reference-list__item" id="c105"><span class="reference__label">105.</span><ol class="reference__authors_list"><li class="reference__author">Strand M. R.</li><li class="reference__author">Brown M. R.</li><li class="reference__author">Vogel K. J.</li></ol><span class="reference__authors_list_suffix">2016</span><span class="reference__title">Mosquito peptide hormones: Diversity, production, and function.</span><span class="reference__origin"><i>Advances in insect physiology</i> <strong>51</strong></span><span class="reference__doi"><a href="https://doi.org/10.1016/BS.AIIP.2016.05.003" class="reference__doi_link">https://doi.org/10.1016/BS.AIIP.2016.05.003</a></span></li><li class="reference-list__item" id="c106"><span class="reference__label">106.</span><ol class="reference__authors_list"><li class="reference__author">Matthews Benjamin J</li><li class="reference__author">McBride Carolyn S</li><li class="reference__author">DeGennaro Matthew</li><li class="reference__author">Despo Orion</li><li class="reference__author">Vosshall Leslie B</li></ol><span class="reference__authors_list_suffix">2016</span><span class="reference__title">The neurotranscriptome of the Aedes aegypti mosquito.</span><span class="reference__origin"><i>BMC Genomics</i> <strong>17</strong>:32</span><span class="reference__doi"><a href="https://doi.org/10.1186/s12864-015-2239-0" class="reference__doi_link">https://doi.org/10.1186/s12864-015-2239-0</a></span></li><li class="reference-list__item" id="c107"><span class="reference__label">107.</span><ol class="reference__authors_list"><li class="reference__author">Riehle Michael A.</li><li class="reference__author">Garczynski Stephen F.</li><li class="reference__author">Crim Joe W.</li><li class="reference__author">Hill Catherine A.</li><li class="reference__author">Brown Mark R.</li></ol><span class="reference__authors_list_suffix">2002</span><span class="reference__title">Neuropeptides and peptide hormones in anopheles gambiae.</span><span class="reference__origin"><i>Science</i> <strong>298</strong>:172‑175</span><span class="reference__doi"><a href="https://doi.org/10.1126/SCIENCE.1076827" class="reference__doi_link">https://doi.org/10.1126/SCIENCE.1076827</a></span></li><li class="reference-list__item" id="c108"><span class="reference__label">108.</span><ol class="reference__authors_list"><li class="reference__author">Prasad T S.Keshava</li><li class="reference__author">Mohanty Ajeet Kumar</li><li class="reference__author">Kumar Manish</li><li class="reference__author">Sreenivasamurthy Sreelakshmi K</li><li class="reference__author">Dey Gourav</li><li class="reference__author">Nirujogi Raja Sekhar</li><li class="reference__author">Pinto Sneha M</li><li class="reference__author">Madugundu Anil K</li><li class="reference__author">Patil Arun H</li><li class="reference__author">Advani Jayshree</li><li class="reference__author"><em>et al.</em></li></ol><span class="reference__authors_list_suffix">2017</span><span class="reference__title">Integrating transcriptomic and proteomic data for accurate assembly and annotation of genomes.</span><span class="reference__origin"><i>Genome Research</i> <strong>27</strong>:133‑144</span><span class="reference__doi"><a href="https://doi.org/10.1101/GR.201368.115/-/DC1" class="reference__doi_link">https://doi.org/10.1101/GR.201368.115/-/DC1</a></span></li><li class="reference-list__item" id="c109"><span class="reference__label">109.</span><ol class="reference__authors_list"><li class="reference__author">Cantalapiedra Carlos P.</li><li class="reference__author">Herńandez-Plaza Ana</li><li class="reference__author">Letunic Ivica</li><li class="reference__author">Bork Peer</li><li class="reference__author">Huerta-Cepas Jaime</li></ol><span class="reference__authors_list_suffix">2021</span><span class="reference__title">eggNOG-mapper v2: Functional Annotation, Orthology Assignments, and Domain Prediction at the Metagenomic Scale.</span><span class="reference__origin"><i>Molecular Biology and Evolution</i> <strong>38</strong>:5825‑5829</span><span class="reference__doi"><a href="https://doi.org/10.1093/MOLBEV/MSAB293" class="reference__doi_link">https://doi.org/10.1093/MOLBEV/MSAB293</a></span></li><li class="reference-list__item" id="c110"><span class="reference__label">110.</span><ol class="reference__authors_list"><li class="reference__author">Peterson Hedi</li><li class="reference__author">Kolberg Liis</li><li class="reference__author">Raudvere Uku</li><li class="reference__author">Kuzmin Ivan</li><li class="reference__author">Vilo Jaak</li></ol><span class="reference__authors_list_suffix">2020</span><span class="reference__title">gprofiler2 – an R package for gene list functional enrichment analysis and namespace conversion toolset g:Profiler.</span><span class="reference__origin"><i>F1000Research</i> <strong>9</strong></span><span class="reference__doi"><a href="https://doi.org/10.12688/F1000RESEARCH.24956.2" class="reference__doi_link">https://doi.org/10.12688/F1000RESEARCH.24956.2</a></span></li><li class="reference-list__item" id="c111"><span class="reference__label">111.</span><ol class="reference__authors_list"><li class="reference__author">Livak Kenneth J.</li><li class="reference__author">Schmittgen Thomas D.</li></ol><span class="reference__authors_list_suffix">2001</span><span class="reference__title">Analysis of relative gene expression data using real-time quantitative pcr and the 2(−delta delta c(t)) method.</span><span class="reference__origin"><i>Methods (San Diego, Calif.)</i> <strong>25</strong>:402‑408</span><span class="reference__doi"><a href="https://doi.org/10.1006/METH.2001.1262" class="reference__doi_link">https://doi.org/10.1006/METH.2001.1262</a></span></li><li class="reference-list__item" id="c112"><span class="reference__label">112.</span><ol class="reference__authors_list"><li class="reference__author">Bruce Heather S</li><li class="reference__author">Jerz Gabby</li><li class="reference__author">Kelly Sophia R</li><li class="reference__author">McCarthy Jenny</li><li class="reference__author">Pomerantz Aaron</li><li class="reference__author">Senevirathne Gayani</li><li class="reference__author">Sherrard Alice</li><li class="reference__author">Sun Dennis A</li><li class="reference__author">Wolff Carsten</li><li class="reference__author">Patel Nipam H</li></ol><span class="reference__authors_list_suffix">2021</span><span class="reference__title">Hybridization Chain Reaction (HCR) In Situ Protocol.</span><span class="reference__origin"><i>protocols.io</i> </span><span class="reference__doi"></span></li><li class="reference-list__item" id="c113"><span class="reference__label">113.</span><ol class="reference__authors_list"><li class="reference__author">Slaidina Maija</li><li class="reference__author">Banisch Torsten U.</li><li class="reference__author">Gupta Selena</li><li class="reference__author">Lehmann Ruth</li></ol><span class="reference__authors_list_suffix">2020</span><span class="reference__title">A single-cell atlas of the developing drosophila ovary identifies follicle stem cell progenitors.</span><span class="reference__origin"><i>Genes and Development</i> <strong>34</strong>:239‑249</span><span class="reference__doi"><a href="https://doi.org/10.1101/GAD.330464.119/-/DC1" class="reference__doi_link">https://doi.org/10.1101/GAD.330464.119/-/DC1</a></span></li><li class="reference-list__item" id="c114"><span class="reference__label">114.</span><ol class="reference__authors_list"><li class="reference__author">Schindelin Johannes</li><li class="reference__author">Arganda-Carreras Ignacio</li><li class="reference__author">Frise Erwin</li><li class="reference__author">Kaynig Verena</li><li class="reference__author">Longair Mark</li><li class="reference__author">Pietzsch Tobias</li><li class="reference__author">Preibisch Stephan</li><li class="reference__author">Rueden Curtis</li><li class="reference__author">Saalfeld Stephan</li><li class="reference__author">Schmid Benjamin</li><li class="reference__author"><em>et al.</em></li></ol><span class="reference__authors_list_suffix">2012</span><span class="reference__title">Fiji: an open-source platform for biological-image analysis.</span><span class="reference__origin"><i>Nature Methods 2012 9:7</i> <strong>9</strong>:676‑682</span><span class="reference__doi"><a href="https://doi.org/10.1038/nmeth.2019" class="reference__doi_link">https://doi.org/10.1038/nmeth.2019</a></span></li></ul></section><div class="pagebreak"></div></section><section id="peer-reviews"><h1>Peer reviews</h1><section id="sa1" class="review-content"><h2>Reviewer #1 (Public review):</h2><div class="review-content__body"><p>Summary:</p><p>Bansal et al. present a study on the fundamental blood and nectar feeding behaviors of the critical disease vector, Anopheles stephensi. The study encompasses not just the fundamental changes in blood feeding behaviors of the crucially understudied vector, but then uses a transcriptomic approach to identify candidate neuromodulation pathways which influence blood feeding behavior in this mosquito species. The authors then provide evidence through RNAi knockdown of candidate pathways that the neuromodulators sNPF and Rya modulate feeding either via their physiological activity in the brain alone or through joint physiological activity along the brain-gut axis (but critically not the gut alone). Overall, I found this study to be built on tractable, well-designed behavioral experiments.</p><p>Their study begins with a well-structured experiment to assess how the feeding behaviors of A. stephensi change over the course of its life history and in response to its age, mating, and oviposition status. The authors are careful and validate their experimental paradigm in the more well-studied Ae. aegypti, and are able to recapitulate the results of prior studies, which show that mating is a prerequisite for blood feeding behaviors in Ae. aegypt. Here they find A. Stephensi, like other Anopheline mosquitoes, has a more nuanced regulation of its blood and nectar feeding behaviors.</p><p>The authors then go on to show in a Y-maze olfactometer that ,to some degree, changes in blood feeding status depend on behavioral modulation to host cues, and this is not likely to be a simple change to the biting behaviors alone. I was especially struck by the swap in valence of the host cues for the blood-fed and mated individuals, which had not yet oviposited. This indicates that there is a change in behavior that is not simply desensitization to host cues while navigating in flight, but something much more exciting is happening.</p><p>The authors then use a transcriptomic approach to identify candidate genes in the blood-feeding stages of the mosquito's life cycle to identify a list of 9 candidates that have a role in regulating the host-seeking status of A. stephensi. Then, through investigations of gene knockdown of candidates, they identify the dual action of RYa and sNPF and candidate neuromodulators of host-seeking in this species. Overall, I found the experiments to be well-designed. I found the molecular approach to be sound. While I do not think the molecular approach is necessarily an all-encompassing mechanism identification (owing mostly to the fact that genetic resources are not yet available in A. stephensi as they are in other dipteran models), I think it sets up a rich line of research questions for the neurobiology of mosquito behavioral plasticity and comparative evolution of neuromodulator action.</p><p>Strengths:</p><p>I am especially impressed by the authors' attention to small details in the course of this article. As I read and evaluated this article, I continued to think about how many crucial details could potentially have been missed if this had not been the approach. The attention to detail paid off in spades and allowed the authors to carefully tease apart molecular candidates of blood-seeking stages. The authors' top-down approach to identifying RYamide and sNPF starting from first principles behavioral experiments is especially comprehensive. The results from both the behavioral and molecular target studies will have broad implications for the vectorial capacity of this species and comparative evolution of neural circuit modulation.</p><p>Weaknesses:</p><p>There are a few elements of data visualizations and methodological reporting that I found confusing on a first few read-throughs. Figure 1F, for example, was initially confusing as it made it seem as though there were multiple 2-choice assays for each of the conditions. I would recommend removing the "X" marker from the x-axis to indicate the mosquitoes did not feed from either nectar, blood, or neither in order to make it clear that there was one assay in which mosquitoes had access to both food sources, and the data quantify if they took both meals, one meal, or no meals.</p><p>I would also like to know more about how the authors achieved tissue-specific knockdown for RNAi experiments. I think this is an intriguing methodology, but I could not figure out from the methods why injections either had whole-body or abdomen-specific knockdown.</p><p>I also found some interpretations of the transcriptomic to be overly broad for what transcriptomes can actually tell us about the organism's state. For example, the authors mention, "Interestingly, we found that after a blood meal, glucose is neither spent nor stored, and that the female brain goes into a state of metabolic 'sugar rest', while actively processing proteins (Figure S2B, S3)".</p><p>This would require a physiological measurement to actually know. It certainly suggests that there are changes in carbohydrate metabolism, but there are too many alternative interpretations to make this broad claim from transcriptomic data alone.</p></div><div class="descriptors"><ul class="descriptors__identifiers"><li class="descriptors__identifier"><a href="https://doi.org/10.7554/eLife.108625.1.sa3">https://doi.org/10.7554/eLife.108625.1.sa3</a></li></ul></div></section><section id="sa2" class="review-content"><h2>Reviewer #2 (Public review):</h2><div class="review-content__body"><p>Summary:</p><p>In this manuscript, Bansal et al examine and characterize feeding behaviour in Anopheles stephensi mosquitoes. While sharing some similarities to the well-studied Aedes aegypti mosquito, the authors demonstrate that mated females, but not unmated (virgin) females, exhibit suppression in their blood-feeding behaviour. Using brain transcriptomic analysis comparing sugar-fed, blood-fed, and starved mosquitoes, several candidate genes potentially responsible for influencing blood-feeding behaviour were identified, including two neuropeptides (short NPF and RYamide) that are known to modulate feeding behaviour in other mosquito species. Using molecular tools, including in situ hybridization, the authors map the distribution of cells producing these neuropeptides in the nervous system and in the gut. Further, by implementing systemic RNA interference (RNAi), the study suggests that both neuropeptides appear to promote blood-feeding (but do not impact sugar feeding), although the impact was observed only after both neuropeptide genes underwent knockdown.</p><p>Strengths and/or weaknesses:</p><p>Overall, the manuscript was well-written; however, the authors should review carefully, as some sections would benefit from restructuring to improve clarity. Some statements need to be rectified as they are factually inaccurate.</p><p>Below are specific concerns and clarifications needed in the opinion of this reviewer:</p><p>(1) What does "central brains" refer to in abstract and in other sections of the manuscript (including methods and results)? This term is ambiguous, and the authors should more clearly define what specific components of the central nervous system was/were used in their study.</p><p>(2) The abstract states that two neuropeptides, sNPF and RYamide are working together, but no evidence is summarized for the latter in this section.</p><p>(3) Figure 1</p><p>
Panel A: This should include mating events in the reproductive cycle to demonstrate differences in the feeding behavior of Ae. aegypti.</p><p>
Panel F: In treatments where insects were not provided either blood or sugar, how is it that some females and males had fed? Also, it is unclear why the y-axis label is % fed when the caption indicates this is a choice assay. Also, it is interesting that sugar-starved females did not increase sugar intake. Is there any explanation for this (was it expected)?</p><p>(4) Figure 3</p><p>
In the neurotranscriptome analysis of the (central) brain involving the two types of comparisons, can the authors clarify what "excluded in males" refers to? Does this imply that only genes not expressed in males were considered in the analysis? If so, what about co-expressed genes that have a specific function in female feeding behaviour?</p><p>(5) Figure 4</p><p>
The authors state that there is more efficient knockdown in the head of unfed females; however, this is not accurate since they only get knockdown in unfed animals, and no evidence of any knockdown in fed animals (panel D). This point should be revised in the results test as well. Relatedly, blood-feeding is decreased when both neuropeptide transcripts are targeted compared to uninjected (panel C) but not compared to dsGFP injected (panel E). Why is this the case if authors showed earlier in this figure (panel B) that dsGFP does not impact blood feeding? In addition, do the uninjected and dsGFP-injected relative mRNA expression data reflect combined RYa and sNPF levels? Why is there no variation in these data, and how do transcript levels of RYa and sNPF compare in the brain versus the abdomen (the presentation of data doesn't make this relationship clear).</p><p>(6) As an overall comment, the figure captions are far too long and include redundant text presented in the methods and results sections.</p><p>(7) Criteria used for identifying neuropeptides promoting blood-feeding: statement that reads "all neuropeptides, since these are known to regulate feeding behaviours". This is not accurate since not all neuropeptides govern feeding behaviors, while certainly a subset do play a role.</p><p>(8) In the section beginning with "Two neuropeptides - sNPF and RYa - showed about 25% and 40% reduced mRNA levels...", the authors state that there was no change in blood-feeding and later state the opposite. The wording should be clarified as it is unclear.</p><p>(9) Just before the conclusions section, the statement that "neuropeptide receptors are often ligand-promiscuous" is unjustified. Indeed, many studies have shown in heterologous systems that high concentrations of structurally related peptides, which are not physiologically relevant, might cross-react and activate a receptor belonging to a different peptide family; however, the natural ligand is often many times more potent (in most cases, orders of magnitude) than structurally related peptides. This is certainly the case for various RYamide and sNPF receptors characterized in various insect species.</p><p>(10) Methods</p><p>
In the dsRNA-mediated gene knockdown section, the authors could more clearly describe how much dsRNA was injected per target. At the moment, the reader must carry out calculations based on the concentrations provided and the injected volume range provided later in this section.</p><p>It is also unclear how tissue-specific knockdown was achieved by performing injection on different days/times. The authors need to explain/support, and justify how temporal differences in injection lead to changes in tissue-specific expression. Does the blood-brain barrier limit knockdown in the brain instead, while leaving expression in the peripheral organs susceptible? For example, in Figure 4, the data support that knockdown in the head/brain is only effective in unfed animals compared to uninjected animals, while there is no evidence of knockdown in the brain relative to dsGFP-injected animals. Comparatively, evidence appears to show stronger evidence of abdominal knockdown mostly for the RYa transcript (&gt;90%) while still significantly for the sNPF transcript (&gt;60%).</p></div><div class="descriptors"><ul class="descriptors__identifiers"><li class="descriptors__identifier"><a href="https://doi.org/10.7554/eLife.108625.1.sa2">https://doi.org/10.7554/eLife.108625.1.sa2</a></li></ul></div></section><section id="sa3" class="review-content"><h2>Reviewer #3 (Public review):</h2><div class="review-content__body"><p>Summary:</p><p>This manuscript investigates the regulation of host-seeking behavior in Anopheles stephensi females across different life stages and mating states. Through transcriptomic profiling, the authors identify differential gene expression between "blood-hungry" and "blood-sated" states. Two neuropeptides, sNPF and RYamide, are highlighted as potential mediators of host-seeking behavior. RNAi knockdown of these peptides alters host-seeking activity, and their expression is anatomically mapped in the mosquito brain (sNPF and RYamide) and midgut (sNPF only).</p><p>Strengths:</p><p>(1) The study addresses an important question in mosquito biology, with relevance to vector control and disease transmission.</p><p>(2) Transcriptomic profiling is used to uncover gene expression changes linked to behavioral states.</p><p>(3) The identification of sNPF and RYamide as candidate regulators provides a clear focus for downstream mechanistic work.</p><p>(3) RNAi experiments demonstrate that these neuropeptides are necessary for normal host-seeking behavior.</p><p>(4) Anatomical localization of neuropeptide expression adds depth to the functional findings.</p><p>Weaknesses:</p><p>(1) The title implies that the neuropeptides promote host-seeking, but sufficiency is not demonstrated (for example, with peptide injection or overexpression experiments).</p><p>(2) The proposed model regarding central versus peripheral (gut) peptide action is inconsistently presented and lacks strong experimental support.</p><p>(3) Some conclusions appear premature based on the current data and would benefit from additional functional validation.</p></div><div class="descriptors"><ul class="descriptors__identifiers"><li class="descriptors__identifier"><a href="https://doi.org/10.7554/eLife.108625.1.sa1">https://doi.org/10.7554/eLife.108625.1.sa1</a></li></ul></div></section><section id="sa4" class="author-response"><h2>Author response:</h2><div class="author-response__body"><blockquote><p><strong>Public Reviews:</strong></p><p><strong>Reviewer #1 (Public review):</strong></p><p>Summary:</p><p>Bansal et al. present a study on the fundamental blood and nectar feeding behaviors of the critical disease vector, Anopheles stephensi. The study encompasses not just the fundamental changes in blood feeding behaviors of the crucially understudied vector, but then uses a transcriptomic approach to identify candidate neuromodulation pathways which influence blood feeding behavior in this mosquito species. The authors then provide evidence through RNAi knockdown of candidate pathways that the neuromodulators sNPF and Rya modulate feeding either via their physiological activity in the brain alone or through joint physiological activity along the brain-gut axis (but critically not the gut alone). Overall, I found this study to be built on tractable, well-designed behavioral experiments.</p><p>Their study begins with a well-structured experiment to assess how the feeding behaviors of A. stephensi change over the course of its life history and in response to its age, mating, and oviposition status. The authors are careful and validate their experimental paradigm in the more well-studied Ae. aegypti, and are able to recapitulate the results of prior studies, which show that mating is a prerequisite for blood feeding behaviors in Ae. aegypt. Here they find A. Stephensi, like other Anopheline mosquitoes, has a more nuanced regulation of its blood and nectar feeding behaviors.</p><p>The authors then go on to show in a Y-maze olfactometer that ,to some degree, changes in blood feeding status depend on behavioral modulation to host cues, and this is not likely to be a simple change to the biting behaviors alone. I was especially struck by the swap in valence of the host cues for the blood-fed and mated individuals, which had not yet oviposited. This indicates that there is a change in behavior that is not simply desensitization to host cues while navigating in flight, but something much more exciting is happening.</p><p>The authors then use a transcriptomic approach to identify candidate genes in the blood-feeding stages of the mosquito's life cycle to identify a list of 9 candidates that have a role in regulating the host-seeking status of A. stephensi. Then, through investigations of gene knockdown of candidates, they identify the dual action of RYa and sNPF and candidate neuromodulators of host-seeking in this species. Overall, I found the experiments to be well-designed. I found the molecular approach to be sound. While I do not think the molecular approach is necessarily an all-encompassing mechanism identification (owing mostly to the fact that genetic resources are not yet available in A. stephensi as they are in other dipteran models), I think it sets up a rich line of research questions for the neurobiology of mosquito behavioral plasticity and comparative evolution of neuromodulator action.</p></blockquote><p>We appreciate the reviewer’s detailed summary of our work. We thank them for their positive comments and agree with them on the shortcomings of our approach.</p><blockquote><p>Strengths:</p><p>I am especially impressed by the authors' attention to small details in the course of this article. As I read and evaluated this article, I continued to think about how many crucial details could potentially have been missed if this had not been the approach. The attention to detail paid off in spades and allowed the authors to carefully tease apart molecular candidates of blood-seeking stages. The authors' top-down approach to identifying RYamide and sNPF starting from first principles behavioral experiments is especially comprehensive. The results from both the behavioral and molecular target studies will have broad implications for the vectorial capacity of this species and comparative evolution of neural circuit modulation.</p></blockquote><p>We really appreciate that the reviewer has recognised the attention to detail we have tried to put, thank you!</p><blockquote><p>Weaknesses:</p><p>There are a few elements of data visualizations and methodological reporting that I found confusing on a first few read-throughs. Figure 1F, for example, was initially confusing as it made it seem as though there were multiple 2-choice assays for each of the conditions. I would recommend removing the "X" marker from the x-axis to indicate the mosquitoes did not feed from either nectar, blood, or neither in order to make it clear that there was one assay in which mosquitoes had access to both food sources, and the data quantify if they took both meals, one meal, or no meals.</p></blockquote><p>We thank the reviewer for flagging the schematic in figure 1F. As suggested, we have removed the “X” markers from the x-axis and revised the axis label from “choice of food” to “choice made” to better reflect what food the mosquitoes chose in the assay. For clarity, we have now also plotted the same data as stacked graphs at the bottom of Fig. 1F, which clearly shows the proportion of mosquitoes fed on each particular choice. We avoid the stacked graph as the sole representation of this data, as it does not capture the variability in the data.</p><blockquote><p>I would also like to know more about how the authors achieved tissue-specific knockdown for RNAi experiments. I think this is an intriguing methodology, but I could not figure out from the methods why injections either had whole-body or abdomen-specific knockdown.</p></blockquote><p>The tissue-specific knockdown (abdomen only or abdomen+head) emerged from initial standardisations where we were unable to achieve knockdown in the head unless we used higher concentrations of dsRNA and did the injections in older females. We realised that this gave us the opportunity to isolate the neuronal contribution of these neuropeptides in the phenotype produced. Further optimisations revealed that injecting dsRNA into 0-10h old females produced abdomen-specific knockdowns without affecting head expression, whereas injections into 4 days old females resulted in knockdowns in both tissues. Moreover, head knockdowns in older females required higher dsRNA concentrations, with knockdown efficiency correlating with the amount injected. In contrast, abdominal knockdowns in younger females could be achieved even with lower dsRNA amounts.</p><p>We have mentioned the knockdown conditions- time of injection and the amount dsRNA injected- for tissue-specific knockdowns in methods but realise now that it does not explain this well enough. We have now edited it to state our methodology more clearly (see lines 932-948).</p><blockquote><p>I also found some interpretations of the transcriptomic to be overly broad for what transcriptomes can actually tell us about the organism's state. For example, the authors mention, "Interestingly, we found that  after a blood meal, glucose is neither spent nor stored, and that the female brain goes into a state of metabolic 'sugar rest', while actively processing proteins (Figure S2B, S3)".</p><p>This would require a physiological measurement to actually know. It certainly suggests that there are changes in carbohydrate metabolism, but there are too many alternative interpretations to make this broad claim from transcriptomic data alone.</p></blockquote><p>We thank the reviewer for pointing this out and agree with them. We have now edited our statement to read:</p><p>“Instead, our data suggests altered carbohydrate metabolism  after a blood meal, with the female brain potentially entering a state of metabolic 'sugar rest' while actively processing proteins (Figure S2B, S3). However, physiological measurements of carbohydrate and protein metabolism will be required to confirm whether glucose is indeed neither spent nor stored during this period.” See lines 271-277.</p><blockquote><p><strong>Reviewer #2 (Public review):</strong></p><p>Summary:</p><p>In this manuscript, Bansal et al examine and characterize feeding behaviour in Anopheles stephensi mosquitoes. While sharing some similarities to the well-studied Aedes aegypti mosquito, the authors demonstrate that mated females, but not unmated (virgin) females, exhibit suppression in their bloodfeeding behaviour. Using brain transcriptomic analysis comparing sugar-fed, blood-fed, and starved mosquitoes, several candidate genes potentially responsible for influencing blood-feeding behaviour were identified, including two neuropeptides (short NPF and RYamide) that are known to modulate feeding behaviour in other mosquito species. Using molecular tools, including in situ hybridization, the authors map the distribution of cells producing these neuropeptides in the nervous system and in the gut. Further, by implementing systemic RNA interference (RNAi), the study suggests that both neuropeptides appear to promote blood-feeding (but do not impact sugar feeding), although the impact was observed only  after both neuropeptide genes underwent knockdown.</p><p>Strengths and/or weaknesses:</p><p>Overall, the manuscript was well-written; however, the authors should review carefully, as some sections would benefit from restructuring to improve clarity. Some statements need to be rectified as they are factually inaccurate.</p><p>Below are specific concerns and clarifications needed in the opinion of this reviewer:</p><p>(1) What does "central brains" refer to in abstract and in other sections of the manuscript (including methods and results)? This term is ambiguous, and the authors should more clearly define what specific components of the central nervous system was/were used in their study.</p></blockquote><p>Central brain, or mid brain, is a commonly used term to refer to brain structures/neuropils without the optic lobes (For example: <span class="fakelink linktoext"><span class="fakelink-text">https://www.nature.com/articles/s41586-024-07686-5</span><a class="fake-icon" href="https://www.nature.com/articles/s41586-024-07686-5"><span class="fakelink-icon">     </span></a></span>). In this study we have focused our analysis on the central brain circuits involved in modulating blood-feeding behaviour and have therefore excluded the optic lobes. As optic lobes account for nearly half of all the neurons in the mosquito brain (<span class="fakelink linktoext"><span class="fakelink-text">https://pmc.ncbi.nlm.nih.gov/articles/PMC8121336/</span><a class="fake-icon" href="https://pmc.ncbi.nlm.nih.gov/articles/PMC8121336/"><span class="fakelink-icon">     </span></a></span>), including them would have disproportionately skewed our transcriptomic data toward visual processing pathways.</p><p>We have indicated this in figure 3A and in the methods (see lines 800-801, 812). We have now also clarified it in the results section for neuro-transcriptomics to avoid confusion (see lines 236-237).</p><blockquote><p>(2) The abstract states that two neuropeptides, sNPF and RYamide are working together, but no evidence is summarized for the latter in this section.</p></blockquote><p>We thank the reviewer for pointing this out. We have now added a statement “This occurs in the context of the action of <em>RYa</em> in the brain” to end of the abstract, for a complete summary of our proposed model.</p><blockquote><p>(3) Figure 1</p><p>Panel A: This should include mating events in the reproductive cycle to demonstrate differences in the feeding behavior of Ae. aegypti.</p></blockquote><p>Our data suggest that mating can occur at any time between eclosion and oviposition in <em>An. stephensi</em> and between eclosion and blood feeding in <em>Ae. aegypti</em>. Adding these into (already busy) 1A, would cloud the purpose of the schematic, which is to indicate the time points used in the behavioural assays and transcriptomics.</p><blockquote><p>Panel F: In treatments where insects were not provided either blood or sugar, how is it that some females and males had fed? Also, it is unclear why the y-axis label is % fed when the caption indicates this is a choice assay. Also, it is interesting that sugar-starved females did not increase sugar intake. Is there any explanation for this (was it expected)?</p></blockquote><p>We apologise for the confusion. The experiment is indeed a choice assay in which sugar-starved or sugar-sated females, co-housed with males, were provided simultaneous access to both blood and sugar, and were assessed for the choice made (indicated on the x-axis): both blood and sugar, blood only, sugar only, or neither. The x-axis indicates the choice made by the mosquitoes, not the choice provided in the assay, and the y-axis indicates the percentage of males or females that made each particular choice. We have now removed the “X” markers from the x-axis and revised the axis label from “choice of food” to “choice made” to better reflect what food the mosquitoes chose to take.</p><p>In this assay, we scored females only for the presence or absence of each meal type (blood or sugar) and are therefore unable to comment on whether sugar-starved females consumed more sugar than sugarsated females. However, when sugar-starved, a higher proportion of females consumed both blood and sugar, while fewer fed on blood alone.</p><p>For clarity, we have now also plotted the same data as stacked graphs at the bottom of Fig. 1F, which clearly shows the proportion of mosquitoes fed on each particular choice. We avoid the stacked graph as the sole representation of this data as it does not capture the variability in the data.</p><blockquote><p>(4) Figure 3</p><p>In the neurotranscriptome analysis of the (central) brain involving the two types of comparisons, can the authors clarify what "excluded in males" refers to? Does this imply that only genes not expressed in males were considered in the analysis? If so, what about co-expressed genes that have a specific function in female feeding behaviour?</p></blockquote><p>This is indeed correct. We reasoned that since blood feeding is exclusive to females, we should focus our analysis on genes that were specifically upregulated in them. As the reviewer points out, it is very likely that genes commonly upregulated in males and females may also promote blood feeding and we will miss out on any such candidates based on our selection criteria.</p><blockquote><p>(5) Figure 4</p><p>The authors state that there is more efficient knockdown in the head of unfed females; however, this is not accurate since they only get knockdown in unfed animals, and no evidence of any knockdown in fed animals (panel D). This point should be revised in the results test as well.</p></blockquote><p>Perhaps we do not understand the reviewer’s point or there has been a misunderstanding. In figure 4D, we show that while there is more robust gene knockdown in unfed females, blood-fed females also showed modest but measurable knockdowns ranging from 5-40% for <em>RYamide</em> and 2-21% for <em>sNPF</em>.</p><blockquote><p>Relatedly, blood-feeding is decreased when both neuropeptide transcripts are targeted compared to uninjected (panel C) but not compared to dsGFP injected (panel E). Why is this the case if authors showed earlier in this figure (panel B) that dsGFP does not impact blood feeding?</p></blockquote><p>We realise this concern stems from our representation of the data. Since we had earlier determined that <em>dsGFP</em>-injected females fed similarly to uninjected females (fig 4B), we used these controls interchangeably in subsequent experiments. To avoid confusion, we have now only used the label ‘control’ in figure 4 (and supplementary figure S9) and specified which control was used for each experiment in the legend.</p><p>In addition to this, we wanted to clarify that fig 4C and 4E are independent experiments. 4C is the behaviour corresponding to when the neuropeptides were knocked down in both heads and abdomens.</p><p>4E is the behaviour corresponding to when the neuropeptides were knocked down in only the abdomens. We have now added a schematic in the plots to make this clearer.</p><blockquote><p>In addition, do the uninjected and dsGFP-injected relative mRNA expression data reflect combined RYa and sNPF levels? Why is there no variation in these data,…</p></blockquote><p>In these qPCRs, we calculated relative mRNA expression using the delta-delta Ct method (see line 975). For each neuropeptide its respective control was used. For simplicity, we combined the <em>RYa</em> and <em>sNPF</em> control data into a single representation. The value of this control is invariant because this method sets the control baseline to a value of 1.</p><blockquote><p>…and how do transcript levels of RYa and sNPF compare in the brain versus the abdomen (the presentation of data doesn't make this relationship clear).</p></blockquote><p>The reviewer is correct in pointing out that we have not clarified this relationship in our current presentation. While we have not performed absolute mRNA quantifications, we extracted relative mRNA levels from qPCR data of 96h old unmanipulated control females. We observed that both <em>sNPF</em> and <em>RYa</em> transcripts are expressed at much lower levels in the abdomens, as compared to those in the heads, as shown in the graphs inserted below.</p><div class="fig-group figure" id="sa4fig1"><img class="child-of-figure has-boundary imageonly" loading="eager" src="https://iiif.elifesciences.org/epp/108625%2Fv1%2Fcontent%2Felife-108625-sa4-fig1.jpg/full/max/0/default.jpg" alt=""><p class="figure__caption"><span class="label figure-name"><label class="figure__label">Author response image 1.</label></span></p></div><blockquote><p>(6) As an overall comment, the figure captions are far too long and include redundant text presented in the methods and results sections.</p></blockquote><p>We thank the reviewer for flagging this and have now edited the legends to remove redundancy.</p><blockquote><p>(7) Criteria used for identifying neuropeptides promoting blood-feeding: statement that reads "all neuropeptides, since these are known to regulate feeding behaviours". This is not accurate since not all neuropeptides govern feeding behaviors, while certainly a subset do play a role.</p></blockquote><p>We agree with the reviewer that not all neuropeptides regulate feeding behaviours. Our statement refers to the screening approach we used: in our shortlist of candidates, we chose to validate <em>all</em> neuropeptides.</p><blockquote><p>(8) In the section beginning with "Two neuropeptides - sNPF and RYa - showed about 25% and 40% reduced mRNA levels...", the authors state that there was no change in blood-feeding and later state the opposite. The wording should be clarified as it is unclear.</p></blockquote><p>Thank you for pointing this out. We were referring to an unchanged proportion of the blood fed females. We have now edited the text to the following:</p><p>“Two neuropeptides - <em>sNPF</em> and <em>RYa</em> - showed about 25% and 40% reduced mRNA levels in the heads but the proportion of females that took blood meals remained unchanged”. See lines 338-340.</p><blockquote><p>(9) Just before the conclusions section, the statement that "neuropeptide receptors are often ligand promiscuous" is unjustified. Indeed, many studies have shown in heterologous systems that high concentrations of structurally related peptides, which are not physiologically relevant, might cross-react and activate a receptor belonging to a different peptide family; however, the natural ligand is often many times more potent (in most cases, orders of magnitude) than structurally related peptides. This is certainly the case for various RYamide and sNPF receptors characterized in various insect species.</p></blockquote><p>We agree with the reviewer and apologise for the mistake. We have now removed the statement.</p><blockquote><p>(10) Methods</p><p>In the dsRNA-mediated gene knockdown section, the authors could more clearly describe how much dsRNA was injected per target. At the moment, the reader must carry out calculations based on the concentrations provided and the injected volume range provided later in this section.</p></blockquote><p>We have now edited the section to reflect the amount of dsRNA injected per target. Please see lines 921-931.</p><blockquote><p>It is also unclear how tissue-specific knockdown was achieved by performing injection on different days/times. The authors need to explain/support, and justify how temporal differences in injection lead to changes in tissue-specific expression. Does the blood-brain barrier limit knockdown in the brain instead, while leaving expression in the peripheral organs susceptible?</p></blockquote><p>To achieve tissue-specific knockdowns of <em>sNPF</em> and <em>RYa</em>, we optimised both the time of injection as well as the dsRNA concentration to be injected. Injecting dsRNA into 0-10h females produced abdomen specific knockdowns without affecting head expression, whereas injections into 96h old females resulted in knockdowns in both tissues. Head knockdowns in older females required higher dsRNA concentrations, with knockdown efficiency correlating with the amount injected. In contrast, abdominal knockdowns in younger females could be achieved even with lower dsRNA amounts, reflecting the lower baseline expression of <em>sNPF</em> in abdomens compared to heads and the age-dependent increase in head expression (as confirmed by qPCR). It is possible that the blood-brain barrier also limits the dsRNA entering the brain, thereby requiring higher amounts to be injected for head knockdowns.</p><p>We have now edited this section to state our methodology more clearly (see lines 932-948).</p><blockquote><p>For example, in Figure 4, the data support that knockdown in the head/brain is only effective in unfed animals compared to uninjected animals, while there is no evidence of knockdown in the brain relative to dsGFP-injected animals. Comparatively, evidence appears to show stronger evidence of abdominal knockdown mostly for the RYa transcript (&gt;90%) while still significantly for the sNPF transcript (&gt;60%).</p></blockquote><p>As we explained earlier, this concern likely stems from our representation of the data. Since we had earlier determined that <em>dsGFP</em>-injected females fed similarly to uninjected females (fig 4B), we used these controls interchangeably in subsequent experiments. To avoid confusion, we have now only used the label ‘control’ in figure 4 (and supplementary figure S9) and specified which control was used for each experiment in the legend.</p><p>In addition to this, we wanted to clarify that fig 4C and 4E are independent experiments. 4C is the behaviour corresponding to when the neuropeptides were knocked down in both heads and abdomens. 4E is the behaviour corresponding to when the neuropeptides were knocked down in only the abdomen. We have now added a schematic in the plots to make this clearer.</p><blockquote><p><strong>Reviewer #3 (Public review):</strong></p><p>Summary:</p><p>This manuscript investigates the regulation of host-seeking behavior in Anopheles stephensi females across different life stages and mating states. Through transcriptomic profiling, the authors identify differential gene expression between "blood-hungry" and "blood-sated" states. Two neuropeptides, sNPF and RYamide, are highlighted as potential mediators of host-seeking behavior. RNAi knockdown of these peptides alters host-seeking activity, and their expression is anatomically mapped in the mosquito brain (sNPF and RYamide) and midgut (sNPF only).</p><p>Strengths:</p><p>(1) The study addresses an important question in mosquito biology, with relevance to vector control and disease transmission.</p><p>(2) Transcriptomic profiling is used to uncover gene expression changes linked to behavioral states.</p><p>(3) The identification of sNPF and RYamide as candidate regulators provides a clear focus for downstream mechanistic work.</p><p>(3) RNAi experiments demonstrate that these neuropeptides are necessary for normal host-seeking behavior.</p><p>(4) Anatomical localization of neuropeptide expression adds depth to the functional findings.</p><p>Weaknesses:</p><p>(1) The title implies that the neuropeptides promote host-seeking, but sufficiency is not demonstrated (for example, with peptide injection or overexpression experiments).</p></blockquote><p>Demonstrating sufficiency would require injecting sNPF peptide or its agonist. To date, no small-molecule agonists (or antagonists) that selectively mimic sNPF or RYa neuropeptides have been identified in insects. An NPY analogue, TM30335, has been reported to activate the <em>Aedes aegypti</em> NPY-like receptor 7 (NPYLR7; Duvall et al., 2019), which is also activated by sNPF peptides at higher doses (Liesch et al., 2013). Unfortunately, the compound is no longer available because its manufacturer, 7TM Pharma, has ceased operations. Synthesising the peptides is a possibility that we will explore in the future.</p><blockquote><p>(2) The proposed model regarding central versus peripheral (gut) peptide action is inconsistently presented and lacks strong experimental support.</p></blockquote><p>The best way to address this would be to conduct tissue-specific manipulations, the tools for which are not available in this species. Our approach to achieve head+abdomen and abdomen only knockdown was the closest we could get to achieving tissue specificity and allowed us to confirm that knockdown in the head was necessary for the phenotype. However, as the reviewer points out, this did not allow us to rule out any involvement of the abdomen. This point has been addressed in lines 364-371.</p><blockquote><p>(3) Some conclusions appear premature based on the current data and would benefit from additional functional validation.</p></blockquote><p>The most definitive way of demonstrating necessity of <em>sNPF</em> and <em>RYa</em> in blood feeding would be to generate mutant lines. While we are pursuing this line of experiments, they lie beyond the scope of a revision. In its absence, we relied on the knockdown of the genes using dsRNA. We would like to posit that despite only partial knockdown, mosquitoes do display defects in blood-feeding behaviour, without affecting sugar-feeding. We think this reflects the importance of <em>sNPF</em> in promoting blood feeding.</p><blockquote><p><strong>Recommendations for the authors:</strong></p><p><strong>Reviewer #1 (Recommendations for the authors):</strong></p><p>Overall, I found this manuscript to be well-prepared, visually the figures are great and clearly were carefully thought out and curated, and the research is impacwul. It was a wonderful read from start to finish. I have the following recommendations:</p></blockquote><p>Thank you very much, we are very pleased to hear that you enjoyed reading our manuscript!</p><blockquote><p>(1) For future manuscripts, it would make things significantly easier on the reviewer side to submit a format that uses line numbers.</p></blockquote><p>We sincerely apologise for the oversight. We have now incorporated line numbers in the revised manuscript.</p><blockquote><p>(2) There are a few statements in the text that I think may need clarification or might be outside the bounds of what was actually studied here. For example, in the introduction "However, mating is dispensable in Anophelines even under conditions of nutritional satiety". I am uncertain what is meant by this statement - please clarify.</p></blockquote><p>We apologise for the lack of clarity in the statement and have now deleted it since we felt it was not necessary.</p><blockquote><p>(3) Typo/Grammatical minutiae:</p><p>a) A small idiosyncrasy of using hyphens in compound words should also be fixed throughout. Typically, you don't hyphenate if the words are being used as a noun, as in the case: e.g. "Age affects blood feeding.". However, you would hyphenate if the two words are used as a compound adjective "Age affects blood-feeding behavior". This may not be an all-inclusive list, but here are some examples where hyphens need to either be removed or added. Some examples:</p><p>"Nutritional state also influences other internal state outputs on blood-feeding": blood-feeding -&gt; blood feeding</p><p>"... the modulation of blood-feeding": blood-feeding -&gt; blood feeding</p><p>"For example, whether virgin females take blood-meals...": blood-meals -&gt; blood meals</p><p>".... how internal and external cues shape meal-choice"-&gt; meal choice</p><p>"blood-meal" is often used throughout the text, but is correctly "blood meal" in the figures.</p><p>There are many more examples throughout.</p></blockquote><p>We apologise for these errors and appreciate the reviewer’s keen eye. We have now fixed them throughout the manuscript.</p><blockquote><p>b) Figure 1 Caption has a typo: "co-housed males were accessed for sugar-feeding" should be "co-housed males were assessed for sugar feeding"</p></blockquote><p>We apologise for the typo and thank the reviewer for spotting it. We have now corrected this.</p><p>c) It would be helpful in some other figure captions to more clearly label which statement is relevant to which part of the text. For example, in Figure 4's caption.</p><blockquote><p>"C,D. Blood-feeding and sugar-feeding behaviour of females when both RYa and sNPF are knocked down in the head (C). Relative mRNA expressions of RYa and sNPF in the heads of dsRYa+dssNPF - injected blood-fed and unfed females, as compared to that in uninjected females, analysed via qPCR (D)."</p><p>I found re-referencing C and D at the end of their statements makes it look as thought C precedes the "Relative mRNA expression" and on a first read through, I thought the figure captions were backwards. I'd recommend reformating here and throughout consistently to only have the figure letter precede its relevant caption information, e.g.:</p><p>"C. Blood-feeding and sugar-feeding behaviour of females when both RYa and sNPF are knocked down in the head. D. Relative mRNA expressions of RYa and sNPF in the heads of dsRYa+dssNPF - injected bloodfed and unfed females, as compared to that in uninjected females, analysed via qPCR."</p></blockquote><p>We have now edited the legends as suggested.</p><blockquote><p><strong>Reviewer #2 (Recommendations for the authors):</strong></p><p>Separately from the clarifications and limitations listed above, the authors could strengthen their study and the conclusions drawn if they could rescue the behavioural phenotype observed following knockdown of sNPF and RYamide. This could be achieved by injection of either sNPF or RYa peptide independently or combined following knockdown to validate the role of these peptides in promoting blood-feeding in An. stephensi. Additionally, the apparent (but unclear) regionalized (or tissue-specific) knockdown of sNPF and RYamide transcripts could be visualized and verified by implementing HCR in situ hyb in knockdown animals (or immunohistochemistry using antibodies specific for these two neuropeptides).</p></blockquote><p>In a follow up of this work, we are generating mutants and peptides for these candidates and are planning to conduct exactly the experiments the reviewer suggests.</p><blockquote><p><strong>Reviewer #3 (Recommendations for the authors):</strong></p><p>The loss-of-function data suggest necessity but not sufficiency. Synthetic peptide injection in non-host seeking (blood-fed mated or juvenile) mosquitoes would provide direct evidence for peptide-induced behavioral activation. The lack of these experiments weakens the central claim of the paper that these neuropeptides directly promote blood feeding.</p></blockquote><p>As noted above, we plan to synthesise the peptide to test rescue in a mutant background and sufficiency.</p><blockquote><p>Some of the claims about knockdown efficiency and interpretation are conflicting; the authors dismiss Hairy and Prp as candidates due to 30-35% knockdown, yet base major conclusions on sNPF and RYamide knockdowns with comparable efficiencies (25-40%). This inconsistency should be addressed, or the justification for different thresholds should be clearly stated.</p></blockquote><p>We have not defined any specific knockdown efficacy thresholds in the manuscript, as these can vary considerably between genes, and in some cases, even modest reductions can be sufficient to produce detectable phenotypes. For example, knockdown efficiencies of even as low as about 25% - 40% gave us observable phenotypes for <em>sNPF</em> and <em>RYa</em> RNAi (Figure S9B-G).</p><p>No such phenotypes were observed for <em>Hairy</em> (30%) or <em>Prp</em> (35%) knockdowns. Either these genes are not involved in blood feeding, or the knockdown was not sufficient for these specific genes to induce phenotypes. We cannot distinguish between these scenarios.</p><blockquote><p>The observation that knockdown animals take smaller blood meals is interesting and could reflect a downstream effect of altered host-seeking or an independent physiological change. The relationship between meal size and host-seeking behavior should be clarified.</p></blockquote><p>We agree with the reviewer that the reduced meal size observed in <em>sNPF</em> and <em>RYa</em> knockdown animals could result from their inability to seek a host or due to an independent effect on blood meal intake. Unfortunately, we did not measure host-seeking in these animals. We plan to distinguish between these possibilities using mutants in future work.</p><blockquote><p>Several figures are difficult to interpret due to cluttered labeling and poorly distinguishable color schemes. Simplifying these and improving contrast (especially for co-housed vs. virgin conditions) would enhance readability.</p></blockquote><p>We regret that the reviewer found the figures difficult to follow. We have now revised our annotations throughout the manuscript for enhanced readability. For example, “D1<sup>B</sup>” is now “D1<sup>PBM</sup>” (post-bloodmeal) and “D1<sup>O</sup>” is now “D1<sup>PO</sup>” (post-oviposition). Wherever mated females were used, we have now appended “(m)” to the annotations and consistently depicted these females with striped abdomens in all the schematics. We believe these changes will improve clarity and readability.</p><blockquote><p>The manuscript does not clearly justify the use of whole-brain RNA sequencing to identify peptides involved in metabolic or peripheral processes. Given that anticipatory feeding signals are often peripheral, the logic for brain transcriptomics should be explained.</p></blockquote><p>The reviewer is correct in pointing out that feeding signals could also emerge from peripheral tissues. Signals from these tissues – in response to both changing nutritional and reproductive states – are then integrated by the central brain to modulate feeding choices. For example, in <em>Drosophila</em>, increased protein intake is mediated by central brain circuitry including those in the SEZ and central complex (Munch et al., 2022; Liu et al., 2017; Goldschmidt et al., 2023). In the context of mating, male-derived sex peptide further increases protein feeding by acting on a dedicated central brain circuitry (Walker et al., 2015). We, therefore focused on the central brain for our studies.</p><blockquote><p>The proposed model suggests brain-derived peptides initiate feeding, while gut peptides provide feedback. However, gut-specific knockdowns had no effect, undermining this hypothesis. Conversely, the authors also suggest abdominal involvement based on RNAi results. These contradictions need to be resolved into a consistent model.</p></blockquote><p>We thank the reviewer for raising this point and recognise their concern. Our reasons for invoking an involvement of the gut were two-fold:</p><p>(1) We find increased sNPF transcript expression in the entero-endocrine cells of the midgut in blood-hungry females, which returns to baseline  after a blood-meal (Fig. 4L, M).</p><p>(2) While the abdomen-only knockdowns did not affect blood feeding, every effective head knockdown that affected blood feeding also abolished abdominal transcript levels (Fig. S9C, F). (Achieving a head-only reduction proved impossible because (i) systemic dsRNA delivery inevitably reaches the abdomen and (ii) abdominal expression of both peptides is low, leaving little dynamic range for selective manipulation.) Consequently, we can only conclude the following: 1) that brain expression is required for the behaviour, 2) that we cannot exclude a contributory role for gut-derived <em>sNPF</em>. We have discussed this in lines 364-371.</p><blockquote><p>The identification of candidate receptors is promising, but the manuscript would be significantly strengthened by testing whether receptor knockdowns phenocopy peptide knockdowns. Without this, it is difficult to conclude that the identified receptors mediate the behavioral effects.</p></blockquote><p>We agree that functional validation of the receptors would strengthen the evidence for <em>sNPF</em> and _RYa_mediated control of blood feeding in <em>An. stephensi.</em> We selected these receptors based on sequence homology. A possibility remains that sNPF neuropeptides activate more than one receptor, each modulating a distinct circuit, as shown in the case of <em>Drosophila</em> Tachykinin (<span class="fakelink linktoext"><span class="fakelink-text">https://pmc.ncbi.nlm.nih.gov/articles/PMC10184743/</span><a class="fake-icon" href="https://pmc.ncbi.nlm.nih.gov/articles/PMC10184743/"><span class="fakelink-icon">     </span></a></span>). This will mean a systematic characterisation and knockdown of each of them to confirm their role. We are planning these experiments in the future.</p><blockquote><p>The authors compared the percentage changes in sugar-fed and blood-fed animals under sugar-sated or sugar-starved conditions. Figure 1F should reflect what was discussed in the results.</p></blockquote><p>Perhaps this concern stems from our representation of the data in figure 1F? We have now edited the xaxis and revised its label from “choice of food” to “choice made” to better reflect what food the mosquitoes chose to take.</p><p>For clarity, we have now also plotted the same data as stacked graphs at the bottom of Fig. 1F, which clearly shows the proportion of mosquitoes fed on each particular choice. We avoid the stacked graph as the sole representation of this data because it does not capture the variability in the data.</p><blockquote><p>Minor issues:</p><p>(1) The authors used mosquitoes with belly stripes to indicate mated females. To be consistent, the post-oviposition females should also have belly stripes.</p></blockquote><p>We thank the reviewer for pointing this out. We have now edited all the figures as suggested.</p><blockquote><p>(2) In the first paragraph on the right column of the second page, the authors state, "Since females took blood-meals regardless of their prior sugar-feeding status and only sugar-feeding was selectively suppressed by prior sugar access." Just because the well-fed animals ate less than the starved animals does not mean their feeding behavior was suppressed.</p></blockquote><p>Perhaps there has been a misunderstanding in the experimental setup of figure 1F, probably stemming from our data representation. The experiment is a choice assay in which sugar-starved or sugar-sated females, co-housed with males, were provided simultaneous access to both blood and sugar, and were assessed for the choice made (indicated on the x-axis): both blood and sugar, blood only, sugar only, or neither. We scored females only for the presence or absence of each meal type (blood or sugar) and did not quantify the amount consumed.</p><blockquote><p>(3) The figure legend for Figure 1A and the naming convention for different experimental groups are difficult to follow. A simplified or consistently abbreviated scheme would help readers navigate the figures and text.</p></blockquote><p>We regret that the reviewer found the figure difficult to follow. We have now revised our annotations throughout the manuscript for enhanced readability. For example, “D1<sup>B</sup>” is now “D1<sup>PBM</sup>” (post-bloodmeal) and “D1<sup>O</sup>” is now “D1<sup>PO</sup>” (post-oviposition).</p><blockquote><p>(4) In the last paragraph of the Y-maze olfactory assay for host-seeking behaviour in An. stephensi in Methods, the authors state, "When testing blood-fed females, aged-matched sugar-fed females (bloodhungry) were included as positive controls where ever possible, with satisfactory results." The authors should explicitly describe what the criteria are for "satisfactory results".</p></blockquote><p>We apologise for the lack of clarity. We have now edited the statement to read:</p><p>“When testing blood-fed females, age-matched sugar-fed females (blood-hungry) were included wherever possible as positive controls. These females consistently showed attraction to host cues, as expected.” See lines 786-790.</p><blockquote><p>(5) In the first paragraph of the dsRNA-mediated gene knockdown section in Methods, dsRNA against GFP is used as a negative control for the injection itself, but not for the potential off-target effect.</p></blockquote><p>We agree with the reviewer that <em>dsGFP</em> injections act as controls only for injection-related behavioural changes, and not for off-target effects of RNAi. We have now corrected the statement. See lines 919-920.</p><p>To control for off-target effects, we could have designed multiple dsRNAs targeting different parts of a given gene. We regret not including these controls for potential off-target effects of dsRNAs injected.</p><blockquote><p>(6) References numbers 48, 89, and 90 are not complete citations.</p></blockquote><p>We thank the reviewer for spotting these. We have now corrected these citations.</p></div><div class="descriptors"><ul class="descriptors__identifiers"><li class="descriptors__identifier"><a href="https://doi.org/10.7554/eLife.108625.1.sa0">https://doi.org/10.7554/eLife.108625.1.sa0</a></li></ul></div></section></section></article></body></html>